{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Goal\n",
    "\n",
    "* Generate a collection of MAGs and isolate genomes for the Christensenellales order"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-02-26T12:19:14.942169Z",
     "iopub.status.busy": "2021-02-26T12:19:14.939928Z",
     "iopub.status.idle": "2021-02-26T12:19:14.966655Z"
    }
   },
   "outputs": [],
   "source": [
    "work_dir = '/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/'\n",
    "clade = 'Christensenellales'\n",
    "taxid = 990719  # Christensenellaceae \n",
    "threads = 8"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Init"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-02-26T12:19:15.044919Z",
     "iopub.status.busy": "2021-02-26T12:19:15.042358Z",
     "iopub.status.idle": "2021-02-26T12:19:15.070224Z"
    }
   },
   "outputs": [],
   "source": [
    "library(dplyr)\n",
    "library(tidyr)\n",
    "library(data.table)\n",
    "library(tidytable)\n",
    "library(ggplot2)\n",
    "library(LeyLabRMisc)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-02-26T12:19:15.149600Z",
     "iopub.status.busy": "2021-02-26T12:19:15.147214Z",
     "iopub.status.idle": "2021-02-26T12:19:15.161765Z"
    }
   },
   "outputs": [],
   "source": [
    "library(curl)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-02-23T13:46:48.166216Z",
     "iopub.status.busy": "2021-02-23T13:46:48.163647Z",
     "iopub.status.idle": "2021-02-23T13:46:48.186797Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Created directory: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/ \n"
     ]
    }
   ],
   "source": [
    "df.dims()\n",
    "setDTthreads(threads)\n",
    "make_dir(work_dir)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# From Genbank"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "* Used the genomes from the Christensenellaceae primer design"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## From Colombian metagenome asesmbly"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-02-23T13:52:27.090246Z",
     "iopub.status.busy": "2021-02-23T13:52:27.087898Z",
     "iopub.status.idle": "2021-02-23T13:52:27.120721Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "218"
      ],
      "text/latex": [
       "218"
      ],
      "text/markdown": [
       "218"
      ],
      "text/plain": [
       "[1] 218"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "list_files(file.path(work_dir, 'Colombian_MG'), '*.fna') %>% length"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## From UHGG"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-02-23T13:56:54.586891Z",
     "iopub.status.busy": "2021-02-23T13:56:54.584051Z",
     "iopub.status.idle": "2021-02-23T13:56:56.265021Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tidytable: 4323 × 30</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Genome</th><th scope=col>Original_name</th><th scope=col>Study_set</th><th scope=col>Genome_type</th><th scope=col>Length</th><th scope=col>N_contigs</th><th scope=col>N50</th><th scope=col>GC_content</th><th scope=col>Completeness</th><th scope=col>Contamination</th><th scope=col>⋯</th><th scope=col>Country</th><th scope=col>Continent</th><th scope=col>FTP_download</th><th scope=col>Domain</th><th scope=col>Phylum</th><th scope=col>Class</th><th scope=col>Order</th><th scope=col>Family</th><th scope=col>Genus</th><th scope=col>Species</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>GUT_GENOME000279</td><td>14207_7_64                          </td><td>HBC  </td><td>Isolate</td><td>2615708</td><td> 20</td><td>268821</td><td>56.68</td><td>99.19</td><td>0.81</td><td>⋯</td><td>United Kingdom</td><td>Europe</td><td>ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-000/MGYG-HGUT-00073/genomes1/GUT_GENOME000279.gff.gz</td><td>d__Bacteria</td><td>p__Firmicutes_A</td><td>c__Clostridia</td><td>o__Christensenellales</td><td>f__GCA-900066905</td><td>g__GCA-900066905</td><td>s__GCA-900066905 sp900066905</td></tr>\n",
       "\t<tr><td>GUT_GENOME001938</td><td>AsnicarF_2017__MV_FEM2_t1Q14__bin.14</td><td>CIBIO</td><td>MAG    </td><td>2965995</td><td>141</td><td> 30137</td><td>58.20</td><td>99.18</td><td>1.17</td><td>⋯</td><td>Italy         </td><td>Europe</td><td>ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-038/MGYG-HGUT-03891/genomes2/GUT_GENOME001938.gff.gz</td><td>d__Bacteria</td><td>p__Firmicutes_A</td><td>c__Clostridia</td><td>o__Christensenellales</td><td>f__CAG-74       </td><td>g__UBA11524     </td><td>s__UBA11524 sp000437595     </td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋱</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>GUT_GENOME286732</td><td>ZellerG_2014__CCMD90311071ST-21-0__bin.32</td><td>CIBIO</td><td>MAG</td><td>1752856</td><td>264</td><td>8460</td><td>49.37</td><td>91.90</td><td>1.88</td><td>⋯</td><td>Germany</td><td>Europe</td><td>ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-005/MGYG-HGUT-00530/genomes1/GUT_GENOME286732.gff.gz</td><td>d__Bacteria</td><td>p__Firmicutes_A</td><td>c__Clostridia</td><td>o__Christensenellales</td><td>f__CAG-138</td><td>g__PeH17</td><td>s__PeH17 sp000435055</td></tr>\n",
       "\t<tr><td>GUT_GENOME286757</td><td>ZellerG_2014__CCMD93344354ST-21-0__bin.14</td><td>CIBIO</td><td>MAG</td><td>1369885</td><td>414</td><td>3602</td><td>49.46</td><td>70.82</td><td>2.42</td><td>⋯</td><td>Germany</td><td>Europe</td><td>ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-005/MGYG-HGUT-00530/genomes2/GUT_GENOME286757.gff.gz</td><td>d__Bacteria</td><td>p__Firmicutes_A</td><td>c__Clostridia</td><td>o__Christensenellales</td><td>f__CAG-138</td><td>g__PeH17</td><td>s__PeH17 sp000435055</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tidytable: 4323 × 30\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " Genome & Original\\_name & Study\\_set & Genome\\_type & Length & N\\_contigs & N50 & GC\\_content & Completeness & Contamination & ⋯ & Country & Continent & FTP\\_download & Domain & Phylum & Class & Order & Family & Genus & Species\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <int> & <int> & <int> & <dbl> & <dbl> & <dbl> & ⋯ & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t GUT\\_GENOME000279 & 14207\\_7\\_64                           & HBC   & Isolate & 2615708 &  20 & 268821 & 56.68 & 99.19 & 0.81 & ⋯ & United Kingdom & Europe & ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify\\_genomes/human-gut/v1.0/all\\_genomes/MGYG-HGUT-000/MGYG-HGUT-00073/genomes1/GUT\\_GENOME000279.gff.gz & d\\_\\_Bacteria & p\\_\\_Firmicutes\\_A & c\\_\\_Clostridia & o\\_\\_Christensenellales & f\\_\\_GCA-900066905 & g\\_\\_GCA-900066905 & s\\_\\_GCA-900066905 sp900066905\\\\\n",
       "\t GUT\\_GENOME001938 & AsnicarF\\_2017\\_\\_MV\\_FEM2\\_t1Q14\\_\\_bin.14 & CIBIO & MAG     & 2965995 & 141 &  30137 & 58.20 & 99.18 & 1.17 & ⋯ & Italy          & Europe & ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify\\_genomes/human-gut/v1.0/all\\_genomes/MGYG-HGUT-038/MGYG-HGUT-03891/genomes2/GUT\\_GENOME001938.gff.gz & d\\_\\_Bacteria & p\\_\\_Firmicutes\\_A & c\\_\\_Clostridia & o\\_\\_Christensenellales & f\\_\\_CAG-74        & g\\_\\_UBA11524      & s\\_\\_UBA11524 sp000437595     \\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋱ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t GUT\\_GENOME286732 & ZellerG\\_2014\\_\\_CCMD90311071ST-21-0\\_\\_bin.32 & CIBIO & MAG & 1752856 & 264 & 8460 & 49.37 & 91.90 & 1.88 & ⋯ & Germany & Europe & ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify\\_genomes/human-gut/v1.0/all\\_genomes/MGYG-HGUT-005/MGYG-HGUT-00530/genomes1/GUT\\_GENOME286732.gff.gz & d\\_\\_Bacteria & p\\_\\_Firmicutes\\_A & c\\_\\_Clostridia & o\\_\\_Christensenellales & f\\_\\_CAG-138 & g\\_\\_PeH17 & s\\_\\_PeH17 sp000435055\\\\\n",
       "\t GUT\\_GENOME286757 & ZellerG\\_2014\\_\\_CCMD93344354ST-21-0\\_\\_bin.14 & CIBIO & MAG & 1369885 & 414 & 3602 & 49.46 & 70.82 & 2.42 & ⋯ & Germany & Europe & ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify\\_genomes/human-gut/v1.0/all\\_genomes/MGYG-HGUT-005/MGYG-HGUT-00530/genomes2/GUT\\_GENOME286757.gff.gz & d\\_\\_Bacteria & p\\_\\_Firmicutes\\_A & c\\_\\_Clostridia & o\\_\\_Christensenellales & f\\_\\_CAG-138 & g\\_\\_PeH17 & s\\_\\_PeH17 sp000435055\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tidytable: 4323 × 30\n",
       "\n",
       "| Genome &lt;chr&gt; | Original_name &lt;chr&gt; | Study_set &lt;chr&gt; | Genome_type &lt;chr&gt; | Length &lt;int&gt; | N_contigs &lt;int&gt; | N50 &lt;int&gt; | GC_content &lt;dbl&gt; | Completeness &lt;dbl&gt; | Contamination &lt;dbl&gt; | ⋯ ⋯ | Country &lt;chr&gt; | Continent &lt;chr&gt; | FTP_download &lt;chr&gt; | Domain &lt;chr&gt; | Phylum &lt;chr&gt; | Class &lt;chr&gt; | Order &lt;chr&gt; | Family &lt;chr&gt; | Genus &lt;chr&gt; | Species &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| GUT_GENOME000279 | 14207_7_64                           | HBC   | Isolate | 2615708 |  20 | 268821 | 56.68 | 99.19 | 0.81 | ⋯ | United Kingdom | Europe | ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-000/MGYG-HGUT-00073/genomes1/GUT_GENOME000279.gff.gz | d__Bacteria | p__Firmicutes_A | c__Clostridia | o__Christensenellales | f__GCA-900066905 | g__GCA-900066905 | s__GCA-900066905 sp900066905 |\n",
       "| GUT_GENOME001938 | AsnicarF_2017__MV_FEM2_t1Q14__bin.14 | CIBIO | MAG     | 2965995 | 141 |  30137 | 58.20 | 99.18 | 1.17 | ⋯ | Italy          | Europe | ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-038/MGYG-HGUT-03891/genomes2/GUT_GENOME001938.gff.gz | d__Bacteria | p__Firmicutes_A | c__Clostridia | o__Christensenellales | f__CAG-74        | g__UBA11524      | s__UBA11524 sp000437595      |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋱ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| GUT_GENOME286732 | ZellerG_2014__CCMD90311071ST-21-0__bin.32 | CIBIO | MAG | 1752856 | 264 | 8460 | 49.37 | 91.90 | 1.88 | ⋯ | Germany | Europe | ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-005/MGYG-HGUT-00530/genomes1/GUT_GENOME286732.gff.gz | d__Bacteria | p__Firmicutes_A | c__Clostridia | o__Christensenellales | f__CAG-138 | g__PeH17 | s__PeH17 sp000435055 |\n",
       "| GUT_GENOME286757 | ZellerG_2014__CCMD93344354ST-21-0__bin.14 | CIBIO | MAG | 1369885 | 414 | 3602 | 49.46 | 70.82 | 2.42 | ⋯ | Germany | Europe | ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-005/MGYG-HGUT-00530/genomes2/GUT_GENOME286757.gff.gz | d__Bacteria | p__Firmicutes_A | c__Clostridia | o__Christensenellales | f__CAG-138 | g__PeH17 | s__PeH17 sp000435055 |\n",
       "\n"
      ],
      "text/plain": [
       "     Genome           Original_name                             Study_set\n",
       "1    GUT_GENOME000279 14207_7_64                                HBC      \n",
       "2    GUT_GENOME001938 AsnicarF_2017__MV_FEM2_t1Q14__bin.14      CIBIO    \n",
       "⋮    ⋮                ⋮                                         ⋮        \n",
       "4322 GUT_GENOME286732 ZellerG_2014__CCMD90311071ST-21-0__bin.32 CIBIO    \n",
       "4323 GUT_GENOME286757 ZellerG_2014__CCMD93344354ST-21-0__bin.14 CIBIO    \n",
       "     Genome_type Length  N_contigs N50    GC_content Completeness Contamination\n",
       "1    Isolate     2615708  20       268821 56.68      99.19        0.81         \n",
       "2    MAG         2965995 141        30137 58.20      99.18        1.17         \n",
       "⋮    ⋮           ⋮       ⋮         ⋮      ⋮          ⋮            ⋮            \n",
       "4322 MAG         1752856 264       8460   49.37      91.90        1.88         \n",
       "4323 MAG         1369885 414       3602   49.46      70.82        2.42         \n",
       "     ⋯ Country        Continent\n",
       "1    ⋯ United Kingdom Europe   \n",
       "2    ⋯ Italy          Europe   \n",
       "⋮    ⋱ ⋮              ⋮        \n",
       "4322 ⋯ Germany        Europe   \n",
       "4323 ⋯ Germany        Europe   \n",
       "     FTP_download                                                                                                                                           \n",
       "1    ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-000/MGYG-HGUT-00073/genomes1/GUT_GENOME000279.gff.gz\n",
       "2    ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-038/MGYG-HGUT-03891/genomes2/GUT_GENOME001938.gff.gz\n",
       "⋮    ⋮                                                                                                                                                      \n",
       "4322 ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-005/MGYG-HGUT-00530/genomes1/GUT_GENOME286732.gff.gz\n",
       "4323 ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-005/MGYG-HGUT-00530/genomes2/GUT_GENOME286757.gff.gz\n",
       "     Domain      Phylum          Class         Order                \n",
       "1    d__Bacteria p__Firmicutes_A c__Clostridia o__Christensenellales\n",
       "2    d__Bacteria p__Firmicutes_A c__Clostridia o__Christensenellales\n",
       "⋮    ⋮           ⋮               ⋮             ⋮                    \n",
       "4322 d__Bacteria p__Firmicutes_A c__Clostridia o__Christensenellales\n",
       "4323 d__Bacteria p__Firmicutes_A c__Clostridia o__Christensenellales\n",
       "     Family           Genus            Species                     \n",
       "1    f__GCA-900066905 g__GCA-900066905 s__GCA-900066905 sp900066905\n",
       "2    f__CAG-74        g__UBA11524      s__UBA11524 sp000437595     \n",
       "⋮    ⋮                ⋮                ⋮                           \n",
       "4322 f__CAG-138       g__PeH17         s__PeH17 sp000435055        \n",
       "4323 f__CAG-138       g__PeH17         s__PeH17 sp000435055        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "F = file.path('/ebio/abt3_projects/databases_no-backup/UHGG/2019_09', 'genomes-nr_metadata.tsv')\n",
    "genomes = Fread(F) %>%\n",
    "    filter.(grepl('o__Christensenellales', Lineage)) %>%\n",
    "    separate.(Lineage, taxonomy_levels(), sep=';') \n",
    "genomes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-02-23T13:57:04.442624Z",
     "iopub.status.busy": "2021-02-23T13:57:04.440469Z",
     "iopub.status.idle": "2021-02-23T13:57:04.495954Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tidytable: 384 × 30</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Genome</th><th scope=col>Original_name</th><th scope=col>Study_set</th><th scope=col>Genome_type</th><th scope=col>Length</th><th scope=col>N_contigs</th><th scope=col>N50</th><th scope=col>GC_content</th><th scope=col>Completeness</th><th scope=col>Contamination</th><th scope=col>⋯</th><th scope=col>Country</th><th scope=col>Continent</th><th scope=col>FTP_download</th><th scope=col>Domain</th><th scope=col>Phylum</th><th scope=col>Class</th><th scope=col>Order</th><th scope=col>Family</th><th scope=col>Genus</th><th scope=col>Species</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>GUT_GENOME001999</td><td>AsnicarF_2017__MV_FEM5_t1Q14__bin.2</td><td>CIBIO</td><td>MAG</td><td>2289322</td><td> 56</td><td>69038</td><td>52.62</td><td>99.19</td><td>0</td><td>⋯</td><td>Italy </td><td>Europe</td><td>ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes2/GUT_GENOME001999.gff.gz</td><td>d__Bacteria</td><td>p__Firmicutes_A</td><td>c__Clostridia</td><td>o__Christensenellales</td><td>f__CAG-138</td><td>g__Phil1</td><td>s__Phil1 sp001940855</td></tr>\n",
       "\t<tr><td>GUT_GENOME004317</td><td>BackhedF_2015__SID343_M__bin.4     </td><td>CIBIO</td><td>MAG</td><td>1124002</td><td>335</td><td> 3591</td><td>50.80</td><td>66.07</td><td>0</td><td>⋯</td><td>Sweden</td><td>Europe</td><td>ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-026/MGYG-HGUT-02681/genomes1/GUT_GENOME004317.gff.gz</td><td>d__Bacteria</td><td>p__Firmicutes_A</td><td>c__Clostridia</td><td>o__Christensenellales</td><td>f__QAND01 </td><td>g__     </td><td>s__                 </td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋱</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>GUT_GENOME286615</td><td>ZellerG_2014__CCMD86114146ST-21-0__bin.70</td><td>CIBIO</td><td>MAG</td><td>2162369</td><td> 61</td><td>51782</td><td>52.79</td><td>96.77</td><td>0</td><td>⋯</td><td>Germany</td><td>Europe</td><td>ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes2/GUT_GENOME286615.gff.gz</td><td>d__Bacteria</td><td>p__Firmicutes_A</td><td>c__Clostridia</td><td>o__Christensenellales</td><td>f__CAG-138</td><td>g__Phil1</td><td>s__Phil1 sp001940855</td></tr>\n",
       "\t<tr><td>GUT_GENOME286691</td><td>ZellerG_2014__CCMD87156761ST-21-0__bin.83</td><td>CIBIO</td><td>MAG</td><td> 923663</td><td>345</td><td> 2702</td><td>49.46</td><td>52.78</td><td>0</td><td>⋯</td><td>Germany</td><td>Europe</td><td>ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-005/MGYG-HGUT-00530/genomes2/GUT_GENOME286691.gff.gz</td><td>d__Bacteria</td><td>p__Firmicutes_A</td><td>c__Clostridia</td><td>o__Christensenellales</td><td>f__CAG-138</td><td>g__PeH17</td><td>s__PeH17 sp000435055</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tidytable: 384 × 30\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " Genome & Original\\_name & Study\\_set & Genome\\_type & Length & N\\_contigs & N50 & GC\\_content & Completeness & Contamination & ⋯ & Country & Continent & FTP\\_download & Domain & Phylum & Class & Order & Family & Genus & Species\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <int> & <int> & <int> & <dbl> & <dbl> & <dbl> & ⋯ & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t GUT\\_GENOME001999 & AsnicarF\\_2017\\_\\_MV\\_FEM5\\_t1Q14\\_\\_bin.2 & CIBIO & MAG & 2289322 &  56 & 69038 & 52.62 & 99.19 & 0 & ⋯ & Italy  & Europe & ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify\\_genomes/human-gut/v1.0/all\\_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes2/GUT\\_GENOME001999.gff.gz & d\\_\\_Bacteria & p\\_\\_Firmicutes\\_A & c\\_\\_Clostridia & o\\_\\_Christensenellales & f\\_\\_CAG-138 & g\\_\\_Phil1 & s\\_\\_Phil1 sp001940855\\\\\n",
       "\t GUT\\_GENOME004317 & BackhedF\\_2015\\_\\_SID343\\_M\\_\\_bin.4      & CIBIO & MAG & 1124002 & 335 &  3591 & 50.80 & 66.07 & 0 & ⋯ & Sweden & Europe & ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify\\_genomes/human-gut/v1.0/all\\_genomes/MGYG-HGUT-026/MGYG-HGUT-02681/genomes1/GUT\\_GENOME004317.gff.gz & d\\_\\_Bacteria & p\\_\\_Firmicutes\\_A & c\\_\\_Clostridia & o\\_\\_Christensenellales & f\\_\\_QAND01  & g\\_\\_      & s\\_\\_                 \\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋱ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t GUT\\_GENOME286615 & ZellerG\\_2014\\_\\_CCMD86114146ST-21-0\\_\\_bin.70 & CIBIO & MAG & 2162369 &  61 & 51782 & 52.79 & 96.77 & 0 & ⋯ & Germany & Europe & ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify\\_genomes/human-gut/v1.0/all\\_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes2/GUT\\_GENOME286615.gff.gz & d\\_\\_Bacteria & p\\_\\_Firmicutes\\_A & c\\_\\_Clostridia & o\\_\\_Christensenellales & f\\_\\_CAG-138 & g\\_\\_Phil1 & s\\_\\_Phil1 sp001940855\\\\\n",
       "\t GUT\\_GENOME286691 & ZellerG\\_2014\\_\\_CCMD87156761ST-21-0\\_\\_bin.83 & CIBIO & MAG &  923663 & 345 &  2702 & 49.46 & 52.78 & 0 & ⋯ & Germany & Europe & ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify\\_genomes/human-gut/v1.0/all\\_genomes/MGYG-HGUT-005/MGYG-HGUT-00530/genomes2/GUT\\_GENOME286691.gff.gz & d\\_\\_Bacteria & p\\_\\_Firmicutes\\_A & c\\_\\_Clostridia & o\\_\\_Christensenellales & f\\_\\_CAG-138 & g\\_\\_PeH17 & s\\_\\_PeH17 sp000435055\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tidytable: 384 × 30\n",
       "\n",
       "| Genome &lt;chr&gt; | Original_name &lt;chr&gt; | Study_set &lt;chr&gt; | Genome_type &lt;chr&gt; | Length &lt;int&gt; | N_contigs &lt;int&gt; | N50 &lt;int&gt; | GC_content &lt;dbl&gt; | Completeness &lt;dbl&gt; | Contamination &lt;dbl&gt; | ⋯ ⋯ | Country &lt;chr&gt; | Continent &lt;chr&gt; | FTP_download &lt;chr&gt; | Domain &lt;chr&gt; | Phylum &lt;chr&gt; | Class &lt;chr&gt; | Order &lt;chr&gt; | Family &lt;chr&gt; | Genus &lt;chr&gt; | Species &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| GUT_GENOME001999 | AsnicarF_2017__MV_FEM5_t1Q14__bin.2 | CIBIO | MAG | 2289322 |  56 | 69038 | 52.62 | 99.19 | 0 | ⋯ | Italy  | Europe | ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes2/GUT_GENOME001999.gff.gz | d__Bacteria | p__Firmicutes_A | c__Clostridia | o__Christensenellales | f__CAG-138 | g__Phil1 | s__Phil1 sp001940855 |\n",
       "| GUT_GENOME004317 | BackhedF_2015__SID343_M__bin.4      | CIBIO | MAG | 1124002 | 335 |  3591 | 50.80 | 66.07 | 0 | ⋯ | Sweden | Europe | ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-026/MGYG-HGUT-02681/genomes1/GUT_GENOME004317.gff.gz | d__Bacteria | p__Firmicutes_A | c__Clostridia | o__Christensenellales | f__QAND01  | g__      | s__                  |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋱ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| GUT_GENOME286615 | ZellerG_2014__CCMD86114146ST-21-0__bin.70 | CIBIO | MAG | 2162369 |  61 | 51782 | 52.79 | 96.77 | 0 | ⋯ | Germany | Europe | ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes2/GUT_GENOME286615.gff.gz | d__Bacteria | p__Firmicutes_A | c__Clostridia | o__Christensenellales | f__CAG-138 | g__Phil1 | s__Phil1 sp001940855 |\n",
       "| GUT_GENOME286691 | ZellerG_2014__CCMD87156761ST-21-0__bin.83 | CIBIO | MAG |  923663 | 345 |  2702 | 49.46 | 52.78 | 0 | ⋯ | Germany | Europe | ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-005/MGYG-HGUT-00530/genomes2/GUT_GENOME286691.gff.gz | d__Bacteria | p__Firmicutes_A | c__Clostridia | o__Christensenellales | f__CAG-138 | g__PeH17 | s__PeH17 sp000435055 |\n",
       "\n"
      ],
      "text/plain": [
       "    Genome           Original_name                             Study_set\n",
       "1   GUT_GENOME001999 AsnicarF_2017__MV_FEM5_t1Q14__bin.2       CIBIO    \n",
       "2   GUT_GENOME004317 BackhedF_2015__SID343_M__bin.4            CIBIO    \n",
       "⋮   ⋮                ⋮                                         ⋮        \n",
       "383 GUT_GENOME286615 ZellerG_2014__CCMD86114146ST-21-0__bin.70 CIBIO    \n",
       "384 GUT_GENOME286691 ZellerG_2014__CCMD87156761ST-21-0__bin.83 CIBIO    \n",
       "    Genome_type Length  N_contigs N50   GC_content Completeness Contamination ⋯\n",
       "1   MAG         2289322  56       69038 52.62      99.19        0             ⋯\n",
       "2   MAG         1124002 335        3591 50.80      66.07        0             ⋯\n",
       "⋮   ⋮           ⋮       ⋮         ⋮     ⋮          ⋮            ⋮             ⋱\n",
       "383 MAG         2162369  61       51782 52.79      96.77        0             ⋯\n",
       "384 MAG          923663 345        2702 49.46      52.78        0             ⋯\n",
       "    Country Continent\n",
       "1   Italy   Europe   \n",
       "2   Sweden  Europe   \n",
       "⋮   ⋮       ⋮        \n",
       "383 Germany Europe   \n",
       "384 Germany Europe   \n",
       "    FTP_download                                                                                                                                           \n",
       "1   ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes2/GUT_GENOME001999.gff.gz\n",
       "2   ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-026/MGYG-HGUT-02681/genomes1/GUT_GENOME004317.gff.gz\n",
       "⋮   ⋮                                                                                                                                                      \n",
       "383 ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes2/GUT_GENOME286615.gff.gz\n",
       "384 ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-005/MGYG-HGUT-00530/genomes2/GUT_GENOME286691.gff.gz\n",
       "    Domain      Phylum          Class         Order                 Family    \n",
       "1   d__Bacteria p__Firmicutes_A c__Clostridia o__Christensenellales f__CAG-138\n",
       "2   d__Bacteria p__Firmicutes_A c__Clostridia o__Christensenellales f__QAND01 \n",
       "⋮   ⋮           ⋮               ⋮             ⋮                     ⋮         \n",
       "383 d__Bacteria p__Firmicutes_A c__Clostridia o__Christensenellales f__CAG-138\n",
       "384 d__Bacteria p__Firmicutes_A c__Clostridia o__Christensenellales f__CAG-138\n",
       "    Genus    Species             \n",
       "1   g__Phil1 s__Phil1 sp001940855\n",
       "2   g__      s__                 \n",
       "⋮   ⋮        ⋮                   \n",
       "383 g__Phil1 s__Phil1 sp001940855\n",
       "384 g__PeH17 s__PeH17 sp000435055"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "genomes_f = genomes %>%\n",
    "    filter.(Completeness >= 0.9,\n",
    "            Contamination < 0.05,\n",
    "            N_contigs <= 500) %>%\n",
    "    filter.(Family != 'f__')\n",
    "genomes_f"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-02-23T13:57:44.939349Z",
     "iopub.status.busy": "2021-02-23T13:57:44.937335Z",
     "iopub.status.idle": "2021-02-23T13:57:45.333443Z"
    }
   },
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA+gAAAKACAIAAACAEcUwAAAACXBIWXMAAB7CAAAewgFu0HU+\nAAAgAElEQVR4nOzdeVxTx/o/8CcLEEBAVMAFQcCiIrihXhR3XGhVimJFq1+3WpeqtT9rpVW7\nWK23Sq1tKdfLxX5drtVWLFIXKH5FKoooKLZaZXEpkT1Ew04SSM7vj3k1NzcsQghC9PP+K86Z\nM/OcnFqfM5mZw+M4jgAAAAAAoGPjt3cAAAAAAADwdEjcAQAAAACMABJ3AAAAAAAjgMQdAAAA\nAMAIIHEHAAAAADACSNwBAAAAAIwAEncAAAAAACOAxB0AAAAAwAggcQcAAAAAMAJI3AEAAAAA\njAASdwAAAAAAI4DEHQAAAADACCBxBwAAAAAwAkjcAQAAAACMgLC9AwCA55lEInn33XcbO6pS\nqdgHgUBgwE45jlOr1Xw+n8fjGbDZNoqWtczj8fh8Q46kqNVqjuOIqC2+B4NHy24ZEbXR96DH\nLbO1tf3mm28MGAkAQOshcQeANqRUKjMyMvr27dulS5f6R2tra9kHExMTA3bKcVxdXZ1AIDBs\nClhXV8dSYaFQaNhUuLa2ls/nG/Z5QKVSsVS4Lb4HIhIKDfnPh1qtZs9FbfQ9tPQ/sJycHKlU\nasAwAAAMAok7ALS5ZcuWTZ06tX45y414PF7Xrl0N2F1dXV1paam1tbWpqakBmy0tLWU5q62t\nrWGTS6lUamFhYWFhYcA2Kysr5XI5EVlZWZmZmRmw5dLSUj6fb21tbcA25XJ5ZWUlEZmbm1ta\nWhqw5crKSqVS2eBzYxN2796dmJhowDAAAAwCc9wBAAAAAIwAEncAAAAAACOAxB0AAAAAwAgg\ncQcAAAAAMAJYnAoA7anuExERFVGlHud2/7aTocMBAADouDDiDgAAAABgBJC4AwAAAAAYASTu\nAAAAAABGAIk7AAAAAIARQOIOAAAAAGAEkLgDAAAAABgBJO4AAAAAAEYAiTsAAAAAgBFA4g4A\nAAAAYATw5lRolFKpPHTo0JUrVyorK7du3Tp48OC27rGsrCwhIeH69esFBQXl5eVWVlbdu3cf\nMWKEn5+fra1t0+fGxcXt27ePiOzt7ffv31+/Qn5+/urVq4nIz89v/fr1DTZy4cKFr776ql+/\nfqGhoZrCdevWicVinZoikcjOzm7o0KEvv/xyr169WnSZAAAAAHpA4g6NOn78+OnTp4moZ8+e\nQmGb/6dy+vTpI0eO1NTUaEpkMplMJsvIyDh+/HhwcHBQUFATp1+8eJF9kEgkWVlZ/fr1a6xm\nQkLCxIkTBw0a1Jpo5XJ5bm5ubm5ubGzs0qVLZ86cWb9OTU1NVFRUSkpKSUmJubl5v379Zs+e\n7eHh0USzEonkzTffXLRoUdMXCwAAAC8gJO7QqPT0dCJas2bNtGnT2rqviIiIs2fPEpGzs3NQ\nUJCHh4eNjU1FRUVGRkZ0dPSDBw8OHTqUl5fX2Ei5RCLJyMggok6dOlVWViYlJTWRuBNReHh4\nWFiYqalp8yNctmxZQEAA+6xSqfLz8x88eBAdHZ2bmxsZGWlqaqrzLZWXl4eEhOTn5xORQCAo\nKytLTU1NS0tbt27d5MmTG+vl3LlzHMc1PyoAAAB4cWCOOzSqqqqKiPr379/WHcXHx7Os/eWX\nX96zZ8+ECRPs7e3NzMy6des2duzY0NDQwMBAHo+XkJAQExPTYAtJSUkcxzk4OLDc+vLly2q1\nurHuunTpUlhY+OOPP7YoSIFAwP+LiYlJnz59/Pz89uzZ4+PjQ0Tffffd48ePteuHh4fn5+fb\n2tru2LEjKioqMjJy0qRJHMeFh4cXFhbWb7+2tjY+Pv7EiRMtigoAAABeHEjc4Sl4PF6btl9d\nXX3w4EEiGjdu3OrVq+uPgguFwmXLlk2dOpWIvv/++/Ly8vqNsHky48aN8/X1JSKZTHb79u3G\nely5ciURRUdH15+53lIikejtt98WiURyuTw+Pl5Tnpubm5KSQkQffvjhoEGDhEKhg4PD22+/\n7eHhoVKpTp48qd1IcnLypk2bFi5cGB4e3sTzBgAAALzgkLhDA/bv3x8QEMAGhteuXRsQEHD9\n+vU26uv8+fNVVVWmpqZLlixpotrChQstLCwUCkVcXJzOIbFYzFLwcePG9e7d29HRkYiSkpIa\na2rUqFE+Pj4qlerbb79t/byUTp06+fv7ExHL1JkrV64QkZubW9++fTWFfD5/ypQpRHTt2jXt\nfouKioqLi0Uika2trYmJSRN9rVy5MiAgYOfOnTrl9+/f37t376pVq+bOnbtx48ZLly4R0fz5\n8+fOndvKqwMAAICOA4k7NMDFxcXX11ckEhHRkCFDfH19O3fu3EZ9sUeCcePGdevWrYlqNjY2\nfn5+RHTjxg2dQ7/++isROTk5OTs7E9GoUaOI6MqVK3V1dY21tnLlSnNz86ysrNjY2NZeANGw\nYcOISHsCDHuQGD58uE7NESNGEJFMJpNKpZrCoKCgQ38ZOXJkS3tPSEgICQlJTEwsKCjg8XjZ\n2dmhoaEREREYvAcAAHjOYHEqNMDPz8/Pz2/lypWFhYXLly93cnJqu76ys7OJiOXcTWNh3Lt3\nj+M4zQQejuPY4Pr48eNZyejRo6OioqqqqtLT0xvLg7t27bpo0aKIiIjDhw//7W9/a/qZ4anY\n6UqlUiaTsW0rZTIZEdnZ2enUtLa2NjU1VSqVBQUF9Y8+1ZQpUyoqKrRvR35+flhYmFqtDgoK\nmjVrlrW1dWFhYXh4OFszwB69nhmpVKpQKOoXEpFarVapVAbvsbE22UOLwTvV/E7SFtfSdtG2\nRcscxxm2Tc1zpsFb1i9arBEHgI4JiTu0p9ra2urqaiJi81uaxuqoVKrKykorKytWePfu3ZKS\nEiIaO3YsK3Fzc3NwcCguLk5KSmpiAPuVV1759ddfs7KyIiIitmzZ0pqr0OT9xcXFLHF/8uQJ\nEWmC1GZtbS2VSisrK/XoaM6cOTolR44cUavV/v7+ixcvZiU9evTYtm3b6tWri4qK9OiiNbZu\n3Vp/SlX37t2JqKqqij3MGFbTber3JTdHgwstWkkul8vlcoM3S0RVVVVsobkBqVSqtrih1Gbf\nQ0ujVSgU+M0KADogTJWB9qQZBjMzM3tqZc26Ve1/UNlwu7u7O0sQGTZbJjU1tYkMgMfjrV27\nViAQXLt2TXt6uh40w/9KpZJ9YImdhYVF/cqssP7ItB44jrt27RoRzZ49W7tcIBDMmDGj9e0D\nAABAh4IRd2hPIpGITR3Jz8/39PRsunJubi4R8Xg8zUi2SqW6fPkyac2TYUaPHh0TEyOXy9PS\n0jQj8fU5OzvPmjXrxIkTERERgwcPbjDPbg7NhHVLS0v2ga0x1X6ZlAb7haHBwfiWkkgkdXV1\nIpFI+6GFcXFxaX37LdWtW7f6L5Ht3LlzUVERn88XCAT1T2nlpIgG2yQijuPUajWfzzfsnkhq\ntZrNoDB4yyqVyuBtsi+B2iBaTbMGbFMTLY/HM2zL+kXb1rtpAQDoB4k7tDMnJ6f79++zpLxp\nrI6Tk5Pm3+D09PSKigoiioyMjIyMrH/KxYsXm0jciWjevHnJycmFhYWHDx9etWqVPhfwV+LO\n4/E0CbSlpWVpaSmLTQebv9GlSxf9+tJWXFxMRGxyjo4GC9vajh076hfm5eUFBgZaWlo2GJL2\nIl09NHaZdXV1paWlnTp1atE7tp6qtLSUrXi2sbFp7JlBP1KpVCQS6f3o2KDKykr2i5OlpWVz\nftFqvtLSUj6fb21tbcA25XI5+6shEok0D8AGUVlZqVQqW/o3wszMzLDPDwAABoH/MUE7Gzp0\nKBFdvny5wfHpoqIiliqp1Wo2ocXb21tzlG3fLhQKreth6zLT09Obnuhsamr61ltvEVFcXFxm\nZqZ+l8BeMduzZ09NwtGzZ08i0nklExFVV1ezXKpr16769aWN5SKlpaX1D5WVlbW+fQAAAOhQ\nkLhDO/P39xcKhU+ePKn/KlOpVBoSEvL+++9LpdKYmJiCggITExPN7G25XM5meAcHBx+ph43+\n1tXVPXX++uDBg9k7Tb/99tva2tqWxl9VVfXLL78Q0bhx4zSF7u7u9FdCr42VWFtb29jYtLSj\n+nr06MHn82tqaiQSic6hR48etb59AAAA6FCQuEM7s7OzCwwMJKKff/45Ojpaexe2zp07jxw5\nMjs7e/369YcPHyai4OBgzRYuV69eVSgUPB5v4sSJ9Zt1d3dnw95NvIlJ44033rC2tn706NGp\nU6daFLxcLv/mm29qamrMzc2nTZumKffx8SGirKws7c3d6a8t50eNGmWQGbRCoZAtDNB5FSvH\ncWfOnGl9+wAAANChIHGH9rdw4cKRI0eqVKqDBw9u3rz5woULOTk5SqWyvLx8yJAhXbt2raio\nUKvVjo6Or732muYsNk/G09PT3t6+wWYnTJhARLdu3XrqTnBWVlbLly+nv6bRP5Varc7Nzf31\n11/fe+89NqL/xhtvaE9bd3Z29vHx4Thux44dbBp3bW3t0aNHU1NThUJhUFBQc3qpLzo6+vDh\nw4mJiZqSJUuW8Hi82NjYY8eOsS3/pFLpZ599VlBQYPAliQAAANC+sDgV2h+fz9+8efORI0di\nYmLu3Llz584dnQomJiZmZmZ5eXmhoaHr1683MzMrKyv77bffiKjB4XZmwoQJR48e5Tju8uXL\nM2fObDqGCRMmXLhwgbXZoMbWvwqFwmXLlk2dOlWnfM2aNWKxODc3d9myZT179pRKpUqlks/n\nr1u3rv4mMM0UGxsrkUh8fHw0V923b9+lS5cePHjw2LFjP/zwQ6dOnSoqKgQCwYYNG8LCwgy7\nJBEAAADaFxJ36BD4fP6iRYv8/f0TEhLS09OLioqqqqpsbW3t7e2HDRs2depUhUKxbds2tvnj\npk2bkpOTVSqVmZmZr69vY2127969f//+mZmZSUlJT03cieitt95au3atZi/2polEom7dug0d\nOnT69OlsTo4OGxubvXv3RkVFpaSklJSUmJubDx06dM6cOf369WtO+80XGBjo5uZ25syZrKws\nuVzu7e29YMECR0dHuVxukL1rAAAAoIPg4cXOYCyqqqr27NmzePFiZ2fn9o6l3Zw+ffru3bsh\nISFNVxOLxevWrfP29v7444+fTWCNYdtB7ty5s/6PEkQklUrrPhHp3Xj3bzs1WM62g7S2tm6j\n7SBtbW0Nvh2khYVFG20HaWVlZUTbQZqbm7fFdpAtfYjdvXt3YmJiXFycASMBAGg9zHEHo2Fp\nafnRRx+9yFk7EclkMjs7O80fd+3atXz58rS0NJ1q8fHxRNS/f/9nGhwAAAC0JSTuAEYjMTEx\nMTFx0qRJmhJXV1eJRHLw4MGSkhJN4fnz5+Pj4wUCgc4LZQEAAMCoYY47NMuBAwd09hw0iP37\n9ze2JwzUV1ZWtmXLlj59+mhKZs2alZ6efufOnRUrVri5ubElvDKZTCgUrlixgq2Cxb0DAAB4\nPiBxh2aZPn0625v8qYqLi7t06WJiYtKcyi19D/kLjm14r00oFG7fvv38+fMXLlwoKiqSy+UO\nDg4eHh5z5851cXFhdXDvAAAAng9I3KFZ7O3tmzm8OmDAgLYOBrQJhUJ/f39/f//GKuDeAQAA\nPB8wxx0AAAAAwAggcQcAAAAAMAJI3AEAAAAAjAASdwAAAAAAI4DEHQAAAADACGBXGQBoT8JP\n5Dwer2vXru0dCAAAQEeHEXcAAAAAACOAxB0AAAAAwAggcQcAAAAAMAKY4w4A7cku2ZKIiGoa\nq8C9av7MggEAAOjIMOIOAAAAAGAEkLgDAAAAABgBJO4AAAAAAEYAiTsAAAAAgBFA4g4AAAAA\nYASQuAMAAAAAGAEk7gAAAAAARgCJOwAAAACAEUDiDgAAAABgBJC4AwAAAAAYAWF7B/BMKZXK\nQ4cOXblypbKycuvWrYMHD27rHsvKyhISEq5fv15QUFBeXm5lZdW9e/cRI0b4+fnZ2to2fW5c\nXNy+ffuIyN7efv/+/fUryOXyuXPnEtEXX3zh7u7eWDuXLl0KDQ0loq+++srV1VXn6Pvvv3/3\n7l1zc/OjR48KBALtQxzHzZs3r6amZsaMGStWrGjeFT9XcPsAAACg43ixEvfjx4+fPn2aiHr2\n7CkUtvm1nz59+siRIzU1NZoSmUwmk8kyMjKOHz8eHBwcFBTUxOkXL15kHyQSSVZWVr9+/fQL\nw8vLi33IyMjQyfxqamqysrLYh8zMzIEDB2ofFYvFLHhNC6xmVFRUSkpKSUmJubl5v379Zs+e\n7eHh0UQAEonkzTffXLRoUdPX29Hg9mniN8bbBwAA8Px5sRL39PR0IlqzZs20adPauq+IiIiz\nZ88SkbOzc1BQkIeHh42NTUVFRUZGRnR09IMHDw4dOpSXl7d+/foGT5dIJBkZGUTUqVOnysrK\npKQkvTO/zp079+7dOzc3NzMzc/r06dqHbt++rVKp2Of09HSdzC8zM5OIeDyepry8vDwkJCQ/\nP5+IBAJBWVlZampqWlraunXrJk+e3FgA586d4zhOv+DbC26fhjHePgAAgOfSizXHvaqqioj6\n9+/f1h3Fx8eztO/ll1/es2fPhAkT7O3tzczMunXrNnbs2NDQ0MDAQB6Pl5CQEBMT02ALSUlJ\nHMc5ODgEBAQQ0eXLl9Vqtd7xsDFXlkpq++2334iod+/eRHTz5k2doyzzc3Z2tra2ZiXh4eH5\n+fm2trY7duyIioqKjIycNGkSx3Hh4eGFhYX1+62trY2Pjz9x4oTekbcL3D7GSG8fAADA8+rF\nStwZHo/Xpu1XV1cfPHiQiMaNG7d69WpTU1OdCkKhcNmyZVOnTiWi77//vry8vH4jbKLFuHHj\nfH19iUgmk92+fVvvkFjmJ5FIHj9+rF3OMr9FixbxeLwHDx6UlZVpH2XTMDQTLXJzc1NSUojo\nww8/HDRokFAodHBwePvttz08PFQq1cmTJ7XPTU5O3rRp08KFC8PDw1uTsz57uH1kzLcPAADg\nOfaiJO779+8PCAhgw4pr164NCAi4fv16G/V1/vz5qqoqU1PTJUuWNFFt4cKFFhYWCoUiLi5O\n55BYLBaLxUQ0bty43r17Ozo6ElFSUpLeIXl5ebHHFTYKy0il0ry8PHNz8+HDh/ft25fjOJYI\nMhUVFWxOhSbzu3LlChG5ubn17dtXU43P50+ZMoWIrl27pj2hoqioqLi4WCQS2drampiYNBHb\nypUrAwICdu7cqVN+//79vXv3rlq1au7cuRs3brx06RIRzZ8/ny3obDu4fdSS2wcAAADPzIuS\nuLu4uPj6+opEIiIaMmSIr69v586d26gv9kgwbty4bt26NVHNxsbGz8+PiG7cuKFz6NdffyUi\nJycnZ2dnIho1ahQRXblypa6uTr+QrK2tnZyc6L+nW7A8z8vLSyAQDBs2jP57ugUbr9WeIc2S\n0eHDh+s0PmLECCKSyWRSqVRTGBQUdOgvI0eObGnACQkJISEhiYmJBQUFPB4vOzs7NDQ0IiLi\nGYz+4vZRq28fAAAAtIUXZXGqn5+fn5/fypUrCwsLly9fztKgNpKdnU1ELGlrGgvj3r17HMdp\nJvBwHMdGZ8ePH89KRo8eHRUVVVVVlZ6erncWNWjQILFYrJ35sTxv6NChRDRs2LAff/zx5s2b\nmkhYTRcXFysrK1ZfJpMRkZ2dnU7L1tbWpqamSqWyoKCg/tGnmjJlSkVFhfYdyc/PDwsLU6vV\nQUFBs2bNsra2LiwsDA8PZ/PO2dNX28Ht09vJkyfz8vJ0CtlAvkKhYCtM9KDHiewBTy6X19bW\n6tdpE80SUU1NjcFn3CmVSsMuAtY8KCoUCr0fGhukVqvVarXeN7RBmmXWtbW1hm25traW47iW\ntsnOMmAYAAAG8aIk7s9MbW1tdXU1EbEJEk1jdVQqVWVlpSbBunv3bklJCRGNHTuWlbi5uTk4\nOBQXFyclJemd+Xl5eZ0+ffrhw4cKhcLMzIzjuFu3bhHRkCFDiMjd3d3CwkImk+Xk5Li4uNBf\nszI8PT01LTx58oSINHFqs7a2lkqllZWVegQ2Z84cnZIjR46o1Wp/f//Fixezkh49emzbtm31\n6tVFRUV6dNF8uH2tER8fX38GWvfu3YlIqVRqb6zZInqfqFQq9TvxqeRyucHbrKurM2x6rdFG\n34Pe96VpbfQ9tDRalUqFxB0AOqAXZarMM6MZNzIzM3tqZc3CR+0ZIGy81t3dnWU8DJtukZqa\nqnfGMHDgQB6Pp1Kp7t27R0R//vlnWVlZt27devXqRUQCgYC9joqN46rValZt0KBBmhbYKkwL\nC4v6jbNChUKhX2zaOI67du0aEc2ePVu7XCAQzJgxo/XtNw23DwAAADosjLgbmEgkYhMP8vPz\ntcc7G5Sbm0tEPB5PMw6qUqkuX75MWhMtmNGjR8fExMjl8rS0NM1QbotYWVm5uLg8fPgwMzPT\n09OTzZBm47XMsGHDUlJSbty4MXv27JycHLlczuPxtF/NwxYpNjhwxUapGxzNbSmJRFJXVycS\nibQTX4aNJbcp3L7WWLBgAdtsR5tCodizZ49IJOrUqVP9U5ozzN/giU1Tq9XV1dUikciw71mr\nqalhj3YWFhZ8viFHPSorK01NTetvYdQaCoWCzRQy+PdQXV3N5/MNO2mttraWPTqamJg057G5\n+dhMIUtLyxadZWJi0tb7jwEA6AGJu+E5OTndv3+fZXVNY3WcnJw0SUB6enpFRQURRUZGRkZG\n1j/l4sWL+mV+ROTl5fXw4cO7d+/Sf8+QZtjnjIwMuVzOlja6uLho50yWlpalpaUsPB0s/erS\npYt+gWkrLi4mIltb2/qHGiw0ONw+vTV4aXl5eXv27DExMWkwz2tO4q5HglhXV1ddXW3wVFjz\ng4mZmZlAIDBgy5WVlUKh0LCpcF1dHUvcDZ4Ky+Vygyfu9NdvPm3xPahUqpa2KRAIkLgDQAeE\nqTKGx1Koy5cvNzi6WVRUxGZwqtVqtq+2t7e35ijb/1soFFrXw/7hSU9P13suMtsZMDMzU6FQ\nZGRk8Hg87akU9vb2jo6OdXV1t2/fZjOktY8SUc+ePYlIZytxIqqurmYJTdeuXfULTBvLzktL\nS+sf0tmnvI3g9gEAAEDHhBF3w/P39z958uSTJ09+/PFHnb3ApVJpSEiInZ3d+++/n5SUVFBQ\nYGJiopm6LZfL2fTu4ODg4OBgnWazs7M3btxYV1eXkpLC9t5uKU9PTz6fX1lZee7cOaVS6erq\namNjo11h6NCheXl56enpbE8Snbki7u7uaWlp6enp8+bN0y5PT08nImtra53W9NOjRw8+n19T\nUyORSOzt7bUPPXr0qPXtPxVuHwAAAHRMGHE3PDs7u8DAQCL6+eefo6Ojtbcm6Ny588iRI7Oz\ns9evX3/48GEiCg4O1uwXfvXqVYVCwePxJk6cWL9Zd3d3Nmiq96t8LCwsXF1diYi9xF57hjTD\ntgO/cuVKUVERj8fTyfx8fHyIKCsri73HSoNtWz5q1CiD/LIsFApZvzrv8uQ47syZM61v/6lw\n+wAAAKBjQuLeJhYuXDhy5EiVSnXw4MHNmzdfuHAhJydHqVSWl5cPGTKka9euFRUVarXa0dHx\ntdde05zFJlp4enrqjDRrTJgwgYhu3brF9uTWA5s+wU7XniHNeHp6mpiYsKNubm46O5A4Ozv7\n+PhwHLdjxw72sp7a2tqjR4+mpqYKhcKgoCD9QoqOjj58+HBiYqKmZMmSJTweLzY29tixY2z3\nZalU+tlnnxUUFPD5/GeQX+L2AQAAQAeEqTJtgs/nb968+ciRIzExMXfu3Llz545OBbZcLC8v\nLzQ0dP369WZmZmVlZWyrkAbHa5kJEyYcPXqU47jLly/PnDlTU75x48bGTomJidHe/sLLyys6\nOpqITE1NBwwYoFPZzMzMw8Pj999/p79mVOtYs2aNWCzOzc1dtmxZz549pVKpUqnk8/nr1q2r\nvwlMM8XGxkokEh8fH82F9+3bd+nSpQcPHjx27NgPP/zQqVOniooKgUCwYcOGsLAwwy6zaxBu\nHwAAAHRASNzbCp/PX7Rokb+/f0JCQnp6elFRUVVVla2trb29/bBhw6ZOnapQKLZt28Z2D9y0\naVNycrJKpTIzM/P19W2sze7du/fv3z8zMzMpKUk782u+gQMHCgQClUrl4eHR4IYb3t7eTWR+\nNjY2e/fujYqKSklJKSkpMTc3Hzp06Jw5c/r166dHME0IDAx0c3M7c+ZMVlaWXC739vZesGCB\no6OjXC43yN41T4XbBwAAAB0NDy+Ha0dVVVV79uxZvHixs7Nze8fSnk6fPn337t2QkJCmq4nF\n4nXr1nl7e3/88cfPJrCm4fY1R15eXmBg4M6dO+tv8U5EUqnULvkpG2xzr5q3tNO6urrS0lJr\na2vDbgdZWlrK9hSytbU17HaQUqnUwsKiwddj6a2yspLtF2RlZWXY36lKS0v5fL61tbUB25TL\n5WzDJXNz85buud60yspKpVLZ0gf+3bt3JyYmxsXFGTASAIDWw4h7e7K0tPzoo4/aO4r2J5PJ\n7OzsNH/ctWvXvXv3Vq5cOWLECO1q8fHxRNS/f/9nHV8jcPsAAADgWcLiVGhniYmJiYmJkyZN\n0pS4urpKJJKDBw+WlJRoCs+fPx8fHy8QCHReSgoAAADwgnhxR9wPHDigs+GgQezfv7+xTUWg\nQWVlZVu2bOnTp4+mZNasWenp6Xfu3FmxYoWbmxtbBiqTyYRC4YoVK9gyStw+AAAAeNG8uIn7\n9OnT2c7WT1VcXNylSxcTE5PmVGYv/oTmY5umaxMKhdu3bz9//vyFCxeKiorkcrmDg4OHh8fc\nuXNdXFxYHdw+AAAAeNG8uIm7vb19M8dW6++7B21NKBT6+/v7+/s3VgG3DwAAAF40mOMOAAAA\nAGAEkLgDAAAAABgBJO4AAAAAAEYAiTsAAAAAgBFA4g4AAAAAYARe3F1lAKAjKPGt4vF4Xbt2\nbe9AAAAAOjqMuAMAAAAAGAEk7gAAAAAARgCJOwAAAACAEcAcdwBoT3bJlkREVO1JeLgAACAA\nSURBVENE3Kvm7RsMAABAR4YRdwAAAAAAI4DEHQAAAADACCBxBwAAAAAwAkjcAQAAAACMABJ3\nAAAAAAAjgMQdAAAAAMAIIHEHAAAAADACSNwBAAAAAIwAEncAAAAAACOAxB0AAAAAwAgI2zsA\n6KCUSuWhQ4euXLlSWVm5devWwYMHt3WPZWVlCQkJ169fLygoKC8vt7Ky6t69+4gRI/z8/Gxt\nbZs+Ny4ubt++fURkb2+/f//++hXkcvncuXOJ6IsvvnB3d2+snUuXLoWGhhLRV1995erqqnP0\n/fffv3v3rrm5+dGjRwUCgfYhjuPmzZtXU1MzY8aMFStWNO+KAQAAAFoAiTs07Pjx46dPnyai\nnj17CoVt/t/J6dOnjxw5UlNToymRyWQymSwjI+P48ePBwcFBQUFNnH7x4kX2QSKRZGVl9evX\nT78wvLy82IeMjAydxL2mpiYrK4t9yMzMHDhwoPZRsVjMgte0wGpGRUWlpKSUlJSYm5v369dv\n9uzZHh4eTQQgkUjefPPNRYsWNX29AAAA8AJC4g4NS09PJ6I1a9ZMmzatrfuKiIg4e/YsETk7\nOwcFBXl4eNjY2FRUVGRkZERHRz948ODQoUN5eXnr169v8HSJRJKRkUFEnTp1qqysTEpK0jtx\n79y5c+/evXNzczMzM6dPn6596Pbt2yqVin1OT0/XSdwzMzOJiMfjacrLy8tDQkLy8/OJSCAQ\nlJWVpaampqWlrVu3bvLkyY0FcO7cOY7j9AseAAAAnm+Y4w4Nq6qqIqL+/fu3dUfx8fEsa3/5\n5Zf37NkzYcIEe3t7MzOzbt26jR07NjQ0NDAwkMfjJSQkxMTENNhCUlISx3EODg4BAQFEdPny\nZbVarXc8bMicPQlo++2334iod+/eRHTz5k2doyxxd3Z2tra2ZiXh4eH5+fm2trY7duyIioqK\njIycNGkSx3Hh4eGFhYX1+62trY2Pjz9x4oTekQMAAMDzDYk7NIXH47Vp+9XV1QcPHiSicePG\nrV692tTUVKeCUChctmzZ1KlTiej7778vLy+v3wibJzNu3DhfX18ikslkt2/f1jsklrhLJJLH\njx9rl7PEfdGiRTwe78GDB2VlZdpH2SwazTyZ3NzclJQUIvrwww8HDRokFAodHBzefvttDw8P\nlUp18uRJ7XOTk5M3bdq0cOHC8PDw1jxyAAAAwPMNiTvo2r9/f0BAABsVXrt2bUBAwPXr19uo\nr/Pnz1dVVZmami5ZsqSJagsXLrSwsFAoFHFxcTqHxGKxWCwmonHjxvXu3dvR0ZGIkpKS9A7J\ny8uLPa6wQXRGKpXm5eWZm5sPHz68b9++HMexPJ6pqKhgU2I0ifuVK1eIyM3NrW/fvppqfD5/\nypQpRHTt2jXt+TBFRUXFxcUikcjW1tbExKSJ2FauXBkQELBz506d8vv37+/du3fVqlVz587d\nuHHjpUuXiGj+/PlsPS4AAAA8H5C4gy4XFxdfX1+RSEREQ4YM8fX17dy5cxv1xR4Jxo0b161b\ntyaq2djY+Pn5EdGNGzd0Dv36669E5OTk5OzsTESjRo0ioitXrtTV1ekXkrW1tZOTE/33bBmW\npnt5eQkEgmHDhtF/z5Zhw+3aE9zZs8Tw4cN1Gh8xYgQRyWQyqVSqKQwKCjr0l5EjR7Y04ISE\nhJCQkMTExIKCAh6Pl52dHRoaGhERgcF7AACA5wwWp4IuPz8/Pz+/lStXFhYWLl++nGWxbSQ7\nO5uIWM7dNBbGvXv3OI7TTODhOI4Nro8fP56VjB49OioqqqqqKj09XY8kmBk0aJBYLNZO3Fma\nPnToUCIaNmzYjz/+ePPmTU0krKaLi4uVlRWrL5PJiMjOzk6nZWtra1NTU6VSWVBQUP/oU02Z\nMqWiokL7juTn54eFhanV6qCgoFmzZllbWxcWFoaHh7NlA+zp65lZtWpV/R9nunfvTkQVFRXa\nzyqNaU6d5mtwYpVBsPtrWNXV1dXV1QZvlogqKioqKioM3qxhb5ZGTU2N9u5ShtLSaOVyOR59\nAaADwog7tJva2lqWqbD5LU1jdVQqVWVlpabw7t27JSUlRDR27FhW4ubm5uDgQK2eLUNEDx8+\nVCgURMRx3K1bt4hoyJAhROTu7m5hYSGTyXJyclh9NqnG09NT08KTJ0+ISJPHa2OrV7Wvovnm\nzJmzdOlS9uMDc+TIEbVa7e/vv3jxYtZyjx49tm3bxtJlAAAAeJ4gcYd2o9ld0czM7KmVNetW\ntYfBWHbu7u6unaey2TKpqalyuVy/wAYOHMjj8VQq1b1794jozz//LCsr69atW69evYhIIBCw\n11GxYXi1Ws2qDRo0SNMCG+u1sLCo3zgrZI8ErcRx3LVr14ho9uzZ2uUCgWDGjBmtbx8AAAA6\nFEyVgXYjEonYvJH8/Hzt4eoG5ebmEhGPx9MMY6tUqsuXL5PWPBlm9OjRMTExcrk8LS1NMxLf\nIlZWVi4uLg8fPszMzPT09GQT3NlwOzNs2LCUlJQbN27Mnj07JydHLpfzeDztNyuxNaYN/uLP\nfmRocDC+pSQSSV1dnUgkqj++7uLi0vr2W+qf//xn/cK8vLzAwEArK6sGlzHoTGBoeqlD89XV\n1ZWWlrKJSQZpkCktLWVrJ2xtbXVendtKUqnUwsKiwSc9vVVWVrJnVysrq+Y8GzdfaWkpn8/X\n7HxqEHK5nP0MZW5ubmlpacCWKysrlUplly5dWnSWSCTi8zGwBQAdDhJ3aE9OTk73799nSXnT\nWB0nJyfNv6bp6els5m5kZGRkZGT9Uy5evKhf4k5EXl5eDx8+vHv3Lv33BHeGfc7IyJDL5Wxl\nqouLS6dOnTQVLC0tS0tLG5xYzLKTlqYRDSouLiYiW1vb+ocaLAQAAACjhhEFaE8sA758+XKD\ng9NFRUVsgFOtVrNt0b29vTVH2fbtQqHQuh62KDM9PV2/qeT01zT3zMxMhUKRkZHB4/G0Z8LY\n29s7OjrW1dXdvn2bTXDXPkpEPXv2JCKdneCJqLq6mg2Cdu3aVb/AtLHsvLS0tP4hnW3mAQAA\n4DmAEXdoT/7+/idPnnzy5MmPP/6os5W7VCoNCQmxs7N7//33k5KSCgoKTExMNFO35XI5m94d\nHBwcHBys02x2dvbGjRvr6upSUlLY1ukt5enpyefzKysrz507p1QqXV1dbWxstCsMHTo0Ly8v\nPT2dbSmjM9XH3d09LS0tPT193rx52uXp6elEZG1trdOafnr06MHn82tqaiQSib29vfahR48e\ntb59AAAA6FAw4g7tyc7OLjAwkIh+/vnn6Oho7dcSde7ceeTIkdnZ2evXrz98+DARBQcHa+ZA\nX716VaFQ8Hi8iRMn1m/W3d2djXnrvbeMhYWFq6srEZ04cYL+e4I7w3Zzv3LlSlFREY/H00nc\nfXx8iCgrK4u9x0qD7To/atQog7ySVigUsn51XsXKcdyZM2da3z4AAAB0KEjcoZ0tXLhw5MiR\nKpXq4MGDmzdvvnDhQk5OjlKpLC8vHzJkSNeuXSsqKtRqtaOj42uvvaY5i82T8fT01Blp1pgw\nYQIR3bp1S+8tt9nsF3a69gR3xtPT08TEhB11c3PTWVbo7Ozs4+PDcdyOHTvY+sva2tqjR4+m\npqYKhcKgoCD9QoqOjj58+HBiYqKmZMmSJTweLzY29tixY1VVVUQklUo/++yzgoICPp9vkMcD\nAAAA6CAwVQbaGZ/P37x585EjR2JiYu7cuXPnzh2dCiYmJmZmZnl5eaGhoevXrzczMysrK2M7\nvTQ43M5MmDDh6NGjHMddvnx55syZmvKNGzc2dkpMTIz2PhJeXl7R0dFEZGpqOmDAAJ3KZmZm\nHh4ev//+O/01IV7HmjVrxGJxbm7usmXLevbsKZVKlUoln89ft26d3pusx8bGSiQSHx8fzYX3\n7dt36dKlBw8ePHbs2A8//NCpU6eKigqBQLBhw4awsDDD7iUCAAAA7QuJO7Q/Pp+/aNEif3//\nhISE9PT0oqKiqqoqW1tbe3v7YcOGTZ06VaFQbNu2jW3+uGnTpuTkZJVKZWZm5uvr21ib3bt3\n79+/f2ZmZlJSknbi3nwDBw4UCAQqlcrDw6PBXQW9vb2bSNxtbGz27t0bFRWVkpJSUlJibm4+\ndOjQOXPm9OvXT49gmhAYGOjm5nbmzJmsrCy5XO7t7b1gwQJHR0e5XG6QvWsAAACgg+BpzyoG\n6LCqqqr27NmzePFiZ2fn9o6lPZ0+ffru3bshISFNVxOLxevWrfP29v7444+fTWCNYfu479y5\nc+rUqfWPSqVSu+T/bNrNvWpukE6xjzthH/e/6LeP++7duxMTE+Pi4gwYCQBA62GOOxgHS0vL\njz766AXP2olIJpPZ2dlp/rhr167ly5enpaXpVIuPjyei/v37P9PgAAAAoC0hcQcwGomJiYmJ\niZMmTdKUuLq6SiSSgwcPlpSUaArPnz8fHx8vEAh03ikLAAAARg1z3OHpDhw4oLPhoEHs37+/\nsT1hoEFlZWVbtmzp06ePpmTWrFnp6el37txZsWKFm5sbW8Urk8mEQuGKFSvYKljcPgAAgOcD\nEnd4uunTp7ONyZ+quLi4S5cuJiYmzanMXvwJzcf2vNcmFAq3b99+/vz5CxcuFBUVyeVyBwcH\nDw+PuXPnuri4sDq4fQDPh4sXL7KNbpsjMDCwLZ7YDcXOzk4qlS5fvjwyMrK9YwEwJkjc4ens\n7e2bObZaf9tEaGtCodDf39/f37+xCrh9ANBe7ty5o3k/HTbDAGg9JO4AAADGZPjw4U08qzN4\nDgd4LiFxBwAAMCY+Pj7bt29v7ygAoB0gcQcAAIA24ebmxl50DQAGgcQdAAAA2oRIJBo8eHB7\nRwHw/MA+7gDQnkp8q6RjqrlXzQ312lQAaND9+/fXrVv3t7/9rVevXiKRqE+fPmPGjNm2bVtx\ncXH9ygMGDODxeJ999hkRJScnBwUF9ejRw8zMzNnZeeXKlYWFhazaH3/8sWjRIkdHR5FI5Orq\nOmvWrN9//12nKRcXFx6P99prrzURW0pKCo/H4/F4jU0BOn78OKsQERGhz8UDPC+QuAMAADzn\nQkND+/Xr9+2336amphYUFCgUCrFYnJyc/Mknnzg7OycmJjZ2YlhY2Pjx48+ePdu1a1cTE5NH\njx7961//Gj58eF5e3r///W9vb+8jR45YWlpaW1v/+eefMTEx3t7eFy5caGl4Pj4+Tk5ORBQV\nFdVghe+//56IzMzM5s6d29LGAZ4nSNwBAACeZ7GxsSEhIWq1WiQSLVq0aO/evfv379+xY8eY\nMWOISKFQzJ49u7y8vP6JZ86ceffdd7dv315eXv7HH388efLk7bffJqKCgoKpU6cuXbp0+vTp\nxcXFWVlZEonk+++/FwgEKpVq7dq1LY2Qx+OxjPz27dtZWVk6R588eRIXF0dEAQEBeIMEvOAw\nxx0AAMCYpKambtu2rYkKy5cv79Wrl+aP3333HcdxIpEoKSlpxIgRmvItW7Zs2rQpNDS0tLQ0\nLS3Nz89Pp52rV69+8sknH3zwAfujqanpl19+efbs2QcPHmRkZIwdO/ann37i8Xjs6Ouvv37+\n/PkDBw5kZGTIZLKWZtjBwcFffPEFEZ04cWLLli3ah44fP15bW0tEixcvblGbAM8fJO4A0J7s\n/viGiLgJn7Z3IABGIzU1NTU1tYkK/v7+2ol7amqqUCh85ZVXtLN2ZsmSJaGhoUSUlZVVP3G3\nsLAICQnRLhEIBCNHjnzw4AERbd++XZO1M2PGjDlw4AARPXz40Nvbu0UXNXz4cFdX14cPH0ZF\nRekk7myejL29/bRp01rUJsDzB1NlAAAAnme5ubm1tbU//fRT/UPV1dXsQ4OvNfX09BSJRDqF\nXbt2JSIejzd8+PAGDxFRXV2dHnGy2TK///77vXv3NIU5OTnJyclEtGDBAqEQo43wokPiDgAA\nYEzWrl3LNelvf/tb0y2UlJRcvXo1PDz89ddfb6Kavb19Y4csLS0tLS31vIBGBAcHsw8nTpzQ\nFB49epQ9VGCeDABhqgwAAMBzr6qq6vvvv79w4UJGRsb9+/c1A+0dypAhQ9zd3bOzs6OiojQT\n648cOUJEgwYNwn7wAITEHQAA4PkWFxe3dOlSzX7t5ubmnp6ebm5ugwcP9vLyanqH9WcsODh4\n+/btN2/efPDggZub282bNzMyMgjD7QB/wVQZAACA51ZWVlZgYGBxcXHnzp0//fTTP/74o6Ki\n4vbt2zExMdu2bau/XLV9abZpZ7Nl2HC7QCBoekoPwIsDiTsAAMBz61//+pdSqSSimJiYDz/8\ncODAgQKBQHP0yZMn7RdaAzw9PT08PIgoKipKrVb/8MMPRDRt2rTu3bu3d2gAHQISdwAAgOdW\nTk4OEQmFwrFjx9Y/evbs2Wcd0NOwJao3btzYv39/QUEBYZ4MgBYk7gAAAM+tvn37ElFdXV16\nerrOoVOnTm3fvp19bvDNqe1Cs7fMu+++S0SdO3cOCAho14gAOhAk7gAAAM+tl19+mb0mKSAg\n4Isvvjh//vy5c+f27ds3adKkV1991crKis2cCQ8P/+KLL9ibldpXv379Bg0aRESVlZVEFBwc\nXH8veYAXFhJ3AACA59aECRPY1oqFhYXvvffelClTpk2b9tZbbyUmJg4ePPjq1auLFi0iovz8\n/Pfee+/69evtHS+R1qA7EbHwAIAxju0glUrloUOHrly5UllZuXXr1mewmatEIklISEhLSysp\nKZHL5d27d+/Zs+fkyZPrL8C/fv36p59+OnjwYM0PjvAsXbp0KTQ01MfHZ/PmzfX/CAAAn332\nmb+//549e+7evfvo0SNbW1tvb+/58+fPnz+fz+fv27fPzc3t4sWLnTt3HjBgQHsHS0QUHBy8\nZcsWIurbt+/o0aPbOxyADsQ4Evfjx4+fPn2aiHr27PkM3ngcExNz5MgRtgyfEYvFYrE4JSXF\ny8vr/ffft7KyMmB3Bw4cOHnyJBGdOnWq+WfNnz+/qqrq8OHDnTt3NmAwAADQMY0fP569Q1QP\nY8eObXBxKhGZmZlt2bKFJcoabPf0BoWFhYWFhTV4aObMmRkZGTdv3uzdu7em8M8//6xfs6Sk\npIloy8rK2AcsSwXQYRyJO1tSs2bNmmnTprV1X4cPH2bbxw4fPnzWrFl9+vQhIolEcvHixbi4\nuNu3b3/55ZcfffQRmzIIAAAADJ/P79+/f//+/VvZTmRkJBHxeLyFCxcaIi6A54dxJO5VVVVE\n1Pr/FzxVeno6y9qDg4Nff/11TXZuZWXl5ubm7e39ySef3Lhx4//+7/+mTp1qqE49PT0N1RQA\nAIBRe/z48dGjR4lo/PjxbOwMADSMI3FnnsEg93fffUdEo0aNWrBgQf2jgwcPnjx5cnx8/MWL\nF1uZuKtUqj/++IPjuCFDhowYMaKjvbsOAADgWaqoqFCr1VKpdNOmTWxvyrfeequ9gwLocDp6\n4r5//37NzO+1a9cS0UcffTR8+PC26Ovu3bu5ubn03+vZdUyePDknJ6eurk4ul+tsUMVx3M8/\n/3z+/PmioiJLS0s3N7d58+a5u7trKrBlrAEBAYGBgZ988smjR48cHR3/8Y9/1F/eqlAoYmNj\nk5OTCwsLVSpVjx49fHx8AgICzM3NiWjXrl3JycmsJltu/+WXX7Kdetm5P/300927d//880+B\nQODk5OTv7+/r66v92MN6nD59+sqVK3/55ZdffvklPz/fwsLCzc1t/vz5L730kvZ1NR2MdjXD\n9tvMNpvj9u3bp0+fzsnJefLkibm5uZ2d3ZgxY/z9/S0sLHRqFhUV/fTTT/fu3SsoKOjWrZu7\nu/u8efN03tjX/Kia328z21y5cmVhYWH9pbf3798/ffp0VlbWkydPnJycXn311bFjx86fP1+l\nUh0/frxF3xUAQLs4deqU9sSYYcOGzZkzpx3jAeiYOnri7uLi4uvre+PGDblcPmTIEEtLy7Zb\ni/n7778TUd++fV1dXRur069fv9DQ0PrlarX6888/T0lJsbCwsLOzKyoqun79+s2bN3ft2qWd\nuxNRbW3tp59++ujRIx6PZ2trW7+pmpqaTZs2icViIjI3N+c47sGDBw8ePEhJSQkNDTUxMenX\nrx8RXb16VaVSjRgxwtTUtFOnTuxcsVj8+eef5+fns3Orqqpu3bp169atUaNGvfvuu6ampjp9\nffvtt2lpaT4+Pr6+vrdu3aof81ODaaN+9WizMWfPno2IiCAiHo9nZWVVVVVVVlZ2//79K1eu\n7Ny5U7udlJSUr7/+urq6mogsLS3z8vLy8vKSkpJ27NjBXsHdoqia328rrzQhIeEf//hHbW0t\nOz07Ozs0NPTu3btqtbqZXxEAQIcyYMCAU6dOYS0ZQH0dPXH38/Pz8/NjA43Lly93cnJqu77u\n379PRD169NDj3D/++EMkEoWEhIwePZrH4xUXF3/66ae5ubk//fQT20BXIzExUSgUvvPOO2PG\njGkwJ/vhhx/EYrGzs/PGjRudnZ2JKDMzc8eOHQ8fPjx37tz06dMDAwPpr11l1q1bp3mSUalU\nu3fvzs/P9/b2fuONNxwdHZVKZUpKyv79+1NSUv7973+/8cYb2h1dvXrVysrqq6++Ys8Pr732\nWnh4eHx8fFRUlGZ7gacG00b9trTNxlRUVOzfv5+IXn/99cDAQJFIxHHczZs39+zZk52dnZaW\n5uvry2pKpdI9e/YolcqZM2fOnTvXxsamtLQ0IiIiOTn5iy++iIiIMDExaX5Uze+3RVc6ZcqU\niooK7b8F+fn5YWFharU6KCho1qxZ1tbWhYWF4eHh7DXmz/itJZcuXaq/U4RCoSCi2tpauVze\nxLlNH20p9tCiVCoN+/Si2dBDoVDw+QZ+CQb7Hc+ADapUKvahtrZW761IGsS+VcNGy548qW2+\nB47jWtomO8uAYcBTzZw58+TJk5mZme7u7gEBAc9gBzkAY4S/GP/BJtU5ODjocS7HcW+88YYm\nG3NwcPif//mfnTt3Pnr0SKemXC7ftGnTmDFjGmvqzp07RBQcHMwSZSLq379/UFDQhQsXSktL\nm4jhl19+yc3Nfemll7Zu3crehGdqajp+/Hh7e/v333//9OnTc+bMsbGx0dR//Pjxhg0btEf9\np0yZEh8fz8bXmx9MW/Tb0jYbk5GRoVKpXnrppXnz5rESHo83bNiwiRMnnjp1KicnR3PL2Aag\nEyZMePPNN1lJ586d33333fv37xcXF2dnZw8cOLD5UTW/3xZdaf0fjo8cOaJWq/39/TWbpvXo\n0WPbtm2rV68uKip66vdjWN9//339t7ewiUZyuZy9BLExTR/Vj2HzP23sZxnDUiqV2lvQGlBb\nfA9qtbotbhkR1dbWapJ4A2pptAZ/2oGnsra2ZsNSANAEvDn1P9i/FmZmZnqcKxKJ/Pz8tEt6\n9epFWsNIGlZWVpqkrUEsgOvXr2sGzIho1qxZYWFhDS6Z1UhLSyOiV199leV/GgMGDHBzc1Or\n1bdu3dIut7a29vLy0i5hg/d1dXUtCqYt+m1pm40ZPnz4Tz/9tHv3bp1ydl+0L+ratWtEpPPP\nhlAoXLZsWXBwMAuj+VE1v9/WXCnHcSzs2bNna5cLBIIZM2Y0dhYAAAAYKYy4/wd7rZJ+45Q9\ne/bUSbwa+yXdzs6u6Xl7r7zyyu3bty9cuPD777+PHDnSw8NjwIAB9vb2T40hLy+PiC5cuMCS\nOW1sdLy4uFi7UGfNpd7BtEW/LW2zMXw+n92I6urqvLy84uLioqKi7Ozs1NRU7WplZWVVVVU8\nHk/zw4LGqFGjRo0a1dKomtlvK69UIpHU1dWJRKL6X6mLi0tjZ7WdadOmDRw4UKeQ47jDhw+b\nmprqrGZmampq2IcGj+pNrVYrFApTU1Odv5WtpFAo2CwRkUhk2Nm3NTU1QqFQs2jEIGpra9nD\ncFt8D6TvGEdjVCoV+8HB4N8DmzHV0mljAoEAE6wBoANC4v4fvXr1+v333wsLC5uu9uWXX1ZV\nVU2ePFmTz9FfSX9zPLWmr6/vp59+evTo0aysrLi4uLi4OCJycnKaNm3aK6+80tg/wBzHSaVS\n+utlVQ3S+cW8OanSU4Npi371aLMJ169fP378eFZWluaHbzs7u969e2vPYmLJsY2NTRP5TUuj\nak6/rbxSFnaDS5wbLGxrs2bNql+Yl5d3+PBhMzMzS0vL+kc1iXuDR/VWV1enUChEIlHzFzE3\nR21tLUvczc3NDZsK19TUmJqa1t9uqDUqKytZ4m5mZmbYJLu2tpbP5xv2lsnlcpa4m5iYGLZl\njuOUSmVL2zQxMUHiDgAdEBL3//D09IyNjb13755EImlshLugoODXX38lotdee63tIhkyZMiQ\nIUNkMtlvv/2WkZFx+/btR48eRUZG3r17NyQkpMFTeDxely5dpFJpWFhY/WHjtgumLfo1YJtJ\nSUl79uzh8XijRo0aOnSoi4tLjx49rKysjh07pp1AszS3srKS47jG/rVuUVTN7LeVV8rCbnDl\ng+aF4QAAAPDcQOL+HyNHjrS2ti4vLz969Og777zTYJ2kpCQiMjc3r7/puEGoVCq2L0f37t1t\nbW0nTpw4ceJEIkpISPj666+Tk5NlMlljg6m9evWSSqU5OTn1U8CSkhKVStWlS5cWDUA2MxiD\n92vAa/nxxx85jlu+fPnMmTO1yzUDvUy3bt1MTU2VSmVBQQFbnKBx/fr1tLQ0T0/PsWPHNj+q\nZvbbyivt0aMHn8+vqamp/6hZf1U0AAAAGDssTv0PU1NT9uqlxMTEmJiY+lsKPHr06MSJE0Q0\nefJkw/5QrqFUKleuXLlixQr2KiiNcePGsR6b2Hdi6NChRBQTE6OzBV5BQcGKFSvWrVunvfrT\ngMEYvF8Dtvn48WMi0pl4rVQq2ZJQDR6PN2jQINajTgvHjx+Pi4tj3TU/qmb228orFQqFnp6e\nRHTy5Entco7jzpw509hZAAAAYKSQuP+XGTNm+Pj4cBz3v//7v9u2bUtOHLgNDQAAIABJREFU\nTi4qKpLL5WKx+MSJExs3blQqlb1799ZsvWdw5ubmbJfuffv2abbELisri4yMVKlUdnZ2OrtV\nshnSmuDt7e0fPHiwc+dOzRLb7Ozs7du3q1SqKVOmtHQGbTODMXi/BmyTxX/u3DnNRi5isXj7\n9u3sbUe5ubmax7MlS5bw+fxz584dPnyY7Rwnl8u/++67zMxMGxsb9rLe5kfV/H5bdKXR0dGH\nDx9OTEzUlCxZsoTH48XGxh47dqyqqoqIpFLpZ599VlBQwOfzMUkXAADgeYKpMv+Fx+OFhIRE\nRkbGxcWlp6fXXzLYv3//9evXG3bFm4633npr8+bNf/zxx/Lly62srHg8Xnl5Ocdxpqam7733\nnqZaly5dqqqqtmzZ0r179w0bNjg7O5uamm7cuHH37t2pqampqamWlpYcx7ENp4cNG9bMNxbp\nEUxb9GuoNl9//fWPP/44Njb20qVLdnZ2paWlT548cXR0fPvtt8PCwq5duzZv3rwDBw5YWFg4\nOTmtWrUqMjLyxIkTJ06csLGxqaioUKvVpqamGzZsYEuKmx9V8/tt0ZXGxsZKJBIfHx82Z4mI\n+vbtu3Tp0oMHDx47duyHH37o1KlTRUWFQCDYsGFDWFiYYZckAgAAQPtC4q5LIBCsWrVq2rRp\nCQkJv/322+PHj1UqVY8ePXr27Dlq1Khx48a1dQADBgz45ptvfvrpp+zsbKlUKhQKXVxcvLy8\nXn311W7dummqrVq1KiIioqCgQCaTaQr79+//zTffsM1MxGKxSCRyc3Pz9/cfM2aMfoOvzQzG\n4P0aqs0hQ4b8/e9/P378+J9//llcXMxamD17tqmpqZmZ2blz5wQCgeb9fP7+/gMGDIiOjn74\n8GFRUZGjo6O7u/u8efO0p483M6oW9dvKKw0MDHRzcztz5kxWVpZcLvf29l6wYIGjo6NcLu/S\npUtzv24AAADo8Hh4ORxoq66urqio0O/1scbYr9E5ffp0E5sLaYjF4nXr1nl7e3/88cfPJrDG\n5OXlBQYG7ty5c+rUqfWPSqVSuz++ISJuwqcG7LSurq60tNTa2tqwP46VlpayJQe2traGXeUi\nlUotLCwMvh0k20vUysrKsL+9lJaW8vl8a2trA7apebeuubm5YbeDrKysVCqVLX2I3b17d2Ji\nItsAFwCg48Acd/gvFhYW7ZI9t1e/Rkcmk9nZ2Wn+uGvXruXLl9df8xofH09E/fv3f6bBAQAA\nQFtC4g5gNBITExMTEydNmqQpcXV1lUgkBw8e1KweJqLz58/Hx8cLBILx48e3R5gAAADQJoxv\njvuBAwd0Nr8ziP379zf20iWADqKsrGzLli19+vTRlMyaNSs9Pf3OnTsrVqxwc3MzMzPLy8uT\nyWRCoXDFihXdu3cn/JUBAAB4Xhhf4j59+nQfH5/m1CwuLu7SpYuJiUlzKrfLK+IBWiQwMFCn\nRCgUbt++/fz58xcuXGBblzo4OHh4eMydO9fFxYXVwV8ZAACA54PxJe729vbNHOcbMGBAWwcD\n0O6EQqG/v7+/v39jFfBXBuD5oAh5u71DILNd37R3CAAvNMxxBwAAAAAwAgZL3MVi8dWrVzV/\nrK2t3bx5c58+fTp37vzyyy/HxsYaqiMAAAAAgBeQARL3vLy8MWPG9OnT58MPP9QUbtiw4e9/\n/7tYLC4rK/vll18CAgIiIyNb3xcAAAAAwIuptYm7QqHw9fVNTk7WLnzw4MG+ffuIyMHBYfr0\n6dbW1iqVav369VKptJXdAQAAAAC8mFqbuP/zn/989OgREU2aNOmdd95hhTExMSqVSiQSXbt2\n7cyZM1evXrW0tKypqQkPD29tvAAAAAAAL6TW7ioTExNDRIMHDz537pzmHeBsRvv06dOdnZ2J\naMCAAXPmzDl06FBSUlIruwOA50yJ59s8Hq+9owAAADACrR1x//PPP4lo7ty5mqy9rq6OrVL1\n8/PTVBs2bBgR5eTktLI7AAAAAIAXU2sT9+LiYiJiL2hkbt26VV1dTUSjRo3SFLJ3tRQUFLSy\nOwAAAACAF1NrE/fevXsTUUlJiabk119/JSJra2svLy9NoUwmI6KuXbu2sjsAAAAAgBdTa+e4\nu7q63rt3LzY2NiQkhJWcOHGCiKZMmaKZPEN/ZfMsywcA0PjlnJ0BW1v4OmfA1gAAADqU1o64\nv/baa0SUlJT0wQcfZGZmfvbZZykpKUQ0ffp0VoHjuO++++7UqVP010x3AAAAAABoqdYm7osX\nL3Z1dSWizz//fMCAAVu3biUiR0fHefPmEZFUKu3Vq9fy5ctVKpVAINiwYUPrIwYAAAAAeAG1\nNnEXCoXnzp0bMGCApsTS0jIiIsLc3JyIFApFYWEhEQkEgs8//9zNza2V3QEAAIDRkcvl77zz\njqOjo6WlZUJCwjPoMScn55NPPhkxYoSDg4OlpeWgQYOCgoLOnDnz1BP37dvH4/F4PJ6Li8tT\nKxcXF+/evXvChAmOjo4ikcjFxWXKlCnvv/8+S34ADK61iTsRubm5paWlnTp1auvWrXv37v3t\nt99eeeUVdojH4/Xr12/BggUJCQkbN25sfV8AAABgdHbs2PH111/n5+f36tXL1NS0rbvbs2eP\nh4fHtm3brl+/LpFIqqurb9++HR0dPXPmzEmTJj158qSJc48ePco+5OTksO2tG/Pll1/27ds3\nJCTk4sWL+fn5CoUiJyfn/Pnzu3btcnV1/eCDDzhOn1U3YrGYz+fv2rWrsQq//fbb3LlzBw4c\naGlp+dJLL82YMeP//u//GqwpkUjWr1/v7e1tZWXl4uIyY8aMc+fO6RESdBytXZzKWFpazpw5\nc+bMmTrlPXv2zMzMNEgXAAAAYKTi4+OJKCIiYsWKFW3d1+bNm//+978T0SuvvLJx48bBgwdz\nHCcWi48ePbpv377ExMSFCxeePXu2wVe/icXi5ORkIrK1tZXJZMeOHfPx8Wmwl9WrV//zn/8k\nIi8vrzVr1gwePHjAgAGPHz/+448/9u3b98svv3z++ec1NTVfffVVS+OPjIxsIuMPCwv7f//v\n/6lUKhbkgwcP7t+/f/bs2bfeekvn/fRXrlx5+eWXy8vLicjc3FwsFufk5Jw9e3bFihUREREt\njQo6CAOMuAMAAAA0obS0lIhGjx7d1h3Fx8ezrH3r1q1nzpyZOHFily5dunbtOmzYsC+++OLU\nqVNCoTAuLu67775r8PRjx45xHOfi4vLOO+8Q0fHjx1mKrCMiIoJl7SEhIdevX1+5cqWPj4+N\njY2rq2tAQEBcXNw//vEPHo/39ddfR0dHNz94hULxr3/96/PPP2+swoMHDzZt2qRSqT744IOy\nsrInT55UVlay6/3H/2fvzuNi3v7HgZ9p2rSH26qSihAhupElKkIipT4Jt+zbJYS42enaqute\nW/bclFuWqLTQKupKY4lokaJ1Km1TpmmW3x/ne+c3j5lpzDQz1tfzr7znvM850zszr/d5n/M6\nJ0/iRCBYa2urp6dna2vrzJkzi4qKOjo6KBTKqVOn5OTkzpw5c/nyZeF7Bb4qELgDAAAA4HPg\nO8gtWTgNhqur6759+3ibs7e39/X1RRzzYbjg415eXu7u7gih2tpanM+aE4VC2b59O0LI29v7\n4MGDfGf+rFq1CmfpwPH9J8XExIwbN+6nn35asWIF31sFLCwsjEqlOjk5BQUFqampIYSUlJQC\nAgK8vLwQQpw3Cffv36+srNTS0oqJiRk4cCAuuXLlSvz7uXr1qjC9Al8hcafKTJ8+XfjCbm5u\nS5cuFbNFAAAAAHwrNmzYwJ4uYmFhgRCKj49n54yWrOzs7MLCQoTQzp07uyuzePHigoICGo3W\n3t6urKzM+VJBQUFBQQFCyMvLa8iQIebm5q9fv46KirK3t+csduXKlQ8fPqirqwcHBwvoTFBQ\n0LRp0wgEAovF+uQdS1lZ2du3b5WVlZWVlZuamjo7O/kWe/nyJeLIuM02YcKEqKgo/CqGfw/m\n5uY4WQiblZUVux7wLRI3cE9KShK+MP4fCwAAAIAfhKWlpbu7e2JiYnt7u6Ojo7q6uo6OjpTa\nunfvHkLIyspqxIgR3ZWxsbHBG87wunLlCkJo6NChOFyZO3duUFDQ9evXT548yTmsjtd3Tp48\nWVtbW0Bn+vfv379/fyF7vnXrVvZGlvPmzcN7WfJasGDBjBkznJycuI7jDez79u3LPoJT4rx4\n8aK1tRWPzWMPHz5kvwq+ReIG7t3936DT6eXl5RQKBf/TycnJ3Nx8woQJYjYHAAAAgG+Ij4+P\nj4+PmZlZaWlpaGjo0KFDpddWfn4+QsjU1LQH57JYrKioKITQ/Pnz8RE3N7egoKDm5uakpCQX\nFxd2STyYLdU30h08JYYTg8FISUkJDQ1FCHEu/J0zZ46tre2DBw+cnZ0PHz48fPjwxsbGy5cv\n//nnnwoKCnv37v2s/QaSI27g/uTJEwGvvn///o8//jhx4sS///7722+/jR8/XszmhEGj0cLD\nwx8+fEihUAIDAy0tLaXdIplMTk1NzcvLq6+vp1KpOjo6enp6Dg4OY8aMEXxiYmLiqVOnEEJa\nWlrnzp0TXLi5uTktLe3x48c1NTWtra2ampq6urqmpqYuLi6ampoSezPfCy8vr/b29s2bN3d3\nu3jlypV//vlnyJAh7GVAbW1t3t7evCWVlJT09fXNzc09PDzU1dW7a1GkqwkAAEDiGhoaEEJ4\nX0hRZWdnv3v3DiGE56YjhEaNGmVsbPz27durV69yBu6NjY18W6mqqurXrx9vzdeuXXNzc+tB\nlwTbs2fP5cuXyWQyhULR0tK6cOGCq6sr+1VZWdmkpKSVK1deuXJl7Nix7OMGBgbR0dHdpcoB\nXz/JpIPsjoGBQXBwsKOj44wZM+bOnfvs2TNdXV2ptogQio6OjouLQwjp6enJykr3DSKEYmNj\nIyIiaDQa+0hFRUVFRUVOTs6wYcMCAgJUVVW7OzczMxP/QCaTi4qKBg0aJKCVyMhIKpXKPkIm\nk8lk8rNnz+Li4lxcXBYuXCjMop/Fixfjz7XujBkzZseOHXxfotPpW7ZsKS0tPXPmjPQedH6F\nOjo6SkpKSkpKUlNT9+zZ091lEv5qsostW7Zs0aJFn/xAF1zy7t27f/31F98T7e3t169fz3mk\ntbX1n3/+KSoqqqqqkpeXNzIysrGxcXJykpERtE79h730AIBvC54arqSk1INz8bLUn3/+mTMi\nnzt3bnBw8O3btzknxOP6yWSyBHoshsrKyrKyMjk5OQKBwGQyyWQyk8nk/DDPzs7OyspCCMnJ\nyRkZGTU2NjY1NVVXV1+9enXUqFGfIZs+kAapx7UIIScnp8WLF58/fz40NPTw4cPSbo5EIiGE\n1qxZM23aNGm3dfnyZTwRbfTo0a6urng2G5lMzszMTExMLCgoCAkJ2blzJ9+Qmkwmv3r1CiGk\noqJCoVCysrK6C/VOnTqVmJiIEDIyMpo5c6axsbGBgUFra+u7d+/u3LlDIpGuXbvW2dm5bNky\n8d+RgADu0qVLpaWl4jfx9Tt69Cj7SSudTi8rK8vOzk5ISOjo6AgODj5+/Djv553wV5MtJSVF\nyL05BJesqqoSphKEUGFh4YEDB9ra2hBCioqKra2tT58+ffr0aUpKyt69ezknQXL5cS49AOCb\n1rt3b4TQmzdvRD2xq6srJiYG8cxFcXNzCw4Obm9vj4uLY4/Ea2lpVVRU8H4q6uvr0+l0ziPO\nzs6cSwE9PDzwZHRMRUUFjzP2zNmzZ8+ePYsQevv27YoVKwICAsrLy/GDX4RQcnIy3l0nNDR0\n9erV+Gvr0aNHy5YtO3bs2Pv3769fv97jpsEX9DkCd4SQnZ3d+fPnb9++/RkC9/b2doSQubm5\ntBvCETNCyNPTc/78+ezoXFVV1cTExMrKavfu3fn5+Xfv3p06dSrv6VlZWSwWS1tb297ePjIy\nMjs7e8mSJbxxc1JSEo7a3dzcvL292c8QlJSUdHR0rK2tExMTT58+HRcXZ2Fhwfk4jK+DBw8y\nmUze4wwGIygoqLq6et68eXxPfPToEWeC2O+bjIwM+0LIy8ubm5ubm5ubmZkFBwfX1taWlZXx\n/nUJeTWxrq6utLS07tYeiVoSB+5Lly61trbmeokzmQCdTj9+/HhbW9vQoUNXr15tYGBAp9Pz\n8vLCwsLKysr+/PPPwMBAvvX/UJceAPBNGzRoUGpq6icHGhYuXNjc3Lx48WL23JLk5GQ8AcbP\nzw9ncOcSFRXFDtxHjx6dl5eH889wIRKJnP/k2oMyJyensrKS/U8NDY1PvyUhGBsbR0RE6Ojo\nnDlzZtu2bYaGhgihTZs20en03bt3c74da2vruLg4c3PzGzduZGVlTZw4USIdAJ/TZ8rjrqWl\nhRDi/HuVts+QLBZv3zB27Fhvb2/e5iwtLR0cHBDHDAou+PjEiRNtbW0RQk1NTbyfAlQq9e+/\n/0YI2dnZ/fLLL3xn/kyfPh1P48bxvWBaWlo6/OTk5Lx//37hwoU42yuXhoaGY8eO9ezh43dj\n0qRJ+DeAJ0FyEeZqIoQePHiwZcuWBQsWnDhxgu8dVA9KVldXI4SGDRvGe1k5Z+RnZ2dXVlYq\nKytv377dwMAAISQrKzt27Njt27cTCIRHjx7xfV9w6QEA35BJkyYhhPLy8ioqKrorU1JSEhER\nER8fz5kTBs+TkZeX78sDz5BJSkpqamrChR0dHRFCjx496u77HWtoaCgvL+c88v79exYHdoVC\nqq2tXbVq1a+//sr7kpaWlp6eHpPJxM9+KRQKTvjIOesdMzQ0xBkhu0utA75ynylwx9NXuBKm\nSty5c+dcXFxqamoQQmvXrnVxcXn8+LGU2iosLHz//j1CyNPTs7syDg4OgwYNotPpnHPTMTwP\nHiE0ceJEAwMDvJwFz0XjlJGR0dbWpqSktHjxYgGdWbRokZ+fn52dnZBTL7iUlpZeuXJlxIgR\nvP/DEUJMJjM4OLitrW3t2rXd1bBixQoXF5egoCDemkNDQ1euXOnh4eHv73///n2EkJeXl4eH\nRw/6+cXhR428U0qEvJoIodra2rq6OkVFRU1NTTk5OQFtCVmSyWTW1NTIyMjwXRHF6e3btwih\ncePGcS26GDhwID63pKSEt/JPXnoAAPh6uLi49O3bt7OzU0Aed5w6RlVVlf2Usr29/datWwih\nHTt21PNIS0tDCNFoNPb2RrNmzTIyMkIIBQQECOiM4Fd7oHfv3hcuXDh+/DjvbQmdTseTcPDd\niIKCgqKiIkKoq6uLtx58UECuBfA1+xyBe2lp6dGjRxFCZmZmUm3I2NjY1tYW/7GOGDHC1tZW\nUs+heD179gwhZGpqKmD1+qBBg44cOXLo0CHcJU54JzZDQ0P8nx9PcXn48CHX9DictGf48OGC\n34iWltaUKVOmTJnSg+cMdDo9ODhYTk7Oz8+P7+l4T4fp06eLulV1amrq1q1b09PTq6urCQRC\ncXHxkSNHwsLCBI8ff7VKS0ubm5uVlJR4U6AKeTURQm5ubuH/4Z3Z0oOSZDKZTqfr6ekVFRUd\nPHhwzZo1GzZsOHbsWF5eHlfJxsZGNTU1vv8H8YwazgXWWI8vPQAAfBGKiop41l9ERERwcDDv\nYNbLly9///13hJCvry/7IXZsbGxHRweBQFiwYAFvndbW1viTE0f8CCFZWdkjR44ghHJzc11d\nXZubm3nPCgsLw4/lJUheXn748OEIoUuXLnG9dPXqVRqNpqCgMGTIEISQnJwcTqnHu0Hsq1ev\ncGjxycR34Osk7hz37jKQYEwms6ysLC4uDs87xxsIS4+9vb29vf2KFStqamqWLl2Kp3lJCZ5C\n17MkOSwWCw/H4od6CKFx48bFxMS0t7eTSCTOKA0P6kv1jSQkJFRVVXl5eeE1PVyeP38eHR1t\nZGS0ZMkSAZU4Ojq2tbVx9rOqquqvv/5iMplubm6urq5qamo1NTUnTpxISEhACPHeyXy1WCzW\nhw8fXrx4ER4eTiQS169fz9V54a+mNOAJ7jU1NXj/bezNmzepqakTJkzw8/Njj9b7+/vzraGu\nrg4v5OLKfCzkpecUFBSEn9JywgP8HR0dfL/bJA63gr+t29vbOzo6JFg5ex/y1tZWiU/Go1Kp\nvPdO4mDfIbe3t3/8+FGCNTMYDAaDIdkLyg6wOjs7+Y4R9hiTyWQymaL2trOzs2cPMMEXt27d\nuoyMjNjYWH9//3v37i1evHjUqFHa2tpv375NSEg4cOAAlUodPHgwOxEw+m/fpUmTJnW3X9KC\nBQt27dqVnp5eW1uLM2vNmzdvw4YNoaGhsbGxz58/DwwMnDJlio6OTlVV1fPnz48fP56amjpg\nwAAzM7Pk5GQJvrstW7Z4eHj8/vvvampqS5YsUVNTa2tru3r16ubNmxFCO3fuZOdOCAwMdHFx\nCQ0NlZeX9/Pz09bWplKpKSkpfn5+NBptxowZo0aNkmDHwGcjbuC+f/9+IUsOHz589erVYjb3\n9WhtbUX/PZMSVWFhIX6kxU4xbmJioq2tXVdXl5WVxRnq4QQgvAn4GhsbfX19eWsOCAgQaXC0\nvb09OjpaQ0OD7ySZlpaW4OBgeXn5LVu2yMvLCxgp570li4iIYDKZTk5Ov/zyCz6iq6u7Z8+e\nVatW1dbWCt/DL2Ljxo18j69evZp3+a/wV1MacODOYDCmTp3q4OBgZGRUX1+fmpoaGxt7//59\nPT09vpnp2VpbWw8ePMhgMCwtLTkH44W/9JzevXvHG7jjv14Gg8H7/EEaOFthx9kSJ42acXwp\n8WqlV7OULuhX0ls8BVni3QCfAYFAiImJWb9+/enTp5OSknj3dx83btyFCxfYa/fr6+vv3r2L\nEFq0aFF3dXp7e+/atYvJZEZHR69btw4fDAkJ0dbW3r17d1lZGe901smTJ1+5ciUpKUmygfu8\nefN++eWX8PDwjRs3bty4UU1NDUcj+CX23qsIIWdn5wMHDuzcufPgwYMHDx5UVVWlUCj4r3rU\nqFHh4eGfYSkgkAapT5UhEAimpqZr1659+PDh95Q0FI8JKSgo9OBcPEA7cOBAzogcR4SPHj3i\nnBCP65feUGV0dHRbW5uXlxfvEDiLxQoNDW1qalq+fDleyyg8Fov177//IoTmzp3LeZxIJDo7\nO4vZ5y/o5MmTISEhXBGA8FdTGtTU1Ozs7JYvX7527Vpzc/NevXoZGhr6+vri+6Xr169/+PCB\n74ksFis9PX3t2rVv3rzR1dXlvFcR59IDAMAXJysre+LECRKJ5OfnN3ToUA0NDWVlZUtLS3d3\n98jIyAcPHnCm642JiaHT6UpKSgImBZiYmOBPdfZsGWzr1q2vX7/+7bffRo4cqampqaCgYGho\nOG/evJs3b6alpenq6k6dOtXPz69nO7l259KlS/Hx8U5OTiYmJl1dXebm5q6ursnJydHR0Vw5\nbbZt21ZQUODj4zNixAgmk2loaDht2rTz58/n5eX17dtXgl0Cn5O4I+6fHMYgEAiC93b5RuEJ\nAD0YPGYwGNnZ2YhjZgU2bty42NhYKpWal5fHHrtVV1cnk8l4uS2nPn36xMbGch7Zu3cvXgGM\nHTp0iH0XjhBSVFTkndTU1taWkJCgqqqKs99wuXHjBolEGj9+PF4+LxI88VpRUZH3WYGxsbGo\ntX1+ISEhnJ+zLBarpaUlLy/v/PnzGRkZysrKK1aswC+JdDWlwc7Ozs7Ojvf47Nmzr1+/3tbW\n9ubNG95JUC9fvjx37hyeITN+/PgVK1ZwrlLq8aXfv38/3v2EU319/dKlS5WVlfnu7ytqUoVP\nwq0wGIzW1lYVFRXBK4BF1draisfa1dTUuL4gxdTU1KSoqMiZvlN8HR0d+HIoKytLdtCktbVV\nRkZGRUVFgnV2dnbieU3S+D3QaDRR1zspKCh8l99cXxDv8ndps7S0DA0N/WSx1atXCzMd4OHD\nh3yPGxkZ7d+/v7vZB/r6+sL0gQvOKC/AzJkzZ86cKUxV5ubmFy9eFLUD4GsmbuAu2W+vb4i+\nvv6zZ894Q2ouISEh7e3tDg4O7CkWJBIJT4Bhb53AJTMzkx3qmZqalpSU8E1rxfWlwpVqs6io\niHOHVL75fO7evUuj0aZNm8Yb3JSXl0dERGhra/csnUhdXR36L4TiwvfgV45AIGhoaDg6OtJo\ntLCwsJSUFB8fH/wwRKSr+TkRicR+/fq9evXq3bt3nCuQOjo6wsPDk5KSWCyWiYnJokWLRo4c\nyXmiOJee7xAOfjIrIyPzeT4rcCtSapT9ZJlIJEr87Uivt9KomUAgSLZO9geaxGvuWW9hFgEA\n4Ov0mTZg+v5YWFjcuXOnpKSETCbjLPW8qqurcb4Rzl2NcNpXWVlZ3tzYNBqNSqWSSCQKhYJH\ns0aMGJGYmFhcXPzixQsLC4vuOtPa2sq19/KFCxcE95/FYuGZf3w3h6qurmYwGHV1dez9Jjgt\nX74cITRmzJjulibj6JzvDJ+WlhbBHZMUZWXl9vZ2vCqaLwqFgkRMUTpmzJiwsLCurq66ujq8\nElekq/mZ4UFWzo41Nzdv27atqqpKU1Nz6dKl48eP541OxLz0AAAAAJAeCNx7yNraGi8KiYyM\n5LvLGvpv9nOvXr3Yy/6oVCqe/O3p6cmbAL64uNjf359Op+fk5OBZCtbW1lpaWmQyOTw8HCef\n4is8PFzU/pNIpNraWjMzM5zBkIusrCxvRMtisfCz7F69esnIyAjIDKOrqysjI/Px40feuxq+\nu/xIw4ABA8hksoD98/BzW5GmHrIXq+GYWNSrKXE0Gm3jxo1dXV27d+/mTXCEH8KwU/3QaLSd\nO3dWVVWNHj3a39+/uz2VxLz0AADwSZs3b8ZJoiWrvLyc7zcaAN8TCQTuLBbr9OnTiYmJhYWF\nn0y2EBQU5OXlJX6jX5y8vLynp+fZs2fT09P79+8/e/ZsrsHLd+/e4Z3qHRwc2E9pc3NzOzs7\nCQTC5MmTeescOHCgnp5edXV1VlYWDvWIRKKvr++hQ4eKioqCgoIozIrwAAAgAElEQVTWr1/P\nG1QlJSXhFfEiwUPF3eU8sba25lqCgxBiMplz5sxBCB07dox38jonWVlZCwuL58+f37x5kz0d\nHCHEYrHi4+NF7WrPDBkyJDc3NyMjw8nJiTc6T0tLKy4uRggJeI7BC89xVFRUxNmERL2aEicv\nL6+jo/Po0aPY2NhVq1ZxvnT//v3GxkZVVVX2PgPp6enl5eWmpqa//fabgGkDYl56AAD4pDVr\n1uCPlE8qLy/X1dUVMg8EfDqBH4G4gTuLxZo9e3ZcXJyQ5b+JtYlCcnZ2LigoyM3NvXDhwtOn\nTx0dHU1MTDQ0NOrq6vLy8qKjo2k0moGBATsfIvovXLawsOhudo2dnV1kZOTz58+bmprwhBNb\nW9vZs2ffunUrNze3vLzcw8Nj+PDhmpqajY2N5eXlCQkJz54909HR0dPT41ycKhiLxcL7L+Cd\nGsR348YNCoViYGDADmF9fHw2bdp0584dNTU1FxcXZWXlhoaG06dPV1dXy8jIfIb5o7Nmzbp/\n/35JSUlgYODs2bPHjRunpaXV1dVVU1OTnJyMd8KbPn36sGHDPlkVi8VqaGjIzs7GuX7t7e1x\n/3twNSVu1qxZeXl5iYmJKioqrq6uKioq7e3tmZmZ+CGMr68ve50fTkk2YcKExsZGvlWpqanB\nUDoA4DPo379/dxnTudja2kq5LwB8Y8QN3K9cuYKjdg0NjdGjR38ywRDe9Ov7QCAQtm7devbs\n2cTERBKJxBs3m5ubr1+/np3PoaWl5enTpwghvgO0GA71WCxWdnb2rFmz8MElS5ZoaGhERUXV\n1tb++eefXKcMGzbM398/Pz9f+MC9pKSkpaWFSCRypsQSx507d8hkso2NDfutmZqa+vr6Xrp0\nKSoq6urVqyoqKm1tbUQicePGjX/99VfP0miKhEgkbt26NTg4+NWrV1FRUVyjyAQCYcqUKUuX\nLuV7bnd53BFCRkZG+E6sx1dTsiwtLRcuXHj58uWYmJiYmBglJSU8p4VAIMyePdve3h4XY7FY\neJLSxYsXu8swsHHjRr4JagAAAADwlRA3cMc7+k6dOvXq1avfYsIQMRGJxJUrV06bNi01NfXp\n06eNjY0MBkNXV1dPT2/s2LETJ07kLPzgwQMGg6GgoCBgCEFHR8fc3Pz169dZWVmcoZ6bm9uE\nCRNSUlIeP35MJpM7Ozs1NDQGDhw4adIkGxsbhNDIkSNdXFyE3Mk1Pz8fIWRmZibVzPpz5swx\nMTGJj48vKiqiUqlWVlbe3t79+vWjUql8d2mVOC0trUOHDv3777+ZmZk1NTU1NTVycnJ6enoG\nBgbOzs5CjvdgqqqqOjo6NjY2rq6ueJdsca6mZLm7u1tYWFy/fr2iouLDhw8GBgb9+/efOXMm\n5+OUpqYmye7KCQAAAIDPjyDm5nD6+vp1dXUVFRX6+vqS6hP4FsXFxRUWFnJu28ZXRUXFr7/+\namVltWvXrs/TMfBlVVZWzpkzJygoiG/+ooaGhqSUnyTY3IL5LIQQnU5vbm5WU1OT7K1pc3Mz\n3rlCU1NTsikLGxoalJSUulsx3DMUCgXv/6WqqirZZ1zNzc0yMjJqamoSrJNKpeJET7169RIp\n19MnUSgUGo0m6mDB4cOH09PTExMTJdgT8XVuXfelu4AUDnE/9QUAfE7ibjDR1NQ0YMAAiNpB\nU1PTTz/9/wjs0KFDS5cuzcvL4yqGZ1qbm5t/1s4BAAAAAHz7xA3c9fX18Zxa8CNLT09PT0+f\nMmUK+wjOxnjp0qX6+nr2wXv37iUnJxOJRK59RgEAAAAAwCeJO8d91qxZoaGh5eXlIs0Y/mwu\nXrx48+ZNiVd77ty57hKJ/JhaWlp+++03zr8BV1dXEon08uXL5cuXm5iYKCgoVFZWNjU1ycrK\nLl++HCftgqsDAAAAACA8cQP3jRs3RkRELF68OCUlBS/a+6rMnDkTr938pLq6ut69e8vJyQlT\n+AdchisYb0ZeWVnZffv23bt3Ly0trba2lkqlamtrDxkyxMPDg50SFK4OAAAAAIDwxA21+/Xr\nFxsbO2fOnClTphw8eHDcuHES6ZakaGlpCTn4OnjwYGl35kcjKyvr5OTk5OTUXQG4OgAAIDxY\nGAoAEDdwP3DgAELI3d09LCzM1tZWW1vb1NRUR0enux12YmJixGwRAAAAAACAH5C4gXtgYCDn\nP+vq6urq6sSsEwAAAAAAAMBF3MD961yTCgAAAAAAwHdG3MD97du3EukHAAAAAASIiOQ/B/Vz\nwnucAQC+lK8uDwwA4IfiNLWeQCD06dPnS3cEAAAA+NqJuwETAAAAAAAA4DOQ5Ij7+/fvMzMz\nnz592tDQ0NnZGRUVhRD6avdmAgAAAAAA4BsimcC9pqZm8+bNV65c4TyIA/exY8daWFj8+eef\nkIobAAAAAACAHpPAVJl3795ZWVlxRe2c7t27N3HiRBKJJH5bAAAAAAAA/JjEHXGn0+kzZsyo\nqalBCE2aNMnf39/Q0NDS0pJdYPPmzbt27WpoaPDw8Hj9+rWsLCyHBQD8f0kpP0mpZg/3TinV\nDAAAAHwR4o64X7t27eXLlwihtWvXpqenOzs76+vrcxbYuHFjfHy8rKzsmzdv/v77bzGbAwAA\nAAAA4MckbuB+8eJFhNDAgQOPHj1KIPBPMTtp0qRZs2YhhG7cuCFmcwAAAAAAAPyYxA3ci4uL\nEULOzs4KCgoCik2cOJFdGAAAAAAAACAqcQP36upqhNDAgQMFF1NXV0cIVVRUiNkcAAAAAAAA\nPyZxA/fevXsjhEpLSwUXe/v2LfovfAcAAAAAAACIStzAfeTIkQih+Ph4Go0moNi9e/cQQkOG\nDBGzOQAAAAAAAH5M4gbu//vf/xBCr1+/3rp1K4vF4lsmPDw8JycHITR37lwxmwMAAADAN4dK\npfr5+fXr109ZWTk1NfUztFheXr579+4xY8Zoa2srKysPHz7czc0tPj7+kyeeOnWKQCAQCARj\nY+NPFq6rqzt8+LCdnV2/fv0UFRWNjY0dHR0DAgJwmmwAJE7cwN3b23vEiBEIoT/++GPy5Mkp\nKSmtra34JSaTWVxcvGHDBl9fX4SQlZXVqlWrxGwOAAAAAN+c/fv3Hzt2rKqqSl9fX15eXtrN\nBQcHDxkyZM+ePY8fPyaTyR0dHQUFBTdu3Jg1a9aUKVM+fPgg4NzIyEj8Q3l5eW5uroCSISEh\npqamW7duzczMrKqq6uzsLC8vv3fv3qFDhwYMGLBt27buBjS5GBgYEATCqfn4otFoo0ePJhAI\nb968EaYt8K0TN3AnEom3b98eMGAAQigzM3PatGmmpqb4JVVV1UGDBv3xxx8sFktJSSk8PBx2\nXwIAAAB+QMnJyQihsLCw4uLiCRMmSLWt7du3+/v7f/z4ccaMGWlpaY2NjQ0NDfn5+Zs2bVJS\nUkpPT1+wYEF3IXVFRcWDBw8QQpqamgihqKio7lpZtWrVpk2bKBTKsGHDTp8+nZOT09zc/ObN\nm1u3bjk5OVGp1IMHD27YsEEi74hIJHb30tatW/Pz8yXSCvgmSCCSNjAwePHiRWho6O+//06h\nUJhMJj7e0dGBf3B1dQ0NDTUyMhK/LfD1oNFo4eHhDx8+pFAogYGBnNvlSgmZTE5NTc3Ly6uv\nr6dSqTo6Onp6eg4ODmPGjBF8YmJi4qlTpxBCWlpa586dE1y4ubk5LS3t8ePHNTU1ra2tmpqa\nurq6pqamLi4u+HMcAACAqJqbmxFC48aNk3ZDycnJv//+O0IoMDBw79697B1m+vTpM2rUqOnT\npzs5OSUmJp4/f37p0qW8p0dFRbFYLGNjYx8fn127dkVHR4eEhPDGzWFhYadPn0YIbd26de/e\nvexnCOrq6gMGDHBxcTl16tSaNWuOHTs2ceLET84Tvn//PoPB4D3OYDBcXV2Li4u3bdvG98S4\nuLg//vhDcOXgOyOZIfBevXpt3759yZIlN2/ezM3Nra2tbW9v19TUHDJkyMyZM6V9bw2+iOjo\n6Li4OISQnp7eZ3iWEhsbGxERwbkGuqKioqKiIicnZ9iwYQEBAaqqqt2dm5mZiX8gk8lFRUWD\nBg0S0EpkZCSVSmUfIZPJZDL52bNncXFxLi4uCxcu7G6jMU6LFy9uaGgQUGDMmDE7duzg+xKd\nTt+yZUtpaemZM2d0dHQ+2RYAAHwrhPn8FNPGjRsRQq6urvv27eN91d7e3tfX9+zZs5GRkXwD\ndzxPxsvLy93dfdeuXbW1tRkZGfb29pxlKBTK9u3bEULe3t4HDx7k241Vq1bdv38/Kirq9OnT\nnwzc+/fvz/f477//XlhYeOjQoZ9//pn31crKSl9fX3V19ZaWFsH1g++JJOMtbW3tlStXrly5\nUoJ1gq8WiURCCK1Zs2batGnSbuvy5cvXrl1DCI0ePdrV1RV/xpHJ5MzMzMTExIKCgpCQkJ07\nd/L9SiCTya9evUIIqaioUCiUrKys7gL3U6dOJSYmIoSMjIxmzpxpbGxsYGDQ2tr67t27O3fu\nkEika9eudXZ2Llu2TPx3JCPT7Sy1S5cufTK/KgAAfCs2bNjAHhW2sLBACMXHx8+cOVMabWVn\nZxcWFiKEdu7c2V2ZxYsXFxQU0Gi09vZ2ZWVlzpcKCgoKCgoQQl5eXkOGDDE3N3/9+nVUVBRX\n4H7lypUPHz6oq6sHBwcL6ExQUNC0adMIBAKLxerBHUt+fv6uXbscHR03b97M+yqDwZg/f35j\nY+M///zj6ekpauXg2wWTzkEPtbe3I4TMzc2l3RCOmBFCnp6e8+fPZ3/8qaqqmpiYWFlZ7d69\nOz8//+7du1OnTuU9PSsri8ViaWtr29vbR0ZGZmdnL1myhDduTkpKwlG7m5ubt7c3+xmCkpKS\njo6OtbV1YmLi6dOn4+LiLCwsxo4dK7jPBw8eZM8Z48RgMIKCgqqrq+fNm8f3xEePHt2+fVtw\n5QAA8A2xtLR0d3dPTExsb293dHRUV1eX3rNEnHvaysoKp83gy8bGBme643XlyhWE0NChQ/EN\nxty5c4OCgq5fv37y5EnOBbUpKSkIocmTJ2trawvoTP/+/bsbSv8kGo3m7e0tLy9/6dIlvkH/\n7t2779+/v2rVKjc3t541Ab5R4i5OBT+4z/Dc8/z58wihsWPHent78zZnaWnp4OCAOObDcMHH\nJ06caGtrixBqamrCAyqcqFTq33//jRCys7P75Zdf+M78mT59Op70heN7wbS0tHT4ycnJef/+\n/cKFC/luNtzQ0HDs2DElJaVP1g8AAN8KHx+fmJgYXV1dhFBoaGhMTIyVlZWU2sLLNNlJMkTC\nYrHwUtT58+fjIzgmbm5uTkpK4iyJB/WHDh0qZm8FOHHiRFFRkb+/v56eHu+raWlpQUFBw4YN\nCwkJkV4fwNdJ3BH3Fy9e4BtThND169fxyGh3vL29nZ2dxWwRfHHnzp1jjwqvXbsWIbRz587R\no0dLo63CwsL3798jhAQ8CnRwcCgvL6fT6VQqVVFRkfMlPA8eITRx4kQDA4N+/fpVVlZmZWVx\nLaXNyMhoa2tTUlJavHixgM4sWrRo1KhRCKGePfcsLS29cuXKiBEjXF1deV9lMpnBwcFtbW1b\ntmw5fPgw3xpWrFhRU1NjY2ODp1dy1hwXF1dUVPThwwdDQ8PZs2dPmDDBy8uLwWBER0eL2k8A\nAPhG4cVFONOdqLKzs9+9e4f+26AGITRq1ChjY+O3b99evXrVxcWFXbKxsZFvK1VVVf369eOt\n+dq1ayKNizc3N+/fv19bW9vf35/3VTKZvGDBAkVFxX/++UdRUZHvqlbwHeth4E6n00+fPn3p\n0qVnz551dXXhgy9evLh69aqAs0aPHg2B+3fA2NjY1tY2Pz+fSqWOGDFCWVlZQ0NDSm09e/YM\nIWRqairgg3jQoEFHjhzh+1JGRgZCyNDQECc1Gjt2bExMzMOHD1etWsU5rP7kyROE0PDhwwW/\nES0trSlTpvTkbSBEp9ODg4Pl5OT8/Pz4Bv1RUVEvX76cPn26qFkXUlNTT548if8b9urVq7i4\n+MiRI4WFhXzn6gAAwHess7MTIdSz55Z4WerPP//M+XUzd+7c4ODg27dvc06Ix/WTyWQJ9Jif\nAwcOfPjw4dSpUyoqKlwvsVisRYsW1dTUnDt3bvDgwVLqAPia9SRwb29vd3d3x0+O+E4qMDMz\nq62tbWtrw//U0NCYPXu2nZ3djBkzxOkr+ErY29vb29vj0d+lS5caGhpKry28TBM/YxUVi8XK\nyspCCE2aNAkfGTduXExMTHt7O4lEsra2ZpfEg/pSfSMJCQlVVVVeXl69e/fmffX58+fR0dFG\nRkZLliwRUImjo2NbWxtnP6uqqv766y8mk+nm5ubq6qqmplZTU3PixImEhASEENfzB2l79eoV\n+389G053wGAw2Hf4n43EG2UnfqbT6RK/L2IymZLtLbuH0vg9sFgsydbJHjWU0u9B1DqZTKaQ\nW+eArwr+gO3BVkRdXV0xMTEIIS8vL87jbm5uwcHB7e3tcXFx7JF4LS2tiooK3iwC+vr6dDqd\n84izszPnNBsPD4/6+nr2P1VUVHByNk6NjY3Hjx/v06cP3rySy+HDh5OTkz08PAR/WYDvmMiB\ne2trq729/ePHj/E/x48fz1umuLgYIfTo0aNNmzZlZ2dTqdQlS5ZAUkjQA3gjXsELgLpTWFiI\nPyLZf3smJiba2tp1dXVZWVmcgTsON3nXSzU2NvL96AwICBBpXLy9vT06OlpDQ4PvJJmWlpbg\n4GB5efktW7bIy8sLiAjd3d25jkRERDCZTCcnp19++QUf0dXV3bNnz6pVq2pra4XvoUQcO3aM\n/cnAhn+rHR0dnz9hGV4/LQ289yfio1KpnHlIJYi9pYYEMRgMKV3Qzs5OPGgqWaL2lkajQeD+\nLRo0aFBqauonE3MtXLiwubl58eLF7M/k5ORkPAHGz8/Pz8+P95SoqCh24D569Oi8vDze5VKI\nZ6ek169fc/4zJyensrKS/U++z3gvXLhApVJXrFihoKDA9dLz588DAwONjY3PnDkj+A2C75jI\ni1NDQkLwd7OxsTGJREpPT++upLW1dXp6up+fH5VKnT17dllZmVg9BT8kPE7G+/klDDzcPnDg\nQM6IHCeEefToEWeQhOvH+4NIQ3R0dFtbm5eXF+8QOIvFCg0NbWpqWr58uYGBgUjVslisf//9\nFyHElSSYSCTCnDQAwA8IP1/Ny8vDq5v4KikpiYiIiI+P5xwSwvNk5OXl+/LAM2SSkpKamppw\nYUdHR4TQo0ePukuKgDU0NJSXl3Meef/+PYsDu0I2JpOJ93XiO6BeUlJCp9Pfvn2roaFB+A97\n4oOpqSmBQJg1a5aALoHvgGgj7k1NTaGhoQghQ0PDzMzMT8YZsrKyoaGhubm5ubm5GzZsuHXr\nVs97Cn5IeFulHgweMxiM7OxsxDFPBhs3blxsbCyVSs3Ly2OPxKurq5PJ5JqaGq5K+vTpExsb\ny3lk7969OIE9dujQIfxMAFNUVOTdVqmtrS0hIUFVVRVnv+Fy48YNEok0fvx4/E0gEjKZTKfT\nFRUVeZ8VGBsbi1qb+MzNzXnzbCopKdXW1hKJRDk5Od5TpDp/hkgkCsiX3wN0Oh2PwsrKyko2\nn1JXV5eMjIyAXc17gMFg4Kc30vg9oG7mSfYYk8nEs2Wk9Hvg++cngIyMzGdImQUkzsXFpW/f\nvg0NDTt37gwPD+dbBqeOUVVVZT93bW9vx/HJjh07AgMDuco/evTo559/ptFoN27cwPH0rFmz\njIyMKioqAgICusssiRAKCAgQtf/JycllZWVjxowZNmwY76tycnK8g/QsFgs/UFJVVSUSiVyZ\n6cH3R7RP3rCwMBym7NmzR/jRwaNHj44fP/727du5ubk2NjYi9xH8wPT19Z89e8YbUnMJCQlp\nb293cHBgZ1gnkUh4PsPZs2fPnj3Le0pmZiY7cDc1NS0pKeE7QsMV8XA+5UQIFRUVce6QyvcT\n8+7duzQabdq0abyhQ3l5eUREhLa2Nk7OI6q6ujqEkKamJu9LfA9KG9/ny5WVlRkZGUpKSurq\n6ryvCt5fVkzKysqcqZfF19zcjGNW/AUpwZobGhoUFRUlmwmUQqHgx0pKSko9e2bVnebmZhkZ\nGTU1NQnWSaVSKRQKQkhBQUGykQeFQqHRaHz//ASQl5eHwP1bpKioGBgY6OfnFxERMXz48I0b\nN3Jdx5cvX/7+++8IIV9fX/bNZ2xsbEdHB4FAWLBgAW+d1tbWZmZmJSUlUVFROHCXlZU9cuSI\nh4dHbm6uq6vrxYsXeePpsLAwnMtYJHjgv7tRcxcXF95BegaDgd/IkydPTExMRG0RfHNEC9yf\nP3+OENLR0fH29hb+LFtbWxMTkzdv3ly/fh0CdyASCwuLO3fulJSUkMlkLS0tvmWqq6tx9hjO\nXY3wE0xZWVneYIhGo1GpVBKJRKFQ8Jr9ESNGJCYmFhcXc6Y35dXa2sqVRuDChQuC+89isfDK\nJL6bQ1VXVzMYjLq6OvbUSU7Lly9HCI0ZM4Z3FB/D0TnfGT6wAzYA4Me0bt26jIyM2NhYf3//\ne/fuLV68eNSoUdra2m/fvk1ISDhw4ACVSh08ePDBgwfZp+B9lyZNmtTdfkkLFizYtWtXenp6\nbW0tfsI5b968DRs2hIaGxsbG4qnnU6ZM0dHRqaqqev78+fHjx1NTUwcMGGBmZpacnCxkz5lM\nJi4MawKBAKIF7njBx4gRI0R97Dho0KA3b96kpqaKdBYA1tbWampqra2tkZGRfAd00X9z2Xv1\n6mVmZoaPUKlUPPnb09OTNwF8cXGxv78/nU7PycnBE1Ssra21tLTIZHJ4eHh3mSURQt09eBWA\nRCLV1taamZnhfJRcZGVleQcXWSwWXk3Yq1cvGRkZAZlhdHV1ZWRkPn78yHtXg7MRAwDAj4ZA\nIMTExKxfv/706dNJSUlceychhMaNG3fhwoVevXrhf9bX19+9exchtGjRou7q9Pb23rVrF5PJ\njI6OXrduHT4YEhKira29e/fusrIy3j1AJk+efOXKlaSkJOED98ePH9fX18vJyf38889CngJ+\nQKJNfCwpKUEIdfcsZsSIET4+Pj4+PrwvDRo0CCFUXV0tcgfBj01eXh5H3unp6bGxsbx5Ht69\ne4e3/XJwcGDPXsjNze3s7CQQCJMnT+atc+DAgXgvOhzxI4SIRCLOHlNUVBQUFMQ3G0lSUhL+\ncBcJHvjnzGDDydraOooHflSKEDp27FhUVNTmzZu7q1xWVhY/H7h58ybncRaLFR8fL2pXAQDg\n+yArK3vixAkSieTn5zd06FANDQ1lZWVLS0t3d/fIyMgHDx7gmASLiYmh0+lKSkq8abvYTExM\n8DxMPD+ebevWra9fv/7tt99GjhypqampoKBgaGg4b968mzdvpqWl6erqTp061c/PT8idXPG2\n3KNHj2bfVADAS7QRdzy/s7vh9tmzZ8+ePVvA6bxzswD4JGdn54KCgtzc3AsXLjx9+tTR0dHE\nxERDQ6Ouri4vLy86OppGoxkYGLDzIaL/wmULC4vuZtfY2dlFRkY+f/68qakJTzixtbWdPXv2\nrVu3cnNzy8vLPTw8hg8frqmp2djYWF5enpCQ8OzZMx0dHT09Pc7FqYKxWCy8tdOQIUPE/S0g\nhBC6ceMGhUIxMDBg35D4+Phs2rTpzp07ampqLi4uysrKDQ0Np0+frq6uhtV1AICvBx74+5ws\nLS1xOg3BVq9evXr16k8We/jwId/jRkZG+/fv379/P99X9fX1hekDtmvXrl27dglZmI1IJELq\n0h+KaIG7jo5OaWlpVVWVqM3gxYV9+/YV9UQACATC1q1bz549m5iYSCKReONmc3Pz9evXs5ch\ntrS0PH36FCHEd7gdw4E7i8XKzs5mLwNasmSJhoZGVFRUbW3tn3/+yXXKsGHD/P398/PzhQ/c\nS0pKWlpaiEQi5+iOOO7cuUMmk21sbNhvzdTU1NfX99KlS1FRUVevXlVRUWlrayMSiRs3bvzr\nr78kuyQRAAAAAF+WaIH7gAEDSktLMzIymEymSPnFcGI+fX190XoHAEIIISKRuHLlymnTpqWm\npj59+rSxsZHBYOjq6urp6Y0dO3bixImchR88eMBgMBQUFGxtbburUEdHx9zc/PXr11lZWZzr\n993c3CZMmJCSkvL48WMymdzZ2amhoTFw4MBJkybhddUjR450cXERcifX/Px8hJCZmZlkc5tw\nmTNnjomJSXx8fFFREZVKtbKy8vb27tevH5VK5btLKwAAAAC+UaIF7s7OzikpKfX19bdv354z\nZ46QZ92/fx8n0ZsyZYrIHQRfq7CwsM/corGx8dKlSz9ZbMaMGTNmzPhkscOHD/M9rqWltWDB\nAr5JwRBCffr0EaYPmJeXF9fu2cKQkZG5ffs235fOnTsXFxdXWFjIdXzYsGFcSX9xakshbzAA\nAOBz2rx589GjRyVebXl5Od80AAB8T0QL3OfOnevv70+j0TZv3mxvb483xxGMwWBs27YN/yx8\nrA8A4Kupqemnn35i//PQoUMlJSUrVqwYM2YMZzGcx8Dc3Pxz9w8AAD5lzZo1QsYD5eXlurq6\nQs76492KDoDvj2iBu76+/q+//hocHFxaWurq6nrt2jXeTQc4MRiMdevWPXjwACE0depUSOIO\ngDjS09PT09M5Vy8NGDDgwYMHly5d6t+/Pzugv3fvXnJyMpFI5No1FgAAvgb9+/fvLmM6FwEz\nHgH4MYm8Z/XOnTvxbNrU1FQzM7P9+/cvW7aM73z37OzstWvXPnv2DCGkoqISEhIigf6Cr9LF\nixe5MhJKxLlz57pLC/Njamlp+e233zi/8FxdXUkk0suXL5cvX25iYqKgoFBZWdnU1CQrK7t8\n+XI8/gRXBwAAAPg+iBy4q6mppaSk2NraVlZWNjQ0rFy58s8//5wzZ461tbW2tra8vHxTU1Nh\nYWFKSgo7k7SsrGxUVNTQoUMl3XnwtZg5c6aQj1Pq6up69zxYNdoAACAASURBVO4t5AZeOFEj\nYON9uCwrK7tv37579+6lpaXV1tZSqVRtbe0hQ4Z4eHgYGxvjMnB1AAAAgO+DyIE7QsjQ0DA/\nP9/X1/fOnTsIocLCQt7Vcmw6OjrR0dGwf+/3TUtLS8jB18GDB0u7Mz8aWVlZJycnJyen7grA\n1QEAAAC+D6LtnMqmpaWVkJAQFxcnYBJt7969d+/e/erVK4jaAQAAAAAAEFNPRtzZnJ2dnZ2d\n6+rqMjMzX7x40dDQQKFQ+vTpo6urO2HCBGtra/YW9AAAAAAAAABxiBW4Y9ra2h4eHh4eHuJX\nBQAAAAAAAOCrh1NlAAAAAAAAAJ+TBEbcAQAAACBtC+azvnQXAABfGATuAIAvyWlqPYFA6NOn\njwTrpNPpzc3NEqwQAAAA+BqIG7ibmZnhH27fvt1dLrmSkpIZM2b8X3uysvr6+jY2NuvXr+fc\nuR0AAAAAAAAggLiBe2lpKf6BRqN1V4ZGo7GLbd++vaWlJTY2Njo6Oj8/X1VVVcwOAAAAAAAA\n8CP43FNlDhw4gBDq6uoaMmTInTt3PD09P3MHAAAAgG8RIWPnl+4CYtnt/dJdAOCHJm7gHhsb\ni39g76/Oq1+/flevXuU8Iicnd/HiRcnOagUAfItUj+xFCHVKvFqEWD2qVuHQn5LuCwAAACAZ\n4gbus2fP/mQZdXV13pH18ePHi9k0AAAAAAAAPw7I4w4AAAAAAMA3QNzA/fTp001NTRLpCgAA\nAAAAAKA74gbuq1at0tXV9fDwSEhIoNPpEukTAAAAAAAAgIsEpsp0dnbGxMQ4Ozvr6+tv3Ljx\n2bNn4tcJAAAAAAAA4CRu4B4dHe3i4iInJ4cQIpPJoaGhI0aMsLS0DAkJqaurk0QPAQAAAAAA\nAGIH7vPmzbt161Ztbe2pU6fGjx9PIBAQQs+fP9+0aZO+vv7MmTOjo6OpVKokugoAAAAAAMCP\nSzJZZXr37r1y5cr79++XlZUdOHBg8ODBCCEGg4G3WNLV1V25cuXDhw8l0hYAAAAAAAA/IAmn\ng+zfv//27dsLCwtJJNLGjRv19PQQQs3NzWFhYba2tmZmZvv37y8vL5dsowAAAAAAAHz3pJXH\nfeTIkcHBwe/fvw8LC1NRUcEHS0tLd+zYMWDAgMmTJ8fExLBYLCm1DgAAAAAAwHdG3J1Tu1Na\nWnr9+vWYmJj8/Hyul1gsVkZGRkZGhpWV1e3bt/GovATRaLTw8PCHDx9SKJTAwEBLS0vJ1s+L\nTCanpqbm5eXV19dTqVQdHR09PT0HB4cxY8YIPjExMfHUqVMIIS0trXPnzgku3NzcnJaW9vjx\n45qamtbWVk1NTV1dXVNTUxcXF01NTYm9mR+Pl5dXe3v75s2bJ0yYwLfAlStX/vnnnyFDhhw8\neBAfaWtr8/b25i2ppKSkr69vbm7u4eGhrq7eXYsiXXcAAAAAAEzCgXtJScm1a9diYmKePHnC\neXz48OGenp6enp4tLS1nz56NjIxsbW3Nz893cHB4/vy5rKwkuxEdHR0XF4cQ0tPTk2zNfMXG\nxkZERNBoNPaRioqKioqKnJycYcOGBQQEqKqqdnduZmYm/oFMJhcVFQ0aNEhAK5GRkZzLfMlk\nMplMfvbsWVxcnIuLy8KFC/HKYJGQyeRly5YtWrTIzc2Nb4GysrKYmJh3796RyeTevXvr6+u7\nuLiMGDGCt2RLS0t0dHRhYWFVVZWampqhoeGsWbNGjhwpape+aR0dHSUlJSUlJampqXv27Onu\nggp/3dnFBF8mhFBubm5iYmJZWRmNRjMwMJgwYYKzszORSOQqJuRlunv37l9//cW3IXt7+/Xr\n1wvuMAAAcKFSqQEBAdeuXWtqarp9+7a9vb20WywvL7906VJCQsK7d+8oFIqJiYmZmZmvr6+z\nszNXyYSEBGdnZwcHh7t370q7V4DXP//887///W/OnDk3b97k/SfgIpm4tqioCMfrXEnchwwZ\nguN1ztDk1KlT+/fvX7Zs2c2bN1+9ehUfHz9nzhyJdAMjkUgIoTVr1kybNk2C1fJ1+fLla9eu\nIYRGjx7t6urav39/hBCZTM7MzExMTCwoKAgJCdm5cyffkJpMJr969QohpKKiQqFQsrKyugvg\nTp06lZiYiBAyMjKaOXOmsbGxgYFBa2vru3fv7ty5QyKRrl271tnZuWzZMlH7n5KSImDCUnx8\n/Llz55hMJu5kbW1tTU3N48ePZ8yYsXLlSs6Sr1+/3r17d0dHB0JIXl6+vr6eTCY/fvx42rRp\na9asEbVX35CjR4+amprin+l0ellZWXZ2dkJCQkdHR3Bw8PHjx+Xl5blOEf66swm+TAihkydP\nJiUlIYQIBAKRSCwuLi4uLs7Jydm/fz/nvavwl6mqqkrYXwEAAAhh//79x44dQwiZmZnxfjBK\nXHBw8I4dOz5+/Mg+UlBQUFBQcOPGjcmTJ1+7dq13794SbG7z5s1Hjx5FCIk0B1hTU7O5ubm2\ntlZbW1uCnQHfN3ED93379sXExBQUFHAeHDhwoKenp4eHh4WFBd+z+vTpc/jwYXwvVVRUJGYf\nuLS3tyOEzM3NJVstLxwxI4Q8PT3nz5/Pjs5VVVVNTEysrKx2796dn59/9+7dqVOn8p6elZXF\nYrG0tbXt7e0jIyOzs7OXLFkiI8O96iApKQlH7W5ubt7e3uw4TElJSUdHx9raOjEx8fTp03Fx\ncRYWFmPHjhWy811dXWlpabj/fNXW1l66dInJZLq7u7u7uyspKXV2dsbFxV2+fPnOnTujRo2y\ntrbGJTs6Og4fPtzR0TF69OglS5bo6+t3dnampaWdOXMmOTl58ODBU6ZMEbJX3xwZGRn2JZOX\nlzc3Nzc3NzczMwsODq6trS0rK+P9OxTyumOfvEwIofj4+KSkJDk5uZUrV9rY2CgrKz979iw0\nNLSwsDAyMnLRokW4mEiXCQfuS5cuZV9ltl69eonyGwIAAIQQSk5ORgiFhYUtX75c2m1t3779\n999/RwjNmDHD39/f0tKSxWJVVFRERkaeOnUqPT19wYIFCQkJPXhMDcAXJ+7i1J07d7Kj9gED\nBgQEBDx58qSoqGjv3r3dRe1Y37598Q9SirA/w3/I8+fPI4TGjh3r7e3N25ylpaWDgwPimBfB\nBR+fOHGira0tQqipqYnr/gchRKVS//77b4SQnZ3dL7/8wnfmz/Tp0/HkbBzff9KDBw+2bNmy\nYMGCEydO4NF0vpKSkmg02qhRoxYtWqSkpIQQUlBQcHd3nzhxIkKIM7lnYWFhQ0ODurp6QECA\nvr4+Ljl9+nT8IOX+/fvC9Op7MmnSJPwbe/fuHe+rwlx3JPRlolKpUVFRCCF/f39HR0dVVVUZ\nGZmRI0euXbsWIXTv3j328I9Il6m6uhohNGzYMB0eAubuAwBAd5qbmxFC48aNk3ZDycnJOGoP\nDAyMj4+fPHly7969+/TpM2rUqKNHj96+fVtWVjYxMRF/g0vKpEmT/P39/f39JVgnAHxJYKqM\noaGhh4eHp6fn6NGjhT+LSCRGREQghCQ4HHvu3Lnbt2/jn3HgsnPnTpF6JbzCwsL3798jhDw9\nPbsr4+DgUF5eTqfTqVSqoqIi50t4HjxCaOLEiQYGBv369ausrMzKyuJaSpuRkdHW1qakpLR4\n8WIBnVm0aNGoUaMQQiwW65N3LLW1tXV1dYqKioqKihQKpauri28xHHTyrq8dOnRoVlYWZ0iK\nf+7Xrx/X0088h4QreF2xYkVNTY2Njc327ds5j5eWlsbFxRUVFX348MHQ0HD27NkTJkzw8vJi\nMBjR0dGC39FXSF5evqOjQ01Njeu4kNcdCX2Z8vLy2tra9PT0uB62jBkzZt++fQihrq4ufF2E\nv0xMJrOmpkZGRqZfv349evcAAMDfZxhT27hxI0LI1dUVfwZysbe39/X1xWvtli5dKk5DdDo9\nMzOTyWQ6Ojo6OzvzTp0HQBrEHXHPyckpLy8/cuSIqPGxqqqqt7e3t7e3gLWbojI2Nra1tcUh\n8ogRI2xtbTU0NCRVORc8m9/U1HTAgAHdlRk0aNCRI0cOHTrEFbUjhDIyMhBChoaGRkZGCCEc\ndT18+JBOp3MWw2t8hw8fLviNaGlpTZkyZcqUKcJ8Jrq5uYX/h3ciBJudnd2KFSusrKy4jre0\ntCCEOENSPDnv3bt3ePI0G57JLczUvdTU1K1bt6anp1dXVxMIhOLi4iNHjoSFhQkYaf6alZaW\nNjc3Kykp8a7iFfK6I6Ev08uXLxFCePCeE4FAsLS0tLS0ZIfpwl8mMplMp9P19PSKiooOHjy4\nZs2aDRs2HDt2LC8vT+jfAQAA/J8NGzYQCITS0lKEkIWFBYFASEhIkFJb2dnZhYWFCKGdO3d2\nV2bx4sU2NjY0Gg1PrOXEZDJDQkIsLCx69eqlp6c3c+bMf//9l7MAnmCzYcOGysrKESNGODg4\nrFu3jn3c0dGRXbKjo+Po0aM///xz3759NTQ0rKys9u3b19bWhl+dN28egUDATyF0dHQIBAJn\nCr6Ojo5du3bZ29v36dNHR0fH3t4+Ojqa6wsRt/jrr78ihMLCwkaOHKmsrKynp+fs7Mz7WS24\nM9JrV8g6hZGenu7q6mpqatqrVy9tbe0xY8YcOXIEByRc3rx5s3z58lGjRqmoqAwePNjHx6es\nrKzHvRK+XUm9008Sd8TdxsZGIv2QCHt7e3t7ezymu3TpUkNDQ+m1hT+DdHV1e3Aui8XKyspC\nCE2aNAkfGTduXExMTHt7O4lE4ozS8KC+VN9Id/CUGE5MJvPJkye3bt1CCHEu/LWxsRk8ePCr\nV6/27dvn4+PTv3//tra2tLS0uLg4OTk5rrSJjo6ObW1tnO+oqqrqr7/+YjKZbm5urq6uampq\nNTU1J06cwJ/svPc8Xy0Wi/Xhw4cXL16Eh4cTicT169dzdV746y68hoYGhJCWlhaZTE5KSsKP\nLIyMjEaOHDl16lTOGznhLxOe4F5TU8P5VOTNmzepqakTJkzw8/OTk5Pj25kzZ87wfj7iwh8/\nfuT9kvg6Cegn+yO4vb1d4gOHNBqNwWBIsEL23eDHjx85016Jj8lkMplMyV5Q9u+WRqNJ9quO\nTqezWCxRe9vV1QU7jUiKpaWlu7t7YmJie3u7o6Ojurq6jo6OlNq6d+8eQsjKyopv9jPMxsYm\nJyeH9zidTnd3d79586a6urqhoWFZWdmdO3fu3r2bnZ3N9fnc2dk5c+bMly9fysjI8A0D2tra\nxo0b9+LFC4SQqqoqi8UikUgkEunGjRu5ubkKCgo4drp161ZXV5ezs7OioiI7rXNBQcG8efPw\n8j9VVdWWlpa0tLS0tDRXV9crV65wrTJisVjLli2Lj493dXWdN29eWlpaQkJCSkoKZ58/2Rkp\ntStqnQKcOHECT6OQkZHp3bt3c3Mzzqxw7dq1jIwMznpu3Ljh6+vb2tqKENLQ0Hj9+vXr16+j\noqJSU1PHjx8vaq+Eb1dS71QYUs+W+L3CfxY9WwleWFhYX1+PEGInDjcxMdHW1q6rq8vKyuL8\no8dfNryfcY2Njb6+vrw1BwQESGMGYVRUVHp6enNzM5VKVVdXX7duHefEDCKRuGfPnpMnT2Zk\nZGzevJl9vG/fvlu3buVKmeLu7s5VeUREBJPJdHJy+uWXX/ARXV3dPXv2rFq1qra2VuLvRbLw\nM1leq1ev5l0oLPx1F15jYyNCqL6+3t/fHw/eIISqqqoePnyYmpoaGBjIfjYi/GXCgTuDwZg6\ndaqDg4ORkVF9fX1qampsbOz9+/f19PT45rBHCJFIpMePH3MdxH+9dDq9s7OzB2/w8xOmn5KN\ngzE6nc774OVrrllKF5TBYEj2BgYTtbcMBgMCd0nx8fHx8fExMzMrLS0NDQ0dOnSo9NrC49bs\nZF8iyczMVFFRiYmJcXNzIxAIb9++dXZ2LiwsPHjw4I0bNzhLXr58WV5e/tKlS56ennxHl/bu\n3fvixYthw4ZduXJl2LBhCKGcnBwXF5enT5+ePXt27dq1mzZtQv9llTl37hw7lqDT6f/73/+K\nioqmT58eEhJibm7+8ePHmzdv+vn53bx587fffgsJCeFsKDY2tk+fPk+ePMGftNu3b1+xYsWZ\nM2eCgoJiY2OF7IyU2hW1zu40Njb6+fkhhPbs2bNp0yZlZWUWi5WSkjJ//vxHjx7Fx8fPmzcP\nl6ysrFywYMHHjx/Xr1+/fft2LS2turq6X3/9NSYmxsvLq7S0VEFBQfheCd+upN6pkEQL3HuQ\ncJDL2bNnxazhK4HnHLNvVUWCh10HDhzIGZGPHTs2Njb20aNHnBPicf3sgOxLaWxsrK2tJRKJ\nBAKBxWK1tLRwTaYvLCzEd/NEIlFLS6utrY1CoXz48CErK8vExERANn0Wi4UfRM6dO5fzOJFI\ndHZ2/nY3Jzp58mRhYeG6des437vw1114OHCPjo42MDD49ddfBw0axGAwnjx5cvHixdevX589\nexZ/PWBCXiY1NTU7O7uBAweyp2waGhr6+vqqq6tfunTp+vXr06dPl2wmNQAAkAj8EFLAFFYB\nWCxWSEgIe3TJ2Nj4wIEDrq6ueEYip/b29gsXLnh4eHRXFV7uHxgYiANlhNDYsWO3bt0aHh5e\nV1cnoA9hYWGFhYVjxozBi2gRQr169Zo/f76RkdHEiRP//PPPgIAALS0tdvmqqqq///6b8ztl\nyZIlZ86c4cx5IExnpNGuqHV2B88mHTNmDHv6E4FAmDZt2qJFi/7444/nz5+zA+jAwMCPHz96\ne3v/8ccf+Ii2tnZERMTjx4/fvn3777//Tpw4UfheCd+upN6pkEQL3MUPpL6bwB1Pze/BkDCD\nwcjOzkYc8yWwcePGxcbGUqnUvLw89oisuro6mUyuqanhqqRPnz7sm1ps7969OIE9dujQIfxM\nAFNUVNyxY4eoXWVbu3Ytvimvq6s7ceJEeHg4mUxetWoVfpVEIuE1QEuXLp0xYwb+qy0uLj5+\n/HhcXFxDQ8O2bdu6qxlPp1ZUVOR9qmBsbNzjDn82ISEhnEM7+K4mLy/v/PnzGRkZysrKK1as\nwC+JdN2Fh/NIampqHjx4kL1cZMqUKVpaWtu3b8/MzJw3bx6emCT8ZbKzs7Ozs+Nta/bs2dev\nX29ra3vz5g3fwH39+vW8ExJaWlq2bdumpKTENx0N35mCX5aAtDkUCgUPBuPsPRJstKWlRVFR\nsWcDAd1hz5BRUlLqbnZTz1AoFBkZGZw6SVJoNBpOua2goCDZCXIfP36k0+miLqaSl5eHXIHf\nIvxopWd/nCoqKj4+PpxH8KNI3idsffr04X16zAlPjbhz546rqyv7f58waWfw9pEbNmzgGvCy\ntbUdNWrU48eP09PTOVNi9O3bd/LkyZwl8eA9Z5+F6Yw02hW1zu7MmDGDSqXyfuTiLSk5Hyfi\nqbycw1UIIXl5+aNHjz558gS/d+F7JXy7knqnQhItcGfncAT6+vrPnj3jDam5hISEtLe3Ozg4\nsCdOkEgkHNycPXuW721MZmYmO4AzNTUtKSnBeUi4cP0xVVZWcv6zqKgIDzxgysrKn35LQtDW\n1t60adOiRYuSkpLc3d1/+uknhNCFCxcYDIaXl5eLiwu75MCBA3fs2LFq1aqcnJyXL19292wU\n3+6z5/Zx4nvwK0cgEDQ0NBwdHWk0WlhYWEpKio+PD47GRLruwuvdu/eHDx+mTJnCFZdYWFjg\nlDVv3rzBgbs4lwkjEon9+vV79erVu3fveNMNIYQGDx7MexD/ZRKJRMnGjtIjoJ/sSE5WVpZ3\nV1oxycjISPZXxJ4cIvFfPoFAIBAIkq2TPT1GSr8HUeuUkZGBwP1bhMcU3rx504NzzczMuAKv\n7u7PDQ0NBd+6r1mzJiMjIzw8/N69e7NmzZowYYKtrS3OSSDY69evEULh4eHsFHls+OuSax2R\niYnJJ+sUpjPSaFfUOrtDJBLx521ra+vr16/fvn375s2bR48e4XCZrb6+vrm5WUZGhv1ggW3u\n3Lnsp/rC90rIdiX4ToUkWuCOZ+gChJCFhcWdO3dKSkrIZHJ3T0Cqq6txFhH28xT0Xxrv/8fe\nvQdClf+P43+Ncb9T7vdcVtHqIpGkO92U2C0pW1uptnZr28TupvtNVr23rXQRZatpUZFyqVAi\nyqV0USjlEmOQO2PMmO8fr1/nN5+ZMQaDLs/HX5x5zTmvmcPM87zO8/V8iYuL8w4JMBgMOp2e\nm5vb3NwsLy+PEBo1alR8fHxhYeHz588F1MVvbGyk0WicW0JDQ3v5whBCCNXV1V2+fFlMTIwY\nMCYoKSmpqqrW1taWlZWpqanR6XRcSZA3pVtNTc3ExCQ/P//Vq1ddRYQ4OuebCzRgY7FycnIt\nLS285QUIzc3NqIcXP+PGjTt16lRHR0dVVRWOm3t03oWHv6X4zrVQV1cvLy/H/7N9PE0EXKNG\ntEOtAAAgKt98801SUhKuHiHAsmXL6uvrf/zxR1dXV2Kj8BmA3bZ0d3e/devWjh07MjMzT548\nefLkSYSQhYXF6tWr169f31X6KJvNxiMdeLEqvri+qoS5ldRtZ/rjuL3YpwBxcXH79u3LzMwk\nZq7r6+sPHz6cM4vp7du3CCE1NTXB2bk96pUwxxXtKxUGTE7tJRsbG0VFxcbGxkuXLuHpC7xw\nTrOMjIypqSneQqfTcUr3okWLeO+bFBYWbtmyhclkZmRk4KpSNjY2uGDI+fPnAwMDu+rM+fPn\nRfKiCAoKCrdv32Yyma6urlyXJSwWC4fUuEKlhISEpKQkg8HgO/sNbxQQ52lpaYmJibW1tfFe\n//Bdvag/DBs2jEajCfigLyoqQj2c7URMa8ORbk/Pu/B0dXUfPXqE10vigi/09fT0UE9OE4PB\n2Lx5c0dHx86dO3mrJeCPp0EpcwQAAN1ydHQ8ceJEVlZWSUlJVyPcRUVFeBkZrjRO4e+xCNNy\nxowZM2bMoFKpt2/fTk9PT0lJefHixaZNm9LT07tanIREImlra5eVlT179kzwEpY9Jbgz/XFc\nEe6TQqF4enqKiYm5uro6OTlZWVmZmJioqqru2rWLM4DGCbd1dXWdnZ1d3Q/pUa+EPG7/nbWu\niDJN86siKSmJI7CUlJTo6Gje+gOlpaV4pfrp06cTd9UzMzPb29tJJBJXchhmZmamra2NPkb8\nCCEymYyrxxQUFOzfv5/vRVtCQsLt27dF9sIQQgiJi4sbGhoihJKSkrgeun//PpPJlJCQwNEb\nmUzGLYk+E8rKyvDtIeK6he+B8B/6tWvXOLez2ewbN270+XUIZcSIEQihu3fv8o3dk5OTCwsL\nEUI9+ofEK8tKS0vjsfCennfh4ZqPKSkpXMnleXl579+/FxMTw2++8KdJUlJSU1OzsrKSaxIF\nQuj+/fu1tbUKCgq9m/gFAAD9zcXFZejQoe3t7QLquOPVphUUFHpXy6tbHR0db968wek6mpqa\ny5YtO3nyZEFBQVhYGEIoMjJSQJKtmZkZQujp06e8D5WWlr558wYnWIu8MyI/rgj3uXfvXjxv\nOCoqavXq1TY2NviOB+dEPoSQnp6ejIwMg8Hg/Sq/efPmunXr/vvvvx71SsjjivCVCgkC996b\nO3eura0tm80ODQ3dtWtXeno6lUql0+klJSVRUVFbtmxhMBh6enpElUP0MV/C0tKyq+waPCnw\n6dOndXV1eIu9vf38+fMRQpmZmb/++uudO3doNFpHRweVSs3MzPT39z9x4oSmpiZeOVWE3Nzc\nEEJRUVExMTF4yZ62trbExER8l23x4sXE3ahFixaRSKSYmJjw8HCc9MJgMB49erR7924mk2lt\nbc2ZDHf16tXw8PCUlBRiy/Lly0kkUlxcHIVCwVcmNTU1+/btq6ioGJg003nz5pmamjIYjG3b\ntlEolJKSkra2tsbGxoKCgqNHjx49ehQhNGvWLN60OV5sNru6uvratWsXL15ECE2bNg33vxfn\nXUja2tqjR49uaGj4888/X7161dnZyWAw0tLS/vrrL4TQggULiHkpwp+mefPmkUik+Pj4f//9\nF6cJtbS0xMXFHTt2DCG0YsUK0ZakBQAAUZGWlt62bRtC6MKFC0FBQbxjai9evDhw4ABCaMWK\nFQJyKvqCTqebmpriFETO7R4eHviIAsK4mTNnIoSCgoK4iqIWFRUZGxt/++23Pa1FK2RnRH5c\nEe4TVyjmmgNGp9O5lvEihsbw1x+nffv2nTx5Eh9O+F4JeVwRvlIh9fivlgiknjx5YmVlxZm9\nLYzIyMieHvGTRSKRfH19z5w5Ex8fj5cz4Gpgbm6+ceNGYunKhoaGJ0+eIIT4DrtikydPvnTp\nEpvNTktLmzdvHt64cuVKZWVlCoVCpVJxHMlp5MiRW7ZsycnJ4e1AX9jb20+dOjU5Ofns2bNn\nz56VlZUlVty0t7fHYT02bty4ZcuWXbx4MSoqKioqSkZGhk6n449LY2PjTZs2cQbfcXFxNBrN\n1taWeBNMTExWrFhx7tw5CoVy+fJleXn5pqYmMpm8efPmf/75R7R1Nvgik8m+vr5BQUEvX76k\nUCh4MIZAIpGmTp3a1crYXdVxRwgZGBjga7Zen3ch/fTTT35+fu/evdu6daukpCSTycTZeOPG\njVuyZAnRTPjTZGVltWzZsvDw8MjIyMjISOLUk0ik+fPnT5s2rUfdAwCAgfTLL7/cvXs3Ojp6\ny5Ytd+7c+fHHH8eMGaOhofH27dubN2/u27ePTqcPHz784MGD/dQBBQUFCwuL58+f//TTT+Hh\n4fjudHV1tb+/P5PJ1NfX56qZVlZWRsxT+uWXX06cOJGbm7tw4cIjR47g25sPHz708vJiMpnr\n1q3jXLZchJ0R+XFFuM8RI0ZkZGSEhIT873//w7PMnz9/vmnTJrza0cuXL4ncmEOHDiUmJoaE\nhAwZMmTr1q0qKiotLS3+/v4ZGRnq6uqzZ8/uUa+EP25/vHsC9PVyE2eDfLXIZPLatWudnJyS\nkpKePHlSW1vLYrG0tLS0tbXt7Oy4Fh9NT09nsVhSo23z8gAAIABJREFUUlK8C9QTNDU1zc3N\nX716lZqayhnAubm5OTg43Lp1Kzs7m0ajtbe3Kysrm5mZOTo64gXYRo8e7eLi0ruVXLuyadOm\niRMn3rhxo6Ki4sOHD7q6unp6es7OzqNHj+Zq6e7ubmtre+XKlbdv31ZUVKipqenq6k6cOJEY\nchZswYIFxsbGN27cKCgooNPpY8eO9fT01NXVpdPpA1MvXF1dPSAg4OHDh/fu3ausrKysrJSQ\nkNDW1tbT05s7dy5OMhGSgoKCpqamra2tq6srHs/oy3kXsvNHjx6Niop6+PBhdXW1qqqqkZHR\n5MmTeWvUCH+a3N3dLS0tr1y5UlJS8uHDBz09PUNDwzlz5uC0IgAA+GSRSKTIyMiNGzeePHky\nISEhISGBq8GECRNCQ0P79c7hyZMnJ0+efO/ePUNDwyFDhoiJiVVXV7PZbGlp6cuXLxPNtLS0\n6uvrp06dOmzYsAsXLlhaWkpLS1+6dGnRokXXr1+/fv26srJyZ2cnzs1wcnIKCgrqp870x3FF\ntc9du3Y5OzufOHHi8uXL+vr6NBqtoqLC3Nz87Nmzq1evjomJUVZWLi8vV1RUtLCwOH78+MaN\nGw8ePHjw4EF1dXUclcnIyPz7779DhgzpUa+EP25/vHsC9DVw71FMMzBOnTo1wEc0MjLqakSW\n0+zZs/EFn2CHDh3iu11dXX3p0qVLly7l++iQIUOE6QMXX19fwQ2sra2tra2F2ZWuru7GjRu7\nbRYSEhIbG8t1zw4hNHLkSK5cFFwEU7SXIoKNHz9+/PjxwrRUUFDgrfrUlT6edyTEacLlh7kq\nEPMl5GlCCJmbm//555/CtAQAgE+KuLj48ePHvb29z507d/v27ffv33d0dJiYmJiami5cuNDD\nw6O/O2Bvb5+XlxcQEPDo0aPy8nIJCQkrK6spU6b8+uuvuGAAduLEiQ0bNhQVFXGuCTNhwoSn\nT5/iYibPnj2Tl5cfO3ast7c3Tnfsv86I/Lii2ueMGTPu3bu3b9++vLy8d+/ejRkzZs2aNT4+\nPjIyMnJyciEhIeLi4kS91zVr1kyYMCEwMPDx48fFxcXm5uY2NjY7duzgnKksZK96dNz+ePe6\nQurpqs7EXFpjY2PRLpMBvhLh4eFMJvPHH3/EvwYEBBQVFa1Zs4arNPjp06dv3Ljh6ekpwmUL\nwMArLy9fsGDB/v37cRYgl5qaGoXA3QPfKwGkAriz0Qj19fW4Ao+Kiopo67jX1NTIysqKts5m\nc3Mzzl5VUFAQbcoZLpYs2pu/dDodT6jAX4oi3HNzczODwejpvbtDhw6lpKTEx8eLsCd9R7rb\n5YTLAcOe/Gn9wwJejY2NtbW1A7+I4WAd92vT4xH3bis9AyBASkpKSkrKjh07iC3Dhg1LT08/\nd+6coaEhXtEJIXTnzp3ExEQymcy1zigAAAAABFBUVBTtdfUnftyvzRdexz0sLIyrzqBIhISE\ndFUeBAiG659wZli5urrm5ua+ePHC29vb2NhYSkqqvLy8rq5OXFzc29sbV2aF8wgAAAAAILLA\nva6urri4mO/yLpz09fUHMmt5zpw5eO5mt6qqqlRVVYVcFhuv9wl6YcGCBVxbxMXF9+zZc+fO\nneTkZFxPU0NDY8SIEd9//z1xxw3OIwAAfDF8fHx4a/b13bt377padAmAL4YIAvekpKQNGza8\nevVKmMYUCmXx4sV9P6iQ1NXVhRxSHT58eH93BnRFXFzc2dnZ2dm5qwZwHgEA4Iuxfv163kEc\nvt69e6elpSXkJA18hxaAL1tfA/e8vDwnJyeumvMC8FYSBAAAAMDXw9DQUMiSdAKq6ALwdepr\n4L5jxw4Wi0Umk728vCZOnIjLZAqAl1UHAAAAAAAA9IgIRtwRQhcuXBjIBBgAAAAAAAC+NmJ9\nfH5FRYW6ujpE7QAAAAAAAPSrvo64Dx06VF5eXiRdAQB8hZp8tpNIpG6z7HqEyWTW19crKipK\nSkqKcLcAAADA4OrriPuUKVOKioqqq6tF0hsAAAAAAAAAX30N3H19fSUkJNauXctms0XSIQAA\nAAAAAACvvgbuI0eOPHPmTExMjKOjY2pqaktLi0i6BQAAAAAAAOAkggWYpk6dampqev/+fUdH\nRxKJJCcnJ6BxU1NT348IAAAAfG3Yk3cPdhcAAIOsr4F7RUWFjY1NZWUl/pXNZjc3N/e5VwCA\nr4VC4G6EUPvHX6UCjg5iZwAAAIBPWV8D9507d+Ko3djY2MbGRrSlIQAAAAAAAABYXwP3+Ph4\nhNDGjRuDgoLIZLIougQAAAAAAADg1tfAnUajycvLHzp0CKJ2AAAAoP+QYtoGuwuIPV9msLsA\nwFetr4G7ioqKmpoarHICAAAAAABAv+prOcgJEya8fv26o6NDJL0BAAAAAAAA8NXXwP23335j\nMBhBQUEi6Q0AAAAAAACAr76mytjb2x8+fNjHx4fBYGzdulVaWlok3QIAAAAAAABw6mvgvm/f\nPoTQ2LFjd+zYcejQIQsLC3V1dQHtY2Nj+3hEAAAAAAAAvkJ9Ddy3bdtG/NzS0vLo0aM+7hAA\nAAAAAADAq6+Bu7m5uUj6MTAYDMb58+cfPHjQ3Ny8bds2Kyur/j4ijUZLSkrKysqqrq6m0+ma\nmpra2trTp08fN26c4CfGx8cHBwcjhNTV1UNCQgQ3rq+vT05Ozs7OrqysbGxsVFFR0dLSMjEx\ncXFxUVFREdmL+fr4+/vn5eXxbieTyWpqatra2rNmzRo/fjzXox4eHi0tLT4+Pg4ODnx3e/Hi\nxf/++2/EiBEHDx7kfRTOJgAAAAD46mvg/vLlS5H0Y2BERETgXB1tbW1x8b6+9m5FR0dfuHCB\nwWAQW0pKSkpKSjIyMkaOHOnn56egoNDVc+/du4d/oNFoBQUF33zzjYCjXLp0iU6nE1toNBqN\nRsvLy4uNjXVxcVm2bBmJROpp52k02urVq728vNzc3Pg2KC4ujoyMLC0tpdFoqqqqOjo6Li4u\no0aN4m3Z0NAQERGRn5///v17RUVFfX39efPmjR49uqdd+nSwWCwqlUqlUnNzc+3t7X19fUW1\n576fTSaTuXXr1tevX58+fVpTU5PY/uOPP9bU1Ah44rhx4/z9/YlfGxsb//vvv4KCgvfv30tK\nShoYGNja2jo7O4uJ9XVGOwAAAAB6p9+D109Kbm4uQmj9+vVOTk79fazw8PCoqCiEkLW1taur\nq6GhIUKIRqPdu3cvPj7+2bNnhw8f3r59O98gjEaj4SsieXn55ubm1NTUrgL34OBgvHitgYHB\nnDlzjIyM9PT0GhsbS0tL4+LicnNzo6Ki2tvbV69e3dP+37p1i81md/XojRs3QkJCOjs7cSep\nVGplZWV2dvbs2bPXrl3L2fLVq1c7d+5sbW1FCElKSlZXV9NotOzsbCcnp/Xr1/e0V4Ni4cKF\nXl5enFtaWloqKyujoqIyMzPT09OTkpKmTZvW9wOJ5GyeO3fu9evXvTg6Z0Sen5+/b9++pqYm\nhJC0tHRjY+OTJ0+ePHly69at3bt3Kyoq9mL/AAAAAOijAQrc37x5k52dPWLEiJEjRw7MEflq\naWlBA5Leg2MshNCiRYuWLFlCROcKCgrGxsZjx47duXNnTk7O7du3Z86cyfv01NRUNputoaEx\nbdq0S5cupaWlrVy5knekMyEhAcd5bm5unp6exD0EWVlZTU1NGxub+Pj4kydPxsbGWlpa2tnZ\nCdn5jo6O5ORk3H++qFTquXPnOjs73d3d3d3dZWVl29vbY2Njw8PD4+LixowZY2Njg1u2trYe\nOnSotbXV2tp65cqVOjo67e3tycnJp0+fTkxMHD58+NSpU4Xs1SASExPjevMVFBQUFBT8/Pw2\nb95cXFz88OHDvgfuIjmbjx49un79Ot+HDh48iC+0uLBYrP3791dUVHz33Xd4C5PJPHbsWFNT\nk4WFxU8//aSnp8dkMrOysk6dOlVcXHz06FHOmS0AAAAAGDADdNf72rVrixcv3rt378AcTrBe\n5I301NmzZxFCdnZ2np6evIezsrKaPn064siH4YK3T5o0yd7eHiFUV1f37NkzrjZ0Ov3ff/9F\nCE2ePPmHH37gm/kza9YsnGaNI8Jupaenb926denSpcePH+cb5GEJCQkMBmPMmDFeXl6ysrII\nISkpKXd390mTJiGEHjx4QLTMz8+vqalRUlLy8/PT0dHBLWfNmrVgwQKE0P3794Xp1SdLTEwM\nT5MoLS3t465EcjZramr+/vtvfEZ4qaura/KTkZFRVla2bNkyMzMz3DItLa28vFxOTu6PP/7Q\n09NDCImLi9vZ2f3xxx8kEunRo0d9f70AAAAA6AXRjLg3NjZmZWUVFBTwTa5obGw8fvw4Qqi+\nvl4kh+uFkJAQYiRyw4YNCKHt27dbW1v3x7Hy8/PLysoQQosWLeqqzfTp09+9e8dkMul0Olfx\ne5wHjxCaNGmSnp6erq5ueXl5amoq11Tau3fvNjU1ycrK/vjjjwI64+XlNWbMGIQQm83u9oqF\nSqVWVVVJS0tLS0s3Nzd3tSAujtt459daWFikpqZyRnX4Z11dXUlJSc6WJiYmiCfeXbNmTWVl\npa2t7R9//MG5/fXr17GxsQUFBR8+fNDX158/f76Dg4OHhweLxYqIiBD8igZG35cv6PvZ7Ozs\nDAoKampq2rp166FDh4Q87uvXry9evDhq1ChXV1di49u3bxFCEyZM4JqDYWZmpqurW1ZWVlRU\npK+vL+QhAAAAACAqIgjcU1JSFi5cKExQPnny5L4frneMjIzs7e1zcnLodPqoUaPk5OSUlZX7\n6Vi4DomJicmwYcO6avPNN98EBgbyfeju3bsIIX19fQMDA4SQnZ1dZGTkgwcP1q1bxzkQ+/jx\nY4TQt99+K/iFqKurC5+O4ubmRkxFDQgISE9P59ts8uTJY8eOHTt2LNf2hoYGhBBnArSGhgZC\nqLS0tLW1lXMkGGfw40cFS0pKOnHiBL6EkJGRKSwsDAwMzM/PF3BDYMCw2Wx8J+Tbb7/t4676\nfjYpFMqLFy9mzZo1YcIEIQ/KZDKDgoIkJCQ2bdrEeRlQW1urqKhoamrK+xQZGRmEEOd8awAA\nEAadTvfz84uKiqqrq7t+/bpI5gUJVl1dHRYWFhcXV1RUVFNTo6qqamxsPGfOnOXLl2tpaQl+\nbnBw8E8//YQQMjQ0xGMZXFpaWuTl5RFCmZmZvLXFCJcvX/bw8EAIPX78mLd4g4ODQ1pamoKC\nQm1trYSEBOdDnZ2dysrKTU1NP//889GjR4V7xeCr0NfAvaqqatasWe3t7YKbkclkLy+vzZs3\n9/FwvTZt2rRp06bhMd1Vq1b163ghnhrY7ecCX2w2OzU1FSHk6OiIt0yYMCEyMrKlpSU3N5fI\nHUcI4UH9QRn4xCkxnDo7Ox8/fhwTE4MQ4pz4a2trO3z48JcvX+7Zs2f58uWGhoZNTU3Jycmx\nsbESEhKenp6cO5kxY0ZTUxPnK3r//v0///zT2dnp5ubm6uqqqKhYWVl5/PjxmzdvIlGMc/da\nW1sbnpz6+vVrbW3trgrvCK+PZ/Pp06cREREGBgYrV64U/lk3b958//69h4eHqqoq5/YtW7bw\nbV9VVfXmzRv08YYJr5qaGt6PAlzKprOzk8ViCdMrIZsJhq/rhD+okIg7iqLdLdZ/ve2PPbPZ\nbNHuk7gUF/mee9dbAVPzQe/s3bv377//RgiZmppy3YPtD3g2Dp5hj+FSYOnp6fv27fP39xdc\nDezSpUv4h3fv3mVmZtra2vauG1OmTME/pKencwXuTU1NDx8+xD9kZGRwfbE+f/4cd55zxLOp\nqenAgQNXrlwpLS1VUFCws7Pz8fGZOHGigA6UlJQYGRkdOHBAhNXPwODqa+AeHByMv6p9fHyW\nL1/e2trq4eHx+vXrqKgoa2vrDx8+3Lx5c8+ePYqKin/++aeUlJQo+vypa2xsRMINJ/PKz8+v\nrq5GCBElwI2NjTU0NKqqqlJTUzkDd/wvzVnvD6utrV2xYgXvnv38/IQfixUehUJJSUmpr6+n\n0+lKSkq//PIL57xJMpm8a9euEydO3L1718fHh9g+dOhQX19frlI57u7uXDu/cOFCZ2ens7Pz\nDz/8gLdoaWnt2rVr3bp1VCpV5K+lK1FRUV1N1XVwcPD29lZSUurjIfpyNhsaGoKCgiQlJbdu\n3SopKSnkvYiWlpaIiAhlZWXOJBkBGhsbDx48yGKxrKys+A7GI4S2bduWnZ3NtRG/qJaWlrq6\nOmEOJGQzYTQ3N4tqV1zw/7ho0el0zjKgItTS0oLn5YsQi8US4Zni1E/vQ097297e/inc1vuS\nJCYmIoROnTrl7e3d38f6+eefjx07hhCytLT09fV1cHBQU1Orra1NT08PDAzMzc318/N79epV\nWFgY36eXlJTgG84qKip1dXUUCqXXgbuGhgYevXrw4AFXIbW7d+8S+aiJiYlcgXtGRgZCiEQi\nEdtramrs7e0LCwsRQhISEtXV1devX8cV3vh+TWBnzpyBq9AvTF8np966dQsh5ODgcOjQoREj\nRlhbW2/cuBEh9PTpUwMDg9GjR2/bti0iIqKmpsbd3f0r+RzE/4q9u0rBw+1mZmacMRwOhR89\nesT5fYb3P4jTBrDa2loqldrR0UEikdhsdkNDA9dnRH5+/vPnzxFCZDJZS0sL31v88OFDamoq\nk8kUsGc2m41HIxYuXMi5nUwmz507V/SvpFdev35dWVnZ9/30+myy2ewjR47U1dV5e3vjiaRC\nioiIaGpq8vDw6PbGBZvNTklJ2bBhw5s3b7S0tAbxvhkA4POFP9/6Y/yIy5kzZ3DUvnbt2qys\nrKVLlxoYGMjKyurp6S1evDgjI2Pz5s0kEuncuXOHDx/muwcKhcJms42MjDZt2oQQioiI6Mtd\nIDzozpt6evv2bYTQiBEj0MerGk44cB85cuTQoUPxFm9v78LCQk1NzaSkpObm5uLiYi8vr87O\nzjVr1vAtAdze3n769Gm+y/yBz1pfR9zfv3+PEJo1axaxBac+P336lNgyf/58R0fHe/fuXb58\necmSJX084qcPT+nrxZAwi8VKS0tDHHky2IQJE6Kjo+l0elZWFjESr6SkRKPReKPGIUOGREdH\nc27ZvXs3LmCPBQQEcI4XSktLcy6701MbNmzAk32rqqqOHz9+/vx5Go22bt06/Ghubu6ePXsQ\nQqtWrZo9ezbO0S8sLDx27FhsbGxNTc3vv//e1Z5pNBqTyZSWluYdhzYyMup1h3uBt457R0dH\nWVlZWFjYs2fP9u7dGxIS0se8nV6fzatXr+bm5k6cOHHGjBnCH66pqenmzZsKCgq4upEAL168\nCAkJwRkyEydOXLNmjYDbC0OHDsW1gzgpKytTqVQxMTEymcz7FN6vQ77NeorNZnd2doqJiYm2\nhFRnZye+LhX5nlkslsj3id8E1A+9JXYrwn0SvSWRSKLdc+96OwDFx75O/f3GNjY2bt26FSHk\n4eGBVx/nIikpGRQU1NzcfPr0aX9/fy8vLyIyJuA8GQ8PD3d39x07dlCp1Lt37/Y6KX/KlCkn\nTpwoKSl5//495yckDtz379+/cOHC3Nzc6upqNTU14lEcuBN5Mvn5+deuXUMIxcbG4roaRkZG\noaGhxcXFaWlpf/3118mTJ4nnRkZGHjlyhEi2AV+YvgbuODzlTAsxNjZGCOFvesL3339/7949\nCoXyNQTuOjo6eXl53Q7EHj58uKWlZfr06URuSW5uLv43O3PmzJkzZ3ifcu/ePSJwNzExKSoq\nwvVnuHB9RZWXl3P+WlBQwLmCppycXPcvSQgaGhq//fabl5dXQkKCu7s7/gAKDQ1lsVgeHh4u\nLi5ESzMzM39//3Xr1mVkZLx48cLCwoLvDquqqhBCKioqvA/x3dh/eOu4S0lJmZiY+Pn5rV69\nurGx8enTp5xZTL3Qu7P57t27CxcuaGho4Gsn4d2+fZvBYDg5OXFNh+LU2tp6/vz5hIQENptt\nbGzs5eXV7WK3fOu9lpeXL1iwQE5Oju9Z413MVSQnl8lk1tfXy8vLizaVtr6+Ht8mUlJSEskF\nBqGmpkZaWrqrUp6909zcjO/RycnJiTZNsb6+XkxMTLTrcNHpdJzaJC0tLaoPJay5uZnBYPT0\n70pKSgoWCRaVX3/99X//+x/+2dLSEiF048aNOXPm9MexQkND6+vrZWRkAgICBDTbu3fvf//9\n19DQEBwczDV09ezZM1x1wMPDY8SIEebm5q9evaJQKL0O3CdPnozvSD948IBYLqOsrOzVq1cK\nCgpz5swZO3ZsVlbW7du3iQCptrYWp8QQKfJXr15FCI0ZM4azGh6ZTF65cmVaWlpMTExwcDBx\nUVRcXPz27Vs5OTk5Obm6urpuZyGCz0tfP5jwtDbOqzp1dXUZGZnXr19zjqXhaJ5zpPALhj+Y\nioqKaDRaV20qKiru3r2blZXF+XWCy7eLi4sr8sADurm5uUTaLp7mUlhYiBNRutLY2MjVjdDQ\n0OscKBRKj15dXV1dcHDwqVOneB9SUlJSVVVls9l4qiWdTscFH3lXC1JTU8MTHF+9etXVgfA7\nwzd7BJevGXQKCgq4cBC+xiDgsENAVjE+iZzRSe/OZkVFBYvFqqqqWrx4sctHuEY+Qsjb29vF\nxQXf8eDEZrMTEhIQQnwX/8Lq6+t/++23+Ph4ZWVlHx+fw4cPdxu1AwAAX1ZWVu7u7vgTb8aM\nGe7u7rz3UUUlLi4OIbR48WLBqYNqamp46hTvshgXL15ECFlYWOCvcpyreeXKlV5X0xo6dCje\nFeciJ3i4fcqUKeLi4s7Ozuj/ZstkZmai/5vgjq8leK92cOIolUrFX7uYr69v5Ufz5s3rXbfB\nJ6uvgbuuri76WMyOYGxs3NbWVlBQQGzBF4IfPnzo4+E+CzY2NoqKih0dHcS0dF44l11GRoaY\n50en03FK96JFiy7wwGOZTCYT3z7DR1FXV0cInT9/XkBnBD/aCwoKCrdv37558ybvZQmLxcIh\nNa5pKCEhgcc7+eay440Chhi1tLTExMTa2tp4D/TpLACE80a4JkHiaJ5v0iFWVFSE/m9tlt6d\nTXFxcTkexFsqIyMjJyfHm8OTm5tLpVJNTU1xvVFeDAZj+/bt79+/t7a2Dg4OdnBwgLQBAECv\nLV++PDIyEldaO3LkSGRkJG81YVF59OgRQkiYNdrxzd6srCzOeVlsNhsPZhGD37hoWH19PR7v\n6B3eNHccuOMUR1yK7datW0RPcIhvZWVFlPzC9/B5K48NHToUV+nFI/Tga9DXwB1nCPz333+c\n1STwZAt8ZwfDpc2/kkVbJCUl8dJLKSkp0dHRvBO6S0tLcZWS6dOnEzfcMzMz29vbSSQScWuM\nk5mZmba2NvoY8SOEyGQynkheUFCwf/9+vuO7CQkJ+NNBhMTFxQ0NDRFCSUlJXA/dv3+fyWRK\nSEjgE00mk3FLos+EsrKy4uJihFBX9UnwgfAoBU7sI7DZ7Bs3bvT5dYgGTp/luhGJ//7v3r3L\nN3ZPTk7Gn7D41WG9O5s2NjYUHsTl4t9//02hUDiL+WD4xo6A3J6UlJR3796ZmJj8+eefok3e\nAACA/tPe3o4Hj8zNzbttPHz4cIQQk8nkrDiUlpaGB4YWL16Mt4wZMwbPqrp8+XKvO4a/1h8/\nftza2ooQYrPZ+AsUB+7jx49XUlKiUqnE5ECuBHf0MXDnKt2LDRkyBIm0Hhf4xPU1cN+4caOY\nmBidTndwcCCqWeM1YoKCgu7du9fZ2ZmdnX3kyBH08f/kazB37lxbW1s2mx0aGrpr16709HQq\nlUqn00tKSqKiorZs2cJgMPT09Igqh+hjOGVpaYlHXnnh/+GnT58S/5/29vbz589HCGVmZv76\n66937tyh0WgdHR1UKjUzM9Pf3//EiROampp4rU0Rwic6KioqJiYGfwy1tbUlJibiyTGLFy8m\nFopatGgRiUSKiYkJDw/HSS8MBuPRo0e7d+9mMpnW1tY4hwq7evVqeHh4SkoKsWX58uUkEiku\nLo5CoeBYtqamZt++fRUVFSKfb9c7+JYCV3HAefPmmZqaMhiMbdu2USiUkpKStra2xsbGgoKC\no0eP4qU0Zs2axTUmNDBnk81m4/tj+OqCL3zH1sHBoba2lsZPP1UtBACAviCqKwoz4kDcjeS8\nJ4wHPsaPH8+5fiLOlrl+/Xqvy6pOmjRJTEyMyWRmZWUhhJ48eVJdXa2np4drIouLi+MEevzZ\ny2KxcDPOUTw8HYjvxBJ84xd/F4OvQV8np5qZmR04cMDPz49Op1+/fh1vXLp06fbt22k02uTJ\nk8XFxYn/it9++62Ph/tckEgkX1/fM2fOxMfH5+bm8ib3m5ubb9y4kZg519DQ8OTJE/R//1G5\nTJ48+dKlS2w2Oy0tjchaW7lypbKyMoVCoVKpvIurjRw5csuWLTk5OaKdXWBvbz916tTk5OSz\nZ8+ePXtWVlaW+Miwt7fnXI1o3Lhxy5Ytu3jxIi6FLiMjQ6fT8S0IY2NjrgU74+LiaDSara0t\n8SaYmJisWLHi3LlzFArl8uXL8vLyTU1NZDJ58+bN//zzz6ewLABOxM/Ly+NcGpZMJvv6+gYF\nBb18+RKPgnM+hUQiTZ06ddWqVbx7G4CzWVRU1NDQQCaTuYroE9hsNh5wCgsL66rI8ebNmwdx\nFWQAAOBLXl5eWlqaTqcXFBRwFWfjhRfwJpFIeMQaIdTR0REZGYkQwmudEtzc3IKCglpaWmJj\nY4mR+B5RVVX99ttvnzx58uDBA0dHR848GczJyenq1asJCQlbt2599uxZc3OzmJgYUYsCfSwZ\nzHdtCjxyRLwK8MXra+COENq6dauFhUVoaCjOLUMIycnJhYWFubq6MhgMHLWTSKTff/+d86/w\ni0cmk9euXevk5JSUlPTkyZPa2loWi6WlpaWtrW1nZ8e11EJ6ejqLxZKSkrK3t+9qh5qamnh6\ne2pqKud0Ezc3NwcHh1u3bmVnZ9NotPb2dmXDHfOBAAAgAElEQVRlZTMzM0dHR7xmxOjRo11c\nXHq3kmtXNm3aNHHixBs3blRUVHz48EFXV1dPT8/Z2Zl3CqO7u7utre2VK1fevn1bUVGhpqam\nq6s7ceLEadOmCTNkvmDBAmNj4xs3bhQUFNDp9LFjx3p6eurq6tLpdL43DQfYiBEjoqOjq6ur\no6KiOEtGqqurBwQEPHz48N69e3iGkISEhLa2tp6e3ty5c3EGEV/9fTZzcnKQwJUL6+rqej0H\nCwAABpeFhUVOTg4OygXLz8/H7YmE1cTExNraWoTQpk2bcAV3LhQKpXeBO0JoypQpT548wWnu\neAEcrsAdIZSent7S0oLzZKysrDhrVygrK1dVVeHuccGzB3EyLfgaiCBwRwjNmTOHa7Lz7Nmz\n8/Lyzpw5g5NlZ8+e3e3l7wDgWwulXxkZGfEdW+Uye/bs2bNnd9vs0KFDfLerq6svXbp06dKl\nfB8dMmSIMH3g0u3yyNbW1px1qQTQ1dXFy3IJFhISEhsbiz9MOY0cOZIrqwSXTRTtpQhfvCVZ\nuNja2hI3mniNHz9+/PjxPT1oH8+mmJiYgC55eHhwDSZxUVVVFfB0AAD4lDk5OeXk5EREROze\nvRuvqcLpzZs3enp6kpKSLBYLT8PjXIUG58lISkryZqS0tbW1tLQkJCTU1dX1rmTtlClTjhw5\n8uDBg9bW1vT0dBKJxFlf0sDAAA/MpaSk4MCd6/a7qalpQUEBXjmHU2NjI07g4V1DA3yp+rFO\nrbm5eVBQ0JUrVwICAj6FqB18+urq6jhXoAgICFi1ahXO9uOEEwGFmX4EAADg67FmzRpJScmK\nigreYZfy8nIHB4dJkyaVl5cfPny4qKhISkrql19+wY+2tLTExMQghPz9/at5JCcnI4QYDAZn\n1Y0emTRpEplMrqurO3PmDJ1OHzVqFOeXHfo46J6YmIhLynClI+JyArwLrOJaN0OHDu1qdhz4\n8sACE+BTkZKSkpKSgmc2Y8OGDaPRaOfOnauuriY23rlzJzExkUwmw9UgAAAATvr6+ng23ZEj\nRwIDA3HhL0xdXX3u3LkPHz4cNWoUXrTb398fl7RGCEVHR7e2tpJIJL53O21sbHANtJ6ufEJQ\nUlLCqaR4ZSjeta5x4B4VFfXmzRsxMTGufFpXV1eEUGZmJlexMlx13tXV9VOo1gAGRs9SZfCf\nznffffcFLIAaFhbGVWdQJEJCQuDCt3caGhr+/PNPzvxvV1fX3NzcFy9eeHt7GxsbS0lJlZeX\n19XViYuLe3t74yU84DwCAAAg7Nmz5/nz57GxsVu3bo2NjV21atWoUaPMzMw+fPgwc+bMuLg4\nnHBibm6Ow3cMR8COjo5dzUFaunTpjh07UlJSqFRq7xaQmjJlSnZ2Ni7syBu4Ozo6SklJ4dXo\nR48ejWvFECwtLRcsWBAdHe3i4pKYmKinp9fe3r5///7r169LSkp2m9oKviQ9C9yjo6PR/60/\nTXj//j1ejL0XSb2DYs6cOXi2X7eqqqpUVVUFLA7PSSQLtn+diCU/CeLi4nv27Llz505ycjKu\np6mhoTFixIjvv/8eF9ZFcB4BAABwIJPJ165d27Zt2+HDh+/fv3///n2uBlJSUrKysq9evfLw\n8AgLC5OVla2ursaVXjhrDHDx9PTcsWNHZ2dnREQEkWCDEBLwBcRkMomZrwihKVOmBAYGIoSk\npaUnTpzI1VhWVtbBweHOnTuoi/pyp0+ffv78+cuXL/X19U1NTcvLy9va2shkckhICGdhZfDF\nE83kVIRQSEjIzp07EUK86w19mtTV1YUcUv16ys9/gvBy0HhFaL7gPAIAAOBEJpMPHDiwdu3a\nsLCwxMTE4uLi+vp6TU1NAwMDZ2fn1atXt7S0zJ49OyIiAiH033//RUZGMplMWVlZd3f3rvZp\nbGxsZ2eXkZFBoVA4A3fhOTg44ALZDg4OvGtaI4ScnZ1x4M633q6amlpOTs7+/fuvXr1aWlqq\noKAwc+ZMPz8/IYeuwBdDZIE7AAAAAABfRUVFA3xEAwODnTt34iFFLmpqag8ePPD09PT390cI\n/fTTTz/99FO3O8TTRjE5ObmeDlPKy8sTS0Tx9dtvvwle7kZRUfHgwYMHDx4U/qC4Mj34kkDg\nDgAAAICvi7Ky8s2bNwe7FwD0GFSVAQAAAAAA4DMAgTsAAAAABo6Pjw+pH+C1+QD4skGqDAAA\nAAAGzvr163nLiPH17t07LS0tKSkpYRr3rkojAJ8XCNwBAIOpyWc7iUQaMmTIYHcEADBADA0N\nuyqXzsXe3r6f+wLAZwZSZQAAAAAAAPgMQOAOAAAAAADAZwACdwAAAAAAAD4DvclxT0hIqK+v\n59r46NEj/MPPP/8s4Ln//PNPL44IAPhSKQTuRgi1i3y3CLFFvVuZjz8wEWKKdM8KCCFR91YC\nIYmPP/fH+yDafZI+vgki3zN+H/juUyrgqEgPBQAA/a43gXt2dnZ2dnZXjx47dkzAcyFwBwAA\nAAAAoBcgVQYAAAAAAIDPQM9G3O/evds/3QAAAAAAAAAI0rPA3dHRsZ/6AQAAAAAB2PNlum8E\nAPiiQaoMAAAAAAAAnwEI3AEAAAAAAPgMQOAOAAAAAADAZ6A35SABAAAAMMCoG5oHuwtI85j8\nYHcBgK8ajLgDAAAAAADwGYDAHQAAAAAAgM8ApMoMKAaDcf78+QcPHjQ3N2/bts3Kyqq/j0ij\n0ZKSkrKysqqrq+l0uqampra29vTp08eNGyf4ifHx8cHBwQghdXX1kJAQwY3r6+uTk5Ozs7Mr\nKysbGxtVVFS0tLRMTExcXFxUVFRE9mK+Pj4+PgUFBVwbxcTENDQ0tLW1nZ2dx48fT2zPzs7e\nvXu3lZXVnj17enSU+/fvBwYG2tra/vHHHyLoNAAAAAD6BwTuAyoiIiI2NhYhpK2tLS7e729+\ndHT0hQsXGAwGsaWkpKSkpCQjI2PkyJF+fn4KCgpdPffevXv4BxqNVlBQ8M033wg4yqVLl+h0\nOrGFRqPRaLS8vLzY2FgXF5dly5aRSKSedp5Go61evdrLy8vNzY1vg+Li4sjIyNLSUhqNpqqq\nqqOj4+LiMmrUKN6WDQ0NERER+fn579+/V1RU1NfXnzdv3ujRo3vapU9EZ2dnZWVlZWVlTk6O\nnZ2dn59fT9/esLCwa9euIYSuX7/ebeNuTwQAAAAABgYE7gMqNzcXIbR+/XonJ6f+PlZ4eHhU\nVBRCyNra2tXV1dDQECFEo9Hu3bsXHx//7Nmzw4cPb9++nW/MR6PRXr58iRCSl5dvbm5OTU3t\nKnAPDg6Oj49HCBkYGMyZM8fIyEhPT6+xsbG0tDQuLi43NzcqKqq9vX316tU97f+tW7fYbHZX\nj964cSMkJKSzsxN3kkqlVlZWZmdnz549e+3atZwtX716tXPnztbWVoSQpKRkdXU1jUbLzs52\ncnJav359T3s1KJYsWfL9998Tv9Lp9PLy8itXrmRkZGRkZCQkJMyaNav/ji74RAAAAABgwEDg\nPqBaWloQQubm5v19IBwxI4QWLVq0ZMkSIjpXUFAwNjYeO3bszp07c3Jybt++PXPmTN6np6am\nstlsDQ2NadOmXbp0KS0tbeXKlWJi3DMiEhIScNTu5ubm6elJ3EOQlZXV1NS0sbGJj48/efJk\nbGyspaWlnZ2dkJ3v6OhITk7G/eeLSqWeO3eus7PT3d3d3d1dVla2vb09NjY2PDw8Li5uzJgx\nNjY2uGVra+uhQ4daW1utra1Xrlypo6PT3t6enJx8+vTpxMTE4cOHT506VcheDSIxMTHON19W\nVtbMzMzPz2/79u15eXm3bt3qaeBuaWkpTLNuTwQAAAAABhJMTh0Evcgb6amzZ88ihOzs7Dw9\nPXkPZ2VlNX36dMSRD8MFb580aZK9vT1CqK6u7tmzZ1xt6HT6v//+ixCaPHnyDz/8wDfzZ9as\nWQ4ODgghHN93Kz09fevWrUuXLj1+/DgeTecrISGBwWCMGTPGy8tLVlYWISQlJeXu7j5p0iSE\n0IMHD4iW+fn5NTU1SkpKfn5+Ojo6uOWsWbMWLFiAELp//74wvfo0kUikiRMnIoTKysp6OiI+\nbty4FStWrFixoqsGQp4IAAAAAAwkGHEfICEhIUQ+8YYNGxBC27dvt7a27o9j5efnl5WVIYQW\nLVrUVZvp06e/e/eOyWTS6XRpaWnOh3AePEJo0qRJenp6urq65eXlqampXFNp796929TUJCsr\n++OPPwrojJeX15gxYxBCbDa72ysWKpVaVVUlLS0tLS3d3Nzc0dHBt1lpaSlCiHd+rYWFRWpq\nKn6Us6Wurq6kpCRnSxMTE+JRwpo1ayorK3nnaL5+/To2NragoODDhw/6+vrz5893cHDw8PBg\nsVgRERGCX1G/wsPwLBaLxWJxXjix2eyYmJg7d+5QqVQ5OTljY+PFixebmZkRDbqdxirkiQAA\nAADAQILAfYAYGRnZ29vn5OTQ6fRRo0bJyckpKyv307Hy8vIQQiYmJsOGDeuqzTfffBMYGMj3\nobt37yKE9PX1DQwMEEJ2dnaRkZEPHjxYt24dZ3T4+PFjhNC3334r+IWoq6sLn47i5uZGzIAM\nCAhIT0/n22zy5Mljx44dO3Ys1/aGhgaEkKKiIrFFQ0MDIVRaWtra2orH5jGcwY8fFSwpKenE\niRM4cpWRkSksLAwMDMzPz/8UxqHx+2NgYMB5Xjo7Ow8ePJiRkSErK6umpkalUrOzsx8/fhwQ\nEMAZuwsm5IkAAAAAwECCwH2ATJs2bdq0aXhMd9WqVfr6+v13rNevXyOEtLS0evFcNpudmpqK\nEHJ0dMRbJkyYEBkZ2dLSkpubS+SOI4TwoH6/vpCu4JQYTp2dnY8fP46JiUEIcU78tbW1HT58\n+MuXL/fs2bN8+XJDQ8Ompqbk5OTY2FgJCQlPT0/OncyYMaOpqYnzFb1///6ff/7p7Ox0c3Nz\ndXVVVFSsrKw8fvz4zZs3EUJcdyoGDIPBwJNT8VznGTNmcD76/PlzaWlpX1/fCRMmkEikqqqq\n3bt3l5WVXbly5ffff+/Xjl27dq28vJxrI07jaW9vxxM8APh0CPib7OjogDnZAIBPEATuX6DG\nxkYk3HAyr/z8/OrqaoQQzk1HCBkbG2toaFRVVaWmpnIG7k1NTQghTU1Nrj3U1tbyTZ728/Ob\nMGFCL7okGIVCSUlJqa+vp9PpSkpKv/zyC+csWDKZvGvXrhMnTty9e9fHx4fYPnToUF9fX65S\nOe7u7lw7v3DhQmdnp7Oz8w8//IC3aGlp7dq1a926dVQqVeSvpSsXLly4cOEC34dsbGxmz57N\nuYXNZq9cuRJPTkAIaWhoLFu2bP/+/Vx5Qf0hMTExOzubayP+C2EwGG1tbf3dAQB6RMDfJIvF\ngsAdAPAJgsmpXyCc1yElJdWL5+LhdjMzM86IHIfCjx494izWjvdfX1/fx972UW1tLZVK7ejo\nIJFIbDa7oaGB6+s2Pz//+fPnCCEymaylpSUvL48Q+vDhQ2pqKpPJFLBnNpv98OFDhNDChQs5\nt5PJ5Llz54r+lfSErKyshYXFzz//vG3bNq6ZA9LS0tOmTePcgmflQp46AAAA8LmDEfcvEF5W\nqRdDwiwWKy0tDXHkyWATJkyIjo6m0+lZWVnESLySkhKNRqusrOTayZAhQ6Kjozm37N69Gyd1\nYAEBAfieACYtLe3v79/TrhI2bNiAJ/tWVVUdP378/PnzNBpt3bp1+NHc3Fw8/3LVqlWzZ8/G\nueCFhYXHjh2LjY2tqakRkD1Co9GYTKa0tDTvXQUjI6Ned7gXuOq4o4/TUvnS1tYmk8lCNhYt\nT09P3uqi7e3tQUFB0tLS+JKJS3Nz84B0DQA++P5NYhISEgNQ/uurQqfT/fz8oqKi6urqrl+/\nzjW+0B+qq6vDwsLi4uKKiopqampUVVWNjY3nzJmzfPnyblNJg4ODf/rpJ4SQoaHh27dveRsU\nFBTgys7Lly8PCwvju5Pz588vX77c1tY2IyOD2Dhy5Eg8lsRJXl5eT0/Pyclp3bp1wk9GAl8n\nCNy/QDo6Onl5ebwhNZfDhw+3tLRMnz6dyC3Jzc3FCTBnzpw5c+YM71Pu3btHBO4mJiZFRUW4\n/gwXrkiRK++5oKCgpqaG+FVOTq77lyQEDQ2N3377zcvLKyEhwd3dXU1NDSEUGhrKYrE8PDxc\nXFyIlmZmZv7+/uvWrcvIyHjx4oWFhQXfHVZVVSGEVFRUeB/iu7H/cNVxF0zAarj9jfjb4FRe\nXh4UFCQhIcF3SgAE7mAQCZimQiaTIXAXrb179/79998IIVNTU64yX/3h6NGj27Ztw99oGJVK\npVKp6enp+/bt8/f39/X1FfD0S5cu4R/evXuXmZlpa2vbVctz584tW7asj0uCNDc3v3z58uXL\nlydOnAgMDPzll1942zQ1NR04cODKlSulpaUKCgp2dnY+Pj64KHBXSkpKjIyMDhw4IPjFgs8L\npMp8gfDyOkVFRTQaras2FRUVd+/ezcrK4oxBcfl2cXFxRR74Gy43N5eItEaNGoUQKiws5B08\n4NTY2MjVjdDQ0OscKBRKj15dXV1dcHDwqVOneB9SUlJSVVVls9l44iydTseJ3bxrP6mpqeGK\nkK9everqQPid4ZsLhMvXAAAAEFJiYiJC6NSpU4WFhXwv8kXo559/3rhxY1NTk6Wl5b///vvu\n3buWlpbS0lIKhTJmzJiWlhY/Pz8BC1mUlJTgUlr4W6DbL6k1a9Zw5pEKIygoiPkRnU7Py8sL\nCwsbMWIEg8HYuHEj78BZTU2NtbX1gQMHCgsLWSxWdXX19evXHR0duxrsx86cOQNTNb48ELh/\ngWxsbBQVFTs6OogxA144l11GRsbU1BRvodPpOKV70aJFF3js3bsXIcRkMolbfjY2Nurq6gih\n8+fPC+iM4Ed7QUFB4fbt2zdv3uS9LGGxWDikxhUqJSQk8LgO31x2vJGzRiQXLS0tMTGxtrY2\n3gMNwERPAAD4kuBBkP4oUcDlzJkzx44dQwitXbs2Kytr6dKlBgYGsrKyenp6ixcvzsjI2Lx5\nM4lEOnfu3OHDh/nugUKhsNlsIyOjTZs2IYQiIiJYLFZXh9PW1n79+nVXa2J0hcxBSkrq22+/\nXb58+cOHD/HigJs3b37//j1ne29v78LCQk1NzaSkpObm5uLiYi8vr87OzjVr1uA6clza29tP\nnz598ODBHvUKfBYgcP8CSUpK4qWXUlJSoqOjeS+4S0tL8Tr206dPJ/KhMzMz29vbSSTSlClT\nePdpZmamra2NPkb8CCEymYxHLAoKCvbv38+3sFpCQsLt27dF9sIQQgiJi4sbGhoihJKSkrge\nun//PpPJlJCQwCUdyWQybkn0mVBWVlZcXIwQIq5b+B4I37u4du0a53Y2m33jxo0+vw4AAPjq\n9HcCUmNj49atWxFCHh4ewcHBvNlQkpKSQUFBq1evRgj5+/tz5m0S8JiXh4cHLjVGpVLx8iZ8\n4YuEwMBA3vXFe0peXj40NFReXr65ufn06dPE9vz8fPw1FBsbO3XqVElJSSMjo9DQ0IkTJ3Z0\ndPz111+cO4mMjJwwYYKamtqaNWsEXG+AzxcE7l+muXPn2trastns0NDQXbt2paenU6lUOp1e\nUlISFRW1ZcsWBoOhp6dHVDlEH/NkLC0t8Tg6r8mTJyOEnj59WldXh7fY29vPnz8fIZSZmfnr\nr7/euXOHRqN1dHRQqdTMzEx/f/8TJ05oamrilVNFCK8NFBUVFRMT09raihBqa2tLTEw8efIk\nQmjx4sXEgkSLFi0ikUgxMTHh4eF4vIfBYDx69Gj37t1MJtPa2trY2JjY7dWrV8PDw1NSUogt\ny5cvJ5FIcXFxFAoFX5nU1NTs27evoqJCTEwMUmABAKBbv/76K4lEwgPDlpaWJBIJL4XRH0JD\nQ+vr62VkZAICAgQ027t3r5KSUmtra3BwMNdDz549wyG4h4fHiBEj8AxUAdkyrq6uCxYs6Ojo\n8Pb27vvCfCoqKt7e3gihq1evEhvxz2PGjOFcbZ1MJq9cuRIhFBMTwzk8V1xc/PbtWzk5OU1N\nzd4VlwOfOJic+mUikUi+vr5nzpyJj4/Pzc3lLOqCmZubb9y4kZgh1NDQ8OTJE4QQ3+F2bPLk\nyZcuXWKz2WlpafPmzcMbV65cqaysTKFQqFTq0aNHuZ4ycuTILVu25OTk8HagL+zt7adOnZqc\nnHz27NmzZ8/Kysri8B0/RCz5iRAaN27csmXLLl68GBUVFRUVJSMjQ6fT8WecsbHxpk2bOIPv\nuLg4Go1ma2tLvAkmJiYrVqw4d+4chUK5fPmyvLx8U1MTmUzevHnzP//8A5+JAADQLSsrK3d3\n9/j4+JaWlhkzZigpKfGW6hKVuLg4hNDixYv19PQENFNTU/vhhx+OHj0aHx/PVdbs4sWLCCEL\nCwt8x3XhwoX79++/cuXKiRMnuppTe+zYsaSkpMzMzODg4PXr1/fxJTg7Ox8+fPjNmzfEFnwh\nMWfOHK6WuDAxlUotKysjlg709fUlpqJ+9913+O46+JJA4P7FIpPJa9eudXJySkpKevLkSW1t\nLYvF0tLS0tbWtrOz41p8ND09ncViSUlJEQv38NLU1DQ3N3/16lVqaioRuCOE3NzcHBwcbt26\nlZ2dTaPR2tvblZWVzczMHB0d8Uz80aNHu7i49G4l165s2rRp4sSJN27cqKio+PDhg66urp6e\nnrOz8+jRo7lauru729raXrly5e3btxUVFWpqarq6uhMnTpw2bZowQ+YLFiwwNja+ceNGQUEB\nnU4fO3asp6enrq4unU5XVVUV4SsCAIAv0vLly5cvX25qavr69esjR450VchLJB49eoQQGjly\nZLctcTeysrLYbDbxXcBms/Hg+pIlS/AWNze3/fv319fXJyQkcFYn46Sjo3PgwIENGzb8/vvv\n8+fP19XV7ctLwJccbW1tVCoVX+HgGnG865QPHTpURkamra2tsLBwUFYxB4MCAvcBxbcWSr8y\nMjJatWpVt81mz57NtQAnX4cOHeK7XV1dfenSpUuXLuX76JAhQ4TpA5duy1dZW1tz3jcUQFdX\nd+PGjd02CwkJiY2Nzc/P59o+cuRIrq8BXARTtJcifAUGBgrf2Nra+vr167zbdXR0uLZztXRw\ncBBQ5AHqiAEAPgvt7e24PgHObxFs+PDhCCEmk1lXV0eMwqSlpeHaA4sXL8ZbxowZY2Rk9Pbt\n28uXL3cVuCOE1q1bd+HChczMzA0bNnCtZNJTxL2C4uJizsCd71DRkCFDysvLifxV8DWAHHcA\n/n91dXW4ADwWEBCwatWqrKwsrma4rpkw3w0AAAAGBrE+tIByYQRi3ipn2TE8LXX8+PHDhg0j\nNuLFs69fv863BgMmJiZ2+vRpCQmJmJgYrnoGPUUM/xMlJvEMWkVFRd7GSkpKCCEiWRR8DSBw\nB+D/k5KSkpKSwrmOxrBhw2g02rlz56qrq4mNd+7cSUxMJJPJXOvLAgAAGETy8vI4HC8oKOi2\n8cuXLxFCJBJpyJAheEtHR0dkZCRCyMPDg7MlnjfV0tISGxsrYId4ThdCaMOGDZyrg/cUXocE\nfaxrjBDC86n4LleHD0S8BPA1gFSZQRMWFtbH63K+QkJCuioLAwRraGj4888/cQVJzNXVNTc3\n98WLF97e3sbGxlJSUvimpLi4uLe3N76JCecRAAA+ERYWFjk5OTgoFwxnRVpYWBA1kRMTE2tr\naxFCmzZtwhXcuVAoFCKFhq/t27dHRka+fv36999/P378eG9ewMfAnUQiEaP+ysrKVVVVuG9c\nPnz4gBDCxZrBVwIC90EzZ84cAasoc6qqqlJVVZWQkBCmMedKqKBH8MoXnMTFxffs2XPnzp3k\n5GRcT1NDQ2PEiBHff/+9kZERbgPnEQAAPhFOTk45OTkRERG7d+9WUFDgevTNmzd6enqSkpIs\nFgvXWJw1axbxKM6TkZSU5E1KaWtra2lpSUhIqKurE/DhLC0tffLkyenTpwcHB3t6evbuJeBU\nTDMzM2LE3dTUtKCggGtJJoRQY2Mjzt7R0dHp3bHA5wgC90Gjrq4u5JAqnkMDBoW4uLizs7Oz\ns3NXDeA8AgDAJ2LNmjV//fVXRUXFnj17uKoplJeXOzg46OvrR0VFUSiUoqIiKSmpX375BT/a\n0tISExODEPL399+2bRvXbh89ejR+/HgGg3H16lVcPb0r06ZN++GHH86fP+/t7S1MUQQu9fX1\nuIgF59C+jY3NjRs3EhMTt2/fztk4ISEBITR06FC4PftVgRx3AAAAAHwJ9PX1f/vtN4TQkSNH\nAgMDOVdEUldXnzt37sOHD0eNGvX7778jhPz9/YnSjdHR0a2trSQSiW95NBsbG7zMtoCVmAhB\nQUFDhw598eLF//73vx51vqWlZeXKlU1NTQoKCngZJszV1RUhlJmZiRexIuCS866urrAa4FcF\nAncAAAAAfCH27Nkzb948JpO5devWyZMnh4eHP336lE6n19TUzJw5U0dHB69qYm5ujsN3DAfB\njo6OnNOcOOGAPiUlhUqlCu7AkCFDjhw5gj6m0XeLxWK9fPny4sWLtra2OIHn8OHDnGnrlpaW\nCxYs6OzsdHFxwRnw7e3tO3bsuH79uqSkJFTs/dpA4A4AAACALwSZTL527Zqfn5+kpOT9+/d/\n+OEHKysrGRkZHR2d77777v3791JSUioqKq9evfLw8MCFFKurq2/fvo0Q8vLy6mq3OGe9s7Mz\nIiKi2z4sXbp0xowZAhrgdbsxcXHxESNGLF269Pnz55KSkkePHuVd+eT06dMmJiYvX77U19c3\nMzNTUVHZvXs3mUwOCQkxNjYW5m0BXwwI3AEAAADw5SCTyQcOHCgsLNyxY4etra26urqkpKS+\nvr6Dg8O+ffvKyspyc3OHDx8eERGxYsUKhFBkZCSTyZSVlXV3d+9qn8bGxnZ2dki4bBmE0MmT\nJ2VkZITssJycnLm5+caNG58/f/7zzw3JUk8AACAASURBVD/zNlBTU8vJyfH19TU1NS0tLZWT\nk5s/f35aWtqyZcuEPAT4YpDYbPZg9wEA8MUqLy9fsGDB/v37Z86cyftoTU2NQuDuge8VAAgh\nqYCjXT106NChlJSU+Pj4gexPt6gb+FTyHmCax+QHuwuiUV9f7+npGRAQYGlpOdh9AaAHoKoM\nAGAwNfls51wDRSSYTGZ9fb2ioqKkpKQId1tfX48XWVRRUSFqP4tETU2NrKysMMs9Cq+5uRmv\nvKigoIAXcBGV+vp6MTExvus49hqdTsfry8jIyMjJyYlwz83NzQwGg+9y8eBrpqysfPPmzcHu\nBQA9BqkyAAAAAAAAfAYgcAcAAADAwPHx8SH1g5KSksF+ZQD0O0iVAQAAAMDAWb9+Pe9K1Xy9\ne/dOS0tLyFwvTU3NvvULgM8ABO4AgMGEJ6e2i3y3CLERQl3PPgQADBZDQ8OuyqVzsbe37+e+\nAPCZgVQZAAAAAAAAPgMQuAMAAAAAAPAZgMAdAAAAAACAzwAE7gAAAAAAAHwGIHAHAAAAAADg\nMwCBOwAAAAAAAJ8BCNwBAAAAAAD4DEAddwAAAOAzoHlMfrC7AAAYZDDiDgAAAAAAwGcAAncA\nAAAAAAA+A5Aq0wMMBuP8+fMPHjxobm7etm2blZVVfx+RRqMlJSVlZWVVV1fT6XRNTU1tbe3p\n06ePGzdO8BPj4+ODg4MRQurq6iEhIYIb19fXJycnZ2dnV1ZWNjY2qqioaGlpmZiYuLi4qKio\niOzFfH18fHwKCgq4NoqJiWloaGhrazs7O48fP753e3Z1dWWxWNu3b7e2tubbIDw8PCoqysrK\nas+ePb07BAAAAAA+NRC490BERERsbCxCSFtbW1y839+66OjoCxcuMBgMYktJSUlJSUlGRsbI\nkSP9/PwUFBS6eu69e/fwDzQaraCg4JtvvhFwlEuXLtHpdGILjUaj0Wh5eXmxsbEuLi7Lli0j\nkUg97TyNRlu9erWXl5ebmxvfBsXFxZGRkaWlpTQaTVVVVUdHx8XFZdSoUbwtGxoaIiIi8vPz\n379/r6ioqK+vP2/evNGjR/e0S5+Izs7OysrKysrKnJwcOzs7Pz+/Xry9fSfgBP344481NTUC\nnjtu3Dh/f//+7B0AAAAA+IDAvQdyc3MRQuvXr3dycurvY+ERU4SQtbW1q6uroaEhQohGo927\ndy8+Pv7Zs2eHDx/evn0735iPRqO9fPkSISQvL9/c3JyamtpV4B4cHBwfH48QMjAwmDNnjpGR\nkZ6eXmNjY2lpaVxcXG5ublRUVHt7++rVq3va/1u3brHZ7K4evXHjRkhISGdnJ+4klUqtrKzM\nzs6ePXv22rVrOVu+evVq586dra2tCCFJScnq6moajZadne3k5LR+/fqe9mpQLFmy5Pvvvyd+\npdPp5eXlV65cycjIyMjISEhImDVr1sD3SvAJEkxMDFLsAAAAgEEAgXsPtLS0IITMzc37+0A4\nYkYILVq0aMmSJUR0rqCgYGxsPHbs2J07d+bk5Ny+fXvmzJm8T09NTWWz2RoaGtOmTbt06VJa\nWtrKlSt5g62EhAQctbu5uXl6ehL3EGRlZTU1NW1sbOLj40+ePBkbG2tpaWlnZydk5zs6OpKT\nk3H/+aJSqefOnevs7HR3d3d3d5eVlW1vb4+NjQ0PD4+LixszZoyNjQ1u2draeujQodbWVmtr\n65UrV+ro6LS3tycnJ58+fToxMXH48OFTp079f+3deVyU1ds/8GsYdgQEZRUUBBEVc0ENxIUE\nEzcUocws0jSXUrOkNEtzyyf1sb6pWE+iKRn4EywMA9xAUcQVt0RBrdiHAWSHmWGW3x/n+c5r\nnpkBWQaY8ft5/wXnPnPu69y3xXWfOefcrYyqG+np6SlefFNTUw8Pj3Xr1m3cuPHu3btnzpzp\n4sT9uTfo66+/Zs9USiQSyfbt24uLi1977bXODBAAAADUw8hZm3XBxIaDBw8Ska+v7/z581VP\nN2zYsMDAQFKYD6OElU+YMMHPz4+IKisr79+/r1RHIBD8/PPPROTv7//OO++onfkzderU8ePH\nExHL758rIyPj008/feuttyIjI9VmfkxKSopIJBo5cmR4eLipqSkRGRkZhYWFTZgwgYiuXLki\nr5mdnV1eXm5pablu3bo+ffqwmlOnTp09ezYRXbp0qTVRaScOhzNu3DgiKigoaPfId1u18gbZ\n2traq5OZmVlQUPD22297eHh0TcAAAACgCCPurRIVFfX777+zn1esWEFELawL7KDs7OyCggIi\nmjt3bnN1AgMD//nnH7FYLBAIjI2NFQ+xefBENGHCBGdnZycnp8LCwvT0dKWltBcuXKitrTU1\nNX333XdbCCY8PHzkyJFEJJPJnvvEwuPxSktLjY2NjY2N6+rqmpqa1FbLz88nItX1tUOGDElP\nT2dHFWs6OTkZGhoq1nR3d5cflVu6dGlJSYmPj8/69esVy588eZKYmJiTk/Ps2bO+ffvOmjVr\n/Pjx8+bNk0gkx48fb7lHnYoNw0skEolEIn9wEgqFJ06cyM7O/vvvv7lcbt++fYOCgvz8/DTy\nuNjKG6TWkydPfvnll+HDh4eEhHQ8EgAAAGgHJO6t4urq6ufnd+vWLYFAMHz4cDMzs549e3bS\nue7evUtE7u7u/fv3b67OwIEDd+3apfbQhQsXiKhv3779+vUjIl9f37i4uCtXrixfvlxxWP32\n7dtE9NJLL7XcEVtb29ZPRwkNDZWvdNyxY0dGRobaav7+/t7e3t7e3krl1dXVRGRhYSEvsbOz\nI6L8/PyGhgY2Ns+wGfzsaMvOnz+/f/9+lqGamJjk5ubu2rUrOzu7hfHmLsOuT79+/eT3JS8v\n7+uvvy4qKiIiExOT+vr6e/fu3bt3z9fXd82aNUpPL+3QyhukSiwW796928DAYPXq1d2ylBYA\nAAAIiXsrBQQEBAQEsDHdxYsX9+3bt/PO9eTJEyJycHBox2dlMll6ejoRTZw4kZWMHTs2Li6u\nvr4+KytLPneciNigfqd2pDlsSowiqVR6+/btkydPEpHiwl8fH59BgwY9fPhw69atCxYscHFx\nqa2tTU1NTUxMNDAwmD9/vmIjkydPrq2tVexRUVHR3r17pVJpaGhoSEiIhYVFSUlJZGTkH3/8\nQURK31R0GZFIxBansrXOkydPZuUSiWTnzp1FRUXe3t6LFi1ycnISiUSZmZlRUVGZmZk///zz\nokWLuiVgIvrjjz+KiormzZtnbW3dQrWPP/74zp07SoW9e/cmorq6uoqKik4MsRmddNKqqiqN\nt9nQ0NDY2KjxZomotra2rq5Ogw2yyV2ddG0bGxsVN7nquPZFKxAItOHxHgBACRJ3rVNTU0Ot\nG05WlZ2dXVZWRkRsbjoRubm52dnZlZaWpqenKybutbW1RGRvb6/UQkVFxcKFC1VbXrdu3dix\nY9sRUstiY2PT0tKqqqoEAoGlpeWqVasUV8FyudzNmzfv37//woULn3zyiby8d+/ea9euVdoq\nJywsTKnxo0ePSqXSoKCgd955h5U4ODhs3rx5+fLlPB5P431pztGjR48ePar20JgxY6ZNm8Z+\nTklJKSgoGDBgwBdffMHlconI0NBw4sSJtra269atS0xMDAsLs7S0VPz4li1bOjt4Iqqvrz9+\n/HjPnj2fO0mmoaGB/etVxL4qkclkXTaVX1EnnVS3mu2klhEtAEDXw+JUrcPmdRgZGbXjs2y4\n3cPDQzEjZ6nw9evXFcexWPudMXDYJhUVFTwer6mpicPhyGSy6upqpb+v2dnZf/75JxFxuVwH\nB4cePXoQ0bNnz9LT08VicQsty2Sya9euEdGcOXMUy7lc7owZMzTfk7YwNTUdMmTIypUrv/ji\nC/nMkxs3bhDRrFmzWNYuN2jQIDc3N6lUeu/evW6Ilej48eO1tbXz5s3rru8oAAAAgMGIu9Zh\nr1Vqx5CwRCK5fPkyKcyTYcaOHZuQkCAQCG7cuCEfibe0tOTz+SUlJUqN9OrVKyEhQbFky5Yt\nbFIHs2PHDsVRVWNj4468i2fFihVssW9paWlkZOSRI0f4fP7y5cvZ0aysLPbiz8WLF0+bNo3N\nBc/Nzd23b19iYmJ5eflnn33WXMt8Pl8sFhsbG6t+q+Dq6trugNtBaR93amYf9MLCQiJKTU1l\nzxuK2PNVaWmpUvlz35za7pjlamtr//jjD3Nzc7aRUct++OEH1cLCwsLZs2ebm5uzOTNKWn7T\nU8epPWm7VVVVscdFKysrpeerDiovLzc1NVVcyNFxdXV17Fnd3Ny8fQMBzamqqtLT01NcjtJx\nAoGAzecxMTExMzPTYMt1dXUikajlWV6qjI2N8b4CANBCSNy1Tp8+fe7evauaUiv55ptv6uvr\nAwMD5XNLsrKy2ASYAwcOHDhwQPUjFy9elCfu7u7ujx8/ZvvPKFH6c8USSrmcnBzFZEtTf2Lt\n7OzWrFkTHh6ekpISFhZmY2NDRIcOHZJIJPPmzQsODpbX9PDw2LBhw/LlyzMzMx88eDBkyBC1\nDbI018rKSvWQ2sLOo7SPu1oymYxdVcVnJCWanfjbSmfPnhWJRFOmTDEwMOj6swMAAIAiJO5a\nx8vLKykp6fHjx3w+39bWVm2d4uJitnuM4qtw2Pbt+vr6quN2IpFIIBBkZWXV1dWx2SbDhw9P\nTk7Ozc39888/vby8mgumpqaGz+crlhw6dKidHSMiosrKymPHjunp6S1dulTpkKWlpbW1dUVF\nRUFBgY2NjUAgYBs+qr77ycbGxt3dPTs7+9GjR80l7iw7VzsXiG1fo1U4HI61tXV5efnevXvZ\ndkDaQCaTpaSkEJHa93wBAABAF0PirnXGjBljYWFRU1MTExOzevVqtXXYXHYTE5MBAwawEoFA\nwKZYzJ07V3UD+Nzc3IiICLFYnJmZybYxGTNmjK2tLZ/PP3LkSHM7SxLRkSNHNNIpOXNz87Nn\nz4rF4pCQEKXHEolEwlJqtkOlgYGBoaGhSCRSO5edFbYwtcDBwUFPT6+xsVH1+UdpA3gt0adP\nn/Ly8n/++Uc1cS8rK5NIJNbW1h3fEbJNsrKyeDzegAEDtOdZAgAA4D8Z5vBpHUNDQ5Z5p6Wl\nJSQkqG6GkJ+fz6YvBwYGyifaXr16VSgUcjicV155RbVNDw8PR0dH+nfGT0RcLpftHpOTk7N9\n+/b6+nrVT6WkpJw9e1ZjHSMiIn19fRcXFyI6f/680qFLly6JxWIDAwO2pSOXy2U15THLFRQU\n/PXXX0Qkf25ReyL2TcJvv/2mWC6TyU6dOtXhfmjeiBEjiCghIUFpE7ri4uIlS5asXLmy5cW4\nnYF9h6O4GREAAAB0IyTu2mjGjBk+Pj4ymezQoUObN2/OyMjg8XgCgSAvLy8+Pj4iIkIkEjk7\nO8t3OaR/51heXl7Nza7x9/cnonv37lVWVrISPz+/WbNmEdHVq1c/+uijc+fO8fn8pqYmHo93\n9erVDRs27N+/397enr05VYPYO4Di4+NPnjzZ0NBARI2NjadPn2ZLG9944w35C4nmzp3L4XBO\nnjwZHR3NJr2IRKLr169v2bJFLBaPGjXKzc1N3uyvv/4aHR2dlpYmL1mwYAGHw0lKSoqNjWVP\nJuXl5V999VVxcbGenp62vUhoxowZtra2T58+3b59u3xpcm5u7tatWyUSyeTJkzW7cvG5ZDIZ\ne0vX4MGDu/K8AAAA0BxMldFGHA5n7dq1Bw4cSE5OzsrKUl2w6Onp+eGHH8onTlRXV7MX36gd\nbmf8/f1jYmJkMtnly5dnzpzJChctWtSzZ8/Y2Fgej7dnzx6ljwwdOjQiIuLWrVstrJhsBz8/\nv0mTJqWmph48ePDgwYOmpqYsfWeH5K/2JKLRo0e//fbbv/zyS3x8fHx8vImJiUAgYF9BuLm5\nKb3FMykpic/n+/j4yC+Cu7v7woULDx8+HBsbe+zYsR49etTW1nK53I8//njv3r2a3Wej4wwN\nDSMiInbu3Hn9+vXr16+bmZnJZDJ2ZUaOHNn1b196/PhxdXU1l8tV2i8fAAAAugsSdy3F5XKX\nLVs2ZcqU8+fP37lzp6KiQiKRODg4ODo6+vr6Kr18NCMjQyKRGBkZ+fn5Ndegvb29p6fno0eP\n0tPT5Yk7EYWGho4fP/7MmTM3b97k8/lCobBnz54eHh4TJ0708fEhohEjRgQHB7fvTa7NWb16\n9bhx406dOlVcXPzs2TMnJydnZ+egoCA2XURRWFiYj4/PiRMn/v777+LiYhsbGycnp3HjxgUE\nBLRmyHz27Nlubm6nTp3KyckRCATe3t7z5893cnISCARt3R6uC3h6eu7Zs+f48eM5OTl5eXnG\nxsZubm5BQUHjxo3r+u8Hbt26RUQDBgzo4on1AAAA0BwO3icHL4zExMTs7Oy1a9e2XC0vL2/l\nypXe3t5ffvll1wT2n4zt4759+3a1W9OUl5eb7+rE978a7VD+HqkjsI8785+wj/vOnTvT0tKS\nk5M1GAkAQMdhjju8OCorK9kG8MyOHTsWL17M3kiq6PTp00Tk6enZpcEBAAAAdAwSd3hBpKWl\npaWlTZo0SV7Sv39/Pp9/+PDhsrIyeeG5c+dOnz7N5XKV3i8LAAAAoOUwx72dfvrpJ6V9BjUi\nKiqquW1hoGXV1dWff/4520GSCQkJycrKevDgwZIlS9zc3IyMjAoLCysrK/X19ZcsWWJvb0+4\njwAAAKA7kLi30/Tp09nazecqLS21trZu5Rvj2fs+oR1mz56tVKKvr79169Zz586lpqay/TTt\n7OwGDx78+uuvu7q6sjq4jwAAAKArkLi3k62tbSuHVAcNGtTZwUBz9PX1g4KCgoKCmquA+wgA\nAAC6AnPcAQAAAAB0ABJ3AAAAAAAdgMQdAAAAAEAHIHEHAAAAANABWJwKAN2p9pONHA6nV69e\nGmxTLBZXVVVp9tWeAAAA3Q4j7gAAAAAAOgCJOwAAAACADkDiDgAAAACgAzDHHQC6k/muLUQk\n7FgjRjv2aCQYAAAAbYYRdwAAAAAAHYDEHQAAAABAByBxBwAAAADQAUjcAQAAAAB0ABJ3AAAA\nAAAdgMQdAAAAAEAHIHEHAAAAANABSNwBAAAAAHQAEncAAAAAAB2AxB0AAAAAQAfod3cAbSYS\niY4cOXLlypW6urovvvhi2LBhnX1GPp9//vz5GzdulJWVCQQCe3t7R0fHwMDA0aNHt/zB5OTk\n77//nohsbW2joqJarlxVVZWamnrz5s2SkpKamhorKysHBwd3d/fg4GArKyuNdeY/GO4jAAAA\n6DTdS9yPHz+emJhIRI6Ojvr6nR5/QkLC0aNHRSKRvCQvLy8vLy8zM3Po0KHr1q0zNzdv7rMX\nL15kP/D5/JycnIEDB7ZwlpiYGIFAIC/h8/l8Pv/u3buJiYnBwcFvv/02h8Npa/B8Pv+9994L\nDw8PDQ1VW+Gvv/6Ki4vLz8/n8/nW1tZ9+vQJDg4ePny4as3q6urjx49nZ2cXFRVZWFj07dt3\n5syZI0aMaGtI3eXFvo+NjY1xcXGZmZllZWUmJiYDBw6cM2fO4MGDVWvW1NT8v//3/3JycoqK\nigwNDfv16+fj4xMUFKSn19KXb2Kx+NNPP33y5MmPP/5ob2/f1vgBAABAI3Qvcc/KyiKiDz74\nYMqUKZ19rujo6Pj4eCIaNWpUSEiIi4sLEfH5/IsXLyYnJ9+/f/+bb77ZuHGj2lSMz+c/fPiQ\niHr06FFXV5eent5cwvf9998nJycTUb9+/aZPn+7q6urs7FxTU5Ofn5+UlJSVlRUfHy8UCt97\n7722xn/mzBmZTNbc0VOnTkVFRUmlUhYkj8crKSm5efPmtGnTli1bpljz0aNHmzZtamhoICJD\nQ8OysjI+n3/z5s0pU6Z88MEHbY2q673Y97Gmpmbt2rVFRUVExOVyq6urr1+/fuPGjZUrVwYG\nBirWzM7O/uqrr2pra4nI2Ni4pqbmzp07d+7cOXPmzJYtWywsLJo7xeHDh588edLWsAEAAECz\ndC9xr6+vJyJPT8/OPhHLtIho7ty5b775pjyrMzc3d3Nz8/b23rRp061bt86ePfvqq6+qfjw9\nPV0mk9nZ2QUEBMTExFy+fHnRokWq45opKSks2wsNDZ0/f778OwRTU1N7e/sxY8YkJyf/8MMP\niYmJXl5evr6+rQy+qakpNTWVxa8Wj8c7fPiwVCoNCwsLCwszNTUVCoWJiYnR0dFJSUkjR44c\nM2YMq9nQ0LBz586GhoZRo0YtWrSoT58+QqEwNTX1xx9/PH369KBBgyZNmtTKqLrFi30fiSgy\nMrKoqMjKymrNmjWDBw+uqKiIjY1NTU2NjIwcMmSIg4MDqyYWi/ft21dbWztkyJD333/f2dlZ\nLBbfuHHjf/7nf/766689e/Z88cUXatu/fv3677//3sqAAQAAoPPo6uLUdsw3aKuDBw8Ska+v\n7/z581VPN2zYMDacKZ9HoYSVT5gwwc/Pj4gqKyvv37+vVEcgEPz8889E5O/v/84776id+TN1\n6tTx48cTEcsLnysjI+PTTz996623IiMj2Wi6WikpKSKRaOTIkeHh4aampkRkZGQUFhY2YcIE\nIrpy5Yq8ZnZ2dnl5uaWl5bp16/r06cNqTp06dfbs2UR06dKl1kTVjV7s+1hQUJCZmUlEGzZs\neOmll/T19e3s7FatWjV48GCJRPLbb7/Ja16+fLmwsNDMzGz9+vXOzs5EpK+v7+vru379eg6H\nc/369fz8fNX2y8vLv/vuO/YvBAAAALqXLiXuUVFRwcHBJSUlRLRixYrg4OCbN2920rmys7ML\nCgqIaO7cuc3VCQwMHDhwoFgsVpzTzLD500Q0YcIEZ2dnJycnIkpPT1eqduHChdraWlNT03ff\nfbeFYMLDw1evXu3v79/CfAk5Ho9XWlpqbGxsZWVlYGDQXDWWpamuyxwyZIj8qGJNJycnQ0ND\nxZru7u5KNYlo6dKlwcHB27dvV2r2yZMn33777bJly15//fWIiAiW7s+bN+/1119/bo864oW/\nj+wRy83Njd0ORk9Pb/LkyUR07do1+bn+/vtvIho7dqzSbH4PDw/Wr8ePHys1LpVKd+/eXVtb\nu2LFiucGDAAAAJ1Nl6bKuLq6+vn53bp1SyAQDB8+3MzMrGfPnp10rrt37xKRu7t7//79m6sz\ncODAXbt2qT104cIFIurbt2+/fv2IyNfXNy4u7sqVK8uXL1ccjr19+zYRvfTSSy13xNbWtvXT\nUUJDQ+VLGHfs2JGRkaG2mr+/v7e3t7e3t1J5dXU1ESlOd7azsyOi/Pz8hoYGxZFXNvObHW3Z\n+fPn9+/f39TUREQmJia5ubm7du3Kzs5uYSBZU174+8ieK0aNGqVUzh7JKisry8vLbWxsiKii\nosLCwmLAgAGqjZiYmBCR4spdJjY29sGDB1OnTh07dmwrwwYAAIDOo0uJe0BAQEBAwNKlS0tK\nShYvXty3b9/OOxdbiiefH9wmMpmMDcpOnDiRlYwdOzYuLq6+vj4rK0s+d5yI2GBwp3akOWxK\njCKpVHr79u2TJ08SkeLCXx8fn0GDBj18+HDr1q0LFixwcXGpra1NTU1NTEw0MDCYP3++YiOT\nJ0+ura1V7FFRUdHevXulUmloaGhISIiFhUVJSUlkZOQff/xBRMbGxp3Yyf+A+1hZWUlELDVX\nZGFhYWhoKBKJiouL2dGIiAi1LZSWlj59+pT+/RWK3L17944fP96vX79Fixa1MphLly6VlZUp\nFQqFQiJqampS/UJDg5QaZ8+EIpFIsw+H8q8vhEJhy/vwtIPa73w6QiKRsB+amppa8yVP67Gr\nqtlo2YM9dc51kMlkbW2TfUqDYQAAaIQuJe5dqaamhlo3nKwqOzubpS9sTjMRubm52dnZlZaW\npqenKyZ8bH8P1f31KioqFi5cqNryunXrOmPsMzY2Ni0traqqSiAQWFparlq1SnH1JJfL3bx5\n8/79+y9cuPDJJ5/Iy3v37r127VqlLVbCwsKUGj969KhUKg0KCnrnnXdYiYODw+bNm5cvX87j\n8TTeFyUv/H189uwZEandy9LCwqK8vLyurq6Fj9fU1Hz99dcSiWTYsGGKg/HV1dW7d+82NDT8\n9NNPDQ0NW5n+/vLLL6qz19hlEQgELUfSQWob77xHBbbDkmaJRCLVLz00ojOug1Qq7aQb2tTU\nJE/iNait0Wr8aQcAQCN0aY57V2J/OYyMjNrxWTZM6+HhoZjJsVT4+vXrin9EWftVVVUdjLaD\nKioqeDxeU1MTh8ORyWTV1dVKf7Gys7P//PNPIuJyuQ4ODj169CCiZ8+epaeni8XiFlqWyWTX\nrl0jojlz5iiWc7ncGTNmaL4nKl74+8ieTNQuHmWFbMBblUwmS0tLW7FixdOnTx0cHD7++GPF\nQ99++21lZeWSJUvYMlYAAADQBhhxV48NYbZjSFgikVy+fJkU5lcwY8eOTUhIEAgEN27ckI/g\nWlpa8vl8ttxWUa9evRISEhRLtmzZwjawZ3bs2MEyNsbY2HjDhg1tDVVuxYoVbPVhaWlpZGTk\nkSNH+Hz+8uXL2dGsrKytW7cS0eLFi6dNm8bmdufm5u7bty8xMbG8vPyzzz5rrmU+ny8Wi42N\njVVHo11dXdsdcOu98PeRrVttbGxUPcRGhdUOxj948CAqKorNkBk3btzSpUstLS3lR3/99des\nrKxx48axFa6tN2XKFLa4WZFMJouOjjY0NGQz6ZWojbwdlBqXSqVCodDQ0JDL5WqkfUYoFLIv\nH4yNjTW7sVVjY6O+vn4Lq5DboampiT1Xd8Z1oPY+DzdHIpGwLxw0fh3YjKm2zsrjcrldsHcZ\nAEBbIXFXr0+fPnfv3lVNxZR888039fX1gYGB8rklWVlZbOLEgQMHDhw4oPqRixcvyhM+d3f3\nx48fs/WFSpRm0BYWFir+mpOTU15eLv/VzMzs+V1qBTs7uzVr1oSHh6ekpISFhbG50YcOHZJI\nJPPmzQsODpbX9PDw2LBhw/LlvH7dSgAAIABJREFUyzMzMx88eKCarjGlpaVEZGVlpXpIbaHG\nvfD30czMrKqqioWqhM0NsLa2VixsaGg4cuRISkqKTCZzc3MLDw9Xev3tP//8c/ToUTs7u3bs\nJBMSEqJaWFhYGB0dbWRkpLZ3mkrclRoXi8VCodDY2FhpK6QOampqYom7iYmJZlPhxsZGQ0ND\nzW67WVdXxxJ3IyMjzSbZTU1Nenp6mvrfDiMQCFjibmBgoNmWZTKZSCRqa5sGBgZI3AFACyFx\nV8/LyyspKenx48d8Pt/W1lZtneLiYrbryGuvvSYvZNt+6+vrq/4NFolEAoEgKyurrq6OzTYZ\nPnx4cnJybm7un3/+6eXl1VwwNTU1fD5fseTQoUPt7BgREVVWVh47dkxPT2/p0qVKhywtLa2t\nrSsqKgoKCmxsbAQCAdvwUfWdQTY2Nu7u7tnZ2Y8ePWoucWfZudo5JGz7ms72Yt9HInJ0dCwq\nKqqoqFAqb2hoYJN5evXqJS+sqqr67LPP2NuaFi9ePG7cONXUpLi4WCKRlJaWvvHGG6qnW7Jk\nCRGNHj26I9/wAAAAQPsgcVdvzJgxFhYWNTU1MTExq1evVluHzYE2MTGRr+oTCARsSvfcuXNV\nNw7Pzc2NiIgQi8WZmZlsEsKYMWNsbW35fP6RI0ea25GQiI4cOaKRTsmZm5ufPXtWLBaHhIQo\npbMSiYSl1GxnQwMDA7Y5idq57KywhWFCBwcHPT29xsZG1bxZ7et+NO7Fvo9E5OHhcePGjays\nLKU8m03IsbCwkM+BEYlEGzduLCoqGjVqVERERHN3TV9fX3VsUiaTsYk3JiYmenp6nb0XEAAA\nAKiFxanqGRoasowtLS0tISFBdXuB/Px89iL6wMBA+ZfmV69eFQqFHA7nlVdeUW3Tw8PD0dGR\nFN7gw+Vy2a4jOTk527dvr6+vV/1USkrK2bNnNdYxIiLS19d3cXEhovPnzysdunTpklgsNjAw\nYJsbcrlcVlP1rUMFBQV//fUXEandGlx+IjYCrfgKTyKSyWSnTp3qcD+e78W+j0Tk4+PDzqs0\nHYh9h+Dr6ysfU09LS/vnn3/c3d0///zzFp61xowZE6siJiaGHf3uu+9iY2MVNxcCAACALoPE\nvVkzZszw8fGRyWSHDh3avHlzRkYGj8cTCAR5eXnx8fEREREikcjZ2Vm+yyH9e36Fl5dXc7My\n/P39iejevXts+20i8vPzmzVrFhFdvXr1o48+OnfuHJ/Pb2pq4vF4V69e3bBhw/79++3t7UeO\nHKnZ3rGX+8THx588eZINpjY2Np4+ffqHH34gojfeeEP+gqG5c+dyOJyTJ09GR0ezSS8ikej6\n9etbtmwRi8WjRo1yc3OTN/vrr79GR0enpaXJSxYsWMDhcJKSkmJjY1lGW15e/tVXXxUXF+vp\n6XXBLNIX+z7269eP9W7btm1sunxTU1NMTMz169f19fXlr3AiotOnTxPR+PHjKyoq+Op06j7r\nAAAA0HGYKtMsDoezdu3aAwcOJCcnZ2VlKW4Gwnh6en744Yfy1W/V1dV37twhIrXDtIy/v39M\nTIxMJrt8+fLMmTNZ4aJFi3r27BkbG8vj8fbs2aP0kaFDh0ZERNy6dUs1gI7w8/ObNGlSamrq\nwYMHDx48aGpqKt+a2s/PTzHhGz169Ntvv/3LL7/Ex8fHx8ebmJgIBAI2dO3m5rZ69WrF5Dsp\nKYnP5/v4+Mgvgru7+8KFCw8fPhwbG3vs2LEePXrU1tZyudyPP/547969ml0zp9aLfR+J6IMP\nPsjLyysoKHj33XcdHR3Ly8tFIpGent7KlSvlm/nIZDI2N+mnn3766aef1Lbz8ccfswcSAAAA\n0E5I3FvC5XKXLVs2ZcqU8+fP37lzp6KiQiKRODg4ODo6+vr6Kr18NCMjQyKRGBkZ+fn5Ndeg\nvb29p6fno0eP0tPT5QkfEYWGho4fP/7MmTM3b97k8/lCobBnz54eHh4TJ05kcyFGjBgRHBzc\nvjeANmf16tXjxo07depUcXHxs2fPnJycnJ2dg4KClLYZIaKwsDAfH58TJ078/fff7E2cTk5O\n48aNCwgIaM2Q+ezZs93c3E6dOpWTkyMQCLy9vefPn+/k5CQQCJT2POkkL/Z9tLS0/Pbbb+Pi\n4jIzM8vKykxMTEaMGBEWFqb4bqzKyspOerkPAAAAdBkOXg4HmpWYmJidnb127dqWq+Xl5a1c\nudLb2/vLL7/smsCgWxQWFs6ePXv79u2vvvqq6tHy8nLzXVs6fhajHf/nKw6xWFxVVWVhYaHZ\n7SCrqqrYgmwrKyvNbgdZXl5uamqq8e0g2fQnc3NzzX61VVVVpaenZ2FhocE25e/WNTEx0ex2\nkHV1dSKRqK1jBDt37kxLS0tOTtZgJAAAHYc57qBhlZWVbAN4ZseOHYsXL75x44ZSNTbl2tPT\ns0uDAwAAANBZSNxBk9LS0tLS0iZNmiQv6d+/P5/PP3z4cFlZmbzw3Llzp0+f5nK5Su8lBQAA\nAIDm6PYc959++klpn0GNiIqKam47EWhZdXX1559/znaQZEJCQrKysh48eLBkyRI3NzcjI6PC\nwsLKykp9ff0lS5aw1ZO4jwAAAADPpduJ+/Tp09mav+cqLS21trY2MDBoTWX2vk9oh9mzZyuV\n6Ovrb9269dy5c6mpqWwfRjs7u8GDB7/++uuurq6sDu4jAAAAwHPpduJua2vbyiHVQYMGdXYw\n0Bx9ff2goKCgoKDmKuA+AgAAADwX5rgDAAAAAOgAJO4AAAAAADoAiTsAAAAAgA5A4g4AAAAA\noAOQuAMAAAAA6ADd3lUGAHRd7ScbORxOr169ujsQAAAAbYcRdwAAAAAAHYDEHQAAAABAByBx\nBwAAAADQAZjjDgDdyXzXFiISarxZItnzmjXasUfTpwUAAOhEGHEHAAAAANABSNwBAAAAAHQA\nEncAAAAAAB2AxB0AAAAAQAcgcQcAAAAA0AFI3AEAAAAAdAASdwAAAAAAHYDEHQAAAABAByBx\nBwAAAADQAUjcAQAAAAB0gH53BwDaRSQSHTly5MqVK3V1dV988cWwYcM6+4x8Pv/8+fM3btwo\nKysTCAT29vaOjo6BgYGjR49u+YPJycnff/89Edna2kZFRbVcuaqqKjU19ebNmyUlJTU1NVZW\nVg4ODu7u7sHBwVZWVhrrDAAAAECnQeIO/8fx48cTExOJyNHRUV+/0/95JCQkHD16VCQSyUvy\n8vLy8vIyMzOHDh26bt06c3Pz5j578eJF9gOfz8/JyRk4cGALZ4mJiREIBPISPp/P5/Pv3r2b\nmJgYHBz89ttvczictgbP5/Pfe++98PDw0NBQtRX++uuvuLi4/Px8Pp9vbW3dp0+f4ODg4cOH\nq9asrq4+fvx4dnZ2UVGRhYVF3759Z86cOWLEiLaGBAAAAC8wJO7wf2RlZRHRBx98MGXKlM4+\nV3R0dHx8PBGNGjUqJCTExcWFiPh8/sWLF5OTk+/fv//NN99s3LhRbUrN5/MfPnxIRD169Kir\nq0tPT28ucf/++++Tk5OJqF+/ftOnT3d1dXV2dq6pqcnPz09KSsrKyoqPjxcKhe+9915b4z9z\n5oxMJmvu6KlTp6KioqRSKQuSx+OVlJTcvHlz2rRpy5YtU6z56NGjTZs2NTQ0EJGhoWFZWRmf\nz7958+aUKVM++OCDtkYFAAAALyok7vB/1NfXE5Gnp2dnn4hlzEQ0d+7cN998U56dm5ubu7m5\neXt7b9q06datW2fPnn311VdVP56eni6Tyezs7AICAmJiYi5fvrxo0SI9PeU1GykpKSxrDw0N\nnT9/vvw7BFNTU3t7+zFjxiQnJ//www+JiYleXl6+vr6tDL6pqSk1NZXFrxaPxzt8+LBUKg0L\nCwsLCzM1NRUKhYmJidHR0UlJSSNHjhwzZgyr2dDQsHPnzoaGhlGjRi1atKhPnz5CoTA1NfXH\nH388ffr0oEGDJk2a1MqoAAAA4MWGxamgRjvmjbTVwYMHicjX13f+/Pmqpxs2bFhgYCApzIdR\nwsonTJjg5+dHRJWVlffv31eqIxAIfv75ZyLy9/d/55131M78mTp16vjx44mI5ffPlZGR8emn\nn7711luRkZFsNF2tlJQUkUg0cuTI8PBwU1NTIjIyMgoLC5swYQIRXblyRV4zOzu7vLzc0tJy\n3bp1ffr0YTWnTp06e/ZsIrp06VJrogIAAID/BBhxh/8VFRX1+++/s59XrFhBRBs3bhw1alRn\nnCs7O7ugoICI5s6d21ydwMDAf/75RywWCwQCY2NjxUNsHjwRTZgwwdnZ2cnJqbCwMD09XWkp\n7YULF2pra01NTd99990WggkPDx85ciQRyWSy5z6x8Hi80tJSY2NjY2Pjurq6pqYmtdXy8/OJ\nSHV97ZAhQ9LT09lRxZpOTk6GhoaKNd3d3eVH5ZYuXVpSUuLj47N+/XrF8idPniQmJubk5Dx7\n9qxv376zZs0aP378vHnzJBLJ8ePHW+4RAAAA6Aok7vC/XF1d/fz8bt26JRAIhg8fbmZm1rNn\nz0461927d4nI3d29f//+zdUZOHDgrl271B66cOECEfXt27dfv35E5OvrGxcXd+XKleXLlysO\nq9++fZuIXnrppZY7Ymtr2/rpKKGhofKlqDt27MjIyFBbzd/f39vb29vbW6m8urqaiCwsLOQl\ndnZ2RJSfn9/Q0MDG5hk2g58dbdn58+f379/PHiFMTExyc3N37dqVnZ3dwhcCAAAAoIuQuMP/\nCggICAgIYGO6ixcv7tu3b+ed68mTJ0Tk4ODQjs/KZLL09HQimjhxIisZO3ZsXFxcfX19VlaW\nfO44EbFB/U7tSHPYlBhFUqn09u3bJ0+eJCLFhb8+Pj6DBg16+PDh1q1bFyxY4OLiUltbm5qa\nmpiYaGBgMH/+fMVGJk+eXFtbq9ijoqKivXv3SqXS0NDQkJAQCwuLkpKSyMjIP/74g4iUvqno\nbNu3b2fPG4rYvkANDQ1VVVVdGUxrtDUkiUTCfqipqdH4dDKBQKC4vVLHyZ/c6uvrGxsbNdiy\nRCKRSCSavaHydd5CobC5L7LaRyqVSqXStkYrFApbWHoOANBdkLhDN6ipqaHWDSerys7OLisr\nIyI2N52I3Nzc7OzsSktL09PTFRP32tpaIrK3t1dqoaKiYuHChaotr1u3buzYse0IqWWxsbFp\naWlVVVUCgcDS0nLVqlWKq2C5XO7mzZv3799/4cKFTz75RF7eu3fvtWvXKm2VExYWptT40aNH\npVJpUFDQO++8w0ocHBw2b968fPlyHo+n8b60LD8/XzVxZ9dfIpGIxeIujue52h2SPIPXIJZf\narzZzmu5k26olkQrk8mQuAOAFsLiVOgGbETNyMioHZ9lw+0eHh6KGTlLha9fv664WTtrv9sH\neisqKng8XlNTE4fDkclk1dXVSglBdnb2n3/+SURcLtfBwaFHjx5E9OzZs/T09JazDZlMdu3a\nNSKaM2eOYjmXy50xY4bmewIAAADdCiPu0A3Y9Il2DAlLJJLLly+TwjwZZuzYsQkJCQKB4MaN\nG/KReEtLSz6fX1JSotRIr169EhISFEu2bNnCNrBnduzYwb4TYIyNjTds2NDWUOVWrFjBFvuW\nlpZGRkYeOXKEz+cvX76cHc3Kytq6dSsRLV68eNq0aWyOfm5u7r59+xITE8vLyz/77LPmWubz\n+WKx2NjYWPVbBVdX13YH3G7btm0TCoVKhWVlZYsXLzYzM1P7htrKysouCU29tr40t6amho21\nW1hYcLlcDUZSWVlpbGxsYmKiwTYbGhrY7TAzM1Na+txBNTU1enp67AlTU4RCIXuVQWdcB5FI\n1NYVO0ZGRqrbywIAdDsk7tAN+vTpc/fuXdWUWsk333xTX18fGBgon1uSlZXFJsAcOHDgwIED\nqh+5ePGiPHF3d3d//Pgx239GidKf5MLCQsVfc3JyysvL5b+amZk9v0utYGdnt2bNmvDw8JSU\nlLCwMBsbGyI6dOiQRCKZN29ecHCwvKaHh8eGDRuWL1+emZn54MGDIUOGqG2wtLSUmsk+25qS\nakTv3r1VC9nXC3p6eprNdDWirSHJ57VzuVyNd0fjl0gebWe0zOFwNNum/D9Jjbfcvmi7YEtc\nAIB2QOIO3cDLyyspKenx48d8Pt/W1lZtneLiYrZ7zGuvvSYvZNu36+vrK27AwohEIoFAkJWV\nVVdXx8YChw8fnpycnJub++eff3p5eTUXTE1NDZ/PVyw5dOhQOztGRESVlZXHjh3T09NbunSp\n0iFLS0tra+uKioqCggIbGxuBQMA2fFR995ONjY27u3t2dvajR4+aS9xZdq52LhDbvgYAAABe\nJEjcoRuMGTPGwsKipqYmJiZm9erVauuwuewmJiYDBgxgJQKBgE3pnjt3ruoG8Lm5uREREWKx\nODMzc/Lkyewstra2fD7/yJEjze0sSURHjhzRSKfkzM3Nz549KxaLQ0JClB5LJBIJS6nZF/cG\nBgaGhoYikUjtXHZWqPqIIufg4KCnp9fY2Kj6/KO0ATwAAAC8ADCHD7qBoaEhy7zT0tISEhJU\nd2/Iz8+Pj48nosDAQPl33FevXhUKhRwO55VXXlFt08PDw9HRkf6d8RMRl8tlu8fk5ORs3769\nvr5e9VMpKSlnz57VWMeIiEhfX9/FxYWIzp8/r3To0qVLYrHYwMCAbenI5XJZTXnMcgUFBX/9\n9RcRyZ9b1J6IfZPw22+/KZbLZLJTp051uB8AAACgXZC4Q/eYMWOGj4+PTCY7dOjQ5s2bMzIy\neDyeQCDIy8uLj4+PiIgQiUTOzs7yXQ7p3/NkvLy8mptd4+/vT0T37t2TL3n08/ObNWsWEV29\nevWjjz46d+4cn89vamri8XhXr17dsGHD/v377e3t2ZtTNYi9pCk+Pv7kyZNsyV1jY+Pp06d/\n+OEHInrjjTfkL4qaO3cuh8M5efJkdHQ0m/QiEomuX7++ZcsWsVg8atQoNzc3ebO//vprdHR0\nWlqavGTBggUcDicpKSk2NpY9mZSXl3/11VfFxcV6enqYpwsAAPAiwVQZ6B4cDmft2rUHDhxI\nTk7OyspS3NSF8fT0/PDDD+W7YVRXV9+5c4eI1A63M/7+/jExMTKZ7PLlyzNnzmSFixYt6tmz\nZ2xsLI/H27Nnj9JHhg4dGhERcevWLdUAOsLPz2/SpEmpqakHDx48ePCgqakpS9/ZIfm7V4lo\n9OjRb7/99i+//BIfHx8fH29iYiIQCNhXEG5ubqtXr1ZMvpOSkvh8vo+Pj/wiuLu7L1y48PDh\nw7GxsceOHevRo0dtbS2Xy/3444/37t3bvg03AQAAQDshcYduw+Vyly1bNmXKlPPnz9+5c6ei\nokIikTg4ODg6Ovr6+iq9fDQjI0MikRgZGfn5+TXXoL29vaen56NHj9LT0+WJOxGFhoaOHz/+\nzJkzN2/e5PP5QqGwZ8+eHh4eEydO9PHxIaIRI0YEBwe3702uzVm9evW4ceNOnTpVXFz87Nkz\nJycnZ2fnoKCgESNGKNUMCwvz8fE5ceLE33//XVxcbGNj4+TkNG7cuICAgNYMmc+ePdvNze3U\nqVM5OTkCgcDb23v+/PlOTk4CgcDa2lqDPQIAAIDuxcHL4QB0SGJiYnZ29tq1a1uulpeXt3Ll\nSm9v7y+//LJrAmtOYWHh7Nmzt2/f/uqrr6oeLS8vN9+1peujYox2KH8D07Kqqiq2YtjKykqz\nWxaWl5ebmpq2sBC5Herq6tj7yMzNzTX73UtVVZWenp6FhYUG2xQIBHV1dURkYmKiqQ1Ymbq6\nOpFI1NaH2J07d6alpSUnJ2swEgCAjsMcdwBdUllZyTaAZ3bs2LF48eIbN24oVTt9+jQReXp6\ndmlwAAAA0JmQuAPojLS0tLS0tEmTJslL+vfvz+fzDx8+XFZWJi88d+7c6dOnuVyu0vtlAQAA\nQKdhjjs066efflLaZ1AjoqKimtsWBlpWXV39+eefsx0kmZCQkKysrAcPHixZssTNzc3IyKiw\nsLCyslJfX3/JkiX29vaE+wgAAPCiQOIOzZo+fTpbu/lcpaWl1tbWBgYGranM3vcJ7TB79myl\nEn19/a1bt547dy41NZXtp2lnZzd48ODXX3/d1dWV1cF9BAAAeDEgcYdm2dratnJIddCgQZ0d\nDDRHX18/KCgoKCiouQq4jwAAAC8GzHEHAAAAANABSNwBAAAAAHQAEncAAAAAAB2AxB0AAAAA\nQAdgcSoAdKfaTzZyOJxevXppsE2xWFxVVWVhYWFoaKjBZgEAALoXRtwBAAAAAHQAEncAAAAA\nAB2AxB0AAAAAQAcgcQcAAAAA0AFYnAoAnS42NjY1NVW1XCgUEhGHw9HsKlKZTCYSiQwMDPT0\nNDk20dTUJJVKicjQ0JDD4WiwZaFQqK+vz+VyNdimWCyWSCRE1BnXgTWrwTYlEolYLCYiLper\nr6/JP0xisVgqlbb1H1hOTo4GYwAA0BQk7gDQiQwNDQcNGiQWi4uKilSP1tfXExGHwzE1NdXg\nSWUymVQq1dPT02x63dDQIJPJiMjExESzqbBEIuFwOJptUygUslTYyMhIs6lwZ0Tb1NQkEomI\nSF9f38jISIMtS6VSmUzW1oeiHj16ODs7azAMAACN4LC/QwAAXaypqcnX15eIzM3N09LSujuc\n53vjjTeePHlCRAkJCU5OTt0dznNs3rw5MTGRiP7rv/5r8uTJ3R3Ocxw/fnznzp1EFB4evmrV\nqu4OBwBAS2GOOwAAAACADkDiDgAAAACgA5C4AwAAAADoACTuAAAAAAA6AIk7AAAAAIAOQOIO\nAAAAAKADsI87AHQPLpe7fv160vSrfDrPe++9V11dTURWVlbdHcvzzZgxY+jQoUQ0ePDg7o7l\n+UaNGsX+MXh4eHR3LAAA2gv7uAMAAAAA6ABMlQEAAAAA0AFI3AEAAAAAdAASdwAAAAAAHYDE\nHQAAAABAByBxBwAAAADQAUjcAQAAAAB0APZxB4Cu1tjYGBcXl5mZWVZWZmJiMnDgwDlz5mjD\nduNnz57du3ev2kMBAQEffvihYkl39YLP57/33nvh4eGhoaFqK7Q+sC7oQsvR6sQFBwDQHkjc\nAaBL1dTUrF27tqioiIi4XG51dfX169dv3LixcuXKwMDA7o2NRdUa3diLM2fOtPD+jdYH1jVd\naDlanbjgAADaA4k7AHSpyMjIoqIiKyurNWvWDB48uKKiIjY2NjU1NTIycsiQIQ4ODt0YG8sL\nFy9ePGbMGKVDJiYmir92Sy+amppSU1Pj4+NbqNP6wDq7C62JVssvOACAtkHiDgBdp6CgIDMz\nk4g2bNjg7u5ORHZ2dqtWreLxeNnZ2b/99tv777/fjeEVFxcT0dChQ+3t7Vuo1vW9yMjIOHny\nZF5eXmNjo0YC69QutDJa0uILDgCgnbA4FQC6zpUrV4jIzc2NpV+Mnp7e5MmTiejatWstTKvo\nbFKptKSkRE9Pz8nJqeWaXd8LHo9XWlpqbGxsZWVlYGDQ8cA6tQutjFabLzgAgHbCiDsAdJ28\nvDwiGjVqlFL56NGjiaiysrK8vNzGxqYbIiPi8/lisdjJySknJ+fUqVMFBQWGhoYuLi5jx45l\n4cl1fS9CQ0Plizt37NiRkZGhtlrrA+vULrQyWm2+4AAA2gmJOwB0ncrKSiJSzbEsLCwMDQ1F\nIlFxcXF3ZWBsvnVJScn69evlhU+fPj1//vz48eNXr14tHzzW2l60PjBt6MILcMEBALoYEncA\n6DrPnj0jInNzc9VDFhYW5eXldXV1XR7U/2J5pEQiefXVVwMDA/v161dWVnb+/PmEhIRLly45\nOjrOnz+f1dTaXrQ+MG3owgtwwQEAuhgSdwDoOjU1NURkamqqeogVCoXCro7p3ywsLPz9/T08\nPGbMmMFK+vbtu3DhQktLy8OHD584cWLq1KnW1takxb1ofWDa0IUX4IIDAHQxJO4A0HXY5Ae1\nm400NDRQM6OqXcPf39/f31+1fNasWSdOnKitrX369CnLI7W2F60PTBu68AJccACALoZdZQCg\n65iZmRFRbW2t6iE224ElalqFy+WybU/y8/NZidb2ovWBaW0XSKcuOABAF0PiDgBdx9HRkYgq\nKiqUyhsaGgQCARH16tWrG8J6HkNDQ1KYqqG1vWh9YFrbBUZXLjgAQBdD4g4AXcfDw4OIsrKy\nlMpZiYWFhaWlZTeERSQSiVasWLF06dKSkhLVo4WFhUTUt29f9qvW9qL1gXV7F16MCw4A0MWQ\nuANA1/Hx8SGinJwcpXTtwoULROTr68vhcLolMENDQ3t7+5KSkoSEBKVDly5dqqioMDc379+/\nPyvR2l60PrBu78KLccEBALoYEncA6Dr9+vXz8fGRyWTbtm0rLy8noqamppiYmOvXr+vr68vf\n2tMtZs6cyeFwkpOTf/75ZzZzur6+Pikpad++fUS0cOFCExMTLe9F6wPThi68ABccAKCLYVcZ\nAOhSH3zwQV5eXkFBwbvvvuvo6FheXi4SifT09FauXGlvb9+NgQ0bNuztt9+Ojo6Oi4uLi4sz\nNTVlO5ZwOJxZs2YFBAQoVtbaXrQ+sG7vwotxwQEAuhJ306ZN3R0DAPwHMTY2DggIkMlk1dXV\nZWVlJiYmw4cP//DDD1VfaN/1Bg8ePGLEiJqaGolEUl9f7+joOGzYsOXLl0+ZMkVpMkY39iIj\nI6OgoGD48OGDBw9WPdr6wLqmCy1HqxMXHABAe3BkMll3xwAAAAAAAM+BOe4AAAAAADoAiTsA\nAAAAgA5A4g4AAAAAoAOQuAMAAAAA6AAk7gAAAAAAOgCJOwAAAACADkDiDgAAAACgA5C4AwAA\nAADoACTuAAAAAAA6AIk7AAAAAIAOQOIOAAAAAKADkLgDAAAAAOgAJO4AAAAAADoAiTsAAAAA\ngA5A4g4A3SkkJITD4XA4nClTpjy3srm5OYfDmTx5chcEpin19fXr1q0bOHCgiYmJr69vd4cD\nAAA6DIk7AGiFM2fOHDl2KBQFAAANIElEQVRypLuj0Ly33nprx44dubm5AoGAx+N1dzgAAKDD\nkLgDgLb46KOPSktLuzsKTfrrr78SEhKIaOjQoYcPHz527Fh3RwQAADoMiTsAaIvKysqVK1d2\ndxSa9OjRI/bDpk2b3nnnnZdffrl74wEAAJ2GxB0AtIKPjw8RxcXFsSHqF0NTUxP7oU+fPt0b\nCQAAvACQuAOAVjh48KChoSERvf/++1VVVd0dDgAAgNZB4g4AWmHw4MGff/45EZWUlHzyySdt\n/XhDQ8PWrVtffvnl3r17GxkZOTk5BQcH//zzzxKJRLNxtvJEu3fv5nA4s2fPZr/6+PhwOBxX\nV9fWn+jBgwdLly51c3MzMTFxcHCYOXPmmTNniCgyMpLD4VhZWal+RCqVRkdHT58+vU+fPsbG\nxv379586derx48fVXgRXV1cOh/PVV18RUVFRUUREhIeHh6mpqZOT06RJk2JiYmQyWUeuADNo\n0CD5WTIyMkJDQx0cHIyMjPr167d06dKSkhJW7c8//wwPD3dycmJhh4SE3L17V+3Z29RHIhKL\nxT/99NO0adMGDhxoamrq6uoaEBCwZ88eoVCotj4AgFaTAQB0H3lqK5PJRCKRl5cX+/X8+fOq\nlXv06EFEgYGBSuXXrl0bMGCA2v/FDR8+vKCgQFPRtv5E//3f/61ax8XFpZUn+vbbbw0MDJQ+\nzuFwPv/88++++46IevbsqfSRgoKCESNGqI3Ny8srLy9Pqb6LiwsRbdu27cqVKzY2NkTk6Ojo\n7u6up/e/AzozZ86USqXtvgKMp6cnO8uePXu4XK6RkdGQIUPMzMzYRxwdHQsKCqKjow0NDTkc\njoeHB4uEiLhcruq/gbb28enTp25ubmrr9+vXr7CwsJW3AwBASyBxB4DupJi4y2Sya9euscTR\nzc2toaFBqbLaxD0rK0tfX5814uvru27dusjIyGXLljk7O7NCT0/PpqamjofaphPV19fzeLyf\nfvqJHfrjjz94PF5ZWVlrThQTEyNPXsPCwnbu3Ll9+/aZM2dyOBwiYmPtSol7Y2Nj//792acG\nDhy4Zs2a77///qOPPpIn2Y6OjsXFxYofYYn78uXLbWxsJk6c+PTpU1ZeUFAg3yl///797b4C\nDEvcfXx8DAwMtm/fLhQKZTKZUChctWoV+8igQYO4XG5ISAifz2cf+eWXX7hcLjvUkT7W1dUN\nHjyYHRo3btz27dujoqK2bdv20ksvyQtbczsAALQHEncA6E5KibtMJvv4449ZSUREhFJltYn7\nxIkTiUhPT2/Xrl2K5ZWVlTNmzGBN7du3r+OhtuNE8oW2V69ebeVZBAJBv379iKhXr16pqamK\nh+Lj401MTFiDSok7m4tCRB9++CFLjpnGxsYPPviAHZo/f77iR1jirqenN27cOKVsu6KiomfP\nnkQ0c+bMDl4BlrgT0aZNmxTLxWKxfCx8/PjxSkP7CxcuZIeePXvW7j7+/vvvrHDt2rVKp37l\nlVfYIdVBegAAbYbEHQC6k2riXl9fz+aCc7nc69evK1ZWTdwTExPVZqVMbW1tr169iKh37971\n9fUdibN9J2pH4v7tt9+yj/zP//yP6tH169erJu6lpaXm5uZENHHiRNXJLVKpdMKECexT+fn5\n8nKWuBPR5cuXVU8UGBhIRG5ubh28AixxNzU1bWxsVPrIvHnzWIMXLlxQOnTw4EF26ObNm+3u\n4xdffMFKHj58qFT/1KlTgYGBgYGBmZmZqn0BANBaWJwKANrF1NT0wIEDRCSRSBYtWiTfUVGt\nS5cusR/kWZqiHj16sCkZ5eXlDx8+7EhUXXaipKQkIrK3t1+wYIHq0dWrV6vOff/1119ra2uJ\n6LPPPmPTaRRxOBz5Yt+LFy8qHbWzs/Pz81M9Ue/evYlIccVnR66Al5eXsbGxUiFL9DkczqhR\no9QeIiKxWNzuPsob+fXXX5XqT58+/ezZs2fPnmWbkAIA6Aok7gCgdQICAt59910iun///o4d\nO1qomZubS0Tm5ubyKRlKRo8ezX548uRJR0Lq4hO99NJLbHNMJTY2NvKRcrmrV68SkZ6enoeH\nR4U6ffv2ZTXv3Lmj9Fl3d/c2Bda+K2Bra9tcs2ZmZvK1qi1oRx9nzJjBWv7888/9/Py+/fbb\n27dva3yXIQCArqTf3QEAAKixe/fu5OTkkpKSrVu3hoaGDho0SG21x48fE5F8zaIq+Q6MrGa7\ndc2JGhsb8/PzFVtT5eLionQKHo9HRFKptIXwmPLycqUSe3v7VsbWZZdarXb00d3d/fjx40uX\nLi0sLLxy5cqVK1eIyMzM7OWXX54+ffrs2bOf2xQAgLbBiDsAaKOePXvu27ePiEQi0eLFi6VS\nqdpqzZXLySeWCASCjsTTNScSCAQymYyIVGeDqJ5Irr6+vvXtK5W0cCIlXXap1WpfH6dNm5aT\nkxMVFTVnzhw2Rb6+vj41NXXNmjUeHh6rVq1qeSIWAIC2QeIOAFpqzpw5oaGhRHTlypXIyEi1\ndTw8PIjo77//bq6Rp0+fsh8GDhzYkWC65kRWVlZsI/N//vmnuTqqMbANIs3MzCQSScurmo4d\nO9bu2LrsUqvV7j6ampouWrToxIkTlZWVWVlZkZGRc+bMMTY2lkgke/fu/fLLLzUeKgBA50Hi\nDgDaa9++fSxjW79+fV5enmoFtod3TU1NczPLb926xX7oYDbZZSdiM8jv3bsnX5epqLa2VjV1\nZvOI6uvrOzi9vmVddgXUakcfi4uLi4qK5DNnuFzuiBEj3n///RMnTjx8+JBNu5dvmQ8AoBOQ\nuAOA9rK3t9+9ezcR1dXVLV26lE0jUTRu3Dj2w7Zt21Q/XldXx3ZXtLS0lL+Lp3267ERsW8Pi\n4uKjR4+qHt23b5/qRBT5Huo//PCD2ja/+eYbBwcHBweHFgbyn6vLroBa7ejjpEmTnJycXnrp\nJdV/Ni4uLmwpbVVVlcZDBQDoPEjcAUCrLVy4kO0pfvr0adWJzrNmzWK7GR49epTNiZerrq5+\n8803y8rKiGjTpk1sD3j5oQcPHjx48KD1w7ftO1E7fPLJJ2wrxrVr17L1lHJnz579+uuvVT8y\nfvx4th3+v/71r59//lnpaHp6+qZNm3g8npeXl+qONK3XZVdArXb0ke0yWVJScujQIaX6jx49\nYrtb+vr6ajxUAIDOg11lAEDb/fjjj0OHDm1ueeK3337r4+MjkUhWrlwZHx8/ceJEJyen+/fv\n//7772x2zaBBg+Rv1mRSUlLeeOMNIvLy8rp//34rw2jHidrB0tJy27Zty5Yt4/P5/v7+r732\n2ujRo2Uy2dWrV+Pj47lc7syZMxMTE5U2i/zXv/515syZhoaG8PDwH374ITAw0NXVtbq6+tKl\nSydPnhSLxb17996zZ08HY+uaK9CctvZx8eLFsbGxUql06dKlycnJEydO7NmzZ1VVVVZWVlxc\nXH19vZ6e3ooVKzopWgCATqGZ9zgBALSL6ptT1ZK/T5T+75tTmV9//ZXtGaIqLCysvLxcqb58\n/aKXl1ebom3ridrx5lQmKipK9Y1FFhYWJ0+e3LlzJxG5uroqfeT+/fsjR45UG1v//v3lryCV\nYyPTYWFhagNgDzYuLi4dvAJsyv6MGTNUT8GS5h49eqgeau66tbWPu3bt4nK5auubmJiofTct\nAIA2w1QZANABq1atevnll5s7GhIS8vTp040bN44aNcrKysrAwMDBwWHGjBnHjx+Pi4uTv0Gz\n47rsRIsWLbp69eqbb77p6OhoaGjo7Oy8ZMmS27dvBwcHs43e2ZpdRV5eXteuXfvXv/41YcIE\nGxsbIyOjAQMGTJs2LSoqKicnx9vbWyOBddkVUKutfYyIiHj06NGyZcvGjBnj6OhoZGTk6urq\n7++/cePGvLy8JUuWdGq0AAAax5GprNoBAACtNXPmzFOnTs2bN69Td0RpaGi4e/fu33///eab\nb3beWZojlUpzc3Nv3749ceJER0fHrg8AAEA7IXEHANAi8fHxJ0+eJKLvvvvO2tpa6SiPx+vf\nv39jY+POnTs/+eST7ggQAAC6DRanAgBoEQsLC7YR5JAhQ9atW6d4SCaTffHFF42NjUQ0Z86c\n7okPAAC6D0bcAQC0iFAoHDFixMOHDzkcTkRExIIFC9jeKdnZ2du2bUtLSyOi+fPnq93lHQAA\nXmxI3AEAtMs///wzduzYkpIStUdHjx6dkpKiOosGAABeeNhVBgBAu7i4uOTm5u7cudPT09PI\nyIgVcrncESNG7N69++LFi8jaAQD+M2HEHQBAe8lkMh6PR0TW1tbyJB4AAP4zIXEHAAAAANAB\nmCoDAAAAAKADkLgDAAAAAOgAJO4AAAAAADoAiTsAAAAAgA5A4g4AAAAAoAOQuAMAAAAA6AAk\n7gAAAAAAOgCJOwAAAACADkDiDgAAAACgA5C4AwAAAADoACTuAAAAAAA6AIk7AAAAAIAOQOIO\nAAAAAKADkLgDAAAAAOgAJO4AAAAAADoAiTsAAAAAgA74/w6lolcsg84CAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "plot without title"
      ]
     },
     "metadata": {
      "image/png": {
       "height": 320,
       "width": 500
      }
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "# summarizing taxonomy\n",
    "p = genomes_f %>%\n",
    "    summarize.(n = n.(), .by=c(Family, Genus))  %>%\n",
    "    mutate.(Taxonomy = paste(Family, Genus, sep=';')) %>%\n",
    "    ggplot(aes(Taxonomy, n, fill=Family)) +\n",
    "    geom_bar(stat='identity') +\n",
    "    labs(y='No. of genomes', x='Family;Genus') +\n",
    "    theme_bw() +\n",
    "    coord_flip()\n",
    "p.dims(5,3.2)\n",
    "plot(p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-02-23T13:58:19.710824Z",
     "iopub.status.busy": "2021-02-23T13:58:19.709799Z",
     "iopub.status.idle": "2021-02-23T14:02:47.821001Z"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes2/GUT_GENOME001999.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-026/MGYG-HGUT-02681/genomes1/GUT_GENOME004317.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-042/MGYG-HGUT-04245/genomes2/GUT_GENOME005637.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes2/GUT_GENOME006149.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-009/MGYG-HGUT-00939/genomes1/GUT_GENOME006702.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-040/MGYG-HGUT-04070/genomes1/GUT_GENOME006711.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME007481.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME009152.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME009895.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-039/MGYG-HGUT-03972/genomes1/GUT_GENOME011026.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME011034.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME012505.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME014750.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME015036.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME018032.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME018884.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes2/GUT_GENOME018997.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes2/GUT_GENOME019075.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME020800.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME021800.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME021845.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME021934.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-042/MGYG-HGUT-04245/genomes1/GUT_GENOME022422.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-008/MGYG-HGUT-00875/genomes1/GUT_GENOME022791.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME023175.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME023308.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME023691.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-011/MGYG-HGUT-01132/genomes1/GUT_GENOME023750.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-018/MGYG-HGUT-01889/genomes1/GUT_GENOME023931.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME024612.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME024865.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME025342.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-040/MGYG-HGUT-04052/genomes1/GUT_GENOME025383.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME025390.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME025396.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00718/genomes1/GUT_GENOME025577.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME025672.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME025895.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME025905.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00727/genomes1/GUT_GENOME025984.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME026012.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME026147.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-011/MGYG-HGUT-01132/genomes1/GUT_GENOME026613.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes2/GUT_GENOME026753.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-011/MGYG-HGUT-01132/genomes1/GUT_GENOME026776.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME026838.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-005/MGYG-HGUT-00530/genomes2/GUT_GENOME031108.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-038/MGYG-HGUT-03891/genomes2/GUT_GENOME033234.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-039/MGYG-HGUT-03972/genomes1/GUT_GENOME035464.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-005/MGYG-HGUT-00530/genomes1/GUT_GENOME038027.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-032/MGYG-HGUT-03224/genomes1/GUT_GENOME043661.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-011/MGYG-HGUT-01132/genomes1/GUT_GENOME044199.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-011/MGYG-HGUT-01132/genomes1/GUT_GENOME044239.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-042/MGYG-HGUT-04245/genomes2/GUT_GENOME047625.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes2/GUT_GENOME048317.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-009/MGYG-HGUT-00939/genomes1/GUT_GENOME048628.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME049881.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME055430.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-026/MGYG-HGUT-02681/genomes1/GUT_GENOME055959.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME057004.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-042/MGYG-HGUT-04245/genomes1/GUT_GENOME057303.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-005/MGYG-HGUT-00530/genomes1/GUT_GENOME057861.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-042/MGYG-HGUT-04245/genomes1/GUT_GENOME067108.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-020/MGYG-HGUT-02072/genomes1/GUT_GENOME067315.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME067942.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-042/MGYG-HGUT-04245/genomes1/GUT_GENOME070737.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-042/MGYG-HGUT-04245/genomes1/GUT_GENOME071621.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-026/MGYG-HGUT-02681/genomes1/GUT_GENOME072725.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME074075.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-011/MGYG-HGUT-01132/genomes1/GUT_GENOME081213.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-011/MGYG-HGUT-01103/genomes1/GUT_GENOME081297.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME081465.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME081799.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME083231.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME083365.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME083987.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-039/MGYG-HGUT-03967/genomes1/GUT_GENOME084056.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME084566.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-011/MGYG-HGUT-01132/genomes1/GUT_GENOME084610.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-038/MGYG-HGUT-03891/genomes1/GUT_GENOME086293.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-039/MGYG-HGUT-03961/genomes1/GUT_GENOME087392.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-011/MGYG-HGUT-01132/genomes1/GUT_GENOME089788.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-042/MGYG-HGUT-04245/genomes2/GUT_GENOME091022.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME091174.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME091251.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00718/genomes1/GUT_GENOME091444.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME092910.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME093203.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes2/GUT_GENOME093433.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME093687.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME094110.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME095220.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME095708.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-005/MGYG-HGUT-00530/genomes1/GUT_GENOME095807.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-026/MGYG-HGUT-02681/genomes1/GUT_GENOME098981.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME102313.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-016/MGYG-HGUT-01675/genomes1/GUT_GENOME102728.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME102818.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-011/MGYG-HGUT-01132/genomes1/GUT_GENOME103985.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00718/genomes1/GUT_GENOME104483.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-019/MGYG-HGUT-01932/genomes1/GUT_GENOME104663.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-045/MGYG-HGUT-04551/genomes1/GUT_GENOME104784.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-008/MGYG-HGUT-00875/genomes1/GUT_GENOME105230.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-009/MGYG-HGUT-00939/genomes1/GUT_GENOME105699.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME105752.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-005/MGYG-HGUT-00530/genomes1/GUT_GENOME107619.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME107801.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-005/MGYG-HGUT-00530/genomes1/GUT_GENOME107816.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes2/GUT_GENOME108244.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-042/MGYG-HGUT-04245/genomes1/GUT_GENOME108859.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-042/MGYG-HGUT-04245/genomes1/GUT_GENOME109505.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME110114.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-008/MGYG-HGUT-00875/genomes1/GUT_GENOME110789.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME110934.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME110954.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes2/GUT_GENOME111365.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME111416.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-026/MGYG-HGUT-02681/genomes1/GUT_GENOME111491.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-042/MGYG-HGUT-04245/genomes2/GUT_GENOME111501.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME112062.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-005/MGYG-HGUT-00530/genomes1/GUT_GENOME112317.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME112403.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes2/GUT_GENOME112898.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-042/MGYG-HGUT-04245/genomes1/GUT_GENOME112935.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME113027.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-042/MGYG-HGUT-04245/genomes1/GUT_GENOME113041.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-005/MGYG-HGUT-00530/genomes1/GUT_GENOME113243.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-005/MGYG-HGUT-00530/genomes3/GUT_GENOME113391.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME113455.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-011/MGYG-HGUT-01132/genomes1/GUT_GENOME113652.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME113942.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-011/MGYG-HGUT-01132/genomes1/GUT_GENOME114047.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME114314.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-026/MGYG-HGUT-02681/genomes1/GUT_GENOME114383.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-042/MGYG-HGUT-04245/genomes2/GUT_GENOME114384.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-005/MGYG-HGUT-00530/genomes1/GUT_GENOME114438.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes2/GUT_GENOME114538.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME114925.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes2/GUT_GENOME115405.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME115664.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME115734.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME116268.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-042/MGYG-HGUT-04245/genomes1/GUT_GENOME116339.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes2/GUT_GENOME116765.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-011/MGYG-HGUT-01132/genomes1/GUT_GENOME117240.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-039/MGYG-HGUT-03961/genomes1/GUT_GENOME117847.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME117964.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-042/MGYG-HGUT-04245/genomes2/GUT_GENOME118007.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-005/MGYG-HGUT-00530/genomes2/GUT_GENOME118193.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-042/MGYG-HGUT-04245/genomes1/GUT_GENOME118498.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME119315.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-005/MGYG-HGUT-00530/genomes1/GUT_GENOME120873.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME122088.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-020/MGYG-HGUT-02072/genomes1/GUT_GENOME123861.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-011/MGYG-HGUT-01132/genomes1/GUT_GENOME123945.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes2/GUT_GENOME124482.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-005/MGYG-HGUT-00530/genomes2/GUT_GENOME124704.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-042/MGYG-HGUT-04245/genomes2/GUT_GENOME124708.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-038/MGYG-HGUT-03891/genomes2/GUT_GENOME125485.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-020/MGYG-HGUT-02072/genomes1/GUT_GENOME125851.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-011/MGYG-HGUT-01132/genomes1/GUT_GENOME126128.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-009/MGYG-HGUT-00939/genomes1/GUT_GENOME126626.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME127157.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME127754.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME129204.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME129433.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-020/MGYG-HGUT-02072/genomes1/GUT_GENOME129467.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME129678.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME132602.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME133038.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-005/MGYG-HGUT-00530/genomes2/GUT_GENOME134096.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-011/MGYG-HGUT-01132/genomes1/GUT_GENOME134119.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-005/MGYG-HGUT-00530/genomes2/GUT_GENOME134648.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME135093.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME135179.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes2/GUT_GENOME135922.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME136346.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME136676.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME138001.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME138477.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-008/MGYG-HGUT-00875/genomes1/GUT_GENOME139071.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-045/MGYG-HGUT-04551/genomes1/GUT_GENOME139869.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes2/GUT_GENOME148347.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-038/MGYG-HGUT-03891/genomes4/GUT_GENOME148387.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME149021.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-011/MGYG-HGUT-01132/genomes1/GUT_GENOME150000.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME151676.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-026/MGYG-HGUT-02629/genomes1/GUT_GENOME155741.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-026/MGYG-HGUT-02679/genomes1/GUT_GENOME158031.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-026/MGYG-HGUT-02681/genomes1/GUT_GENOME158142.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-011/MGYG-HGUT-01132/genomes1/GUT_GENOME158277.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME158309.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-011/MGYG-HGUT-01132/genomes1/GUT_GENOME158323.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-009/MGYG-HGUT-00939/genomes1/GUT_GENOME158370.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-009/MGYG-HGUT-00939/genomes1/GUT_GENOME158436.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-009/MGYG-HGUT-00939/genomes1/GUT_GENOME158456.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-011/MGYG-HGUT-01132/genomes1/GUT_GENOME158543.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-011/MGYG-HGUT-01132/genomes1/GUT_GENOME158610.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-038/MGYG-HGUT-03891/genomes1/GUT_GENOME158947.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-038/MGYG-HGUT-03891/genomes1/GUT_GENOME159103.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-042/MGYG-HGUT-04245/genomes1/GUT_GENOME159777.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-042/MGYG-HGUT-04245/genomes2/GUT_GENOME160252.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-005/MGYG-HGUT-00530/genomes3/GUT_GENOME162740.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00718/genomes1/GUT_GENOME170651.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-020/MGYG-HGUT-02072/genomes1/GUT_GENOME173659.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-026/MGYG-HGUT-02679/genomes1/GUT_GENOME175025.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes2/GUT_GENOME175257.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME176010.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-008/MGYG-HGUT-00875/genomes1/GUT_GENOME179638.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-016/MGYG-HGUT-01658/genomes1/GUT_GENOME180128.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-016/MGYG-HGUT-01658/genomes1/GUT_GENOME180773.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-038/MGYG-HGUT-03891/genomes2/GUT_GENOME186446.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-038/MGYG-HGUT-03891/genomes2/GUT_GENOME188950.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME191092.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-017/MGYG-HGUT-01747/genomes1/GUT_GENOME191624.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME193107.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME193901.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-038/MGYG-HGUT-03891/genomes4/GUT_GENOME196897.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-042/MGYG-HGUT-04245/genomes1/GUT_GENOME197544.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-038/MGYG-HGUT-03891/genomes1/GUT_GENOME198224.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME198593.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME206070.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-038/MGYG-HGUT-03891/genomes1/GUT_GENOME207134.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-038/MGYG-HGUT-03891/genomes4/GUT_GENOME207194.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-038/MGYG-HGUT-03891/genomes1/GUT_GENOME208933.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-011/MGYG-HGUT-01132/genomes1/GUT_GENOME209311.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-038/MGYG-HGUT-03891/genomes4/GUT_GENOME209384.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-038/MGYG-HGUT-03891/genomes1/GUT_GENOME211577.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-005/MGYG-HGUT-00530/genomes1/GUT_GENOME211643.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-011/MGYG-HGUT-01132/genomes1/GUT_GENOME211803.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-039/MGYG-HGUT-03972/genomes1/GUT_GENOME213158.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-039/MGYG-HGUT-03972/genomes1/GUT_GENOME214515.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-026/MGYG-HGUT-02681/genomes1/GUT_GENOME214773.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-026/MGYG-HGUT-02681/genomes1/GUT_GENOME217538.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-039/MGYG-HGUT-03972/genomes1/GUT_GENOME217857.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-042/MGYG-HGUT-04245/genomes2/GUT_GENOME217871.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-039/MGYG-HGUT-03972/genomes1/GUT_GENOME217887.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME218690.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-026/MGYG-HGUT-02681/genomes1/GUT_GENOME219111.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-032/MGYG-HGUT-03224/genomes1/GUT_GENOME219688.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-020/MGYG-HGUT-02072/genomes1/GUT_GENOME220052.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-020/MGYG-HGUT-02072/genomes1/GUT_GENOME220179.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME221658.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-042/MGYG-HGUT-04245/genomes2/GUT_GENOME226773.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME229160.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME229523.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-011/MGYG-HGUT-01132/genomes1/GUT_GENOME231159.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-035/MGYG-HGUT-03558/genomes1/GUT_GENOME237426.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME237619.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-032/MGYG-HGUT-03224/genomes1/GUT_GENOME243616.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-038/MGYG-HGUT-03875/genomes1/GUT_GENOME243997.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME245548.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-039/MGYG-HGUT-03961/genomes1/GUT_GENOME245676.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME246707.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-019/MGYG-HGUT-01932/genomes1/GUT_GENOME246796.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-011/MGYG-HGUT-01132/genomes1/GUT_GENOME247127.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-042/MGYG-HGUT-04245/genomes1/GUT_GENOME247253.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME247368.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes2/GUT_GENOME249008.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-040/MGYG-HGUT-04034/genomes1/GUT_GENOME251006.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-011/MGYG-HGUT-01156/genomes1/GUT_GENOME251370.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes2/GUT_GENOME251395.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-039/MGYG-HGUT-03945/genomes1/GUT_GENOME251449.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME251481.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-011/MGYG-HGUT-01132/genomes1/GUT_GENOME251502.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes2/GUT_GENOME251715.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME251769.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-026/MGYG-HGUT-02681/genomes1/GUT_GENOME251866.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-042/MGYG-HGUT-04245/genomes2/GUT_GENOME251979.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME251995.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-038/MGYG-HGUT-03891/genomes4/GUT_GENOME252229.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-005/MGYG-HGUT-00530/genomes2/GUT_GENOME252265.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME252323.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME252447.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME252532.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-038/MGYG-HGUT-03891/genomes3/GUT_GENOME252572.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-042/MGYG-HGUT-04245/genomes2/GUT_GENOME252763.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-008/MGYG-HGUT-00875/genomes1/GUT_GENOME253102.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-011/MGYG-HGUT-01132/genomes1/GUT_GENOME253389.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME253599.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-038/MGYG-HGUT-03891/genomes3/GUT_GENOME253664.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME254184.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-012/MGYG-HGUT-01237/genomes1/GUT_GENOME254327.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-040/MGYG-HGUT-04034/genomes1/GUT_GENOME254353.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME254480.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME254551.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-032/MGYG-HGUT-03224/genomes1/GUT_GENOME254685.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME255044.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-005/MGYG-HGUT-00530/genomes2/GUT_GENOME255058.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-005/MGYG-HGUT-00530/genomes2/GUT_GENOME255127.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME255169.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME255357.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-009/MGYG-HGUT-00939/genomes1/GUT_GENOME255516.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-011/MGYG-HGUT-01132/genomes1/GUT_GENOME255520.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-011/MGYG-HGUT-01132/genomes1/GUT_GENOME255625.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME255693.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME255920.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME256026.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME256080.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-020/MGYG-HGUT-02072/genomes1/GUT_GENOME256472.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME256501.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-040/MGYG-HGUT-04052/genomes1/GUT_GENOME256891.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-039/MGYG-HGUT-03972/genomes1/GUT_GENOME256928.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-012/MGYG-HGUT-01237/genomes1/GUT_GENOME256935.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME257064.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME257240.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-011/MGYG-HGUT-01132/genomes1/GUT_GENOME257346.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME257380.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME257491.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-011/MGYG-HGUT-01132/genomes1/GUT_GENOME257687.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-011/MGYG-HGUT-01132/genomes1/GUT_GENOME257761.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME258061.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-042/MGYG-HGUT-04245/genomes2/GUT_GENOME258296.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-009/MGYG-HGUT-00939/genomes1/GUT_GENOME258401.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-011/MGYG-HGUT-01132/genomes1/GUT_GENOME258517.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-009/MGYG-HGUT-00939/genomes1/GUT_GENOME258570.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME258620.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-041/MGYG-HGUT-04179/genomes1/GUT_GENOME258665.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes2/GUT_GENOME258671.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-011/MGYG-HGUT-01132/genomes1/GUT_GENOME258678.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-042/MGYG-HGUT-04245/genomes1/GUT_GENOME259078.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME259116.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-020/MGYG-HGUT-02072/genomes1/GUT_GENOME259347.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME259475.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-039/MGYG-HGUT-03967/genomes1/GUT_GENOME259485.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-005/MGYG-HGUT-00530/genomes1/GUT_GENOME259575.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME259624.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00718/genomes1/GUT_GENOME259815.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME259848.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-011/MGYG-HGUT-01132/genomes1/GUT_GENOME260244.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-038/MGYG-HGUT-03891/genomes1/GUT_GENOME260260.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME260362.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-039/MGYG-HGUT-03972/genomes1/GUT_GENOME260711.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME260852.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME260912.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-032/MGYG-HGUT-03224/genomes1/GUT_GENOME260971.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-038/MGYG-HGUT-03891/genomes2/GUT_GENOME260994.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME261098.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-011/MGYG-HGUT-01132/genomes1/GUT_GENOME261099.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-040/MGYG-HGUT-04052/genomes1/GUT_GENOME261125.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME261147.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME261177.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME262360.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME263396.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-005/MGYG-HGUT-00530/genomes2/GUT_GENOME265850.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-039/MGYG-HGUT-03961/genomes1/GUT_GENOME265851.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME266061.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME268393.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME268512.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-041/MGYG-HGUT-04179/genomes1/GUT_GENOME269902.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME270724.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME274044.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME274160.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME274720.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-038/MGYG-HGUT-03891/genomes2/GUT_GENOME275434.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-011/MGYG-HGUT-01132/genomes1/GUT_GENOME275671.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME276507.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-005/MGYG-HGUT-00530/genomes2/GUT_GENOME277111.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME277241.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-011/MGYG-HGUT-01132/genomes1/GUT_GENOME277680.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME278880.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME279410.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME279435.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-005/MGYG-HGUT-00530/genomes1/GUT_GENOME279772.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME280640.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00718/genomes1/GUT_GENOME281443.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME281451.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME281578.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME282452.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME282719.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-045/MGYG-HGUT-04551/genomes1/GUT_GENOME282790.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-026/MGYG-HGUT-02681/genomes1/GUT_GENOME282987.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME283318.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes2/GUT_GENOME283538.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME283719.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-005/MGYG-HGUT-00530/genomes2/GUT_GENOME284294.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-011/MGYG-HGUT-01132/genomes1/GUT_GENOME285124.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME285532.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME285613.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME285799.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes1/GUT_GENOME286150.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-038/MGYG-HGUT-03891/genomes4/GUT_GENOME286223.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-007/MGYG-HGUT-00730/genomes2/GUT_GENOME286615.gff.gz\n",
      "\n",
      "Downloading: ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v1.0/all_genomes/MGYG-HGUT-005/MGYG-HGUT-00530/genomes2/GUT_GENOME286691.gff.gz\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "384"
      ],
      "text/latex": [
       "384"
      ],
      "text/markdown": [
       "384"
      ],
      "text/plain": [
       "[1] 384"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# downloading\n",
    "get_file = function(url, base_dir){\n",
    "    outfile = file.path(base_dir,  'UHGG', gsub('.+/', '', url))\n",
    "    message('Downloading: ', url)\n",
    "    curl_download(url, outfile, mode = \"wb\")\n",
    "}\n",
    "\n",
    "ret = genomes_f$FTP_download %>%\n",
    "    lapply(get_file, work_dir)\n",
    "ret %>% length"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Parsing gff files\n",
    "\n",
    "```\n",
    "(genome) @ rick:/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/UHGG\n",
    "$ find . -name \"*.gff.gz\" | xargs -I % /ebio/abt3_projects/databases_no-backup/UHGG/2019_09/prokka_gff2fasta.py %\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## TUK MAGs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-02-23T14:03:17.063502Z",
     "iopub.status.busy": "2021-02-23T14:03:17.061344Z",
     "iopub.status.idle": "2021-02-23T14:03:17.155148Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 12357 × 20</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Name</th><th scope=col>Fasta</th><th scope=col>Domain</th><th scope=col>Phylum</th><th scope=col>Class</th><th scope=col>Order</th><th scope=col>Family</th><th scope=col>Genus</th><th scope=col>Species</th><th scope=col>Completeness</th><th scope=col>Contamination</th><th scope=col>Strain.heterogeneity</th><th scope=col>Genome.size..bp.</th><th scope=col>X..contigs</th><th scope=col>N50..contigs.</th><th scope=col>Mean.contig.length..bp.</th><th scope=col>Longest.contig..bp.</th><th scope=col>GC</th><th scope=col>Coding.density</th><th scope=col>X..predicted.genes</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1002320__metabat2__HighNoCov.035</td><td>/ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/TUK-5projects/LLMGA/v0.12/LLG/rnd1/drep/drep/dereplicated_genomes/1002320__metabat2__HighNoCov.035.fna</td><td>Bacteria</td><td>Bacteroidota</td><td>Bacteroidia</td><td>Bacteroidales</td><td>Rikenellaceae</td><td>Tidjanibacter</td><td>Tidjanibacter inops</td><td> 92.2</td><td>4.61</td><td>7.14</td><td>2261725</td><td>365</td><td>  8314</td><td> 6196</td><td> 65550</td><td>56.9</td><td>86.29</td><td>2235</td></tr>\n",
       "\t<tr><td>1002335__maxbin2__High.035      </td><td>/ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/TUK-5projects/LLMGA/v0.12/LLG/rnd1/drep/drep/dereplicated_genomes/1002335__maxbin2__High.035.fna      </td><td>Bacteria</td><td>Bacteroidota</td><td>Bacteroidia</td><td>Bacteroidales</td><td>Rikenellaceae</td><td>Alistipes_A  </td><td>Alistipes_A ihumii </td><td>100.0</td><td>0.00</td><td>0.00</td><td>2620923</td><td> 29</td><td>339045</td><td>90376</td><td>457503</td><td>58.5</td><td>83.91</td><td>2100</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>ERR3451530__metabat2__High.008</td><td>/ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/TUK-5projects/LLMGA/v0.12/LLG/rnd1/drep/drep/dereplicated_genomes/ERR3451530__metabat2__High.008.fna</td><td>Bacteria</td><td>Firmicutes_A</td><td>Clostridia</td><td>Oscillospirales</td><td>Oscillospiraceae</td><td>CAG-103</td><td>CAG-103 sp000432375</td><td>88.26</td><td>0.67</td><td>100</td><td>1925297</td><td> 28</td><td>172712</td><td>68760</td><td>283992</td><td>62.2</td><td>89.32</td><td>1782</td></tr>\n",
       "\t<tr><td>ERR3451530__vamb__High.213    </td><td>/ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/TUK-5projects/LLMGA/v0.12/LLG/rnd1/drep/drep/dereplicated_genomes/ERR3451530__vamb__High.213.fna    </td><td>Bacteria</td><td>Firmicutes_A</td><td>Clostridia</td><td>Oscillospirales</td><td>Ruminococcaceae </td><td>CAG-115</td><td>CAG-115 sp003531585</td><td>95.97</td><td>0.84</td><td>100</td><td>2811853</td><td>271</td><td> 15050</td><td>10375</td><td> 90239</td><td>52.4</td><td>88.48</td><td>2685</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 12357 × 20\n",
       "\\begin{tabular}{llllllllllllllllllll}\n",
       " Name & Fasta & Domain & Phylum & Class & Order & Family & Genus & Species & Completeness & Contamination & Strain.heterogeneity & Genome.size..bp. & X..contigs & N50..contigs. & Mean.contig.length..bp. & Longest.contig..bp. & GC & Coding.density & X..predicted.genes\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <int> & <int> & <int> & <int> & <int> & <dbl> & <dbl> & <int>\\\\\n",
       "\\hline\n",
       "\t 1002320\\_\\_metabat2\\_\\_HighNoCov.035 & /ebio/abt3\\_projects/Anxiety\\_Twins\\_Metagenomes/data/metagenome/TUK-5projects/LLMGA/v0.12/LLG/rnd1/drep/drep/dereplicated\\_genomes/1002320\\_\\_metabat2\\_\\_HighNoCov.035.fna & Bacteria & Bacteroidota & Bacteroidia & Bacteroidales & Rikenellaceae & Tidjanibacter & Tidjanibacter inops &  92.2 & 4.61 & 7.14 & 2261725 & 365 &   8314 &  6196 &  65550 & 56.9 & 86.29 & 2235\\\\\n",
       "\t 1002335\\_\\_maxbin2\\_\\_High.035       & /ebio/abt3\\_projects/Anxiety\\_Twins\\_Metagenomes/data/metagenome/TUK-5projects/LLMGA/v0.12/LLG/rnd1/drep/drep/dereplicated\\_genomes/1002335\\_\\_maxbin2\\_\\_High.035.fna       & Bacteria & Bacteroidota & Bacteroidia & Bacteroidales & Rikenellaceae & Alistipes\\_A   & Alistipes\\_A ihumii  & 100.0 & 0.00 & 0.00 & 2620923 &  29 & 339045 & 90376 & 457503 & 58.5 & 83.91 & 2100\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t ERR3451530\\_\\_metabat2\\_\\_High.008 & /ebio/abt3\\_projects/Anxiety\\_Twins\\_Metagenomes/data/metagenome/TUK-5projects/LLMGA/v0.12/LLG/rnd1/drep/drep/dereplicated\\_genomes/ERR3451530\\_\\_metabat2\\_\\_High.008.fna & Bacteria & Firmicutes\\_A & Clostridia & Oscillospirales & Oscillospiraceae & CAG-103 & CAG-103 sp000432375 & 88.26 & 0.67 & 100 & 1925297 &  28 & 172712 & 68760 & 283992 & 62.2 & 89.32 & 1782\\\\\n",
       "\t ERR3451530\\_\\_vamb\\_\\_High.213     & /ebio/abt3\\_projects/Anxiety\\_Twins\\_Metagenomes/data/metagenome/TUK-5projects/LLMGA/v0.12/LLG/rnd1/drep/drep/dereplicated\\_genomes/ERR3451530\\_\\_vamb\\_\\_High.213.fna     & Bacteria & Firmicutes\\_A & Clostridia & Oscillospirales & Ruminococcaceae  & CAG-115 & CAG-115 sp003531585 & 95.97 & 0.84 & 100 & 2811853 & 271 &  15050 & 10375 &  90239 & 52.4 & 88.48 & 2685\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 12357 × 20\n",
       "\n",
       "| Name &lt;chr&gt; | Fasta &lt;chr&gt; | Domain &lt;chr&gt; | Phylum &lt;chr&gt; | Class &lt;chr&gt; | Order &lt;chr&gt; | Family &lt;chr&gt; | Genus &lt;chr&gt; | Species &lt;chr&gt; | Completeness &lt;dbl&gt; | Contamination &lt;dbl&gt; | Strain.heterogeneity &lt;dbl&gt; | Genome.size..bp. &lt;int&gt; | X..contigs &lt;int&gt; | N50..contigs. &lt;int&gt; | Mean.contig.length..bp. &lt;int&gt; | Longest.contig..bp. &lt;int&gt; | GC &lt;dbl&gt; | Coding.density &lt;dbl&gt; | X..predicted.genes &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1002320__metabat2__HighNoCov.035 | /ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/TUK-5projects/LLMGA/v0.12/LLG/rnd1/drep/drep/dereplicated_genomes/1002320__metabat2__HighNoCov.035.fna | Bacteria | Bacteroidota | Bacteroidia | Bacteroidales | Rikenellaceae | Tidjanibacter | Tidjanibacter inops |  92.2 | 4.61 | 7.14 | 2261725 | 365 |   8314 |  6196 |  65550 | 56.9 | 86.29 | 2235 |\n",
       "| 1002335__maxbin2__High.035       | /ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/TUK-5projects/LLMGA/v0.12/LLG/rnd1/drep/drep/dereplicated_genomes/1002335__maxbin2__High.035.fna       | Bacteria | Bacteroidota | Bacteroidia | Bacteroidales | Rikenellaceae | Alistipes_A   | Alistipes_A ihumii  | 100.0 | 0.00 | 0.00 | 2620923 |  29 | 339045 | 90376 | 457503 | 58.5 | 83.91 | 2100 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| ERR3451530__metabat2__High.008 | /ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/TUK-5projects/LLMGA/v0.12/LLG/rnd1/drep/drep/dereplicated_genomes/ERR3451530__metabat2__High.008.fna | Bacteria | Firmicutes_A | Clostridia | Oscillospirales | Oscillospiraceae | CAG-103 | CAG-103 sp000432375 | 88.26 | 0.67 | 100 | 1925297 |  28 | 172712 | 68760 | 283992 | 62.2 | 89.32 | 1782 |\n",
       "| ERR3451530__vamb__High.213     | /ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/TUK-5projects/LLMGA/v0.12/LLG/rnd1/drep/drep/dereplicated_genomes/ERR3451530__vamb__High.213.fna     | Bacteria | Firmicutes_A | Clostridia | Oscillospirales | Ruminococcaceae  | CAG-115 | CAG-115 sp003531585 | 95.97 | 0.84 | 100 | 2811853 | 271 |  15050 | 10375 |  90239 | 52.4 | 88.48 | 2685 |\n",
       "\n"
      ],
      "text/plain": [
       "      Name                            \n",
       "1     1002320__metabat2__HighNoCov.035\n",
       "2     1002335__maxbin2__High.035      \n",
       "⋮     ⋮                               \n",
       "12356 ERR3451530__metabat2__High.008  \n",
       "12357 ERR3451530__vamb__High.213      \n",
       "      Fasta                                                                                                                                                               \n",
       "1     /ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/TUK-5projects/LLMGA/v0.12/LLG/rnd1/drep/drep/dereplicated_genomes/1002320__metabat2__HighNoCov.035.fna\n",
       "2     /ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/TUK-5projects/LLMGA/v0.12/LLG/rnd1/drep/drep/dereplicated_genomes/1002335__maxbin2__High.035.fna      \n",
       "⋮     ⋮                                                                                                                                                                   \n",
       "12356 /ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/TUK-5projects/LLMGA/v0.12/LLG/rnd1/drep/drep/dereplicated_genomes/ERR3451530__metabat2__High.008.fna  \n",
       "12357 /ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/TUK-5projects/LLMGA/v0.12/LLG/rnd1/drep/drep/dereplicated_genomes/ERR3451530__vamb__High.213.fna      \n",
       "      Domain   Phylum       Class       Order           Family          \n",
       "1     Bacteria Bacteroidota Bacteroidia Bacteroidales   Rikenellaceae   \n",
       "2     Bacteria Bacteroidota Bacteroidia Bacteroidales   Rikenellaceae   \n",
       "⋮     ⋮        ⋮            ⋮           ⋮               ⋮               \n",
       "12356 Bacteria Firmicutes_A Clostridia  Oscillospirales Oscillospiraceae\n",
       "12357 Bacteria Firmicutes_A Clostridia  Oscillospirales Ruminococcaceae \n",
       "      Genus         Species             Completeness Contamination\n",
       "1     Tidjanibacter Tidjanibacter inops  92.2        4.61         \n",
       "2     Alistipes_A   Alistipes_A ihumii  100.0        0.00         \n",
       "⋮     ⋮             ⋮                   ⋮            ⋮            \n",
       "12356 CAG-103       CAG-103 sp000432375 88.26        0.67         \n",
       "12357 CAG-115       CAG-115 sp003531585 95.97        0.84         \n",
       "      Strain.heterogeneity Genome.size..bp. X..contigs N50..contigs.\n",
       "1     7.14                 2261725          365          8314       \n",
       "2     0.00                 2620923           29        339045       \n",
       "⋮     ⋮                    ⋮                ⋮          ⋮            \n",
       "12356 100                  1925297           28        172712       \n",
       "12357 100                  2811853          271         15050       \n",
       "      Mean.contig.length..bp. Longest.contig..bp. GC   Coding.density\n",
       "1      6196                    65550              56.9 86.29         \n",
       "2     90376                   457503              58.5 83.91         \n",
       "⋮     ⋮                       ⋮                   ⋮    ⋮             \n",
       "12356 68760                   283992              62.2 89.32         \n",
       "12357 10375                    90239              52.4 88.48         \n",
       "      X..predicted.genes\n",
       "1     2235              \n",
       "2     2100              \n",
       "⋮     ⋮                 \n",
       "12356 1782              \n",
       "12357 2685              "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "F = '/ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/TUK-5projects/LLMGA/v0.12/LLG//rnd1/final_MAGs.tsv'\n",
    "TUK = Fread(F)\n",
    "TUK"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-02-23T14:03:18.507856Z",
     "iopub.status.busy": "2021-02-23T14:03:18.505554Z",
     "iopub.status.idle": "2021-02-23T14:03:18.555015Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tidytable: 729 × 20</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Name</th><th scope=col>Fasta</th><th scope=col>Domain</th><th scope=col>Phylum</th><th scope=col>Class</th><th scope=col>Order</th><th scope=col>Family</th><th scope=col>Genus</th><th scope=col>Species</th><th scope=col>Completeness</th><th scope=col>Contamination</th><th scope=col>Strain.heterogeneity</th><th scope=col>Genome.size..bp.</th><th scope=col>X..contigs</th><th scope=col>N50..contigs.</th><th scope=col>Mean.contig.length..bp.</th><th scope=col>Longest.contig..bp.</th><th scope=col>GC</th><th scope=col>Coding.density</th><th scope=col>X..predicted.genes</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1002795__metabat2__High.054</td><td>/ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/TUK-5projects/LLMGA/v0.12/LLG/rnd1/drep/drep/dereplicated_genomes/1002795__metabat2__High.054.fna</td><td>Bacteria</td><td>Firmicutes_A</td><td>Clostridia_A</td><td>Christensenellales</td><td>CAG-138</td><td>PeH17</td><td>PeH17 sp000435055</td><td>93.94</td><td>1.34</td><td>0</td><td>2081772</td><td>47</td><td>74861</td><td>44293</td><td>274868</td><td>49.1</td><td>86.23</td><td>1845</td></tr>\n",
       "\t<tr><td>1003355__metabat2__High.078</td><td>/ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/TUK-5projects/LLMGA/v0.12/LLG/rnd1/drep/drep/dereplicated_genomes/1003355__metabat2__High.078.fna</td><td>Bacteria</td><td>Firmicutes_A</td><td>Clostridia_A</td><td>Christensenellales</td><td>CAG-138</td><td>PeH17</td><td>PeH17 sp000435055</td><td>92.93</td><td>1.34</td><td>0</td><td>2045922</td><td>29</td><td>98160</td><td>70549</td><td>275567</td><td>49.5</td><td>86.12</td><td>1809</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>ERR3451182__metabat2__High.040</td><td>/ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/TUK-5projects/LLMGA/v0.12/LLG/rnd1/drep/drep/dereplicated_genomes/ERR3451182__metabat2__High.040.fna</td><td>Bacteria</td><td>Firmicutes_A</td><td>Clostridia_A</td><td>Christensenellales</td><td>CAG-74 </td><td>UBA11524</td><td>UBA11524 sp000437595</td><td>98.60</td><td>1.17</td><td>  0</td><td>3151119</td><td> 47</td><td>102152</td><td>67045</td><td>524993</td><td>57.7</td><td>87.32</td><td>2684</td></tr>\n",
       "\t<tr><td>ERR3451511__metabat2__High.016</td><td>/ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/TUK-5projects/LLMGA/v0.12/LLG/rnd1/drep/drep/dereplicated_genomes/ERR3451511__metabat2__High.016.fna</td><td>Bacteria</td><td>Firmicutes_A</td><td>Clostridia_A</td><td>Christensenellales</td><td>CAG-138</td><td>Phil1   </td><td>Phil1 sp001940855   </td><td>99.19</td><td>0.81</td><td>100</td><td>2112719</td><td>182</td><td> 18021</td><td>11608</td><td> 67673</td><td>52.6</td><td>90.65</td><td>2086</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tidytable: 729 × 20\n",
       "\\begin{tabular}{llllllllllllllllllll}\n",
       " Name & Fasta & Domain & Phylum & Class & Order & Family & Genus & Species & Completeness & Contamination & Strain.heterogeneity & Genome.size..bp. & X..contigs & N50..contigs. & Mean.contig.length..bp. & Longest.contig..bp. & GC & Coding.density & X..predicted.genes\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <int> & <int> & <int> & <int> & <int> & <dbl> & <dbl> & <int>\\\\\n",
       "\\hline\n",
       "\t 1002795\\_\\_metabat2\\_\\_High.054 & /ebio/abt3\\_projects/Anxiety\\_Twins\\_Metagenomes/data/metagenome/TUK-5projects/LLMGA/v0.12/LLG/rnd1/drep/drep/dereplicated\\_genomes/1002795\\_\\_metabat2\\_\\_High.054.fna & Bacteria & Firmicutes\\_A & Clostridia\\_A & Christensenellales & CAG-138 & PeH17 & PeH17 sp000435055 & 93.94 & 1.34 & 0 & 2081772 & 47 & 74861 & 44293 & 274868 & 49.1 & 86.23 & 1845\\\\\n",
       "\t 1003355\\_\\_metabat2\\_\\_High.078 & /ebio/abt3\\_projects/Anxiety\\_Twins\\_Metagenomes/data/metagenome/TUK-5projects/LLMGA/v0.12/LLG/rnd1/drep/drep/dereplicated\\_genomes/1003355\\_\\_metabat2\\_\\_High.078.fna & Bacteria & Firmicutes\\_A & Clostridia\\_A & Christensenellales & CAG-138 & PeH17 & PeH17 sp000435055 & 92.93 & 1.34 & 0 & 2045922 & 29 & 98160 & 70549 & 275567 & 49.5 & 86.12 & 1809\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t ERR3451182\\_\\_metabat2\\_\\_High.040 & /ebio/abt3\\_projects/Anxiety\\_Twins\\_Metagenomes/data/metagenome/TUK-5projects/LLMGA/v0.12/LLG/rnd1/drep/drep/dereplicated\\_genomes/ERR3451182\\_\\_metabat2\\_\\_High.040.fna & Bacteria & Firmicutes\\_A & Clostridia\\_A & Christensenellales & CAG-74  & UBA11524 & UBA11524 sp000437595 & 98.60 & 1.17 &   0 & 3151119 &  47 & 102152 & 67045 & 524993 & 57.7 & 87.32 & 2684\\\\\n",
       "\t ERR3451511\\_\\_metabat2\\_\\_High.016 & /ebio/abt3\\_projects/Anxiety\\_Twins\\_Metagenomes/data/metagenome/TUK-5projects/LLMGA/v0.12/LLG/rnd1/drep/drep/dereplicated\\_genomes/ERR3451511\\_\\_metabat2\\_\\_High.016.fna & Bacteria & Firmicutes\\_A & Clostridia\\_A & Christensenellales & CAG-138 & Phil1    & Phil1 sp001940855    & 99.19 & 0.81 & 100 & 2112719 & 182 &  18021 & 11608 &  67673 & 52.6 & 90.65 & 2086\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tidytable: 729 × 20\n",
       "\n",
       "| Name &lt;chr&gt; | Fasta &lt;chr&gt; | Domain &lt;chr&gt; | Phylum &lt;chr&gt; | Class &lt;chr&gt; | Order &lt;chr&gt; | Family &lt;chr&gt; | Genus &lt;chr&gt; | Species &lt;chr&gt; | Completeness &lt;dbl&gt; | Contamination &lt;dbl&gt; | Strain.heterogeneity &lt;dbl&gt; | Genome.size..bp. &lt;int&gt; | X..contigs &lt;int&gt; | N50..contigs. &lt;int&gt; | Mean.contig.length..bp. &lt;int&gt; | Longest.contig..bp. &lt;int&gt; | GC &lt;dbl&gt; | Coding.density &lt;dbl&gt; | X..predicted.genes &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1002795__metabat2__High.054 | /ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/TUK-5projects/LLMGA/v0.12/LLG/rnd1/drep/drep/dereplicated_genomes/1002795__metabat2__High.054.fna | Bacteria | Firmicutes_A | Clostridia_A | Christensenellales | CAG-138 | PeH17 | PeH17 sp000435055 | 93.94 | 1.34 | 0 | 2081772 | 47 | 74861 | 44293 | 274868 | 49.1 | 86.23 | 1845 |\n",
       "| 1003355__metabat2__High.078 | /ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/TUK-5projects/LLMGA/v0.12/LLG/rnd1/drep/drep/dereplicated_genomes/1003355__metabat2__High.078.fna | Bacteria | Firmicutes_A | Clostridia_A | Christensenellales | CAG-138 | PeH17 | PeH17 sp000435055 | 92.93 | 1.34 | 0 | 2045922 | 29 | 98160 | 70549 | 275567 | 49.5 | 86.12 | 1809 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| ERR3451182__metabat2__High.040 | /ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/TUK-5projects/LLMGA/v0.12/LLG/rnd1/drep/drep/dereplicated_genomes/ERR3451182__metabat2__High.040.fna | Bacteria | Firmicutes_A | Clostridia_A | Christensenellales | CAG-74  | UBA11524 | UBA11524 sp000437595 | 98.60 | 1.17 |   0 | 3151119 |  47 | 102152 | 67045 | 524993 | 57.7 | 87.32 | 2684 |\n",
       "| ERR3451511__metabat2__High.016 | /ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/TUK-5projects/LLMGA/v0.12/LLG/rnd1/drep/drep/dereplicated_genomes/ERR3451511__metabat2__High.016.fna | Bacteria | Firmicutes_A | Clostridia_A | Christensenellales | CAG-138 | Phil1    | Phil1 sp001940855    | 99.19 | 0.81 | 100 | 2112719 | 182 |  18021 | 11608 |  67673 | 52.6 | 90.65 | 2086 |\n",
       "\n"
      ],
      "text/plain": [
       "    Name                          \n",
       "1   1002795__metabat2__High.054   \n",
       "2   1003355__metabat2__High.078   \n",
       "⋮   ⋮                             \n",
       "728 ERR3451182__metabat2__High.040\n",
       "729 ERR3451511__metabat2__High.016\n",
       "    Fasta                                                                                                                                                             \n",
       "1   /ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/TUK-5projects/LLMGA/v0.12/LLG/rnd1/drep/drep/dereplicated_genomes/1002795__metabat2__High.054.fna   \n",
       "2   /ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/TUK-5projects/LLMGA/v0.12/LLG/rnd1/drep/drep/dereplicated_genomes/1003355__metabat2__High.078.fna   \n",
       "⋮   ⋮                                                                                                                                                                 \n",
       "728 /ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/TUK-5projects/LLMGA/v0.12/LLG/rnd1/drep/drep/dereplicated_genomes/ERR3451182__metabat2__High.040.fna\n",
       "729 /ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/TUK-5projects/LLMGA/v0.12/LLG/rnd1/drep/drep/dereplicated_genomes/ERR3451511__metabat2__High.016.fna\n",
       "    Domain   Phylum       Class        Order              Family  Genus   \n",
       "1   Bacteria Firmicutes_A Clostridia_A Christensenellales CAG-138 PeH17   \n",
       "2   Bacteria Firmicutes_A Clostridia_A Christensenellales CAG-138 PeH17   \n",
       "⋮   ⋮        ⋮            ⋮            ⋮                  ⋮       ⋮       \n",
       "728 Bacteria Firmicutes_A Clostridia_A Christensenellales CAG-74  UBA11524\n",
       "729 Bacteria Firmicutes_A Clostridia_A Christensenellales CAG-138 Phil1   \n",
       "    Species              Completeness Contamination Strain.heterogeneity\n",
       "1   PeH17 sp000435055    93.94        1.34          0                   \n",
       "2   PeH17 sp000435055    92.93        1.34          0                   \n",
       "⋮   ⋮                    ⋮            ⋮             ⋮                   \n",
       "728 UBA11524 sp000437595 98.60        1.17            0                 \n",
       "729 Phil1 sp001940855    99.19        0.81          100                 \n",
       "    Genome.size..bp. X..contigs N50..contigs. Mean.contig.length..bp.\n",
       "1   2081772          47         74861         44293                  \n",
       "2   2045922          29         98160         70549                  \n",
       "⋮   ⋮                ⋮          ⋮             ⋮                      \n",
       "728 3151119           47        102152        67045                  \n",
       "729 2112719          182         18021        11608                  \n",
       "    Longest.contig..bp. GC   Coding.density X..predicted.genes\n",
       "1   274868              49.1 86.23          1845              \n",
       "2   275567              49.5 86.12          1809              \n",
       "⋮   ⋮                   ⋮    ⋮              ⋮                 \n",
       "728 524993              57.7 87.32          2684              \n",
       "729  67673              52.6 90.65          2086              "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "TUK = TUK %>%\n",
    "    filter.(Order == 'Christensenellales')\n",
    "TUK"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-02-23T14:04:04.237553Z",
     "iopub.status.busy": "2021-02-23T14:04:04.234259Z",
     "iopub.status.idle": "2021-02-23T14:04:04.294830Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tidytable: 76 × 20</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Name</th><th scope=col>Fasta</th><th scope=col>Domain</th><th scope=col>Phylum</th><th scope=col>Class</th><th scope=col>Order</th><th scope=col>Family</th><th scope=col>Genus</th><th scope=col>Species</th><th scope=col>Completeness</th><th scope=col>Contamination</th><th scope=col>Strain.heterogeneity</th><th scope=col>Genome.size..bp.</th><th scope=col>X..contigs</th><th scope=col>N50..contigs.</th><th scope=col>Mean.contig.length..bp.</th><th scope=col>Longest.contig..bp.</th><th scope=col>GC</th><th scope=col>Coding.density</th><th scope=col>X..predicted.genes</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1002469__metabat2__HighNoCov.011</td><td>/ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/TUK-5projects/LLMGA/v0.12/LLG/rnd1/drep/drep/dereplicated_genomes/1002469__metabat2__HighNoCov.011.fna</td><td>Bacteria</td><td>Firmicutes_A</td><td>Clostridia_A</td><td>Christensenellales</td><td>CAG-138</td><td>Phil1</td><td>Phil1 sp001940855</td><td>98.39</td><td>0</td><td>0</td><td>2229828</td><td>63</td><td>59908</td><td>35394</td><td>166984</td><td>52.8</td><td>90.20</td><td>2136</td></tr>\n",
       "\t<tr><td>1006002__maxbin2__Low.057       </td><td>/ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/TUK-5projects/LLMGA/v0.12/LLG/rnd1/drep/drep/dereplicated_genomes/1006002__maxbin2__Low.057.fna       </td><td>Bacteria</td><td>Firmicutes_A</td><td>Clostridia_A</td><td>Christensenellales</td><td>CAG-138</td><td>Phil1</td><td>Phil1 sp001940855</td><td>99.19</td><td>0</td><td>0</td><td>2183900</td><td>98</td><td>36095</td><td>22284</td><td>109973</td><td>52.6</td><td>89.84</td><td>2113</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>1003478__vamb__Low.002         </td><td>/ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/TUK-5projects/LLMGA/v0.12/LLG/rnd1/drep/drep/dereplicated_genomes/1003478__vamb__Low.002.fna         </td><td>Bacteria</td><td>Firmicutes_A</td><td>Clostridia_A</td><td>Christensenellales</td><td>Borkfalkiaceae</td><td>UBA4636</td><td>unclassified     </td><td>63.46</td><td>0.03</td><td>100</td><td>1294155</td><td>397</td><td> 3402</td><td> 3259</td><td> 16255</td><td>45.5</td><td>88.72</td><td>1454</td></tr>\n",
       "\t<tr><td>1006313__metabat2__LowNoCov.067</td><td>/ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/TUK-5projects/LLMGA/v0.12/LLG/rnd1/drep/drep/dereplicated_genomes/1006313__metabat2__LowNoCov.067.fna</td><td>Bacteria</td><td>Firmicutes_A</td><td>Clostridia_A</td><td>Christensenellales</td><td>CAG-138       </td><td>Phil1  </td><td>Phil1 sp001940855</td><td>99.19</td><td>0.00</td><td>  0</td><td>1924531</td><td> 61</td><td>47515</td><td>31549</td><td>136878</td><td>53.3</td><td>90.10</td><td>1794</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tidytable: 76 × 20\n",
       "\\begin{tabular}{llllllllllllllllllll}\n",
       " Name & Fasta & Domain & Phylum & Class & Order & Family & Genus & Species & Completeness & Contamination & Strain.heterogeneity & Genome.size..bp. & X..contigs & N50..contigs. & Mean.contig.length..bp. & Longest.contig..bp. & GC & Coding.density & X..predicted.genes\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <int> & <int> & <int> & <int> & <int> & <dbl> & <dbl> & <int>\\\\\n",
       "\\hline\n",
       "\t 1002469\\_\\_metabat2\\_\\_HighNoCov.011 & /ebio/abt3\\_projects/Anxiety\\_Twins\\_Metagenomes/data/metagenome/TUK-5projects/LLMGA/v0.12/LLG/rnd1/drep/drep/dereplicated\\_genomes/1002469\\_\\_metabat2\\_\\_HighNoCov.011.fna & Bacteria & Firmicutes\\_A & Clostridia\\_A & Christensenellales & CAG-138 & Phil1 & Phil1 sp001940855 & 98.39 & 0 & 0 & 2229828 & 63 & 59908 & 35394 & 166984 & 52.8 & 90.20 & 2136\\\\\n",
       "\t 1006002\\_\\_maxbin2\\_\\_Low.057        & /ebio/abt3\\_projects/Anxiety\\_Twins\\_Metagenomes/data/metagenome/TUK-5projects/LLMGA/v0.12/LLG/rnd1/drep/drep/dereplicated\\_genomes/1006002\\_\\_maxbin2\\_\\_Low.057.fna        & Bacteria & Firmicutes\\_A & Clostridia\\_A & Christensenellales & CAG-138 & Phil1 & Phil1 sp001940855 & 99.19 & 0 & 0 & 2183900 & 98 & 36095 & 22284 & 109973 & 52.6 & 89.84 & 2113\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t 1003478\\_\\_vamb\\_\\_Low.002          & /ebio/abt3\\_projects/Anxiety\\_Twins\\_Metagenomes/data/metagenome/TUK-5projects/LLMGA/v0.12/LLG/rnd1/drep/drep/dereplicated\\_genomes/1003478\\_\\_vamb\\_\\_Low.002.fna          & Bacteria & Firmicutes\\_A & Clostridia\\_A & Christensenellales & Borkfalkiaceae & UBA4636 & unclassified      & 63.46 & 0.03 & 100 & 1294155 & 397 &  3402 &  3259 &  16255 & 45.5 & 88.72 & 1454\\\\\n",
       "\t 1006313\\_\\_metabat2\\_\\_LowNoCov.067 & /ebio/abt3\\_projects/Anxiety\\_Twins\\_Metagenomes/data/metagenome/TUK-5projects/LLMGA/v0.12/LLG/rnd1/drep/drep/dereplicated\\_genomes/1006313\\_\\_metabat2\\_\\_LowNoCov.067.fna & Bacteria & Firmicutes\\_A & Clostridia\\_A & Christensenellales & CAG-138        & Phil1   & Phil1 sp001940855 & 99.19 & 0.00 &   0 & 1924531 &  61 & 47515 & 31549 & 136878 & 53.3 & 90.10 & 1794\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tidytable: 76 × 20\n",
       "\n",
       "| Name &lt;chr&gt; | Fasta &lt;chr&gt; | Domain &lt;chr&gt; | Phylum &lt;chr&gt; | Class &lt;chr&gt; | Order &lt;chr&gt; | Family &lt;chr&gt; | Genus &lt;chr&gt; | Species &lt;chr&gt; | Completeness &lt;dbl&gt; | Contamination &lt;dbl&gt; | Strain.heterogeneity &lt;dbl&gt; | Genome.size..bp. &lt;int&gt; | X..contigs &lt;int&gt; | N50..contigs. &lt;int&gt; | Mean.contig.length..bp. &lt;int&gt; | Longest.contig..bp. &lt;int&gt; | GC &lt;dbl&gt; | Coding.density &lt;dbl&gt; | X..predicted.genes &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1002469__metabat2__HighNoCov.011 | /ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/TUK-5projects/LLMGA/v0.12/LLG/rnd1/drep/drep/dereplicated_genomes/1002469__metabat2__HighNoCov.011.fna | Bacteria | Firmicutes_A | Clostridia_A | Christensenellales | CAG-138 | Phil1 | Phil1 sp001940855 | 98.39 | 0 | 0 | 2229828 | 63 | 59908 | 35394 | 166984 | 52.8 | 90.20 | 2136 |\n",
       "| 1006002__maxbin2__Low.057        | /ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/TUK-5projects/LLMGA/v0.12/LLG/rnd1/drep/drep/dereplicated_genomes/1006002__maxbin2__Low.057.fna        | Bacteria | Firmicutes_A | Clostridia_A | Christensenellales | CAG-138 | Phil1 | Phil1 sp001940855 | 99.19 | 0 | 0 | 2183900 | 98 | 36095 | 22284 | 109973 | 52.6 | 89.84 | 2113 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| 1003478__vamb__Low.002          | /ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/TUK-5projects/LLMGA/v0.12/LLG/rnd1/drep/drep/dereplicated_genomes/1003478__vamb__Low.002.fna          | Bacteria | Firmicutes_A | Clostridia_A | Christensenellales | Borkfalkiaceae | UBA4636 | unclassified      | 63.46 | 0.03 | 100 | 1294155 | 397 |  3402 |  3259 |  16255 | 45.5 | 88.72 | 1454 |\n",
       "| 1006313__metabat2__LowNoCov.067 | /ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/TUK-5projects/LLMGA/v0.12/LLG/rnd1/drep/drep/dereplicated_genomes/1006313__metabat2__LowNoCov.067.fna | Bacteria | Firmicutes_A | Clostridia_A | Christensenellales | CAG-138        | Phil1   | Phil1 sp001940855 | 99.19 | 0.00 |   0 | 1924531 |  61 | 47515 | 31549 | 136878 | 53.3 | 90.10 | 1794 |\n",
       "\n"
      ],
      "text/plain": [
       "   Name                            \n",
       "1  1002469__metabat2__HighNoCov.011\n",
       "2  1006002__maxbin2__Low.057       \n",
       "⋮  ⋮                               \n",
       "75 1003478__vamb__Low.002          \n",
       "76 1006313__metabat2__LowNoCov.067 \n",
       "   Fasta                                                                                                                                                               \n",
       "1  /ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/TUK-5projects/LLMGA/v0.12/LLG/rnd1/drep/drep/dereplicated_genomes/1002469__metabat2__HighNoCov.011.fna\n",
       "2  /ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/TUK-5projects/LLMGA/v0.12/LLG/rnd1/drep/drep/dereplicated_genomes/1006002__maxbin2__Low.057.fna       \n",
       "⋮  ⋮                                                                                                                                                                   \n",
       "75 /ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/TUK-5projects/LLMGA/v0.12/LLG/rnd1/drep/drep/dereplicated_genomes/1003478__vamb__Low.002.fna          \n",
       "76 /ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/TUK-5projects/LLMGA/v0.12/LLG/rnd1/drep/drep/dereplicated_genomes/1006313__metabat2__LowNoCov.067.fna \n",
       "   Domain   Phylum       Class        Order              Family         Genus  \n",
       "1  Bacteria Firmicutes_A Clostridia_A Christensenellales CAG-138        Phil1  \n",
       "2  Bacteria Firmicutes_A Clostridia_A Christensenellales CAG-138        Phil1  \n",
       "⋮  ⋮        ⋮            ⋮            ⋮                  ⋮              ⋮      \n",
       "75 Bacteria Firmicutes_A Clostridia_A Christensenellales Borkfalkiaceae UBA4636\n",
       "76 Bacteria Firmicutes_A Clostridia_A Christensenellales CAG-138        Phil1  \n",
       "   Species           Completeness Contamination Strain.heterogeneity\n",
       "1  Phil1 sp001940855 98.39        0             0                   \n",
       "2  Phil1 sp001940855 99.19        0             0                   \n",
       "⋮  ⋮                 ⋮            ⋮             ⋮                   \n",
       "75 unclassified      63.46        0.03          100                 \n",
       "76 Phil1 sp001940855 99.19        0.00            0                 \n",
       "   Genome.size..bp. X..contigs N50..contigs. Mean.contig.length..bp.\n",
       "1  2229828          63         59908         35394                  \n",
       "2  2183900          98         36095         22284                  \n",
       "⋮  ⋮                ⋮          ⋮             ⋮                      \n",
       "75 1294155          397         3402          3259                  \n",
       "76 1924531           61        47515         31549                  \n",
       "   Longest.contig..bp. GC   Coding.density X..predicted.genes\n",
       "1  166984              52.8 90.20          2136              \n",
       "2  109973              52.6 89.84          2113              \n",
       "⋮  ⋮                   ⋮    ⋮              ⋮                 \n",
       "75  16255              45.5 88.72          1454              \n",
       "76 136878              53.3 90.10          1794              "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "TUK_f = TUK %>%\n",
    "    filter.(Completeness >= 0.9,\n",
    "            Contamination < 0.05,\n",
    "            X..contigs <= 500) %>%\n",
    "    filter.(Family != 'f__')\n",
    "TUK_f"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-02-23T14:04:21.909399Z",
     "iopub.status.busy": "2021-02-23T14:04:21.905435Z",
     "iopub.status.idle": "2021-02-23T14:04:21.964595Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "76"
      ],
      "text/latex": [
       "76"
      ],
      "text/markdown": [
       "76"
      ],
      "text/plain": [
       "[1] 76"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "copy_file = function(F, base_dir){\n",
    "    outfile = file.path(base_dir, basename(F))\n",
    "    stopifnot(F != outfile)\n",
    "    file.copy(F, outfile)\n",
    "}\n",
    "res = TUK_f$Fasta %>%\n",
    "    lapply(copy_file, base_dir=file.path(work_dir, 'TUK'))\n",
    "res %>% length"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## T1k MAGs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-02-23T14:04:25.485113Z",
     "iopub.status.busy": "2021-02-23T14:04:25.481773Z",
     "iopub.status.idle": "2021-02-23T14:04:25.670245Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 2522 × 20</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Name</th><th scope=col>Fasta</th><th scope=col>Domain</th><th scope=col>Phylum</th><th scope=col>Class</th><th scope=col>Order</th><th scope=col>Family</th><th scope=col>Genus</th><th scope=col>Species</th><th scope=col>Completeness</th><th scope=col>Contamination</th><th scope=col>Strain.heterogeneity</th><th scope=col>Genome.size..bp.</th><th scope=col>X..contigs</th><th scope=col>N50..contigs.</th><th scope=col>Mean.contig.length..bp.</th><th scope=col>Longest.contig..bp.</th><th scope=col>GC</th><th scope=col>Coding.density</th><th scope=col>X..predicted.genes</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>AM140__metabat2__High.003   </td><td>/ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/Tulsa1000/LLMGA_v0.12/HiSeq_R160-163-169/LLG/rnd1/drep/drep/dereplicated_genomes/AM140__metabat2__High.003.fna   </td><td>Bacteria</td><td>Firmicutes_A</td><td>Clostridia   </td><td>Oscillospirales</td><td>Ruminococcaceae</td><td>Ruminococcus_D</td><td>Ruminococcus_D bicirculans</td><td>96.58</td><td>0.00</td><td> 0.0</td><td>2420017</td><td>157</td><td>27447</td><td>15414</td><td>70866</td><td>43.1</td><td>87.97</td><td>2239</td></tr>\n",
       "\t<tr><td>AP416__metabat2__Med.004_sub</td><td>/ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/Tulsa1000/LLMGA_v0.12/HiSeq_R160-163-169/LLG/rnd1/drep/drep/dereplicated_genomes/AP416__metabat2__Med.004_sub.fna</td><td>Bacteria</td><td>Firmicutes_C</td><td>Negativicutes</td><td>Veillonellales </td><td>Dialisteraceae </td><td>Dialister     </td><td>Dialister sp900541485     </td><td>63.98</td><td>3.42</td><td>12.5</td><td> 737017</td><td>148</td><td> 6693</td><td> 4979</td><td>33090</td><td>39.6</td><td>91.84</td><td> 807</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>BB280__vamb__Low.145_sub</td><td>/ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/Tulsa1000/LLMGA_v0.12/HiSeq_R160-163-169/LLG/rnd1/drep/drep/dereplicated_genomes/BB280__vamb__Low.145_sub.fna</td><td>Bacteria</td><td>Firmicutes  </td><td>Bacilli   </td><td>Erysipelotrichales</td><td>Erysipelotrichaceae</td><td>Holdemanella</td><td>Holdemanella sp002299315</td><td>66.51</td><td>0.41</td><td> 0.00</td><td>1299578</td><td>455</td><td> 3036</td><td> 2856</td><td>10694</td><td>34.2</td><td>91.17</td><td>1639</td></tr>\n",
       "\t<tr><td>BB280__vamb__Med.096    </td><td>/ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/Tulsa1000/LLMGA_v0.12/HiSeq_R160-163-169/LLG/rnd1/drep/drep/dereplicated_genomes/BB280__vamb__Med.096.fna    </td><td>Bacteria</td><td>Firmicutes_A</td><td>Clostridia</td><td>Lachnospirales    </td><td>Lachnospiraceae    </td><td>Agathobacter</td><td>Agathobacter faecis     </td><td>99.05</td><td>1.98</td><td>14.29</td><td>2762368</td><td>135</td><td>27606</td><td>20461</td><td>97447</td><td>43.7</td><td>90.78</td><td>2681</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 2522 × 20\n",
       "\\begin{tabular}{llllllllllllllllllll}\n",
       " Name & Fasta & Domain & Phylum & Class & Order & Family & Genus & Species & Completeness & Contamination & Strain.heterogeneity & Genome.size..bp. & X..contigs & N50..contigs. & Mean.contig.length..bp. & Longest.contig..bp. & GC & Coding.density & X..predicted.genes\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <int> & <int> & <int> & <int> & <int> & <dbl> & <dbl> & <int>\\\\\n",
       "\\hline\n",
       "\t AM140\\_\\_metabat2\\_\\_High.003    & /ebio/abt3\\_projects/Anxiety\\_Twins\\_Metagenomes/data/metagenome/Tulsa1000/LLMGA\\_v0.12/HiSeq\\_R160-163-169/LLG/rnd1/drep/drep/dereplicated\\_genomes/AM140\\_\\_metabat2\\_\\_High.003.fna    & Bacteria & Firmicutes\\_A & Clostridia    & Oscillospirales & Ruminococcaceae & Ruminococcus\\_D & Ruminococcus\\_D bicirculans & 96.58 & 0.00 &  0.0 & 2420017 & 157 & 27447 & 15414 & 70866 & 43.1 & 87.97 & 2239\\\\\n",
       "\t AP416\\_\\_metabat2\\_\\_Med.004\\_sub & /ebio/abt3\\_projects/Anxiety\\_Twins\\_Metagenomes/data/metagenome/Tulsa1000/LLMGA\\_v0.12/HiSeq\\_R160-163-169/LLG/rnd1/drep/drep/dereplicated\\_genomes/AP416\\_\\_metabat2\\_\\_Med.004\\_sub.fna & Bacteria & Firmicutes\\_C & Negativicutes & Veillonellales  & Dialisteraceae  & Dialister      & Dialister sp900541485      & 63.98 & 3.42 & 12.5 &  737017 & 148 &  6693 &  4979 & 33090 & 39.6 & 91.84 &  807\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t BB280\\_\\_vamb\\_\\_Low.145\\_sub & /ebio/abt3\\_projects/Anxiety\\_Twins\\_Metagenomes/data/metagenome/Tulsa1000/LLMGA\\_v0.12/HiSeq\\_R160-163-169/LLG/rnd1/drep/drep/dereplicated\\_genomes/BB280\\_\\_vamb\\_\\_Low.145\\_sub.fna & Bacteria & Firmicutes   & Bacilli    & Erysipelotrichales & Erysipelotrichaceae & Holdemanella & Holdemanella sp002299315 & 66.51 & 0.41 &  0.00 & 1299578 & 455 &  3036 &  2856 & 10694 & 34.2 & 91.17 & 1639\\\\\n",
       "\t BB280\\_\\_vamb\\_\\_Med.096     & /ebio/abt3\\_projects/Anxiety\\_Twins\\_Metagenomes/data/metagenome/Tulsa1000/LLMGA\\_v0.12/HiSeq\\_R160-163-169/LLG/rnd1/drep/drep/dereplicated\\_genomes/BB280\\_\\_vamb\\_\\_Med.096.fna     & Bacteria & Firmicutes\\_A & Clostridia & Lachnospirales     & Lachnospiraceae     & Agathobacter & Agathobacter faecis      & 99.05 & 1.98 & 14.29 & 2762368 & 135 & 27606 & 20461 & 97447 & 43.7 & 90.78 & 2681\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 2522 × 20\n",
       "\n",
       "| Name &lt;chr&gt; | Fasta &lt;chr&gt; | Domain &lt;chr&gt; | Phylum &lt;chr&gt; | Class &lt;chr&gt; | Order &lt;chr&gt; | Family &lt;chr&gt; | Genus &lt;chr&gt; | Species &lt;chr&gt; | Completeness &lt;dbl&gt; | Contamination &lt;dbl&gt; | Strain.heterogeneity &lt;dbl&gt; | Genome.size..bp. &lt;int&gt; | X..contigs &lt;int&gt; | N50..contigs. &lt;int&gt; | Mean.contig.length..bp. &lt;int&gt; | Longest.contig..bp. &lt;int&gt; | GC &lt;dbl&gt; | Coding.density &lt;dbl&gt; | X..predicted.genes &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| AM140__metabat2__High.003    | /ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/Tulsa1000/LLMGA_v0.12/HiSeq_R160-163-169/LLG/rnd1/drep/drep/dereplicated_genomes/AM140__metabat2__High.003.fna    | Bacteria | Firmicutes_A | Clostridia    | Oscillospirales | Ruminococcaceae | Ruminococcus_D | Ruminococcus_D bicirculans | 96.58 | 0.00 |  0.0 | 2420017 | 157 | 27447 | 15414 | 70866 | 43.1 | 87.97 | 2239 |\n",
       "| AP416__metabat2__Med.004_sub | /ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/Tulsa1000/LLMGA_v0.12/HiSeq_R160-163-169/LLG/rnd1/drep/drep/dereplicated_genomes/AP416__metabat2__Med.004_sub.fna | Bacteria | Firmicutes_C | Negativicutes | Veillonellales  | Dialisteraceae  | Dialister      | Dialister sp900541485      | 63.98 | 3.42 | 12.5 |  737017 | 148 |  6693 |  4979 | 33090 | 39.6 | 91.84 |  807 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| BB280__vamb__Low.145_sub | /ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/Tulsa1000/LLMGA_v0.12/HiSeq_R160-163-169/LLG/rnd1/drep/drep/dereplicated_genomes/BB280__vamb__Low.145_sub.fna | Bacteria | Firmicutes   | Bacilli    | Erysipelotrichales | Erysipelotrichaceae | Holdemanella | Holdemanella sp002299315 | 66.51 | 0.41 |  0.00 | 1299578 | 455 |  3036 |  2856 | 10694 | 34.2 | 91.17 | 1639 |\n",
       "| BB280__vamb__Med.096     | /ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/Tulsa1000/LLMGA_v0.12/HiSeq_R160-163-169/LLG/rnd1/drep/drep/dereplicated_genomes/BB280__vamb__Med.096.fna     | Bacteria | Firmicutes_A | Clostridia | Lachnospirales     | Lachnospiraceae     | Agathobacter | Agathobacter faecis      | 99.05 | 1.98 | 14.29 | 2762368 | 135 | 27606 | 20461 | 97447 | 43.7 | 90.78 | 2681 |\n",
       "\n"
      ],
      "text/plain": [
       "     Name                        \n",
       "1    AM140__metabat2__High.003   \n",
       "2    AP416__metabat2__Med.004_sub\n",
       "⋮    ⋮                           \n",
       "2521 BB280__vamb__Low.145_sub    \n",
       "2522 BB280__vamb__Med.096        \n",
       "     Fasta                                                                                                                                                                          \n",
       "1    /ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/Tulsa1000/LLMGA_v0.12/HiSeq_R160-163-169/LLG/rnd1/drep/drep/dereplicated_genomes/AM140__metabat2__High.003.fna   \n",
       "2    /ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/Tulsa1000/LLMGA_v0.12/HiSeq_R160-163-169/LLG/rnd1/drep/drep/dereplicated_genomes/AP416__metabat2__Med.004_sub.fna\n",
       "⋮    ⋮                                                                                                                                                                              \n",
       "2521 /ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/Tulsa1000/LLMGA_v0.12/HiSeq_R160-163-169/LLG/rnd1/drep/drep/dereplicated_genomes/BB280__vamb__Low.145_sub.fna    \n",
       "2522 /ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/Tulsa1000/LLMGA_v0.12/HiSeq_R160-163-169/LLG/rnd1/drep/drep/dereplicated_genomes/BB280__vamb__Med.096.fna        \n",
       "     Domain   Phylum       Class         Order              Family             \n",
       "1    Bacteria Firmicutes_A Clostridia    Oscillospirales    Ruminococcaceae    \n",
       "2    Bacteria Firmicutes_C Negativicutes Veillonellales     Dialisteraceae     \n",
       "⋮    ⋮        ⋮            ⋮             ⋮                  ⋮                  \n",
       "2521 Bacteria Firmicutes   Bacilli       Erysipelotrichales Erysipelotrichaceae\n",
       "2522 Bacteria Firmicutes_A Clostridia    Lachnospirales     Lachnospiraceae    \n",
       "     Genus          Species                    Completeness Contamination\n",
       "1    Ruminococcus_D Ruminococcus_D bicirculans 96.58        0.00         \n",
       "2    Dialister      Dialister sp900541485      63.98        3.42         \n",
       "⋮    ⋮              ⋮                          ⋮            ⋮            \n",
       "2521 Holdemanella   Holdemanella sp002299315   66.51        0.41         \n",
       "2522 Agathobacter   Agathobacter faecis        99.05        1.98         \n",
       "     Strain.heterogeneity Genome.size..bp. X..contigs N50..contigs.\n",
       "1     0.0                 2420017          157        27447        \n",
       "2    12.5                  737017          148         6693        \n",
       "⋮    ⋮                    ⋮                ⋮          ⋮            \n",
       "2521  0.00                1299578          455         3036        \n",
       "2522 14.29                2762368          135        27606        \n",
       "     Mean.contig.length..bp. Longest.contig..bp. GC   Coding.density\n",
       "1    15414                   70866               43.1 87.97         \n",
       "2     4979                   33090               39.6 91.84         \n",
       "⋮    ⋮                       ⋮                   ⋮    ⋮             \n",
       "2521  2856                   10694               34.2 91.17         \n",
       "2522 20461                   97447               43.7 90.78         \n",
       "     X..predicted.genes\n",
       "1    2239              \n",
       "2     807              \n",
       "⋮    ⋮                 \n",
       "2521 1639              \n",
       "2522 2681              "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "F =  '/ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/Tulsa1000/LLMGA_v0.12/HiSeq_R160-163-169/LLG//rnd1/final_MAGs.tsv'\n",
    "T1k = Fread(F)\n",
    "T1k"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-02-23T14:04:27.563450Z",
     "iopub.status.busy": "2021-02-23T14:04:27.561576Z",
     "iopub.status.idle": "2021-02-23T14:04:27.623681Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tidytable: 33 × 20</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Name</th><th scope=col>Fasta</th><th scope=col>Domain</th><th scope=col>Phylum</th><th scope=col>Class</th><th scope=col>Order</th><th scope=col>Family</th><th scope=col>Genus</th><th scope=col>Species</th><th scope=col>Completeness</th><th scope=col>Contamination</th><th scope=col>Strain.heterogeneity</th><th scope=col>Genome.size..bp.</th><th scope=col>X..contigs</th><th scope=col>N50..contigs.</th><th scope=col>Mean.contig.length..bp.</th><th scope=col>Longest.contig..bp.</th><th scope=col>GC</th><th scope=col>Coding.density</th><th scope=col>X..predicted.genes</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>AT573__metabat2__HighNoCov.002</td><td>/ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/Tulsa1000/LLMGA_v0.12/HiSeq_R160-163-169/LLG/rnd1/drep/drep/dereplicated_genomes/AT573__metabat2__HighNoCov.002.fna</td><td>Bacteria</td><td>Firmicutes_A</td><td>Clostridia_A</td><td>Christensenellales</td><td>CAG-138</td><td>Phil1  </td><td>Phil1 sp001940855  </td><td>99.19</td><td>0.00</td><td>0</td><td>2369272</td><td>65</td><td>66039</td><td>36450</td><td>232224</td><td>52.5</td><td>90.05</td><td>2292</td></tr>\n",
       "\t<tr><td>AX681__metabat2__Med.017      </td><td>/ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/Tulsa1000/LLMGA_v0.12/HiSeq_R160-163-169/LLG/rnd1/drep/drep/dereplicated_genomes/AX681__metabat2__Med.017.fna      </td><td>Bacteria</td><td>Firmicutes_A</td><td>Clostridia_A</td><td>Christensenellales</td><td>CAG-314</td><td>CAG-314</td><td>CAG-314 sp000437915</td><td>93.95</td><td>1.61</td><td>0</td><td>1443486</td><td>45</td><td>70685</td><td>32077</td><td>148713</td><td>49.2</td><td>90.72</td><td>1298</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>AT573__maxbin2__Low.011_sub</td><td>/ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/Tulsa1000/LLMGA_v0.12/HiSeq_R160-163-169/LLG/rnd1/drep/drep/dereplicated_genomes/AT573__maxbin2__Low.011_sub.fna</td><td>Bacteria</td><td>Firmicutes_A</td><td>Clostridia_A</td><td>Christensenellales</td><td>CAG-917</td><td>CAG-349</td><td>CAG-349 sp003539515</td><td>92.40</td><td>2.10</td><td>0</td><td>1769393</td><td>27</td><td>366008</td><td>65533</td><td>577376</td><td>47.3</td><td>90.06</td><td>1608</td></tr>\n",
       "\t<tr><td>AV958__metabat2__Med.056   </td><td>/ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/Tulsa1000/LLMGA_v0.12/HiSeq_R160-163-169/LLG/rnd1/drep/drep/dereplicated_genomes/AV958__metabat2__Med.056.fna   </td><td>Bacteria</td><td>Firmicutes_A</td><td>Clostridia_A</td><td>Christensenellales</td><td>CAG-314</td><td>CAG-314</td><td>CAG-314 sp000437915</td><td>93.95</td><td>0.81</td><td>0</td><td>1588990</td><td>28</td><td>114674</td><td>56749</td><td>246228</td><td>48.4</td><td>90.25</td><td>1415</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tidytable: 33 × 20\n",
       "\\begin{tabular}{llllllllllllllllllll}\n",
       " Name & Fasta & Domain & Phylum & Class & Order & Family & Genus & Species & Completeness & Contamination & Strain.heterogeneity & Genome.size..bp. & X..contigs & N50..contigs. & Mean.contig.length..bp. & Longest.contig..bp. & GC & Coding.density & X..predicted.genes\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <int> & <int> & <int> & <int> & <int> & <dbl> & <dbl> & <int>\\\\\n",
       "\\hline\n",
       "\t AT573\\_\\_metabat2\\_\\_HighNoCov.002 & /ebio/abt3\\_projects/Anxiety\\_Twins\\_Metagenomes/data/metagenome/Tulsa1000/LLMGA\\_v0.12/HiSeq\\_R160-163-169/LLG/rnd1/drep/drep/dereplicated\\_genomes/AT573\\_\\_metabat2\\_\\_HighNoCov.002.fna & Bacteria & Firmicutes\\_A & Clostridia\\_A & Christensenellales & CAG-138 & Phil1   & Phil1 sp001940855   & 99.19 & 0.00 & 0 & 2369272 & 65 & 66039 & 36450 & 232224 & 52.5 & 90.05 & 2292\\\\\n",
       "\t AX681\\_\\_metabat2\\_\\_Med.017       & /ebio/abt3\\_projects/Anxiety\\_Twins\\_Metagenomes/data/metagenome/Tulsa1000/LLMGA\\_v0.12/HiSeq\\_R160-163-169/LLG/rnd1/drep/drep/dereplicated\\_genomes/AX681\\_\\_metabat2\\_\\_Med.017.fna       & Bacteria & Firmicutes\\_A & Clostridia\\_A & Christensenellales & CAG-314 & CAG-314 & CAG-314 sp000437915 & 93.95 & 1.61 & 0 & 1443486 & 45 & 70685 & 32077 & 148713 & 49.2 & 90.72 & 1298\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t AT573\\_\\_maxbin2\\_\\_Low.011\\_sub & /ebio/abt3\\_projects/Anxiety\\_Twins\\_Metagenomes/data/metagenome/Tulsa1000/LLMGA\\_v0.12/HiSeq\\_R160-163-169/LLG/rnd1/drep/drep/dereplicated\\_genomes/AT573\\_\\_maxbin2\\_\\_Low.011\\_sub.fna & Bacteria & Firmicutes\\_A & Clostridia\\_A & Christensenellales & CAG-917 & CAG-349 & CAG-349 sp003539515 & 92.40 & 2.10 & 0 & 1769393 & 27 & 366008 & 65533 & 577376 & 47.3 & 90.06 & 1608\\\\\n",
       "\t AV958\\_\\_metabat2\\_\\_Med.056    & /ebio/abt3\\_projects/Anxiety\\_Twins\\_Metagenomes/data/metagenome/Tulsa1000/LLMGA\\_v0.12/HiSeq\\_R160-163-169/LLG/rnd1/drep/drep/dereplicated\\_genomes/AV958\\_\\_metabat2\\_\\_Med.056.fna    & Bacteria & Firmicutes\\_A & Clostridia\\_A & Christensenellales & CAG-314 & CAG-314 & CAG-314 sp000437915 & 93.95 & 0.81 & 0 & 1588990 & 28 & 114674 & 56749 & 246228 & 48.4 & 90.25 & 1415\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tidytable: 33 × 20\n",
       "\n",
       "| Name &lt;chr&gt; | Fasta &lt;chr&gt; | Domain &lt;chr&gt; | Phylum &lt;chr&gt; | Class &lt;chr&gt; | Order &lt;chr&gt; | Family &lt;chr&gt; | Genus &lt;chr&gt; | Species &lt;chr&gt; | Completeness &lt;dbl&gt; | Contamination &lt;dbl&gt; | Strain.heterogeneity &lt;dbl&gt; | Genome.size..bp. &lt;int&gt; | X..contigs &lt;int&gt; | N50..contigs. &lt;int&gt; | Mean.contig.length..bp. &lt;int&gt; | Longest.contig..bp. &lt;int&gt; | GC &lt;dbl&gt; | Coding.density &lt;dbl&gt; | X..predicted.genes &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| AT573__metabat2__HighNoCov.002 | /ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/Tulsa1000/LLMGA_v0.12/HiSeq_R160-163-169/LLG/rnd1/drep/drep/dereplicated_genomes/AT573__metabat2__HighNoCov.002.fna | Bacteria | Firmicutes_A | Clostridia_A | Christensenellales | CAG-138 | Phil1   | Phil1 sp001940855   | 99.19 | 0.00 | 0 | 2369272 | 65 | 66039 | 36450 | 232224 | 52.5 | 90.05 | 2292 |\n",
       "| AX681__metabat2__Med.017       | /ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/Tulsa1000/LLMGA_v0.12/HiSeq_R160-163-169/LLG/rnd1/drep/drep/dereplicated_genomes/AX681__metabat2__Med.017.fna       | Bacteria | Firmicutes_A | Clostridia_A | Christensenellales | CAG-314 | CAG-314 | CAG-314 sp000437915 | 93.95 | 1.61 | 0 | 1443486 | 45 | 70685 | 32077 | 148713 | 49.2 | 90.72 | 1298 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| AT573__maxbin2__Low.011_sub | /ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/Tulsa1000/LLMGA_v0.12/HiSeq_R160-163-169/LLG/rnd1/drep/drep/dereplicated_genomes/AT573__maxbin2__Low.011_sub.fna | Bacteria | Firmicutes_A | Clostridia_A | Christensenellales | CAG-917 | CAG-349 | CAG-349 sp003539515 | 92.40 | 2.10 | 0 | 1769393 | 27 | 366008 | 65533 | 577376 | 47.3 | 90.06 | 1608 |\n",
       "| AV958__metabat2__Med.056    | /ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/Tulsa1000/LLMGA_v0.12/HiSeq_R160-163-169/LLG/rnd1/drep/drep/dereplicated_genomes/AV958__metabat2__Med.056.fna    | Bacteria | Firmicutes_A | Clostridia_A | Christensenellales | CAG-314 | CAG-314 | CAG-314 sp000437915 | 93.95 | 0.81 | 0 | 1588990 | 28 | 114674 | 56749 | 246228 | 48.4 | 90.25 | 1415 |\n",
       "\n"
      ],
      "text/plain": [
       "   Name                          \n",
       "1  AT573__metabat2__HighNoCov.002\n",
       "2  AX681__metabat2__Med.017      \n",
       "⋮  ⋮                             \n",
       "32 AT573__maxbin2__Low.011_sub   \n",
       "33 AV958__metabat2__Med.056      \n",
       "   Fasta                                                                                                                                                                            \n",
       "1  /ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/Tulsa1000/LLMGA_v0.12/HiSeq_R160-163-169/LLG/rnd1/drep/drep/dereplicated_genomes/AT573__metabat2__HighNoCov.002.fna\n",
       "2  /ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/Tulsa1000/LLMGA_v0.12/HiSeq_R160-163-169/LLG/rnd1/drep/drep/dereplicated_genomes/AX681__metabat2__Med.017.fna      \n",
       "⋮  ⋮                                                                                                                                                                                \n",
       "32 /ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/Tulsa1000/LLMGA_v0.12/HiSeq_R160-163-169/LLG/rnd1/drep/drep/dereplicated_genomes/AT573__maxbin2__Low.011_sub.fna   \n",
       "33 /ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/Tulsa1000/LLMGA_v0.12/HiSeq_R160-163-169/LLG/rnd1/drep/drep/dereplicated_genomes/AV958__metabat2__Med.056.fna      \n",
       "   Domain   Phylum       Class        Order              Family  Genus  \n",
       "1  Bacteria Firmicutes_A Clostridia_A Christensenellales CAG-138 Phil1  \n",
       "2  Bacteria Firmicutes_A Clostridia_A Christensenellales CAG-314 CAG-314\n",
       "⋮  ⋮        ⋮            ⋮            ⋮                  ⋮       ⋮      \n",
       "32 Bacteria Firmicutes_A Clostridia_A Christensenellales CAG-917 CAG-349\n",
       "33 Bacteria Firmicutes_A Clostridia_A Christensenellales CAG-314 CAG-314\n",
       "   Species             Completeness Contamination Strain.heterogeneity\n",
       "1  Phil1 sp001940855   99.19        0.00          0                   \n",
       "2  CAG-314 sp000437915 93.95        1.61          0                   \n",
       "⋮  ⋮                   ⋮            ⋮             ⋮                   \n",
       "32 CAG-349 sp003539515 92.40        2.10          0                   \n",
       "33 CAG-314 sp000437915 93.95        0.81          0                   \n",
       "   Genome.size..bp. X..contigs N50..contigs. Mean.contig.length..bp.\n",
       "1  2369272          65         66039         36450                  \n",
       "2  1443486          45         70685         32077                  \n",
       "⋮  ⋮                ⋮          ⋮             ⋮                      \n",
       "32 1769393          27         366008        65533                  \n",
       "33 1588990          28         114674        56749                  \n",
       "   Longest.contig..bp. GC   Coding.density X..predicted.genes\n",
       "1  232224              52.5 90.05          2292              \n",
       "2  148713              49.2 90.72          1298              \n",
       "⋮  ⋮                   ⋮    ⋮              ⋮                 \n",
       "32 577376              47.3 90.06          1608              \n",
       "33 246228              48.4 90.25          1415              "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "T1k = T1k %>%\n",
    "    filter.(Order == 'Christensenellales')\n",
    "T1k"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-02-23T14:04:45.925676Z",
     "iopub.status.busy": "2021-02-23T14:04:45.923146Z",
     "iopub.status.idle": "2021-02-23T14:04:45.982046Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tidytable: 6 × 20</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Name</th><th scope=col>Fasta</th><th scope=col>Domain</th><th scope=col>Phylum</th><th scope=col>Class</th><th scope=col>Order</th><th scope=col>Family</th><th scope=col>Genus</th><th scope=col>Species</th><th scope=col>Completeness</th><th scope=col>Contamination</th><th scope=col>Strain.heterogeneity</th><th scope=col>Genome.size..bp.</th><th scope=col>X..contigs</th><th scope=col>N50..contigs.</th><th scope=col>Mean.contig.length..bp.</th><th scope=col>Longest.contig..bp.</th><th scope=col>GC</th><th scope=col>Coding.density</th><th scope=col>X..predicted.genes</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>AT573__metabat2__HighNoCov.002</td><td>/ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/Tulsa1000/LLMGA_v0.12/HiSeq_R160-163-169/LLG/rnd1/drep/drep/dereplicated_genomes/AT573__metabat2__HighNoCov.002.fna</td><td>Bacteria</td><td>Firmicutes_A</td><td>Clostridia_A</td><td>Christensenellales</td><td>CAG-138</td><td>Phil1  </td><td>Phil1 sp001940855  </td><td>99.19</td><td>0</td><td>0</td><td>2369272</td><td>65</td><td>66039</td><td>36450</td><td>232224</td><td>52.5</td><td>90.05</td><td>2292</td></tr>\n",
       "\t<tr><td>AW108__metabat2__HighNoCov.004</td><td>/ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/Tulsa1000/LLMGA_v0.12/HiSeq_R160-163-169/LLG/rnd1/drep/drep/dereplicated_genomes/AW108__metabat2__HighNoCov.004.fna</td><td>Bacteria</td><td>Firmicutes_A</td><td>Clostridia_A</td><td>Christensenellales</td><td>CAG-917</td><td>CAG-475</td><td>CAG-475 sp000434435</td><td>83.56</td><td>0</td><td>0</td><td>1563909</td><td>41</td><td>50516</td><td>38144</td><td>148342</td><td>46.8</td><td>87.09</td><td>1298</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>AS988__metabat2__MedNoCov.013 </td><td>/ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/Tulsa1000/LLMGA_v0.12/HiSeq_R160-163-169/LLG/rnd1/drep/drep/dereplicated_genomes/AS988__metabat2__MedNoCov.013.fna </td><td>Bacteria</td><td>Firmicutes_A</td><td>Clostridia_A</td><td>Christensenellales</td><td>CAG-314</td><td>CAG-314</td><td>                   </td><td>83.47</td><td>0.03</td><td>0</td><td>1390176</td><td>184</td><td> 11215</td><td> 7555</td><td> 46136</td><td>49.3</td><td>89.24</td><td>1345</td></tr>\n",
       "\t<tr><td>AR644__metabat2__HighNoCov.010</td><td>/ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/Tulsa1000/LLMGA_v0.12/HiSeq_R160-163-169/LLG/rnd1/drep/drep/dereplicated_genomes/AR644__metabat2__HighNoCov.010.fna</td><td>Bacteria</td><td>Firmicutes_A</td><td>Clostridia_A</td><td>Christensenellales</td><td>CAG-917</td><td>CAG-475</td><td>CAG-475 sp000434435</td><td>85.46</td><td>0.00</td><td>0</td><td>1569791</td><td> 26</td><td>102156</td><td>60376</td><td>263332</td><td>46.4</td><td>86.72</td><td>1327</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tidytable: 6 × 20\n",
       "\\begin{tabular}{llllllllllllllllllll}\n",
       " Name & Fasta & Domain & Phylum & Class & Order & Family & Genus & Species & Completeness & Contamination & Strain.heterogeneity & Genome.size..bp. & X..contigs & N50..contigs. & Mean.contig.length..bp. & Longest.contig..bp. & GC & Coding.density & X..predicted.genes\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <int> & <int> & <int> & <int> & <int> & <dbl> & <dbl> & <int>\\\\\n",
       "\\hline\n",
       "\t AT573\\_\\_metabat2\\_\\_HighNoCov.002 & /ebio/abt3\\_projects/Anxiety\\_Twins\\_Metagenomes/data/metagenome/Tulsa1000/LLMGA\\_v0.12/HiSeq\\_R160-163-169/LLG/rnd1/drep/drep/dereplicated\\_genomes/AT573\\_\\_metabat2\\_\\_HighNoCov.002.fna & Bacteria & Firmicutes\\_A & Clostridia\\_A & Christensenellales & CAG-138 & Phil1   & Phil1 sp001940855   & 99.19 & 0 & 0 & 2369272 & 65 & 66039 & 36450 & 232224 & 52.5 & 90.05 & 2292\\\\\n",
       "\t AW108\\_\\_metabat2\\_\\_HighNoCov.004 & /ebio/abt3\\_projects/Anxiety\\_Twins\\_Metagenomes/data/metagenome/Tulsa1000/LLMGA\\_v0.12/HiSeq\\_R160-163-169/LLG/rnd1/drep/drep/dereplicated\\_genomes/AW108\\_\\_metabat2\\_\\_HighNoCov.004.fna & Bacteria & Firmicutes\\_A & Clostridia\\_A & Christensenellales & CAG-917 & CAG-475 & CAG-475 sp000434435 & 83.56 & 0 & 0 & 1563909 & 41 & 50516 & 38144 & 148342 & 46.8 & 87.09 & 1298\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t AS988\\_\\_metabat2\\_\\_MedNoCov.013  & /ebio/abt3\\_projects/Anxiety\\_Twins\\_Metagenomes/data/metagenome/Tulsa1000/LLMGA\\_v0.12/HiSeq\\_R160-163-169/LLG/rnd1/drep/drep/dereplicated\\_genomes/AS988\\_\\_metabat2\\_\\_MedNoCov.013.fna  & Bacteria & Firmicutes\\_A & Clostridia\\_A & Christensenellales & CAG-314 & CAG-314 &                     & 83.47 & 0.03 & 0 & 1390176 & 184 &  11215 &  7555 &  46136 & 49.3 & 89.24 & 1345\\\\\n",
       "\t AR644\\_\\_metabat2\\_\\_HighNoCov.010 & /ebio/abt3\\_projects/Anxiety\\_Twins\\_Metagenomes/data/metagenome/Tulsa1000/LLMGA\\_v0.12/HiSeq\\_R160-163-169/LLG/rnd1/drep/drep/dereplicated\\_genomes/AR644\\_\\_metabat2\\_\\_HighNoCov.010.fna & Bacteria & Firmicutes\\_A & Clostridia\\_A & Christensenellales & CAG-917 & CAG-475 & CAG-475 sp000434435 & 85.46 & 0.00 & 0 & 1569791 &  26 & 102156 & 60376 & 263332 & 46.4 & 86.72 & 1327\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tidytable: 6 × 20\n",
       "\n",
       "| Name &lt;chr&gt; | Fasta &lt;chr&gt; | Domain &lt;chr&gt; | Phylum &lt;chr&gt; | Class &lt;chr&gt; | Order &lt;chr&gt; | Family &lt;chr&gt; | Genus &lt;chr&gt; | Species &lt;chr&gt; | Completeness &lt;dbl&gt; | Contamination &lt;dbl&gt; | Strain.heterogeneity &lt;dbl&gt; | Genome.size..bp. &lt;int&gt; | X..contigs &lt;int&gt; | N50..contigs. &lt;int&gt; | Mean.contig.length..bp. &lt;int&gt; | Longest.contig..bp. &lt;int&gt; | GC &lt;dbl&gt; | Coding.density &lt;dbl&gt; | X..predicted.genes &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| AT573__metabat2__HighNoCov.002 | /ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/Tulsa1000/LLMGA_v0.12/HiSeq_R160-163-169/LLG/rnd1/drep/drep/dereplicated_genomes/AT573__metabat2__HighNoCov.002.fna | Bacteria | Firmicutes_A | Clostridia_A | Christensenellales | CAG-138 | Phil1   | Phil1 sp001940855   | 99.19 | 0 | 0 | 2369272 | 65 | 66039 | 36450 | 232224 | 52.5 | 90.05 | 2292 |\n",
       "| AW108__metabat2__HighNoCov.004 | /ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/Tulsa1000/LLMGA_v0.12/HiSeq_R160-163-169/LLG/rnd1/drep/drep/dereplicated_genomes/AW108__metabat2__HighNoCov.004.fna | Bacteria | Firmicutes_A | Clostridia_A | Christensenellales | CAG-917 | CAG-475 | CAG-475 sp000434435 | 83.56 | 0 | 0 | 1563909 | 41 | 50516 | 38144 | 148342 | 46.8 | 87.09 | 1298 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| AS988__metabat2__MedNoCov.013  | /ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/Tulsa1000/LLMGA_v0.12/HiSeq_R160-163-169/LLG/rnd1/drep/drep/dereplicated_genomes/AS988__metabat2__MedNoCov.013.fna  | Bacteria | Firmicutes_A | Clostridia_A | Christensenellales | CAG-314 | CAG-314 | <!----> | 83.47 | 0.03 | 0 | 1390176 | 184 |  11215 |  7555 |  46136 | 49.3 | 89.24 | 1345 |\n",
       "| AR644__metabat2__HighNoCov.010 | /ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/Tulsa1000/LLMGA_v0.12/HiSeq_R160-163-169/LLG/rnd1/drep/drep/dereplicated_genomes/AR644__metabat2__HighNoCov.010.fna | Bacteria | Firmicutes_A | Clostridia_A | Christensenellales | CAG-917 | CAG-475 | CAG-475 sp000434435 | 85.46 | 0.00 | 0 | 1569791 |  26 | 102156 | 60376 | 263332 | 46.4 | 86.72 | 1327 |\n",
       "\n"
      ],
      "text/plain": [
       "  Name                          \n",
       "1 AT573__metabat2__HighNoCov.002\n",
       "2 AW108__metabat2__HighNoCov.004\n",
       "⋮ ⋮                             \n",
       "5 AS988__metabat2__MedNoCov.013 \n",
       "6 AR644__metabat2__HighNoCov.010\n",
       "  Fasta                                                                                                                                                                            \n",
       "1 /ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/Tulsa1000/LLMGA_v0.12/HiSeq_R160-163-169/LLG/rnd1/drep/drep/dereplicated_genomes/AT573__metabat2__HighNoCov.002.fna\n",
       "2 /ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/Tulsa1000/LLMGA_v0.12/HiSeq_R160-163-169/LLG/rnd1/drep/drep/dereplicated_genomes/AW108__metabat2__HighNoCov.004.fna\n",
       "⋮ ⋮                                                                                                                                                                                \n",
       "5 /ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/Tulsa1000/LLMGA_v0.12/HiSeq_R160-163-169/LLG/rnd1/drep/drep/dereplicated_genomes/AS988__metabat2__MedNoCov.013.fna \n",
       "6 /ebio/abt3_projects/Anxiety_Twins_Metagenomes/data/metagenome/Tulsa1000/LLMGA_v0.12/HiSeq_R160-163-169/LLG/rnd1/drep/drep/dereplicated_genomes/AR644__metabat2__HighNoCov.010.fna\n",
       "  Domain   Phylum       Class        Order              Family  Genus  \n",
       "1 Bacteria Firmicutes_A Clostridia_A Christensenellales CAG-138 Phil1  \n",
       "2 Bacteria Firmicutes_A Clostridia_A Christensenellales CAG-917 CAG-475\n",
       "⋮ ⋮        ⋮            ⋮            ⋮                  ⋮       ⋮      \n",
       "5 Bacteria Firmicutes_A Clostridia_A Christensenellales CAG-314 CAG-314\n",
       "6 Bacteria Firmicutes_A Clostridia_A Christensenellales CAG-917 CAG-475\n",
       "  Species             Completeness Contamination Strain.heterogeneity\n",
       "1 Phil1 sp001940855   99.19        0             0                   \n",
       "2 CAG-475 sp000434435 83.56        0             0                   \n",
       "⋮ ⋮                   ⋮            ⋮             ⋮                   \n",
       "5                     83.47        0.03          0                   \n",
       "6 CAG-475 sp000434435 85.46        0.00          0                   \n",
       "  Genome.size..bp. X..contigs N50..contigs. Mean.contig.length..bp.\n",
       "1 2369272          65         66039         36450                  \n",
       "2 1563909          41         50516         38144                  \n",
       "⋮ ⋮                ⋮          ⋮             ⋮                      \n",
       "5 1390176          184         11215         7555                  \n",
       "6 1569791           26        102156        60376                  \n",
       "  Longest.contig..bp. GC   Coding.density X..predicted.genes\n",
       "1 232224              52.5 90.05          2292              \n",
       "2 148342              46.8 87.09          1298              \n",
       "⋮ ⋮                   ⋮    ⋮              ⋮                 \n",
       "5  46136              49.3 89.24          1345              \n",
       "6 263332              46.4 86.72          1327              "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "T1k_f = T1k %>%\n",
    "    filter.(Completeness >= 0.9,\n",
    "            Contamination < 0.05,\n",
    "            X..contigs <= 500) %>%\n",
    "    filter.(Family != 'f__')\n",
    "T1k_f"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-02-23T14:05:18.585459Z",
     "iopub.status.busy": "2021-02-23T14:05:18.583135Z",
     "iopub.status.idle": "2021-02-23T14:05:18.643166Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "6"
      ],
      "text/latex": [
       "6"
      ],
      "text/markdown": [
       "6"
      ],
      "text/plain": [
       "[1] 6"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# copying files\n",
    "res = T1k_f$Fasta %>%\n",
    "    lapply(copy_file, base_dir=file.path(work_dir, 'T1k'))\n",
    "res %>% length"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## List of all genomes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-02-26T12:25:59.652152Z",
     "iopub.status.busy": "2021-02-26T12:25:59.648421Z",
     "iopub.status.idle": "2021-02-26T12:25:59.807222Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 731 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Name</th><th scope=col>Fasta</th><th scope=col>Domain</th><th scope=col>Taxid</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>MI-001-H__metabat2_high_PE.003.fna.gz</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/Colombian_MG/MI-001-H__metabat2_high_PE.003.fna.gz</td><td>Bacteria</td><td>990719</td></tr>\n",
       "\t<tr><td>MI-002-H__metabat2_high_PE.005.fna.gz</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/Colombian_MG/MI-002-H__metabat2_high_PE.005.fna.gz</td><td>Bacteria</td><td>990719</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>GUT_GENOME286615.fna.gz</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/UHGG/GUT_GENOME286615.fna.gz</td><td>Bacteria</td><td>990719</td></tr>\n",
       "\t<tr><td>GUT_GENOME286691.fna.gz</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/UHGG/GUT_GENOME286691.fna.gz</td><td>Bacteria</td><td>990719</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 731 × 4\n",
       "\\begin{tabular}{llll}\n",
       " Name & Fasta & Domain & Taxid\\\\\n",
       " <chr> & <chr> & <chr> & <dbl>\\\\\n",
       "\\hline\n",
       "\t MI-001-H\\_\\_metabat2\\_high\\_PE.003.fna.gz & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/christensenellales/genomes/Colombian\\_MG/MI-001-H\\_\\_metabat2\\_high\\_PE.003.fna.gz & Bacteria & 990719\\\\\n",
       "\t MI-002-H\\_\\_metabat2\\_high\\_PE.005.fna.gz & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/christensenellales/genomes/Colombian\\_MG/MI-002-H\\_\\_metabat2\\_high\\_PE.005.fna.gz & Bacteria & 990719\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t GUT\\_GENOME286615.fna.gz & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/christensenellales/genomes/UHGG/GUT\\_GENOME286615.fna.gz & Bacteria & 990719\\\\\n",
       "\t GUT\\_GENOME286691.fna.gz & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/christensenellales/genomes/UHGG/GUT\\_GENOME286691.fna.gz & Bacteria & 990719\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 731 × 4\n",
       "\n",
       "| Name &lt;chr&gt; | Fasta &lt;chr&gt; | Domain &lt;chr&gt; | Taxid &lt;dbl&gt; |\n",
       "|---|---|---|---|\n",
       "| MI-001-H__metabat2_high_PE.003.fna.gz | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/Colombian_MG/MI-001-H__metabat2_high_PE.003.fna.gz | Bacteria | 990719 |\n",
       "| MI-002-H__metabat2_high_PE.005.fna.gz | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/Colombian_MG/MI-002-H__metabat2_high_PE.005.fna.gz | Bacteria | 990719 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| GUT_GENOME286615.fna.gz | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/UHGG/GUT_GENOME286615.fna.gz | Bacteria | 990719 |\n",
       "| GUT_GENOME286691.fna.gz | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/UHGG/GUT_GENOME286691.fna.gz | Bacteria | 990719 |\n",
       "\n"
      ],
      "text/plain": [
       "    Name                                 \n",
       "1   MI-001-H__metabat2_high_PE.003.fna.gz\n",
       "2   MI-002-H__metabat2_high_PE.005.fna.gz\n",
       "⋮   ⋮                                    \n",
       "730 GUT_GENOME286615.fna.gz              \n",
       "731 GUT_GENOME286691.fna.gz              \n",
       "    Fasta                                                                                                                                           \n",
       "1   /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/Colombian_MG/MI-001-H__metabat2_high_PE.003.fna.gz\n",
       "2   /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/Colombian_MG/MI-002-H__metabat2_high_PE.005.fna.gz\n",
       "⋮   ⋮                                                                                                                                               \n",
       "730 /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/UHGG/GUT_GENOME286615.fna.gz                      \n",
       "731 /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/UHGG/GUT_GENOME286691.fna.gz                      \n",
       "    Domain   Taxid \n",
       "1   Bacteria 990719\n",
       "2   Bacteria 990719\n",
       "⋮   ⋮        ⋮     \n",
       "730 Bacteria 990719\n",
       "731 Bacteria 990719"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "files = list_files(work_dir, '.fna')\n",
    "samps = data.frame(Name = files %>% as.character %>% basename,\n",
    "                   Fasta = files,\n",
    "                   Domain = 'Bacteria',\n",
    "                   Taxid = taxid) %>%\n",
    "    mutate(Fasta = gsub('/+', '/', Fasta))\n",
    "samps"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-02-26T12:26:01.740272Z",
     "iopub.status.busy": "2021-02-26T12:26:01.737339Z",
     "iopub.status.idle": "2021-02-26T12:26:01.769190Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "File written: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes//genomes_raw.txt \n"
     ]
    }
   ],
   "source": [
    "# writing file\n",
    "outfile = file.path(work_dir, 'genomes_raw.txt')\n",
    "write_table(samps, outfile)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### WARNING! \n",
    "\n",
    "> The taxid will need to be changed for the primer design.\n",
    "  The proper taxids can be determined by: `GTDB-Tk => ncbi-lineage => names.dmp`"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# LLG"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Config"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-02-23T14:47:20.533730Z",
     "iopub.status.busy": "2021-02-23T14:47:20.531674Z",
     "iopub.status.idle": "2021-02-23T14:47:20.591174Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# table with genome --> fasta_file information\n",
      "samples_file: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/genomes_raw.txt\n",
      "\n",
      "# output location\n",
      "output_dir: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/\n",
      "\n",
      "# temporary file directory (your username will be added automatically)\n",
      "tmp_dir: /ebio/abt3_scratch/\n",
      "\n",
      "# batch processing of genomes for certain steps\n",
      "## increase to better parallelize\n",
      "batches: 5\n",
      "\n",
      "# Domain of genomes ('Archaea' or 'Bacteria)\n",
      "## Use \"Skip\" if provided as a \"Domain\" column in the genome table\n",
      "Domain: Skip\n",
      "\n",
      "# software parameters\n",
      "# Use \"Skip\" to skip any of these steps. If no params for rule, use \"\"\n",
      "# dRep MAGs are not further analyzed, but you can de-rep & then use the de-rep genome table as input.\n",
      "params:\n",
      "  ionice: -c 3\n",
      "  # assembly assessment\n",
      "  seqkit: \"\"\n",
      "  quast: Skip #\"\"\n",
      "  multiqc_on_quast: \"\" \n",
      "  checkm: \"\"\n",
      "  # de-replication (CheckM recommended)\n",
      "  drep:\n",
      "    algorithm: auto     # will select fastANI if >1000 genomes, else accurate mode\n",
      "    params: -comp 50 -con 5 -sa 0.999\n",
      "  # taxonomy\n",
      "  sourmash:\n",
      "    compute: Skip #--scaled 10000 -k 31\n",
      "    gather: -k 31\n",
      "  gtdbtk:\n",
      "    classify_wf: --min_perc_aa 10\n",
      "  # genome pairwise ANI  \n",
      "  fastani: Skip #--fragLen 3000 --minFraction 0.2 -k 16\n",
      "  dashing: Skip # -k 31 --full-tsv\n",
      "  comparem_aai: Skip # --evalue 0.001\n",
      "  # gene annotation\n",
      "  gene_call:\n",
      "    prokka: Skip #\"\" \n",
      "    multiqc_on_prokka: \"\"\n",
      "    prodigal: Skip #\"\"\n",
      "  eggnog_mapper: Skip #\"\"\n",
      "  eggnog_mapper_annot: \"\"\n",
      "  # rRNA (16S alignment & phylogeny)\n",
      "  barrnap: Skip #--lencutoff 0.8\n",
      "  vsearch_per_genome_drep: --id 0.95    # Skip to prevent drep of 16S copies within each genome\n",
      "  qiime2_fasttree: \"\"\n",
      "  qiime2_iqtree: --p-alrt 1000 --p-abayes --p-lbp 1000  --p-substitution-model 'GTR+I+G'\n",
      "  # genome phylogeny\n",
      "  phylophlan_config: Skip #--map_dna diamond --db_aa diamond --map_aa diamond --msa mafft --trim trimal --tree1 fasttree --tree2 raxml\n",
      "  phylophlan:\n",
      "    accuracy: --auto       # --auto will select --fast if >2000 genomes, otherwise --accurate \n",
      "    other_params: --diversity high --min_num_markers 50 \n",
      "  # phenotype\n",
      "  traitar: Skip #\"\"\n",
      "  # biosynthetic gene clusters (BGCs)\n",
      "  antismash: Skip #--cb-knownclusters --cb-subclusters --asf  \n",
      "  DeepBGC: Skip #--score 0.5 --classifier-score 0.5 --prodigal-meta-mode\n",
      "  # antimicrobial resistance (AMR)\n",
      "  abricate: Skip #--minid 75 --mincov 80\n",
      "  # CRISPRs\n",
      "  cctyper: Skip #--prodigal meta\n",
      "  \n",
      "# databases\n",
      "databases:\n",
      "  checkM_data: /ebio/abt3_projects/databases_no-backup/checkM/\n",
      "  sourmash: /ebio/abt3_projects/databases_no-backup/sourmash/genbank-k31.sbt.json\n",
      "  sourmash_lca: /ebio/abt3_projects/databases_no-backup/sourmash/genbank-k31.lca.json.gz\n",
      "  gtdbtk: /ebio/abt3_projects/databases_no-backup/GTDB/release95/gtdbtk/db_info.md\n",
      "  phylophlan: /ebio/abt3_projects/databases_no-backup/phylophlan/PhyloPhlan/phylophlan.faa.bz2\n",
      "  eggnog: /ebio/abt3_projects/databases_no-backup/Eggnog/v2/eggnog.db\n",
      "  eggnog_diamond: /ebio/abt3_projects/databases_no-backup/Eggnog/v2/eggnog_proteins.dmnd\n",
      "  antismash: /ebio/abt3_projects/databases_no-backup/antismash/v5/\n",
      "  deepbgc: /ebio/abt3_projects/databases_no-backup/DeepBGC/\n",
      "  traitar: /ebio/abt3_projects/databases_no-backup/pfam/traitar/\n",
      "  taxdump:   # used for adding taxids to GTDB-Tk classifications\n",
      "    names: /ebio/abt3_projects/databases_no-backup/GTDB/release95/taxdump/names.dmp\n",
      "    nodes: /ebio/abt3_projects/databases_no-backup/GTDB/release95/taxdump/nodes.dmp\n",
      "  abricate:\n",
      "    ncbi: /ebio/abt3_projects/databases_no-backup/abricate/ncbi/sequences\n",
      "    card: /ebio/abt3_projects/databases_no-backup/abricate/card/sequences\n",
      "    resfinder: /ebio/abt3_projects/databases_no-backup/abricate/resfinder/sequences\n",
      "    argannot: /ebio/abt3_projects/databases_no-backup/abricate/argannot/sequences\n",
      "    bacmet2: /ebio/abt3_projects/databases_no-backup/abricate/bacmet2/sequences\n",
      "    vfdb: /ebio/abt3_projects/databases_no-backup/abricate/vfdb/sequences\n",
      "    megares: /ebio/abt3_projects/databases_no-backup/abricate/megares/sequences\n",
      "    plasmidfinder: /ebio/abt3_projects/databases_no-backup/abricate/plasmidfinder/sequences\n",
      "\n",
      "# snakemake pipeline\n",
      "pipeline:\n",
      "  snakemake_folder: ./\n",
      "  script_folder: ./bin/scripts/\n",
      "  use_shared_mem: True\n",
      "  name: LLG"
     ]
    }
   ],
   "source": [
    "cat_file(file.path(work_dir, 'config_llg.yaml'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Run"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "(snakemake) @ rick:/ebio/abt3_projects/software/dev/ll_pipelines/llg\n",
    "$ screen -L -s llg-christ ./snakemake_sge.sh /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/config_llg.yaml 30 -F\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Samples table of high quality genomes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-02-27T09:52:38.326191Z",
     "iopub.status.busy": "2021-02-27T09:52:38.323791Z",
     "iopub.status.idle": "2021-02-27T09:52:38.399975Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 731 × 33</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Bin.Id</th><th scope=col>Marker.lineage</th><th scope=col>X..genomes</th><th scope=col>X..markers</th><th scope=col>X..marker.sets</th><th scope=col>Completeness</th><th scope=col>Contamination</th><th scope=col>Strain.heterogeneity</th><th scope=col>Genome.size..bp.</th><th scope=col>X..ambiguous.bases</th><th scope=col>⋯</th><th scope=col>X0</th><th scope=col>X1</th><th scope=col>X2</th><th scope=col>X3</th><th scope=col>X4</th><th scope=col>X5.</th><th scope=col>assembly.Id</th><th scope=col>assembler.Id</th><th scope=col>taxon.Id</th><th scope=col>File</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1002469__metabat2__HighNoCov.011.fna.gz</td><td>c__Clostridia (UID1118)   </td><td>387</td><td>223</td><td>124</td><td>98.39</td><td>0.00</td><td>0</td><td>2229828</td><td>0</td><td>⋯</td><td> 3</td><td>220</td><td>0</td><td>0</td><td>0</td><td>0</td><td>|ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|christensenellales|genomes|LLG_output|checkM|1|checkm|markers_qa_summary.tsv.1</td><td>markers_qa_summary.tsv.1</td><td>NA</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/LLG_output/checkM/1/checkm/markers_qa_summary.tsv.1</td></tr>\n",
       "\t<tr><td>1002756__metabat2__Low.024.fna.gz      </td><td>o__Clostridiales (UID1120)</td><td>304</td><td>250</td><td>143</td><td>72.41</td><td>0.02</td><td>0</td><td>1268686</td><td>0</td><td>⋯</td><td>53</td><td>196</td><td>1</td><td>0</td><td>0</td><td>0</td><td>|ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|christensenellales|genomes|LLG_output|checkM|1|checkm|markers_qa_summary.tsv.2</td><td>markers_qa_summary.tsv.2</td><td>NA</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/LLG_output/checkM/1/checkm/markers_qa_summary.tsv.2</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋱</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>MI-408-H__metabat2_high_PE.004.fna.gz</td><td>c__Clostridia (UID1118)</td><td>387</td><td>223</td><td>124</td><td>96.75</td><td>2.69</td><td>0</td><td>3065865</td><td>0</td><td>⋯</td><td> 6</td><td>213</td><td>4</td><td>0</td><td>0</td><td>0</td><td>|ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|christensenellales|genomes|LLG_output|checkM|10|checkm|markers_qa_summary.tsv.72</td><td>markers_qa_summary.tsv.72</td><td>NA</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/LLG_output/checkM/10/checkm/markers_qa_summary.tsv.72</td></tr>\n",
       "\t<tr><td>MI-423-H__metabat2_high_PE.025.fna.gz</td><td>c__Clostridia (UID1118)</td><td>387</td><td>223</td><td>124</td><td>79.38</td><td>0.00</td><td>0</td><td>1843857</td><td>0</td><td>⋯</td><td>35</td><td>188</td><td>0</td><td>0</td><td>0</td><td>0</td><td>|ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|christensenellales|genomes|LLG_output|checkM|10|checkm|markers_qa_summary.tsv.73</td><td>markers_qa_summary.tsv.73</td><td>NA</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/LLG_output/checkM/10/checkm/markers_qa_summary.tsv.73</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 731 × 33\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " Bin.Id & Marker.lineage & X..genomes & X..markers & X..marker.sets & Completeness & Contamination & Strain.heterogeneity & Genome.size..bp. & X..ambiguous.bases & ⋯ & X0 & X1 & X2 & X3 & X4 & X5. & assembly.Id & assembler.Id & taxon.Id & File\\\\\n",
       " <chr> & <chr> & <int> & <int> & <int> & <dbl> & <dbl> & <dbl> & <int> & <int> & ⋯ & <int> & <int> & <int> & <int> & <int> & <int> & <chr> & <chr> & <lgl> & <chr>\\\\\n",
       "\\hline\n",
       "\t 1002469\\_\\_metabat2\\_\\_HighNoCov.011.fna.gz & c\\_\\_Clostridia (UID1118)    & 387 & 223 & 124 & 98.39 & 0.00 & 0 & 2229828 & 0 & ⋯ &  3 & 220 & 0 & 0 & 0 & 0 & \\textbar{}ebio\\textbar{}abt3\\_projects\\textbar{}software\\textbar{}dev\\textbar{}ll\\_pipelines\\textbar{}llprimer\\textbar{}experiments\\textbar{}christensenellales\\textbar{}genomes\\textbar{}LLG\\_output\\textbar{}checkM\\textbar{}1\\textbar{}checkm\\textbar{}markers\\_qa\\_summary.tsv.1 & markers\\_qa\\_summary.tsv.1 & NA & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/christensenellales/genomes/LLG\\_output/checkM/1/checkm/markers\\_qa\\_summary.tsv.1\\\\\n",
       "\t 1002756\\_\\_metabat2\\_\\_Low.024.fna.gz       & o\\_\\_Clostridiales (UID1120) & 304 & 250 & 143 & 72.41 & 0.02 & 0 & 1268686 & 0 & ⋯ & 53 & 196 & 1 & 0 & 0 & 0 & \\textbar{}ebio\\textbar{}abt3\\_projects\\textbar{}software\\textbar{}dev\\textbar{}ll\\_pipelines\\textbar{}llprimer\\textbar{}experiments\\textbar{}christensenellales\\textbar{}genomes\\textbar{}LLG\\_output\\textbar{}checkM\\textbar{}1\\textbar{}checkm\\textbar{}markers\\_qa\\_summary.tsv.2 & markers\\_qa\\_summary.tsv.2 & NA & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/christensenellales/genomes/LLG\\_output/checkM/1/checkm/markers\\_qa\\_summary.tsv.2\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋱ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t MI-408-H\\_\\_metabat2\\_high\\_PE.004.fna.gz & c\\_\\_Clostridia (UID1118) & 387 & 223 & 124 & 96.75 & 2.69 & 0 & 3065865 & 0 & ⋯ &  6 & 213 & 4 & 0 & 0 & 0 & \\textbar{}ebio\\textbar{}abt3\\_projects\\textbar{}software\\textbar{}dev\\textbar{}ll\\_pipelines\\textbar{}llprimer\\textbar{}experiments\\textbar{}christensenellales\\textbar{}genomes\\textbar{}LLG\\_output\\textbar{}checkM\\textbar{}10\\textbar{}checkm\\textbar{}markers\\_qa\\_summary.tsv.72 & markers\\_qa\\_summary.tsv.72 & NA & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/christensenellales/genomes/LLG\\_output/checkM/10/checkm/markers\\_qa\\_summary.tsv.72\\\\\n",
       "\t MI-423-H\\_\\_metabat2\\_high\\_PE.025.fna.gz & c\\_\\_Clostridia (UID1118) & 387 & 223 & 124 & 79.38 & 0.00 & 0 & 1843857 & 0 & ⋯ & 35 & 188 & 0 & 0 & 0 & 0 & \\textbar{}ebio\\textbar{}abt3\\_projects\\textbar{}software\\textbar{}dev\\textbar{}ll\\_pipelines\\textbar{}llprimer\\textbar{}experiments\\textbar{}christensenellales\\textbar{}genomes\\textbar{}LLG\\_output\\textbar{}checkM\\textbar{}10\\textbar{}checkm\\textbar{}markers\\_qa\\_summary.tsv.73 & markers\\_qa\\_summary.tsv.73 & NA & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/christensenellales/genomes/LLG\\_output/checkM/10/checkm/markers\\_qa\\_summary.tsv.73\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 731 × 33\n",
       "\n",
       "| Bin.Id &lt;chr&gt; | Marker.lineage &lt;chr&gt; | X..genomes &lt;int&gt; | X..markers &lt;int&gt; | X..marker.sets &lt;int&gt; | Completeness &lt;dbl&gt; | Contamination &lt;dbl&gt; | Strain.heterogeneity &lt;dbl&gt; | Genome.size..bp. &lt;int&gt; | X..ambiguous.bases &lt;int&gt; | ⋯ ⋯ | X0 &lt;int&gt; | X1 &lt;int&gt; | X2 &lt;int&gt; | X3 &lt;int&gt; | X4 &lt;int&gt; | X5. &lt;int&gt; | assembly.Id &lt;chr&gt; | assembler.Id &lt;chr&gt; | taxon.Id &lt;lgl&gt; | File &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1002469__metabat2__HighNoCov.011.fna.gz | c__Clostridia (UID1118)    | 387 | 223 | 124 | 98.39 | 0.00 | 0 | 2229828 | 0 | ⋯ |  3 | 220 | 0 | 0 | 0 | 0 | |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|christensenellales|genomes|LLG_output|checkM|1|checkm|markers_qa_summary.tsv.1 | markers_qa_summary.tsv.1 | NA | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/LLG_output/checkM/1/checkm/markers_qa_summary.tsv.1 |\n",
       "| 1002756__metabat2__Low.024.fna.gz       | o__Clostridiales (UID1120) | 304 | 250 | 143 | 72.41 | 0.02 | 0 | 1268686 | 0 | ⋯ | 53 | 196 | 1 | 0 | 0 | 0 | |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|christensenellales|genomes|LLG_output|checkM|1|checkm|markers_qa_summary.tsv.2 | markers_qa_summary.tsv.2 | NA | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/LLG_output/checkM/1/checkm/markers_qa_summary.tsv.2 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋱ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| MI-408-H__metabat2_high_PE.004.fna.gz | c__Clostridia (UID1118) | 387 | 223 | 124 | 96.75 | 2.69 | 0 | 3065865 | 0 | ⋯ |  6 | 213 | 4 | 0 | 0 | 0 | |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|christensenellales|genomes|LLG_output|checkM|10|checkm|markers_qa_summary.tsv.72 | markers_qa_summary.tsv.72 | NA | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/LLG_output/checkM/10/checkm/markers_qa_summary.tsv.72 |\n",
       "| MI-423-H__metabat2_high_PE.025.fna.gz | c__Clostridia (UID1118) | 387 | 223 | 124 | 79.38 | 0.00 | 0 | 1843857 | 0 | ⋯ | 35 | 188 | 0 | 0 | 0 | 0 | |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|christensenellales|genomes|LLG_output|checkM|10|checkm|markers_qa_summary.tsv.73 | markers_qa_summary.tsv.73 | NA | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/LLG_output/checkM/10/checkm/markers_qa_summary.tsv.73 |\n",
       "\n"
      ],
      "text/plain": [
       "    Bin.Id                                  Marker.lineage            \n",
       "1   1002469__metabat2__HighNoCov.011.fna.gz c__Clostridia (UID1118)   \n",
       "2   1002756__metabat2__Low.024.fna.gz       o__Clostridiales (UID1120)\n",
       "⋮   ⋮                                       ⋮                         \n",
       "730 MI-408-H__metabat2_high_PE.004.fna.gz   c__Clostridia (UID1118)   \n",
       "731 MI-423-H__metabat2_high_PE.025.fna.gz   c__Clostridia (UID1118)   \n",
       "    X..genomes X..markers X..marker.sets Completeness Contamination\n",
       "1   387        223        124            98.39        0.00         \n",
       "2   304        250        143            72.41        0.02         \n",
       "⋮   ⋮          ⋮          ⋮              ⋮            ⋮            \n",
       "730 387        223        124            96.75        2.69         \n",
       "731 387        223        124            79.38        0.00         \n",
       "    Strain.heterogeneity Genome.size..bp. X..ambiguous.bases ⋯ X0 X1  X2 X3 X4\n",
       "1   0                    2229828          0                  ⋯  3 220 0  0  0 \n",
       "2   0                    1268686          0                  ⋯ 53 196 1  0  0 \n",
       "⋮   ⋮                    ⋮                ⋮                  ⋱ ⋮  ⋮   ⋮  ⋮  ⋮ \n",
       "730 0                    3065865          0                  ⋯  6 213 4  0  0 \n",
       "731 0                    1843857          0                  ⋯ 35 188 0  0  0 \n",
       "    X5.\n",
       "1   0  \n",
       "2   0  \n",
       "⋮   ⋮  \n",
       "730 0  \n",
       "731 0  \n",
       "    assembly.Id                                                                                                                                        \n",
       "1   |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|christensenellales|genomes|LLG_output|checkM|1|checkm|markers_qa_summary.tsv.1  \n",
       "2   |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|christensenellales|genomes|LLG_output|checkM|1|checkm|markers_qa_summary.tsv.2  \n",
       "⋮   ⋮                                                                                                                                                  \n",
       "730 |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|christensenellales|genomes|LLG_output|checkM|10|checkm|markers_qa_summary.tsv.72\n",
       "731 |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|christensenellales|genomes|LLG_output|checkM|10|checkm|markers_qa_summary.tsv.73\n",
       "    assembler.Id              taxon.Id\n",
       "1   markers_qa_summary.tsv.1  NA      \n",
       "2   markers_qa_summary.tsv.2  NA      \n",
       "⋮   ⋮                         ⋮       \n",
       "730 markers_qa_summary.tsv.72 NA      \n",
       "731 markers_qa_summary.tsv.73 NA      \n",
       "    File                                                                                                                                               \n",
       "1   /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/LLG_output/checkM/1/checkm/markers_qa_summary.tsv.1  \n",
       "2   /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/LLG_output/checkM/1/checkm/markers_qa_summary.tsv.2  \n",
       "⋮   ⋮                                                                                                                                                  \n",
       "730 /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/LLG_output/checkM/10/checkm/markers_qa_summary.tsv.72\n",
       "731 /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/LLG_output/checkM/10/checkm/markers_qa_summary.tsv.73"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# checkM summary\n",
    "checkm = file.path(work_dir, 'LLG_output', 'checkM', 'checkm_qa_summary.tsv') %>%\n",
    "    read.delim(sep='\\t') \n",
    "checkm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-02-27T09:52:38.480769Z",
     "iopub.status.busy": "2021-02-27T09:52:38.477998Z",
     "iopub.status.idle": "2021-02-27T09:52:38.525059Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 731 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>genome</th><th scope=col>completeness</th><th scope=col>contamination</th><th scope=col>Bin.Id</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/1002469__metabat2__HighNoCov.011.fna.gz.fna</td><td>98.39</td><td>0.00</td><td>1002469__metabat2__HighNoCov.011.fna.gz</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/1002756__metabat2__Low.024.fna.gz.fna      </td><td>72.41</td><td>0.02</td><td>1002756__metabat2__Low.024.fna.gz      </td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/MI-408-H__metabat2_high_PE.004.fna.gz.fna</td><td>96.75</td><td>2.69</td><td>MI-408-H__metabat2_high_PE.004.fna.gz</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/MI-423-H__metabat2_high_PE.025.fna.gz.fna</td><td>79.38</td><td>0.00</td><td>MI-423-H__metabat2_high_PE.025.fna.gz</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 731 × 4\n",
       "\\begin{tabular}{llll}\n",
       " genome & completeness & contamination & Bin.Id\\\\\n",
       " <chr> & <dbl> & <dbl> & <chr>\\\\\n",
       "\\hline\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/1002469\\_\\_metabat2\\_\\_HighNoCov.011.fna.gz.fna & 98.39 & 0.00 & 1002469\\_\\_metabat2\\_\\_HighNoCov.011.fna.gz\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/1002756\\_\\_metabat2\\_\\_Low.024.fna.gz.fna       & 72.41 & 0.02 & 1002756\\_\\_metabat2\\_\\_Low.024.fna.gz      \\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/MI-408-H\\_\\_metabat2\\_high\\_PE.004.fna.gz.fna & 96.75 & 2.69 & MI-408-H\\_\\_metabat2\\_high\\_PE.004.fna.gz\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/MI-423-H\\_\\_metabat2\\_high\\_PE.025.fna.gz.fna & 79.38 & 0.00 & MI-423-H\\_\\_metabat2\\_high\\_PE.025.fna.gz\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 731 × 4\n",
       "\n",
       "| genome &lt;chr&gt; | completeness &lt;dbl&gt; | contamination &lt;dbl&gt; | Bin.Id &lt;chr&gt; |\n",
       "|---|---|---|---|\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/1002469__metabat2__HighNoCov.011.fna.gz.fna | 98.39 | 0.00 | 1002469__metabat2__HighNoCov.011.fna.gz |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/1002756__metabat2__Low.024.fna.gz.fna       | 72.41 | 0.02 | 1002756__metabat2__Low.024.fna.gz       |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/MI-408-H__metabat2_high_PE.004.fna.gz.fna | 96.75 | 2.69 | MI-408-H__metabat2_high_PE.004.fna.gz |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/MI-423-H__metabat2_high_PE.025.fna.gz.fna | 79.38 | 0.00 | MI-423-H__metabat2_high_PE.025.fna.gz |\n",
       "\n"
      ],
      "text/plain": [
       "    genome                                                                                           \n",
       "1   /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/1002469__metabat2__HighNoCov.011.fna.gz.fna\n",
       "2   /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/1002756__metabat2__Low.024.fna.gz.fna      \n",
       "⋮   ⋮                                                                                                \n",
       "730 /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/MI-408-H__metabat2_high_PE.004.fna.gz.fna  \n",
       "731 /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/MI-423-H__metabat2_high_PE.025.fna.gz.fna  \n",
       "    completeness contamination Bin.Id                                 \n",
       "1   98.39        0.00          1002469__metabat2__HighNoCov.011.fna.gz\n",
       "2   72.41        0.02          1002756__metabat2__Low.024.fna.gz      \n",
       "⋮   ⋮            ⋮             ⋮                                      \n",
       "730 96.75        2.69          MI-408-H__metabat2_high_PE.004.fna.gz  \n",
       "731 79.38        0.00          MI-423-H__metabat2_high_PE.025.fna.gz  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# dRep summary\n",
    "drep = file.path(work_dir, 'LLG_output', 'drep', 'checkm_markers_qa_summary.tsv') %>%\n",
    "    read.delim(sep='\\t') %>%\n",
    "    mutate(Bin.Id = gsub('.+/', '', genome),\n",
    "           Bin.Id = gsub('\\\\.fna$', '', Bin.Id))\n",
    "drep"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-02-27T09:52:38.605097Z",
     "iopub.status.busy": "2021-02-27T09:52:38.602431Z",
     "iopub.status.idle": "2021-02-27T09:52:38.639347Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 665 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Name</th><th scope=col>Fasta</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>GCA_012518615.1_ASM1251861v1_genomic.fna</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/LLG_output/drep/drep/dereplicated_genomes/GCA_012518615.1_ASM1251861v1_genomic.fna.fna</td></tr>\n",
       "\t<tr><td>GCA_012837835.1_ASM1283783v1_genomic.fna</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/LLG_output/drep/drep/dereplicated_genomes/GCA_012837835.1_ASM1283783v1_genomic.fna.fna</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>AT573__metabat2__High.025.fna.gz </td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/LLG_output/drep/drep/dereplicated_genomes/AT573__metabat2__High.025.fna.gz.fna </td></tr>\n",
       "\t<tr><td>1002756__metabat2__Low.024.fna.gz</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/LLG_output/drep/drep/dereplicated_genomes/1002756__metabat2__Low.024.fna.gz.fna</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 665 × 2\n",
       "\\begin{tabular}{ll}\n",
       " Name & Fasta\\\\\n",
       " <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t GCA\\_012518615.1\\_ASM1251861v1\\_genomic.fna & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/christensenellales/genomes/LLG\\_output/drep/drep/dereplicated\\_genomes/GCA\\_012518615.1\\_ASM1251861v1\\_genomic.fna.fna\\\\\n",
       "\t GCA\\_012837835.1\\_ASM1283783v1\\_genomic.fna & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/christensenellales/genomes/LLG\\_output/drep/drep/dereplicated\\_genomes/GCA\\_012837835.1\\_ASM1283783v1\\_genomic.fna.fna\\\\\n",
       "\t ⋮ & ⋮\\\\\n",
       "\t AT573\\_\\_metabat2\\_\\_High.025.fna.gz  & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/christensenellales/genomes/LLG\\_output/drep/drep/dereplicated\\_genomes/AT573\\_\\_metabat2\\_\\_High.025.fna.gz.fna \\\\\n",
       "\t 1002756\\_\\_metabat2\\_\\_Low.024.fna.gz & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/christensenellales/genomes/LLG\\_output/drep/drep/dereplicated\\_genomes/1002756\\_\\_metabat2\\_\\_Low.024.fna.gz.fna\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 665 × 2\n",
       "\n",
       "| Name &lt;chr&gt; | Fasta &lt;chr&gt; |\n",
       "|---|---|\n",
       "| GCA_012518615.1_ASM1251861v1_genomic.fna | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/LLG_output/drep/drep/dereplicated_genomes/GCA_012518615.1_ASM1251861v1_genomic.fna.fna |\n",
       "| GCA_012837835.1_ASM1283783v1_genomic.fna | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/LLG_output/drep/drep/dereplicated_genomes/GCA_012837835.1_ASM1283783v1_genomic.fna.fna |\n",
       "| ⋮ | ⋮ |\n",
       "| AT573__metabat2__High.025.fna.gz  | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/LLG_output/drep/drep/dereplicated_genomes/AT573__metabat2__High.025.fna.gz.fna  |\n",
       "| 1002756__metabat2__Low.024.fna.gz | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/LLG_output/drep/drep/dereplicated_genomes/1002756__metabat2__Low.024.fna.gz.fna |\n",
       "\n"
      ],
      "text/plain": [
       "    Name                                    \n",
       "1   GCA_012518615.1_ASM1251861v1_genomic.fna\n",
       "2   GCA_012837835.1_ASM1283783v1_genomic.fna\n",
       "⋮   ⋮                                       \n",
       "664 AT573__metabat2__High.025.fna.gz        \n",
       "665 1002756__metabat2__Low.024.fna.gz       \n",
       "    Fasta                                                                                                                                                                               \n",
       "1   /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/LLG_output/drep/drep/dereplicated_genomes/GCA_012518615.1_ASM1251861v1_genomic.fna.fna\n",
       "2   /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/LLG_output/drep/drep/dereplicated_genomes/GCA_012837835.1_ASM1283783v1_genomic.fna.fna\n",
       "⋮   ⋮                                                                                                                                                                                   \n",
       "664 /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/LLG_output/drep/drep/dereplicated_genomes/AT573__metabat2__High.025.fna.gz.fna        \n",
       "665 /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/LLG_output/drep/drep/dereplicated_genomes/1002756__metabat2__Low.024.fna.gz.fna       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# de-replicated genomes\n",
    "drep_gen = file.path(work_dir, 'LLG_output', 'drep', 'dereplicated_genomes.tsv') %>%\n",
    "    read.delim(sep='\\t')\n",
    "drep_gen"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-02-27T09:52:38.719709Z",
     "iopub.status.busy": "2021-02-27T09:52:38.717015Z",
     "iopub.status.idle": "2021-02-27T09:52:38.802950Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 731 × 24</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>user_genome</th><th scope=col>Domain</th><th scope=col>Phylum</th><th scope=col>Class</th><th scope=col>Order</th><th scope=col>Family</th><th scope=col>Genus</th><th scope=col>Species</th><th scope=col>fastani_reference</th><th scope=col>fastani_reference_radius</th><th scope=col>⋯</th><th scope=col>closest_placement_radius</th><th scope=col>closest_placement_taxonomy</th><th scope=col>closest_placement_ani</th><th scope=col>closest_placement_af</th><th scope=col>msa_percent</th><th scope=col>translation_table</th><th scope=col>red_value</th><th scope=col>warnings</th><th scope=col>taxid</th><th scope=col>taxid_rank</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1002469__metabat2__HighNoCov.011.fna.gz</td><td>d__Bacteria</td><td>p__Firmicutes_A</td><td>c__Clostridia_A</td><td>o__Christensenellales</td><td>f__CAG-138 </td><td>g__Phil1   </td><td>s__Phil1 sp001940855</td><td>GCA_001940855.1</td><td>95.0</td><td>⋯</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia_A;o__Christensenellales;f__CAG-138;g__Phil1;s__Phil1 sp001940855       </td><td>97.1 </td><td>0.9 </td><td>92.90</td><td>11</td><td>N/A               </td><td>N/A</td><td>153230</td><td>species</td></tr>\n",
       "\t<tr><td>1002756__metabat2__Low.024.fna.gz      </td><td>d__Bacteria</td><td>p__Firmicutes_A</td><td>c__Clostridia_A</td><td>o__Christensenellales</td><td>f__UMGS1908</td><td>g__UMGS1908</td><td>s__                 </td><td>N/A            </td><td>N/A </td><td>⋯</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia_A;o__Christensenellales;f__UMGS1908;g__UMGS1908;s__UMGS1908 sp900556185</td><td>77.88</td><td>0.13</td><td>82.62</td><td>11</td><td>0.8811857065570132</td><td>N/A</td><td>232276</td><td>genus  </td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋱</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>MI-408-H__metabat2_high_PE.004.fna.gz</td><td>d__Bacteria</td><td>p__Firmicutes_A</td><td>c__Clostridia_A</td><td>o__Christensenellales</td><td>f__CAG-74 </td><td>g__SFHK01</td><td>s__SFHK01 sp004556395</td><td>GCA_004556395.1</td><td>95.0</td><td>⋯</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia_A;o__Christensenellales;f__CAG-74;g__SFHK01;s__SFHK01 sp004556395 </td><td>98.47</td><td>0.89</td><td>93.85</td><td>11</td><td>N/A</td><td>N/A</td><td>225050</td><td>species</td></tr>\n",
       "\t<tr><td>MI-423-H__metabat2_high_PE.025.fna.gz</td><td>d__Bacteria</td><td>p__Firmicutes_A</td><td>c__Clostridia_A</td><td>o__Christensenellales</td><td>f__CAG-138</td><td>g__RUG472</td><td>s__RUG472 sp900545265</td><td>GCA_900545265.1</td><td>95.0</td><td>⋯</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia_A;o__Christensenellales;f__CAG-138;g__RUG472;s__RUG472 sp900545265</td><td>98.69</td><td>0.94</td><td>84.07</td><td>11</td><td>N/A</td><td>N/A</td><td>218859</td><td>species</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 731 × 24\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " user\\_genome & Domain & Phylum & Class & Order & Family & Genus & Species & fastani\\_reference & fastani\\_reference\\_radius & ⋯ & closest\\_placement\\_radius & closest\\_placement\\_taxonomy & closest\\_placement\\_ani & closest\\_placement\\_af & msa\\_percent & translation\\_table & red\\_value & warnings & taxid & taxid\\_rank\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & ⋯ & <chr> & <chr> & <chr> & <chr> & <dbl> & <int> & <chr> & <chr> & <int> & <chr>\\\\\n",
       "\\hline\n",
       "\t 1002469\\_\\_metabat2\\_\\_HighNoCov.011.fna.gz & d\\_\\_Bacteria & p\\_\\_Firmicutes\\_A & c\\_\\_Clostridia\\_A & o\\_\\_Christensenellales & f\\_\\_CAG-138  & g\\_\\_Phil1    & s\\_\\_Phil1 sp001940855 & GCA\\_001940855.1 & 95.0 & ⋯ & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia\\_A;o\\_\\_Christensenellales;f\\_\\_CAG-138;g\\_\\_Phil1;s\\_\\_Phil1 sp001940855        & 97.1  & 0.9  & 92.90 & 11 & N/A                & N/A & 153230 & species\\\\\n",
       "\t 1002756\\_\\_metabat2\\_\\_Low.024.fna.gz       & d\\_\\_Bacteria & p\\_\\_Firmicutes\\_A & c\\_\\_Clostridia\\_A & o\\_\\_Christensenellales & f\\_\\_UMGS1908 & g\\_\\_UMGS1908 & s\\_\\_                  & N/A             & N/A  & ⋯ & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia\\_A;o\\_\\_Christensenellales;f\\_\\_UMGS1908;g\\_\\_UMGS1908;s\\_\\_UMGS1908 sp900556185 & 77.88 & 0.13 & 82.62 & 11 & 0.8811857065570132 & N/A & 232276 & genus  \\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋱ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t MI-408-H\\_\\_metabat2\\_high\\_PE.004.fna.gz & d\\_\\_Bacteria & p\\_\\_Firmicutes\\_A & c\\_\\_Clostridia\\_A & o\\_\\_Christensenellales & f\\_\\_CAG-74  & g\\_\\_SFHK01 & s\\_\\_SFHK01 sp004556395 & GCA\\_004556395.1 & 95.0 & ⋯ & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia\\_A;o\\_\\_Christensenellales;f\\_\\_CAG-74;g\\_\\_SFHK01;s\\_\\_SFHK01 sp004556395  & 98.47 & 0.89 & 93.85 & 11 & N/A & N/A & 225050 & species\\\\\n",
       "\t MI-423-H\\_\\_metabat2\\_high\\_PE.025.fna.gz & d\\_\\_Bacteria & p\\_\\_Firmicutes\\_A & c\\_\\_Clostridia\\_A & o\\_\\_Christensenellales & f\\_\\_CAG-138 & g\\_\\_RUG472 & s\\_\\_RUG472 sp900545265 & GCA\\_900545265.1 & 95.0 & ⋯ & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia\\_A;o\\_\\_Christensenellales;f\\_\\_CAG-138;g\\_\\_RUG472;s\\_\\_RUG472 sp900545265 & 98.69 & 0.94 & 84.07 & 11 & N/A & N/A & 218859 & species\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 731 × 24\n",
       "\n",
       "| user_genome &lt;chr&gt; | Domain &lt;chr&gt; | Phylum &lt;chr&gt; | Class &lt;chr&gt; | Order &lt;chr&gt; | Family &lt;chr&gt; | Genus &lt;chr&gt; | Species &lt;chr&gt; | fastani_reference &lt;chr&gt; | fastani_reference_radius &lt;chr&gt; | ⋯ ⋯ | closest_placement_radius &lt;chr&gt; | closest_placement_taxonomy &lt;chr&gt; | closest_placement_ani &lt;chr&gt; | closest_placement_af &lt;chr&gt; | msa_percent &lt;dbl&gt; | translation_table &lt;int&gt; | red_value &lt;chr&gt; | warnings &lt;chr&gt; | taxid &lt;int&gt; | taxid_rank &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1002469__metabat2__HighNoCov.011.fna.gz | d__Bacteria | p__Firmicutes_A | c__Clostridia_A | o__Christensenellales | f__CAG-138  | g__Phil1    | s__Phil1 sp001940855 | GCA_001940855.1 | 95.0 | ⋯ | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia_A;o__Christensenellales;f__CAG-138;g__Phil1;s__Phil1 sp001940855        | 97.1  | 0.9  | 92.90 | 11 | N/A                | N/A | 153230 | species |\n",
       "| 1002756__metabat2__Low.024.fna.gz       | d__Bacteria | p__Firmicutes_A | c__Clostridia_A | o__Christensenellales | f__UMGS1908 | g__UMGS1908 | s__                  | N/A             | N/A  | ⋯ | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia_A;o__Christensenellales;f__UMGS1908;g__UMGS1908;s__UMGS1908 sp900556185 | 77.88 | 0.13 | 82.62 | 11 | 0.8811857065570132 | N/A | 232276 | genus   |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋱ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| MI-408-H__metabat2_high_PE.004.fna.gz | d__Bacteria | p__Firmicutes_A | c__Clostridia_A | o__Christensenellales | f__CAG-74  | g__SFHK01 | s__SFHK01 sp004556395 | GCA_004556395.1 | 95.0 | ⋯ | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia_A;o__Christensenellales;f__CAG-74;g__SFHK01;s__SFHK01 sp004556395  | 98.47 | 0.89 | 93.85 | 11 | N/A | N/A | 225050 | species |\n",
       "| MI-423-H__metabat2_high_PE.025.fna.gz | d__Bacteria | p__Firmicutes_A | c__Clostridia_A | o__Christensenellales | f__CAG-138 | g__RUG472 | s__RUG472 sp900545265 | GCA_900545265.1 | 95.0 | ⋯ | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia_A;o__Christensenellales;f__CAG-138;g__RUG472;s__RUG472 sp900545265 | 98.69 | 0.94 | 84.07 | 11 | N/A | N/A | 218859 | species |\n",
       "\n"
      ],
      "text/plain": [
       "    user_genome                             Domain      Phylum         \n",
       "1   1002469__metabat2__HighNoCov.011.fna.gz d__Bacteria p__Firmicutes_A\n",
       "2   1002756__metabat2__Low.024.fna.gz       d__Bacteria p__Firmicutes_A\n",
       "⋮   ⋮                                       ⋮           ⋮              \n",
       "730 MI-408-H__metabat2_high_PE.004.fna.gz   d__Bacteria p__Firmicutes_A\n",
       "731 MI-423-H__metabat2_high_PE.025.fna.gz   d__Bacteria p__Firmicutes_A\n",
       "    Class           Order                 Family      Genus      \n",
       "1   c__Clostridia_A o__Christensenellales f__CAG-138  g__Phil1   \n",
       "2   c__Clostridia_A o__Christensenellales f__UMGS1908 g__UMGS1908\n",
       "⋮   ⋮               ⋮                     ⋮           ⋮          \n",
       "730 c__Clostridia_A o__Christensenellales f__CAG-74   g__SFHK01  \n",
       "731 c__Clostridia_A o__Christensenellales f__CAG-138  g__RUG472  \n",
       "    Species               fastani_reference fastani_reference_radius ⋯\n",
       "1   s__Phil1 sp001940855  GCA_001940855.1   95.0                     ⋯\n",
       "2   s__                   N/A               N/A                      ⋯\n",
       "⋮   ⋮                     ⋮                 ⋮                        ⋱\n",
       "730 s__SFHK01 sp004556395 GCA_004556395.1   95.0                     ⋯\n",
       "731 s__RUG472 sp900545265 GCA_900545265.1   95.0                     ⋯\n",
       "    closest_placement_radius\n",
       "1   95.0                    \n",
       "2   95.0                    \n",
       "⋮   ⋮                       \n",
       "730 95.0                    \n",
       "731 95.0                    \n",
       "    closest_placement_taxonomy                                                                                       \n",
       "1   d__Bacteria;p__Firmicutes_A;c__Clostridia_A;o__Christensenellales;f__CAG-138;g__Phil1;s__Phil1 sp001940855       \n",
       "2   d__Bacteria;p__Firmicutes_A;c__Clostridia_A;o__Christensenellales;f__UMGS1908;g__UMGS1908;s__UMGS1908 sp900556185\n",
       "⋮   ⋮                                                                                                                \n",
       "730 d__Bacteria;p__Firmicutes_A;c__Clostridia_A;o__Christensenellales;f__CAG-74;g__SFHK01;s__SFHK01 sp004556395      \n",
       "731 d__Bacteria;p__Firmicutes_A;c__Clostridia_A;o__Christensenellales;f__CAG-138;g__RUG472;s__RUG472 sp900545265     \n",
       "    closest_placement_ani closest_placement_af msa_percent translation_table\n",
       "1   97.1                  0.9                  92.90       11               \n",
       "2   77.88                 0.13                 82.62       11               \n",
       "⋮   ⋮                     ⋮                    ⋮           ⋮                \n",
       "730 98.47                 0.89                 93.85       11               \n",
       "731 98.69                 0.94                 84.07       11               \n",
       "    red_value          warnings taxid  taxid_rank\n",
       "1   N/A                N/A      153230 species   \n",
       "2   0.8811857065570132 N/A      232276 genus     \n",
       "⋮   ⋮                  ⋮        ⋮      ⋮         \n",
       "730 N/A                N/A      225050 species   \n",
       "731 N/A                N/A      218859 species   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# GTDBTk summary\n",
    "tax = file.path(work_dir, 'LLG_output', 'gtdbtk', 'gtdbtk_summary_wTaxid.tsv') %>%\n",
    "    read.delim(, sep='\\t') %>%\n",
    "    separate(classification, \n",
    "             c('Domain', 'Phylum', 'Class', 'Order', 'Family', 'Genus', 'Species'),\n",
    "             sep=';') %>%\n",
    "    select(-note, -classification_method, -pplacer_taxonomy,\n",
    "           -other_related_references.genome_id.species_name.radius.ANI.AF.)\n",
    "tax"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-02-27T09:52:38.840721Z",
     "iopub.status.busy": "2021-02-27T09:52:38.837873Z",
     "iopub.status.idle": "2021-02-27T09:52:56.605217Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No. of unique taxids: 23 \n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 719 × 9</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>user_genome</th><th scope=col>Domain</th><th scope=col>Phylum</th><th scope=col>Class</th><th scope=col>Order</th><th scope=col>Family</th><th scope=col>Genus</th><th scope=col>Species</th><th scope=col>ncbi_taxid</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>1002379__metabat2__High.033.fna.gz    </td><td>d__Bacteria</td><td>p__Firmicutes_A</td><td>c__Clostridia_A</td><td>o__Christensenellales</td><td>f__CAG-917</td><td>g__CAG-475 </td><td>s__CAG-475 sp000434435 </td><td>1263026</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>1002404__metabat2__LowNoCov.002.fna.gz</td><td>d__Bacteria</td><td>p__Firmicutes_A</td><td>c__Clostridia_A</td><td>o__Christensenellales</td><td>f__CAG-917</td><td>g__UMGS1688</td><td>s__UMGS1688 sp900544575</td><td> 172733</td></tr>\n",
       "\t<tr><th scope=row>⋮</th><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><th scope=row>718</th><td>MI-448-H__maxbin2_low_prob.001.fna.gz</td><td>d__Bacteria</td><td>p__Firmicutes_A</td><td>c__Clostridia_A</td><td>o__Christensenellales</td><td>f__CAG-138</td><td>g__Phil1 </td><td>s__Phil1 sp001940855 </td><td>1262781</td></tr>\n",
       "\t<tr><th scope=row>719</th><td>MI-451-H__metabat2_low_PE.014.fna.gz </td><td>d__Bacteria</td><td>p__Firmicutes_A</td><td>c__Clostridia_A</td><td>o__Christensenellales</td><td>f__CAG-74 </td><td>g__SFMI01</td><td>s__SFMI01 sp004556155</td><td>1898207</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 719 × 9\n",
       "\\begin{tabular}{r|lllllllll}\n",
       "  & user\\_genome & Domain & Phylum & Class & Order & Family & Genus & Species & ncbi\\_taxid\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & 1002379\\_\\_metabat2\\_\\_High.033.fna.gz     & d\\_\\_Bacteria & p\\_\\_Firmicutes\\_A & c\\_\\_Clostridia\\_A & o\\_\\_Christensenellales & f\\_\\_CAG-917 & g\\_\\_CAG-475  & s\\_\\_CAG-475 sp000434435  & 1263026\\\\\n",
       "\t2 & 1002404\\_\\_metabat2\\_\\_LowNoCov.002.fna.gz & d\\_\\_Bacteria & p\\_\\_Firmicutes\\_A & c\\_\\_Clostridia\\_A & o\\_\\_Christensenellales & f\\_\\_CAG-917 & g\\_\\_UMGS1688 & s\\_\\_UMGS1688 sp900544575 &  172733\\\\\n",
       "\t⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t718 & MI-448-H\\_\\_maxbin2\\_low\\_prob.001.fna.gz & d\\_\\_Bacteria & p\\_\\_Firmicutes\\_A & c\\_\\_Clostridia\\_A & o\\_\\_Christensenellales & f\\_\\_CAG-138 & g\\_\\_Phil1  & s\\_\\_Phil1 sp001940855  & 1262781\\\\\n",
       "\t719 & MI-451-H\\_\\_metabat2\\_low\\_PE.014.fna.gz  & d\\_\\_Bacteria & p\\_\\_Firmicutes\\_A & c\\_\\_Clostridia\\_A & o\\_\\_Christensenellales & f\\_\\_CAG-74  & g\\_\\_SFMI01 & s\\_\\_SFMI01 sp004556155 & 1898207\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 719 × 9\n",
       "\n",
       "| <!--/--> | user_genome &lt;chr&gt; | Domain &lt;chr&gt; | Phylum &lt;chr&gt; | Class &lt;chr&gt; | Order &lt;chr&gt; | Family &lt;chr&gt; | Genus &lt;chr&gt; | Species &lt;chr&gt; | ncbi_taxid &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 1002379__metabat2__High.033.fna.gz     | d__Bacteria | p__Firmicutes_A | c__Clostridia_A | o__Christensenellales | f__CAG-917 | g__CAG-475  | s__CAG-475 sp000434435  | 1263026 |\n",
       "| 2 | 1002404__metabat2__LowNoCov.002.fna.gz | d__Bacteria | p__Firmicutes_A | c__Clostridia_A | o__Christensenellales | f__CAG-917 | g__UMGS1688 | s__UMGS1688 sp900544575 |  172733 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| 718 | MI-448-H__maxbin2_low_prob.001.fna.gz | d__Bacteria | p__Firmicutes_A | c__Clostridia_A | o__Christensenellales | f__CAG-138 | g__Phil1  | s__Phil1 sp001940855  | 1262781 |\n",
       "| 719 | MI-451-H__metabat2_low_PE.014.fna.gz  | d__Bacteria | p__Firmicutes_A | c__Clostridia_A | o__Christensenellales | f__CAG-74  | g__SFMI01 | s__SFMI01 sp004556155 | 1898207 |\n",
       "\n"
      ],
      "text/plain": [
       "    user_genome                            Domain      Phylum         \n",
       "1   1002379__metabat2__High.033.fna.gz     d__Bacteria p__Firmicutes_A\n",
       "2   1002404__metabat2__LowNoCov.002.fna.gz d__Bacteria p__Firmicutes_A\n",
       "⋮   ⋮                                      ⋮           ⋮              \n",
       "718 MI-448-H__maxbin2_low_prob.001.fna.gz  d__Bacteria p__Firmicutes_A\n",
       "719 MI-451-H__metabat2_low_PE.014.fna.gz   d__Bacteria p__Firmicutes_A\n",
       "    Class           Order                 Family     Genus      \n",
       "1   c__Clostridia_A o__Christensenellales f__CAG-917 g__CAG-475 \n",
       "2   c__Clostridia_A o__Christensenellales f__CAG-917 g__UMGS1688\n",
       "⋮   ⋮               ⋮                     ⋮          ⋮          \n",
       "718 c__Clostridia_A o__Christensenellales f__CAG-138 g__Phil1   \n",
       "719 c__Clostridia_A o__Christensenellales f__CAG-74  g__SFMI01  \n",
       "    Species                 ncbi_taxid\n",
       "1   s__CAG-475 sp000434435  1263026   \n",
       "2   s__UMGS1688 sp900544575  172733   \n",
       "⋮   ⋮                       ⋮         \n",
       "718 s__Phil1 sp001940855    1262781   \n",
       "719 s__SFMI01 sp004556155   1898207   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# getting NCBI taxid from GTDB metadata\n",
    "D = '/ebio/abt3_projects/databases_no-backup/GTDB/release95/metadata/'\n",
    "F = file.path(D, 'bac120_metadata_r95.tsv')\n",
    "tax = tax %>%\n",
    "    inner_join(read.delim(F, sep='\\t') %>% select(gtdb_taxonomy, ncbi_taxonomy, ncbi_taxid), c('closest_placement_taxonomy'='gtdb_taxonomy')) %>%\n",
    "    group_by(user_genome, Domain, Phylum, Class, Order, Family, Genus, Species) %>%\n",
    "    summarize(ncbi_taxid = first(ncbi_taxid), .groups='drop')\n",
    "tax$ncbi_taxid %>% unique_n('taxids')\n",
    "tax"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-02-27T10:56:58.545213Z",
     "iopub.status.busy": "2021-02-27T10:56:58.542218Z",
     "iopub.status.idle": "2021-02-27T10:56:58.569669Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       ".\n",
       " 244328  512298 1262970 1805714 1930013 1930014 2044939 2086585 1262797 1676160 \n",
       "      1       1       1       1       1       1       1       1       2       2 \n",
       "1816678 1851429 1946276   59620 2137878 1263026 1262770 1262775 1946247 1262897 \n",
       "      2       2       2       5       9      11      18      32      45     112 \n",
       "1898207  172733 1262781 \n",
       "    122     130     217 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "tax$ncbi_taxid %>% table %>% sort"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-02-27T09:52:56.609347Z",
     "iopub.status.busy": "2021-02-27T09:52:56.607893Z",
     "iopub.status.idle": "2021-02-27T09:52:56.644434Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- drep --\n",
      "intersect(x,y): 665 \n",
      "setdiff(x,y): 0 \n",
      "setdiff(y,x): 66 \n",
      "union(x,y): 731 \n",
      "-- checkm --\n",
      "intersect(x,y): 731 \n",
      "setdiff(x,y): 0 \n",
      "setdiff(y,x): 0 \n",
      "union(x,y): 731 \n",
      "-- gtdbtk --\n",
      "intersect(x,y): 719 \n",
      "setdiff(x,y): 12 \n",
      "setdiff(y,x): 0 \n",
      "union(x,y): 731 \n"
     ]
    }
   ],
   "source": [
    "# checking overlap\n",
    "cat('-- drep --\\n')\n",
    "overlap(basename(as.character(drep_gen$Fasta)), \n",
    "        basename(as.character(drep$genome)))\n",
    "cat('-- checkm --\\n')\n",
    "overlap(drep$Bin.Id, checkm$Bin.Id)\n",
    "cat('-- gtdbtk --\\n')\n",
    "overlap(drep$Bin.Id, tax$user_genome)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-02-27T09:52:56.648844Z",
     "iopub.status.busy": "2021-02-27T09:52:56.646919Z",
     "iopub.status.idle": "2021-02-27T09:52:56.715159Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 653 × 47</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>genome</th><th scope=col>completeness</th><th scope=col>contamination</th><th scope=col>Bin.Id</th><th scope=col>Marker.lineage</th><th scope=col>X..genomes</th><th scope=col>X..markers</th><th scope=col>X..marker.sets</th><th scope=col>Completeness</th><th scope=col>Contamination</th><th scope=col>⋯</th><th scope=col>Name</th><th scope=col>Fasta</th><th scope=col>Domain</th><th scope=col>Phylum</th><th scope=col>Class</th><th scope=col>Order</th><th scope=col>Family</th><th scope=col>Genus</th><th scope=col>Species</th><th scope=col>ncbi_taxid</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/1002756__metabat2__Low.024.fna.gz.fna </td><td>72.41</td><td>0.02</td><td>1002756__metabat2__Low.024.fna.gz </td><td>o__Clostridiales (UID1120)</td><td>304</td><td>250</td><td>143</td><td>72.41</td><td>0.02</td><td>⋯</td><td>1002756__metabat2__Low.024.fna.gz </td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/LLG_output/drep/drep/dereplicated_genomes/1002756__metabat2__Low.024.fna.gz.fna </td><td>d__Bacteria</td><td>p__Firmicutes_A</td><td>c__Clostridia_A</td><td>o__Christensenellales</td><td>f__UMGS1908</td><td>g__UMGS1908</td><td>s__                 </td><td> 172733</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/1006216__metabat2__High.023.fna.gz.fna</td><td>94.35</td><td>0.00</td><td>1006216__metabat2__High.023.fna.gz</td><td>c__Clostridia (UID1118)   </td><td>387</td><td>223</td><td>124</td><td>94.35</td><td>0.00</td><td>⋯</td><td>1006216__metabat2__High.023.fna.gz</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/LLG_output/drep/drep/dereplicated_genomes/1006216__metabat2__High.023.fna.gz.fna</td><td>d__Bacteria</td><td>p__Firmicutes_A</td><td>c__Clostridia_A</td><td>o__Christensenellales</td><td>f__CAG-138 </td><td>g__Phil1   </td><td>s__Phil1 sp001940855</td><td>1262781</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋱</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/MI-408-H__metabat2_high_PE.004.fna.gz.fna</td><td>96.75</td><td>2.69</td><td>MI-408-H__metabat2_high_PE.004.fna.gz</td><td>c__Clostridia (UID1118)</td><td>387</td><td>223</td><td>124</td><td>96.75</td><td>2.69</td><td>⋯</td><td>MI-408-H__metabat2_high_PE.004.fna.gz</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/LLG_output/drep/drep/dereplicated_genomes/MI-408-H__metabat2_high_PE.004.fna.gz.fna</td><td>d__Bacteria</td><td>p__Firmicutes_A</td><td>c__Clostridia_A</td><td>o__Christensenellales</td><td>f__CAG-74 </td><td>g__SFHK01</td><td>s__SFHK01 sp004556395</td><td>1898207</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/MI-423-H__metabat2_high_PE.025.fna.gz.fna</td><td>79.38</td><td>0.00</td><td>MI-423-H__metabat2_high_PE.025.fna.gz</td><td>c__Clostridia (UID1118)</td><td>387</td><td>223</td><td>124</td><td>79.38</td><td>0.00</td><td>⋯</td><td>MI-423-H__metabat2_high_PE.025.fna.gz</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/LLG_output/drep/drep/dereplicated_genomes/MI-423-H__metabat2_high_PE.025.fna.gz.fna</td><td>d__Bacteria</td><td>p__Firmicutes_A</td><td>c__Clostridia_A</td><td>o__Christensenellales</td><td>f__CAG-138</td><td>g__RUG472</td><td>s__RUG472 sp900545265</td><td> 172733</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 653 × 47\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " genome & completeness & contamination & Bin.Id & Marker.lineage & X..genomes & X..markers & X..marker.sets & Completeness & Contamination & ⋯ & Name & Fasta & Domain & Phylum & Class & Order & Family & Genus & Species & ncbi\\_taxid\\\\\n",
       " <chr> & <dbl> & <dbl> & <chr> & <chr> & <int> & <int> & <int> & <dbl> & <dbl> & ⋯ & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/1002756\\_\\_metabat2\\_\\_Low.024.fna.gz.fna  & 72.41 & 0.02 & 1002756\\_\\_metabat2\\_\\_Low.024.fna.gz  & o\\_\\_Clostridiales (UID1120) & 304 & 250 & 143 & 72.41 & 0.02 & ⋯ & 1002756\\_\\_metabat2\\_\\_Low.024.fna.gz  & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/christensenellales/genomes/LLG\\_output/drep/drep/dereplicated\\_genomes/1002756\\_\\_metabat2\\_\\_Low.024.fna.gz.fna  & d\\_\\_Bacteria & p\\_\\_Firmicutes\\_A & c\\_\\_Clostridia\\_A & o\\_\\_Christensenellales & f\\_\\_UMGS1908 & g\\_\\_UMGS1908 & s\\_\\_                  &  172733\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/1006216\\_\\_metabat2\\_\\_High.023.fna.gz.fna & 94.35 & 0.00 & 1006216\\_\\_metabat2\\_\\_High.023.fna.gz & c\\_\\_Clostridia (UID1118)    & 387 & 223 & 124 & 94.35 & 0.00 & ⋯ & 1006216\\_\\_metabat2\\_\\_High.023.fna.gz & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/christensenellales/genomes/LLG\\_output/drep/drep/dereplicated\\_genomes/1006216\\_\\_metabat2\\_\\_High.023.fna.gz.fna & d\\_\\_Bacteria & p\\_\\_Firmicutes\\_A & c\\_\\_Clostridia\\_A & o\\_\\_Christensenellales & f\\_\\_CAG-138  & g\\_\\_Phil1    & s\\_\\_Phil1 sp001940855 & 1262781\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋱ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/MI-408-H\\_\\_metabat2\\_high\\_PE.004.fna.gz.fna & 96.75 & 2.69 & MI-408-H\\_\\_metabat2\\_high\\_PE.004.fna.gz & c\\_\\_Clostridia (UID1118) & 387 & 223 & 124 & 96.75 & 2.69 & ⋯ & MI-408-H\\_\\_metabat2\\_high\\_PE.004.fna.gz & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/christensenellales/genomes/LLG\\_output/drep/drep/dereplicated\\_genomes/MI-408-H\\_\\_metabat2\\_high\\_PE.004.fna.gz.fna & d\\_\\_Bacteria & p\\_\\_Firmicutes\\_A & c\\_\\_Clostridia\\_A & o\\_\\_Christensenellales & f\\_\\_CAG-74  & g\\_\\_SFHK01 & s\\_\\_SFHK01 sp004556395 & 1898207\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/MI-423-H\\_\\_metabat2\\_high\\_PE.025.fna.gz.fna & 79.38 & 0.00 & MI-423-H\\_\\_metabat2\\_high\\_PE.025.fna.gz & c\\_\\_Clostridia (UID1118) & 387 & 223 & 124 & 79.38 & 0.00 & ⋯ & MI-423-H\\_\\_metabat2\\_high\\_PE.025.fna.gz & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/christensenellales/genomes/LLG\\_output/drep/drep/dereplicated\\_genomes/MI-423-H\\_\\_metabat2\\_high\\_PE.025.fna.gz.fna & d\\_\\_Bacteria & p\\_\\_Firmicutes\\_A & c\\_\\_Clostridia\\_A & o\\_\\_Christensenellales & f\\_\\_CAG-138 & g\\_\\_RUG472 & s\\_\\_RUG472 sp900545265 &  172733\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 653 × 47\n",
       "\n",
       "| genome &lt;chr&gt; | completeness &lt;dbl&gt; | contamination &lt;dbl&gt; | Bin.Id &lt;chr&gt; | Marker.lineage &lt;chr&gt; | X..genomes &lt;int&gt; | X..markers &lt;int&gt; | X..marker.sets &lt;int&gt; | Completeness &lt;dbl&gt; | Contamination &lt;dbl&gt; | ⋯ ⋯ | Name &lt;chr&gt; | Fasta &lt;chr&gt; | Domain &lt;chr&gt; | Phylum &lt;chr&gt; | Class &lt;chr&gt; | Order &lt;chr&gt; | Family &lt;chr&gt; | Genus &lt;chr&gt; | Species &lt;chr&gt; | ncbi_taxid &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/1002756__metabat2__Low.024.fna.gz.fna  | 72.41 | 0.02 | 1002756__metabat2__Low.024.fna.gz  | o__Clostridiales (UID1120) | 304 | 250 | 143 | 72.41 | 0.02 | ⋯ | 1002756__metabat2__Low.024.fna.gz  | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/LLG_output/drep/drep/dereplicated_genomes/1002756__metabat2__Low.024.fna.gz.fna  | d__Bacteria | p__Firmicutes_A | c__Clostridia_A | o__Christensenellales | f__UMGS1908 | g__UMGS1908 | s__                  |  172733 |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/1006216__metabat2__High.023.fna.gz.fna | 94.35 | 0.00 | 1006216__metabat2__High.023.fna.gz | c__Clostridia (UID1118)    | 387 | 223 | 124 | 94.35 | 0.00 | ⋯ | 1006216__metabat2__High.023.fna.gz | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/LLG_output/drep/drep/dereplicated_genomes/1006216__metabat2__High.023.fna.gz.fna | d__Bacteria | p__Firmicutes_A | c__Clostridia_A | o__Christensenellales | f__CAG-138  | g__Phil1    | s__Phil1 sp001940855 | 1262781 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋱ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/MI-408-H__metabat2_high_PE.004.fna.gz.fna | 96.75 | 2.69 | MI-408-H__metabat2_high_PE.004.fna.gz | c__Clostridia (UID1118) | 387 | 223 | 124 | 96.75 | 2.69 | ⋯ | MI-408-H__metabat2_high_PE.004.fna.gz | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/LLG_output/drep/drep/dereplicated_genomes/MI-408-H__metabat2_high_PE.004.fna.gz.fna | d__Bacteria | p__Firmicutes_A | c__Clostridia_A | o__Christensenellales | f__CAG-74  | g__SFHK01 | s__SFHK01 sp004556395 | 1898207 |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/MI-423-H__metabat2_high_PE.025.fna.gz.fna | 79.38 | 0.00 | MI-423-H__metabat2_high_PE.025.fna.gz | c__Clostridia (UID1118) | 387 | 223 | 124 | 79.38 | 0.00 | ⋯ | MI-423-H__metabat2_high_PE.025.fna.gz | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/LLG_output/drep/drep/dereplicated_genomes/MI-423-H__metabat2_high_PE.025.fna.gz.fna | d__Bacteria | p__Firmicutes_A | c__Clostridia_A | o__Christensenellales | f__CAG-138 | g__RUG472 | s__RUG472 sp900545265 |  172733 |\n",
       "\n"
      ],
      "text/plain": [
       "    genome                                                                                         \n",
       "1   /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/1002756__metabat2__Low.024.fna.gz.fna    \n",
       "2   /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/1006216__metabat2__High.023.fna.gz.fna   \n",
       "⋮   ⋮                                                                                              \n",
       "652 /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/MI-408-H__metabat2_high_PE.004.fna.gz.fna\n",
       "653 /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/MI-423-H__metabat2_high_PE.025.fna.gz.fna\n",
       "    completeness contamination Bin.Id                               \n",
       "1   72.41        0.02          1002756__metabat2__Low.024.fna.gz    \n",
       "2   94.35        0.00          1006216__metabat2__High.023.fna.gz   \n",
       "⋮   ⋮            ⋮             ⋮                                    \n",
       "652 96.75        2.69          MI-408-H__metabat2_high_PE.004.fna.gz\n",
       "653 79.38        0.00          MI-423-H__metabat2_high_PE.025.fna.gz\n",
       "    Marker.lineage             X..genomes X..markers X..marker.sets\n",
       "1   o__Clostridiales (UID1120) 304        250        143           \n",
       "2   c__Clostridia (UID1118)    387        223        124           \n",
       "⋮   ⋮                          ⋮          ⋮          ⋮             \n",
       "652 c__Clostridia (UID1118)    387        223        124           \n",
       "653 c__Clostridia (UID1118)    387        223        124           \n",
       "    Completeness Contamination ⋯ Name                                 \n",
       "1   72.41        0.02          ⋯ 1002756__metabat2__Low.024.fna.gz    \n",
       "2   94.35        0.00          ⋯ 1006216__metabat2__High.023.fna.gz   \n",
       "⋮   ⋮            ⋮             ⋱ ⋮                                    \n",
       "652 96.75        2.69          ⋯ MI-408-H__metabat2_high_PE.004.fna.gz\n",
       "653 79.38        0.00          ⋯ MI-423-H__metabat2_high_PE.025.fna.gz\n",
       "    Fasta                                                                                                                                                                            \n",
       "1   /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/LLG_output/drep/drep/dereplicated_genomes/1002756__metabat2__Low.024.fna.gz.fna    \n",
       "2   /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/LLG_output/drep/drep/dereplicated_genomes/1006216__metabat2__High.023.fna.gz.fna   \n",
       "⋮   ⋮                                                                                                                                                                                \n",
       "652 /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/LLG_output/drep/drep/dereplicated_genomes/MI-408-H__metabat2_high_PE.004.fna.gz.fna\n",
       "653 /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/LLG_output/drep/drep/dereplicated_genomes/MI-423-H__metabat2_high_PE.025.fna.gz.fna\n",
       "    Domain      Phylum          Class           Order                \n",
       "1   d__Bacteria p__Firmicutes_A c__Clostridia_A o__Christensenellales\n",
       "2   d__Bacteria p__Firmicutes_A c__Clostridia_A o__Christensenellales\n",
       "⋮   ⋮           ⋮               ⋮               ⋮                    \n",
       "652 d__Bacteria p__Firmicutes_A c__Clostridia_A o__Christensenellales\n",
       "653 d__Bacteria p__Firmicutes_A c__Clostridia_A o__Christensenellales\n",
       "    Family      Genus       Species               ncbi_taxid\n",
       "1   f__UMGS1908 g__UMGS1908 s__                    172733   \n",
       "2   f__CAG-138  g__Phil1    s__Phil1 sp001940855  1262781   \n",
       "⋮   ⋮           ⋮           ⋮                     ⋮         \n",
       "652 f__CAG-74   g__SFHK01   s__SFHK01 sp004556395 1898207   \n",
       "653 f__CAG-138  g__RUG472   s__RUG472 sp900545265  172733   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# joining based on Bin.Id\n",
    "drep = drep %>%\n",
    "    inner_join(checkm, c('Bin.Id')) %>%\n",
    "    mutate(GEN = genome %>% as.character %>% basename) %>%\n",
    "    inner_join(drep_gen %>% mutate(GEN = Fasta %>% as.character %>% basename),\n",
    "               by=c('GEN')) %>%\n",
    "    inner_join(tax, c('Bin.Id'='user_genome')) #%>%\n",
    "drep"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-02-27T09:52:56.719695Z",
     "iopub.status.busy": "2021-02-27T09:52:56.717975Z",
     "iopub.status.idle": "2021-02-27T09:52:56.761857Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 13 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Order</th><th scope=col>Family</th><th scope=col>n_genomes</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>o__Christensenellales</td><td>f__Borkfalkiaceae     </td><td>  6</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>o__Christensenellales</td><td>f__CAG-138            </td><td>348</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>o__Christensenellales</td><td>f__CAG-314            </td><td>  3</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>o__Christensenellales</td><td>f__CAG-552            </td><td>  4</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>o__Christensenellales</td><td>f__CAG-74             </td><td>230</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>o__Christensenellales</td><td>f__CAG-917            </td><td> 18</td></tr>\n",
       "\t<tr><th scope=row>7</th><td>o__Christensenellales</td><td>f__Christensenellaceae</td><td>  9</td></tr>\n",
       "\t<tr><th scope=row>8</th><td>o__Christensenellales</td><td>f__GCA-900066905      </td><td>  1</td></tr>\n",
       "\t<tr><th scope=row>9</th><td>o__Christensenellales</td><td>f__MARSEILLE-P3954    </td><td>  1</td></tr>\n",
       "\t<tr><th scope=row>10</th><td>o__Christensenellales</td><td>f__QALW01             </td><td> 15</td></tr>\n",
       "\t<tr><th scope=row>11</th><td>o__Christensenellales</td><td>f__QAND01             </td><td> 16</td></tr>\n",
       "\t<tr><th scope=row>12</th><td>o__Christensenellales</td><td>f__SZUA-584           </td><td>  1</td></tr>\n",
       "\t<tr><th scope=row>13</th><td>o__Christensenellales</td><td>f__UMGS1908           </td><td>  1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 13 × 3\n",
       "\\begin{tabular}{r|lll}\n",
       "  & Order & Family & n\\_genomes\\\\\n",
       "  & <chr> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & o\\_\\_Christensenellales & f\\_\\_Borkfalkiaceae      &   6\\\\\n",
       "\t2 & o\\_\\_Christensenellales & f\\_\\_CAG-138             & 348\\\\\n",
       "\t3 & o\\_\\_Christensenellales & f\\_\\_CAG-314             &   3\\\\\n",
       "\t4 & o\\_\\_Christensenellales & f\\_\\_CAG-552             &   4\\\\\n",
       "\t5 & o\\_\\_Christensenellales & f\\_\\_CAG-74              & 230\\\\\n",
       "\t6 & o\\_\\_Christensenellales & f\\_\\_CAG-917             &  18\\\\\n",
       "\t7 & o\\_\\_Christensenellales & f\\_\\_Christensenellaceae &   9\\\\\n",
       "\t8 & o\\_\\_Christensenellales & f\\_\\_GCA-900066905       &   1\\\\\n",
       "\t9 & o\\_\\_Christensenellales & f\\_\\_MARSEILLE-P3954     &   1\\\\\n",
       "\t10 & o\\_\\_Christensenellales & f\\_\\_QALW01              &  15\\\\\n",
       "\t11 & o\\_\\_Christensenellales & f\\_\\_QAND01              &  16\\\\\n",
       "\t12 & o\\_\\_Christensenellales & f\\_\\_SZUA-584            &   1\\\\\n",
       "\t13 & o\\_\\_Christensenellales & f\\_\\_UMGS1908            &   1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 13 × 3\n",
       "\n",
       "| <!--/--> | Order &lt;chr&gt; | Family &lt;chr&gt; | n_genomes &lt;int&gt; |\n",
       "|---|---|---|---|\n",
       "| 1 | o__Christensenellales | f__Borkfalkiaceae      |   6 |\n",
       "| 2 | o__Christensenellales | f__CAG-138             | 348 |\n",
       "| 3 | o__Christensenellales | f__CAG-314             |   3 |\n",
       "| 4 | o__Christensenellales | f__CAG-552             |   4 |\n",
       "| 5 | o__Christensenellales | f__CAG-74              | 230 |\n",
       "| 6 | o__Christensenellales | f__CAG-917             |  18 |\n",
       "| 7 | o__Christensenellales | f__Christensenellaceae |   9 |\n",
       "| 8 | o__Christensenellales | f__GCA-900066905       |   1 |\n",
       "| 9 | o__Christensenellales | f__MARSEILLE-P3954     |   1 |\n",
       "| 10 | o__Christensenellales | f__QALW01              |  15 |\n",
       "| 11 | o__Christensenellales | f__QAND01              |  16 |\n",
       "| 12 | o__Christensenellales | f__SZUA-584            |   1 |\n",
       "| 13 | o__Christensenellales | f__UMGS1908            |   1 |\n",
       "\n"
      ],
      "text/plain": [
       "   Order                 Family                 n_genomes\n",
       "1  o__Christensenellales f__Borkfalkiaceae        6      \n",
       "2  o__Christensenellales f__CAG-138             348      \n",
       "3  o__Christensenellales f__CAG-314               3      \n",
       "4  o__Christensenellales f__CAG-552               4      \n",
       "5  o__Christensenellales f__CAG-74              230      \n",
       "6  o__Christensenellales f__CAG-917              18      \n",
       "7  o__Christensenellales f__Christensenellaceae   9      \n",
       "8  o__Christensenellales f__GCA-900066905         1      \n",
       "9  o__Christensenellales f__MARSEILLE-P3954       1      \n",
       "10 o__Christensenellales f__QALW01               15      \n",
       "11 o__Christensenellales f__QAND01               16      \n",
       "12 o__Christensenellales f__SZUA-584              1      \n",
       "13 o__Christensenellales f__UMGS1908              1      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# summarizing the taxonomy\n",
    "df.dims(20)\n",
    "drep %>%\n",
    "    group_by(Order, Family) %>%\n",
    "    summarize(n_genomes = n(), .groups='drop')\n",
    "df.dims()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-02-27T09:52:56.767201Z",
     "iopub.status.busy": "2021-02-27T09:52:56.764281Z",
     "iopub.status.idle": "2021-02-27T09:52:56.814865Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 265 × 47</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>genome</th><th scope=col>completeness</th><th scope=col>contamination</th><th scope=col>Bin.Id</th><th scope=col>Marker.lineage</th><th scope=col>X..genomes</th><th scope=col>X..markers</th><th scope=col>X..marker.sets</th><th scope=col>Completeness</th><th scope=col>Contamination</th><th scope=col>⋯</th><th scope=col>Name</th><th scope=col>Fasta</th><th scope=col>Domain</th><th scope=col>Phylum</th><th scope=col>Class</th><th scope=col>Order</th><th scope=col>Family</th><th scope=col>Genus</th><th scope=col>Species</th><th scope=col>ncbi_taxid</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/1006216__metabat2__High.023.fna.gz.fna   </td><td>94.35</td><td>0</td><td>1006216__metabat2__High.023.fna.gz   </td><td>c__Clostridia (UID1118)</td><td>387</td><td>223</td><td>124</td><td>94.35</td><td>0</td><td>⋯</td><td>1006216__metabat2__High.023.fna.gz   </td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/LLG_output/drep/drep/dereplicated_genomes/1006216__metabat2__High.023.fna.gz.fna   </td><td>d__Bacteria</td><td>p__Firmicutes_A</td><td>c__Clostridia_A</td><td>o__Christensenellales</td><td>f__CAG-138</td><td>g__Phil1</td><td>s__Phil1 sp001940855</td><td>1262781</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/ERR3450721__metabat2__High.043.fna.gz.fna</td><td>96.77</td><td>0</td><td>ERR3450721__metabat2__High.043.fna.gz</td><td>c__Clostridia (UID1118)</td><td>387</td><td>223</td><td>124</td><td>96.77</td><td>0</td><td>⋯</td><td>ERR3450721__metabat2__High.043.fna.gz</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/LLG_output/drep/drep/dereplicated_genomes/ERR3450721__metabat2__High.043.fna.gz.fna</td><td>d__Bacteria</td><td>p__Firmicutes_A</td><td>c__Clostridia_A</td><td>o__Christensenellales</td><td>f__CAG-138</td><td>g__Phil1</td><td>s__Phil1 sp001940855</td><td>1262781</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋱</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/MI-326-H__metabat2_low_PE.034.fna.gz.fna </td><td>99.19</td><td>1.01</td><td>MI-326-H__metabat2_low_PE.034.fna.gz </td><td>c__Clostridia (UID1118)</td><td>387</td><td>223</td><td>124</td><td>99.19</td><td>1.01</td><td>⋯</td><td>MI-326-H__metabat2_low_PE.034.fna.gz </td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/LLG_output/drep/drep/dereplicated_genomes/MI-326-H__metabat2_low_PE.034.fna.gz.fna </td><td>d__Bacteria</td><td>p__Firmicutes_A</td><td>c__Clostridia_A</td><td>o__Christensenellales</td><td>f__CAG-74</td><td>g__UBA11524</td><td>s__UBA11524 sp000437595</td><td>1262897</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/MI-408-H__metabat2_high_PE.004.fna.gz.fna</td><td>96.75</td><td>2.69</td><td>MI-408-H__metabat2_high_PE.004.fna.gz</td><td>c__Clostridia (UID1118)</td><td>387</td><td>223</td><td>124</td><td>96.75</td><td>2.69</td><td>⋯</td><td>MI-408-H__metabat2_high_PE.004.fna.gz</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/LLG_output/drep/drep/dereplicated_genomes/MI-408-H__metabat2_high_PE.004.fna.gz.fna</td><td>d__Bacteria</td><td>p__Firmicutes_A</td><td>c__Clostridia_A</td><td>o__Christensenellales</td><td>f__CAG-74</td><td>g__SFHK01  </td><td>s__SFHK01 sp004556395  </td><td>1898207</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 265 × 47\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " genome & completeness & contamination & Bin.Id & Marker.lineage & X..genomes & X..markers & X..marker.sets & Completeness & Contamination & ⋯ & Name & Fasta & Domain & Phylum & Class & Order & Family & Genus & Species & ncbi\\_taxid\\\\\n",
       " <chr> & <dbl> & <dbl> & <chr> & <chr> & <int> & <int> & <int> & <dbl> & <dbl> & ⋯ & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/1006216\\_\\_metabat2\\_\\_High.023.fna.gz.fna    & 94.35 & 0 & 1006216\\_\\_metabat2\\_\\_High.023.fna.gz    & c\\_\\_Clostridia (UID1118) & 387 & 223 & 124 & 94.35 & 0 & ⋯ & 1006216\\_\\_metabat2\\_\\_High.023.fna.gz    & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/christensenellales/genomes/LLG\\_output/drep/drep/dereplicated\\_genomes/1006216\\_\\_metabat2\\_\\_High.023.fna.gz.fna    & d\\_\\_Bacteria & p\\_\\_Firmicutes\\_A & c\\_\\_Clostridia\\_A & o\\_\\_Christensenellales & f\\_\\_CAG-138 & g\\_\\_Phil1 & s\\_\\_Phil1 sp001940855 & 1262781\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/ERR3450721\\_\\_metabat2\\_\\_High.043.fna.gz.fna & 96.77 & 0 & ERR3450721\\_\\_metabat2\\_\\_High.043.fna.gz & c\\_\\_Clostridia (UID1118) & 387 & 223 & 124 & 96.77 & 0 & ⋯ & ERR3450721\\_\\_metabat2\\_\\_High.043.fna.gz & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/christensenellales/genomes/LLG\\_output/drep/drep/dereplicated\\_genomes/ERR3450721\\_\\_metabat2\\_\\_High.043.fna.gz.fna & d\\_\\_Bacteria & p\\_\\_Firmicutes\\_A & c\\_\\_Clostridia\\_A & o\\_\\_Christensenellales & f\\_\\_CAG-138 & g\\_\\_Phil1 & s\\_\\_Phil1 sp001940855 & 1262781\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋱ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/MI-326-H\\_\\_metabat2\\_low\\_PE.034.fna.gz.fna  & 99.19 & 1.01 & MI-326-H\\_\\_metabat2\\_low\\_PE.034.fna.gz  & c\\_\\_Clostridia (UID1118) & 387 & 223 & 124 & 99.19 & 1.01 & ⋯ & MI-326-H\\_\\_metabat2\\_low\\_PE.034.fna.gz  & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/christensenellales/genomes/LLG\\_output/drep/drep/dereplicated\\_genomes/MI-326-H\\_\\_metabat2\\_low\\_PE.034.fna.gz.fna  & d\\_\\_Bacteria & p\\_\\_Firmicutes\\_A & c\\_\\_Clostridia\\_A & o\\_\\_Christensenellales & f\\_\\_CAG-74 & g\\_\\_UBA11524 & s\\_\\_UBA11524 sp000437595 & 1262897\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/MI-408-H\\_\\_metabat2\\_high\\_PE.004.fna.gz.fna & 96.75 & 2.69 & MI-408-H\\_\\_metabat2\\_high\\_PE.004.fna.gz & c\\_\\_Clostridia (UID1118) & 387 & 223 & 124 & 96.75 & 2.69 & ⋯ & MI-408-H\\_\\_metabat2\\_high\\_PE.004.fna.gz & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/christensenellales/genomes/LLG\\_output/drep/drep/dereplicated\\_genomes/MI-408-H\\_\\_metabat2\\_high\\_PE.004.fna.gz.fna & d\\_\\_Bacteria & p\\_\\_Firmicutes\\_A & c\\_\\_Clostridia\\_A & o\\_\\_Christensenellales & f\\_\\_CAG-74 & g\\_\\_SFHK01   & s\\_\\_SFHK01 sp004556395   & 1898207\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 265 × 47\n",
       "\n",
       "| genome &lt;chr&gt; | completeness &lt;dbl&gt; | contamination &lt;dbl&gt; | Bin.Id &lt;chr&gt; | Marker.lineage &lt;chr&gt; | X..genomes &lt;int&gt; | X..markers &lt;int&gt; | X..marker.sets &lt;int&gt; | Completeness &lt;dbl&gt; | Contamination &lt;dbl&gt; | ⋯ ⋯ | Name &lt;chr&gt; | Fasta &lt;chr&gt; | Domain &lt;chr&gt; | Phylum &lt;chr&gt; | Class &lt;chr&gt; | Order &lt;chr&gt; | Family &lt;chr&gt; | Genus &lt;chr&gt; | Species &lt;chr&gt; | ncbi_taxid &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/1006216__metabat2__High.023.fna.gz.fna    | 94.35 | 0 | 1006216__metabat2__High.023.fna.gz    | c__Clostridia (UID1118) | 387 | 223 | 124 | 94.35 | 0 | ⋯ | 1006216__metabat2__High.023.fna.gz    | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/LLG_output/drep/drep/dereplicated_genomes/1006216__metabat2__High.023.fna.gz.fna    | d__Bacteria | p__Firmicutes_A | c__Clostridia_A | o__Christensenellales | f__CAG-138 | g__Phil1 | s__Phil1 sp001940855 | 1262781 |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/ERR3450721__metabat2__High.043.fna.gz.fna | 96.77 | 0 | ERR3450721__metabat2__High.043.fna.gz | c__Clostridia (UID1118) | 387 | 223 | 124 | 96.77 | 0 | ⋯ | ERR3450721__metabat2__High.043.fna.gz | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/LLG_output/drep/drep/dereplicated_genomes/ERR3450721__metabat2__High.043.fna.gz.fna | d__Bacteria | p__Firmicutes_A | c__Clostridia_A | o__Christensenellales | f__CAG-138 | g__Phil1 | s__Phil1 sp001940855 | 1262781 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋱ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/MI-326-H__metabat2_low_PE.034.fna.gz.fna  | 99.19 | 1.01 | MI-326-H__metabat2_low_PE.034.fna.gz  | c__Clostridia (UID1118) | 387 | 223 | 124 | 99.19 | 1.01 | ⋯ | MI-326-H__metabat2_low_PE.034.fna.gz  | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/LLG_output/drep/drep/dereplicated_genomes/MI-326-H__metabat2_low_PE.034.fna.gz.fna  | d__Bacteria | p__Firmicutes_A | c__Clostridia_A | o__Christensenellales | f__CAG-74 | g__UBA11524 | s__UBA11524 sp000437595 | 1262897 |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/MI-408-H__metabat2_high_PE.004.fna.gz.fna | 96.75 | 2.69 | MI-408-H__metabat2_high_PE.004.fna.gz | c__Clostridia (UID1118) | 387 | 223 | 124 | 96.75 | 2.69 | ⋯ | MI-408-H__metabat2_high_PE.004.fna.gz | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/LLG_output/drep/drep/dereplicated_genomes/MI-408-H__metabat2_high_PE.004.fna.gz.fna | d__Bacteria | p__Firmicutes_A | c__Clostridia_A | o__Christensenellales | f__CAG-74 | g__SFHK01   | s__SFHK01 sp004556395   | 1898207 |\n",
       "\n"
      ],
      "text/plain": [
       "    genome                                                                                         \n",
       "1   /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/1006216__metabat2__High.023.fna.gz.fna   \n",
       "2   /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/ERR3450721__metabat2__High.043.fna.gz.fna\n",
       "⋮   ⋮                                                                                              \n",
       "264 /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/MI-326-H__metabat2_low_PE.034.fna.gz.fna \n",
       "265 /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/MI-408-H__metabat2_high_PE.004.fna.gz.fna\n",
       "    completeness contamination Bin.Id                               \n",
       "1   94.35        0             1006216__metabat2__High.023.fna.gz   \n",
       "2   96.77        0             ERR3450721__metabat2__High.043.fna.gz\n",
       "⋮   ⋮            ⋮             ⋮                                    \n",
       "264 99.19        1.01          MI-326-H__metabat2_low_PE.034.fna.gz \n",
       "265 96.75        2.69          MI-408-H__metabat2_high_PE.004.fna.gz\n",
       "    Marker.lineage          X..genomes X..markers X..marker.sets Completeness\n",
       "1   c__Clostridia (UID1118) 387        223        124            94.35       \n",
       "2   c__Clostridia (UID1118) 387        223        124            96.77       \n",
       "⋮   ⋮                       ⋮          ⋮          ⋮              ⋮           \n",
       "264 c__Clostridia (UID1118) 387        223        124            99.19       \n",
       "265 c__Clostridia (UID1118) 387        223        124            96.75       \n",
       "    Contamination ⋯ Name                                 \n",
       "1   0             ⋯ 1006216__metabat2__High.023.fna.gz   \n",
       "2   0             ⋯ ERR3450721__metabat2__High.043.fna.gz\n",
       "⋮   ⋮             ⋱ ⋮                                    \n",
       "264 1.01          ⋯ MI-326-H__metabat2_low_PE.034.fna.gz \n",
       "265 2.69          ⋯ MI-408-H__metabat2_high_PE.004.fna.gz\n",
       "    Fasta                                                                                                                                                                            \n",
       "1   /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/LLG_output/drep/drep/dereplicated_genomes/1006216__metabat2__High.023.fna.gz.fna   \n",
       "2   /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/LLG_output/drep/drep/dereplicated_genomes/ERR3450721__metabat2__High.043.fna.gz.fna\n",
       "⋮   ⋮                                                                                                                                                                                \n",
       "264 /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/LLG_output/drep/drep/dereplicated_genomes/MI-326-H__metabat2_low_PE.034.fna.gz.fna \n",
       "265 /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes/LLG_output/drep/drep/dereplicated_genomes/MI-408-H__metabat2_high_PE.004.fna.gz.fna\n",
       "    Domain      Phylum          Class           Order                \n",
       "1   d__Bacteria p__Firmicutes_A c__Clostridia_A o__Christensenellales\n",
       "2   d__Bacteria p__Firmicutes_A c__Clostridia_A o__Christensenellales\n",
       "⋮   ⋮           ⋮               ⋮               ⋮                    \n",
       "264 d__Bacteria p__Firmicutes_A c__Clostridia_A o__Christensenellales\n",
       "265 d__Bacteria p__Firmicutes_A c__Clostridia_A o__Christensenellales\n",
       "    Family     Genus       Species                 ncbi_taxid\n",
       "1   f__CAG-138 g__Phil1    s__Phil1 sp001940855    1262781   \n",
       "2   f__CAG-138 g__Phil1    s__Phil1 sp001940855    1262781   \n",
       "⋮   ⋮          ⋮           ⋮                       ⋮         \n",
       "264 f__CAG-74  g__UBA11524 s__UBA11524 sp000437595 1262897   \n",
       "265 f__CAG-74  g__SFHK01   s__SFHK01 sp004556395   1898207   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# filtering by quality\n",
    "hq_genomes = drep %>%\n",
    "    filter(completeness >= 90,\n",
    "           contamination < 5,\n",
    "           Strain.heterogeneity < 50,\n",
    "           N50..contigs. >= 20000)\n",
    "hq_genomes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-02-27T09:52:56.818995Z",
     "iopub.status.busy": "2021-02-27T09:52:56.817504Z",
     "iopub.status.idle": "2021-02-27T09:52:56.857209Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 8 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Order</th><th scope=col>Family</th><th scope=col>n_genomes</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>o__Christensenellales</td><td>f__Borkfalkiaceae     </td><td>  3</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>o__Christensenellales</td><td>f__CAG-138            </td><td>190</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>o__Christensenellales</td><td>f__CAG-74             </td><td> 61</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>o__Christensenellales</td><td>f__Christensenellaceae</td><td>  4</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>o__Christensenellales</td><td>f__GCA-900066905      </td><td>  1</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>o__Christensenellales</td><td>f__MARSEILLE-P3954    </td><td>  1</td></tr>\n",
       "\t<tr><th scope=row>7</th><td>o__Christensenellales</td><td>f__QALW01             </td><td>  2</td></tr>\n",
       "\t<tr><th scope=row>8</th><td>o__Christensenellales</td><td>f__QAND01             </td><td>  3</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 8 × 3\n",
       "\\begin{tabular}{r|lll}\n",
       "  & Order & Family & n\\_genomes\\\\\n",
       "  & <chr> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & o\\_\\_Christensenellales & f\\_\\_Borkfalkiaceae      &   3\\\\\n",
       "\t2 & o\\_\\_Christensenellales & f\\_\\_CAG-138             & 190\\\\\n",
       "\t3 & o\\_\\_Christensenellales & f\\_\\_CAG-74              &  61\\\\\n",
       "\t4 & o\\_\\_Christensenellales & f\\_\\_Christensenellaceae &   4\\\\\n",
       "\t5 & o\\_\\_Christensenellales & f\\_\\_GCA-900066905       &   1\\\\\n",
       "\t6 & o\\_\\_Christensenellales & f\\_\\_MARSEILLE-P3954     &   1\\\\\n",
       "\t7 & o\\_\\_Christensenellales & f\\_\\_QALW01              &   2\\\\\n",
       "\t8 & o\\_\\_Christensenellales & f\\_\\_QAND01              &   3\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 8 × 3\n",
       "\n",
       "| <!--/--> | Order &lt;chr&gt; | Family &lt;chr&gt; | n_genomes &lt;int&gt; |\n",
       "|---|---|---|---|\n",
       "| 1 | o__Christensenellales | f__Borkfalkiaceae      |   3 |\n",
       "| 2 | o__Christensenellales | f__CAG-138             | 190 |\n",
       "| 3 | o__Christensenellales | f__CAG-74              |  61 |\n",
       "| 4 | o__Christensenellales | f__Christensenellaceae |   4 |\n",
       "| 5 | o__Christensenellales | f__GCA-900066905       |   1 |\n",
       "| 6 | o__Christensenellales | f__MARSEILLE-P3954     |   1 |\n",
       "| 7 | o__Christensenellales | f__QALW01              |   2 |\n",
       "| 8 | o__Christensenellales | f__QAND01              |   3 |\n",
       "\n"
      ],
      "text/plain": [
       "  Order                 Family                 n_genomes\n",
       "1 o__Christensenellales f__Borkfalkiaceae        3      \n",
       "2 o__Christensenellales f__CAG-138             190      \n",
       "3 o__Christensenellales f__CAG-74               61      \n",
       "4 o__Christensenellales f__Christensenellaceae   4      \n",
       "5 o__Christensenellales f__GCA-900066905         1      \n",
       "6 o__Christensenellales f__MARSEILLE-P3954       1      \n",
       "7 o__Christensenellales f__QALW01                2      \n",
       "8 o__Christensenellales f__QAND01                3      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# summarizing the taxonomy\n",
    "df.dims(30)\n",
    "hq_genomes %>%\n",
    "    group_by(Order, Family) %>%\n",
    "    summarize(n_genomes = n(), .groups='drop')\n",
    "df.dims()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-02-27T09:52:56.861530Z",
     "iopub.status.busy": "2021-02-27T09:52:56.859830Z",
     "iopub.status.idle": "2021-02-27T09:52:56.922207Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A matrix: 1 × 8 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Min.</th><th scope=col>1st Qu.</th><th scope=col>Median</th><th scope=col>Mean</th><th scope=col>3rd Qu.</th><th scope=col>Max.</th><th scope=col>sd</th><th scope=col>sd_err_of_mean</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Completeness</th><td>90.25</td><td>96.37</td><td>97.58</td><td>97.10242</td><td>98.6</td><td>99.6</td><td>3.304</td><td>1.349</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 1 × 8 of type dbl\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & Min. & 1st Qu. & Median & Mean & 3rd Qu. & Max. & sd & sd\\_err\\_of\\_mean\\\\\n",
       "\\hline\n",
       "\tCompleteness & 90.25 & 96.37 & 97.58 & 97.10242 & 98.6 & 99.6 & 3.304 & 1.349\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 1 × 8 of type dbl\n",
       "\n",
       "| <!--/--> | Min. | 1st Qu. | Median | Mean | 3rd Qu. | Max. | sd | sd_err_of_mean |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Completeness | 90.25 | 96.37 | 97.58 | 97.10242 | 98.6 | 99.6 | 3.304 | 1.349 |\n",
       "\n"
      ],
      "text/plain": [
       "             Min.  1st Qu. Median Mean     3rd Qu. Max. sd    sd_err_of_mean\n",
       "Completeness 90.25 96.37   97.58  97.10242 98.6    99.6 3.304 1.349         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A matrix: 1 × 8 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Min.</th><th scope=col>1st Qu.</th><th scope=col>Median</th><th scope=col>Mean</th><th scope=col>3rd Qu.</th><th scope=col>Max.</th><th scope=col>sd</th><th scope=col>sd_err_of_mean</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>No. of contigs</th><td>1</td><td>45</td><td>66</td><td>75.53585</td><td>98</td><td>222</td><td>74.891</td><td>30.574</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 1 × 8 of type dbl\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & Min. & 1st Qu. & Median & Mean & 3rd Qu. & Max. & sd & sd\\_err\\_of\\_mean\\\\\n",
       "\\hline\n",
       "\tNo. of contigs & 1 & 45 & 66 & 75.53585 & 98 & 222 & 74.891 & 30.574\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 1 × 8 of type dbl\n",
       "\n",
       "| <!--/--> | Min. | 1st Qu. | Median | Mean | 3rd Qu. | Max. | sd | sd_err_of_mean |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| No. of contigs | 1 | 45 | 66 | 75.53585 | 98 | 222 | 74.891 | 30.574 |\n",
       "\n"
      ],
      "text/plain": [
       "               Min. 1st Qu. Median Mean     3rd Qu. Max. sd     sd_err_of_mean\n",
       "No. of contigs 1    45      66     75.53585 98      222  74.891 30.574        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A matrix: 1 × 8 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Min.</th><th scope=col>1st Qu.</th><th scope=col>Median</th><th scope=col>Mean</th><th scope=col>3rd Qu.</th><th scope=col>Max.</th><th scope=col>sd</th><th scope=col>sd_err_of_mean</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Mean contig length</th><td>12558</td><td>23536</td><td>33875</td><td>72444.58</td><td>48014</td><td>2969292</td><td>1196841</td><td>488608.1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 1 × 8 of type dbl\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & Min. & 1st Qu. & Median & Mean & 3rd Qu. & Max. & sd & sd\\_err\\_of\\_mean\\\\\n",
       "\\hline\n",
       "\tMean contig length & 12558 & 23536 & 33875 & 72444.58 & 48014 & 2969292 & 1196841 & 488608.1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 1 × 8 of type dbl\n",
       "\n",
       "| <!--/--> | Min. | 1st Qu. | Median | Mean | 3rd Qu. | Max. | sd | sd_err_of_mean |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Mean contig length | 12558 | 23536 | 33875 | 72444.58 | 48014 | 2969292 | 1196841 | 488608.1 |\n",
       "\n"
      ],
      "text/plain": [
       "                   Min.  1st Qu. Median Mean     3rd Qu. Max.    sd     \n",
       "Mean contig length 12558 23536   33875  72444.58 48014   2969292 1196841\n",
       "                   sd_err_of_mean\n",
       "Mean contig length 488608.1      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A matrix: 1 × 8 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Min.</th><th scope=col>1st Qu.</th><th scope=col>Median</th><th scope=col>Mean</th><th scope=col>3rd Qu.</th><th scope=col>Max.</th><th scope=col>sd</th><th scope=col>sd_err_of_mean</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>No. of genes</th><td>1449</td><td>1894</td><td>2059</td><td>2125.928</td><td>2420</td><td>3023</td><td>529.148</td><td>216.024</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 1 × 8 of type dbl\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & Min. & 1st Qu. & Median & Mean & 3rd Qu. & Max. & sd & sd\\_err\\_of\\_mean\\\\\n",
       "\\hline\n",
       "\tNo. of genes & 1449 & 1894 & 2059 & 2125.928 & 2420 & 3023 & 529.148 & 216.024\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 1 × 8 of type dbl\n",
       "\n",
       "| <!--/--> | Min. | 1st Qu. | Median | Mean | 3rd Qu. | Max. | sd | sd_err_of_mean |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| No. of genes | 1449 | 1894 | 2059 | 2125.928 | 2420 | 3023 | 529.148 | 216.024 |\n",
       "\n"
      ],
      "text/plain": [
       "             Min. 1st Qu. Median Mean     3rd Qu. Max. sd      sd_err_of_mean\n",
       "No. of genes 1449 1894    2059   2125.928 2420    3023 529.148 216.024       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A matrix: 1 × 8 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Min.</th><th scope=col>1st Qu.</th><th scope=col>Median</th><th scope=col>Mean</th><th scope=col>3rd Qu.</th><th scope=col>Max.</th><th scope=col>sd</th><th scope=col>sd_err_of_mean</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>N50</th><td>20160</td><td>36095</td><td>55481</td><td>102360</td><td>82483</td><td>2969292</td><td>1188370</td><td>485149.8</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 1 × 8 of type dbl\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & Min. & 1st Qu. & Median & Mean & 3rd Qu. & Max. & sd & sd\\_err\\_of\\_mean\\\\\n",
       "\\hline\n",
       "\tN50 & 20160 & 36095 & 55481 & 102360 & 82483 & 2969292 & 1188370 & 485149.8\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 1 × 8 of type dbl\n",
       "\n",
       "| <!--/--> | Min. | 1st Qu. | Median | Mean | 3rd Qu. | Max. | sd | sd_err_of_mean |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| N50 | 20160 | 36095 | 55481 | 102360 | 82483 | 2969292 | 1188370 | 485149.8 |\n",
       "\n"
      ],
      "text/plain": [
       "    Min.  1st Qu. Median Mean   3rd Qu. Max.    sd      sd_err_of_mean\n",
       "N50 20160 36095   55481  102360 82483   2969292 1188370 485149.8      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# summarizing\n",
    "hq_genomes$Completeness %>% summary_x('Completeness')\n",
    "hq_genomes$X..contigs %>% summary_x('No. of contigs')\n",
    "hq_genomes$Mean.contig.length..bp. %>% summary_x('Mean contig length')\n",
    "hq_genomes$X..predicted.genes %>% summary_x('No. of genes')\n",
    "hq_genomes$N50..contigs. %>% summary_x('N50')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-02-27T09:52:56.926579Z",
     "iopub.status.busy": "2021-02-27T09:52:56.924945Z",
     "iopub.status.idle": "2021-02-27T09:52:56.956361Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "File written: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/christensenellales/genomes//LLG_output/samples_genomes_hq.txt \n"
     ]
    }
   ],
   "source": [
    " # writing samples table for LLPRIMER\n",
    "outfile = file.path(work_dir, 'LLG_output', 'samples_genomes_hq.txt')\n",
    "hq_genomes %>%\n",
    "    select(Bin.Id, Fasta, Domain, ncbi_taxid) %>%\n",
    "    rename('Taxon' = Bin.Id,\n",
    "           'Taxid' = ncbi_taxid) %>%\n",
    "    mutate(Taxon = gsub('_chromosome.+', '', Taxon),\n",
    "           Taxon = gsub('_bin_.+', '', Taxon),\n",
    "           Taxon = gsub('_genomic', '', Taxon),\n",
    "           Taxon = gsub('_annotated_assembly', '', Taxon)) %>%\n",
    "    write_table(outfile)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# sessionInfo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-02-26T12:29:39.974255Z",
     "iopub.status.busy": "2021-02-26T12:29:39.972564Z",
     "iopub.status.idle": "2021-02-26T12:29:41.049449Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "R version 4.0.3 (2020-10-10)\n",
       "Platform: x86_64-conda-linux-gnu (64-bit)\n",
       "Running under: Ubuntu 18.04.5 LTS\n",
       "\n",
       "Matrix products: default\n",
       "BLAS/LAPACK: /ebio/abt3_projects/Anxiety_Twins_Metagenomes/envs/tidyverse2/lib/libopenblasp-r0.3.12.so\n",
       "\n",
       "locale:\n",
       " [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              \n",
       " [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    \n",
       " [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   \n",
       " [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 \n",
       " [9] LC_ADDRESS=C               LC_TELEPHONE=C            \n",
       "[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       \n",
       "\n",
       "attached base packages:\n",
       "[1] stats     graphics  grDevices utils     datasets  methods   base     \n",
       "\n",
       "other attached packages:\n",
       "[1] curl_4.3          LeyLabRMisc_0.1.8 ggplot2_3.3.3     tidytable_0.5.8  \n",
       "[5] data.table_1.13.6 tidyr_1.1.2       dplyr_1.0.3      \n",
       "\n",
       "loaded via a namespace (and not attached):\n",
       " [1] magrittr_2.0.1    munsell_0.5.0     tidyselect_1.1.0  uuid_0.1-4       \n",
       " [5] colorspace_2.0-0  R6_2.5.0          rlang_0.4.10      tools_4.0.3      \n",
       " [9] grid_4.0.3        gtable_0.3.0      withr_2.4.1       htmltools_0.5.1.1\n",
       "[13] ellipsis_0.3.1    digest_0.6.27     tibble_3.0.6      lifecycle_0.2.0  \n",
       "[17] crayon_1.4.0      IRdisplay_1.0     farver_2.0.3      purrr_0.3.4      \n",
       "[21] repr_1.1.3        base64enc_0.1-3   vctrs_0.3.6       IRkernel_1.1.1   \n",
       "[25] glue_1.4.2        evaluate_0.14     labeling_0.4.2    pbdZMQ_0.3-4     \n",
       "[29] compiler_4.0.3    pillar_1.4.7      scales_1.1.1      generics_0.1.0   \n",
       "[33] jsonlite_1.7.2    pkgconfig_2.0.3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "sessionInfo()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R [conda env:Anxiety_Twins_Metagenomes-tidyverse2]",
   "language": "R",
   "name": "conda-env-Anxiety_Twins_Metagenomes-tidyverse2-r"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.0.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
