{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Goal\n",
    "\n",
    "* Primer design for clade of interest"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Var"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-15T09:35:33.456316Z",
     "iopub.status.busy": "2021-01-15T09:35:33.452866Z",
     "iopub.status.idle": "2021-01-15T09:35:33.476102Z"
    }
   },
   "outputs": [],
   "source": [
    "base_dir = '/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/'\n",
    "clade = 'Anaerotruncus'\n",
    "taxid = 244127 "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Init"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-15T09:35:33.605624Z",
     "iopub.status.busy": "2021-01-15T09:35:33.603562Z",
     "iopub.status.idle": "2021-01-15T09:35:33.630921Z"
    }
   },
   "outputs": [],
   "source": [
    "library(dplyr)\n",
    "library(tidyr)\n",
    "library(ggplot2)\n",
    "library(LeyLabRMisc)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-15T09:35:33.928966Z",
     "iopub.status.busy": "2021-01-15T09:35:33.928147Z",
     "iopub.status.idle": "2021-01-15T09:35:33.953655Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Directory already exists: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted//Anaerotruncus \n"
     ]
    }
   ],
   "source": [
    "df.dims()\n",
    "work_dir = file.path(base_dir, clade)\n",
    "make_dir(work_dir)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Genome download\n",
    "\n",
    "* Downloading genomes from NCBI"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-12T13:07:32.874718Z",
     "iopub.status.busy": "2021-01-12T13:07:32.867938Z",
     "iopub.status.idle": "2021-01-12T13:07:32.985785Z",
     "shell.execute_reply": "2021-01-12T13:07:32.983225Z"
    }
   },
   "source": [
    "```\n",
    "OUTDIR=/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/\n",
    "mkdir -p $OUTDIR\n",
    "ncbi-genome-download -p 12 -s genbank -F fasta --genera Anaerotruncus -o $OUTDIR bacteria\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Genome quality\n",
    "\n",
    "* Filtering genomes by quality"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-12T13:39:08.270473Z",
     "iopub.status.busy": "2021-01-12T13:39:08.268233Z",
     "iopub.status.idle": "2021-01-12T13:39:08.415063Z",
     "shell.execute_reply": "2021-01-12T13:39:08.413708Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 31 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Name</th><th scope=col>Fasta</th><th scope=col>Domain</th><th scope=col>Taxid</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>GCA_000154565.1_ASM15456v1_genomic     </td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/genbank/bacteria/GCA_000154565.1/GCA_000154565.1_ASM15456v1_genomic.fna.gz     </td><td>Bacteria</td><td>244127</td></tr>\n",
       "\t<tr><td>GCA_000403395.2_Anae_bact_G3_V1_genomic</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/genbank/bacteria/GCA_000403395.2/GCA_000403395.2_Anae_bact_G3_V1_genomic.fna.gz</td><td>Bacteria</td><td>244127</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>GCA_902363605.1_MGYG-HGUT-00131_genomic   </td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/genbank/bacteria/GCA_902363605.1/GCA_902363605.1_MGYG-HGUT-00131_genomic.fna.gz   </td><td>Bacteria</td><td>244127</td></tr>\n",
       "\t<tr><td>GCA_904384175.1_Chicken_20_mag_154_genomic</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/genbank/bacteria/GCA_904384175.1/GCA_904384175.1_Chicken_20_mag_154_genomic.fna.gz</td><td>Bacteria</td><td>244127</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 31 × 4\n",
       "\\begin{tabular}{llll}\n",
       " Name & Fasta & Domain & Taxid\\\\\n",
       " <chr> & <chr> & <fct> & <dbl>\\\\\n",
       "\\hline\n",
       "\t GCA\\_000154565.1\\_ASM15456v1\\_genomic      & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/Anaerotruncus/genbank/bacteria/GCA\\_000154565.1/GCA\\_000154565.1\\_ASM15456v1\\_genomic.fna.gz      & Bacteria & 244127\\\\\n",
       "\t GCA\\_000403395.2\\_Anae\\_bact\\_G3\\_V1\\_genomic & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/Anaerotruncus/genbank/bacteria/GCA\\_000403395.2/GCA\\_000403395.2\\_Anae\\_bact\\_G3\\_V1\\_genomic.fna.gz & Bacteria & 244127\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t GCA\\_902363605.1\\_MGYG-HGUT-00131\\_genomic    & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/Anaerotruncus/genbank/bacteria/GCA\\_902363605.1/GCA\\_902363605.1\\_MGYG-HGUT-00131\\_genomic.fna.gz    & Bacteria & 244127\\\\\n",
       "\t GCA\\_904384175.1\\_Chicken\\_20\\_mag\\_154\\_genomic & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/Anaerotruncus/genbank/bacteria/GCA\\_904384175.1/GCA\\_904384175.1\\_Chicken\\_20\\_mag\\_154\\_genomic.fna.gz & Bacteria & 244127\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 31 × 4\n",
       "\n",
       "| Name &lt;chr&gt; | Fasta &lt;chr&gt; | Domain &lt;fct&gt; | Taxid &lt;dbl&gt; |\n",
       "|---|---|---|---|\n",
       "| GCA_000154565.1_ASM15456v1_genomic      | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/genbank/bacteria/GCA_000154565.1/GCA_000154565.1_ASM15456v1_genomic.fna.gz      | Bacteria | 244127 |\n",
       "| GCA_000403395.2_Anae_bact_G3_V1_genomic | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/genbank/bacteria/GCA_000403395.2/GCA_000403395.2_Anae_bact_G3_V1_genomic.fna.gz | Bacteria | 244127 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| GCA_902363605.1_MGYG-HGUT-00131_genomic    | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/genbank/bacteria/GCA_902363605.1/GCA_902363605.1_MGYG-HGUT-00131_genomic.fna.gz    | Bacteria | 244127 |\n",
       "| GCA_904384175.1_Chicken_20_mag_154_genomic | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/genbank/bacteria/GCA_904384175.1/GCA_904384175.1_Chicken_20_mag_154_genomic.fna.gz | Bacteria | 244127 |\n",
       "\n"
      ],
      "text/plain": [
       "   Name                                      \n",
       "1  GCA_000154565.1_ASM15456v1_genomic        \n",
       "2  GCA_000403395.2_Anae_bact_G3_V1_genomic   \n",
       "⋮  ⋮                                         \n",
       "30 GCA_902363605.1_MGYG-HGUT-00131_genomic   \n",
       "31 GCA_904384175.1_Chicken_20_mag_154_genomic\n",
       "   Fasta                                                                                                                                                                              \n",
       "1  /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/genbank/bacteria/GCA_000154565.1/GCA_000154565.1_ASM15456v1_genomic.fna.gz        \n",
       "2  /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/genbank/bacteria/GCA_000403395.2/GCA_000403395.2_Anae_bact_G3_V1_genomic.fna.gz   \n",
       "⋮  ⋮                                                                                                                                                                                  \n",
       "30 /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/genbank/bacteria/GCA_902363605.1/GCA_902363605.1_MGYG-HGUT-00131_genomic.fna.gz   \n",
       "31 /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/genbank/bacteria/GCA_904384175.1/GCA_904384175.1_Chicken_20_mag_154_genomic.fna.gz\n",
       "   Domain   Taxid \n",
       "1  Bacteria 244127\n",
       "2  Bacteria 244127\n",
       "⋮  ⋮        ⋮     \n",
       "30 Bacteria 244127\n",
       "31 Bacteria 244127"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "File written: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted//Anaerotruncus/genbank/samples.txt \n"
     ]
    }
   ],
   "source": [
    "D = file.path(base_dir, clade, 'genbank')\n",
    "files = list_files(D, '.fna.gz')\n",
    "samps = data.frame(Name = files %>% as.character %>% basename,\n",
    "                   Fasta = files,\n",
    "                   Domain = 'Bacteria',\n",
    "                   Taxid = taxid) %>%\n",
    "    mutate(Name = gsub('\\\\.fna\\\\.gz$', '', Name),\n",
    "           Fasta = gsub('/+', '/', Fasta))\n",
    "samps\n",
    "\n",
    "# writing file\n",
    "outfile = file.path(D, 'samples.txt')\n",
    "write_table(samps, outfile)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### LLG"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Config"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-12T13:41:55.331362Z",
     "iopub.status.busy": "2021-01-12T13:41:55.329331Z",
     "iopub.status.idle": "2021-01-12T13:41:55.365215Z",
     "shell.execute_reply": "2021-01-12T13:41:55.362970Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# table with genome --> fasta_file information\n",
      "samples_file: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/genbank/samples.txt\n",
      "\n",
      "# output location\n",
      "output_dir: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/LLG_output/\n",
      "\n",
      "# temporary file directory (your username will be added automatically)\n",
      "tmp_dir: /ebio/abt3_scratch/\n",
      "\n",
      "# batch processing of genomes for certain steps\n",
      "## increase to better parallelize\n",
      "batches: 2 \n",
      "\n",
      "# Domain of genomes ('Archaea' or 'Bacteria)\n",
      "## Use \"Skip\" if provided as a \"Domain\" column in the genome table\n",
      "Domain: Skip\n",
      "\n",
      "# software parameters\n",
      "# Use \"Skip\" to skip any of these steps. If no params for rule, use \"\"\n",
      "# dRep MAGs are not further analyzed, but you can de-rep & then use the de-rep genome table as input.\n",
      "params:\n",
      "  ionice: -c 3\n",
      "  # assembly assessment\n",
      "  seqkit: \"\"\n",
      "  quast: Skip #\"\"\n",
      "  multiqc_on_quast: \"\" \n",
      "  checkm: \"\"\n",
      "  # de-replication (requires checkm)\n",
      "  drep: -comp 90 -con 5 -sa 0.999\n",
      "  # taxonomy\n",
      "  sourmash:\n",
      "    compute: Skip #--scaled 10000 -k 31\n",
      "    gather: -k 31\n",
      "  gtdbtk:\n",
      "    classify_wf: --min_perc_aa 10\n",
      "  # genome pairwise ANI  \n",
      "  fastani: Skip #--fragLen 3000 --minFraction 0.2 -k 16\n",
      "  dashing: Skip #-k 31 --full-tsv\n",
      "  comparem_aai: Skip #--evalue 0.001\n",
      "  # gene annotation\n",
      "  gene_call:\n",
      "    prokka: Skip #\"\" \n",
      "    multiqc_on_prokka: \"\"\n",
      "    prodigal: Skip #\"\"\n",
      "  eggnog_mapper: \"\"\n",
      "  eggnog_mapper_annot: \"\"\n",
      "  # rRNA (16S alignment & phylogeny)\n",
      "  barrnap: Skip #--lencutoff 0.8\n",
      "  vsearch_per_genome_drep: --id 0.95    # Skip to prevent drep of 16S copies within each genome\n",
      "  qiime2_fasttree: \"\"\n",
      "  qiime2_iqtree: --p-alrt 1000 --p-abayes --p-lbp 1000  --p-substitution-model 'GTR+I+G'\n",
      "  # genome phylogeny\n",
      "  phylophlan_config: Skip #--map_dna diamond --db_aa diamond --map_aa diamond --msa mafft --trim trimal --tree1 fasttree --tree2 raxml\n",
      "  phylophlan: --diversity high --min_num_markers 50 --accurate\n",
      "  # phenotype\n",
      "  traitar: Skip #\"\"\n",
      "  # biosynthetic gene clusters (BGCs)\n",
      "  antismash: Skip #--cb-knownclusters --cb-subclusters --asf  \n",
      "  DeepBGC: Skip #--score 0.5 --classifier-score 0.5 --prodigal-meta-mode\n",
      "  # antimicrobial resistance (AMR)\n",
      "  abricate: Skip #--minid 75 --mincov 80\n",
      "  # CRISPRs\n",
      "  cctyper: Skip #--prodigal meta\n",
      "  \n",
      "# databases\n",
      "databases:\n",
      "  checkM_data: /ebio/abt3_projects/databases_no-backup/checkM/\n",
      "  sourmash: /ebio/abt3_projects/databases_no-backup/sourmash/genbank-k31.sbt.json\n",
      "  sourmash_lca: /ebio/abt3_projects/databases_no-backup/sourmash/genbank-k31.lca.json.gz\n",
      "  gtdbtk: /ebio/abt3_projects/databases_no-backup/GTDB/release95/gtdbtk/db_info.md\n",
      "  phylophlan: /ebio/abt3_projects/databases_no-backup/phylophlan/PhyloPhlan/phylophlan.faa.bz2\n",
      "  eggnog: /ebio/abt3_projects/databases_no-backup/Eggnog/v2/eggnog.db\n",
      "  eggnog_diamond: /ebio/abt3_projects/databases_no-backup/Eggnog/v2/eggnog_proteins.dmnd\n",
      "  antismash: /ebio/abt3_projects/databases_no-backup/antismash/v5/\n",
      "  deepbgc: /ebio/abt3_projects/databases_no-backup/DeepBGC/\n",
      "  traitar: /ebio/abt3_projects/databases_no-backup/pfam/traitar/\n",
      "  abricate:\n",
      "    ncbi: /ebio/abt3_projects/databases_no-backup/abricate/ncbi/sequences\n",
      "    card: /ebio/abt3_projects/databases_no-backup/abricate/card/sequences\n",
      "    resfinder: /ebio/abt3_projects/databases_no-backup/abricate/resfinder/sequences\n",
      "    argannot: /ebio/abt3_projects/databases_no-backup/abricate/argannot/sequences\n",
      "    bacmet2: /ebio/abt3_projects/databases_no-backup/abricate/bacmet2/sequences\n",
      "    vfdb: /ebio/abt3_projects/databases_no-backup/abricate/vfdb/sequences\n",
      "    megares: /ebio/abt3_projects/databases_no-backup/abricate/megares/sequences\n",
      "    plasmidfinder: /ebio/abt3_projects/databases_no-backup/abricate/plasmidfinder/sequences\n",
      "\n",
      "# snakemake pipeline\n",
      "pipeline:\n",
      "  snakemake_folder: ./\n",
      "  script_folder: ./bin/scripts/\n",
      "  use_shared_mem: True\n",
      "  name: LLG"
     ]
    }
   ],
   "source": [
    "cat_file(file.path(work_dir, 'config_llg.yaml'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-12T13:14:50.416604Z",
     "iopub.status.busy": "2021-01-12T13:14:50.414179Z",
     "iopub.status.idle": "2021-01-12T13:14:50.431925Z",
     "shell.execute_reply": "2021-01-12T13:14:50.430260Z"
    }
   },
   "source": [
    "#### Run\n",
    "\n",
    "```\n",
    "(snakemake) @ rick:/ebio/abt3_projects/software/dev/ll_pipelines/llg\n",
    "$ screen -L -S llg ./snakemake_sge.sh /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/config_llg.yaml 20 -F\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-12T13:14:10.995042Z",
     "iopub.status.busy": "2021-01-12T13:14:10.992773Z",
     "iopub.status.idle": "2021-01-12T13:14:11.009275Z",
     "shell.execute_reply": "2021-01-12T13:14:11.008255Z"
    }
   },
   "source": [
    "### Samples table of high-quality genomes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-15T09:35:38.914961Z",
     "iopub.status.busy": "2021-01-15T09:35:38.912889Z",
     "iopub.status.idle": "2021-01-15T09:35:38.969821Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 31 × 33</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Bin.Id</th><th scope=col>Marker.lineage</th><th scope=col>X..genomes</th><th scope=col>X..markers</th><th scope=col>X..marker.sets</th><th scope=col>Completeness</th><th scope=col>Contamination</th><th scope=col>Strain.heterogeneity</th><th scope=col>Genome.size..bp.</th><th scope=col>X..ambiguous.bases</th><th scope=col>⋯</th><th scope=col>X0</th><th scope=col>X1</th><th scope=col>X2</th><th scope=col>X3</th><th scope=col>X4</th><th scope=col>X5.</th><th scope=col>assembly.Id</th><th scope=col>assembler.Id</th><th scope=col>taxon.Id</th><th scope=col>File</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>GCA_000154565.1_ASM15456v1_genomic</td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>97.99</td><td>0.22</td><td>0</td><td>3719688</td><td>800</td><td>⋯</td><td>3</td><td>259</td><td>1</td><td>0</td><td>0</td><td>0</td><td>|ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Anaerotruncus|LLG_output|checkM|1|checkm|markers_qa_summary.tsv.1</td><td>markers_qa_summary.tsv.1</td><td>NA</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/LLG_output/checkM/1/checkm/markers_qa_summary.tsv.1</td></tr>\n",
       "\t<tr><td>GCA_000433975.1_MGS528_genomic    </td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>97.99</td><td>0.00</td><td>0</td><td>1881083</td><td>732</td><td>⋯</td><td>3</td><td>260</td><td>0</td><td>0</td><td>0</td><td>0</td><td>|ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Anaerotruncus|LLG_output|checkM|1|checkm|markers_qa_summary.tsv.2</td><td>markers_qa_summary.tsv.2</td><td>NA</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/LLG_output/checkM/1/checkm/markers_qa_summary.tsv.2</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋱</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>GCA_900199635.1_PRJEB19460_genomic     </td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>99.33</td><td>0.00</td><td>0</td><td>3145951</td><td> 14</td><td>⋯</td><td>1</td><td>262</td><td>0</td><td>0</td><td>0</td><td>0</td><td>|ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Anaerotruncus|LLG_output|checkM|2|checkm|markers_qa_summary.tsv.14</td><td>markers_qa_summary.tsv.14</td><td>NA</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/LLG_output/checkM/2/checkm/markers_qa_summary.tsv.14</td></tr>\n",
       "\t<tr><td>GCA_902363605.1_MGYG-HGUT-00131_genomic</td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>97.99</td><td>1.34</td><td>0</td><td>3306065</td><td>265</td><td>⋯</td><td>3</td><td>257</td><td>3</td><td>0</td><td>0</td><td>0</td><td>|ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Anaerotruncus|LLG_output|checkM|2|checkm|markers_qa_summary.tsv.15</td><td>markers_qa_summary.tsv.15</td><td>NA</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/LLG_output/checkM/2/checkm/markers_qa_summary.tsv.15</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 31 × 33\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " Bin.Id & Marker.lineage & X..genomes & X..markers & X..marker.sets & Completeness & Contamination & Strain.heterogeneity & Genome.size..bp. & X..ambiguous.bases & ⋯ & X0 & X1 & X2 & X3 & X4 & X5. & assembly.Id & assembler.Id & taxon.Id & File\\\\\n",
       " <fct> & <fct> & <int> & <int> & <int> & <dbl> & <dbl> & <dbl> & <int> & <int> & ⋯ & <int> & <int> & <int> & <int> & <int> & <int> & <fct> & <fct> & <lgl> & <fct>\\\\\n",
       "\\hline\n",
       "\t GCA\\_000154565.1\\_ASM15456v1\\_genomic & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 97.99 & 0.22 & 0 & 3719688 & 800 & ⋯ & 3 & 259 & 1 & 0 & 0 & 0 & \\textbar{}ebio\\textbar{}abt3\\_projects\\textbar{}software\\textbar{}dev\\textbar{}ll\\_pipelines\\textbar{}llprimer\\textbar{}experiments\\textbar{}HMP\\_most-wanted\\textbar{}Anaerotruncus\\textbar{}LLG\\_output\\textbar{}checkM\\textbar{}1\\textbar{}checkm\\textbar{}markers\\_qa\\_summary.tsv.1 & markers\\_qa\\_summary.tsv.1 & NA & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/Anaerotruncus/LLG\\_output/checkM/1/checkm/markers\\_qa\\_summary.tsv.1\\\\\n",
       "\t GCA\\_000433975.1\\_MGS528\\_genomic     & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 97.99 & 0.00 & 0 & 1881083 & 732 & ⋯ & 3 & 260 & 0 & 0 & 0 & 0 & \\textbar{}ebio\\textbar{}abt3\\_projects\\textbar{}software\\textbar{}dev\\textbar{}ll\\_pipelines\\textbar{}llprimer\\textbar{}experiments\\textbar{}HMP\\_most-wanted\\textbar{}Anaerotruncus\\textbar{}LLG\\_output\\textbar{}checkM\\textbar{}1\\textbar{}checkm\\textbar{}markers\\_qa\\_summary.tsv.2 & markers\\_qa\\_summary.tsv.2 & NA & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/Anaerotruncus/LLG\\_output/checkM/1/checkm/markers\\_qa\\_summary.tsv.2\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋱ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t GCA\\_900199635.1\\_PRJEB19460\\_genomic      & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 99.33 & 0.00 & 0 & 3145951 &  14 & ⋯ & 1 & 262 & 0 & 0 & 0 & 0 & \\textbar{}ebio\\textbar{}abt3\\_projects\\textbar{}software\\textbar{}dev\\textbar{}ll\\_pipelines\\textbar{}llprimer\\textbar{}experiments\\textbar{}HMP\\_most-wanted\\textbar{}Anaerotruncus\\textbar{}LLG\\_output\\textbar{}checkM\\textbar{}2\\textbar{}checkm\\textbar{}markers\\_qa\\_summary.tsv.14 & markers\\_qa\\_summary.tsv.14 & NA & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/Anaerotruncus/LLG\\_output/checkM/2/checkm/markers\\_qa\\_summary.tsv.14\\\\\n",
       "\t GCA\\_902363605.1\\_MGYG-HGUT-00131\\_genomic & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 97.99 & 1.34 & 0 & 3306065 & 265 & ⋯ & 3 & 257 & 3 & 0 & 0 & 0 & \\textbar{}ebio\\textbar{}abt3\\_projects\\textbar{}software\\textbar{}dev\\textbar{}ll\\_pipelines\\textbar{}llprimer\\textbar{}experiments\\textbar{}HMP\\_most-wanted\\textbar{}Anaerotruncus\\textbar{}LLG\\_output\\textbar{}checkM\\textbar{}2\\textbar{}checkm\\textbar{}markers\\_qa\\_summary.tsv.15 & markers\\_qa\\_summary.tsv.15 & NA & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/Anaerotruncus/LLG\\_output/checkM/2/checkm/markers\\_qa\\_summary.tsv.15\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 31 × 33\n",
       "\n",
       "| Bin.Id &lt;fct&gt; | Marker.lineage &lt;fct&gt; | X..genomes &lt;int&gt; | X..markers &lt;int&gt; | X..marker.sets &lt;int&gt; | Completeness &lt;dbl&gt; | Contamination &lt;dbl&gt; | Strain.heterogeneity &lt;dbl&gt; | Genome.size..bp. &lt;int&gt; | X..ambiguous.bases &lt;int&gt; | ⋯ ⋯ | X0 &lt;int&gt; | X1 &lt;int&gt; | X2 &lt;int&gt; | X3 &lt;int&gt; | X4 &lt;int&gt; | X5. &lt;int&gt; | assembly.Id &lt;fct&gt; | assembler.Id &lt;fct&gt; | taxon.Id &lt;lgl&gt; | File &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| GCA_000154565.1_ASM15456v1_genomic | o__Clostridiales (UID1212) | 172 | 263 | 149 | 97.99 | 0.22 | 0 | 3719688 | 800 | ⋯ | 3 | 259 | 1 | 0 | 0 | 0 | |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Anaerotruncus|LLG_output|checkM|1|checkm|markers_qa_summary.tsv.1 | markers_qa_summary.tsv.1 | NA | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/LLG_output/checkM/1/checkm/markers_qa_summary.tsv.1 |\n",
       "| GCA_000433975.1_MGS528_genomic     | o__Clostridiales (UID1212) | 172 | 263 | 149 | 97.99 | 0.00 | 0 | 1881083 | 732 | ⋯ | 3 | 260 | 0 | 0 | 0 | 0 | |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Anaerotruncus|LLG_output|checkM|1|checkm|markers_qa_summary.tsv.2 | markers_qa_summary.tsv.2 | NA | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/LLG_output/checkM/1/checkm/markers_qa_summary.tsv.2 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋱ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| GCA_900199635.1_PRJEB19460_genomic      | o__Clostridiales (UID1212) | 172 | 263 | 149 | 99.33 | 0.00 | 0 | 3145951 |  14 | ⋯ | 1 | 262 | 0 | 0 | 0 | 0 | |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Anaerotruncus|LLG_output|checkM|2|checkm|markers_qa_summary.tsv.14 | markers_qa_summary.tsv.14 | NA | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/LLG_output/checkM/2/checkm/markers_qa_summary.tsv.14 |\n",
       "| GCA_902363605.1_MGYG-HGUT-00131_genomic | o__Clostridiales (UID1212) | 172 | 263 | 149 | 97.99 | 1.34 | 0 | 3306065 | 265 | ⋯ | 3 | 257 | 3 | 0 | 0 | 0 | |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Anaerotruncus|LLG_output|checkM|2|checkm|markers_qa_summary.tsv.15 | markers_qa_summary.tsv.15 | NA | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/LLG_output/checkM/2/checkm/markers_qa_summary.tsv.15 |\n",
       "\n"
      ],
      "text/plain": [
       "   Bin.Id                                  Marker.lineage            \n",
       "1  GCA_000154565.1_ASM15456v1_genomic      o__Clostridiales (UID1212)\n",
       "2  GCA_000433975.1_MGS528_genomic          o__Clostridiales (UID1212)\n",
       "⋮  ⋮                                       ⋮                         \n",
       "30 GCA_900199635.1_PRJEB19460_genomic      o__Clostridiales (UID1212)\n",
       "31 GCA_902363605.1_MGYG-HGUT-00131_genomic o__Clostridiales (UID1212)\n",
       "   X..genomes X..markers X..marker.sets Completeness Contamination\n",
       "1  172        263        149            97.99        0.22         \n",
       "2  172        263        149            97.99        0.00         \n",
       "⋮  ⋮          ⋮          ⋮              ⋮            ⋮            \n",
       "30 172        263        149            99.33        0.00         \n",
       "31 172        263        149            97.99        1.34         \n",
       "   Strain.heterogeneity Genome.size..bp. X..ambiguous.bases ⋯ X0 X1  X2 X3 X4\n",
       "1  0                    3719688          800                ⋯ 3  259 1  0  0 \n",
       "2  0                    1881083          732                ⋯ 3  260 0  0  0 \n",
       "⋮  ⋮                    ⋮                ⋮                  ⋱ ⋮  ⋮   ⋮  ⋮  ⋮ \n",
       "30 0                    3145951           14                ⋯ 1  262 0  0  0 \n",
       "31 0                    3306065          265                ⋯ 3  257 3  0  0 \n",
       "   X5.\n",
       "1  0  \n",
       "2  0  \n",
       "⋮  ⋮  \n",
       "30 0  \n",
       "31 0  \n",
       "   assembly.Id                                                                                                                                          \n",
       "1  |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Anaerotruncus|LLG_output|checkM|1|checkm|markers_qa_summary.tsv.1 \n",
       "2  |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Anaerotruncus|LLG_output|checkM|1|checkm|markers_qa_summary.tsv.2 \n",
       "⋮  ⋮                                                                                                                                                    \n",
       "30 |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Anaerotruncus|LLG_output|checkM|2|checkm|markers_qa_summary.tsv.14\n",
       "31 |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Anaerotruncus|LLG_output|checkM|2|checkm|markers_qa_summary.tsv.15\n",
       "   assembler.Id              taxon.Id\n",
       "1  markers_qa_summary.tsv.1  NA      \n",
       "2  markers_qa_summary.tsv.2  NA      \n",
       "⋮  ⋮                         ⋮       \n",
       "30 markers_qa_summary.tsv.14 NA      \n",
       "31 markers_qa_summary.tsv.15 NA      \n",
       "   File                                                                                                                                                 \n",
       "1  /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/LLG_output/checkM/1/checkm/markers_qa_summary.tsv.1 \n",
       "2  /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/LLG_output/checkM/1/checkm/markers_qa_summary.tsv.2 \n",
       "⋮  ⋮                                                                                                                                                    \n",
       "30 /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/LLG_output/checkM/2/checkm/markers_qa_summary.tsv.14\n",
       "31 /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/LLG_output/checkM/2/checkm/markers_qa_summary.tsv.15"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# checkM summary\n",
    "checkm = file.path(work_dir, 'LLG_output', 'checkM', 'checkm_qa_summary.tsv') %>%\n",
    "    read.delim(sep='\\t') \n",
    "checkm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-15T09:35:39.466298Z",
     "iopub.status.busy": "2021-01-15T09:35:39.464447Z",
     "iopub.status.idle": "2021-01-15T09:35:39.495956Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 31 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>genome</th><th scope=col>completeness</th><th scope=col>contamination</th><th scope=col>Bin.Id</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000154565.1_ASM15456v1_genomic.fna</td><td>97.99</td><td>0.22</td><td>GCA_000154565.1_ASM15456v1_genomic</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000433975.1_MGS528_genomic.fna    </td><td>97.99</td><td>0.00</td><td>GCA_000433975.1_MGS528_genomic    </td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_900199635.1_PRJEB19460_genomic.fna     </td><td>99.33</td><td>0.00</td><td>GCA_900199635.1_PRJEB19460_genomic     </td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_902363605.1_MGYG-HGUT-00131_genomic.fna</td><td>97.99</td><td>1.34</td><td>GCA_902363605.1_MGYG-HGUT-00131_genomic</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 31 × 4\n",
       "\\begin{tabular}{llll}\n",
       " genome & completeness & contamination & Bin.Id\\\\\n",
       " <fct> & <dbl> & <dbl> & <chr>\\\\\n",
       "\\hline\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_000154565.1\\_ASM15456v1\\_genomic.fna & 97.99 & 0.22 & GCA\\_000154565.1\\_ASM15456v1\\_genomic\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_000433975.1\\_MGS528\\_genomic.fna     & 97.99 & 0.00 & GCA\\_000433975.1\\_MGS528\\_genomic    \\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_900199635.1\\_PRJEB19460\\_genomic.fna      & 99.33 & 0.00 & GCA\\_900199635.1\\_PRJEB19460\\_genomic     \\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_902363605.1\\_MGYG-HGUT-00131\\_genomic.fna & 97.99 & 1.34 & GCA\\_902363605.1\\_MGYG-HGUT-00131\\_genomic\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 31 × 4\n",
       "\n",
       "| genome &lt;fct&gt; | completeness &lt;dbl&gt; | contamination &lt;dbl&gt; | Bin.Id &lt;chr&gt; |\n",
       "|---|---|---|---|\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000154565.1_ASM15456v1_genomic.fna | 97.99 | 0.22 | GCA_000154565.1_ASM15456v1_genomic |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000433975.1_MGS528_genomic.fna     | 97.99 | 0.00 | GCA_000433975.1_MGS528_genomic     |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_900199635.1_PRJEB19460_genomic.fna      | 99.33 | 0.00 | GCA_900199635.1_PRJEB19460_genomic      |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_902363605.1_MGYG-HGUT-00131_genomic.fna | 97.99 | 1.34 | GCA_902363605.1_MGYG-HGUT-00131_genomic |\n",
       "\n"
      ],
      "text/plain": [
       "   genome                                                                                           \n",
       "1  /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000154565.1_ASM15456v1_genomic.fna     \n",
       "2  /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000433975.1_MGS528_genomic.fna         \n",
       "⋮  ⋮                                                                                                \n",
       "30 /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_900199635.1_PRJEB19460_genomic.fna     \n",
       "31 /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_902363605.1_MGYG-HGUT-00131_genomic.fna\n",
       "   completeness contamination Bin.Id                                 \n",
       "1  97.99        0.22          GCA_000154565.1_ASM15456v1_genomic     \n",
       "2  97.99        0.00          GCA_000433975.1_MGS528_genomic         \n",
       "⋮  ⋮            ⋮             ⋮                                      \n",
       "30 99.33        0.00          GCA_900199635.1_PRJEB19460_genomic     \n",
       "31 97.99        1.34          GCA_902363605.1_MGYG-HGUT-00131_genomic"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# dRep summary\n",
    "drep = file.path(work_dir, 'LLG_output', 'drep', 'checkm_markers_qa_summary.tsv') %>%\n",
    "    read.delim(sep='\\t') %>%\n",
    "    mutate(Bin.Id = gsub('.+/', '', genome),\n",
    "           Bin.Id = gsub('\\\\.fna$', '', Bin.Id))\n",
    "drep"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-15T09:35:39.810192Z",
     "iopub.status.busy": "2021-01-15T09:35:39.809209Z",
     "iopub.status.idle": "2021-01-15T09:35:39.848107Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 27 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Name</th><th scope=col>Fasta</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>GCA_014284405.1_ASM1428440v1_genomic</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/LLG_output/drep/drep/dereplicated_genomes/GCA_014284405.1_ASM1428440v1_genomic.fna</td></tr>\n",
       "\t<tr><td>GCA_900199635.1_PRJEB19460_genomic  </td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/LLG_output/drep/drep/dereplicated_genomes/GCA_900199635.1_PRJEB19460_genomic.fna  </td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>GCA_015554285.1_ASM1555428v1_genomic</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/LLG_output/drep/drep/dereplicated_genomes/GCA_015554285.1_ASM1555428v1_genomic.fna</td></tr>\n",
       "\t<tr><td>GCA_000433975.1_MGS528_genomic      </td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/LLG_output/drep/drep/dereplicated_genomes/GCA_000433975.1_MGS528_genomic.fna      </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 27 × 2\n",
       "\\begin{tabular}{ll}\n",
       " Name & Fasta\\\\\n",
       " <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\t GCA\\_014284405.1\\_ASM1428440v1\\_genomic & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/Anaerotruncus/LLG\\_output/drep/drep/dereplicated\\_genomes/GCA\\_014284405.1\\_ASM1428440v1\\_genomic.fna\\\\\n",
       "\t GCA\\_900199635.1\\_PRJEB19460\\_genomic   & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/Anaerotruncus/LLG\\_output/drep/drep/dereplicated\\_genomes/GCA\\_900199635.1\\_PRJEB19460\\_genomic.fna  \\\\\n",
       "\t ⋮ & ⋮\\\\\n",
       "\t GCA\\_015554285.1\\_ASM1555428v1\\_genomic & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/Anaerotruncus/LLG\\_output/drep/drep/dereplicated\\_genomes/GCA\\_015554285.1\\_ASM1555428v1\\_genomic.fna\\\\\n",
       "\t GCA\\_000433975.1\\_MGS528\\_genomic       & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/Anaerotruncus/LLG\\_output/drep/drep/dereplicated\\_genomes/GCA\\_000433975.1\\_MGS528\\_genomic.fna      \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 27 × 2\n",
       "\n",
       "| Name &lt;fct&gt; | Fasta &lt;fct&gt; |\n",
       "|---|---|\n",
       "| GCA_014284405.1_ASM1428440v1_genomic | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/LLG_output/drep/drep/dereplicated_genomes/GCA_014284405.1_ASM1428440v1_genomic.fna |\n",
       "| GCA_900199635.1_PRJEB19460_genomic   | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/LLG_output/drep/drep/dereplicated_genomes/GCA_900199635.1_PRJEB19460_genomic.fna   |\n",
       "| ⋮ | ⋮ |\n",
       "| GCA_015554285.1_ASM1555428v1_genomic | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/LLG_output/drep/drep/dereplicated_genomes/GCA_015554285.1_ASM1555428v1_genomic.fna |\n",
       "| GCA_000433975.1_MGS528_genomic       | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/LLG_output/drep/drep/dereplicated_genomes/GCA_000433975.1_MGS528_genomic.fna       |\n",
       "\n"
      ],
      "text/plain": [
       "   Name                                \n",
       "1  GCA_014284405.1_ASM1428440v1_genomic\n",
       "2  GCA_900199635.1_PRJEB19460_genomic  \n",
       "⋮  ⋮                                   \n",
       "26 GCA_015554285.1_ASM1555428v1_genomic\n",
       "27 GCA_000433975.1_MGS528_genomic      \n",
       "   Fasta                                                                                                                                                                              \n",
       "1  /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/LLG_output/drep/drep/dereplicated_genomes/GCA_014284405.1_ASM1428440v1_genomic.fna\n",
       "2  /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/LLG_output/drep/drep/dereplicated_genomes/GCA_900199635.1_PRJEB19460_genomic.fna  \n",
       "⋮  ⋮                                                                                                                                                                                  \n",
       "26 /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/LLG_output/drep/drep/dereplicated_genomes/GCA_015554285.1_ASM1555428v1_genomic.fna\n",
       "27 /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/LLG_output/drep/drep/dereplicated_genomes/GCA_000433975.1_MGS528_genomic.fna      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# de-replicated genomes\n",
    "drep_gen = file.path(work_dir, 'LLG_output', 'drep', 'dereplicated_genomes.tsv') %>%\n",
    "    read.delim(sep='\\t')\n",
    "drep_gen"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-15T09:35:40.122011Z",
     "iopub.status.busy": "2021-01-15T09:35:40.120051Z",
     "iopub.status.idle": "2021-01-15T09:35:40.172866Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 30 × 22</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>user_genome</th><th scope=col>Domain</th><th scope=col>Phylum</th><th scope=col>Class</th><th scope=col>Order</th><th scope=col>Family</th><th scope=col>Genus</th><th scope=col>Species</th><th scope=col>fastani_reference</th><th scope=col>fastani_reference_radius</th><th scope=col>⋯</th><th scope=col>fastani_af</th><th scope=col>closest_placement_reference</th><th scope=col>closest_placement_radius</th><th scope=col>closest_placement_taxonomy</th><th scope=col>closest_placement_ani</th><th scope=col>closest_placement_af</th><th scope=col>msa_percent</th><th scope=col>translation_table</th><th scope=col>red_value</th><th scope=col>warnings</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>⋯</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>GCA_000154565.1_ASM15456v1_genomic</td><td>d__Bacteria</td><td>p__Firmicutes_A</td><td>c__Clostridia</td><td>o__Oscillospirales</td><td>f__Ruminococcaceae   </td><td>g__Anaerotruncus</td><td>s__Anaerotruncus colihominis</td><td>GCF_000154565.1</td><td>95.0</td><td>⋯</td><td>1.0</td><td>GCF_000154565.1</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus colihominis   </td><td>100.0</td><td>1.0</td><td>97.18</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>GCA_000433975.1_MGS528_genomic    </td><td>d__Bacteria</td><td>p__Firmicutes_A</td><td>c__Clostridia</td><td>o__Oscillospirales</td><td>f__Acutalibacteraceae</td><td>g__Eubacterium_R</td><td>s__Eubacterium_R sp000433975</td><td>GCA_000433975.1</td><td>95.0</td><td>⋯</td><td>1.0</td><td>GCA_000433975.1</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Acutalibacteraceae;g__Eubacterium_R;s__Eubacterium_R sp000433975</td><td>100.0</td><td>1.0</td><td>96.57</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋱</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>GCA_900199635.1_PRJEB19460_genomic     </td><td>d__Bacteria</td><td>p__Firmicutes_A</td><td>c__Clostridia</td><td>o__Oscillospirales</td><td>f__Ruminococcaceae</td><td>g__Anaerotruncus</td><td>s__Anaerotruncus massiliensis</td><td>GCF_900199635.1</td><td>95.0</td><td>⋯</td><td>1.0 </td><td>GCF_900199635.1</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus massiliensis</td><td>100.0</td><td>1.0 </td><td>96.43</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>GCA_902363605.1_MGYG-HGUT-00131_genomic</td><td>d__Bacteria</td><td>p__Firmicutes_A</td><td>c__Clostridia</td><td>o__Oscillospirales</td><td>f__Ruminococcaceae</td><td>g__Anaerotruncus</td><td>s__Anaerotruncus rubiinfantis</td><td>GCF_900078395.1</td><td>95.0</td><td>⋯</td><td>0.91</td><td>GCF_900078395.1</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus rubiinfantis</td><td>99.26</td><td>0.91</td><td>96.35</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 30 × 22\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " user\\_genome & Domain & Phylum & Class & Order & Family & Genus & Species & fastani\\_reference & fastani\\_reference\\_radius & ⋯ & fastani\\_af & closest\\_placement\\_reference & closest\\_placement\\_radius & closest\\_placement\\_taxonomy & closest\\_placement\\_ani & closest\\_placement\\_af & msa\\_percent & translation\\_table & red\\_value & warnings\\\\\n",
       " <fct> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <fct> & <fct> & ⋯ & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <dbl> & <int> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\t GCA\\_000154565.1\\_ASM15456v1\\_genomic & d\\_\\_Bacteria & p\\_\\_Firmicutes\\_A & c\\_\\_Clostridia & o\\_\\_Oscillospirales & f\\_\\_Ruminococcaceae    & g\\_\\_Anaerotruncus & s\\_\\_Anaerotruncus colihominis & GCF\\_000154565.1 & 95.0 & ⋯ & 1.0 & GCF\\_000154565.1 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Ruminococcaceae;g\\_\\_Anaerotruncus;s\\_\\_Anaerotruncus colihominis    & 100.0 & 1.0 & 97.18 & 11 & N/A & N/A\\\\\n",
       "\t GCA\\_000433975.1\\_MGS528\\_genomic     & d\\_\\_Bacteria & p\\_\\_Firmicutes\\_A & c\\_\\_Clostridia & o\\_\\_Oscillospirales & f\\_\\_Acutalibacteraceae & g\\_\\_Eubacterium\\_R & s\\_\\_Eubacterium\\_R sp000433975 & GCA\\_000433975.1 & 95.0 & ⋯ & 1.0 & GCA\\_000433975.1 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Acutalibacteraceae;g\\_\\_Eubacterium\\_R;s\\_\\_Eubacterium\\_R sp000433975 & 100.0 & 1.0 & 96.57 & 11 & N/A & N/A\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋱ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t GCA\\_900199635.1\\_PRJEB19460\\_genomic      & d\\_\\_Bacteria & p\\_\\_Firmicutes\\_A & c\\_\\_Clostridia & o\\_\\_Oscillospirales & f\\_\\_Ruminococcaceae & g\\_\\_Anaerotruncus & s\\_\\_Anaerotruncus massiliensis & GCF\\_900199635.1 & 95.0 & ⋯ & 1.0  & GCF\\_900199635.1 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Ruminococcaceae;g\\_\\_Anaerotruncus;s\\_\\_Anaerotruncus massiliensis & 100.0 & 1.0  & 96.43 & 11 & N/A & N/A\\\\\n",
       "\t GCA\\_902363605.1\\_MGYG-HGUT-00131\\_genomic & d\\_\\_Bacteria & p\\_\\_Firmicutes\\_A & c\\_\\_Clostridia & o\\_\\_Oscillospirales & f\\_\\_Ruminococcaceae & g\\_\\_Anaerotruncus & s\\_\\_Anaerotruncus rubiinfantis & GCF\\_900078395.1 & 95.0 & ⋯ & 0.91 & GCF\\_900078395.1 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Ruminococcaceae;g\\_\\_Anaerotruncus;s\\_\\_Anaerotruncus rubiinfantis & 99.26 & 0.91 & 96.35 & 11 & N/A & N/A\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 30 × 22\n",
       "\n",
       "| user_genome &lt;fct&gt; | Domain &lt;chr&gt; | Phylum &lt;chr&gt; | Class &lt;chr&gt; | Order &lt;chr&gt; | Family &lt;chr&gt; | Genus &lt;chr&gt; | Species &lt;chr&gt; | fastani_reference &lt;fct&gt; | fastani_reference_radius &lt;fct&gt; | ⋯ ⋯ | fastani_af &lt;fct&gt; | closest_placement_reference &lt;fct&gt; | closest_placement_radius &lt;fct&gt; | closest_placement_taxonomy &lt;fct&gt; | closest_placement_ani &lt;fct&gt; | closest_placement_af &lt;fct&gt; | msa_percent &lt;dbl&gt; | translation_table &lt;int&gt; | red_value &lt;fct&gt; | warnings &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| GCA_000154565.1_ASM15456v1_genomic | d__Bacteria | p__Firmicutes_A | c__Clostridia | o__Oscillospirales | f__Ruminococcaceae    | g__Anaerotruncus | s__Anaerotruncus colihominis | GCF_000154565.1 | 95.0 | ⋯ | 1.0 | GCF_000154565.1 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus colihominis    | 100.0 | 1.0 | 97.18 | 11 | N/A | N/A |\n",
       "| GCA_000433975.1_MGS528_genomic     | d__Bacteria | p__Firmicutes_A | c__Clostridia | o__Oscillospirales | f__Acutalibacteraceae | g__Eubacterium_R | s__Eubacterium_R sp000433975 | GCA_000433975.1 | 95.0 | ⋯ | 1.0 | GCA_000433975.1 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Acutalibacteraceae;g__Eubacterium_R;s__Eubacterium_R sp000433975 | 100.0 | 1.0 | 96.57 | 11 | N/A | N/A |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋱ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| GCA_900199635.1_PRJEB19460_genomic      | d__Bacteria | p__Firmicutes_A | c__Clostridia | o__Oscillospirales | f__Ruminococcaceae | g__Anaerotruncus | s__Anaerotruncus massiliensis | GCF_900199635.1 | 95.0 | ⋯ | 1.0  | GCF_900199635.1 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus massiliensis | 100.0 | 1.0  | 96.43 | 11 | N/A | N/A |\n",
       "| GCA_902363605.1_MGYG-HGUT-00131_genomic | d__Bacteria | p__Firmicutes_A | c__Clostridia | o__Oscillospirales | f__Ruminococcaceae | g__Anaerotruncus | s__Anaerotruncus rubiinfantis | GCF_900078395.1 | 95.0 | ⋯ | 0.91 | GCF_900078395.1 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus rubiinfantis | 99.26 | 0.91 | 96.35 | 11 | N/A | N/A |\n",
       "\n"
      ],
      "text/plain": [
       "   user_genome                             Domain      Phylum         \n",
       "1  GCA_000154565.1_ASM15456v1_genomic      d__Bacteria p__Firmicutes_A\n",
       "2  GCA_000433975.1_MGS528_genomic          d__Bacteria p__Firmicutes_A\n",
       "⋮  ⋮                                       ⋮           ⋮              \n",
       "29 GCA_900199635.1_PRJEB19460_genomic      d__Bacteria p__Firmicutes_A\n",
       "30 GCA_902363605.1_MGYG-HGUT-00131_genomic d__Bacteria p__Firmicutes_A\n",
       "   Class         Order              Family                Genus           \n",
       "1  c__Clostridia o__Oscillospirales f__Ruminococcaceae    g__Anaerotruncus\n",
       "2  c__Clostridia o__Oscillospirales f__Acutalibacteraceae g__Eubacterium_R\n",
       "⋮  ⋮             ⋮                  ⋮                     ⋮               \n",
       "29 c__Clostridia o__Oscillospirales f__Ruminococcaceae    g__Anaerotruncus\n",
       "30 c__Clostridia o__Oscillospirales f__Ruminococcaceae    g__Anaerotruncus\n",
       "   Species                       fastani_reference fastani_reference_radius ⋯\n",
       "1  s__Anaerotruncus colihominis  GCF_000154565.1   95.0                     ⋯\n",
       "2  s__Eubacterium_R sp000433975  GCA_000433975.1   95.0                     ⋯\n",
       "⋮  ⋮                             ⋮                 ⋮                        ⋱\n",
       "29 s__Anaerotruncus massiliensis GCF_900199635.1   95.0                     ⋯\n",
       "30 s__Anaerotruncus rubiinfantis GCF_900078395.1   95.0                     ⋯\n",
       "   fastani_af closest_placement_reference closest_placement_radius\n",
       "1  1.0        GCF_000154565.1             95.0                    \n",
       "2  1.0        GCA_000433975.1             95.0                    \n",
       "⋮  ⋮          ⋮                           ⋮                       \n",
       "29 1.0        GCF_900199635.1             95.0                    \n",
       "30 0.91       GCF_900078395.1             95.0                    \n",
       "   closest_placement_taxonomy                                                                                                      \n",
       "1  d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus colihominis   \n",
       "2  d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Acutalibacteraceae;g__Eubacterium_R;s__Eubacterium_R sp000433975\n",
       "⋮  ⋮                                                                                                                               \n",
       "29 d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus massiliensis  \n",
       "30 d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus rubiinfantis  \n",
       "   closest_placement_ani closest_placement_af msa_percent translation_table\n",
       "1  100.0                 1.0                  97.18       11               \n",
       "2  100.0                 1.0                  96.57       11               \n",
       "⋮  ⋮                     ⋮                    ⋮           ⋮                \n",
       "29 100.0                 1.0                  96.43       11               \n",
       "30 99.26                 0.91                 96.35       11               \n",
       "   red_value warnings\n",
       "1  N/A       N/A     \n",
       "2  N/A       N/A     \n",
       "⋮  ⋮         ⋮       \n",
       "29 N/A       N/A     \n",
       "30 N/A       N/A     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# GTDBTk summary\n",
    "tax = file.path(work_dir, 'LLG_output', 'gtdbtk', 'gtdbtk_bac_summary.tsv') %>%\n",
    "    read.delim(, sep='\\t') %>%\n",
    "    separate(classification, \n",
    "             c('Domain', 'Phylum', 'Class', 'Order', 'Family', 'Genus', 'Species'),\n",
    "             sep=';') %>%\n",
    "    select(-note, -classification_method, -pplacer_taxonomy,\n",
    "           -other_related_references.genome_id.species_name.radius.ANI.AF.)\n",
    "tax"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-15T09:35:41.061197Z",
     "iopub.status.busy": "2021-01-15T09:35:41.059171Z",
     "iopub.status.idle": "2021-01-15T09:35:41.098756Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- drep --\n",
      "intersect(x,y): 27 \n",
      "setdiff(x,y): 0 \n",
      "setdiff(y,x): 4 \n",
      "union(x,y): 31 \n",
      "-- checkm --\n",
      "intersect(x,y): 31 \n",
      "setdiff(x,y): 0 \n",
      "setdiff(y,x): 0 \n",
      "union(x,y): 31 \n",
      "-- gtdbtk --\n",
      "intersect(x,y): 30 \n",
      "setdiff(x,y): 1 \n",
      "setdiff(y,x): 0 \n",
      "union(x,y): 31 \n"
     ]
    }
   ],
   "source": [
    "# checking overlap\n",
    "cat('-- drep --\\n')\n",
    "overlap(basename(as.character(drep_gen$Fasta)), \n",
    "        basename(as.character(drep$genome)))\n",
    "cat('-- checkm --\\n')\n",
    "overlap(drep$Bin.Id, checkm$Bin.Id)\n",
    "cat('-- gtdbtk --\\n')\n",
    "overlap(drep$Bin.Id, tax$user_genome)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-15T09:35:42.321312Z",
     "iopub.status.busy": "2021-01-15T09:35:42.319086Z",
     "iopub.status.idle": "2021-01-15T09:35:42.391817Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 27 × 60</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>genome</th><th scope=col>completeness</th><th scope=col>contamination</th><th scope=col>Bin.Id</th><th scope=col>Marker.lineage</th><th scope=col>X..genomes</th><th scope=col>X..markers</th><th scope=col>X..marker.sets</th><th scope=col>Completeness</th><th scope=col>Contamination</th><th scope=col>⋯</th><th scope=col>fastani_af</th><th scope=col>closest_placement_reference</th><th scope=col>closest_placement_radius</th><th scope=col>closest_placement_taxonomy</th><th scope=col>closest_placement_ani</th><th scope=col>closest_placement_af</th><th scope=col>msa_percent</th><th scope=col>translation_table</th><th scope=col>red_value</th><th scope=col>warnings</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000154565.1_ASM15456v1_genomic.fna</td><td>97.99</td><td>0.22</td><td>GCA_000154565.1_ASM15456v1_genomic</td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>97.99</td><td>0.22</td><td>⋯</td><td>1.0</td><td>GCF_000154565.1</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus colihominis   </td><td>100.0</td><td>1.0</td><td>97.18</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000433975.1_MGS528_genomic.fna    </td><td>97.99</td><td>0.00</td><td>GCA_000433975.1_MGS528_genomic    </td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>97.99</td><td>0.00</td><td>⋯</td><td>1.0</td><td>GCA_000433975.1</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Acutalibacteraceae;g__Eubacterium_R;s__Eubacterium_R sp000433975</td><td>100.0</td><td>1.0</td><td>96.57</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋱</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_900199635.1_PRJEB19460_genomic.fna     </td><td>99.33</td><td>0.00</td><td>GCA_900199635.1_PRJEB19460_genomic     </td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>99.33</td><td>0.00</td><td>⋯</td><td>1.0 </td><td>GCF_900199635.1</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus massiliensis</td><td>100.0</td><td>1.0 </td><td>96.43</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_902363605.1_MGYG-HGUT-00131_genomic.fna</td><td>97.99</td><td>1.34</td><td>GCA_902363605.1_MGYG-HGUT-00131_genomic</td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>97.99</td><td>1.34</td><td>⋯</td><td>0.91</td><td>GCF_900078395.1</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus rubiinfantis</td><td>99.26</td><td>0.91</td><td>96.35</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 27 × 60\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " genome & completeness & contamination & Bin.Id & Marker.lineage & X..genomes & X..markers & X..marker.sets & Completeness & Contamination & ⋯ & fastani\\_af & closest\\_placement\\_reference & closest\\_placement\\_radius & closest\\_placement\\_taxonomy & closest\\_placement\\_ani & closest\\_placement\\_af & msa\\_percent & translation\\_table & red\\_value & warnings\\\\\n",
       " <fct> & <dbl> & <dbl> & <chr> & <fct> & <int> & <int> & <int> & <dbl> & <dbl> & ⋯ & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <dbl> & <int> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_000154565.1\\_ASM15456v1\\_genomic.fna & 97.99 & 0.22 & GCA\\_000154565.1\\_ASM15456v1\\_genomic & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 97.99 & 0.22 & ⋯ & 1.0 & GCF\\_000154565.1 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Ruminococcaceae;g\\_\\_Anaerotruncus;s\\_\\_Anaerotruncus colihominis    & 100.0 & 1.0 & 97.18 & 11 & N/A & N/A\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_000433975.1\\_MGS528\\_genomic.fna     & 97.99 & 0.00 & GCA\\_000433975.1\\_MGS528\\_genomic     & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 97.99 & 0.00 & ⋯ & 1.0 & GCA\\_000433975.1 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Acutalibacteraceae;g\\_\\_Eubacterium\\_R;s\\_\\_Eubacterium\\_R sp000433975 & 100.0 & 1.0 & 96.57 & 11 & N/A & N/A\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋱ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_900199635.1\\_PRJEB19460\\_genomic.fna      & 99.33 & 0.00 & GCA\\_900199635.1\\_PRJEB19460\\_genomic      & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 99.33 & 0.00 & ⋯ & 1.0  & GCF\\_900199635.1 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Ruminococcaceae;g\\_\\_Anaerotruncus;s\\_\\_Anaerotruncus massiliensis & 100.0 & 1.0  & 96.43 & 11 & N/A & N/A\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_902363605.1\\_MGYG-HGUT-00131\\_genomic.fna & 97.99 & 1.34 & GCA\\_902363605.1\\_MGYG-HGUT-00131\\_genomic & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 97.99 & 1.34 & ⋯ & 0.91 & GCF\\_900078395.1 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Ruminococcaceae;g\\_\\_Anaerotruncus;s\\_\\_Anaerotruncus rubiinfantis & 99.26 & 0.91 & 96.35 & 11 & N/A & N/A\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 27 × 60\n",
       "\n",
       "| genome &lt;fct&gt; | completeness &lt;dbl&gt; | contamination &lt;dbl&gt; | Bin.Id &lt;chr&gt; | Marker.lineage &lt;fct&gt; | X..genomes &lt;int&gt; | X..markers &lt;int&gt; | X..marker.sets &lt;int&gt; | Completeness &lt;dbl&gt; | Contamination &lt;dbl&gt; | ⋯ ⋯ | fastani_af &lt;fct&gt; | closest_placement_reference &lt;fct&gt; | closest_placement_radius &lt;fct&gt; | closest_placement_taxonomy &lt;fct&gt; | closest_placement_ani &lt;fct&gt; | closest_placement_af &lt;fct&gt; | msa_percent &lt;dbl&gt; | translation_table &lt;int&gt; | red_value &lt;fct&gt; | warnings &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000154565.1_ASM15456v1_genomic.fna | 97.99 | 0.22 | GCA_000154565.1_ASM15456v1_genomic | o__Clostridiales (UID1212) | 172 | 263 | 149 | 97.99 | 0.22 | ⋯ | 1.0 | GCF_000154565.1 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus colihominis    | 100.0 | 1.0 | 97.18 | 11 | N/A | N/A |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000433975.1_MGS528_genomic.fna     | 97.99 | 0.00 | GCA_000433975.1_MGS528_genomic     | o__Clostridiales (UID1212) | 172 | 263 | 149 | 97.99 | 0.00 | ⋯ | 1.0 | GCA_000433975.1 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Acutalibacteraceae;g__Eubacterium_R;s__Eubacterium_R sp000433975 | 100.0 | 1.0 | 96.57 | 11 | N/A | N/A |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋱ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_900199635.1_PRJEB19460_genomic.fna      | 99.33 | 0.00 | GCA_900199635.1_PRJEB19460_genomic      | o__Clostridiales (UID1212) | 172 | 263 | 149 | 99.33 | 0.00 | ⋯ | 1.0  | GCF_900199635.1 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus massiliensis | 100.0 | 1.0  | 96.43 | 11 | N/A | N/A |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_902363605.1_MGYG-HGUT-00131_genomic.fna | 97.99 | 1.34 | GCA_902363605.1_MGYG-HGUT-00131_genomic | o__Clostridiales (UID1212) | 172 | 263 | 149 | 97.99 | 1.34 | ⋯ | 0.91 | GCF_900078395.1 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus rubiinfantis | 99.26 | 0.91 | 96.35 | 11 | N/A | N/A |\n",
       "\n"
      ],
      "text/plain": [
       "   genome                                                                                           \n",
       "1  /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000154565.1_ASM15456v1_genomic.fna     \n",
       "2  /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000433975.1_MGS528_genomic.fna         \n",
       "⋮  ⋮                                                                                                \n",
       "26 /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_900199635.1_PRJEB19460_genomic.fna     \n",
       "27 /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_902363605.1_MGYG-HGUT-00131_genomic.fna\n",
       "   completeness contamination Bin.Id                                 \n",
       "1  97.99        0.22          GCA_000154565.1_ASM15456v1_genomic     \n",
       "2  97.99        0.00          GCA_000433975.1_MGS528_genomic         \n",
       "⋮  ⋮            ⋮             ⋮                                      \n",
       "26 99.33        0.00          GCA_900199635.1_PRJEB19460_genomic     \n",
       "27 97.99        1.34          GCA_902363605.1_MGYG-HGUT-00131_genomic\n",
       "   Marker.lineage             X..genomes X..markers X..marker.sets Completeness\n",
       "1  o__Clostridiales (UID1212) 172        263        149            97.99       \n",
       "2  o__Clostridiales (UID1212) 172        263        149            97.99       \n",
       "⋮  ⋮                          ⋮          ⋮          ⋮              ⋮           \n",
       "26 o__Clostridiales (UID1212) 172        263        149            99.33       \n",
       "27 o__Clostridiales (UID1212) 172        263        149            97.99       \n",
       "   Contamination ⋯ fastani_af closest_placement_reference\n",
       "1  0.22          ⋯ 1.0        GCF_000154565.1            \n",
       "2  0.00          ⋯ 1.0        GCA_000433975.1            \n",
       "⋮  ⋮             ⋱ ⋮          ⋮                          \n",
       "26 0.00          ⋯ 1.0        GCF_900199635.1            \n",
       "27 1.34          ⋯ 0.91       GCF_900078395.1            \n",
       "   closest_placement_radius\n",
       "1  95.0                    \n",
       "2  95.0                    \n",
       "⋮  ⋮                       \n",
       "26 95.0                    \n",
       "27 95.0                    \n",
       "   closest_placement_taxonomy                                                                                                      \n",
       "1  d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus colihominis   \n",
       "2  d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Acutalibacteraceae;g__Eubacterium_R;s__Eubacterium_R sp000433975\n",
       "⋮  ⋮                                                                                                                               \n",
       "26 d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus massiliensis  \n",
       "27 d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus rubiinfantis  \n",
       "   closest_placement_ani closest_placement_af msa_percent translation_table\n",
       "1  100.0                 1.0                  97.18       11               \n",
       "2  100.0                 1.0                  96.57       11               \n",
       "⋮  ⋮                     ⋮                    ⋮           ⋮                \n",
       "26 100.0                 1.0                  96.43       11               \n",
       "27 99.26                 0.91                 96.35       11               \n",
       "   red_value warnings\n",
       "1  N/A       N/A     \n",
       "2  N/A       N/A     \n",
       "⋮  ⋮         ⋮       \n",
       "26 N/A       N/A     \n",
       "27 N/A       N/A     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# joining based on Bin.Id\n",
    "drep = drep %>%\n",
    "    inner_join(checkm, c('Bin.Id')) %>%\n",
    "    mutate(GEN = genome %>% as.character %>% basename) %>%\n",
    "    inner_join(drep_gen %>% mutate(GEN = Fasta %>% as.character %>% basename),\n",
    "               by=c('GEN')) %>%\n",
    "    inner_join(tax, c('Bin.Id'='user_genome')) #%>%\n",
    "drep"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-15T09:39:23.524993Z",
     "iopub.status.busy": "2021-01-15T09:39:23.523165Z",
     "iopub.status.idle": "2021-01-15T09:39:23.577306Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 26 × 60</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>genome</th><th scope=col>completeness</th><th scope=col>contamination</th><th scope=col>Bin.Id</th><th scope=col>Marker.lineage</th><th scope=col>X..genomes</th><th scope=col>X..markers</th><th scope=col>X..marker.sets</th><th scope=col>Completeness</th><th scope=col>Contamination</th><th scope=col>⋯</th><th scope=col>fastani_af</th><th scope=col>closest_placement_reference</th><th scope=col>closest_placement_radius</th><th scope=col>closest_placement_taxonomy</th><th scope=col>closest_placement_ani</th><th scope=col>closest_placement_af</th><th scope=col>msa_percent</th><th scope=col>translation_table</th><th scope=col>red_value</th><th scope=col>warnings</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000154565.1_ASM15456v1_genomic.fna</td><td>97.99</td><td>0.22</td><td>GCA_000154565.1_ASM15456v1_genomic</td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>97.99</td><td>0.22</td><td>⋯</td><td>1.0</td><td>GCF_000154565.1</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus colihominis   </td><td>100.0</td><td>1.0</td><td>97.18</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000433975.1_MGS528_genomic.fna    </td><td>97.99</td><td>0.00</td><td>GCA_000433975.1_MGS528_genomic    </td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>97.99</td><td>0.00</td><td>⋯</td><td>1.0</td><td>GCA_000433975.1</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Acutalibacteraceae;g__Eubacterium_R;s__Eubacterium_R sp000433975</td><td>100.0</td><td>1.0</td><td>96.57</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋱</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_900199635.1_PRJEB19460_genomic.fna     </td><td>99.33</td><td>0.00</td><td>GCA_900199635.1_PRJEB19460_genomic     </td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>99.33</td><td>0.00</td><td>⋯</td><td>1.0 </td><td>GCF_900199635.1</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus massiliensis</td><td>100.0</td><td>1.0 </td><td>96.43</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_902363605.1_MGYG-HGUT-00131_genomic.fna</td><td>97.99</td><td>1.34</td><td>GCA_902363605.1_MGYG-HGUT-00131_genomic</td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>97.99</td><td>1.34</td><td>⋯</td><td>0.91</td><td>GCF_900078395.1</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus rubiinfantis</td><td>99.26</td><td>0.91</td><td>96.35</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 26 × 60\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " genome & completeness & contamination & Bin.Id & Marker.lineage & X..genomes & X..markers & X..marker.sets & Completeness & Contamination & ⋯ & fastani\\_af & closest\\_placement\\_reference & closest\\_placement\\_radius & closest\\_placement\\_taxonomy & closest\\_placement\\_ani & closest\\_placement\\_af & msa\\_percent & translation\\_table & red\\_value & warnings\\\\\n",
       " <fct> & <dbl> & <dbl> & <chr> & <fct> & <int> & <int> & <int> & <dbl> & <dbl> & ⋯ & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <dbl> & <int> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_000154565.1\\_ASM15456v1\\_genomic.fna & 97.99 & 0.22 & GCA\\_000154565.1\\_ASM15456v1\\_genomic & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 97.99 & 0.22 & ⋯ & 1.0 & GCF\\_000154565.1 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Ruminococcaceae;g\\_\\_Anaerotruncus;s\\_\\_Anaerotruncus colihominis    & 100.0 & 1.0 & 97.18 & 11 & N/A & N/A\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_000433975.1\\_MGS528\\_genomic.fna     & 97.99 & 0.00 & GCA\\_000433975.1\\_MGS528\\_genomic     & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 97.99 & 0.00 & ⋯ & 1.0 & GCA\\_000433975.1 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Acutalibacteraceae;g\\_\\_Eubacterium\\_R;s\\_\\_Eubacterium\\_R sp000433975 & 100.0 & 1.0 & 96.57 & 11 & N/A & N/A\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋱ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_900199635.1\\_PRJEB19460\\_genomic.fna      & 99.33 & 0.00 & GCA\\_900199635.1\\_PRJEB19460\\_genomic      & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 99.33 & 0.00 & ⋯ & 1.0  & GCF\\_900199635.1 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Ruminococcaceae;g\\_\\_Anaerotruncus;s\\_\\_Anaerotruncus massiliensis & 100.0 & 1.0  & 96.43 & 11 & N/A & N/A\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_902363605.1\\_MGYG-HGUT-00131\\_genomic.fna & 97.99 & 1.34 & GCA\\_902363605.1\\_MGYG-HGUT-00131\\_genomic & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 97.99 & 1.34 & ⋯ & 0.91 & GCF\\_900078395.1 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Ruminococcaceae;g\\_\\_Anaerotruncus;s\\_\\_Anaerotruncus rubiinfantis & 99.26 & 0.91 & 96.35 & 11 & N/A & N/A\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 26 × 60\n",
       "\n",
       "| genome &lt;fct&gt; | completeness &lt;dbl&gt; | contamination &lt;dbl&gt; | Bin.Id &lt;chr&gt; | Marker.lineage &lt;fct&gt; | X..genomes &lt;int&gt; | X..markers &lt;int&gt; | X..marker.sets &lt;int&gt; | Completeness &lt;dbl&gt; | Contamination &lt;dbl&gt; | ⋯ ⋯ | fastani_af &lt;fct&gt; | closest_placement_reference &lt;fct&gt; | closest_placement_radius &lt;fct&gt; | closest_placement_taxonomy &lt;fct&gt; | closest_placement_ani &lt;fct&gt; | closest_placement_af &lt;fct&gt; | msa_percent &lt;dbl&gt; | translation_table &lt;int&gt; | red_value &lt;fct&gt; | warnings &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000154565.1_ASM15456v1_genomic.fna | 97.99 | 0.22 | GCA_000154565.1_ASM15456v1_genomic | o__Clostridiales (UID1212) | 172 | 263 | 149 | 97.99 | 0.22 | ⋯ | 1.0 | GCF_000154565.1 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus colihominis    | 100.0 | 1.0 | 97.18 | 11 | N/A | N/A |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000433975.1_MGS528_genomic.fna     | 97.99 | 0.00 | GCA_000433975.1_MGS528_genomic     | o__Clostridiales (UID1212) | 172 | 263 | 149 | 97.99 | 0.00 | ⋯ | 1.0 | GCA_000433975.1 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Acutalibacteraceae;g__Eubacterium_R;s__Eubacterium_R sp000433975 | 100.0 | 1.0 | 96.57 | 11 | N/A | N/A |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋱ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_900199635.1_PRJEB19460_genomic.fna      | 99.33 | 0.00 | GCA_900199635.1_PRJEB19460_genomic      | o__Clostridiales (UID1212) | 172 | 263 | 149 | 99.33 | 0.00 | ⋯ | 1.0  | GCF_900199635.1 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus massiliensis | 100.0 | 1.0  | 96.43 | 11 | N/A | N/A |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_902363605.1_MGYG-HGUT-00131_genomic.fna | 97.99 | 1.34 | GCA_902363605.1_MGYG-HGUT-00131_genomic | o__Clostridiales (UID1212) | 172 | 263 | 149 | 97.99 | 1.34 | ⋯ | 0.91 | GCF_900078395.1 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus rubiinfantis | 99.26 | 0.91 | 96.35 | 11 | N/A | N/A |\n",
       "\n"
      ],
      "text/plain": [
       "   genome                                                                                           \n",
       "1  /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000154565.1_ASM15456v1_genomic.fna     \n",
       "2  /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000433975.1_MGS528_genomic.fna         \n",
       "⋮  ⋮                                                                                                \n",
       "25 /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_900199635.1_PRJEB19460_genomic.fna     \n",
       "26 /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_902363605.1_MGYG-HGUT-00131_genomic.fna\n",
       "   completeness contamination Bin.Id                                 \n",
       "1  97.99        0.22          GCA_000154565.1_ASM15456v1_genomic     \n",
       "2  97.99        0.00          GCA_000433975.1_MGS528_genomic         \n",
       "⋮  ⋮            ⋮             ⋮                                      \n",
       "25 99.33        0.00          GCA_900199635.1_PRJEB19460_genomic     \n",
       "26 97.99        1.34          GCA_902363605.1_MGYG-HGUT-00131_genomic\n",
       "   Marker.lineage             X..genomes X..markers X..marker.sets Completeness\n",
       "1  o__Clostridiales (UID1212) 172        263        149            97.99       \n",
       "2  o__Clostridiales (UID1212) 172        263        149            97.99       \n",
       "⋮  ⋮                          ⋮          ⋮          ⋮              ⋮           \n",
       "25 o__Clostridiales (UID1212) 172        263        149            99.33       \n",
       "26 o__Clostridiales (UID1212) 172        263        149            97.99       \n",
       "   Contamination ⋯ fastani_af closest_placement_reference\n",
       "1  0.22          ⋯ 1.0        GCF_000154565.1            \n",
       "2  0.00          ⋯ 1.0        GCA_000433975.1            \n",
       "⋮  ⋮             ⋱ ⋮          ⋮                          \n",
       "25 0.00          ⋯ 1.0        GCF_900199635.1            \n",
       "26 1.34          ⋯ 0.91       GCF_900078395.1            \n",
       "   closest_placement_radius\n",
       "1  95.0                    \n",
       "2  95.0                    \n",
       "⋮  ⋮                       \n",
       "25 95.0                    \n",
       "26 95.0                    \n",
       "   closest_placement_taxonomy                                                                                                      \n",
       "1  d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus colihominis   \n",
       "2  d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Acutalibacteraceae;g__Eubacterium_R;s__Eubacterium_R sp000433975\n",
       "⋮  ⋮                                                                                                                               \n",
       "25 d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus massiliensis  \n",
       "26 d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus rubiinfantis  \n",
       "   closest_placement_ani closest_placement_af msa_percent translation_table\n",
       "1  100.0                 1.0                  97.18       11               \n",
       "2  100.0                 1.0                  96.57       11               \n",
       "⋮  ⋮                     ⋮                    ⋮           ⋮                \n",
       "25 100.0                 1.0                  96.43       11               \n",
       "26 99.26                 0.91                 96.35       11               \n",
       "   red_value warnings\n",
       "1  N/A       N/A     \n",
       "2  N/A       N/A     \n",
       "⋮  ⋮         ⋮       \n",
       "25 N/A       N/A     \n",
       "26 N/A       N/A     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# filtering by quality\n",
    "hq_genomes = drep %>%\n",
    "    filter(completeness >= 90,\n",
    "           contamination < 5,\n",
    "           Strain.heterogeneity < 50,\n",
    "           X..contigs < 300,\n",
    "           Mean.contig.length..bp. > 10000)\n",
    "hq_genomes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-15T09:39:25.176314Z",
     "iopub.status.busy": "2021-01-15T09:39:25.174231Z",
     "iopub.status.idle": "2021-01-15T09:39:25.234104Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 1 × 8 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Min.</th><th scope=col>1st Qu.</th><th scope=col>Median</th><th scope=col>Mean</th><th scope=col>3rd Qu.</th><th scope=col>Max.</th><th scope=col>sd</th><th scope=col>sd_err_of_mean</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>completeness</th><td>97.32</td><td>97.99</td><td>97.99</td><td>98.24538</td><td>98.66</td><td>99.33</td><td>0.683</td><td>0.279</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 1 × 8 of type dbl\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & Min. & 1st Qu. & Median & Mean & 3rd Qu. & Max. & sd & sd\\_err\\_of\\_mean\\\\\n",
       "\\hline\n",
       "\tcompleteness & 97.32 & 97.99 & 97.99 & 98.24538 & 98.66 & 99.33 & 0.683 & 0.279\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 1 × 8 of type dbl\n",
       "\n",
       "| <!--/--> | Min. | 1st Qu. | Median | Mean | 3rd Qu. | Max. | sd | sd_err_of_mean |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| completeness | 97.32 | 97.99 | 97.99 | 98.24538 | 98.66 | 99.33 | 0.683 | 0.279 |\n",
       "\n"
      ],
      "text/plain": [
       "             Min.  1st Qu. Median Mean     3rd Qu. Max.  sd    sd_err_of_mean\n",
       "completeness 97.32 97.99   97.99  98.24538 98.66   99.33 0.683 0.279         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 1 × 8 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Min.</th><th scope=col>1st Qu.</th><th scope=col>Median</th><th scope=col>Mean</th><th scope=col>3rd Qu.</th><th scope=col>Max.</th><th scope=col>sd</th><th scope=col>sd_err_of_mean</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>contamination</th><td>0</td><td>0.055</td><td>0.22</td><td>0.4280769</td><td>0.59</td><td>1.34</td><td>0.495</td><td>0.202</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 1 × 8 of type dbl\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & Min. & 1st Qu. & Median & Mean & 3rd Qu. & Max. & sd & sd\\_err\\_of\\_mean\\\\\n",
       "\\hline\n",
       "\tcontamination & 0 & 0.055 & 0.22 & 0.4280769 & 0.59 & 1.34 & 0.495 & 0.202\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 1 × 8 of type dbl\n",
       "\n",
       "| <!--/--> | Min. | 1st Qu. | Median | Mean | 3rd Qu. | Max. | sd | sd_err_of_mean |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| contamination | 0 | 0.055 | 0.22 | 0.4280769 | 0.59 | 1.34 | 0.495 | 0.202 |\n",
       "\n"
      ],
      "text/plain": [
       "              Min. 1st Qu. Median Mean      3rd Qu. Max. sd    sd_err_of_mean\n",
       "contamination 0    0.055   0.22   0.4280769 0.59    1.34 0.495 0.202         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 1 × 8 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Min.</th><th scope=col>1st Qu.</th><th scope=col>Median</th><th scope=col>Mean</th><th scope=col>3rd Qu.</th><th scope=col>Max.</th><th scope=col>sd</th><th scope=col>sd_err_of_mean</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>No. of contigs</th><td>8</td><td>36</td><td>68</td><td>72.15385</td><td>96</td><td>251</td><td>85.307</td><td>34.826</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 1 × 8 of type dbl\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & Min. & 1st Qu. & Median & Mean & 3rd Qu. & Max. & sd & sd\\_err\\_of\\_mean\\\\\n",
       "\\hline\n",
       "\tNo. of contigs & 8 & 36 & 68 & 72.15385 & 96 & 251 & 85.307 & 34.826\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 1 × 8 of type dbl\n",
       "\n",
       "| <!--/--> | Min. | 1st Qu. | Median | Mean | 3rd Qu. | Max. | sd | sd_err_of_mean |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| No. of contigs | 8 | 36 | 68 | 72.15385 | 96 | 251 | 85.307 | 34.826 |\n",
       "\n"
      ],
      "text/plain": [
       "               Min. 1st Qu. Median Mean     3rd Qu. Max. sd     sd_err_of_mean\n",
       "No. of contigs 8    36      68     72.15385 96      251  85.307 34.826        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 1 × 8 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Min.</th><th scope=col>1st Qu.</th><th scope=col>Median</th><th scope=col>Mean</th><th scope=col>3rd Qu.</th><th scope=col>Max.</th><th scope=col>sd</th><th scope=col>sd_err_of_mean</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>N50</th><td>48108</td><td>87859.5</td><td>112468.5</td><td>303948.2</td><td>248682.5</td><td>2375707</td><td>909825.5</td><td>371434.7</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 1 × 8 of type dbl\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & Min. & 1st Qu. & Median & Mean & 3rd Qu. & Max. & sd & sd\\_err\\_of\\_mean\\\\\n",
       "\\hline\n",
       "\tN50 & 48108 & 87859.5 & 112468.5 & 303948.2 & 248682.5 & 2375707 & 909825.5 & 371434.7\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 1 × 8 of type dbl\n",
       "\n",
       "| <!--/--> | Min. | 1st Qu. | Median | Mean | 3rd Qu. | Max. | sd | sd_err_of_mean |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| N50 | 48108 | 87859.5 | 112468.5 | 303948.2 | 248682.5 | 2375707 | 909825.5 | 371434.7 |\n",
       "\n"
      ],
      "text/plain": [
       "    Min.  1st Qu. Median   Mean     3rd Qu.  Max.    sd       sd_err_of_mean\n",
       "N50 48108 87859.5 112468.5 303948.2 248682.5 2375707 909825.5 371434.7      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 1 × 8 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Min.</th><th scope=col>1st Qu.</th><th scope=col>Median</th><th scope=col>Mean</th><th scope=col>3rd Qu.</th><th scope=col>Max.</th><th scope=col>sd</th><th scope=col>sd_err_of_mean</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Mean contig length</th><td>14550</td><td>37144.25</td><td>45687</td><td>87416.5</td><td>101514.8</td><td>393242</td><td>140924.9</td><td>57532.35</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 1 × 8 of type dbl\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & Min. & 1st Qu. & Median & Mean & 3rd Qu. & Max. & sd & sd\\_err\\_of\\_mean\\\\\n",
       "\\hline\n",
       "\tMean contig length & 14550 & 37144.25 & 45687 & 87416.5 & 101514.8 & 393242 & 140924.9 & 57532.35\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 1 × 8 of type dbl\n",
       "\n",
       "| <!--/--> | Min. | 1st Qu. | Median | Mean | 3rd Qu. | Max. | sd | sd_err_of_mean |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Mean contig length | 14550 | 37144.25 | 45687 | 87416.5 | 101514.8 | 393242 | 140924.9 | 57532.35 |\n",
       "\n"
      ],
      "text/plain": [
       "                   Min.  1st Qu.  Median Mean    3rd Qu.  Max.   sd      \n",
       "Mean contig length 14550 37144.25 45687  87416.5 101514.8 393242 140924.9\n",
       "                   sd_err_of_mean\n",
       "Mean contig length 57532.35      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# summarizing the genome stats\n",
    "hq_genomes %>% summary_x('completeness', completeness)\n",
    "hq_genomes %>% summary_x('contamination', contamination)\n",
    "hq_genomes %>% summary_x('No. of contigs', X..contigs)\n",
    "hq_genomes %>% summary_x('N50', N50..contigs.)\n",
    "hq_genomes %>% summary_x('Mean contig length', Mean.contig.length..bp.)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-15T09:39:41.025198Z",
     "iopub.status.busy": "2021-01-15T09:39:41.023218Z",
     "iopub.status.idle": "2021-01-15T09:39:41.057391Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A tibble: 3 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Family</th><th scope=col>Genus</th><th scope=col>n_genomes</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>f__Acutalibacteraceae</td><td>g__Eubacterium_R</td><td> 1</td></tr>\n",
       "\t<tr><td>f__Anaerovoracaceae  </td><td>g__Emergencia   </td><td> 1</td></tr>\n",
       "\t<tr><td>f__Ruminococcaceae   </td><td>g__Anaerotruncus</td><td>24</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 3 × 3\n",
       "\\begin{tabular}{lll}\n",
       " Family & Genus & n\\_genomes\\\\\n",
       " <chr> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t f\\_\\_Acutalibacteraceae & g\\_\\_Eubacterium\\_R &  1\\\\\n",
       "\t f\\_\\_Anaerovoracaceae   & g\\_\\_Emergencia    &  1\\\\\n",
       "\t f\\_\\_Ruminococcaceae    & g\\_\\_Anaerotruncus & 24\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 3 × 3\n",
       "\n",
       "| Family &lt;chr&gt; | Genus &lt;chr&gt; | n_genomes &lt;int&gt; |\n",
       "|---|---|---|\n",
       "| f__Acutalibacteraceae | g__Eubacterium_R |  1 |\n",
       "| f__Anaerovoracaceae   | g__Emergencia    |  1 |\n",
       "| f__Ruminococcaceae    | g__Anaerotruncus | 24 |\n",
       "\n"
      ],
      "text/plain": [
       "  Family                Genus            n_genomes\n",
       "1 f__Acutalibacteraceae g__Eubacterium_R  1       \n",
       "2 f__Anaerovoracaceae   g__Emergencia     1       \n",
       "3 f__Ruminococcaceae    g__Anaerotruncus 24       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# summarizing the taxonomy\n",
    "df.dims(20)\n",
    "hq_genomes %>%\n",
    "    group_by(Family, Genus) %>%\n",
    "    summarize(n_genomes = n(), .groups='drop')\n",
    "df.dims()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-15T09:40:53.456570Z",
     "iopub.status.busy": "2021-01-15T09:40:53.454365Z",
     "iopub.status.idle": "2021-01-15T09:40:53.539337Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 24 × 60</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>genome</th><th scope=col>completeness</th><th scope=col>contamination</th><th scope=col>Bin.Id</th><th scope=col>Marker.lineage</th><th scope=col>X..genomes</th><th scope=col>X..markers</th><th scope=col>X..marker.sets</th><th scope=col>Completeness</th><th scope=col>Contamination</th><th scope=col>⋯</th><th scope=col>fastani_af</th><th scope=col>closest_placement_reference</th><th scope=col>closest_placement_radius</th><th scope=col>closest_placement_taxonomy</th><th scope=col>closest_placement_ani</th><th scope=col>closest_placement_af</th><th scope=col>msa_percent</th><th scope=col>translation_table</th><th scope=col>red_value</th><th scope=col>warnings</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000154565.1_ASM15456v1_genomic.fna</td><td>97.99</td><td>0.22</td><td>GCA_000154565.1_ASM15456v1_genomic</td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>97.99</td><td>0.22</td><td>⋯</td><td>1.0 </td><td>GCF_000154565.1</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus colihominis</td><td>100.0</td><td>1.0 </td><td>97.18</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_001404495.1_14207_7_62_genomic.fna</td><td>97.99</td><td>0.36</td><td>GCA_001404495.1_14207_7_62_genomic</td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>97.99</td><td>0.36</td><td>⋯</td><td>0.83</td><td>GCF_000154565.1</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus colihominis</td><td>99.07</td><td>0.83</td><td>95.50</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋱</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_900199635.1_PRJEB19460_genomic.fna     </td><td>99.33</td><td>0.00</td><td>GCA_900199635.1_PRJEB19460_genomic     </td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>99.33</td><td>0.00</td><td>⋯</td><td>1.0 </td><td>GCF_900199635.1</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus massiliensis</td><td>100.0</td><td>1.0 </td><td>96.43</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_902363605.1_MGYG-HGUT-00131_genomic.fna</td><td>97.99</td><td>1.34</td><td>GCA_902363605.1_MGYG-HGUT-00131_genomic</td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>97.99</td><td>1.34</td><td>⋯</td><td>0.91</td><td>GCF_900078395.1</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus rubiinfantis</td><td>99.26</td><td>0.91</td><td>96.35</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 24 × 60\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " genome & completeness & contamination & Bin.Id & Marker.lineage & X..genomes & X..markers & X..marker.sets & Completeness & Contamination & ⋯ & fastani\\_af & closest\\_placement\\_reference & closest\\_placement\\_radius & closest\\_placement\\_taxonomy & closest\\_placement\\_ani & closest\\_placement\\_af & msa\\_percent & translation\\_table & red\\_value & warnings\\\\\n",
       " <fct> & <dbl> & <dbl> & <chr> & <fct> & <int> & <int> & <int> & <dbl> & <dbl> & ⋯ & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <dbl> & <int> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_000154565.1\\_ASM15456v1\\_genomic.fna & 97.99 & 0.22 & GCA\\_000154565.1\\_ASM15456v1\\_genomic & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 97.99 & 0.22 & ⋯ & 1.0  & GCF\\_000154565.1 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Ruminococcaceae;g\\_\\_Anaerotruncus;s\\_\\_Anaerotruncus colihominis & 100.0 & 1.0  & 97.18 & 11 & N/A & N/A\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_001404495.1\\_14207\\_7\\_62\\_genomic.fna & 97.99 & 0.36 & GCA\\_001404495.1\\_14207\\_7\\_62\\_genomic & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 97.99 & 0.36 & ⋯ & 0.83 & GCF\\_000154565.1 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Ruminococcaceae;g\\_\\_Anaerotruncus;s\\_\\_Anaerotruncus colihominis & 99.07 & 0.83 & 95.50 & 11 & N/A & N/A\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋱ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_900199635.1\\_PRJEB19460\\_genomic.fna      & 99.33 & 0.00 & GCA\\_900199635.1\\_PRJEB19460\\_genomic      & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 99.33 & 0.00 & ⋯ & 1.0  & GCF\\_900199635.1 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Ruminococcaceae;g\\_\\_Anaerotruncus;s\\_\\_Anaerotruncus massiliensis & 100.0 & 1.0  & 96.43 & 11 & N/A & N/A\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_902363605.1\\_MGYG-HGUT-00131\\_genomic.fna & 97.99 & 1.34 & GCA\\_902363605.1\\_MGYG-HGUT-00131\\_genomic & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 97.99 & 1.34 & ⋯ & 0.91 & GCF\\_900078395.1 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Ruminococcaceae;g\\_\\_Anaerotruncus;s\\_\\_Anaerotruncus rubiinfantis & 99.26 & 0.91 & 96.35 & 11 & N/A & N/A\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 24 × 60\n",
       "\n",
       "| genome &lt;fct&gt; | completeness &lt;dbl&gt; | contamination &lt;dbl&gt; | Bin.Id &lt;chr&gt; | Marker.lineage &lt;fct&gt; | X..genomes &lt;int&gt; | X..markers &lt;int&gt; | X..marker.sets &lt;int&gt; | Completeness &lt;dbl&gt; | Contamination &lt;dbl&gt; | ⋯ ⋯ | fastani_af &lt;fct&gt; | closest_placement_reference &lt;fct&gt; | closest_placement_radius &lt;fct&gt; | closest_placement_taxonomy &lt;fct&gt; | closest_placement_ani &lt;fct&gt; | closest_placement_af &lt;fct&gt; | msa_percent &lt;dbl&gt; | translation_table &lt;int&gt; | red_value &lt;fct&gt; | warnings &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000154565.1_ASM15456v1_genomic.fna | 97.99 | 0.22 | GCA_000154565.1_ASM15456v1_genomic | o__Clostridiales (UID1212) | 172 | 263 | 149 | 97.99 | 0.22 | ⋯ | 1.0  | GCF_000154565.1 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus colihominis | 100.0 | 1.0  | 97.18 | 11 | N/A | N/A |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_001404495.1_14207_7_62_genomic.fna | 97.99 | 0.36 | GCA_001404495.1_14207_7_62_genomic | o__Clostridiales (UID1212) | 172 | 263 | 149 | 97.99 | 0.36 | ⋯ | 0.83 | GCF_000154565.1 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus colihominis | 99.07 | 0.83 | 95.50 | 11 | N/A | N/A |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋱ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_900199635.1_PRJEB19460_genomic.fna      | 99.33 | 0.00 | GCA_900199635.1_PRJEB19460_genomic      | o__Clostridiales (UID1212) | 172 | 263 | 149 | 99.33 | 0.00 | ⋯ | 1.0  | GCF_900199635.1 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus massiliensis | 100.0 | 1.0  | 96.43 | 11 | N/A | N/A |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_902363605.1_MGYG-HGUT-00131_genomic.fna | 97.99 | 1.34 | GCA_902363605.1_MGYG-HGUT-00131_genomic | o__Clostridiales (UID1212) | 172 | 263 | 149 | 97.99 | 1.34 | ⋯ | 0.91 | GCF_900078395.1 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus rubiinfantis | 99.26 | 0.91 | 96.35 | 11 | N/A | N/A |\n",
       "\n"
      ],
      "text/plain": [
       "   genome                                                                                           \n",
       "1  /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000154565.1_ASM15456v1_genomic.fna     \n",
       "2  /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_001404495.1_14207_7_62_genomic.fna     \n",
       "⋮  ⋮                                                                                                \n",
       "23 /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_900199635.1_PRJEB19460_genomic.fna     \n",
       "24 /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_902363605.1_MGYG-HGUT-00131_genomic.fna\n",
       "   completeness contamination Bin.Id                                 \n",
       "1  97.99        0.22          GCA_000154565.1_ASM15456v1_genomic     \n",
       "2  97.99        0.36          GCA_001404495.1_14207_7_62_genomic     \n",
       "⋮  ⋮            ⋮             ⋮                                      \n",
       "23 99.33        0.00          GCA_900199635.1_PRJEB19460_genomic     \n",
       "24 97.99        1.34          GCA_902363605.1_MGYG-HGUT-00131_genomic\n",
       "   Marker.lineage             X..genomes X..markers X..marker.sets Completeness\n",
       "1  o__Clostridiales (UID1212) 172        263        149            97.99       \n",
       "2  o__Clostridiales (UID1212) 172        263        149            97.99       \n",
       "⋮  ⋮                          ⋮          ⋮          ⋮              ⋮           \n",
       "23 o__Clostridiales (UID1212) 172        263        149            99.33       \n",
       "24 o__Clostridiales (UID1212) 172        263        149            97.99       \n",
       "   Contamination ⋯ fastani_af closest_placement_reference\n",
       "1  0.22          ⋯ 1.0        GCF_000154565.1            \n",
       "2  0.36          ⋯ 0.83       GCF_000154565.1            \n",
       "⋮  ⋮             ⋱ ⋮          ⋮                          \n",
       "23 0.00          ⋯ 1.0        GCF_900199635.1            \n",
       "24 1.34          ⋯ 0.91       GCF_900078395.1            \n",
       "   closest_placement_radius\n",
       "1  95.0                    \n",
       "2  95.0                    \n",
       "⋮  ⋮                       \n",
       "23 95.0                    \n",
       "24 95.0                    \n",
       "   closest_placement_taxonomy                                                                                                    \n",
       "1  d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus colihominis \n",
       "2  d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus colihominis \n",
       "⋮  ⋮                                                                                                                             \n",
       "23 d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus massiliensis\n",
       "24 d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus rubiinfantis\n",
       "   closest_placement_ani closest_placement_af msa_percent translation_table\n",
       "1  100.0                 1.0                  97.18       11               \n",
       "2  99.07                 0.83                 95.50       11               \n",
       "⋮  ⋮                     ⋮                    ⋮           ⋮                \n",
       "23 100.0                 1.0                  96.43       11               \n",
       "24 99.26                 0.91                 96.35       11               \n",
       "   red_value warnings\n",
       "1  N/A       N/A     \n",
       "2  N/A       N/A     \n",
       "⋮  ⋮         ⋮       \n",
       "23 N/A       N/A     \n",
       "24 N/A       N/A     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# filtering by taxonomy\n",
    "hq_genomes = hq_genomes %>%\n",
    "    filter(Genus == 'g__Anaerotruncus')\n",
    "hq_genomes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-15T09:40:58.742599Z",
     "iopub.status.busy": "2021-01-15T09:40:58.740434Z",
     "iopub.status.idle": "2021-01-15T09:40:58.769678Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "File written: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted//Anaerotruncus/samples_genomes_hq.txt \n"
     ]
    }
   ],
   "source": [
    "# writing samples table for LLPRIMER\n",
    "outfile = file.path(work_dir, 'samples_genomes_hq.txt')\n",
    "hq_genomes %>%\n",
    "    select(Bin.Id, Fasta) %>%\n",
    "    rename('Taxon' = Bin.Id) %>%\n",
    "    mutate(Taxon = gsub('_chromosome.+', '', Taxon),\n",
    "           Taxon = gsub('_bin_.+', '', Taxon),\n",
    "           Taxon = gsub('_genomic', '', Taxon),\n",
    "           Taxon = gsub('_annotated_assembly', '', Taxon),\n",
    "           Taxid = taxid) %>%\n",
    "    write_table(outfile)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Primer design"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Config"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-15T09:40:05.356269Z",
     "iopub.status.busy": "2021-01-15T09:40:05.354229Z",
     "iopub.status.idle": "2021-01-15T09:40:05.382719Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#-- I/O --#\n",
      "samples_file: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/samples_genomes_hq.txt\n",
      "\n",
      "# output location\n",
      "output_dir: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/primers/\n",
      "\n",
      "# temporary file directory (your username will be added automatically)\n",
      "tmp_dir: /ebio/abt3_scratch/\n",
      "\n",
      "#-- software parameters --#\n",
      "# See the README for a description\n",
      "params:\n",
      "  ionice: -c 3\n",
      "  cgp:\n",
      "    prodigal: \"\"    \n",
      "    mmseqs:\n",
      "      method: cluster    # or linclust (faster)\n",
      "      run: --min-seq-id 0.8 -c 0.8\n",
      "    core_genes: --frac 1 --max-clusters 500\n",
      "    blastx: -evalue 1e-10 -max_target_seqs 3\n",
      "    blastx_nontarget: -evalue 1e-5 -max_target_seqs 30\n",
      "    align:\n",
      "      method: linsi\n",
      "      params: --auto --maxiterate 1000\n",
      "    primer3:\n",
      "      number: --num-primers 500\n",
      "      size: --opt-size 20 --min-size 18 --max-size 24\n",
      "      product: --opt-prod-size 150 --min-prod-size 100 --max-prod-size 200\n",
      "      Tm: --opt-tm 62 --min-tm 55 --max-tm 70 --max-tm-diff 1\n",
      "      PCR: --oligo-DNA 50 --salt-monovalent 50 --salt-divalent 1.5 --dNTPs 0.2\n",
      "      GC: --opt-gc 50 --min-gc 25 --max-gc 75\n",
      "      degeneracy: --max-degeneracy 128 --max-degeneracy-3prime 1 --window-3prime 5\n",
      "      internal_oligo: --make-oligo 0 --int-opt-size 20 --int-opt-tm 60\n",
      "    blastn: -max_target_seqs 500 -perc_identity 90\n",
      "    blast_filter: --perc-len 80 --min-amplicon-len 30 --max-amplicon-len 2000    \n",
      "\n",
      "#-- database --#\n",
      "databases:\n",
      "  blast:\n",
      "    nucl: /ebio/abt3_projects/databases_no-backup/NCBI_blastdb/nt\n",
      "    prot: /ebio/abt3_projects/databases_no-backup/NCBI_blastdb/nr\n",
      "    gene_names: /ebio/abt3_projects/databases_no-backup/NCBI_nr_geneIDs/2021-01/nr_accession-geneNames.pkl\n",
      "  taxonomy:\n",
      "    names: /ebio/abt3_projects/databases_no-backup/NCBI_taxonomy/2020-12-23/names.dmp\n",
      "    nodes: /ebio/abt3_projects/databases_no-backup/NCBI_taxonomy/2020-12-23/nodes.dmp\n",
      "\n",
      "#-- snakemake pipeline --#\n",
      "pipeline:\n",
      "  snakemake_folder: ./\n",
      "  script_folder: bin/scripts/\n",
      "  name: LLPRIMER"
     ]
    }
   ],
   "source": [
    "F = file.path(work_dir, 'primers', 'config.yaml')\n",
    "cat_file(F)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Run"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "(snakemake) @ rick:/ebio/abt3_projects/software/dev/ll_pipelines/llprimer\n",
    "$ screen -L -S llprimer-aner ./snakemake_sge.sh experiments/HMP_most-wanted/Anaerotruncus/primers/config.yaml 50 --notemp -F\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-17T07:43:57.016878Z",
     "iopub.status.busy": "2021-01-17T07:43:57.015090Z",
     "iopub.status.idle": "2021-01-17T07:43:57.084572Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No. of unique primer sets: 276 \n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 774 × 21</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>cluster_id</th><th scope=col>primer_set</th><th scope=col>amplicon_size_consensus</th><th scope=col>amplicon_size_avg</th><th scope=col>amplicon_size_sd</th><th scope=col>primer_id</th><th scope=col>primer_type</th><th scope=col>sequence</th><th scope=col>length</th><th scope=col>degeneracy</th><th scope=col>⋯</th><th scope=col>position_start</th><th scope=col>position_end</th><th scope=col>Tm_avg</th><th scope=col>Tm_sd</th><th scope=col>GC_avg</th><th scope=col>GC_sd</th><th scope=col>hairpin_avg</th><th scope=col>hairpin_sd</th><th scope=col>homodimer_avg</th><th scope=col>homodimer_sd</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>4</td><td>267</td><td>121</td><td>121</td><td>0</td><td>267f</td><td>PRIMER_LEFT </td><td>KMSAAGCTGCTSAARTCGGC</td><td>20</td><td>32</td><td>⋯</td><td>141</td><td>161</td><td>62.20092</td><td>1.817715</td><td>57.50000</td><td>4.330127</td><td>44.12935</td><td> 7.786871</td><td>6.123101</td><td> 9.199705</td></tr>\n",
       "\t<tr><td>4</td><td>267</td><td>121</td><td>121</td><td>0</td><td>267r</td><td>PRIMER_RIGHT</td><td>GGGGRCGRATVCGYTTCAT </td><td>19</td><td>24</td><td>⋯</td><td>243</td><td>262</td><td>61.25268</td><td>2.574313</td><td>58.77193</td><td>5.189544</td><td>53.52857</td><td>12.395231</td><td>7.186602</td><td>22.559500</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋱</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>246</td><td>438</td><td>165</td><td>165</td><td>0</td><td>438f</td><td>PRIMER_LEFT </td><td>YGAYTTYCCGCTBTCMAACTG</td><td>21</td><td>48</td><td>⋯</td><td>122</td><td>143</td><td>60.35574</td><td>1.733370</td><td>50.79365</td><td>5.264484</td><td> 4.452419</td><td>11.83836</td><td>-38.53541</td><td> 5.906476</td></tr>\n",
       "\t<tr><td>246</td><td>438</td><td>165</td><td>165</td><td>0</td><td>438r</td><td>PRIMER_RIGHT</td><td>GAGSWYTTCTGRTAVGGCGG </td><td>20</td><td>48</td><td>⋯</td><td>267</td><td>287</td><td>60.29829</td><td>1.778744</td><td>58.33333</td><td>4.249183</td><td>13.355395</td><td>18.42953</td><td>-13.88619</td><td>23.840250</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 774 × 21\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " cluster\\_id & primer\\_set & amplicon\\_size\\_consensus & amplicon\\_size\\_avg & amplicon\\_size\\_sd & primer\\_id & primer\\_type & sequence & length & degeneracy & ⋯ & position\\_start & position\\_end & Tm\\_avg & Tm\\_sd & GC\\_avg & GC\\_sd & hairpin\\_avg & hairpin\\_sd & homodimer\\_avg & homodimer\\_sd\\\\\n",
       " <int> & <int> & <int> & <dbl> & <dbl> & <fct> & <fct> & <fct> & <int> & <int> & ⋯ & <int> & <int> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t 4 & 267 & 121 & 121 & 0 & 267f & PRIMER\\_LEFT  & KMSAAGCTGCTSAARTCGGC & 20 & 32 & ⋯ & 141 & 161 & 62.20092 & 1.817715 & 57.50000 & 4.330127 & 44.12935 &  7.786871 & 6.123101 &  9.199705\\\\\n",
       "\t 4 & 267 & 121 & 121 & 0 & 267r & PRIMER\\_RIGHT & GGGGRCGRATVCGYTTCAT  & 19 & 24 & ⋯ & 243 & 262 & 61.25268 & 2.574313 & 58.77193 & 5.189544 & 53.52857 & 12.395231 & 7.186602 & 22.559500\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋱ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t 246 & 438 & 165 & 165 & 0 & 438f & PRIMER\\_LEFT  & YGAYTTYCCGCTBTCMAACTG & 21 & 48 & ⋯ & 122 & 143 & 60.35574 & 1.733370 & 50.79365 & 5.264484 &  4.452419 & 11.83836 & -38.53541 &  5.906476\\\\\n",
       "\t 246 & 438 & 165 & 165 & 0 & 438r & PRIMER\\_RIGHT & GAGSWYTTCTGRTAVGGCGG  & 20 & 48 & ⋯ & 267 & 287 & 60.29829 & 1.778744 & 58.33333 & 4.249183 & 13.355395 & 18.42953 & -13.88619 & 23.840250\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 774 × 21\n",
       "\n",
       "| cluster_id &lt;int&gt; | primer_set &lt;int&gt; | amplicon_size_consensus &lt;int&gt; | amplicon_size_avg &lt;dbl&gt; | amplicon_size_sd &lt;dbl&gt; | primer_id &lt;fct&gt; | primer_type &lt;fct&gt; | sequence &lt;fct&gt; | length &lt;int&gt; | degeneracy &lt;int&gt; | ⋯ ⋯ | position_start &lt;int&gt; | position_end &lt;int&gt; | Tm_avg &lt;dbl&gt; | Tm_sd &lt;dbl&gt; | GC_avg &lt;dbl&gt; | GC_sd &lt;dbl&gt; | hairpin_avg &lt;dbl&gt; | hairpin_sd &lt;dbl&gt; | homodimer_avg &lt;dbl&gt; | homodimer_sd &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 4 | 267 | 121 | 121 | 0 | 267f | PRIMER_LEFT  | KMSAAGCTGCTSAARTCGGC | 20 | 32 | ⋯ | 141 | 161 | 62.20092 | 1.817715 | 57.50000 | 4.330127 | 44.12935 |  7.786871 | 6.123101 |  9.199705 |\n",
       "| 4 | 267 | 121 | 121 | 0 | 267r | PRIMER_RIGHT | GGGGRCGRATVCGYTTCAT  | 19 | 24 | ⋯ | 243 | 262 | 61.25268 | 2.574313 | 58.77193 | 5.189544 | 53.52857 | 12.395231 | 7.186602 | 22.559500 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋱ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| 246 | 438 | 165 | 165 | 0 | 438f | PRIMER_LEFT  | YGAYTTYCCGCTBTCMAACTG | 21 | 48 | ⋯ | 122 | 143 | 60.35574 | 1.733370 | 50.79365 | 5.264484 |  4.452419 | 11.83836 | -38.53541 |  5.906476 |\n",
       "| 246 | 438 | 165 | 165 | 0 | 438r | PRIMER_RIGHT | GAGSWYTTCTGRTAVGGCGG  | 20 | 48 | ⋯ | 267 | 287 | 60.29829 | 1.778744 | 58.33333 | 4.249183 | 13.355395 | 18.42953 | -13.88619 | 23.840250 |\n",
       "\n"
      ],
      "text/plain": [
       "    cluster_id primer_set amplicon_size_consensus amplicon_size_avg\n",
       "1   4          267        121                     121              \n",
       "2   4          267        121                     121              \n",
       "⋮   ⋮          ⋮          ⋮                       ⋮                \n",
       "773 246        438        165                     165              \n",
       "774 246        438        165                     165              \n",
       "    amplicon_size_sd primer_id primer_type  sequence              length\n",
       "1   0                267f      PRIMER_LEFT  KMSAAGCTGCTSAARTCGGC  20    \n",
       "2   0                267r      PRIMER_RIGHT GGGGRCGRATVCGYTTCAT   19    \n",
       "⋮   ⋮                ⋮         ⋮            ⋮                     ⋮     \n",
       "773 0                438f      PRIMER_LEFT  YGAYTTYCCGCTBTCMAACTG 21    \n",
       "774 0                438r      PRIMER_RIGHT GAGSWYTTCTGRTAVGGCGG  20    \n",
       "    degeneracy ⋯ position_start position_end Tm_avg   Tm_sd    GC_avg  \n",
       "1   32         ⋯ 141            161          62.20092 1.817715 57.50000\n",
       "2   24         ⋯ 243            262          61.25268 2.574313 58.77193\n",
       "⋮   ⋮          ⋱ ⋮              ⋮            ⋮        ⋮        ⋮       \n",
       "773 48         ⋯ 122            143          60.35574 1.733370 50.79365\n",
       "774 48         ⋯ 267            287          60.29829 1.778744 58.33333\n",
       "    GC_sd    hairpin_avg hairpin_sd homodimer_avg homodimer_sd\n",
       "1   4.330127 44.12935     7.786871  6.123101       9.199705   \n",
       "2   5.189544 53.52857    12.395231  7.186602      22.559500   \n",
       "⋮   ⋮        ⋮           ⋮          ⋮             ⋮           \n",
       "773 5.264484  4.452419   11.83836   -38.53541      5.906476   \n",
       "774 4.249183 13.355395   18.42953   -13.88619     23.840250   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "primer_info = read.delim(file.path(work_dir, 'primers', 'cgp', 'primers_final_info.tsv'), sep='\\t')\n",
    "primer_info %>% unique_n('primer sets', primer_set)\n",
    "primer_info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Gene cluster annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-17T07:38:55.943674Z",
     "iopub.status.busy": "2021-01-17T07:38:55.941540Z",
     "iopub.status.idle": "2021-01-17T07:38:56.001963Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 228 × 19</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>cluster_id</th><th scope=col>query</th><th scope=col>subject</th><th scope=col>subject_name</th><th scope=col>pident</th><th scope=col>length</th><th scope=col>mismatch</th><th scope=col>qstart</th><th scope=col>qend</th><th scope=col>sstart</th><th scope=col>send</th><th scope=col>evalue</th><th scope=col>slen</th><th scope=col>qlen</th><th scope=col>sscinames</th><th scope=col>staxids</th><th scope=col>pident_rank</th><th scope=col>gene_name</th><th scope=col>gene_taxonomy</th></tr>\n",
       "\t<tr><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>4</td><td>abf8b17f07404c7183f7f0f8c044a48f</td><td>WP_066455364.1</td><td>MULTISPECIES: 50S ribosomal protein L22 [Anaerotruncus]</td><td>100.000</td><td>111</td><td>0</td><td>1</td><td>333</td><td>1</td><td>111</td><td>3.92e-75</td><td>111</td><td>336</td><td>Anaerotruncus;Anaerotruncus rubiinfantis;Anaerotruncus sp. AF02-27</td><td>244127;1720200;2292191</td><td>3</td><td>MULTISPECIES: 50S ribosomal protein L22</td><td>Anaerotruncus             </td></tr>\n",
       "\t<tr><td>4</td><td>abf8b17f07404c7183f7f0f8c044a48f</td><td>WP_147539073.1</td><td>50S ribosomal protein L22 [Anaerotruncus rubiinfantis] </td><td> 99.099</td><td>111</td><td>1</td><td>1</td><td>333</td><td>1</td><td>111</td><td>1.26e-74</td><td>111</td><td>336</td><td>Anaerotruncus rubiinfantis                                        </td><td>1720200               </td><td>1</td><td>50S ribosomal protein L22              </td><td>Anaerotruncus rubiinfantis</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>246</td><td>ce2e4df7569647278c025e0d6334cd91</td><td>WP_006874432.1</td><td>glucose-1-phosphate adenylyltransferase [Anaerotruncus colihominis]</td><td>99.762</td><td>420</td><td>1</td><td>1</td><td>1260</td><td>1</td><td>420</td><td>0</td><td>420</td><td>1263</td><td>Anaerotruncus colihominis;Anaerotruncus colihominis DSM 17241</td><td>169435;445972</td><td>1</td><td>glucose-1-phosphate adenylyltransferase</td><td>Anaerotruncus colihominis</td></tr>\n",
       "\t<tr><td>246</td><td>ce2e4df7569647278c025e0d6334cd91</td><td>WP_195462589.1</td><td>glucose-1-phosphate adenylyltransferase [Anaerotruncus colihominis]</td><td>99.048</td><td>420</td><td>4</td><td>1</td><td>1260</td><td>1</td><td>420</td><td>0</td><td>420</td><td>1263</td><td>Anaerotruncus colihominis                                    </td><td>169435       </td><td>2</td><td>glucose-1-phosphate adenylyltransferase</td><td>Anaerotruncus colihominis</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 228 × 19\n",
       "\\begin{tabular}{lllllllllllllllllll}\n",
       " cluster\\_id & query & subject & subject\\_name & pident & length & mismatch & qstart & qend & sstart & send & evalue & slen & qlen & sscinames & staxids & pident\\_rank & gene\\_name & gene\\_taxonomy\\\\\n",
       " <dbl> & <fct> & <fct> & <fct> & <dbl> & <int> & <int> & <int> & <int> & <int> & <int> & <dbl> & <int> & <int> & <fct> & <fct> & <int> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t 4 & abf8b17f07404c7183f7f0f8c044a48f & WP\\_066455364.1 & MULTISPECIES: 50S ribosomal protein L22 {[}Anaerotruncus{]} & 100.000 & 111 & 0 & 1 & 333 & 1 & 111 & 3.92e-75 & 111 & 336 & Anaerotruncus;Anaerotruncus rubiinfantis;Anaerotruncus sp. AF02-27 & 244127;1720200;2292191 & 3 & MULTISPECIES: 50S ribosomal protein L22 & Anaerotruncus             \\\\\n",
       "\t 4 & abf8b17f07404c7183f7f0f8c044a48f & WP\\_147539073.1 & 50S ribosomal protein L22 {[}Anaerotruncus rubiinfantis{]}  &  99.099 & 111 & 1 & 1 & 333 & 1 & 111 & 1.26e-74 & 111 & 336 & Anaerotruncus rubiinfantis                                         & 1720200                & 1 & 50S ribosomal protein L22               & Anaerotruncus rubiinfantis\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t 246 & ce2e4df7569647278c025e0d6334cd91 & WP\\_006874432.1 & glucose-1-phosphate adenylyltransferase {[}Anaerotruncus colihominis{]} & 99.762 & 420 & 1 & 1 & 1260 & 1 & 420 & 0 & 420 & 1263 & Anaerotruncus colihominis;Anaerotruncus colihominis DSM 17241 & 169435;445972 & 1 & glucose-1-phosphate adenylyltransferase & Anaerotruncus colihominis\\\\\n",
       "\t 246 & ce2e4df7569647278c025e0d6334cd91 & WP\\_195462589.1 & glucose-1-phosphate adenylyltransferase {[}Anaerotruncus colihominis{]} & 99.048 & 420 & 4 & 1 & 1260 & 1 & 420 & 0 & 420 & 1263 & Anaerotruncus colihominis                                     & 169435        & 2 & glucose-1-phosphate adenylyltransferase & Anaerotruncus colihominis\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 228 × 19\n",
       "\n",
       "| cluster_id &lt;dbl&gt; | query &lt;fct&gt; | subject &lt;fct&gt; | subject_name &lt;fct&gt; | pident &lt;dbl&gt; | length &lt;int&gt; | mismatch &lt;int&gt; | qstart &lt;int&gt; | qend &lt;int&gt; | sstart &lt;int&gt; | send &lt;int&gt; | evalue &lt;dbl&gt; | slen &lt;int&gt; | qlen &lt;int&gt; | sscinames &lt;fct&gt; | staxids &lt;fct&gt; | pident_rank &lt;int&gt; | gene_name &lt;chr&gt; | gene_taxonomy &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 4 | abf8b17f07404c7183f7f0f8c044a48f | WP_066455364.1 | MULTISPECIES: 50S ribosomal protein L22 [Anaerotruncus] | 100.000 | 111 | 0 | 1 | 333 | 1 | 111 | 3.92e-75 | 111 | 336 | Anaerotruncus;Anaerotruncus rubiinfantis;Anaerotruncus sp. AF02-27 | 244127;1720200;2292191 | 3 | MULTISPECIES: 50S ribosomal protein L22 | Anaerotruncus              |\n",
       "| 4 | abf8b17f07404c7183f7f0f8c044a48f | WP_147539073.1 | 50S ribosomal protein L22 [Anaerotruncus rubiinfantis]  |  99.099 | 111 | 1 | 1 | 333 | 1 | 111 | 1.26e-74 | 111 | 336 | Anaerotruncus rubiinfantis                                         | 1720200                | 1 | 50S ribosomal protein L22               | Anaerotruncus rubiinfantis |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| 246 | ce2e4df7569647278c025e0d6334cd91 | WP_006874432.1 | glucose-1-phosphate adenylyltransferase [Anaerotruncus colihominis] | 99.762 | 420 | 1 | 1 | 1260 | 1 | 420 | 0 | 420 | 1263 | Anaerotruncus colihominis;Anaerotruncus colihominis DSM 17241 | 169435;445972 | 1 | glucose-1-phosphate adenylyltransferase | Anaerotruncus colihominis |\n",
       "| 246 | ce2e4df7569647278c025e0d6334cd91 | WP_195462589.1 | glucose-1-phosphate adenylyltransferase [Anaerotruncus colihominis] | 99.048 | 420 | 4 | 1 | 1260 | 1 | 420 | 0 | 420 | 1263 | Anaerotruncus colihominis                                     | 169435        | 2 | glucose-1-phosphate adenylyltransferase | Anaerotruncus colihominis |\n",
       "\n"
      ],
      "text/plain": [
       "    cluster_id query                            subject       \n",
       "1   4          abf8b17f07404c7183f7f0f8c044a48f WP_066455364.1\n",
       "2   4          abf8b17f07404c7183f7f0f8c044a48f WP_147539073.1\n",
       "⋮   ⋮          ⋮                                ⋮             \n",
       "227 246        ce2e4df7569647278c025e0d6334cd91 WP_006874432.1\n",
       "228 246        ce2e4df7569647278c025e0d6334cd91 WP_195462589.1\n",
       "    subject_name                                                        pident \n",
       "1   MULTISPECIES: 50S ribosomal protein L22 [Anaerotruncus]             100.000\n",
       "2   50S ribosomal protein L22 [Anaerotruncus rubiinfantis]               99.099\n",
       "⋮   ⋮                                                                   ⋮      \n",
       "227 glucose-1-phosphate adenylyltransferase [Anaerotruncus colihominis] 99.762 \n",
       "228 glucose-1-phosphate adenylyltransferase [Anaerotruncus colihominis] 99.048 \n",
       "    length mismatch qstart qend sstart send evalue   slen qlen\n",
       "1   111    0        1      333  1      111  3.92e-75 111  336 \n",
       "2   111    1        1      333  1      111  1.26e-74 111  336 \n",
       "⋮   ⋮      ⋮        ⋮      ⋮    ⋮      ⋮    ⋮        ⋮    ⋮   \n",
       "227 420    1        1      1260 1      420  0        420  1263\n",
       "228 420    4        1      1260 1      420  0        420  1263\n",
       "    sscinames                                                         \n",
       "1   Anaerotruncus;Anaerotruncus rubiinfantis;Anaerotruncus sp. AF02-27\n",
       "2   Anaerotruncus rubiinfantis                                        \n",
       "⋮   ⋮                                                                 \n",
       "227 Anaerotruncus colihominis;Anaerotruncus colihominis DSM 17241     \n",
       "228 Anaerotruncus colihominis                                         \n",
       "    staxids                pident_rank gene_name                              \n",
       "1   244127;1720200;2292191 3           MULTISPECIES: 50S ribosomal protein L22\n",
       "2   1720200                1           50S ribosomal protein L22              \n",
       "⋮   ⋮                      ⋮           ⋮                                      \n",
       "227 169435;445972          1           glucose-1-phosphate adenylyltransferase\n",
       "228 169435                 2           glucose-1-phosphate adenylyltransferase\n",
       "    gene_taxonomy             \n",
       "1   Anaerotruncus             \n",
       "2   Anaerotruncus rubiinfantis\n",
       "⋮   ⋮                         \n",
       "227 Anaerotruncus colihominis \n",
       "228 Anaerotruncus colihominis "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "gene_annot = read.delim(file.path(work_dir, 'primers', 'cgp', 'core_clusters_blastx.tsv'), \n",
    "                        sep='\\t') %>%\n",
    "    mutate(cluster_id = gsub('cluster_', '', cluster_id) %>% as.Num) %>%\n",
    "    semi_join(primer_info, c('cluster_id')) %>%\n",
    "    mutate(gene_name = gsub(' \\\\[.+', '', subject_name),\n",
    "           gene_taxonomy = gsub('.+\\\\[', '', subject_name),\n",
    "           gene_taxonomy = gsub('\\\\]', '', gene_taxonomy))\n",
    "gene_annot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-17T07:39:05.117049Z",
     "iopub.status.busy": "2021-01-17T07:39:05.108779Z",
     "iopub.status.idle": "2021-01-17T07:39:05.163153Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 139 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>cluster_id</th><th scope=col>gene_name</th></tr>\n",
       "\t<tr><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td> 4</td><td>MULTISPECIES: 50S ribosomal protein L22                </td></tr>\n",
       "\t<tr><td> 4</td><td>50S ribosomal protein L22                              </td></tr>\n",
       "\t<tr><td> 5</td><td>MULTISPECIES: type I DNA topoisomerase                 </td></tr>\n",
       "\t<tr><td> 5</td><td>type I DNA topoisomerase                               </td></tr>\n",
       "\t<tr><td>10</td><td>Catabolite repression HPr                              </td></tr>\n",
       "\t<tr><td>10</td><td>HPr family phosphocarrier protein                      </td></tr>\n",
       "\t<tr><td>10</td><td>MULTISPECIES: HPr family phosphocarrier protein        </td></tr>\n",
       "\t<tr><td>11</td><td>50S ribosomal protein L30                              </td></tr>\n",
       "\t<tr><td>11</td><td>MULTISPECIES: 50S ribosomal protein L30                </td></tr>\n",
       "\t<tr><td>13</td><td>MULTISPECIES: 30S ribosomal protein S12                </td></tr>\n",
       "\t<tr><td>13</td><td>30S ribosomal protein S12                              </td></tr>\n",
       "\t<tr><td>19</td><td>tRNA (guanosine(37)-N1)-methyltransferase TrmD         </td></tr>\n",
       "\t<tr><td>19</td><td>tRNA (guanine-N(1)-)-methyltransferase                 </td></tr>\n",
       "\t<tr><td>20</td><td>MULTISPECIES: 30S ribosomal protein S3                 </td></tr>\n",
       "\t<tr><td>20</td><td>30S ribosomal protein S3                               </td></tr>\n",
       "\t<tr><td>22</td><td>MULTISPECIES: DNA-directed RNA polymerase subunit beta'</td></tr>\n",
       "\t<tr><td>22</td><td>DNA-directed RNA polymerase subunit beta'              </td></tr>\n",
       "\t<tr><td>24</td><td>MULTISPECIES: glycosyltransferase family 2 protein     </td></tr>\n",
       "\t<tr><td>33</td><td>MULTISPECIES: 50S ribosomal protein L11                </td></tr>\n",
       "\t<tr><td>33</td><td>50S ribosomal protein L11                              </td></tr>\n",
       "\t<tr><td>34</td><td>MULTISPECIES: 30S ribosomal protein S7                 </td></tr>\n",
       "\t<tr><td>34</td><td>30S ribosomal protein S7                               </td></tr>\n",
       "\t<tr><td>35</td><td>30S ribosomal protein S13                              </td></tr>\n",
       "\t<tr><td>35</td><td>MULTISPECIES: 30S ribosomal protein S13                </td></tr>\n",
       "\t<tr><td>37</td><td>DNA-directed RNA polymerase subunit alpha              </td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>200</td><td>MULTISPECIES: UMP kinase                             </td></tr>\n",
       "\t<tr><td>200</td><td>UMP kinase                                           </td></tr>\n",
       "\t<tr><td>201</td><td>NADH-quinone oxidoreductase subunit NuoF             </td></tr>\n",
       "\t<tr><td>204</td><td>preprotein translocase subunit SecA                  </td></tr>\n",
       "\t<tr><td>206</td><td>MULTISPECIES: translation initiation factor IF-1     </td></tr>\n",
       "\t<tr><td>206</td><td>translation initiation factor IF-1                   </td></tr>\n",
       "\t<tr><td>208</td><td>MULTISPECIES: uracil phosphoribosyltransferase       </td></tr>\n",
       "\t<tr><td>208</td><td>uracil phosphoribosyltransferase                     </td></tr>\n",
       "\t<tr><td>214</td><td>chaperonin GroS                                      </td></tr>\n",
       "\t<tr><td>214</td><td>co-chaperone GroES                                   </td></tr>\n",
       "\t<tr><td>214</td><td>MULTISPECIES: co-chaperone GroES                     </td></tr>\n",
       "\t<tr><td>220</td><td>phosphoenolpyruvate carboxykinase (GTP)              </td></tr>\n",
       "\t<tr><td>220</td><td>MULTISPECIES: phosphoenolpyruvate carboxykinase (GTP)</td></tr>\n",
       "\t<tr><td>221</td><td>phenylacetate--CoA ligase                            </td></tr>\n",
       "\t<tr><td>221</td><td>MULTISPECIES: phenylacetate--CoA ligase              </td></tr>\n",
       "\t<tr><td>227</td><td>MULTISPECIES: cell wall hydrolase                    </td></tr>\n",
       "\t<tr><td>227</td><td>cell wall hydrolase                                  </td></tr>\n",
       "\t<tr><td>237</td><td>MULTISPECIES: 30S ribosomal protein S11              </td></tr>\n",
       "\t<tr><td>237</td><td>30S ribosomal protein S11                            </td></tr>\n",
       "\t<tr><td>240</td><td>elongation factor G                                  </td></tr>\n",
       "\t<tr><td>241</td><td>MULTISPECIES: 50S ribosomal protein L19              </td></tr>\n",
       "\t<tr><td>241</td><td>50S ribosomal protein L19                            </td></tr>\n",
       "\t<tr><td>243</td><td>MULTISPECIES: response regulator transcription factor</td></tr>\n",
       "\t<tr><td>243</td><td>response regulator transcription factor              </td></tr>\n",
       "\t<tr><td>246</td><td>glucose-1-phosphate adenylyltransferase              </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 139 × 2\n",
       "\\begin{tabular}{ll}\n",
       " cluster\\_id & gene\\_name\\\\\n",
       " <dbl> & <chr>\\\\\n",
       "\\hline\n",
       "\t  4 & MULTISPECIES: 50S ribosomal protein L22                \\\\\n",
       "\t  4 & 50S ribosomal protein L22                              \\\\\n",
       "\t  5 & MULTISPECIES: type I DNA topoisomerase                 \\\\\n",
       "\t  5 & type I DNA topoisomerase                               \\\\\n",
       "\t 10 & Catabolite repression HPr                              \\\\\n",
       "\t 10 & HPr family phosphocarrier protein                      \\\\\n",
       "\t 10 & MULTISPECIES: HPr family phosphocarrier protein        \\\\\n",
       "\t 11 & 50S ribosomal protein L30                              \\\\\n",
       "\t 11 & MULTISPECIES: 50S ribosomal protein L30                \\\\\n",
       "\t 13 & MULTISPECIES: 30S ribosomal protein S12                \\\\\n",
       "\t 13 & 30S ribosomal protein S12                              \\\\\n",
       "\t 19 & tRNA (guanosine(37)-N1)-methyltransferase TrmD         \\\\\n",
       "\t 19 & tRNA (guanine-N(1)-)-methyltransferase                 \\\\\n",
       "\t 20 & MULTISPECIES: 30S ribosomal protein S3                 \\\\\n",
       "\t 20 & 30S ribosomal protein S3                               \\\\\n",
       "\t 22 & MULTISPECIES: DNA-directed RNA polymerase subunit beta'\\\\\n",
       "\t 22 & DNA-directed RNA polymerase subunit beta'              \\\\\n",
       "\t 24 & MULTISPECIES: glycosyltransferase family 2 protein     \\\\\n",
       "\t 33 & MULTISPECIES: 50S ribosomal protein L11                \\\\\n",
       "\t 33 & 50S ribosomal protein L11                              \\\\\n",
       "\t 34 & MULTISPECIES: 30S ribosomal protein S7                 \\\\\n",
       "\t 34 & 30S ribosomal protein S7                               \\\\\n",
       "\t 35 & 30S ribosomal protein S13                              \\\\\n",
       "\t 35 & MULTISPECIES: 30S ribosomal protein S13                \\\\\n",
       "\t 37 & DNA-directed RNA polymerase subunit alpha              \\\\\n",
       "\t ⋮ & ⋮\\\\\n",
       "\t 200 & MULTISPECIES: UMP kinase                             \\\\\n",
       "\t 200 & UMP kinase                                           \\\\\n",
       "\t 201 & NADH-quinone oxidoreductase subunit NuoF             \\\\\n",
       "\t 204 & preprotein translocase subunit SecA                  \\\\\n",
       "\t 206 & MULTISPECIES: translation initiation factor IF-1     \\\\\n",
       "\t 206 & translation initiation factor IF-1                   \\\\\n",
       "\t 208 & MULTISPECIES: uracil phosphoribosyltransferase       \\\\\n",
       "\t 208 & uracil phosphoribosyltransferase                     \\\\\n",
       "\t 214 & chaperonin GroS                                      \\\\\n",
       "\t 214 & co-chaperone GroES                                   \\\\\n",
       "\t 214 & MULTISPECIES: co-chaperone GroES                     \\\\\n",
       "\t 220 & phosphoenolpyruvate carboxykinase (GTP)              \\\\\n",
       "\t 220 & MULTISPECIES: phosphoenolpyruvate carboxykinase (GTP)\\\\\n",
       "\t 221 & phenylacetate--CoA ligase                            \\\\\n",
       "\t 221 & MULTISPECIES: phenylacetate--CoA ligase              \\\\\n",
       "\t 227 & MULTISPECIES: cell wall hydrolase                    \\\\\n",
       "\t 227 & cell wall hydrolase                                  \\\\\n",
       "\t 237 & MULTISPECIES: 30S ribosomal protein S11              \\\\\n",
       "\t 237 & 30S ribosomal protein S11                            \\\\\n",
       "\t 240 & elongation factor G                                  \\\\\n",
       "\t 241 & MULTISPECIES: 50S ribosomal protein L19              \\\\\n",
       "\t 241 & 50S ribosomal protein L19                            \\\\\n",
       "\t 243 & MULTISPECIES: response regulator transcription factor\\\\\n",
       "\t 243 & response regulator transcription factor              \\\\\n",
       "\t 246 & glucose-1-phosphate adenylyltransferase              \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 139 × 2\n",
       "\n",
       "| cluster_id &lt;dbl&gt; | gene_name &lt;chr&gt; |\n",
       "|---|---|\n",
       "|  4 | MULTISPECIES: 50S ribosomal protein L22                 |\n",
       "|  4 | 50S ribosomal protein L22                               |\n",
       "|  5 | MULTISPECIES: type I DNA topoisomerase                  |\n",
       "|  5 | type I DNA topoisomerase                                |\n",
       "| 10 | Catabolite repression HPr                               |\n",
       "| 10 | HPr family phosphocarrier protein                       |\n",
       "| 10 | MULTISPECIES: HPr family phosphocarrier protein         |\n",
       "| 11 | 50S ribosomal protein L30                               |\n",
       "| 11 | MULTISPECIES: 50S ribosomal protein L30                 |\n",
       "| 13 | MULTISPECIES: 30S ribosomal protein S12                 |\n",
       "| 13 | 30S ribosomal protein S12                               |\n",
       "| 19 | tRNA (guanosine(37)-N1)-methyltransferase TrmD          |\n",
       "| 19 | tRNA (guanine-N(1)-)-methyltransferase                  |\n",
       "| 20 | MULTISPECIES: 30S ribosomal protein S3                  |\n",
       "| 20 | 30S ribosomal protein S3                                |\n",
       "| 22 | MULTISPECIES: DNA-directed RNA polymerase subunit beta' |\n",
       "| 22 | DNA-directed RNA polymerase subunit beta'               |\n",
       "| 24 | MULTISPECIES: glycosyltransferase family 2 protein      |\n",
       "| 33 | MULTISPECIES: 50S ribosomal protein L11                 |\n",
       "| 33 | 50S ribosomal protein L11                               |\n",
       "| 34 | MULTISPECIES: 30S ribosomal protein S7                  |\n",
       "| 34 | 30S ribosomal protein S7                                |\n",
       "| 35 | 30S ribosomal protein S13                               |\n",
       "| 35 | MULTISPECIES: 30S ribosomal protein S13                 |\n",
       "| 37 | DNA-directed RNA polymerase subunit alpha               |\n",
       "| ⋮ | ⋮ |\n",
       "| 200 | MULTISPECIES: UMP kinase                              |\n",
       "| 200 | UMP kinase                                            |\n",
       "| 201 | NADH-quinone oxidoreductase subunit NuoF              |\n",
       "| 204 | preprotein translocase subunit SecA                   |\n",
       "| 206 | MULTISPECIES: translation initiation factor IF-1      |\n",
       "| 206 | translation initiation factor IF-1                    |\n",
       "| 208 | MULTISPECIES: uracil phosphoribosyltransferase        |\n",
       "| 208 | uracil phosphoribosyltransferase                      |\n",
       "| 214 | chaperonin GroS                                       |\n",
       "| 214 | co-chaperone GroES                                    |\n",
       "| 214 | MULTISPECIES: co-chaperone GroES                      |\n",
       "| 220 | phosphoenolpyruvate carboxykinase (GTP)               |\n",
       "| 220 | MULTISPECIES: phosphoenolpyruvate carboxykinase (GTP) |\n",
       "| 221 | phenylacetate--CoA ligase                             |\n",
       "| 221 | MULTISPECIES: phenylacetate--CoA ligase               |\n",
       "| 227 | MULTISPECIES: cell wall hydrolase                     |\n",
       "| 227 | cell wall hydrolase                                   |\n",
       "| 237 | MULTISPECIES: 30S ribosomal protein S11               |\n",
       "| 237 | 30S ribosomal protein S11                             |\n",
       "| 240 | elongation factor G                                   |\n",
       "| 241 | MULTISPECIES: 50S ribosomal protein L19               |\n",
       "| 241 | 50S ribosomal protein L19                             |\n",
       "| 243 | MULTISPECIES: response regulator transcription factor |\n",
       "| 243 | response regulator transcription factor               |\n",
       "| 246 | glucose-1-phosphate adenylyltransferase               |\n",
       "\n"
      ],
      "text/plain": [
       "    cluster_id gene_name                                              \n",
       "1    4         MULTISPECIES: 50S ribosomal protein L22                \n",
       "2    4         50S ribosomal protein L22                              \n",
       "3    5         MULTISPECIES: type I DNA topoisomerase                 \n",
       "4    5         type I DNA topoisomerase                               \n",
       "5   10         Catabolite repression HPr                              \n",
       "6   10         HPr family phosphocarrier protein                      \n",
       "7   10         MULTISPECIES: HPr family phosphocarrier protein        \n",
       "8   11         50S ribosomal protein L30                              \n",
       "9   11         MULTISPECIES: 50S ribosomal protein L30                \n",
       "10  13         MULTISPECIES: 30S ribosomal protein S12                \n",
       "11  13         30S ribosomal protein S12                              \n",
       "12  19         tRNA (guanosine(37)-N1)-methyltransferase TrmD         \n",
       "13  19         tRNA (guanine-N(1)-)-methyltransferase                 \n",
       "14  20         MULTISPECIES: 30S ribosomal protein S3                 \n",
       "15  20         30S ribosomal protein S3                               \n",
       "16  22         MULTISPECIES: DNA-directed RNA polymerase subunit beta'\n",
       "17  22         DNA-directed RNA polymerase subunit beta'              \n",
       "18  24         MULTISPECIES: glycosyltransferase family 2 protein     \n",
       "19  33         MULTISPECIES: 50S ribosomal protein L11                \n",
       "20  33         50S ribosomal protein L11                              \n",
       "21  34         MULTISPECIES: 30S ribosomal protein S7                 \n",
       "22  34         30S ribosomal protein S7                               \n",
       "23  35         30S ribosomal protein S13                              \n",
       "24  35         MULTISPECIES: 30S ribosomal protein S13                \n",
       "25  37         DNA-directed RNA polymerase subunit alpha              \n",
       "⋮   ⋮          ⋮                                                      \n",
       "115 200        MULTISPECIES: UMP kinase                               \n",
       "116 200        UMP kinase                                             \n",
       "117 201        NADH-quinone oxidoreductase subunit NuoF               \n",
       "118 204        preprotein translocase subunit SecA                    \n",
       "119 206        MULTISPECIES: translation initiation factor IF-1       \n",
       "120 206        translation initiation factor IF-1                     \n",
       "121 208        MULTISPECIES: uracil phosphoribosyltransferase         \n",
       "122 208        uracil phosphoribosyltransferase                       \n",
       "123 214        chaperonin GroS                                        \n",
       "124 214        co-chaperone GroES                                     \n",
       "125 214        MULTISPECIES: co-chaperone GroES                       \n",
       "126 220        phosphoenolpyruvate carboxykinase (GTP)                \n",
       "127 220        MULTISPECIES: phosphoenolpyruvate carboxykinase (GTP)  \n",
       "128 221        phenylacetate--CoA ligase                              \n",
       "129 221        MULTISPECIES: phenylacetate--CoA ligase                \n",
       "130 227        MULTISPECIES: cell wall hydrolase                      \n",
       "131 227        cell wall hydrolase                                    \n",
       "132 237        MULTISPECIES: 30S ribosomal protein S11                \n",
       "133 237        30S ribosomal protein S11                              \n",
       "134 240        elongation factor G                                    \n",
       "135 241        MULTISPECIES: 50S ribosomal protein L19                \n",
       "136 241        50S ribosomal protein L19                              \n",
       "137 243        MULTISPECIES: response regulator transcription factor  \n",
       "138 243        response regulator transcription factor                \n",
       "139 246        glucose-1-phosphate adenylyltransferase                "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df.dims(50)\n",
    "gene_annot %>%\n",
    "    distinct(cluster_id, gene_name) \n",
    "df.dims()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-17T07:39:23.661788Z",
     "iopub.status.busy": "2021-01-17T07:39:23.660044Z",
     "iopub.status.idle": "2021-01-17T07:39:23.714572Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 166 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>cluster_id</th><th scope=col>gene_taxonomy</th></tr>\n",
       "\t<tr><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td> 4</td><td>Anaerotruncus                      </td></tr>\n",
       "\t<tr><td> 4</td><td>Anaerotruncus rubiinfantis         </td></tr>\n",
       "\t<tr><td> 4</td><td>unclassified Anaerotruncus         </td></tr>\n",
       "\t<tr><td> 5</td><td>Anaerotruncus                      </td></tr>\n",
       "\t<tr><td> 5</td><td>Anaerotruncus rubiinfantis         </td></tr>\n",
       "\t<tr><td> 5</td><td>Anaerotruncus sp. G3(2012)         </td></tr>\n",
       "\t<tr><td>10</td><td>Anaerotruncus colihominis          </td></tr>\n",
       "\t<tr><td>10</td><td>Anaerotruncus                      </td></tr>\n",
       "\t<tr><td>11</td><td>Anaerotruncus colihominis          </td></tr>\n",
       "\t<tr><td>11</td><td>unclassified Anaerotruncus         </td></tr>\n",
       "\t<tr><td>13</td><td>Anaerotruncus                      </td></tr>\n",
       "\t<tr><td>13</td><td>Anaerotruncus colihominis          </td></tr>\n",
       "\t<tr><td>19</td><td>Anaerotruncus colihominis          </td></tr>\n",
       "\t<tr><td>19</td><td>Anaerotruncus colihominis DSM 17241</td></tr>\n",
       "\t<tr><td>20</td><td>Anaerotruncus                      </td></tr>\n",
       "\t<tr><td>20</td><td>Anaerotruncus colihominis          </td></tr>\n",
       "\t<tr><td>22</td><td>unclassified Anaerotruncus         </td></tr>\n",
       "\t<tr><td>22</td><td>Marasmitruncus massiliensis        </td></tr>\n",
       "\t<tr><td>22</td><td>Anaerotruncus colihominis          </td></tr>\n",
       "\t<tr><td>24</td><td>Anaerotruncus                      </td></tr>\n",
       "\t<tr><td>24</td><td>unclassified Anaerotruncus         </td></tr>\n",
       "\t<tr><td>33</td><td>Anaerotruncus                      </td></tr>\n",
       "\t<tr><td>33</td><td>Anaerotruncus colihominis          </td></tr>\n",
       "\t<tr><td>34</td><td>Anaerotruncus                      </td></tr>\n",
       "\t<tr><td>34</td><td>Anaerotruncus rubiinfantis         </td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>206</td><td>Anaerotruncus                         </td></tr>\n",
       "\t<tr><td>206</td><td>Marasmitruncus massiliensis           </td></tr>\n",
       "\t<tr><td>208</td><td>Anaerotruncus                         </td></tr>\n",
       "\t<tr><td>208</td><td>Anaerotruncus rubiinfantis            </td></tr>\n",
       "\t<tr><td>208</td><td>Anaerotruncus sp. G3(2012)            </td></tr>\n",
       "\t<tr><td>214</td><td>Anaerotruncus colihominis DSM 17241   </td></tr>\n",
       "\t<tr><td>214</td><td>Anaerotruncus colihominis             </td></tr>\n",
       "\t<tr><td>214</td><td>unclassified Anaerotruncus            </td></tr>\n",
       "\t<tr><td>220</td><td>Anaerotruncus rubiinfantis            </td></tr>\n",
       "\t<tr><td>220</td><td>Anaerotruncus sp. AF02-27             </td></tr>\n",
       "\t<tr><td>220</td><td>Anaerotruncus                         </td></tr>\n",
       "\t<tr><td>221</td><td>Anaerotruncus sp. G3(2012)            </td></tr>\n",
       "\t<tr><td>221</td><td>unclassified Anaerotruncus            </td></tr>\n",
       "\t<tr><td>221</td><td>Anaerotruncus                         </td></tr>\n",
       "\t<tr><td>227</td><td>unclassified Anaerotruncus            </td></tr>\n",
       "\t<tr><td>227</td><td>Anaerotruncus                         </td></tr>\n",
       "\t<tr><td>227</td><td>Anaerotruncus colihominis             </td></tr>\n",
       "\t<tr><td>237</td><td>Anaerotruncus                         </td></tr>\n",
       "\t<tr><td>237</td><td>Candidatus Pararuminococcus gallinarum</td></tr>\n",
       "\t<tr><td>240</td><td>Anaerotruncus colihominis             </td></tr>\n",
       "\t<tr><td>241</td><td>Anaerotruncus                         </td></tr>\n",
       "\t<tr><td>241</td><td>Anaerotruncus colihominis             </td></tr>\n",
       "\t<tr><td>243</td><td>Anaerotruncus                         </td></tr>\n",
       "\t<tr><td>243</td><td>Marasmitruncus massiliensis           </td></tr>\n",
       "\t<tr><td>246</td><td>Anaerotruncus colihominis             </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 166 × 2\n",
       "\\begin{tabular}{ll}\n",
       " cluster\\_id & gene\\_taxonomy\\\\\n",
       " <dbl> & <chr>\\\\\n",
       "\\hline\n",
       "\t  4 & Anaerotruncus                      \\\\\n",
       "\t  4 & Anaerotruncus rubiinfantis         \\\\\n",
       "\t  4 & unclassified Anaerotruncus         \\\\\n",
       "\t  5 & Anaerotruncus                      \\\\\n",
       "\t  5 & Anaerotruncus rubiinfantis         \\\\\n",
       "\t  5 & Anaerotruncus sp. G3(2012)         \\\\\n",
       "\t 10 & Anaerotruncus colihominis          \\\\\n",
       "\t 10 & Anaerotruncus                      \\\\\n",
       "\t 11 & Anaerotruncus colihominis          \\\\\n",
       "\t 11 & unclassified Anaerotruncus         \\\\\n",
       "\t 13 & Anaerotruncus                      \\\\\n",
       "\t 13 & Anaerotruncus colihominis          \\\\\n",
       "\t 19 & Anaerotruncus colihominis          \\\\\n",
       "\t 19 & Anaerotruncus colihominis DSM 17241\\\\\n",
       "\t 20 & Anaerotruncus                      \\\\\n",
       "\t 20 & Anaerotruncus colihominis          \\\\\n",
       "\t 22 & unclassified Anaerotruncus         \\\\\n",
       "\t 22 & Marasmitruncus massiliensis        \\\\\n",
       "\t 22 & Anaerotruncus colihominis          \\\\\n",
       "\t 24 & Anaerotruncus                      \\\\\n",
       "\t 24 & unclassified Anaerotruncus         \\\\\n",
       "\t 33 & Anaerotruncus                      \\\\\n",
       "\t 33 & Anaerotruncus colihominis          \\\\\n",
       "\t 34 & Anaerotruncus                      \\\\\n",
       "\t 34 & Anaerotruncus rubiinfantis         \\\\\n",
       "\t ⋮ & ⋮\\\\\n",
       "\t 206 & Anaerotruncus                         \\\\\n",
       "\t 206 & Marasmitruncus massiliensis           \\\\\n",
       "\t 208 & Anaerotruncus                         \\\\\n",
       "\t 208 & Anaerotruncus rubiinfantis            \\\\\n",
       "\t 208 & Anaerotruncus sp. G3(2012)            \\\\\n",
       "\t 214 & Anaerotruncus colihominis DSM 17241   \\\\\n",
       "\t 214 & Anaerotruncus colihominis             \\\\\n",
       "\t 214 & unclassified Anaerotruncus            \\\\\n",
       "\t 220 & Anaerotruncus rubiinfantis            \\\\\n",
       "\t 220 & Anaerotruncus sp. AF02-27             \\\\\n",
       "\t 220 & Anaerotruncus                         \\\\\n",
       "\t 221 & Anaerotruncus sp. G3(2012)            \\\\\n",
       "\t 221 & unclassified Anaerotruncus            \\\\\n",
       "\t 221 & Anaerotruncus                         \\\\\n",
       "\t 227 & unclassified Anaerotruncus            \\\\\n",
       "\t 227 & Anaerotruncus                         \\\\\n",
       "\t 227 & Anaerotruncus colihominis             \\\\\n",
       "\t 237 & Anaerotruncus                         \\\\\n",
       "\t 237 & Candidatus Pararuminococcus gallinarum\\\\\n",
       "\t 240 & Anaerotruncus colihominis             \\\\\n",
       "\t 241 & Anaerotruncus                         \\\\\n",
       "\t 241 & Anaerotruncus colihominis             \\\\\n",
       "\t 243 & Anaerotruncus                         \\\\\n",
       "\t 243 & Marasmitruncus massiliensis           \\\\\n",
       "\t 246 & Anaerotruncus colihominis             \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 166 × 2\n",
       "\n",
       "| cluster_id &lt;dbl&gt; | gene_taxonomy &lt;chr&gt; |\n",
       "|---|---|\n",
       "|  4 | Anaerotruncus                       |\n",
       "|  4 | Anaerotruncus rubiinfantis          |\n",
       "|  4 | unclassified Anaerotruncus          |\n",
       "|  5 | Anaerotruncus                       |\n",
       "|  5 | Anaerotruncus rubiinfantis          |\n",
       "|  5 | Anaerotruncus sp. G3(2012)          |\n",
       "| 10 | Anaerotruncus colihominis           |\n",
       "| 10 | Anaerotruncus                       |\n",
       "| 11 | Anaerotruncus colihominis           |\n",
       "| 11 | unclassified Anaerotruncus          |\n",
       "| 13 | Anaerotruncus                       |\n",
       "| 13 | Anaerotruncus colihominis           |\n",
       "| 19 | Anaerotruncus colihominis           |\n",
       "| 19 | Anaerotruncus colihominis DSM 17241 |\n",
       "| 20 | Anaerotruncus                       |\n",
       "| 20 | Anaerotruncus colihominis           |\n",
       "| 22 | unclassified Anaerotruncus          |\n",
       "| 22 | Marasmitruncus massiliensis         |\n",
       "| 22 | Anaerotruncus colihominis           |\n",
       "| 24 | Anaerotruncus                       |\n",
       "| 24 | unclassified Anaerotruncus          |\n",
       "| 33 | Anaerotruncus                       |\n",
       "| 33 | Anaerotruncus colihominis           |\n",
       "| 34 | Anaerotruncus                       |\n",
       "| 34 | Anaerotruncus rubiinfantis          |\n",
       "| ⋮ | ⋮ |\n",
       "| 206 | Anaerotruncus                          |\n",
       "| 206 | Marasmitruncus massiliensis            |\n",
       "| 208 | Anaerotruncus                          |\n",
       "| 208 | Anaerotruncus rubiinfantis             |\n",
       "| 208 | Anaerotruncus sp. G3(2012)             |\n",
       "| 214 | Anaerotruncus colihominis DSM 17241    |\n",
       "| 214 | Anaerotruncus colihominis              |\n",
       "| 214 | unclassified Anaerotruncus             |\n",
       "| 220 | Anaerotruncus rubiinfantis             |\n",
       "| 220 | Anaerotruncus sp. AF02-27              |\n",
       "| 220 | Anaerotruncus                          |\n",
       "| 221 | Anaerotruncus sp. G3(2012)             |\n",
       "| 221 | unclassified Anaerotruncus             |\n",
       "| 221 | Anaerotruncus                          |\n",
       "| 227 | unclassified Anaerotruncus             |\n",
       "| 227 | Anaerotruncus                          |\n",
       "| 227 | Anaerotruncus colihominis              |\n",
       "| 237 | Anaerotruncus                          |\n",
       "| 237 | Candidatus Pararuminococcus gallinarum |\n",
       "| 240 | Anaerotruncus colihominis              |\n",
       "| 241 | Anaerotruncus                          |\n",
       "| 241 | Anaerotruncus colihominis              |\n",
       "| 243 | Anaerotruncus                          |\n",
       "| 243 | Marasmitruncus massiliensis            |\n",
       "| 246 | Anaerotruncus colihominis              |\n",
       "\n"
      ],
      "text/plain": [
       "    cluster_id gene_taxonomy                         \n",
       "1    4         Anaerotruncus                         \n",
       "2    4         Anaerotruncus rubiinfantis            \n",
       "3    4         unclassified Anaerotruncus            \n",
       "4    5         Anaerotruncus                         \n",
       "5    5         Anaerotruncus rubiinfantis            \n",
       "6    5         Anaerotruncus sp. G3(2012)            \n",
       "7   10         Anaerotruncus colihominis             \n",
       "8   10         Anaerotruncus                         \n",
       "9   11         Anaerotruncus colihominis             \n",
       "10  11         unclassified Anaerotruncus            \n",
       "11  13         Anaerotruncus                         \n",
       "12  13         Anaerotruncus colihominis             \n",
       "13  19         Anaerotruncus colihominis             \n",
       "14  19         Anaerotruncus colihominis DSM 17241   \n",
       "15  20         Anaerotruncus                         \n",
       "16  20         Anaerotruncus colihominis             \n",
       "17  22         unclassified Anaerotruncus            \n",
       "18  22         Marasmitruncus massiliensis           \n",
       "19  22         Anaerotruncus colihominis             \n",
       "20  24         Anaerotruncus                         \n",
       "21  24         unclassified Anaerotruncus            \n",
       "22  33         Anaerotruncus                         \n",
       "23  33         Anaerotruncus colihominis             \n",
       "24  34         Anaerotruncus                         \n",
       "25  34         Anaerotruncus rubiinfantis            \n",
       "⋮   ⋮          ⋮                                     \n",
       "142 206        Anaerotruncus                         \n",
       "143 206        Marasmitruncus massiliensis           \n",
       "144 208        Anaerotruncus                         \n",
       "145 208        Anaerotruncus rubiinfantis            \n",
       "146 208        Anaerotruncus sp. G3(2012)            \n",
       "147 214        Anaerotruncus colihominis DSM 17241   \n",
       "148 214        Anaerotruncus colihominis             \n",
       "149 214        unclassified Anaerotruncus            \n",
       "150 220        Anaerotruncus rubiinfantis            \n",
       "151 220        Anaerotruncus sp. AF02-27             \n",
       "152 220        Anaerotruncus                         \n",
       "153 221        Anaerotruncus sp. G3(2012)            \n",
       "154 221        unclassified Anaerotruncus            \n",
       "155 221        Anaerotruncus                         \n",
       "156 227        unclassified Anaerotruncus            \n",
       "157 227        Anaerotruncus                         \n",
       "158 227        Anaerotruncus colihominis             \n",
       "159 237        Anaerotruncus                         \n",
       "160 237        Candidatus Pararuminococcus gallinarum\n",
       "161 240        Anaerotruncus colihominis             \n",
       "162 241        Anaerotruncus                         \n",
       "163 241        Anaerotruncus colihominis             \n",
       "164 243        Anaerotruncus                         \n",
       "165 243        Marasmitruncus massiliensis           \n",
       "166 246        Anaerotruncus colihominis             "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df.dims(50)\n",
    "gene_annot %>%\n",
    "    distinct(cluster_id, gene_taxonomy) \n",
    "df.dims()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Gene cluster: closest related"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-17T07:39:50.045492Z",
     "iopub.status.busy": "2021-01-17T07:39:50.043405Z",
     "iopub.status.idle": "2021-01-17T07:39:50.306080Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 2280 × 19</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>cluster_id</th><th scope=col>query</th><th scope=col>subject</th><th scope=col>subject_name</th><th scope=col>pident</th><th scope=col>length</th><th scope=col>mismatch</th><th scope=col>qstart</th><th scope=col>qend</th><th scope=col>sstart</th><th scope=col>send</th><th scope=col>evalue</th><th scope=col>slen</th><th scope=col>qlen</th><th scope=col>sscinames</th><th scope=col>staxids</th><th scope=col>pident_rank</th><th scope=col>gene_name</th><th scope=col>gene_taxonomy</th></tr>\n",
       "\t<tr><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>4</td><td>abf8b17f07404c7183f7f0f8c044a48f</td><td>WP_101908516.1</td><td>50S ribosomal protein L22 [Marasmitruncus massiliensis]      </td><td>91.892</td><td>111</td><td> 9</td><td>1</td><td>333</td><td>1</td><td>111</td><td>1.45e-70</td><td>111</td><td>336</td><td>Ruminococcaceae bacterium;Marasmitruncus massiliensis</td><td>1898205;1944642</td><td>1</td><td>50S ribosomal protein L22</td><td>Marasmitruncus massiliensis      </td></tr>\n",
       "\t<tr><td>4</td><td>abf8b17f07404c7183f7f0f8c044a48f</td><td>MBC8585943.1  </td><td>50S ribosomal protein L22 [Oscillospiraceae bacterium NSJ-64]</td><td>90.090</td><td>111</td><td>11</td><td>1</td><td>333</td><td>1</td><td>111</td><td>1.60e-69</td><td>111</td><td>336</td><td>Oscillospiraceae bacterium NSJ-64                    </td><td>2763678        </td><td>2</td><td>50S ribosomal protein L22</td><td>Oscillospiraceae bacterium NSJ-64</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>246</td><td>ce2e4df7569647278c025e0d6334cd91</td><td>WP_090266788.1</td><td>glucose-1-phosphate adenylyltransferase [Anaeromassilibacillus sp. Marseille-P4683]</td><td>73.580</td><td>405</td><td>107</td><td> 1</td><td>1215</td><td>1</td><td>405</td><td>0</td><td>405</td><td>1263</td><td>Eubacteriaceae bacterium CHKCI005;Anaeromassilibacillus sp. Marseille-P4683</td><td>1780381;2041843</td><td>28</td><td>glucose-1-phosphate adenylyltransferase</td><td>Anaeromassilibacillus sp. Marseille-P4683</td></tr>\n",
       "\t<tr><td>246</td><td>ce2e4df7569647278c025e0d6334cd91</td><td>MBE6902255.1  </td><td>glucose-1-phosphate adenylyltransferase [Ruminococcaceae bacterium]                </td><td>73.048</td><td>397</td><td>107</td><td>10</td><td>1200</td><td>5</td><td>401</td><td>0</td><td>401</td><td>1263</td><td>Ruminococcaceae bacterium                                                  </td><td>1898205        </td><td>30</td><td>glucose-1-phosphate adenylyltransferase</td><td>Ruminococcaceae bacterium                </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2280 × 19\n",
       "\\begin{tabular}{lllllllllllllllllll}\n",
       " cluster\\_id & query & subject & subject\\_name & pident & length & mismatch & qstart & qend & sstart & send & evalue & slen & qlen & sscinames & staxids & pident\\_rank & gene\\_name & gene\\_taxonomy\\\\\n",
       " <dbl> & <fct> & <fct> & <fct> & <dbl> & <int> & <int> & <int> & <int> & <int> & <int> & <dbl> & <int> & <int> & <fct> & <fct> & <int> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t 4 & abf8b17f07404c7183f7f0f8c044a48f & WP\\_101908516.1 & 50S ribosomal protein L22 {[}Marasmitruncus massiliensis{]}       & 91.892 & 111 &  9 & 1 & 333 & 1 & 111 & 1.45e-70 & 111 & 336 & Ruminococcaceae bacterium;Marasmitruncus massiliensis & 1898205;1944642 & 1 & 50S ribosomal protein L22 & Marasmitruncus massiliensis      \\\\\n",
       "\t 4 & abf8b17f07404c7183f7f0f8c044a48f & MBC8585943.1   & 50S ribosomal protein L22 {[}Oscillospiraceae bacterium NSJ-64{]} & 90.090 & 111 & 11 & 1 & 333 & 1 & 111 & 1.60e-69 & 111 & 336 & Oscillospiraceae bacterium NSJ-64                     & 2763678         & 2 & 50S ribosomal protein L22 & Oscillospiraceae bacterium NSJ-64\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t 246 & ce2e4df7569647278c025e0d6334cd91 & WP\\_090266788.1 & glucose-1-phosphate adenylyltransferase {[}Anaeromassilibacillus sp. Marseille-P4683{]} & 73.580 & 405 & 107 &  1 & 1215 & 1 & 405 & 0 & 405 & 1263 & Eubacteriaceae bacterium CHKCI005;Anaeromassilibacillus sp. Marseille-P4683 & 1780381;2041843 & 28 & glucose-1-phosphate adenylyltransferase & Anaeromassilibacillus sp. Marseille-P4683\\\\\n",
       "\t 246 & ce2e4df7569647278c025e0d6334cd91 & MBE6902255.1   & glucose-1-phosphate adenylyltransferase {[}Ruminococcaceae bacterium{]}                 & 73.048 & 397 & 107 & 10 & 1200 & 5 & 401 & 0 & 401 & 1263 & Ruminococcaceae bacterium                                                   & 1898205         & 30 & glucose-1-phosphate adenylyltransferase & Ruminococcaceae bacterium                \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2280 × 19\n",
       "\n",
       "| cluster_id &lt;dbl&gt; | query &lt;fct&gt; | subject &lt;fct&gt; | subject_name &lt;fct&gt; | pident &lt;dbl&gt; | length &lt;int&gt; | mismatch &lt;int&gt; | qstart &lt;int&gt; | qend &lt;int&gt; | sstart &lt;int&gt; | send &lt;int&gt; | evalue &lt;dbl&gt; | slen &lt;int&gt; | qlen &lt;int&gt; | sscinames &lt;fct&gt; | staxids &lt;fct&gt; | pident_rank &lt;int&gt; | gene_name &lt;chr&gt; | gene_taxonomy &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 4 | abf8b17f07404c7183f7f0f8c044a48f | WP_101908516.1 | 50S ribosomal protein L22 [Marasmitruncus massiliensis]       | 91.892 | 111 |  9 | 1 | 333 | 1 | 111 | 1.45e-70 | 111 | 336 | Ruminococcaceae bacterium;Marasmitruncus massiliensis | 1898205;1944642 | 1 | 50S ribosomal protein L22 | Marasmitruncus massiliensis       |\n",
       "| 4 | abf8b17f07404c7183f7f0f8c044a48f | MBC8585943.1   | 50S ribosomal protein L22 [Oscillospiraceae bacterium NSJ-64] | 90.090 | 111 | 11 | 1 | 333 | 1 | 111 | 1.60e-69 | 111 | 336 | Oscillospiraceae bacterium NSJ-64                     | 2763678         | 2 | 50S ribosomal protein L22 | Oscillospiraceae bacterium NSJ-64 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| 246 | ce2e4df7569647278c025e0d6334cd91 | WP_090266788.1 | glucose-1-phosphate adenylyltransferase [Anaeromassilibacillus sp. Marseille-P4683] | 73.580 | 405 | 107 |  1 | 1215 | 1 | 405 | 0 | 405 | 1263 | Eubacteriaceae bacterium CHKCI005;Anaeromassilibacillus sp. Marseille-P4683 | 1780381;2041843 | 28 | glucose-1-phosphate adenylyltransferase | Anaeromassilibacillus sp. Marseille-P4683 |\n",
       "| 246 | ce2e4df7569647278c025e0d6334cd91 | MBE6902255.1   | glucose-1-phosphate adenylyltransferase [Ruminococcaceae bacterium]                 | 73.048 | 397 | 107 | 10 | 1200 | 5 | 401 | 0 | 401 | 1263 | Ruminococcaceae bacterium                                                   | 1898205         | 30 | glucose-1-phosphate adenylyltransferase | Ruminococcaceae bacterium                 |\n",
       "\n"
      ],
      "text/plain": [
       "     cluster_id query                            subject       \n",
       "1    4          abf8b17f07404c7183f7f0f8c044a48f WP_101908516.1\n",
       "2    4          abf8b17f07404c7183f7f0f8c044a48f MBC8585943.1  \n",
       "⋮    ⋮          ⋮                                ⋮             \n",
       "2279 246        ce2e4df7569647278c025e0d6334cd91 WP_090266788.1\n",
       "2280 246        ce2e4df7569647278c025e0d6334cd91 MBE6902255.1  \n",
       "     subject_name                                                                       \n",
       "1    50S ribosomal protein L22 [Marasmitruncus massiliensis]                            \n",
       "2    50S ribosomal protein L22 [Oscillospiraceae bacterium NSJ-64]                      \n",
       "⋮    ⋮                                                                                  \n",
       "2279 glucose-1-phosphate adenylyltransferase [Anaeromassilibacillus sp. Marseille-P4683]\n",
       "2280 glucose-1-phosphate adenylyltransferase [Ruminococcaceae bacterium]                \n",
       "     pident length mismatch qstart qend sstart send evalue   slen qlen\n",
       "1    91.892 111     9       1      333  1      111  1.45e-70 111  336 \n",
       "2    90.090 111    11       1      333  1      111  1.60e-69 111  336 \n",
       "⋮    ⋮      ⋮      ⋮        ⋮      ⋮    ⋮      ⋮    ⋮        ⋮    ⋮   \n",
       "2279 73.580 405    107       1     1215 1      405  0        405  1263\n",
       "2280 73.048 397    107      10     1200 5      401  0        401  1263\n",
       "     sscinames                                                                  \n",
       "1    Ruminococcaceae bacterium;Marasmitruncus massiliensis                      \n",
       "2    Oscillospiraceae bacterium NSJ-64                                          \n",
       "⋮    ⋮                                                                          \n",
       "2279 Eubacteriaceae bacterium CHKCI005;Anaeromassilibacillus sp. Marseille-P4683\n",
       "2280 Ruminococcaceae bacterium                                                  \n",
       "     staxids         pident_rank gene_name                              \n",
       "1    1898205;1944642 1           50S ribosomal protein L22              \n",
       "2    2763678         2           50S ribosomal protein L22              \n",
       "⋮    ⋮               ⋮           ⋮                                      \n",
       "2279 1780381;2041843 28          glucose-1-phosphate adenylyltransferase\n",
       "2280 1898205         30          glucose-1-phosphate adenylyltransferase\n",
       "     gene_taxonomy                            \n",
       "1    Marasmitruncus massiliensis              \n",
       "2    Oscillospiraceae bacterium NSJ-64        \n",
       "⋮    ⋮                                        \n",
       "2279 Anaeromassilibacillus sp. Marseille-P4683\n",
       "2280 Ruminococcaceae bacterium                "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "gene_annot = read.delim(file.path(work_dir, 'primers', 'cgp', 'core_clusters_blastx_nontarget.tsv'), \n",
    "                        sep='\\t') %>%\n",
    "    mutate(cluster_id = gsub('cluster_', '', cluster_id) %>% as.Num) %>%\n",
    "    semi_join(primer_info, c('cluster_id')) %>%\n",
    "    mutate(gene_name = gsub(' \\\\[.+', '', subject_name),\n",
    "           gene_taxonomy = gsub('.+\\\\[', '', subject_name),\n",
    "           gene_taxonomy = gsub('\\\\]', '', gene_taxonomy))\n",
    "gene_annot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-17T07:39:58.120882Z",
     "iopub.status.busy": "2021-01-17T07:39:58.118914Z",
     "iopub.status.idle": "2021-01-17T07:39:58.199238Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 226 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>cluster_id</th><th scope=col>gene_name</th><th scope=col>gene_taxonomy</th><th scope=col>pident</th></tr>\n",
       "\t<tr><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td> 4</td><td>50S ribosomal protein L22                     </td><td>Marasmitruncus massiliensis            </td><td>91.892</td></tr>\n",
       "\t<tr><td> 4</td><td>50S ribosomal protein L22                     </td><td>Oscillospiraceae bacterium NSJ-64      </td><td>90.090</td></tr>\n",
       "\t<tr><td> 4</td><td>50S ribosomal protein L22                     </td><td>bacterium 1XD42-8                      </td><td>89.189</td></tr>\n",
       "\t<tr><td> 5</td><td>type I DNA topoisomerase                      </td><td>Ruminococcaceae bacterium              </td><td>84.173</td></tr>\n",
       "\t<tr><td> 5</td><td>type I DNA topoisomerase                      </td><td>Marasmitruncus massiliensis            </td><td>84.029</td></tr>\n",
       "\t<tr><td> 5</td><td>type I DNA topoisomerase                      </td><td>Oscillospiraceae bacterium NSJ-64      </td><td>81.727</td></tr>\n",
       "\t<tr><td>10</td><td>HPr family phosphocarrier protein             </td><td>Marasmitruncus massiliensis            </td><td>97.701</td></tr>\n",
       "\t<tr><td>10</td><td>HPr family phosphocarrier protein             </td><td>bacterium 1XD42-8                      </td><td>97.701</td></tr>\n",
       "\t<tr><td>10</td><td>HPr family phosphocarrier protein             </td><td>Hydrogenoanaerobacterium saccharovorans</td><td>95.402</td></tr>\n",
       "\t<tr><td>11</td><td>50S ribosomal protein L30                     </td><td>Oscillospiraceae bacterium NSJ-64      </td><td>88.136</td></tr>\n",
       "\t<tr><td>11</td><td>50S ribosomal protein L30                     </td><td>Marasmitruncus massiliensis            </td><td>88.136</td></tr>\n",
       "\t<tr><td>11</td><td>50S ribosomal protein L30                     </td><td>Phocea massiliensis                    </td><td>87.500</td></tr>\n",
       "\t<tr><td>13</td><td>30S ribosomal protein S12                     </td><td>bacterium 1XD42-8                      </td><td>97.656</td></tr>\n",
       "\t<tr><td>13</td><td>30S ribosomal protein S12                     </td><td>Marasmitruncus massiliensis            </td><td>96.094</td></tr>\n",
       "\t<tr><td>13</td><td>30S ribosomal protein S12                     </td><td>Clostridiaceae bacterium NSJ-31        </td><td>94.531</td></tr>\n",
       "\t<tr><td>13</td><td>30S ribosomal protein S12                     </td><td>Angelakisella massiliensis             </td><td>94.531</td></tr>\n",
       "\t<tr><td>19</td><td>tRNA (guanosine(37)-N1)-methyltransferase TrmD</td><td>Ruminococcaceae bacterium              </td><td>86.290</td></tr>\n",
       "\t<tr><td>19</td><td>tRNA (guanosine(37)-N1)-methyltransferase TrmD</td><td>Marasmitruncus massiliensis            </td><td>85.887</td></tr>\n",
       "\t<tr><td>19</td><td>tRNA (guanosine(37)-N1)-methyltransferase TrmD</td><td>bacterium 1XD42-8                      </td><td>80.328</td></tr>\n",
       "\t<tr><td>20</td><td>30S ribosomal protein S3                      </td><td>Marasmitruncus massiliensis            </td><td>90.667</td></tr>\n",
       "\t<tr><td>20</td><td>30S ribosomal protein S3                      </td><td>Oscillospiraceae bacterium NSJ-64      </td><td>84.681</td></tr>\n",
       "\t<tr><td>20</td><td>30S ribosomal protein S3                      </td><td>Phocea massiliensis                    </td><td>82.667</td></tr>\n",
       "\t<tr><td>22</td><td>DNA-directed RNA polymerase subunit beta'     </td><td>Marasmitruncus massiliensis            </td><td>90.886</td></tr>\n",
       "\t<tr><td>22</td><td>DNA-directed RNA polymerase subunit beta'     </td><td>bacterium 1XD42-8                      </td><td>89.228</td></tr>\n",
       "\t<tr><td>22</td><td>DNA-directed RNA polymerase subunit beta'     </td><td>Oscillospiraceae bacterium NSJ-64      </td><td>89.076</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>214</td><td>co-chaperonin GroES                     </td><td>uncultured Flavonifractor sp.         </td><td>89.362</td></tr>\n",
       "\t<tr><td>214</td><td>co-chaperone GroES                      </td><td>bacterium 1XD42-8                     </td><td>89.362</td></tr>\n",
       "\t<tr><td>220</td><td>phosphoenolpyruvate carboxykinase (GTP) </td><td>Ruminococcaceae bacterium             </td><td>87.544</td></tr>\n",
       "\t<tr><td>220</td><td>phosphoenolpyruvate carboxykinase (GTP) </td><td>bacterium 1XD42-8                     </td><td>87.895</td></tr>\n",
       "\t<tr><td>220</td><td>phosphoenolpyruvate carboxykinase (GTP) </td><td>Marasmitruncus massiliensis           </td><td>87.193</td></tr>\n",
       "\t<tr><td>221</td><td>AMP-binding protein                     </td><td>Marasmitruncus massiliensis           </td><td>84.562</td></tr>\n",
       "\t<tr><td>221</td><td>phenylacetate--CoA ligase               </td><td>Ruminococcaceae bacterium             </td><td>84.778</td></tr>\n",
       "\t<tr><td>221</td><td>phenylacetate--CoA ligase family protein</td><td>bacterium 1XD42-8                     </td><td>82.679</td></tr>\n",
       "\t<tr><td>237</td><td>30S ribosomal protein S11               </td><td>Candidatus Pararuminococcus gallinarum</td><td>98.261</td></tr>\n",
       "\t<tr><td>237</td><td>30S ribosomal protein S11               </td><td>Ruminococcaceae bacterium             </td><td>98.261</td></tr>\n",
       "\t<tr><td>237</td><td>30S ribosomal protein S11               </td><td>Clostridiaceae bacterium NSJ-31       </td><td>98.261</td></tr>\n",
       "\t<tr><td>237</td><td>30S ribosomal protein S11               </td><td>bacterium 1XD42-8                     </td><td>99.130</td></tr>\n",
       "\t<tr><td>237</td><td>30S ribosomal protein S11               </td><td>Angelakisella massiliensis            </td><td>98.261</td></tr>\n",
       "\t<tr><td>240</td><td>elongation factor G                     </td><td>Ruminococcaceae bacterium             </td><td>88.399</td></tr>\n",
       "\t<tr><td>240</td><td>elongation factor G                     </td><td>Marasmitruncus massiliensis           </td><td>88.399</td></tr>\n",
       "\t<tr><td>240</td><td>elongation factor G                     </td><td>Oscillospiraceae bacterium NSJ-64     </td><td>84.435</td></tr>\n",
       "\t<tr><td>241</td><td>50S ribosomal protein L19               </td><td>bacterium 1XD42-8                     </td><td>91.228</td></tr>\n",
       "\t<tr><td>241</td><td>50S ribosomal protein L19               </td><td>Marasmitruncus massiliensis           </td><td>89.474</td></tr>\n",
       "\t<tr><td>241</td><td>50S ribosomal protein L19               </td><td>Oscillospiraceae bacterium NSJ-64     </td><td>82.456</td></tr>\n",
       "\t<tr><td>243</td><td>response regulator transcription factor </td><td>Marasmitruncus massiliensis           </td><td>94.348</td></tr>\n",
       "\t<tr><td>243</td><td>response regulator transcription factor </td><td>Oscillospiraceae bacterium NSJ-64     </td><td>90.435</td></tr>\n",
       "\t<tr><td>243</td><td>DNA-binding response regulator          </td><td>bacterium 1XD42-8                     </td><td>90.870</td></tr>\n",
       "\t<tr><td>246</td><td>glucose-1-phosphate adenylyltransferase </td><td>Oscillospiraceae bacterium NSJ-64     </td><td>86.104</td></tr>\n",
       "\t<tr><td>246</td><td>glucose-1-phosphate adenylyltransferase </td><td>Marasmitruncus massiliensis           </td><td>85.396</td></tr>\n",
       "\t<tr><td>246</td><td>glucose-1-phosphate adenylyltransferase </td><td>Ruminococcaceae bacterium             </td><td>85.396</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 226 × 4\n",
       "\\begin{tabular}{llll}\n",
       " cluster\\_id & gene\\_name & gene\\_taxonomy & pident\\\\\n",
       " <dbl> & <chr> & <chr> & <dbl>\\\\\n",
       "\\hline\n",
       "\t  4 & 50S ribosomal protein L22                      & Marasmitruncus massiliensis             & 91.892\\\\\n",
       "\t  4 & 50S ribosomal protein L22                      & Oscillospiraceae bacterium NSJ-64       & 90.090\\\\\n",
       "\t  4 & 50S ribosomal protein L22                      & bacterium 1XD42-8                       & 89.189\\\\\n",
       "\t  5 & type I DNA topoisomerase                       & Ruminococcaceae bacterium               & 84.173\\\\\n",
       "\t  5 & type I DNA topoisomerase                       & Marasmitruncus massiliensis             & 84.029\\\\\n",
       "\t  5 & type I DNA topoisomerase                       & Oscillospiraceae bacterium NSJ-64       & 81.727\\\\\n",
       "\t 10 & HPr family phosphocarrier protein              & Marasmitruncus massiliensis             & 97.701\\\\\n",
       "\t 10 & HPr family phosphocarrier protein              & bacterium 1XD42-8                       & 97.701\\\\\n",
       "\t 10 & HPr family phosphocarrier protein              & Hydrogenoanaerobacterium saccharovorans & 95.402\\\\\n",
       "\t 11 & 50S ribosomal protein L30                      & Oscillospiraceae bacterium NSJ-64       & 88.136\\\\\n",
       "\t 11 & 50S ribosomal protein L30                      & Marasmitruncus massiliensis             & 88.136\\\\\n",
       "\t 11 & 50S ribosomal protein L30                      & Phocea massiliensis                     & 87.500\\\\\n",
       "\t 13 & 30S ribosomal protein S12                      & bacterium 1XD42-8                       & 97.656\\\\\n",
       "\t 13 & 30S ribosomal protein S12                      & Marasmitruncus massiliensis             & 96.094\\\\\n",
       "\t 13 & 30S ribosomal protein S12                      & Clostridiaceae bacterium NSJ-31         & 94.531\\\\\n",
       "\t 13 & 30S ribosomal protein S12                      & Angelakisella massiliensis              & 94.531\\\\\n",
       "\t 19 & tRNA (guanosine(37)-N1)-methyltransferase TrmD & Ruminococcaceae bacterium               & 86.290\\\\\n",
       "\t 19 & tRNA (guanosine(37)-N1)-methyltransferase TrmD & Marasmitruncus massiliensis             & 85.887\\\\\n",
       "\t 19 & tRNA (guanosine(37)-N1)-methyltransferase TrmD & bacterium 1XD42-8                       & 80.328\\\\\n",
       "\t 20 & 30S ribosomal protein S3                       & Marasmitruncus massiliensis             & 90.667\\\\\n",
       "\t 20 & 30S ribosomal protein S3                       & Oscillospiraceae bacterium NSJ-64       & 84.681\\\\\n",
       "\t 20 & 30S ribosomal protein S3                       & Phocea massiliensis                     & 82.667\\\\\n",
       "\t 22 & DNA-directed RNA polymerase subunit beta'      & Marasmitruncus massiliensis             & 90.886\\\\\n",
       "\t 22 & DNA-directed RNA polymerase subunit beta'      & bacterium 1XD42-8                       & 89.228\\\\\n",
       "\t 22 & DNA-directed RNA polymerase subunit beta'      & Oscillospiraceae bacterium NSJ-64       & 89.076\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t 214 & co-chaperonin GroES                      & uncultured Flavonifractor sp.          & 89.362\\\\\n",
       "\t 214 & co-chaperone GroES                       & bacterium 1XD42-8                      & 89.362\\\\\n",
       "\t 220 & phosphoenolpyruvate carboxykinase (GTP)  & Ruminococcaceae bacterium              & 87.544\\\\\n",
       "\t 220 & phosphoenolpyruvate carboxykinase (GTP)  & bacterium 1XD42-8                      & 87.895\\\\\n",
       "\t 220 & phosphoenolpyruvate carboxykinase (GTP)  & Marasmitruncus massiliensis            & 87.193\\\\\n",
       "\t 221 & AMP-binding protein                      & Marasmitruncus massiliensis            & 84.562\\\\\n",
       "\t 221 & phenylacetate--CoA ligase                & Ruminococcaceae bacterium              & 84.778\\\\\n",
       "\t 221 & phenylacetate--CoA ligase family protein & bacterium 1XD42-8                      & 82.679\\\\\n",
       "\t 237 & 30S ribosomal protein S11                & Candidatus Pararuminococcus gallinarum & 98.261\\\\\n",
       "\t 237 & 30S ribosomal protein S11                & Ruminococcaceae bacterium              & 98.261\\\\\n",
       "\t 237 & 30S ribosomal protein S11                & Clostridiaceae bacterium NSJ-31        & 98.261\\\\\n",
       "\t 237 & 30S ribosomal protein S11                & bacterium 1XD42-8                      & 99.130\\\\\n",
       "\t 237 & 30S ribosomal protein S11                & Angelakisella massiliensis             & 98.261\\\\\n",
       "\t 240 & elongation factor G                      & Ruminococcaceae bacterium              & 88.399\\\\\n",
       "\t 240 & elongation factor G                      & Marasmitruncus massiliensis            & 88.399\\\\\n",
       "\t 240 & elongation factor G                      & Oscillospiraceae bacterium NSJ-64      & 84.435\\\\\n",
       "\t 241 & 50S ribosomal protein L19                & bacterium 1XD42-8                      & 91.228\\\\\n",
       "\t 241 & 50S ribosomal protein L19                & Marasmitruncus massiliensis            & 89.474\\\\\n",
       "\t 241 & 50S ribosomal protein L19                & Oscillospiraceae bacterium NSJ-64      & 82.456\\\\\n",
       "\t 243 & response regulator transcription factor  & Marasmitruncus massiliensis            & 94.348\\\\\n",
       "\t 243 & response regulator transcription factor  & Oscillospiraceae bacterium NSJ-64      & 90.435\\\\\n",
       "\t 243 & DNA-binding response regulator           & bacterium 1XD42-8                      & 90.870\\\\\n",
       "\t 246 & glucose-1-phosphate adenylyltransferase  & Oscillospiraceae bacterium NSJ-64      & 86.104\\\\\n",
       "\t 246 & glucose-1-phosphate adenylyltransferase  & Marasmitruncus massiliensis            & 85.396\\\\\n",
       "\t 246 & glucose-1-phosphate adenylyltransferase  & Ruminococcaceae bacterium              & 85.396\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 226 × 4\n",
       "\n",
       "| cluster_id &lt;dbl&gt; | gene_name &lt;chr&gt; | gene_taxonomy &lt;chr&gt; | pident &lt;dbl&gt; |\n",
       "|---|---|---|---|\n",
       "|  4 | 50S ribosomal protein L22                      | Marasmitruncus massiliensis             | 91.892 |\n",
       "|  4 | 50S ribosomal protein L22                      | Oscillospiraceae bacterium NSJ-64       | 90.090 |\n",
       "|  4 | 50S ribosomal protein L22                      | bacterium 1XD42-8                       | 89.189 |\n",
       "|  5 | type I DNA topoisomerase                       | Ruminococcaceae bacterium               | 84.173 |\n",
       "|  5 | type I DNA topoisomerase                       | Marasmitruncus massiliensis             | 84.029 |\n",
       "|  5 | type I DNA topoisomerase                       | Oscillospiraceae bacterium NSJ-64       | 81.727 |\n",
       "| 10 | HPr family phosphocarrier protein              | Marasmitruncus massiliensis             | 97.701 |\n",
       "| 10 | HPr family phosphocarrier protein              | bacterium 1XD42-8                       | 97.701 |\n",
       "| 10 | HPr family phosphocarrier protein              | Hydrogenoanaerobacterium saccharovorans | 95.402 |\n",
       "| 11 | 50S ribosomal protein L30                      | Oscillospiraceae bacterium NSJ-64       | 88.136 |\n",
       "| 11 | 50S ribosomal protein L30                      | Marasmitruncus massiliensis             | 88.136 |\n",
       "| 11 | 50S ribosomal protein L30                      | Phocea massiliensis                     | 87.500 |\n",
       "| 13 | 30S ribosomal protein S12                      | bacterium 1XD42-8                       | 97.656 |\n",
       "| 13 | 30S ribosomal protein S12                      | Marasmitruncus massiliensis             | 96.094 |\n",
       "| 13 | 30S ribosomal protein S12                      | Clostridiaceae bacterium NSJ-31         | 94.531 |\n",
       "| 13 | 30S ribosomal protein S12                      | Angelakisella massiliensis              | 94.531 |\n",
       "| 19 | tRNA (guanosine(37)-N1)-methyltransferase TrmD | Ruminococcaceae bacterium               | 86.290 |\n",
       "| 19 | tRNA (guanosine(37)-N1)-methyltransferase TrmD | Marasmitruncus massiliensis             | 85.887 |\n",
       "| 19 | tRNA (guanosine(37)-N1)-methyltransferase TrmD | bacterium 1XD42-8                       | 80.328 |\n",
       "| 20 | 30S ribosomal protein S3                       | Marasmitruncus massiliensis             | 90.667 |\n",
       "| 20 | 30S ribosomal protein S3                       | Oscillospiraceae bacterium NSJ-64       | 84.681 |\n",
       "| 20 | 30S ribosomal protein S3                       | Phocea massiliensis                     | 82.667 |\n",
       "| 22 | DNA-directed RNA polymerase subunit beta'      | Marasmitruncus massiliensis             | 90.886 |\n",
       "| 22 | DNA-directed RNA polymerase subunit beta'      | bacterium 1XD42-8                       | 89.228 |\n",
       "| 22 | DNA-directed RNA polymerase subunit beta'      | Oscillospiraceae bacterium NSJ-64       | 89.076 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| 214 | co-chaperonin GroES                      | uncultured Flavonifractor sp.          | 89.362 |\n",
       "| 214 | co-chaperone GroES                       | bacterium 1XD42-8                      | 89.362 |\n",
       "| 220 | phosphoenolpyruvate carboxykinase (GTP)  | Ruminococcaceae bacterium              | 87.544 |\n",
       "| 220 | phosphoenolpyruvate carboxykinase (GTP)  | bacterium 1XD42-8                      | 87.895 |\n",
       "| 220 | phosphoenolpyruvate carboxykinase (GTP)  | Marasmitruncus massiliensis            | 87.193 |\n",
       "| 221 | AMP-binding protein                      | Marasmitruncus massiliensis            | 84.562 |\n",
       "| 221 | phenylacetate--CoA ligase                | Ruminococcaceae bacterium              | 84.778 |\n",
       "| 221 | phenylacetate--CoA ligase family protein | bacterium 1XD42-8                      | 82.679 |\n",
       "| 237 | 30S ribosomal protein S11                | Candidatus Pararuminococcus gallinarum | 98.261 |\n",
       "| 237 | 30S ribosomal protein S11                | Ruminococcaceae bacterium              | 98.261 |\n",
       "| 237 | 30S ribosomal protein S11                | Clostridiaceae bacterium NSJ-31        | 98.261 |\n",
       "| 237 | 30S ribosomal protein S11                | bacterium 1XD42-8                      | 99.130 |\n",
       "| 237 | 30S ribosomal protein S11                | Angelakisella massiliensis             | 98.261 |\n",
       "| 240 | elongation factor G                      | Ruminococcaceae bacterium              | 88.399 |\n",
       "| 240 | elongation factor G                      | Marasmitruncus massiliensis            | 88.399 |\n",
       "| 240 | elongation factor G                      | Oscillospiraceae bacterium NSJ-64      | 84.435 |\n",
       "| 241 | 50S ribosomal protein L19                | bacterium 1XD42-8                      | 91.228 |\n",
       "| 241 | 50S ribosomal protein L19                | Marasmitruncus massiliensis            | 89.474 |\n",
       "| 241 | 50S ribosomal protein L19                | Oscillospiraceae bacterium NSJ-64      | 82.456 |\n",
       "| 243 | response regulator transcription factor  | Marasmitruncus massiliensis            | 94.348 |\n",
       "| 243 | response regulator transcription factor  | Oscillospiraceae bacterium NSJ-64      | 90.435 |\n",
       "| 243 | DNA-binding response regulator           | bacterium 1XD42-8                      | 90.870 |\n",
       "| 246 | glucose-1-phosphate adenylyltransferase  | Oscillospiraceae bacterium NSJ-64      | 86.104 |\n",
       "| 246 | glucose-1-phosphate adenylyltransferase  | Marasmitruncus massiliensis            | 85.396 |\n",
       "| 246 | glucose-1-phosphate adenylyltransferase  | Ruminococcaceae bacterium              | 85.396 |\n",
       "\n"
      ],
      "text/plain": [
       "    cluster_id gene_name                                     \n",
       "1    4         50S ribosomal protein L22                     \n",
       "2    4         50S ribosomal protein L22                     \n",
       "3    4         50S ribosomal protein L22                     \n",
       "4    5         type I DNA topoisomerase                      \n",
       "5    5         type I DNA topoisomerase                      \n",
       "6    5         type I DNA topoisomerase                      \n",
       "7   10         HPr family phosphocarrier protein             \n",
       "8   10         HPr family phosphocarrier protein             \n",
       "9   10         HPr family phosphocarrier protein             \n",
       "10  11         50S ribosomal protein L30                     \n",
       "11  11         50S ribosomal protein L30                     \n",
       "12  11         50S ribosomal protein L30                     \n",
       "13  13         30S ribosomal protein S12                     \n",
       "14  13         30S ribosomal protein S12                     \n",
       "15  13         30S ribosomal protein S12                     \n",
       "16  13         30S ribosomal protein S12                     \n",
       "17  19         tRNA (guanosine(37)-N1)-methyltransferase TrmD\n",
       "18  19         tRNA (guanosine(37)-N1)-methyltransferase TrmD\n",
       "19  19         tRNA (guanosine(37)-N1)-methyltransferase TrmD\n",
       "20  20         30S ribosomal protein S3                      \n",
       "21  20         30S ribosomal protein S3                      \n",
       "22  20         30S ribosomal protein S3                      \n",
       "23  22         DNA-directed RNA polymerase subunit beta'     \n",
       "24  22         DNA-directed RNA polymerase subunit beta'     \n",
       "25  22         DNA-directed RNA polymerase subunit beta'     \n",
       "⋮   ⋮          ⋮                                             \n",
       "202 214        co-chaperonin GroES                           \n",
       "203 214        co-chaperone GroES                            \n",
       "204 220        phosphoenolpyruvate carboxykinase (GTP)       \n",
       "205 220        phosphoenolpyruvate carboxykinase (GTP)       \n",
       "206 220        phosphoenolpyruvate carboxykinase (GTP)       \n",
       "207 221        AMP-binding protein                           \n",
       "208 221        phenylacetate--CoA ligase                     \n",
       "209 221        phenylacetate--CoA ligase family protein      \n",
       "210 237        30S ribosomal protein S11                     \n",
       "211 237        30S ribosomal protein S11                     \n",
       "212 237        30S ribosomal protein S11                     \n",
       "213 237        30S ribosomal protein S11                     \n",
       "214 237        30S ribosomal protein S11                     \n",
       "215 240        elongation factor G                           \n",
       "216 240        elongation factor G                           \n",
       "217 240        elongation factor G                           \n",
       "218 241        50S ribosomal protein L19                     \n",
       "219 241        50S ribosomal protein L19                     \n",
       "220 241        50S ribosomal protein L19                     \n",
       "221 243        response regulator transcription factor       \n",
       "222 243        response regulator transcription factor       \n",
       "223 243        DNA-binding response regulator                \n",
       "224 246        glucose-1-phosphate adenylyltransferase       \n",
       "225 246        glucose-1-phosphate adenylyltransferase       \n",
       "226 246        glucose-1-phosphate adenylyltransferase       \n",
       "    gene_taxonomy                           pident\n",
       "1   Marasmitruncus massiliensis             91.892\n",
       "2   Oscillospiraceae bacterium NSJ-64       90.090\n",
       "3   bacterium 1XD42-8                       89.189\n",
       "4   Ruminococcaceae bacterium               84.173\n",
       "5   Marasmitruncus massiliensis             84.029\n",
       "6   Oscillospiraceae bacterium NSJ-64       81.727\n",
       "7   Marasmitruncus massiliensis             97.701\n",
       "8   bacterium 1XD42-8                       97.701\n",
       "9   Hydrogenoanaerobacterium saccharovorans 95.402\n",
       "10  Oscillospiraceae bacterium NSJ-64       88.136\n",
       "11  Marasmitruncus massiliensis             88.136\n",
       "12  Phocea massiliensis                     87.500\n",
       "13  bacterium 1XD42-8                       97.656\n",
       "14  Marasmitruncus massiliensis             96.094\n",
       "15  Clostridiaceae bacterium NSJ-31         94.531\n",
       "16  Angelakisella massiliensis              94.531\n",
       "17  Ruminococcaceae bacterium               86.290\n",
       "18  Marasmitruncus massiliensis             85.887\n",
       "19  bacterium 1XD42-8                       80.328\n",
       "20  Marasmitruncus massiliensis             90.667\n",
       "21  Oscillospiraceae bacterium NSJ-64       84.681\n",
       "22  Phocea massiliensis                     82.667\n",
       "23  Marasmitruncus massiliensis             90.886\n",
       "24  bacterium 1XD42-8                       89.228\n",
       "25  Oscillospiraceae bacterium NSJ-64       89.076\n",
       "⋮   ⋮                                       ⋮     \n",
       "202 uncultured Flavonifractor sp.           89.362\n",
       "203 bacterium 1XD42-8                       89.362\n",
       "204 Ruminococcaceae bacterium               87.544\n",
       "205 bacterium 1XD42-8                       87.895\n",
       "206 Marasmitruncus massiliensis             87.193\n",
       "207 Marasmitruncus massiliensis             84.562\n",
       "208 Ruminococcaceae bacterium               84.778\n",
       "209 bacterium 1XD42-8                       82.679\n",
       "210 Candidatus Pararuminococcus gallinarum  98.261\n",
       "211 Ruminococcaceae bacterium               98.261\n",
       "212 Clostridiaceae bacterium NSJ-31         98.261\n",
       "213 bacterium 1XD42-8                       99.130\n",
       "214 Angelakisella massiliensis              98.261\n",
       "215 Ruminococcaceae bacterium               88.399\n",
       "216 Marasmitruncus massiliensis             88.399\n",
       "217 Oscillospiraceae bacterium NSJ-64       84.435\n",
       "218 bacterium 1XD42-8                       91.228\n",
       "219 Marasmitruncus massiliensis             89.474\n",
       "220 Oscillospiraceae bacterium NSJ-64       82.456\n",
       "221 Marasmitruncus massiliensis             94.348\n",
       "222 Oscillospiraceae bacterium NSJ-64       90.435\n",
       "223 bacterium 1XD42-8                       90.870\n",
       "224 Oscillospiraceae bacterium NSJ-64       86.104\n",
       "225 Marasmitruncus massiliensis             85.396\n",
       "226 Ruminococcaceae bacterium               85.396"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df.dims(50)\n",
    "gene_annot %>%\n",
    "    filter(pident > 80,\n",
    "           pident_rank <= 3) %>%\n",
    "    select(cluster_id, gene_name, gene_taxonomy, pident)\n",
    "    \n",
    "df.dims()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-17T07:40:32.917987Z",
     "iopub.status.busy": "2021-01-17T07:40:32.916078Z",
     "iopub.status.idle": "2021-01-17T07:40:32.991597Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 284 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>cluster_id</th><th scope=col>gene_name</th></tr>\n",
       "\t<tr><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td> 4</td><td>50S ribosomal protein L22                              </td></tr>\n",
       "\t<tr><td> 4</td><td>MULTISPECIES: 50S ribosomal protein L22                </td></tr>\n",
       "\t<tr><td> 5</td><td>type I DNA topoisomerase                               </td></tr>\n",
       "\t<tr><td> 5</td><td>MULTISPECIES: type I DNA topoisomerase                 </td></tr>\n",
       "\t<tr><td> 5</td><td>unnamed protein product                                </td></tr>\n",
       "\t<tr><td> 5</td><td>dNA topoisomerase                                      </td></tr>\n",
       "\t<tr><td> 5</td><td>DNA topoisomerase-1                                    </td></tr>\n",
       "\t<tr><td> 5</td><td>DNA topoisomerase 1                                    </td></tr>\n",
       "\t<tr><td>10</td><td>HPr family phosphocarrier protein                      </td></tr>\n",
       "\t<tr><td>10</td><td>phosphocarrier, HPr family                             </td></tr>\n",
       "\t<tr><td>10</td><td>phosphotransferase System HPr (HPr) Family             </td></tr>\n",
       "\t<tr><td>10</td><td>MULTISPECIES: HPr family phosphocarrier protein        </td></tr>\n",
       "\t<tr><td>11</td><td>50S ribosomal protein L30                              </td></tr>\n",
       "\t<tr><td>11</td><td>ribosomal protein L30                                  </td></tr>\n",
       "\t<tr><td>11</td><td>MULTISPECIES: 50S ribosomal protein L30                </td></tr>\n",
       "\t<tr><td>13</td><td>30S ribosomal protein S12                              </td></tr>\n",
       "\t<tr><td>13</td><td>ribosomal protein S12                                  </td></tr>\n",
       "\t<tr><td>13</td><td>MULTISPECIES: 30S ribosomal protein S12                </td></tr>\n",
       "\t<tr><td>19</td><td>tRNA (guanosine(37)-N1)-methyltransferase TrmD         </td></tr>\n",
       "\t<tr><td>19</td><td>tRNA (guanine-N(1)-)-methyltransferase                 </td></tr>\n",
       "\t<tr><td>20</td><td>30S ribosomal protein S3                               </td></tr>\n",
       "\t<tr><td>20</td><td>MULTISPECIES: 30S ribosomal protein S3                 </td></tr>\n",
       "\t<tr><td>20</td><td>30S ribosomal protein S3, partial                      </td></tr>\n",
       "\t<tr><td>22</td><td>DNA-directed RNA polymerase subunit beta'              </td></tr>\n",
       "\t<tr><td>22</td><td>MULTISPECIES: DNA-directed RNA polymerase subunit beta'</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>221</td><td>MULTISPECIES: phenylacetate--CoA ligase                                                 </td></tr>\n",
       "\t<tr><td>221</td><td>hypothetical protein EUBSIR_00342                                                       </td></tr>\n",
       "\t<tr><td>221</td><td>phenylacetate-CoA ligase                                                                </td></tr>\n",
       "\t<tr><td>227</td><td>cell wall hydrolase                                                                     </td></tr>\n",
       "\t<tr><td>227</td><td>cell Wall Hydrolase                                                                     </td></tr>\n",
       "\t<tr><td>227</td><td>MULTISPECIES: cell wall hydrolase                                                       </td></tr>\n",
       "\t<tr><td>227</td><td>Cell wall hydrolase CwlJ                                                                </td></tr>\n",
       "\t<tr><td>237</td><td>30S ribosomal protein S11                                                               </td></tr>\n",
       "\t<tr><td>237</td><td>MULTISPECIES: 30S ribosomal protein S11                                                 </td></tr>\n",
       "\t<tr><td>240</td><td>elongation factor G                                                                     </td></tr>\n",
       "\t<tr><td>240</td><td>MULTISPECIES: elongation factor G                                                       </td></tr>\n",
       "\t<tr><td>240</td><td>putative translation elongation factor G                                                </td></tr>\n",
       "\t<tr><td>240</td><td>elongation factor G, partial                                                            </td></tr>\n",
       "\t<tr><td>240</td><td>translation elongation factor 2 (EF-2/EF-G)                                             </td></tr>\n",
       "\t<tr><td>241</td><td>50S ribosomal protein L19                                                               </td></tr>\n",
       "\t<tr><td>241</td><td>large subunit ribosomal protein L19                                                     </td></tr>\n",
       "\t<tr><td>241</td><td>unnamed protein product                                                                 </td></tr>\n",
       "\t<tr><td>243</td><td>response regulator transcription factor                                                 </td></tr>\n",
       "\t<tr><td>243</td><td>DNA-binding response regulator                                                          </td></tr>\n",
       "\t<tr><td>243</td><td>response regulator                                                                      </td></tr>\n",
       "\t<tr><td>243</td><td>MULTISPECIES: response regulator transcription factor                                   </td></tr>\n",
       "\t<tr><td>243</td><td>response regulator receiver domain protein                                              </td></tr>\n",
       "\t<tr><td>243</td><td>DNA-binding response regulator, OmpR family, contains REC and winged-helix (wHTH) domain</td></tr>\n",
       "\t<tr><td>246</td><td>glucose-1-phosphate adenylyltransferase                                                 </td></tr>\n",
       "\t<tr><td>246</td><td>MULTISPECIES: glucose-1-phosphate adenylyltransferase                                   </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 284 × 2\n",
       "\\begin{tabular}{ll}\n",
       " cluster\\_id & gene\\_name\\\\\n",
       " <dbl> & <chr>\\\\\n",
       "\\hline\n",
       "\t  4 & 50S ribosomal protein L22                              \\\\\n",
       "\t  4 & MULTISPECIES: 50S ribosomal protein L22                \\\\\n",
       "\t  5 & type I DNA topoisomerase                               \\\\\n",
       "\t  5 & MULTISPECIES: type I DNA topoisomerase                 \\\\\n",
       "\t  5 & unnamed protein product                                \\\\\n",
       "\t  5 & dNA topoisomerase                                      \\\\\n",
       "\t  5 & DNA topoisomerase-1                                    \\\\\n",
       "\t  5 & DNA topoisomerase 1                                    \\\\\n",
       "\t 10 & HPr family phosphocarrier protein                      \\\\\n",
       "\t 10 & phosphocarrier, HPr family                             \\\\\n",
       "\t 10 & phosphotransferase System HPr (HPr) Family             \\\\\n",
       "\t 10 & MULTISPECIES: HPr family phosphocarrier protein        \\\\\n",
       "\t 11 & 50S ribosomal protein L30                              \\\\\n",
       "\t 11 & ribosomal protein L30                                  \\\\\n",
       "\t 11 & MULTISPECIES: 50S ribosomal protein L30                \\\\\n",
       "\t 13 & 30S ribosomal protein S12                              \\\\\n",
       "\t 13 & ribosomal protein S12                                  \\\\\n",
       "\t 13 & MULTISPECIES: 30S ribosomal protein S12                \\\\\n",
       "\t 19 & tRNA (guanosine(37)-N1)-methyltransferase TrmD         \\\\\n",
       "\t 19 & tRNA (guanine-N(1)-)-methyltransferase                 \\\\\n",
       "\t 20 & 30S ribosomal protein S3                               \\\\\n",
       "\t 20 & MULTISPECIES: 30S ribosomal protein S3                 \\\\\n",
       "\t 20 & 30S ribosomal protein S3, partial                      \\\\\n",
       "\t 22 & DNA-directed RNA polymerase subunit beta'              \\\\\n",
       "\t 22 & MULTISPECIES: DNA-directed RNA polymerase subunit beta'\\\\\n",
       "\t ⋮ & ⋮\\\\\n",
       "\t 221 & MULTISPECIES: phenylacetate--CoA ligase                                                 \\\\\n",
       "\t 221 & hypothetical protein EUBSIR\\_00342                                                       \\\\\n",
       "\t 221 & phenylacetate-CoA ligase                                                                \\\\\n",
       "\t 227 & cell wall hydrolase                                                                     \\\\\n",
       "\t 227 & cell Wall Hydrolase                                                                     \\\\\n",
       "\t 227 & MULTISPECIES: cell wall hydrolase                                                       \\\\\n",
       "\t 227 & Cell wall hydrolase CwlJ                                                                \\\\\n",
       "\t 237 & 30S ribosomal protein S11                                                               \\\\\n",
       "\t 237 & MULTISPECIES: 30S ribosomal protein S11                                                 \\\\\n",
       "\t 240 & elongation factor G                                                                     \\\\\n",
       "\t 240 & MULTISPECIES: elongation factor G                                                       \\\\\n",
       "\t 240 & putative translation elongation factor G                                                \\\\\n",
       "\t 240 & elongation factor G, partial                                                            \\\\\n",
       "\t 240 & translation elongation factor 2 (EF-2/EF-G)                                             \\\\\n",
       "\t 241 & 50S ribosomal protein L19                                                               \\\\\n",
       "\t 241 & large subunit ribosomal protein L19                                                     \\\\\n",
       "\t 241 & unnamed protein product                                                                 \\\\\n",
       "\t 243 & response regulator transcription factor                                                 \\\\\n",
       "\t 243 & DNA-binding response regulator                                                          \\\\\n",
       "\t 243 & response regulator                                                                      \\\\\n",
       "\t 243 & MULTISPECIES: response regulator transcription factor                                   \\\\\n",
       "\t 243 & response regulator receiver domain protein                                              \\\\\n",
       "\t 243 & DNA-binding response regulator, OmpR family, contains REC and winged-helix (wHTH) domain\\\\\n",
       "\t 246 & glucose-1-phosphate adenylyltransferase                                                 \\\\\n",
       "\t 246 & MULTISPECIES: glucose-1-phosphate adenylyltransferase                                   \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 284 × 2\n",
       "\n",
       "| cluster_id &lt;dbl&gt; | gene_name &lt;chr&gt; |\n",
       "|---|---|\n",
       "|  4 | 50S ribosomal protein L22                               |\n",
       "|  4 | MULTISPECIES: 50S ribosomal protein L22                 |\n",
       "|  5 | type I DNA topoisomerase                                |\n",
       "|  5 | MULTISPECIES: type I DNA topoisomerase                  |\n",
       "|  5 | unnamed protein product                                 |\n",
       "|  5 | dNA topoisomerase                                       |\n",
       "|  5 | DNA topoisomerase-1                                     |\n",
       "|  5 | DNA topoisomerase 1                                     |\n",
       "| 10 | HPr family phosphocarrier protein                       |\n",
       "| 10 | phosphocarrier, HPr family                              |\n",
       "| 10 | phosphotransferase System HPr (HPr) Family              |\n",
       "| 10 | MULTISPECIES: HPr family phosphocarrier protein         |\n",
       "| 11 | 50S ribosomal protein L30                               |\n",
       "| 11 | ribosomal protein L30                                   |\n",
       "| 11 | MULTISPECIES: 50S ribosomal protein L30                 |\n",
       "| 13 | 30S ribosomal protein S12                               |\n",
       "| 13 | ribosomal protein S12                                   |\n",
       "| 13 | MULTISPECIES: 30S ribosomal protein S12                 |\n",
       "| 19 | tRNA (guanosine(37)-N1)-methyltransferase TrmD          |\n",
       "| 19 | tRNA (guanine-N(1)-)-methyltransferase                  |\n",
       "| 20 | 30S ribosomal protein S3                                |\n",
       "| 20 | MULTISPECIES: 30S ribosomal protein S3                  |\n",
       "| 20 | 30S ribosomal protein S3, partial                       |\n",
       "| 22 | DNA-directed RNA polymerase subunit beta'               |\n",
       "| 22 | MULTISPECIES: DNA-directed RNA polymerase subunit beta' |\n",
       "| ⋮ | ⋮ |\n",
       "| 221 | MULTISPECIES: phenylacetate--CoA ligase                                                  |\n",
       "| 221 | hypothetical protein EUBSIR_00342                                                        |\n",
       "| 221 | phenylacetate-CoA ligase                                                                 |\n",
       "| 227 | cell wall hydrolase                                                                      |\n",
       "| 227 | cell Wall Hydrolase                                                                      |\n",
       "| 227 | MULTISPECIES: cell wall hydrolase                                                        |\n",
       "| 227 | Cell wall hydrolase CwlJ                                                                 |\n",
       "| 237 | 30S ribosomal protein S11                                                                |\n",
       "| 237 | MULTISPECIES: 30S ribosomal protein S11                                                  |\n",
       "| 240 | elongation factor G                                                                      |\n",
       "| 240 | MULTISPECIES: elongation factor G                                                        |\n",
       "| 240 | putative translation elongation factor G                                                 |\n",
       "| 240 | elongation factor G, partial                                                             |\n",
       "| 240 | translation elongation factor 2 (EF-2/EF-G)                                              |\n",
       "| 241 | 50S ribosomal protein L19                                                                |\n",
       "| 241 | large subunit ribosomal protein L19                                                      |\n",
       "| 241 | unnamed protein product                                                                  |\n",
       "| 243 | response regulator transcription factor                                                  |\n",
       "| 243 | DNA-binding response regulator                                                           |\n",
       "| 243 | response regulator                                                                       |\n",
       "| 243 | MULTISPECIES: response regulator transcription factor                                    |\n",
       "| 243 | response regulator receiver domain protein                                               |\n",
       "| 243 | DNA-binding response regulator, OmpR family, contains REC and winged-helix (wHTH) domain |\n",
       "| 246 | glucose-1-phosphate adenylyltransferase                                                  |\n",
       "| 246 | MULTISPECIES: glucose-1-phosphate adenylyltransferase                                    |\n",
       "\n"
      ],
      "text/plain": [
       "    cluster_id\n",
       "1    4        \n",
       "2    4        \n",
       "3    5        \n",
       "4    5        \n",
       "5    5        \n",
       "6    5        \n",
       "7    5        \n",
       "8    5        \n",
       "9   10        \n",
       "10  10        \n",
       "11  10        \n",
       "12  10        \n",
       "13  11        \n",
       "14  11        \n",
       "15  11        \n",
       "16  13        \n",
       "17  13        \n",
       "18  13        \n",
       "19  19        \n",
       "20  19        \n",
       "21  20        \n",
       "22  20        \n",
       "23  20        \n",
       "24  22        \n",
       "25  22        \n",
       "⋮   ⋮         \n",
       "260 221       \n",
       "261 221       \n",
       "262 221       \n",
       "263 227       \n",
       "264 227       \n",
       "265 227       \n",
       "266 227       \n",
       "267 237       \n",
       "268 237       \n",
       "269 240       \n",
       "270 240       \n",
       "271 240       \n",
       "272 240       \n",
       "273 240       \n",
       "274 241       \n",
       "275 241       \n",
       "276 241       \n",
       "277 243       \n",
       "278 243       \n",
       "279 243       \n",
       "280 243       \n",
       "281 243       \n",
       "282 243       \n",
       "283 246       \n",
       "284 246       \n",
       "    gene_name                                                                               \n",
       "1   50S ribosomal protein L22                                                               \n",
       "2   MULTISPECIES: 50S ribosomal protein L22                                                 \n",
       "3   type I DNA topoisomerase                                                                \n",
       "4   MULTISPECIES: type I DNA topoisomerase                                                  \n",
       "5   unnamed protein product                                                                 \n",
       "6   dNA topoisomerase                                                                       \n",
       "7   DNA topoisomerase-1                                                                     \n",
       "8   DNA topoisomerase 1                                                                     \n",
       "9   HPr family phosphocarrier protein                                                       \n",
       "10  phosphocarrier, HPr family                                                              \n",
       "11  phosphotransferase System HPr (HPr) Family                                              \n",
       "12  MULTISPECIES: HPr family phosphocarrier protein                                         \n",
       "13  50S ribosomal protein L30                                                               \n",
       "14  ribosomal protein L30                                                                   \n",
       "15  MULTISPECIES: 50S ribosomal protein L30                                                 \n",
       "16  30S ribosomal protein S12                                                               \n",
       "17  ribosomal protein S12                                                                   \n",
       "18  MULTISPECIES: 30S ribosomal protein S12                                                 \n",
       "19  tRNA (guanosine(37)-N1)-methyltransferase TrmD                                          \n",
       "20  tRNA (guanine-N(1)-)-methyltransferase                                                  \n",
       "21  30S ribosomal protein S3                                                                \n",
       "22  MULTISPECIES: 30S ribosomal protein S3                                                  \n",
       "23  30S ribosomal protein S3, partial                                                       \n",
       "24  DNA-directed RNA polymerase subunit beta'                                               \n",
       "25  MULTISPECIES: DNA-directed RNA polymerase subunit beta'                                 \n",
       "⋮   ⋮                                                                                       \n",
       "260 MULTISPECIES: phenylacetate--CoA ligase                                                 \n",
       "261 hypothetical protein EUBSIR_00342                                                       \n",
       "262 phenylacetate-CoA ligase                                                                \n",
       "263 cell wall hydrolase                                                                     \n",
       "264 cell Wall Hydrolase                                                                     \n",
       "265 MULTISPECIES: cell wall hydrolase                                                       \n",
       "266 Cell wall hydrolase CwlJ                                                                \n",
       "267 30S ribosomal protein S11                                                               \n",
       "268 MULTISPECIES: 30S ribosomal protein S11                                                 \n",
       "269 elongation factor G                                                                     \n",
       "270 MULTISPECIES: elongation factor G                                                       \n",
       "271 putative translation elongation factor G                                                \n",
       "272 elongation factor G, partial                                                            \n",
       "273 translation elongation factor 2 (EF-2/EF-G)                                             \n",
       "274 50S ribosomal protein L19                                                               \n",
       "275 large subunit ribosomal protein L19                                                     \n",
       "276 unnamed protein product                                                                 \n",
       "277 response regulator transcription factor                                                 \n",
       "278 DNA-binding response regulator                                                          \n",
       "279 response regulator                                                                      \n",
       "280 MULTISPECIES: response regulator transcription factor                                   \n",
       "281 response regulator receiver domain protein                                              \n",
       "282 DNA-binding response regulator, OmpR family, contains REC and winged-helix (wHTH) domain\n",
       "283 glucose-1-phosphate adenylyltransferase                                                 \n",
       "284 MULTISPECIES: glucose-1-phosphate adenylyltransferase                                   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df.dims(50)\n",
    "gene_annot %>%\n",
    "    distinct(cluster_id, gene_name) \n",
    "df.dims()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# sessionInfo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "collapsed": true,
    "execution": {
     "iopub.execute_input": "2021-01-12T18:39:31.062497Z",
     "iopub.status.busy": "2021-01-12T18:39:31.060301Z",
     "iopub.status.idle": "2021-01-12T18:39:31.119705Z",
     "shell.execute_reply": "2021-01-12T18:39:31.118844Z"
    },
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LLG\n",
      "===\n",
      "\n",
      "Ley Lab Genome analysis pipeline (LLG)\n",
      "\n",
      "* Version: 0.1.9\n",
      "* Authors:\n",
      "  * Nick Youngblut <nyoungb2@gmail.com>\n",
      "* Maintainers:\n",
      "  * Nick Youngblut <nyoungb2@gmail.com>\n",
      "\n",
      "--- conda envs ---\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/gtdbtk.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::gtdbtk\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/checkm.yaml <==\n",
      "channels:\n",
      "- bioconda\n",
      "dependencies:\n",
      "- python=2.7\n",
      "- pigz\n",
      "- bioconda::prodigal\n",
      "- bioconda::pplacer\n",
      "- bioconda::checkm-genome\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/quast.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- bioconda::seqkit\n",
      "- bioconda::quast>=5.0.0\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/sourmash.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- bioconda::sourmash=2.0.0a4\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/fastqc.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- bioconda::seqtk\n",
      "- bioconda::fastqc=0.11.7\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/multiqc.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- bioconda::multiqc=1.4\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/hadley.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- conda-forge::r-ape\n",
      "- conda-forge::r-dplyr\n",
      "- conda-forge::r-tidyr\n",
      "- conda-forge::r-ggplot2\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/bbmap.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::bbmap=37.78\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/ani.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::dashing\n",
      "- bioconda::fastani\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/prokka.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- bioconda::prokka\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/prodigal.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- bioconda::prodigal\n",
      "- pip:\n",
      "  - biopython-convert\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/eggnog-mapper.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::eggnog-mapper\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/phylophlan.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::phylophlan\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/drep.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::mash\n",
      "- bioconda::checkm-genome\n",
      "- bioconda::drep\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/antismash.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- biopython=1.76\n",
      "- bioconda::antismash>=5.1.2\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/deepbgc.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- conda-forge::biopython=1.76\n",
      "- bioconda::deepbgc=0.1.18\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/abricate.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- bioconda::abricate\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/rRNA.yaml <==\n",
      "channels:\n",
      "- qiime2/label/r2020.2\n",
      "- conda-forge\n",
      "- bioconda\n",
      "- defaults\n",
      "dependencies:\n",
      "- barrnap\n",
      "- _libgcc_mutex=0.1\n",
      "- _openmp_mutex=4.5\n",
      "- _r-mutex=1.0.1\n",
      "- alsa-lib=1.1.5\n",
      "- arb-bio-tools=6.0.6\n",
      "- attrs=19.3.0\n",
      "- backcall=0.1.0\n",
      "- bibtexparser=1.1.0\n",
      "- binutils_impl_linux-64=2.33.1\n",
      "- binutils_linux-64=2.33.1\n",
      "- bioconductor-biobase=2.42.0\n",
      "- bioconductor-biocgenerics=0.28.0\n",
      "- bioconductor-biocparallel=1.16.6\n",
      "- bioconductor-biostrings=2.50.2\n",
      "- bioconductor-dada2=1.10.0\n",
      "- bioconductor-delayedarray=0.8.0\n",
      "- bioconductor-genomeinfodb=1.18.1\n",
      "- bioconductor-genomeinfodbdata=1.2.1\n",
      "- bioconductor-genomicalignments=1.18.1\n",
      "- bioconductor-genomicranges=1.34.0\n",
      "- bioconductor-iranges=2.16.0\n",
      "- bioconductor-rsamtools=1.34.0\n",
      "- bioconductor-s4vectors=0.20.1\n",
      "- bioconductor-shortread=1.40.0\n",
      "- bioconductor-summarizedexperiment=1.12.0\n",
      "- bioconductor-xvector=0.22.0\n",
      "- bioconductor-zlibbioc=1.28.0\n",
      "- biom-format=2.1.8\n",
      "- blas=2.14\n",
      "- blast=2.9.0\n",
      "- bleach=3.1.1\n",
      "- bokeh=1.4.0\n",
      "- boost=1.68.0\n",
      "- boost-cpp=1.68.0\n",
      "- bwidget=1.9.14\n",
      "- bzip2=1.0.8\n",
      "- ca-certificates=2019.11.28\n",
      "- cachecontrol=0.12.5\n",
      "- cairo=1.16.0\n",
      "- certifi=2019.11.28\n",
      "- cffi=1.13.2\n",
      "- chardet=3.0.4\n",
      "- click=7.0\n",
      "- cryptography=2.8\n",
      "- curl=7.68.0\n",
      "- cutadapt=2.8\n",
      "- cycler=0.10.0\n",
      "- cython=0.29.15\n",
      "- dbus=1.13.6\n",
      "- deblur=1.1.0\n",
      "- decorator=4.4.1\n",
      "- defusedxml=0.6.0\n",
      "- dendropy=4.4.0\n",
      "- dnaio=0.4.1\n",
      "- emperor=1.0.0\n",
      "- entrypoints=0.3\n",
      "- expat=2.2.9\n",
      "- fastcluster=1.1.26\n",
      "- fasttree=2.1.10\n",
      "- fontconfig=2.13.1\n",
      "- freetype=2.10.0\n",
      "- future=0.18.2\n",
      "- gcc_impl_linux-64=7.3.0\n",
      "- gcc_linux-64=7.3.0\n",
      "- gettext=0.19.8.1\n",
      "- gfortran_impl_linux-64=7.3.0\n",
      "- gfortran_linux-64=7.3.0\n",
      "- giflib=5.2.1\n",
      "- glib=2.58.3\n",
      "- gmp=6.2.0\n",
      "- gneiss=0.4.6\n",
      "- gnutls=3.6.5\n",
      "- graphite2=1.3.13\n",
      "- gsl=2.5\n",
      "- gst-plugins-base=1.14.5\n",
      "- gstreamer=1.14.5\n",
      "- gxx_impl_linux-64=7.3.0\n",
      "- gxx_linux-64=7.3.0\n",
      "- h5py=2.10.0\n",
      "- harfbuzz=2.4.0\n",
      "- hdf5=1.10.5\n",
      "- hdmedians=0.13\n",
      "- hmmer=3.1b2\n",
      "- icu=58.2\n",
      "- idna=2.9\n",
      "- ijson=2.6.1\n",
      "- importlib_metadata=1.5.0\n",
      "- ipykernel=5.1.4\n",
      "- ipython=7.12.0\n",
      "- ipython_genutils=0.2.0\n",
      "- ipywidgets=7.5.1\n",
      "- iqtree=1.6.12\n",
      "- jedi=0.16.0\n",
      "- jinja2=2.11.1\n",
      "- joblib=0.14.1\n",
      "- jpeg=9c\n",
      "- jsonschema=3.2.0\n",
      "- jupyter_client=6.0.0\n",
      "- jupyter_core=4.6.3\n",
      "- kiwisolver=1.1.0\n",
      "- krb5=1.16.4\n",
      "- lcms2=2.9\n",
      "- ld_impl_linux-64=2.33.1\n",
      "- libarbdb=6.0.6\n",
      "- libblas=3.8.0\n",
      "- libcblas=3.8.0\n",
      "- libcurl=7.68.0\n",
      "- libedit=3.1.20170329\n",
      "- libffi=3.2.1\n",
      "- libgcc=7.2.0\n",
      "- libgcc-ng=9.2.0\n",
      "- libgfortran-ng=7.3.0\n",
      "- libgomp=9.2.0\n",
      "- libiconv=1.15\n",
      "- liblapack=3.8.0\n",
      "- liblapacke=3.8.0\n",
      "- libopenblas=0.3.7\n",
      "- libpng=1.6.37\n",
      "- libsodium=1.0.17\n",
      "- libssh2=1.8.2\n",
      "- libstdcxx-ng=9.2.0\n",
      "- libtiff=4.1.0\n",
      "- libuuid=2.32.1\n",
      "- libxcb=1.13\n",
      "- libxml2=2.9.9\n",
      "- lockfile=0.12.2\n",
      "- lz4-c=1.8.3\n",
      "- mafft=7.310\n",
      "- make=4.3\n",
      "- markupsafe=1.1.1\n",
      "- matplotlib=3.1.1\n",
      "- matplotlib-base=3.1.1\n",
      "- mistune=0.8.4\n",
      "- more-itertools=8.2.0\n",
      "- msgpack-python=1.0.0\n",
      "- natsort=7.0.1\n",
      "- nbconvert=5.6.1\n",
      "- nbformat=5.0.4\n",
      "- ncurses=6.1\n",
      "- nettle=3.4.1\n",
      "- networkx=2.4\n",
      "- nose=1.3.7\n",
      "- notebook=6.0.3\n",
      "- numpy=1.18.1\n",
      "- olefile=0.46\n",
      "- openjdk=11.0.1\n",
      "- openssl=1.1.1d\n",
      "- packaging=20.1\n",
      "- pandas=0.25.3\n",
      "- pandoc=2.9.2\n",
      "- pandocfilters=1.4.2\n",
      "- pango=1.40.14\n",
      "- parso=0.6.1\n",
      "- patsy=0.5.1\n",
      "- pcre=8.44\n",
      "- perl=5.26.2\n",
      "- perl-archive-tar=2.32\n",
      "- perl-carp=1.38\n",
      "- perl-common-sense=3.74\n",
      "- perl-compress-raw-bzip2=2.087\n",
      "- perl-compress-raw-zlib=2.087\n",
      "- perl-exporter=5.72\n",
      "- perl-exporter-tiny=1.002001\n",
      "- perl-extutils-makemaker=7.36\n",
      "- perl-io-compress=2.087\n",
      "- perl-io-zlib=1.10\n",
      "- perl-json=4.02\n",
      "- perl-json-xs=2.34\n",
      "- perl-list-moreutils=0.428\n",
      "- perl-list-moreutils-xs=0.428\n",
      "- perl-pathtools=3.75\n",
      "- perl-scalar-list-utils=1.52\n",
      "- perl-types-serialiser=1.0\n",
      "- perl-xsloader=0.24\n",
      "- pexpect=4.8.0\n",
      "- pickleshare=0.7.5\n",
      "- pigz=2.3.4\n",
      "- pillow=7.0.0\n",
      "- pip=20.0.2\n",
      "- pixman=0.38.0\n",
      "- pluggy=0.12.0\n",
      "- prometheus_client=0.7.1\n",
      "- prompt_toolkit=3.0.3\n",
      "- psutil=5.7.0\n",
      "- pthread-stubs=0.4\n",
      "- ptyprocess=0.6.0\n",
      "- py=1.8.1\n",
      "- pycparser=2.19\n",
      "- pygments=2.5.2\n",
      "- pyopenssl=19.1.0\n",
      "- pyparsing=2.4.6\n",
      "- pyqt=5.9.2\n",
      "- pyrsistent=0.15.7\n",
      "- pysocks=1.7.1\n",
      "- pytest=5.3.5\n",
      "- python=3.6.7\n",
      "- python-dateutil=2.8.1\n",
      "- pytz=2019.3\n",
      "- pyyaml=5.3\n",
      "- pyzmq=19.0.0\n",
      "- q2-alignment=2020.2.0\n",
      "- q2-composition=2020.2.0\n",
      "- q2-cutadapt=2020.2.0\n",
      "- q2-dada2=2020.2.0\n",
      "- q2-deblur=2020.2.0\n",
      "- q2-demux=2020.2.0\n",
      "- q2-diversity=2020.2.0\n",
      "- q2-emperor=2020.2.0\n",
      "- q2-feature-classifier=2020.2.0\n",
      "- q2-feature-table=2020.2.0\n",
      "- q2-fragment-insertion=2020.2.0\n",
      "- q2-gneiss=2020.2.0\n",
      "- q2-longitudinal=2020.2.0\n",
      "- q2-metadata=2020.2.0\n",
      "- q2-phylogeny=2020.2.0\n",
      "- q2-quality-control=2020.2.0\n",
      "- q2-quality-filter=2020.2.0\n",
      "- q2-sample-classifier=2020.2.0\n",
      "- q2-taxa=2020.2.0\n",
      "- q2-types=2020.2.0\n",
      "- q2-vsearch=2020.2.0\n",
      "- q2cli=2020.2.0\n",
      "- q2templates=2020.2.0\n",
      "- qiime2=2020.2.0\n",
      "- qt=5.9.7\n",
      "- r-assertthat=0.2.1\n",
      "- r-backports=1.1.5\n",
      "- r-base=3.5.1\n",
      "- r-bh=1.72.0_3\n",
      "- r-bitops=1.0_6\n",
      "- r-cli=2.0.2\n",
      "- r-cluster=2.1.0\n",
      "- r-colorspace=1.4_1\n",
      "- r-crayon=1.3.4\n",
      "- r-data.table=1.12.6\n",
      "- r-digest=0.6.25\n",
      "- r-ellipsis=0.3.0\n",
      "- r-fansi=0.4.1\n",
      "- r-farver=2.0.3\n",
      "- r-formatr=1.7\n",
      "- r-futile.logger=1.4.3\n",
      "- r-futile.options=1.0.1\n",
      "- r-ggplot2=3.2.1\n",
      "- r-glue=1.3.1\n",
      "- r-gtable=0.3.0\n",
      "- r-hwriter=1.3.2\n",
      "- r-labeling=0.3\n",
      "- r-lambda.r=1.2.4\n",
      "- r-lattice=0.20_40\n",
      "- r-latticeextra=0.6_28\n",
      "- r-lazyeval=0.2.2\n",
      "- r-lifecycle=0.1.0\n",
      "- r-magrittr=1.5\n",
      "- r-mass=7.3_51.5\n",
      "- r-matrix=1.2_18\n",
      "- r-matrixstats=0.55.0\n",
      "- r-mgcv=1.8_31\n",
      "- r-munsell=0.5.0\n",
      "- r-nlme=3.1_144\n",
      "- r-permute=0.9_5\n",
      "- r-pillar=1.4.3\n",
      "- r-pkgconfig=2.0.3\n",
      "- r-plyr=1.8.5\n",
      "- r-r6=2.4.1\n",
      "- r-rcolorbrewer=1.1_2\n",
      "- r-rcpp=1.0.3\n",
      "- r-rcppparallel=4.4.4\n",
      "- r-rcurl=1.98_1.1\n",
      "- r-reshape2=1.4.3\n",
      "- r-rlang=0.4.4\n",
      "- r-scales=1.1.0\n",
      "- r-snow=0.4_3\n",
      "- r-stringi=1.4.3\n",
      "- r-stringr=1.4.0\n",
      "- r-tibble=2.1.3\n",
      "- r-utf8=1.1.4\n",
      "- r-vctrs=0.2.3\n",
      "- r-vegan=2.5_6\n",
      "- r-viridislite=0.3.0\n",
      "- r-withr=2.1.2\n",
      "- r-zeallot=0.1.0\n",
      "- raxml=8.2.12\n",
      "- readline=8.0\n",
      "- requests=2.23.0\n",
      "- scikit-bio=0.5.5\n",
      "- scikit-learn=0.22.1\n",
      "- scipy=1.4.1\n",
      "- seaborn=0.10.0\n",
      "- send2trash=1.5.0\n",
      "- sepp=4.3.10\n",
      "- setuptools=45.2.0\n",
      "- sina=1.6.0\n",
      "- sip=4.19.8\n",
      "- six=1.14.0\n",
      "- sortmerna=2.0\n",
      "- sqlite=3.30.1\n",
      "- statsmodels=0.11.1\n",
      "- tbb=2019.9\n",
      "- terminado=0.8.3\n",
      "- testpath=0.4.4\n",
      "- tk=8.6.10\n",
      "- tktable=2.10\n",
      "- tornado=6.0.3\n",
      "- traitlets=4.3.3\n",
      "- tzlocal=2.0.0\n",
      "- unifrac=0.10.0\n",
      "- urllib3=1.25.7\n",
      "- vsearch=2.7.0\n",
      "- wcwidth=0.1.8\n",
      "- webencodings=0.5.1\n",
      "- wheel=0.34.2\n",
      "- widgetsnbextension=3.5.1\n",
      "- xopen=0.8.4\n",
      "- xorg-fixesproto=5.0\n",
      "- xorg-inputproto=2.3.2\n",
      "- xorg-kbproto=1.0.7\n",
      "- xorg-libice=1.0.10\n",
      "- xorg-libsm=1.2.3\n",
      "- xorg-libx11=1.6.9\n",
      "- xorg-libxau=1.0.9\n",
      "- xorg-libxdmcp=1.1.3\n",
      "- xorg-libxext=1.3.4\n",
      "- xorg-libxfixes=5.0.3\n",
      "- xorg-libxi=1.7.10\n",
      "- xorg-libxrender=0.9.10\n",
      "- xorg-libxtst=1.2.3\n",
      "- xorg-recordproto=1.14.2\n",
      "- xorg-renderproto=0.11.1\n",
      "- xorg-xextproto=7.3.0\n",
      "- xorg-xproto=7.0.31\n",
      "- xz=5.2.4\n",
      "- yaml=0.2.2\n",
      "- zeromq=4.3.2\n",
      "- zipp=3.0.0\n",
      "- zlib=1.2.11\n",
      "- zstd=1.4.4\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/traitar.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- scipy\n",
      "- pandas\n",
      "- numpy\n",
      "- matplotlib\n",
      "- pigz\n",
      "- bioconda::hmmer\n",
      "- pip:\n",
      "  - traitar3>=0.1.1\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/cctyper.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "- russel88\n",
      "dependencies:\n",
      "- pigz\n",
      "- cctyper\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/comparem.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::comparem"
     ]
    }
   ],
   "source": [
    "pipelineInfo('/ebio/abt3_projects/software/dev/ll_pipelines/llg/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-12T18:39:31.218756Z",
     "iopub.status.busy": "2021-01-12T18:39:31.201101Z",
     "iopub.status.idle": "2021-01-12T18:39:31.274232Z",
     "shell.execute_reply": "2021-01-12T18:39:31.272154Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LLPRIMER\n",
      "========\n",
      "\n",
      "Ley Lab Primer generation pipeline (LLPRIMER)\n",
      "\n",
      "* Version: 0.2.2\n",
      "* Authors:\n",
      "  * Nick Youngblut <nyoungb2@gmail.com>\n",
      "* Maintainers:\n",
      "  * Nick Youngblut <nyoungb2@gmail.com>\n",
      "\n",
      "--- conda envs ---\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llprimer//bin/envs/pdp.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- python=3.7\n",
      "- intervaltree\n",
      "- prodigal\n",
      "- blast\n",
      "- bedtools\n",
      "- mafft\n",
      "- mummer=3.23\n",
      "- emboss\n",
      "- primer3=1.1.4\n",
      "- biopython<1.78\n",
      "- pybedtools\n",
      "- joblib\n",
      "- tqdm\n",
      "- openpyxl\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llprimer//bin/envs/genes.yaml <==\n",
      "channels:\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- python=3\n",
      "- numpy\n",
      "- pyfaidx\n",
      "- bioconda::seqkit\n",
      "- bioconda::fasta-splitter\n",
      "- bioconda::vsearch\n",
      "- bioconda::prodigal\n",
      "- bioconda::mmseqs2\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llprimer//bin/envs/aln.yaml <==\n",
      "channels:\n",
      "- bioconda\n",
      "- conda-forge\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::kalign3\n",
      "- bioconda::mafft\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llprimer//bin/envs/blast.yaml <==\n",
      "channels:\n",
      "- bioconda\n",
      "- conda-forge\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::seqkit\n",
      "- bioconda::biopython<1.78\n",
      "- bioconda::blast>=2.10.1\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llprimer//bin/envs/taxids.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- intervaltree\n",
      "- taxonkit\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llprimer//bin/envs/primer3.yaml <==\n",
      "channels:\n",
      "- bioconda\n",
      "- conda-forge\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::biopython<1.78\n",
      "- bioconda::primer3-py"
     ]
    }
   ],
   "source": [
    "pipelineInfo('/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-12T18:39:31.299767Z",
     "iopub.status.busy": "2021-01-12T18:39:31.297629Z",
     "iopub.status.idle": "2021-01-12T18:39:31.420355Z",
     "shell.execute_reply": "2021-01-12T18:39:31.417388Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "R version 3.6.3 (2020-02-29)\n",
       "Platform: x86_64-conda_cos6-linux-gnu (64-bit)\n",
       "Running under: Ubuntu 18.04.5 LTS\n",
       "\n",
       "Matrix products: default\n",
       "BLAS/LAPACK: /ebio/abt3_projects/Georg_animal_feces/envs/tidyverse/lib/libopenblasp-r0.3.9.so\n",
       "\n",
       "locale:\n",
       " [1] LC_CTYPE=C.UTF-8    LC_NUMERIC=C        LC_TIME=C          \n",
       " [4] LC_COLLATE=C        LC_MONETARY=C       LC_MESSAGES=C      \n",
       " [7] LC_PAPER=C          LC_NAME=C           LC_ADDRESS=C       \n",
       "[10] LC_TELEPHONE=C      LC_MEASUREMENT=C    LC_IDENTIFICATION=C\n",
       "\n",
       "attached base packages:\n",
       "[1] stats     graphics  grDevices utils     datasets  methods   base     \n",
       "\n",
       "other attached packages:\n",
       "[1] LeyLabRMisc_0.1.6 ggplot2_3.3.1     tidyr_1.1.0       dplyr_1.0.0      \n",
       "\n",
       "loaded via a namespace (and not attached):\n",
       " [1] Rcpp_1.0.4.6     magrittr_1.5     munsell_0.5.0    tidyselect_1.1.0\n",
       " [5] uuid_0.1-4       colorspace_1.4-1 R6_2.4.1         rlang_0.4.6     \n",
       " [9] tools_3.6.3      grid_3.6.3       gtable_0.3.0     withr_2.2.0     \n",
       "[13] htmltools_0.4.0  ellipsis_0.3.1   digest_0.6.25    tibble_3.0.1    \n",
       "[17] lifecycle_0.2.0  crayon_1.3.4     IRdisplay_0.7.0  purrr_0.3.4     \n",
       "[21] repr_1.1.0       base64enc_0.1-3  vctrs_0.3.1      IRkernel_1.1    \n",
       "[25] glue_1.4.1       evaluate_0.14    stringi_1.4.6    pbdZMQ_0.3-3    \n",
       "[29] compiler_3.6.3   pillar_1.4.4     scales_1.1.1     generics_0.0.2  \n",
       "[33] jsonlite_1.6.1   pkgconfig_2.0.3 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "sessionInfo()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R [conda env:Georg_animal_feces-tidyverse]",
   "language": "R",
   "name": "conda-env-Georg_animal_feces-tidyverse-r"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.3"
  },
  "toc-autonumbering": true,
  "toc-showmarkdowntxt": false,
  "toc-showtags": false
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
