{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Goal\n",
    "\n",
    "* Primer design for clade of interest"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Var"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-16T07:47:51.170109Z",
     "iopub.status.busy": "2021-01-16T07:47:51.166835Z",
     "iopub.status.idle": "2021-01-16T07:47:51.188090Z"
    }
   },
   "outputs": [],
   "source": [
    "base_dir = '/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/'\n",
    "clade = 'Butyricimonas'\n",
    "taxid = 574697 "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Init"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-16T07:47:51.562993Z",
     "iopub.status.busy": "2021-01-16T07:47:51.560457Z",
     "iopub.status.idle": "2021-01-16T07:47:51.581538Z"
    }
   },
   "outputs": [],
   "source": [
    "library(dplyr)\n",
    "library(tidyr)\n",
    "library(ggplot2)\n",
    "library(LeyLabRMisc)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-16T07:47:51.678557Z",
     "iopub.status.busy": "2021-01-16T07:47:51.675437Z",
     "iopub.status.idle": "2021-01-16T07:47:51.698435Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Directory already exists: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted//Butyricimonas \n"
     ]
    }
   ],
   "source": [
    "df.dims()\n",
    "work_dir = file.path(base_dir, clade)\n",
    "make_dir(work_dir)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Genome download\n",
    "\n",
    "* Downloading genomes from NCBI"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-12T13:07:32.874718Z",
     "iopub.status.busy": "2021-01-12T13:07:32.867938Z",
     "iopub.status.idle": "2021-01-12T13:07:32.985785Z",
     "shell.execute_reply": "2021-01-12T13:07:32.983225Z"
    }
   },
   "source": [
    "```\n",
    "OUTDIR=/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/\n",
    "mkdir -p $OUTDIR\n",
    "ncbi-genome-download -p 12 -s genbank -F fasta --genera Butyricimonas -o $OUTDIR bacteria\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Genome quality\n",
    "\n",
    "* Filtering genomes by quality"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-12T18:43:18.459453Z",
     "iopub.status.busy": "2021-01-12T18:43:18.457315Z",
     "iopub.status.idle": "2021-01-12T18:43:18.582607Z",
     "shell.execute_reply": "2021-01-12T18:43:18.581292Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 29 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Name</th><th scope=col>Fasta</th><th scope=col>Domain</th><th scope=col>Taxid</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>GCA_000379665.1_ASM37966v1_genomic</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/genbank/bacteria/GCA_000379665.1/GCA_000379665.1_ASM37966v1_genomic.fna.gz</td><td>Bacteria</td><td>244127</td></tr>\n",
       "\t<tr><td>GCA_000519105.1_ASM51910v1_genomic</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/genbank/bacteria/GCA_000519105.1/GCA_000519105.1_ASM51910v1_genomic.fna.gz</td><td>Bacteria</td><td>244127</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>GCA_902364335.1_MGYG-HGUT-00203_genomic</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/genbank/bacteria/GCA_902364335.1/GCA_902364335.1_MGYG-HGUT-00203_genomic.fna.gz</td><td>Bacteria</td><td>244127</td></tr>\n",
       "\t<tr><td>GCA_902376195.1_MGYG-HGUT-01552_genomic</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/genbank/bacteria/GCA_902376195.1/GCA_902376195.1_MGYG-HGUT-01552_genomic.fna.gz</td><td>Bacteria</td><td>244127</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 29 × 4\n",
       "\\begin{tabular}{llll}\n",
       " Name & Fasta & Domain & Taxid\\\\\n",
       " <chr> & <chr> & <fct> & <dbl>\\\\\n",
       "\\hline\n",
       "\t GCA\\_000379665.1\\_ASM37966v1\\_genomic & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/Butyricimonas/genbank/bacteria/GCA\\_000379665.1/GCA\\_000379665.1\\_ASM37966v1\\_genomic.fna.gz & Bacteria & 244127\\\\\n",
       "\t GCA\\_000519105.1\\_ASM51910v1\\_genomic & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/Butyricimonas/genbank/bacteria/GCA\\_000519105.1/GCA\\_000519105.1\\_ASM51910v1\\_genomic.fna.gz & Bacteria & 244127\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t GCA\\_902364335.1\\_MGYG-HGUT-00203\\_genomic & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/Butyricimonas/genbank/bacteria/GCA\\_902364335.1/GCA\\_902364335.1\\_MGYG-HGUT-00203\\_genomic.fna.gz & Bacteria & 244127\\\\\n",
       "\t GCA\\_902376195.1\\_MGYG-HGUT-01552\\_genomic & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/Butyricimonas/genbank/bacteria/GCA\\_902376195.1/GCA\\_902376195.1\\_MGYG-HGUT-01552\\_genomic.fna.gz & Bacteria & 244127\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 29 × 4\n",
       "\n",
       "| Name &lt;chr&gt; | Fasta &lt;chr&gt; | Domain &lt;fct&gt; | Taxid &lt;dbl&gt; |\n",
       "|---|---|---|---|\n",
       "| GCA_000379665.1_ASM37966v1_genomic | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/genbank/bacteria/GCA_000379665.1/GCA_000379665.1_ASM37966v1_genomic.fna.gz | Bacteria | 244127 |\n",
       "| GCA_000519105.1_ASM51910v1_genomic | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/genbank/bacteria/GCA_000519105.1/GCA_000519105.1_ASM51910v1_genomic.fna.gz | Bacteria | 244127 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| GCA_902364335.1_MGYG-HGUT-00203_genomic | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/genbank/bacteria/GCA_902364335.1/GCA_902364335.1_MGYG-HGUT-00203_genomic.fna.gz | Bacteria | 244127 |\n",
       "| GCA_902376195.1_MGYG-HGUT-01552_genomic | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/genbank/bacteria/GCA_902376195.1/GCA_902376195.1_MGYG-HGUT-01552_genomic.fna.gz | Bacteria | 244127 |\n",
       "\n"
      ],
      "text/plain": [
       "   Name                                   \n",
       "1  GCA_000379665.1_ASM37966v1_genomic     \n",
       "2  GCA_000519105.1_ASM51910v1_genomic     \n",
       "⋮  ⋮                                      \n",
       "28 GCA_902364335.1_MGYG-HGUT-00203_genomic\n",
       "29 GCA_902376195.1_MGYG-HGUT-01552_genomic\n",
       "   Fasta                                                                                                                                                                           \n",
       "1  /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/genbank/bacteria/GCA_000379665.1/GCA_000379665.1_ASM37966v1_genomic.fna.gz     \n",
       "2  /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/genbank/bacteria/GCA_000519105.1/GCA_000519105.1_ASM51910v1_genomic.fna.gz     \n",
       "⋮  ⋮                                                                                                                                                                               \n",
       "28 /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/genbank/bacteria/GCA_902364335.1/GCA_902364335.1_MGYG-HGUT-00203_genomic.fna.gz\n",
       "29 /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/genbank/bacteria/GCA_902376195.1/GCA_902376195.1_MGYG-HGUT-01552_genomic.fna.gz\n",
       "   Domain   Taxid \n",
       "1  Bacteria 244127\n",
       "2  Bacteria 244127\n",
       "⋮  ⋮        ⋮     \n",
       "28 Bacteria 244127\n",
       "29 Bacteria 244127"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "File written: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted//Butyricimonas/genbank/samples.txt \n"
     ]
    }
   ],
   "source": [
    "D = file.path(base_dir, clade, 'genbank')\n",
    "files = list_files(D, '.fna.gz')\n",
    "samps = data.frame(Name = files %>% as.character %>% basename,\n",
    "                   Fasta = files,\n",
    "                   Domain = 'Bacteria',\n",
    "                   Taxid = taxid) %>%\n",
    "    mutate(Name = gsub('\\\\.fna\\\\.gz$', '', Name),\n",
    "           Fasta = gsub('/+', '/', Fasta))\n",
    "samps\n",
    "\n",
    "# writing file\n",
    "outfile = file.path(D, 'samples.txt')\n",
    "write_table(samps, outfile)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### LLG"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Config"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-12T18:44:47.887578Z",
     "iopub.status.busy": "2021-01-12T18:44:47.885583Z",
     "iopub.status.idle": "2021-01-12T18:44:47.916028Z",
     "shell.execute_reply": "2021-01-12T18:44:47.914263Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# table with genome --> fasta_file information\n",
      "samples_file: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/genbank/samples.txt\n",
      "\n",
      "# output location\n",
      "output_dir: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/LLG_output/\n",
      "\n",
      "# temporary file directory (your username will be added automatically)\n",
      "tmp_dir: /ebio/abt3_scratch/\n",
      "\n",
      "# batch processing of genomes for certain steps\n",
      "## increase to better parallelize\n",
      "batches: 2 \n",
      "\n",
      "# Domain of genomes ('Archaea' or 'Bacteria)\n",
      "## Use \"Skip\" if provided as a \"Domain\" column in the genome table\n",
      "Domain: Skip\n",
      "\n",
      "# software parameters\n",
      "# Use \"Skip\" to skip any of these steps. If no params for rule, use \"\"\n",
      "# dRep MAGs are not further analyzed, but you can de-rep & then use the de-rep genome table as input.\n",
      "params:\n",
      "  ionice: -c 3\n",
      "  # assembly assessment\n",
      "  seqkit: \"\"\n",
      "  quast: Skip #\"\"\n",
      "  multiqc_on_quast: \"\" \n",
      "  checkm: \"\"\n",
      "  # de-replication (requires checkm)\n",
      "  drep: -comp 90 -con 5 -sa 0.999\n",
      "  # taxonomy\n",
      "  sourmash:\n",
      "    compute: Skip #--scaled 10000 -k 31\n",
      "    gather: -k 31\n",
      "  gtdbtk:\n",
      "    classify_wf: --min_perc_aa 10\n",
      "  # genome pairwise ANI  \n",
      "  fastani: Skip #--fragLen 3000 --minFraction 0.2 -k 16\n",
      "  dashing: Skip #-k 31 --full-tsv\n",
      "  comparem_aai: Skip #--evalue 0.001\n",
      "  # gene annotation\n",
      "  gene_call:\n",
      "    prokka: Skip #\"\" \n",
      "    multiqc_on_prokka: \"\"\n",
      "    prodigal: Skip #\"\"\n",
      "  eggnog_mapper: \"\"\n",
      "  eggnog_mapper_annot: \"\"\n",
      "  # rRNA (16S alignment & phylogeny)\n",
      "  barrnap: Skip #--lencutoff 0.8\n",
      "  vsearch_per_genome_drep: --id 0.95    # Skip to prevent drep of 16S copies within each genome\n",
      "  qiime2_fasttree: \"\"\n",
      "  qiime2_iqtree: --p-alrt 1000 --p-abayes --p-lbp 1000  --p-substitution-model 'GTR+I+G'\n",
      "  # genome phylogeny\n",
      "  phylophlan_config: Skip #--map_dna diamond --db_aa diamond --map_aa diamond --msa mafft --trim trimal --tree1 fasttree --tree2 raxml\n",
      "  phylophlan: --diversity high --min_num_markers 50 --accurate\n",
      "  # phenotype\n",
      "  traitar: Skip #\"\"\n",
      "  # biosynthetic gene clusters (BGCs)\n",
      "  antismash: Skip #--cb-knownclusters --cb-subclusters --asf  \n",
      "  DeepBGC: Skip #--score 0.5 --classifier-score 0.5 --prodigal-meta-mode\n",
      "  # antimicrobial resistance (AMR)\n",
      "  abricate: Skip #--minid 75 --mincov 80\n",
      "  # CRISPRs\n",
      "  cctyper: Skip #--prodigal meta\n",
      "  \n",
      "# databases\n",
      "databases:\n",
      "  checkM_data: /ebio/abt3_projects/databases_no-backup/checkM/\n",
      "  sourmash: /ebio/abt3_projects/databases_no-backup/sourmash/genbank-k31.sbt.json\n",
      "  sourmash_lca: /ebio/abt3_projects/databases_no-backup/sourmash/genbank-k31.lca.json.gz\n",
      "  gtdbtk: /ebio/abt3_projects/databases_no-backup/GTDB/release95/gtdbtk/db_info.md\n",
      "  phylophlan: /ebio/abt3_projects/databases_no-backup/phylophlan/PhyloPhlan/phylophlan.faa.bz2\n",
      "  eggnog: /ebio/abt3_projects/databases_no-backup/Eggnog/v2/eggnog.db\n",
      "  eggnog_diamond: /ebio/abt3_projects/databases_no-backup/Eggnog/v2/eggnog_proteins.dmnd\n",
      "  antismash: /ebio/abt3_projects/databases_no-backup/antismash/v5/\n",
      "  deepbgc: /ebio/abt3_projects/databases_no-backup/DeepBGC/\n",
      "  traitar: /ebio/abt3_projects/databases_no-backup/pfam/traitar/\n",
      "  abricate:\n",
      "    ncbi: /ebio/abt3_projects/databases_no-backup/abricate/ncbi/sequences\n",
      "    card: /ebio/abt3_projects/databases_no-backup/abricate/card/sequences\n",
      "    resfinder: /ebio/abt3_projects/databases_no-backup/abricate/resfinder/sequences\n",
      "    argannot: /ebio/abt3_projects/databases_no-backup/abricate/argannot/sequences\n",
      "    bacmet2: /ebio/abt3_projects/databases_no-backup/abricate/bacmet2/sequences\n",
      "    vfdb: /ebio/abt3_projects/databases_no-backup/abricate/vfdb/sequences\n",
      "    megares: /ebio/abt3_projects/databases_no-backup/abricate/megares/sequences\n",
      "    plasmidfinder: /ebio/abt3_projects/databases_no-backup/abricate/plasmidfinder/sequences\n",
      "\n",
      "# snakemake pipeline\n",
      "pipeline:\n",
      "  snakemake_folder: ./\n",
      "  script_folder: ./bin/scripts/\n",
      "  use_shared_mem: True\n",
      "  name: LLG"
     ]
    }
   ],
   "source": [
    "cat_file(file.path(work_dir, 'config_llg.yaml'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-12T13:14:50.416604Z",
     "iopub.status.busy": "2021-01-12T13:14:50.414179Z",
     "iopub.status.idle": "2021-01-12T13:14:50.431925Z",
     "shell.execute_reply": "2021-01-12T13:14:50.430260Z"
    }
   },
   "source": [
    "#### Run\n",
    "\n",
    "```\n",
    "(snakemake) @ rick:/ebio/abt3_projects/software/dev/ll_pipelines/llg\n",
    "$ screen -L -S llg ./snakemake_sge.sh /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/config_llg.yaml 20 -F\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-12T13:14:10.995042Z",
     "iopub.status.busy": "2021-01-12T13:14:10.992773Z",
     "iopub.status.idle": "2021-01-12T13:14:11.009275Z",
     "shell.execute_reply": "2021-01-12T13:14:11.008255Z"
    }
   },
   "source": [
    "### Samples table of high-quality genomes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-13T14:43:18.665723Z",
     "iopub.status.busy": "2021-01-13T14:43:18.663772Z",
     "iopub.status.idle": "2021-01-13T14:43:18.724202Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 29 × 33</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Bin.Id</th><th scope=col>Marker.lineage</th><th scope=col>X..genomes</th><th scope=col>X..markers</th><th scope=col>X..marker.sets</th><th scope=col>Completeness</th><th scope=col>Contamination</th><th scope=col>Strain.heterogeneity</th><th scope=col>Genome.size..bp.</th><th scope=col>X..ambiguous.bases</th><th scope=col>⋯</th><th scope=col>X0</th><th scope=col>X1</th><th scope=col>X2</th><th scope=col>X3</th><th scope=col>X4</th><th scope=col>X5.</th><th scope=col>assembly.Id</th><th scope=col>assembler.Id</th><th scope=col>taxon.Id</th><th scope=col>File</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>GCA_000379665.1_ASM37966v1_genomic </td><td>k__Bacteria (UID2569)</td><td>434</td><td>278</td><td>186</td><td>100.00</td><td>0.00</td><td>0</td><td>4770838</td><td> 131</td><td>⋯</td><td> 0</td><td>278</td><td>0</td><td>0</td><td>0</td><td>0</td><td>|ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Butyricimonas|LLG_output|checkM|1|checkm|markers_qa_summary.tsv.1</td><td>markers_qa_summary.tsv.1</td><td>NA</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/LLG_output/checkM/1/checkm/markers_qa_summary.tsv.1</td></tr>\n",
       "\t<tr><td>GCA_001915615.1_ASM191561v1_genomic</td><td>k__Bacteria (UID2569)</td><td>434</td><td>278</td><td>186</td><td> 88.69</td><td>0.27</td><td>0</td><td>4167409</td><td>2477</td><td>⋯</td><td>44</td><td>233</td><td>1</td><td>0</td><td>0</td><td>0</td><td>|ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Butyricimonas|LLG_output|checkM|1|checkm|markers_qa_summary.tsv.2</td><td>markers_qa_summary.tsv.2</td><td>NA</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/LLG_output/checkM/1/checkm/markers_qa_summary.tsv.2</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋱</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>GCA_900184685.1_PRJEB20857_genomic     </td><td>k__Bacteria (UID2569)</td><td>434</td><td>278</td><td>186</td><td>100.00</td><td>0</td><td>0</td><td>5300003</td><td>  0</td><td>⋯</td><td>0</td><td>278</td><td>0</td><td>0</td><td>0</td><td>0</td><td>|ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Butyricimonas|LLG_output|checkM|2|checkm|markers_qa_summary.tsv.13</td><td>markers_qa_summary.tsv.13</td><td>NA</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/LLG_output/checkM/2/checkm/markers_qa_summary.tsv.13</td></tr>\n",
       "\t<tr><td>GCA_902364335.1_MGYG-HGUT-00203_genomic</td><td>k__Bacteria (UID2569)</td><td>434</td><td>278</td><td>186</td><td> 99.46</td><td>0</td><td>0</td><td>4537596</td><td>250</td><td>⋯</td><td>1</td><td>277</td><td>0</td><td>0</td><td>0</td><td>0</td><td>|ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Butyricimonas|LLG_output|checkM|2|checkm|markers_qa_summary.tsv.14</td><td>markers_qa_summary.tsv.14</td><td>NA</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/LLG_output/checkM/2/checkm/markers_qa_summary.tsv.14</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 29 × 33\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " Bin.Id & Marker.lineage & X..genomes & X..markers & X..marker.sets & Completeness & Contamination & Strain.heterogeneity & Genome.size..bp. & X..ambiguous.bases & ⋯ & X0 & X1 & X2 & X3 & X4 & X5. & assembly.Id & assembler.Id & taxon.Id & File\\\\\n",
       " <fct> & <fct> & <int> & <int> & <int> & <dbl> & <dbl> & <int> & <int> & <int> & ⋯ & <int> & <int> & <int> & <int> & <int> & <int> & <fct> & <fct> & <lgl> & <fct>\\\\\n",
       "\\hline\n",
       "\t GCA\\_000379665.1\\_ASM37966v1\\_genomic  & k\\_\\_Bacteria (UID2569) & 434 & 278 & 186 & 100.00 & 0.00 & 0 & 4770838 &  131 & ⋯ &  0 & 278 & 0 & 0 & 0 & 0 & \\textbar{}ebio\\textbar{}abt3\\_projects\\textbar{}software\\textbar{}dev\\textbar{}ll\\_pipelines\\textbar{}llprimer\\textbar{}experiments\\textbar{}HMP\\_most-wanted\\textbar{}Butyricimonas\\textbar{}LLG\\_output\\textbar{}checkM\\textbar{}1\\textbar{}checkm\\textbar{}markers\\_qa\\_summary.tsv.1 & markers\\_qa\\_summary.tsv.1 & NA & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/Butyricimonas/LLG\\_output/checkM/1/checkm/markers\\_qa\\_summary.tsv.1\\\\\n",
       "\t GCA\\_001915615.1\\_ASM191561v1\\_genomic & k\\_\\_Bacteria (UID2569) & 434 & 278 & 186 &  88.69 & 0.27 & 0 & 4167409 & 2477 & ⋯ & 44 & 233 & 1 & 0 & 0 & 0 & \\textbar{}ebio\\textbar{}abt3\\_projects\\textbar{}software\\textbar{}dev\\textbar{}ll\\_pipelines\\textbar{}llprimer\\textbar{}experiments\\textbar{}HMP\\_most-wanted\\textbar{}Butyricimonas\\textbar{}LLG\\_output\\textbar{}checkM\\textbar{}1\\textbar{}checkm\\textbar{}markers\\_qa\\_summary.tsv.2 & markers\\_qa\\_summary.tsv.2 & NA & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/Butyricimonas/LLG\\_output/checkM/1/checkm/markers\\_qa\\_summary.tsv.2\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋱ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t GCA\\_900184685.1\\_PRJEB20857\\_genomic      & k\\_\\_Bacteria (UID2569) & 434 & 278 & 186 & 100.00 & 0 & 0 & 5300003 &   0 & ⋯ & 0 & 278 & 0 & 0 & 0 & 0 & \\textbar{}ebio\\textbar{}abt3\\_projects\\textbar{}software\\textbar{}dev\\textbar{}ll\\_pipelines\\textbar{}llprimer\\textbar{}experiments\\textbar{}HMP\\_most-wanted\\textbar{}Butyricimonas\\textbar{}LLG\\_output\\textbar{}checkM\\textbar{}2\\textbar{}checkm\\textbar{}markers\\_qa\\_summary.tsv.13 & markers\\_qa\\_summary.tsv.13 & NA & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/Butyricimonas/LLG\\_output/checkM/2/checkm/markers\\_qa\\_summary.tsv.13\\\\\n",
       "\t GCA\\_902364335.1\\_MGYG-HGUT-00203\\_genomic & k\\_\\_Bacteria (UID2569) & 434 & 278 & 186 &  99.46 & 0 & 0 & 4537596 & 250 & ⋯ & 1 & 277 & 0 & 0 & 0 & 0 & \\textbar{}ebio\\textbar{}abt3\\_projects\\textbar{}software\\textbar{}dev\\textbar{}ll\\_pipelines\\textbar{}llprimer\\textbar{}experiments\\textbar{}HMP\\_most-wanted\\textbar{}Butyricimonas\\textbar{}LLG\\_output\\textbar{}checkM\\textbar{}2\\textbar{}checkm\\textbar{}markers\\_qa\\_summary.tsv.14 & markers\\_qa\\_summary.tsv.14 & NA & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/Butyricimonas/LLG\\_output/checkM/2/checkm/markers\\_qa\\_summary.tsv.14\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 29 × 33\n",
       "\n",
       "| Bin.Id &lt;fct&gt; | Marker.lineage &lt;fct&gt; | X..genomes &lt;int&gt; | X..markers &lt;int&gt; | X..marker.sets &lt;int&gt; | Completeness &lt;dbl&gt; | Contamination &lt;dbl&gt; | Strain.heterogeneity &lt;int&gt; | Genome.size..bp. &lt;int&gt; | X..ambiguous.bases &lt;int&gt; | ⋯ ⋯ | X0 &lt;int&gt; | X1 &lt;int&gt; | X2 &lt;int&gt; | X3 &lt;int&gt; | X4 &lt;int&gt; | X5. &lt;int&gt; | assembly.Id &lt;fct&gt; | assembler.Id &lt;fct&gt; | taxon.Id &lt;lgl&gt; | File &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| GCA_000379665.1_ASM37966v1_genomic  | k__Bacteria (UID2569) | 434 | 278 | 186 | 100.00 | 0.00 | 0 | 4770838 |  131 | ⋯ |  0 | 278 | 0 | 0 | 0 | 0 | |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Butyricimonas|LLG_output|checkM|1|checkm|markers_qa_summary.tsv.1 | markers_qa_summary.tsv.1 | NA | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/LLG_output/checkM/1/checkm/markers_qa_summary.tsv.1 |\n",
       "| GCA_001915615.1_ASM191561v1_genomic | k__Bacteria (UID2569) | 434 | 278 | 186 |  88.69 | 0.27 | 0 | 4167409 | 2477 | ⋯ | 44 | 233 | 1 | 0 | 0 | 0 | |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Butyricimonas|LLG_output|checkM|1|checkm|markers_qa_summary.tsv.2 | markers_qa_summary.tsv.2 | NA | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/LLG_output/checkM/1/checkm/markers_qa_summary.tsv.2 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋱ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| GCA_900184685.1_PRJEB20857_genomic      | k__Bacteria (UID2569) | 434 | 278 | 186 | 100.00 | 0 | 0 | 5300003 |   0 | ⋯ | 0 | 278 | 0 | 0 | 0 | 0 | |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Butyricimonas|LLG_output|checkM|2|checkm|markers_qa_summary.tsv.13 | markers_qa_summary.tsv.13 | NA | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/LLG_output/checkM/2/checkm/markers_qa_summary.tsv.13 |\n",
       "| GCA_902364335.1_MGYG-HGUT-00203_genomic | k__Bacteria (UID2569) | 434 | 278 | 186 |  99.46 | 0 | 0 | 4537596 | 250 | ⋯ | 1 | 277 | 0 | 0 | 0 | 0 | |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Butyricimonas|LLG_output|checkM|2|checkm|markers_qa_summary.tsv.14 | markers_qa_summary.tsv.14 | NA | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/LLG_output/checkM/2/checkm/markers_qa_summary.tsv.14 |\n",
       "\n"
      ],
      "text/plain": [
       "   Bin.Id                                  Marker.lineage        X..genomes\n",
       "1  GCA_000379665.1_ASM37966v1_genomic      k__Bacteria (UID2569) 434       \n",
       "2  GCA_001915615.1_ASM191561v1_genomic     k__Bacteria (UID2569) 434       \n",
       "⋮  ⋮                                       ⋮                     ⋮         \n",
       "28 GCA_900184685.1_PRJEB20857_genomic      k__Bacteria (UID2569) 434       \n",
       "29 GCA_902364335.1_MGYG-HGUT-00203_genomic k__Bacteria (UID2569) 434       \n",
       "   X..markers X..marker.sets Completeness Contamination Strain.heterogeneity\n",
       "1  278        186            100.00       0.00          0                   \n",
       "2  278        186             88.69       0.27          0                   \n",
       "⋮  ⋮          ⋮              ⋮            ⋮             ⋮                   \n",
       "28 278        186            100.00       0             0                   \n",
       "29 278        186             99.46       0             0                   \n",
       "   Genome.size..bp. X..ambiguous.bases ⋯ X0 X1  X2 X3 X4 X5.\n",
       "1  4770838           131               ⋯  0 278 0  0  0  0  \n",
       "2  4167409          2477               ⋯ 44 233 1  0  0  0  \n",
       "⋮  ⋮                ⋮                  ⋱ ⋮  ⋮   ⋮  ⋮  ⋮  ⋮  \n",
       "28 5300003            0                ⋯ 0  278 0  0  0  0  \n",
       "29 4537596          250                ⋯ 1  277 0  0  0  0  \n",
       "   assembly.Id                                                                                                                                          \n",
       "1  |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Butyricimonas|LLG_output|checkM|1|checkm|markers_qa_summary.tsv.1 \n",
       "2  |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Butyricimonas|LLG_output|checkM|1|checkm|markers_qa_summary.tsv.2 \n",
       "⋮  ⋮                                                                                                                                                    \n",
       "28 |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Butyricimonas|LLG_output|checkM|2|checkm|markers_qa_summary.tsv.13\n",
       "29 |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Butyricimonas|LLG_output|checkM|2|checkm|markers_qa_summary.tsv.14\n",
       "   assembler.Id              taxon.Id\n",
       "1  markers_qa_summary.tsv.1  NA      \n",
       "2  markers_qa_summary.tsv.2  NA      \n",
       "⋮  ⋮                         ⋮       \n",
       "28 markers_qa_summary.tsv.13 NA      \n",
       "29 markers_qa_summary.tsv.14 NA      \n",
       "   File                                                                                                                                                 \n",
       "1  /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/LLG_output/checkM/1/checkm/markers_qa_summary.tsv.1 \n",
       "2  /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/LLG_output/checkM/1/checkm/markers_qa_summary.tsv.2 \n",
       "⋮  ⋮                                                                                                                                                    \n",
       "28 /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/LLG_output/checkM/2/checkm/markers_qa_summary.tsv.13\n",
       "29 /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/LLG_output/checkM/2/checkm/markers_qa_summary.tsv.14"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# checkM summary\n",
    "checkm = file.path(work_dir, 'LLG_output', 'checkM', 'checkm_qa_summary.tsv') %>%\n",
    "    read.delim(sep='\\t') \n",
    "checkm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-13T14:43:19.329651Z",
     "iopub.status.busy": "2021-01-13T14:43:19.327502Z",
     "iopub.status.idle": "2021-01-13T14:43:19.385980Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 29 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>genome</th><th scope=col>completeness</th><th scope=col>contamination</th><th scope=col>Bin.Id</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000379665.1_ASM37966v1_genomic.fna </td><td>100.00</td><td>0.00</td><td>GCA_000379665.1_ASM37966v1_genomic </td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_001915615.1_ASM191561v1_genomic.fna</td><td> 88.69</td><td>0.27</td><td>GCA_001915615.1_ASM191561v1_genomic</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_900184685.1_PRJEB20857_genomic.fna     </td><td>100.00</td><td>0</td><td>GCA_900184685.1_PRJEB20857_genomic     </td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_902364335.1_MGYG-HGUT-00203_genomic.fna</td><td> 99.46</td><td>0</td><td>GCA_902364335.1_MGYG-HGUT-00203_genomic</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 29 × 4\n",
       "\\begin{tabular}{llll}\n",
       " genome & completeness & contamination & Bin.Id\\\\\n",
       " <fct> & <dbl> & <dbl> & <chr>\\\\\n",
       "\\hline\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_000379665.1\\_ASM37966v1\\_genomic.fna  & 100.00 & 0.00 & GCA\\_000379665.1\\_ASM37966v1\\_genomic \\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_001915615.1\\_ASM191561v1\\_genomic.fna &  88.69 & 0.27 & GCA\\_001915615.1\\_ASM191561v1\\_genomic\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_900184685.1\\_PRJEB20857\\_genomic.fna      & 100.00 & 0 & GCA\\_900184685.1\\_PRJEB20857\\_genomic     \\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_902364335.1\\_MGYG-HGUT-00203\\_genomic.fna &  99.46 & 0 & GCA\\_902364335.1\\_MGYG-HGUT-00203\\_genomic\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 29 × 4\n",
       "\n",
       "| genome &lt;fct&gt; | completeness &lt;dbl&gt; | contamination &lt;dbl&gt; | Bin.Id &lt;chr&gt; |\n",
       "|---|---|---|---|\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000379665.1_ASM37966v1_genomic.fna  | 100.00 | 0.00 | GCA_000379665.1_ASM37966v1_genomic  |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_001915615.1_ASM191561v1_genomic.fna |  88.69 | 0.27 | GCA_001915615.1_ASM191561v1_genomic |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_900184685.1_PRJEB20857_genomic.fna      | 100.00 | 0 | GCA_900184685.1_PRJEB20857_genomic      |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_902364335.1_MGYG-HGUT-00203_genomic.fna |  99.46 | 0 | GCA_902364335.1_MGYG-HGUT-00203_genomic |\n",
       "\n"
      ],
      "text/plain": [
       "   genome                                                                                           \n",
       "1  /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000379665.1_ASM37966v1_genomic.fna     \n",
       "2  /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_001915615.1_ASM191561v1_genomic.fna    \n",
       "⋮  ⋮                                                                                                \n",
       "28 /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_900184685.1_PRJEB20857_genomic.fna     \n",
       "29 /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_902364335.1_MGYG-HGUT-00203_genomic.fna\n",
       "   completeness contamination Bin.Id                                 \n",
       "1  100.00       0.00          GCA_000379665.1_ASM37966v1_genomic     \n",
       "2   88.69       0.27          GCA_001915615.1_ASM191561v1_genomic    \n",
       "⋮  ⋮            ⋮             ⋮                                      \n",
       "28 100.00       0             GCA_900184685.1_PRJEB20857_genomic     \n",
       "29  99.46       0             GCA_902364335.1_MGYG-HGUT-00203_genomic"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# dRep summary\n",
    "drep = file.path(work_dir, 'LLG_output', 'drep', 'checkm_markers_qa_summary.tsv') %>%\n",
    "    read.delim(sep='\\t') %>%\n",
    "    mutate(Bin.Id = gsub('.+/', '', genome),\n",
    "           Bin.Id = gsub('\\\\.fna$', '', Bin.Id))\n",
    "drep"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-13T14:43:19.890745Z",
     "iopub.status.busy": "2021-01-13T14:43:19.888534Z",
     "iopub.status.idle": "2021-01-13T14:43:19.925313Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 17 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Name</th><th scope=col>Fasta</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>GCA_003991565.1_ASM399156v1_genomic</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/LLG_output/drep/drep/dereplicated_genomes/GCA_003991565.1_ASM399156v1_genomic.fna</td></tr>\n",
       "\t<tr><td>GCA_900258545.1_PRJEB22910_genomic </td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/LLG_output/drep/drep/dereplicated_genomes/GCA_900258545.1_PRJEB22910_genomic.fna </td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>GCA_014287455.1_ASM1428745v1_genomic</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/LLG_output/drep/drep/dereplicated_genomes/GCA_014287455.1_ASM1428745v1_genomic.fna</td></tr>\n",
       "\t<tr><td>GCA_900184685.1_PRJEB20857_genomic  </td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/LLG_output/drep/drep/dereplicated_genomes/GCA_900184685.1_PRJEB20857_genomic.fna  </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 17 × 2\n",
       "\\begin{tabular}{ll}\n",
       " Name & Fasta\\\\\n",
       " <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\t GCA\\_003991565.1\\_ASM399156v1\\_genomic & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/Butyricimonas/LLG\\_output/drep/drep/dereplicated\\_genomes/GCA\\_003991565.1\\_ASM399156v1\\_genomic.fna\\\\\n",
       "\t GCA\\_900258545.1\\_PRJEB22910\\_genomic  & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/Butyricimonas/LLG\\_output/drep/drep/dereplicated\\_genomes/GCA\\_900258545.1\\_PRJEB22910\\_genomic.fna \\\\\n",
       "\t ⋮ & ⋮\\\\\n",
       "\t GCA\\_014287455.1\\_ASM1428745v1\\_genomic & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/Butyricimonas/LLG\\_output/drep/drep/dereplicated\\_genomes/GCA\\_014287455.1\\_ASM1428745v1\\_genomic.fna\\\\\n",
       "\t GCA\\_900184685.1\\_PRJEB20857\\_genomic   & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/Butyricimonas/LLG\\_output/drep/drep/dereplicated\\_genomes/GCA\\_900184685.1\\_PRJEB20857\\_genomic.fna  \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 17 × 2\n",
       "\n",
       "| Name &lt;fct&gt; | Fasta &lt;fct&gt; |\n",
       "|---|---|\n",
       "| GCA_003991565.1_ASM399156v1_genomic | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/LLG_output/drep/drep/dereplicated_genomes/GCA_003991565.1_ASM399156v1_genomic.fna |\n",
       "| GCA_900258545.1_PRJEB22910_genomic  | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/LLG_output/drep/drep/dereplicated_genomes/GCA_900258545.1_PRJEB22910_genomic.fna  |\n",
       "| ⋮ | ⋮ |\n",
       "| GCA_014287455.1_ASM1428745v1_genomic | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/LLG_output/drep/drep/dereplicated_genomes/GCA_014287455.1_ASM1428745v1_genomic.fna |\n",
       "| GCA_900184685.1_PRJEB20857_genomic   | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/LLG_output/drep/drep/dereplicated_genomes/GCA_900184685.1_PRJEB20857_genomic.fna   |\n",
       "\n"
      ],
      "text/plain": [
       "   Name                                \n",
       "1  GCA_003991565.1_ASM399156v1_genomic \n",
       "2  GCA_900258545.1_PRJEB22910_genomic  \n",
       "⋮  ⋮                                   \n",
       "16 GCA_014287455.1_ASM1428745v1_genomic\n",
       "17 GCA_900184685.1_PRJEB20857_genomic  \n",
       "   Fasta                                                                                                                                                                              \n",
       "1  /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/LLG_output/drep/drep/dereplicated_genomes/GCA_003991565.1_ASM399156v1_genomic.fna \n",
       "2  /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/LLG_output/drep/drep/dereplicated_genomes/GCA_900258545.1_PRJEB22910_genomic.fna  \n",
       "⋮  ⋮                                                                                                                                                                                  \n",
       "16 /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/LLG_output/drep/drep/dereplicated_genomes/GCA_014287455.1_ASM1428745v1_genomic.fna\n",
       "17 /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/LLG_output/drep/drep/dereplicated_genomes/GCA_900184685.1_PRJEB20857_genomic.fna  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# de-replicated genomes\n",
    "drep_gen = file.path(work_dir, 'LLG_output', 'drep', 'dereplicated_genomes.tsv') %>%\n",
    "    read.delim(sep='\\t')\n",
    "drep_gen"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-13T14:43:20.531024Z",
     "iopub.status.busy": "2021-01-13T14:43:20.528622Z",
     "iopub.status.idle": "2021-01-13T14:43:20.749210Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 29 × 22</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>user_genome</th><th scope=col>Domain</th><th scope=col>Phylum</th><th scope=col>Class</th><th scope=col>Order</th><th scope=col>Family</th><th scope=col>Genus</th><th scope=col>Species</th><th scope=col>fastani_reference</th><th scope=col>fastani_reference_radius</th><th scope=col>⋯</th><th scope=col>fastani_af</th><th scope=col>closest_placement_reference</th><th scope=col>closest_placement_radius</th><th scope=col>closest_placement_taxonomy</th><th scope=col>closest_placement_ani</th><th scope=col>closest_placement_af</th><th scope=col>msa_percent</th><th scope=col>translation_table</th><th scope=col>red_value</th><th scope=col>warnings</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>GCA_000379665.1_ASM37966v1_genomic </td><td>d__Bacteria</td><td>p__Bacteroidota</td><td>c__Bacteroidia</td><td>o__Bacteroidales</td><td>f__Marinifilaceae</td><td>g__Butyricimonas</td><td>s__Butyricimonas synergistica</td><td>GCF_000379665.1</td><td>95</td><td>⋯</td><td>1.00</td><td>GCF_000379665.1</td><td>95</td><td>d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas synergistica</td><td>100.00</td><td>1.00</td><td>96.57</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>GCA_001915615.1_ASM191561v1_genomic</td><td>d__Bacteria</td><td>p__Bacteroidota</td><td>c__Bacteroidia</td><td>o__Bacteroidales</td><td>f__Marinifilaceae</td><td>g__Butyricimonas</td><td>s__Butyricimonas faecihominis</td><td>GCF_003851945.2</td><td>95</td><td>⋯</td><td>0.78</td><td>GCF_003851945.2</td><td>95</td><td>d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas faecihominis</td><td> 98.09</td><td>0.78</td><td>84.37</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋱</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>GCA_900184685.1_PRJEB20857_genomic     </td><td>d__Bacteria</td><td>p__Bacteroidota</td><td>c__Bacteroidia</td><td>o__Bacteroidales</td><td>f__Marinifilaceae</td><td>g__Butyricimonas</td><td>s__Butyricimonas sp900184685</td><td>GCF_900184685.1</td><td>95</td><td>⋯</td><td>1.00</td><td>GCF_900184685.1</td><td>95</td><td>d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas sp900184685</td><td>100.00</td><td>1.00</td><td>96.71</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>GCA_902364335.1_MGYG-HGUT-00203_genomic</td><td>d__Bacteria</td><td>p__Bacteroidota</td><td>c__Bacteroidia</td><td>o__Bacteroidales</td><td>f__Marinifilaceae</td><td>g__Butyricimonas</td><td>s__Butyricimonas virosa     </td><td>GCF_000519105.1</td><td>95</td><td>⋯</td><td>0.85</td><td>GCF_000519105.1</td><td>95</td><td>d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas virosa     </td><td> 97.98</td><td>0.85</td><td>96.79</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 29 × 22\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " user\\_genome & Domain & Phylum & Class & Order & Family & Genus & Species & fastani\\_reference & fastani\\_reference\\_radius & ⋯ & fastani\\_af & closest\\_placement\\_reference & closest\\_placement\\_radius & closest\\_placement\\_taxonomy & closest\\_placement\\_ani & closest\\_placement\\_af & msa\\_percent & translation\\_table & red\\_value & warnings\\\\\n",
       " <fct> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <fct> & <dbl> & ⋯ & <dbl> & <fct> & <dbl> & <fct> & <dbl> & <dbl> & <dbl> & <int> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\t GCA\\_000379665.1\\_ASM37966v1\\_genomic  & d\\_\\_Bacteria & p\\_\\_Bacteroidota & c\\_\\_Bacteroidia & o\\_\\_Bacteroidales & f\\_\\_Marinifilaceae & g\\_\\_Butyricimonas & s\\_\\_Butyricimonas synergistica & GCF\\_000379665.1 & 95 & ⋯ & 1.00 & GCF\\_000379665.1 & 95 & d\\_\\_Bacteria;p\\_\\_Bacteroidota;c\\_\\_Bacteroidia;o\\_\\_Bacteroidales;f\\_\\_Marinifilaceae;g\\_\\_Butyricimonas;s\\_\\_Butyricimonas synergistica & 100.00 & 1.00 & 96.57 & 11 & N/A & N/A\\\\\n",
       "\t GCA\\_001915615.1\\_ASM191561v1\\_genomic & d\\_\\_Bacteria & p\\_\\_Bacteroidota & c\\_\\_Bacteroidia & o\\_\\_Bacteroidales & f\\_\\_Marinifilaceae & g\\_\\_Butyricimonas & s\\_\\_Butyricimonas faecihominis & GCF\\_003851945.2 & 95 & ⋯ & 0.78 & GCF\\_003851945.2 & 95 & d\\_\\_Bacteria;p\\_\\_Bacteroidota;c\\_\\_Bacteroidia;o\\_\\_Bacteroidales;f\\_\\_Marinifilaceae;g\\_\\_Butyricimonas;s\\_\\_Butyricimonas faecihominis &  98.09 & 0.78 & 84.37 & 11 & N/A & N/A\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋱ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t GCA\\_900184685.1\\_PRJEB20857\\_genomic      & d\\_\\_Bacteria & p\\_\\_Bacteroidota & c\\_\\_Bacteroidia & o\\_\\_Bacteroidales & f\\_\\_Marinifilaceae & g\\_\\_Butyricimonas & s\\_\\_Butyricimonas sp900184685 & GCF\\_900184685.1 & 95 & ⋯ & 1.00 & GCF\\_900184685.1 & 95 & d\\_\\_Bacteria;p\\_\\_Bacteroidota;c\\_\\_Bacteroidia;o\\_\\_Bacteroidales;f\\_\\_Marinifilaceae;g\\_\\_Butyricimonas;s\\_\\_Butyricimonas sp900184685 & 100.00 & 1.00 & 96.71 & 11 & N/A & N/A\\\\\n",
       "\t GCA\\_902364335.1\\_MGYG-HGUT-00203\\_genomic & d\\_\\_Bacteria & p\\_\\_Bacteroidota & c\\_\\_Bacteroidia & o\\_\\_Bacteroidales & f\\_\\_Marinifilaceae & g\\_\\_Butyricimonas & s\\_\\_Butyricimonas virosa      & GCF\\_000519105.1 & 95 & ⋯ & 0.85 & GCF\\_000519105.1 & 95 & d\\_\\_Bacteria;p\\_\\_Bacteroidota;c\\_\\_Bacteroidia;o\\_\\_Bacteroidales;f\\_\\_Marinifilaceae;g\\_\\_Butyricimonas;s\\_\\_Butyricimonas virosa      &  97.98 & 0.85 & 96.79 & 11 & N/A & N/A\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 29 × 22\n",
       "\n",
       "| user_genome &lt;fct&gt; | Domain &lt;chr&gt; | Phylum &lt;chr&gt; | Class &lt;chr&gt; | Order &lt;chr&gt; | Family &lt;chr&gt; | Genus &lt;chr&gt; | Species &lt;chr&gt; | fastani_reference &lt;fct&gt; | fastani_reference_radius &lt;dbl&gt; | ⋯ ⋯ | fastani_af &lt;dbl&gt; | closest_placement_reference &lt;fct&gt; | closest_placement_radius &lt;dbl&gt; | closest_placement_taxonomy &lt;fct&gt; | closest_placement_ani &lt;dbl&gt; | closest_placement_af &lt;dbl&gt; | msa_percent &lt;dbl&gt; | translation_table &lt;int&gt; | red_value &lt;fct&gt; | warnings &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| GCA_000379665.1_ASM37966v1_genomic  | d__Bacteria | p__Bacteroidota | c__Bacteroidia | o__Bacteroidales | f__Marinifilaceae | g__Butyricimonas | s__Butyricimonas synergistica | GCF_000379665.1 | 95 | ⋯ | 1.00 | GCF_000379665.1 | 95 | d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas synergistica | 100.00 | 1.00 | 96.57 | 11 | N/A | N/A |\n",
       "| GCA_001915615.1_ASM191561v1_genomic | d__Bacteria | p__Bacteroidota | c__Bacteroidia | o__Bacteroidales | f__Marinifilaceae | g__Butyricimonas | s__Butyricimonas faecihominis | GCF_003851945.2 | 95 | ⋯ | 0.78 | GCF_003851945.2 | 95 | d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas faecihominis |  98.09 | 0.78 | 84.37 | 11 | N/A | N/A |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋱ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| GCA_900184685.1_PRJEB20857_genomic      | d__Bacteria | p__Bacteroidota | c__Bacteroidia | o__Bacteroidales | f__Marinifilaceae | g__Butyricimonas | s__Butyricimonas sp900184685 | GCF_900184685.1 | 95 | ⋯ | 1.00 | GCF_900184685.1 | 95 | d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas sp900184685 | 100.00 | 1.00 | 96.71 | 11 | N/A | N/A |\n",
       "| GCA_902364335.1_MGYG-HGUT-00203_genomic | d__Bacteria | p__Bacteroidota | c__Bacteroidia | o__Bacteroidales | f__Marinifilaceae | g__Butyricimonas | s__Butyricimonas virosa      | GCF_000519105.1 | 95 | ⋯ | 0.85 | GCF_000519105.1 | 95 | d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas virosa      |  97.98 | 0.85 | 96.79 | 11 | N/A | N/A |\n",
       "\n"
      ],
      "text/plain": [
       "   user_genome                             Domain      Phylum         \n",
       "1  GCA_000379665.1_ASM37966v1_genomic      d__Bacteria p__Bacteroidota\n",
       "2  GCA_001915615.1_ASM191561v1_genomic     d__Bacteria p__Bacteroidota\n",
       "⋮  ⋮                                       ⋮           ⋮              \n",
       "28 GCA_900184685.1_PRJEB20857_genomic      d__Bacteria p__Bacteroidota\n",
       "29 GCA_902364335.1_MGYG-HGUT-00203_genomic d__Bacteria p__Bacteroidota\n",
       "   Class          Order            Family            Genus           \n",
       "1  c__Bacteroidia o__Bacteroidales f__Marinifilaceae g__Butyricimonas\n",
       "2  c__Bacteroidia o__Bacteroidales f__Marinifilaceae g__Butyricimonas\n",
       "⋮  ⋮              ⋮                ⋮                 ⋮               \n",
       "28 c__Bacteroidia o__Bacteroidales f__Marinifilaceae g__Butyricimonas\n",
       "29 c__Bacteroidia o__Bacteroidales f__Marinifilaceae g__Butyricimonas\n",
       "   Species                       fastani_reference fastani_reference_radius ⋯\n",
       "1  s__Butyricimonas synergistica GCF_000379665.1   95                       ⋯\n",
       "2  s__Butyricimonas faecihominis GCF_003851945.2   95                       ⋯\n",
       "⋮  ⋮                             ⋮                 ⋮                        ⋱\n",
       "28 s__Butyricimonas sp900184685  GCF_900184685.1   95                       ⋯\n",
       "29 s__Butyricimonas virosa       GCF_000519105.1   95                       ⋯\n",
       "   fastani_af closest_placement_reference closest_placement_radius\n",
       "1  1.00       GCF_000379665.1             95                      \n",
       "2  0.78       GCF_003851945.2             95                      \n",
       "⋮  ⋮          ⋮                           ⋮                       \n",
       "28 1.00       GCF_900184685.1             95                      \n",
       "29 0.85       GCF_000519105.1             95                      \n",
       "   closest_placement_taxonomy                                                                                                  \n",
       "1  d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas synergistica\n",
       "2  d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas faecihominis\n",
       "⋮  ⋮                                                                                                                           \n",
       "28 d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas sp900184685 \n",
       "29 d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas virosa      \n",
       "   closest_placement_ani closest_placement_af msa_percent translation_table\n",
       "1  100.00                1.00                 96.57       11               \n",
       "2   98.09                0.78                 84.37       11               \n",
       "⋮  ⋮                     ⋮                    ⋮           ⋮                \n",
       "28 100.00                1.00                 96.71       11               \n",
       "29  97.98                0.85                 96.79       11               \n",
       "   red_value warnings\n",
       "1  N/A       N/A     \n",
       "2  N/A       N/A     \n",
       "⋮  ⋮         ⋮       \n",
       "28 N/A       N/A     \n",
       "29 N/A       N/A     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# GTDBTk summary\n",
    "tax = file.path(work_dir, 'LLG_output', 'gtdbtk', 'gtdbtk_bac_summary.tsv') %>%\n",
    "    read.delim(, sep='\\t') %>%\n",
    "    separate(classification, \n",
    "             c('Domain', 'Phylum', 'Class', 'Order', 'Family', 'Genus', 'Species'),\n",
    "             sep=';') %>%\n",
    "    select(-note, -classification_method, -pplacer_taxonomy,\n",
    "           -other_related_references.genome_id.species_name.radius.ANI.AF.)\n",
    "tax"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-13T14:43:21.314991Z",
     "iopub.status.busy": "2021-01-13T14:43:21.313024Z",
     "iopub.status.idle": "2021-01-13T14:43:21.352154Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- drep --\n",
      "intersect(x,y): 17 \n",
      "setdiff(x,y): 0 \n",
      "setdiff(y,x): 12 \n",
      "union(x,y): 29 \n",
      "-- checkm --\n",
      "intersect(x,y): 29 \n",
      "setdiff(x,y): 0 \n",
      "setdiff(y,x): 0 \n",
      "union(x,y): 29 \n",
      "-- gtdbtk --\n",
      "intersect(x,y): 29 \n",
      "setdiff(x,y): 0 \n",
      "setdiff(y,x): 0 \n",
      "union(x,y): 29 \n"
     ]
    }
   ],
   "source": [
    "# checking overlap\n",
    "cat('-- drep --\\n')\n",
    "overlap(basename(as.character(drep_gen$Fasta)), \n",
    "        basename(as.character(drep$genome)))\n",
    "cat('-- checkm --\\n')\n",
    "overlap(drep$Bin.Id, checkm$Bin.Id)\n",
    "cat('-- gtdbtk --\\n')\n",
    "overlap(drep$Bin.Id, tax$user_genome)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-13T14:43:21.834030Z",
     "iopub.status.busy": "2021-01-13T14:43:21.831780Z",
     "iopub.status.idle": "2021-01-13T14:43:21.919643Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 17 × 60</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>genome</th><th scope=col>completeness</th><th scope=col>contamination</th><th scope=col>Bin.Id</th><th scope=col>Marker.lineage</th><th scope=col>X..genomes</th><th scope=col>X..markers</th><th scope=col>X..marker.sets</th><th scope=col>Completeness</th><th scope=col>Contamination</th><th scope=col>⋯</th><th scope=col>fastani_af</th><th scope=col>closest_placement_reference</th><th scope=col>closest_placement_radius</th><th scope=col>closest_placement_taxonomy</th><th scope=col>closest_placement_ani</th><th scope=col>closest_placement_af</th><th scope=col>msa_percent</th><th scope=col>translation_table</th><th scope=col>red_value</th><th scope=col>warnings</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000379665.1_ASM37966v1_genomic.fna </td><td>100.00</td><td>0</td><td>GCA_000379665.1_ASM37966v1_genomic </td><td>k__Bacteria (UID2569)</td><td>434</td><td>278</td><td>186</td><td>100.00</td><td>0</td><td>⋯</td><td>1.00</td><td>GCF_000379665.1</td><td>95</td><td>d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas synergistica</td><td>100.00</td><td>1.00</td><td>96.57</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_002320175.1_ASM232017v1_genomic.fna</td><td> 98.39</td><td>0</td><td>GCA_002320175.1_ASM232017v1_genomic</td><td>k__Bacteria (UID2569)</td><td>434</td><td>278</td><td>186</td><td> 98.39</td><td>0</td><td>⋯</td><td>0.87</td><td>GCF_000519105.1</td><td>95</td><td>d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas virosa      </td><td> 98.27</td><td>0.87</td><td>94.37</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋱</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_014287455.1_ASM1428745v1_genomic.fna</td><td>100</td><td>0</td><td>GCA_014287455.1_ASM1428745v1_genomic</td><td>k__Bacteria (UID2569)</td><td>434</td><td>278</td><td>186</td><td>100</td><td>0</td><td>⋯</td><td>0.81</td><td>GCF_900184685.1</td><td>95</td><td>d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas sp900184685</td><td> 97.68</td><td>0.81</td><td>96.73</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_900184685.1_PRJEB20857_genomic.fna  </td><td>100</td><td>0</td><td>GCA_900184685.1_PRJEB20857_genomic  </td><td>k__Bacteria (UID2569)</td><td>434</td><td>278</td><td>186</td><td>100</td><td>0</td><td>⋯</td><td>1.00</td><td>GCF_900184685.1</td><td>95</td><td>d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas sp900184685</td><td>100.00</td><td>1.00</td><td>96.71</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 17 × 60\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " genome & completeness & contamination & Bin.Id & Marker.lineage & X..genomes & X..markers & X..marker.sets & Completeness & Contamination & ⋯ & fastani\\_af & closest\\_placement\\_reference & closest\\_placement\\_radius & closest\\_placement\\_taxonomy & closest\\_placement\\_ani & closest\\_placement\\_af & msa\\_percent & translation\\_table & red\\_value & warnings\\\\\n",
       " <fct> & <dbl> & <dbl> & <chr> & <fct> & <int> & <int> & <int> & <dbl> & <dbl> & ⋯ & <dbl> & <fct> & <dbl> & <fct> & <dbl> & <dbl> & <dbl> & <int> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_000379665.1\\_ASM37966v1\\_genomic.fna  & 100.00 & 0 & GCA\\_000379665.1\\_ASM37966v1\\_genomic  & k\\_\\_Bacteria (UID2569) & 434 & 278 & 186 & 100.00 & 0 & ⋯ & 1.00 & GCF\\_000379665.1 & 95 & d\\_\\_Bacteria;p\\_\\_Bacteroidota;c\\_\\_Bacteroidia;o\\_\\_Bacteroidales;f\\_\\_Marinifilaceae;g\\_\\_Butyricimonas;s\\_\\_Butyricimonas synergistica & 100.00 & 1.00 & 96.57 & 11 & N/A & N/A\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_002320175.1\\_ASM232017v1\\_genomic.fna &  98.39 & 0 & GCA\\_002320175.1\\_ASM232017v1\\_genomic & k\\_\\_Bacteria (UID2569) & 434 & 278 & 186 &  98.39 & 0 & ⋯ & 0.87 & GCF\\_000519105.1 & 95 & d\\_\\_Bacteria;p\\_\\_Bacteroidota;c\\_\\_Bacteroidia;o\\_\\_Bacteroidales;f\\_\\_Marinifilaceae;g\\_\\_Butyricimonas;s\\_\\_Butyricimonas virosa       &  98.27 & 0.87 & 94.37 & 11 & N/A & N/A\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋱ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_014287455.1\\_ASM1428745v1\\_genomic.fna & 100 & 0 & GCA\\_014287455.1\\_ASM1428745v1\\_genomic & k\\_\\_Bacteria (UID2569) & 434 & 278 & 186 & 100 & 0 & ⋯ & 0.81 & GCF\\_900184685.1 & 95 & d\\_\\_Bacteria;p\\_\\_Bacteroidota;c\\_\\_Bacteroidia;o\\_\\_Bacteroidales;f\\_\\_Marinifilaceae;g\\_\\_Butyricimonas;s\\_\\_Butyricimonas sp900184685 &  97.68 & 0.81 & 96.73 & 11 & N/A & N/A\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_900184685.1\\_PRJEB20857\\_genomic.fna   & 100 & 0 & GCA\\_900184685.1\\_PRJEB20857\\_genomic   & k\\_\\_Bacteria (UID2569) & 434 & 278 & 186 & 100 & 0 & ⋯ & 1.00 & GCF\\_900184685.1 & 95 & d\\_\\_Bacteria;p\\_\\_Bacteroidota;c\\_\\_Bacteroidia;o\\_\\_Bacteroidales;f\\_\\_Marinifilaceae;g\\_\\_Butyricimonas;s\\_\\_Butyricimonas sp900184685 & 100.00 & 1.00 & 96.71 & 11 & N/A & N/A\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 17 × 60\n",
       "\n",
       "| genome &lt;fct&gt; | completeness &lt;dbl&gt; | contamination &lt;dbl&gt; | Bin.Id &lt;chr&gt; | Marker.lineage &lt;fct&gt; | X..genomes &lt;int&gt; | X..markers &lt;int&gt; | X..marker.sets &lt;int&gt; | Completeness &lt;dbl&gt; | Contamination &lt;dbl&gt; | ⋯ ⋯ | fastani_af &lt;dbl&gt; | closest_placement_reference &lt;fct&gt; | closest_placement_radius &lt;dbl&gt; | closest_placement_taxonomy &lt;fct&gt; | closest_placement_ani &lt;dbl&gt; | closest_placement_af &lt;dbl&gt; | msa_percent &lt;dbl&gt; | translation_table &lt;int&gt; | red_value &lt;fct&gt; | warnings &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000379665.1_ASM37966v1_genomic.fna  | 100.00 | 0 | GCA_000379665.1_ASM37966v1_genomic  | k__Bacteria (UID2569) | 434 | 278 | 186 | 100.00 | 0 | ⋯ | 1.00 | GCF_000379665.1 | 95 | d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas synergistica | 100.00 | 1.00 | 96.57 | 11 | N/A | N/A |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_002320175.1_ASM232017v1_genomic.fna |  98.39 | 0 | GCA_002320175.1_ASM232017v1_genomic | k__Bacteria (UID2569) | 434 | 278 | 186 |  98.39 | 0 | ⋯ | 0.87 | GCF_000519105.1 | 95 | d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas virosa       |  98.27 | 0.87 | 94.37 | 11 | N/A | N/A |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋱ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_014287455.1_ASM1428745v1_genomic.fna | 100 | 0 | GCA_014287455.1_ASM1428745v1_genomic | k__Bacteria (UID2569) | 434 | 278 | 186 | 100 | 0 | ⋯ | 0.81 | GCF_900184685.1 | 95 | d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas sp900184685 |  97.68 | 0.81 | 96.73 | 11 | N/A | N/A |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_900184685.1_PRJEB20857_genomic.fna   | 100 | 0 | GCA_900184685.1_PRJEB20857_genomic   | k__Bacteria (UID2569) | 434 | 278 | 186 | 100 | 0 | ⋯ | 1.00 | GCF_900184685.1 | 95 | d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas sp900184685 | 100.00 | 1.00 | 96.71 | 11 | N/A | N/A |\n",
       "\n"
      ],
      "text/plain": [
       "   genome                                                                                        \n",
       "1  /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000379665.1_ASM37966v1_genomic.fna  \n",
       "2  /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_002320175.1_ASM232017v1_genomic.fna \n",
       "⋮  ⋮                                                                                             \n",
       "16 /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_014287455.1_ASM1428745v1_genomic.fna\n",
       "17 /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_900184685.1_PRJEB20857_genomic.fna  \n",
       "   completeness contamination Bin.Id                              \n",
       "1  100.00       0             GCA_000379665.1_ASM37966v1_genomic  \n",
       "2   98.39       0             GCA_002320175.1_ASM232017v1_genomic \n",
       "⋮  ⋮            ⋮             ⋮                                   \n",
       "16 100          0             GCA_014287455.1_ASM1428745v1_genomic\n",
       "17 100          0             GCA_900184685.1_PRJEB20857_genomic  \n",
       "   Marker.lineage        X..genomes X..markers X..marker.sets Completeness\n",
       "1  k__Bacteria (UID2569) 434        278        186            100.00      \n",
       "2  k__Bacteria (UID2569) 434        278        186             98.39      \n",
       "⋮  ⋮                     ⋮          ⋮          ⋮              ⋮           \n",
       "16 k__Bacteria (UID2569) 434        278        186            100         \n",
       "17 k__Bacteria (UID2569) 434        278        186            100         \n",
       "   Contamination ⋯ fastani_af closest_placement_reference\n",
       "1  0             ⋯ 1.00       GCF_000379665.1            \n",
       "2  0             ⋯ 0.87       GCF_000519105.1            \n",
       "⋮  ⋮             ⋱ ⋮          ⋮                          \n",
       "16 0             ⋯ 0.81       GCF_900184685.1            \n",
       "17 0             ⋯ 1.00       GCF_900184685.1            \n",
       "   closest_placement_radius\n",
       "1  95                      \n",
       "2  95                      \n",
       "⋮  ⋮                       \n",
       "16 95                      \n",
       "17 95                      \n",
       "   closest_placement_taxonomy                                                                                                  \n",
       "1  d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas synergistica\n",
       "2  d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas virosa      \n",
       "⋮  ⋮                                                                                                                           \n",
       "16 d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas sp900184685 \n",
       "17 d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas sp900184685 \n",
       "   closest_placement_ani closest_placement_af msa_percent translation_table\n",
       "1  100.00                1.00                 96.57       11               \n",
       "2   98.27                0.87                 94.37       11               \n",
       "⋮  ⋮                     ⋮                    ⋮           ⋮                \n",
       "16  97.68                0.81                 96.73       11               \n",
       "17 100.00                1.00                 96.71       11               \n",
       "   red_value warnings\n",
       "1  N/A       N/A     \n",
       "2  N/A       N/A     \n",
       "⋮  ⋮         ⋮       \n",
       "16 N/A       N/A     \n",
       "17 N/A       N/A     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# joining based on Bin.Id\n",
    "drep = drep %>%\n",
    "    inner_join(checkm, c('Bin.Id')) %>%\n",
    "    mutate(GEN = genome %>% as.character %>% basename) %>%\n",
    "    inner_join(drep_gen %>% mutate(GEN = Fasta %>% as.character %>% basename),\n",
    "               by=c('GEN')) %>%\n",
    "    inner_join(tax, c('Bin.Id'='user_genome')) #%>%\n",
    "drep"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-13T14:43:22.506684Z",
     "iopub.status.busy": "2021-01-13T14:43:22.504459Z",
     "iopub.status.idle": "2021-01-13T14:43:22.551161Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 17 × 60</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>genome</th><th scope=col>completeness</th><th scope=col>contamination</th><th scope=col>Bin.Id</th><th scope=col>Marker.lineage</th><th scope=col>X..genomes</th><th scope=col>X..markers</th><th scope=col>X..marker.sets</th><th scope=col>Completeness</th><th scope=col>Contamination</th><th scope=col>⋯</th><th scope=col>fastani_af</th><th scope=col>closest_placement_reference</th><th scope=col>closest_placement_radius</th><th scope=col>closest_placement_taxonomy</th><th scope=col>closest_placement_ani</th><th scope=col>closest_placement_af</th><th scope=col>msa_percent</th><th scope=col>translation_table</th><th scope=col>red_value</th><th scope=col>warnings</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000379665.1_ASM37966v1_genomic.fna </td><td>100.00</td><td>0</td><td>GCA_000379665.1_ASM37966v1_genomic </td><td>k__Bacteria (UID2569)</td><td>434</td><td>278</td><td>186</td><td>100.00</td><td>0</td><td>⋯</td><td>1.00</td><td>GCF_000379665.1</td><td>95</td><td>d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas synergistica</td><td>100.00</td><td>1.00</td><td>96.57</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_002320175.1_ASM232017v1_genomic.fna</td><td> 98.39</td><td>0</td><td>GCA_002320175.1_ASM232017v1_genomic</td><td>k__Bacteria (UID2569)</td><td>434</td><td>278</td><td>186</td><td> 98.39</td><td>0</td><td>⋯</td><td>0.87</td><td>GCF_000519105.1</td><td>95</td><td>d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas virosa      </td><td> 98.27</td><td>0.87</td><td>94.37</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋱</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_014287455.1_ASM1428745v1_genomic.fna</td><td>100</td><td>0</td><td>GCA_014287455.1_ASM1428745v1_genomic</td><td>k__Bacteria (UID2569)</td><td>434</td><td>278</td><td>186</td><td>100</td><td>0</td><td>⋯</td><td>0.81</td><td>GCF_900184685.1</td><td>95</td><td>d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas sp900184685</td><td> 97.68</td><td>0.81</td><td>96.73</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_900184685.1_PRJEB20857_genomic.fna  </td><td>100</td><td>0</td><td>GCA_900184685.1_PRJEB20857_genomic  </td><td>k__Bacteria (UID2569)</td><td>434</td><td>278</td><td>186</td><td>100</td><td>0</td><td>⋯</td><td>1.00</td><td>GCF_900184685.1</td><td>95</td><td>d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas sp900184685</td><td>100.00</td><td>1.00</td><td>96.71</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 17 × 60\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " genome & completeness & contamination & Bin.Id & Marker.lineage & X..genomes & X..markers & X..marker.sets & Completeness & Contamination & ⋯ & fastani\\_af & closest\\_placement\\_reference & closest\\_placement\\_radius & closest\\_placement\\_taxonomy & closest\\_placement\\_ani & closest\\_placement\\_af & msa\\_percent & translation\\_table & red\\_value & warnings\\\\\n",
       " <fct> & <dbl> & <dbl> & <chr> & <fct> & <int> & <int> & <int> & <dbl> & <dbl> & ⋯ & <dbl> & <fct> & <dbl> & <fct> & <dbl> & <dbl> & <dbl> & <int> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_000379665.1\\_ASM37966v1\\_genomic.fna  & 100.00 & 0 & GCA\\_000379665.1\\_ASM37966v1\\_genomic  & k\\_\\_Bacteria (UID2569) & 434 & 278 & 186 & 100.00 & 0 & ⋯ & 1.00 & GCF\\_000379665.1 & 95 & d\\_\\_Bacteria;p\\_\\_Bacteroidota;c\\_\\_Bacteroidia;o\\_\\_Bacteroidales;f\\_\\_Marinifilaceae;g\\_\\_Butyricimonas;s\\_\\_Butyricimonas synergistica & 100.00 & 1.00 & 96.57 & 11 & N/A & N/A\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_002320175.1\\_ASM232017v1\\_genomic.fna &  98.39 & 0 & GCA\\_002320175.1\\_ASM232017v1\\_genomic & k\\_\\_Bacteria (UID2569) & 434 & 278 & 186 &  98.39 & 0 & ⋯ & 0.87 & GCF\\_000519105.1 & 95 & d\\_\\_Bacteria;p\\_\\_Bacteroidota;c\\_\\_Bacteroidia;o\\_\\_Bacteroidales;f\\_\\_Marinifilaceae;g\\_\\_Butyricimonas;s\\_\\_Butyricimonas virosa       &  98.27 & 0.87 & 94.37 & 11 & N/A & N/A\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋱ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_014287455.1\\_ASM1428745v1\\_genomic.fna & 100 & 0 & GCA\\_014287455.1\\_ASM1428745v1\\_genomic & k\\_\\_Bacteria (UID2569) & 434 & 278 & 186 & 100 & 0 & ⋯ & 0.81 & GCF\\_900184685.1 & 95 & d\\_\\_Bacteria;p\\_\\_Bacteroidota;c\\_\\_Bacteroidia;o\\_\\_Bacteroidales;f\\_\\_Marinifilaceae;g\\_\\_Butyricimonas;s\\_\\_Butyricimonas sp900184685 &  97.68 & 0.81 & 96.73 & 11 & N/A & N/A\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_900184685.1\\_PRJEB20857\\_genomic.fna   & 100 & 0 & GCA\\_900184685.1\\_PRJEB20857\\_genomic   & k\\_\\_Bacteria (UID2569) & 434 & 278 & 186 & 100 & 0 & ⋯ & 1.00 & GCF\\_900184685.1 & 95 & d\\_\\_Bacteria;p\\_\\_Bacteroidota;c\\_\\_Bacteroidia;o\\_\\_Bacteroidales;f\\_\\_Marinifilaceae;g\\_\\_Butyricimonas;s\\_\\_Butyricimonas sp900184685 & 100.00 & 1.00 & 96.71 & 11 & N/A & N/A\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 17 × 60\n",
       "\n",
       "| genome &lt;fct&gt; | completeness &lt;dbl&gt; | contamination &lt;dbl&gt; | Bin.Id &lt;chr&gt; | Marker.lineage &lt;fct&gt; | X..genomes &lt;int&gt; | X..markers &lt;int&gt; | X..marker.sets &lt;int&gt; | Completeness &lt;dbl&gt; | Contamination &lt;dbl&gt; | ⋯ ⋯ | fastani_af &lt;dbl&gt; | closest_placement_reference &lt;fct&gt; | closest_placement_radius &lt;dbl&gt; | closest_placement_taxonomy &lt;fct&gt; | closest_placement_ani &lt;dbl&gt; | closest_placement_af &lt;dbl&gt; | msa_percent &lt;dbl&gt; | translation_table &lt;int&gt; | red_value &lt;fct&gt; | warnings &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000379665.1_ASM37966v1_genomic.fna  | 100.00 | 0 | GCA_000379665.1_ASM37966v1_genomic  | k__Bacteria (UID2569) | 434 | 278 | 186 | 100.00 | 0 | ⋯ | 1.00 | GCF_000379665.1 | 95 | d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas synergistica | 100.00 | 1.00 | 96.57 | 11 | N/A | N/A |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_002320175.1_ASM232017v1_genomic.fna |  98.39 | 0 | GCA_002320175.1_ASM232017v1_genomic | k__Bacteria (UID2569) | 434 | 278 | 186 |  98.39 | 0 | ⋯ | 0.87 | GCF_000519105.1 | 95 | d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas virosa       |  98.27 | 0.87 | 94.37 | 11 | N/A | N/A |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋱ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_014287455.1_ASM1428745v1_genomic.fna | 100 | 0 | GCA_014287455.1_ASM1428745v1_genomic | k__Bacteria (UID2569) | 434 | 278 | 186 | 100 | 0 | ⋯ | 0.81 | GCF_900184685.1 | 95 | d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas sp900184685 |  97.68 | 0.81 | 96.73 | 11 | N/A | N/A |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_900184685.1_PRJEB20857_genomic.fna   | 100 | 0 | GCA_900184685.1_PRJEB20857_genomic   | k__Bacteria (UID2569) | 434 | 278 | 186 | 100 | 0 | ⋯ | 1.00 | GCF_900184685.1 | 95 | d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas sp900184685 | 100.00 | 1.00 | 96.71 | 11 | N/A | N/A |\n",
       "\n"
      ],
      "text/plain": [
       "   genome                                                                                        \n",
       "1  /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000379665.1_ASM37966v1_genomic.fna  \n",
       "2  /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_002320175.1_ASM232017v1_genomic.fna \n",
       "⋮  ⋮                                                                                             \n",
       "16 /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_014287455.1_ASM1428745v1_genomic.fna\n",
       "17 /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_900184685.1_PRJEB20857_genomic.fna  \n",
       "   completeness contamination Bin.Id                              \n",
       "1  100.00       0             GCA_000379665.1_ASM37966v1_genomic  \n",
       "2   98.39       0             GCA_002320175.1_ASM232017v1_genomic \n",
       "⋮  ⋮            ⋮             ⋮                                   \n",
       "16 100          0             GCA_014287455.1_ASM1428745v1_genomic\n",
       "17 100          0             GCA_900184685.1_PRJEB20857_genomic  \n",
       "   Marker.lineage        X..genomes X..markers X..marker.sets Completeness\n",
       "1  k__Bacteria (UID2569) 434        278        186            100.00      \n",
       "2  k__Bacteria (UID2569) 434        278        186             98.39      \n",
       "⋮  ⋮                     ⋮          ⋮          ⋮              ⋮           \n",
       "16 k__Bacteria (UID2569) 434        278        186            100         \n",
       "17 k__Bacteria (UID2569) 434        278        186            100         \n",
       "   Contamination ⋯ fastani_af closest_placement_reference\n",
       "1  0             ⋯ 1.00       GCF_000379665.1            \n",
       "2  0             ⋯ 0.87       GCF_000519105.1            \n",
       "⋮  ⋮             ⋱ ⋮          ⋮                          \n",
       "16 0             ⋯ 0.81       GCF_900184685.1            \n",
       "17 0             ⋯ 1.00       GCF_900184685.1            \n",
       "   closest_placement_radius\n",
       "1  95                      \n",
       "2  95                      \n",
       "⋮  ⋮                       \n",
       "16 95                      \n",
       "17 95                      \n",
       "   closest_placement_taxonomy                                                                                                  \n",
       "1  d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas synergistica\n",
       "2  d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas virosa      \n",
       "⋮  ⋮                                                                                                                           \n",
       "16 d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas sp900184685 \n",
       "17 d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas sp900184685 \n",
       "   closest_placement_ani closest_placement_af msa_percent translation_table\n",
       "1  100.00                1.00                 96.57       11               \n",
       "2   98.27                0.87                 94.37       11               \n",
       "⋮  ⋮                     ⋮                    ⋮           ⋮                \n",
       "16  97.68                0.81                 96.73       11               \n",
       "17 100.00                1.00                 96.71       11               \n",
       "   red_value warnings\n",
       "1  N/A       N/A     \n",
       "2  N/A       N/A     \n",
       "⋮  ⋮         ⋮       \n",
       "16 N/A       N/A     \n",
       "17 N/A       N/A     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# filtering by quality\n",
    "hq_genomes = drep %>%\n",
    "    filter(completeness >= 90,\n",
    "           contamination < 5,\n",
    "           Strain.heterogeneity < 50)\n",
    "hq_genomes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-13T14:43:23.010076Z",
     "iopub.status.busy": "2021-01-13T14:43:23.007961Z",
     "iopub.status.idle": "2021-01-13T14:43:23.054654Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A tibble: 1 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Family</th><th scope=col>Genus</th><th scope=col>n_genomes</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>f__Marinifilaceae</td><td>g__Butyricimonas</td><td>17</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 1 × 3\n",
       "\\begin{tabular}{lll}\n",
       " Family & Genus & n\\_genomes\\\\\n",
       " <chr> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t f\\_\\_Marinifilaceae & g\\_\\_Butyricimonas & 17\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 1 × 3\n",
       "\n",
       "| Family &lt;chr&gt; | Genus &lt;chr&gt; | n_genomes &lt;int&gt; |\n",
       "|---|---|---|\n",
       "| f__Marinifilaceae | g__Butyricimonas | 17 |\n",
       "\n"
      ],
      "text/plain": [
       "  Family            Genus            n_genomes\n",
       "1 f__Marinifilaceae g__Butyricimonas 17       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# summarizing the taxonomy\n",
    "df.dims(20)\n",
    "hq_genomes %>%\n",
    "    group_by(Family, Genus) %>%\n",
    "    summarize(n_genomes = n(), .groups='drop')\n",
    "df.dims()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-13T14:43:24.514531Z",
     "iopub.status.busy": "2021-01-13T14:43:24.512543Z",
     "iopub.status.idle": "2021-01-13T14:43:24.542241Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "File written: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted//Butyricimonas/samples_genomes_hq.txt \n"
     ]
    }
   ],
   "source": [
    "# writing samples table for LLPRIMER\n",
    "outfile = file.path(work_dir, 'samples_genomes_hq.txt')\n",
    "hq_genomes %>%\n",
    "    select(Bin.Id, Fasta) %>%\n",
    "    rename('Taxon' = Bin.Id) %>%\n",
    "    mutate(Taxon = gsub('_chromosome.+', '', Taxon),\n",
    "           Taxon = gsub('_bin_.+', '', Taxon),\n",
    "           Taxon = gsub('_genomic', '', Taxon),\n",
    "           Taxon = gsub('_annotated_assembly', '', Taxon),\n",
    "           Taxid = taxid) %>%\n",
    "    write_table(outfile)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Primer design"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Config"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-17T07:44:06.406362Z",
     "iopub.status.busy": "2021-01-17T07:44:06.404355Z",
     "iopub.status.idle": "2021-01-17T07:44:06.436544Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#-- I/O --#\n",
      "samples_file: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/samples_genomes_hq.txt\n",
      "\n",
      "# output location\n",
      "output_dir: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/primers/\n",
      "\n",
      "# temporary file directory (your username will be added automatically)\n",
      "tmp_dir: /ebio/abt3_scratch/\n",
      "\n",
      "#-- software parameters --#\n",
      "# See the README for a description\n",
      "params:\n",
      "  ionice: -c 3\n",
      "  cgp:\n",
      "    prodigal: \"\"    \n",
      "    mmseqs:\n",
      "      method: cluster    # or linclust (faster)\n",
      "      run: --min-seq-id 0.8 -c 0.8\n",
      "    core_genes: --frac 1 --max-clusters 500\n",
      "    blastx: -evalue 1e-10 -max_target_seqs 3\n",
      "    blastx_nontarget: -evalue 1e-5 -max_target_seqs 30\n",
      "    align:\n",
      "      method: linsi\n",
      "      params: --auto --maxiterate 1000\n",
      "    primer3:\n",
      "      number: --num-primers 500\n",
      "      size: --opt-size 20 --min-size 18 --max-size 24\n",
      "      product: --opt-prod-size 150 --min-prod-size 100 --max-prod-size 200\n",
      "      Tm: --opt-tm 62 --min-tm 55 --max-tm 70 --max-tm-diff 1\n",
      "      PCR: --oligo-DNA 50 --salt-monovalent 50 --salt-divalent 1.5 --dNTPs 0.2\n",
      "      GC: --opt-gc 50 --min-gc 25 --max-gc 75\n",
      "      degeneracy: --max-degeneracy 128 --max-degeneracy-3prime 1 --window-3prime 5\n",
      "      internal_oligo: --make-oligo 0 --int-opt-size 20 --int-opt-tm 60\n",
      "    blastn: -max_target_seqs 500 -perc_identity 90\n",
      "    blast_filter: --perc-len 80 --min-amplicon-len 30 --max-amplicon-len 2000    \n",
      "\n",
      "#-- database --#\n",
      "databases:\n",
      "  blast:\n",
      "    nucl: /ebio/abt3_projects/databases_no-backup/NCBI_blastdb/nt\n",
      "    prot: /ebio/abt3_projects/databases_no-backup/NCBI_blastdb/nr\n",
      "    gene_names: /ebio/abt3_projects/databases_no-backup/NCBI_nr_geneIDs/2021-01/nr_accession-geneNames.pkl\n",
      "  taxonomy:\n",
      "    names: /ebio/abt3_projects/databases_no-backup/NCBI_taxonomy/2020-12-23/names.dmp\n",
      "    nodes: /ebio/abt3_projects/databases_no-backup/NCBI_taxonomy/2020-12-23/nodes.dmp\n",
      "\n",
      "#-- snakemake pipeline --#\n",
      "pipeline:\n",
      "  snakemake_folder: ./\n",
      "  script_folder: bin/scripts/\n",
      "  name: LLPRIMER"
     ]
    }
   ],
   "source": [
    "F = file.path(work_dir, 'primers', 'config.yaml')\n",
    "cat_file(F)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Run"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "(snakemake) @ rick:/ebio/abt3_projects/software/dev/ll_pipelines/llprimer\n",
    "$ screen -L -S llprimer-But ./snakemake_sge.sh /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/primers/config.yaml 50 -F\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-19T14:37:59.168527Z",
     "iopub.status.busy": "2021-01-19T14:37:59.166829Z",
     "iopub.status.idle": "2021-01-19T14:37:59.311799Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No. of unique primer sets: 495 \n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 4208 × 21</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>cluster_id</th><th scope=col>primer_set</th><th scope=col>amplicon_size_consensus</th><th scope=col>amplicon_size_avg</th><th scope=col>amplicon_size_sd</th><th scope=col>primer_id</th><th scope=col>primer_type</th><th scope=col>sequence</th><th scope=col>length</th><th scope=col>degeneracy</th><th scope=col>⋯</th><th scope=col>position_start</th><th scope=col>position_end</th><th scope=col>Tm_avg</th><th scope=col>Tm_sd</th><th scope=col>GC_avg</th><th scope=col>GC_sd</th><th scope=col>hairpin_avg</th><th scope=col>hairpin_sd</th><th scope=col>homodimer_avg</th><th scope=col>homodimer_sd</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>4</td><td>13</td><td>172</td><td>172</td><td>0</td><td>13f</td><td>PRIMER_LEFT </td><td>GAAYCARGGACAYWGCCACC</td><td>20</td><td>16</td><td>⋯</td><td>161</td><td>181</td><td>60.40519</td><td>1.951478</td><td>57.50000</td><td>4.330127</td><td>10.70543</td><td>18.661427</td><td>-23.02055</td><td>15.93829</td></tr>\n",
       "\t<tr><td>4</td><td>13</td><td>172</td><td>172</td><td>0</td><td>13r</td><td>PRIMER_RIGHT</td><td>MGCYTCDGCACCRCTATTCA</td><td>20</td><td>24</td><td>⋯</td><td>313</td><td>333</td><td>61.39287</td><td>2.160181</td><td>54.16667</td><td>4.930066</td><td>42.22950</td><td> 2.453775</td><td>-19.36984</td><td>24.02227</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋱</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>500</td><td>499</td><td>133</td><td>133</td><td>0</td><td>499f</td><td>PRIMER_LEFT </td><td>ACBCCMTGYGCBGCAAAAAT</td><td>20</td><td>36</td><td>⋯</td><td>573</td><td>593</td><td>63.11378</td><td>2.487889</td><td>51.66667</td><td>4.859127</td><td>31.85494</td><td>24.19495</td><td> 6.798857</td><td>10.52647</td></tr>\n",
       "\t<tr><td>500</td><td>499</td><td>133</td><td>133</td><td>0</td><td>499r</td><td>PRIMER_RIGHT</td><td>GTTGRTCCCGGTGTTGCGTG</td><td>20</td><td> 2</td><td>⋯</td><td>686</td><td>706</td><td>63.96520</td><td>1.143588</td><td>62.50000</td><td>2.500000</td><td> 0.00000</td><td> 0.00000</td><td>-1.736426</td><td> 0.00000</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 4208 × 21\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " cluster\\_id & primer\\_set & amplicon\\_size\\_consensus & amplicon\\_size\\_avg & amplicon\\_size\\_sd & primer\\_id & primer\\_type & sequence & length & degeneracy & ⋯ & position\\_start & position\\_end & Tm\\_avg & Tm\\_sd & GC\\_avg & GC\\_sd & hairpin\\_avg & hairpin\\_sd & homodimer\\_avg & homodimer\\_sd\\\\\n",
       " <int> & <int> & <int> & <dbl> & <dbl> & <fct> & <fct> & <fct> & <int> & <int> & ⋯ & <int> & <int> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t 4 & 13 & 172 & 172 & 0 & 13f & PRIMER\\_LEFT  & GAAYCARGGACAYWGCCACC & 20 & 16 & ⋯ & 161 & 181 & 60.40519 & 1.951478 & 57.50000 & 4.330127 & 10.70543 & 18.661427 & -23.02055 & 15.93829\\\\\n",
       "\t 4 & 13 & 172 & 172 & 0 & 13r & PRIMER\\_RIGHT & MGCYTCDGCACCRCTATTCA & 20 & 24 & ⋯ & 313 & 333 & 61.39287 & 2.160181 & 54.16667 & 4.930066 & 42.22950 &  2.453775 & -19.36984 & 24.02227\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋱ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t 500 & 499 & 133 & 133 & 0 & 499f & PRIMER\\_LEFT  & ACBCCMTGYGCBGCAAAAAT & 20 & 36 & ⋯ & 573 & 593 & 63.11378 & 2.487889 & 51.66667 & 4.859127 & 31.85494 & 24.19495 &  6.798857 & 10.52647\\\\\n",
       "\t 500 & 499 & 133 & 133 & 0 & 499r & PRIMER\\_RIGHT & GTTGRTCCCGGTGTTGCGTG & 20 &  2 & ⋯ & 686 & 706 & 63.96520 & 1.143588 & 62.50000 & 2.500000 &  0.00000 &  0.00000 & -1.736426 &  0.00000\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 4208 × 21\n",
       "\n",
       "| cluster_id &lt;int&gt; | primer_set &lt;int&gt; | amplicon_size_consensus &lt;int&gt; | amplicon_size_avg &lt;dbl&gt; | amplicon_size_sd &lt;dbl&gt; | primer_id &lt;fct&gt; | primer_type &lt;fct&gt; | sequence &lt;fct&gt; | length &lt;int&gt; | degeneracy &lt;int&gt; | ⋯ ⋯ | position_start &lt;int&gt; | position_end &lt;int&gt; | Tm_avg &lt;dbl&gt; | Tm_sd &lt;dbl&gt; | GC_avg &lt;dbl&gt; | GC_sd &lt;dbl&gt; | hairpin_avg &lt;dbl&gt; | hairpin_sd &lt;dbl&gt; | homodimer_avg &lt;dbl&gt; | homodimer_sd &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 4 | 13 | 172 | 172 | 0 | 13f | PRIMER_LEFT  | GAAYCARGGACAYWGCCACC | 20 | 16 | ⋯ | 161 | 181 | 60.40519 | 1.951478 | 57.50000 | 4.330127 | 10.70543 | 18.661427 | -23.02055 | 15.93829 |\n",
       "| 4 | 13 | 172 | 172 | 0 | 13r | PRIMER_RIGHT | MGCYTCDGCACCRCTATTCA | 20 | 24 | ⋯ | 313 | 333 | 61.39287 | 2.160181 | 54.16667 | 4.930066 | 42.22950 |  2.453775 | -19.36984 | 24.02227 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋱ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| 500 | 499 | 133 | 133 | 0 | 499f | PRIMER_LEFT  | ACBCCMTGYGCBGCAAAAAT | 20 | 36 | ⋯ | 573 | 593 | 63.11378 | 2.487889 | 51.66667 | 4.859127 | 31.85494 | 24.19495 |  6.798857 | 10.52647 |\n",
       "| 500 | 499 | 133 | 133 | 0 | 499r | PRIMER_RIGHT | GTTGRTCCCGGTGTTGCGTG | 20 |  2 | ⋯ | 686 | 706 | 63.96520 | 1.143588 | 62.50000 | 2.500000 |  0.00000 |  0.00000 | -1.736426 |  0.00000 |\n",
       "\n"
      ],
      "text/plain": [
       "     cluster_id primer_set amplicon_size_consensus amplicon_size_avg\n",
       "1    4          13         172                     172              \n",
       "2    4          13         172                     172              \n",
       "⋮    ⋮          ⋮          ⋮                       ⋮                \n",
       "4207 500        499        133                     133              \n",
       "4208 500        499        133                     133              \n",
       "     amplicon_size_sd primer_id primer_type  sequence             length\n",
       "1    0                13f       PRIMER_LEFT  GAAYCARGGACAYWGCCACC 20    \n",
       "2    0                13r       PRIMER_RIGHT MGCYTCDGCACCRCTATTCA 20    \n",
       "⋮    ⋮                ⋮         ⋮            ⋮                    ⋮     \n",
       "4207 0                499f      PRIMER_LEFT  ACBCCMTGYGCBGCAAAAAT 20    \n",
       "4208 0                499r      PRIMER_RIGHT GTTGRTCCCGGTGTTGCGTG 20    \n",
       "     degeneracy ⋯ position_start position_end Tm_avg   Tm_sd    GC_avg  \n",
       "1    16         ⋯ 161            181          60.40519 1.951478 57.50000\n",
       "2    24         ⋯ 313            333          61.39287 2.160181 54.16667\n",
       "⋮    ⋮          ⋱ ⋮              ⋮            ⋮        ⋮        ⋮       \n",
       "4207 36         ⋯ 573            593          63.11378 2.487889 51.66667\n",
       "4208  2         ⋯ 686            706          63.96520 1.143588 62.50000\n",
       "     GC_sd    hairpin_avg hairpin_sd homodimer_avg homodimer_sd\n",
       "1    4.330127 10.70543    18.661427  -23.02055     15.93829    \n",
       "2    4.930066 42.22950     2.453775  -19.36984     24.02227    \n",
       "⋮    ⋮        ⋮           ⋮          ⋮             ⋮           \n",
       "4207 4.859127 31.85494    24.19495    6.798857     10.52647    \n",
       "4208 2.500000  0.00000     0.00000   -1.736426      0.00000    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "primer_info = read.delim(file.path(work_dir, 'primers', 'cgp', 'primers_final_info.tsv'), sep='\\t')\n",
    "primer_info %>% unique_n('primer sets', primer_set)\n",
    "primer_info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Gene cluster annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-19T14:37:59.979958Z",
     "iopub.status.busy": "2021-01-19T14:37:59.977989Z",
     "iopub.status.idle": "2021-01-19T14:38:00.068237Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 690 × 19</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>cluster_id</th><th scope=col>query</th><th scope=col>subject</th><th scope=col>subject_name</th><th scope=col>pident</th><th scope=col>length</th><th scope=col>mismatch</th><th scope=col>qstart</th><th scope=col>qend</th><th scope=col>sstart</th><th scope=col>send</th><th scope=col>evalue</th><th scope=col>slen</th><th scope=col>qlen</th><th scope=col>sscinames</th><th scope=col>staxids</th><th scope=col>pident_rank</th><th scope=col>gene_name</th><th scope=col>gene_taxonomy</th></tr>\n",
       "\t<tr><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>4</td><td>4dd2fadf7df94a0babc664b2d3c2b220</td><td>NJC17858.1  </td><td>ornithine--oxo-acid transaminase [Butyricimonas paravirosa]  </td><td>99.015</td><td>406</td><td>4</td><td>1</td><td>1218</td><td>1</td><td>406</td><td>0</td><td>406</td><td>1221</td><td>Butyricimonas paravirosa  </td><td>1472417</td><td>1</td><td>ornithine--oxo-acid transaminase</td><td>Butyricimonas paravirosa  </td></tr>\n",
       "\t<tr><td>4</td><td>4dd2fadf7df94a0babc664b2d3c2b220</td><td>MBB4027972.1</td><td>ornithine--oxo-acid transaminase [Butyricimonas faecihominis]</td><td>98.768</td><td>406</td><td>5</td><td>1</td><td>1218</td><td>1</td><td>406</td><td>0</td><td>406</td><td>1221</td><td>Butyricimonas faecihominis</td><td>1472416</td><td>2</td><td>ornithine--oxo-acid transaminase</td><td>Butyricimonas faecihominis</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>500</td><td>466e10a5838d4b9ebeea2eb8446771b8</td><td>WP_099293516.1</td><td>4Fe-4S binding protein [Butyricimonas sp. Marseille-P3923]</td><td>99.776</td><td>447</td><td> 1</td><td>1</td><td>1341</td><td>1</td><td>447</td><td>0</td><td>447</td><td>1344</td><td>Butyricimonas sp. Marseille-P3923                                                                              </td><td>1987504                      </td><td>1</td><td>4Fe-4S binding protein              </td><td>Butyricimonas sp. Marseille-P3923</td></tr>\n",
       "\t<tr><td>500</td><td>466e10a5838d4b9ebeea2eb8446771b8</td><td>WP_040371279.1</td><td>MULTISPECIES: 4Fe-4S binding protein [Butyricimonas]      </td><td>95.302</td><td>447</td><td>21</td><td>1</td><td>1341</td><td>1</td><td>447</td><td>0</td><td>447</td><td>1344</td><td>Butyricimonas synergistica;Butyricimonas;Butyricimonas synergistica DSM 23225;Butyricimonas sp. Marseille-P2440</td><td>544644;574697;1121129;1816677</td><td>2</td><td>MULTISPECIES: 4Fe-4S binding protein</td><td>Butyricimonas                    </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 690 × 19\n",
       "\\begin{tabular}{lllllllllllllllllll}\n",
       " cluster\\_id & query & subject & subject\\_name & pident & length & mismatch & qstart & qend & sstart & send & evalue & slen & qlen & sscinames & staxids & pident\\_rank & gene\\_name & gene\\_taxonomy\\\\\n",
       " <dbl> & <fct> & <fct> & <fct> & <dbl> & <int> & <int> & <int> & <int> & <int> & <int> & <dbl> & <int> & <int> & <fct> & <fct> & <int> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t 4 & 4dd2fadf7df94a0babc664b2d3c2b220 & NJC17858.1   & ornithine--oxo-acid transaminase {[}Butyricimonas paravirosa{]}   & 99.015 & 406 & 4 & 1 & 1218 & 1 & 406 & 0 & 406 & 1221 & Butyricimonas paravirosa   & 1472417 & 1 & ornithine--oxo-acid transaminase & Butyricimonas paravirosa  \\\\\n",
       "\t 4 & 4dd2fadf7df94a0babc664b2d3c2b220 & MBB4027972.1 & ornithine--oxo-acid transaminase {[}Butyricimonas faecihominis{]} & 98.768 & 406 & 5 & 1 & 1218 & 1 & 406 & 0 & 406 & 1221 & Butyricimonas faecihominis & 1472416 & 2 & ornithine--oxo-acid transaminase & Butyricimonas faecihominis\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t 500 & 466e10a5838d4b9ebeea2eb8446771b8 & WP\\_099293516.1 & 4Fe-4S binding protein {[}Butyricimonas sp. Marseille-P3923{]} & 99.776 & 447 &  1 & 1 & 1341 & 1 & 447 & 0 & 447 & 1344 & Butyricimonas sp. Marseille-P3923                                                                               & 1987504                       & 1 & 4Fe-4S binding protein               & Butyricimonas sp. Marseille-P3923\\\\\n",
       "\t 500 & 466e10a5838d4b9ebeea2eb8446771b8 & WP\\_040371279.1 & MULTISPECIES: 4Fe-4S binding protein {[}Butyricimonas{]}       & 95.302 & 447 & 21 & 1 & 1341 & 1 & 447 & 0 & 447 & 1344 & Butyricimonas synergistica;Butyricimonas;Butyricimonas synergistica DSM 23225;Butyricimonas sp. Marseille-P2440 & 544644;574697;1121129;1816677 & 2 & MULTISPECIES: 4Fe-4S binding protein & Butyricimonas                    \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 690 × 19\n",
       "\n",
       "| cluster_id &lt;dbl&gt; | query &lt;fct&gt; | subject &lt;fct&gt; | subject_name &lt;fct&gt; | pident &lt;dbl&gt; | length &lt;int&gt; | mismatch &lt;int&gt; | qstart &lt;int&gt; | qend &lt;int&gt; | sstart &lt;int&gt; | send &lt;int&gt; | evalue &lt;dbl&gt; | slen &lt;int&gt; | qlen &lt;int&gt; | sscinames &lt;fct&gt; | staxids &lt;fct&gt; | pident_rank &lt;int&gt; | gene_name &lt;chr&gt; | gene_taxonomy &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 4 | 4dd2fadf7df94a0babc664b2d3c2b220 | NJC17858.1   | ornithine--oxo-acid transaminase [Butyricimonas paravirosa]   | 99.015 | 406 | 4 | 1 | 1218 | 1 | 406 | 0 | 406 | 1221 | Butyricimonas paravirosa   | 1472417 | 1 | ornithine--oxo-acid transaminase | Butyricimonas paravirosa   |\n",
       "| 4 | 4dd2fadf7df94a0babc664b2d3c2b220 | MBB4027972.1 | ornithine--oxo-acid transaminase [Butyricimonas faecihominis] | 98.768 | 406 | 5 | 1 | 1218 | 1 | 406 | 0 | 406 | 1221 | Butyricimonas faecihominis | 1472416 | 2 | ornithine--oxo-acid transaminase | Butyricimonas faecihominis |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| 500 | 466e10a5838d4b9ebeea2eb8446771b8 | WP_099293516.1 | 4Fe-4S binding protein [Butyricimonas sp. Marseille-P3923] | 99.776 | 447 |  1 | 1 | 1341 | 1 | 447 | 0 | 447 | 1344 | Butyricimonas sp. Marseille-P3923                                                                               | 1987504                       | 1 | 4Fe-4S binding protein               | Butyricimonas sp. Marseille-P3923 |\n",
       "| 500 | 466e10a5838d4b9ebeea2eb8446771b8 | WP_040371279.1 | MULTISPECIES: 4Fe-4S binding protein [Butyricimonas]       | 95.302 | 447 | 21 | 1 | 1341 | 1 | 447 | 0 | 447 | 1344 | Butyricimonas synergistica;Butyricimonas;Butyricimonas synergistica DSM 23225;Butyricimonas sp. Marseille-P2440 | 544644;574697;1121129;1816677 | 2 | MULTISPECIES: 4Fe-4S binding protein | Butyricimonas                     |\n",
       "\n"
      ],
      "text/plain": [
       "    cluster_id query                            subject       \n",
       "1   4          4dd2fadf7df94a0babc664b2d3c2b220 NJC17858.1    \n",
       "2   4          4dd2fadf7df94a0babc664b2d3c2b220 MBB4027972.1  \n",
       "⋮   ⋮          ⋮                                ⋮             \n",
       "689 500        466e10a5838d4b9ebeea2eb8446771b8 WP_099293516.1\n",
       "690 500        466e10a5838d4b9ebeea2eb8446771b8 WP_040371279.1\n",
       "    subject_name                                                  pident length\n",
       "1   ornithine--oxo-acid transaminase [Butyricimonas paravirosa]   99.015 406   \n",
       "2   ornithine--oxo-acid transaminase [Butyricimonas faecihominis] 98.768 406   \n",
       "⋮   ⋮                                                             ⋮      ⋮     \n",
       "689 4Fe-4S binding protein [Butyricimonas sp. Marseille-P3923]    99.776 447   \n",
       "690 MULTISPECIES: 4Fe-4S binding protein [Butyricimonas]          95.302 447   \n",
       "    mismatch qstart qend sstart send evalue slen qlen\n",
       "1   4        1      1218 1      406  0      406  1221\n",
       "2   5        1      1218 1      406  0      406  1221\n",
       "⋮   ⋮        ⋮      ⋮    ⋮      ⋮    ⋮      ⋮    ⋮   \n",
       "689  1       1      1341 1      447  0      447  1344\n",
       "690 21       1      1341 1      447  0      447  1344\n",
       "    sscinames                                                                                                      \n",
       "1   Butyricimonas paravirosa                                                                                       \n",
       "2   Butyricimonas faecihominis                                                                                     \n",
       "⋮   ⋮                                                                                                              \n",
       "689 Butyricimonas sp. Marseille-P3923                                                                              \n",
       "690 Butyricimonas synergistica;Butyricimonas;Butyricimonas synergistica DSM 23225;Butyricimonas sp. Marseille-P2440\n",
       "    staxids                       pident_rank\n",
       "1   1472417                       1          \n",
       "2   1472416                       2          \n",
       "⋮   ⋮                             ⋮          \n",
       "689 1987504                       1          \n",
       "690 544644;574697;1121129;1816677 2          \n",
       "    gene_name                            gene_taxonomy                    \n",
       "1   ornithine--oxo-acid transaminase     Butyricimonas paravirosa         \n",
       "2   ornithine--oxo-acid transaminase     Butyricimonas faecihominis       \n",
       "⋮   ⋮                                    ⋮                                \n",
       "689 4Fe-4S binding protein               Butyricimonas sp. Marseille-P3923\n",
       "690 MULTISPECIES: 4Fe-4S binding protein Butyricimonas                    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "gene_annot = read.delim(file.path(work_dir, 'primers', 'cgp', 'core_clusters_blastx.tsv'), \n",
    "                        sep='\\t') %>%\n",
    "    mutate(cluster_id = gsub('cluster_', '', cluster_id) %>% as.Num) %>%\n",
    "    semi_join(primer_info, c('cluster_id')) %>%\n",
    "    mutate(gene_name = gsub(' \\\\[.+', '', subject_name),\n",
    "           gene_taxonomy = gsub('.+\\\\[', '', subject_name),\n",
    "           gene_taxonomy = gsub('\\\\]', '', gene_taxonomy))\n",
    "gene_annot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-19T14:38:44.084661Z",
     "iopub.status.busy": "2021-01-19T14:38:44.082569Z",
     "iopub.status.idle": "2021-01-19T14:38:44.143946Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 430 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>cluster_id</th><th scope=col>gene_name</th></tr>\n",
       "\t<tr><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td> 4</td><td>ornithine--oxo-acid transaminase                                                                               </td></tr>\n",
       "\t<tr><td> 7</td><td>MULTISPECIES: 50S ribosomal protein L27                                                                        </td></tr>\n",
       "\t<tr><td> 9</td><td>MULTISPECIES: bifunctional methylenetetrahydrofolate dehydrogenase/methenyltetrahydrofolate cyclohydrolase FolD</td></tr>\n",
       "\t<tr><td> 9</td><td>bifunctional methylenetetrahydrofolate dehydrogenase/methenyltetrahydrofolate cyclohydrolase FolD              </td></tr>\n",
       "\t<tr><td>13</td><td>NADH:ubiquinone reductase (Na(+)-transporting) subunit E                                                       </td></tr>\n",
       "\t<tr><td>14</td><td>MULTISPECIES: homocysteine biosynthesis protein                                                                </td></tr>\n",
       "\t<tr><td>14</td><td>hypothetical protein BHV81_15115                                                                               </td></tr>\n",
       "\t<tr><td>14</td><td>homocysteine biosynthesis protein                                                                              </td></tr>\n",
       "\t<tr><td>16</td><td>MULTISPECIES: DNA polymerase III subunit delta                                                                 </td></tr>\n",
       "\t<tr><td>16</td><td>DNA polymerase III subunit delta                                                                               </td></tr>\n",
       "\t<tr><td>18</td><td>V-type ATP synthase subunit A                                                                                  </td></tr>\n",
       "\t<tr><td>18</td><td>MULTISPECIES: V-type ATP synthase subunit A                                                                    </td></tr>\n",
       "\t<tr><td>22</td><td>MULTISPECIES: 30S ribosomal protein S20                                                                        </td></tr>\n",
       "\t<tr><td>22</td><td>30S ribosomal protein S20                                                                                      </td></tr>\n",
       "\t<tr><td>24</td><td>MULTISPECIES: glycine cleavage system protein GcvH                                                             </td></tr>\n",
       "\t<tr><td>24</td><td>glycine cleavage system protein GcvH                                                                           </td></tr>\n",
       "\t<tr><td>25</td><td>MULTISPECIES: ATP synthase F0 subunit C                                                                        </td></tr>\n",
       "\t<tr><td>25</td><td>ATP synthase F0 subunit C                                                                                      </td></tr>\n",
       "\t<tr><td>26</td><td>MULTISPECIES: NADH-quinone oxidoreductase subunit L                                                            </td></tr>\n",
       "\t<tr><td>26</td><td>NADH-quinone oxidoreductase subunit L                                                                          </td></tr>\n",
       "\t<tr><td>33</td><td>inorganic phosphate transporter                                                                                </td></tr>\n",
       "\t<tr><td>33</td><td>MULTISPECIES: inorganic phosphate transporter                                                                  </td></tr>\n",
       "\t<tr><td>35</td><td>MULTISPECIES: adenylosuccinate synthase                                                                        </td></tr>\n",
       "\t<tr><td>35</td><td>adenylosuccinate synthase                                                                                      </td></tr>\n",
       "\t<tr><td>37</td><td>MULTISPECIES: lysophospholipid acyltransferase family protein                                                  </td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>479</td><td>MULTISPECIES: translation initiation factor IF-2         </td></tr>\n",
       "\t<tr><td>479</td><td>translation initiation factor IF-2                       </td></tr>\n",
       "\t<tr><td>480</td><td>MULTISPECIES: phosphoribosylglycinamide formyltransferase</td></tr>\n",
       "\t<tr><td>480</td><td>phosphoribosylglycinamide formyltransferase              </td></tr>\n",
       "\t<tr><td>481</td><td>magnesium transporter CorA family protein                </td></tr>\n",
       "\t<tr><td>481</td><td>MULTISPECIES: magnesium transporter CorA family protein  </td></tr>\n",
       "\t<tr><td>486</td><td>30S ribosomal protein S16                                </td></tr>\n",
       "\t<tr><td>486</td><td>MULTISPECIES: 30S ribosomal protein S16                  </td></tr>\n",
       "\t<tr><td>487</td><td>tryptophan--tRNA ligase                                  </td></tr>\n",
       "\t<tr><td>487</td><td>MULTISPECIES: tryptophan--tRNA ligase                    </td></tr>\n",
       "\t<tr><td>489</td><td>MULTISPECIES: 30S ribosomal protein S10                  </td></tr>\n",
       "\t<tr><td>489</td><td>30S ribosomal protein S10                                </td></tr>\n",
       "\t<tr><td>492</td><td>phosphoribosylformylglycinamidine synthase               </td></tr>\n",
       "\t<tr><td>494</td><td>MULTISPECIES: cell division protein FtsZ                 </td></tr>\n",
       "\t<tr><td>494</td><td>cell division protein FtsZ                               </td></tr>\n",
       "\t<tr><td>495</td><td>MULTISPECIES: M23 family metallopeptidase                </td></tr>\n",
       "\t<tr><td>495</td><td>M23 family metallopeptidase                              </td></tr>\n",
       "\t<tr><td>496</td><td>acyl-CoA carboxylase subunit beta                        </td></tr>\n",
       "\t<tr><td>496</td><td>MULTISPECIES: acyl-CoA carboxylase subunit beta          </td></tr>\n",
       "\t<tr><td>498</td><td>MULTISPECIES: 50S ribosomal protein L17                  </td></tr>\n",
       "\t<tr><td>498</td><td>50S ribosomal protein L17                                </td></tr>\n",
       "\t<tr><td>499</td><td>DUF2149 domain-containing protein                        </td></tr>\n",
       "\t<tr><td>499</td><td>MULTISPECIES: DUF2149 domain-containing protein          </td></tr>\n",
       "\t<tr><td>500</td><td>4Fe-4S binding protein                                   </td></tr>\n",
       "\t<tr><td>500</td><td>MULTISPECIES: 4Fe-4S binding protein                     </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 430 × 2\n",
       "\\begin{tabular}{ll}\n",
       " cluster\\_id & gene\\_name\\\\\n",
       " <dbl> & <chr>\\\\\n",
       "\\hline\n",
       "\t  4 & ornithine--oxo-acid transaminase                                                                               \\\\\n",
       "\t  7 & MULTISPECIES: 50S ribosomal protein L27                                                                        \\\\\n",
       "\t  9 & MULTISPECIES: bifunctional methylenetetrahydrofolate dehydrogenase/methenyltetrahydrofolate cyclohydrolase FolD\\\\\n",
       "\t  9 & bifunctional methylenetetrahydrofolate dehydrogenase/methenyltetrahydrofolate cyclohydrolase FolD              \\\\\n",
       "\t 13 & NADH:ubiquinone reductase (Na(+)-transporting) subunit E                                                       \\\\\n",
       "\t 14 & MULTISPECIES: homocysteine biosynthesis protein                                                                \\\\\n",
       "\t 14 & hypothetical protein BHV81\\_15115                                                                               \\\\\n",
       "\t 14 & homocysteine biosynthesis protein                                                                              \\\\\n",
       "\t 16 & MULTISPECIES: DNA polymerase III subunit delta                                                                 \\\\\n",
       "\t 16 & DNA polymerase III subunit delta                                                                               \\\\\n",
       "\t 18 & V-type ATP synthase subunit A                                                                                  \\\\\n",
       "\t 18 & MULTISPECIES: V-type ATP synthase subunit A                                                                    \\\\\n",
       "\t 22 & MULTISPECIES: 30S ribosomal protein S20                                                                        \\\\\n",
       "\t 22 & 30S ribosomal protein S20                                                                                      \\\\\n",
       "\t 24 & MULTISPECIES: glycine cleavage system protein GcvH                                                             \\\\\n",
       "\t 24 & glycine cleavage system protein GcvH                                                                           \\\\\n",
       "\t 25 & MULTISPECIES: ATP synthase F0 subunit C                                                                        \\\\\n",
       "\t 25 & ATP synthase F0 subunit C                                                                                      \\\\\n",
       "\t 26 & MULTISPECIES: NADH-quinone oxidoreductase subunit L                                                            \\\\\n",
       "\t 26 & NADH-quinone oxidoreductase subunit L                                                                          \\\\\n",
       "\t 33 & inorganic phosphate transporter                                                                                \\\\\n",
       "\t 33 & MULTISPECIES: inorganic phosphate transporter                                                                  \\\\\n",
       "\t 35 & MULTISPECIES: adenylosuccinate synthase                                                                        \\\\\n",
       "\t 35 & adenylosuccinate synthase                                                                                      \\\\\n",
       "\t 37 & MULTISPECIES: lysophospholipid acyltransferase family protein                                                  \\\\\n",
       "\t ⋮ & ⋮\\\\\n",
       "\t 479 & MULTISPECIES: translation initiation factor IF-2         \\\\\n",
       "\t 479 & translation initiation factor IF-2                       \\\\\n",
       "\t 480 & MULTISPECIES: phosphoribosylglycinamide formyltransferase\\\\\n",
       "\t 480 & phosphoribosylglycinamide formyltransferase              \\\\\n",
       "\t 481 & magnesium transporter CorA family protein                \\\\\n",
       "\t 481 & MULTISPECIES: magnesium transporter CorA family protein  \\\\\n",
       "\t 486 & 30S ribosomal protein S16                                \\\\\n",
       "\t 486 & MULTISPECIES: 30S ribosomal protein S16                  \\\\\n",
       "\t 487 & tryptophan--tRNA ligase                                  \\\\\n",
       "\t 487 & MULTISPECIES: tryptophan--tRNA ligase                    \\\\\n",
       "\t 489 & MULTISPECIES: 30S ribosomal protein S10                  \\\\\n",
       "\t 489 & 30S ribosomal protein S10                                \\\\\n",
       "\t 492 & phosphoribosylformylglycinamidine synthase               \\\\\n",
       "\t 494 & MULTISPECIES: cell division protein FtsZ                 \\\\\n",
       "\t 494 & cell division protein FtsZ                               \\\\\n",
       "\t 495 & MULTISPECIES: M23 family metallopeptidase                \\\\\n",
       "\t 495 & M23 family metallopeptidase                              \\\\\n",
       "\t 496 & acyl-CoA carboxylase subunit beta                        \\\\\n",
       "\t 496 & MULTISPECIES: acyl-CoA carboxylase subunit beta          \\\\\n",
       "\t 498 & MULTISPECIES: 50S ribosomal protein L17                  \\\\\n",
       "\t 498 & 50S ribosomal protein L17                                \\\\\n",
       "\t 499 & DUF2149 domain-containing protein                        \\\\\n",
       "\t 499 & MULTISPECIES: DUF2149 domain-containing protein          \\\\\n",
       "\t 500 & 4Fe-4S binding protein                                   \\\\\n",
       "\t 500 & MULTISPECIES: 4Fe-4S binding protein                     \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 430 × 2\n",
       "\n",
       "| cluster_id &lt;dbl&gt; | gene_name &lt;chr&gt; |\n",
       "|---|---|\n",
       "|  4 | ornithine--oxo-acid transaminase                                                                                |\n",
       "|  7 | MULTISPECIES: 50S ribosomal protein L27                                                                         |\n",
       "|  9 | MULTISPECIES: bifunctional methylenetetrahydrofolate dehydrogenase/methenyltetrahydrofolate cyclohydrolase FolD |\n",
       "|  9 | bifunctional methylenetetrahydrofolate dehydrogenase/methenyltetrahydrofolate cyclohydrolase FolD               |\n",
       "| 13 | NADH:ubiquinone reductase (Na(+)-transporting) subunit E                                                        |\n",
       "| 14 | MULTISPECIES: homocysteine biosynthesis protein                                                                 |\n",
       "| 14 | hypothetical protein BHV81_15115                                                                                |\n",
       "| 14 | homocysteine biosynthesis protein                                                                               |\n",
       "| 16 | MULTISPECIES: DNA polymerase III subunit delta                                                                  |\n",
       "| 16 | DNA polymerase III subunit delta                                                                                |\n",
       "| 18 | V-type ATP synthase subunit A                                                                                   |\n",
       "| 18 | MULTISPECIES: V-type ATP synthase subunit A                                                                     |\n",
       "| 22 | MULTISPECIES: 30S ribosomal protein S20                                                                         |\n",
       "| 22 | 30S ribosomal protein S20                                                                                       |\n",
       "| 24 | MULTISPECIES: glycine cleavage system protein GcvH                                                              |\n",
       "| 24 | glycine cleavage system protein GcvH                                                                            |\n",
       "| 25 | MULTISPECIES: ATP synthase F0 subunit C                                                                         |\n",
       "| 25 | ATP synthase F0 subunit C                                                                                       |\n",
       "| 26 | MULTISPECIES: NADH-quinone oxidoreductase subunit L                                                             |\n",
       "| 26 | NADH-quinone oxidoreductase subunit L                                                                           |\n",
       "| 33 | inorganic phosphate transporter                                                                                 |\n",
       "| 33 | MULTISPECIES: inorganic phosphate transporter                                                                   |\n",
       "| 35 | MULTISPECIES: adenylosuccinate synthase                                                                         |\n",
       "| 35 | adenylosuccinate synthase                                                                                       |\n",
       "| 37 | MULTISPECIES: lysophospholipid acyltransferase family protein                                                   |\n",
       "| ⋮ | ⋮ |\n",
       "| 479 | MULTISPECIES: translation initiation factor IF-2          |\n",
       "| 479 | translation initiation factor IF-2                        |\n",
       "| 480 | MULTISPECIES: phosphoribosylglycinamide formyltransferase |\n",
       "| 480 | phosphoribosylglycinamide formyltransferase               |\n",
       "| 481 | magnesium transporter CorA family protein                 |\n",
       "| 481 | MULTISPECIES: magnesium transporter CorA family protein   |\n",
       "| 486 | 30S ribosomal protein S16                                 |\n",
       "| 486 | MULTISPECIES: 30S ribosomal protein S16                   |\n",
       "| 487 | tryptophan--tRNA ligase                                   |\n",
       "| 487 | MULTISPECIES: tryptophan--tRNA ligase                     |\n",
       "| 489 | MULTISPECIES: 30S ribosomal protein S10                   |\n",
       "| 489 | 30S ribosomal protein S10                                 |\n",
       "| 492 | phosphoribosylformylglycinamidine synthase                |\n",
       "| 494 | MULTISPECIES: cell division protein FtsZ                  |\n",
       "| 494 | cell division protein FtsZ                                |\n",
       "| 495 | MULTISPECIES: M23 family metallopeptidase                 |\n",
       "| 495 | M23 family metallopeptidase                               |\n",
       "| 496 | acyl-CoA carboxylase subunit beta                         |\n",
       "| 496 | MULTISPECIES: acyl-CoA carboxylase subunit beta           |\n",
       "| 498 | MULTISPECIES: 50S ribosomal protein L17                   |\n",
       "| 498 | 50S ribosomal protein L17                                 |\n",
       "| 499 | DUF2149 domain-containing protein                         |\n",
       "| 499 | MULTISPECIES: DUF2149 domain-containing protein           |\n",
       "| 500 | 4Fe-4S binding protein                                    |\n",
       "| 500 | MULTISPECIES: 4Fe-4S binding protein                      |\n",
       "\n"
      ],
      "text/plain": [
       "    cluster_id\n",
       "1    4        \n",
       "2    7        \n",
       "3    9        \n",
       "4    9        \n",
       "5   13        \n",
       "6   14        \n",
       "7   14        \n",
       "8   14        \n",
       "9   16        \n",
       "10  16        \n",
       "11  18        \n",
       "12  18        \n",
       "13  22        \n",
       "14  22        \n",
       "15  24        \n",
       "16  24        \n",
       "17  25        \n",
       "18  25        \n",
       "19  26        \n",
       "20  26        \n",
       "21  33        \n",
       "22  33        \n",
       "23  35        \n",
       "24  35        \n",
       "25  37        \n",
       "⋮   ⋮         \n",
       "406 479       \n",
       "407 479       \n",
       "408 480       \n",
       "409 480       \n",
       "410 481       \n",
       "411 481       \n",
       "412 486       \n",
       "413 486       \n",
       "414 487       \n",
       "415 487       \n",
       "416 489       \n",
       "417 489       \n",
       "418 492       \n",
       "419 494       \n",
       "420 494       \n",
       "421 495       \n",
       "422 495       \n",
       "423 496       \n",
       "424 496       \n",
       "425 498       \n",
       "426 498       \n",
       "427 499       \n",
       "428 499       \n",
       "429 500       \n",
       "430 500       \n",
       "    gene_name                                                                                                      \n",
       "1   ornithine--oxo-acid transaminase                                                                               \n",
       "2   MULTISPECIES: 50S ribosomal protein L27                                                                        \n",
       "3   MULTISPECIES: bifunctional methylenetetrahydrofolate dehydrogenase/methenyltetrahydrofolate cyclohydrolase FolD\n",
       "4   bifunctional methylenetetrahydrofolate dehydrogenase/methenyltetrahydrofolate cyclohydrolase FolD              \n",
       "5   NADH:ubiquinone reductase (Na(+)-transporting) subunit E                                                       \n",
       "6   MULTISPECIES: homocysteine biosynthesis protein                                                                \n",
       "7   hypothetical protein BHV81_15115                                                                               \n",
       "8   homocysteine biosynthesis protein                                                                              \n",
       "9   MULTISPECIES: DNA polymerase III subunit delta                                                                 \n",
       "10  DNA polymerase III subunit delta                                                                               \n",
       "11  V-type ATP synthase subunit A                                                                                  \n",
       "12  MULTISPECIES: V-type ATP synthase subunit A                                                                    \n",
       "13  MULTISPECIES: 30S ribosomal protein S20                                                                        \n",
       "14  30S ribosomal protein S20                                                                                      \n",
       "15  MULTISPECIES: glycine cleavage system protein GcvH                                                             \n",
       "16  glycine cleavage system protein GcvH                                                                           \n",
       "17  MULTISPECIES: ATP synthase F0 subunit C                                                                        \n",
       "18  ATP synthase F0 subunit C                                                                                      \n",
       "19  MULTISPECIES: NADH-quinone oxidoreductase subunit L                                                            \n",
       "20  NADH-quinone oxidoreductase subunit L                                                                          \n",
       "21  inorganic phosphate transporter                                                                                \n",
       "22  MULTISPECIES: inorganic phosphate transporter                                                                  \n",
       "23  MULTISPECIES: adenylosuccinate synthase                                                                        \n",
       "24  adenylosuccinate synthase                                                                                      \n",
       "25  MULTISPECIES: lysophospholipid acyltransferase family protein                                                  \n",
       "⋮   ⋮                                                                                                              \n",
       "406 MULTISPECIES: translation initiation factor IF-2                                                               \n",
       "407 translation initiation factor IF-2                                                                             \n",
       "408 MULTISPECIES: phosphoribosylglycinamide formyltransferase                                                      \n",
       "409 phosphoribosylglycinamide formyltransferase                                                                    \n",
       "410 magnesium transporter CorA family protein                                                                      \n",
       "411 MULTISPECIES: magnesium transporter CorA family protein                                                        \n",
       "412 30S ribosomal protein S16                                                                                      \n",
       "413 MULTISPECIES: 30S ribosomal protein S16                                                                        \n",
       "414 tryptophan--tRNA ligase                                                                                        \n",
       "415 MULTISPECIES: tryptophan--tRNA ligase                                                                          \n",
       "416 MULTISPECIES: 30S ribosomal protein S10                                                                        \n",
       "417 30S ribosomal protein S10                                                                                      \n",
       "418 phosphoribosylformylglycinamidine synthase                                                                     \n",
       "419 MULTISPECIES: cell division protein FtsZ                                                                       \n",
       "420 cell division protein FtsZ                                                                                     \n",
       "421 MULTISPECIES: M23 family metallopeptidase                                                                      \n",
       "422 M23 family metallopeptidase                                                                                    \n",
       "423 acyl-CoA carboxylase subunit beta                                                                              \n",
       "424 MULTISPECIES: acyl-CoA carboxylase subunit beta                                                                \n",
       "425 MULTISPECIES: 50S ribosomal protein L17                                                                        \n",
       "426 50S ribosomal protein L17                                                                                      \n",
       "427 DUF2149 domain-containing protein                                                                              \n",
       "428 MULTISPECIES: DUF2149 domain-containing protein                                                                \n",
       "429 4Fe-4S binding protein                                                                                         \n",
       "430 MULTISPECIES: 4Fe-4S binding protein                                                                           "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df.dims(50)\n",
    "gene_annot %>%\n",
    "    distinct(cluster_id, gene_name) \n",
    "df.dims()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-19T14:38:52.046725Z",
     "iopub.status.busy": "2021-01-19T14:38:52.044929Z",
     "iopub.status.idle": "2021-01-19T14:38:52.102197Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 581 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>cluster_id</th><th scope=col>gene_taxonomy</th></tr>\n",
       "\t<tr><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td> 4</td><td>Butyricimonas paravirosa  </td></tr>\n",
       "\t<tr><td> 4</td><td>Butyricimonas faecihominis</td></tr>\n",
       "\t<tr><td> 4</td><td>Butyricimonas virosa      </td></tr>\n",
       "\t<tr><td> 7</td><td>Butyricimonas             </td></tr>\n",
       "\t<tr><td> 7</td><td>unclassified Butyricimonas</td></tr>\n",
       "\t<tr><td> 9</td><td>Butyricimonas             </td></tr>\n",
       "\t<tr><td> 9</td><td>Butyricimonas virosa      </td></tr>\n",
       "\t<tr><td>13</td><td>Butyricimonas paravirosa  </td></tr>\n",
       "\t<tr><td>13</td><td>Butyricimonas virosa      </td></tr>\n",
       "\t<tr><td>13</td><td>Odoribacter sp. AF15-53   </td></tr>\n",
       "\t<tr><td>14</td><td>Butyricimonas             </td></tr>\n",
       "\t<tr><td>14</td><td>Butyricimonas synergistica</td></tr>\n",
       "\t<tr><td>14</td><td>Butyricimonas virosa      </td></tr>\n",
       "\t<tr><td>16</td><td>Butyricimonas             </td></tr>\n",
       "\t<tr><td>16</td><td>Butyricimonas virosa      </td></tr>\n",
       "\t<tr><td>16</td><td>Butyricimonas faecalis    </td></tr>\n",
       "\t<tr><td>18</td><td>Butyricimonas paravirosa  </td></tr>\n",
       "\t<tr><td>18</td><td>unclassified Butyricimonas</td></tr>\n",
       "\t<tr><td>18</td><td>Butyricimonas faecihominis</td></tr>\n",
       "\t<tr><td>22</td><td>Butyricimonas             </td></tr>\n",
       "\t<tr><td>22</td><td>unclassified Butyricimonas</td></tr>\n",
       "\t<tr><td>22</td><td>Odoribacter sp. AF15-53   </td></tr>\n",
       "\t<tr><td>24</td><td>Butyricimonas             </td></tr>\n",
       "\t<tr><td>24</td><td>Butyricimonas sp. NSJ-56  </td></tr>\n",
       "\t<tr><td>24</td><td>Odoribacter sp. AF15-53   </td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>486</td><td>Butyricimonas synergistica       </td></tr>\n",
       "\t<tr><td>486</td><td>Butyricimonas                    </td></tr>\n",
       "\t<tr><td>486</td><td>Odoribacter sp. AF15-53          </td></tr>\n",
       "\t<tr><td>487</td><td>Butyricimonas virosa             </td></tr>\n",
       "\t<tr><td>487</td><td>unclassified Butyricimonas       </td></tr>\n",
       "\t<tr><td>489</td><td>Odoribacteraceae                 </td></tr>\n",
       "\t<tr><td>489</td><td>Odoribacter sp. AF15-53          </td></tr>\n",
       "\t<tr><td>489</td><td>Sanguibacteroides justesenii     </td></tr>\n",
       "\t<tr><td>492</td><td>Butyricimonas sp. Marseille-P3923</td></tr>\n",
       "\t<tr><td>492</td><td>Butyricimonas sp. NSJ-56         </td></tr>\n",
       "\t<tr><td>492</td><td>Butyricimonas synergistica       </td></tr>\n",
       "\t<tr><td>494</td><td>Butyricimonas                    </td></tr>\n",
       "\t<tr><td>494</td><td>Butyricimonas virosa             </td></tr>\n",
       "\t<tr><td>495</td><td>Butyricimonas                    </td></tr>\n",
       "\t<tr><td>495</td><td>Butyricimonas virosa             </td></tr>\n",
       "\t<tr><td>496</td><td>Butyricimonas virosa             </td></tr>\n",
       "\t<tr><td>496</td><td>unclassified Butyricimonas       </td></tr>\n",
       "\t<tr><td>496</td><td>Butyricimonas paravirosa         </td></tr>\n",
       "\t<tr><td>498</td><td>Butyricimonas                    </td></tr>\n",
       "\t<tr><td>498</td><td>Odoribacter sp. AF15-53          </td></tr>\n",
       "\t<tr><td>499</td><td>Butyricimonas virosa             </td></tr>\n",
       "\t<tr><td>499</td><td>Butyricimonas                    </td></tr>\n",
       "\t<tr><td>500</td><td>Butyricimonas sp. NSJ-56         </td></tr>\n",
       "\t<tr><td>500</td><td>Butyricimonas sp. Marseille-P3923</td></tr>\n",
       "\t<tr><td>500</td><td>Butyricimonas                    </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 581 × 2\n",
       "\\begin{tabular}{ll}\n",
       " cluster\\_id & gene\\_taxonomy\\\\\n",
       " <dbl> & <chr>\\\\\n",
       "\\hline\n",
       "\t  4 & Butyricimonas paravirosa  \\\\\n",
       "\t  4 & Butyricimonas faecihominis\\\\\n",
       "\t  4 & Butyricimonas virosa      \\\\\n",
       "\t  7 & Butyricimonas             \\\\\n",
       "\t  7 & unclassified Butyricimonas\\\\\n",
       "\t  9 & Butyricimonas             \\\\\n",
       "\t  9 & Butyricimonas virosa      \\\\\n",
       "\t 13 & Butyricimonas paravirosa  \\\\\n",
       "\t 13 & Butyricimonas virosa      \\\\\n",
       "\t 13 & Odoribacter sp. AF15-53   \\\\\n",
       "\t 14 & Butyricimonas             \\\\\n",
       "\t 14 & Butyricimonas synergistica\\\\\n",
       "\t 14 & Butyricimonas virosa      \\\\\n",
       "\t 16 & Butyricimonas             \\\\\n",
       "\t 16 & Butyricimonas virosa      \\\\\n",
       "\t 16 & Butyricimonas faecalis    \\\\\n",
       "\t 18 & Butyricimonas paravirosa  \\\\\n",
       "\t 18 & unclassified Butyricimonas\\\\\n",
       "\t 18 & Butyricimonas faecihominis\\\\\n",
       "\t 22 & Butyricimonas             \\\\\n",
       "\t 22 & unclassified Butyricimonas\\\\\n",
       "\t 22 & Odoribacter sp. AF15-53   \\\\\n",
       "\t 24 & Butyricimonas             \\\\\n",
       "\t 24 & Butyricimonas sp. NSJ-56  \\\\\n",
       "\t 24 & Odoribacter sp. AF15-53   \\\\\n",
       "\t ⋮ & ⋮\\\\\n",
       "\t 486 & Butyricimonas synergistica       \\\\\n",
       "\t 486 & Butyricimonas                    \\\\\n",
       "\t 486 & Odoribacter sp. AF15-53          \\\\\n",
       "\t 487 & Butyricimonas virosa             \\\\\n",
       "\t 487 & unclassified Butyricimonas       \\\\\n",
       "\t 489 & Odoribacteraceae                 \\\\\n",
       "\t 489 & Odoribacter sp. AF15-53          \\\\\n",
       "\t 489 & Sanguibacteroides justesenii     \\\\\n",
       "\t 492 & Butyricimonas sp. Marseille-P3923\\\\\n",
       "\t 492 & Butyricimonas sp. NSJ-56         \\\\\n",
       "\t 492 & Butyricimonas synergistica       \\\\\n",
       "\t 494 & Butyricimonas                    \\\\\n",
       "\t 494 & Butyricimonas virosa             \\\\\n",
       "\t 495 & Butyricimonas                    \\\\\n",
       "\t 495 & Butyricimonas virosa             \\\\\n",
       "\t 496 & Butyricimonas virosa             \\\\\n",
       "\t 496 & unclassified Butyricimonas       \\\\\n",
       "\t 496 & Butyricimonas paravirosa         \\\\\n",
       "\t 498 & Butyricimonas                    \\\\\n",
       "\t 498 & Odoribacter sp. AF15-53          \\\\\n",
       "\t 499 & Butyricimonas virosa             \\\\\n",
       "\t 499 & Butyricimonas                    \\\\\n",
       "\t 500 & Butyricimonas sp. NSJ-56         \\\\\n",
       "\t 500 & Butyricimonas sp. Marseille-P3923\\\\\n",
       "\t 500 & Butyricimonas                    \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 581 × 2\n",
       "\n",
       "| cluster_id &lt;dbl&gt; | gene_taxonomy &lt;chr&gt; |\n",
       "|---|---|\n",
       "|  4 | Butyricimonas paravirosa   |\n",
       "|  4 | Butyricimonas faecihominis |\n",
       "|  4 | Butyricimonas virosa       |\n",
       "|  7 | Butyricimonas              |\n",
       "|  7 | unclassified Butyricimonas |\n",
       "|  9 | Butyricimonas              |\n",
       "|  9 | Butyricimonas virosa       |\n",
       "| 13 | Butyricimonas paravirosa   |\n",
       "| 13 | Butyricimonas virosa       |\n",
       "| 13 | Odoribacter sp. AF15-53    |\n",
       "| 14 | Butyricimonas              |\n",
       "| 14 | Butyricimonas synergistica |\n",
       "| 14 | Butyricimonas virosa       |\n",
       "| 16 | Butyricimonas              |\n",
       "| 16 | Butyricimonas virosa       |\n",
       "| 16 | Butyricimonas faecalis     |\n",
       "| 18 | Butyricimonas paravirosa   |\n",
       "| 18 | unclassified Butyricimonas |\n",
       "| 18 | Butyricimonas faecihominis |\n",
       "| 22 | Butyricimonas              |\n",
       "| 22 | unclassified Butyricimonas |\n",
       "| 22 | Odoribacter sp. AF15-53    |\n",
       "| 24 | Butyricimonas              |\n",
       "| 24 | Butyricimonas sp. NSJ-56   |\n",
       "| 24 | Odoribacter sp. AF15-53    |\n",
       "| ⋮ | ⋮ |\n",
       "| 486 | Butyricimonas synergistica        |\n",
       "| 486 | Butyricimonas                     |\n",
       "| 486 | Odoribacter sp. AF15-53           |\n",
       "| 487 | Butyricimonas virosa              |\n",
       "| 487 | unclassified Butyricimonas        |\n",
       "| 489 | Odoribacteraceae                  |\n",
       "| 489 | Odoribacter sp. AF15-53           |\n",
       "| 489 | Sanguibacteroides justesenii      |\n",
       "| 492 | Butyricimonas sp. Marseille-P3923 |\n",
       "| 492 | Butyricimonas sp. NSJ-56          |\n",
       "| 492 | Butyricimonas synergistica        |\n",
       "| 494 | Butyricimonas                     |\n",
       "| 494 | Butyricimonas virosa              |\n",
       "| 495 | Butyricimonas                     |\n",
       "| 495 | Butyricimonas virosa              |\n",
       "| 496 | Butyricimonas virosa              |\n",
       "| 496 | unclassified Butyricimonas        |\n",
       "| 496 | Butyricimonas paravirosa          |\n",
       "| 498 | Butyricimonas                     |\n",
       "| 498 | Odoribacter sp. AF15-53           |\n",
       "| 499 | Butyricimonas virosa              |\n",
       "| 499 | Butyricimonas                     |\n",
       "| 500 | Butyricimonas sp. NSJ-56          |\n",
       "| 500 | Butyricimonas sp. Marseille-P3923 |\n",
       "| 500 | Butyricimonas                     |\n",
       "\n"
      ],
      "text/plain": [
       "    cluster_id gene_taxonomy                    \n",
       "1    4         Butyricimonas paravirosa         \n",
       "2    4         Butyricimonas faecihominis       \n",
       "3    4         Butyricimonas virosa             \n",
       "4    7         Butyricimonas                    \n",
       "5    7         unclassified Butyricimonas       \n",
       "6    9         Butyricimonas                    \n",
       "7    9         Butyricimonas virosa             \n",
       "8   13         Butyricimonas paravirosa         \n",
       "9   13         Butyricimonas virosa             \n",
       "10  13         Odoribacter sp. AF15-53          \n",
       "11  14         Butyricimonas                    \n",
       "12  14         Butyricimonas synergistica       \n",
       "13  14         Butyricimonas virosa             \n",
       "14  16         Butyricimonas                    \n",
       "15  16         Butyricimonas virosa             \n",
       "16  16         Butyricimonas faecalis           \n",
       "17  18         Butyricimonas paravirosa         \n",
       "18  18         unclassified Butyricimonas       \n",
       "19  18         Butyricimonas faecihominis       \n",
       "20  22         Butyricimonas                    \n",
       "21  22         unclassified Butyricimonas       \n",
       "22  22         Odoribacter sp. AF15-53          \n",
       "23  24         Butyricimonas                    \n",
       "24  24         Butyricimonas sp. NSJ-56         \n",
       "25  24         Odoribacter sp. AF15-53          \n",
       "⋮   ⋮          ⋮                                \n",
       "557 486        Butyricimonas synergistica       \n",
       "558 486        Butyricimonas                    \n",
       "559 486        Odoribacter sp. AF15-53          \n",
       "560 487        Butyricimonas virosa             \n",
       "561 487        unclassified Butyricimonas       \n",
       "562 489        Odoribacteraceae                 \n",
       "563 489        Odoribacter sp. AF15-53          \n",
       "564 489        Sanguibacteroides justesenii     \n",
       "565 492        Butyricimonas sp. Marseille-P3923\n",
       "566 492        Butyricimonas sp. NSJ-56         \n",
       "567 492        Butyricimonas synergistica       \n",
       "568 494        Butyricimonas                    \n",
       "569 494        Butyricimonas virosa             \n",
       "570 495        Butyricimonas                    \n",
       "571 495        Butyricimonas virosa             \n",
       "572 496        Butyricimonas virosa             \n",
       "573 496        unclassified Butyricimonas       \n",
       "574 496        Butyricimonas paravirosa         \n",
       "575 498        Butyricimonas                    \n",
       "576 498        Odoribacter sp. AF15-53          \n",
       "577 499        Butyricimonas virosa             \n",
       "578 499        Butyricimonas                    \n",
       "579 500        Butyricimonas sp. NSJ-56         \n",
       "580 500        Butyricimonas sp. Marseille-P3923\n",
       "581 500        Butyricimonas                    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df.dims(50)\n",
    "gene_annot %>%\n",
    "    distinct(cluster_id, gene_taxonomy) \n",
    "df.dims()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Gene cluster: clostest related"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-19T14:39:09.445330Z",
     "iopub.status.busy": "2021-01-19T14:39:09.443349Z",
     "iopub.status.idle": "2021-01-19T14:39:09.882919Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 6955 × 19</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>cluster_id</th><th scope=col>query</th><th scope=col>subject</th><th scope=col>subject_name</th><th scope=col>pident</th><th scope=col>length</th><th scope=col>mismatch</th><th scope=col>qstart</th><th scope=col>qend</th><th scope=col>sstart</th><th scope=col>send</th><th scope=col>evalue</th><th scope=col>slen</th><th scope=col>qlen</th><th scope=col>sscinames</th><th scope=col>staxids</th><th scope=col>pident_rank</th><th scope=col>gene_name</th><th scope=col>gene_taxonomy</th></tr>\n",
       "\t<tr><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>4</td><td>4dd2fadf7df94a0babc664b2d3c2b220</td><td>RHR78904.1</td><td>None</td><td>98.276</td><td>406</td><td>7</td><td> 1</td><td>1218</td><td>1</td><td>406</td><td>0</td><td>406</td><td>1221</td><td>Odoribacter sp. AF15-53                        </td><td>2292236        </td><td>2</td><td>None</td><td>None</td></tr>\n",
       "\t<tr><td>4</td><td>4dd2fadf7df94a0babc664b2d3c2b220</td><td>RGG42580.1</td><td>None</td><td>99.007</td><td>403</td><td>4</td><td>10</td><td>1218</td><td>1</td><td>403</td><td>0</td><td>403</td><td>1221</td><td>Odoribacter sp. AM16-33;Odoribacter sp. AF21-41</td><td>2292048;2293111</td><td>1</td><td>None</td><td>None</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>500</td><td>466e10a5838d4b9ebeea2eb8446771b8</td><td>OHD46869.1  </td><td>ferredoxin [Spirochaetes bacterium GWF1_31_7]   </td><td>49.888</td><td>445</td><td>221</td><td>10</td><td>1341</td><td>6</td><td>449</td><td>1.12e-167</td><td>461</td><td>1344</td><td>Spirochaetes bacterium GWF1_31_7</td><td>1802187</td><td>29</td><td>ferredoxin            </td><td>Spirochaetes bacterium GWF1_31_7</td></tr>\n",
       "\t<tr><td>500</td><td>466e10a5838d4b9ebeea2eb8446771b8</td><td>MBI9053771.1</td><td>4Fe-4S binding protein [Bacteroidales bacterium]</td><td>49.213</td><td>445</td><td>223</td><td>10</td><td>1341</td><td>6</td><td>448</td><td>1.18e-167</td><td>450</td><td>1344</td><td>Bacteroidales bacterium         </td><td>2030927</td><td>30</td><td>4Fe-4S binding protein</td><td>Bacteroidales bacterium         </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6955 × 19\n",
       "\\begin{tabular}{lllllllllllllllllll}\n",
       " cluster\\_id & query & subject & subject\\_name & pident & length & mismatch & qstart & qend & sstart & send & evalue & slen & qlen & sscinames & staxids & pident\\_rank & gene\\_name & gene\\_taxonomy\\\\\n",
       " <dbl> & <fct> & <fct> & <fct> & <dbl> & <int> & <int> & <int> & <int> & <int> & <int> & <dbl> & <int> & <int> & <fct> & <fct> & <int> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t 4 & 4dd2fadf7df94a0babc664b2d3c2b220 & RHR78904.1 & None & 98.276 & 406 & 7 &  1 & 1218 & 1 & 406 & 0 & 406 & 1221 & Odoribacter sp. AF15-53                         & 2292236         & 2 & None & None\\\\\n",
       "\t 4 & 4dd2fadf7df94a0babc664b2d3c2b220 & RGG42580.1 & None & 99.007 & 403 & 4 & 10 & 1218 & 1 & 403 & 0 & 403 & 1221 & Odoribacter sp. AM16-33;Odoribacter sp. AF21-41 & 2292048;2293111 & 1 & None & None\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t 500 & 466e10a5838d4b9ebeea2eb8446771b8 & OHD46869.1   & ferredoxin {[}Spirochaetes bacterium GWF1\\_31\\_7{]}    & 49.888 & 445 & 221 & 10 & 1341 & 6 & 449 & 1.12e-167 & 461 & 1344 & Spirochaetes bacterium GWF1\\_31\\_7 & 1802187 & 29 & ferredoxin             & Spirochaetes bacterium GWF1\\_31\\_7\\\\\n",
       "\t 500 & 466e10a5838d4b9ebeea2eb8446771b8 & MBI9053771.1 & 4Fe-4S binding protein {[}Bacteroidales bacterium{]} & 49.213 & 445 & 223 & 10 & 1341 & 6 & 448 & 1.18e-167 & 450 & 1344 & Bacteroidales bacterium          & 2030927 & 30 & 4Fe-4S binding protein & Bacteroidales bacterium         \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6955 × 19\n",
       "\n",
       "| cluster_id &lt;dbl&gt; | query &lt;fct&gt; | subject &lt;fct&gt; | subject_name &lt;fct&gt; | pident &lt;dbl&gt; | length &lt;int&gt; | mismatch &lt;int&gt; | qstart &lt;int&gt; | qend &lt;int&gt; | sstart &lt;int&gt; | send &lt;int&gt; | evalue &lt;dbl&gt; | slen &lt;int&gt; | qlen &lt;int&gt; | sscinames &lt;fct&gt; | staxids &lt;fct&gt; | pident_rank &lt;int&gt; | gene_name &lt;chr&gt; | gene_taxonomy &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 4 | 4dd2fadf7df94a0babc664b2d3c2b220 | RHR78904.1 | None | 98.276 | 406 | 7 |  1 | 1218 | 1 | 406 | 0 | 406 | 1221 | Odoribacter sp. AF15-53                         | 2292236         | 2 | None | None |\n",
       "| 4 | 4dd2fadf7df94a0babc664b2d3c2b220 | RGG42580.1 | None | 99.007 | 403 | 4 | 10 | 1218 | 1 | 403 | 0 | 403 | 1221 | Odoribacter sp. AM16-33;Odoribacter sp. AF21-41 | 2292048;2293111 | 1 | None | None |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| 500 | 466e10a5838d4b9ebeea2eb8446771b8 | OHD46869.1   | ferredoxin [Spirochaetes bacterium GWF1_31_7]    | 49.888 | 445 | 221 | 10 | 1341 | 6 | 449 | 1.12e-167 | 461 | 1344 | Spirochaetes bacterium GWF1_31_7 | 1802187 | 29 | ferredoxin             | Spirochaetes bacterium GWF1_31_7 |\n",
       "| 500 | 466e10a5838d4b9ebeea2eb8446771b8 | MBI9053771.1 | 4Fe-4S binding protein [Bacteroidales bacterium] | 49.213 | 445 | 223 | 10 | 1341 | 6 | 448 | 1.18e-167 | 450 | 1344 | Bacteroidales bacterium          | 2030927 | 30 | 4Fe-4S binding protein | Bacteroidales bacterium          |\n",
       "\n"
      ],
      "text/plain": [
       "     cluster_id query                            subject     \n",
       "1    4          4dd2fadf7df94a0babc664b2d3c2b220 RHR78904.1  \n",
       "2    4          4dd2fadf7df94a0babc664b2d3c2b220 RGG42580.1  \n",
       "⋮    ⋮          ⋮                                ⋮           \n",
       "6954 500        466e10a5838d4b9ebeea2eb8446771b8 OHD46869.1  \n",
       "6955 500        466e10a5838d4b9ebeea2eb8446771b8 MBI9053771.1\n",
       "     subject_name                                     pident length mismatch\n",
       "1    None                                             98.276 406    7       \n",
       "2    None                                             99.007 403    4       \n",
       "⋮    ⋮                                                ⋮      ⋮      ⋮       \n",
       "6954 ferredoxin [Spirochaetes bacterium GWF1_31_7]    49.888 445    221     \n",
       "6955 4Fe-4S binding protein [Bacteroidales bacterium] 49.213 445    223     \n",
       "     qstart qend sstart send evalue    slen qlen\n",
       "1     1     1218 1      406  0         406  1221\n",
       "2    10     1218 1      403  0         403  1221\n",
       "⋮    ⋮      ⋮    ⋮      ⋮    ⋮         ⋮    ⋮   \n",
       "6954 10     1341 6      449  1.12e-167 461  1344\n",
       "6955 10     1341 6      448  1.18e-167 450  1344\n",
       "     sscinames                                       staxids        \n",
       "1    Odoribacter sp. AF15-53                         2292236        \n",
       "2    Odoribacter sp. AM16-33;Odoribacter sp. AF21-41 2292048;2293111\n",
       "⋮    ⋮                                               ⋮              \n",
       "6954 Spirochaetes bacterium GWF1_31_7                1802187        \n",
       "6955 Bacteroidales bacterium                         2030927        \n",
       "     pident_rank gene_name              gene_taxonomy                   \n",
       "1    2           None                   None                            \n",
       "2    1           None                   None                            \n",
       "⋮    ⋮           ⋮                      ⋮                               \n",
       "6954 29          ferredoxin             Spirochaetes bacterium GWF1_31_7\n",
       "6955 30          4Fe-4S binding protein Bacteroidales bacterium         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "gene_annot = read.delim(file.path(work_dir, 'primers', 'cgp', 'core_clusters_blastx_nontarget.tsv'), \n",
    "                        sep='\\t') %>%\n",
    "    mutate(cluster_id = gsub('cluster_', '', cluster_id) %>% as.Num) %>%\n",
    "    semi_join(primer_info, c('cluster_id')) %>%\n",
    "    mutate(gene_name = gsub(' \\\\[.+', '', subject_name),\n",
    "           gene_taxonomy = gsub('.+\\\\[', '', subject_name),\n",
    "           gene_taxonomy = gsub('\\\\]', '', gene_taxonomy))\n",
    "gene_annot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-19T14:39:15.354645Z",
     "iopub.status.busy": "2021-01-19T14:39:15.352215Z",
     "iopub.status.idle": "2021-01-19T14:39:15.496792Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 647 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>cluster_id</th><th scope=col>gene_name</th><th scope=col>gene_taxonomy</th><th scope=col>pident</th></tr>\n",
       "\t<tr><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td> 4</td><td>None                                                                                             </td><td>None                        </td><td>98.276</td></tr>\n",
       "\t<tr><td> 4</td><td>None                                                                                             </td><td>None                        </td><td>99.007</td></tr>\n",
       "\t<tr><td> 4</td><td>None                                                                                             </td><td>None                        </td><td>98.263</td></tr>\n",
       "\t<tr><td> 7</td><td>None                                                                                             </td><td>None                        </td><td>91.860</td></tr>\n",
       "\t<tr><td> 7</td><td>50S ribosomal protein L27                                                                        </td><td>Odoribacter sp. AF15-53     </td><td>89.535</td></tr>\n",
       "\t<tr><td> 7</td><td>50S ribosomal protein L27                                                                        </td><td>Sanguibacteroides justesenii</td><td>84.884</td></tr>\n",
       "\t<tr><td> 7</td><td>50S ribosomal protein L27                                                                        </td><td>bacterium                   </td><td>84.884</td></tr>\n",
       "\t<tr><td> 9</td><td>None                                                                                             </td><td>None                        </td><td>97.938</td></tr>\n",
       "\t<tr><td> 9</td><td>bifunctional methylenetetrahydrofolate dehydrogenase/methenyltetrahydrofolate cyclohydrolase FolD</td><td>Odoribacter sp. AF15-53     </td><td>94.158</td></tr>\n",
       "\t<tr><td> 9</td><td>bifunctional methylenetetrahydrofolate dehydrogenase/methenyltetrahydrofolate cyclohydrolase FolD</td><td>Sanguibacteroides justesenii</td><td>83.505</td></tr>\n",
       "\t<tr><td>13</td><td>NADH:ubiquinone reductase (Na(+)-transporting) subunit E                                         </td><td>Odoribacter sp. AF15-53     </td><td>99.029</td></tr>\n",
       "\t<tr><td>13</td><td>NADH:ubiquinone reductase (Na(+)-transporting) subunit E                                         </td><td>Sanguibacteroides justesenii</td><td>95.631</td></tr>\n",
       "\t<tr><td>13</td><td>NADH:ubiquinone reductase (Na(+)-transporting) subunit E                                         </td><td>Odoribacter splanchnicus    </td><td>93.204</td></tr>\n",
       "\t<tr><td>14</td><td>homocysteine biosynthesis protein                                                                </td><td>Odoribacter sp. AF15-53     </td><td>97.090</td></tr>\n",
       "\t<tr><td>14</td><td>homocysteine biosynthesis protein                                                                </td><td>Sanguibacteroides justesenii</td><td>88.684</td></tr>\n",
       "\t<tr><td>14</td><td>hypothetical protein                                                                             </td><td>Odoribacter sp. Z80         </td><td>86.364</td></tr>\n",
       "\t<tr><td>16</td><td>DNA polymerase III subunit delta                                                                 </td><td>Odoribacter sp. AF15-53     </td><td>97.024</td></tr>\n",
       "\t<tr><td>16</td><td>DNA polymerase III subunit delta                                                                 </td><td>Sanguibacteroides justesenii</td><td>83.929</td></tr>\n",
       "\t<tr><td>16</td><td>DNA polymerase III subunit delta                                                                 </td><td>Sanguibacteroides justesenii</td><td>83.333</td></tr>\n",
       "\t<tr><td>18</td><td>V-type ATP synthase subunit A                                                                    </td><td>Odoribacter sp. AF21-41     </td><td>99.658</td></tr>\n",
       "\t<tr><td>18</td><td>V-type ATP synthase subunit A                                                                    </td><td>Odoribacter sp. AF15-53     </td><td>96.928</td></tr>\n",
       "\t<tr><td>18</td><td>V-type ATP synthase subunit A                                                                    </td><td>Sanguibacteroides justesenii</td><td>93.162</td></tr>\n",
       "\t<tr><td>22</td><td>30S ribosomal protein S20                                                                        </td><td>Odoribacter sp. AF15-53     </td><td>91.566</td></tr>\n",
       "\t<tr><td>22</td><td>None                                                                                             </td><td>None                        </td><td>91.566</td></tr>\n",
       "\t<tr><td>22</td><td>30S ribosomal protein S20                                                                        </td><td>Sanguibacteroides justesenii</td><td>90.361</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>486</td><td>30S ribosomal protein S16                 </td><td>Sanguibacteroides justesenii        </td><td>85.714</td></tr>\n",
       "\t<tr><td>487</td><td>None                                      </td><td>None                                </td><td>99.083</td></tr>\n",
       "\t<tr><td>487</td><td>tryptophan--tRNA ligase                   </td><td>Odoribacter sp. AF15-53             </td><td>92.355</td></tr>\n",
       "\t<tr><td>487</td><td>tryptophan--tRNA ligase                   </td><td>Sanguibacteroides justesenii        </td><td>83.486</td></tr>\n",
       "\t<tr><td>489</td><td>30S ribosomal protein S10                 </td><td>Odoribacter sp. AF15-53             </td><td>99.010</td></tr>\n",
       "\t<tr><td>489</td><td>30S ribosomal protein S10                 </td><td>Sanguibacteroides justesenii        </td><td>99.010</td></tr>\n",
       "\t<tr><td>489</td><td>30S ribosomal protein S10                 </td><td>Odoribacter sp. Z80                 </td><td>98.020</td></tr>\n",
       "\t<tr><td>492</td><td>phosphoribosylformylglycinamidine synthase</td><td>Odoribacter sp. AF15-53             </td><td>90.968</td></tr>\n",
       "\t<tr><td>492</td><td>None                                      </td><td>None                                </td><td>90.480</td></tr>\n",
       "\t<tr><td>492</td><td>phosphoribosylformylglycinamidine synthase</td><td>Sanguibacteroides justesenii        </td><td>84.459</td></tr>\n",
       "\t<tr><td>494</td><td>cell division protein FtsZ                </td><td>Odoribacter sp. AF15-53             </td><td>93.246</td></tr>\n",
       "\t<tr><td>495</td><td>M23 family metallopeptidase               </td><td>Odoribacter sp. AF15-53             </td><td>92.708</td></tr>\n",
       "\t<tr><td>496</td><td>None                                      </td><td>None                                </td><td>99.615</td></tr>\n",
       "\t<tr><td>496</td><td>acyl-CoA carboxylase subunit beta         </td><td>Sanguibacteroides justesenii        </td><td>97.110</td></tr>\n",
       "\t<tr><td>496</td><td>acyl-CoA carboxylase subunit beta         </td><td>Sanguibacteroides justesenii        </td><td>97.110</td></tr>\n",
       "\t<tr><td>498</td><td>None                                      </td><td>None                                </td><td>98.319</td></tr>\n",
       "\t<tr><td>498</td><td>50S ribosomal protein L17                 </td><td>Sanguibacteroides justesenii        </td><td>83.333</td></tr>\n",
       "\t<tr><td>498</td><td>50S ribosomal protein L17                 </td><td>Bacteroidia bacterium               </td><td>83.193</td></tr>\n",
       "\t<tr><td>498</td><td>50S ribosomal protein L17                 </td><td>Sphingobacteriales bacterium TSM_CSS</td><td>83.193</td></tr>\n",
       "\t<tr><td>499</td><td>None                                      </td><td>None                                </td><td>99.065</td></tr>\n",
       "\t<tr><td>499</td><td>DUF2149 domain-containing protein         </td><td>Odoribacter sp. AF15-53             </td><td>96.262</td></tr>\n",
       "\t<tr><td>499</td><td>DUF2149 domain-containing protein         </td><td>Lentimicrobiaceae bacterium         </td><td>88.889</td></tr>\n",
       "\t<tr><td>500</td><td>4Fe-4S binding protein                    </td><td>Odoribacter sp. AF15-53             </td><td>87.946</td></tr>\n",
       "\t<tr><td>500</td><td>None                                      </td><td>None                                </td><td>86.771</td></tr>\n",
       "\t<tr><td>500</td><td>4Fe-4S binding protein                    </td><td>Sanguibacteroides justesenii        </td><td>80.269</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 647 × 4\n",
       "\\begin{tabular}{llll}\n",
       " cluster\\_id & gene\\_name & gene\\_taxonomy & pident\\\\\n",
       " <dbl> & <chr> & <chr> & <dbl>\\\\\n",
       "\\hline\n",
       "\t  4 & None                                                                                              & None                         & 98.276\\\\\n",
       "\t  4 & None                                                                                              & None                         & 99.007\\\\\n",
       "\t  4 & None                                                                                              & None                         & 98.263\\\\\n",
       "\t  7 & None                                                                                              & None                         & 91.860\\\\\n",
       "\t  7 & 50S ribosomal protein L27                                                                         & Odoribacter sp. AF15-53      & 89.535\\\\\n",
       "\t  7 & 50S ribosomal protein L27                                                                         & Sanguibacteroides justesenii & 84.884\\\\\n",
       "\t  7 & 50S ribosomal protein L27                                                                         & bacterium                    & 84.884\\\\\n",
       "\t  9 & None                                                                                              & None                         & 97.938\\\\\n",
       "\t  9 & bifunctional methylenetetrahydrofolate dehydrogenase/methenyltetrahydrofolate cyclohydrolase FolD & Odoribacter sp. AF15-53      & 94.158\\\\\n",
       "\t  9 & bifunctional methylenetetrahydrofolate dehydrogenase/methenyltetrahydrofolate cyclohydrolase FolD & Sanguibacteroides justesenii & 83.505\\\\\n",
       "\t 13 & NADH:ubiquinone reductase (Na(+)-transporting) subunit E                                          & Odoribacter sp. AF15-53      & 99.029\\\\\n",
       "\t 13 & NADH:ubiquinone reductase (Na(+)-transporting) subunit E                                          & Sanguibacteroides justesenii & 95.631\\\\\n",
       "\t 13 & NADH:ubiquinone reductase (Na(+)-transporting) subunit E                                          & Odoribacter splanchnicus     & 93.204\\\\\n",
       "\t 14 & homocysteine biosynthesis protein                                                                 & Odoribacter sp. AF15-53      & 97.090\\\\\n",
       "\t 14 & homocysteine biosynthesis protein                                                                 & Sanguibacteroides justesenii & 88.684\\\\\n",
       "\t 14 & hypothetical protein                                                                              & Odoribacter sp. Z80          & 86.364\\\\\n",
       "\t 16 & DNA polymerase III subunit delta                                                                  & Odoribacter sp. AF15-53      & 97.024\\\\\n",
       "\t 16 & DNA polymerase III subunit delta                                                                  & Sanguibacteroides justesenii & 83.929\\\\\n",
       "\t 16 & DNA polymerase III subunit delta                                                                  & Sanguibacteroides justesenii & 83.333\\\\\n",
       "\t 18 & V-type ATP synthase subunit A                                                                     & Odoribacter sp. AF21-41      & 99.658\\\\\n",
       "\t 18 & V-type ATP synthase subunit A                                                                     & Odoribacter sp. AF15-53      & 96.928\\\\\n",
       "\t 18 & V-type ATP synthase subunit A                                                                     & Sanguibacteroides justesenii & 93.162\\\\\n",
       "\t 22 & 30S ribosomal protein S20                                                                         & Odoribacter sp. AF15-53      & 91.566\\\\\n",
       "\t 22 & None                                                                                              & None                         & 91.566\\\\\n",
       "\t 22 & 30S ribosomal protein S20                                                                         & Sanguibacteroides justesenii & 90.361\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t 486 & 30S ribosomal protein S16                  & Sanguibacteroides justesenii         & 85.714\\\\\n",
       "\t 487 & None                                       & None                                 & 99.083\\\\\n",
       "\t 487 & tryptophan--tRNA ligase                    & Odoribacter sp. AF15-53              & 92.355\\\\\n",
       "\t 487 & tryptophan--tRNA ligase                    & Sanguibacteroides justesenii         & 83.486\\\\\n",
       "\t 489 & 30S ribosomal protein S10                  & Odoribacter sp. AF15-53              & 99.010\\\\\n",
       "\t 489 & 30S ribosomal protein S10                  & Sanguibacteroides justesenii         & 99.010\\\\\n",
       "\t 489 & 30S ribosomal protein S10                  & Odoribacter sp. Z80                  & 98.020\\\\\n",
       "\t 492 & phosphoribosylformylglycinamidine synthase & Odoribacter sp. AF15-53              & 90.968\\\\\n",
       "\t 492 & None                                       & None                                 & 90.480\\\\\n",
       "\t 492 & phosphoribosylformylglycinamidine synthase & Sanguibacteroides justesenii         & 84.459\\\\\n",
       "\t 494 & cell division protein FtsZ                 & Odoribacter sp. AF15-53              & 93.246\\\\\n",
       "\t 495 & M23 family metallopeptidase                & Odoribacter sp. AF15-53              & 92.708\\\\\n",
       "\t 496 & None                                       & None                                 & 99.615\\\\\n",
       "\t 496 & acyl-CoA carboxylase subunit beta          & Sanguibacteroides justesenii         & 97.110\\\\\n",
       "\t 496 & acyl-CoA carboxylase subunit beta          & Sanguibacteroides justesenii         & 97.110\\\\\n",
       "\t 498 & None                                       & None                                 & 98.319\\\\\n",
       "\t 498 & 50S ribosomal protein L17                  & Sanguibacteroides justesenii         & 83.333\\\\\n",
       "\t 498 & 50S ribosomal protein L17                  & Bacteroidia bacterium                & 83.193\\\\\n",
       "\t 498 & 50S ribosomal protein L17                  & Sphingobacteriales bacterium TSM\\_CSS & 83.193\\\\\n",
       "\t 499 & None                                       & None                                 & 99.065\\\\\n",
       "\t 499 & DUF2149 domain-containing protein          & Odoribacter sp. AF15-53              & 96.262\\\\\n",
       "\t 499 & DUF2149 domain-containing protein          & Lentimicrobiaceae bacterium          & 88.889\\\\\n",
       "\t 500 & 4Fe-4S binding protein                     & Odoribacter sp. AF15-53              & 87.946\\\\\n",
       "\t 500 & None                                       & None                                 & 86.771\\\\\n",
       "\t 500 & 4Fe-4S binding protein                     & Sanguibacteroides justesenii         & 80.269\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 647 × 4\n",
       "\n",
       "| cluster_id &lt;dbl&gt; | gene_name &lt;chr&gt; | gene_taxonomy &lt;chr&gt; | pident &lt;dbl&gt; |\n",
       "|---|---|---|---|\n",
       "|  4 | None                                                                                              | None                         | 98.276 |\n",
       "|  4 | None                                                                                              | None                         | 99.007 |\n",
       "|  4 | None                                                                                              | None                         | 98.263 |\n",
       "|  7 | None                                                                                              | None                         | 91.860 |\n",
       "|  7 | 50S ribosomal protein L27                                                                         | Odoribacter sp. AF15-53      | 89.535 |\n",
       "|  7 | 50S ribosomal protein L27                                                                         | Sanguibacteroides justesenii | 84.884 |\n",
       "|  7 | 50S ribosomal protein L27                                                                         | bacterium                    | 84.884 |\n",
       "|  9 | None                                                                                              | None                         | 97.938 |\n",
       "|  9 | bifunctional methylenetetrahydrofolate dehydrogenase/methenyltetrahydrofolate cyclohydrolase FolD | Odoribacter sp. AF15-53      | 94.158 |\n",
       "|  9 | bifunctional methylenetetrahydrofolate dehydrogenase/methenyltetrahydrofolate cyclohydrolase FolD | Sanguibacteroides justesenii | 83.505 |\n",
       "| 13 | NADH:ubiquinone reductase (Na(+)-transporting) subunit E                                          | Odoribacter sp. AF15-53      | 99.029 |\n",
       "| 13 | NADH:ubiquinone reductase (Na(+)-transporting) subunit E                                          | Sanguibacteroides justesenii | 95.631 |\n",
       "| 13 | NADH:ubiquinone reductase (Na(+)-transporting) subunit E                                          | Odoribacter splanchnicus     | 93.204 |\n",
       "| 14 | homocysteine biosynthesis protein                                                                 | Odoribacter sp. AF15-53      | 97.090 |\n",
       "| 14 | homocysteine biosynthesis protein                                                                 | Sanguibacteroides justesenii | 88.684 |\n",
       "| 14 | hypothetical protein                                                                              | Odoribacter sp. Z80          | 86.364 |\n",
       "| 16 | DNA polymerase III subunit delta                                                                  | Odoribacter sp. AF15-53      | 97.024 |\n",
       "| 16 | DNA polymerase III subunit delta                                                                  | Sanguibacteroides justesenii | 83.929 |\n",
       "| 16 | DNA polymerase III subunit delta                                                                  | Sanguibacteroides justesenii | 83.333 |\n",
       "| 18 | V-type ATP synthase subunit A                                                                     | Odoribacter sp. AF21-41      | 99.658 |\n",
       "| 18 | V-type ATP synthase subunit A                                                                     | Odoribacter sp. AF15-53      | 96.928 |\n",
       "| 18 | V-type ATP synthase subunit A                                                                     | Sanguibacteroides justesenii | 93.162 |\n",
       "| 22 | 30S ribosomal protein S20                                                                         | Odoribacter sp. AF15-53      | 91.566 |\n",
       "| 22 | None                                                                                              | None                         | 91.566 |\n",
       "| 22 | 30S ribosomal protein S20                                                                         | Sanguibacteroides justesenii | 90.361 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| 486 | 30S ribosomal protein S16                  | Sanguibacteroides justesenii         | 85.714 |\n",
       "| 487 | None                                       | None                                 | 99.083 |\n",
       "| 487 | tryptophan--tRNA ligase                    | Odoribacter sp. AF15-53              | 92.355 |\n",
       "| 487 | tryptophan--tRNA ligase                    | Sanguibacteroides justesenii         | 83.486 |\n",
       "| 489 | 30S ribosomal protein S10                  | Odoribacter sp. AF15-53              | 99.010 |\n",
       "| 489 | 30S ribosomal protein S10                  | Sanguibacteroides justesenii         | 99.010 |\n",
       "| 489 | 30S ribosomal protein S10                  | Odoribacter sp. Z80                  | 98.020 |\n",
       "| 492 | phosphoribosylformylglycinamidine synthase | Odoribacter sp. AF15-53              | 90.968 |\n",
       "| 492 | None                                       | None                                 | 90.480 |\n",
       "| 492 | phosphoribosylformylglycinamidine synthase | Sanguibacteroides justesenii         | 84.459 |\n",
       "| 494 | cell division protein FtsZ                 | Odoribacter sp. AF15-53              | 93.246 |\n",
       "| 495 | M23 family metallopeptidase                | Odoribacter sp. AF15-53              | 92.708 |\n",
       "| 496 | None                                       | None                                 | 99.615 |\n",
       "| 496 | acyl-CoA carboxylase subunit beta          | Sanguibacteroides justesenii         | 97.110 |\n",
       "| 496 | acyl-CoA carboxylase subunit beta          | Sanguibacteroides justesenii         | 97.110 |\n",
       "| 498 | None                                       | None                                 | 98.319 |\n",
       "| 498 | 50S ribosomal protein L17                  | Sanguibacteroides justesenii         | 83.333 |\n",
       "| 498 | 50S ribosomal protein L17                  | Bacteroidia bacterium                | 83.193 |\n",
       "| 498 | 50S ribosomal protein L17                  | Sphingobacteriales bacterium TSM_CSS | 83.193 |\n",
       "| 499 | None                                       | None                                 | 99.065 |\n",
       "| 499 | DUF2149 domain-containing protein          | Odoribacter sp. AF15-53              | 96.262 |\n",
       "| 499 | DUF2149 domain-containing protein          | Lentimicrobiaceae bacterium          | 88.889 |\n",
       "| 500 | 4Fe-4S binding protein                     | Odoribacter sp. AF15-53              | 87.946 |\n",
       "| 500 | None                                       | None                                 | 86.771 |\n",
       "| 500 | 4Fe-4S binding protein                     | Sanguibacteroides justesenii         | 80.269 |\n",
       "\n"
      ],
      "text/plain": [
       "    cluster_id\n",
       "1    4        \n",
       "2    4        \n",
       "3    4        \n",
       "4    7        \n",
       "5    7        \n",
       "6    7        \n",
       "7    7        \n",
       "8    9        \n",
       "9    9        \n",
       "10   9        \n",
       "11  13        \n",
       "12  13        \n",
       "13  13        \n",
       "14  14        \n",
       "15  14        \n",
       "16  14        \n",
       "17  16        \n",
       "18  16        \n",
       "19  16        \n",
       "20  18        \n",
       "21  18        \n",
       "22  18        \n",
       "23  22        \n",
       "24  22        \n",
       "25  22        \n",
       "⋮   ⋮         \n",
       "623 486       \n",
       "624 487       \n",
       "625 487       \n",
       "626 487       \n",
       "627 489       \n",
       "628 489       \n",
       "629 489       \n",
       "630 492       \n",
       "631 492       \n",
       "632 492       \n",
       "633 494       \n",
       "634 495       \n",
       "635 496       \n",
       "636 496       \n",
       "637 496       \n",
       "638 498       \n",
       "639 498       \n",
       "640 498       \n",
       "641 498       \n",
       "642 499       \n",
       "643 499       \n",
       "644 499       \n",
       "645 500       \n",
       "646 500       \n",
       "647 500       \n",
       "    gene_name                                                                                        \n",
       "1   None                                                                                             \n",
       "2   None                                                                                             \n",
       "3   None                                                                                             \n",
       "4   None                                                                                             \n",
       "5   50S ribosomal protein L27                                                                        \n",
       "6   50S ribosomal protein L27                                                                        \n",
       "7   50S ribosomal protein L27                                                                        \n",
       "8   None                                                                                             \n",
       "9   bifunctional methylenetetrahydrofolate dehydrogenase/methenyltetrahydrofolate cyclohydrolase FolD\n",
       "10  bifunctional methylenetetrahydrofolate dehydrogenase/methenyltetrahydrofolate cyclohydrolase FolD\n",
       "11  NADH:ubiquinone reductase (Na(+)-transporting) subunit E                                         \n",
       "12  NADH:ubiquinone reductase (Na(+)-transporting) subunit E                                         \n",
       "13  NADH:ubiquinone reductase (Na(+)-transporting) subunit E                                         \n",
       "14  homocysteine biosynthesis protein                                                                \n",
       "15  homocysteine biosynthesis protein                                                                \n",
       "16  hypothetical protein                                                                             \n",
       "17  DNA polymerase III subunit delta                                                                 \n",
       "18  DNA polymerase III subunit delta                                                                 \n",
       "19  DNA polymerase III subunit delta                                                                 \n",
       "20  V-type ATP synthase subunit A                                                                    \n",
       "21  V-type ATP synthase subunit A                                                                    \n",
       "22  V-type ATP synthase subunit A                                                                    \n",
       "23  30S ribosomal protein S20                                                                        \n",
       "24  None                                                                                             \n",
       "25  30S ribosomal protein S20                                                                        \n",
       "⋮   ⋮                                                                                                \n",
       "623 30S ribosomal protein S16                                                                        \n",
       "624 None                                                                                             \n",
       "625 tryptophan--tRNA ligase                                                                          \n",
       "626 tryptophan--tRNA ligase                                                                          \n",
       "627 30S ribosomal protein S10                                                                        \n",
       "628 30S ribosomal protein S10                                                                        \n",
       "629 30S ribosomal protein S10                                                                        \n",
       "630 phosphoribosylformylglycinamidine synthase                                                       \n",
       "631 None                                                                                             \n",
       "632 phosphoribosylformylglycinamidine synthase                                                       \n",
       "633 cell division protein FtsZ                                                                       \n",
       "634 M23 family metallopeptidase                                                                      \n",
       "635 None                                                                                             \n",
       "636 acyl-CoA carboxylase subunit beta                                                                \n",
       "637 acyl-CoA carboxylase subunit beta                                                                \n",
       "638 None                                                                                             \n",
       "639 50S ribosomal protein L17                                                                        \n",
       "640 50S ribosomal protein L17                                                                        \n",
       "641 50S ribosomal protein L17                                                                        \n",
       "642 None                                                                                             \n",
       "643 DUF2149 domain-containing protein                                                                \n",
       "644 DUF2149 domain-containing protein                                                                \n",
       "645 4Fe-4S binding protein                                                                           \n",
       "646 None                                                                                             \n",
       "647 4Fe-4S binding protein                                                                           \n",
       "    gene_taxonomy                        pident\n",
       "1   None                                 98.276\n",
       "2   None                                 99.007\n",
       "3   None                                 98.263\n",
       "4   None                                 91.860\n",
       "5   Odoribacter sp. AF15-53              89.535\n",
       "6   Sanguibacteroides justesenii         84.884\n",
       "7   bacterium                            84.884\n",
       "8   None                                 97.938\n",
       "9   Odoribacter sp. AF15-53              94.158\n",
       "10  Sanguibacteroides justesenii         83.505\n",
       "11  Odoribacter sp. AF15-53              99.029\n",
       "12  Sanguibacteroides justesenii         95.631\n",
       "13  Odoribacter splanchnicus             93.204\n",
       "14  Odoribacter sp. AF15-53              97.090\n",
       "15  Sanguibacteroides justesenii         88.684\n",
       "16  Odoribacter sp. Z80                  86.364\n",
       "17  Odoribacter sp. AF15-53              97.024\n",
       "18  Sanguibacteroides justesenii         83.929\n",
       "19  Sanguibacteroides justesenii         83.333\n",
       "20  Odoribacter sp. AF21-41              99.658\n",
       "21  Odoribacter sp. AF15-53              96.928\n",
       "22  Sanguibacteroides justesenii         93.162\n",
       "23  Odoribacter sp. AF15-53              91.566\n",
       "24  None                                 91.566\n",
       "25  Sanguibacteroides justesenii         90.361\n",
       "⋮   ⋮                                    ⋮     \n",
       "623 Sanguibacteroides justesenii         85.714\n",
       "624 None                                 99.083\n",
       "625 Odoribacter sp. AF15-53              92.355\n",
       "626 Sanguibacteroides justesenii         83.486\n",
       "627 Odoribacter sp. AF15-53              99.010\n",
       "628 Sanguibacteroides justesenii         99.010\n",
       "629 Odoribacter sp. Z80                  98.020\n",
       "630 Odoribacter sp. AF15-53              90.968\n",
       "631 None                                 90.480\n",
       "632 Sanguibacteroides justesenii         84.459\n",
       "633 Odoribacter sp. AF15-53              93.246\n",
       "634 Odoribacter sp. AF15-53              92.708\n",
       "635 None                                 99.615\n",
       "636 Sanguibacteroides justesenii         97.110\n",
       "637 Sanguibacteroides justesenii         97.110\n",
       "638 None                                 98.319\n",
       "639 Sanguibacteroides justesenii         83.333\n",
       "640 Bacteroidia bacterium                83.193\n",
       "641 Sphingobacteriales bacterium TSM_CSS 83.193\n",
       "642 None                                 99.065\n",
       "643 Odoribacter sp. AF15-53              96.262\n",
       "644 Lentimicrobiaceae bacterium          88.889\n",
       "645 Odoribacter sp. AF15-53              87.946\n",
       "646 None                                 86.771\n",
       "647 Sanguibacteroides justesenii         80.269"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df.dims(50)\n",
    "gene_annot %>%\n",
    "    filter(pident > 80,\n",
    "           pident_rank <= 3) %>%\n",
    "    select(cluster_id, gene_name, gene_taxonomy, pident)\n",
    "    \n",
    "df.dims()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# sessionInfo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "pipelineInfo('/ebio/abt3_projects/software/dev/ll_pipelines/llg/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "pipelineInfo('/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "sessionInfo()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R [conda env:Georg_animal_feces-tidyverse]",
   "language": "R",
   "name": "conda-env-Georg_animal_feces-tidyverse-r"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.3"
  },
  "toc-autonumbering": true
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
