{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Goal\n",
    "\n",
    "* Primer design for clade of interest"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Var"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-22T08:53:07.606391Z",
     "iopub.status.busy": "2021-04-22T08:53:07.604302Z",
     "iopub.status.idle": "2021-04-22T08:53:07.633142Z"
    }
   },
   "outputs": [],
   "source": [
    "base_dir = '/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/'\n",
    "clade = 'Oscillibacter'\n",
    "domain = 'Bacteria'\n",
    "taxid = 459786"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Init"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-22T08:42:46.407964Z",
     "iopub.status.busy": "2021-04-22T08:42:46.405994Z",
     "iopub.status.idle": "2021-04-22T08:42:46.432087Z"
    }
   },
   "outputs": [],
   "source": [
    "library(dplyr)\n",
    "library(tidyr)\n",
    "library(ggplot2)\n",
    "library(LeyLabRMisc)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-22T08:42:46.836315Z",
     "iopub.status.busy": "2021-04-22T08:42:46.834149Z",
     "iopub.status.idle": "2021-04-22T08:42:46.854010Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Directory already exists: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3//Oscillibacter \n"
     ]
    }
   ],
   "source": [
    "df.dims()\n",
    "work_dir = file.path(base_dir, clade)\n",
    "make_dir(work_dir)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Genome download\n",
    "\n",
    "* Downloading genomes from NCBI"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-12T13:07:32.874718Z",
     "iopub.status.busy": "2021-01-12T13:07:32.867938Z",
     "iopub.status.idle": "2021-01-12T13:07:32.985785Z",
     "shell.execute_reply": "2021-01-12T13:07:32.983225Z"
    }
   },
   "source": [
    "```\n",
    "OUTDIR=/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Oscillibacter/\n",
    "mkdir -p $OUTDIR\n",
    "ncbi-genome-download -p 12 -s genbank -F fasta --genera Oscillibacter -o $OUTDIR bacteria\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Genome quality\n",
    "\n",
    "* Filtering genomes by quality"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-22T08:45:37.477602Z",
     "iopub.status.busy": "2021-04-22T08:45:37.475567Z",
     "iopub.status.idle": "2021-04-22T08:45:37.567284Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 39 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Name</th><th scope=col>Fasta</th><th scope=col>Domain</th><th scope=col>Taxid</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>GCA_000283575.1_ASM28357v1_genomic</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Oscillibacter/genbank/bacteria/GCA_000283575.1/GCA_000283575.1_ASM28357v1_genomic.fna.gz</td><td>Bacteria</td><td>459786</td></tr>\n",
       "\t<tr><td>GCA_000307265.1_ASM30726v1_genomic</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Oscillibacter/genbank/bacteria/GCA_000307265.1/GCA_000307265.1_ASM30726v1_genomic.fna.gz</td><td>Bacteria</td><td>459786</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>GCA_015052085.1_ASM1505208v1_genomic                           </td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Oscillibacter/genbank/bacteria/GCA_015052085.1/GCA_015052085.1_ASM1505208v1_genomic.fna.gz                           </td><td>Bacteria</td><td>459786</td></tr>\n",
       "\t<tr><td>GCA_900115635.1_IMG-taxon_2623620509_annotated_assembly_genomic</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Oscillibacter/genbank/bacteria/GCA_900115635.1/GCA_900115635.1_IMG-taxon_2623620509_annotated_assembly_genomic.fna.gz</td><td>Bacteria</td><td>459786</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 39 × 4\n",
       "\\begin{tabular}{llll}\n",
       " Name & Fasta & Domain & Taxid\\\\\n",
       " <chr> & <chr> & <fct> & <dbl>\\\\\n",
       "\\hline\n",
       "\t GCA\\_000283575.1\\_ASM28357v1\\_genomic & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/v0.3/Oscillibacter/genbank/bacteria/GCA\\_000283575.1/GCA\\_000283575.1\\_ASM28357v1\\_genomic.fna.gz & Bacteria & 459786\\\\\n",
       "\t GCA\\_000307265.1\\_ASM30726v1\\_genomic & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/v0.3/Oscillibacter/genbank/bacteria/GCA\\_000307265.1/GCA\\_000307265.1\\_ASM30726v1\\_genomic.fna.gz & Bacteria & 459786\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t GCA\\_015052085.1\\_ASM1505208v1\\_genomic                            & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/v0.3/Oscillibacter/genbank/bacteria/GCA\\_015052085.1/GCA\\_015052085.1\\_ASM1505208v1\\_genomic.fna.gz                            & Bacteria & 459786\\\\\n",
       "\t GCA\\_900115635.1\\_IMG-taxon\\_2623620509\\_annotated\\_assembly\\_genomic & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/v0.3/Oscillibacter/genbank/bacteria/GCA\\_900115635.1/GCA\\_900115635.1\\_IMG-taxon\\_2623620509\\_annotated\\_assembly\\_genomic.fna.gz & Bacteria & 459786\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 39 × 4\n",
       "\n",
       "| Name &lt;chr&gt; | Fasta &lt;chr&gt; | Domain &lt;fct&gt; | Taxid &lt;dbl&gt; |\n",
       "|---|---|---|---|\n",
       "| GCA_000283575.1_ASM28357v1_genomic | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Oscillibacter/genbank/bacteria/GCA_000283575.1/GCA_000283575.1_ASM28357v1_genomic.fna.gz | Bacteria | 459786 |\n",
       "| GCA_000307265.1_ASM30726v1_genomic | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Oscillibacter/genbank/bacteria/GCA_000307265.1/GCA_000307265.1_ASM30726v1_genomic.fna.gz | Bacteria | 459786 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| GCA_015052085.1_ASM1505208v1_genomic                            | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Oscillibacter/genbank/bacteria/GCA_015052085.1/GCA_015052085.1_ASM1505208v1_genomic.fna.gz                            | Bacteria | 459786 |\n",
       "| GCA_900115635.1_IMG-taxon_2623620509_annotated_assembly_genomic | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Oscillibacter/genbank/bacteria/GCA_900115635.1/GCA_900115635.1_IMG-taxon_2623620509_annotated_assembly_genomic.fna.gz | Bacteria | 459786 |\n",
       "\n"
      ],
      "text/plain": [
       "   Name                                                           \n",
       "1  GCA_000283575.1_ASM28357v1_genomic                             \n",
       "2  GCA_000307265.1_ASM30726v1_genomic                             \n",
       "⋮  ⋮                                                              \n",
       "38 GCA_015052085.1_ASM1505208v1_genomic                           \n",
       "39 GCA_900115635.1_IMG-taxon_2623620509_annotated_assembly_genomic\n",
       "   Fasta                                                                                                                                                                                                        \n",
       "1  /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Oscillibacter/genbank/bacteria/GCA_000283575.1/GCA_000283575.1_ASM28357v1_genomic.fna.gz                             \n",
       "2  /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Oscillibacter/genbank/bacteria/GCA_000307265.1/GCA_000307265.1_ASM30726v1_genomic.fna.gz                             \n",
       "⋮  ⋮                                                                                                                                                                                                            \n",
       "38 /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Oscillibacter/genbank/bacteria/GCA_015052085.1/GCA_015052085.1_ASM1505208v1_genomic.fna.gz                           \n",
       "39 /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Oscillibacter/genbank/bacteria/GCA_900115635.1/GCA_900115635.1_IMG-taxon_2623620509_annotated_assembly_genomic.fna.gz\n",
       "   Domain   Taxid \n",
       "1  Bacteria 459786\n",
       "2  Bacteria 459786\n",
       "⋮  ⋮        ⋮     \n",
       "38 Bacteria 459786\n",
       "39 Bacteria 459786"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "File written: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3//Oscillibacter/genbank/samples.txt \n"
     ]
    }
   ],
   "source": [
    "D = file.path(base_dir, clade, 'genbank')\n",
    "files = list_files(D, '.fna.gz')\n",
    "samps = data.frame(Name = files %>% as.character %>% basename,\n",
    "                   Fasta = files,\n",
    "                   Domain = 'Bacteria',\n",
    "                   Taxid = taxid) %>%\n",
    "    mutate(Name = gsub('\\\\.fna\\\\.gz$', '', Name),\n",
    "           Fasta = gsub('/+', '/', Fasta))\n",
    "samps\n",
    "\n",
    "# writing file\n",
    "outfile = file.path(D, 'samples.txt')\n",
    "write_table(samps, outfile)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### LLG"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Config"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-22T08:45:51.380795Z",
     "iopub.status.busy": "2021-04-22T08:45:51.378646Z",
     "iopub.status.idle": "2021-04-22T08:45:51.409930Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# table with genome --> fasta_file information\n",
      "samples_file: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Oscillibacter/genbank/samples.txt\n",
      "\n",
      "# output location\n",
      "output_dir: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Oscillibacter/LLG/\n",
      "\n",
      "# temporary file directory (your username will be added automatically)\n",
      "tmp_dir: /ebio/abt3_scratch/\n",
      "\n",
      "# batch processing of genomes for certain steps\n",
      "## increase to better parallelize\n",
      "batches: 2 \n",
      "\n",
      "# Domain of genomes ('Archaea' or 'Bacteria)\n",
      "## Use \"Skip\" if provided as a \"Domain\" column in the genome table\n",
      "Domain: Skip\n",
      "\n",
      "# software parameters\n",
      "# Use \"Skip\" to skip any of these steps. If no params for rule, use \"\"\n",
      "# dRep MAGs are not further analyzed, but you can de-rep & then use the de-rep genome table as input.\n",
      "params:\n",
      "  ionice: -c 3\n",
      "  # assembly assessment\n",
      "  seqkit: \"\"\n",
      "  quast: Skip #\"\"\n",
      "  multiqc_on_quast: \"\" \n",
      "  checkm: \"\"\n",
      "  # de-replication (requires checkm)\n",
      "  drep: -comp 90 -con 5 -sa 0.999\n",
      "  # taxonomy\n",
      "  sourmash:\n",
      "    compute: Skip #--scaled 10000 -k 31\n",
      "    gather: -k 31\n",
      "  gtdbtk:\n",
      "    classify_wf: --min_perc_aa 10\n",
      "  # genome pairwise ANI  \n",
      "  fastani: Skip #--fragLen 3000 --minFraction 0.2 -k 16\n",
      "  dashing: Skip #-k 31 --full-tsv\n",
      "  comparem_aai: Skip #--evalue 0.001\n",
      "  # gene annotation\n",
      "  gene_call:\n",
      "    prokka: Skip #\"\" \n",
      "    multiqc_on_prokka: \"\"\n",
      "    prodigal: Skip #\"\"\n",
      "  eggnog_mapper: \"\"\n",
      "  eggnog_mapper_annot: \"\"\n",
      "  # rRNA (16S alignment & phylogeny)\n",
      "  barrnap: Skip #--lencutoff 0.8\n",
      "  vsearch_per_genome_drep: --id 0.95    # Skip to prevent drep of 16S copies within each genome\n",
      "  qiime2_fasttree: \"\"\n",
      "  qiime2_iqtree: --p-alrt 1000 --p-abayes --p-lbp 1000  --p-substitution-model 'GTR+I+G'\n",
      "  # genome phylogeny\n",
      "  phylophlan_config: Skip #--map_dna diamond --db_aa diamond --map_aa diamond --msa mafft --trim trimal --tree1 fasttree --tree2 raxml\n",
      "  phylophlan: --diversity high --min_num_markers 50 --accurate\n",
      "  # phenotype\n",
      "  traitar: Skip #\"\"\n",
      "  # biosynthetic gene clusters (BGCs)\n",
      "  antismash: Skip #--cb-knownclusters --cb-subclusters --asf  \n",
      "  DeepBGC: Skip #--score 0.5 --classifier-score 0.5 --prodigal-meta-mode\n",
      "  # antimicrobial resistance (AMR)\n",
      "  abricate: Skip #--minid 75 --mincov 80\n",
      "  # CRISPRs\n",
      "  cctyper: Skip #--prodigal meta\n",
      "  \n",
      "# databases\n",
      "databases:\n",
      "  checkM_data: /ebio/abt3_projects/databases_no-backup/checkM/\n",
      "  sourmash: /ebio/abt3_projects/databases_no-backup/sourmash/genbank-k31.sbt.json\n",
      "  sourmash_lca: /ebio/abt3_projects/databases_no-backup/sourmash/genbank-k31.lca.json.gz\n",
      "  gtdbtk: /ebio/abt3_projects/databases_no-backup/GTDB/release95/gtdbtk/db_info.md\n",
      "  phylophlan: /ebio/abt3_projects/databases_no-backup/phylophlan/PhyloPhlan/phylophlan.faa.bz2\n",
      "  eggnog: /ebio/abt3_projects/databases_no-backup/Eggnog/v2/eggnog.db\n",
      "  eggnog_diamond: /ebio/abt3_projects/databases_no-backup/Eggnog/v2/eggnog_proteins.dmnd\n",
      "  antismash: /ebio/abt3_projects/databases_no-backup/antismash/v5/\n",
      "  deepbgc: /ebio/abt3_projects/databases_no-backup/DeepBGC/\n",
      "  traitar: /ebio/abt3_projects/databases_no-backup/pfam/traitar/\n",
      "  abricate:\n",
      "    ncbi: /ebio/abt3_projects/databases_no-backup/abricate/ncbi/sequences\n",
      "    card: /ebio/abt3_projects/databases_no-backup/abricate/card/sequences\n",
      "    resfinder: /ebio/abt3_projects/databases_no-backup/abricate/resfinder/sequences\n",
      "    argannot: /ebio/abt3_projects/databases_no-backup/abricate/argannot/sequences\n",
      "    bacmet2: /ebio/abt3_projects/databases_no-backup/abricate/bacmet2/sequences\n",
      "    vfdb: /ebio/abt3_projects/databases_no-backup/abricate/vfdb/sequences\n",
      "    megares: /ebio/abt3_projects/databases_no-backup/abricate/megares/sequences\n",
      "    plasmidfinder: /ebio/abt3_projects/databases_no-backup/abricate/plasmidfinder/sequences\n",
      "\n",
      "# snakemake pipeline\n",
      "pipeline:\n",
      "  snakemake_folder: ./\n",
      "  script_folder: ./bin/scripts/\n",
      "  use_shared_mem: True\n",
      "  name: LLG"
     ]
    }
   ],
   "source": [
    "cat_file(file.path(work_dir, 'LLG', 'config.yaml'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-12T13:14:50.416604Z",
     "iopub.status.busy": "2021-01-12T13:14:50.414179Z",
     "iopub.status.idle": "2021-01-12T13:14:50.431925Z",
     "shell.execute_reply": "2021-01-12T13:14:50.430260Z"
    }
   },
   "source": [
    "#### Run\n",
    "\n",
    "```\n",
    "(snakemake) @ rick:/ebio/abt3_projects/software/dev/ll_pipelines/llg\n",
    "$ screen -L -S llg-osc ./snakemake_sge.sh /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Oscillibacter/LLG/llg.yaml 20 -F\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-12T13:14:10.995042Z",
     "iopub.status.busy": "2021-01-12T13:14:10.992773Z",
     "iopub.status.idle": "2021-01-12T13:14:11.009275Z",
     "shell.execute_reply": "2021-01-12T13:14:11.008255Z"
    }
   },
   "source": [
    "### Samples table of high-quality genomes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-22T08:46:15.557780Z",
     "iopub.status.busy": "2021-04-22T08:46:15.555373Z",
     "iopub.status.idle": "2021-04-22T08:46:15.612622Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 39 × 33</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Bin.Id</th><th scope=col>Marker.lineage</th><th scope=col>X..genomes</th><th scope=col>X..markers</th><th scope=col>X..marker.sets</th><th scope=col>Completeness</th><th scope=col>Contamination</th><th scope=col>Strain.heterogeneity</th><th scope=col>Genome.size..bp.</th><th scope=col>X..ambiguous.bases</th><th scope=col>⋯</th><th scope=col>X0</th><th scope=col>X1</th><th scope=col>X2</th><th scope=col>X3</th><th scope=col>X4</th><th scope=col>X5.</th><th scope=col>assembly.Id</th><th scope=col>assembler.Id</th><th scope=col>taxon.Id</th><th scope=col>File</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>GCA_000283575.1_ASM28357v1_genomic      </td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>98.99</td><td>1.01</td><td>0</td><td>4470622</td><td>    0</td><td>⋯</td><td>2</td><td>259</td><td>2</td><td>0</td><td>0</td><td>0</td><td>|ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Oscillibacter|LLG_output|checkM|1|checkm|markers_qa_summary.tsv.1</td><td>markers_qa_summary.tsv.1</td><td>NA</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Oscillibacter/LLG_output/checkM/1/checkm/markers_qa_summary.tsv.1</td></tr>\n",
       "\t<tr><td>GCA_000403435.2_Osci_bact_1-3_V1_genomic</td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>99.33</td><td>1.45</td><td>0</td><td>4467686</td><td>37481</td><td>⋯</td><td>1</td><td>259</td><td>3</td><td>0</td><td>0</td><td>0</td><td>|ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Oscillibacter|LLG_output|checkM|1|checkm|markers_qa_summary.tsv.2</td><td>markers_qa_summary.tsv.2</td><td>NA</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Oscillibacter/LLG_output/checkM/1/checkm/markers_qa_summary.tsv.2</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋱</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>GCA_014799915.1_ASM1479991v1_genomic</td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>87.74</td><td> 2.35</td><td>50.00</td><td>2875254</td><td>19077</td><td>⋯</td><td>50</td><td>209</td><td> 4</td><td>0</td><td>0</td><td>0</td><td>|ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Oscillibacter|LLG_output|checkM|2|checkm|markers_qa_summary.tsv.18</td><td>markers_qa_summary.tsv.18</td><td>NA</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Oscillibacter/LLG_output/checkM/2/checkm/markers_qa_summary.tsv.18</td></tr>\n",
       "\t<tr><td>GCA_015052085.1_ASM1505208v1_genomic</td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>89.56</td><td>11.63</td><td>82.61</td><td>2465477</td><td>    0</td><td>⋯</td><td>45</td><td>195</td><td>23</td><td>0</td><td>0</td><td>0</td><td>|ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Oscillibacter|LLG_output|checkM|2|checkm|markers_qa_summary.tsv.19</td><td>markers_qa_summary.tsv.19</td><td>NA</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Oscillibacter/LLG_output/checkM/2/checkm/markers_qa_summary.tsv.19</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 39 × 33\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " Bin.Id & Marker.lineage & X..genomes & X..markers & X..marker.sets & Completeness & Contamination & Strain.heterogeneity & Genome.size..bp. & X..ambiguous.bases & ⋯ & X0 & X1 & X2 & X3 & X4 & X5. & assembly.Id & assembler.Id & taxon.Id & File\\\\\n",
       " <fct> & <fct> & <int> & <int> & <int> & <dbl> & <dbl> & <dbl> & <int> & <int> & ⋯ & <int> & <int> & <int> & <int> & <int> & <int> & <fct> & <fct> & <lgl> & <fct>\\\\\n",
       "\\hline\n",
       "\t GCA\\_000283575.1\\_ASM28357v1\\_genomic       & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 98.99 & 1.01 & 0 & 4470622 &     0 & ⋯ & 2 & 259 & 2 & 0 & 0 & 0 & \\textbar{}ebio\\textbar{}abt3\\_projects\\textbar{}software\\textbar{}dev\\textbar{}ll\\_pipelines\\textbar{}llprimer\\textbar{}experiments\\textbar{}HMP\\_most-wanted\\textbar{}Oscillibacter\\textbar{}LLG\\_output\\textbar{}checkM\\textbar{}1\\textbar{}checkm\\textbar{}markers\\_qa\\_summary.tsv.1 & markers\\_qa\\_summary.tsv.1 & NA & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/Oscillibacter/LLG\\_output/checkM/1/checkm/markers\\_qa\\_summary.tsv.1\\\\\n",
       "\t GCA\\_000403435.2\\_Osci\\_bact\\_1-3\\_V1\\_genomic & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 99.33 & 1.45 & 0 & 4467686 & 37481 & ⋯ & 1 & 259 & 3 & 0 & 0 & 0 & \\textbar{}ebio\\textbar{}abt3\\_projects\\textbar{}software\\textbar{}dev\\textbar{}ll\\_pipelines\\textbar{}llprimer\\textbar{}experiments\\textbar{}HMP\\_most-wanted\\textbar{}Oscillibacter\\textbar{}LLG\\_output\\textbar{}checkM\\textbar{}1\\textbar{}checkm\\textbar{}markers\\_qa\\_summary.tsv.2 & markers\\_qa\\_summary.tsv.2 & NA & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/Oscillibacter/LLG\\_output/checkM/1/checkm/markers\\_qa\\_summary.tsv.2\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋱ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t GCA\\_014799915.1\\_ASM1479991v1\\_genomic & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 87.74 &  2.35 & 50.00 & 2875254 & 19077 & ⋯ & 50 & 209 &  4 & 0 & 0 & 0 & \\textbar{}ebio\\textbar{}abt3\\_projects\\textbar{}software\\textbar{}dev\\textbar{}ll\\_pipelines\\textbar{}llprimer\\textbar{}experiments\\textbar{}HMP\\_most-wanted\\textbar{}Oscillibacter\\textbar{}LLG\\_output\\textbar{}checkM\\textbar{}2\\textbar{}checkm\\textbar{}markers\\_qa\\_summary.tsv.18 & markers\\_qa\\_summary.tsv.18 & NA & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/Oscillibacter/LLG\\_output/checkM/2/checkm/markers\\_qa\\_summary.tsv.18\\\\\n",
       "\t GCA\\_015052085.1\\_ASM1505208v1\\_genomic & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 89.56 & 11.63 & 82.61 & 2465477 &     0 & ⋯ & 45 & 195 & 23 & 0 & 0 & 0 & \\textbar{}ebio\\textbar{}abt3\\_projects\\textbar{}software\\textbar{}dev\\textbar{}ll\\_pipelines\\textbar{}llprimer\\textbar{}experiments\\textbar{}HMP\\_most-wanted\\textbar{}Oscillibacter\\textbar{}LLG\\_output\\textbar{}checkM\\textbar{}2\\textbar{}checkm\\textbar{}markers\\_qa\\_summary.tsv.19 & markers\\_qa\\_summary.tsv.19 & NA & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/Oscillibacter/LLG\\_output/checkM/2/checkm/markers\\_qa\\_summary.tsv.19\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 39 × 33\n",
       "\n",
       "| Bin.Id &lt;fct&gt; | Marker.lineage &lt;fct&gt; | X..genomes &lt;int&gt; | X..markers &lt;int&gt; | X..marker.sets &lt;int&gt; | Completeness &lt;dbl&gt; | Contamination &lt;dbl&gt; | Strain.heterogeneity &lt;dbl&gt; | Genome.size..bp. &lt;int&gt; | X..ambiguous.bases &lt;int&gt; | ⋯ ⋯ | X0 &lt;int&gt; | X1 &lt;int&gt; | X2 &lt;int&gt; | X3 &lt;int&gt; | X4 &lt;int&gt; | X5. &lt;int&gt; | assembly.Id &lt;fct&gt; | assembler.Id &lt;fct&gt; | taxon.Id &lt;lgl&gt; | File &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| GCA_000283575.1_ASM28357v1_genomic       | o__Clostridiales (UID1212) | 172 | 263 | 149 | 98.99 | 1.01 | 0 | 4470622 |     0 | ⋯ | 2 | 259 | 2 | 0 | 0 | 0 | |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Oscillibacter|LLG_output|checkM|1|checkm|markers_qa_summary.tsv.1 | markers_qa_summary.tsv.1 | NA | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Oscillibacter/LLG_output/checkM/1/checkm/markers_qa_summary.tsv.1 |\n",
       "| GCA_000403435.2_Osci_bact_1-3_V1_genomic | o__Clostridiales (UID1212) | 172 | 263 | 149 | 99.33 | 1.45 | 0 | 4467686 | 37481 | ⋯ | 1 | 259 | 3 | 0 | 0 | 0 | |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Oscillibacter|LLG_output|checkM|1|checkm|markers_qa_summary.tsv.2 | markers_qa_summary.tsv.2 | NA | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Oscillibacter/LLG_output/checkM/1/checkm/markers_qa_summary.tsv.2 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋱ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| GCA_014799915.1_ASM1479991v1_genomic | o__Clostridiales (UID1212) | 172 | 263 | 149 | 87.74 |  2.35 | 50.00 | 2875254 | 19077 | ⋯ | 50 | 209 |  4 | 0 | 0 | 0 | |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Oscillibacter|LLG_output|checkM|2|checkm|markers_qa_summary.tsv.18 | markers_qa_summary.tsv.18 | NA | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Oscillibacter/LLG_output/checkM/2/checkm/markers_qa_summary.tsv.18 |\n",
       "| GCA_015052085.1_ASM1505208v1_genomic | o__Clostridiales (UID1212) | 172 | 263 | 149 | 89.56 | 11.63 | 82.61 | 2465477 |     0 | ⋯ | 45 | 195 | 23 | 0 | 0 | 0 | |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Oscillibacter|LLG_output|checkM|2|checkm|markers_qa_summary.tsv.19 | markers_qa_summary.tsv.19 | NA | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Oscillibacter/LLG_output/checkM/2/checkm/markers_qa_summary.tsv.19 |\n",
       "\n"
      ],
      "text/plain": [
       "   Bin.Id                                   Marker.lineage            \n",
       "1  GCA_000283575.1_ASM28357v1_genomic       o__Clostridiales (UID1212)\n",
       "2  GCA_000403435.2_Osci_bact_1-3_V1_genomic o__Clostridiales (UID1212)\n",
       "⋮  ⋮                                        ⋮                         \n",
       "38 GCA_014799915.1_ASM1479991v1_genomic     o__Clostridiales (UID1212)\n",
       "39 GCA_015052085.1_ASM1505208v1_genomic     o__Clostridiales (UID1212)\n",
       "   X..genomes X..markers X..marker.sets Completeness Contamination\n",
       "1  172        263        149            98.99        1.01         \n",
       "2  172        263        149            99.33        1.45         \n",
       "⋮  ⋮          ⋮          ⋮              ⋮            ⋮            \n",
       "38 172        263        149            87.74         2.35        \n",
       "39 172        263        149            89.56        11.63        \n",
       "   Strain.heterogeneity Genome.size..bp. X..ambiguous.bases ⋯ X0 X1  X2 X3 X4\n",
       "1  0                    4470622              0              ⋯ 2  259 2  0  0 \n",
       "2  0                    4467686          37481              ⋯ 1  259 3  0  0 \n",
       "⋮  ⋮                    ⋮                ⋮                  ⋱ ⋮  ⋮   ⋮  ⋮  ⋮ \n",
       "38 50.00                2875254          19077              ⋯ 50 209  4 0  0 \n",
       "39 82.61                2465477              0              ⋯ 45 195 23 0  0 \n",
       "   X5.\n",
       "1  0  \n",
       "2  0  \n",
       "⋮  ⋮  \n",
       "38 0  \n",
       "39 0  \n",
       "   assembly.Id                                                                                                                                          \n",
       "1  |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Oscillibacter|LLG_output|checkM|1|checkm|markers_qa_summary.tsv.1 \n",
       "2  |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Oscillibacter|LLG_output|checkM|1|checkm|markers_qa_summary.tsv.2 \n",
       "⋮  ⋮                                                                                                                                                    \n",
       "38 |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Oscillibacter|LLG_output|checkM|2|checkm|markers_qa_summary.tsv.18\n",
       "39 |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Oscillibacter|LLG_output|checkM|2|checkm|markers_qa_summary.tsv.19\n",
       "   assembler.Id              taxon.Id\n",
       "1  markers_qa_summary.tsv.1  NA      \n",
       "2  markers_qa_summary.tsv.2  NA      \n",
       "⋮  ⋮                         ⋮       \n",
       "38 markers_qa_summary.tsv.18 NA      \n",
       "39 markers_qa_summary.tsv.19 NA      \n",
       "   File                                                                                                                                                 \n",
       "1  /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Oscillibacter/LLG_output/checkM/1/checkm/markers_qa_summary.tsv.1 \n",
       "2  /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Oscillibacter/LLG_output/checkM/1/checkm/markers_qa_summary.tsv.2 \n",
       "⋮  ⋮                                                                                                                                                    \n",
       "38 /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Oscillibacter/LLG_output/checkM/2/checkm/markers_qa_summary.tsv.18\n",
       "39 /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Oscillibacter/LLG_output/checkM/2/checkm/markers_qa_summary.tsv.19"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# checkM summary\n",
    "checkm = file.path(work_dir, 'LLG', 'checkM', 'checkm_qa_summary.tsv') %>%\n",
    "    read.delim(sep='\\t') \n",
    "checkm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-22T08:46:20.210758Z",
     "iopub.status.busy": "2021-04-22T08:46:20.208701Z",
     "iopub.status.idle": "2021-04-22T08:46:20.242119Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 39 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>genome</th><th scope=col>completeness</th><th scope=col>contamination</th><th scope=col>Bin.Id</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000283575.1_ASM28357v1_genomic.fna      </td><td>98.99</td><td>1.01</td><td>GCA_000283575.1_ASM28357v1_genomic      </td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000403435.2_Osci_bact_1-3_V1_genomic.fna</td><td>99.33</td><td>1.45</td><td>GCA_000403435.2_Osci_bact_1-3_V1_genomic</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_014799915.1_ASM1479991v1_genomic.fna</td><td>87.74</td><td> 2.35</td><td>GCA_014799915.1_ASM1479991v1_genomic</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_015052085.1_ASM1505208v1_genomic.fna</td><td>89.56</td><td>11.63</td><td>GCA_015052085.1_ASM1505208v1_genomic</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 39 × 4\n",
       "\\begin{tabular}{llll}\n",
       " genome & completeness & contamination & Bin.Id\\\\\n",
       " <fct> & <dbl> & <dbl> & <chr>\\\\\n",
       "\\hline\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_000283575.1\\_ASM28357v1\\_genomic.fna       & 98.99 & 1.01 & GCA\\_000283575.1\\_ASM28357v1\\_genomic      \\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_000403435.2\\_Osci\\_bact\\_1-3\\_V1\\_genomic.fna & 99.33 & 1.45 & GCA\\_000403435.2\\_Osci\\_bact\\_1-3\\_V1\\_genomic\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_014799915.1\\_ASM1479991v1\\_genomic.fna & 87.74 &  2.35 & GCA\\_014799915.1\\_ASM1479991v1\\_genomic\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_015052085.1\\_ASM1505208v1\\_genomic.fna & 89.56 & 11.63 & GCA\\_015052085.1\\_ASM1505208v1\\_genomic\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 39 × 4\n",
       "\n",
       "| genome &lt;fct&gt; | completeness &lt;dbl&gt; | contamination &lt;dbl&gt; | Bin.Id &lt;chr&gt; |\n",
       "|---|---|---|---|\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000283575.1_ASM28357v1_genomic.fna       | 98.99 | 1.01 | GCA_000283575.1_ASM28357v1_genomic       |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000403435.2_Osci_bact_1-3_V1_genomic.fna | 99.33 | 1.45 | GCA_000403435.2_Osci_bact_1-3_V1_genomic |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_014799915.1_ASM1479991v1_genomic.fna | 87.74 |  2.35 | GCA_014799915.1_ASM1479991v1_genomic |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_015052085.1_ASM1505208v1_genomic.fna | 89.56 | 11.63 | GCA_015052085.1_ASM1505208v1_genomic |\n",
       "\n"
      ],
      "text/plain": [
       "   genome                                                                                            \n",
       "1  /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000283575.1_ASM28357v1_genomic.fna      \n",
       "2  /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000403435.2_Osci_bact_1-3_V1_genomic.fna\n",
       "⋮  ⋮                                                                                                 \n",
       "38 /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_014799915.1_ASM1479991v1_genomic.fna    \n",
       "39 /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_015052085.1_ASM1505208v1_genomic.fna    \n",
       "   completeness contamination Bin.Id                                  \n",
       "1  98.99        1.01          GCA_000283575.1_ASM28357v1_genomic      \n",
       "2  99.33        1.45          GCA_000403435.2_Osci_bact_1-3_V1_genomic\n",
       "⋮  ⋮            ⋮             ⋮                                       \n",
       "38 87.74         2.35         GCA_014799915.1_ASM1479991v1_genomic    \n",
       "39 89.56        11.63         GCA_015052085.1_ASM1505208v1_genomic    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# dRep summary\n",
    "drep = file.path(work_dir, 'LLG', 'drep', 'checkm_markers_qa_summary.tsv') %>%\n",
    "    read.delim(sep='\\t') %>%\n",
    "    mutate(Bin.Id = gsub('.+/', '', genome),\n",
    "           Bin.Id = gsub('\\\\.fna$', '', Bin.Id))\n",
    "drep"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-22T08:46:23.605801Z",
     "iopub.status.busy": "2021-04-22T08:46:23.602057Z",
     "iopub.status.idle": "2021-04-22T08:46:23.633355Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 16 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Name</th><th scope=col>Fasta</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>GCA_000765235.1_ASM76523v1_genomic </td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Oscillibacter/LLG_output/drep/drep/dereplicated_genomes/GCA_000765235.1_ASM76523v1_genomic.fna </td></tr>\n",
       "\t<tr><td>GCA_003525445.1_ASM352544v1_genomic</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Oscillibacter/LLG_output/drep/drep/dereplicated_genomes/GCA_003525445.1_ASM352544v1_genomic.fna</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>GCA_001916835.1_ASM191683v1_genomic </td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Oscillibacter/LLG_output/drep/drep/dereplicated_genomes/GCA_001916835.1_ASM191683v1_genomic.fna </td></tr>\n",
       "\t<tr><td>GCA_014799925.1_ASM1479992v1_genomic</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Oscillibacter/LLG_output/drep/drep/dereplicated_genomes/GCA_014799925.1_ASM1479992v1_genomic.fna</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 16 × 2\n",
       "\\begin{tabular}{ll}\n",
       " Name & Fasta\\\\\n",
       " <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\t GCA\\_000765235.1\\_ASM76523v1\\_genomic  & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/Oscillibacter/LLG\\_output/drep/drep/dereplicated\\_genomes/GCA\\_000765235.1\\_ASM76523v1\\_genomic.fna \\\\\n",
       "\t GCA\\_003525445.1\\_ASM352544v1\\_genomic & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/Oscillibacter/LLG\\_output/drep/drep/dereplicated\\_genomes/GCA\\_003525445.1\\_ASM352544v1\\_genomic.fna\\\\\n",
       "\t ⋮ & ⋮\\\\\n",
       "\t GCA\\_001916835.1\\_ASM191683v1\\_genomic  & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/Oscillibacter/LLG\\_output/drep/drep/dereplicated\\_genomes/GCA\\_001916835.1\\_ASM191683v1\\_genomic.fna \\\\\n",
       "\t GCA\\_014799925.1\\_ASM1479992v1\\_genomic & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/Oscillibacter/LLG\\_output/drep/drep/dereplicated\\_genomes/GCA\\_014799925.1\\_ASM1479992v1\\_genomic.fna\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 16 × 2\n",
       "\n",
       "| Name &lt;fct&gt; | Fasta &lt;fct&gt; |\n",
       "|---|---|\n",
       "| GCA_000765235.1_ASM76523v1_genomic  | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Oscillibacter/LLG_output/drep/drep/dereplicated_genomes/GCA_000765235.1_ASM76523v1_genomic.fna  |\n",
       "| GCA_003525445.1_ASM352544v1_genomic | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Oscillibacter/LLG_output/drep/drep/dereplicated_genomes/GCA_003525445.1_ASM352544v1_genomic.fna |\n",
       "| ⋮ | ⋮ |\n",
       "| GCA_001916835.1_ASM191683v1_genomic  | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Oscillibacter/LLG_output/drep/drep/dereplicated_genomes/GCA_001916835.1_ASM191683v1_genomic.fna  |\n",
       "| GCA_014799925.1_ASM1479992v1_genomic | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Oscillibacter/LLG_output/drep/drep/dereplicated_genomes/GCA_014799925.1_ASM1479992v1_genomic.fna |\n",
       "\n"
      ],
      "text/plain": [
       "   Name                                \n",
       "1  GCA_000765235.1_ASM76523v1_genomic  \n",
       "2  GCA_003525445.1_ASM352544v1_genomic \n",
       "⋮  ⋮                                   \n",
       "15 GCA_001916835.1_ASM191683v1_genomic \n",
       "16 GCA_014799925.1_ASM1479992v1_genomic\n",
       "   Fasta                                                                                                                                                                              \n",
       "1  /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Oscillibacter/LLG_output/drep/drep/dereplicated_genomes/GCA_000765235.1_ASM76523v1_genomic.fna  \n",
       "2  /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Oscillibacter/LLG_output/drep/drep/dereplicated_genomes/GCA_003525445.1_ASM352544v1_genomic.fna \n",
       "⋮  ⋮                                                                                                                                                                                  \n",
       "15 /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Oscillibacter/LLG_output/drep/drep/dereplicated_genomes/GCA_001916835.1_ASM191683v1_genomic.fna \n",
       "16 /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Oscillibacter/LLG_output/drep/drep/dereplicated_genomes/GCA_014799925.1_ASM1479992v1_genomic.fna"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# de-replicated genomes\n",
    "drep_gen = file.path(work_dir, 'LLG', 'drep', 'dereplicated_genomes.tsv') %>%\n",
    "    read.delim(sep='\\t')\n",
    "drep_gen"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-22T08:46:27.657831Z",
     "iopub.status.busy": "2021-04-22T08:46:27.655621Z",
     "iopub.status.idle": "2021-04-22T08:46:27.933782Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 39 × 22</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>user_genome</th><th scope=col>Domain</th><th scope=col>Phylum</th><th scope=col>Class</th><th scope=col>Order</th><th scope=col>Family</th><th scope=col>Genus</th><th scope=col>Species</th><th scope=col>fastani_reference</th><th scope=col>fastani_reference_radius</th><th scope=col>⋯</th><th scope=col>fastani_af</th><th scope=col>closest_placement_reference</th><th scope=col>closest_placement_radius</th><th scope=col>closest_placement_taxonomy</th><th scope=col>closest_placement_ani</th><th scope=col>closest_placement_af</th><th scope=col>msa_percent</th><th scope=col>translation_table</th><th scope=col>red_value</th><th scope=col>warnings</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>⋯</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>GCA_000283575.1_ASM28357v1_genomic      </td><td>d__Bacteria</td><td>p__Firmicutes_A</td><td>c__Clostridia</td><td>o__Oscillospirales</td><td>f__Oscillospiraceae</td><td>g__Oscillibacter</td><td>s__Oscillibacter valericigenes</td><td>GCF_000283575.1</td><td>95.0</td><td>⋯</td><td>1.0 </td><td>GCF_000283575.1</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Oscillospiraceae;g__Oscillibacter;s__Oscillibacter valericigenes</td><td>100.0</td><td>1.0 </td><td>95.38</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>GCA_000403435.2_Osci_bact_1-3_V1_genomic</td><td>d__Bacteria</td><td>p__Firmicutes_A</td><td>c__Clostridia</td><td>o__Oscillospirales</td><td>f__Oscillospiraceae</td><td>g__Oscillibacter</td><td>s__Oscillibacter sp000403435  </td><td>GCF_000403435.2</td><td>95.0</td><td>⋯</td><td>0.99</td><td>GCF_000403435.2</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Oscillospiraceae;g__Oscillibacter;s__Oscillibacter sp000403435  </td><td>99.99</td><td>0.99</td><td>95.67</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋱</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>GCA_014799915.1_ASM1479991v1_genomic</td><td>d__Bacteria</td><td>p__Firmicutes_A</td><td>c__Clostridia</td><td>o__Oscillospirales</td><td>f__Oscillospiraceae</td><td>g__             </td><td>s__                       </td><td>N/A            </td><td>N/A </td><td>⋯</td><td>N/A </td><td>N/A            </td><td>N/A </td><td>N/A                                                                                                                         </td><td>N/A  </td><td>N/A </td><td>79.35</td><td>11</td><td>0.9173648727242681</td><td>N/A</td></tr>\n",
       "\t<tr><td>GCA_015052085.1_ASM1505208v1_genomic</td><td>d__Bacteria</td><td>p__Firmicutes_A</td><td>c__Clostridia</td><td>o__Oscillospirales</td><td>f__Oscillospiraceae</td><td>g__Oscillibacter</td><td>s__Oscillibacter welbionis</td><td>GCF_005121165.1</td><td>95.0</td><td>⋯</td><td>0.88</td><td>GCF_005121165.1</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Oscillospiraceae;g__Oscillibacter;s__Oscillibacter welbionis</td><td>98.68</td><td>0.88</td><td>76.73</td><td>11</td><td>N/A               </td><td>N/A</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 39 × 22\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " user\\_genome & Domain & Phylum & Class & Order & Family & Genus & Species & fastani\\_reference & fastani\\_reference\\_radius & ⋯ & fastani\\_af & closest\\_placement\\_reference & closest\\_placement\\_radius & closest\\_placement\\_taxonomy & closest\\_placement\\_ani & closest\\_placement\\_af & msa\\_percent & translation\\_table & red\\_value & warnings\\\\\n",
       " <fct> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <fct> & <fct> & ⋯ & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <dbl> & <int> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\t GCA\\_000283575.1\\_ASM28357v1\\_genomic       & d\\_\\_Bacteria & p\\_\\_Firmicutes\\_A & c\\_\\_Clostridia & o\\_\\_Oscillospirales & f\\_\\_Oscillospiraceae & g\\_\\_Oscillibacter & s\\_\\_Oscillibacter valericigenes & GCF\\_000283575.1 & 95.0 & ⋯ & 1.0  & GCF\\_000283575.1 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Oscillospiraceae;g\\_\\_Oscillibacter;s\\_\\_Oscillibacter valericigenes & 100.0 & 1.0  & 95.38 & 11 & N/A & N/A\\\\\n",
       "\t GCA\\_000403435.2\\_Osci\\_bact\\_1-3\\_V1\\_genomic & d\\_\\_Bacteria & p\\_\\_Firmicutes\\_A & c\\_\\_Clostridia & o\\_\\_Oscillospirales & f\\_\\_Oscillospiraceae & g\\_\\_Oscillibacter & s\\_\\_Oscillibacter sp000403435   & GCF\\_000403435.2 & 95.0 & ⋯ & 0.99 & GCF\\_000403435.2 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Oscillospiraceae;g\\_\\_Oscillibacter;s\\_\\_Oscillibacter sp000403435   & 99.99 & 0.99 & 95.67 & 11 & N/A & N/A\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋱ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t GCA\\_014799915.1\\_ASM1479991v1\\_genomic & d\\_\\_Bacteria & p\\_\\_Firmicutes\\_A & c\\_\\_Clostridia & o\\_\\_Oscillospirales & f\\_\\_Oscillospiraceae & g\\_\\_              & s\\_\\_                        & N/A             & N/A  & ⋯ & N/A  & N/A             & N/A  & N/A                                                                                                                          & N/A   & N/A  & 79.35 & 11 & 0.9173648727242681 & N/A\\\\\n",
       "\t GCA\\_015052085.1\\_ASM1505208v1\\_genomic & d\\_\\_Bacteria & p\\_\\_Firmicutes\\_A & c\\_\\_Clostridia & o\\_\\_Oscillospirales & f\\_\\_Oscillospiraceae & g\\_\\_Oscillibacter & s\\_\\_Oscillibacter welbionis & GCF\\_005121165.1 & 95.0 & ⋯ & 0.88 & GCF\\_005121165.1 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Oscillospiraceae;g\\_\\_Oscillibacter;s\\_\\_Oscillibacter welbionis & 98.68 & 0.88 & 76.73 & 11 & N/A                & N/A\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 39 × 22\n",
       "\n",
       "| user_genome &lt;fct&gt; | Domain &lt;chr&gt; | Phylum &lt;chr&gt; | Class &lt;chr&gt; | Order &lt;chr&gt; | Family &lt;chr&gt; | Genus &lt;chr&gt; | Species &lt;chr&gt; | fastani_reference &lt;fct&gt; | fastani_reference_radius &lt;fct&gt; | ⋯ ⋯ | fastani_af &lt;fct&gt; | closest_placement_reference &lt;fct&gt; | closest_placement_radius &lt;fct&gt; | closest_placement_taxonomy &lt;fct&gt; | closest_placement_ani &lt;fct&gt; | closest_placement_af &lt;fct&gt; | msa_percent &lt;dbl&gt; | translation_table &lt;int&gt; | red_value &lt;fct&gt; | warnings &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| GCA_000283575.1_ASM28357v1_genomic       | d__Bacteria | p__Firmicutes_A | c__Clostridia | o__Oscillospirales | f__Oscillospiraceae | g__Oscillibacter | s__Oscillibacter valericigenes | GCF_000283575.1 | 95.0 | ⋯ | 1.0  | GCF_000283575.1 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Oscillospiraceae;g__Oscillibacter;s__Oscillibacter valericigenes | 100.0 | 1.0  | 95.38 | 11 | N/A | N/A |\n",
       "| GCA_000403435.2_Osci_bact_1-3_V1_genomic | d__Bacteria | p__Firmicutes_A | c__Clostridia | o__Oscillospirales | f__Oscillospiraceae | g__Oscillibacter | s__Oscillibacter sp000403435   | GCF_000403435.2 | 95.0 | ⋯ | 0.99 | GCF_000403435.2 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Oscillospiraceae;g__Oscillibacter;s__Oscillibacter sp000403435   | 99.99 | 0.99 | 95.67 | 11 | N/A | N/A |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋱ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| GCA_014799915.1_ASM1479991v1_genomic | d__Bacteria | p__Firmicutes_A | c__Clostridia | o__Oscillospirales | f__Oscillospiraceae | g__              | s__                        | N/A             | N/A  | ⋯ | N/A  | N/A             | N/A  | N/A                                                                                                                          | N/A   | N/A  | 79.35 | 11 | 0.9173648727242681 | N/A |\n",
       "| GCA_015052085.1_ASM1505208v1_genomic | d__Bacteria | p__Firmicutes_A | c__Clostridia | o__Oscillospirales | f__Oscillospiraceae | g__Oscillibacter | s__Oscillibacter welbionis | GCF_005121165.1 | 95.0 | ⋯ | 0.88 | GCF_005121165.1 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Oscillospiraceae;g__Oscillibacter;s__Oscillibacter welbionis | 98.68 | 0.88 | 76.73 | 11 | N/A                | N/A |\n",
       "\n"
      ],
      "text/plain": [
       "   user_genome                              Domain      Phylum         \n",
       "1  GCA_000283575.1_ASM28357v1_genomic       d__Bacteria p__Firmicutes_A\n",
       "2  GCA_000403435.2_Osci_bact_1-3_V1_genomic d__Bacteria p__Firmicutes_A\n",
       "⋮  ⋮                                        ⋮           ⋮              \n",
       "38 GCA_014799915.1_ASM1479991v1_genomic     d__Bacteria p__Firmicutes_A\n",
       "39 GCA_015052085.1_ASM1505208v1_genomic     d__Bacteria p__Firmicutes_A\n",
       "   Class         Order              Family              Genus           \n",
       "1  c__Clostridia o__Oscillospirales f__Oscillospiraceae g__Oscillibacter\n",
       "2  c__Clostridia o__Oscillospirales f__Oscillospiraceae g__Oscillibacter\n",
       "⋮  ⋮             ⋮                  ⋮                   ⋮               \n",
       "38 c__Clostridia o__Oscillospirales f__Oscillospiraceae g__             \n",
       "39 c__Clostridia o__Oscillospirales f__Oscillospiraceae g__Oscillibacter\n",
       "   Species                        fastani_reference fastani_reference_radius ⋯\n",
       "1  s__Oscillibacter valericigenes GCF_000283575.1   95.0                     ⋯\n",
       "2  s__Oscillibacter sp000403435   GCF_000403435.2   95.0                     ⋯\n",
       "⋮  ⋮                              ⋮                 ⋮                        ⋱\n",
       "38 s__                            N/A               N/A                      ⋯\n",
       "39 s__Oscillibacter welbionis     GCF_005121165.1   95.0                     ⋯\n",
       "   fastani_af closest_placement_reference closest_placement_radius\n",
       "1  1.0        GCF_000283575.1             95.0                    \n",
       "2  0.99       GCF_000403435.2             95.0                    \n",
       "⋮  ⋮          ⋮                           ⋮                       \n",
       "38 N/A        N/A                         N/A                     \n",
       "39 0.88       GCF_005121165.1             95.0                    \n",
       "   closest_placement_taxonomy                                                                                                      \n",
       "1  d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Oscillospiraceae;g__Oscillibacter;s__Oscillibacter valericigenes\n",
       "2  d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Oscillospiraceae;g__Oscillibacter;s__Oscillibacter sp000403435  \n",
       "⋮  ⋮                                                                                                                               \n",
       "38 N/A                                                                                                                             \n",
       "39 d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Oscillospiraceae;g__Oscillibacter;s__Oscillibacter welbionis    \n",
       "   closest_placement_ani closest_placement_af msa_percent translation_table\n",
       "1  100.0                 1.0                  95.38       11               \n",
       "2  99.99                 0.99                 95.67       11               \n",
       "⋮  ⋮                     ⋮                    ⋮           ⋮                \n",
       "38 N/A                   N/A                  79.35       11               \n",
       "39 98.68                 0.88                 76.73       11               \n",
       "   red_value          warnings\n",
       "1  N/A                N/A     \n",
       "2  N/A                N/A     \n",
       "⋮  ⋮                  ⋮       \n",
       "38 0.9173648727242681 N/A     \n",
       "39 N/A                N/A     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# GTDBTk summary\n",
    "tax = file.path(work_dir, 'LLG', 'gtdbtk', 'gtdbtk_bac_summary.tsv') %>%\n",
    "    read.delim(, sep='\\t') %>%\n",
    "    separate(classification, \n",
    "             c('Domain', 'Phylum', 'Class', 'Order', 'Family', 'Genus', 'Species'),\n",
    "             sep=';') %>%\n",
    "    select(-note, -classification_method, -pplacer_taxonomy,\n",
    "           -other_related_references.genome_id.species_name.radius.ANI.AF.)\n",
    "tax"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-22T08:46:30.672052Z",
     "iopub.status.busy": "2021-04-22T08:46:30.669934Z",
     "iopub.status.idle": "2021-04-22T08:46:30.709127Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- drep --\n",
      "intersect(x,y): 16 \n",
      "setdiff(x,y): 0 \n",
      "setdiff(y,x): 23 \n",
      "union(x,y): 39 \n",
      "-- checkm --\n",
      "intersect(x,y): 39 \n",
      "setdiff(x,y): 0 \n",
      "setdiff(y,x): 0 \n",
      "union(x,y): 39 \n",
      "-- gtdbtk --\n",
      "intersect(x,y): 39 \n",
      "setdiff(x,y): 0 \n",
      "setdiff(y,x): 0 \n",
      "union(x,y): 39 \n"
     ]
    }
   ],
   "source": [
    "# checking overlap\n",
    "cat('-- drep --\\n')\n",
    "overlap(basename(as.character(drep_gen$Fasta)), \n",
    "        basename(as.character(drep$genome)))\n",
    "cat('-- checkm --\\n')\n",
    "overlap(drep$Bin.Id, checkm$Bin.Id)\n",
    "cat('-- gtdbtk --\\n')\n",
    "overlap(drep$Bin.Id, tax$user_genome)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-22T08:46:32.198129Z",
     "iopub.status.busy": "2021-04-22T08:46:32.196063Z",
     "iopub.status.idle": "2021-04-22T08:46:32.281905Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 16 × 60</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>genome</th><th scope=col>completeness</th><th scope=col>contamination</th><th scope=col>Bin.Id</th><th scope=col>Marker.lineage</th><th scope=col>X..genomes</th><th scope=col>X..markers</th><th scope=col>X..marker.sets</th><th scope=col>Completeness</th><th scope=col>Contamination</th><th scope=col>⋯</th><th scope=col>fastani_af</th><th scope=col>closest_placement_reference</th><th scope=col>closest_placement_radius</th><th scope=col>closest_placement_taxonomy</th><th scope=col>closest_placement_ani</th><th scope=col>closest_placement_af</th><th scope=col>msa_percent</th><th scope=col>translation_table</th><th scope=col>red_value</th><th scope=col>warnings</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000283575.1_ASM28357v1_genomic.fna      </td><td>98.99</td><td>1.01</td><td>GCA_000283575.1_ASM28357v1_genomic      </td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>98.99</td><td>1.01</td><td>⋯</td><td>1.0 </td><td>GCF_000283575.1</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Oscillospiraceae;g__Oscillibacter;s__Oscillibacter valericigenes</td><td>100.0</td><td>1.0 </td><td>95.38</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000403435.2_Osci_bact_1-3_V1_genomic.fna</td><td>99.33</td><td>1.45</td><td>GCA_000403435.2_Osci_bact_1-3_V1_genomic</td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>99.33</td><td>1.45</td><td>⋯</td><td>0.99</td><td>GCF_000403435.2</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Oscillospiraceae;g__Oscillibacter;s__Oscillibacter sp000403435  </td><td>99.99</td><td>0.99</td><td>95.67</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋱</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_003497655.1_ASM349765v1_genomic.fna</td><td>91.14</td><td>0.00</td><td>GCA_003497655.1_ASM349765v1_genomic</td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>91.14</td><td>0.00</td><td>⋯</td><td>0.82</td><td>GCF_000765235.1</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Oscillospiraceae;g__ER4;s__ER4 sp000765235                    </td><td>96.71</td><td>0.82</td><td>91.90</td><td>11</td><td>N/A               </td><td>N/A</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_009774015.1_ASM977401v1_genomic.fna</td><td>91.03</td><td>3.68</td><td>GCA_009774015.1_ASM977401v1_genomic</td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>91.03</td><td>3.68</td><td>⋯</td><td>N/A </td><td>GCF_000403435.2</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Oscillospiraceae;g__Oscillibacter;s__Oscillibacter sp000403435</td><td>82.6 </td><td>0.59</td><td>83.51</td><td>11</td><td>0.9623815263719147</td><td>N/A</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 16 × 60\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " genome & completeness & contamination & Bin.Id & Marker.lineage & X..genomes & X..markers & X..marker.sets & Completeness & Contamination & ⋯ & fastani\\_af & closest\\_placement\\_reference & closest\\_placement\\_radius & closest\\_placement\\_taxonomy & closest\\_placement\\_ani & closest\\_placement\\_af & msa\\_percent & translation\\_table & red\\_value & warnings\\\\\n",
       " <fct> & <dbl> & <dbl> & <chr> & <fct> & <int> & <int> & <int> & <dbl> & <dbl> & ⋯ & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <dbl> & <int> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_000283575.1\\_ASM28357v1\\_genomic.fna       & 98.99 & 1.01 & GCA\\_000283575.1\\_ASM28357v1\\_genomic       & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 98.99 & 1.01 & ⋯ & 1.0  & GCF\\_000283575.1 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Oscillospiraceae;g\\_\\_Oscillibacter;s\\_\\_Oscillibacter valericigenes & 100.0 & 1.0  & 95.38 & 11 & N/A & N/A\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_000403435.2\\_Osci\\_bact\\_1-3\\_V1\\_genomic.fna & 99.33 & 1.45 & GCA\\_000403435.2\\_Osci\\_bact\\_1-3\\_V1\\_genomic & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 99.33 & 1.45 & ⋯ & 0.99 & GCF\\_000403435.2 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Oscillospiraceae;g\\_\\_Oscillibacter;s\\_\\_Oscillibacter sp000403435   & 99.99 & 0.99 & 95.67 & 11 & N/A & N/A\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋱ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_003497655.1\\_ASM349765v1\\_genomic.fna & 91.14 & 0.00 & GCA\\_003497655.1\\_ASM349765v1\\_genomic & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 91.14 & 0.00 & ⋯ & 0.82 & GCF\\_000765235.1 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Oscillospiraceae;g\\_\\_ER4;s\\_\\_ER4 sp000765235                     & 96.71 & 0.82 & 91.90 & 11 & N/A                & N/A\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_009774015.1\\_ASM977401v1\\_genomic.fna & 91.03 & 3.68 & GCA\\_009774015.1\\_ASM977401v1\\_genomic & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 91.03 & 3.68 & ⋯ & N/A  & GCF\\_000403435.2 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Oscillospiraceae;g\\_\\_Oscillibacter;s\\_\\_Oscillibacter sp000403435 & 82.6  & 0.59 & 83.51 & 11 & 0.9623815263719147 & N/A\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 16 × 60\n",
       "\n",
       "| genome &lt;fct&gt; | completeness &lt;dbl&gt; | contamination &lt;dbl&gt; | Bin.Id &lt;chr&gt; | Marker.lineage &lt;fct&gt; | X..genomes &lt;int&gt; | X..markers &lt;int&gt; | X..marker.sets &lt;int&gt; | Completeness &lt;dbl&gt; | Contamination &lt;dbl&gt; | ⋯ ⋯ | fastani_af &lt;fct&gt; | closest_placement_reference &lt;fct&gt; | closest_placement_radius &lt;fct&gt; | closest_placement_taxonomy &lt;fct&gt; | closest_placement_ani &lt;fct&gt; | closest_placement_af &lt;fct&gt; | msa_percent &lt;dbl&gt; | translation_table &lt;int&gt; | red_value &lt;fct&gt; | warnings &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000283575.1_ASM28357v1_genomic.fna       | 98.99 | 1.01 | GCA_000283575.1_ASM28357v1_genomic       | o__Clostridiales (UID1212) | 172 | 263 | 149 | 98.99 | 1.01 | ⋯ | 1.0  | GCF_000283575.1 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Oscillospiraceae;g__Oscillibacter;s__Oscillibacter valericigenes | 100.0 | 1.0  | 95.38 | 11 | N/A | N/A |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000403435.2_Osci_bact_1-3_V1_genomic.fna | 99.33 | 1.45 | GCA_000403435.2_Osci_bact_1-3_V1_genomic | o__Clostridiales (UID1212) | 172 | 263 | 149 | 99.33 | 1.45 | ⋯ | 0.99 | GCF_000403435.2 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Oscillospiraceae;g__Oscillibacter;s__Oscillibacter sp000403435   | 99.99 | 0.99 | 95.67 | 11 | N/A | N/A |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋱ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_003497655.1_ASM349765v1_genomic.fna | 91.14 | 0.00 | GCA_003497655.1_ASM349765v1_genomic | o__Clostridiales (UID1212) | 172 | 263 | 149 | 91.14 | 0.00 | ⋯ | 0.82 | GCF_000765235.1 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Oscillospiraceae;g__ER4;s__ER4 sp000765235                     | 96.71 | 0.82 | 91.90 | 11 | N/A                | N/A |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_009774015.1_ASM977401v1_genomic.fna | 91.03 | 3.68 | GCA_009774015.1_ASM977401v1_genomic | o__Clostridiales (UID1212) | 172 | 263 | 149 | 91.03 | 3.68 | ⋯ | N/A  | GCF_000403435.2 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Oscillospiraceae;g__Oscillibacter;s__Oscillibacter sp000403435 | 82.6  | 0.59 | 83.51 | 11 | 0.9623815263719147 | N/A |\n",
       "\n"
      ],
      "text/plain": [
       "   genome                                                                                            \n",
       "1  /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000283575.1_ASM28357v1_genomic.fna      \n",
       "2  /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000403435.2_Osci_bact_1-3_V1_genomic.fna\n",
       "⋮  ⋮                                                                                                 \n",
       "15 /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_003497655.1_ASM349765v1_genomic.fna     \n",
       "16 /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_009774015.1_ASM977401v1_genomic.fna     \n",
       "   completeness contamination Bin.Id                                  \n",
       "1  98.99        1.01          GCA_000283575.1_ASM28357v1_genomic      \n",
       "2  99.33        1.45          GCA_000403435.2_Osci_bact_1-3_V1_genomic\n",
       "⋮  ⋮            ⋮             ⋮                                       \n",
       "15 91.14        0.00          GCA_003497655.1_ASM349765v1_genomic     \n",
       "16 91.03        3.68          GCA_009774015.1_ASM977401v1_genomic     \n",
       "   Marker.lineage             X..genomes X..markers X..marker.sets Completeness\n",
       "1  o__Clostridiales (UID1212) 172        263        149            98.99       \n",
       "2  o__Clostridiales (UID1212) 172        263        149            99.33       \n",
       "⋮  ⋮                          ⋮          ⋮          ⋮              ⋮           \n",
       "15 o__Clostridiales (UID1212) 172        263        149            91.14       \n",
       "16 o__Clostridiales (UID1212) 172        263        149            91.03       \n",
       "   Contamination ⋯ fastani_af closest_placement_reference\n",
       "1  1.01          ⋯ 1.0        GCF_000283575.1            \n",
       "2  1.45          ⋯ 0.99       GCF_000403435.2            \n",
       "⋮  ⋮             ⋱ ⋮          ⋮                          \n",
       "15 0.00          ⋯ 0.82       GCF_000765235.1            \n",
       "16 3.68          ⋯ N/A        GCF_000403435.2            \n",
       "   closest_placement_radius\n",
       "1  95.0                    \n",
       "2  95.0                    \n",
       "⋮  ⋮                       \n",
       "15 95.0                    \n",
       "16 95.0                    \n",
       "   closest_placement_taxonomy                                                                                                      \n",
       "1  d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Oscillospiraceae;g__Oscillibacter;s__Oscillibacter valericigenes\n",
       "2  d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Oscillospiraceae;g__Oscillibacter;s__Oscillibacter sp000403435  \n",
       "⋮  ⋮                                                                                                                               \n",
       "15 d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Oscillospiraceae;g__ER4;s__ER4 sp000765235                      \n",
       "16 d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Oscillospiraceae;g__Oscillibacter;s__Oscillibacter sp000403435  \n",
       "   closest_placement_ani closest_placement_af msa_percent translation_table\n",
       "1  100.0                 1.0                  95.38       11               \n",
       "2  99.99                 0.99                 95.67       11               \n",
       "⋮  ⋮                     ⋮                    ⋮           ⋮                \n",
       "15 96.71                 0.82                 91.90       11               \n",
       "16 82.6                  0.59                 83.51       11               \n",
       "   red_value          warnings\n",
       "1  N/A                N/A     \n",
       "2  N/A                N/A     \n",
       "⋮  ⋮                  ⋮       \n",
       "15 N/A                N/A     \n",
       "16 0.9623815263719147 N/A     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# joining based on Bin.Id\n",
    "drep = drep %>%\n",
    "    inner_join(checkm, c('Bin.Id')) %>%\n",
    "    mutate(GEN = genome %>% as.character %>% basename) %>%\n",
    "    inner_join(drep_gen %>% mutate(GEN = Fasta %>% as.character %>% basename),\n",
    "               by=c('GEN')) %>%\n",
    "    inner_join(tax, c('Bin.Id'='user_genome')) #%>%\n",
    "drep"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-22T08:46:34.424094Z",
     "iopub.status.busy": "2021-04-22T08:46:34.421979Z",
     "iopub.status.idle": "2021-04-22T08:46:34.477785Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 14 × 60</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>genome</th><th scope=col>completeness</th><th scope=col>contamination</th><th scope=col>Bin.Id</th><th scope=col>Marker.lineage</th><th scope=col>X..genomes</th><th scope=col>X..markers</th><th scope=col>X..marker.sets</th><th scope=col>Completeness</th><th scope=col>Contamination</th><th scope=col>⋯</th><th scope=col>fastani_af</th><th scope=col>closest_placement_reference</th><th scope=col>closest_placement_radius</th><th scope=col>closest_placement_taxonomy</th><th scope=col>closest_placement_ani</th><th scope=col>closest_placement_af</th><th scope=col>msa_percent</th><th scope=col>translation_table</th><th scope=col>red_value</th><th scope=col>warnings</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000283575.1_ASM28357v1_genomic.fna      </td><td>98.99</td><td>1.01</td><td>GCA_000283575.1_ASM28357v1_genomic      </td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>98.99</td><td>1.01</td><td>⋯</td><td>1.0 </td><td>GCF_000283575.1</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Oscillospiraceae;g__Oscillibacter;s__Oscillibacter valericigenes</td><td>100.0</td><td>1.0 </td><td>95.38</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000403435.2_Osci_bact_1-3_V1_genomic.fna</td><td>99.33</td><td>1.45</td><td>GCA_000403435.2_Osci_bact_1-3_V1_genomic</td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>99.33</td><td>1.45</td><td>⋯</td><td>0.99</td><td>GCF_000403435.2</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Oscillospiraceae;g__Oscillibacter;s__Oscillibacter sp000403435  </td><td>99.99</td><td>0.99</td><td>95.67</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋱</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_003497655.1_ASM349765v1_genomic.fna</td><td>91.14</td><td>0.00</td><td>GCA_003497655.1_ASM349765v1_genomic</td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>91.14</td><td>0.00</td><td>⋯</td><td>0.82</td><td>GCF_000765235.1</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Oscillospiraceae;g__ER4;s__ER4 sp000765235                    </td><td>96.71</td><td>0.82</td><td>91.90</td><td>11</td><td>N/A               </td><td>N/A</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_009774015.1_ASM977401v1_genomic.fna</td><td>91.03</td><td>3.68</td><td>GCA_009774015.1_ASM977401v1_genomic</td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>91.03</td><td>3.68</td><td>⋯</td><td>N/A </td><td>GCF_000403435.2</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Oscillospiraceae;g__Oscillibacter;s__Oscillibacter sp000403435</td><td>82.6 </td><td>0.59</td><td>83.51</td><td>11</td><td>0.9623815263719147</td><td>N/A</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 14 × 60\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " genome & completeness & contamination & Bin.Id & Marker.lineage & X..genomes & X..markers & X..marker.sets & Completeness & Contamination & ⋯ & fastani\\_af & closest\\_placement\\_reference & closest\\_placement\\_radius & closest\\_placement\\_taxonomy & closest\\_placement\\_ani & closest\\_placement\\_af & msa\\_percent & translation\\_table & red\\_value & warnings\\\\\n",
       " <fct> & <dbl> & <dbl> & <chr> & <fct> & <int> & <int> & <int> & <dbl> & <dbl> & ⋯ & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <dbl> & <int> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_000283575.1\\_ASM28357v1\\_genomic.fna       & 98.99 & 1.01 & GCA\\_000283575.1\\_ASM28357v1\\_genomic       & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 98.99 & 1.01 & ⋯ & 1.0  & GCF\\_000283575.1 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Oscillospiraceae;g\\_\\_Oscillibacter;s\\_\\_Oscillibacter valericigenes & 100.0 & 1.0  & 95.38 & 11 & N/A & N/A\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_000403435.2\\_Osci\\_bact\\_1-3\\_V1\\_genomic.fna & 99.33 & 1.45 & GCA\\_000403435.2\\_Osci\\_bact\\_1-3\\_V1\\_genomic & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 99.33 & 1.45 & ⋯ & 0.99 & GCF\\_000403435.2 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Oscillospiraceae;g\\_\\_Oscillibacter;s\\_\\_Oscillibacter sp000403435   & 99.99 & 0.99 & 95.67 & 11 & N/A & N/A\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋱ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_003497655.1\\_ASM349765v1\\_genomic.fna & 91.14 & 0.00 & GCA\\_003497655.1\\_ASM349765v1\\_genomic & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 91.14 & 0.00 & ⋯ & 0.82 & GCF\\_000765235.1 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Oscillospiraceae;g\\_\\_ER4;s\\_\\_ER4 sp000765235                     & 96.71 & 0.82 & 91.90 & 11 & N/A                & N/A\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_009774015.1\\_ASM977401v1\\_genomic.fna & 91.03 & 3.68 & GCA\\_009774015.1\\_ASM977401v1\\_genomic & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 91.03 & 3.68 & ⋯ & N/A  & GCF\\_000403435.2 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Oscillospiraceae;g\\_\\_Oscillibacter;s\\_\\_Oscillibacter sp000403435 & 82.6  & 0.59 & 83.51 & 11 & 0.9623815263719147 & N/A\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 14 × 60\n",
       "\n",
       "| genome &lt;fct&gt; | completeness &lt;dbl&gt; | contamination &lt;dbl&gt; | Bin.Id &lt;chr&gt; | Marker.lineage &lt;fct&gt; | X..genomes &lt;int&gt; | X..markers &lt;int&gt; | X..marker.sets &lt;int&gt; | Completeness &lt;dbl&gt; | Contamination &lt;dbl&gt; | ⋯ ⋯ | fastani_af &lt;fct&gt; | closest_placement_reference &lt;fct&gt; | closest_placement_radius &lt;fct&gt; | closest_placement_taxonomy &lt;fct&gt; | closest_placement_ani &lt;fct&gt; | closest_placement_af &lt;fct&gt; | msa_percent &lt;dbl&gt; | translation_table &lt;int&gt; | red_value &lt;fct&gt; | warnings &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000283575.1_ASM28357v1_genomic.fna       | 98.99 | 1.01 | GCA_000283575.1_ASM28357v1_genomic       | o__Clostridiales (UID1212) | 172 | 263 | 149 | 98.99 | 1.01 | ⋯ | 1.0  | GCF_000283575.1 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Oscillospiraceae;g__Oscillibacter;s__Oscillibacter valericigenes | 100.0 | 1.0  | 95.38 | 11 | N/A | N/A |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000403435.2_Osci_bact_1-3_V1_genomic.fna | 99.33 | 1.45 | GCA_000403435.2_Osci_bact_1-3_V1_genomic | o__Clostridiales (UID1212) | 172 | 263 | 149 | 99.33 | 1.45 | ⋯ | 0.99 | GCF_000403435.2 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Oscillospiraceae;g__Oscillibacter;s__Oscillibacter sp000403435   | 99.99 | 0.99 | 95.67 | 11 | N/A | N/A |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋱ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_003497655.1_ASM349765v1_genomic.fna | 91.14 | 0.00 | GCA_003497655.1_ASM349765v1_genomic | o__Clostridiales (UID1212) | 172 | 263 | 149 | 91.14 | 0.00 | ⋯ | 0.82 | GCF_000765235.1 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Oscillospiraceae;g__ER4;s__ER4 sp000765235                     | 96.71 | 0.82 | 91.90 | 11 | N/A                | N/A |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_009774015.1_ASM977401v1_genomic.fna | 91.03 | 3.68 | GCA_009774015.1_ASM977401v1_genomic | o__Clostridiales (UID1212) | 172 | 263 | 149 | 91.03 | 3.68 | ⋯ | N/A  | GCF_000403435.2 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Oscillospiraceae;g__Oscillibacter;s__Oscillibacter sp000403435 | 82.6  | 0.59 | 83.51 | 11 | 0.9623815263719147 | N/A |\n",
       "\n"
      ],
      "text/plain": [
       "   genome                                                                                            \n",
       "1  /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000283575.1_ASM28357v1_genomic.fna      \n",
       "2  /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000403435.2_Osci_bact_1-3_V1_genomic.fna\n",
       "⋮  ⋮                                                                                                 \n",
       "13 /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_003497655.1_ASM349765v1_genomic.fna     \n",
       "14 /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_009774015.1_ASM977401v1_genomic.fna     \n",
       "   completeness contamination Bin.Id                                  \n",
       "1  98.99        1.01          GCA_000283575.1_ASM28357v1_genomic      \n",
       "2  99.33        1.45          GCA_000403435.2_Osci_bact_1-3_V1_genomic\n",
       "⋮  ⋮            ⋮             ⋮                                       \n",
       "13 91.14        0.00          GCA_003497655.1_ASM349765v1_genomic     \n",
       "14 91.03        3.68          GCA_009774015.1_ASM977401v1_genomic     \n",
       "   Marker.lineage             X..genomes X..markers X..marker.sets Completeness\n",
       "1  o__Clostridiales (UID1212) 172        263        149            98.99       \n",
       "2  o__Clostridiales (UID1212) 172        263        149            99.33       \n",
       "⋮  ⋮                          ⋮          ⋮          ⋮              ⋮           \n",
       "13 o__Clostridiales (UID1212) 172        263        149            91.14       \n",
       "14 o__Clostridiales (UID1212) 172        263        149            91.03       \n",
       "   Contamination ⋯ fastani_af closest_placement_reference\n",
       "1  1.01          ⋯ 1.0        GCF_000283575.1            \n",
       "2  1.45          ⋯ 0.99       GCF_000403435.2            \n",
       "⋮  ⋮             ⋱ ⋮          ⋮                          \n",
       "13 0.00          ⋯ 0.82       GCF_000765235.1            \n",
       "14 3.68          ⋯ N/A        GCF_000403435.2            \n",
       "   closest_placement_radius\n",
       "1  95.0                    \n",
       "2  95.0                    \n",
       "⋮  ⋮                       \n",
       "13 95.0                    \n",
       "14 95.0                    \n",
       "   closest_placement_taxonomy                                                                                                      \n",
       "1  d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Oscillospiraceae;g__Oscillibacter;s__Oscillibacter valericigenes\n",
       "2  d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Oscillospiraceae;g__Oscillibacter;s__Oscillibacter sp000403435  \n",
       "⋮  ⋮                                                                                                                               \n",
       "13 d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Oscillospiraceae;g__ER4;s__ER4 sp000765235                      \n",
       "14 d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Oscillospiraceae;g__Oscillibacter;s__Oscillibacter sp000403435  \n",
       "   closest_placement_ani closest_placement_af msa_percent translation_table\n",
       "1  100.0                 1.0                  95.38       11               \n",
       "2  99.99                 0.99                 95.67       11               \n",
       "⋮  ⋮                     ⋮                    ⋮           ⋮                \n",
       "13 96.71                 0.82                 91.90       11               \n",
       "14 82.6                  0.59                 83.51       11               \n",
       "   red_value          warnings\n",
       "1  N/A                N/A     \n",
       "2  N/A                N/A     \n",
       "⋮  ⋮                  ⋮       \n",
       "13 N/A                N/A     \n",
       "14 0.9623815263719147 N/A     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# filtering by quality\n",
    "hq_genomes = drep %>%\n",
    "    filter(completeness >= 90,\n",
    "           contamination < 5,\n",
    "           Strain.heterogeneity < 50)\n",
    "hq_genomes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-22T08:46:37.375278Z",
     "iopub.status.busy": "2021-04-22T08:46:37.373157Z",
     "iopub.status.idle": "2021-04-22T08:46:37.410321Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A tibble: 4 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Family</th><th scope=col>Genus</th><th scope=col>n_genomes</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>f__Oscillospiraceae</td><td>g__             </td><td>1</td></tr>\n",
       "\t<tr><td>f__Oscillospiraceae</td><td>g__CAG-83       </td><td>1</td></tr>\n",
       "\t<tr><td>f__Oscillospiraceae</td><td>g__ER4          </td><td>3</td></tr>\n",
       "\t<tr><td>f__Oscillospiraceae</td><td>g__Oscillibacter</td><td>9</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 4 × 3\n",
       "\\begin{tabular}{lll}\n",
       " Family & Genus & n\\_genomes\\\\\n",
       " <chr> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t f\\_\\_Oscillospiraceae & g\\_\\_              & 1\\\\\n",
       "\t f\\_\\_Oscillospiraceae & g\\_\\_CAG-83        & 1\\\\\n",
       "\t f\\_\\_Oscillospiraceae & g\\_\\_ER4           & 3\\\\\n",
       "\t f\\_\\_Oscillospiraceae & g\\_\\_Oscillibacter & 9\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 4 × 3\n",
       "\n",
       "| Family &lt;chr&gt; | Genus &lt;chr&gt; | n_genomes &lt;int&gt; |\n",
       "|---|---|---|\n",
       "| f__Oscillospiraceae | g__              | 1 |\n",
       "| f__Oscillospiraceae | g__CAG-83        | 1 |\n",
       "| f__Oscillospiraceae | g__ER4           | 3 |\n",
       "| f__Oscillospiraceae | g__Oscillibacter | 9 |\n",
       "\n"
      ],
      "text/plain": [
       "  Family              Genus            n_genomes\n",
       "1 f__Oscillospiraceae g__              1        \n",
       "2 f__Oscillospiraceae g__CAG-83        1        \n",
       "3 f__Oscillospiraceae g__ER4           3        \n",
       "4 f__Oscillospiraceae g__Oscillibacter 9        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# summarizing the taxonomy\n",
    "df.dims(20)\n",
    "hq_genomes %>%\n",
    "    group_by(Family, Genus) %>%\n",
    "    summarize(n_genomes = n(), .groups='drop')\n",
    "df.dims()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-22T08:53:48.529461Z",
     "iopub.status.busy": "2021-04-22T08:53:48.525461Z",
     "iopub.status.idle": "2021-04-22T08:53:48.561572Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "File written: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3//Oscillibacter/samples_genomes_hq.txt \n"
     ]
    }
   ],
   "source": [
    "# writing samples table for LLPRIMER\n",
    "outfile = file.path(work_dir, 'samples_genomes_hq.txt')\n",
    "hq_genomes %>%\n",
    "    select(Bin.Id, Fasta) %>%\n",
    "    rename('Taxon' = Bin.Id) %>%\n",
    "    mutate(Taxon = gsub('_chromosome.+', '', Taxon),\n",
    "           Taxon = gsub('_bin_.+', '', Taxon),\n",
    "           Taxon = gsub('_genomic', '', Taxon),\n",
    "           Taxon = gsub('_annotated_assembly', '', Taxon),\n",
    "           Taxid = taxid,\n",
    "           Domain = domain) %>%\n",
    "    write_table(outfile)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Primer design"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Config"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-22T08:47:59.935010Z",
     "iopub.status.busy": "2021-04-22T08:47:59.933095Z",
     "iopub.status.idle": "2021-04-22T08:47:59.965868Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#-- I/O --#\n",
      "samples_file: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Oscillibacter/samples_genomes_hq.txt\n",
      "\n",
      "# output location\n",
      "output_dir: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Oscillibacter/primers/\n",
      "\n",
      "# temporary file directory (your username will be added automatically)\n",
      "tmp_dir: /ebio/abt3_scratch/\n",
      "\n",
      "#-- software parameters --#\n",
      "# See the README for a description\n",
      "params:\n",
      "  ionice: -c 3\n",
      "  cgp:\n",
      "    prokka: \"\"\n",
      "    mmseqs:\n",
      "      method: cluster    # or linclust (faster)\n",
      "      run: --min-seq-id 0.9 -c 0.8\n",
      "    vsearch: --id 0.94\n",
      "    core_genes:\n",
      "      cds: --perc-genomes-cds 100 --copies-per-genome-cds 1 --max-clusters-cds 10 #500\n",
      "      rRNA: --perc-genomes-rrna 100 --copies-per-genome-rrna 10 --max-clusters-rrna 500\n",
      "    align:\n",
      "      method: linsi\n",
      "      params: --auto --maxiterate 1000\n",
      "    primer3:\n",
      "      consensus: --consensus-threshold 0.34\n",
      "      number: --num-raw-primers 1000 --num-final-primers 10\n",
      "      size: --opt-size 20 --min-size 18 --max-size 24\n",
      "      product: --opt-prod-size 150 --min-prod-size 100 --max-prod-size 250\n",
      "      Tm: --opt-tm 62 --min-tm 54 --max-tm 70 --max-tm-diff 2\n",
      "      PCR: --oligo-DNA 50 --salt-monovalent 50 --salt-divalent 1.5 --dNTPs 0.2\n",
      "      GC: --opt-gc 50 --min-gc 25 --max-gc 75\n",
      "      degeneracy: --max-degeneracy 128 --max-degeneracy-3prime 6 --window-3prime 5\n",
      "      internal_oligo: --make-oligo 0 --int-opt-size 20 --int-opt-tm 60\n",
      "    blast_nontarget:\n",
      "      run_locally: False\n",
      "      cds: -evalue 1e-5 -max_target_seqs 30\n",
      "      rRNA: -evalue 1e-5 -max_target_seqs 30\n",
      "    blastn: -max_target_seqs 500 -perc_identity 90\n",
      "    blast_filter: --perc-len 80 --min-amplicon-len 30 --max-amplicon-len 2000\n",
      "    seqkit_amplicon: \"\"\n",
      "\n",
      "#-- database --#\n",
      "databases:\n",
      "  blast:\n",
      "    nucl: /ebio/abt3_projects/databases_no-backup/NCBI_blastdb/nt\n",
      "    prot: /ebio/abt3_projects/databases_no-backup/NCBI_blastdb/nr\n",
      "    rRNA:\n",
      "      SSU: /ebio/abt3_projects/databases_no-backup/NCBI_rRNA/2021-01/SSU/16S_ribosomal_RNA\n",
      "      LSU: /ebio/abt3_projects/databases_no-backup/NCBI_rRNA/2021-01/LSU/LSU_prokaryote_rRNA\n",
      "    gene_names: /ebio/abt3_projects/databases_no-backup/NCBI_nr_geneIDs/2021-01/nr_accession-geneNames.pkl\n",
      "  taxonomy:\n",
      "    names: /ebio/abt3_projects/databases_no-backup/NCBI_taxonomy/2020-12-23/names.dmp\n",
      "    nodes: /ebio/abt3_projects/databases_no-backup/NCBI_taxonomy/2020-12-23/nodes.dmp\n",
      "\n",
      "#-- snakemake pipeline --#\n",
      "pipeline:\n",
      "  snakemake_folder: ./\n",
      "  script_folder: bin/scripts/\n",
      "  name: LLPRIMER"
     ]
    }
   ],
   "source": [
    "F = file.path(work_dir, 'primers', 'config.yaml')\n",
    "cat_file(F)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Run"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "(snakemake) @ rick:/ebio/abt3_projects/software/dev/ll_pipelines/llprimer\n",
    "$ screen -L -S llprimer-Osc ./snakemake_sge.sh /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Oscillibacter/primers/config.yaml 30 --notemp\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Summary"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Primers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-24T13:16:25.987102Z",
     "iopub.status.busy": "2021-04-24T13:16:25.984557Z",
     "iopub.status.idle": "2021-04-24T13:16:26.056027Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No. of unique primers: 162 \n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 370 × 22</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>gene_type</th><th scope=col>cluster_id</th><th scope=col>primer_set</th><th scope=col>amplicon_size_consensus</th><th scope=col>amplicon_size_avg</th><th scope=col>amplicon_size_sd</th><th scope=col>primer_id</th><th scope=col>primer_type</th><th scope=col>sequence</th><th scope=col>length</th><th scope=col>⋯</th><th scope=col>position_start</th><th scope=col>position_end</th><th scope=col>Tm_avg</th><th scope=col>Tm_sd</th><th scope=col>GC_avg</th><th scope=col>GC_sd</th><th scope=col>hairpin_avg</th><th scope=col>hairpin_sd</th><th scope=col>homodimer_avg</th><th scope=col>homodimer_sd</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>cds</td><td>1</td><td>468</td><td>189</td><td>189</td><td>0</td><td>468f</td><td>PRIMER_LEFT </td><td>AARAACGGYCGYGAYTCYRA</td><td>20</td><td>⋯</td><td> 54</td><td> 74</td><td>60.76051</td><td>2.848347</td><td>50.00000</td><td>6.123724</td><td>16.30579</td><td>21.51507</td><td>22.824341</td><td>16.40584</td></tr>\n",
       "\t<tr><td>cds</td><td>1</td><td>468</td><td>189</td><td>189</td><td>0</td><td>468r</td><td>PRIMER_RIGHT</td><td>RCCSAGVYGCTCRAARCG  </td><td>18</td><td>⋯</td><td>225</td><td>243</td><td>62.16555</td><td>2.574813</td><td>64.81481</td><td>6.141898</td><td>35.16456</td><td>27.78734</td><td> 4.565427</td><td>23.59421</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋱</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>cds</td><td>38</td><td>911</td><td>204</td><td>204</td><td>0</td><td>911f</td><td>PRIMER_LEFT </td><td>GGHAAGATGCTSAAYGTSGA </td><td>20</td><td>⋯</td><td>1401</td><td>1421</td><td>58.39153</td><td>1.903507</td><td>49.16667</td><td>3.435921</td><td>21.547320</td><td>21.60791</td><td>-18.44452</td><td>20.099912</td></tr>\n",
       "\t<tr><td>cds</td><td>38</td><td>911</td><td>204</td><td>204</td><td>0</td><td>911r</td><td>PRIMER_RIGHT</td><td>SCGGAARAARAAVGTCARMAG</td><td>21</td><td>⋯</td><td>1584</td><td>1605</td><td>57.91049</td><td>1.924208</td><td>46.03175</td><td>5.264484</td><td> 4.451632</td><td>11.86112</td><td>-24.35675</td><td> 9.648227</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 370 × 22\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " gene\\_type & cluster\\_id & primer\\_set & amplicon\\_size\\_consensus & amplicon\\_size\\_avg & amplicon\\_size\\_sd & primer\\_id & primer\\_type & sequence & length & ⋯ & position\\_start & position\\_end & Tm\\_avg & Tm\\_sd & GC\\_avg & GC\\_sd & hairpin\\_avg & hairpin\\_sd & homodimer\\_avg & homodimer\\_sd\\\\\n",
       " <fct> & <int> & <int> & <int> & <dbl> & <dbl> & <fct> & <fct> & <fct> & <int> & ⋯ & <int> & <int> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t cds & 1 & 468 & 189 & 189 & 0 & 468f & PRIMER\\_LEFT  & AARAACGGYCGYGAYTCYRA & 20 & ⋯ &  54 &  74 & 60.76051 & 2.848347 & 50.00000 & 6.123724 & 16.30579 & 21.51507 & 22.824341 & 16.40584\\\\\n",
       "\t cds & 1 & 468 & 189 & 189 & 0 & 468r & PRIMER\\_RIGHT & RCCSAGVYGCTCRAARCG   & 18 & ⋯ & 225 & 243 & 62.16555 & 2.574813 & 64.81481 & 6.141898 & 35.16456 & 27.78734 &  4.565427 & 23.59421\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋱ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t cds & 38 & 911 & 204 & 204 & 0 & 911f & PRIMER\\_LEFT  & GGHAAGATGCTSAAYGTSGA  & 20 & ⋯ & 1401 & 1421 & 58.39153 & 1.903507 & 49.16667 & 3.435921 & 21.547320 & 21.60791 & -18.44452 & 20.099912\\\\\n",
       "\t cds & 38 & 911 & 204 & 204 & 0 & 911r & PRIMER\\_RIGHT & SCGGAARAARAAVGTCARMAG & 21 & ⋯ & 1584 & 1605 & 57.91049 & 1.924208 & 46.03175 & 5.264484 &  4.451632 & 11.86112 & -24.35675 &  9.648227\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 370 × 22\n",
       "\n",
       "| gene_type &lt;fct&gt; | cluster_id &lt;int&gt; | primer_set &lt;int&gt; | amplicon_size_consensus &lt;int&gt; | amplicon_size_avg &lt;dbl&gt; | amplicon_size_sd &lt;dbl&gt; | primer_id &lt;fct&gt; | primer_type &lt;fct&gt; | sequence &lt;fct&gt; | length &lt;int&gt; | ⋯ ⋯ | position_start &lt;int&gt; | position_end &lt;int&gt; | Tm_avg &lt;dbl&gt; | Tm_sd &lt;dbl&gt; | GC_avg &lt;dbl&gt; | GC_sd &lt;dbl&gt; | hairpin_avg &lt;dbl&gt; | hairpin_sd &lt;dbl&gt; | homodimer_avg &lt;dbl&gt; | homodimer_sd &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| cds | 1 | 468 | 189 | 189 | 0 | 468f | PRIMER_LEFT  | AARAACGGYCGYGAYTCYRA | 20 | ⋯ |  54 |  74 | 60.76051 | 2.848347 | 50.00000 | 6.123724 | 16.30579 | 21.51507 | 22.824341 | 16.40584 |\n",
       "| cds | 1 | 468 | 189 | 189 | 0 | 468r | PRIMER_RIGHT | RCCSAGVYGCTCRAARCG   | 18 | ⋯ | 225 | 243 | 62.16555 | 2.574813 | 64.81481 | 6.141898 | 35.16456 | 27.78734 |  4.565427 | 23.59421 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋱ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| cds | 38 | 911 | 204 | 204 | 0 | 911f | PRIMER_LEFT  | GGHAAGATGCTSAAYGTSGA  | 20 | ⋯ | 1401 | 1421 | 58.39153 | 1.903507 | 49.16667 | 3.435921 | 21.547320 | 21.60791 | -18.44452 | 20.099912 |\n",
       "| cds | 38 | 911 | 204 | 204 | 0 | 911r | PRIMER_RIGHT | SCGGAARAARAAVGTCARMAG | 21 | ⋯ | 1584 | 1605 | 57.91049 | 1.924208 | 46.03175 | 5.264484 |  4.451632 | 11.86112 | -24.35675 |  9.648227 |\n",
       "\n"
      ],
      "text/plain": [
       "    gene_type cluster_id primer_set amplicon_size_consensus amplicon_size_avg\n",
       "1   cds       1          468        189                     189              \n",
       "2   cds       1          468        189                     189              \n",
       "⋮   ⋮         ⋮          ⋮          ⋮                       ⋮                \n",
       "369 cds       38         911        204                     204              \n",
       "370 cds       38         911        204                     204              \n",
       "    amplicon_size_sd primer_id primer_type  sequence              length ⋯\n",
       "1   0                468f      PRIMER_LEFT  AARAACGGYCGYGAYTCYRA  20     ⋯\n",
       "2   0                468r      PRIMER_RIGHT RCCSAGVYGCTCRAARCG    18     ⋯\n",
       "⋮   ⋮                ⋮         ⋮            ⋮                     ⋮      ⋱\n",
       "369 0                911f      PRIMER_LEFT  GGHAAGATGCTSAAYGTSGA  20     ⋯\n",
       "370 0                911r      PRIMER_RIGHT SCGGAARAARAAVGTCARMAG 21     ⋯\n",
       "    position_start position_end Tm_avg   Tm_sd    GC_avg   GC_sd    hairpin_avg\n",
       "1    54             74          60.76051 2.848347 50.00000 6.123724 16.30579   \n",
       "2   225            243          62.16555 2.574813 64.81481 6.141898 35.16456   \n",
       "⋮   ⋮              ⋮            ⋮        ⋮        ⋮        ⋮        ⋮          \n",
       "369 1401           1421         58.39153 1.903507 49.16667 3.435921 21.547320  \n",
       "370 1584           1605         57.91049 1.924208 46.03175 5.264484  4.451632  \n",
       "    hairpin_sd homodimer_avg homodimer_sd\n",
       "1   21.51507   22.824341     16.40584    \n",
       "2   27.78734    4.565427     23.59421    \n",
       "⋮   ⋮          ⋮             ⋮           \n",
       "369 21.60791   -18.44452     20.099912   \n",
       "370 11.86112   -24.35675      9.648227   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "primer_info = read.delim(file.path(work_dir, 'primers', 'cgp', 'primers_final_info.tsv'), sep='\\t')\n",
    "primer_info %>% unique_n('primers', primer_set)\n",
    "primer_info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Gene clusters"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-23T07:49:06.804811Z",
     "iopub.status.busy": "2021-04-23T07:49:06.802457Z",
     "iopub.status.idle": "2021-04-23T07:49:06.851780Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No. of clusters 26 \n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 364 × 13</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>gene_type</th><th scope=col>cluster_id</th><th scope=col>seq_uuid</th><th scope=col>seq_orig_name</th><th scope=col>contig_id</th><th scope=col>taxon</th><th scope=col>start</th><th scope=col>end</th><th scope=col>score</th><th scope=col>strand</th><th scope=col>annotation</th><th scope=col>cluster_name</th><th scope=col>clust_id</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>cds</td><td>1</td><td>0d69d3fe0064481cbd2260b6630da5ef</td><td>FKNECIIB_01044</td><td>AP012044.1</td><td>GCA_000283575.1_ASM28357v1      </td><td>1072042</td><td>1072323</td><td>.</td><td>-</td><td>50S ribosomal protein L27</td><td>05b02d57263741a486e49b34bde0fa28</td><td>1</td></tr>\n",
       "\t<tr><td>cds</td><td>1</td><td>18f7efb1fe36464b80562c6f3fa295b7</td><td>MKEBFIFM_00152</td><td>KE159703.1</td><td>GCA_000403435.2_Osci_bact_1-3_V1</td><td> 150345</td><td> 150626</td><td>.</td><td>+</td><td>50S ribosomal protein L27</td><td>05b02d57263741a486e49b34bde0fa28</td><td>1</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>cds</td><td>38</td><td>3bfb278fd5594012b6ce303878008895</td><td>MINOPNOJ_01286</td><td>DNON01000002.1</td><td>GCA_003497655.1_ASM349765v1</td><td> 9130</td><td>11097</td><td>.</td><td>+</td><td>DNA gyrase subunit B</td><td>5730334f42ee4013a044d98cfa4e837c</td><td>38</td></tr>\n",
       "\t<tr><td>cds</td><td>38</td><td>f51bc6153d974eca99492cceaa1dfaf3</td><td>ONMBBDAA_00481</td><td>VSMZ01000014.1</td><td>GCA_009774015.1_ASM977401v1</td><td>18854</td><td>20803</td><td>.</td><td>+</td><td>DNA gyrase subunit B</td><td>5730334f42ee4013a044d98cfa4e837c</td><td>38</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 364 × 13\n",
       "\\begin{tabular}{lllllllllllll}\n",
       " gene\\_type & cluster\\_id & seq\\_uuid & seq\\_orig\\_name & contig\\_id & taxon & start & end & score & strand & annotation & cluster\\_name & clust\\_id\\\\\n",
       " <fct> & <int> & <fct> & <fct> & <fct> & <fct> & <int> & <int> & <fct> & <fct> & <fct> & <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t cds & 1 & 0d69d3fe0064481cbd2260b6630da5ef & FKNECIIB\\_01044 & AP012044.1 & GCA\\_000283575.1\\_ASM28357v1       & 1072042 & 1072323 & . & - & 50S ribosomal protein L27 & 05b02d57263741a486e49b34bde0fa28 & 1\\\\\n",
       "\t cds & 1 & 18f7efb1fe36464b80562c6f3fa295b7 & MKEBFIFM\\_00152 & KE159703.1 & GCA\\_000403435.2\\_Osci\\_bact\\_1-3\\_V1 &  150345 &  150626 & . & + & 50S ribosomal protein L27 & 05b02d57263741a486e49b34bde0fa28 & 1\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t cds & 38 & 3bfb278fd5594012b6ce303878008895 & MINOPNOJ\\_01286 & DNON01000002.1 & GCA\\_003497655.1\\_ASM349765v1 &  9130 & 11097 & . & + & DNA gyrase subunit B & 5730334f42ee4013a044d98cfa4e837c & 38\\\\\n",
       "\t cds & 38 & f51bc6153d974eca99492cceaa1dfaf3 & ONMBBDAA\\_00481 & VSMZ01000014.1 & GCA\\_009774015.1\\_ASM977401v1 & 18854 & 20803 & . & + & DNA gyrase subunit B & 5730334f42ee4013a044d98cfa4e837c & 38\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 364 × 13\n",
       "\n",
       "| gene_type &lt;fct&gt; | cluster_id &lt;int&gt; | seq_uuid &lt;fct&gt; | seq_orig_name &lt;fct&gt; | contig_id &lt;fct&gt; | taxon &lt;fct&gt; | start &lt;int&gt; | end &lt;int&gt; | score &lt;fct&gt; | strand &lt;fct&gt; | annotation &lt;fct&gt; | cluster_name &lt;fct&gt; | clust_id &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| cds | 1 | 0d69d3fe0064481cbd2260b6630da5ef | FKNECIIB_01044 | AP012044.1 | GCA_000283575.1_ASM28357v1       | 1072042 | 1072323 | . | - | 50S ribosomal protein L27 | 05b02d57263741a486e49b34bde0fa28 | 1 |\n",
       "| cds | 1 | 18f7efb1fe36464b80562c6f3fa295b7 | MKEBFIFM_00152 | KE159703.1 | GCA_000403435.2_Osci_bact_1-3_V1 |  150345 |  150626 | . | + | 50S ribosomal protein L27 | 05b02d57263741a486e49b34bde0fa28 | 1 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| cds | 38 | 3bfb278fd5594012b6ce303878008895 | MINOPNOJ_01286 | DNON01000002.1 | GCA_003497655.1_ASM349765v1 |  9130 | 11097 | . | + | DNA gyrase subunit B | 5730334f42ee4013a044d98cfa4e837c | 38 |\n",
       "| cds | 38 | f51bc6153d974eca99492cceaa1dfaf3 | ONMBBDAA_00481 | VSMZ01000014.1 | GCA_009774015.1_ASM977401v1 | 18854 | 20803 | . | + | DNA gyrase subunit B | 5730334f42ee4013a044d98cfa4e837c | 38 |\n",
       "\n"
      ],
      "text/plain": [
       "    gene_type cluster_id seq_uuid                         seq_orig_name \n",
       "1   cds       1          0d69d3fe0064481cbd2260b6630da5ef FKNECIIB_01044\n",
       "2   cds       1          18f7efb1fe36464b80562c6f3fa295b7 MKEBFIFM_00152\n",
       "⋮   ⋮         ⋮          ⋮                                ⋮             \n",
       "363 cds       38         3bfb278fd5594012b6ce303878008895 MINOPNOJ_01286\n",
       "364 cds       38         f51bc6153d974eca99492cceaa1dfaf3 ONMBBDAA_00481\n",
       "    contig_id      taxon                            start   end     score\n",
       "1   AP012044.1     GCA_000283575.1_ASM28357v1       1072042 1072323 .    \n",
       "2   KE159703.1     GCA_000403435.2_Osci_bact_1-3_V1  150345  150626 .    \n",
       "⋮   ⋮              ⋮                                ⋮       ⋮       ⋮    \n",
       "363 DNON01000002.1 GCA_003497655.1_ASM349765v1       9130   11097   .    \n",
       "364 VSMZ01000014.1 GCA_009774015.1_ASM977401v1      18854   20803   .    \n",
       "    strand annotation                cluster_name                     clust_id\n",
       "1   -      50S ribosomal protein L27 05b02d57263741a486e49b34bde0fa28 1       \n",
       "2   +      50S ribosomal protein L27 05b02d57263741a486e49b34bde0fa28 1       \n",
       "⋮   ⋮      ⋮                         ⋮                                ⋮       \n",
       "363 +      DNA gyrase subunit B      5730334f42ee4013a044d98cfa4e837c 38      \n",
       "364 +      DNA gyrase subunit B      5730334f42ee4013a044d98cfa4e837c 38      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# target gene info\n",
    "gene_annot = read.delim(file.path(work_dir, 'primers', 'cgp', 'core_clusters_info.tsv'), \n",
    "                        sep='\\t') %>%\n",
    "    semi_join(primer_info, c('cluster_id')) \n",
    "gene_annot %>% distinct(gene_type, cluster_id) %>% nrow %>% cat('No. of clusters', ., '\\n')\n",
    "gene_annot"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Gene cluster annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-23T07:49:08.105020Z",
     "iopub.status.busy": "2021-04-23T07:49:08.102882Z",
     "iopub.status.idle": "2021-04-23T07:49:08.165289Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 780 × 17</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>cluster_id</th><th scope=col>query</th><th scope=col>subject</th><th scope=col>subject_name</th><th scope=col>pident</th><th scope=col>length</th><th scope=col>mismatch</th><th scope=col>qstart</th><th scope=col>qend</th><th scope=col>sstart</th><th scope=col>send</th><th scope=col>evalue</th><th scope=col>slen</th><th scope=col>qlen</th><th scope=col>sscinames</th><th scope=col>staxids</th><th scope=col>pident_rank</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td>05b02d57263741a486e49b34bde0fa28</td><td>WP_025543765.1</td><td>MULTISPECIES: 50S ribosomal protein L27 [Oscillospiraceae]</td><td>100.000</td><td>93</td><td> 0</td><td>1</td><td>279</td><td>1</td><td>93</td><td>1.43e-59</td><td>93</td><td>282</td><td>Oscillospiraceae;Oscillibacter valericigenes;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis;Oscillibacter sp. PEA192</td><td>216572;351091;1226322;1226323;2093857;2109687</td><td>30</td></tr>\n",
       "\t<tr><td>1</td><td>05b02d57263741a486e49b34bde0fa28</td><td>WP_187032256.1</td><td>50S ribosomal protein L27 [Dysosmobacter sp. NSJ-60]      </td><td> 86.022</td><td>93</td><td>13</td><td>1</td><td>279</td><td>1</td><td>93</td><td>1.68e-50</td><td>93</td><td>282</td><td>Dysosmobacter sp. NSJ-60                                                                                                                           </td><td>2763041                                      </td><td> 2</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>38</td><td>5730334f42ee4013a044d98cfa4e837c</td><td>CDA81968.1    </td><td>dNA gyrase subunit B [Firmicutes bacterium CAG:176]</td><td>81.985</td><td>655</td><td>107</td><td>1</td><td>1965</td><td>1</td><td>644</td><td>0</td><td>644</td><td>1968</td><td>Firmicutes bacterium CAG:176     </td><td>1263007</td><td>26</td></tr>\n",
       "\t<tr><td>38</td><td>5730334f42ee4013a044d98cfa4e837c</td><td>WP_199648700.1</td><td>None                                               </td><td>82.290</td><td>655</td><td>107</td><td>1</td><td>1965</td><td>1</td><td>646</td><td>0</td><td>646</td><td>1968</td><td>Pseudoflavonifractor sp. AF19-9AC</td><td>2292244</td><td>22</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 780 × 17\n",
       "\\begin{tabular}{lllllllllllllllll}\n",
       " cluster\\_id & query & subject & subject\\_name & pident & length & mismatch & qstart & qend & sstart & send & evalue & slen & qlen & sscinames & staxids & pident\\_rank\\\\\n",
       " <int> & <fct> & <fct> & <fct> & <dbl> & <int> & <int> & <int> & <int> & <int> & <int> & <dbl> & <int> & <int> & <fct> & <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t 1 & 05b02d57263741a486e49b34bde0fa28 & WP\\_025543765.1 & MULTISPECIES: 50S ribosomal protein L27 {[}Oscillospiraceae{]} & 100.000 & 93 &  0 & 1 & 279 & 1 & 93 & 1.43e-59 & 93 & 282 & Oscillospiraceae;Oscillibacter valericigenes;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis;Oscillibacter sp. PEA192 & 216572;351091;1226322;1226323;2093857;2109687 & 30\\\\\n",
       "\t 1 & 05b02d57263741a486e49b34bde0fa28 & WP\\_187032256.1 & 50S ribosomal protein L27 {[}Dysosmobacter sp. NSJ-60{]}       &  86.022 & 93 & 13 & 1 & 279 & 1 & 93 & 1.68e-50 & 93 & 282 & Dysosmobacter sp. NSJ-60                                                                                                                            & 2763041                                       &  2\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t 38 & 5730334f42ee4013a044d98cfa4e837c & CDA81968.1     & dNA gyrase subunit B {[}Firmicutes bacterium CAG:176{]} & 81.985 & 655 & 107 & 1 & 1965 & 1 & 644 & 0 & 644 & 1968 & Firmicutes bacterium CAG:176      & 1263007 & 26\\\\\n",
       "\t 38 & 5730334f42ee4013a044d98cfa4e837c & WP\\_199648700.1 & None                                                & 82.290 & 655 & 107 & 1 & 1965 & 1 & 646 & 0 & 646 & 1968 & Pseudoflavonifractor sp. AF19-9AC & 2292244 & 22\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 780 × 17\n",
       "\n",
       "| cluster_id &lt;int&gt; | query &lt;fct&gt; | subject &lt;fct&gt; | subject_name &lt;fct&gt; | pident &lt;dbl&gt; | length &lt;int&gt; | mismatch &lt;int&gt; | qstart &lt;int&gt; | qend &lt;int&gt; | sstart &lt;int&gt; | send &lt;int&gt; | evalue &lt;dbl&gt; | slen &lt;int&gt; | qlen &lt;int&gt; | sscinames &lt;fct&gt; | staxids &lt;fct&gt; | pident_rank &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 05b02d57263741a486e49b34bde0fa28 | WP_025543765.1 | MULTISPECIES: 50S ribosomal protein L27 [Oscillospiraceae] | 100.000 | 93 |  0 | 1 | 279 | 1 | 93 | 1.43e-59 | 93 | 282 | Oscillospiraceae;Oscillibacter valericigenes;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis;Oscillibacter sp. PEA192 | 216572;351091;1226322;1226323;2093857;2109687 | 30 |\n",
       "| 1 | 05b02d57263741a486e49b34bde0fa28 | WP_187032256.1 | 50S ribosomal protein L27 [Dysosmobacter sp. NSJ-60]       |  86.022 | 93 | 13 | 1 | 279 | 1 | 93 | 1.68e-50 | 93 | 282 | Dysosmobacter sp. NSJ-60                                                                                                                            | 2763041                                       |  2 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| 38 | 5730334f42ee4013a044d98cfa4e837c | CDA81968.1     | dNA gyrase subunit B [Firmicutes bacterium CAG:176] | 81.985 | 655 | 107 | 1 | 1965 | 1 | 644 | 0 | 644 | 1968 | Firmicutes bacterium CAG:176      | 1263007 | 26 |\n",
       "| 38 | 5730334f42ee4013a044d98cfa4e837c | WP_199648700.1 | None                                                | 82.290 | 655 | 107 | 1 | 1965 | 1 | 646 | 0 | 646 | 1968 | Pseudoflavonifractor sp. AF19-9AC | 2292244 | 22 |\n",
       "\n"
      ],
      "text/plain": [
       "    cluster_id query                            subject       \n",
       "1   1          05b02d57263741a486e49b34bde0fa28 WP_025543765.1\n",
       "2   1          05b02d57263741a486e49b34bde0fa28 WP_187032256.1\n",
       "⋮   ⋮          ⋮                                ⋮             \n",
       "779 38         5730334f42ee4013a044d98cfa4e837c CDA81968.1    \n",
       "780 38         5730334f42ee4013a044d98cfa4e837c WP_199648700.1\n",
       "    subject_name                                               pident  length\n",
       "1   MULTISPECIES: 50S ribosomal protein L27 [Oscillospiraceae] 100.000 93    \n",
       "2   50S ribosomal protein L27 [Dysosmobacter sp. NSJ-60]        86.022 93    \n",
       "⋮   ⋮                                                          ⋮       ⋮     \n",
       "779 dNA gyrase subunit B [Firmicutes bacterium CAG:176]        81.985  655   \n",
       "780 None                                                       82.290  655   \n",
       "    mismatch qstart qend sstart send evalue   slen qlen\n",
       "1    0       1      279  1      93   1.43e-59 93   282 \n",
       "2   13       1      279  1      93   1.68e-50 93   282 \n",
       "⋮   ⋮        ⋮      ⋮    ⋮      ⋮    ⋮        ⋮    ⋮   \n",
       "779 107      1      1965 1      644  0        644  1968\n",
       "780 107      1      1965 1      646  0        646  1968\n",
       "    sscinames                                                                                                                                          \n",
       "1   Oscillospiraceae;Oscillibacter valericigenes;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis;Oscillibacter sp. PEA192\n",
       "2   Dysosmobacter sp. NSJ-60                                                                                                                           \n",
       "⋮   ⋮                                                                                                                                                  \n",
       "779 Firmicutes bacterium CAG:176                                                                                                                       \n",
       "780 Pseudoflavonifractor sp. AF19-9AC                                                                                                                  \n",
       "    staxids                                       pident_rank\n",
       "1   216572;351091;1226322;1226323;2093857;2109687 30         \n",
       "2   2763041                                        2         \n",
       "⋮   ⋮                                             ⋮          \n",
       "779 1263007                                       26         \n",
       "780 2292244                                       22         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# non-target cds gene annotations\n",
    "gene_nontarget = read.delim(file.path(work_dir, 'primers', 'cgp', 'nontarget', 'cds_blastx.tsv'), \n",
    "                        sep='\\t') %>%\n",
    "    semi_join(primer_info, c('cluster_id')) \n",
    "gene_nontarget"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Cluster taxonomic uniqueness"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-23T07:53:39.056935Z",
     "iopub.status.busy": "2021-04-23T07:53:39.054701Z",
     "iopub.status.idle": "2021-04-23T07:53:39.080196Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 1 × 8 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Min.</th><th scope=col>1st Qu.</th><th scope=col>Median</th><th scope=col>Mean</th><th scope=col>3rd Qu.</th><th scope=col>Max.</th><th scope=col>sd</th><th scope=col>sd_err_of_mean</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>seq-ID to closest non-target</th><td>88.413</td><td>92.683</td><td>95.868</td><td>95.28045</td><td>98.462</td><td>99.806</td><td>4.114</td><td>1.68</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 1 × 8 of type dbl\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & Min. & 1st Qu. & Median & Mean & 3rd Qu. & Max. & sd & sd\\_err\\_of\\_mean\\\\\n",
       "\\hline\n",
       "\tseq-ID to closest non-target & 88.413 & 92.683 & 95.868 & 95.28045 & 98.462 & 99.806 & 4.114 & 1.68\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 1 × 8 of type dbl\n",
       "\n",
       "| <!--/--> | Min. | 1st Qu. | Median | Mean | 3rd Qu. | Max. | sd | sd_err_of_mean |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| seq-ID to closest non-target | 88.413 | 92.683 | 95.868 | 95.28045 | 98.462 | 99.806 | 4.114 | 1.68 |\n",
       "\n"
      ],
      "text/plain": [
       "                             Min.   1st Qu. Median Mean     3rd Qu. Max.  \n",
       "seq-ID to closest non-target 88.413 92.683  95.868 95.28045 98.462  99.806\n",
       "                             sd    sd_err_of_mean\n",
       "seq-ID to closest non-target 4.114 1.68          "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# closest seqID\n",
    "gene_nontarget %>%\n",
    "    filter(pident_rank == 1) %>%\n",
    "    .$pident %>% summary_x('seq-ID to closest non-target')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Primer quality"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-23T07:49:08.482677Z",
     "iopub.status.busy": "2021-04-23T07:49:08.480388Z",
     "iopub.status.idle": "2021-04-23T07:49:08.502067Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No. of unique primer sets: 162 \n",
      "No. of unique clusters: 26 \n"
     ]
    }
   ],
   "source": [
    "# summary\n",
    "primer_info %>% unique_n('primer sets', primer_set)\n",
    "primer_info %>% distinct(gene_type, cluster_id) %>% nrow %>% cat('No. of unique clusters:', ., '\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-23T07:49:08.639805Z",
     "iopub.status.busy": "2021-04-23T07:49:08.637707Z",
     "iopub.status.idle": "2021-04-23T07:49:08.664101Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 1 × 8 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Min.</th><th scope=col>1st Qu.</th><th scope=col>Median</th><th scope=col>Mean</th><th scope=col>3rd Qu.</th><th scope=col>Max.</th><th scope=col>sd</th><th scope=col>sd_err_of_mean</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>primer sets per cluster</th><td>1</td><td>4.25</td><td>9</td><td>7.115385</td><td>10</td><td>10</td><td>3.62</td><td>1.478</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 1 × 8 of type dbl\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & Min. & 1st Qu. & Median & Mean & 3rd Qu. & Max. & sd & sd\\_err\\_of\\_mean\\\\\n",
       "\\hline\n",
       "\tprimer sets per cluster & 1 & 4.25 & 9 & 7.115385 & 10 & 10 & 3.62 & 1.478\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 1 × 8 of type dbl\n",
       "\n",
       "| <!--/--> | Min. | 1st Qu. | Median | Mean | 3rd Qu. | Max. | sd | sd_err_of_mean |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| primer sets per cluster | 1 | 4.25 | 9 | 7.115385 | 10 | 10 | 3.62 | 1.478 |\n",
       "\n"
      ],
      "text/plain": [
       "                        Min. 1st Qu. Median Mean     3rd Qu. Max. sd  \n",
       "primer sets per cluster 1    4.25    9      7.115385 10      10   3.62\n",
       "                        sd_err_of_mean\n",
       "primer sets per cluster 1.478         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# primers per cluster\n",
    "primer_info %>%\n",
    "    distinct(cluster_id, primer_set) %>%\n",
    "    group_by(cluster_id) %>%\n",
    "    summarize(n_primer_sets = n(), .groups='drop') %>%\n",
    "    .$n_primer_sets %>% summary_x('primer sets per cluster')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-23T07:49:08.913751Z",
     "iopub.status.busy": "2021-04-23T07:49:08.911611Z",
     "iopub.status.idle": "2021-04-23T07:49:08.956363Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 1 × 8 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Min.</th><th scope=col>1st Qu.</th><th scope=col>Median</th><th scope=col>Mean</th><th scope=col>3rd Qu.</th><th scope=col>Max.</th><th scope=col>sd</th><th scope=col>sd_err_of_mean</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>mean amplicon size</th><td>100</td><td>106</td><td>124</td><td>147.6475</td><td>194</td><td>250</td><td>58.267</td><td>23.788</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 1 × 8 of type dbl\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & Min. & 1st Qu. & Median & Mean & 3rd Qu. & Max. & sd & sd\\_err\\_of\\_mean\\\\\n",
       "\\hline\n",
       "\tmean amplicon size & 100 & 106 & 124 & 147.6475 & 194 & 250 & 58.267 & 23.788\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 1 × 8 of type dbl\n",
       "\n",
       "| <!--/--> | Min. | 1st Qu. | Median | Mean | 3rd Qu. | Max. | sd | sd_err_of_mean |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| mean amplicon size | 100 | 106 | 124 | 147.6475 | 194 | 250 | 58.267 | 23.788 |\n",
       "\n"
      ],
      "text/plain": [
       "                   Min. 1st Qu. Median Mean     3rd Qu. Max. sd    \n",
       "mean amplicon size 100  106     124    147.6475 194     250  58.267\n",
       "                   sd_err_of_mean\n",
       "mean amplicon size 23.788        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 1 × 8 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Min.</th><th scope=col>1st Qu.</th><th scope=col>Median</th><th scope=col>Mean</th><th scope=col>3rd Qu.</th><th scope=col>Max.</th><th scope=col>sd</th><th scope=col>sd_err_of_mean</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>degeneracy</th><td>1</td><td>16</td><td>24</td><td>33.53514</td><td>48</td><td>128</td><td>45.132</td><td>18.425</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 1 × 8 of type dbl\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & Min. & 1st Qu. & Median & Mean & 3rd Qu. & Max. & sd & sd\\_err\\_of\\_mean\\\\\n",
       "\\hline\n",
       "\tdegeneracy & 1 & 16 & 24 & 33.53514 & 48 & 128 & 45.132 & 18.425\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 1 × 8 of type dbl\n",
       "\n",
       "| <!--/--> | Min. | 1st Qu. | Median | Mean | 3rd Qu. | Max. | sd | sd_err_of_mean |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| degeneracy | 1 | 16 | 24 | 33.53514 | 48 | 128 | 45.132 | 18.425 |\n",
       "\n"
      ],
      "text/plain": [
       "           Min. 1st Qu. Median Mean     3rd Qu. Max. sd     sd_err_of_mean\n",
       "degeneracy 1    16      24     33.53514 48      128  45.132 18.425        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 1 × 8 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Min.</th><th scope=col>1st Qu.</th><th scope=col>Median</th><th scope=col>Mean</th><th scope=col>3rd Qu.</th><th scope=col>Max.</th><th scope=col>sd</th><th scope=col>sd_err_of_mean</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>degeneracy (3-prime)</th><td>1</td><td>1</td><td>2</td><td>2.108108</td><td>2</td><td>6</td><td>1.859</td><td>0.759</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 1 × 8 of type dbl\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & Min. & 1st Qu. & Median & Mean & 3rd Qu. & Max. & sd & sd\\_err\\_of\\_mean\\\\\n",
       "\\hline\n",
       "\tdegeneracy (3-prime) & 1 & 1 & 2 & 2.108108 & 2 & 6 & 1.859 & 0.759\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 1 × 8 of type dbl\n",
       "\n",
       "| <!--/--> | Min. | 1st Qu. | Median | Mean | 3rd Qu. | Max. | sd | sd_err_of_mean |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| degeneracy (3-prime) | 1 | 1 | 2 | 2.108108 | 2 | 6 | 1.859 | 0.759 |\n",
       "\n"
      ],
      "text/plain": [
       "                     Min. 1st Qu. Median Mean     3rd Qu. Max. sd   \n",
       "degeneracy (3-prime) 1    1       2      2.108108 2       6    1.859\n",
       "                     sd_err_of_mean\n",
       "degeneracy (3-prime) 0.759         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# primer quality\n",
    "primer_info %>% filter(primer_type == 'PRIMER_LEFT') %>% .$amplicon_size_avg %>% summary_x('mean amplicon size')\n",
    "primer_info %>% .$degeneracy %>% summary_x('degeneracy')\n",
    "primer_info %>% .$degeneracy_3prime %>% summary_x('degeneracy (3-prime)')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-23T07:56:25.062933Z",
     "iopub.status.busy": "2021-04-23T07:56:25.060912Z",
     "iopub.status.idle": "2021-04-23T07:56:25.119689Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 494 × 24</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>gene_type</th><th scope=col>cluster_id</th><th scope=col>primer_set</th><th scope=col>amplicon_size_consensus</th><th scope=col>amplicon_size_avg</th><th scope=col>amplicon_size_sd</th><th scope=col>primer_id</th><th scope=col>primer_type</th><th scope=col>sequence</th><th scope=col>length</th><th scope=col>⋯</th><th scope=col>Tm_avg</th><th scope=col>Tm_sd</th><th scope=col>GC_avg</th><th scope=col>GC_sd</th><th scope=col>hairpin_avg</th><th scope=col>hairpin_sd</th><th scope=col>homodimer_avg</th><th scope=col>homodimer_sd</th><th scope=col>pident</th><th scope=col>sscinames</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>cds</td><td>1</td><td>468</td><td>189</td><td>189</td><td>0</td><td>468f</td><td>PRIMER_LEFT </td><td>AARAACGGYCGYGAYTCYRA</td><td>20</td><td>⋯</td><td>60.76051</td><td>2.848347</td><td>50.00000</td><td>6.123724</td><td>16.30579</td><td>21.51507</td><td>22.824341</td><td>16.40584</td><td>91.011</td><td>Oscillospiraceae bacterium</td></tr>\n",
       "\t<tr><td>cds</td><td>1</td><td>468</td><td>189</td><td>189</td><td>0</td><td>468r</td><td>PRIMER_RIGHT</td><td>RCCSAGVYGCTCRAARCG  </td><td>18</td><td>⋯</td><td>62.16555</td><td>2.574813</td><td>64.81481</td><td>6.141898</td><td>35.16456</td><td>27.78734</td><td> 4.565427</td><td>23.59421</td><td>91.011</td><td>Oscillospiraceae bacterium</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋱</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>cds</td><td>38</td><td>911</td><td>204</td><td>204</td><td>0</td><td>911f</td><td>PRIMER_LEFT </td><td>GGHAAGATGCTSAAYGTSGA </td><td>20</td><td>⋯</td><td>58.39153</td><td>1.903507</td><td>49.16667</td><td>3.435921</td><td>21.547320</td><td>21.60791</td><td>-18.44452</td><td>20.099912</td><td>98.626</td><td>Ruminococcaceae bacterium TF06-43</td></tr>\n",
       "\t<tr><td>cds</td><td>38</td><td>911</td><td>204</td><td>204</td><td>0</td><td>911r</td><td>PRIMER_RIGHT</td><td>SCGGAARAARAAVGTCARMAG</td><td>21</td><td>⋯</td><td>57.91049</td><td>1.924208</td><td>46.03175</td><td>5.264484</td><td> 4.451632</td><td>11.86112</td><td>-24.35675</td><td> 9.648227</td><td>98.626</td><td>Ruminococcaceae bacterium TF06-43</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 494 × 24\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " gene\\_type & cluster\\_id & primer\\_set & amplicon\\_size\\_consensus & amplicon\\_size\\_avg & amplicon\\_size\\_sd & primer\\_id & primer\\_type & sequence & length & ⋯ & Tm\\_avg & Tm\\_sd & GC\\_avg & GC\\_sd & hairpin\\_avg & hairpin\\_sd & homodimer\\_avg & homodimer\\_sd & pident & sscinames\\\\\n",
       " <chr> & <int> & <int> & <int> & <dbl> & <dbl> & <fct> & <fct> & <fct> & <int> & ⋯ & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <fct>\\\\\n",
       "\\hline\n",
       "\t cds & 1 & 468 & 189 & 189 & 0 & 468f & PRIMER\\_LEFT  & AARAACGGYCGYGAYTCYRA & 20 & ⋯ & 60.76051 & 2.848347 & 50.00000 & 6.123724 & 16.30579 & 21.51507 & 22.824341 & 16.40584 & 91.011 & Oscillospiraceae bacterium\\\\\n",
       "\t cds & 1 & 468 & 189 & 189 & 0 & 468r & PRIMER\\_RIGHT & RCCSAGVYGCTCRAARCG   & 18 & ⋯ & 62.16555 & 2.574813 & 64.81481 & 6.141898 & 35.16456 & 27.78734 &  4.565427 & 23.59421 & 91.011 & Oscillospiraceae bacterium\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋱ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t cds & 38 & 911 & 204 & 204 & 0 & 911f & PRIMER\\_LEFT  & GGHAAGATGCTSAAYGTSGA  & 20 & ⋯ & 58.39153 & 1.903507 & 49.16667 & 3.435921 & 21.547320 & 21.60791 & -18.44452 & 20.099912 & 98.626 & Ruminococcaceae bacterium TF06-43\\\\\n",
       "\t cds & 38 & 911 & 204 & 204 & 0 & 911r & PRIMER\\_RIGHT & SCGGAARAARAAVGTCARMAG & 21 & ⋯ & 57.91049 & 1.924208 & 46.03175 & 5.264484 &  4.451632 & 11.86112 & -24.35675 &  9.648227 & 98.626 & Ruminococcaceae bacterium TF06-43\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 494 × 24\n",
       "\n",
       "| gene_type &lt;chr&gt; | cluster_id &lt;int&gt; | primer_set &lt;int&gt; | amplicon_size_consensus &lt;int&gt; | amplicon_size_avg &lt;dbl&gt; | amplicon_size_sd &lt;dbl&gt; | primer_id &lt;fct&gt; | primer_type &lt;fct&gt; | sequence &lt;fct&gt; | length &lt;int&gt; | ⋯ ⋯ | Tm_avg &lt;dbl&gt; | Tm_sd &lt;dbl&gt; | GC_avg &lt;dbl&gt; | GC_sd &lt;dbl&gt; | hairpin_avg &lt;dbl&gt; | hairpin_sd &lt;dbl&gt; | homodimer_avg &lt;dbl&gt; | homodimer_sd &lt;dbl&gt; | pident &lt;dbl&gt; | sscinames &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| cds | 1 | 468 | 189 | 189 | 0 | 468f | PRIMER_LEFT  | AARAACGGYCGYGAYTCYRA | 20 | ⋯ | 60.76051 | 2.848347 | 50.00000 | 6.123724 | 16.30579 | 21.51507 | 22.824341 | 16.40584 | 91.011 | Oscillospiraceae bacterium |\n",
       "| cds | 1 | 468 | 189 | 189 | 0 | 468r | PRIMER_RIGHT | RCCSAGVYGCTCRAARCG   | 18 | ⋯ | 62.16555 | 2.574813 | 64.81481 | 6.141898 | 35.16456 | 27.78734 |  4.565427 | 23.59421 | 91.011 | Oscillospiraceae bacterium |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋱ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| cds | 38 | 911 | 204 | 204 | 0 | 911f | PRIMER_LEFT  | GGHAAGATGCTSAAYGTSGA  | 20 | ⋯ | 58.39153 | 1.903507 | 49.16667 | 3.435921 | 21.547320 | 21.60791 | -18.44452 | 20.099912 | 98.626 | Ruminococcaceae bacterium TF06-43 |\n",
       "| cds | 38 | 911 | 204 | 204 | 0 | 911r | PRIMER_RIGHT | SCGGAARAARAAVGTCARMAG | 21 | ⋯ | 57.91049 | 1.924208 | 46.03175 | 5.264484 |  4.451632 | 11.86112 | -24.35675 |  9.648227 | 98.626 | Ruminococcaceae bacterium TF06-43 |\n",
       "\n"
      ],
      "text/plain": [
       "    gene_type cluster_id primer_set amplicon_size_consensus amplicon_size_avg\n",
       "1   cds       1          468        189                     189              \n",
       "2   cds       1          468        189                     189              \n",
       "⋮   ⋮         ⋮          ⋮          ⋮                       ⋮                \n",
       "493 cds       38         911        204                     204              \n",
       "494 cds       38         911        204                     204              \n",
       "    amplicon_size_sd primer_id primer_type  sequence              length ⋯\n",
       "1   0                468f      PRIMER_LEFT  AARAACGGYCGYGAYTCYRA  20     ⋯\n",
       "2   0                468r      PRIMER_RIGHT RCCSAGVYGCTCRAARCG    18     ⋯\n",
       "⋮   ⋮                ⋮         ⋮            ⋮                     ⋮      ⋱\n",
       "493 0                911f      PRIMER_LEFT  GGHAAGATGCTSAAYGTSGA  20     ⋯\n",
       "494 0                911r      PRIMER_RIGHT SCGGAARAARAAVGTCARMAG 21     ⋯\n",
       "    Tm_avg   Tm_sd    GC_avg   GC_sd    hairpin_avg hairpin_sd homodimer_avg\n",
       "1   60.76051 2.848347 50.00000 6.123724 16.30579    21.51507   22.824341    \n",
       "2   62.16555 2.574813 64.81481 6.141898 35.16456    27.78734    4.565427    \n",
       "⋮   ⋮        ⋮        ⋮        ⋮        ⋮           ⋮          ⋮            \n",
       "493 58.39153 1.903507 49.16667 3.435921 21.547320   21.60791   -18.44452    \n",
       "494 57.91049 1.924208 46.03175 5.264484  4.451632   11.86112   -24.35675    \n",
       "    homodimer_sd pident sscinames                        \n",
       "1   16.40584     91.011 Oscillospiraceae bacterium       \n",
       "2   23.59421     91.011 Oscillospiraceae bacterium       \n",
       "⋮   ⋮            ⋮      ⋮                                \n",
       "493 20.099912    98.626 Ruminococcaceae bacterium TF06-43\n",
       "494  9.648227    98.626 Ruminococcaceae bacterium TF06-43"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# adding non-target pident info\n",
    "tmp1 = gene_nontarget %>%\n",
    "    filter(pident_rank == 1) %>%\n",
    "    distinct(cluster_id, pident, sscinames) %>%\n",
    "    mutate(gene_type = 'cds')\n",
    "\n",
    "primer_info_j = primer_info %>%\n",
    "    inner_join(tmp1, c('gene_type', 'cluster_id'))\n",
    "primer_info_j"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-23T07:58:47.546864Z",
     "iopub.status.busy": "2021-04-23T07:58:47.544707Z",
     "iopub.status.idle": "2021-04-23T07:58:47.658034Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A tibble: 30 × 26</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>gene_type</th><th scope=col>cluster_id</th><th scope=col>primer_set</th><th scope=col>amplicon_size_consensus</th><th scope=col>amplicon_size_avg</th><th scope=col>amplicon_size_sd</th><th scope=col>primer_id</th><th scope=col>primer_type</th><th scope=col>sequence</th><th scope=col>length</th><th scope=col>degeneracy</th><th scope=col>degeneracy_3prime</th><th scope=col>position_start</th><th scope=col>position_end</th><th scope=col>Tm_avg</th><th scope=col>Tm_sd</th><th scope=col>GC_avg</th><th scope=col>GC_sd</th><th scope=col>hairpin_avg</th><th scope=col>hairpin_sd</th><th scope=col>homodimer_avg</th><th scope=col>homodimer_sd</th><th scope=col>pident</th><th scope=col>sscinames</th><th scope=col>max_degeneracy_3prime</th><th scope=col>max_degeneracy</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>cds</td><td>24</td><td>101</td><td>160</td><td>160</td><td>0</td><td>101f</td><td>PRIMER_LEFT </td><td>TBTTCCGBGAYAAYTCCACS </td><td>20</td><td> 72</td><td>2</td><td>196</td><td>216</td><td>59.40978</td><td>2.2201857</td><td>51.66667</td><td>4.859127</td><td>23.170159</td><td>23.39333</td><td>  3.8426830</td><td>12.757158</td><td>88.413</td><td>Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         </td><td>2</td><td> 72</td></tr>\n",
       "\t<tr><td>cds</td><td>24</td><td>101</td><td>160</td><td>160</td><td>0</td><td>101r</td><td>PRIMER_RIGHT</td><td>ATSACRTCVGCCATRAASGA </td><td>20</td><td> 48</td><td>2</td><td>336</td><td>356</td><td>59.29837</td><td>2.1329283</td><td>48.33333</td><td>4.249183</td><td>37.237596</td><td>11.36689</td><td> -0.2072119</td><td>21.774421</td><td>88.413</td><td>Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         </td><td>2</td><td> 72</td></tr>\n",
       "\t<tr><td>cds</td><td>24</td><td>925</td><td>159</td><td>159</td><td>0</td><td>925f</td><td>PRIMER_LEFT </td><td>TBTTCCGBGAYAAYTCCACS </td><td>20</td><td> 72</td><td>2</td><td>196</td><td>216</td><td>59.40978</td><td>2.2201857</td><td>51.66667</td><td>4.859127</td><td>23.170159</td><td>23.39333</td><td>  3.8426830</td><td>12.757158</td><td>88.413</td><td>Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         </td><td>2</td><td> 72</td></tr>\n",
       "\t<tr><td>cds</td><td>24</td><td>925</td><td>159</td><td>159</td><td>0</td><td>925r</td><td>PRIMER_RIGHT</td><td>TSACRTCVGCCATRAASGA  </td><td>19</td><td> 48</td><td>2</td><td>336</td><td>355</td><td>58.81608</td><td>2.2386673</td><td>50.87719</td><td>4.472824</td><td>37.237596</td><td>11.36689</td><td> -0.9210938</td><td>21.070817</td><td>88.413</td><td>Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         </td><td>2</td><td> 72</td></tr>\n",
       "\t<tr><td>cds</td><td>24</td><td>252</td><td>205</td><td>205</td><td>0</td><td>252f</td><td>PRIMER_LEFT </td><td>ACSGCSAACATGGTBTCSTT </td><td>20</td><td> 24</td><td>2</td><td>321</td><td>341</td><td>62.52437</td><td>0.8325446</td><td>53.33333</td><td>2.357023</td><td>34.079778</td><td>15.71851</td><td>  8.6676870</td><td> 8.532017</td><td>88.413</td><td>Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         </td><td>2</td><td> 96</td></tr>\n",
       "\t<tr><td>cds</td><td>24</td><td>252</td><td>205</td><td>205</td><td>0</td><td>252r</td><td>PRIMER_RIGHT</td><td>YRTGSAGCATRTCVGCCATR </td><td>20</td><td> 96</td><td>2</td><td>506</td><td>526</td><td>60.37872</td><td>2.5635953</td><td>53.33333</td><td>5.527708</td><td>39.814057</td><td>14.18557</td><td>  3.8855806</td><td>13.889514</td><td>88.413</td><td>Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         </td><td>2</td><td> 96</td></tr>\n",
       "\t<tr><td>cds</td><td>24</td><td>852</td><td>135</td><td>135</td><td>0</td><td>852f</td><td>PRIMER_LEFT </td><td>GBTTCTCVTTYGCCTCYGC  </td><td>19</td><td> 36</td><td>2</td><td>223</td><td>242</td><td>60.45917</td><td>2.1671838</td><td>59.64912</td><td>5.114870</td><td> 9.565666</td><td>16.56822</td><td>-52.4259519</td><td> 8.602116</td><td>88.413</td><td>Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         </td><td>4</td><td> 48</td></tr>\n",
       "\t<tr><td>cds</td><td>24</td><td>852</td><td>135</td><td>135</td><td>0</td><td>852r</td><td>PRIMER_RIGHT</td><td>CGATSACRTCVGCCATRAAS </td><td>20</td><td> 48</td><td>4</td><td>338</td><td>358</td><td>59.09106</td><td>2.0362332</td><td>53.33333</td><td>4.249183</td><td>34.242766</td><td>16.18458</td><td>  5.7775265</td><td>14.765420</td><td>88.413</td><td>Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         </td><td>4</td><td> 48</td></tr>\n",
       "\t<tr><td>cds</td><td>24</td><td>962</td><td>133</td><td>133</td><td>0</td><td>962f</td><td>PRIMER_LEFT </td><td>TTCTCVTTYGCCTCYGCST  </td><td>19</td><td> 24</td><td>4</td><td>225</td><td>244</td><td>61.47875</td><td>2.0007686</td><td>56.14035</td><td>4.472824</td><td>29.250102</td><td>23.92131</td><td>-27.4141990</td><td>26.541144</td><td>88.413</td><td>Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         </td><td>4</td><td> 48</td></tr>\n",
       "\t<tr><td>cds</td><td>24</td><td>962</td><td>133</td><td>133</td><td>0</td><td>962r</td><td>PRIMER_RIGHT</td><td>CGATSACRTCVGCCATRAAS </td><td>20</td><td> 48</td><td>4</td><td>338</td><td>358</td><td>59.09106</td><td>2.0362332</td><td>53.33333</td><td>4.249183</td><td>34.242766</td><td>16.18458</td><td>  5.7775265</td><td>14.765420</td><td>88.413</td><td>Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         </td><td>4</td><td> 48</td></tr>\n",
       "\t<tr><td>cds</td><td>24</td><td>997</td><td>149</td><td>149</td><td>0</td><td>997f</td><td>PRIMER_LEFT </td><td>CSTTYATGGCBGAYGTSATC </td><td>20</td><td> 48</td><td>2</td><td>337</td><td>357</td><td>58.90509</td><td>2.0743193</td><td>53.33333</td><td>4.249183</td><td>29.056470</td><td>16.26418</td><td>  1.3684155</td><td>16.456230</td><td>88.413</td><td>Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         </td><td>4</td><td> 64</td></tr>\n",
       "\t<tr><td>cds</td><td>24</td><td>997</td><td>149</td><td>149</td><td>0</td><td>997r</td><td>PRIMER_RIGHT</td><td>RTCRCACTGSAGRTTYACMA </td><td>20</td><td> 64</td><td>4</td><td>466</td><td>486</td><td>58.27347</td><td>2.5143072</td><td>47.50000</td><td>5.590170</td><td>15.209464</td><td>18.47412</td><td> -4.6224649</td><td>33.375152</td><td>88.413</td><td>Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         </td><td>4</td><td> 64</td></tr>\n",
       "\t<tr><td>cds</td><td>24</td><td> 34</td><td>162</td><td>162</td><td>0</td><td>34f </td><td>PRIMER_LEFT </td><td>TBTTCCGBGAYAAYTCCACS </td><td>20</td><td> 72</td><td>2</td><td>196</td><td>216</td><td>59.40978</td><td>2.2201857</td><td>51.66667</td><td>4.859127</td><td>23.170159</td><td>23.39333</td><td>  3.8426830</td><td>12.757158</td><td>88.413</td><td>Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         </td><td>4</td><td> 72</td></tr>\n",
       "\t<tr><td>cds</td><td>24</td><td> 34</td><td>162</td><td>162</td><td>0</td><td>34r </td><td>PRIMER_RIGHT</td><td>CGATSACRTCVGCCATRAAS </td><td>20</td><td> 48</td><td>4</td><td>338</td><td>358</td><td>59.09106</td><td>2.0362332</td><td>53.33333</td><td>4.249183</td><td>34.242766</td><td>16.18458</td><td>  5.7775265</td><td>14.765420</td><td>88.413</td><td>Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         </td><td>4</td><td> 72</td></tr>\n",
       "\t<tr><td>cds</td><td>24</td><td>949</td><td>137</td><td>137</td><td>0</td><td>949f</td><td>PRIMER_LEFT </td><td>CBTGGGAYAAGASYGAYGA  </td><td>19</td><td> 48</td><td>2</td><td> 79</td><td> 98</td><td>57.43879</td><td>2.7226357</td><td>53.50877</td><td>5.189544</td><td>23.628755</td><td>22.96581</td><td>-25.9893377</td><td>26.595037</td><td>88.413</td><td>Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         </td><td>4</td><td> 72</td></tr>\n",
       "\t<tr><td>cds</td><td>24</td><td>949</td><td>137</td><td>137</td><td>0</td><td>949r</td><td>PRIMER_RIGHT</td><td>SGTGGARTTRTCVCGGAAVA </td><td>20</td><td> 72</td><td>3</td><td>196</td><td>216</td><td>59.40978</td><td>2.2201857</td><td>51.66667</td><td>4.859127</td><td>32.446352</td><td>19.32925</td><td>  2.2146646</td><td>13.318311</td><td>88.413</td><td>Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         </td><td>4</td><td> 72</td></tr>\n",
       "\t<tr><td>cds</td><td>36</td><td>952</td><td>114</td><td>114</td><td>0</td><td>952f</td><td>PRIMER_LEFT </td><td>YGARATYCAYCGSCTGAACC </td><td>20</td><td> 32</td><td>1</td><td>356</td><td>376</td><td>60.32997</td><td>1.6997727</td><td>55.00000</td><td>5.000000</td><td>27.075589</td><td>21.15220</td><td>-11.3804235</td><td>24.602761</td><td>89.118</td><td>Clostridiales bacterium                                                                                                                       </td><td>3</td><td> 32</td></tr>\n",
       "\t<tr><td>cds</td><td>36</td><td>952</td><td>114</td><td>114</td><td>0</td><td>952r</td><td>PRIMER_RIGHT</td><td>TCSAGVCGGATGGAGTTBG  </td><td>19</td><td> 18</td><td>3</td><td>451</td><td>470</td><td>60.74049</td><td>1.6644375</td><td>59.64912</td><td>3.508772</td><td>28.880424</td><td>20.58490</td><td> -4.3656091</td><td>26.977200</td><td>89.118</td><td>Clostridiales bacterium                                                                                                                       </td><td>3</td><td> 32</td></tr>\n",
       "\t<tr><td>cds</td><td>36</td><td>627</td><td>113</td><td>113</td><td>0</td><td>627f</td><td>PRIMER_LEFT </td><td>GAYYTKGCKGCRCTKYTGA  </td><td>19</td><td>128</td><td>4</td><td>303</td><td>322</td><td>60.57837</td><td>2.9830748</td><td>55.26316</td><td>6.962503</td><td> 4.689779</td><td>12.40799</td><td> 15.8095996</td><td>13.575059</td><td>89.118</td><td>Clostridiales bacterium                                                                                                                       </td><td>6</td><td>128</td></tr>\n",
       "\t<tr><td>cds</td><td>36</td><td>627</td><td>113</td><td>113</td><td>0</td><td>627r</td><td>PRIMER_RIGHT</td><td>WARTCCTCCATVGCVGGRTA </td><td>20</td><td> 72</td><td>2</td><td>396</td><td>416</td><td>59.04885</td><td>2.4997399</td><td>51.66667</td><td>4.859127</td><td>28.991987</td><td>16.18955</td><td>  3.0088796</td><td>20.436287</td><td>89.118</td><td>Clostridiales bacterium                                                                                                                       </td><td>6</td><td>128</td></tr>\n",
       "\t<tr><td>cds</td><td> 1</td><td>468</td><td>189</td><td>189</td><td>0</td><td>468f</td><td>PRIMER_LEFT </td><td>AARAACGGYCGYGAYTCYRA </td><td>20</td><td> 64</td><td>4</td><td> 54</td><td> 74</td><td>60.76051</td><td>2.8483468</td><td>50.00000</td><td>6.123724</td><td>16.305789</td><td>21.51507</td><td> 22.8243414</td><td>16.405838</td><td>91.011</td><td>Oscillospiraceae bacterium                                                                                                                    </td><td>4</td><td> 96</td></tr>\n",
       "\t<tr><td>cds</td><td> 1</td><td>468</td><td>189</td><td>189</td><td>0</td><td>468r</td><td>PRIMER_RIGHT</td><td>RCCSAGVYGCTCRAARCG   </td><td>18</td><td> 96</td><td>2</td><td>225</td><td>243</td><td>62.16555</td><td>2.5748132</td><td>64.81481</td><td>6.141898</td><td>35.164559</td><td>27.78734</td><td>  4.5654274</td><td>23.594207</td><td>91.011</td><td>Oscillospiraceae bacterium                                                                                                                    </td><td>4</td><td> 96</td></tr>\n",
       "\t<tr><td>cds</td><td> 1</td><td>927</td><td>188</td><td>188</td><td>0</td><td>927f</td><td>PRIMER_LEFT </td><td>ARAACGGYCGYGAYTCYRA  </td><td>19</td><td> 64</td><td>4</td><td> 55</td><td> 74</td><td>60.13246</td><td>2.9968465</td><td>52.63158</td><td>6.446026</td><td>16.118781</td><td>21.37324</td><td> 22.7879011</td><td>16.410793</td><td>91.011</td><td>Oscillospiraceae bacterium                                                                                                                    </td><td>4</td><td> 96</td></tr>\n",
       "\t<tr><td>cds</td><td> 1</td><td>927</td><td>188</td><td>188</td><td>0</td><td>927r</td><td>PRIMER_RIGHT</td><td>RCCSAGVYGCTCRAARCG   </td><td>18</td><td> 96</td><td>2</td><td>225</td><td>243</td><td>62.16555</td><td>2.5748132</td><td>64.81481</td><td>6.141898</td><td>35.164559</td><td>27.78734</td><td>  4.5654274</td><td>23.594207</td><td>91.011</td><td>Oscillospiraceae bacterium                                                                                                                    </td><td>4</td><td> 96</td></tr>\n",
       "\t<tr><td>cds</td><td> 8</td><td>780</td><td>126</td><td>126</td><td>0</td><td>780f</td><td>PRIMER_LEFT </td><td>AARCCVCGVTAYCTBATGGG </td><td>20</td><td>108</td><td>1</td><td>738</td><td>758</td><td>60.35110</td><td>2.6283001</td><td>55.00000</td><td>5.400617</td><td>34.293877</td><td>14.01021</td><td>  2.3512552</td><td>19.402971</td><td>91.247</td><td>Dysosmobacter welbionis                                                                                                                       </td><td>6</td><td>108</td></tr>\n",
       "\t<tr><td>cds</td><td> 8</td><td>780</td><td>126</td><td>126</td><td>0</td><td>780r</td><td>PRIMER_RIGHT</td><td>CCAVGTRAABAGCTTBCCRT </td><td>20</td><td>108</td><td>6</td><td>844</td><td>864</td><td>58.71495</td><td>2.6684202</td><td>50.00000</td><td>5.400617</td><td>18.377478</td><td>19.37307</td><td>-16.3267600</td><td>21.842517</td><td>91.247</td><td>Dysosmobacter welbionis                                                                                                                       </td><td>6</td><td>108</td></tr>\n",
       "\t<tr><td>cds</td><td>13</td><td>265</td><td>216</td><td>216</td><td>0</td><td>265f</td><td>PRIMER_LEFT </td><td>CRHTGATGRCGCGYAARAGA </td><td>20</td><td> 48</td><td>2</td><td> 19</td><td> 39</td><td>60.46714</td><td>2.6839022</td><td>51.66667</td><td>5.527708</td><td>27.836024</td><td>21.98608</td><td> 15.9462087</td><td>11.271684</td><td>91.379</td><td>Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis;Oscillibacter sp. PEA192</td><td>2</td><td> 48</td></tr>\n",
       "\t<tr><td>cds</td><td>13</td><td>265</td><td>216</td><td>216</td><td>0</td><td>265r</td><td>PRIMER_RIGHT</td><td>AVSTGGARTAGTTCATSCCGT</td><td>21</td><td> 24</td><td>2</td><td>214</td><td>235</td><td>60.08197</td><td>1.3894420</td><td>48.41270</td><td>3.272306</td><td>30.039833</td><td>16.01142</td><td>-27.4416464</td><td>14.647714</td><td>91.379</td><td>Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis;Oscillibacter sp. PEA192</td><td>2</td><td> 48</td></tr>\n",
       "\t<tr><td>cds</td><td>13</td><td>295</td><td>125</td><td>125</td><td>0</td><td>295f</td><td>PRIMER_LEFT </td><td>RSMVGTSATGAAGTCCCTGA </td><td>20</td><td> 48</td><td>1</td><td>110</td><td>130</td><td>59.84114</td><td>1.8496320</td><td>53.33333</td><td>4.249183</td><td>11.670800</td><td>18.48426</td><td> -7.5781759</td><td>28.282381</td><td>91.379</td><td>Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis;Oscillibacter sp. PEA192</td><td>2</td><td> 48</td></tr>\n",
       "\t<tr><td>cds</td><td>13</td><td>295</td><td>125</td><td>125</td><td>0</td><td>295r</td><td>PRIMER_RIGHT</td><td>AVSTGGARTAGTTCATSCCGT</td><td>21</td><td> 24</td><td>2</td><td>214</td><td>235</td><td>60.08197</td><td>1.3894420</td><td>48.41270</td><td>3.272306</td><td>30.039833</td><td>16.01142</td><td>-27.4416464</td><td>14.647714</td><td>91.379</td><td>Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis;Oscillibacter sp. PEA192</td><td>2</td><td> 48</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 30 × 26\n",
       "\\begin{tabular}{llllllllllllllllllllllllll}\n",
       " gene\\_type & cluster\\_id & primer\\_set & amplicon\\_size\\_consensus & amplicon\\_size\\_avg & amplicon\\_size\\_sd & primer\\_id & primer\\_type & sequence & length & degeneracy & degeneracy\\_3prime & position\\_start & position\\_end & Tm\\_avg & Tm\\_sd & GC\\_avg & GC\\_sd & hairpin\\_avg & hairpin\\_sd & homodimer\\_avg & homodimer\\_sd & pident & sscinames & max\\_degeneracy\\_3prime & max\\_degeneracy\\\\\n",
       " <chr> & <int> & <int> & <int> & <dbl> & <dbl> & <fct> & <fct> & <fct> & <int> & <int> & <int> & <int> & <int> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <fct> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t cds & 24 & 101 & 160 & 160 & 0 & 101f & PRIMER\\_LEFT  & TBTTCCGBGAYAAYTCCACS  & 20 &  72 & 2 & 196 & 216 & 59.40978 & 2.2201857 & 51.66667 & 4.859127 & 23.170159 & 23.39333 &   3.8426830 & 12.757158 & 88.413 & Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          & 2 &  72\\\\\n",
       "\t cds & 24 & 101 & 160 & 160 & 0 & 101r & PRIMER\\_RIGHT & ATSACRTCVGCCATRAASGA  & 20 &  48 & 2 & 336 & 356 & 59.29837 & 2.1329283 & 48.33333 & 4.249183 & 37.237596 & 11.36689 &  -0.2072119 & 21.774421 & 88.413 & Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          & 2 &  72\\\\\n",
       "\t cds & 24 & 925 & 159 & 159 & 0 & 925f & PRIMER\\_LEFT  & TBTTCCGBGAYAAYTCCACS  & 20 &  72 & 2 & 196 & 216 & 59.40978 & 2.2201857 & 51.66667 & 4.859127 & 23.170159 & 23.39333 &   3.8426830 & 12.757158 & 88.413 & Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          & 2 &  72\\\\\n",
       "\t cds & 24 & 925 & 159 & 159 & 0 & 925r & PRIMER\\_RIGHT & TSACRTCVGCCATRAASGA   & 19 &  48 & 2 & 336 & 355 & 58.81608 & 2.2386673 & 50.87719 & 4.472824 & 37.237596 & 11.36689 &  -0.9210938 & 21.070817 & 88.413 & Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          & 2 &  72\\\\\n",
       "\t cds & 24 & 252 & 205 & 205 & 0 & 252f & PRIMER\\_LEFT  & ACSGCSAACATGGTBTCSTT  & 20 &  24 & 2 & 321 & 341 & 62.52437 & 0.8325446 & 53.33333 & 2.357023 & 34.079778 & 15.71851 &   8.6676870 &  8.532017 & 88.413 & Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          & 2 &  96\\\\\n",
       "\t cds & 24 & 252 & 205 & 205 & 0 & 252r & PRIMER\\_RIGHT & YRTGSAGCATRTCVGCCATR  & 20 &  96 & 2 & 506 & 526 & 60.37872 & 2.5635953 & 53.33333 & 5.527708 & 39.814057 & 14.18557 &   3.8855806 & 13.889514 & 88.413 & Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          & 2 &  96\\\\\n",
       "\t cds & 24 & 852 & 135 & 135 & 0 & 852f & PRIMER\\_LEFT  & GBTTCTCVTTYGCCTCYGC   & 19 &  36 & 2 & 223 & 242 & 60.45917 & 2.1671838 & 59.64912 & 5.114870 &  9.565666 & 16.56822 & -52.4259519 &  8.602116 & 88.413 & Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          & 4 &  48\\\\\n",
       "\t cds & 24 & 852 & 135 & 135 & 0 & 852r & PRIMER\\_RIGHT & CGATSACRTCVGCCATRAAS  & 20 &  48 & 4 & 338 & 358 & 59.09106 & 2.0362332 & 53.33333 & 4.249183 & 34.242766 & 16.18458 &   5.7775265 & 14.765420 & 88.413 & Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          & 4 &  48\\\\\n",
       "\t cds & 24 & 962 & 133 & 133 & 0 & 962f & PRIMER\\_LEFT  & TTCTCVTTYGCCTCYGCST   & 19 &  24 & 4 & 225 & 244 & 61.47875 & 2.0007686 & 56.14035 & 4.472824 & 29.250102 & 23.92131 & -27.4141990 & 26.541144 & 88.413 & Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          & 4 &  48\\\\\n",
       "\t cds & 24 & 962 & 133 & 133 & 0 & 962r & PRIMER\\_RIGHT & CGATSACRTCVGCCATRAAS  & 20 &  48 & 4 & 338 & 358 & 59.09106 & 2.0362332 & 53.33333 & 4.249183 & 34.242766 & 16.18458 &   5.7775265 & 14.765420 & 88.413 & Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          & 4 &  48\\\\\n",
       "\t cds & 24 & 997 & 149 & 149 & 0 & 997f & PRIMER\\_LEFT  & CSTTYATGGCBGAYGTSATC  & 20 &  48 & 2 & 337 & 357 & 58.90509 & 2.0743193 & 53.33333 & 4.249183 & 29.056470 & 16.26418 &   1.3684155 & 16.456230 & 88.413 & Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          & 4 &  64\\\\\n",
       "\t cds & 24 & 997 & 149 & 149 & 0 & 997r & PRIMER\\_RIGHT & RTCRCACTGSAGRTTYACMA  & 20 &  64 & 4 & 466 & 486 & 58.27347 & 2.5143072 & 47.50000 & 5.590170 & 15.209464 & 18.47412 &  -4.6224649 & 33.375152 & 88.413 & Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          & 4 &  64\\\\\n",
       "\t cds & 24 &  34 & 162 & 162 & 0 & 34f  & PRIMER\\_LEFT  & TBTTCCGBGAYAAYTCCACS  & 20 &  72 & 2 & 196 & 216 & 59.40978 & 2.2201857 & 51.66667 & 4.859127 & 23.170159 & 23.39333 &   3.8426830 & 12.757158 & 88.413 & Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          & 4 &  72\\\\\n",
       "\t cds & 24 &  34 & 162 & 162 & 0 & 34r  & PRIMER\\_RIGHT & CGATSACRTCVGCCATRAAS  & 20 &  48 & 4 & 338 & 358 & 59.09106 & 2.0362332 & 53.33333 & 4.249183 & 34.242766 & 16.18458 &   5.7775265 & 14.765420 & 88.413 & Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          & 4 &  72\\\\\n",
       "\t cds & 24 & 949 & 137 & 137 & 0 & 949f & PRIMER\\_LEFT  & CBTGGGAYAAGASYGAYGA   & 19 &  48 & 2 &  79 &  98 & 57.43879 & 2.7226357 & 53.50877 & 5.189544 & 23.628755 & 22.96581 & -25.9893377 & 26.595037 & 88.413 & Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          & 4 &  72\\\\\n",
       "\t cds & 24 & 949 & 137 & 137 & 0 & 949r & PRIMER\\_RIGHT & SGTGGARTTRTCVCGGAAVA  & 20 &  72 & 3 & 196 & 216 & 59.40978 & 2.2201857 & 51.66667 & 4.859127 & 32.446352 & 19.32925 &   2.2146646 & 13.318311 & 88.413 & Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          & 4 &  72\\\\\n",
       "\t cds & 36 & 952 & 114 & 114 & 0 & 952f & PRIMER\\_LEFT  & YGARATYCAYCGSCTGAACC  & 20 &  32 & 1 & 356 & 376 & 60.32997 & 1.6997727 & 55.00000 & 5.000000 & 27.075589 & 21.15220 & -11.3804235 & 24.602761 & 89.118 & Clostridiales bacterium                                                                                                                        & 3 &  32\\\\\n",
       "\t cds & 36 & 952 & 114 & 114 & 0 & 952r & PRIMER\\_RIGHT & TCSAGVCGGATGGAGTTBG   & 19 &  18 & 3 & 451 & 470 & 60.74049 & 1.6644375 & 59.64912 & 3.508772 & 28.880424 & 20.58490 &  -4.3656091 & 26.977200 & 89.118 & Clostridiales bacterium                                                                                                                        & 3 &  32\\\\\n",
       "\t cds & 36 & 627 & 113 & 113 & 0 & 627f & PRIMER\\_LEFT  & GAYYTKGCKGCRCTKYTGA   & 19 & 128 & 4 & 303 & 322 & 60.57837 & 2.9830748 & 55.26316 & 6.962503 &  4.689779 & 12.40799 &  15.8095996 & 13.575059 & 89.118 & Clostridiales bacterium                                                                                                                        & 6 & 128\\\\\n",
       "\t cds & 36 & 627 & 113 & 113 & 0 & 627r & PRIMER\\_RIGHT & WARTCCTCCATVGCVGGRTA  & 20 &  72 & 2 & 396 & 416 & 59.04885 & 2.4997399 & 51.66667 & 4.859127 & 28.991987 & 16.18955 &   3.0088796 & 20.436287 & 89.118 & Clostridiales bacterium                                                                                                                        & 6 & 128\\\\\n",
       "\t cds &  1 & 468 & 189 & 189 & 0 & 468f & PRIMER\\_LEFT  & AARAACGGYCGYGAYTCYRA  & 20 &  64 & 4 &  54 &  74 & 60.76051 & 2.8483468 & 50.00000 & 6.123724 & 16.305789 & 21.51507 &  22.8243414 & 16.405838 & 91.011 & Oscillospiraceae bacterium                                                                                                                     & 4 &  96\\\\\n",
       "\t cds &  1 & 468 & 189 & 189 & 0 & 468r & PRIMER\\_RIGHT & RCCSAGVYGCTCRAARCG    & 18 &  96 & 2 & 225 & 243 & 62.16555 & 2.5748132 & 64.81481 & 6.141898 & 35.164559 & 27.78734 &   4.5654274 & 23.594207 & 91.011 & Oscillospiraceae bacterium                                                                                                                     & 4 &  96\\\\\n",
       "\t cds &  1 & 927 & 188 & 188 & 0 & 927f & PRIMER\\_LEFT  & ARAACGGYCGYGAYTCYRA   & 19 &  64 & 4 &  55 &  74 & 60.13246 & 2.9968465 & 52.63158 & 6.446026 & 16.118781 & 21.37324 &  22.7879011 & 16.410793 & 91.011 & Oscillospiraceae bacterium                                                                                                                     & 4 &  96\\\\\n",
       "\t cds &  1 & 927 & 188 & 188 & 0 & 927r & PRIMER\\_RIGHT & RCCSAGVYGCTCRAARCG    & 18 &  96 & 2 & 225 & 243 & 62.16555 & 2.5748132 & 64.81481 & 6.141898 & 35.164559 & 27.78734 &   4.5654274 & 23.594207 & 91.011 & Oscillospiraceae bacterium                                                                                                                     & 4 &  96\\\\\n",
       "\t cds &  8 & 780 & 126 & 126 & 0 & 780f & PRIMER\\_LEFT  & AARCCVCGVTAYCTBATGGG  & 20 & 108 & 1 & 738 & 758 & 60.35110 & 2.6283001 & 55.00000 & 5.400617 & 34.293877 & 14.01021 &   2.3512552 & 19.402971 & 91.247 & Dysosmobacter welbionis                                                                                                                        & 6 & 108\\\\\n",
       "\t cds &  8 & 780 & 126 & 126 & 0 & 780r & PRIMER\\_RIGHT & CCAVGTRAABAGCTTBCCRT  & 20 & 108 & 6 & 844 & 864 & 58.71495 & 2.6684202 & 50.00000 & 5.400617 & 18.377478 & 19.37307 & -16.3267600 & 21.842517 & 91.247 & Dysosmobacter welbionis                                                                                                                        & 6 & 108\\\\\n",
       "\t cds & 13 & 265 & 216 & 216 & 0 & 265f & PRIMER\\_LEFT  & CRHTGATGRCGCGYAARAGA  & 20 &  48 & 2 &  19 &  39 & 60.46714 & 2.6839022 & 51.66667 & 5.527708 & 27.836024 & 21.98608 &  15.9462087 & 11.271684 & 91.379 & Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis;Oscillibacter sp. PEA192 & 2 &  48\\\\\n",
       "\t cds & 13 & 265 & 216 & 216 & 0 & 265r & PRIMER\\_RIGHT & AVSTGGARTAGTTCATSCCGT & 21 &  24 & 2 & 214 & 235 & 60.08197 & 1.3894420 & 48.41270 & 3.272306 & 30.039833 & 16.01142 & -27.4416464 & 14.647714 & 91.379 & Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis;Oscillibacter sp. PEA192 & 2 &  48\\\\\n",
       "\t cds & 13 & 295 & 125 & 125 & 0 & 295f & PRIMER\\_LEFT  & RSMVGTSATGAAGTCCCTGA  & 20 &  48 & 1 & 110 & 130 & 59.84114 & 1.8496320 & 53.33333 & 4.249183 & 11.670800 & 18.48426 &  -7.5781759 & 28.282381 & 91.379 & Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis;Oscillibacter sp. PEA192 & 2 &  48\\\\\n",
       "\t cds & 13 & 295 & 125 & 125 & 0 & 295r & PRIMER\\_RIGHT & AVSTGGARTAGTTCATSCCGT & 21 &  24 & 2 & 214 & 235 & 60.08197 & 1.3894420 & 48.41270 & 3.272306 & 30.039833 & 16.01142 & -27.4416464 & 14.647714 & 91.379 & Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis;Oscillibacter sp. PEA192 & 2 &  48\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 30 × 26\n",
       "\n",
       "| gene_type &lt;chr&gt; | cluster_id &lt;int&gt; | primer_set &lt;int&gt; | amplicon_size_consensus &lt;int&gt; | amplicon_size_avg &lt;dbl&gt; | amplicon_size_sd &lt;dbl&gt; | primer_id &lt;fct&gt; | primer_type &lt;fct&gt; | sequence &lt;fct&gt; | length &lt;int&gt; | degeneracy &lt;int&gt; | degeneracy_3prime &lt;int&gt; | position_start &lt;int&gt; | position_end &lt;int&gt; | Tm_avg &lt;dbl&gt; | Tm_sd &lt;dbl&gt; | GC_avg &lt;dbl&gt; | GC_sd &lt;dbl&gt; | hairpin_avg &lt;dbl&gt; | hairpin_sd &lt;dbl&gt; | homodimer_avg &lt;dbl&gt; | homodimer_sd &lt;dbl&gt; | pident &lt;dbl&gt; | sscinames &lt;fct&gt; | max_degeneracy_3prime &lt;int&gt; | max_degeneracy &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| cds | 24 | 101 | 160 | 160 | 0 | 101f | PRIMER_LEFT  | TBTTCCGBGAYAAYTCCACS  | 20 |  72 | 2 | 196 | 216 | 59.40978 | 2.2201857 | 51.66667 | 4.859127 | 23.170159 | 23.39333 |   3.8426830 | 12.757158 | 88.413 | Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          | 2 |  72 |\n",
       "| cds | 24 | 101 | 160 | 160 | 0 | 101r | PRIMER_RIGHT | ATSACRTCVGCCATRAASGA  | 20 |  48 | 2 | 336 | 356 | 59.29837 | 2.1329283 | 48.33333 | 4.249183 | 37.237596 | 11.36689 |  -0.2072119 | 21.774421 | 88.413 | Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          | 2 |  72 |\n",
       "| cds | 24 | 925 | 159 | 159 | 0 | 925f | PRIMER_LEFT  | TBTTCCGBGAYAAYTCCACS  | 20 |  72 | 2 | 196 | 216 | 59.40978 | 2.2201857 | 51.66667 | 4.859127 | 23.170159 | 23.39333 |   3.8426830 | 12.757158 | 88.413 | Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          | 2 |  72 |\n",
       "| cds | 24 | 925 | 159 | 159 | 0 | 925r | PRIMER_RIGHT | TSACRTCVGCCATRAASGA   | 19 |  48 | 2 | 336 | 355 | 58.81608 | 2.2386673 | 50.87719 | 4.472824 | 37.237596 | 11.36689 |  -0.9210938 | 21.070817 | 88.413 | Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          | 2 |  72 |\n",
       "| cds | 24 | 252 | 205 | 205 | 0 | 252f | PRIMER_LEFT  | ACSGCSAACATGGTBTCSTT  | 20 |  24 | 2 | 321 | 341 | 62.52437 | 0.8325446 | 53.33333 | 2.357023 | 34.079778 | 15.71851 |   8.6676870 |  8.532017 | 88.413 | Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          | 2 |  96 |\n",
       "| cds | 24 | 252 | 205 | 205 | 0 | 252r | PRIMER_RIGHT | YRTGSAGCATRTCVGCCATR  | 20 |  96 | 2 | 506 | 526 | 60.37872 | 2.5635953 | 53.33333 | 5.527708 | 39.814057 | 14.18557 |   3.8855806 | 13.889514 | 88.413 | Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          | 2 |  96 |\n",
       "| cds | 24 | 852 | 135 | 135 | 0 | 852f | PRIMER_LEFT  | GBTTCTCVTTYGCCTCYGC   | 19 |  36 | 2 | 223 | 242 | 60.45917 | 2.1671838 | 59.64912 | 5.114870 |  9.565666 | 16.56822 | -52.4259519 |  8.602116 | 88.413 | Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          | 4 |  48 |\n",
       "| cds | 24 | 852 | 135 | 135 | 0 | 852r | PRIMER_RIGHT | CGATSACRTCVGCCATRAAS  | 20 |  48 | 4 | 338 | 358 | 59.09106 | 2.0362332 | 53.33333 | 4.249183 | 34.242766 | 16.18458 |   5.7775265 | 14.765420 | 88.413 | Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          | 4 |  48 |\n",
       "| cds | 24 | 962 | 133 | 133 | 0 | 962f | PRIMER_LEFT  | TTCTCVTTYGCCTCYGCST   | 19 |  24 | 4 | 225 | 244 | 61.47875 | 2.0007686 | 56.14035 | 4.472824 | 29.250102 | 23.92131 | -27.4141990 | 26.541144 | 88.413 | Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          | 4 |  48 |\n",
       "| cds | 24 | 962 | 133 | 133 | 0 | 962r | PRIMER_RIGHT | CGATSACRTCVGCCATRAAS  | 20 |  48 | 4 | 338 | 358 | 59.09106 | 2.0362332 | 53.33333 | 4.249183 | 34.242766 | 16.18458 |   5.7775265 | 14.765420 | 88.413 | Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          | 4 |  48 |\n",
       "| cds | 24 | 997 | 149 | 149 | 0 | 997f | PRIMER_LEFT  | CSTTYATGGCBGAYGTSATC  | 20 |  48 | 2 | 337 | 357 | 58.90509 | 2.0743193 | 53.33333 | 4.249183 | 29.056470 | 16.26418 |   1.3684155 | 16.456230 | 88.413 | Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          | 4 |  64 |\n",
       "| cds | 24 | 997 | 149 | 149 | 0 | 997r | PRIMER_RIGHT | RTCRCACTGSAGRTTYACMA  | 20 |  64 | 4 | 466 | 486 | 58.27347 | 2.5143072 | 47.50000 | 5.590170 | 15.209464 | 18.47412 |  -4.6224649 | 33.375152 | 88.413 | Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          | 4 |  64 |\n",
       "| cds | 24 |  34 | 162 | 162 | 0 | 34f  | PRIMER_LEFT  | TBTTCCGBGAYAAYTCCACS  | 20 |  72 | 2 | 196 | 216 | 59.40978 | 2.2201857 | 51.66667 | 4.859127 | 23.170159 | 23.39333 |   3.8426830 | 12.757158 | 88.413 | Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          | 4 |  72 |\n",
       "| cds | 24 |  34 | 162 | 162 | 0 | 34r  | PRIMER_RIGHT | CGATSACRTCVGCCATRAAS  | 20 |  48 | 4 | 338 | 358 | 59.09106 | 2.0362332 | 53.33333 | 4.249183 | 34.242766 | 16.18458 |   5.7775265 | 14.765420 | 88.413 | Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          | 4 |  72 |\n",
       "| cds | 24 | 949 | 137 | 137 | 0 | 949f | PRIMER_LEFT  | CBTGGGAYAAGASYGAYGA   | 19 |  48 | 2 |  79 |  98 | 57.43879 | 2.7226357 | 53.50877 | 5.189544 | 23.628755 | 22.96581 | -25.9893377 | 26.595037 | 88.413 | Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          | 4 |  72 |\n",
       "| cds | 24 | 949 | 137 | 137 | 0 | 949r | PRIMER_RIGHT | SGTGGARTTRTCVCGGAAVA  | 20 |  72 | 3 | 196 | 216 | 59.40978 | 2.2201857 | 51.66667 | 4.859127 | 32.446352 | 19.32925 |   2.2146646 | 13.318311 | 88.413 | Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          | 4 |  72 |\n",
       "| cds | 36 | 952 | 114 | 114 | 0 | 952f | PRIMER_LEFT  | YGARATYCAYCGSCTGAACC  | 20 |  32 | 1 | 356 | 376 | 60.32997 | 1.6997727 | 55.00000 | 5.000000 | 27.075589 | 21.15220 | -11.3804235 | 24.602761 | 89.118 | Clostridiales bacterium                                                                                                                        | 3 |  32 |\n",
       "| cds | 36 | 952 | 114 | 114 | 0 | 952r | PRIMER_RIGHT | TCSAGVCGGATGGAGTTBG   | 19 |  18 | 3 | 451 | 470 | 60.74049 | 1.6644375 | 59.64912 | 3.508772 | 28.880424 | 20.58490 |  -4.3656091 | 26.977200 | 89.118 | Clostridiales bacterium                                                                                                                        | 3 |  32 |\n",
       "| cds | 36 | 627 | 113 | 113 | 0 | 627f | PRIMER_LEFT  | GAYYTKGCKGCRCTKYTGA   | 19 | 128 | 4 | 303 | 322 | 60.57837 | 2.9830748 | 55.26316 | 6.962503 |  4.689779 | 12.40799 |  15.8095996 | 13.575059 | 89.118 | Clostridiales bacterium                                                                                                                        | 6 | 128 |\n",
       "| cds | 36 | 627 | 113 | 113 | 0 | 627r | PRIMER_RIGHT | WARTCCTCCATVGCVGGRTA  | 20 |  72 | 2 | 396 | 416 | 59.04885 | 2.4997399 | 51.66667 | 4.859127 | 28.991987 | 16.18955 |   3.0088796 | 20.436287 | 89.118 | Clostridiales bacterium                                                                                                                        | 6 | 128 |\n",
       "| cds |  1 | 468 | 189 | 189 | 0 | 468f | PRIMER_LEFT  | AARAACGGYCGYGAYTCYRA  | 20 |  64 | 4 |  54 |  74 | 60.76051 | 2.8483468 | 50.00000 | 6.123724 | 16.305789 | 21.51507 |  22.8243414 | 16.405838 | 91.011 | Oscillospiraceae bacterium                                                                                                                     | 4 |  96 |\n",
       "| cds |  1 | 468 | 189 | 189 | 0 | 468r | PRIMER_RIGHT | RCCSAGVYGCTCRAARCG    | 18 |  96 | 2 | 225 | 243 | 62.16555 | 2.5748132 | 64.81481 | 6.141898 | 35.164559 | 27.78734 |   4.5654274 | 23.594207 | 91.011 | Oscillospiraceae bacterium                                                                                                                     | 4 |  96 |\n",
       "| cds |  1 | 927 | 188 | 188 | 0 | 927f | PRIMER_LEFT  | ARAACGGYCGYGAYTCYRA   | 19 |  64 | 4 |  55 |  74 | 60.13246 | 2.9968465 | 52.63158 | 6.446026 | 16.118781 | 21.37324 |  22.7879011 | 16.410793 | 91.011 | Oscillospiraceae bacterium                                                                                                                     | 4 |  96 |\n",
       "| cds |  1 | 927 | 188 | 188 | 0 | 927r | PRIMER_RIGHT | RCCSAGVYGCTCRAARCG    | 18 |  96 | 2 | 225 | 243 | 62.16555 | 2.5748132 | 64.81481 | 6.141898 | 35.164559 | 27.78734 |   4.5654274 | 23.594207 | 91.011 | Oscillospiraceae bacterium                                                                                                                     | 4 |  96 |\n",
       "| cds |  8 | 780 | 126 | 126 | 0 | 780f | PRIMER_LEFT  | AARCCVCGVTAYCTBATGGG  | 20 | 108 | 1 | 738 | 758 | 60.35110 | 2.6283001 | 55.00000 | 5.400617 | 34.293877 | 14.01021 |   2.3512552 | 19.402971 | 91.247 | Dysosmobacter welbionis                                                                                                                        | 6 | 108 |\n",
       "| cds |  8 | 780 | 126 | 126 | 0 | 780r | PRIMER_RIGHT | CCAVGTRAABAGCTTBCCRT  | 20 | 108 | 6 | 844 | 864 | 58.71495 | 2.6684202 | 50.00000 | 5.400617 | 18.377478 | 19.37307 | -16.3267600 | 21.842517 | 91.247 | Dysosmobacter welbionis                                                                                                                        | 6 | 108 |\n",
       "| cds | 13 | 265 | 216 | 216 | 0 | 265f | PRIMER_LEFT  | CRHTGATGRCGCGYAARAGA  | 20 |  48 | 2 |  19 |  39 | 60.46714 | 2.6839022 | 51.66667 | 5.527708 | 27.836024 | 21.98608 |  15.9462087 | 11.271684 | 91.379 | Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis;Oscillibacter sp. PEA192 | 2 |  48 |\n",
       "| cds | 13 | 265 | 216 | 216 | 0 | 265r | PRIMER_RIGHT | AVSTGGARTAGTTCATSCCGT | 21 |  24 | 2 | 214 | 235 | 60.08197 | 1.3894420 | 48.41270 | 3.272306 | 30.039833 | 16.01142 | -27.4416464 | 14.647714 | 91.379 | Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis;Oscillibacter sp. PEA192 | 2 |  48 |\n",
       "| cds | 13 | 295 | 125 | 125 | 0 | 295f | PRIMER_LEFT  | RSMVGTSATGAAGTCCCTGA  | 20 |  48 | 1 | 110 | 130 | 59.84114 | 1.8496320 | 53.33333 | 4.249183 | 11.670800 | 18.48426 |  -7.5781759 | 28.282381 | 91.379 | Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis;Oscillibacter sp. PEA192 | 2 |  48 |\n",
       "| cds | 13 | 295 | 125 | 125 | 0 | 295r | PRIMER_RIGHT | AVSTGGARTAGTTCATSCCGT | 21 |  24 | 2 | 214 | 235 | 60.08197 | 1.3894420 | 48.41270 | 3.272306 | 30.039833 | 16.01142 | -27.4416464 | 14.647714 | 91.379 | Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis;Oscillibacter sp. PEA192 | 2 |  48 |\n",
       "\n"
      ],
      "text/plain": [
       "   gene_type cluster_id primer_set amplicon_size_consensus amplicon_size_avg\n",
       "1  cds       24         101        160                     160              \n",
       "2  cds       24         101        160                     160              \n",
       "3  cds       24         925        159                     159              \n",
       "4  cds       24         925        159                     159              \n",
       "5  cds       24         252        205                     205              \n",
       "6  cds       24         252        205                     205              \n",
       "7  cds       24         852        135                     135              \n",
       "8  cds       24         852        135                     135              \n",
       "9  cds       24         962        133                     133              \n",
       "10 cds       24         962        133                     133              \n",
       "11 cds       24         997        149                     149              \n",
       "12 cds       24         997        149                     149              \n",
       "13 cds       24          34        162                     162              \n",
       "14 cds       24          34        162                     162              \n",
       "15 cds       24         949        137                     137              \n",
       "16 cds       24         949        137                     137              \n",
       "17 cds       36         952        114                     114              \n",
       "18 cds       36         952        114                     114              \n",
       "19 cds       36         627        113                     113              \n",
       "20 cds       36         627        113                     113              \n",
       "21 cds        1         468        189                     189              \n",
       "22 cds        1         468        189                     189              \n",
       "23 cds        1         927        188                     188              \n",
       "24 cds        1         927        188                     188              \n",
       "25 cds        8         780        126                     126              \n",
       "26 cds        8         780        126                     126              \n",
       "27 cds       13         265        216                     216              \n",
       "28 cds       13         265        216                     216              \n",
       "29 cds       13         295        125                     125              \n",
       "30 cds       13         295        125                     125              \n",
       "   amplicon_size_sd primer_id primer_type  sequence              length\n",
       "1  0                101f      PRIMER_LEFT  TBTTCCGBGAYAAYTCCACS  20    \n",
       "2  0                101r      PRIMER_RIGHT ATSACRTCVGCCATRAASGA  20    \n",
       "3  0                925f      PRIMER_LEFT  TBTTCCGBGAYAAYTCCACS  20    \n",
       "4  0                925r      PRIMER_RIGHT TSACRTCVGCCATRAASGA   19    \n",
       "5  0                252f      PRIMER_LEFT  ACSGCSAACATGGTBTCSTT  20    \n",
       "6  0                252r      PRIMER_RIGHT YRTGSAGCATRTCVGCCATR  20    \n",
       "7  0                852f      PRIMER_LEFT  GBTTCTCVTTYGCCTCYGC   19    \n",
       "8  0                852r      PRIMER_RIGHT CGATSACRTCVGCCATRAAS  20    \n",
       "9  0                962f      PRIMER_LEFT  TTCTCVTTYGCCTCYGCST   19    \n",
       "10 0                962r      PRIMER_RIGHT CGATSACRTCVGCCATRAAS  20    \n",
       "11 0                997f      PRIMER_LEFT  CSTTYATGGCBGAYGTSATC  20    \n",
       "12 0                997r      PRIMER_RIGHT RTCRCACTGSAGRTTYACMA  20    \n",
       "13 0                34f       PRIMER_LEFT  TBTTCCGBGAYAAYTCCACS  20    \n",
       "14 0                34r       PRIMER_RIGHT CGATSACRTCVGCCATRAAS  20    \n",
       "15 0                949f      PRIMER_LEFT  CBTGGGAYAAGASYGAYGA   19    \n",
       "16 0                949r      PRIMER_RIGHT SGTGGARTTRTCVCGGAAVA  20    \n",
       "17 0                952f      PRIMER_LEFT  YGARATYCAYCGSCTGAACC  20    \n",
       "18 0                952r      PRIMER_RIGHT TCSAGVCGGATGGAGTTBG   19    \n",
       "19 0                627f      PRIMER_LEFT  GAYYTKGCKGCRCTKYTGA   19    \n",
       "20 0                627r      PRIMER_RIGHT WARTCCTCCATVGCVGGRTA  20    \n",
       "21 0                468f      PRIMER_LEFT  AARAACGGYCGYGAYTCYRA  20    \n",
       "22 0                468r      PRIMER_RIGHT RCCSAGVYGCTCRAARCG    18    \n",
       "23 0                927f      PRIMER_LEFT  ARAACGGYCGYGAYTCYRA   19    \n",
       "24 0                927r      PRIMER_RIGHT RCCSAGVYGCTCRAARCG    18    \n",
       "25 0                780f      PRIMER_LEFT  AARCCVCGVTAYCTBATGGG  20    \n",
       "26 0                780r      PRIMER_RIGHT CCAVGTRAABAGCTTBCCRT  20    \n",
       "27 0                265f      PRIMER_LEFT  CRHTGATGRCGCGYAARAGA  20    \n",
       "28 0                265r      PRIMER_RIGHT AVSTGGARTAGTTCATSCCGT 21    \n",
       "29 0                295f      PRIMER_LEFT  RSMVGTSATGAAGTCCCTGA  20    \n",
       "30 0                295r      PRIMER_RIGHT AVSTGGARTAGTTCATSCCGT 21    \n",
       "   degeneracy degeneracy_3prime position_start position_end Tm_avg   Tm_sd    \n",
       "1   72        2                 196            216          59.40978 2.2201857\n",
       "2   48        2                 336            356          59.29837 2.1329283\n",
       "3   72        2                 196            216          59.40978 2.2201857\n",
       "4   48        2                 336            355          58.81608 2.2386673\n",
       "5   24        2                 321            341          62.52437 0.8325446\n",
       "6   96        2                 506            526          60.37872 2.5635953\n",
       "7   36        2                 223            242          60.45917 2.1671838\n",
       "8   48        4                 338            358          59.09106 2.0362332\n",
       "9   24        4                 225            244          61.47875 2.0007686\n",
       "10  48        4                 338            358          59.09106 2.0362332\n",
       "11  48        2                 337            357          58.90509 2.0743193\n",
       "12  64        4                 466            486          58.27347 2.5143072\n",
       "13  72        2                 196            216          59.40978 2.2201857\n",
       "14  48        4                 338            358          59.09106 2.0362332\n",
       "15  48        2                  79             98          57.43879 2.7226357\n",
       "16  72        3                 196            216          59.40978 2.2201857\n",
       "17  32        1                 356            376          60.32997 1.6997727\n",
       "18  18        3                 451            470          60.74049 1.6644375\n",
       "19 128        4                 303            322          60.57837 2.9830748\n",
       "20  72        2                 396            416          59.04885 2.4997399\n",
       "21  64        4                  54             74          60.76051 2.8483468\n",
       "22  96        2                 225            243          62.16555 2.5748132\n",
       "23  64        4                  55             74          60.13246 2.9968465\n",
       "24  96        2                 225            243          62.16555 2.5748132\n",
       "25 108        1                 738            758          60.35110 2.6283001\n",
       "26 108        6                 844            864          58.71495 2.6684202\n",
       "27  48        2                  19             39          60.46714 2.6839022\n",
       "28  24        2                 214            235          60.08197 1.3894420\n",
       "29  48        1                 110            130          59.84114 1.8496320\n",
       "30  24        2                 214            235          60.08197 1.3894420\n",
       "   GC_avg   GC_sd    hairpin_avg hairpin_sd homodimer_avg homodimer_sd pident\n",
       "1  51.66667 4.859127 23.170159   23.39333     3.8426830   12.757158    88.413\n",
       "2  48.33333 4.249183 37.237596   11.36689    -0.2072119   21.774421    88.413\n",
       "3  51.66667 4.859127 23.170159   23.39333     3.8426830   12.757158    88.413\n",
       "4  50.87719 4.472824 37.237596   11.36689    -0.9210938   21.070817    88.413\n",
       "5  53.33333 2.357023 34.079778   15.71851     8.6676870    8.532017    88.413\n",
       "6  53.33333 5.527708 39.814057   14.18557     3.8855806   13.889514    88.413\n",
       "7  59.64912 5.114870  9.565666   16.56822   -52.4259519    8.602116    88.413\n",
       "8  53.33333 4.249183 34.242766   16.18458     5.7775265   14.765420    88.413\n",
       "9  56.14035 4.472824 29.250102   23.92131   -27.4141990   26.541144    88.413\n",
       "10 53.33333 4.249183 34.242766   16.18458     5.7775265   14.765420    88.413\n",
       "11 53.33333 4.249183 29.056470   16.26418     1.3684155   16.456230    88.413\n",
       "12 47.50000 5.590170 15.209464   18.47412    -4.6224649   33.375152    88.413\n",
       "13 51.66667 4.859127 23.170159   23.39333     3.8426830   12.757158    88.413\n",
       "14 53.33333 4.249183 34.242766   16.18458     5.7775265   14.765420    88.413\n",
       "15 53.50877 5.189544 23.628755   22.96581   -25.9893377   26.595037    88.413\n",
       "16 51.66667 4.859127 32.446352   19.32925     2.2146646   13.318311    88.413\n",
       "17 55.00000 5.000000 27.075589   21.15220   -11.3804235   24.602761    89.118\n",
       "18 59.64912 3.508772 28.880424   20.58490    -4.3656091   26.977200    89.118\n",
       "19 55.26316 6.962503  4.689779   12.40799    15.8095996   13.575059    89.118\n",
       "20 51.66667 4.859127 28.991987   16.18955     3.0088796   20.436287    89.118\n",
       "21 50.00000 6.123724 16.305789   21.51507    22.8243414   16.405838    91.011\n",
       "22 64.81481 6.141898 35.164559   27.78734     4.5654274   23.594207    91.011\n",
       "23 52.63158 6.446026 16.118781   21.37324    22.7879011   16.410793    91.011\n",
       "24 64.81481 6.141898 35.164559   27.78734     4.5654274   23.594207    91.011\n",
       "25 55.00000 5.400617 34.293877   14.01021     2.3512552   19.402971    91.247\n",
       "26 50.00000 5.400617 18.377478   19.37307   -16.3267600   21.842517    91.247\n",
       "27 51.66667 5.527708 27.836024   21.98608    15.9462087   11.271684    91.379\n",
       "28 48.41270 3.272306 30.039833   16.01142   -27.4416464   14.647714    91.379\n",
       "29 53.33333 4.249183 11.670800   18.48426    -7.5781759   28.282381    91.379\n",
       "30 48.41270 3.272306 30.039833   16.01142   -27.4416464   14.647714    91.379\n",
       "   sscinames                                                                                                                                     \n",
       "1  Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         \n",
       "2  Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         \n",
       "3  Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         \n",
       "4  Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         \n",
       "5  Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         \n",
       "6  Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         \n",
       "7  Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         \n",
       "8  Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         \n",
       "9  Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         \n",
       "10 Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         \n",
       "11 Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         \n",
       "12 Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         \n",
       "13 Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         \n",
       "14 Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         \n",
       "15 Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         \n",
       "16 Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         \n",
       "17 Clostridiales bacterium                                                                                                                       \n",
       "18 Clostridiales bacterium                                                                                                                       \n",
       "19 Clostridiales bacterium                                                                                                                       \n",
       "20 Clostridiales bacterium                                                                                                                       \n",
       "21 Oscillospiraceae bacterium                                                                                                                    \n",
       "22 Oscillospiraceae bacterium                                                                                                                    \n",
       "23 Oscillospiraceae bacterium                                                                                                                    \n",
       "24 Oscillospiraceae bacterium                                                                                                                    \n",
       "25 Dysosmobacter welbionis                                                                                                                       \n",
       "26 Dysosmobacter welbionis                                                                                                                       \n",
       "27 Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis;Oscillibacter sp. PEA192\n",
       "28 Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis;Oscillibacter sp. PEA192\n",
       "29 Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis;Oscillibacter sp. PEA192\n",
       "30 Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis;Oscillibacter sp. PEA192\n",
       "   max_degeneracy_3prime max_degeneracy\n",
       "1  2                      72           \n",
       "2  2                      72           \n",
       "3  2                      72           \n",
       "4  2                      72           \n",
       "5  2                      96           \n",
       "6  2                      96           \n",
       "7  4                      48           \n",
       "8  4                      48           \n",
       "9  4                      48           \n",
       "10 4                      48           \n",
       "11 4                      64           \n",
       "12 4                      64           \n",
       "13 4                      72           \n",
       "14 4                      72           \n",
       "15 4                      72           \n",
       "16 4                      72           \n",
       "17 3                      32           \n",
       "18 3                      32           \n",
       "19 6                     128           \n",
       "20 6                     128           \n",
       "21 4                      96           \n",
       "22 4                      96           \n",
       "23 4                      96           \n",
       "24 4                      96           \n",
       "25 6                     108           \n",
       "26 6                     108           \n",
       "27 2                      48           \n",
       "28 2                      48           \n",
       "29 2                      48           \n",
       "30 2                      48           "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# arrange by \"best\" primers\n",
    "primer_info_jf = primer_info_j %>%\n",
    "    group_by(primer_set) %>%\n",
    "    mutate(max_degeneracy_3prime = max(degeneracy_3prime),  # max of [Fwd,Rev]\n",
    "           max_degeneracy = max(degeneracy)) %>%\n",
    "    ungroup() %>%\n",
    "    arrange(pident, max_degeneracy_3prime, max_degeneracy) %>%\n",
    "    head(n=30)\n",
    "\n",
    "df.dims(30,40)\n",
    "primer_info_jf\n",
    "df.dims()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### w/ gene annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-23T08:00:29.911161Z",
     "iopub.status.busy": "2021-04-23T08:00:29.909221Z",
     "iopub.status.idle": "2021-04-23T08:00:30.019072Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A tibble: 30 × 27</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>gene_type</th><th scope=col>cluster_id</th><th scope=col>primer_set</th><th scope=col>amplicon_size_consensus</th><th scope=col>amplicon_size_avg</th><th scope=col>amplicon_size_sd</th><th scope=col>primer_id</th><th scope=col>primer_type</th><th scope=col>sequence</th><th scope=col>length</th><th scope=col>degeneracy</th><th scope=col>degeneracy_3prime</th><th scope=col>position_start</th><th scope=col>position_end</th><th scope=col>Tm_avg</th><th scope=col>Tm_sd</th><th scope=col>GC_avg</th><th scope=col>GC_sd</th><th scope=col>hairpin_avg</th><th scope=col>hairpin_sd</th><th scope=col>homodimer_avg</th><th scope=col>homodimer_sd</th><th scope=col>pident</th><th scope=col>sscinames</th><th scope=col>max_degeneracy_3prime</th><th scope=col>max_degeneracy</th><th scope=col>annotation</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>cds</td><td>24</td><td>101</td><td>160</td><td>160</td><td>0</td><td>101f</td><td>PRIMER_LEFT </td><td>TBTTCCGBGAYAAYTCCACS </td><td>20</td><td> 72</td><td>2</td><td>196</td><td>216</td><td>59.40978</td><td>2.2201857</td><td>51.66667</td><td>4.859127</td><td>23.170159</td><td>23.39333</td><td>  3.8426830</td><td>12.757158</td><td>88.413</td><td>Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         </td><td>2</td><td> 72</td><td>Putative carbamoyltransferase YgeW               </td></tr>\n",
       "\t<tr><td>cds</td><td>24</td><td>101</td><td>160</td><td>160</td><td>0</td><td>101r</td><td>PRIMER_RIGHT</td><td>ATSACRTCVGCCATRAASGA </td><td>20</td><td> 48</td><td>2</td><td>336</td><td>356</td><td>59.29837</td><td>2.1329283</td><td>48.33333</td><td>4.249183</td><td>37.237596</td><td>11.36689</td><td> -0.2072119</td><td>21.774421</td><td>88.413</td><td>Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         </td><td>2</td><td> 72</td><td>Putative carbamoyltransferase YgeW               </td></tr>\n",
       "\t<tr><td>cds</td><td>24</td><td>925</td><td>159</td><td>159</td><td>0</td><td>925f</td><td>PRIMER_LEFT </td><td>TBTTCCGBGAYAAYTCCACS </td><td>20</td><td> 72</td><td>2</td><td>196</td><td>216</td><td>59.40978</td><td>2.2201857</td><td>51.66667</td><td>4.859127</td><td>23.170159</td><td>23.39333</td><td>  3.8426830</td><td>12.757158</td><td>88.413</td><td>Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         </td><td>2</td><td> 72</td><td>Putative carbamoyltransferase YgeW               </td></tr>\n",
       "\t<tr><td>cds</td><td>24</td><td>925</td><td>159</td><td>159</td><td>0</td><td>925r</td><td>PRIMER_RIGHT</td><td>TSACRTCVGCCATRAASGA  </td><td>19</td><td> 48</td><td>2</td><td>336</td><td>355</td><td>58.81608</td><td>2.2386673</td><td>50.87719</td><td>4.472824</td><td>37.237596</td><td>11.36689</td><td> -0.9210938</td><td>21.070817</td><td>88.413</td><td>Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         </td><td>2</td><td> 72</td><td>Putative carbamoyltransferase YgeW               </td></tr>\n",
       "\t<tr><td>cds</td><td>24</td><td>252</td><td>205</td><td>205</td><td>0</td><td>252f</td><td>PRIMER_LEFT </td><td>ACSGCSAACATGGTBTCSTT </td><td>20</td><td> 24</td><td>2</td><td>321</td><td>341</td><td>62.52437</td><td>0.8325446</td><td>53.33333</td><td>2.357023</td><td>34.079778</td><td>15.71851</td><td>  8.6676870</td><td> 8.532017</td><td>88.413</td><td>Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         </td><td>2</td><td> 96</td><td>Putative carbamoyltransferase YgeW               </td></tr>\n",
       "\t<tr><td>cds</td><td>24</td><td>252</td><td>205</td><td>205</td><td>0</td><td>252r</td><td>PRIMER_RIGHT</td><td>YRTGSAGCATRTCVGCCATR </td><td>20</td><td> 96</td><td>2</td><td>506</td><td>526</td><td>60.37872</td><td>2.5635953</td><td>53.33333</td><td>5.527708</td><td>39.814057</td><td>14.18557</td><td>  3.8855806</td><td>13.889514</td><td>88.413</td><td>Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         </td><td>2</td><td> 96</td><td>Putative carbamoyltransferase YgeW               </td></tr>\n",
       "\t<tr><td>cds</td><td>24</td><td>852</td><td>135</td><td>135</td><td>0</td><td>852f</td><td>PRIMER_LEFT </td><td>GBTTCTCVTTYGCCTCYGC  </td><td>19</td><td> 36</td><td>2</td><td>223</td><td>242</td><td>60.45917</td><td>2.1671838</td><td>59.64912</td><td>5.114870</td><td> 9.565666</td><td>16.56822</td><td>-52.4259519</td><td> 8.602116</td><td>88.413</td><td>Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         </td><td>4</td><td> 48</td><td>Putative carbamoyltransferase YgeW               </td></tr>\n",
       "\t<tr><td>cds</td><td>24</td><td>852</td><td>135</td><td>135</td><td>0</td><td>852r</td><td>PRIMER_RIGHT</td><td>CGATSACRTCVGCCATRAAS </td><td>20</td><td> 48</td><td>4</td><td>338</td><td>358</td><td>59.09106</td><td>2.0362332</td><td>53.33333</td><td>4.249183</td><td>34.242766</td><td>16.18458</td><td>  5.7775265</td><td>14.765420</td><td>88.413</td><td>Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         </td><td>4</td><td> 48</td><td>Putative carbamoyltransferase YgeW               </td></tr>\n",
       "\t<tr><td>cds</td><td>24</td><td>962</td><td>133</td><td>133</td><td>0</td><td>962f</td><td>PRIMER_LEFT </td><td>TTCTCVTTYGCCTCYGCST  </td><td>19</td><td> 24</td><td>4</td><td>225</td><td>244</td><td>61.47875</td><td>2.0007686</td><td>56.14035</td><td>4.472824</td><td>29.250102</td><td>23.92131</td><td>-27.4141990</td><td>26.541144</td><td>88.413</td><td>Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         </td><td>4</td><td> 48</td><td>Putative carbamoyltransferase YgeW               </td></tr>\n",
       "\t<tr><td>cds</td><td>24</td><td>962</td><td>133</td><td>133</td><td>0</td><td>962r</td><td>PRIMER_RIGHT</td><td>CGATSACRTCVGCCATRAAS </td><td>20</td><td> 48</td><td>4</td><td>338</td><td>358</td><td>59.09106</td><td>2.0362332</td><td>53.33333</td><td>4.249183</td><td>34.242766</td><td>16.18458</td><td>  5.7775265</td><td>14.765420</td><td>88.413</td><td>Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         </td><td>4</td><td> 48</td><td>Putative carbamoyltransferase YgeW               </td></tr>\n",
       "\t<tr><td>cds</td><td>24</td><td>997</td><td>149</td><td>149</td><td>0</td><td>997f</td><td>PRIMER_LEFT </td><td>CSTTYATGGCBGAYGTSATC </td><td>20</td><td> 48</td><td>2</td><td>337</td><td>357</td><td>58.90509</td><td>2.0743193</td><td>53.33333</td><td>4.249183</td><td>29.056470</td><td>16.26418</td><td>  1.3684155</td><td>16.456230</td><td>88.413</td><td>Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         </td><td>4</td><td> 64</td><td>Putative carbamoyltransferase YgeW               </td></tr>\n",
       "\t<tr><td>cds</td><td>24</td><td>997</td><td>149</td><td>149</td><td>0</td><td>997r</td><td>PRIMER_RIGHT</td><td>RTCRCACTGSAGRTTYACMA </td><td>20</td><td> 64</td><td>4</td><td>466</td><td>486</td><td>58.27347</td><td>2.5143072</td><td>47.50000</td><td>5.590170</td><td>15.209464</td><td>18.47412</td><td> -4.6224649</td><td>33.375152</td><td>88.413</td><td>Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         </td><td>4</td><td> 64</td><td>Putative carbamoyltransferase YgeW               </td></tr>\n",
       "\t<tr><td>cds</td><td>24</td><td> 34</td><td>162</td><td>162</td><td>0</td><td>34f </td><td>PRIMER_LEFT </td><td>TBTTCCGBGAYAAYTCCACS </td><td>20</td><td> 72</td><td>2</td><td>196</td><td>216</td><td>59.40978</td><td>2.2201857</td><td>51.66667</td><td>4.859127</td><td>23.170159</td><td>23.39333</td><td>  3.8426830</td><td>12.757158</td><td>88.413</td><td>Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         </td><td>4</td><td> 72</td><td>Putative carbamoyltransferase YgeW               </td></tr>\n",
       "\t<tr><td>cds</td><td>24</td><td> 34</td><td>162</td><td>162</td><td>0</td><td>34r </td><td>PRIMER_RIGHT</td><td>CGATSACRTCVGCCATRAAS </td><td>20</td><td> 48</td><td>4</td><td>338</td><td>358</td><td>59.09106</td><td>2.0362332</td><td>53.33333</td><td>4.249183</td><td>34.242766</td><td>16.18458</td><td>  5.7775265</td><td>14.765420</td><td>88.413</td><td>Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         </td><td>4</td><td> 72</td><td>Putative carbamoyltransferase YgeW               </td></tr>\n",
       "\t<tr><td>cds</td><td>24</td><td>949</td><td>137</td><td>137</td><td>0</td><td>949f</td><td>PRIMER_LEFT </td><td>CBTGGGAYAAGASYGAYGA  </td><td>19</td><td> 48</td><td>2</td><td> 79</td><td> 98</td><td>57.43879</td><td>2.7226357</td><td>53.50877</td><td>5.189544</td><td>23.628755</td><td>22.96581</td><td>-25.9893377</td><td>26.595037</td><td>88.413</td><td>Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         </td><td>4</td><td> 72</td><td>Putative carbamoyltransferase YgeW               </td></tr>\n",
       "\t<tr><td>cds</td><td>24</td><td>949</td><td>137</td><td>137</td><td>0</td><td>949r</td><td>PRIMER_RIGHT</td><td>SGTGGARTTRTCVCGGAAVA </td><td>20</td><td> 72</td><td>3</td><td>196</td><td>216</td><td>59.40978</td><td>2.2201857</td><td>51.66667</td><td>4.859127</td><td>32.446352</td><td>19.32925</td><td>  2.2146646</td><td>13.318311</td><td>88.413</td><td>Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         </td><td>4</td><td> 72</td><td>Putative carbamoyltransferase YgeW               </td></tr>\n",
       "\t<tr><td>cds</td><td>36</td><td>952</td><td>114</td><td>114</td><td>0</td><td>952f</td><td>PRIMER_LEFT </td><td>YGARATYCAYCGSCTGAACC </td><td>20</td><td> 32</td><td>1</td><td>356</td><td>376</td><td>60.32997</td><td>1.6997727</td><td>55.00000</td><td>5.000000</td><td>27.075589</td><td>21.15220</td><td>-11.3804235</td><td>24.602761</td><td>89.118</td><td>Clostridiales bacterium                                                                                                                       </td><td>3</td><td> 32</td><td>Holliday junction ATP-dependent DNA helicase RuvB</td></tr>\n",
       "\t<tr><td>cds</td><td>36</td><td>952</td><td>114</td><td>114</td><td>0</td><td>952r</td><td>PRIMER_RIGHT</td><td>TCSAGVCGGATGGAGTTBG  </td><td>19</td><td> 18</td><td>3</td><td>451</td><td>470</td><td>60.74049</td><td>1.6644375</td><td>59.64912</td><td>3.508772</td><td>28.880424</td><td>20.58490</td><td> -4.3656091</td><td>26.977200</td><td>89.118</td><td>Clostridiales bacterium                                                                                                                       </td><td>3</td><td> 32</td><td>Holliday junction ATP-dependent DNA helicase RuvB</td></tr>\n",
       "\t<tr><td>cds</td><td>36</td><td>627</td><td>113</td><td>113</td><td>0</td><td>627f</td><td>PRIMER_LEFT </td><td>GAYYTKGCKGCRCTKYTGA  </td><td>19</td><td>128</td><td>4</td><td>303</td><td>322</td><td>60.57837</td><td>2.9830748</td><td>55.26316</td><td>6.962503</td><td> 4.689779</td><td>12.40799</td><td> 15.8095996</td><td>13.575059</td><td>89.118</td><td>Clostridiales bacterium                                                                                                                       </td><td>6</td><td>128</td><td>Holliday junction ATP-dependent DNA helicase RuvB</td></tr>\n",
       "\t<tr><td>cds</td><td>36</td><td>627</td><td>113</td><td>113</td><td>0</td><td>627r</td><td>PRIMER_RIGHT</td><td>WARTCCTCCATVGCVGGRTA </td><td>20</td><td> 72</td><td>2</td><td>396</td><td>416</td><td>59.04885</td><td>2.4997399</td><td>51.66667</td><td>4.859127</td><td>28.991987</td><td>16.18955</td><td>  3.0088796</td><td>20.436287</td><td>89.118</td><td>Clostridiales bacterium                                                                                                                       </td><td>6</td><td>128</td><td>Holliday junction ATP-dependent DNA helicase RuvB</td></tr>\n",
       "\t<tr><td>cds</td><td> 1</td><td>468</td><td>189</td><td>189</td><td>0</td><td>468f</td><td>PRIMER_LEFT </td><td>AARAACGGYCGYGAYTCYRA </td><td>20</td><td> 64</td><td>4</td><td> 54</td><td> 74</td><td>60.76051</td><td>2.8483468</td><td>50.00000</td><td>6.123724</td><td>16.305789</td><td>21.51507</td><td> 22.8243414</td><td>16.405838</td><td>91.011</td><td>Oscillospiraceae bacterium                                                                                                                    </td><td>4</td><td> 96</td><td>50S ribosomal protein L27                        </td></tr>\n",
       "\t<tr><td>cds</td><td> 1</td><td>468</td><td>189</td><td>189</td><td>0</td><td>468r</td><td>PRIMER_RIGHT</td><td>RCCSAGVYGCTCRAARCG   </td><td>18</td><td> 96</td><td>2</td><td>225</td><td>243</td><td>62.16555</td><td>2.5748132</td><td>64.81481</td><td>6.141898</td><td>35.164559</td><td>27.78734</td><td>  4.5654274</td><td>23.594207</td><td>91.011</td><td>Oscillospiraceae bacterium                                                                                                                    </td><td>4</td><td> 96</td><td>50S ribosomal protein L27                        </td></tr>\n",
       "\t<tr><td>cds</td><td> 1</td><td>927</td><td>188</td><td>188</td><td>0</td><td>927f</td><td>PRIMER_LEFT </td><td>ARAACGGYCGYGAYTCYRA  </td><td>19</td><td> 64</td><td>4</td><td> 55</td><td> 74</td><td>60.13246</td><td>2.9968465</td><td>52.63158</td><td>6.446026</td><td>16.118781</td><td>21.37324</td><td> 22.7879011</td><td>16.410793</td><td>91.011</td><td>Oscillospiraceae bacterium                                                                                                                    </td><td>4</td><td> 96</td><td>50S ribosomal protein L27                        </td></tr>\n",
       "\t<tr><td>cds</td><td> 1</td><td>927</td><td>188</td><td>188</td><td>0</td><td>927r</td><td>PRIMER_RIGHT</td><td>RCCSAGVYGCTCRAARCG   </td><td>18</td><td> 96</td><td>2</td><td>225</td><td>243</td><td>62.16555</td><td>2.5748132</td><td>64.81481</td><td>6.141898</td><td>35.164559</td><td>27.78734</td><td>  4.5654274</td><td>23.594207</td><td>91.011</td><td>Oscillospiraceae bacterium                                                                                                                    </td><td>4</td><td> 96</td><td>50S ribosomal protein L27                        </td></tr>\n",
       "\t<tr><td>cds</td><td> 8</td><td>780</td><td>126</td><td>126</td><td>0</td><td>780f</td><td>PRIMER_LEFT </td><td>AARCCVCGVTAYCTBATGGG </td><td>20</td><td>108</td><td>1</td><td>738</td><td>758</td><td>60.35110</td><td>2.6283001</td><td>55.00000</td><td>5.400617</td><td>34.293877</td><td>14.01021</td><td>  2.3512552</td><td>19.402971</td><td>91.247</td><td>Dysosmobacter welbionis                                                                                                                       </td><td>6</td><td>108</td><td>Queuine tRNA-ribosyltransferase                  </td></tr>\n",
       "\t<tr><td>cds</td><td> 8</td><td>780</td><td>126</td><td>126</td><td>0</td><td>780r</td><td>PRIMER_RIGHT</td><td>CCAVGTRAABAGCTTBCCRT </td><td>20</td><td>108</td><td>6</td><td>844</td><td>864</td><td>58.71495</td><td>2.6684202</td><td>50.00000</td><td>5.400617</td><td>18.377478</td><td>19.37307</td><td>-16.3267600</td><td>21.842517</td><td>91.247</td><td>Dysosmobacter welbionis                                                                                                                       </td><td>6</td><td>108</td><td>Queuine tRNA-ribosyltransferase                  </td></tr>\n",
       "\t<tr><td>cds</td><td>13</td><td>265</td><td>216</td><td>216</td><td>0</td><td>265f</td><td>PRIMER_LEFT </td><td>CRHTGATGRCGCGYAARAGA </td><td>20</td><td> 48</td><td>2</td><td> 19</td><td> 39</td><td>60.46714</td><td>2.6839022</td><td>51.66667</td><td>5.527708</td><td>27.836024</td><td>21.98608</td><td> 15.9462087</td><td>11.271684</td><td>91.379</td><td>Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis;Oscillibacter sp. PEA192</td><td>2</td><td> 48</td><td>50S ribosomal protein L20                        </td></tr>\n",
       "\t<tr><td>cds</td><td>13</td><td>265</td><td>216</td><td>216</td><td>0</td><td>265r</td><td>PRIMER_RIGHT</td><td>AVSTGGARTAGTTCATSCCGT</td><td>21</td><td> 24</td><td>2</td><td>214</td><td>235</td><td>60.08197</td><td>1.3894420</td><td>48.41270</td><td>3.272306</td><td>30.039833</td><td>16.01142</td><td>-27.4416464</td><td>14.647714</td><td>91.379</td><td>Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis;Oscillibacter sp. PEA192</td><td>2</td><td> 48</td><td>50S ribosomal protein L20                        </td></tr>\n",
       "\t<tr><td>cds</td><td>13</td><td>295</td><td>125</td><td>125</td><td>0</td><td>295f</td><td>PRIMER_LEFT </td><td>RSMVGTSATGAAGTCCCTGA </td><td>20</td><td> 48</td><td>1</td><td>110</td><td>130</td><td>59.84114</td><td>1.8496320</td><td>53.33333</td><td>4.249183</td><td>11.670800</td><td>18.48426</td><td> -7.5781759</td><td>28.282381</td><td>91.379</td><td>Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis;Oscillibacter sp. PEA192</td><td>2</td><td> 48</td><td>50S ribosomal protein L20                        </td></tr>\n",
       "\t<tr><td>cds</td><td>13</td><td>295</td><td>125</td><td>125</td><td>0</td><td>295r</td><td>PRIMER_RIGHT</td><td>AVSTGGARTAGTTCATSCCGT</td><td>21</td><td> 24</td><td>2</td><td>214</td><td>235</td><td>60.08197</td><td>1.3894420</td><td>48.41270</td><td>3.272306</td><td>30.039833</td><td>16.01142</td><td>-27.4416464</td><td>14.647714</td><td>91.379</td><td>Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis;Oscillibacter sp. PEA192</td><td>2</td><td> 48</td><td>50S ribosomal protein L20                        </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 30 × 27\n",
       "\\begin{tabular}{lllllllllllllllllllllllllll}\n",
       " gene\\_type & cluster\\_id & primer\\_set & amplicon\\_size\\_consensus & amplicon\\_size\\_avg & amplicon\\_size\\_sd & primer\\_id & primer\\_type & sequence & length & degeneracy & degeneracy\\_3prime & position\\_start & position\\_end & Tm\\_avg & Tm\\_sd & GC\\_avg & GC\\_sd & hairpin\\_avg & hairpin\\_sd & homodimer\\_avg & homodimer\\_sd & pident & sscinames & max\\_degeneracy\\_3prime & max\\_degeneracy & annotation\\\\\n",
       " <chr> & <int> & <int> & <int> & <dbl> & <dbl> & <fct> & <fct> & <fct> & <int> & <int> & <int> & <int> & <int> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <fct> & <int> & <int> & <fct>\\\\\n",
       "\\hline\n",
       "\t cds & 24 & 101 & 160 & 160 & 0 & 101f & PRIMER\\_LEFT  & TBTTCCGBGAYAAYTCCACS  & 20 &  72 & 2 & 196 & 216 & 59.40978 & 2.2201857 & 51.66667 & 4.859127 & 23.170159 & 23.39333 &   3.8426830 & 12.757158 & 88.413 & Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          & 2 &  72 & Putative carbamoyltransferase YgeW               \\\\\n",
       "\t cds & 24 & 101 & 160 & 160 & 0 & 101r & PRIMER\\_RIGHT & ATSACRTCVGCCATRAASGA  & 20 &  48 & 2 & 336 & 356 & 59.29837 & 2.1329283 & 48.33333 & 4.249183 & 37.237596 & 11.36689 &  -0.2072119 & 21.774421 & 88.413 & Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          & 2 &  72 & Putative carbamoyltransferase YgeW               \\\\\n",
       "\t cds & 24 & 925 & 159 & 159 & 0 & 925f & PRIMER\\_LEFT  & TBTTCCGBGAYAAYTCCACS  & 20 &  72 & 2 & 196 & 216 & 59.40978 & 2.2201857 & 51.66667 & 4.859127 & 23.170159 & 23.39333 &   3.8426830 & 12.757158 & 88.413 & Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          & 2 &  72 & Putative carbamoyltransferase YgeW               \\\\\n",
       "\t cds & 24 & 925 & 159 & 159 & 0 & 925r & PRIMER\\_RIGHT & TSACRTCVGCCATRAASGA   & 19 &  48 & 2 & 336 & 355 & 58.81608 & 2.2386673 & 50.87719 & 4.472824 & 37.237596 & 11.36689 &  -0.9210938 & 21.070817 & 88.413 & Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          & 2 &  72 & Putative carbamoyltransferase YgeW               \\\\\n",
       "\t cds & 24 & 252 & 205 & 205 & 0 & 252f & PRIMER\\_LEFT  & ACSGCSAACATGGTBTCSTT  & 20 &  24 & 2 & 321 & 341 & 62.52437 & 0.8325446 & 53.33333 & 2.357023 & 34.079778 & 15.71851 &   8.6676870 &  8.532017 & 88.413 & Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          & 2 &  96 & Putative carbamoyltransferase YgeW               \\\\\n",
       "\t cds & 24 & 252 & 205 & 205 & 0 & 252r & PRIMER\\_RIGHT & YRTGSAGCATRTCVGCCATR  & 20 &  96 & 2 & 506 & 526 & 60.37872 & 2.5635953 & 53.33333 & 5.527708 & 39.814057 & 14.18557 &   3.8855806 & 13.889514 & 88.413 & Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          & 2 &  96 & Putative carbamoyltransferase YgeW               \\\\\n",
       "\t cds & 24 & 852 & 135 & 135 & 0 & 852f & PRIMER\\_LEFT  & GBTTCTCVTTYGCCTCYGC   & 19 &  36 & 2 & 223 & 242 & 60.45917 & 2.1671838 & 59.64912 & 5.114870 &  9.565666 & 16.56822 & -52.4259519 &  8.602116 & 88.413 & Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          & 4 &  48 & Putative carbamoyltransferase YgeW               \\\\\n",
       "\t cds & 24 & 852 & 135 & 135 & 0 & 852r & PRIMER\\_RIGHT & CGATSACRTCVGCCATRAAS  & 20 &  48 & 4 & 338 & 358 & 59.09106 & 2.0362332 & 53.33333 & 4.249183 & 34.242766 & 16.18458 &   5.7775265 & 14.765420 & 88.413 & Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          & 4 &  48 & Putative carbamoyltransferase YgeW               \\\\\n",
       "\t cds & 24 & 962 & 133 & 133 & 0 & 962f & PRIMER\\_LEFT  & TTCTCVTTYGCCTCYGCST   & 19 &  24 & 4 & 225 & 244 & 61.47875 & 2.0007686 & 56.14035 & 4.472824 & 29.250102 & 23.92131 & -27.4141990 & 26.541144 & 88.413 & Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          & 4 &  48 & Putative carbamoyltransferase YgeW               \\\\\n",
       "\t cds & 24 & 962 & 133 & 133 & 0 & 962r & PRIMER\\_RIGHT & CGATSACRTCVGCCATRAAS  & 20 &  48 & 4 & 338 & 358 & 59.09106 & 2.0362332 & 53.33333 & 4.249183 & 34.242766 & 16.18458 &   5.7775265 & 14.765420 & 88.413 & Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          & 4 &  48 & Putative carbamoyltransferase YgeW               \\\\\n",
       "\t cds & 24 & 997 & 149 & 149 & 0 & 997f & PRIMER\\_LEFT  & CSTTYATGGCBGAYGTSATC  & 20 &  48 & 2 & 337 & 357 & 58.90509 & 2.0743193 & 53.33333 & 4.249183 & 29.056470 & 16.26418 &   1.3684155 & 16.456230 & 88.413 & Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          & 4 &  64 & Putative carbamoyltransferase YgeW               \\\\\n",
       "\t cds & 24 & 997 & 149 & 149 & 0 & 997r & PRIMER\\_RIGHT & RTCRCACTGSAGRTTYACMA  & 20 &  64 & 4 & 466 & 486 & 58.27347 & 2.5143072 & 47.50000 & 5.590170 & 15.209464 & 18.47412 &  -4.6224649 & 33.375152 & 88.413 & Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          & 4 &  64 & Putative carbamoyltransferase YgeW               \\\\\n",
       "\t cds & 24 &  34 & 162 & 162 & 0 & 34f  & PRIMER\\_LEFT  & TBTTCCGBGAYAAYTCCACS  & 20 &  72 & 2 & 196 & 216 & 59.40978 & 2.2201857 & 51.66667 & 4.859127 & 23.170159 & 23.39333 &   3.8426830 & 12.757158 & 88.413 & Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          & 4 &  72 & Putative carbamoyltransferase YgeW               \\\\\n",
       "\t cds & 24 &  34 & 162 & 162 & 0 & 34r  & PRIMER\\_RIGHT & CGATSACRTCVGCCATRAAS  & 20 &  48 & 4 & 338 & 358 & 59.09106 & 2.0362332 & 53.33333 & 4.249183 & 34.242766 & 16.18458 &   5.7775265 & 14.765420 & 88.413 & Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          & 4 &  72 & Putative carbamoyltransferase YgeW               \\\\\n",
       "\t cds & 24 & 949 & 137 & 137 & 0 & 949f & PRIMER\\_LEFT  & CBTGGGAYAAGASYGAYGA   & 19 &  48 & 2 &  79 &  98 & 57.43879 & 2.7226357 & 53.50877 & 5.189544 & 23.628755 & 22.96581 & -25.9893377 & 26.595037 & 88.413 & Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          & 4 &  72 & Putative carbamoyltransferase YgeW               \\\\\n",
       "\t cds & 24 & 949 & 137 & 137 & 0 & 949r & PRIMER\\_RIGHT & SGTGGARTTRTCVCGGAAVA  & 20 &  72 & 3 & 196 & 216 & 59.40978 & 2.2201857 & 51.66667 & 4.859127 & 32.446352 & 19.32925 &   2.2146646 & 13.318311 & 88.413 & Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          & 4 &  72 & Putative carbamoyltransferase YgeW               \\\\\n",
       "\t cds & 36 & 952 & 114 & 114 & 0 & 952f & PRIMER\\_LEFT  & YGARATYCAYCGSCTGAACC  & 20 &  32 & 1 & 356 & 376 & 60.32997 & 1.6997727 & 55.00000 & 5.000000 & 27.075589 & 21.15220 & -11.3804235 & 24.602761 & 89.118 & Clostridiales bacterium                                                                                                                        & 3 &  32 & Holliday junction ATP-dependent DNA helicase RuvB\\\\\n",
       "\t cds & 36 & 952 & 114 & 114 & 0 & 952r & PRIMER\\_RIGHT & TCSAGVCGGATGGAGTTBG   & 19 &  18 & 3 & 451 & 470 & 60.74049 & 1.6644375 & 59.64912 & 3.508772 & 28.880424 & 20.58490 &  -4.3656091 & 26.977200 & 89.118 & Clostridiales bacterium                                                                                                                        & 3 &  32 & Holliday junction ATP-dependent DNA helicase RuvB\\\\\n",
       "\t cds & 36 & 627 & 113 & 113 & 0 & 627f & PRIMER\\_LEFT  & GAYYTKGCKGCRCTKYTGA   & 19 & 128 & 4 & 303 & 322 & 60.57837 & 2.9830748 & 55.26316 & 6.962503 &  4.689779 & 12.40799 &  15.8095996 & 13.575059 & 89.118 & Clostridiales bacterium                                                                                                                        & 6 & 128 & Holliday junction ATP-dependent DNA helicase RuvB\\\\\n",
       "\t cds & 36 & 627 & 113 & 113 & 0 & 627r & PRIMER\\_RIGHT & WARTCCTCCATVGCVGGRTA  & 20 &  72 & 2 & 396 & 416 & 59.04885 & 2.4997399 & 51.66667 & 4.859127 & 28.991987 & 16.18955 &   3.0088796 & 20.436287 & 89.118 & Clostridiales bacterium                                                                                                                        & 6 & 128 & Holliday junction ATP-dependent DNA helicase RuvB\\\\\n",
       "\t cds &  1 & 468 & 189 & 189 & 0 & 468f & PRIMER\\_LEFT  & AARAACGGYCGYGAYTCYRA  & 20 &  64 & 4 &  54 &  74 & 60.76051 & 2.8483468 & 50.00000 & 6.123724 & 16.305789 & 21.51507 &  22.8243414 & 16.405838 & 91.011 & Oscillospiraceae bacterium                                                                                                                     & 4 &  96 & 50S ribosomal protein L27                        \\\\\n",
       "\t cds &  1 & 468 & 189 & 189 & 0 & 468r & PRIMER\\_RIGHT & RCCSAGVYGCTCRAARCG    & 18 &  96 & 2 & 225 & 243 & 62.16555 & 2.5748132 & 64.81481 & 6.141898 & 35.164559 & 27.78734 &   4.5654274 & 23.594207 & 91.011 & Oscillospiraceae bacterium                                                                                                                     & 4 &  96 & 50S ribosomal protein L27                        \\\\\n",
       "\t cds &  1 & 927 & 188 & 188 & 0 & 927f & PRIMER\\_LEFT  & ARAACGGYCGYGAYTCYRA   & 19 &  64 & 4 &  55 &  74 & 60.13246 & 2.9968465 & 52.63158 & 6.446026 & 16.118781 & 21.37324 &  22.7879011 & 16.410793 & 91.011 & Oscillospiraceae bacterium                                                                                                                     & 4 &  96 & 50S ribosomal protein L27                        \\\\\n",
       "\t cds &  1 & 927 & 188 & 188 & 0 & 927r & PRIMER\\_RIGHT & RCCSAGVYGCTCRAARCG    & 18 &  96 & 2 & 225 & 243 & 62.16555 & 2.5748132 & 64.81481 & 6.141898 & 35.164559 & 27.78734 &   4.5654274 & 23.594207 & 91.011 & Oscillospiraceae bacterium                                                                                                                     & 4 &  96 & 50S ribosomal protein L27                        \\\\\n",
       "\t cds &  8 & 780 & 126 & 126 & 0 & 780f & PRIMER\\_LEFT  & AARCCVCGVTAYCTBATGGG  & 20 & 108 & 1 & 738 & 758 & 60.35110 & 2.6283001 & 55.00000 & 5.400617 & 34.293877 & 14.01021 &   2.3512552 & 19.402971 & 91.247 & Dysosmobacter welbionis                                                                                                                        & 6 & 108 & Queuine tRNA-ribosyltransferase                  \\\\\n",
       "\t cds &  8 & 780 & 126 & 126 & 0 & 780r & PRIMER\\_RIGHT & CCAVGTRAABAGCTTBCCRT  & 20 & 108 & 6 & 844 & 864 & 58.71495 & 2.6684202 & 50.00000 & 5.400617 & 18.377478 & 19.37307 & -16.3267600 & 21.842517 & 91.247 & Dysosmobacter welbionis                                                                                                                        & 6 & 108 & Queuine tRNA-ribosyltransferase                  \\\\\n",
       "\t cds & 13 & 265 & 216 & 216 & 0 & 265f & PRIMER\\_LEFT  & CRHTGATGRCGCGYAARAGA  & 20 &  48 & 2 &  19 &  39 & 60.46714 & 2.6839022 & 51.66667 & 5.527708 & 27.836024 & 21.98608 &  15.9462087 & 11.271684 & 91.379 & Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis;Oscillibacter sp. PEA192 & 2 &  48 & 50S ribosomal protein L20                        \\\\\n",
       "\t cds & 13 & 265 & 216 & 216 & 0 & 265r & PRIMER\\_RIGHT & AVSTGGARTAGTTCATSCCGT & 21 &  24 & 2 & 214 & 235 & 60.08197 & 1.3894420 & 48.41270 & 3.272306 & 30.039833 & 16.01142 & -27.4416464 & 14.647714 & 91.379 & Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis;Oscillibacter sp. PEA192 & 2 &  48 & 50S ribosomal protein L20                        \\\\\n",
       "\t cds & 13 & 295 & 125 & 125 & 0 & 295f & PRIMER\\_LEFT  & RSMVGTSATGAAGTCCCTGA  & 20 &  48 & 1 & 110 & 130 & 59.84114 & 1.8496320 & 53.33333 & 4.249183 & 11.670800 & 18.48426 &  -7.5781759 & 28.282381 & 91.379 & Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis;Oscillibacter sp. PEA192 & 2 &  48 & 50S ribosomal protein L20                        \\\\\n",
       "\t cds & 13 & 295 & 125 & 125 & 0 & 295r & PRIMER\\_RIGHT & AVSTGGARTAGTTCATSCCGT & 21 &  24 & 2 & 214 & 235 & 60.08197 & 1.3894420 & 48.41270 & 3.272306 & 30.039833 & 16.01142 & -27.4416464 & 14.647714 & 91.379 & Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis;Oscillibacter sp. PEA192 & 2 &  48 & 50S ribosomal protein L20                        \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 30 × 27\n",
       "\n",
       "| gene_type &lt;chr&gt; | cluster_id &lt;int&gt; | primer_set &lt;int&gt; | amplicon_size_consensus &lt;int&gt; | amplicon_size_avg &lt;dbl&gt; | amplicon_size_sd &lt;dbl&gt; | primer_id &lt;fct&gt; | primer_type &lt;fct&gt; | sequence &lt;fct&gt; | length &lt;int&gt; | degeneracy &lt;int&gt; | degeneracy_3prime &lt;int&gt; | position_start &lt;int&gt; | position_end &lt;int&gt; | Tm_avg &lt;dbl&gt; | Tm_sd &lt;dbl&gt; | GC_avg &lt;dbl&gt; | GC_sd &lt;dbl&gt; | hairpin_avg &lt;dbl&gt; | hairpin_sd &lt;dbl&gt; | homodimer_avg &lt;dbl&gt; | homodimer_sd &lt;dbl&gt; | pident &lt;dbl&gt; | sscinames &lt;fct&gt; | max_degeneracy_3prime &lt;int&gt; | max_degeneracy &lt;int&gt; | annotation &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| cds | 24 | 101 | 160 | 160 | 0 | 101f | PRIMER_LEFT  | TBTTCCGBGAYAAYTCCACS  | 20 |  72 | 2 | 196 | 216 | 59.40978 | 2.2201857 | 51.66667 | 4.859127 | 23.170159 | 23.39333 |   3.8426830 | 12.757158 | 88.413 | Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          | 2 |  72 | Putative carbamoyltransferase YgeW                |\n",
       "| cds | 24 | 101 | 160 | 160 | 0 | 101r | PRIMER_RIGHT | ATSACRTCVGCCATRAASGA  | 20 |  48 | 2 | 336 | 356 | 59.29837 | 2.1329283 | 48.33333 | 4.249183 | 37.237596 | 11.36689 |  -0.2072119 | 21.774421 | 88.413 | Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          | 2 |  72 | Putative carbamoyltransferase YgeW                |\n",
       "| cds | 24 | 925 | 159 | 159 | 0 | 925f | PRIMER_LEFT  | TBTTCCGBGAYAAYTCCACS  | 20 |  72 | 2 | 196 | 216 | 59.40978 | 2.2201857 | 51.66667 | 4.859127 | 23.170159 | 23.39333 |   3.8426830 | 12.757158 | 88.413 | Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          | 2 |  72 | Putative carbamoyltransferase YgeW                |\n",
       "| cds | 24 | 925 | 159 | 159 | 0 | 925r | PRIMER_RIGHT | TSACRTCVGCCATRAASGA   | 19 |  48 | 2 | 336 | 355 | 58.81608 | 2.2386673 | 50.87719 | 4.472824 | 37.237596 | 11.36689 |  -0.9210938 | 21.070817 | 88.413 | Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          | 2 |  72 | Putative carbamoyltransferase YgeW                |\n",
       "| cds | 24 | 252 | 205 | 205 | 0 | 252f | PRIMER_LEFT  | ACSGCSAACATGGTBTCSTT  | 20 |  24 | 2 | 321 | 341 | 62.52437 | 0.8325446 | 53.33333 | 2.357023 | 34.079778 | 15.71851 |   8.6676870 |  8.532017 | 88.413 | Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          | 2 |  96 | Putative carbamoyltransferase YgeW                |\n",
       "| cds | 24 | 252 | 205 | 205 | 0 | 252r | PRIMER_RIGHT | YRTGSAGCATRTCVGCCATR  | 20 |  96 | 2 | 506 | 526 | 60.37872 | 2.5635953 | 53.33333 | 5.527708 | 39.814057 | 14.18557 |   3.8855806 | 13.889514 | 88.413 | Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          | 2 |  96 | Putative carbamoyltransferase YgeW                |\n",
       "| cds | 24 | 852 | 135 | 135 | 0 | 852f | PRIMER_LEFT  | GBTTCTCVTTYGCCTCYGC   | 19 |  36 | 2 | 223 | 242 | 60.45917 | 2.1671838 | 59.64912 | 5.114870 |  9.565666 | 16.56822 | -52.4259519 |  8.602116 | 88.413 | Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          | 4 |  48 | Putative carbamoyltransferase YgeW                |\n",
       "| cds | 24 | 852 | 135 | 135 | 0 | 852r | PRIMER_RIGHT | CGATSACRTCVGCCATRAAS  | 20 |  48 | 4 | 338 | 358 | 59.09106 | 2.0362332 | 53.33333 | 4.249183 | 34.242766 | 16.18458 |   5.7775265 | 14.765420 | 88.413 | Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          | 4 |  48 | Putative carbamoyltransferase YgeW                |\n",
       "| cds | 24 | 962 | 133 | 133 | 0 | 962f | PRIMER_LEFT  | TTCTCVTTYGCCTCYGCST   | 19 |  24 | 4 | 225 | 244 | 61.47875 | 2.0007686 | 56.14035 | 4.472824 | 29.250102 | 23.92131 | -27.4141990 | 26.541144 | 88.413 | Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          | 4 |  48 | Putative carbamoyltransferase YgeW                |\n",
       "| cds | 24 | 962 | 133 | 133 | 0 | 962r | PRIMER_RIGHT | CGATSACRTCVGCCATRAAS  | 20 |  48 | 4 | 338 | 358 | 59.09106 | 2.0362332 | 53.33333 | 4.249183 | 34.242766 | 16.18458 |   5.7775265 | 14.765420 | 88.413 | Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          | 4 |  48 | Putative carbamoyltransferase YgeW                |\n",
       "| cds | 24 | 997 | 149 | 149 | 0 | 997f | PRIMER_LEFT  | CSTTYATGGCBGAYGTSATC  | 20 |  48 | 2 | 337 | 357 | 58.90509 | 2.0743193 | 53.33333 | 4.249183 | 29.056470 | 16.26418 |   1.3684155 | 16.456230 | 88.413 | Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          | 4 |  64 | Putative carbamoyltransferase YgeW                |\n",
       "| cds | 24 | 997 | 149 | 149 | 0 | 997r | PRIMER_RIGHT | RTCRCACTGSAGRTTYACMA  | 20 |  64 | 4 | 466 | 486 | 58.27347 | 2.5143072 | 47.50000 | 5.590170 | 15.209464 | 18.47412 |  -4.6224649 | 33.375152 | 88.413 | Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          | 4 |  64 | Putative carbamoyltransferase YgeW                |\n",
       "| cds | 24 |  34 | 162 | 162 | 0 | 34f  | PRIMER_LEFT  | TBTTCCGBGAYAAYTCCACS  | 20 |  72 | 2 | 196 | 216 | 59.40978 | 2.2201857 | 51.66667 | 4.859127 | 23.170159 | 23.39333 |   3.8426830 | 12.757158 | 88.413 | Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          | 4 |  72 | Putative carbamoyltransferase YgeW                |\n",
       "| cds | 24 |  34 | 162 | 162 | 0 | 34r  | PRIMER_RIGHT | CGATSACRTCVGCCATRAAS  | 20 |  48 | 4 | 338 | 358 | 59.09106 | 2.0362332 | 53.33333 | 4.249183 | 34.242766 | 16.18458 |   5.7775265 | 14.765420 | 88.413 | Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          | 4 |  72 | Putative carbamoyltransferase YgeW                |\n",
       "| cds | 24 | 949 | 137 | 137 | 0 | 949f | PRIMER_LEFT  | CBTGGGAYAAGASYGAYGA   | 19 |  48 | 2 |  79 |  98 | 57.43879 | 2.7226357 | 53.50877 | 5.189544 | 23.628755 | 22.96581 | -25.9893377 | 26.595037 | 88.413 | Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          | 4 |  72 | Putative carbamoyltransferase YgeW                |\n",
       "| cds | 24 | 949 | 137 | 137 | 0 | 949r | PRIMER_RIGHT | SGTGGARTTRTCVCGGAAVA  | 20 |  72 | 3 | 196 | 216 | 59.40978 | 2.2201857 | 51.66667 | 4.859127 | 32.446352 | 19.32925 |   2.2146646 | 13.318311 | 88.413 | Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                          | 4 |  72 | Putative carbamoyltransferase YgeW                |\n",
       "| cds | 36 | 952 | 114 | 114 | 0 | 952f | PRIMER_LEFT  | YGARATYCAYCGSCTGAACC  | 20 |  32 | 1 | 356 | 376 | 60.32997 | 1.6997727 | 55.00000 | 5.000000 | 27.075589 | 21.15220 | -11.3804235 | 24.602761 | 89.118 | Clostridiales bacterium                                                                                                                        | 3 |  32 | Holliday junction ATP-dependent DNA helicase RuvB |\n",
       "| cds | 36 | 952 | 114 | 114 | 0 | 952r | PRIMER_RIGHT | TCSAGVCGGATGGAGTTBG   | 19 |  18 | 3 | 451 | 470 | 60.74049 | 1.6644375 | 59.64912 | 3.508772 | 28.880424 | 20.58490 |  -4.3656091 | 26.977200 | 89.118 | Clostridiales bacterium                                                                                                                        | 3 |  32 | Holliday junction ATP-dependent DNA helicase RuvB |\n",
       "| cds | 36 | 627 | 113 | 113 | 0 | 627f | PRIMER_LEFT  | GAYYTKGCKGCRCTKYTGA   | 19 | 128 | 4 | 303 | 322 | 60.57837 | 2.9830748 | 55.26316 | 6.962503 |  4.689779 | 12.40799 |  15.8095996 | 13.575059 | 89.118 | Clostridiales bacterium                                                                                                                        | 6 | 128 | Holliday junction ATP-dependent DNA helicase RuvB |\n",
       "| cds | 36 | 627 | 113 | 113 | 0 | 627r | PRIMER_RIGHT | WARTCCTCCATVGCVGGRTA  | 20 |  72 | 2 | 396 | 416 | 59.04885 | 2.4997399 | 51.66667 | 4.859127 | 28.991987 | 16.18955 |   3.0088796 | 20.436287 | 89.118 | Clostridiales bacterium                                                                                                                        | 6 | 128 | Holliday junction ATP-dependent DNA helicase RuvB |\n",
       "| cds |  1 | 468 | 189 | 189 | 0 | 468f | PRIMER_LEFT  | AARAACGGYCGYGAYTCYRA  | 20 |  64 | 4 |  54 |  74 | 60.76051 | 2.8483468 | 50.00000 | 6.123724 | 16.305789 | 21.51507 |  22.8243414 | 16.405838 | 91.011 | Oscillospiraceae bacterium                                                                                                                     | 4 |  96 | 50S ribosomal protein L27                         |\n",
       "| cds |  1 | 468 | 189 | 189 | 0 | 468r | PRIMER_RIGHT | RCCSAGVYGCTCRAARCG    | 18 |  96 | 2 | 225 | 243 | 62.16555 | 2.5748132 | 64.81481 | 6.141898 | 35.164559 | 27.78734 |   4.5654274 | 23.594207 | 91.011 | Oscillospiraceae bacterium                                                                                                                     | 4 |  96 | 50S ribosomal protein L27                         |\n",
       "| cds |  1 | 927 | 188 | 188 | 0 | 927f | PRIMER_LEFT  | ARAACGGYCGYGAYTCYRA   | 19 |  64 | 4 |  55 |  74 | 60.13246 | 2.9968465 | 52.63158 | 6.446026 | 16.118781 | 21.37324 |  22.7879011 | 16.410793 | 91.011 | Oscillospiraceae bacterium                                                                                                                     | 4 |  96 | 50S ribosomal protein L27                         |\n",
       "| cds |  1 | 927 | 188 | 188 | 0 | 927r | PRIMER_RIGHT | RCCSAGVYGCTCRAARCG    | 18 |  96 | 2 | 225 | 243 | 62.16555 | 2.5748132 | 64.81481 | 6.141898 | 35.164559 | 27.78734 |   4.5654274 | 23.594207 | 91.011 | Oscillospiraceae bacterium                                                                                                                     | 4 |  96 | 50S ribosomal protein L27                         |\n",
       "| cds |  8 | 780 | 126 | 126 | 0 | 780f | PRIMER_LEFT  | AARCCVCGVTAYCTBATGGG  | 20 | 108 | 1 | 738 | 758 | 60.35110 | 2.6283001 | 55.00000 | 5.400617 | 34.293877 | 14.01021 |   2.3512552 | 19.402971 | 91.247 | Dysosmobacter welbionis                                                                                                                        | 6 | 108 | Queuine tRNA-ribosyltransferase                   |\n",
       "| cds |  8 | 780 | 126 | 126 | 0 | 780r | PRIMER_RIGHT | CCAVGTRAABAGCTTBCCRT  | 20 | 108 | 6 | 844 | 864 | 58.71495 | 2.6684202 | 50.00000 | 5.400617 | 18.377478 | 19.37307 | -16.3267600 | 21.842517 | 91.247 | Dysosmobacter welbionis                                                                                                                        | 6 | 108 | Queuine tRNA-ribosyltransferase                   |\n",
       "| cds | 13 | 265 | 216 | 216 | 0 | 265f | PRIMER_LEFT  | CRHTGATGRCGCGYAARAGA  | 20 |  48 | 2 |  19 |  39 | 60.46714 | 2.6839022 | 51.66667 | 5.527708 | 27.836024 | 21.98608 |  15.9462087 | 11.271684 | 91.379 | Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis;Oscillibacter sp. PEA192 | 2 |  48 | 50S ribosomal protein L20                         |\n",
       "| cds | 13 | 265 | 216 | 216 | 0 | 265r | PRIMER_RIGHT | AVSTGGARTAGTTCATSCCGT | 21 |  24 | 2 | 214 | 235 | 60.08197 | 1.3894420 | 48.41270 | 3.272306 | 30.039833 | 16.01142 | -27.4416464 | 14.647714 | 91.379 | Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis;Oscillibacter sp. PEA192 | 2 |  48 | 50S ribosomal protein L20                         |\n",
       "| cds | 13 | 295 | 125 | 125 | 0 | 295f | PRIMER_LEFT  | RSMVGTSATGAAGTCCCTGA  | 20 |  48 | 1 | 110 | 130 | 59.84114 | 1.8496320 | 53.33333 | 4.249183 | 11.670800 | 18.48426 |  -7.5781759 | 28.282381 | 91.379 | Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis;Oscillibacter sp. PEA192 | 2 |  48 | 50S ribosomal protein L20                         |\n",
       "| cds | 13 | 295 | 125 | 125 | 0 | 295r | PRIMER_RIGHT | AVSTGGARTAGTTCATSCCGT | 21 |  24 | 2 | 214 | 235 | 60.08197 | 1.3894420 | 48.41270 | 3.272306 | 30.039833 | 16.01142 | -27.4416464 | 14.647714 | 91.379 | Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis;Oscillibacter sp. PEA192 | 2 |  48 | 50S ribosomal protein L20                         |\n",
       "\n"
      ],
      "text/plain": [
       "   gene_type cluster_id primer_set amplicon_size_consensus amplicon_size_avg\n",
       "1  cds       24         101        160                     160              \n",
       "2  cds       24         101        160                     160              \n",
       "3  cds       24         925        159                     159              \n",
       "4  cds       24         925        159                     159              \n",
       "5  cds       24         252        205                     205              \n",
       "6  cds       24         252        205                     205              \n",
       "7  cds       24         852        135                     135              \n",
       "8  cds       24         852        135                     135              \n",
       "9  cds       24         962        133                     133              \n",
       "10 cds       24         962        133                     133              \n",
       "11 cds       24         997        149                     149              \n",
       "12 cds       24         997        149                     149              \n",
       "13 cds       24          34        162                     162              \n",
       "14 cds       24          34        162                     162              \n",
       "15 cds       24         949        137                     137              \n",
       "16 cds       24         949        137                     137              \n",
       "17 cds       36         952        114                     114              \n",
       "18 cds       36         952        114                     114              \n",
       "19 cds       36         627        113                     113              \n",
       "20 cds       36         627        113                     113              \n",
       "21 cds        1         468        189                     189              \n",
       "22 cds        1         468        189                     189              \n",
       "23 cds        1         927        188                     188              \n",
       "24 cds        1         927        188                     188              \n",
       "25 cds        8         780        126                     126              \n",
       "26 cds        8         780        126                     126              \n",
       "27 cds       13         265        216                     216              \n",
       "28 cds       13         265        216                     216              \n",
       "29 cds       13         295        125                     125              \n",
       "30 cds       13         295        125                     125              \n",
       "   amplicon_size_sd primer_id primer_type  sequence              length\n",
       "1  0                101f      PRIMER_LEFT  TBTTCCGBGAYAAYTCCACS  20    \n",
       "2  0                101r      PRIMER_RIGHT ATSACRTCVGCCATRAASGA  20    \n",
       "3  0                925f      PRIMER_LEFT  TBTTCCGBGAYAAYTCCACS  20    \n",
       "4  0                925r      PRIMER_RIGHT TSACRTCVGCCATRAASGA   19    \n",
       "5  0                252f      PRIMER_LEFT  ACSGCSAACATGGTBTCSTT  20    \n",
       "6  0                252r      PRIMER_RIGHT YRTGSAGCATRTCVGCCATR  20    \n",
       "7  0                852f      PRIMER_LEFT  GBTTCTCVTTYGCCTCYGC   19    \n",
       "8  0                852r      PRIMER_RIGHT CGATSACRTCVGCCATRAAS  20    \n",
       "9  0                962f      PRIMER_LEFT  TTCTCVTTYGCCTCYGCST   19    \n",
       "10 0                962r      PRIMER_RIGHT CGATSACRTCVGCCATRAAS  20    \n",
       "11 0                997f      PRIMER_LEFT  CSTTYATGGCBGAYGTSATC  20    \n",
       "12 0                997r      PRIMER_RIGHT RTCRCACTGSAGRTTYACMA  20    \n",
       "13 0                34f       PRIMER_LEFT  TBTTCCGBGAYAAYTCCACS  20    \n",
       "14 0                34r       PRIMER_RIGHT CGATSACRTCVGCCATRAAS  20    \n",
       "15 0                949f      PRIMER_LEFT  CBTGGGAYAAGASYGAYGA   19    \n",
       "16 0                949r      PRIMER_RIGHT SGTGGARTTRTCVCGGAAVA  20    \n",
       "17 0                952f      PRIMER_LEFT  YGARATYCAYCGSCTGAACC  20    \n",
       "18 0                952r      PRIMER_RIGHT TCSAGVCGGATGGAGTTBG   19    \n",
       "19 0                627f      PRIMER_LEFT  GAYYTKGCKGCRCTKYTGA   19    \n",
       "20 0                627r      PRIMER_RIGHT WARTCCTCCATVGCVGGRTA  20    \n",
       "21 0                468f      PRIMER_LEFT  AARAACGGYCGYGAYTCYRA  20    \n",
       "22 0                468r      PRIMER_RIGHT RCCSAGVYGCTCRAARCG    18    \n",
       "23 0                927f      PRIMER_LEFT  ARAACGGYCGYGAYTCYRA   19    \n",
       "24 0                927r      PRIMER_RIGHT RCCSAGVYGCTCRAARCG    18    \n",
       "25 0                780f      PRIMER_LEFT  AARCCVCGVTAYCTBATGGG  20    \n",
       "26 0                780r      PRIMER_RIGHT CCAVGTRAABAGCTTBCCRT  20    \n",
       "27 0                265f      PRIMER_LEFT  CRHTGATGRCGCGYAARAGA  20    \n",
       "28 0                265r      PRIMER_RIGHT AVSTGGARTAGTTCATSCCGT 21    \n",
       "29 0                295f      PRIMER_LEFT  RSMVGTSATGAAGTCCCTGA  20    \n",
       "30 0                295r      PRIMER_RIGHT AVSTGGARTAGTTCATSCCGT 21    \n",
       "   degeneracy degeneracy_3prime position_start position_end Tm_avg   Tm_sd    \n",
       "1   72        2                 196            216          59.40978 2.2201857\n",
       "2   48        2                 336            356          59.29837 2.1329283\n",
       "3   72        2                 196            216          59.40978 2.2201857\n",
       "4   48        2                 336            355          58.81608 2.2386673\n",
       "5   24        2                 321            341          62.52437 0.8325446\n",
       "6   96        2                 506            526          60.37872 2.5635953\n",
       "7   36        2                 223            242          60.45917 2.1671838\n",
       "8   48        4                 338            358          59.09106 2.0362332\n",
       "9   24        4                 225            244          61.47875 2.0007686\n",
       "10  48        4                 338            358          59.09106 2.0362332\n",
       "11  48        2                 337            357          58.90509 2.0743193\n",
       "12  64        4                 466            486          58.27347 2.5143072\n",
       "13  72        2                 196            216          59.40978 2.2201857\n",
       "14  48        4                 338            358          59.09106 2.0362332\n",
       "15  48        2                  79             98          57.43879 2.7226357\n",
       "16  72        3                 196            216          59.40978 2.2201857\n",
       "17  32        1                 356            376          60.32997 1.6997727\n",
       "18  18        3                 451            470          60.74049 1.6644375\n",
       "19 128        4                 303            322          60.57837 2.9830748\n",
       "20  72        2                 396            416          59.04885 2.4997399\n",
       "21  64        4                  54             74          60.76051 2.8483468\n",
       "22  96        2                 225            243          62.16555 2.5748132\n",
       "23  64        4                  55             74          60.13246 2.9968465\n",
       "24  96        2                 225            243          62.16555 2.5748132\n",
       "25 108        1                 738            758          60.35110 2.6283001\n",
       "26 108        6                 844            864          58.71495 2.6684202\n",
       "27  48        2                  19             39          60.46714 2.6839022\n",
       "28  24        2                 214            235          60.08197 1.3894420\n",
       "29  48        1                 110            130          59.84114 1.8496320\n",
       "30  24        2                 214            235          60.08197 1.3894420\n",
       "   GC_avg   GC_sd    hairpin_avg hairpin_sd homodimer_avg homodimer_sd pident\n",
       "1  51.66667 4.859127 23.170159   23.39333     3.8426830   12.757158    88.413\n",
       "2  48.33333 4.249183 37.237596   11.36689    -0.2072119   21.774421    88.413\n",
       "3  51.66667 4.859127 23.170159   23.39333     3.8426830   12.757158    88.413\n",
       "4  50.87719 4.472824 37.237596   11.36689    -0.9210938   21.070817    88.413\n",
       "5  53.33333 2.357023 34.079778   15.71851     8.6676870    8.532017    88.413\n",
       "6  53.33333 5.527708 39.814057   14.18557     3.8855806   13.889514    88.413\n",
       "7  59.64912 5.114870  9.565666   16.56822   -52.4259519    8.602116    88.413\n",
       "8  53.33333 4.249183 34.242766   16.18458     5.7775265   14.765420    88.413\n",
       "9  56.14035 4.472824 29.250102   23.92131   -27.4141990   26.541144    88.413\n",
       "10 53.33333 4.249183 34.242766   16.18458     5.7775265   14.765420    88.413\n",
       "11 53.33333 4.249183 29.056470   16.26418     1.3684155   16.456230    88.413\n",
       "12 47.50000 5.590170 15.209464   18.47412    -4.6224649   33.375152    88.413\n",
       "13 51.66667 4.859127 23.170159   23.39333     3.8426830   12.757158    88.413\n",
       "14 53.33333 4.249183 34.242766   16.18458     5.7775265   14.765420    88.413\n",
       "15 53.50877 5.189544 23.628755   22.96581   -25.9893377   26.595037    88.413\n",
       "16 51.66667 4.859127 32.446352   19.32925     2.2146646   13.318311    88.413\n",
       "17 55.00000 5.000000 27.075589   21.15220   -11.3804235   24.602761    89.118\n",
       "18 59.64912 3.508772 28.880424   20.58490    -4.3656091   26.977200    89.118\n",
       "19 55.26316 6.962503  4.689779   12.40799    15.8095996   13.575059    89.118\n",
       "20 51.66667 4.859127 28.991987   16.18955     3.0088796   20.436287    89.118\n",
       "21 50.00000 6.123724 16.305789   21.51507    22.8243414   16.405838    91.011\n",
       "22 64.81481 6.141898 35.164559   27.78734     4.5654274   23.594207    91.011\n",
       "23 52.63158 6.446026 16.118781   21.37324    22.7879011   16.410793    91.011\n",
       "24 64.81481 6.141898 35.164559   27.78734     4.5654274   23.594207    91.011\n",
       "25 55.00000 5.400617 34.293877   14.01021     2.3512552   19.402971    91.247\n",
       "26 50.00000 5.400617 18.377478   19.37307   -16.3267600   21.842517    91.247\n",
       "27 51.66667 5.527708 27.836024   21.98608    15.9462087   11.271684    91.379\n",
       "28 48.41270 3.272306 30.039833   16.01142   -27.4416464   14.647714    91.379\n",
       "29 53.33333 4.249183 11.670800   18.48426    -7.5781759   28.282381    91.379\n",
       "30 48.41270 3.272306 30.039833   16.01142   -27.4416464   14.647714    91.379\n",
       "   sscinames                                                                                                                                     \n",
       "1  Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         \n",
       "2  Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         \n",
       "3  Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         \n",
       "4  Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         \n",
       "5  Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         \n",
       "6  Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         \n",
       "7  Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         \n",
       "8  Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         \n",
       "9  Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         \n",
       "10 Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         \n",
       "11 Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         \n",
       "12 Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         \n",
       "13 Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         \n",
       "14 Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         \n",
       "15 Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         \n",
       "16 Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis                         \n",
       "17 Clostridiales bacterium                                                                                                                       \n",
       "18 Clostridiales bacterium                                                                                                                       \n",
       "19 Clostridiales bacterium                                                                                                                       \n",
       "20 Clostridiales bacterium                                                                                                                       \n",
       "21 Oscillospiraceae bacterium                                                                                                                    \n",
       "22 Oscillospiraceae bacterium                                                                                                                    \n",
       "23 Oscillospiraceae bacterium                                                                                                                    \n",
       "24 Oscillospiraceae bacterium                                                                                                                    \n",
       "25 Dysosmobacter welbionis                                                                                                                       \n",
       "26 Dysosmobacter welbionis                                                                                                                       \n",
       "27 Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis;Oscillibacter sp. PEA192\n",
       "28 Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis;Oscillibacter sp. PEA192\n",
       "29 Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis;Oscillibacter sp. PEA192\n",
       "30 Oscillospiraceae;uncultured Blautia sp.;Oscillibacter sp. KLE 1728;Oscillibacter sp. KLE 1745;Dysosmobacter welbionis;Oscillibacter sp. PEA192\n",
       "   max_degeneracy_3prime max_degeneracy\n",
       "1  2                      72           \n",
       "2  2                      72           \n",
       "3  2                      72           \n",
       "4  2                      72           \n",
       "5  2                      96           \n",
       "6  2                      96           \n",
       "7  4                      48           \n",
       "8  4                      48           \n",
       "9  4                      48           \n",
       "10 4                      48           \n",
       "11 4                      64           \n",
       "12 4                      64           \n",
       "13 4                      72           \n",
       "14 4                      72           \n",
       "15 4                      72           \n",
       "16 4                      72           \n",
       "17 3                      32           \n",
       "18 3                      32           \n",
       "19 6                     128           \n",
       "20 6                     128           \n",
       "21 4                      96           \n",
       "22 4                      96           \n",
       "23 4                      96           \n",
       "24 4                      96           \n",
       "25 6                     108           \n",
       "26 6                     108           \n",
       "27 2                      48           \n",
       "28 2                      48           \n",
       "29 2                      48           \n",
       "30 2                      48           \n",
       "   annotation                                       \n",
       "1  Putative carbamoyltransferase YgeW               \n",
       "2  Putative carbamoyltransferase YgeW               \n",
       "3  Putative carbamoyltransferase YgeW               \n",
       "4  Putative carbamoyltransferase YgeW               \n",
       "5  Putative carbamoyltransferase YgeW               \n",
       "6  Putative carbamoyltransferase YgeW               \n",
       "7  Putative carbamoyltransferase YgeW               \n",
       "8  Putative carbamoyltransferase YgeW               \n",
       "9  Putative carbamoyltransferase YgeW               \n",
       "10 Putative carbamoyltransferase YgeW               \n",
       "11 Putative carbamoyltransferase YgeW               \n",
       "12 Putative carbamoyltransferase YgeW               \n",
       "13 Putative carbamoyltransferase YgeW               \n",
       "14 Putative carbamoyltransferase YgeW               \n",
       "15 Putative carbamoyltransferase YgeW               \n",
       "16 Putative carbamoyltransferase YgeW               \n",
       "17 Holliday junction ATP-dependent DNA helicase RuvB\n",
       "18 Holliday junction ATP-dependent DNA helicase RuvB\n",
       "19 Holliday junction ATP-dependent DNA helicase RuvB\n",
       "20 Holliday junction ATP-dependent DNA helicase RuvB\n",
       "21 50S ribosomal protein L27                        \n",
       "22 50S ribosomal protein L27                        \n",
       "23 50S ribosomal protein L27                        \n",
       "24 50S ribosomal protein L27                        \n",
       "25 Queuine tRNA-ribosyltransferase                  \n",
       "26 Queuine tRNA-ribosyltransferase                  \n",
       "27 50S ribosomal protein L20                        \n",
       "28 50S ribosomal protein L20                        \n",
       "29 50S ribosomal protein L20                        \n",
       "30 50S ribosomal protein L20                        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# joining\n",
    "primer_info_jfj = primer_info_jf %>%\n",
    "    left_join(gene_annot %>% distinct(gene_type, cluster_id, annotation),\n",
    "              c('gene_type', 'cluster_id'))\n",
    "\n",
    "df.dims(30,50)\n",
    "primer_info_jfj\n",
    "df.dims()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Writing table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-23T08:05:20.857491Z",
     "iopub.status.busy": "2021-04-23T08:05:20.847256Z",
     "iopub.status.idle": "2021-04-23T08:05:20.876460Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "File written: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3//Oscillibacter/best_primers.tsv \n"
     ]
    }
   ],
   "source": [
    "F = file.path(work_dir, 'best_primers.tsv')\n",
    "write_table(primer_info_jfj, F)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# sessionInfo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-23T08:05:22.854468Z",
     "iopub.status.busy": "2021-04-23T08:05:22.852379Z",
     "iopub.status.idle": "2021-04-23T08:05:22.936059Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LLG\n",
      "===\n",
      "\n",
      "Ley Lab Genome analysis pipeline (LLG)\n",
      "\n",
      "* Version: 0.2.4\n",
      "* Authors:\n",
      "  * Nick Youngblut <nyoungb2@gmail.com>\n",
      "* Maintainers:\n",
      "  * Nick Youngblut <nyoungb2@gmail.com>\n",
      "\n",
      "--- conda envs ---\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/gtdbtk.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- conda-forge::networkx\n",
      "- bioconda::gtdbtk\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/checkm.yaml <==\n",
      "channels:\n",
      "- bioconda\n",
      "dependencies:\n",
      "- python=2.7\n",
      "- pigz\n",
      "- bioconda::prodigal\n",
      "- bioconda::pplacer\n",
      "- bioconda::checkm-genome\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/quast.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- bioconda::seqkit\n",
      "- bioconda::quast>=5.0.0\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/sourmash.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- bioconda::sourmash=2.0.0a4\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/fastqc.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- bioconda::seqtk\n",
      "- bioconda::fastqc=0.11.7\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/multiqc.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- bioconda::multiqc=1.4\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/hadley.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- conda-forge::r-ape\n",
      "- conda-forge::r-dplyr\n",
      "- conda-forge::r-tidyr\n",
      "- conda-forge::r-ggplot2\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/bbmap.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::bbmap=37.78\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/ani.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::dashing\n",
      "- bioconda::fastani\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/prokka.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- bioconda::prokka\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/prodigal.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- bioconda::prodigal\n",
      "- pip:\n",
      "  - biopython-convert\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/eggnog-mapper.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::eggnog-mapper\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/phylophlan.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::phylophlan\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/drep.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::mash\n",
      "- bioconda::checkm-genome\n",
      "- bioconda::fastani\n",
      "- bioconda::drep\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/antismash.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- biopython=1.77\n",
      "- bioconda::blast=2.9\n",
      "- bioconda::antismash>=5.1.2\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/deepbgc.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::deepbgc\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/abricate.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- bioconda::abricate\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/rRNA.yaml <==\n",
      "channels:\n",
      "- qiime2/label/r2020.2\n",
      "- conda-forge\n",
      "- bioconda\n",
      "- defaults\n",
      "dependencies:\n",
      "- barrnap\n",
      "- _libgcc_mutex=0.1\n",
      "- _openmp_mutex=4.5\n",
      "- _r-mutex=1.0.1\n",
      "- alsa-lib=1.1.5\n",
      "- arb-bio-tools=6.0.6\n",
      "- attrs=19.3.0\n",
      "- backcall=0.1.0\n",
      "- bibtexparser=1.1.0\n",
      "- binutils_impl_linux-64=2.33.1\n",
      "- binutils_linux-64=2.33.1\n",
      "- bioconductor-biobase=2.42.0\n",
      "- bioconductor-biocgenerics=0.28.0\n",
      "- bioconductor-biocparallel=1.16.6\n",
      "- bioconductor-biostrings=2.50.2\n",
      "- bioconductor-dada2=1.10.0\n",
      "- bioconductor-delayedarray=0.8.0\n",
      "- bioconductor-genomeinfodb=1.18.1\n",
      "- bioconductor-genomeinfodbdata=1.2.1\n",
      "- bioconductor-genomicalignments=1.18.1\n",
      "- bioconductor-genomicranges=1.34.0\n",
      "- bioconductor-iranges=2.16.0\n",
      "- bioconductor-rsamtools=1.34.0\n",
      "- bioconductor-s4vectors=0.20.1\n",
      "- bioconductor-shortread=1.40.0\n",
      "- bioconductor-summarizedexperiment=1.12.0\n",
      "- bioconductor-xvector=0.22.0\n",
      "- bioconductor-zlibbioc=1.28.0\n",
      "- biom-format=2.1.8\n",
      "- blas=2.14\n",
      "- blast=2.9.0\n",
      "- bleach=3.1.1\n",
      "- bokeh=1.4.0\n",
      "- boost=1.68.0\n",
      "- boost-cpp=1.68.0\n",
      "- bwidget=1.9.14\n",
      "- bzip2=1.0.8\n",
      "- ca-certificates=2019.11.28\n",
      "- cachecontrol=0.12.5\n",
      "- cairo=1.16.0\n",
      "- certifi=2019.11.28\n",
      "- cffi=1.13.2\n",
      "- chardet=3.0.4\n",
      "- click=7.0\n",
      "- cryptography=2.8\n",
      "- curl=7.68.0\n",
      "- cutadapt=2.8\n",
      "- cycler=0.10.0\n",
      "- cython=0.29.15\n",
      "- dbus=1.13.6\n",
      "- deblur=1.1.0\n",
      "- decorator=4.4.1\n",
      "- defusedxml=0.6.0\n",
      "- dendropy=4.4.0\n",
      "- dnaio=0.4.1\n",
      "- emperor=1.0.0\n",
      "- entrypoints=0.3\n",
      "- expat=2.2.9\n",
      "- fastcluster=1.1.26\n",
      "- fasttree=2.1.10\n",
      "- fontconfig=2.13.1\n",
      "- freetype=2.10.0\n",
      "- future=0.18.2\n",
      "- gcc_impl_linux-64=7.3.0\n",
      "- gcc_linux-64=7.3.0\n",
      "- gettext=0.19.8.1\n",
      "- gfortran_impl_linux-64=7.3.0\n",
      "- gfortran_linux-64=7.3.0\n",
      "- giflib=5.2.1\n",
      "- glib=2.58.3\n",
      "- gmp=6.2.0\n",
      "- gneiss=0.4.6\n",
      "- gnutls=3.6.5\n",
      "- graphite2=1.3.13\n",
      "- gsl=2.5\n",
      "- gst-plugins-base=1.14.5\n",
      "- gstreamer=1.14.5\n",
      "- gxx_impl_linux-64=7.3.0\n",
      "- gxx_linux-64=7.3.0\n",
      "- h5py=2.10.0\n",
      "- harfbuzz=2.4.0\n",
      "- hdf5=1.10.5\n",
      "- hdmedians=0.13\n",
      "- hmmer=3.1b2\n",
      "- icu=58.2\n",
      "- idna=2.9\n",
      "- ijson=2.6.1\n",
      "- importlib_metadata=1.5.0\n",
      "- ipykernel=5.1.4\n",
      "- ipython=7.12.0\n",
      "- ipython_genutils=0.2.0\n",
      "- ipywidgets=7.5.1\n",
      "- iqtree=1.6.12\n",
      "- jedi=0.16.0\n",
      "- jinja2=2.11.1\n",
      "- joblib=0.14.1\n",
      "- jpeg=9c\n",
      "- jsonschema=3.2.0\n",
      "- jupyter_client=6.0.0\n",
      "- jupyter_core=4.6.3\n",
      "- kiwisolver=1.1.0\n",
      "- krb5=1.16.4\n",
      "- lcms2=2.9\n",
      "- ld_impl_linux-64=2.33.1\n",
      "- libarbdb=6.0.6\n",
      "- libblas=3.8.0\n",
      "- libcblas=3.8.0\n",
      "- libcurl=7.68.0\n",
      "- libedit=3.1.20170329\n",
      "- libffi=3.2.1\n",
      "- libgcc=7.2.0\n",
      "- libgcc-ng=9.2.0\n",
      "- libgfortran-ng=7.3.0\n",
      "- libgomp=9.2.0\n",
      "- libiconv=1.15\n",
      "- liblapack=3.8.0\n",
      "- liblapacke=3.8.0\n",
      "- libopenblas=0.3.7\n",
      "- libpng=1.6.37\n",
      "- libsodium=1.0.17\n",
      "- libssh2=1.8.2\n",
      "- libstdcxx-ng=9.2.0\n",
      "- libtiff=4.1.0\n",
      "- libuuid=2.32.1\n",
      "- libxcb=1.13\n",
      "- libxml2=2.9.9\n",
      "- lockfile=0.12.2\n",
      "- lz4-c=1.8.3\n",
      "- mafft=7.310\n",
      "- make=4.3\n",
      "- markupsafe=1.1.1\n",
      "- matplotlib=3.1.1\n",
      "- matplotlib-base=3.1.1\n",
      "- mistune=0.8.4\n",
      "- more-itertools=8.2.0\n",
      "- msgpack-python=1.0.0\n",
      "- natsort=7.0.1\n",
      "- nbconvert=5.6.1\n",
      "- nbformat=5.0.4\n",
      "- ncurses=6.1\n",
      "- nettle=3.4.1\n",
      "- networkx=2.4\n",
      "- nose=1.3.7\n",
      "- notebook=6.0.3\n",
      "- numpy=1.18.1\n",
      "- olefile=0.46\n",
      "- openjdk=11.0.1\n",
      "- openssl=1.1.1d\n",
      "- packaging=20.1\n",
      "- pandas=0.25.3\n",
      "- pandoc=2.9.2\n",
      "- pandocfilters=1.4.2\n",
      "- pango=1.40.14\n",
      "- parso=0.6.1\n",
      "- patsy=0.5.1\n",
      "- pcre=8.44\n",
      "- perl=5.26.2\n",
      "- perl-archive-tar=2.32\n",
      "- perl-carp=1.38\n",
      "- perl-common-sense=3.74\n",
      "- perl-compress-raw-bzip2=2.087\n",
      "- perl-compress-raw-zlib=2.087\n",
      "- perl-exporter=5.72\n",
      "- perl-exporter-tiny=1.002001\n",
      "- perl-extutils-makemaker=7.36\n",
      "- perl-io-compress=2.087\n",
      "- perl-io-zlib=1.10\n",
      "- perl-json=4.02\n",
      "- perl-json-xs=2.34\n",
      "- perl-list-moreutils=0.428\n",
      "- perl-list-moreutils-xs=0.428\n",
      "- perl-pathtools=3.75\n",
      "- perl-scalar-list-utils=1.52\n",
      "- perl-types-serialiser=1.0\n",
      "- perl-xsloader=0.24\n",
      "- pexpect=4.8.0\n",
      "- pickleshare=0.7.5\n",
      "- pigz=2.3.4\n",
      "- pillow=7.0.0\n",
      "- pip=20.0.2\n",
      "- pixman=0.38.0\n",
      "- pluggy=0.12.0\n",
      "- prometheus_client=0.7.1\n",
      "- prompt_toolkit=3.0.3\n",
      "- psutil=5.7.0\n",
      "- pthread-stubs=0.4\n",
      "- ptyprocess=0.6.0\n",
      "- py=1.8.1\n",
      "- pycparser=2.19\n",
      "- pygments=2.5.2\n",
      "- pyopenssl=19.1.0\n",
      "- pyparsing=2.4.6\n",
      "- pyqt=5.9.2\n",
      "- pyrsistent=0.15.7\n",
      "- pysocks=1.7.1\n",
      "- pytest=5.3.5\n",
      "- python=3.6.7\n",
      "- python-dateutil=2.8.1\n",
      "- pytz=2019.3\n",
      "- pyyaml=5.3\n",
      "- pyzmq=19.0.0\n",
      "- q2-alignment=2020.2.0\n",
      "- q2-composition=2020.2.0\n",
      "- q2-cutadapt=2020.2.0\n",
      "- q2-dada2=2020.2.0\n",
      "- q2-deblur=2020.2.0\n",
      "- q2-demux=2020.2.0\n",
      "- q2-diversity=2020.2.0\n",
      "- q2-emperor=2020.2.0\n",
      "- q2-feature-classifier=2020.2.0\n",
      "- q2-feature-table=2020.2.0\n",
      "- q2-fragment-insertion=2020.2.0\n",
      "- q2-gneiss=2020.2.0\n",
      "- q2-longitudinal=2020.2.0\n",
      "- q2-metadata=2020.2.0\n",
      "- q2-phylogeny=2020.2.0\n",
      "- q2-quality-control=2020.2.0\n",
      "- q2-quality-filter=2020.2.0\n",
      "- q2-sample-classifier=2020.2.0\n",
      "- q2-taxa=2020.2.0\n",
      "- q2-types=2020.2.0\n",
      "- q2-vsearch=2020.2.0\n",
      "- q2cli=2020.2.0\n",
      "- q2templates=2020.2.0\n",
      "- qiime2=2020.2.0\n",
      "- qt=5.9.7\n",
      "- r-assertthat=0.2.1\n",
      "- r-backports=1.1.5\n",
      "- r-base=3.5.1\n",
      "- r-bh=1.72.0_3\n",
      "- r-bitops=1.0_6\n",
      "- r-cli=2.0.2\n",
      "- r-cluster=2.1.0\n",
      "- r-colorspace=1.4_1\n",
      "- r-crayon=1.3.4\n",
      "- r-data.table=1.12.6\n",
      "- r-digest=0.6.25\n",
      "- r-ellipsis=0.3.0\n",
      "- r-fansi=0.4.1\n",
      "- r-farver=2.0.3\n",
      "- r-formatr=1.7\n",
      "- r-futile.logger=1.4.3\n",
      "- r-futile.options=1.0.1\n",
      "- r-ggplot2=3.2.1\n",
      "- r-glue=1.3.1\n",
      "- r-gtable=0.3.0\n",
      "- r-hwriter=1.3.2\n",
      "- r-labeling=0.3\n",
      "- r-lambda.r=1.2.4\n",
      "- r-lattice=0.20_40\n",
      "- r-latticeextra=0.6_28\n",
      "- r-lazyeval=0.2.2\n",
      "- r-lifecycle=0.1.0\n",
      "- r-magrittr=1.5\n",
      "- r-mass=7.3_51.5\n",
      "- r-matrix=1.2_18\n",
      "- r-matrixstats=0.55.0\n",
      "- r-mgcv=1.8_31\n",
      "- r-munsell=0.5.0\n",
      "- r-nlme=3.1_144\n",
      "- r-permute=0.9_5\n",
      "- r-pillar=1.4.3\n",
      "- r-pkgconfig=2.0.3\n",
      "- r-plyr=1.8.5\n",
      "- r-r6=2.4.1\n",
      "- r-rcolorbrewer=1.1_2\n",
      "- r-rcpp=1.0.3\n",
      "- r-rcppparallel=4.4.4\n",
      "- r-rcurl=1.98_1.1\n",
      "- r-reshape2=1.4.3\n",
      "- r-rlang=0.4.4\n",
      "- r-scales=1.1.0\n",
      "- r-snow=0.4_3\n",
      "- r-stringi=1.4.3\n",
      "- r-stringr=1.4.0\n",
      "- r-tibble=2.1.3\n",
      "- r-utf8=1.1.4\n",
      "- r-vctrs=0.2.3\n",
      "- r-vegan=2.5_6\n",
      "- r-viridislite=0.3.0\n",
      "- r-withr=2.1.2\n",
      "- r-zeallot=0.1.0\n",
      "- raxml=8.2.12\n",
      "- readline=8.0\n",
      "- requests=2.23.0\n",
      "- scikit-bio=0.5.5\n",
      "- scikit-learn=0.22.1\n",
      "- scipy=1.4.1\n",
      "- seaborn=0.10.0\n",
      "- send2trash=1.5.0\n",
      "- sepp=4.3.10\n",
      "- setuptools=45.2.0\n",
      "- sina=1.6.0\n",
      "- sip=4.19.8\n",
      "- six=1.14.0\n",
      "- sortmerna=2.0\n",
      "- sqlite=3.30.1\n",
      "- statsmodels=0.11.1\n",
      "- tbb=2019.9\n",
      "- terminado=0.8.3\n",
      "- testpath=0.4.4\n",
      "- tk=8.6.10\n",
      "- tktable=2.10\n",
      "- tornado=6.0.3\n",
      "- traitlets=4.3.3\n",
      "- tzlocal=2.0.0\n",
      "- unifrac=0.10.0\n",
      "- urllib3=1.25.7\n",
      "- vsearch=2.7.0\n",
      "- wcwidth=0.1.8\n",
      "- webencodings=0.5.1\n",
      "- wheel=0.34.2\n",
      "- widgetsnbextension=3.5.1\n",
      "- xopen=0.8.4\n",
      "- xorg-fixesproto=5.0\n",
      "- xorg-inputproto=2.3.2\n",
      "- xorg-kbproto=1.0.7\n",
      "- xorg-libice=1.0.10\n",
      "- xorg-libsm=1.2.3\n",
      "- xorg-libx11=1.6.9\n",
      "- xorg-libxau=1.0.9\n",
      "- xorg-libxdmcp=1.1.3\n",
      "- xorg-libxext=1.3.4\n",
      "- xorg-libxfixes=5.0.3\n",
      "- xorg-libxi=1.7.10\n",
      "- xorg-libxrender=0.9.10\n",
      "- xorg-libxtst=1.2.3\n",
      "- xorg-recordproto=1.14.2\n",
      "- xorg-renderproto=0.11.1\n",
      "- xorg-xextproto=7.3.0\n",
      "- xorg-xproto=7.0.31\n",
      "- xz=5.2.4\n",
      "- yaml=0.2.2\n",
      "- zeromq=4.3.2\n",
      "- zipp=3.0.0\n",
      "- zlib=1.2.11\n",
      "- zstd=1.4.4\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/traitar.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- scipy\n",
      "- pandas\n",
      "- numpy\n",
      "- matplotlib\n",
      "- pigz\n",
      "- bioconda::hmmer\n",
      "- pip:\n",
      "  - traitar3>=0.1.1\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/cctyper.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "- russel88\n",
      "dependencies:\n",
      "- pigz\n",
      "- cctyper\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/comparem.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::comparem\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/bigscape.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- python=3.6\n",
      "- hmmer\n",
      "- biopython\n",
      "- fasttree\n",
      "- numpy\n",
      "- scipy\n",
      "- networkx\n",
      "- scikit-learn=0.19.1\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/rgi.yaml <==\n",
      "name: rgi\n",
      "channels:\n",
      "  - bioconda\n",
      "  - conda-forge\n",
      "  - defaults\n",
      "dependencies:\n",
      "  - _libgcc_mutex=0.1=conda_forge\n",
      "  - _openmp_mutex=4.5=1_llvm\n",
      "  - bamtools=2.5.1=he860b03_5\n",
      "  - bedtools=2.29.2=hc088bd4_0\n",
      "  - biopython=1.72=py36h14c3975_1000\n",
      "  - blast=2.9.0=pl526h3066fca_4\n",
      "  - bowtie2=2.4.1=py36h7f0b59b_2\n",
      "  - bwa=0.7.17=hed695b0_7\n",
      "  - bzip2=1.0.8=h516909a_2\n",
      "  - ca-certificates=2020.4.5.2=hecda079_0\n",
      "  - certifi=2020.4.5.2=py36h9f0ad1d_0\n",
      "  - curl=7.69.1=h33f0ec9_0\n",
      "  - cycler=0.10.0=py_2\n",
      "  - diamond=0.8.36=h8b12597_4\n",
      "  - entrez-direct=13.3=pl526h375a9b1_0\n",
      "  - expat=2.2.9=he1b5a44_2\n",
      "  - filetype=1.0.7=pyh9f0ad1d_0\n",
      "  - freetype=2.10.2=he06d7ca_0\n",
      "  - icu=64.2=he1b5a44_1\n",
      "  - kiwisolver=1.2.0=py36hdb11119_0\n",
      "  - krb5=1.17.1=h2fd8d38_0\n",
      "  - ld_impl_linux-64=2.34=h53a641e_5\n",
      "  - libblas=3.8.0=16_openblas\n",
      "  - libcblas=3.8.0=16_openblas\n",
      "  - libcurl=7.69.1=hf7181ac_0\n",
      "  - libdeflate=1.0=h14c3975_1\n",
      "  - libedit=3.1.20191231=h46ee950_0\n",
      "  - libffi=3.2.1=he1b5a44_1007\n",
      "  - libgcc-ng=9.2.0=h24d8f2e_2\n",
      "  - libgfortran-ng=7.5.0\n",
      "  - liblapack=3.8.0=16_openblas\n",
      "  - libopenblas=0.3.9=h5ec1e0e_0\n",
      "  - libpng=1.6.37=hed695b0_1\n",
      "  - libssh2=1.9.0=hab1572f_2\n",
      "  - libstdcxx-ng=9.2.0=hdf63c60_2\n",
      "  - llvm-openmp=10.0.0=hc9558a2_0\n",
      "  - matplotlib-base=3.2.1=py36hb8e4980_0\n",
      "  - ncurses=6.1=hf484d3e_1002\n",
      "  - numpy=1.18.5=py36h7314795_0\n",
      "  - oligoarrayaux=3.8=hc9558a2_0\n",
      "  - openssl=1.1.1g=h516909a_0\n",
      "  - pandas=1.0.4=py36h830a2c2_0\n",
      "  - patsy=0.5.1=py_0\n",
      "  - pcre=8.44=he1b5a44_0\n",
      "  - perl=5.26.2=h516909a_1006\n",
      "  - perl-app-cpanminus=1.7044=pl526_1\n",
      "  - perl-archive-tar=2.32=pl526_0\n",
      "  - perl-base=2.23=pl526_1\n",
      "  - perl-business-isbn=3.004=pl526_0\n",
      "  - perl-business-isbn-data=20140910.003=pl526_0\n",
      "  - perl-carp=1.38=pl526_3\n",
      "  - perl-common-sense=3.74=pl526_2\n",
      "  - perl-compress-raw-bzip2=2.087=pl526he1b5a44_0\n",
      "  - perl-compress-raw-zlib=2.087=pl526hc9558a2_0\n",
      "  - perl-constant=1.33=pl526_1\n",
      "  - perl-data-dumper=2.173=pl526_0\n",
      "  - perl-digest-hmac=1.03=pl526_3\n",
      "  - perl-digest-md5=2.55=pl526_0\n",
      "  - perl-encode=2.88=pl526_1\n",
      "  - perl-encode-locale=1.05=pl526_6\n",
      "  - perl-exporter=5.72=pl526_1\n",
      "  - perl-exporter-tiny=1.002001=pl526_0\n",
      "  - perl-extutils-makemaker=7.36=pl526_1\n",
      "  - perl-file-listing=6.04=pl526_1\n",
      "  - perl-file-path=2.16=pl526_0\n",
      "  - perl-file-temp=0.2304=pl526_2\n",
      "  - perl-html-parser=3.72=pl526h6bb024c_5\n",
      "  - perl-html-tagset=3.20=pl526_3\n",
      "  - perl-html-tree=5.07=pl526_1\n",
      "  - perl-http-cookies=6.04=pl526_0\n",
      "  - perl-http-daemon=6.01=pl526_1\n",
      "  - perl-http-date=6.02=pl526_3\n",
      "  - perl-http-message=6.18=pl526_0\n",
      "  - perl-http-negotiate=6.01=pl526_3\n",
      "  - perl-io-compress=2.087=pl526he1b5a44_0\n",
      "  - perl-io-html=1.001=pl526_2\n",
      "  - perl-io-socket-ssl=2.066=pl526_0\n",
      "  - perl-io-zlib=1.10=pl526_2\n",
      "  - perl-json=4.02=pl526_0\n",
      "  - perl-json-xs=2.34=pl526h6bb024c_3\n",
      "  - perl-libwww-perl=6.39=pl526_0\n",
      "  - perl-list-moreutils=0.428=pl526_1\n",
      "  - perl-list-moreutils-xs=0.428=pl526_0\n",
      "  - perl-lwp-mediatypes=6.04=pl526_0\n",
      "  - perl-lwp-protocol-https=6.07=pl526_4\n",
      "  - perl-mime-base64=3.15=pl526_1\n",
      "  - perl-mozilla-ca=20180117=pl526_1\n",
      "  - perl-net-http=6.19=pl526_0\n",
      "  - perl-net-ssleay=1.88=pl526h90d6eec_0\n",
      "  - perl-ntlm=1.09=pl526_4\n",
      "  - perl-parent=0.236=pl526_1\n",
      "  - perl-pathtools=3.75=pl526h14c3975_1\n",
      "  - perl-scalar-list-utils=1.52=pl526h516909a_0\n",
      "  - perl-socket=2.027=pl526_1\n",
      "  - perl-storable=3.15=pl526h14c3975_0\n",
      "  - perl-test-requiresinternet=0.05=pl526_0\n",
      "  - perl-time-local=1.28=pl526_1\n",
      "  - perl-try-tiny=0.30=pl526_1\n",
      "  - perl-types-serialiser=1.0=pl526_2\n",
      "  - perl-uri=1.76=pl526_0\n",
      "  - perl-www-robotrules=6.02=pl526_3\n",
      "  - perl-xml-namespacesupport=1.12=pl526_0\n",
      "  - perl-xml-parser=2.44=pl526h4e0c4b3_7\n",
      "  - perl-xml-sax=1.02=pl526_0\n",
      "  - perl-xml-sax-base=1.09=pl526_0\n",
      "  - perl-xml-sax-expat=0.51=pl526_3\n",
      "  - perl-xml-simple=2.25=pl526_1\n",
      "  - perl-xsloader=0.24=pl526_0\n",
      "  - pip=20.1.1=py_1\n",
      "  - prodigal=2.6.3=h516909a_2\n",
      "  - pyahocorasick=1.4.0=py36h8c4c3a4_1\n",
      "  - pyfaidx=0.5.8=py_1\n",
      "  - pyparsing=2.4.7=pyh9f0ad1d_0\n",
      "  - python=3.6.10=h8356626_1011_cpython\n",
      "  - python-dateutil=2.8.1=py_0\n",
      "  - python_abi=3.6=1_cp36m\n",
      "  - pytz=2020.1=pyh9f0ad1d_0\n",
      "  - readline=8.0=hf8c457e_0\n",
      "  - rgi=5.1.0=py_1\n",
      "  - samtools=1.9=h8571acd_11\n",
      "  - scipy=1.4.1=py36h2d22cac_3\n",
      "  - seaborn=0.10.1=py_0\n",
      "  - setuptools=47.1.1=py36h9f0ad1d_0\n",
      "  - six=1.15.0=pyh9f0ad1d_0\n",
      "  - sqlite=3.30.1=hcee41ef_0\n",
      "  - statsmodels=0.11.1=py36h8c4c3a4_1\n",
      "  - tbb=2020.1=hc9558a2_0\n",
      "  - tk=8.6.10=hed695b0_0\n",
      "  - tornado=6.0.4=py36h8c4c3a4_1\n",
      "  - wheel=0.34.2=py_1\n",
      "  - xz=5.2.5=h516909a_0\n",
      "  - zlib=1.2.11=h516909a_1006\n",
      "prefix: /home/users/ldenies/apps/miniconda3/envs/rgi\n",
      "\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/pathofact_hmm.yaml <==\n",
      "channels:\n",
      "  - bioconda\n",
      "  - conda-forge\n",
      "  - defaults\n",
      "dependencies:\n",
      "  - _libgcc_mutex=0.1=conda_forge\n",
      "  - _openmp_mutex=4.5=0_gnu\n",
      "  - hmmer=3.2.1=he1b5a44_2\n",
      "  - libgcc-ng=9.2.0=h24d8f2e_2\n",
      "  - libgomp=9.2.0=h24d8f2e_2\n",
      "  - libstdcxx-ng=9.2.0=hdf63c60_2\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/pathofact_biopython.yaml <==\n",
      "channels:\n",
      "  - conda-forge\n",
      "  - bioconda\n",
      "  - defaults\n",
      "dependencies:\n",
      "  - _libgcc_mutex=0.1=main\n",
      "  - biopython=1.74=py36h516909a_0\n",
      "  - ca-certificates=2019.6.16=hecc5488_0\n",
      "  - certifi=2019.6.16=py36_1\n",
      "  - cycler=0.10.0=py_1\n",
      "  - dbus=1.13.6=he372182_0\n",
      "  - expat=2.2.5=he1b5a44_1003\n",
      "  - fontconfig=2.13.1=he4413a7_1000\n",
      "  - freetype=2.10.0=he983fc9_0\n",
      "  - gettext=0.19.8.1=hc5be6a0_1002\n",
      "  - glib=2.58.3=h6f030ca_1002\n",
      "  - gst-plugins-base=1.14.5=h0935bb2_0\n",
      "  - gstreamer=1.14.5=h36ae1b5_0\n",
      "  - icu=58.2=hf484d3e_1000\n",
      "  - joblib=0.13.2=py_0\n",
      "  - jpeg=9c=h14c3975_1001\n",
      "  - kiwisolver=1.1.0=py36hc9558a2_0\n",
      "  - libblas=3.8.0=11_openblas\n",
      "  - libcblas=3.8.0=11_openblas\n",
      "  - libedit=3.1.20181209=hc058e9b_0\n",
      "  - libffi=3.2.1=hd88cf55_4\n",
      "  - libgcc-ng=9.1.0=hdf63c60_0\n",
      "  - libgfortran-ng=7.3.0=hdf63c60_0\n",
      "  - libiconv=1.15=h516909a_1005\n",
      "  - liblapack=3.8.0=11_openblas\n",
      "  - libopenblas=0.3.6=h6e990d7_6\n",
      "  - libpng=1.6.37=hed695b0_0\n",
      "  - libstdcxx-ng=9.1.0=hdf63c60_0\n",
      "  - libuuid=2.32.1=h14c3975_1000\n",
      "  - libxcb=1.13=h14c3975_1002\n",
      "  - libxml2=2.9.9=h13577e0_2\n",
      "  - matplotlib=3.1.0=py36h5429711_0\n",
      "  - ncurses=6.1=he6710b0_1\n",
      "  - numpy=1.17.0=py36h95a1406_0\n",
      "  - openssl=1.1.1c=h516909a_0\n",
      "  - pandas=0.25.0=py36hb3f55d8_0\n",
      "  - patsy=0.5.1=py_0\n",
      "  - pcre=8.41=hf484d3e_1003\n",
      "  - pip=19.1.1=py36_0\n",
      "  - pthread-stubs=0.4=h14c3975_1001\n",
      "  - pyparsing=2.4.2=py_0\n",
      "  - pyqt=5.9.2=py36hcca6a23_2\n",
      "  - python=3.6.9=h265db76_0\n",
      "  - python-dateutil=2.8.0=py_0\n",
      "  - pytz=2019.2=py_0\n",
      "  - qt=5.9.7=h52cfd70_2\n",
      "  - readline=7.0=h7b6447c_5\n",
      "  - scikit-learn=0.21.3=py36hcdab131_0\n",
      "  - scipy=1.3.0=py36h921218d_1\n",
      "  - seaborn=0.9.0=py_0\n",
      "  - setuptools=41.0.1=py36_0\n",
      "  - sip=4.19.8=py36hf484d3e_1000\n",
      "  - six=1.12.0=py36_1000\n",
      "  - sqlite=3.29.0=h7b6447c_0\n",
      "  - statsmodels=0.10.1=py36hc1659b7_0\n",
      "  - tk=8.6.8=hbc83047_0\n",
      "  - tornado=6.0.3=py36h516909a_0\n",
      "  - wheel=0.33.4=py36_0\n",
      "  - xorg-libxau=1.0.9=h14c3975_0\n",
      "  - xorg-libxdmcp=1.1.3=h516909a_0\n",
      "  - xz=5.2.4=h14c3975_4\n",
      "  - zlib=1.2.11=h7b6447c_3\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/platon.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::platon\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/plasflow.yaml <==\n",
      "channels:\n",
      "  # for PlasFlow\n",
      "  - smaegol\n",
      "  # rest\n",
      "  - bioconda\n",
      "  - conda-forge\n",
      "  - defaults\n",
      "dependencies:\n",
      "  - seqkit\n",
      "  - _libgcc_mutex=0.1=conda_forge\n",
      "  - _openmp_mutex=4.5=1_llvm\n",
      "  - _r-mutex=1.0.1=anacondar_1\n",
      "  - backports=1.0=py_2\n",
      "  - backports.weakref=1.0rc1=py35_1\n",
      "  - bioconductor-biocgenerics=0.22.0=r3.3.2_0\n",
      "  - bioconductor-biostrings=2.42.1=r3.3.2_0\n",
      "  - bioconductor-iranges=2.8.2=r3.3.2_0\n",
      "  - bioconductor-s4vectors=0.12.2=r3.3.2_0\n",
      "  - bioconductor-xvector=0.14.1=r3.3.2_0\n",
      "  - bioconductor-zlibbioc=1.20.0=r3.3.2_1\n",
      "  - biopython=1.68=py35_0\n",
      "  - blas=2.16=openblas\n",
      "  - bleach=1.5.0=py35_0\n",
      "  - bzip2=1.0.8=h516909a_2\n",
      "  - ca-certificates=2020.4.5.1=hecc5488_0\n",
      "  - cairo=1.14.6=4\n",
      "  - certifi=2018.8.24=py35_1001\n",
      "  - curl=7.52.1=0\n",
      "  - fontconfig=2.12.1=6\n",
      "  - freetype=2.7=1\n",
      "  - gettext=0.19.8.1=hc5be6a0_1002\n",
      "  - glib=2.51.4=0\n",
      "  - graphite2=1.3.13=he1b5a44_1001\n",
      "  - gsl=2.6=h294904e_0\n",
      "  - harfbuzz=1.4.3=0\n",
      "  - html5lib=0.9999999=py35_0\n",
      "  - icu=58.2=hf484d3e_1000\n",
      "  - jpeg=9c=h14c3975_1001\n",
      "  - libblas=3.8.0=16_openblas\n",
      "  - libcblas=3.8.0=16_openblas\n",
      "  - libffi=3.2.1=he1b5a44_1007\n",
      "  - libgcc=7.2.0=h69d50b8_2\n",
      "  - libgcc-ng=9.2.0=h24d8f2e_2\n",
      "  - libgfortran-ng=7.3.0=hdf63c60_5\n",
      "  - libiconv=1.15=h516909a_1006\n",
      "  - liblapack=3.8.0=16_openblas\n",
      "  - liblapacke=3.8.0=16_openblas\n",
      "  - libopenblas=0.3.9=h5ec1e0e_0\n",
      "  - libpng=1.6.28=1\n",
      "  - libstdcxx-ng=9.2.0=hdf63c60_2\n",
      "  - libtiff=4.0.7=0\n",
      "  - libxml2=2.9.5=0\n",
      "  - llvm-openmp=10.0.0=hc9558a2_0\n",
      "  - markdown=3.2.1=py_0\n",
      "  - mmtf-python=1.0.2=py35_0\n",
      "  - mock=2.0.0=py35_0\n",
      "  - msgpack-python=0.5.6=py35h2d50403_3\n",
      "  - ncurses=5.9=10\n",
      "  - numpy=1.15.2=py35h99e49ec_0\n",
      "  - numpy-base=1.15.2=py35h2f8d375_0\n",
      "  - olefile=0.46=py_0\n",
      "  - openssl=1.0.2u=h516909a_0\n",
      "  - pandas=0.23.4=py35hf8a1672_0\n",
      "  - pango=1.40.4=0\n",
      "  - pbr=5.4.2=py_0\n",
      "  - pcre=8.44=he1b5a44_0\n",
      "  - pillow=4.3.0=py35_0\n",
      "  - pip=20.0.2=py_2\n",
      "  - pixman=0.34.0=h14c3975_1003\n",
      "  - protobuf=3.4.0=py35_0\n",
      "  - python=3.5.4=0\n",
      "  - python-dateutil=2.8.1=py_0\n",
      "  - pytz=2019.3=py_0\n",
      "  - r-base=3.3.2=5\n",
      "  - readline=6.2=0\n",
      "  - reportlab=3.4.0=py35_0\n",
      "  - rpy2=2.7.8=py35r3.3.2_1\n",
      "  - scikit-learn=0.20.0=py35h22eb022_1\n",
      "  - scipy=1.1.0=py35he2b7bc3_1\n",
      "  - setuptools=40.4.3=py35_0\n",
      "  - six=1.14.0=py_1\n",
      "  - sqlite=3.13.0=1\n",
      "  - tensorflow=1.3.0=py35_0\n",
      "  - tk=8.5.19=2\n",
      "  - webencodings=0.5.1=py_1\n",
      "  - werkzeug=1.0.1=pyh9f0ad1d_0\n",
      "  - wheel=0.34.2=py_1\n",
      "  - xz=5.2.5=h516909a_0\n",
      "  - zlib=1.2.8=3\n",
      "  # PlasFlow: should be installed from smaegol, NOT from pip\n",
      "  - plasflow=1.1.0"
     ]
    }
   ],
   "source": [
    "pipelineInfo('/ebio/abt3_projects/software/dev/ll_pipelines/llg/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-23T08:05:23.186727Z",
     "iopub.status.busy": "2021-04-23T08:05:23.184337Z",
     "iopub.status.idle": "2021-04-23T08:05:23.257930Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LLPRIMER\n",
      "========\n",
      "\n",
      "Ley Lab Primer generation pipeline (LLPRIMER)\n",
      "\n",
      "* Version: 0.3.0\n",
      "* Authors:\n",
      "  * Nick Youngblut <nyoungb2@gmail.com>\n",
      "* Maintainers:\n",
      "  * Nick Youngblut <nyoungb2@gmail.com>\n",
      "\n",
      "--- conda envs ---\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llprimer//bin/envs/pdp.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- python=3.7\n",
      "- intervaltree\n",
      "- prodigal\n",
      "- blast\n",
      "- bedtools\n",
      "- mafft\n",
      "- mummer=3.23\n",
      "- emboss\n",
      "- primer3=1.1.4\n",
      "- biopython<1.78\n",
      "- pybedtools\n",
      "- joblib\n",
      "- tqdm\n",
      "- openpyxl\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llprimer//bin/envs/genes.yaml <==\n",
      "channels:\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- python=3\n",
      "- numpy\n",
      "- pyfaidx\n",
      "- bioconda::seqkit\n",
      "- bioconda::fasta-splitter\n",
      "- bioconda::vsearch\n",
      "- bioconda::mmseqs2\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llprimer//bin/envs/aln.yaml <==\n",
      "channels:\n",
      "- bioconda\n",
      "- conda-forge\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::revtrans\n",
      "- bioconda::kalign3\n",
      "- bioconda::mafft\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llprimer//bin/envs/blast.yaml <==\n",
      "channels:\n",
      "- bioconda\n",
      "- conda-forge\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::seqkit\n",
      "- bioconda::biopython<1.78\n",
      "- bioconda::blast>=2.10.1\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llprimer//bin/envs/primer3.yaml <==\n",
      "channels:\n",
      "- bioconda\n",
      "- conda-forge\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::biopython<1.78\n",
      "- bioconda::primer3-py\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llprimer//bin/envs/prokka.yaml <==\n",
      "channels:\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::prokka"
     ]
    }
   ],
   "source": [
    "pipelineInfo('/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-23T08:05:23.754128Z",
     "iopub.status.busy": "2021-04-23T08:05:23.751844Z",
     "iopub.status.idle": "2021-04-23T08:05:24.060133Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "R version 3.6.3 (2020-02-29)\n",
       "Platform: x86_64-conda_cos6-linux-gnu (64-bit)\n",
       "Running under: Ubuntu 18.04.5 LTS\n",
       "\n",
       "Matrix products: default\n",
       "BLAS/LAPACK: /ebio/abt3_projects/Georg_animal_feces/envs/tidyverse/lib/libopenblasp-r0.3.9.so\n",
       "\n",
       "locale:\n",
       " [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              \n",
       " [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    \n",
       " [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   \n",
       " [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 \n",
       " [9] LC_ADDRESS=C               LC_TELEPHONE=C            \n",
       "[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       \n",
       "\n",
       "attached base packages:\n",
       "[1] stats     graphics  grDevices utils     datasets  methods   base     \n",
       "\n",
       "other attached packages:\n",
       "[1] LeyLabRMisc_0.1.6 ggplot2_3.3.1     tidyr_1.1.0       dplyr_1.0.0      \n",
       "\n",
       "loaded via a namespace (and not attached):\n",
       " [1] Rcpp_1.0.4.6     magrittr_1.5     munsell_0.5.0    tidyselect_1.1.0\n",
       " [5] uuid_0.1-4       colorspace_1.4-1 R6_2.4.1         rlang_0.4.6     \n",
       " [9] fansi_0.4.1      tools_3.6.3      grid_3.6.3       gtable_0.3.0    \n",
       "[13] cli_2.0.2        withr_2.2.0      htmltools_0.4.0  ellipsis_0.3.1  \n",
       "[17] assertthat_0.2.1 digest_0.6.25    tibble_3.0.1     lifecycle_0.2.0 \n",
       "[21] crayon_1.3.4     IRdisplay_0.7.0  purrr_0.3.4      repr_1.1.0      \n",
       "[25] base64enc_0.1-3  vctrs_0.3.1      IRkernel_1.1     glue_1.4.1      \n",
       "[29] evaluate_0.14    stringi_1.4.6    pbdZMQ_0.3-3     compiler_3.6.3  \n",
       "[33] pillar_1.4.4     scales_1.1.1     generics_0.0.2   jsonlite_1.6.1  \n",
       "[37] pkgconfig_2.0.3 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "sessionInfo()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R [conda env:Georg_animal_feces-tidyverse]",
   "language": "R",
   "name": "conda-env-Georg_animal_feces-tidyverse-r"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.3"
  },
  "toc-autonumbering": true
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
