{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Goal\n",
    "\n",
    "* Primer design for clade of interest"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Var"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-24T05:30:09.726923Z",
     "iopub.status.busy": "2021-04-24T05:30:09.717861Z",
     "iopub.status.idle": "2021-04-24T05:30:09.790744Z"
    }
   },
   "outputs": [],
   "source": [
    "base_dir = '/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/'\n",
    "clade = 'Anaerotruncus'\n",
    "domain = 'Bacteria'\n",
    "taxid = 244127 "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Init"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-24T05:30:11.708953Z",
     "iopub.status.busy": "2021-04-24T05:30:11.659846Z",
     "iopub.status.idle": "2021-04-24T05:30:12.326689Z"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "Attaching package: ‘dplyr’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(dplyr)\n",
    "library(tidyr)\n",
    "library(ggplot2)\n",
    "library(LeyLabRMisc)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-24T05:30:13.098570Z",
     "iopub.status.busy": "2021-04-24T05:30:13.097645Z",
     "iopub.status.idle": "2021-04-24T05:30:13.115825Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Created directory: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3//Anaerotruncus \n"
     ]
    }
   ],
   "source": [
    "df.dims()\n",
    "work_dir = file.path(base_dir, clade)\n",
    "make_dir(work_dir)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Genome download\n",
    "\n",
    "* Downloading genomes from NCBI"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-12T13:07:32.874718Z",
     "iopub.status.busy": "2021-01-12T13:07:32.867938Z",
     "iopub.status.idle": "2021-01-12T13:07:32.985785Z",
     "shell.execute_reply": "2021-01-12T13:07:32.983225Z"
    }
   },
   "source": [
    "```\n",
    "OUTDIR=/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/\n",
    "mkdir -p $OUTDIR\n",
    "ncbi-genome-download -p 12 -s genbank -F fasta --genera Anaerotruncus -o $OUTDIR bacteria\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Genome quality\n",
    "\n",
    "* Filtering genomes by quality"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-24T05:33:37.504631Z",
     "iopub.status.busy": "2021-04-24T05:33:37.502694Z",
     "iopub.status.idle": "2021-04-24T05:33:37.581318Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 31 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Name</th><th scope=col>Fasta</th><th scope=col>Domain</th><th scope=col>Taxid</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>GCA_000154565.1_ASM15456v1_genomic     </td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Anaerotruncus/genbank/bacteria/GCA_000154565.1/GCA_000154565.1_ASM15456v1_genomic.fna.gz     </td><td>Bacteria</td><td>244127</td></tr>\n",
       "\t<tr><td>GCA_000403395.2_Anae_bact_G3_V1_genomic</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Anaerotruncus/genbank/bacteria/GCA_000403395.2/GCA_000403395.2_Anae_bact_G3_V1_genomic.fna.gz</td><td>Bacteria</td><td>244127</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>GCA_902363605.1_MGYG-HGUT-00131_genomic   </td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Anaerotruncus/genbank/bacteria/GCA_902363605.1/GCA_902363605.1_MGYG-HGUT-00131_genomic.fna.gz   </td><td>Bacteria</td><td>244127</td></tr>\n",
       "\t<tr><td>GCA_904384175.1_Chicken_20_mag_154_genomic</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Anaerotruncus/genbank/bacteria/GCA_904384175.1/GCA_904384175.1_Chicken_20_mag_154_genomic.fna.gz</td><td>Bacteria</td><td>244127</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 31 × 4\n",
       "\\begin{tabular}{llll}\n",
       " Name & Fasta & Domain & Taxid\\\\\n",
       " <chr> & <chr> & <fct> & <dbl>\\\\\n",
       "\\hline\n",
       "\t GCA\\_000154565.1\\_ASM15456v1\\_genomic      & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/v0.3/Anaerotruncus/genbank/bacteria/GCA\\_000154565.1/GCA\\_000154565.1\\_ASM15456v1\\_genomic.fna.gz      & Bacteria & 244127\\\\\n",
       "\t GCA\\_000403395.2\\_Anae\\_bact\\_G3\\_V1\\_genomic & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/v0.3/Anaerotruncus/genbank/bacteria/GCA\\_000403395.2/GCA\\_000403395.2\\_Anae\\_bact\\_G3\\_V1\\_genomic.fna.gz & Bacteria & 244127\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t GCA\\_902363605.1\\_MGYG-HGUT-00131\\_genomic    & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/v0.3/Anaerotruncus/genbank/bacteria/GCA\\_902363605.1/GCA\\_902363605.1\\_MGYG-HGUT-00131\\_genomic.fna.gz    & Bacteria & 244127\\\\\n",
       "\t GCA\\_904384175.1\\_Chicken\\_20\\_mag\\_154\\_genomic & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/v0.3/Anaerotruncus/genbank/bacteria/GCA\\_904384175.1/GCA\\_904384175.1\\_Chicken\\_20\\_mag\\_154\\_genomic.fna.gz & Bacteria & 244127\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 31 × 4\n",
       "\n",
       "| Name &lt;chr&gt; | Fasta &lt;chr&gt; | Domain &lt;fct&gt; | Taxid &lt;dbl&gt; |\n",
       "|---|---|---|---|\n",
       "| GCA_000154565.1_ASM15456v1_genomic      | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Anaerotruncus/genbank/bacteria/GCA_000154565.1/GCA_000154565.1_ASM15456v1_genomic.fna.gz      | Bacteria | 244127 |\n",
       "| GCA_000403395.2_Anae_bact_G3_V1_genomic | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Anaerotruncus/genbank/bacteria/GCA_000403395.2/GCA_000403395.2_Anae_bact_G3_V1_genomic.fna.gz | Bacteria | 244127 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| GCA_902363605.1_MGYG-HGUT-00131_genomic    | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Anaerotruncus/genbank/bacteria/GCA_902363605.1/GCA_902363605.1_MGYG-HGUT-00131_genomic.fna.gz    | Bacteria | 244127 |\n",
       "| GCA_904384175.1_Chicken_20_mag_154_genomic | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Anaerotruncus/genbank/bacteria/GCA_904384175.1/GCA_904384175.1_Chicken_20_mag_154_genomic.fna.gz | Bacteria | 244127 |\n",
       "\n"
      ],
      "text/plain": [
       "   Name                                      \n",
       "1  GCA_000154565.1_ASM15456v1_genomic        \n",
       "2  GCA_000403395.2_Anae_bact_G3_V1_genomic   \n",
       "⋮  ⋮                                         \n",
       "30 GCA_902363605.1_MGYG-HGUT-00131_genomic   \n",
       "31 GCA_904384175.1_Chicken_20_mag_154_genomic\n",
       "   Fasta                                                                                                                                                                                   \n",
       "1  /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Anaerotruncus/genbank/bacteria/GCA_000154565.1/GCA_000154565.1_ASM15456v1_genomic.fna.gz        \n",
       "2  /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Anaerotruncus/genbank/bacteria/GCA_000403395.2/GCA_000403395.2_Anae_bact_G3_V1_genomic.fna.gz   \n",
       "⋮  ⋮                                                                                                                                                                                       \n",
       "30 /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Anaerotruncus/genbank/bacteria/GCA_902363605.1/GCA_902363605.1_MGYG-HGUT-00131_genomic.fna.gz   \n",
       "31 /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Anaerotruncus/genbank/bacteria/GCA_904384175.1/GCA_904384175.1_Chicken_20_mag_154_genomic.fna.gz\n",
       "   Domain   Taxid \n",
       "1  Bacteria 244127\n",
       "2  Bacteria 244127\n",
       "⋮  ⋮        ⋮     \n",
       "30 Bacteria 244127\n",
       "31 Bacteria 244127"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "File written: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3//Anaerotruncus/genbank/samples.txt \n"
     ]
    }
   ],
   "source": [
    "D = file.path(base_dir, clade, 'genbank')\n",
    "files = list_files(D, '.fna.gz')\n",
    "samps = data.frame(Name = files %>% as.character %>% basename,\n",
    "                   Fasta = files,\n",
    "                   Domain = 'Bacteria',\n",
    "                   Taxid = taxid) %>%\n",
    "    mutate(Name = gsub('\\\\.fna\\\\.gz$', '', Name),\n",
    "           Fasta = gsub('/+', '/', Fasta))\n",
    "samps\n",
    "\n",
    "# writing file\n",
    "outfile = file.path(D, 'samples.txt')\n",
    "write_table(samps, outfile)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### LLG"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Config"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-24T05:34:23.076324Z",
     "iopub.status.busy": "2021-04-24T05:34:23.074301Z",
     "iopub.status.idle": "2021-04-24T05:34:23.109554Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# table with genome --> fasta_file information\n",
      "samples_file: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Anaerotruncus/genbank/samples.txt\n",
      "\n",
      "# output location\n",
      "output_dir: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Anaerotruncus/LLG/\n",
      "\n",
      "# temporary file directory (your username will be added automatically)\n",
      "tmp_dir: /ebio/abt3_scratch/\n",
      "\n",
      "# batch processing of genomes for certain steps\n",
      "## increase to better parallelize\n",
      "batches: 2 \n",
      "\n",
      "# Domain of genomes ('Archaea' or 'Bacteria)\n",
      "## Use \"Skip\" if provided as a \"Domain\" column in the genome table\n",
      "Domain: Skip\n",
      "\n",
      "# software parameters\n",
      "# Use \"Skip\" to skip any of these steps. If no params for rule, use \"\"\n",
      "# dRep MAGs are not further analyzed, but you can de-rep & then use the de-rep genome table as input.\n",
      "params:\n",
      "  ionice: -c 3\n",
      "  # assembly assessment\n",
      "  seqkit: \"\"\n",
      "  quast: Skip #\"\"\n",
      "  multiqc_on_quast: \"\" \n",
      "  checkm: \"\"\n",
      "  # de-replication (requires checkm)\n",
      "  drep: -comp 90 -con 5 -sa 0.999\n",
      "  # taxonomy\n",
      "  sourmash:\n",
      "    compute: Skip #--scaled 10000 -k 31\n",
      "    gather: -k 31\n",
      "  gtdbtk:\n",
      "    classify_wf: --min_perc_aa 10\n",
      "  # genome pairwise ANI  \n",
      "  fastani: Skip #--fragLen 3000 --minFraction 0.2 -k 16\n",
      "  dashing: Skip #-k 31 --full-tsv\n",
      "  comparem_aai: Skip #--evalue 0.001\n",
      "  # gene annotation\n",
      "  gene_call:\n",
      "    prokka: Skip #\"\" \n",
      "    multiqc_on_prokka: \"\"\n",
      "    prodigal: Skip #\"\"\n",
      "  eggnog_mapper: \"\"\n",
      "  eggnog_mapper_annot: \"\"\n",
      "  # rRNA (16S alignment & phylogeny)\n",
      "  barrnap: Skip #--lencutoff 0.8\n",
      "  vsearch_per_genome_drep: --id 0.95    # Skip to prevent drep of 16S copies within each genome\n",
      "  qiime2_fasttree: \"\"\n",
      "  qiime2_iqtree: --p-alrt 1000 --p-abayes --p-lbp 1000  --p-substitution-model 'GTR+I+G'\n",
      "  # genome phylogeny\n",
      "  phylophlan_config: Skip #--map_dna diamond --db_aa diamond --map_aa diamond --msa mafft --trim trimal --tree1 fasttree --tree2 raxml\n",
      "  phylophlan: --diversity high --min_num_markers 50 --accurate\n",
      "  # phenotype\n",
      "  traitar: Skip #\"\"\n",
      "  # biosynthetic gene clusters (BGCs)\n",
      "  antismash: Skip #--cb-knownclusters --cb-subclusters --asf  \n",
      "  DeepBGC: Skip #--score 0.5 --classifier-score 0.5 --prodigal-meta-mode\n",
      "  # antimicrobial resistance (AMR)\n",
      "  abricate: Skip #--minid 75 --mincov 80\n",
      "  # CRISPRs\n",
      "  cctyper: Skip #--prodigal meta\n",
      "  \n",
      "# databases\n",
      "databases:\n",
      "  checkM_data: /ebio/abt3_projects/databases_no-backup/checkM/\n",
      "  sourmash: /ebio/abt3_projects/databases_no-backup/sourmash/genbank-k31.sbt.json\n",
      "  sourmash_lca: /ebio/abt3_projects/databases_no-backup/sourmash/genbank-k31.lca.json.gz\n",
      "  gtdbtk: /ebio/abt3_projects/databases_no-backup/GTDB/release95/gtdbtk/db_info.md\n",
      "  phylophlan: /ebio/abt3_projects/databases_no-backup/phylophlan/PhyloPhlan/phylophlan.faa.bz2\n",
      "  eggnog: /ebio/abt3_projects/databases_no-backup/Eggnog/v2/eggnog.db\n",
      "  eggnog_diamond: /ebio/abt3_projects/databases_no-backup/Eggnog/v2/eggnog_proteins.dmnd\n",
      "  antismash: /ebio/abt3_projects/databases_no-backup/antismash/v5/\n",
      "  deepbgc: /ebio/abt3_projects/databases_no-backup/DeepBGC/\n",
      "  traitar: /ebio/abt3_projects/databases_no-backup/pfam/traitar/\n",
      "  abricate:\n",
      "    ncbi: /ebio/abt3_projects/databases_no-backup/abricate/ncbi/sequences\n",
      "    card: /ebio/abt3_projects/databases_no-backup/abricate/card/sequences\n",
      "    resfinder: /ebio/abt3_projects/databases_no-backup/abricate/resfinder/sequences\n",
      "    argannot: /ebio/abt3_projects/databases_no-backup/abricate/argannot/sequences\n",
      "    bacmet2: /ebio/abt3_projects/databases_no-backup/abricate/bacmet2/sequences\n",
      "    vfdb: /ebio/abt3_projects/databases_no-backup/abricate/vfdb/sequences\n",
      "    megares: /ebio/abt3_projects/databases_no-backup/abricate/megares/sequences\n",
      "    plasmidfinder: /ebio/abt3_projects/databases_no-backup/abricate/plasmidfinder/sequences\n",
      "\n",
      "# snakemake pipeline\n",
      "pipeline:\n",
      "  snakemake_folder: ./\n",
      "  script_folder: ./bin/scripts/\n",
      "  use_shared_mem: True\n",
      "  name: LLG"
     ]
    }
   ],
   "source": [
    "cat_file(file.path(work_dir, 'LLG', 'config.yaml'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-12T13:14:50.416604Z",
     "iopub.status.busy": "2021-01-12T13:14:50.414179Z",
     "iopub.status.idle": "2021-01-12T13:14:50.431925Z",
     "shell.execute_reply": "2021-01-12T13:14:50.430260Z"
    }
   },
   "source": [
    "#### Run\n",
    "\n",
    "```\n",
    "(snakemake) @ rick:/ebio/abt3_projects/software/dev/ll_pipelines/llg\n",
    "$ screen -L -S llg ./snakemake_sge.sh /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Anaerotruncus/config_llg.yaml 20 -F\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-12T13:14:10.995042Z",
     "iopub.status.busy": "2021-01-12T13:14:10.992773Z",
     "iopub.status.idle": "2021-01-12T13:14:11.009275Z",
     "shell.execute_reply": "2021-01-12T13:14:11.008255Z"
    }
   },
   "source": [
    "### Samples table of high-quality genomes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-24T05:34:47.206452Z",
     "iopub.status.busy": "2021-04-24T05:34:47.204439Z",
     "iopub.status.idle": "2021-04-24T05:34:47.258517Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 31 × 33</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Bin.Id</th><th scope=col>Marker.lineage</th><th scope=col>X..genomes</th><th scope=col>X..markers</th><th scope=col>X..marker.sets</th><th scope=col>Completeness</th><th scope=col>Contamination</th><th scope=col>Strain.heterogeneity</th><th scope=col>Genome.size..bp.</th><th scope=col>X..ambiguous.bases</th><th scope=col>⋯</th><th scope=col>X0</th><th scope=col>X1</th><th scope=col>X2</th><th scope=col>X3</th><th scope=col>X4</th><th scope=col>X5.</th><th scope=col>assembly.Id</th><th scope=col>assembler.Id</th><th scope=col>taxon.Id</th><th scope=col>File</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>GCA_000154565.1_ASM15456v1_genomic</td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>97.99</td><td>0.22</td><td>0</td><td>3719688</td><td>800</td><td>⋯</td><td>3</td><td>259</td><td>1</td><td>0</td><td>0</td><td>0</td><td>|ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Anaerotruncus|LLG_output|checkM|1|checkm|markers_qa_summary.tsv.1</td><td>markers_qa_summary.tsv.1</td><td>NA</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/LLG_output/checkM/1/checkm/markers_qa_summary.tsv.1</td></tr>\n",
       "\t<tr><td>GCA_000433975.1_MGS528_genomic    </td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>97.99</td><td>0.00</td><td>0</td><td>1881083</td><td>732</td><td>⋯</td><td>3</td><td>260</td><td>0</td><td>0</td><td>0</td><td>0</td><td>|ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Anaerotruncus|LLG_output|checkM|1|checkm|markers_qa_summary.tsv.2</td><td>markers_qa_summary.tsv.2</td><td>NA</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/LLG_output/checkM/1/checkm/markers_qa_summary.tsv.2</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋱</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>GCA_900199635.1_PRJEB19460_genomic     </td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>99.33</td><td>0.00</td><td>0</td><td>3145951</td><td> 14</td><td>⋯</td><td>1</td><td>262</td><td>0</td><td>0</td><td>0</td><td>0</td><td>|ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Anaerotruncus|LLG_output|checkM|2|checkm|markers_qa_summary.tsv.14</td><td>markers_qa_summary.tsv.14</td><td>NA</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/LLG_output/checkM/2/checkm/markers_qa_summary.tsv.14</td></tr>\n",
       "\t<tr><td>GCA_902363605.1_MGYG-HGUT-00131_genomic</td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>97.99</td><td>1.34</td><td>0</td><td>3306065</td><td>265</td><td>⋯</td><td>3</td><td>257</td><td>3</td><td>0</td><td>0</td><td>0</td><td>|ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Anaerotruncus|LLG_output|checkM|2|checkm|markers_qa_summary.tsv.15</td><td>markers_qa_summary.tsv.15</td><td>NA</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/LLG_output/checkM/2/checkm/markers_qa_summary.tsv.15</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 31 × 33\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " Bin.Id & Marker.lineage & X..genomes & X..markers & X..marker.sets & Completeness & Contamination & Strain.heterogeneity & Genome.size..bp. & X..ambiguous.bases & ⋯ & X0 & X1 & X2 & X3 & X4 & X5. & assembly.Id & assembler.Id & taxon.Id & File\\\\\n",
       " <fct> & <fct> & <int> & <int> & <int> & <dbl> & <dbl> & <dbl> & <int> & <int> & ⋯ & <int> & <int> & <int> & <int> & <int> & <int> & <fct> & <fct> & <lgl> & <fct>\\\\\n",
       "\\hline\n",
       "\t GCA\\_000154565.1\\_ASM15456v1\\_genomic & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 97.99 & 0.22 & 0 & 3719688 & 800 & ⋯ & 3 & 259 & 1 & 0 & 0 & 0 & \\textbar{}ebio\\textbar{}abt3\\_projects\\textbar{}software\\textbar{}dev\\textbar{}ll\\_pipelines\\textbar{}llprimer\\textbar{}experiments\\textbar{}HMP\\_most-wanted\\textbar{}Anaerotruncus\\textbar{}LLG\\_output\\textbar{}checkM\\textbar{}1\\textbar{}checkm\\textbar{}markers\\_qa\\_summary.tsv.1 & markers\\_qa\\_summary.tsv.1 & NA & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/Anaerotruncus/LLG\\_output/checkM/1/checkm/markers\\_qa\\_summary.tsv.1\\\\\n",
       "\t GCA\\_000433975.1\\_MGS528\\_genomic     & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 97.99 & 0.00 & 0 & 1881083 & 732 & ⋯ & 3 & 260 & 0 & 0 & 0 & 0 & \\textbar{}ebio\\textbar{}abt3\\_projects\\textbar{}software\\textbar{}dev\\textbar{}ll\\_pipelines\\textbar{}llprimer\\textbar{}experiments\\textbar{}HMP\\_most-wanted\\textbar{}Anaerotruncus\\textbar{}LLG\\_output\\textbar{}checkM\\textbar{}1\\textbar{}checkm\\textbar{}markers\\_qa\\_summary.tsv.2 & markers\\_qa\\_summary.tsv.2 & NA & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/Anaerotruncus/LLG\\_output/checkM/1/checkm/markers\\_qa\\_summary.tsv.2\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋱ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t GCA\\_900199635.1\\_PRJEB19460\\_genomic      & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 99.33 & 0.00 & 0 & 3145951 &  14 & ⋯ & 1 & 262 & 0 & 0 & 0 & 0 & \\textbar{}ebio\\textbar{}abt3\\_projects\\textbar{}software\\textbar{}dev\\textbar{}ll\\_pipelines\\textbar{}llprimer\\textbar{}experiments\\textbar{}HMP\\_most-wanted\\textbar{}Anaerotruncus\\textbar{}LLG\\_output\\textbar{}checkM\\textbar{}2\\textbar{}checkm\\textbar{}markers\\_qa\\_summary.tsv.14 & markers\\_qa\\_summary.tsv.14 & NA & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/Anaerotruncus/LLG\\_output/checkM/2/checkm/markers\\_qa\\_summary.tsv.14\\\\\n",
       "\t GCA\\_902363605.1\\_MGYG-HGUT-00131\\_genomic & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 97.99 & 1.34 & 0 & 3306065 & 265 & ⋯ & 3 & 257 & 3 & 0 & 0 & 0 & \\textbar{}ebio\\textbar{}abt3\\_projects\\textbar{}software\\textbar{}dev\\textbar{}ll\\_pipelines\\textbar{}llprimer\\textbar{}experiments\\textbar{}HMP\\_most-wanted\\textbar{}Anaerotruncus\\textbar{}LLG\\_output\\textbar{}checkM\\textbar{}2\\textbar{}checkm\\textbar{}markers\\_qa\\_summary.tsv.15 & markers\\_qa\\_summary.tsv.15 & NA & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/Anaerotruncus/LLG\\_output/checkM/2/checkm/markers\\_qa\\_summary.tsv.15\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 31 × 33\n",
       "\n",
       "| Bin.Id &lt;fct&gt; | Marker.lineage &lt;fct&gt; | X..genomes &lt;int&gt; | X..markers &lt;int&gt; | X..marker.sets &lt;int&gt; | Completeness &lt;dbl&gt; | Contamination &lt;dbl&gt; | Strain.heterogeneity &lt;dbl&gt; | Genome.size..bp. &lt;int&gt; | X..ambiguous.bases &lt;int&gt; | ⋯ ⋯ | X0 &lt;int&gt; | X1 &lt;int&gt; | X2 &lt;int&gt; | X3 &lt;int&gt; | X4 &lt;int&gt; | X5. &lt;int&gt; | assembly.Id &lt;fct&gt; | assembler.Id &lt;fct&gt; | taxon.Id &lt;lgl&gt; | File &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| GCA_000154565.1_ASM15456v1_genomic | o__Clostridiales (UID1212) | 172 | 263 | 149 | 97.99 | 0.22 | 0 | 3719688 | 800 | ⋯ | 3 | 259 | 1 | 0 | 0 | 0 | |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Anaerotruncus|LLG_output|checkM|1|checkm|markers_qa_summary.tsv.1 | markers_qa_summary.tsv.1 | NA | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/LLG_output/checkM/1/checkm/markers_qa_summary.tsv.1 |\n",
       "| GCA_000433975.1_MGS528_genomic     | o__Clostridiales (UID1212) | 172 | 263 | 149 | 97.99 | 0.00 | 0 | 1881083 | 732 | ⋯ | 3 | 260 | 0 | 0 | 0 | 0 | |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Anaerotruncus|LLG_output|checkM|1|checkm|markers_qa_summary.tsv.2 | markers_qa_summary.tsv.2 | NA | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/LLG_output/checkM/1/checkm/markers_qa_summary.tsv.2 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋱ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| GCA_900199635.1_PRJEB19460_genomic      | o__Clostridiales (UID1212) | 172 | 263 | 149 | 99.33 | 0.00 | 0 | 3145951 |  14 | ⋯ | 1 | 262 | 0 | 0 | 0 | 0 | |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Anaerotruncus|LLG_output|checkM|2|checkm|markers_qa_summary.tsv.14 | markers_qa_summary.tsv.14 | NA | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/LLG_output/checkM/2/checkm/markers_qa_summary.tsv.14 |\n",
       "| GCA_902363605.1_MGYG-HGUT-00131_genomic | o__Clostridiales (UID1212) | 172 | 263 | 149 | 97.99 | 1.34 | 0 | 3306065 | 265 | ⋯ | 3 | 257 | 3 | 0 | 0 | 0 | |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Anaerotruncus|LLG_output|checkM|2|checkm|markers_qa_summary.tsv.15 | markers_qa_summary.tsv.15 | NA | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/LLG_output/checkM/2/checkm/markers_qa_summary.tsv.15 |\n",
       "\n"
      ],
      "text/plain": [
       "   Bin.Id                                  Marker.lineage            \n",
       "1  GCA_000154565.1_ASM15456v1_genomic      o__Clostridiales (UID1212)\n",
       "2  GCA_000433975.1_MGS528_genomic          o__Clostridiales (UID1212)\n",
       "⋮  ⋮                                       ⋮                         \n",
       "30 GCA_900199635.1_PRJEB19460_genomic      o__Clostridiales (UID1212)\n",
       "31 GCA_902363605.1_MGYG-HGUT-00131_genomic o__Clostridiales (UID1212)\n",
       "   X..genomes X..markers X..marker.sets Completeness Contamination\n",
       "1  172        263        149            97.99        0.22         \n",
       "2  172        263        149            97.99        0.00         \n",
       "⋮  ⋮          ⋮          ⋮              ⋮            ⋮            \n",
       "30 172        263        149            99.33        0.00         \n",
       "31 172        263        149            97.99        1.34         \n",
       "   Strain.heterogeneity Genome.size..bp. X..ambiguous.bases ⋯ X0 X1  X2 X3 X4\n",
       "1  0                    3719688          800                ⋯ 3  259 1  0  0 \n",
       "2  0                    1881083          732                ⋯ 3  260 0  0  0 \n",
       "⋮  ⋮                    ⋮                ⋮                  ⋱ ⋮  ⋮   ⋮  ⋮  ⋮ \n",
       "30 0                    3145951           14                ⋯ 1  262 0  0  0 \n",
       "31 0                    3306065          265                ⋯ 3  257 3  0  0 \n",
       "   X5.\n",
       "1  0  \n",
       "2  0  \n",
       "⋮  ⋮  \n",
       "30 0  \n",
       "31 0  \n",
       "   assembly.Id                                                                                                                                          \n",
       "1  |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Anaerotruncus|LLG_output|checkM|1|checkm|markers_qa_summary.tsv.1 \n",
       "2  |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Anaerotruncus|LLG_output|checkM|1|checkm|markers_qa_summary.tsv.2 \n",
       "⋮  ⋮                                                                                                                                                    \n",
       "30 |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Anaerotruncus|LLG_output|checkM|2|checkm|markers_qa_summary.tsv.14\n",
       "31 |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Anaerotruncus|LLG_output|checkM|2|checkm|markers_qa_summary.tsv.15\n",
       "   assembler.Id              taxon.Id\n",
       "1  markers_qa_summary.tsv.1  NA      \n",
       "2  markers_qa_summary.tsv.2  NA      \n",
       "⋮  ⋮                         ⋮       \n",
       "30 markers_qa_summary.tsv.14 NA      \n",
       "31 markers_qa_summary.tsv.15 NA      \n",
       "   File                                                                                                                                                 \n",
       "1  /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/LLG_output/checkM/1/checkm/markers_qa_summary.tsv.1 \n",
       "2  /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/LLG_output/checkM/1/checkm/markers_qa_summary.tsv.2 \n",
       "⋮  ⋮                                                                                                                                                    \n",
       "30 /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/LLG_output/checkM/2/checkm/markers_qa_summary.tsv.14\n",
       "31 /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Anaerotruncus/LLG_output/checkM/2/checkm/markers_qa_summary.tsv.15"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# checkM summary\n",
    "checkm = file.path(work_dir, 'LLG', 'checkM', 'checkm_qa_summary.tsv') %>%\n",
    "    read.delim(sep='\\t') \n",
    "checkm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-24T05:34:51.595069Z",
     "iopub.status.busy": "2021-04-24T05:34:51.592880Z",
     "iopub.status.idle": "2021-04-24T05:34:51.635720Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 31 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>genome</th><th scope=col>completeness</th><th scope=col>contamination</th><th scope=col>Bin.Id</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000154565.1_ASM15456v1_genomic.fna</td><td>97.99</td><td>0.22</td><td>GCA_000154565.1_ASM15456v1_genomic</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000433975.1_MGS528_genomic.fna    </td><td>97.99</td><td>0.00</td><td>GCA_000433975.1_MGS528_genomic    </td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_900199635.1_PRJEB19460_genomic.fna     </td><td>99.33</td><td>0.00</td><td>GCA_900199635.1_PRJEB19460_genomic     </td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_902363605.1_MGYG-HGUT-00131_genomic.fna</td><td>97.99</td><td>1.34</td><td>GCA_902363605.1_MGYG-HGUT-00131_genomic</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 31 × 4\n",
       "\\begin{tabular}{llll}\n",
       " genome & completeness & contamination & Bin.Id\\\\\n",
       " <fct> & <dbl> & <dbl> & <chr>\\\\\n",
       "\\hline\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_000154565.1\\_ASM15456v1\\_genomic.fna & 97.99 & 0.22 & GCA\\_000154565.1\\_ASM15456v1\\_genomic\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_000433975.1\\_MGS528\\_genomic.fna     & 97.99 & 0.00 & GCA\\_000433975.1\\_MGS528\\_genomic    \\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_900199635.1\\_PRJEB19460\\_genomic.fna      & 99.33 & 0.00 & GCA\\_900199635.1\\_PRJEB19460\\_genomic     \\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_902363605.1\\_MGYG-HGUT-00131\\_genomic.fna & 97.99 & 1.34 & GCA\\_902363605.1\\_MGYG-HGUT-00131\\_genomic\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 31 × 4\n",
       "\n",
       "| genome &lt;fct&gt; | completeness &lt;dbl&gt; | contamination &lt;dbl&gt; | Bin.Id &lt;chr&gt; |\n",
       "|---|---|---|---|\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000154565.1_ASM15456v1_genomic.fna | 97.99 | 0.22 | GCA_000154565.1_ASM15456v1_genomic |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000433975.1_MGS528_genomic.fna     | 97.99 | 0.00 | GCA_000433975.1_MGS528_genomic     |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_900199635.1_PRJEB19460_genomic.fna      | 99.33 | 0.00 | GCA_900199635.1_PRJEB19460_genomic      |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_902363605.1_MGYG-HGUT-00131_genomic.fna | 97.99 | 1.34 | GCA_902363605.1_MGYG-HGUT-00131_genomic |\n",
       "\n"
      ],
      "text/plain": [
       "   genome                                                                                           \n",
       "1  /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000154565.1_ASM15456v1_genomic.fna     \n",
       "2  /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000433975.1_MGS528_genomic.fna         \n",
       "⋮  ⋮                                                                                                \n",
       "30 /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_900199635.1_PRJEB19460_genomic.fna     \n",
       "31 /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_902363605.1_MGYG-HGUT-00131_genomic.fna\n",
       "   completeness contamination Bin.Id                                 \n",
       "1  97.99        0.22          GCA_000154565.1_ASM15456v1_genomic     \n",
       "2  97.99        0.00          GCA_000433975.1_MGS528_genomic         \n",
       "⋮  ⋮            ⋮             ⋮                                      \n",
       "30 99.33        0.00          GCA_900199635.1_PRJEB19460_genomic     \n",
       "31 97.99        1.34          GCA_902363605.1_MGYG-HGUT-00131_genomic"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# dRep summary\n",
    "drep = file.path(work_dir, 'LLG', 'drep', 'checkm_markers_qa_summary.tsv') %>%\n",
    "    read.delim(sep='\\t') %>%\n",
    "    mutate(Bin.Id = gsub('.+/', '', genome),\n",
    "           Bin.Id = gsub('\\\\.fna$', '', Bin.Id))\n",
    "drep"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-24T05:34:57.519460Z",
     "iopub.status.busy": "2021-04-24T05:34:57.517401Z",
     "iopub.status.idle": "2021-04-24T05:34:57.546143Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 27 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Name</th><th scope=col>Fasta</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>GCA_014284405.1_ASM1428440v1_genomic</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Anaerotruncus/LLG/drep/drep/dereplicated_genomes/GCA_014284405.1_ASM1428440v1_genomic.fna</td></tr>\n",
       "\t<tr><td>GCA_900199635.1_PRJEB19460_genomic  </td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Anaerotruncus/LLG/drep/drep/dereplicated_genomes/GCA_900199635.1_PRJEB19460_genomic.fna  </td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>GCA_015554285.1_ASM1555428v1_genomic</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Anaerotruncus/LLG/drep/drep/dereplicated_genomes/GCA_015554285.1_ASM1555428v1_genomic.fna</td></tr>\n",
       "\t<tr><td>GCA_000433975.1_MGS528_genomic      </td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Anaerotruncus/LLG/drep/drep/dereplicated_genomes/GCA_000433975.1_MGS528_genomic.fna      </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 27 × 2\n",
       "\\begin{tabular}{ll}\n",
       " Name & Fasta\\\\\n",
       " <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\t GCA\\_014284405.1\\_ASM1428440v1\\_genomic & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/v0.3/Anaerotruncus/LLG/drep/drep/dereplicated\\_genomes/GCA\\_014284405.1\\_ASM1428440v1\\_genomic.fna\\\\\n",
       "\t GCA\\_900199635.1\\_PRJEB19460\\_genomic   & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/v0.3/Anaerotruncus/LLG/drep/drep/dereplicated\\_genomes/GCA\\_900199635.1\\_PRJEB19460\\_genomic.fna  \\\\\n",
       "\t ⋮ & ⋮\\\\\n",
       "\t GCA\\_015554285.1\\_ASM1555428v1\\_genomic & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/v0.3/Anaerotruncus/LLG/drep/drep/dereplicated\\_genomes/GCA\\_015554285.1\\_ASM1555428v1\\_genomic.fna\\\\\n",
       "\t GCA\\_000433975.1\\_MGS528\\_genomic       & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/v0.3/Anaerotruncus/LLG/drep/drep/dereplicated\\_genomes/GCA\\_000433975.1\\_MGS528\\_genomic.fna      \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 27 × 2\n",
       "\n",
       "| Name &lt;fct&gt; | Fasta &lt;fct&gt; |\n",
       "|---|---|\n",
       "| GCA_014284405.1_ASM1428440v1_genomic | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Anaerotruncus/LLG/drep/drep/dereplicated_genomes/GCA_014284405.1_ASM1428440v1_genomic.fna |\n",
       "| GCA_900199635.1_PRJEB19460_genomic   | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Anaerotruncus/LLG/drep/drep/dereplicated_genomes/GCA_900199635.1_PRJEB19460_genomic.fna   |\n",
       "| ⋮ | ⋮ |\n",
       "| GCA_015554285.1_ASM1555428v1_genomic | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Anaerotruncus/LLG/drep/drep/dereplicated_genomes/GCA_015554285.1_ASM1555428v1_genomic.fna |\n",
       "| GCA_000433975.1_MGS528_genomic       | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Anaerotruncus/LLG/drep/drep/dereplicated_genomes/GCA_000433975.1_MGS528_genomic.fna       |\n",
       "\n"
      ],
      "text/plain": [
       "   Name                                \n",
       "1  GCA_014284405.1_ASM1428440v1_genomic\n",
       "2  GCA_900199635.1_PRJEB19460_genomic  \n",
       "⋮  ⋮                                   \n",
       "26 GCA_015554285.1_ASM1555428v1_genomic\n",
       "27 GCA_000433975.1_MGS528_genomic      \n",
       "   Fasta                                                                                                                                                                            \n",
       "1  /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Anaerotruncus/LLG/drep/drep/dereplicated_genomes/GCA_014284405.1_ASM1428440v1_genomic.fna\n",
       "2  /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Anaerotruncus/LLG/drep/drep/dereplicated_genomes/GCA_900199635.1_PRJEB19460_genomic.fna  \n",
       "⋮  ⋮                                                                                                                                                                                \n",
       "26 /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Anaerotruncus/LLG/drep/drep/dereplicated_genomes/GCA_015554285.1_ASM1555428v1_genomic.fna\n",
       "27 /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Anaerotruncus/LLG/drep/drep/dereplicated_genomes/GCA_000433975.1_MGS528_genomic.fna      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# de-replicated genomes\n",
    "drep_gen = file.path(work_dir, 'LLG', 'drep', 'dereplicated_genomes.tsv') %>%\n",
    "    read.delim(sep='\\t')\n",
    "drep_gen"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-24T05:35:02.363983Z",
     "iopub.status.busy": "2021-04-24T05:35:02.361909Z",
     "iopub.status.idle": "2021-04-24T05:35:02.445819Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 30 × 22</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>user_genome</th><th scope=col>Domain</th><th scope=col>Phylum</th><th scope=col>Class</th><th scope=col>Order</th><th scope=col>Family</th><th scope=col>Genus</th><th scope=col>Species</th><th scope=col>fastani_reference</th><th scope=col>fastani_reference_radius</th><th scope=col>⋯</th><th scope=col>fastani_af</th><th scope=col>closest_placement_reference</th><th scope=col>closest_placement_radius</th><th scope=col>closest_placement_taxonomy</th><th scope=col>closest_placement_ani</th><th scope=col>closest_placement_af</th><th scope=col>msa_percent</th><th scope=col>translation_table</th><th scope=col>red_value</th><th scope=col>warnings</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>⋯</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>GCA_000154565.1_ASM15456v1_genomic</td><td>d__Bacteria</td><td>p__Firmicutes_A</td><td>c__Clostridia</td><td>o__Oscillospirales</td><td>f__Ruminococcaceae   </td><td>g__Anaerotruncus</td><td>s__Anaerotruncus colihominis</td><td>GCF_000154565.1</td><td>95.0</td><td>⋯</td><td>1.0</td><td>GCF_000154565.1</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus colihominis   </td><td>100.0</td><td>1.0</td><td>97.18</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>GCA_000433975.1_MGS528_genomic    </td><td>d__Bacteria</td><td>p__Firmicutes_A</td><td>c__Clostridia</td><td>o__Oscillospirales</td><td>f__Acutalibacteraceae</td><td>g__Eubacterium_R</td><td>s__Eubacterium_R sp000433975</td><td>GCA_000433975.1</td><td>95.0</td><td>⋯</td><td>1.0</td><td>GCA_000433975.1</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Acutalibacteraceae;g__Eubacterium_R;s__Eubacterium_R sp000433975</td><td>100.0</td><td>1.0</td><td>96.57</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋱</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>GCA_900199635.1_PRJEB19460_genomic     </td><td>d__Bacteria</td><td>p__Firmicutes_A</td><td>c__Clostridia</td><td>o__Oscillospirales</td><td>f__Ruminococcaceae</td><td>g__Anaerotruncus</td><td>s__Anaerotruncus massiliensis</td><td>GCF_900199635.1</td><td>95.0</td><td>⋯</td><td>1.0 </td><td>GCF_900199635.1</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus massiliensis</td><td>100.0</td><td>1.0 </td><td>96.43</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>GCA_902363605.1_MGYG-HGUT-00131_genomic</td><td>d__Bacteria</td><td>p__Firmicutes_A</td><td>c__Clostridia</td><td>o__Oscillospirales</td><td>f__Ruminococcaceae</td><td>g__Anaerotruncus</td><td>s__Anaerotruncus rubiinfantis</td><td>GCF_900078395.1</td><td>95.0</td><td>⋯</td><td>0.91</td><td>GCF_900078395.1</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus rubiinfantis</td><td>99.26</td><td>0.91</td><td>96.35</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 30 × 22\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " user\\_genome & Domain & Phylum & Class & Order & Family & Genus & Species & fastani\\_reference & fastani\\_reference\\_radius & ⋯ & fastani\\_af & closest\\_placement\\_reference & closest\\_placement\\_radius & closest\\_placement\\_taxonomy & closest\\_placement\\_ani & closest\\_placement\\_af & msa\\_percent & translation\\_table & red\\_value & warnings\\\\\n",
       " <fct> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <fct> & <fct> & ⋯ & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <dbl> & <int> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\t GCA\\_000154565.1\\_ASM15456v1\\_genomic & d\\_\\_Bacteria & p\\_\\_Firmicutes\\_A & c\\_\\_Clostridia & o\\_\\_Oscillospirales & f\\_\\_Ruminococcaceae    & g\\_\\_Anaerotruncus & s\\_\\_Anaerotruncus colihominis & GCF\\_000154565.1 & 95.0 & ⋯ & 1.0 & GCF\\_000154565.1 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Ruminococcaceae;g\\_\\_Anaerotruncus;s\\_\\_Anaerotruncus colihominis    & 100.0 & 1.0 & 97.18 & 11 & N/A & N/A\\\\\n",
       "\t GCA\\_000433975.1\\_MGS528\\_genomic     & d\\_\\_Bacteria & p\\_\\_Firmicutes\\_A & c\\_\\_Clostridia & o\\_\\_Oscillospirales & f\\_\\_Acutalibacteraceae & g\\_\\_Eubacterium\\_R & s\\_\\_Eubacterium\\_R sp000433975 & GCA\\_000433975.1 & 95.0 & ⋯ & 1.0 & GCA\\_000433975.1 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Acutalibacteraceae;g\\_\\_Eubacterium\\_R;s\\_\\_Eubacterium\\_R sp000433975 & 100.0 & 1.0 & 96.57 & 11 & N/A & N/A\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋱ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t GCA\\_900199635.1\\_PRJEB19460\\_genomic      & d\\_\\_Bacteria & p\\_\\_Firmicutes\\_A & c\\_\\_Clostridia & o\\_\\_Oscillospirales & f\\_\\_Ruminococcaceae & g\\_\\_Anaerotruncus & s\\_\\_Anaerotruncus massiliensis & GCF\\_900199635.1 & 95.0 & ⋯ & 1.0  & GCF\\_900199635.1 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Ruminococcaceae;g\\_\\_Anaerotruncus;s\\_\\_Anaerotruncus massiliensis & 100.0 & 1.0  & 96.43 & 11 & N/A & N/A\\\\\n",
       "\t GCA\\_902363605.1\\_MGYG-HGUT-00131\\_genomic & d\\_\\_Bacteria & p\\_\\_Firmicutes\\_A & c\\_\\_Clostridia & o\\_\\_Oscillospirales & f\\_\\_Ruminococcaceae & g\\_\\_Anaerotruncus & s\\_\\_Anaerotruncus rubiinfantis & GCF\\_900078395.1 & 95.0 & ⋯ & 0.91 & GCF\\_900078395.1 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Ruminococcaceae;g\\_\\_Anaerotruncus;s\\_\\_Anaerotruncus rubiinfantis & 99.26 & 0.91 & 96.35 & 11 & N/A & N/A\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 30 × 22\n",
       "\n",
       "| user_genome &lt;fct&gt; | Domain &lt;chr&gt; | Phylum &lt;chr&gt; | Class &lt;chr&gt; | Order &lt;chr&gt; | Family &lt;chr&gt; | Genus &lt;chr&gt; | Species &lt;chr&gt; | fastani_reference &lt;fct&gt; | fastani_reference_radius &lt;fct&gt; | ⋯ ⋯ | fastani_af &lt;fct&gt; | closest_placement_reference &lt;fct&gt; | closest_placement_radius &lt;fct&gt; | closest_placement_taxonomy &lt;fct&gt; | closest_placement_ani &lt;fct&gt; | closest_placement_af &lt;fct&gt; | msa_percent &lt;dbl&gt; | translation_table &lt;int&gt; | red_value &lt;fct&gt; | warnings &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| GCA_000154565.1_ASM15456v1_genomic | d__Bacteria | p__Firmicutes_A | c__Clostridia | o__Oscillospirales | f__Ruminococcaceae    | g__Anaerotruncus | s__Anaerotruncus colihominis | GCF_000154565.1 | 95.0 | ⋯ | 1.0 | GCF_000154565.1 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus colihominis    | 100.0 | 1.0 | 97.18 | 11 | N/A | N/A |\n",
       "| GCA_000433975.1_MGS528_genomic     | d__Bacteria | p__Firmicutes_A | c__Clostridia | o__Oscillospirales | f__Acutalibacteraceae | g__Eubacterium_R | s__Eubacterium_R sp000433975 | GCA_000433975.1 | 95.0 | ⋯ | 1.0 | GCA_000433975.1 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Acutalibacteraceae;g__Eubacterium_R;s__Eubacterium_R sp000433975 | 100.0 | 1.0 | 96.57 | 11 | N/A | N/A |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋱ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| GCA_900199635.1_PRJEB19460_genomic      | d__Bacteria | p__Firmicutes_A | c__Clostridia | o__Oscillospirales | f__Ruminococcaceae | g__Anaerotruncus | s__Anaerotruncus massiliensis | GCF_900199635.1 | 95.0 | ⋯ | 1.0  | GCF_900199635.1 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus massiliensis | 100.0 | 1.0  | 96.43 | 11 | N/A | N/A |\n",
       "| GCA_902363605.1_MGYG-HGUT-00131_genomic | d__Bacteria | p__Firmicutes_A | c__Clostridia | o__Oscillospirales | f__Ruminococcaceae | g__Anaerotruncus | s__Anaerotruncus rubiinfantis | GCF_900078395.1 | 95.0 | ⋯ | 0.91 | GCF_900078395.1 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus rubiinfantis | 99.26 | 0.91 | 96.35 | 11 | N/A | N/A |\n",
       "\n"
      ],
      "text/plain": [
       "   user_genome                             Domain      Phylum         \n",
       "1  GCA_000154565.1_ASM15456v1_genomic      d__Bacteria p__Firmicutes_A\n",
       "2  GCA_000433975.1_MGS528_genomic          d__Bacteria p__Firmicutes_A\n",
       "⋮  ⋮                                       ⋮           ⋮              \n",
       "29 GCA_900199635.1_PRJEB19460_genomic      d__Bacteria p__Firmicutes_A\n",
       "30 GCA_902363605.1_MGYG-HGUT-00131_genomic d__Bacteria p__Firmicutes_A\n",
       "   Class         Order              Family                Genus           \n",
       "1  c__Clostridia o__Oscillospirales f__Ruminococcaceae    g__Anaerotruncus\n",
       "2  c__Clostridia o__Oscillospirales f__Acutalibacteraceae g__Eubacterium_R\n",
       "⋮  ⋮             ⋮                  ⋮                     ⋮               \n",
       "29 c__Clostridia o__Oscillospirales f__Ruminococcaceae    g__Anaerotruncus\n",
       "30 c__Clostridia o__Oscillospirales f__Ruminococcaceae    g__Anaerotruncus\n",
       "   Species                       fastani_reference fastani_reference_radius ⋯\n",
       "1  s__Anaerotruncus colihominis  GCF_000154565.1   95.0                     ⋯\n",
       "2  s__Eubacterium_R sp000433975  GCA_000433975.1   95.0                     ⋯\n",
       "⋮  ⋮                             ⋮                 ⋮                        ⋱\n",
       "29 s__Anaerotruncus massiliensis GCF_900199635.1   95.0                     ⋯\n",
       "30 s__Anaerotruncus rubiinfantis GCF_900078395.1   95.0                     ⋯\n",
       "   fastani_af closest_placement_reference closest_placement_radius\n",
       "1  1.0        GCF_000154565.1             95.0                    \n",
       "2  1.0        GCA_000433975.1             95.0                    \n",
       "⋮  ⋮          ⋮                           ⋮                       \n",
       "29 1.0        GCF_900199635.1             95.0                    \n",
       "30 0.91       GCF_900078395.1             95.0                    \n",
       "   closest_placement_taxonomy                                                                                                      \n",
       "1  d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus colihominis   \n",
       "2  d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Acutalibacteraceae;g__Eubacterium_R;s__Eubacterium_R sp000433975\n",
       "⋮  ⋮                                                                                                                               \n",
       "29 d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus massiliensis  \n",
       "30 d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus rubiinfantis  \n",
       "   closest_placement_ani closest_placement_af msa_percent translation_table\n",
       "1  100.0                 1.0                  97.18       11               \n",
       "2  100.0                 1.0                  96.57       11               \n",
       "⋮  ⋮                     ⋮                    ⋮           ⋮                \n",
       "29 100.0                 1.0                  96.43       11               \n",
       "30 99.26                 0.91                 96.35       11               \n",
       "   red_value warnings\n",
       "1  N/A       N/A     \n",
       "2  N/A       N/A     \n",
       "⋮  ⋮         ⋮       \n",
       "29 N/A       N/A     \n",
       "30 N/A       N/A     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# GTDBTk summary\n",
    "tax = file.path(work_dir, 'LLG', 'gtdbtk', 'gtdbtk_bac_summary.tsv') %>%\n",
    "    read.delim(, sep='\\t') %>%\n",
    "    separate(classification, \n",
    "             c('Domain', 'Phylum', 'Class', 'Order', 'Family', 'Genus', 'Species'),\n",
    "             sep=';') %>%\n",
    "    select(-note, -classification_method, -pplacer_taxonomy,\n",
    "           -other_related_references.genome_id.species_name.radius.ANI.AF.)\n",
    "tax"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-24T05:35:07.269834Z",
     "iopub.status.busy": "2021-04-24T05:35:07.267903Z",
     "iopub.status.idle": "2021-04-24T05:35:07.299237Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- drep --\n",
      "intersect(x,y): 27 \n",
      "setdiff(x,y): 0 \n",
      "setdiff(y,x): 4 \n",
      "union(x,y): 31 \n",
      "-- checkm --\n",
      "intersect(x,y): 31 \n",
      "setdiff(x,y): 0 \n",
      "setdiff(y,x): 0 \n",
      "union(x,y): 31 \n",
      "-- gtdbtk --\n",
      "intersect(x,y): 30 \n",
      "setdiff(x,y): 1 \n",
      "setdiff(y,x): 0 \n",
      "union(x,y): 31 \n"
     ]
    }
   ],
   "source": [
    "# checking overlap\n",
    "cat('-- drep --\\n')\n",
    "overlap(basename(as.character(drep_gen$Fasta)), \n",
    "        basename(as.character(drep$genome)))\n",
    "cat('-- checkm --\\n')\n",
    "overlap(drep$Bin.Id, checkm$Bin.Id)\n",
    "cat('-- gtdbtk --\\n')\n",
    "overlap(drep$Bin.Id, tax$user_genome)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-24T05:35:11.535849Z",
     "iopub.status.busy": "2021-04-24T05:35:11.533849Z",
     "iopub.status.idle": "2021-04-24T05:35:11.606620Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 27 × 60</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>genome</th><th scope=col>completeness</th><th scope=col>contamination</th><th scope=col>Bin.Id</th><th scope=col>Marker.lineage</th><th scope=col>X..genomes</th><th scope=col>X..markers</th><th scope=col>X..marker.sets</th><th scope=col>Completeness</th><th scope=col>Contamination</th><th scope=col>⋯</th><th scope=col>fastani_af</th><th scope=col>closest_placement_reference</th><th scope=col>closest_placement_radius</th><th scope=col>closest_placement_taxonomy</th><th scope=col>closest_placement_ani</th><th scope=col>closest_placement_af</th><th scope=col>msa_percent</th><th scope=col>translation_table</th><th scope=col>red_value</th><th scope=col>warnings</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000154565.1_ASM15456v1_genomic.fna</td><td>97.99</td><td>0.22</td><td>GCA_000154565.1_ASM15456v1_genomic</td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>97.99</td><td>0.22</td><td>⋯</td><td>1.0</td><td>GCF_000154565.1</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus colihominis   </td><td>100.0</td><td>1.0</td><td>97.18</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000433975.1_MGS528_genomic.fna    </td><td>97.99</td><td>0.00</td><td>GCA_000433975.1_MGS528_genomic    </td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>97.99</td><td>0.00</td><td>⋯</td><td>1.0</td><td>GCA_000433975.1</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Acutalibacteraceae;g__Eubacterium_R;s__Eubacterium_R sp000433975</td><td>100.0</td><td>1.0</td><td>96.57</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋱</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_900199635.1_PRJEB19460_genomic.fna     </td><td>99.33</td><td>0.00</td><td>GCA_900199635.1_PRJEB19460_genomic     </td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>99.33</td><td>0.00</td><td>⋯</td><td>1.0 </td><td>GCF_900199635.1</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus massiliensis</td><td>100.0</td><td>1.0 </td><td>96.43</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_902363605.1_MGYG-HGUT-00131_genomic.fna</td><td>97.99</td><td>1.34</td><td>GCA_902363605.1_MGYG-HGUT-00131_genomic</td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>97.99</td><td>1.34</td><td>⋯</td><td>0.91</td><td>GCF_900078395.1</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus rubiinfantis</td><td>99.26</td><td>0.91</td><td>96.35</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 27 × 60\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " genome & completeness & contamination & Bin.Id & Marker.lineage & X..genomes & X..markers & X..marker.sets & Completeness & Contamination & ⋯ & fastani\\_af & closest\\_placement\\_reference & closest\\_placement\\_radius & closest\\_placement\\_taxonomy & closest\\_placement\\_ani & closest\\_placement\\_af & msa\\_percent & translation\\_table & red\\_value & warnings\\\\\n",
       " <fct> & <dbl> & <dbl> & <chr> & <fct> & <int> & <int> & <int> & <dbl> & <dbl> & ⋯ & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <dbl> & <int> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_000154565.1\\_ASM15456v1\\_genomic.fna & 97.99 & 0.22 & GCA\\_000154565.1\\_ASM15456v1\\_genomic & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 97.99 & 0.22 & ⋯ & 1.0 & GCF\\_000154565.1 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Ruminococcaceae;g\\_\\_Anaerotruncus;s\\_\\_Anaerotruncus colihominis    & 100.0 & 1.0 & 97.18 & 11 & N/A & N/A\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_000433975.1\\_MGS528\\_genomic.fna     & 97.99 & 0.00 & GCA\\_000433975.1\\_MGS528\\_genomic     & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 97.99 & 0.00 & ⋯ & 1.0 & GCA\\_000433975.1 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Acutalibacteraceae;g\\_\\_Eubacterium\\_R;s\\_\\_Eubacterium\\_R sp000433975 & 100.0 & 1.0 & 96.57 & 11 & N/A & N/A\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋱ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_900199635.1\\_PRJEB19460\\_genomic.fna      & 99.33 & 0.00 & GCA\\_900199635.1\\_PRJEB19460\\_genomic      & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 99.33 & 0.00 & ⋯ & 1.0  & GCF\\_900199635.1 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Ruminococcaceae;g\\_\\_Anaerotruncus;s\\_\\_Anaerotruncus massiliensis & 100.0 & 1.0  & 96.43 & 11 & N/A & N/A\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_902363605.1\\_MGYG-HGUT-00131\\_genomic.fna & 97.99 & 1.34 & GCA\\_902363605.1\\_MGYG-HGUT-00131\\_genomic & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 97.99 & 1.34 & ⋯ & 0.91 & GCF\\_900078395.1 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Ruminococcaceae;g\\_\\_Anaerotruncus;s\\_\\_Anaerotruncus rubiinfantis & 99.26 & 0.91 & 96.35 & 11 & N/A & N/A\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 27 × 60\n",
       "\n",
       "| genome &lt;fct&gt; | completeness &lt;dbl&gt; | contamination &lt;dbl&gt; | Bin.Id &lt;chr&gt; | Marker.lineage &lt;fct&gt; | X..genomes &lt;int&gt; | X..markers &lt;int&gt; | X..marker.sets &lt;int&gt; | Completeness &lt;dbl&gt; | Contamination &lt;dbl&gt; | ⋯ ⋯ | fastani_af &lt;fct&gt; | closest_placement_reference &lt;fct&gt; | closest_placement_radius &lt;fct&gt; | closest_placement_taxonomy &lt;fct&gt; | closest_placement_ani &lt;fct&gt; | closest_placement_af &lt;fct&gt; | msa_percent &lt;dbl&gt; | translation_table &lt;int&gt; | red_value &lt;fct&gt; | warnings &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000154565.1_ASM15456v1_genomic.fna | 97.99 | 0.22 | GCA_000154565.1_ASM15456v1_genomic | o__Clostridiales (UID1212) | 172 | 263 | 149 | 97.99 | 0.22 | ⋯ | 1.0 | GCF_000154565.1 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus colihominis    | 100.0 | 1.0 | 97.18 | 11 | N/A | N/A |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000433975.1_MGS528_genomic.fna     | 97.99 | 0.00 | GCA_000433975.1_MGS528_genomic     | o__Clostridiales (UID1212) | 172 | 263 | 149 | 97.99 | 0.00 | ⋯ | 1.0 | GCA_000433975.1 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Acutalibacteraceae;g__Eubacterium_R;s__Eubacterium_R sp000433975 | 100.0 | 1.0 | 96.57 | 11 | N/A | N/A |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋱ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_900199635.1_PRJEB19460_genomic.fna      | 99.33 | 0.00 | GCA_900199635.1_PRJEB19460_genomic      | o__Clostridiales (UID1212) | 172 | 263 | 149 | 99.33 | 0.00 | ⋯ | 1.0  | GCF_900199635.1 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus massiliensis | 100.0 | 1.0  | 96.43 | 11 | N/A | N/A |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_902363605.1_MGYG-HGUT-00131_genomic.fna | 97.99 | 1.34 | GCA_902363605.1_MGYG-HGUT-00131_genomic | o__Clostridiales (UID1212) | 172 | 263 | 149 | 97.99 | 1.34 | ⋯ | 0.91 | GCF_900078395.1 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus rubiinfantis | 99.26 | 0.91 | 96.35 | 11 | N/A | N/A |\n",
       "\n"
      ],
      "text/plain": [
       "   genome                                                                                           \n",
       "1  /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000154565.1_ASM15456v1_genomic.fna     \n",
       "2  /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000433975.1_MGS528_genomic.fna         \n",
       "⋮  ⋮                                                                                                \n",
       "26 /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_900199635.1_PRJEB19460_genomic.fna     \n",
       "27 /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_902363605.1_MGYG-HGUT-00131_genomic.fna\n",
       "   completeness contamination Bin.Id                                 \n",
       "1  97.99        0.22          GCA_000154565.1_ASM15456v1_genomic     \n",
       "2  97.99        0.00          GCA_000433975.1_MGS528_genomic         \n",
       "⋮  ⋮            ⋮             ⋮                                      \n",
       "26 99.33        0.00          GCA_900199635.1_PRJEB19460_genomic     \n",
       "27 97.99        1.34          GCA_902363605.1_MGYG-HGUT-00131_genomic\n",
       "   Marker.lineage             X..genomes X..markers X..marker.sets Completeness\n",
       "1  o__Clostridiales (UID1212) 172        263        149            97.99       \n",
       "2  o__Clostridiales (UID1212) 172        263        149            97.99       \n",
       "⋮  ⋮                          ⋮          ⋮          ⋮              ⋮           \n",
       "26 o__Clostridiales (UID1212) 172        263        149            99.33       \n",
       "27 o__Clostridiales (UID1212) 172        263        149            97.99       \n",
       "   Contamination ⋯ fastani_af closest_placement_reference\n",
       "1  0.22          ⋯ 1.0        GCF_000154565.1            \n",
       "2  0.00          ⋯ 1.0        GCA_000433975.1            \n",
       "⋮  ⋮             ⋱ ⋮          ⋮                          \n",
       "26 0.00          ⋯ 1.0        GCF_900199635.1            \n",
       "27 1.34          ⋯ 0.91       GCF_900078395.1            \n",
       "   closest_placement_radius\n",
       "1  95.0                    \n",
       "2  95.0                    \n",
       "⋮  ⋮                       \n",
       "26 95.0                    \n",
       "27 95.0                    \n",
       "   closest_placement_taxonomy                                                                                                      \n",
       "1  d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus colihominis   \n",
       "2  d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Acutalibacteraceae;g__Eubacterium_R;s__Eubacterium_R sp000433975\n",
       "⋮  ⋮                                                                                                                               \n",
       "26 d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus massiliensis  \n",
       "27 d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus rubiinfantis  \n",
       "   closest_placement_ani closest_placement_af msa_percent translation_table\n",
       "1  100.0                 1.0                  97.18       11               \n",
       "2  100.0                 1.0                  96.57       11               \n",
       "⋮  ⋮                     ⋮                    ⋮           ⋮                \n",
       "26 100.0                 1.0                  96.43       11               \n",
       "27 99.26                 0.91                 96.35       11               \n",
       "   red_value warnings\n",
       "1  N/A       N/A     \n",
       "2  N/A       N/A     \n",
       "⋮  ⋮         ⋮       \n",
       "26 N/A       N/A     \n",
       "27 N/A       N/A     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# joining based on Bin.Id\n",
    "drep = drep %>%\n",
    "    inner_join(checkm, c('Bin.Id')) %>%\n",
    "    mutate(GEN = genome %>% as.character %>% basename) %>%\n",
    "    inner_join(drep_gen %>% mutate(GEN = Fasta %>% as.character %>% basename),\n",
    "               by=c('GEN')) %>%\n",
    "    inner_join(tax, c('Bin.Id'='user_genome')) #%>%\n",
    "drep"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-24T05:35:41.675698Z",
     "iopub.status.busy": "2021-04-24T05:35:41.673878Z",
     "iopub.status.idle": "2021-04-24T05:35:41.726713Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 26 × 60</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>genome</th><th scope=col>completeness</th><th scope=col>contamination</th><th scope=col>Bin.Id</th><th scope=col>Marker.lineage</th><th scope=col>X..genomes</th><th scope=col>X..markers</th><th scope=col>X..marker.sets</th><th scope=col>Completeness</th><th scope=col>Contamination</th><th scope=col>⋯</th><th scope=col>fastani_af</th><th scope=col>closest_placement_reference</th><th scope=col>closest_placement_radius</th><th scope=col>closest_placement_taxonomy</th><th scope=col>closest_placement_ani</th><th scope=col>closest_placement_af</th><th scope=col>msa_percent</th><th scope=col>translation_table</th><th scope=col>red_value</th><th scope=col>warnings</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000154565.1_ASM15456v1_genomic.fna</td><td>97.99</td><td>0.22</td><td>GCA_000154565.1_ASM15456v1_genomic</td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>97.99</td><td>0.22</td><td>⋯</td><td>1.0</td><td>GCF_000154565.1</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus colihominis   </td><td>100.0</td><td>1.0</td><td>97.18</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000433975.1_MGS528_genomic.fna    </td><td>97.99</td><td>0.00</td><td>GCA_000433975.1_MGS528_genomic    </td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>97.99</td><td>0.00</td><td>⋯</td><td>1.0</td><td>GCA_000433975.1</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Acutalibacteraceae;g__Eubacterium_R;s__Eubacterium_R sp000433975</td><td>100.0</td><td>1.0</td><td>96.57</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋱</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_900199635.1_PRJEB19460_genomic.fna     </td><td>99.33</td><td>0.00</td><td>GCA_900199635.1_PRJEB19460_genomic     </td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>99.33</td><td>0.00</td><td>⋯</td><td>1.0 </td><td>GCF_900199635.1</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus massiliensis</td><td>100.0</td><td>1.0 </td><td>96.43</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_902363605.1_MGYG-HGUT-00131_genomic.fna</td><td>97.99</td><td>1.34</td><td>GCA_902363605.1_MGYG-HGUT-00131_genomic</td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>97.99</td><td>1.34</td><td>⋯</td><td>0.91</td><td>GCF_900078395.1</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus rubiinfantis</td><td>99.26</td><td>0.91</td><td>96.35</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 26 × 60\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " genome & completeness & contamination & Bin.Id & Marker.lineage & X..genomes & X..markers & X..marker.sets & Completeness & Contamination & ⋯ & fastani\\_af & closest\\_placement\\_reference & closest\\_placement\\_radius & closest\\_placement\\_taxonomy & closest\\_placement\\_ani & closest\\_placement\\_af & msa\\_percent & translation\\_table & red\\_value & warnings\\\\\n",
       " <fct> & <dbl> & <dbl> & <chr> & <fct> & <int> & <int> & <int> & <dbl> & <dbl> & ⋯ & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <dbl> & <int> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_000154565.1\\_ASM15456v1\\_genomic.fna & 97.99 & 0.22 & GCA\\_000154565.1\\_ASM15456v1\\_genomic & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 97.99 & 0.22 & ⋯ & 1.0 & GCF\\_000154565.1 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Ruminococcaceae;g\\_\\_Anaerotruncus;s\\_\\_Anaerotruncus colihominis    & 100.0 & 1.0 & 97.18 & 11 & N/A & N/A\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_000433975.1\\_MGS528\\_genomic.fna     & 97.99 & 0.00 & GCA\\_000433975.1\\_MGS528\\_genomic     & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 97.99 & 0.00 & ⋯ & 1.0 & GCA\\_000433975.1 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Acutalibacteraceae;g\\_\\_Eubacterium\\_R;s\\_\\_Eubacterium\\_R sp000433975 & 100.0 & 1.0 & 96.57 & 11 & N/A & N/A\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋱ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_900199635.1\\_PRJEB19460\\_genomic.fna      & 99.33 & 0.00 & GCA\\_900199635.1\\_PRJEB19460\\_genomic      & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 99.33 & 0.00 & ⋯ & 1.0  & GCF\\_900199635.1 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Ruminococcaceae;g\\_\\_Anaerotruncus;s\\_\\_Anaerotruncus massiliensis & 100.0 & 1.0  & 96.43 & 11 & N/A & N/A\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_902363605.1\\_MGYG-HGUT-00131\\_genomic.fna & 97.99 & 1.34 & GCA\\_902363605.1\\_MGYG-HGUT-00131\\_genomic & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 97.99 & 1.34 & ⋯ & 0.91 & GCF\\_900078395.1 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Ruminococcaceae;g\\_\\_Anaerotruncus;s\\_\\_Anaerotruncus rubiinfantis & 99.26 & 0.91 & 96.35 & 11 & N/A & N/A\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 26 × 60\n",
       "\n",
       "| genome &lt;fct&gt; | completeness &lt;dbl&gt; | contamination &lt;dbl&gt; | Bin.Id &lt;chr&gt; | Marker.lineage &lt;fct&gt; | X..genomes &lt;int&gt; | X..markers &lt;int&gt; | X..marker.sets &lt;int&gt; | Completeness &lt;dbl&gt; | Contamination &lt;dbl&gt; | ⋯ ⋯ | fastani_af &lt;fct&gt; | closest_placement_reference &lt;fct&gt; | closest_placement_radius &lt;fct&gt; | closest_placement_taxonomy &lt;fct&gt; | closest_placement_ani &lt;fct&gt; | closest_placement_af &lt;fct&gt; | msa_percent &lt;dbl&gt; | translation_table &lt;int&gt; | red_value &lt;fct&gt; | warnings &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000154565.1_ASM15456v1_genomic.fna | 97.99 | 0.22 | GCA_000154565.1_ASM15456v1_genomic | o__Clostridiales (UID1212) | 172 | 263 | 149 | 97.99 | 0.22 | ⋯ | 1.0 | GCF_000154565.1 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus colihominis    | 100.0 | 1.0 | 97.18 | 11 | N/A | N/A |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000433975.1_MGS528_genomic.fna     | 97.99 | 0.00 | GCA_000433975.1_MGS528_genomic     | o__Clostridiales (UID1212) | 172 | 263 | 149 | 97.99 | 0.00 | ⋯ | 1.0 | GCA_000433975.1 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Acutalibacteraceae;g__Eubacterium_R;s__Eubacterium_R sp000433975 | 100.0 | 1.0 | 96.57 | 11 | N/A | N/A |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋱ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_900199635.1_PRJEB19460_genomic.fna      | 99.33 | 0.00 | GCA_900199635.1_PRJEB19460_genomic      | o__Clostridiales (UID1212) | 172 | 263 | 149 | 99.33 | 0.00 | ⋯ | 1.0  | GCF_900199635.1 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus massiliensis | 100.0 | 1.0  | 96.43 | 11 | N/A | N/A |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_902363605.1_MGYG-HGUT-00131_genomic.fna | 97.99 | 1.34 | GCA_902363605.1_MGYG-HGUT-00131_genomic | o__Clostridiales (UID1212) | 172 | 263 | 149 | 97.99 | 1.34 | ⋯ | 0.91 | GCF_900078395.1 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus rubiinfantis | 99.26 | 0.91 | 96.35 | 11 | N/A | N/A |\n",
       "\n"
      ],
      "text/plain": [
       "   genome                                                                                           \n",
       "1  /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000154565.1_ASM15456v1_genomic.fna     \n",
       "2  /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000433975.1_MGS528_genomic.fna         \n",
       "⋮  ⋮                                                                                                \n",
       "25 /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_900199635.1_PRJEB19460_genomic.fna     \n",
       "26 /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_902363605.1_MGYG-HGUT-00131_genomic.fna\n",
       "   completeness contamination Bin.Id                                 \n",
       "1  97.99        0.22          GCA_000154565.1_ASM15456v1_genomic     \n",
       "2  97.99        0.00          GCA_000433975.1_MGS528_genomic         \n",
       "⋮  ⋮            ⋮             ⋮                                      \n",
       "25 99.33        0.00          GCA_900199635.1_PRJEB19460_genomic     \n",
       "26 97.99        1.34          GCA_902363605.1_MGYG-HGUT-00131_genomic\n",
       "   Marker.lineage             X..genomes X..markers X..marker.sets Completeness\n",
       "1  o__Clostridiales (UID1212) 172        263        149            97.99       \n",
       "2  o__Clostridiales (UID1212) 172        263        149            97.99       \n",
       "⋮  ⋮                          ⋮          ⋮          ⋮              ⋮           \n",
       "25 o__Clostridiales (UID1212) 172        263        149            99.33       \n",
       "26 o__Clostridiales (UID1212) 172        263        149            97.99       \n",
       "   Contamination ⋯ fastani_af closest_placement_reference\n",
       "1  0.22          ⋯ 1.0        GCF_000154565.1            \n",
       "2  0.00          ⋯ 1.0        GCA_000433975.1            \n",
       "⋮  ⋮             ⋱ ⋮          ⋮                          \n",
       "25 0.00          ⋯ 1.0        GCF_900199635.1            \n",
       "26 1.34          ⋯ 0.91       GCF_900078395.1            \n",
       "   closest_placement_radius\n",
       "1  95.0                    \n",
       "2  95.0                    \n",
       "⋮  ⋮                       \n",
       "25 95.0                    \n",
       "26 95.0                    \n",
       "   closest_placement_taxonomy                                                                                                      \n",
       "1  d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus colihominis   \n",
       "2  d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Acutalibacteraceae;g__Eubacterium_R;s__Eubacterium_R sp000433975\n",
       "⋮  ⋮                                                                                                                               \n",
       "25 d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus massiliensis  \n",
       "26 d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Anaerotruncus;s__Anaerotruncus rubiinfantis  \n",
       "   closest_placement_ani closest_placement_af msa_percent translation_table\n",
       "1  100.0                 1.0                  97.18       11               \n",
       "2  100.0                 1.0                  96.57       11               \n",
       "⋮  ⋮                     ⋮                    ⋮           ⋮                \n",
       "25 100.0                 1.0                  96.43       11               \n",
       "26 99.26                 0.91                 96.35       11               \n",
       "   red_value warnings\n",
       "1  N/A       N/A     \n",
       "2  N/A       N/A     \n",
       "⋮  ⋮         ⋮       \n",
       "25 N/A       N/A     \n",
       "26 N/A       N/A     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# filtering by quality\n",
    "hq_genomes = drep %>%\n",
    "    filter(completeness >= 90,\n",
    "           contamination < 5,\n",
    "           Strain.heterogeneity < 50,\n",
    "           X..contigs < 300,\n",
    "           Mean.contig.length..bp. > 10000)\n",
    "hq_genomes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-24T05:35:43.258991Z",
     "iopub.status.busy": "2021-04-24T05:35:43.257053Z",
     "iopub.status.idle": "2021-04-24T05:35:43.291300Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A tibble: 3 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Family</th><th scope=col>Genus</th><th scope=col>n_genomes</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>f__Acutalibacteraceae</td><td>g__Eubacterium_R</td><td> 1</td></tr>\n",
       "\t<tr><td>f__Anaerovoracaceae  </td><td>g__Emergencia   </td><td> 1</td></tr>\n",
       "\t<tr><td>f__Ruminococcaceae   </td><td>g__Anaerotruncus</td><td>24</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 3 × 3\n",
       "\\begin{tabular}{lll}\n",
       " Family & Genus & n\\_genomes\\\\\n",
       " <chr> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t f\\_\\_Acutalibacteraceae & g\\_\\_Eubacterium\\_R &  1\\\\\n",
       "\t f\\_\\_Anaerovoracaceae   & g\\_\\_Emergencia    &  1\\\\\n",
       "\t f\\_\\_Ruminococcaceae    & g\\_\\_Anaerotruncus & 24\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 3 × 3\n",
       "\n",
       "| Family &lt;chr&gt; | Genus &lt;chr&gt; | n_genomes &lt;int&gt; |\n",
       "|---|---|---|\n",
       "| f__Acutalibacteraceae | g__Eubacterium_R |  1 |\n",
       "| f__Anaerovoracaceae   | g__Emergencia    |  1 |\n",
       "| f__Ruminococcaceae    | g__Anaerotruncus | 24 |\n",
       "\n"
      ],
      "text/plain": [
       "  Family                Genus            n_genomes\n",
       "1 f__Acutalibacteraceae g__Eubacterium_R  1       \n",
       "2 f__Anaerovoracaceae   g__Emergencia     1       \n",
       "3 f__Ruminococcaceae    g__Anaerotruncus 24       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# summarizing the taxonomy\n",
    "df.dims(20)\n",
    "hq_genomes %>%\n",
    "    group_by(Family, Genus) %>%\n",
    "    summarize(n_genomes = n(), .groups='drop')\n",
    "df.dims()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-24T05:35:49.120045Z",
     "iopub.status.busy": "2021-04-24T05:35:49.117946Z",
     "iopub.status.idle": "2021-04-24T05:35:49.154844Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "File written: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3//Anaerotruncus/samples_genomes_hq.txt \n"
     ]
    }
   ],
   "source": [
    "# writing samples table for LLPRIMER\n",
    "outfile = file.path(work_dir, 'samples_genomes_hq.txt')\n",
    "hq_genomes %>%\n",
    "    select(Bin.Id, Fasta) %>%\n",
    "    rename('Taxon' = Bin.Id) %>%\n",
    "    mutate(Taxon = gsub('_chromosome.+', '', Taxon),\n",
    "           Taxon = gsub('_bin_.+', '', Taxon),\n",
    "           Taxon = gsub('_genomic', '', Taxon),\n",
    "           Taxon = gsub('_annotated_assembly', '', Taxon),\n",
    "           Taxid = taxid,\n",
    "           Domain = domain) %>%\n",
    "    write_table(outfile)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Primer design"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Config"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-24T05:36:49.596772Z",
     "iopub.status.busy": "2021-04-24T05:36:49.595702Z",
     "iopub.status.idle": "2021-04-24T05:36:49.624887Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#-- I/O --#\n",
      "samples_file: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Anaerotruncus/samples_genomes_hq.txt\n",
      "\n",
      "# output location\n",
      "output_dir: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Anaerotruncus/primers/\n",
      "\n",
      "# temporary file directory (your username will be added automatically)\n",
      "tmp_dir: /ebio/abt3_scratch/\n",
      "\n",
      "#-- software parameters --#\n",
      "# See the README for a description\n",
      "params:\n",
      "  ionice: -c 3\n",
      "  cgp:\n",
      "    prokka: \"\"\n",
      "    mmseqs:\n",
      "      method: cluster    # or linclust (faster)\n",
      "      run: --min-seq-id 0.8 -c 0.8\n",
      "    vsearch: --id 0.94\n",
      "    core_genes:\n",
      "      cds: --perc-genomes-cds 100 --copies-per-genome-cds 1 --max-clusters-cds 500\n",
      "      rRNA: --perc-genomes-rrna 100 --copies-per-genome-rrna 10 --max-clusters-rrna 500\n",
      "    align:\n",
      "      method: linsi\n",
      "      params: --auto --maxiterate 1000\n",
      "    primer3:\n",
      "      consensus: --consensus-threshold 0.34\n",
      "      number: --num-raw-primers 1000 --num-final-primers 10\n",
      "      size: --opt-size 20 --min-size 18 --max-size 26\n",
      "      product: --opt-prod-size 150 --min-prod-size 100 --max-prod-size 250\n",
      "      Tm: --opt-tm 62 --min-tm 54 --max-tm 70 --max-tm-diff 2.5\n",
      "      PCR: --oligo-DNA 50 --salt-monovalent 50 --salt-divalent 1.5 --dNTPs 0.2\n",
      "      GC: --opt-gc 50 --min-gc 25 --max-gc 75\n",
      "      degeneracy: --max-degeneracy 128 --max-degeneracy-3prime 6 --window-3prime 5\n",
      "      internal_oligo: --make-oligo 0 --int-opt-size 20 --int-opt-tm 60\n",
      "    blast_nontarget:\n",
      "      run_locally: False\n",
      "      cds: -evalue 1e-5 -max_target_seqs 30\n",
      "      rRNA: -evalue 1e-5 -max_target_seqs 30\n",
      "    blastn: -max_target_seqs 500 -perc_identity 90\n",
      "    blast_filter: --perc-len 80 --min-amplicon-len 30 --max-amplicon-len 2000\n",
      "    seqkit_amplicon: \"\"\n",
      "\n",
      "#-- database --#\n",
      "databases:\n",
      "  blast:\n",
      "    nucl: /ebio/abt3_projects/databases_no-backup/NCBI_blastdb/nt\n",
      "    prot: /ebio/abt3_projects/databases_no-backup/NCBI_blastdb/nr\n",
      "    rRNA:\n",
      "      SSU: /ebio/abt3_projects/databases_no-backup/NCBI_rRNA/2021-01/SSU/16S_ribosomal_RNA\n",
      "      LSU: /ebio/abt3_projects/databases_no-backup/NCBI_rRNA/2021-01/LSU/LSU_prokaryote_rRNA\n",
      "    gene_names: /ebio/abt3_projects/databases_no-backup/NCBI_nr_geneIDs/2021-01/nr_accession-geneNames.pkl\n",
      "  taxonomy:\n",
      "    names: /ebio/abt3_projects/databases_no-backup/NCBI_taxonomy/2020-12-23/names.dmp\n",
      "    nodes: /ebio/abt3_projects/databases_no-backup/NCBI_taxonomy/2020-12-23/nodes.dmp\n",
      "\n",
      "#-- snakemake pipeline --#\n",
      "pipeline:\n",
      "  snakemake_folder: ./\n",
      "  script_folder: bin/scripts/\n",
      "  name: LLPRIMER"
     ]
    }
   ],
   "source": [
    "F = file.path(work_dir, 'primers', 'config.yaml')\n",
    "cat_file(F)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Run"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "(snakemake) @ rick:/ebio/abt3_projects/software/dev/ll_pipelines/llprimer\n",
    "$ screen -L -S llprimer-An ./snakemake_sge.sh /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Anaerotruncus/primers/config.yaml 30 --notemp\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Summary"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Primers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-24T13:17:46.070697Z",
     "iopub.status.busy": "2021-04-24T13:17:46.068574Z",
     "iopub.status.idle": "2021-04-24T13:17:46.124841Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No. of unique primers: 10 \n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 20 × 22</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>gene_type</th><th scope=col>cluster_id</th><th scope=col>primer_set</th><th scope=col>amplicon_size_consensus</th><th scope=col>amplicon_size_avg</th><th scope=col>amplicon_size_sd</th><th scope=col>primer_id</th><th scope=col>primer_type</th><th scope=col>sequence</th><th scope=col>length</th><th scope=col>⋯</th><th scope=col>position_start</th><th scope=col>position_end</th><th scope=col>Tm_avg</th><th scope=col>Tm_sd</th><th scope=col>GC_avg</th><th scope=col>GC_sd</th><th scope=col>hairpin_avg</th><th scope=col>hairpin_sd</th><th scope=col>homodimer_avg</th><th scope=col>homodimer_sd</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>cds</td><td>2</td><td>73</td><td>109</td><td>109</td><td>0</td><td>73f</td><td>PRIMER_LEFT </td><td>TGAAAGTHAGASCTTCYGTDAA</td><td>22</td><td>⋯</td><td> 1</td><td> 23</td><td>56.23671</td><td>1.915052</td><td>37.12121</td><td>3.787879</td><td>7.511429</td><td>16.79607</td><td>-27.56715</td><td>2.442498e+01</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>73</td><td>109</td><td>109</td><td>0</td><td>73r</td><td>PRIMER_RIGHT</td><td>CCYTGWCKCTGTTTRTGCTT  </td><td>20</td><td>⋯</td><td>90</td><td>110</td><td>57.99806</td><td>2.451058</td><td>47.50000</td><td>4.330127</td><td>0.000000</td><td> 0.00000</td><td>-33.75316</td><td>1.421085e-14</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋱</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>925</td><td>107</td><td>107</td><td>0</td><td>925f</td><td>PRIMER_LEFT </td><td>GAAAGTHAGASCTTCYGTDA</td><td>20</td><td>⋯</td><td> 2</td><td> 22</td><td>53.48922</td><td>2.087977</td><td>40.83333</td><td>4.166667</td><td>7.511429</td><td>16.79607</td><td>-28.87287</td><td>2.245080e+01</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>925</td><td>107</td><td>107</td><td>0</td><td>925r</td><td>PRIMER_RIGHT</td><td>CYTGWCKCTGTTTRTGCTT </td><td>19</td><td>⋯</td><td>90</td><td>109</td><td>55.29904</td><td>2.585134</td><td>44.73684</td><td>4.558028</td><td>0.000000</td><td> 0.00000</td><td>-33.75316</td><td>1.421085e-14</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 22\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " gene\\_type & cluster\\_id & primer\\_set & amplicon\\_size\\_consensus & amplicon\\_size\\_avg & amplicon\\_size\\_sd & primer\\_id & primer\\_type & sequence & length & ⋯ & position\\_start & position\\_end & Tm\\_avg & Tm\\_sd & GC\\_avg & GC\\_sd & hairpin\\_avg & hairpin\\_sd & homodimer\\_avg & homodimer\\_sd\\\\\n",
       " <fct> & <int> & <int> & <int> & <dbl> & <dbl> & <fct> & <fct> & <fct> & <int> & ⋯ & <int> & <int> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t cds & 2 & 73 & 109 & 109 & 0 & 73f & PRIMER\\_LEFT  & TGAAAGTHAGASCTTCYGTDAA & 22 & ⋯ &  1 &  23 & 56.23671 & 1.915052 & 37.12121 & 3.787879 & 7.511429 & 16.79607 & -27.56715 & 2.442498e+01\\\\\n",
       "\t cds & 2 & 73 & 109 & 109 & 0 & 73r & PRIMER\\_RIGHT & CCYTGWCKCTGTTTRTGCTT   & 20 & ⋯ & 90 & 110 & 57.99806 & 2.451058 & 47.50000 & 4.330127 & 0.000000 &  0.00000 & -33.75316 & 1.421085e-14\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋱ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t cds & 2 & 925 & 107 & 107 & 0 & 925f & PRIMER\\_LEFT  & GAAAGTHAGASCTTCYGTDA & 20 & ⋯ &  2 &  22 & 53.48922 & 2.087977 & 40.83333 & 4.166667 & 7.511429 & 16.79607 & -28.87287 & 2.245080e+01\\\\\n",
       "\t cds & 2 & 925 & 107 & 107 & 0 & 925r & PRIMER\\_RIGHT & CYTGWCKCTGTTTRTGCTT  & 19 & ⋯ & 90 & 109 & 55.29904 & 2.585134 & 44.73684 & 4.558028 & 0.000000 &  0.00000 & -33.75316 & 1.421085e-14\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 22\n",
       "\n",
       "| gene_type &lt;fct&gt; | cluster_id &lt;int&gt; | primer_set &lt;int&gt; | amplicon_size_consensus &lt;int&gt; | amplicon_size_avg &lt;dbl&gt; | amplicon_size_sd &lt;dbl&gt; | primer_id &lt;fct&gt; | primer_type &lt;fct&gt; | sequence &lt;fct&gt; | length &lt;int&gt; | ⋯ ⋯ | position_start &lt;int&gt; | position_end &lt;int&gt; | Tm_avg &lt;dbl&gt; | Tm_sd &lt;dbl&gt; | GC_avg &lt;dbl&gt; | GC_sd &lt;dbl&gt; | hairpin_avg &lt;dbl&gt; | hairpin_sd &lt;dbl&gt; | homodimer_avg &lt;dbl&gt; | homodimer_sd &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| cds | 2 | 73 | 109 | 109 | 0 | 73f | PRIMER_LEFT  | TGAAAGTHAGASCTTCYGTDAA | 22 | ⋯ |  1 |  23 | 56.23671 | 1.915052 | 37.12121 | 3.787879 | 7.511429 | 16.79607 | -27.56715 | 2.442498e+01 |\n",
       "| cds | 2 | 73 | 109 | 109 | 0 | 73r | PRIMER_RIGHT | CCYTGWCKCTGTTTRTGCTT   | 20 | ⋯ | 90 | 110 | 57.99806 | 2.451058 | 47.50000 | 4.330127 | 0.000000 |  0.00000 | -33.75316 | 1.421085e-14 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋱ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| cds | 2 | 925 | 107 | 107 | 0 | 925f | PRIMER_LEFT  | GAAAGTHAGASCTTCYGTDA | 20 | ⋯ |  2 |  22 | 53.48922 | 2.087977 | 40.83333 | 4.166667 | 7.511429 | 16.79607 | -28.87287 | 2.245080e+01 |\n",
       "| cds | 2 | 925 | 107 | 107 | 0 | 925r | PRIMER_RIGHT | CYTGWCKCTGTTTRTGCTT  | 19 | ⋯ | 90 | 109 | 55.29904 | 2.585134 | 44.73684 | 4.558028 | 0.000000 |  0.00000 | -33.75316 | 1.421085e-14 |\n",
       "\n"
      ],
      "text/plain": [
       "   gene_type cluster_id primer_set amplicon_size_consensus amplicon_size_avg\n",
       "1  cds       2          73         109                     109              \n",
       "2  cds       2          73         109                     109              \n",
       "⋮  ⋮         ⋮          ⋮          ⋮                       ⋮                \n",
       "19 cds       2          925        107                     107              \n",
       "20 cds       2          925        107                     107              \n",
       "   amplicon_size_sd primer_id primer_type  sequence               length ⋯\n",
       "1  0                73f       PRIMER_LEFT  TGAAAGTHAGASCTTCYGTDAA 22     ⋯\n",
       "2  0                73r       PRIMER_RIGHT CCYTGWCKCTGTTTRTGCTT   20     ⋯\n",
       "⋮  ⋮                ⋮         ⋮            ⋮                      ⋮      ⋱\n",
       "19 0                925f      PRIMER_LEFT  GAAAGTHAGASCTTCYGTDA   20     ⋯\n",
       "20 0                925r      PRIMER_RIGHT CYTGWCKCTGTTTRTGCTT    19     ⋯\n",
       "   position_start position_end Tm_avg   Tm_sd    GC_avg   GC_sd    hairpin_avg\n",
       "1   1              23          56.23671 1.915052 37.12121 3.787879 7.511429   \n",
       "2  90             110          57.99806 2.451058 47.50000 4.330127 0.000000   \n",
       "⋮  ⋮              ⋮            ⋮        ⋮        ⋮        ⋮        ⋮          \n",
       "19  2              22          53.48922 2.087977 40.83333 4.166667 7.511429   \n",
       "20 90             109          55.29904 2.585134 44.73684 4.558028 0.000000   \n",
       "   hairpin_sd homodimer_avg homodimer_sd\n",
       "1  16.79607   -27.56715     2.442498e+01\n",
       "2   0.00000   -33.75316     1.421085e-14\n",
       "⋮  ⋮          ⋮             ⋮           \n",
       "19 16.79607   -28.87287     2.245080e+01\n",
       "20  0.00000   -33.75316     1.421085e-14"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "primer_info = read.delim(file.path(work_dir, 'primers', 'cgp', 'primers_final_info.tsv'), sep='\\t')\n",
    "primer_info %>% unique_n('primers', primer_set)\n",
    "primer_info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Gene clusters"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-24T13:17:47.415226Z",
     "iopub.status.busy": "2021-04-24T13:17:47.412911Z",
     "iopub.status.idle": "2021-04-24T13:17:47.459896Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No. of clusters 1 \n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 26 × 13</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>gene_type</th><th scope=col>cluster_id</th><th scope=col>seq_uuid</th><th scope=col>seq_orig_name</th><th scope=col>contig_id</th><th scope=col>taxon</th><th scope=col>start</th><th scope=col>end</th><th scope=col>score</th><th scope=col>strand</th><th scope=col>annotation</th><th scope=col>cluster_name</th><th scope=col>clust_id</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>cds</td><td>2</td><td>95cf4c86c7cf4893b51f3ba76fc4d9e9</td><td>KHIIGKDO_03304</td><td>DS544172.1</td><td>GCA_000154565.1_ASM15456v1</td><td> 5547</td><td> 5660</td><td>.</td><td>-</td><td>50S ribosomal protein L36</td><td>492e52027403480f8ebe523409d9267f</td><td>2</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>663b3413e3c644b0bf9751a313c1cb7a</td><td>AAEGJLIP_00392</td><td>HF999344.1</td><td>GCA_000433975.1_MGS528    </td><td>28097</td><td>28210</td><td>.</td><td>+</td><td>50S ribosomal protein L36</td><td>492e52027403480f8ebe523409d9267f</td><td>2</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>5c515b50cccc4c8cb8d4d071d478865d</td><td>HDMAFNJB_00208</td><td>LT962687.1       </td><td>GCA_900199635.1_PRJEB19460     </td><td>100615</td><td>100728</td><td>.</td><td>+</td><td>50S ribosomal protein L36</td><td>492e52027403480f8ebe523409d9267f</td><td>2</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>1f2a10cd3735420c99aa8058f1c4a798</td><td>LFEHCDIH_00979</td><td>CABJAH010000003.1</td><td>GCA_902363605.1_MGYG-HGUT-00131</td><td> 74458</td><td> 74571</td><td>.</td><td>+</td><td>50S ribosomal protein L36</td><td>492e52027403480f8ebe523409d9267f</td><td>2</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 26 × 13\n",
       "\\begin{tabular}{lllllllllllll}\n",
       " gene\\_type & cluster\\_id & seq\\_uuid & seq\\_orig\\_name & contig\\_id & taxon & start & end & score & strand & annotation & cluster\\_name & clust\\_id\\\\\n",
       " <fct> & <int> & <fct> & <fct> & <fct> & <fct> & <int> & <int> & <fct> & <fct> & <fct> & <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t cds & 2 & 95cf4c86c7cf4893b51f3ba76fc4d9e9 & KHIIGKDO\\_03304 & DS544172.1 & GCA\\_000154565.1\\_ASM15456v1 &  5547 &  5660 & . & - & 50S ribosomal protein L36 & 492e52027403480f8ebe523409d9267f & 2\\\\\n",
       "\t cds & 2 & 663b3413e3c644b0bf9751a313c1cb7a & AAEGJLIP\\_00392 & HF999344.1 & GCA\\_000433975.1\\_MGS528     & 28097 & 28210 & . & + & 50S ribosomal protein L36 & 492e52027403480f8ebe523409d9267f & 2\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t cds & 2 & 5c515b50cccc4c8cb8d4d071d478865d & HDMAFNJB\\_00208 & LT962687.1        & GCA\\_900199635.1\\_PRJEB19460      & 100615 & 100728 & . & + & 50S ribosomal protein L36 & 492e52027403480f8ebe523409d9267f & 2\\\\\n",
       "\t cds & 2 & 1f2a10cd3735420c99aa8058f1c4a798 & LFEHCDIH\\_00979 & CABJAH010000003.1 & GCA\\_902363605.1\\_MGYG-HGUT-00131 &  74458 &  74571 & . & + & 50S ribosomal protein L36 & 492e52027403480f8ebe523409d9267f & 2\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 26 × 13\n",
       "\n",
       "| gene_type &lt;fct&gt; | cluster_id &lt;int&gt; | seq_uuid &lt;fct&gt; | seq_orig_name &lt;fct&gt; | contig_id &lt;fct&gt; | taxon &lt;fct&gt; | start &lt;int&gt; | end &lt;int&gt; | score &lt;fct&gt; | strand &lt;fct&gt; | annotation &lt;fct&gt; | cluster_name &lt;fct&gt; | clust_id &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| cds | 2 | 95cf4c86c7cf4893b51f3ba76fc4d9e9 | KHIIGKDO_03304 | DS544172.1 | GCA_000154565.1_ASM15456v1 |  5547 |  5660 | . | - | 50S ribosomal protein L36 | 492e52027403480f8ebe523409d9267f | 2 |\n",
       "| cds | 2 | 663b3413e3c644b0bf9751a313c1cb7a | AAEGJLIP_00392 | HF999344.1 | GCA_000433975.1_MGS528     | 28097 | 28210 | . | + | 50S ribosomal protein L36 | 492e52027403480f8ebe523409d9267f | 2 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| cds | 2 | 5c515b50cccc4c8cb8d4d071d478865d | HDMAFNJB_00208 | LT962687.1        | GCA_900199635.1_PRJEB19460      | 100615 | 100728 | . | + | 50S ribosomal protein L36 | 492e52027403480f8ebe523409d9267f | 2 |\n",
       "| cds | 2 | 1f2a10cd3735420c99aa8058f1c4a798 | LFEHCDIH_00979 | CABJAH010000003.1 | GCA_902363605.1_MGYG-HGUT-00131 |  74458 |  74571 | . | + | 50S ribosomal protein L36 | 492e52027403480f8ebe523409d9267f | 2 |\n",
       "\n"
      ],
      "text/plain": [
       "   gene_type cluster_id seq_uuid                         seq_orig_name \n",
       "1  cds       2          95cf4c86c7cf4893b51f3ba76fc4d9e9 KHIIGKDO_03304\n",
       "2  cds       2          663b3413e3c644b0bf9751a313c1cb7a AAEGJLIP_00392\n",
       "⋮  ⋮         ⋮          ⋮                                ⋮             \n",
       "25 cds       2          5c515b50cccc4c8cb8d4d071d478865d HDMAFNJB_00208\n",
       "26 cds       2          1f2a10cd3735420c99aa8058f1c4a798 LFEHCDIH_00979\n",
       "   contig_id         taxon                           start  end    score strand\n",
       "1  DS544172.1        GCA_000154565.1_ASM15456v1       5547   5660  .     -     \n",
       "2  HF999344.1        GCA_000433975.1_MGS528          28097  28210  .     +     \n",
       "⋮  ⋮                 ⋮                               ⋮      ⋮      ⋮     ⋮     \n",
       "25 LT962687.1        GCA_900199635.1_PRJEB19460      100615 100728 .     +     \n",
       "26 CABJAH010000003.1 GCA_902363605.1_MGYG-HGUT-00131  74458  74571 .     +     \n",
       "   annotation                cluster_name                     clust_id\n",
       "1  50S ribosomal protein L36 492e52027403480f8ebe523409d9267f 2       \n",
       "2  50S ribosomal protein L36 492e52027403480f8ebe523409d9267f 2       \n",
       "⋮  ⋮                         ⋮                                ⋮       \n",
       "25 50S ribosomal protein L36 492e52027403480f8ebe523409d9267f 2       \n",
       "26 50S ribosomal protein L36 492e52027403480f8ebe523409d9267f 2       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# target gene info\n",
    "gene_annot = read.delim(file.path(work_dir, 'primers', 'cgp', 'core_clusters_info.tsv'), \n",
    "                        sep='\\t') %>%\n",
    "    semi_join(primer_info, c('cluster_id')) \n",
    "gene_annot %>% distinct(gene_type, cluster_id) %>% nrow %>% cat('No. of clusters', ., '\\n')\n",
    "gene_annot"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Gene cluster annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-24T13:17:48.528219Z",
     "iopub.status.busy": "2021-04-24T13:17:48.525876Z",
     "iopub.status.idle": "2021-04-24T13:17:48.609306Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 30 × 17</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>cluster_id</th><th scope=col>query</th><th scope=col>subject</th><th scope=col>subject_name</th><th scope=col>pident</th><th scope=col>length</th><th scope=col>mismatch</th><th scope=col>qstart</th><th scope=col>qend</th><th scope=col>sstart</th><th scope=col>send</th><th scope=col>evalue</th><th scope=col>slen</th><th scope=col>qlen</th><th scope=col>sscinames</th><th scope=col>staxids</th><th scope=col>pident_rank</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>2</td><td>492e52027403480f8ebe523409d9267f</td><td>WP_024729495.1</td><td>MULTISPECIES: 50S ribosomal protein L36 [Ruminococcaceae]</td><td>100.000</td><td>37</td><td>0</td><td>1</td><td>111</td><td>1</td><td>37</td><td>4.88e-16</td><td>37</td><td>114</td><td>Anaerotruncus colihominis;Oscillospiraceae;Anaerotruncus colihominis DSM 17241;Provencibacterium massiliense;Harryflintia acetispora;Oscillospiraceae bacterium;Clostridiales bacterium;bacterium 1XD42-1;bacterium 1XD42-8;Oscillospiraceae bacterium NSJ-64</td><td>169435;216572;445972;1841868;1849041;1898205;1898207;2320101;2320102;2763678</td><td>30</td></tr>\n",
       "\t<tr><td>2</td><td>492e52027403480f8ebe523409d9267f</td><td>MBC8570276.1  </td><td>None                                                     </td><td> 97.297</td><td>37</td><td>1</td><td>1</td><td>111</td><td>1</td><td>37</td><td>7.32e-16</td><td>37</td><td>114</td><td>Oscillospiraceae bacterium NSJ-54                                                                                                                                                                                                                            </td><td>2763677                                                                     </td><td> 1</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>2</td><td>492e52027403480f8ebe523409d9267f</td><td>EFB76111.1    </td><td>ribosomal protein L36 [Subdoligranulum variabile DSM 15176]</td><td>89.189</td><td>37</td><td>4</td><td>1</td><td>111</td><td>10</td><td>46</td><td>6.75e-15</td><td>46</td><td>114</td><td>Subdoligranulum variabile DSM 15176                             </td><td>411471             </td><td>28</td></tr>\n",
       "\t<tr><td>2</td><td>492e52027403480f8ebe523409d9267f</td><td>WP_073261210.1</td><td>MULTISPECIES: 50S ribosomal protein L36 [Clostridiales]    </td><td>91.892</td><td>37</td><td>3</td><td>1</td><td>111</td><td> 1</td><td>37</td><td>6.89e-15</td><td>37</td><td>114</td><td>Clostridium sp. ATCC 29733;Eubacteriales;Bittarella massiliensis</td><td>1507;186802;1720313</td><td>20</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 30 × 17\n",
       "\\begin{tabular}{lllllllllllllllll}\n",
       " cluster\\_id & query & subject & subject\\_name & pident & length & mismatch & qstart & qend & sstart & send & evalue & slen & qlen & sscinames & staxids & pident\\_rank\\\\\n",
       " <int> & <fct> & <fct> & <fct> & <dbl> & <int> & <int> & <int> & <int> & <int> & <int> & <dbl> & <int> & <int> & <fct> & <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t 2 & 492e52027403480f8ebe523409d9267f & WP\\_024729495.1 & MULTISPECIES: 50S ribosomal protein L36 {[}Ruminococcaceae{]} & 100.000 & 37 & 0 & 1 & 111 & 1 & 37 & 4.88e-16 & 37 & 114 & Anaerotruncus colihominis;Oscillospiraceae;Anaerotruncus colihominis DSM 17241;Provencibacterium massiliense;Harryflintia acetispora;Oscillospiraceae bacterium;Clostridiales bacterium;bacterium 1XD42-1;bacterium 1XD42-8;Oscillospiraceae bacterium NSJ-64 & 169435;216572;445972;1841868;1849041;1898205;1898207;2320101;2320102;2763678 & 30\\\\\n",
       "\t 2 & 492e52027403480f8ebe523409d9267f & MBC8570276.1   & None                                                      &  97.297 & 37 & 1 & 1 & 111 & 1 & 37 & 7.32e-16 & 37 & 114 & Oscillospiraceae bacterium NSJ-54                                                                                                                                                                                                                             & 2763677                                                                      &  1\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t 2 & 492e52027403480f8ebe523409d9267f & EFB76111.1     & ribosomal protein L36 {[}Subdoligranulum variabile DSM 15176{]} & 89.189 & 37 & 4 & 1 & 111 & 10 & 46 & 6.75e-15 & 46 & 114 & Subdoligranulum variabile DSM 15176                              & 411471              & 28\\\\\n",
       "\t 2 & 492e52027403480f8ebe523409d9267f & WP\\_073261210.1 & MULTISPECIES: 50S ribosomal protein L36 {[}Clostridiales{]}     & 91.892 & 37 & 3 & 1 & 111 &  1 & 37 & 6.89e-15 & 37 & 114 & Clostridium sp. ATCC 29733;Eubacteriales;Bittarella massiliensis & 1507;186802;1720313 & 20\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 30 × 17\n",
       "\n",
       "| cluster_id &lt;int&gt; | query &lt;fct&gt; | subject &lt;fct&gt; | subject_name &lt;fct&gt; | pident &lt;dbl&gt; | length &lt;int&gt; | mismatch &lt;int&gt; | qstart &lt;int&gt; | qend &lt;int&gt; | sstart &lt;int&gt; | send &lt;int&gt; | evalue &lt;dbl&gt; | slen &lt;int&gt; | qlen &lt;int&gt; | sscinames &lt;fct&gt; | staxids &lt;fct&gt; | pident_rank &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 2 | 492e52027403480f8ebe523409d9267f | WP_024729495.1 | MULTISPECIES: 50S ribosomal protein L36 [Ruminococcaceae] | 100.000 | 37 | 0 | 1 | 111 | 1 | 37 | 4.88e-16 | 37 | 114 | Anaerotruncus colihominis;Oscillospiraceae;Anaerotruncus colihominis DSM 17241;Provencibacterium massiliense;Harryflintia acetispora;Oscillospiraceae bacterium;Clostridiales bacterium;bacterium 1XD42-1;bacterium 1XD42-8;Oscillospiraceae bacterium NSJ-64 | 169435;216572;445972;1841868;1849041;1898205;1898207;2320101;2320102;2763678 | 30 |\n",
       "| 2 | 492e52027403480f8ebe523409d9267f | MBC8570276.1   | None                                                      |  97.297 | 37 | 1 | 1 | 111 | 1 | 37 | 7.32e-16 | 37 | 114 | Oscillospiraceae bacterium NSJ-54                                                                                                                                                                                                                             | 2763677                                                                      |  1 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| 2 | 492e52027403480f8ebe523409d9267f | EFB76111.1     | ribosomal protein L36 [Subdoligranulum variabile DSM 15176] | 89.189 | 37 | 4 | 1 | 111 | 10 | 46 | 6.75e-15 | 46 | 114 | Subdoligranulum variabile DSM 15176                              | 411471              | 28 |\n",
       "| 2 | 492e52027403480f8ebe523409d9267f | WP_073261210.1 | MULTISPECIES: 50S ribosomal protein L36 [Clostridiales]     | 91.892 | 37 | 3 | 1 | 111 |  1 | 37 | 6.89e-15 | 37 | 114 | Clostridium sp. ATCC 29733;Eubacteriales;Bittarella massiliensis | 1507;186802;1720313 | 20 |\n",
       "\n"
      ],
      "text/plain": [
       "   cluster_id query                            subject       \n",
       "1  2          492e52027403480f8ebe523409d9267f WP_024729495.1\n",
       "2  2          492e52027403480f8ebe523409d9267f MBC8570276.1  \n",
       "⋮  ⋮          ⋮                                ⋮             \n",
       "29 2          492e52027403480f8ebe523409d9267f EFB76111.1    \n",
       "30 2          492e52027403480f8ebe523409d9267f WP_073261210.1\n",
       "   subject_name                                                pident  length\n",
       "1  MULTISPECIES: 50S ribosomal protein L36 [Ruminococcaceae]   100.000 37    \n",
       "2  None                                                         97.297 37    \n",
       "⋮  ⋮                                                           ⋮       ⋮     \n",
       "29 ribosomal protein L36 [Subdoligranulum variabile DSM 15176] 89.189  37    \n",
       "30 MULTISPECIES: 50S ribosomal protein L36 [Clostridiales]     91.892  37    \n",
       "   mismatch qstart qend sstart send evalue   slen qlen\n",
       "1  0        1      111  1      37   4.88e-16 37   114 \n",
       "2  1        1      111  1      37   7.32e-16 37   114 \n",
       "⋮  ⋮        ⋮      ⋮    ⋮      ⋮    ⋮        ⋮    ⋮   \n",
       "29 4        1      111  10     46   6.75e-15 46   114 \n",
       "30 3        1      111   1     37   6.89e-15 37   114 \n",
       "   sscinames                                                                                                                                                                                                                                                    \n",
       "1  Anaerotruncus colihominis;Oscillospiraceae;Anaerotruncus colihominis DSM 17241;Provencibacterium massiliense;Harryflintia acetispora;Oscillospiraceae bacterium;Clostridiales bacterium;bacterium 1XD42-1;bacterium 1XD42-8;Oscillospiraceae bacterium NSJ-64\n",
       "2  Oscillospiraceae bacterium NSJ-54                                                                                                                                                                                                                            \n",
       "⋮  ⋮                                                                                                                                                                                                                                                            \n",
       "29 Subdoligranulum variabile DSM 15176                                                                                                                                                                                                                          \n",
       "30 Clostridium sp. ATCC 29733;Eubacteriales;Bittarella massiliensis                                                                                                                                                                                             \n",
       "   staxids                                                                     \n",
       "1  169435;216572;445972;1841868;1849041;1898205;1898207;2320101;2320102;2763678\n",
       "2  2763677                                                                     \n",
       "⋮  ⋮                                                                           \n",
       "29 411471                                                                      \n",
       "30 1507;186802;1720313                                                         \n",
       "   pident_rank\n",
       "1  30         \n",
       "2   1         \n",
       "⋮  ⋮          \n",
       "29 28         \n",
       "30 20         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# non-target cds gene annotations\n",
    "gene_nontarget = read.delim(file.path(work_dir, 'primers', 'cgp', 'nontarget', 'cds_blastx.tsv'), \n",
    "                        sep='\\t') %>%\n",
    "    semi_join(primer_info, c('cluster_id')) \n",
    "gene_nontarget"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Cluster taxonomic uniqueness"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-24T13:17:50.857887Z",
     "iopub.status.busy": "2021-04-24T13:17:50.855788Z",
     "iopub.status.idle": "2021-04-24T13:17:50.883193Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 1 × 8 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Min.</th><th scope=col>1st Qu.</th><th scope=col>Median</th><th scope=col>Mean</th><th scope=col>3rd Qu.</th><th scope=col>Max.</th><th scope=col>sd</th><th scope=col>sd_err_of_mean</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>seq-ID to closest non-target</th><td>97.297</td><td>97.297</td><td>97.297</td><td>97.297</td><td>97.297</td><td>97.297</td><td>0</td><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 1 × 8 of type dbl\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & Min. & 1st Qu. & Median & Mean & 3rd Qu. & Max. & sd & sd\\_err\\_of\\_mean\\\\\n",
       "\\hline\n",
       "\tseq-ID to closest non-target & 97.297 & 97.297 & 97.297 & 97.297 & 97.297 & 97.297 & 0 & 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 1 × 8 of type dbl\n",
       "\n",
       "| <!--/--> | Min. | 1st Qu. | Median | Mean | 3rd Qu. | Max. | sd | sd_err_of_mean |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| seq-ID to closest non-target | 97.297 | 97.297 | 97.297 | 97.297 | 97.297 | 97.297 | 0 | 0 |\n",
       "\n"
      ],
      "text/plain": [
       "                             Min.   1st Qu. Median Mean   3rd Qu. Max.   sd\n",
       "seq-ID to closest non-target 97.297 97.297  97.297 97.297 97.297  97.297 0 \n",
       "                             sd_err_of_mean\n",
       "seq-ID to closest non-target 0             "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# closest seqID\n",
    "gene_nontarget %>%\n",
    "    filter(pident_rank == 1) %>%\n",
    "    .$pident %>% summary_x('seq-ID to closest non-target')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Primer quality"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-24T13:18:15.514775Z",
     "iopub.status.busy": "2021-04-24T13:18:15.512617Z",
     "iopub.status.idle": "2021-04-24T13:18:15.537388Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No. of unique primer sets: 10 \n",
      "No. of unique clusters: 1 \n"
     ]
    }
   ],
   "source": [
    "# summary\n",
    "primer_info %>% unique_n('primer sets', primer_set)\n",
    "primer_info %>% distinct(gene_type, cluster_id) %>% nrow %>% cat('No. of unique clusters:', ., '\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-24T13:18:20.743672Z",
     "iopub.status.busy": "2021-04-24T13:18:20.741325Z",
     "iopub.status.idle": "2021-04-24T13:18:20.772788Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 1 × 8 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Min.</th><th scope=col>1st Qu.</th><th scope=col>Median</th><th scope=col>Mean</th><th scope=col>3rd Qu.</th><th scope=col>Max.</th><th scope=col>sd</th><th scope=col>sd_err_of_mean</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>primer sets per cluster</th><td>10</td><td>10</td><td>10</td><td>10</td><td>10</td><td>10</td><td>0</td><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 1 × 8 of type dbl\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & Min. & 1st Qu. & Median & Mean & 3rd Qu. & Max. & sd & sd\\_err\\_of\\_mean\\\\\n",
       "\\hline\n",
       "\tprimer sets per cluster & 10 & 10 & 10 & 10 & 10 & 10 & 0 & 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 1 × 8 of type dbl\n",
       "\n",
       "| <!--/--> | Min. | 1st Qu. | Median | Mean | 3rd Qu. | Max. | sd | sd_err_of_mean |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| primer sets per cluster | 10 | 10 | 10 | 10 | 10 | 10 | 0 | 0 |\n",
       "\n"
      ],
      "text/plain": [
       "                        Min. 1st Qu. Median Mean 3rd Qu. Max. sd sd_err_of_mean\n",
       "primer sets per cluster 10   10      10     10   10      10   0  0             "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# primers per cluster\n",
    "primer_info %>%\n",
    "    distinct(cluster_id, primer_set) %>%\n",
    "    group_by(cluster_id) %>%\n",
    "    summarize(n_primer_sets = n(), .groups='drop') %>%\n",
    "    .$n_primer_sets %>% summary_x('primer sets per cluster')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-24T13:18:26.387987Z",
     "iopub.status.busy": "2021-04-24T13:18:26.385746Z",
     "iopub.status.idle": "2021-04-24T13:18:26.431031Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 1 × 8 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Min.</th><th scope=col>1st Qu.</th><th scope=col>Median</th><th scope=col>Mean</th><th scope=col>3rd Qu.</th><th scope=col>Max.</th><th scope=col>sd</th><th scope=col>sd_err_of_mean</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>mean amplicon size</th><td>103</td><td>107.25</td><td>109</td><td>108.1</td><td>109.75</td><td>110</td><td>2.588</td><td>1.057</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 1 × 8 of type dbl\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & Min. & 1st Qu. & Median & Mean & 3rd Qu. & Max. & sd & sd\\_err\\_of\\_mean\\\\\n",
       "\\hline\n",
       "\tmean amplicon size & 103 & 107.25 & 109 & 108.1 & 109.75 & 110 & 2.588 & 1.057\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 1 × 8 of type dbl\n",
       "\n",
       "| <!--/--> | Min. | 1st Qu. | Median | Mean | 3rd Qu. | Max. | sd | sd_err_of_mean |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| mean amplicon size | 103 | 107.25 | 109 | 108.1 | 109.75 | 110 | 2.588 | 1.057 |\n",
       "\n"
      ],
      "text/plain": [
       "                   Min. 1st Qu. Median Mean  3rd Qu. Max. sd    sd_err_of_mean\n",
       "mean amplicon size 103  107.25  109    108.1 109.75  110  2.588 1.057         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 1 × 8 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Min.</th><th scope=col>1st Qu.</th><th scope=col>Median</th><th scope=col>Mean</th><th scope=col>3rd Qu.</th><th scope=col>Max.</th><th scope=col>sd</th><th scope=col>sd_err_of_mean</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>degeneracy</th><td>12</td><td>16</td><td>20</td><td>24.4</td><td>36</td><td>36</td><td>10.121</td><td>4.132</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 1 × 8 of type dbl\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & Min. & 1st Qu. & Median & Mean & 3rd Qu. & Max. & sd & sd\\_err\\_of\\_mean\\\\\n",
       "\\hline\n",
       "\tdegeneracy & 12 & 16 & 20 & 24.4 & 36 & 36 & 10.121 & 4.132\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 1 × 8 of type dbl\n",
       "\n",
       "| <!--/--> | Min. | 1st Qu. | Median | Mean | 3rd Qu. | Max. | sd | sd_err_of_mean |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| degeneracy | 12 | 16 | 20 | 24.4 | 36 | 36 | 10.121 | 4.132 |\n",
       "\n"
      ],
      "text/plain": [
       "           Min. 1st Qu. Median Mean 3rd Qu. Max. sd     sd_err_of_mean\n",
       "degeneracy 12   16      20     24.4 36      36   10.121 4.132         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 1 × 8 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Min.</th><th scope=col>1st Qu.</th><th scope=col>Median</th><th scope=col>Mean</th><th scope=col>3rd Qu.</th><th scope=col>Max.</th><th scope=col>sd</th><th scope=col>sd_err_of_mean</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>degeneracy (3-prime)</th><td>1</td><td>1</td><td>2</td><td>2.6</td><td>3</td><td>6</td><td>1.855</td><td>0.757</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 1 × 8 of type dbl\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & Min. & 1st Qu. & Median & Mean & 3rd Qu. & Max. & sd & sd\\_err\\_of\\_mean\\\\\n",
       "\\hline\n",
       "\tdegeneracy (3-prime) & 1 & 1 & 2 & 2.6 & 3 & 6 & 1.855 & 0.757\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 1 × 8 of type dbl\n",
       "\n",
       "| <!--/--> | Min. | 1st Qu. | Median | Mean | 3rd Qu. | Max. | sd | sd_err_of_mean |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| degeneracy (3-prime) | 1 | 1 | 2 | 2.6 | 3 | 6 | 1.855 | 0.757 |\n",
       "\n"
      ],
      "text/plain": [
       "                     Min. 1st Qu. Median Mean 3rd Qu. Max. sd    sd_err_of_mean\n",
       "degeneracy (3-prime) 1    1       2      2.6  3       6    1.855 0.757         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# primer quality\n",
    "primer_info %>% filter(primer_type == 'PRIMER_LEFT') %>% .$amplicon_size_avg %>% summary_x('mean amplicon size')\n",
    "primer_info %>% .$degeneracy %>% summary_x('degeneracy')\n",
    "primer_info %>% .$degeneracy_3prime %>% summary_x('degeneracy (3-prime)')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-24T13:18:40.202871Z",
     "iopub.status.busy": "2021-04-24T13:18:40.192868Z",
     "iopub.status.idle": "2021-04-24T13:18:40.256666Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 140 × 24</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>gene_type</th><th scope=col>cluster_id</th><th scope=col>primer_set</th><th scope=col>amplicon_size_consensus</th><th scope=col>amplicon_size_avg</th><th scope=col>amplicon_size_sd</th><th scope=col>primer_id</th><th scope=col>primer_type</th><th scope=col>sequence</th><th scope=col>length</th><th scope=col>⋯</th><th scope=col>Tm_avg</th><th scope=col>Tm_sd</th><th scope=col>GC_avg</th><th scope=col>GC_sd</th><th scope=col>hairpin_avg</th><th scope=col>hairpin_sd</th><th scope=col>homodimer_avg</th><th scope=col>homodimer_sd</th><th scope=col>pident</th><th scope=col>sscinames</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>cds</td><td>2</td><td>73</td><td>109</td><td>109</td><td>0</td><td>73f</td><td>PRIMER_LEFT</td><td>TGAAAGTHAGASCTTCYGTDAA</td><td>22</td><td>⋯</td><td>56.23671</td><td>1.915052</td><td>37.12121</td><td>3.787879</td><td>7.511429</td><td>16.79607</td><td>-27.56715</td><td>24.42498</td><td>97.297</td><td>Oscillospiraceae bacterium NSJ-54                                                                                                                                                                                     </td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>73</td><td>109</td><td>109</td><td>0</td><td>73f</td><td>PRIMER_LEFT</td><td>TGAAAGTHAGASCTTCYGTDAA</td><td>22</td><td>⋯</td><td>56.23671</td><td>1.915052</td><td>37.12121</td><td>3.787879</td><td>7.511429</td><td>16.79607</td><td>-27.56715</td><td>24.42498</td><td>97.297</td><td>Oscillospiraceae;Hydrogenoanaerobacterium saccharovorans;Anaerotruncus sp. G3(2012);Oscillospiraceae bacterium;Marasmitruncus massiliensis;Anaerotruncus sp. X29;Anaerotruncus sp. 1XD22-93;Anaerotruncus sp. 1XD42-93</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋱</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>925</td><td>107</td><td>107</td><td>0</td><td>925r</td><td>PRIMER_RIGHT</td><td>CYTGWCKCTGTTTRTGCTT</td><td>19</td><td>⋯</td><td>55.29904</td><td>2.585134</td><td>44.73684</td><td>4.558028</td><td>0</td><td>0</td><td>-33.75316</td><td>1.421085e-14</td><td>97.297</td><td>Eubacteriales;Candidatus Soleaferrea massiliensis AP7;Anaerotruncus massiliensis;Anaerotruncus rubiinfantis;Anaerotruncus sp. AF02-27;Anaerotruncus sp. 22A2-44;Oscillospiraceae bacterium;Clostridiaceae bacterium NSJ-31</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>925</td><td>107</td><td>107</td><td>0</td><td>925r</td><td>PRIMER_RIGHT</td><td>CYTGWCKCTGTTTRTGCTT</td><td>19</td><td>⋯</td><td>55.29904</td><td>2.585134</td><td>44.73684</td><td>4.558028</td><td>0</td><td>0</td><td>-33.75316</td><td>1.421085e-14</td><td>97.297</td><td>Acetanaerobacterium elongatum                                                                                                                                                                                             </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 140 × 24\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " gene\\_type & cluster\\_id & primer\\_set & amplicon\\_size\\_consensus & amplicon\\_size\\_avg & amplicon\\_size\\_sd & primer\\_id & primer\\_type & sequence & length & ⋯ & Tm\\_avg & Tm\\_sd & GC\\_avg & GC\\_sd & hairpin\\_avg & hairpin\\_sd & homodimer\\_avg & homodimer\\_sd & pident & sscinames\\\\\n",
       " <chr> & <int> & <int> & <int> & <dbl> & <dbl> & <fct> & <fct> & <fct> & <int> & ⋯ & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <fct>\\\\\n",
       "\\hline\n",
       "\t cds & 2 & 73 & 109 & 109 & 0 & 73f & PRIMER\\_LEFT & TGAAAGTHAGASCTTCYGTDAA & 22 & ⋯ & 56.23671 & 1.915052 & 37.12121 & 3.787879 & 7.511429 & 16.79607 & -27.56715 & 24.42498 & 97.297 & Oscillospiraceae bacterium NSJ-54                                                                                                                                                                                     \\\\\n",
       "\t cds & 2 & 73 & 109 & 109 & 0 & 73f & PRIMER\\_LEFT & TGAAAGTHAGASCTTCYGTDAA & 22 & ⋯ & 56.23671 & 1.915052 & 37.12121 & 3.787879 & 7.511429 & 16.79607 & -27.56715 & 24.42498 & 97.297 & Oscillospiraceae;Hydrogenoanaerobacterium saccharovorans;Anaerotruncus sp. G3(2012);Oscillospiraceae bacterium;Marasmitruncus massiliensis;Anaerotruncus sp. X29;Anaerotruncus sp. 1XD22-93;Anaerotruncus sp. 1XD42-93\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋱ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t cds & 2 & 925 & 107 & 107 & 0 & 925r & PRIMER\\_RIGHT & CYTGWCKCTGTTTRTGCTT & 19 & ⋯ & 55.29904 & 2.585134 & 44.73684 & 4.558028 & 0 & 0 & -33.75316 & 1.421085e-14 & 97.297 & Eubacteriales;Candidatus Soleaferrea massiliensis AP7;Anaerotruncus massiliensis;Anaerotruncus rubiinfantis;Anaerotruncus sp. AF02-27;Anaerotruncus sp. 22A2-44;Oscillospiraceae bacterium;Clostridiaceae bacterium NSJ-31\\\\\n",
       "\t cds & 2 & 925 & 107 & 107 & 0 & 925r & PRIMER\\_RIGHT & CYTGWCKCTGTTTRTGCTT & 19 & ⋯ & 55.29904 & 2.585134 & 44.73684 & 4.558028 & 0 & 0 & -33.75316 & 1.421085e-14 & 97.297 & Acetanaerobacterium elongatum                                                                                                                                                                                             \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 140 × 24\n",
       "\n",
       "| gene_type &lt;chr&gt; | cluster_id &lt;int&gt; | primer_set &lt;int&gt; | amplicon_size_consensus &lt;int&gt; | amplicon_size_avg &lt;dbl&gt; | amplicon_size_sd &lt;dbl&gt; | primer_id &lt;fct&gt; | primer_type &lt;fct&gt; | sequence &lt;fct&gt; | length &lt;int&gt; | ⋯ ⋯ | Tm_avg &lt;dbl&gt; | Tm_sd &lt;dbl&gt; | GC_avg &lt;dbl&gt; | GC_sd &lt;dbl&gt; | hairpin_avg &lt;dbl&gt; | hairpin_sd &lt;dbl&gt; | homodimer_avg &lt;dbl&gt; | homodimer_sd &lt;dbl&gt; | pident &lt;dbl&gt; | sscinames &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| cds | 2 | 73 | 109 | 109 | 0 | 73f | PRIMER_LEFT | TGAAAGTHAGASCTTCYGTDAA | 22 | ⋯ | 56.23671 | 1.915052 | 37.12121 | 3.787879 | 7.511429 | 16.79607 | -27.56715 | 24.42498 | 97.297 | Oscillospiraceae bacterium NSJ-54                                                                                                                                                                                      |\n",
       "| cds | 2 | 73 | 109 | 109 | 0 | 73f | PRIMER_LEFT | TGAAAGTHAGASCTTCYGTDAA | 22 | ⋯ | 56.23671 | 1.915052 | 37.12121 | 3.787879 | 7.511429 | 16.79607 | -27.56715 | 24.42498 | 97.297 | Oscillospiraceae;Hydrogenoanaerobacterium saccharovorans;Anaerotruncus sp. G3(2012);Oscillospiraceae bacterium;Marasmitruncus massiliensis;Anaerotruncus sp. X29;Anaerotruncus sp. 1XD22-93;Anaerotruncus sp. 1XD42-93 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋱ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| cds | 2 | 925 | 107 | 107 | 0 | 925r | PRIMER_RIGHT | CYTGWCKCTGTTTRTGCTT | 19 | ⋯ | 55.29904 | 2.585134 | 44.73684 | 4.558028 | 0 | 0 | -33.75316 | 1.421085e-14 | 97.297 | Eubacteriales;Candidatus Soleaferrea massiliensis AP7;Anaerotruncus massiliensis;Anaerotruncus rubiinfantis;Anaerotruncus sp. AF02-27;Anaerotruncus sp. 22A2-44;Oscillospiraceae bacterium;Clostridiaceae bacterium NSJ-31 |\n",
       "| cds | 2 | 925 | 107 | 107 | 0 | 925r | PRIMER_RIGHT | CYTGWCKCTGTTTRTGCTT | 19 | ⋯ | 55.29904 | 2.585134 | 44.73684 | 4.558028 | 0 | 0 | -33.75316 | 1.421085e-14 | 97.297 | Acetanaerobacterium elongatum                                                                                                                                                                                              |\n",
       "\n"
      ],
      "text/plain": [
       "    gene_type cluster_id primer_set amplicon_size_consensus amplicon_size_avg\n",
       "1   cds       2          73         109                     109              \n",
       "2   cds       2          73         109                     109              \n",
       "⋮   ⋮         ⋮          ⋮          ⋮                       ⋮                \n",
       "139 cds       2          925        107                     107              \n",
       "140 cds       2          925        107                     107              \n",
       "    amplicon_size_sd primer_id primer_type  sequence               length ⋯\n",
       "1   0                73f       PRIMER_LEFT  TGAAAGTHAGASCTTCYGTDAA 22     ⋯\n",
       "2   0                73f       PRIMER_LEFT  TGAAAGTHAGASCTTCYGTDAA 22     ⋯\n",
       "⋮   ⋮                ⋮         ⋮            ⋮                      ⋮      ⋱\n",
       "139 0                925r      PRIMER_RIGHT CYTGWCKCTGTTTRTGCTT    19     ⋯\n",
       "140 0                925r      PRIMER_RIGHT CYTGWCKCTGTTTRTGCTT    19     ⋯\n",
       "    Tm_avg   Tm_sd    GC_avg   GC_sd    hairpin_avg hairpin_sd homodimer_avg\n",
       "1   56.23671 1.915052 37.12121 3.787879 7.511429    16.79607   -27.56715    \n",
       "2   56.23671 1.915052 37.12121 3.787879 7.511429    16.79607   -27.56715    \n",
       "⋮   ⋮        ⋮        ⋮        ⋮        ⋮           ⋮          ⋮            \n",
       "139 55.29904 2.585134 44.73684 4.558028 0           0          -33.75316    \n",
       "140 55.29904 2.585134 44.73684 4.558028 0           0          -33.75316    \n",
       "    homodimer_sd pident\n",
       "1   24.42498     97.297\n",
       "2   24.42498     97.297\n",
       "⋮   ⋮            ⋮     \n",
       "139 1.421085e-14 97.297\n",
       "140 1.421085e-14 97.297\n",
       "    sscinames                                                                                                                                                                                                                 \n",
       "1   Oscillospiraceae bacterium NSJ-54                                                                                                                                                                                         \n",
       "2   Oscillospiraceae;Hydrogenoanaerobacterium saccharovorans;Anaerotruncus sp. G3(2012);Oscillospiraceae bacterium;Marasmitruncus massiliensis;Anaerotruncus sp. X29;Anaerotruncus sp. 1XD22-93;Anaerotruncus sp. 1XD42-93    \n",
       "⋮   ⋮                                                                                                                                                                                                                         \n",
       "139 Eubacteriales;Candidatus Soleaferrea massiliensis AP7;Anaerotruncus massiliensis;Anaerotruncus rubiinfantis;Anaerotruncus sp. AF02-27;Anaerotruncus sp. 22A2-44;Oscillospiraceae bacterium;Clostridiaceae bacterium NSJ-31\n",
       "140 Acetanaerobacterium elongatum                                                                                                                                                                                             "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# adding non-target pident info\n",
    "tmp1 = gene_nontarget %>%\n",
    "    filter(pident_rank == 1) %>%\n",
    "    distinct(cluster_id, pident, sscinames) %>%\n",
    "    mutate(gene_type = 'cds')\n",
    "\n",
    "primer_info_j = primer_info %>%\n",
    "    inner_join(tmp1, c('gene_type', 'cluster_id'))\n",
    "primer_info_j"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-24T13:18:49.557433Z",
     "iopub.status.busy": "2021-04-24T13:18:49.555114Z",
     "iopub.status.idle": "2021-04-24T13:18:49.680787Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A tibble: 30 × 26</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>gene_type</th><th scope=col>cluster_id</th><th scope=col>primer_set</th><th scope=col>amplicon_size_consensus</th><th scope=col>amplicon_size_avg</th><th scope=col>amplicon_size_sd</th><th scope=col>primer_id</th><th scope=col>primer_type</th><th scope=col>sequence</th><th scope=col>length</th><th scope=col>degeneracy</th><th scope=col>degeneracy_3prime</th><th scope=col>position_start</th><th scope=col>position_end</th><th scope=col>Tm_avg</th><th scope=col>Tm_sd</th><th scope=col>GC_avg</th><th scope=col>GC_sd</th><th scope=col>hairpin_avg</th><th scope=col>hairpin_sd</th><th scope=col>homodimer_avg</th><th scope=col>homodimer_sd</th><th scope=col>pident</th><th scope=col>sscinames</th><th scope=col>max_degeneracy_3prime</th><th scope=col>max_degeneracy</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>cds</td><td>2</td><td>546</td><td>103</td><td>103</td><td>0</td><td>546f</td><td>PRIMER_LEFT </td><td>ATGAAAGTHAGASCTTCYGT  </td><td>20</td><td>12</td><td>2</td><td> 0</td><td> 20</td><td>54.37386</td><td>1.763376</td><td>39.16667</td><td>3.435921</td><td>7.511429</td><td>16.79607</td><td>-27.56715</td><td>2.442498e+01</td><td>97.297</td><td>Oscillospiraceae bacterium NSJ-54                                                                                                                                                                                         </td><td>2</td><td>24</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>546</td><td>103</td><td>103</td><td>0</td><td>546f</td><td>PRIMER_LEFT </td><td>ATGAAAGTHAGASCTTCYGT  </td><td>20</td><td>12</td><td>2</td><td> 0</td><td> 20</td><td>54.37386</td><td>1.763376</td><td>39.16667</td><td>3.435921</td><td>7.511429</td><td>16.79607</td><td>-27.56715</td><td>2.442498e+01</td><td>97.297</td><td>Oscillospiraceae;Hydrogenoanaerobacterium saccharovorans;Anaerotruncus sp. G3(2012);Oscillospiraceae bacterium;Marasmitruncus massiliensis;Anaerotruncus sp. X29;Anaerotruncus sp. 1XD22-93;Anaerotruncus sp. 1XD42-93    </td><td>2</td><td>24</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>546</td><td>103</td><td>103</td><td>0</td><td>546f</td><td>PRIMER_LEFT </td><td>ATGAAAGTHAGASCTTCYGT  </td><td>20</td><td>12</td><td>2</td><td> 0</td><td> 20</td><td>54.37386</td><td>1.763376</td><td>39.16667</td><td>3.435921</td><td>7.511429</td><td>16.79607</td><td>-27.56715</td><td>2.442498e+01</td><td>97.297</td><td>Oscillospiraceae bacterium                                                                                                                                                                                                </td><td>2</td><td>24</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>546</td><td>103</td><td>103</td><td>0</td><td>546f</td><td>PRIMER_LEFT </td><td>ATGAAAGTHAGASCTTCYGT  </td><td>20</td><td>12</td><td>2</td><td> 0</td><td> 20</td><td>54.37386</td><td>1.763376</td><td>39.16667</td><td>3.435921</td><td>7.511429</td><td>16.79607</td><td>-27.56715</td><td>2.442498e+01</td><td>97.297</td><td>Clostridium sp. CAG:169;Clostridium sp. CAG:242;Negativibacillus massiliensis                                                                                                                                             </td><td>2</td><td>24</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>546</td><td>103</td><td>103</td><td>0</td><td>546f</td><td>PRIMER_LEFT </td><td>ATGAAAGTHAGASCTTCYGT  </td><td>20</td><td>12</td><td>2</td><td> 0</td><td> 20</td><td>54.37386</td><td>1.763376</td><td>39.16667</td><td>3.435921</td><td>7.511429</td><td>16.79607</td><td>-27.56715</td><td>2.442498e+01</td><td>97.297</td><td>Ruminococcus flavefaciens;Anaerofustis stercorihominis;Oscillospiraceae bacterium;Clostridiales bacterium                                                                                                                 </td><td>2</td><td>24</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>546</td><td>103</td><td>103</td><td>0</td><td>546f</td><td>PRIMER_LEFT </td><td>ATGAAAGTHAGASCTTCYGT  </td><td>20</td><td>12</td><td>2</td><td> 0</td><td> 20</td><td>54.37386</td><td>1.763376</td><td>39.16667</td><td>3.435921</td><td>7.511429</td><td>16.79607</td><td>-27.56715</td><td>2.442498e+01</td><td>97.297</td><td>Eubacteriales;Candidatus Soleaferrea massiliensis AP7;Anaerotruncus massiliensis;Anaerotruncus rubiinfantis;Anaerotruncus sp. AF02-27;Anaerotruncus sp. 22A2-44;Oscillospiraceae bacterium;Clostridiaceae bacterium NSJ-31</td><td>2</td><td>24</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>546</td><td>103</td><td>103</td><td>0</td><td>546f</td><td>PRIMER_LEFT </td><td>ATGAAAGTHAGASCTTCYGT  </td><td>20</td><td>12</td><td>2</td><td> 0</td><td> 20</td><td>54.37386</td><td>1.763376</td><td>39.16667</td><td>3.435921</td><td>7.511429</td><td>16.79607</td><td>-27.56715</td><td>2.442498e+01</td><td>97.297</td><td>Acetanaerobacterium elongatum                                                                                                                                                                                             </td><td>2</td><td>24</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>546</td><td>103</td><td>103</td><td>0</td><td>546r</td><td>PRIMER_RIGHT</td><td>KCTGTTTRTGCTTBGGRTTT  </td><td>20</td><td>24</td><td>2</td><td>83</td><td>103</td><td>56.37333</td><td>2.279991</td><td>40.83333</td><td>4.930066</td><td>0.000000</td><td> 0.00000</td><td>-33.75316</td><td>2.131628e-14</td><td>97.297</td><td>Oscillospiraceae bacterium NSJ-54                                                                                                                                                                                         </td><td>2</td><td>24</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>546</td><td>103</td><td>103</td><td>0</td><td>546r</td><td>PRIMER_RIGHT</td><td>KCTGTTTRTGCTTBGGRTTT  </td><td>20</td><td>24</td><td>2</td><td>83</td><td>103</td><td>56.37333</td><td>2.279991</td><td>40.83333</td><td>4.930066</td><td>0.000000</td><td> 0.00000</td><td>-33.75316</td><td>2.131628e-14</td><td>97.297</td><td>Oscillospiraceae;Hydrogenoanaerobacterium saccharovorans;Anaerotruncus sp. G3(2012);Oscillospiraceae bacterium;Marasmitruncus massiliensis;Anaerotruncus sp. X29;Anaerotruncus sp. 1XD22-93;Anaerotruncus sp. 1XD42-93    </td><td>2</td><td>24</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>546</td><td>103</td><td>103</td><td>0</td><td>546r</td><td>PRIMER_RIGHT</td><td>KCTGTTTRTGCTTBGGRTTT  </td><td>20</td><td>24</td><td>2</td><td>83</td><td>103</td><td>56.37333</td><td>2.279991</td><td>40.83333</td><td>4.930066</td><td>0.000000</td><td> 0.00000</td><td>-33.75316</td><td>2.131628e-14</td><td>97.297</td><td>Oscillospiraceae bacterium                                                                                                                                                                                                </td><td>2</td><td>24</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>546</td><td>103</td><td>103</td><td>0</td><td>546r</td><td>PRIMER_RIGHT</td><td>KCTGTTTRTGCTTBGGRTTT  </td><td>20</td><td>24</td><td>2</td><td>83</td><td>103</td><td>56.37333</td><td>2.279991</td><td>40.83333</td><td>4.930066</td><td>0.000000</td><td> 0.00000</td><td>-33.75316</td><td>2.131628e-14</td><td>97.297</td><td>Clostridium sp. CAG:169;Clostridium sp. CAG:242;Negativibacillus massiliensis                                                                                                                                             </td><td>2</td><td>24</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>546</td><td>103</td><td>103</td><td>0</td><td>546r</td><td>PRIMER_RIGHT</td><td>KCTGTTTRTGCTTBGGRTTT  </td><td>20</td><td>24</td><td>2</td><td>83</td><td>103</td><td>56.37333</td><td>2.279991</td><td>40.83333</td><td>4.930066</td><td>0.000000</td><td> 0.00000</td><td>-33.75316</td><td>2.131628e-14</td><td>97.297</td><td>Ruminococcus flavefaciens;Anaerofustis stercorihominis;Oscillospiraceae bacterium;Clostridiales bacterium                                                                                                                 </td><td>2</td><td>24</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>546</td><td>103</td><td>103</td><td>0</td><td>546r</td><td>PRIMER_RIGHT</td><td>KCTGTTTRTGCTTBGGRTTT  </td><td>20</td><td>24</td><td>2</td><td>83</td><td>103</td><td>56.37333</td><td>2.279991</td><td>40.83333</td><td>4.930066</td><td>0.000000</td><td> 0.00000</td><td>-33.75316</td><td>2.131628e-14</td><td>97.297</td><td>Eubacteriales;Candidatus Soleaferrea massiliensis AP7;Anaerotruncus massiliensis;Anaerotruncus rubiinfantis;Anaerotruncus sp. AF02-27;Anaerotruncus sp. 22A2-44;Oscillospiraceae bacterium;Clostridiaceae bacterium NSJ-31</td><td>2</td><td>24</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>546</td><td>103</td><td>103</td><td>0</td><td>546r</td><td>PRIMER_RIGHT</td><td>KCTGTTTRTGCTTBGGRTTT  </td><td>20</td><td>24</td><td>2</td><td>83</td><td>103</td><td>56.37333</td><td>2.279991</td><td>40.83333</td><td>4.930066</td><td>0.000000</td><td> 0.00000</td><td>-33.75316</td><td>2.131628e-14</td><td>97.297</td><td>Acetanaerobacterium elongatum                                                                                                                                                                                             </td><td>2</td><td>24</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>809</td><td>106</td><td>106</td><td>0</td><td>809f</td><td>PRIMER_LEFT </td><td>ATGAAAGTHAGASCTTCYGT  </td><td>20</td><td>12</td><td>2</td><td> 0</td><td> 20</td><td>54.37386</td><td>1.763376</td><td>39.16667</td><td>3.435921</td><td>7.511429</td><td>16.79607</td><td>-27.56715</td><td>2.442498e+01</td><td>97.297</td><td>Oscillospiraceae bacterium NSJ-54                                                                                                                                                                                         </td><td>3</td><td>24</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>809</td><td>106</td><td>106</td><td>0</td><td>809f</td><td>PRIMER_LEFT </td><td>ATGAAAGTHAGASCTTCYGT  </td><td>20</td><td>12</td><td>2</td><td> 0</td><td> 20</td><td>54.37386</td><td>1.763376</td><td>39.16667</td><td>3.435921</td><td>7.511429</td><td>16.79607</td><td>-27.56715</td><td>2.442498e+01</td><td>97.297</td><td>Oscillospiraceae;Hydrogenoanaerobacterium saccharovorans;Anaerotruncus sp. G3(2012);Oscillospiraceae bacterium;Marasmitruncus massiliensis;Anaerotruncus sp. X29;Anaerotruncus sp. 1XD22-93;Anaerotruncus sp. 1XD42-93    </td><td>3</td><td>24</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>809</td><td>106</td><td>106</td><td>0</td><td>809f</td><td>PRIMER_LEFT </td><td>ATGAAAGTHAGASCTTCYGT  </td><td>20</td><td>12</td><td>2</td><td> 0</td><td> 20</td><td>54.37386</td><td>1.763376</td><td>39.16667</td><td>3.435921</td><td>7.511429</td><td>16.79607</td><td>-27.56715</td><td>2.442498e+01</td><td>97.297</td><td>Oscillospiraceae bacterium                                                                                                                                                                                                </td><td>3</td><td>24</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>809</td><td>106</td><td>106</td><td>0</td><td>809f</td><td>PRIMER_LEFT </td><td>ATGAAAGTHAGASCTTCYGT  </td><td>20</td><td>12</td><td>2</td><td> 0</td><td> 20</td><td>54.37386</td><td>1.763376</td><td>39.16667</td><td>3.435921</td><td>7.511429</td><td>16.79607</td><td>-27.56715</td><td>2.442498e+01</td><td>97.297</td><td>Clostridium sp. CAG:169;Clostridium sp. CAG:242;Negativibacillus massiliensis                                                                                                                                             </td><td>3</td><td>24</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>809</td><td>106</td><td>106</td><td>0</td><td>809f</td><td>PRIMER_LEFT </td><td>ATGAAAGTHAGASCTTCYGT  </td><td>20</td><td>12</td><td>2</td><td> 0</td><td> 20</td><td>54.37386</td><td>1.763376</td><td>39.16667</td><td>3.435921</td><td>7.511429</td><td>16.79607</td><td>-27.56715</td><td>2.442498e+01</td><td>97.297</td><td>Ruminococcus flavefaciens;Anaerofustis stercorihominis;Oscillospiraceae bacterium;Clostridiales bacterium                                                                                                                 </td><td>3</td><td>24</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>809</td><td>106</td><td>106</td><td>0</td><td>809f</td><td>PRIMER_LEFT </td><td>ATGAAAGTHAGASCTTCYGT  </td><td>20</td><td>12</td><td>2</td><td> 0</td><td> 20</td><td>54.37386</td><td>1.763376</td><td>39.16667</td><td>3.435921</td><td>7.511429</td><td>16.79607</td><td>-27.56715</td><td>2.442498e+01</td><td>97.297</td><td>Eubacteriales;Candidatus Soleaferrea massiliensis AP7;Anaerotruncus massiliensis;Anaerotruncus rubiinfantis;Anaerotruncus sp. AF02-27;Anaerotruncus sp. 22A2-44;Oscillospiraceae bacterium;Clostridiaceae bacterium NSJ-31</td><td>3</td><td>24</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>809</td><td>106</td><td>106</td><td>0</td><td>809f</td><td>PRIMER_LEFT </td><td>ATGAAAGTHAGASCTTCYGT  </td><td>20</td><td>12</td><td>2</td><td> 0</td><td> 20</td><td>54.37386</td><td>1.763376</td><td>39.16667</td><td>3.435921</td><td>7.511429</td><td>16.79607</td><td>-27.56715</td><td>2.442498e+01</td><td>97.297</td><td>Acetanaerobacterium elongatum                                                                                                                                                                                             </td><td>3</td><td>24</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>809</td><td>106</td><td>106</td><td>0</td><td>809r</td><td>PRIMER_RIGHT</td><td>GWCKCTGTTTRTGCTTBGG   </td><td>19</td><td>24</td><td>3</td><td>87</td><td>106</td><td>56.74707</td><td>2.521808</td><td>50.87719</td><td>4.472824</td><td>0.000000</td><td> 0.00000</td><td>-33.75316</td><td>2.131628e-14</td><td>97.297</td><td>Oscillospiraceae bacterium NSJ-54                                                                                                                                                                                         </td><td>3</td><td>24</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>809</td><td>106</td><td>106</td><td>0</td><td>809r</td><td>PRIMER_RIGHT</td><td>GWCKCTGTTTRTGCTTBGG   </td><td>19</td><td>24</td><td>3</td><td>87</td><td>106</td><td>56.74707</td><td>2.521808</td><td>50.87719</td><td>4.472824</td><td>0.000000</td><td> 0.00000</td><td>-33.75316</td><td>2.131628e-14</td><td>97.297</td><td>Oscillospiraceae;Hydrogenoanaerobacterium saccharovorans;Anaerotruncus sp. G3(2012);Oscillospiraceae bacterium;Marasmitruncus massiliensis;Anaerotruncus sp. X29;Anaerotruncus sp. 1XD22-93;Anaerotruncus sp. 1XD42-93    </td><td>3</td><td>24</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>809</td><td>106</td><td>106</td><td>0</td><td>809r</td><td>PRIMER_RIGHT</td><td>GWCKCTGTTTRTGCTTBGG   </td><td>19</td><td>24</td><td>3</td><td>87</td><td>106</td><td>56.74707</td><td>2.521808</td><td>50.87719</td><td>4.472824</td><td>0.000000</td><td> 0.00000</td><td>-33.75316</td><td>2.131628e-14</td><td>97.297</td><td>Oscillospiraceae bacterium                                                                                                                                                                                                </td><td>3</td><td>24</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>809</td><td>106</td><td>106</td><td>0</td><td>809r</td><td>PRIMER_RIGHT</td><td>GWCKCTGTTTRTGCTTBGG   </td><td>19</td><td>24</td><td>3</td><td>87</td><td>106</td><td>56.74707</td><td>2.521808</td><td>50.87719</td><td>4.472824</td><td>0.000000</td><td> 0.00000</td><td>-33.75316</td><td>2.131628e-14</td><td>97.297</td><td>Clostridium sp. CAG:169;Clostridium sp. CAG:242;Negativibacillus massiliensis                                                                                                                                             </td><td>3</td><td>24</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>809</td><td>106</td><td>106</td><td>0</td><td>809r</td><td>PRIMER_RIGHT</td><td>GWCKCTGTTTRTGCTTBGG   </td><td>19</td><td>24</td><td>3</td><td>87</td><td>106</td><td>56.74707</td><td>2.521808</td><td>50.87719</td><td>4.472824</td><td>0.000000</td><td> 0.00000</td><td>-33.75316</td><td>2.131628e-14</td><td>97.297</td><td>Ruminococcus flavefaciens;Anaerofustis stercorihominis;Oscillospiraceae bacterium;Clostridiales bacterium                                                                                                                 </td><td>3</td><td>24</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>809</td><td>106</td><td>106</td><td>0</td><td>809r</td><td>PRIMER_RIGHT</td><td>GWCKCTGTTTRTGCTTBGG   </td><td>19</td><td>24</td><td>3</td><td>87</td><td>106</td><td>56.74707</td><td>2.521808</td><td>50.87719</td><td>4.472824</td><td>0.000000</td><td> 0.00000</td><td>-33.75316</td><td>2.131628e-14</td><td>97.297</td><td>Eubacteriales;Candidatus Soleaferrea massiliensis AP7;Anaerotruncus massiliensis;Anaerotruncus rubiinfantis;Anaerotruncus sp. AF02-27;Anaerotruncus sp. 22A2-44;Oscillospiraceae bacterium;Clostridiaceae bacterium NSJ-31</td><td>3</td><td>24</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>809</td><td>106</td><td>106</td><td>0</td><td>809r</td><td>PRIMER_RIGHT</td><td>GWCKCTGTTTRTGCTTBGG   </td><td>19</td><td>24</td><td>3</td><td>87</td><td>106</td><td>56.74707</td><td>2.521808</td><td>50.87719</td><td>4.472824</td><td>0.000000</td><td> 0.00000</td><td>-33.75316</td><td>2.131628e-14</td><td>97.297</td><td>Acetanaerobacterium elongatum                                                                                                                                                                                             </td><td>3</td><td>24</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td> 73</td><td>109</td><td>109</td><td>0</td><td>73f </td><td>PRIMER_LEFT </td><td>TGAAAGTHAGASCTTCYGTDAA</td><td>22</td><td>36</td><td>3</td><td> 1</td><td> 23</td><td>56.23671</td><td>1.915052</td><td>37.12121</td><td>3.787879</td><td>7.511429</td><td>16.79607</td><td>-27.56715</td><td>2.442498e+01</td><td>97.297</td><td>Oscillospiraceae bacterium NSJ-54                                                                                                                                                                                         </td><td>3</td><td>36</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td> 73</td><td>109</td><td>109</td><td>0</td><td>73f </td><td>PRIMER_LEFT </td><td>TGAAAGTHAGASCTTCYGTDAA</td><td>22</td><td>36</td><td>3</td><td> 1</td><td> 23</td><td>56.23671</td><td>1.915052</td><td>37.12121</td><td>3.787879</td><td>7.511429</td><td>16.79607</td><td>-27.56715</td><td>2.442498e+01</td><td>97.297</td><td>Oscillospiraceae;Hydrogenoanaerobacterium saccharovorans;Anaerotruncus sp. G3(2012);Oscillospiraceae bacterium;Marasmitruncus massiliensis;Anaerotruncus sp. X29;Anaerotruncus sp. 1XD22-93;Anaerotruncus sp. 1XD42-93    </td><td>3</td><td>36</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 30 × 26\n",
       "\\begin{tabular}{llllllllllllllllllllllllll}\n",
       " gene\\_type & cluster\\_id & primer\\_set & amplicon\\_size\\_consensus & amplicon\\_size\\_avg & amplicon\\_size\\_sd & primer\\_id & primer\\_type & sequence & length & degeneracy & degeneracy\\_3prime & position\\_start & position\\_end & Tm\\_avg & Tm\\_sd & GC\\_avg & GC\\_sd & hairpin\\_avg & hairpin\\_sd & homodimer\\_avg & homodimer\\_sd & pident & sscinames & max\\_degeneracy\\_3prime & max\\_degeneracy\\\\\n",
       " <chr> & <int> & <int> & <int> & <dbl> & <dbl> & <fct> & <fct> & <fct> & <int> & <int> & <int> & <int> & <int> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <fct> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t cds & 2 & 546 & 103 & 103 & 0 & 546f & PRIMER\\_LEFT  & ATGAAAGTHAGASCTTCYGT   & 20 & 12 & 2 &  0 &  20 & 54.37386 & 1.763376 & 39.16667 & 3.435921 & 7.511429 & 16.79607 & -27.56715 & 2.442498e+01 & 97.297 & Oscillospiraceae bacterium NSJ-54                                                                                                                                                                                          & 2 & 24\\\\\n",
       "\t cds & 2 & 546 & 103 & 103 & 0 & 546f & PRIMER\\_LEFT  & ATGAAAGTHAGASCTTCYGT   & 20 & 12 & 2 &  0 &  20 & 54.37386 & 1.763376 & 39.16667 & 3.435921 & 7.511429 & 16.79607 & -27.56715 & 2.442498e+01 & 97.297 & Oscillospiraceae;Hydrogenoanaerobacterium saccharovorans;Anaerotruncus sp. G3(2012);Oscillospiraceae bacterium;Marasmitruncus massiliensis;Anaerotruncus sp. X29;Anaerotruncus sp. 1XD22-93;Anaerotruncus sp. 1XD42-93     & 2 & 24\\\\\n",
       "\t cds & 2 & 546 & 103 & 103 & 0 & 546f & PRIMER\\_LEFT  & ATGAAAGTHAGASCTTCYGT   & 20 & 12 & 2 &  0 &  20 & 54.37386 & 1.763376 & 39.16667 & 3.435921 & 7.511429 & 16.79607 & -27.56715 & 2.442498e+01 & 97.297 & Oscillospiraceae bacterium                                                                                                                                                                                                 & 2 & 24\\\\\n",
       "\t cds & 2 & 546 & 103 & 103 & 0 & 546f & PRIMER\\_LEFT  & ATGAAAGTHAGASCTTCYGT   & 20 & 12 & 2 &  0 &  20 & 54.37386 & 1.763376 & 39.16667 & 3.435921 & 7.511429 & 16.79607 & -27.56715 & 2.442498e+01 & 97.297 & Clostridium sp. CAG:169;Clostridium sp. CAG:242;Negativibacillus massiliensis                                                                                                                                              & 2 & 24\\\\\n",
       "\t cds & 2 & 546 & 103 & 103 & 0 & 546f & PRIMER\\_LEFT  & ATGAAAGTHAGASCTTCYGT   & 20 & 12 & 2 &  0 &  20 & 54.37386 & 1.763376 & 39.16667 & 3.435921 & 7.511429 & 16.79607 & -27.56715 & 2.442498e+01 & 97.297 & Ruminococcus flavefaciens;Anaerofustis stercorihominis;Oscillospiraceae bacterium;Clostridiales bacterium                                                                                                                  & 2 & 24\\\\\n",
       "\t cds & 2 & 546 & 103 & 103 & 0 & 546f & PRIMER\\_LEFT  & ATGAAAGTHAGASCTTCYGT   & 20 & 12 & 2 &  0 &  20 & 54.37386 & 1.763376 & 39.16667 & 3.435921 & 7.511429 & 16.79607 & -27.56715 & 2.442498e+01 & 97.297 & Eubacteriales;Candidatus Soleaferrea massiliensis AP7;Anaerotruncus massiliensis;Anaerotruncus rubiinfantis;Anaerotruncus sp. AF02-27;Anaerotruncus sp. 22A2-44;Oscillospiraceae bacterium;Clostridiaceae bacterium NSJ-31 & 2 & 24\\\\\n",
       "\t cds & 2 & 546 & 103 & 103 & 0 & 546f & PRIMER\\_LEFT  & ATGAAAGTHAGASCTTCYGT   & 20 & 12 & 2 &  0 &  20 & 54.37386 & 1.763376 & 39.16667 & 3.435921 & 7.511429 & 16.79607 & -27.56715 & 2.442498e+01 & 97.297 & Acetanaerobacterium elongatum                                                                                                                                                                                              & 2 & 24\\\\\n",
       "\t cds & 2 & 546 & 103 & 103 & 0 & 546r & PRIMER\\_RIGHT & KCTGTTTRTGCTTBGGRTTT   & 20 & 24 & 2 & 83 & 103 & 56.37333 & 2.279991 & 40.83333 & 4.930066 & 0.000000 &  0.00000 & -33.75316 & 2.131628e-14 & 97.297 & Oscillospiraceae bacterium NSJ-54                                                                                                                                                                                          & 2 & 24\\\\\n",
       "\t cds & 2 & 546 & 103 & 103 & 0 & 546r & PRIMER\\_RIGHT & KCTGTTTRTGCTTBGGRTTT   & 20 & 24 & 2 & 83 & 103 & 56.37333 & 2.279991 & 40.83333 & 4.930066 & 0.000000 &  0.00000 & -33.75316 & 2.131628e-14 & 97.297 & Oscillospiraceae;Hydrogenoanaerobacterium saccharovorans;Anaerotruncus sp. G3(2012);Oscillospiraceae bacterium;Marasmitruncus massiliensis;Anaerotruncus sp. X29;Anaerotruncus sp. 1XD22-93;Anaerotruncus sp. 1XD42-93     & 2 & 24\\\\\n",
       "\t cds & 2 & 546 & 103 & 103 & 0 & 546r & PRIMER\\_RIGHT & KCTGTTTRTGCTTBGGRTTT   & 20 & 24 & 2 & 83 & 103 & 56.37333 & 2.279991 & 40.83333 & 4.930066 & 0.000000 &  0.00000 & -33.75316 & 2.131628e-14 & 97.297 & Oscillospiraceae bacterium                                                                                                                                                                                                 & 2 & 24\\\\\n",
       "\t cds & 2 & 546 & 103 & 103 & 0 & 546r & PRIMER\\_RIGHT & KCTGTTTRTGCTTBGGRTTT   & 20 & 24 & 2 & 83 & 103 & 56.37333 & 2.279991 & 40.83333 & 4.930066 & 0.000000 &  0.00000 & -33.75316 & 2.131628e-14 & 97.297 & Clostridium sp. CAG:169;Clostridium sp. CAG:242;Negativibacillus massiliensis                                                                                                                                              & 2 & 24\\\\\n",
       "\t cds & 2 & 546 & 103 & 103 & 0 & 546r & PRIMER\\_RIGHT & KCTGTTTRTGCTTBGGRTTT   & 20 & 24 & 2 & 83 & 103 & 56.37333 & 2.279991 & 40.83333 & 4.930066 & 0.000000 &  0.00000 & -33.75316 & 2.131628e-14 & 97.297 & Ruminococcus flavefaciens;Anaerofustis stercorihominis;Oscillospiraceae bacterium;Clostridiales bacterium                                                                                                                  & 2 & 24\\\\\n",
       "\t cds & 2 & 546 & 103 & 103 & 0 & 546r & PRIMER\\_RIGHT & KCTGTTTRTGCTTBGGRTTT   & 20 & 24 & 2 & 83 & 103 & 56.37333 & 2.279991 & 40.83333 & 4.930066 & 0.000000 &  0.00000 & -33.75316 & 2.131628e-14 & 97.297 & Eubacteriales;Candidatus Soleaferrea massiliensis AP7;Anaerotruncus massiliensis;Anaerotruncus rubiinfantis;Anaerotruncus sp. AF02-27;Anaerotruncus sp. 22A2-44;Oscillospiraceae bacterium;Clostridiaceae bacterium NSJ-31 & 2 & 24\\\\\n",
       "\t cds & 2 & 546 & 103 & 103 & 0 & 546r & PRIMER\\_RIGHT & KCTGTTTRTGCTTBGGRTTT   & 20 & 24 & 2 & 83 & 103 & 56.37333 & 2.279991 & 40.83333 & 4.930066 & 0.000000 &  0.00000 & -33.75316 & 2.131628e-14 & 97.297 & Acetanaerobacterium elongatum                                                                                                                                                                                              & 2 & 24\\\\\n",
       "\t cds & 2 & 809 & 106 & 106 & 0 & 809f & PRIMER\\_LEFT  & ATGAAAGTHAGASCTTCYGT   & 20 & 12 & 2 &  0 &  20 & 54.37386 & 1.763376 & 39.16667 & 3.435921 & 7.511429 & 16.79607 & -27.56715 & 2.442498e+01 & 97.297 & Oscillospiraceae bacterium NSJ-54                                                                                                                                                                                          & 3 & 24\\\\\n",
       "\t cds & 2 & 809 & 106 & 106 & 0 & 809f & PRIMER\\_LEFT  & ATGAAAGTHAGASCTTCYGT   & 20 & 12 & 2 &  0 &  20 & 54.37386 & 1.763376 & 39.16667 & 3.435921 & 7.511429 & 16.79607 & -27.56715 & 2.442498e+01 & 97.297 & Oscillospiraceae;Hydrogenoanaerobacterium saccharovorans;Anaerotruncus sp. G3(2012);Oscillospiraceae bacterium;Marasmitruncus massiliensis;Anaerotruncus sp. X29;Anaerotruncus sp. 1XD22-93;Anaerotruncus sp. 1XD42-93     & 3 & 24\\\\\n",
       "\t cds & 2 & 809 & 106 & 106 & 0 & 809f & PRIMER\\_LEFT  & ATGAAAGTHAGASCTTCYGT   & 20 & 12 & 2 &  0 &  20 & 54.37386 & 1.763376 & 39.16667 & 3.435921 & 7.511429 & 16.79607 & -27.56715 & 2.442498e+01 & 97.297 & Oscillospiraceae bacterium                                                                                                                                                                                                 & 3 & 24\\\\\n",
       "\t cds & 2 & 809 & 106 & 106 & 0 & 809f & PRIMER\\_LEFT  & ATGAAAGTHAGASCTTCYGT   & 20 & 12 & 2 &  0 &  20 & 54.37386 & 1.763376 & 39.16667 & 3.435921 & 7.511429 & 16.79607 & -27.56715 & 2.442498e+01 & 97.297 & Clostridium sp. CAG:169;Clostridium sp. CAG:242;Negativibacillus massiliensis                                                                                                                                              & 3 & 24\\\\\n",
       "\t cds & 2 & 809 & 106 & 106 & 0 & 809f & PRIMER\\_LEFT  & ATGAAAGTHAGASCTTCYGT   & 20 & 12 & 2 &  0 &  20 & 54.37386 & 1.763376 & 39.16667 & 3.435921 & 7.511429 & 16.79607 & -27.56715 & 2.442498e+01 & 97.297 & Ruminococcus flavefaciens;Anaerofustis stercorihominis;Oscillospiraceae bacterium;Clostridiales bacterium                                                                                                                  & 3 & 24\\\\\n",
       "\t cds & 2 & 809 & 106 & 106 & 0 & 809f & PRIMER\\_LEFT  & ATGAAAGTHAGASCTTCYGT   & 20 & 12 & 2 &  0 &  20 & 54.37386 & 1.763376 & 39.16667 & 3.435921 & 7.511429 & 16.79607 & -27.56715 & 2.442498e+01 & 97.297 & Eubacteriales;Candidatus Soleaferrea massiliensis AP7;Anaerotruncus massiliensis;Anaerotruncus rubiinfantis;Anaerotruncus sp. AF02-27;Anaerotruncus sp. 22A2-44;Oscillospiraceae bacterium;Clostridiaceae bacterium NSJ-31 & 3 & 24\\\\\n",
       "\t cds & 2 & 809 & 106 & 106 & 0 & 809f & PRIMER\\_LEFT  & ATGAAAGTHAGASCTTCYGT   & 20 & 12 & 2 &  0 &  20 & 54.37386 & 1.763376 & 39.16667 & 3.435921 & 7.511429 & 16.79607 & -27.56715 & 2.442498e+01 & 97.297 & Acetanaerobacterium elongatum                                                                                                                                                                                              & 3 & 24\\\\\n",
       "\t cds & 2 & 809 & 106 & 106 & 0 & 809r & PRIMER\\_RIGHT & GWCKCTGTTTRTGCTTBGG    & 19 & 24 & 3 & 87 & 106 & 56.74707 & 2.521808 & 50.87719 & 4.472824 & 0.000000 &  0.00000 & -33.75316 & 2.131628e-14 & 97.297 & Oscillospiraceae bacterium NSJ-54                                                                                                                                                                                          & 3 & 24\\\\\n",
       "\t cds & 2 & 809 & 106 & 106 & 0 & 809r & PRIMER\\_RIGHT & GWCKCTGTTTRTGCTTBGG    & 19 & 24 & 3 & 87 & 106 & 56.74707 & 2.521808 & 50.87719 & 4.472824 & 0.000000 &  0.00000 & -33.75316 & 2.131628e-14 & 97.297 & Oscillospiraceae;Hydrogenoanaerobacterium saccharovorans;Anaerotruncus sp. G3(2012);Oscillospiraceae bacterium;Marasmitruncus massiliensis;Anaerotruncus sp. X29;Anaerotruncus sp. 1XD22-93;Anaerotruncus sp. 1XD42-93     & 3 & 24\\\\\n",
       "\t cds & 2 & 809 & 106 & 106 & 0 & 809r & PRIMER\\_RIGHT & GWCKCTGTTTRTGCTTBGG    & 19 & 24 & 3 & 87 & 106 & 56.74707 & 2.521808 & 50.87719 & 4.472824 & 0.000000 &  0.00000 & -33.75316 & 2.131628e-14 & 97.297 & Oscillospiraceae bacterium                                                                                                                                                                                                 & 3 & 24\\\\\n",
       "\t cds & 2 & 809 & 106 & 106 & 0 & 809r & PRIMER\\_RIGHT & GWCKCTGTTTRTGCTTBGG    & 19 & 24 & 3 & 87 & 106 & 56.74707 & 2.521808 & 50.87719 & 4.472824 & 0.000000 &  0.00000 & -33.75316 & 2.131628e-14 & 97.297 & Clostridium sp. CAG:169;Clostridium sp. CAG:242;Negativibacillus massiliensis                                                                                                                                              & 3 & 24\\\\\n",
       "\t cds & 2 & 809 & 106 & 106 & 0 & 809r & PRIMER\\_RIGHT & GWCKCTGTTTRTGCTTBGG    & 19 & 24 & 3 & 87 & 106 & 56.74707 & 2.521808 & 50.87719 & 4.472824 & 0.000000 &  0.00000 & -33.75316 & 2.131628e-14 & 97.297 & Ruminococcus flavefaciens;Anaerofustis stercorihominis;Oscillospiraceae bacterium;Clostridiales bacterium                                                                                                                  & 3 & 24\\\\\n",
       "\t cds & 2 & 809 & 106 & 106 & 0 & 809r & PRIMER\\_RIGHT & GWCKCTGTTTRTGCTTBGG    & 19 & 24 & 3 & 87 & 106 & 56.74707 & 2.521808 & 50.87719 & 4.472824 & 0.000000 &  0.00000 & -33.75316 & 2.131628e-14 & 97.297 & Eubacteriales;Candidatus Soleaferrea massiliensis AP7;Anaerotruncus massiliensis;Anaerotruncus rubiinfantis;Anaerotruncus sp. AF02-27;Anaerotruncus sp. 22A2-44;Oscillospiraceae bacterium;Clostridiaceae bacterium NSJ-31 & 3 & 24\\\\\n",
       "\t cds & 2 & 809 & 106 & 106 & 0 & 809r & PRIMER\\_RIGHT & GWCKCTGTTTRTGCTTBGG    & 19 & 24 & 3 & 87 & 106 & 56.74707 & 2.521808 & 50.87719 & 4.472824 & 0.000000 &  0.00000 & -33.75316 & 2.131628e-14 & 97.297 & Acetanaerobacterium elongatum                                                                                                                                                                                              & 3 & 24\\\\\n",
       "\t cds & 2 &  73 & 109 & 109 & 0 & 73f  & PRIMER\\_LEFT  & TGAAAGTHAGASCTTCYGTDAA & 22 & 36 & 3 &  1 &  23 & 56.23671 & 1.915052 & 37.12121 & 3.787879 & 7.511429 & 16.79607 & -27.56715 & 2.442498e+01 & 97.297 & Oscillospiraceae bacterium NSJ-54                                                                                                                                                                                          & 3 & 36\\\\\n",
       "\t cds & 2 &  73 & 109 & 109 & 0 & 73f  & PRIMER\\_LEFT  & TGAAAGTHAGASCTTCYGTDAA & 22 & 36 & 3 &  1 &  23 & 56.23671 & 1.915052 & 37.12121 & 3.787879 & 7.511429 & 16.79607 & -27.56715 & 2.442498e+01 & 97.297 & Oscillospiraceae;Hydrogenoanaerobacterium saccharovorans;Anaerotruncus sp. G3(2012);Oscillospiraceae bacterium;Marasmitruncus massiliensis;Anaerotruncus sp. X29;Anaerotruncus sp. 1XD22-93;Anaerotruncus sp. 1XD42-93     & 3 & 36\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 30 × 26\n",
       "\n",
       "| gene_type &lt;chr&gt; | cluster_id &lt;int&gt; | primer_set &lt;int&gt; | amplicon_size_consensus &lt;int&gt; | amplicon_size_avg &lt;dbl&gt; | amplicon_size_sd &lt;dbl&gt; | primer_id &lt;fct&gt; | primer_type &lt;fct&gt; | sequence &lt;fct&gt; | length &lt;int&gt; | degeneracy &lt;int&gt; | degeneracy_3prime &lt;int&gt; | position_start &lt;int&gt; | position_end &lt;int&gt; | Tm_avg &lt;dbl&gt; | Tm_sd &lt;dbl&gt; | GC_avg &lt;dbl&gt; | GC_sd &lt;dbl&gt; | hairpin_avg &lt;dbl&gt; | hairpin_sd &lt;dbl&gt; | homodimer_avg &lt;dbl&gt; | homodimer_sd &lt;dbl&gt; | pident &lt;dbl&gt; | sscinames &lt;fct&gt; | max_degeneracy_3prime &lt;int&gt; | max_degeneracy &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| cds | 2 | 546 | 103 | 103 | 0 | 546f | PRIMER_LEFT  | ATGAAAGTHAGASCTTCYGT   | 20 | 12 | 2 |  0 |  20 | 54.37386 | 1.763376 | 39.16667 | 3.435921 | 7.511429 | 16.79607 | -27.56715 | 2.442498e+01 | 97.297 | Oscillospiraceae bacterium NSJ-54                                                                                                                                                                                          | 2 | 24 |\n",
       "| cds | 2 | 546 | 103 | 103 | 0 | 546f | PRIMER_LEFT  | ATGAAAGTHAGASCTTCYGT   | 20 | 12 | 2 |  0 |  20 | 54.37386 | 1.763376 | 39.16667 | 3.435921 | 7.511429 | 16.79607 | -27.56715 | 2.442498e+01 | 97.297 | Oscillospiraceae;Hydrogenoanaerobacterium saccharovorans;Anaerotruncus sp. G3(2012);Oscillospiraceae bacterium;Marasmitruncus massiliensis;Anaerotruncus sp. X29;Anaerotruncus sp. 1XD22-93;Anaerotruncus sp. 1XD42-93     | 2 | 24 |\n",
       "| cds | 2 | 546 | 103 | 103 | 0 | 546f | PRIMER_LEFT  | ATGAAAGTHAGASCTTCYGT   | 20 | 12 | 2 |  0 |  20 | 54.37386 | 1.763376 | 39.16667 | 3.435921 | 7.511429 | 16.79607 | -27.56715 | 2.442498e+01 | 97.297 | Oscillospiraceae bacterium                                                                                                                                                                                                 | 2 | 24 |\n",
       "| cds | 2 | 546 | 103 | 103 | 0 | 546f | PRIMER_LEFT  | ATGAAAGTHAGASCTTCYGT   | 20 | 12 | 2 |  0 |  20 | 54.37386 | 1.763376 | 39.16667 | 3.435921 | 7.511429 | 16.79607 | -27.56715 | 2.442498e+01 | 97.297 | Clostridium sp. CAG:169;Clostridium sp. CAG:242;Negativibacillus massiliensis                                                                                                                                              | 2 | 24 |\n",
       "| cds | 2 | 546 | 103 | 103 | 0 | 546f | PRIMER_LEFT  | ATGAAAGTHAGASCTTCYGT   | 20 | 12 | 2 |  0 |  20 | 54.37386 | 1.763376 | 39.16667 | 3.435921 | 7.511429 | 16.79607 | -27.56715 | 2.442498e+01 | 97.297 | Ruminococcus flavefaciens;Anaerofustis stercorihominis;Oscillospiraceae bacterium;Clostridiales bacterium                                                                                                                  | 2 | 24 |\n",
       "| cds | 2 | 546 | 103 | 103 | 0 | 546f | PRIMER_LEFT  | ATGAAAGTHAGASCTTCYGT   | 20 | 12 | 2 |  0 |  20 | 54.37386 | 1.763376 | 39.16667 | 3.435921 | 7.511429 | 16.79607 | -27.56715 | 2.442498e+01 | 97.297 | Eubacteriales;Candidatus Soleaferrea massiliensis AP7;Anaerotruncus massiliensis;Anaerotruncus rubiinfantis;Anaerotruncus sp. AF02-27;Anaerotruncus sp. 22A2-44;Oscillospiraceae bacterium;Clostridiaceae bacterium NSJ-31 | 2 | 24 |\n",
       "| cds | 2 | 546 | 103 | 103 | 0 | 546f | PRIMER_LEFT  | ATGAAAGTHAGASCTTCYGT   | 20 | 12 | 2 |  0 |  20 | 54.37386 | 1.763376 | 39.16667 | 3.435921 | 7.511429 | 16.79607 | -27.56715 | 2.442498e+01 | 97.297 | Acetanaerobacterium elongatum                                                                                                                                                                                              | 2 | 24 |\n",
       "| cds | 2 | 546 | 103 | 103 | 0 | 546r | PRIMER_RIGHT | KCTGTTTRTGCTTBGGRTTT   | 20 | 24 | 2 | 83 | 103 | 56.37333 | 2.279991 | 40.83333 | 4.930066 | 0.000000 |  0.00000 | -33.75316 | 2.131628e-14 | 97.297 | Oscillospiraceae bacterium NSJ-54                                                                                                                                                                                          | 2 | 24 |\n",
       "| cds | 2 | 546 | 103 | 103 | 0 | 546r | PRIMER_RIGHT | KCTGTTTRTGCTTBGGRTTT   | 20 | 24 | 2 | 83 | 103 | 56.37333 | 2.279991 | 40.83333 | 4.930066 | 0.000000 |  0.00000 | -33.75316 | 2.131628e-14 | 97.297 | Oscillospiraceae;Hydrogenoanaerobacterium saccharovorans;Anaerotruncus sp. G3(2012);Oscillospiraceae bacterium;Marasmitruncus massiliensis;Anaerotruncus sp. X29;Anaerotruncus sp. 1XD22-93;Anaerotruncus sp. 1XD42-93     | 2 | 24 |\n",
       "| cds | 2 | 546 | 103 | 103 | 0 | 546r | PRIMER_RIGHT | KCTGTTTRTGCTTBGGRTTT   | 20 | 24 | 2 | 83 | 103 | 56.37333 | 2.279991 | 40.83333 | 4.930066 | 0.000000 |  0.00000 | -33.75316 | 2.131628e-14 | 97.297 | Oscillospiraceae bacterium                                                                                                                                                                                                 | 2 | 24 |\n",
       "| cds | 2 | 546 | 103 | 103 | 0 | 546r | PRIMER_RIGHT | KCTGTTTRTGCTTBGGRTTT   | 20 | 24 | 2 | 83 | 103 | 56.37333 | 2.279991 | 40.83333 | 4.930066 | 0.000000 |  0.00000 | -33.75316 | 2.131628e-14 | 97.297 | Clostridium sp. CAG:169;Clostridium sp. CAG:242;Negativibacillus massiliensis                                                                                                                                              | 2 | 24 |\n",
       "| cds | 2 | 546 | 103 | 103 | 0 | 546r | PRIMER_RIGHT | KCTGTTTRTGCTTBGGRTTT   | 20 | 24 | 2 | 83 | 103 | 56.37333 | 2.279991 | 40.83333 | 4.930066 | 0.000000 |  0.00000 | -33.75316 | 2.131628e-14 | 97.297 | Ruminococcus flavefaciens;Anaerofustis stercorihominis;Oscillospiraceae bacterium;Clostridiales bacterium                                                                                                                  | 2 | 24 |\n",
       "| cds | 2 | 546 | 103 | 103 | 0 | 546r | PRIMER_RIGHT | KCTGTTTRTGCTTBGGRTTT   | 20 | 24 | 2 | 83 | 103 | 56.37333 | 2.279991 | 40.83333 | 4.930066 | 0.000000 |  0.00000 | -33.75316 | 2.131628e-14 | 97.297 | Eubacteriales;Candidatus Soleaferrea massiliensis AP7;Anaerotruncus massiliensis;Anaerotruncus rubiinfantis;Anaerotruncus sp. AF02-27;Anaerotruncus sp. 22A2-44;Oscillospiraceae bacterium;Clostridiaceae bacterium NSJ-31 | 2 | 24 |\n",
       "| cds | 2 | 546 | 103 | 103 | 0 | 546r | PRIMER_RIGHT | KCTGTTTRTGCTTBGGRTTT   | 20 | 24 | 2 | 83 | 103 | 56.37333 | 2.279991 | 40.83333 | 4.930066 | 0.000000 |  0.00000 | -33.75316 | 2.131628e-14 | 97.297 | Acetanaerobacterium elongatum                                                                                                                                                                                              | 2 | 24 |\n",
       "| cds | 2 | 809 | 106 | 106 | 0 | 809f | PRIMER_LEFT  | ATGAAAGTHAGASCTTCYGT   | 20 | 12 | 2 |  0 |  20 | 54.37386 | 1.763376 | 39.16667 | 3.435921 | 7.511429 | 16.79607 | -27.56715 | 2.442498e+01 | 97.297 | Oscillospiraceae bacterium NSJ-54                                                                                                                                                                                          | 3 | 24 |\n",
       "| cds | 2 | 809 | 106 | 106 | 0 | 809f | PRIMER_LEFT  | ATGAAAGTHAGASCTTCYGT   | 20 | 12 | 2 |  0 |  20 | 54.37386 | 1.763376 | 39.16667 | 3.435921 | 7.511429 | 16.79607 | -27.56715 | 2.442498e+01 | 97.297 | Oscillospiraceae;Hydrogenoanaerobacterium saccharovorans;Anaerotruncus sp. G3(2012);Oscillospiraceae bacterium;Marasmitruncus massiliensis;Anaerotruncus sp. X29;Anaerotruncus sp. 1XD22-93;Anaerotruncus sp. 1XD42-93     | 3 | 24 |\n",
       "| cds | 2 | 809 | 106 | 106 | 0 | 809f | PRIMER_LEFT  | ATGAAAGTHAGASCTTCYGT   | 20 | 12 | 2 |  0 |  20 | 54.37386 | 1.763376 | 39.16667 | 3.435921 | 7.511429 | 16.79607 | -27.56715 | 2.442498e+01 | 97.297 | Oscillospiraceae bacterium                                                                                                                                                                                                 | 3 | 24 |\n",
       "| cds | 2 | 809 | 106 | 106 | 0 | 809f | PRIMER_LEFT  | ATGAAAGTHAGASCTTCYGT   | 20 | 12 | 2 |  0 |  20 | 54.37386 | 1.763376 | 39.16667 | 3.435921 | 7.511429 | 16.79607 | -27.56715 | 2.442498e+01 | 97.297 | Clostridium sp. CAG:169;Clostridium sp. CAG:242;Negativibacillus massiliensis                                                                                                                                              | 3 | 24 |\n",
       "| cds | 2 | 809 | 106 | 106 | 0 | 809f | PRIMER_LEFT  | ATGAAAGTHAGASCTTCYGT   | 20 | 12 | 2 |  0 |  20 | 54.37386 | 1.763376 | 39.16667 | 3.435921 | 7.511429 | 16.79607 | -27.56715 | 2.442498e+01 | 97.297 | Ruminococcus flavefaciens;Anaerofustis stercorihominis;Oscillospiraceae bacterium;Clostridiales bacterium                                                                                                                  | 3 | 24 |\n",
       "| cds | 2 | 809 | 106 | 106 | 0 | 809f | PRIMER_LEFT  | ATGAAAGTHAGASCTTCYGT   | 20 | 12 | 2 |  0 |  20 | 54.37386 | 1.763376 | 39.16667 | 3.435921 | 7.511429 | 16.79607 | -27.56715 | 2.442498e+01 | 97.297 | Eubacteriales;Candidatus Soleaferrea massiliensis AP7;Anaerotruncus massiliensis;Anaerotruncus rubiinfantis;Anaerotruncus sp. AF02-27;Anaerotruncus sp. 22A2-44;Oscillospiraceae bacterium;Clostridiaceae bacterium NSJ-31 | 3 | 24 |\n",
       "| cds | 2 | 809 | 106 | 106 | 0 | 809f | PRIMER_LEFT  | ATGAAAGTHAGASCTTCYGT   | 20 | 12 | 2 |  0 |  20 | 54.37386 | 1.763376 | 39.16667 | 3.435921 | 7.511429 | 16.79607 | -27.56715 | 2.442498e+01 | 97.297 | Acetanaerobacterium elongatum                                                                                                                                                                                              | 3 | 24 |\n",
       "| cds | 2 | 809 | 106 | 106 | 0 | 809r | PRIMER_RIGHT | GWCKCTGTTTRTGCTTBGG    | 19 | 24 | 3 | 87 | 106 | 56.74707 | 2.521808 | 50.87719 | 4.472824 | 0.000000 |  0.00000 | -33.75316 | 2.131628e-14 | 97.297 | Oscillospiraceae bacterium NSJ-54                                                                                                                                                                                          | 3 | 24 |\n",
       "| cds | 2 | 809 | 106 | 106 | 0 | 809r | PRIMER_RIGHT | GWCKCTGTTTRTGCTTBGG    | 19 | 24 | 3 | 87 | 106 | 56.74707 | 2.521808 | 50.87719 | 4.472824 | 0.000000 |  0.00000 | -33.75316 | 2.131628e-14 | 97.297 | Oscillospiraceae;Hydrogenoanaerobacterium saccharovorans;Anaerotruncus sp. G3(2012);Oscillospiraceae bacterium;Marasmitruncus massiliensis;Anaerotruncus sp. X29;Anaerotruncus sp. 1XD22-93;Anaerotruncus sp. 1XD42-93     | 3 | 24 |\n",
       "| cds | 2 | 809 | 106 | 106 | 0 | 809r | PRIMER_RIGHT | GWCKCTGTTTRTGCTTBGG    | 19 | 24 | 3 | 87 | 106 | 56.74707 | 2.521808 | 50.87719 | 4.472824 | 0.000000 |  0.00000 | -33.75316 | 2.131628e-14 | 97.297 | Oscillospiraceae bacterium                                                                                                                                                                                                 | 3 | 24 |\n",
       "| cds | 2 | 809 | 106 | 106 | 0 | 809r | PRIMER_RIGHT | GWCKCTGTTTRTGCTTBGG    | 19 | 24 | 3 | 87 | 106 | 56.74707 | 2.521808 | 50.87719 | 4.472824 | 0.000000 |  0.00000 | -33.75316 | 2.131628e-14 | 97.297 | Clostridium sp. CAG:169;Clostridium sp. CAG:242;Negativibacillus massiliensis                                                                                                                                              | 3 | 24 |\n",
       "| cds | 2 | 809 | 106 | 106 | 0 | 809r | PRIMER_RIGHT | GWCKCTGTTTRTGCTTBGG    | 19 | 24 | 3 | 87 | 106 | 56.74707 | 2.521808 | 50.87719 | 4.472824 | 0.000000 |  0.00000 | -33.75316 | 2.131628e-14 | 97.297 | Ruminococcus flavefaciens;Anaerofustis stercorihominis;Oscillospiraceae bacterium;Clostridiales bacterium                                                                                                                  | 3 | 24 |\n",
       "| cds | 2 | 809 | 106 | 106 | 0 | 809r | PRIMER_RIGHT | GWCKCTGTTTRTGCTTBGG    | 19 | 24 | 3 | 87 | 106 | 56.74707 | 2.521808 | 50.87719 | 4.472824 | 0.000000 |  0.00000 | -33.75316 | 2.131628e-14 | 97.297 | Eubacteriales;Candidatus Soleaferrea massiliensis AP7;Anaerotruncus massiliensis;Anaerotruncus rubiinfantis;Anaerotruncus sp. AF02-27;Anaerotruncus sp. 22A2-44;Oscillospiraceae bacterium;Clostridiaceae bacterium NSJ-31 | 3 | 24 |\n",
       "| cds | 2 | 809 | 106 | 106 | 0 | 809r | PRIMER_RIGHT | GWCKCTGTTTRTGCTTBGG    | 19 | 24 | 3 | 87 | 106 | 56.74707 | 2.521808 | 50.87719 | 4.472824 | 0.000000 |  0.00000 | -33.75316 | 2.131628e-14 | 97.297 | Acetanaerobacterium elongatum                                                                                                                                                                                              | 3 | 24 |\n",
       "| cds | 2 |  73 | 109 | 109 | 0 | 73f  | PRIMER_LEFT  | TGAAAGTHAGASCTTCYGTDAA | 22 | 36 | 3 |  1 |  23 | 56.23671 | 1.915052 | 37.12121 | 3.787879 | 7.511429 | 16.79607 | -27.56715 | 2.442498e+01 | 97.297 | Oscillospiraceae bacterium NSJ-54                                                                                                                                                                                          | 3 | 36 |\n",
       "| cds | 2 |  73 | 109 | 109 | 0 | 73f  | PRIMER_LEFT  | TGAAAGTHAGASCTTCYGTDAA | 22 | 36 | 3 |  1 |  23 | 56.23671 | 1.915052 | 37.12121 | 3.787879 | 7.511429 | 16.79607 | -27.56715 | 2.442498e+01 | 97.297 | Oscillospiraceae;Hydrogenoanaerobacterium saccharovorans;Anaerotruncus sp. G3(2012);Oscillospiraceae bacterium;Marasmitruncus massiliensis;Anaerotruncus sp. X29;Anaerotruncus sp. 1XD22-93;Anaerotruncus sp. 1XD42-93     | 3 | 36 |\n",
       "\n"
      ],
      "text/plain": [
       "   gene_type cluster_id primer_set amplicon_size_consensus amplicon_size_avg\n",
       "1  cds       2          546        103                     103              \n",
       "2  cds       2          546        103                     103              \n",
       "3  cds       2          546        103                     103              \n",
       "4  cds       2          546        103                     103              \n",
       "5  cds       2          546        103                     103              \n",
       "6  cds       2          546        103                     103              \n",
       "7  cds       2          546        103                     103              \n",
       "8  cds       2          546        103                     103              \n",
       "9  cds       2          546        103                     103              \n",
       "10 cds       2          546        103                     103              \n",
       "11 cds       2          546        103                     103              \n",
       "12 cds       2          546        103                     103              \n",
       "13 cds       2          546        103                     103              \n",
       "14 cds       2          546        103                     103              \n",
       "15 cds       2          809        106                     106              \n",
       "16 cds       2          809        106                     106              \n",
       "17 cds       2          809        106                     106              \n",
       "18 cds       2          809        106                     106              \n",
       "19 cds       2          809        106                     106              \n",
       "20 cds       2          809        106                     106              \n",
       "21 cds       2          809        106                     106              \n",
       "22 cds       2          809        106                     106              \n",
       "23 cds       2          809        106                     106              \n",
       "24 cds       2          809        106                     106              \n",
       "25 cds       2          809        106                     106              \n",
       "26 cds       2          809        106                     106              \n",
       "27 cds       2          809        106                     106              \n",
       "28 cds       2          809        106                     106              \n",
       "29 cds       2           73        109                     109              \n",
       "30 cds       2           73        109                     109              \n",
       "   amplicon_size_sd primer_id primer_type  sequence               length\n",
       "1  0                546f      PRIMER_LEFT  ATGAAAGTHAGASCTTCYGT   20    \n",
       "2  0                546f      PRIMER_LEFT  ATGAAAGTHAGASCTTCYGT   20    \n",
       "3  0                546f      PRIMER_LEFT  ATGAAAGTHAGASCTTCYGT   20    \n",
       "4  0                546f      PRIMER_LEFT  ATGAAAGTHAGASCTTCYGT   20    \n",
       "5  0                546f      PRIMER_LEFT  ATGAAAGTHAGASCTTCYGT   20    \n",
       "6  0                546f      PRIMER_LEFT  ATGAAAGTHAGASCTTCYGT   20    \n",
       "7  0                546f      PRIMER_LEFT  ATGAAAGTHAGASCTTCYGT   20    \n",
       "8  0                546r      PRIMER_RIGHT KCTGTTTRTGCTTBGGRTTT   20    \n",
       "9  0                546r      PRIMER_RIGHT KCTGTTTRTGCTTBGGRTTT   20    \n",
       "10 0                546r      PRIMER_RIGHT KCTGTTTRTGCTTBGGRTTT   20    \n",
       "11 0                546r      PRIMER_RIGHT KCTGTTTRTGCTTBGGRTTT   20    \n",
       "12 0                546r      PRIMER_RIGHT KCTGTTTRTGCTTBGGRTTT   20    \n",
       "13 0                546r      PRIMER_RIGHT KCTGTTTRTGCTTBGGRTTT   20    \n",
       "14 0                546r      PRIMER_RIGHT KCTGTTTRTGCTTBGGRTTT   20    \n",
       "15 0                809f      PRIMER_LEFT  ATGAAAGTHAGASCTTCYGT   20    \n",
       "16 0                809f      PRIMER_LEFT  ATGAAAGTHAGASCTTCYGT   20    \n",
       "17 0                809f      PRIMER_LEFT  ATGAAAGTHAGASCTTCYGT   20    \n",
       "18 0                809f      PRIMER_LEFT  ATGAAAGTHAGASCTTCYGT   20    \n",
       "19 0                809f      PRIMER_LEFT  ATGAAAGTHAGASCTTCYGT   20    \n",
       "20 0                809f      PRIMER_LEFT  ATGAAAGTHAGASCTTCYGT   20    \n",
       "21 0                809f      PRIMER_LEFT  ATGAAAGTHAGASCTTCYGT   20    \n",
       "22 0                809r      PRIMER_RIGHT GWCKCTGTTTRTGCTTBGG    19    \n",
       "23 0                809r      PRIMER_RIGHT GWCKCTGTTTRTGCTTBGG    19    \n",
       "24 0                809r      PRIMER_RIGHT GWCKCTGTTTRTGCTTBGG    19    \n",
       "25 0                809r      PRIMER_RIGHT GWCKCTGTTTRTGCTTBGG    19    \n",
       "26 0                809r      PRIMER_RIGHT GWCKCTGTTTRTGCTTBGG    19    \n",
       "27 0                809r      PRIMER_RIGHT GWCKCTGTTTRTGCTTBGG    19    \n",
       "28 0                809r      PRIMER_RIGHT GWCKCTGTTTRTGCTTBGG    19    \n",
       "29 0                73f       PRIMER_LEFT  TGAAAGTHAGASCTTCYGTDAA 22    \n",
       "30 0                73f       PRIMER_LEFT  TGAAAGTHAGASCTTCYGTDAA 22    \n",
       "   degeneracy degeneracy_3prime position_start position_end Tm_avg   Tm_sd   \n",
       "1  12         2                  0              20          54.37386 1.763376\n",
       "2  12         2                  0              20          54.37386 1.763376\n",
       "3  12         2                  0              20          54.37386 1.763376\n",
       "4  12         2                  0              20          54.37386 1.763376\n",
       "5  12         2                  0              20          54.37386 1.763376\n",
       "6  12         2                  0              20          54.37386 1.763376\n",
       "7  12         2                  0              20          54.37386 1.763376\n",
       "8  24         2                 83             103          56.37333 2.279991\n",
       "9  24         2                 83             103          56.37333 2.279991\n",
       "10 24         2                 83             103          56.37333 2.279991\n",
       "11 24         2                 83             103          56.37333 2.279991\n",
       "12 24         2                 83             103          56.37333 2.279991\n",
       "13 24         2                 83             103          56.37333 2.279991\n",
       "14 24         2                 83             103          56.37333 2.279991\n",
       "15 12         2                  0              20          54.37386 1.763376\n",
       "16 12         2                  0              20          54.37386 1.763376\n",
       "17 12         2                  0              20          54.37386 1.763376\n",
       "18 12         2                  0              20          54.37386 1.763376\n",
       "19 12         2                  0              20          54.37386 1.763376\n",
       "20 12         2                  0              20          54.37386 1.763376\n",
       "21 12         2                  0              20          54.37386 1.763376\n",
       "22 24         3                 87             106          56.74707 2.521808\n",
       "23 24         3                 87             106          56.74707 2.521808\n",
       "24 24         3                 87             106          56.74707 2.521808\n",
       "25 24         3                 87             106          56.74707 2.521808\n",
       "26 24         3                 87             106          56.74707 2.521808\n",
       "27 24         3                 87             106          56.74707 2.521808\n",
       "28 24         3                 87             106          56.74707 2.521808\n",
       "29 36         3                  1              23          56.23671 1.915052\n",
       "30 36         3                  1              23          56.23671 1.915052\n",
       "   GC_avg   GC_sd    hairpin_avg hairpin_sd homodimer_avg homodimer_sd pident\n",
       "1  39.16667 3.435921 7.511429    16.79607   -27.56715     2.442498e+01 97.297\n",
       "2  39.16667 3.435921 7.511429    16.79607   -27.56715     2.442498e+01 97.297\n",
       "3  39.16667 3.435921 7.511429    16.79607   -27.56715     2.442498e+01 97.297\n",
       "4  39.16667 3.435921 7.511429    16.79607   -27.56715     2.442498e+01 97.297\n",
       "5  39.16667 3.435921 7.511429    16.79607   -27.56715     2.442498e+01 97.297\n",
       "6  39.16667 3.435921 7.511429    16.79607   -27.56715     2.442498e+01 97.297\n",
       "7  39.16667 3.435921 7.511429    16.79607   -27.56715     2.442498e+01 97.297\n",
       "8  40.83333 4.930066 0.000000     0.00000   -33.75316     2.131628e-14 97.297\n",
       "9  40.83333 4.930066 0.000000     0.00000   -33.75316     2.131628e-14 97.297\n",
       "10 40.83333 4.930066 0.000000     0.00000   -33.75316     2.131628e-14 97.297\n",
       "11 40.83333 4.930066 0.000000     0.00000   -33.75316     2.131628e-14 97.297\n",
       "12 40.83333 4.930066 0.000000     0.00000   -33.75316     2.131628e-14 97.297\n",
       "13 40.83333 4.930066 0.000000     0.00000   -33.75316     2.131628e-14 97.297\n",
       "14 40.83333 4.930066 0.000000     0.00000   -33.75316     2.131628e-14 97.297\n",
       "15 39.16667 3.435921 7.511429    16.79607   -27.56715     2.442498e+01 97.297\n",
       "16 39.16667 3.435921 7.511429    16.79607   -27.56715     2.442498e+01 97.297\n",
       "17 39.16667 3.435921 7.511429    16.79607   -27.56715     2.442498e+01 97.297\n",
       "18 39.16667 3.435921 7.511429    16.79607   -27.56715     2.442498e+01 97.297\n",
       "19 39.16667 3.435921 7.511429    16.79607   -27.56715     2.442498e+01 97.297\n",
       "20 39.16667 3.435921 7.511429    16.79607   -27.56715     2.442498e+01 97.297\n",
       "21 39.16667 3.435921 7.511429    16.79607   -27.56715     2.442498e+01 97.297\n",
       "22 50.87719 4.472824 0.000000     0.00000   -33.75316     2.131628e-14 97.297\n",
       "23 50.87719 4.472824 0.000000     0.00000   -33.75316     2.131628e-14 97.297\n",
       "24 50.87719 4.472824 0.000000     0.00000   -33.75316     2.131628e-14 97.297\n",
       "25 50.87719 4.472824 0.000000     0.00000   -33.75316     2.131628e-14 97.297\n",
       "26 50.87719 4.472824 0.000000     0.00000   -33.75316     2.131628e-14 97.297\n",
       "27 50.87719 4.472824 0.000000     0.00000   -33.75316     2.131628e-14 97.297\n",
       "28 50.87719 4.472824 0.000000     0.00000   -33.75316     2.131628e-14 97.297\n",
       "29 37.12121 3.787879 7.511429    16.79607   -27.56715     2.442498e+01 97.297\n",
       "30 37.12121 3.787879 7.511429    16.79607   -27.56715     2.442498e+01 97.297\n",
       "   sscinames                                                                                                                                                                                                                 \n",
       "1  Oscillospiraceae bacterium NSJ-54                                                                                                                                                                                         \n",
       "2  Oscillospiraceae;Hydrogenoanaerobacterium saccharovorans;Anaerotruncus sp. G3(2012);Oscillospiraceae bacterium;Marasmitruncus massiliensis;Anaerotruncus sp. X29;Anaerotruncus sp. 1XD22-93;Anaerotruncus sp. 1XD42-93    \n",
       "3  Oscillospiraceae bacterium                                                                                                                                                                                                \n",
       "4  Clostridium sp. CAG:169;Clostridium sp. CAG:242;Negativibacillus massiliensis                                                                                                                                             \n",
       "5  Ruminococcus flavefaciens;Anaerofustis stercorihominis;Oscillospiraceae bacterium;Clostridiales bacterium                                                                                                                 \n",
       "6  Eubacteriales;Candidatus Soleaferrea massiliensis AP7;Anaerotruncus massiliensis;Anaerotruncus rubiinfantis;Anaerotruncus sp. AF02-27;Anaerotruncus sp. 22A2-44;Oscillospiraceae bacterium;Clostridiaceae bacterium NSJ-31\n",
       "7  Acetanaerobacterium elongatum                                                                                                                                                                                             \n",
       "8  Oscillospiraceae bacterium NSJ-54                                                                                                                                                                                         \n",
       "9  Oscillospiraceae;Hydrogenoanaerobacterium saccharovorans;Anaerotruncus sp. G3(2012);Oscillospiraceae bacterium;Marasmitruncus massiliensis;Anaerotruncus sp. X29;Anaerotruncus sp. 1XD22-93;Anaerotruncus sp. 1XD42-93    \n",
       "10 Oscillospiraceae bacterium                                                                                                                                                                                                \n",
       "11 Clostridium sp. CAG:169;Clostridium sp. CAG:242;Negativibacillus massiliensis                                                                                                                                             \n",
       "12 Ruminococcus flavefaciens;Anaerofustis stercorihominis;Oscillospiraceae bacterium;Clostridiales bacterium                                                                                                                 \n",
       "13 Eubacteriales;Candidatus Soleaferrea massiliensis AP7;Anaerotruncus massiliensis;Anaerotruncus rubiinfantis;Anaerotruncus sp. AF02-27;Anaerotruncus sp. 22A2-44;Oscillospiraceae bacterium;Clostridiaceae bacterium NSJ-31\n",
       "14 Acetanaerobacterium elongatum                                                                                                                                                                                             \n",
       "15 Oscillospiraceae bacterium NSJ-54                                                                                                                                                                                         \n",
       "16 Oscillospiraceae;Hydrogenoanaerobacterium saccharovorans;Anaerotruncus sp. G3(2012);Oscillospiraceae bacterium;Marasmitruncus massiliensis;Anaerotruncus sp. X29;Anaerotruncus sp. 1XD22-93;Anaerotruncus sp. 1XD42-93    \n",
       "17 Oscillospiraceae bacterium                                                                                                                                                                                                \n",
       "18 Clostridium sp. CAG:169;Clostridium sp. CAG:242;Negativibacillus massiliensis                                                                                                                                             \n",
       "19 Ruminococcus flavefaciens;Anaerofustis stercorihominis;Oscillospiraceae bacterium;Clostridiales bacterium                                                                                                                 \n",
       "20 Eubacteriales;Candidatus Soleaferrea massiliensis AP7;Anaerotruncus massiliensis;Anaerotruncus rubiinfantis;Anaerotruncus sp. AF02-27;Anaerotruncus sp. 22A2-44;Oscillospiraceae bacterium;Clostridiaceae bacterium NSJ-31\n",
       "21 Acetanaerobacterium elongatum                                                                                                                                                                                             \n",
       "22 Oscillospiraceae bacterium NSJ-54                                                                                                                                                                                         \n",
       "23 Oscillospiraceae;Hydrogenoanaerobacterium saccharovorans;Anaerotruncus sp. G3(2012);Oscillospiraceae bacterium;Marasmitruncus massiliensis;Anaerotruncus sp. X29;Anaerotruncus sp. 1XD22-93;Anaerotruncus sp. 1XD42-93    \n",
       "24 Oscillospiraceae bacterium                                                                                                                                                                                                \n",
       "25 Clostridium sp. CAG:169;Clostridium sp. CAG:242;Negativibacillus massiliensis                                                                                                                                             \n",
       "26 Ruminococcus flavefaciens;Anaerofustis stercorihominis;Oscillospiraceae bacterium;Clostridiales bacterium                                                                                                                 \n",
       "27 Eubacteriales;Candidatus Soleaferrea massiliensis AP7;Anaerotruncus massiliensis;Anaerotruncus rubiinfantis;Anaerotruncus sp. AF02-27;Anaerotruncus sp. 22A2-44;Oscillospiraceae bacterium;Clostridiaceae bacterium NSJ-31\n",
       "28 Acetanaerobacterium elongatum                                                                                                                                                                                             \n",
       "29 Oscillospiraceae bacterium NSJ-54                                                                                                                                                                                         \n",
       "30 Oscillospiraceae;Hydrogenoanaerobacterium saccharovorans;Anaerotruncus sp. G3(2012);Oscillospiraceae bacterium;Marasmitruncus massiliensis;Anaerotruncus sp. X29;Anaerotruncus sp. 1XD22-93;Anaerotruncus sp. 1XD42-93    \n",
       "   max_degeneracy_3prime max_degeneracy\n",
       "1  2                     24            \n",
       "2  2                     24            \n",
       "3  2                     24            \n",
       "4  2                     24            \n",
       "5  2                     24            \n",
       "6  2                     24            \n",
       "7  2                     24            \n",
       "8  2                     24            \n",
       "9  2                     24            \n",
       "10 2                     24            \n",
       "11 2                     24            \n",
       "12 2                     24            \n",
       "13 2                     24            \n",
       "14 2                     24            \n",
       "15 3                     24            \n",
       "16 3                     24            \n",
       "17 3                     24            \n",
       "18 3                     24            \n",
       "19 3                     24            \n",
       "20 3                     24            \n",
       "21 3                     24            \n",
       "22 3                     24            \n",
       "23 3                     24            \n",
       "24 3                     24            \n",
       "25 3                     24            \n",
       "26 3                     24            \n",
       "27 3                     24            \n",
       "28 3                     24            \n",
       "29 3                     36            \n",
       "30 3                     36            "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# arrange by \"best\" primers\n",
    "primer_info_jf = primer_info_j %>%\n",
    "    group_by(primer_set) %>%\n",
    "    mutate(max_degeneracy_3prime = max(degeneracy_3prime),  # max of [Fwd,Rev]\n",
    "           max_degeneracy = max(degeneracy)) %>%\n",
    "    ungroup() %>%\n",
    "    arrange(pident, max_degeneracy_3prime, max_degeneracy) %>%\n",
    "    head(n=30)\n",
    "\n",
    "df.dims(30,40)\n",
    "primer_info_jf\n",
    "df.dims()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### w/ gene annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-24T13:19:08.886736Z",
     "iopub.status.busy": "2021-04-24T13:19:08.884725Z",
     "iopub.status.idle": "2021-04-24T13:19:09.004416Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A tibble: 30 × 27</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>gene_type</th><th scope=col>cluster_id</th><th scope=col>primer_set</th><th scope=col>amplicon_size_consensus</th><th scope=col>amplicon_size_avg</th><th scope=col>amplicon_size_sd</th><th scope=col>primer_id</th><th scope=col>primer_type</th><th scope=col>sequence</th><th scope=col>length</th><th scope=col>degeneracy</th><th scope=col>degeneracy_3prime</th><th scope=col>position_start</th><th scope=col>position_end</th><th scope=col>Tm_avg</th><th scope=col>Tm_sd</th><th scope=col>GC_avg</th><th scope=col>GC_sd</th><th scope=col>hairpin_avg</th><th scope=col>hairpin_sd</th><th scope=col>homodimer_avg</th><th scope=col>homodimer_sd</th><th scope=col>pident</th><th scope=col>sscinames</th><th scope=col>max_degeneracy_3prime</th><th scope=col>max_degeneracy</th><th scope=col>annotation</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>cds</td><td>2</td><td>546</td><td>103</td><td>103</td><td>0</td><td>546f</td><td>PRIMER_LEFT </td><td>ATGAAAGTHAGASCTTCYGT  </td><td>20</td><td>12</td><td>2</td><td> 0</td><td> 20</td><td>54.37386</td><td>1.763376</td><td>39.16667</td><td>3.435921</td><td>7.511429</td><td>16.79607</td><td>-27.56715</td><td>2.442498e+01</td><td>97.297</td><td>Oscillospiraceae bacterium NSJ-54                                                                                                                                                                                         </td><td>2</td><td>24</td><td>50S ribosomal protein L36</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>546</td><td>103</td><td>103</td><td>0</td><td>546f</td><td>PRIMER_LEFT </td><td>ATGAAAGTHAGASCTTCYGT  </td><td>20</td><td>12</td><td>2</td><td> 0</td><td> 20</td><td>54.37386</td><td>1.763376</td><td>39.16667</td><td>3.435921</td><td>7.511429</td><td>16.79607</td><td>-27.56715</td><td>2.442498e+01</td><td>97.297</td><td>Oscillospiraceae;Hydrogenoanaerobacterium saccharovorans;Anaerotruncus sp. G3(2012);Oscillospiraceae bacterium;Marasmitruncus massiliensis;Anaerotruncus sp. X29;Anaerotruncus sp. 1XD22-93;Anaerotruncus sp. 1XD42-93    </td><td>2</td><td>24</td><td>50S ribosomal protein L36</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>546</td><td>103</td><td>103</td><td>0</td><td>546f</td><td>PRIMER_LEFT </td><td>ATGAAAGTHAGASCTTCYGT  </td><td>20</td><td>12</td><td>2</td><td> 0</td><td> 20</td><td>54.37386</td><td>1.763376</td><td>39.16667</td><td>3.435921</td><td>7.511429</td><td>16.79607</td><td>-27.56715</td><td>2.442498e+01</td><td>97.297</td><td>Oscillospiraceae bacterium                                                                                                                                                                                                </td><td>2</td><td>24</td><td>50S ribosomal protein L36</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>546</td><td>103</td><td>103</td><td>0</td><td>546f</td><td>PRIMER_LEFT </td><td>ATGAAAGTHAGASCTTCYGT  </td><td>20</td><td>12</td><td>2</td><td> 0</td><td> 20</td><td>54.37386</td><td>1.763376</td><td>39.16667</td><td>3.435921</td><td>7.511429</td><td>16.79607</td><td>-27.56715</td><td>2.442498e+01</td><td>97.297</td><td>Clostridium sp. CAG:169;Clostridium sp. CAG:242;Negativibacillus massiliensis                                                                                                                                             </td><td>2</td><td>24</td><td>50S ribosomal protein L36</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>546</td><td>103</td><td>103</td><td>0</td><td>546f</td><td>PRIMER_LEFT </td><td>ATGAAAGTHAGASCTTCYGT  </td><td>20</td><td>12</td><td>2</td><td> 0</td><td> 20</td><td>54.37386</td><td>1.763376</td><td>39.16667</td><td>3.435921</td><td>7.511429</td><td>16.79607</td><td>-27.56715</td><td>2.442498e+01</td><td>97.297</td><td>Ruminococcus flavefaciens;Anaerofustis stercorihominis;Oscillospiraceae bacterium;Clostridiales bacterium                                                                                                                 </td><td>2</td><td>24</td><td>50S ribosomal protein L36</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>546</td><td>103</td><td>103</td><td>0</td><td>546f</td><td>PRIMER_LEFT </td><td>ATGAAAGTHAGASCTTCYGT  </td><td>20</td><td>12</td><td>2</td><td> 0</td><td> 20</td><td>54.37386</td><td>1.763376</td><td>39.16667</td><td>3.435921</td><td>7.511429</td><td>16.79607</td><td>-27.56715</td><td>2.442498e+01</td><td>97.297</td><td>Eubacteriales;Candidatus Soleaferrea massiliensis AP7;Anaerotruncus massiliensis;Anaerotruncus rubiinfantis;Anaerotruncus sp. AF02-27;Anaerotruncus sp. 22A2-44;Oscillospiraceae bacterium;Clostridiaceae bacterium NSJ-31</td><td>2</td><td>24</td><td>50S ribosomal protein L36</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>546</td><td>103</td><td>103</td><td>0</td><td>546f</td><td>PRIMER_LEFT </td><td>ATGAAAGTHAGASCTTCYGT  </td><td>20</td><td>12</td><td>2</td><td> 0</td><td> 20</td><td>54.37386</td><td>1.763376</td><td>39.16667</td><td>3.435921</td><td>7.511429</td><td>16.79607</td><td>-27.56715</td><td>2.442498e+01</td><td>97.297</td><td>Acetanaerobacterium elongatum                                                                                                                                                                                             </td><td>2</td><td>24</td><td>50S ribosomal protein L36</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>546</td><td>103</td><td>103</td><td>0</td><td>546r</td><td>PRIMER_RIGHT</td><td>KCTGTTTRTGCTTBGGRTTT  </td><td>20</td><td>24</td><td>2</td><td>83</td><td>103</td><td>56.37333</td><td>2.279991</td><td>40.83333</td><td>4.930066</td><td>0.000000</td><td> 0.00000</td><td>-33.75316</td><td>2.131628e-14</td><td>97.297</td><td>Oscillospiraceae bacterium NSJ-54                                                                                                                                                                                         </td><td>2</td><td>24</td><td>50S ribosomal protein L36</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>546</td><td>103</td><td>103</td><td>0</td><td>546r</td><td>PRIMER_RIGHT</td><td>KCTGTTTRTGCTTBGGRTTT  </td><td>20</td><td>24</td><td>2</td><td>83</td><td>103</td><td>56.37333</td><td>2.279991</td><td>40.83333</td><td>4.930066</td><td>0.000000</td><td> 0.00000</td><td>-33.75316</td><td>2.131628e-14</td><td>97.297</td><td>Oscillospiraceae;Hydrogenoanaerobacterium saccharovorans;Anaerotruncus sp. G3(2012);Oscillospiraceae bacterium;Marasmitruncus massiliensis;Anaerotruncus sp. X29;Anaerotruncus sp. 1XD22-93;Anaerotruncus sp. 1XD42-93    </td><td>2</td><td>24</td><td>50S ribosomal protein L36</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>546</td><td>103</td><td>103</td><td>0</td><td>546r</td><td>PRIMER_RIGHT</td><td>KCTGTTTRTGCTTBGGRTTT  </td><td>20</td><td>24</td><td>2</td><td>83</td><td>103</td><td>56.37333</td><td>2.279991</td><td>40.83333</td><td>4.930066</td><td>0.000000</td><td> 0.00000</td><td>-33.75316</td><td>2.131628e-14</td><td>97.297</td><td>Oscillospiraceae bacterium                                                                                                                                                                                                </td><td>2</td><td>24</td><td>50S ribosomal protein L36</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>546</td><td>103</td><td>103</td><td>0</td><td>546r</td><td>PRIMER_RIGHT</td><td>KCTGTTTRTGCTTBGGRTTT  </td><td>20</td><td>24</td><td>2</td><td>83</td><td>103</td><td>56.37333</td><td>2.279991</td><td>40.83333</td><td>4.930066</td><td>0.000000</td><td> 0.00000</td><td>-33.75316</td><td>2.131628e-14</td><td>97.297</td><td>Clostridium sp. CAG:169;Clostridium sp. CAG:242;Negativibacillus massiliensis                                                                                                                                             </td><td>2</td><td>24</td><td>50S ribosomal protein L36</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>546</td><td>103</td><td>103</td><td>0</td><td>546r</td><td>PRIMER_RIGHT</td><td>KCTGTTTRTGCTTBGGRTTT  </td><td>20</td><td>24</td><td>2</td><td>83</td><td>103</td><td>56.37333</td><td>2.279991</td><td>40.83333</td><td>4.930066</td><td>0.000000</td><td> 0.00000</td><td>-33.75316</td><td>2.131628e-14</td><td>97.297</td><td>Ruminococcus flavefaciens;Anaerofustis stercorihominis;Oscillospiraceae bacterium;Clostridiales bacterium                                                                                                                 </td><td>2</td><td>24</td><td>50S ribosomal protein L36</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>546</td><td>103</td><td>103</td><td>0</td><td>546r</td><td>PRIMER_RIGHT</td><td>KCTGTTTRTGCTTBGGRTTT  </td><td>20</td><td>24</td><td>2</td><td>83</td><td>103</td><td>56.37333</td><td>2.279991</td><td>40.83333</td><td>4.930066</td><td>0.000000</td><td> 0.00000</td><td>-33.75316</td><td>2.131628e-14</td><td>97.297</td><td>Eubacteriales;Candidatus Soleaferrea massiliensis AP7;Anaerotruncus massiliensis;Anaerotruncus rubiinfantis;Anaerotruncus sp. AF02-27;Anaerotruncus sp. 22A2-44;Oscillospiraceae bacterium;Clostridiaceae bacterium NSJ-31</td><td>2</td><td>24</td><td>50S ribosomal protein L36</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>546</td><td>103</td><td>103</td><td>0</td><td>546r</td><td>PRIMER_RIGHT</td><td>KCTGTTTRTGCTTBGGRTTT  </td><td>20</td><td>24</td><td>2</td><td>83</td><td>103</td><td>56.37333</td><td>2.279991</td><td>40.83333</td><td>4.930066</td><td>0.000000</td><td> 0.00000</td><td>-33.75316</td><td>2.131628e-14</td><td>97.297</td><td>Acetanaerobacterium elongatum                                                                                                                                                                                             </td><td>2</td><td>24</td><td>50S ribosomal protein L36</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>809</td><td>106</td><td>106</td><td>0</td><td>809f</td><td>PRIMER_LEFT </td><td>ATGAAAGTHAGASCTTCYGT  </td><td>20</td><td>12</td><td>2</td><td> 0</td><td> 20</td><td>54.37386</td><td>1.763376</td><td>39.16667</td><td>3.435921</td><td>7.511429</td><td>16.79607</td><td>-27.56715</td><td>2.442498e+01</td><td>97.297</td><td>Oscillospiraceae bacterium NSJ-54                                                                                                                                                                                         </td><td>3</td><td>24</td><td>50S ribosomal protein L36</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>809</td><td>106</td><td>106</td><td>0</td><td>809f</td><td>PRIMER_LEFT </td><td>ATGAAAGTHAGASCTTCYGT  </td><td>20</td><td>12</td><td>2</td><td> 0</td><td> 20</td><td>54.37386</td><td>1.763376</td><td>39.16667</td><td>3.435921</td><td>7.511429</td><td>16.79607</td><td>-27.56715</td><td>2.442498e+01</td><td>97.297</td><td>Oscillospiraceae;Hydrogenoanaerobacterium saccharovorans;Anaerotruncus sp. G3(2012);Oscillospiraceae bacterium;Marasmitruncus massiliensis;Anaerotruncus sp. X29;Anaerotruncus sp. 1XD22-93;Anaerotruncus sp. 1XD42-93    </td><td>3</td><td>24</td><td>50S ribosomal protein L36</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>809</td><td>106</td><td>106</td><td>0</td><td>809f</td><td>PRIMER_LEFT </td><td>ATGAAAGTHAGASCTTCYGT  </td><td>20</td><td>12</td><td>2</td><td> 0</td><td> 20</td><td>54.37386</td><td>1.763376</td><td>39.16667</td><td>3.435921</td><td>7.511429</td><td>16.79607</td><td>-27.56715</td><td>2.442498e+01</td><td>97.297</td><td>Oscillospiraceae bacterium                                                                                                                                                                                                </td><td>3</td><td>24</td><td>50S ribosomal protein L36</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>809</td><td>106</td><td>106</td><td>0</td><td>809f</td><td>PRIMER_LEFT </td><td>ATGAAAGTHAGASCTTCYGT  </td><td>20</td><td>12</td><td>2</td><td> 0</td><td> 20</td><td>54.37386</td><td>1.763376</td><td>39.16667</td><td>3.435921</td><td>7.511429</td><td>16.79607</td><td>-27.56715</td><td>2.442498e+01</td><td>97.297</td><td>Clostridium sp. CAG:169;Clostridium sp. CAG:242;Negativibacillus massiliensis                                                                                                                                             </td><td>3</td><td>24</td><td>50S ribosomal protein L36</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>809</td><td>106</td><td>106</td><td>0</td><td>809f</td><td>PRIMER_LEFT </td><td>ATGAAAGTHAGASCTTCYGT  </td><td>20</td><td>12</td><td>2</td><td> 0</td><td> 20</td><td>54.37386</td><td>1.763376</td><td>39.16667</td><td>3.435921</td><td>7.511429</td><td>16.79607</td><td>-27.56715</td><td>2.442498e+01</td><td>97.297</td><td>Ruminococcus flavefaciens;Anaerofustis stercorihominis;Oscillospiraceae bacterium;Clostridiales bacterium                                                                                                                 </td><td>3</td><td>24</td><td>50S ribosomal protein L36</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>809</td><td>106</td><td>106</td><td>0</td><td>809f</td><td>PRIMER_LEFT </td><td>ATGAAAGTHAGASCTTCYGT  </td><td>20</td><td>12</td><td>2</td><td> 0</td><td> 20</td><td>54.37386</td><td>1.763376</td><td>39.16667</td><td>3.435921</td><td>7.511429</td><td>16.79607</td><td>-27.56715</td><td>2.442498e+01</td><td>97.297</td><td>Eubacteriales;Candidatus Soleaferrea massiliensis AP7;Anaerotruncus massiliensis;Anaerotruncus rubiinfantis;Anaerotruncus sp. AF02-27;Anaerotruncus sp. 22A2-44;Oscillospiraceae bacterium;Clostridiaceae bacterium NSJ-31</td><td>3</td><td>24</td><td>50S ribosomal protein L36</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>809</td><td>106</td><td>106</td><td>0</td><td>809f</td><td>PRIMER_LEFT </td><td>ATGAAAGTHAGASCTTCYGT  </td><td>20</td><td>12</td><td>2</td><td> 0</td><td> 20</td><td>54.37386</td><td>1.763376</td><td>39.16667</td><td>3.435921</td><td>7.511429</td><td>16.79607</td><td>-27.56715</td><td>2.442498e+01</td><td>97.297</td><td>Acetanaerobacterium elongatum                                                                                                                                                                                             </td><td>3</td><td>24</td><td>50S ribosomal protein L36</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>809</td><td>106</td><td>106</td><td>0</td><td>809r</td><td>PRIMER_RIGHT</td><td>GWCKCTGTTTRTGCTTBGG   </td><td>19</td><td>24</td><td>3</td><td>87</td><td>106</td><td>56.74707</td><td>2.521808</td><td>50.87719</td><td>4.472824</td><td>0.000000</td><td> 0.00000</td><td>-33.75316</td><td>2.131628e-14</td><td>97.297</td><td>Oscillospiraceae bacterium NSJ-54                                                                                                                                                                                         </td><td>3</td><td>24</td><td>50S ribosomal protein L36</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>809</td><td>106</td><td>106</td><td>0</td><td>809r</td><td>PRIMER_RIGHT</td><td>GWCKCTGTTTRTGCTTBGG   </td><td>19</td><td>24</td><td>3</td><td>87</td><td>106</td><td>56.74707</td><td>2.521808</td><td>50.87719</td><td>4.472824</td><td>0.000000</td><td> 0.00000</td><td>-33.75316</td><td>2.131628e-14</td><td>97.297</td><td>Oscillospiraceae;Hydrogenoanaerobacterium saccharovorans;Anaerotruncus sp. G3(2012);Oscillospiraceae bacterium;Marasmitruncus massiliensis;Anaerotruncus sp. X29;Anaerotruncus sp. 1XD22-93;Anaerotruncus sp. 1XD42-93    </td><td>3</td><td>24</td><td>50S ribosomal protein L36</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>809</td><td>106</td><td>106</td><td>0</td><td>809r</td><td>PRIMER_RIGHT</td><td>GWCKCTGTTTRTGCTTBGG   </td><td>19</td><td>24</td><td>3</td><td>87</td><td>106</td><td>56.74707</td><td>2.521808</td><td>50.87719</td><td>4.472824</td><td>0.000000</td><td> 0.00000</td><td>-33.75316</td><td>2.131628e-14</td><td>97.297</td><td>Oscillospiraceae bacterium                                                                                                                                                                                                </td><td>3</td><td>24</td><td>50S ribosomal protein L36</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>809</td><td>106</td><td>106</td><td>0</td><td>809r</td><td>PRIMER_RIGHT</td><td>GWCKCTGTTTRTGCTTBGG   </td><td>19</td><td>24</td><td>3</td><td>87</td><td>106</td><td>56.74707</td><td>2.521808</td><td>50.87719</td><td>4.472824</td><td>0.000000</td><td> 0.00000</td><td>-33.75316</td><td>2.131628e-14</td><td>97.297</td><td>Clostridium sp. CAG:169;Clostridium sp. CAG:242;Negativibacillus massiliensis                                                                                                                                             </td><td>3</td><td>24</td><td>50S ribosomal protein L36</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>809</td><td>106</td><td>106</td><td>0</td><td>809r</td><td>PRIMER_RIGHT</td><td>GWCKCTGTTTRTGCTTBGG   </td><td>19</td><td>24</td><td>3</td><td>87</td><td>106</td><td>56.74707</td><td>2.521808</td><td>50.87719</td><td>4.472824</td><td>0.000000</td><td> 0.00000</td><td>-33.75316</td><td>2.131628e-14</td><td>97.297</td><td>Ruminococcus flavefaciens;Anaerofustis stercorihominis;Oscillospiraceae bacterium;Clostridiales bacterium                                                                                                                 </td><td>3</td><td>24</td><td>50S ribosomal protein L36</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>809</td><td>106</td><td>106</td><td>0</td><td>809r</td><td>PRIMER_RIGHT</td><td>GWCKCTGTTTRTGCTTBGG   </td><td>19</td><td>24</td><td>3</td><td>87</td><td>106</td><td>56.74707</td><td>2.521808</td><td>50.87719</td><td>4.472824</td><td>0.000000</td><td> 0.00000</td><td>-33.75316</td><td>2.131628e-14</td><td>97.297</td><td>Eubacteriales;Candidatus Soleaferrea massiliensis AP7;Anaerotruncus massiliensis;Anaerotruncus rubiinfantis;Anaerotruncus sp. AF02-27;Anaerotruncus sp. 22A2-44;Oscillospiraceae bacterium;Clostridiaceae bacterium NSJ-31</td><td>3</td><td>24</td><td>50S ribosomal protein L36</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td>809</td><td>106</td><td>106</td><td>0</td><td>809r</td><td>PRIMER_RIGHT</td><td>GWCKCTGTTTRTGCTTBGG   </td><td>19</td><td>24</td><td>3</td><td>87</td><td>106</td><td>56.74707</td><td>2.521808</td><td>50.87719</td><td>4.472824</td><td>0.000000</td><td> 0.00000</td><td>-33.75316</td><td>2.131628e-14</td><td>97.297</td><td>Acetanaerobacterium elongatum                                                                                                                                                                                             </td><td>3</td><td>24</td><td>50S ribosomal protein L36</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td> 73</td><td>109</td><td>109</td><td>0</td><td>73f </td><td>PRIMER_LEFT </td><td>TGAAAGTHAGASCTTCYGTDAA</td><td>22</td><td>36</td><td>3</td><td> 1</td><td> 23</td><td>56.23671</td><td>1.915052</td><td>37.12121</td><td>3.787879</td><td>7.511429</td><td>16.79607</td><td>-27.56715</td><td>2.442498e+01</td><td>97.297</td><td>Oscillospiraceae bacterium NSJ-54                                                                                                                                                                                         </td><td>3</td><td>36</td><td>50S ribosomal protein L36</td></tr>\n",
       "\t<tr><td>cds</td><td>2</td><td> 73</td><td>109</td><td>109</td><td>0</td><td>73f </td><td>PRIMER_LEFT </td><td>TGAAAGTHAGASCTTCYGTDAA</td><td>22</td><td>36</td><td>3</td><td> 1</td><td> 23</td><td>56.23671</td><td>1.915052</td><td>37.12121</td><td>3.787879</td><td>7.511429</td><td>16.79607</td><td>-27.56715</td><td>2.442498e+01</td><td>97.297</td><td>Oscillospiraceae;Hydrogenoanaerobacterium saccharovorans;Anaerotruncus sp. G3(2012);Oscillospiraceae bacterium;Marasmitruncus massiliensis;Anaerotruncus sp. X29;Anaerotruncus sp. 1XD22-93;Anaerotruncus sp. 1XD42-93    </td><td>3</td><td>36</td><td>50S ribosomal protein L36</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 30 × 27\n",
       "\\begin{tabular}{lllllllllllllllllllllllllll}\n",
       " gene\\_type & cluster\\_id & primer\\_set & amplicon\\_size\\_consensus & amplicon\\_size\\_avg & amplicon\\_size\\_sd & primer\\_id & primer\\_type & sequence & length & degeneracy & degeneracy\\_3prime & position\\_start & position\\_end & Tm\\_avg & Tm\\_sd & GC\\_avg & GC\\_sd & hairpin\\_avg & hairpin\\_sd & homodimer\\_avg & homodimer\\_sd & pident & sscinames & max\\_degeneracy\\_3prime & max\\_degeneracy & annotation\\\\\n",
       " <chr> & <int> & <int> & <int> & <dbl> & <dbl> & <fct> & <fct> & <fct> & <int> & <int> & <int> & <int> & <int> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <fct> & <int> & <int> & <fct>\\\\\n",
       "\\hline\n",
       "\t cds & 2 & 546 & 103 & 103 & 0 & 546f & PRIMER\\_LEFT  & ATGAAAGTHAGASCTTCYGT   & 20 & 12 & 2 &  0 &  20 & 54.37386 & 1.763376 & 39.16667 & 3.435921 & 7.511429 & 16.79607 & -27.56715 & 2.442498e+01 & 97.297 & Oscillospiraceae bacterium NSJ-54                                                                                                                                                                                          & 2 & 24 & 50S ribosomal protein L36\\\\\n",
       "\t cds & 2 & 546 & 103 & 103 & 0 & 546f & PRIMER\\_LEFT  & ATGAAAGTHAGASCTTCYGT   & 20 & 12 & 2 &  0 &  20 & 54.37386 & 1.763376 & 39.16667 & 3.435921 & 7.511429 & 16.79607 & -27.56715 & 2.442498e+01 & 97.297 & Oscillospiraceae;Hydrogenoanaerobacterium saccharovorans;Anaerotruncus sp. G3(2012);Oscillospiraceae bacterium;Marasmitruncus massiliensis;Anaerotruncus sp. X29;Anaerotruncus sp. 1XD22-93;Anaerotruncus sp. 1XD42-93     & 2 & 24 & 50S ribosomal protein L36\\\\\n",
       "\t cds & 2 & 546 & 103 & 103 & 0 & 546f & PRIMER\\_LEFT  & ATGAAAGTHAGASCTTCYGT   & 20 & 12 & 2 &  0 &  20 & 54.37386 & 1.763376 & 39.16667 & 3.435921 & 7.511429 & 16.79607 & -27.56715 & 2.442498e+01 & 97.297 & Oscillospiraceae bacterium                                                                                                                                                                                                 & 2 & 24 & 50S ribosomal protein L36\\\\\n",
       "\t cds & 2 & 546 & 103 & 103 & 0 & 546f & PRIMER\\_LEFT  & ATGAAAGTHAGASCTTCYGT   & 20 & 12 & 2 &  0 &  20 & 54.37386 & 1.763376 & 39.16667 & 3.435921 & 7.511429 & 16.79607 & -27.56715 & 2.442498e+01 & 97.297 & Clostridium sp. CAG:169;Clostridium sp. CAG:242;Negativibacillus massiliensis                                                                                                                                              & 2 & 24 & 50S ribosomal protein L36\\\\\n",
       "\t cds & 2 & 546 & 103 & 103 & 0 & 546f & PRIMER\\_LEFT  & ATGAAAGTHAGASCTTCYGT   & 20 & 12 & 2 &  0 &  20 & 54.37386 & 1.763376 & 39.16667 & 3.435921 & 7.511429 & 16.79607 & -27.56715 & 2.442498e+01 & 97.297 & Ruminococcus flavefaciens;Anaerofustis stercorihominis;Oscillospiraceae bacterium;Clostridiales bacterium                                                                                                                  & 2 & 24 & 50S ribosomal protein L36\\\\\n",
       "\t cds & 2 & 546 & 103 & 103 & 0 & 546f & PRIMER\\_LEFT  & ATGAAAGTHAGASCTTCYGT   & 20 & 12 & 2 &  0 &  20 & 54.37386 & 1.763376 & 39.16667 & 3.435921 & 7.511429 & 16.79607 & -27.56715 & 2.442498e+01 & 97.297 & Eubacteriales;Candidatus Soleaferrea massiliensis AP7;Anaerotruncus massiliensis;Anaerotruncus rubiinfantis;Anaerotruncus sp. AF02-27;Anaerotruncus sp. 22A2-44;Oscillospiraceae bacterium;Clostridiaceae bacterium NSJ-31 & 2 & 24 & 50S ribosomal protein L36\\\\\n",
       "\t cds & 2 & 546 & 103 & 103 & 0 & 546f & PRIMER\\_LEFT  & ATGAAAGTHAGASCTTCYGT   & 20 & 12 & 2 &  0 &  20 & 54.37386 & 1.763376 & 39.16667 & 3.435921 & 7.511429 & 16.79607 & -27.56715 & 2.442498e+01 & 97.297 & Acetanaerobacterium elongatum                                                                                                                                                                                              & 2 & 24 & 50S ribosomal protein L36\\\\\n",
       "\t cds & 2 & 546 & 103 & 103 & 0 & 546r & PRIMER\\_RIGHT & KCTGTTTRTGCTTBGGRTTT   & 20 & 24 & 2 & 83 & 103 & 56.37333 & 2.279991 & 40.83333 & 4.930066 & 0.000000 &  0.00000 & -33.75316 & 2.131628e-14 & 97.297 & Oscillospiraceae bacterium NSJ-54                                                                                                                                                                                          & 2 & 24 & 50S ribosomal protein L36\\\\\n",
       "\t cds & 2 & 546 & 103 & 103 & 0 & 546r & PRIMER\\_RIGHT & KCTGTTTRTGCTTBGGRTTT   & 20 & 24 & 2 & 83 & 103 & 56.37333 & 2.279991 & 40.83333 & 4.930066 & 0.000000 &  0.00000 & -33.75316 & 2.131628e-14 & 97.297 & Oscillospiraceae;Hydrogenoanaerobacterium saccharovorans;Anaerotruncus sp. G3(2012);Oscillospiraceae bacterium;Marasmitruncus massiliensis;Anaerotruncus sp. X29;Anaerotruncus sp. 1XD22-93;Anaerotruncus sp. 1XD42-93     & 2 & 24 & 50S ribosomal protein L36\\\\\n",
       "\t cds & 2 & 546 & 103 & 103 & 0 & 546r & PRIMER\\_RIGHT & KCTGTTTRTGCTTBGGRTTT   & 20 & 24 & 2 & 83 & 103 & 56.37333 & 2.279991 & 40.83333 & 4.930066 & 0.000000 &  0.00000 & -33.75316 & 2.131628e-14 & 97.297 & Oscillospiraceae bacterium                                                                                                                                                                                                 & 2 & 24 & 50S ribosomal protein L36\\\\\n",
       "\t cds & 2 & 546 & 103 & 103 & 0 & 546r & PRIMER\\_RIGHT & KCTGTTTRTGCTTBGGRTTT   & 20 & 24 & 2 & 83 & 103 & 56.37333 & 2.279991 & 40.83333 & 4.930066 & 0.000000 &  0.00000 & -33.75316 & 2.131628e-14 & 97.297 & Clostridium sp. CAG:169;Clostridium sp. CAG:242;Negativibacillus massiliensis                                                                                                                                              & 2 & 24 & 50S ribosomal protein L36\\\\\n",
       "\t cds & 2 & 546 & 103 & 103 & 0 & 546r & PRIMER\\_RIGHT & KCTGTTTRTGCTTBGGRTTT   & 20 & 24 & 2 & 83 & 103 & 56.37333 & 2.279991 & 40.83333 & 4.930066 & 0.000000 &  0.00000 & -33.75316 & 2.131628e-14 & 97.297 & Ruminococcus flavefaciens;Anaerofustis stercorihominis;Oscillospiraceae bacterium;Clostridiales bacterium                                                                                                                  & 2 & 24 & 50S ribosomal protein L36\\\\\n",
       "\t cds & 2 & 546 & 103 & 103 & 0 & 546r & PRIMER\\_RIGHT & KCTGTTTRTGCTTBGGRTTT   & 20 & 24 & 2 & 83 & 103 & 56.37333 & 2.279991 & 40.83333 & 4.930066 & 0.000000 &  0.00000 & -33.75316 & 2.131628e-14 & 97.297 & Eubacteriales;Candidatus Soleaferrea massiliensis AP7;Anaerotruncus massiliensis;Anaerotruncus rubiinfantis;Anaerotruncus sp. AF02-27;Anaerotruncus sp. 22A2-44;Oscillospiraceae bacterium;Clostridiaceae bacterium NSJ-31 & 2 & 24 & 50S ribosomal protein L36\\\\\n",
       "\t cds & 2 & 546 & 103 & 103 & 0 & 546r & PRIMER\\_RIGHT & KCTGTTTRTGCTTBGGRTTT   & 20 & 24 & 2 & 83 & 103 & 56.37333 & 2.279991 & 40.83333 & 4.930066 & 0.000000 &  0.00000 & -33.75316 & 2.131628e-14 & 97.297 & Acetanaerobacterium elongatum                                                                                                                                                                                              & 2 & 24 & 50S ribosomal protein L36\\\\\n",
       "\t cds & 2 & 809 & 106 & 106 & 0 & 809f & PRIMER\\_LEFT  & ATGAAAGTHAGASCTTCYGT   & 20 & 12 & 2 &  0 &  20 & 54.37386 & 1.763376 & 39.16667 & 3.435921 & 7.511429 & 16.79607 & -27.56715 & 2.442498e+01 & 97.297 & Oscillospiraceae bacterium NSJ-54                                                                                                                                                                                          & 3 & 24 & 50S ribosomal protein L36\\\\\n",
       "\t cds & 2 & 809 & 106 & 106 & 0 & 809f & PRIMER\\_LEFT  & ATGAAAGTHAGASCTTCYGT   & 20 & 12 & 2 &  0 &  20 & 54.37386 & 1.763376 & 39.16667 & 3.435921 & 7.511429 & 16.79607 & -27.56715 & 2.442498e+01 & 97.297 & Oscillospiraceae;Hydrogenoanaerobacterium saccharovorans;Anaerotruncus sp. G3(2012);Oscillospiraceae bacterium;Marasmitruncus massiliensis;Anaerotruncus sp. X29;Anaerotruncus sp. 1XD22-93;Anaerotruncus sp. 1XD42-93     & 3 & 24 & 50S ribosomal protein L36\\\\\n",
       "\t cds & 2 & 809 & 106 & 106 & 0 & 809f & PRIMER\\_LEFT  & ATGAAAGTHAGASCTTCYGT   & 20 & 12 & 2 &  0 &  20 & 54.37386 & 1.763376 & 39.16667 & 3.435921 & 7.511429 & 16.79607 & -27.56715 & 2.442498e+01 & 97.297 & Oscillospiraceae bacterium                                                                                                                                                                                                 & 3 & 24 & 50S ribosomal protein L36\\\\\n",
       "\t cds & 2 & 809 & 106 & 106 & 0 & 809f & PRIMER\\_LEFT  & ATGAAAGTHAGASCTTCYGT   & 20 & 12 & 2 &  0 &  20 & 54.37386 & 1.763376 & 39.16667 & 3.435921 & 7.511429 & 16.79607 & -27.56715 & 2.442498e+01 & 97.297 & Clostridium sp. CAG:169;Clostridium sp. CAG:242;Negativibacillus massiliensis                                                                                                                                              & 3 & 24 & 50S ribosomal protein L36\\\\\n",
       "\t cds & 2 & 809 & 106 & 106 & 0 & 809f & PRIMER\\_LEFT  & ATGAAAGTHAGASCTTCYGT   & 20 & 12 & 2 &  0 &  20 & 54.37386 & 1.763376 & 39.16667 & 3.435921 & 7.511429 & 16.79607 & -27.56715 & 2.442498e+01 & 97.297 & Ruminococcus flavefaciens;Anaerofustis stercorihominis;Oscillospiraceae bacterium;Clostridiales bacterium                                                                                                                  & 3 & 24 & 50S ribosomal protein L36\\\\\n",
       "\t cds & 2 & 809 & 106 & 106 & 0 & 809f & PRIMER\\_LEFT  & ATGAAAGTHAGASCTTCYGT   & 20 & 12 & 2 &  0 &  20 & 54.37386 & 1.763376 & 39.16667 & 3.435921 & 7.511429 & 16.79607 & -27.56715 & 2.442498e+01 & 97.297 & Eubacteriales;Candidatus Soleaferrea massiliensis AP7;Anaerotruncus massiliensis;Anaerotruncus rubiinfantis;Anaerotruncus sp. AF02-27;Anaerotruncus sp. 22A2-44;Oscillospiraceae bacterium;Clostridiaceae bacterium NSJ-31 & 3 & 24 & 50S ribosomal protein L36\\\\\n",
       "\t cds & 2 & 809 & 106 & 106 & 0 & 809f & PRIMER\\_LEFT  & ATGAAAGTHAGASCTTCYGT   & 20 & 12 & 2 &  0 &  20 & 54.37386 & 1.763376 & 39.16667 & 3.435921 & 7.511429 & 16.79607 & -27.56715 & 2.442498e+01 & 97.297 & Acetanaerobacterium elongatum                                                                                                                                                                                              & 3 & 24 & 50S ribosomal protein L36\\\\\n",
       "\t cds & 2 & 809 & 106 & 106 & 0 & 809r & PRIMER\\_RIGHT & GWCKCTGTTTRTGCTTBGG    & 19 & 24 & 3 & 87 & 106 & 56.74707 & 2.521808 & 50.87719 & 4.472824 & 0.000000 &  0.00000 & -33.75316 & 2.131628e-14 & 97.297 & Oscillospiraceae bacterium NSJ-54                                                                                                                                                                                          & 3 & 24 & 50S ribosomal protein L36\\\\\n",
       "\t cds & 2 & 809 & 106 & 106 & 0 & 809r & PRIMER\\_RIGHT & GWCKCTGTTTRTGCTTBGG    & 19 & 24 & 3 & 87 & 106 & 56.74707 & 2.521808 & 50.87719 & 4.472824 & 0.000000 &  0.00000 & -33.75316 & 2.131628e-14 & 97.297 & Oscillospiraceae;Hydrogenoanaerobacterium saccharovorans;Anaerotruncus sp. G3(2012);Oscillospiraceae bacterium;Marasmitruncus massiliensis;Anaerotruncus sp. X29;Anaerotruncus sp. 1XD22-93;Anaerotruncus sp. 1XD42-93     & 3 & 24 & 50S ribosomal protein L36\\\\\n",
       "\t cds & 2 & 809 & 106 & 106 & 0 & 809r & PRIMER\\_RIGHT & GWCKCTGTTTRTGCTTBGG    & 19 & 24 & 3 & 87 & 106 & 56.74707 & 2.521808 & 50.87719 & 4.472824 & 0.000000 &  0.00000 & -33.75316 & 2.131628e-14 & 97.297 & Oscillospiraceae bacterium                                                                                                                                                                                                 & 3 & 24 & 50S ribosomal protein L36\\\\\n",
       "\t cds & 2 & 809 & 106 & 106 & 0 & 809r & PRIMER\\_RIGHT & GWCKCTGTTTRTGCTTBGG    & 19 & 24 & 3 & 87 & 106 & 56.74707 & 2.521808 & 50.87719 & 4.472824 & 0.000000 &  0.00000 & -33.75316 & 2.131628e-14 & 97.297 & Clostridium sp. CAG:169;Clostridium sp. CAG:242;Negativibacillus massiliensis                                                                                                                                              & 3 & 24 & 50S ribosomal protein L36\\\\\n",
       "\t cds & 2 & 809 & 106 & 106 & 0 & 809r & PRIMER\\_RIGHT & GWCKCTGTTTRTGCTTBGG    & 19 & 24 & 3 & 87 & 106 & 56.74707 & 2.521808 & 50.87719 & 4.472824 & 0.000000 &  0.00000 & -33.75316 & 2.131628e-14 & 97.297 & Ruminococcus flavefaciens;Anaerofustis stercorihominis;Oscillospiraceae bacterium;Clostridiales bacterium                                                                                                                  & 3 & 24 & 50S ribosomal protein L36\\\\\n",
       "\t cds & 2 & 809 & 106 & 106 & 0 & 809r & PRIMER\\_RIGHT & GWCKCTGTTTRTGCTTBGG    & 19 & 24 & 3 & 87 & 106 & 56.74707 & 2.521808 & 50.87719 & 4.472824 & 0.000000 &  0.00000 & -33.75316 & 2.131628e-14 & 97.297 & Eubacteriales;Candidatus Soleaferrea massiliensis AP7;Anaerotruncus massiliensis;Anaerotruncus rubiinfantis;Anaerotruncus sp. AF02-27;Anaerotruncus sp. 22A2-44;Oscillospiraceae bacterium;Clostridiaceae bacterium NSJ-31 & 3 & 24 & 50S ribosomal protein L36\\\\\n",
       "\t cds & 2 & 809 & 106 & 106 & 0 & 809r & PRIMER\\_RIGHT & GWCKCTGTTTRTGCTTBGG    & 19 & 24 & 3 & 87 & 106 & 56.74707 & 2.521808 & 50.87719 & 4.472824 & 0.000000 &  0.00000 & -33.75316 & 2.131628e-14 & 97.297 & Acetanaerobacterium elongatum                                                                                                                                                                                              & 3 & 24 & 50S ribosomal protein L36\\\\\n",
       "\t cds & 2 &  73 & 109 & 109 & 0 & 73f  & PRIMER\\_LEFT  & TGAAAGTHAGASCTTCYGTDAA & 22 & 36 & 3 &  1 &  23 & 56.23671 & 1.915052 & 37.12121 & 3.787879 & 7.511429 & 16.79607 & -27.56715 & 2.442498e+01 & 97.297 & Oscillospiraceae bacterium NSJ-54                                                                                                                                                                                          & 3 & 36 & 50S ribosomal protein L36\\\\\n",
       "\t cds & 2 &  73 & 109 & 109 & 0 & 73f  & PRIMER\\_LEFT  & TGAAAGTHAGASCTTCYGTDAA & 22 & 36 & 3 &  1 &  23 & 56.23671 & 1.915052 & 37.12121 & 3.787879 & 7.511429 & 16.79607 & -27.56715 & 2.442498e+01 & 97.297 & Oscillospiraceae;Hydrogenoanaerobacterium saccharovorans;Anaerotruncus sp. G3(2012);Oscillospiraceae bacterium;Marasmitruncus massiliensis;Anaerotruncus sp. X29;Anaerotruncus sp. 1XD22-93;Anaerotruncus sp. 1XD42-93     & 3 & 36 & 50S ribosomal protein L36\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 30 × 27\n",
       "\n",
       "| gene_type &lt;chr&gt; | cluster_id &lt;int&gt; | primer_set &lt;int&gt; | amplicon_size_consensus &lt;int&gt; | amplicon_size_avg &lt;dbl&gt; | amplicon_size_sd &lt;dbl&gt; | primer_id &lt;fct&gt; | primer_type &lt;fct&gt; | sequence &lt;fct&gt; | length &lt;int&gt; | degeneracy &lt;int&gt; | degeneracy_3prime &lt;int&gt; | position_start &lt;int&gt; | position_end &lt;int&gt; | Tm_avg &lt;dbl&gt; | Tm_sd &lt;dbl&gt; | GC_avg &lt;dbl&gt; | GC_sd &lt;dbl&gt; | hairpin_avg &lt;dbl&gt; | hairpin_sd &lt;dbl&gt; | homodimer_avg &lt;dbl&gt; | homodimer_sd &lt;dbl&gt; | pident &lt;dbl&gt; | sscinames &lt;fct&gt; | max_degeneracy_3prime &lt;int&gt; | max_degeneracy &lt;int&gt; | annotation &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| cds | 2 | 546 | 103 | 103 | 0 | 546f | PRIMER_LEFT  | ATGAAAGTHAGASCTTCYGT   | 20 | 12 | 2 |  0 |  20 | 54.37386 | 1.763376 | 39.16667 | 3.435921 | 7.511429 | 16.79607 | -27.56715 | 2.442498e+01 | 97.297 | Oscillospiraceae bacterium NSJ-54                                                                                                                                                                                          | 2 | 24 | 50S ribosomal protein L36 |\n",
       "| cds | 2 | 546 | 103 | 103 | 0 | 546f | PRIMER_LEFT  | ATGAAAGTHAGASCTTCYGT   | 20 | 12 | 2 |  0 |  20 | 54.37386 | 1.763376 | 39.16667 | 3.435921 | 7.511429 | 16.79607 | -27.56715 | 2.442498e+01 | 97.297 | Oscillospiraceae;Hydrogenoanaerobacterium saccharovorans;Anaerotruncus sp. G3(2012);Oscillospiraceae bacterium;Marasmitruncus massiliensis;Anaerotruncus sp. X29;Anaerotruncus sp. 1XD22-93;Anaerotruncus sp. 1XD42-93     | 2 | 24 | 50S ribosomal protein L36 |\n",
       "| cds | 2 | 546 | 103 | 103 | 0 | 546f | PRIMER_LEFT  | ATGAAAGTHAGASCTTCYGT   | 20 | 12 | 2 |  0 |  20 | 54.37386 | 1.763376 | 39.16667 | 3.435921 | 7.511429 | 16.79607 | -27.56715 | 2.442498e+01 | 97.297 | Oscillospiraceae bacterium                                                                                                                                                                                                 | 2 | 24 | 50S ribosomal protein L36 |\n",
       "| cds | 2 | 546 | 103 | 103 | 0 | 546f | PRIMER_LEFT  | ATGAAAGTHAGASCTTCYGT   | 20 | 12 | 2 |  0 |  20 | 54.37386 | 1.763376 | 39.16667 | 3.435921 | 7.511429 | 16.79607 | -27.56715 | 2.442498e+01 | 97.297 | Clostridium sp. CAG:169;Clostridium sp. CAG:242;Negativibacillus massiliensis                                                                                                                                              | 2 | 24 | 50S ribosomal protein L36 |\n",
       "| cds | 2 | 546 | 103 | 103 | 0 | 546f | PRIMER_LEFT  | ATGAAAGTHAGASCTTCYGT   | 20 | 12 | 2 |  0 |  20 | 54.37386 | 1.763376 | 39.16667 | 3.435921 | 7.511429 | 16.79607 | -27.56715 | 2.442498e+01 | 97.297 | Ruminococcus flavefaciens;Anaerofustis stercorihominis;Oscillospiraceae bacterium;Clostridiales bacterium                                                                                                                  | 2 | 24 | 50S ribosomal protein L36 |\n",
       "| cds | 2 | 546 | 103 | 103 | 0 | 546f | PRIMER_LEFT  | ATGAAAGTHAGASCTTCYGT   | 20 | 12 | 2 |  0 |  20 | 54.37386 | 1.763376 | 39.16667 | 3.435921 | 7.511429 | 16.79607 | -27.56715 | 2.442498e+01 | 97.297 | Eubacteriales;Candidatus Soleaferrea massiliensis AP7;Anaerotruncus massiliensis;Anaerotruncus rubiinfantis;Anaerotruncus sp. AF02-27;Anaerotruncus sp. 22A2-44;Oscillospiraceae bacterium;Clostridiaceae bacterium NSJ-31 | 2 | 24 | 50S ribosomal protein L36 |\n",
       "| cds | 2 | 546 | 103 | 103 | 0 | 546f | PRIMER_LEFT  | ATGAAAGTHAGASCTTCYGT   | 20 | 12 | 2 |  0 |  20 | 54.37386 | 1.763376 | 39.16667 | 3.435921 | 7.511429 | 16.79607 | -27.56715 | 2.442498e+01 | 97.297 | Acetanaerobacterium elongatum                                                                                                                                                                                              | 2 | 24 | 50S ribosomal protein L36 |\n",
       "| cds | 2 | 546 | 103 | 103 | 0 | 546r | PRIMER_RIGHT | KCTGTTTRTGCTTBGGRTTT   | 20 | 24 | 2 | 83 | 103 | 56.37333 | 2.279991 | 40.83333 | 4.930066 | 0.000000 |  0.00000 | -33.75316 | 2.131628e-14 | 97.297 | Oscillospiraceae bacterium NSJ-54                                                                                                                                                                                          | 2 | 24 | 50S ribosomal protein L36 |\n",
       "| cds | 2 | 546 | 103 | 103 | 0 | 546r | PRIMER_RIGHT | KCTGTTTRTGCTTBGGRTTT   | 20 | 24 | 2 | 83 | 103 | 56.37333 | 2.279991 | 40.83333 | 4.930066 | 0.000000 |  0.00000 | -33.75316 | 2.131628e-14 | 97.297 | Oscillospiraceae;Hydrogenoanaerobacterium saccharovorans;Anaerotruncus sp. G3(2012);Oscillospiraceae bacterium;Marasmitruncus massiliensis;Anaerotruncus sp. X29;Anaerotruncus sp. 1XD22-93;Anaerotruncus sp. 1XD42-93     | 2 | 24 | 50S ribosomal protein L36 |\n",
       "| cds | 2 | 546 | 103 | 103 | 0 | 546r | PRIMER_RIGHT | KCTGTTTRTGCTTBGGRTTT   | 20 | 24 | 2 | 83 | 103 | 56.37333 | 2.279991 | 40.83333 | 4.930066 | 0.000000 |  0.00000 | -33.75316 | 2.131628e-14 | 97.297 | Oscillospiraceae bacterium                                                                                                                                                                                                 | 2 | 24 | 50S ribosomal protein L36 |\n",
       "| cds | 2 | 546 | 103 | 103 | 0 | 546r | PRIMER_RIGHT | KCTGTTTRTGCTTBGGRTTT   | 20 | 24 | 2 | 83 | 103 | 56.37333 | 2.279991 | 40.83333 | 4.930066 | 0.000000 |  0.00000 | -33.75316 | 2.131628e-14 | 97.297 | Clostridium sp. CAG:169;Clostridium sp. CAG:242;Negativibacillus massiliensis                                                                                                                                              | 2 | 24 | 50S ribosomal protein L36 |\n",
       "| cds | 2 | 546 | 103 | 103 | 0 | 546r | PRIMER_RIGHT | KCTGTTTRTGCTTBGGRTTT   | 20 | 24 | 2 | 83 | 103 | 56.37333 | 2.279991 | 40.83333 | 4.930066 | 0.000000 |  0.00000 | -33.75316 | 2.131628e-14 | 97.297 | Ruminococcus flavefaciens;Anaerofustis stercorihominis;Oscillospiraceae bacterium;Clostridiales bacterium                                                                                                                  | 2 | 24 | 50S ribosomal protein L36 |\n",
       "| cds | 2 | 546 | 103 | 103 | 0 | 546r | PRIMER_RIGHT | KCTGTTTRTGCTTBGGRTTT   | 20 | 24 | 2 | 83 | 103 | 56.37333 | 2.279991 | 40.83333 | 4.930066 | 0.000000 |  0.00000 | -33.75316 | 2.131628e-14 | 97.297 | Eubacteriales;Candidatus Soleaferrea massiliensis AP7;Anaerotruncus massiliensis;Anaerotruncus rubiinfantis;Anaerotruncus sp. AF02-27;Anaerotruncus sp. 22A2-44;Oscillospiraceae bacterium;Clostridiaceae bacterium NSJ-31 | 2 | 24 | 50S ribosomal protein L36 |\n",
       "| cds | 2 | 546 | 103 | 103 | 0 | 546r | PRIMER_RIGHT | KCTGTTTRTGCTTBGGRTTT   | 20 | 24 | 2 | 83 | 103 | 56.37333 | 2.279991 | 40.83333 | 4.930066 | 0.000000 |  0.00000 | -33.75316 | 2.131628e-14 | 97.297 | Acetanaerobacterium elongatum                                                                                                                                                                                              | 2 | 24 | 50S ribosomal protein L36 |\n",
       "| cds | 2 | 809 | 106 | 106 | 0 | 809f | PRIMER_LEFT  | ATGAAAGTHAGASCTTCYGT   | 20 | 12 | 2 |  0 |  20 | 54.37386 | 1.763376 | 39.16667 | 3.435921 | 7.511429 | 16.79607 | -27.56715 | 2.442498e+01 | 97.297 | Oscillospiraceae bacterium NSJ-54                                                                                                                                                                                          | 3 | 24 | 50S ribosomal protein L36 |\n",
       "| cds | 2 | 809 | 106 | 106 | 0 | 809f | PRIMER_LEFT  | ATGAAAGTHAGASCTTCYGT   | 20 | 12 | 2 |  0 |  20 | 54.37386 | 1.763376 | 39.16667 | 3.435921 | 7.511429 | 16.79607 | -27.56715 | 2.442498e+01 | 97.297 | Oscillospiraceae;Hydrogenoanaerobacterium saccharovorans;Anaerotruncus sp. G3(2012);Oscillospiraceae bacterium;Marasmitruncus massiliensis;Anaerotruncus sp. X29;Anaerotruncus sp. 1XD22-93;Anaerotruncus sp. 1XD42-93     | 3 | 24 | 50S ribosomal protein L36 |\n",
       "| cds | 2 | 809 | 106 | 106 | 0 | 809f | PRIMER_LEFT  | ATGAAAGTHAGASCTTCYGT   | 20 | 12 | 2 |  0 |  20 | 54.37386 | 1.763376 | 39.16667 | 3.435921 | 7.511429 | 16.79607 | -27.56715 | 2.442498e+01 | 97.297 | Oscillospiraceae bacterium                                                                                                                                                                                                 | 3 | 24 | 50S ribosomal protein L36 |\n",
       "| cds | 2 | 809 | 106 | 106 | 0 | 809f | PRIMER_LEFT  | ATGAAAGTHAGASCTTCYGT   | 20 | 12 | 2 |  0 |  20 | 54.37386 | 1.763376 | 39.16667 | 3.435921 | 7.511429 | 16.79607 | -27.56715 | 2.442498e+01 | 97.297 | Clostridium sp. CAG:169;Clostridium sp. CAG:242;Negativibacillus massiliensis                                                                                                                                              | 3 | 24 | 50S ribosomal protein L36 |\n",
       "| cds | 2 | 809 | 106 | 106 | 0 | 809f | PRIMER_LEFT  | ATGAAAGTHAGASCTTCYGT   | 20 | 12 | 2 |  0 |  20 | 54.37386 | 1.763376 | 39.16667 | 3.435921 | 7.511429 | 16.79607 | -27.56715 | 2.442498e+01 | 97.297 | Ruminococcus flavefaciens;Anaerofustis stercorihominis;Oscillospiraceae bacterium;Clostridiales bacterium                                                                                                                  | 3 | 24 | 50S ribosomal protein L36 |\n",
       "| cds | 2 | 809 | 106 | 106 | 0 | 809f | PRIMER_LEFT  | ATGAAAGTHAGASCTTCYGT   | 20 | 12 | 2 |  0 |  20 | 54.37386 | 1.763376 | 39.16667 | 3.435921 | 7.511429 | 16.79607 | -27.56715 | 2.442498e+01 | 97.297 | Eubacteriales;Candidatus Soleaferrea massiliensis AP7;Anaerotruncus massiliensis;Anaerotruncus rubiinfantis;Anaerotruncus sp. AF02-27;Anaerotruncus sp. 22A2-44;Oscillospiraceae bacterium;Clostridiaceae bacterium NSJ-31 | 3 | 24 | 50S ribosomal protein L36 |\n",
       "| cds | 2 | 809 | 106 | 106 | 0 | 809f | PRIMER_LEFT  | ATGAAAGTHAGASCTTCYGT   | 20 | 12 | 2 |  0 |  20 | 54.37386 | 1.763376 | 39.16667 | 3.435921 | 7.511429 | 16.79607 | -27.56715 | 2.442498e+01 | 97.297 | Acetanaerobacterium elongatum                                                                                                                                                                                              | 3 | 24 | 50S ribosomal protein L36 |\n",
       "| cds | 2 | 809 | 106 | 106 | 0 | 809r | PRIMER_RIGHT | GWCKCTGTTTRTGCTTBGG    | 19 | 24 | 3 | 87 | 106 | 56.74707 | 2.521808 | 50.87719 | 4.472824 | 0.000000 |  0.00000 | -33.75316 | 2.131628e-14 | 97.297 | Oscillospiraceae bacterium NSJ-54                                                                                                                                                                                          | 3 | 24 | 50S ribosomal protein L36 |\n",
       "| cds | 2 | 809 | 106 | 106 | 0 | 809r | PRIMER_RIGHT | GWCKCTGTTTRTGCTTBGG    | 19 | 24 | 3 | 87 | 106 | 56.74707 | 2.521808 | 50.87719 | 4.472824 | 0.000000 |  0.00000 | -33.75316 | 2.131628e-14 | 97.297 | Oscillospiraceae;Hydrogenoanaerobacterium saccharovorans;Anaerotruncus sp. G3(2012);Oscillospiraceae bacterium;Marasmitruncus massiliensis;Anaerotruncus sp. X29;Anaerotruncus sp. 1XD22-93;Anaerotruncus sp. 1XD42-93     | 3 | 24 | 50S ribosomal protein L36 |\n",
       "| cds | 2 | 809 | 106 | 106 | 0 | 809r | PRIMER_RIGHT | GWCKCTGTTTRTGCTTBGG    | 19 | 24 | 3 | 87 | 106 | 56.74707 | 2.521808 | 50.87719 | 4.472824 | 0.000000 |  0.00000 | -33.75316 | 2.131628e-14 | 97.297 | Oscillospiraceae bacterium                                                                                                                                                                                                 | 3 | 24 | 50S ribosomal protein L36 |\n",
       "| cds | 2 | 809 | 106 | 106 | 0 | 809r | PRIMER_RIGHT | GWCKCTGTTTRTGCTTBGG    | 19 | 24 | 3 | 87 | 106 | 56.74707 | 2.521808 | 50.87719 | 4.472824 | 0.000000 |  0.00000 | -33.75316 | 2.131628e-14 | 97.297 | Clostridium sp. CAG:169;Clostridium sp. CAG:242;Negativibacillus massiliensis                                                                                                                                              | 3 | 24 | 50S ribosomal protein L36 |\n",
       "| cds | 2 | 809 | 106 | 106 | 0 | 809r | PRIMER_RIGHT | GWCKCTGTTTRTGCTTBGG    | 19 | 24 | 3 | 87 | 106 | 56.74707 | 2.521808 | 50.87719 | 4.472824 | 0.000000 |  0.00000 | -33.75316 | 2.131628e-14 | 97.297 | Ruminococcus flavefaciens;Anaerofustis stercorihominis;Oscillospiraceae bacterium;Clostridiales bacterium                                                                                                                  | 3 | 24 | 50S ribosomal protein L36 |\n",
       "| cds | 2 | 809 | 106 | 106 | 0 | 809r | PRIMER_RIGHT | GWCKCTGTTTRTGCTTBGG    | 19 | 24 | 3 | 87 | 106 | 56.74707 | 2.521808 | 50.87719 | 4.472824 | 0.000000 |  0.00000 | -33.75316 | 2.131628e-14 | 97.297 | Eubacteriales;Candidatus Soleaferrea massiliensis AP7;Anaerotruncus massiliensis;Anaerotruncus rubiinfantis;Anaerotruncus sp. AF02-27;Anaerotruncus sp. 22A2-44;Oscillospiraceae bacterium;Clostridiaceae bacterium NSJ-31 | 3 | 24 | 50S ribosomal protein L36 |\n",
       "| cds | 2 | 809 | 106 | 106 | 0 | 809r | PRIMER_RIGHT | GWCKCTGTTTRTGCTTBGG    | 19 | 24 | 3 | 87 | 106 | 56.74707 | 2.521808 | 50.87719 | 4.472824 | 0.000000 |  0.00000 | -33.75316 | 2.131628e-14 | 97.297 | Acetanaerobacterium elongatum                                                                                                                                                                                              | 3 | 24 | 50S ribosomal protein L36 |\n",
       "| cds | 2 |  73 | 109 | 109 | 0 | 73f  | PRIMER_LEFT  | TGAAAGTHAGASCTTCYGTDAA | 22 | 36 | 3 |  1 |  23 | 56.23671 | 1.915052 | 37.12121 | 3.787879 | 7.511429 | 16.79607 | -27.56715 | 2.442498e+01 | 97.297 | Oscillospiraceae bacterium NSJ-54                                                                                                                                                                                          | 3 | 36 | 50S ribosomal protein L36 |\n",
       "| cds | 2 |  73 | 109 | 109 | 0 | 73f  | PRIMER_LEFT  | TGAAAGTHAGASCTTCYGTDAA | 22 | 36 | 3 |  1 |  23 | 56.23671 | 1.915052 | 37.12121 | 3.787879 | 7.511429 | 16.79607 | -27.56715 | 2.442498e+01 | 97.297 | Oscillospiraceae;Hydrogenoanaerobacterium saccharovorans;Anaerotruncus sp. G3(2012);Oscillospiraceae bacterium;Marasmitruncus massiliensis;Anaerotruncus sp. X29;Anaerotruncus sp. 1XD22-93;Anaerotruncus sp. 1XD42-93     | 3 | 36 | 50S ribosomal protein L36 |\n",
       "\n"
      ],
      "text/plain": [
       "   gene_type cluster_id primer_set amplicon_size_consensus amplicon_size_avg\n",
       "1  cds       2          546        103                     103              \n",
       "2  cds       2          546        103                     103              \n",
       "3  cds       2          546        103                     103              \n",
       "4  cds       2          546        103                     103              \n",
       "5  cds       2          546        103                     103              \n",
       "6  cds       2          546        103                     103              \n",
       "7  cds       2          546        103                     103              \n",
       "8  cds       2          546        103                     103              \n",
       "9  cds       2          546        103                     103              \n",
       "10 cds       2          546        103                     103              \n",
       "11 cds       2          546        103                     103              \n",
       "12 cds       2          546        103                     103              \n",
       "13 cds       2          546        103                     103              \n",
       "14 cds       2          546        103                     103              \n",
       "15 cds       2          809        106                     106              \n",
       "16 cds       2          809        106                     106              \n",
       "17 cds       2          809        106                     106              \n",
       "18 cds       2          809        106                     106              \n",
       "19 cds       2          809        106                     106              \n",
       "20 cds       2          809        106                     106              \n",
       "21 cds       2          809        106                     106              \n",
       "22 cds       2          809        106                     106              \n",
       "23 cds       2          809        106                     106              \n",
       "24 cds       2          809        106                     106              \n",
       "25 cds       2          809        106                     106              \n",
       "26 cds       2          809        106                     106              \n",
       "27 cds       2          809        106                     106              \n",
       "28 cds       2          809        106                     106              \n",
       "29 cds       2           73        109                     109              \n",
       "30 cds       2           73        109                     109              \n",
       "   amplicon_size_sd primer_id primer_type  sequence               length\n",
       "1  0                546f      PRIMER_LEFT  ATGAAAGTHAGASCTTCYGT   20    \n",
       "2  0                546f      PRIMER_LEFT  ATGAAAGTHAGASCTTCYGT   20    \n",
       "3  0                546f      PRIMER_LEFT  ATGAAAGTHAGASCTTCYGT   20    \n",
       "4  0                546f      PRIMER_LEFT  ATGAAAGTHAGASCTTCYGT   20    \n",
       "5  0                546f      PRIMER_LEFT  ATGAAAGTHAGASCTTCYGT   20    \n",
       "6  0                546f      PRIMER_LEFT  ATGAAAGTHAGASCTTCYGT   20    \n",
       "7  0                546f      PRIMER_LEFT  ATGAAAGTHAGASCTTCYGT   20    \n",
       "8  0                546r      PRIMER_RIGHT KCTGTTTRTGCTTBGGRTTT   20    \n",
       "9  0                546r      PRIMER_RIGHT KCTGTTTRTGCTTBGGRTTT   20    \n",
       "10 0                546r      PRIMER_RIGHT KCTGTTTRTGCTTBGGRTTT   20    \n",
       "11 0                546r      PRIMER_RIGHT KCTGTTTRTGCTTBGGRTTT   20    \n",
       "12 0                546r      PRIMER_RIGHT KCTGTTTRTGCTTBGGRTTT   20    \n",
       "13 0                546r      PRIMER_RIGHT KCTGTTTRTGCTTBGGRTTT   20    \n",
       "14 0                546r      PRIMER_RIGHT KCTGTTTRTGCTTBGGRTTT   20    \n",
       "15 0                809f      PRIMER_LEFT  ATGAAAGTHAGASCTTCYGT   20    \n",
       "16 0                809f      PRIMER_LEFT  ATGAAAGTHAGASCTTCYGT   20    \n",
       "17 0                809f      PRIMER_LEFT  ATGAAAGTHAGASCTTCYGT   20    \n",
       "18 0                809f      PRIMER_LEFT  ATGAAAGTHAGASCTTCYGT   20    \n",
       "19 0                809f      PRIMER_LEFT  ATGAAAGTHAGASCTTCYGT   20    \n",
       "20 0                809f      PRIMER_LEFT  ATGAAAGTHAGASCTTCYGT   20    \n",
       "21 0                809f      PRIMER_LEFT  ATGAAAGTHAGASCTTCYGT   20    \n",
       "22 0                809r      PRIMER_RIGHT GWCKCTGTTTRTGCTTBGG    19    \n",
       "23 0                809r      PRIMER_RIGHT GWCKCTGTTTRTGCTTBGG    19    \n",
       "24 0                809r      PRIMER_RIGHT GWCKCTGTTTRTGCTTBGG    19    \n",
       "25 0                809r      PRIMER_RIGHT GWCKCTGTTTRTGCTTBGG    19    \n",
       "26 0                809r      PRIMER_RIGHT GWCKCTGTTTRTGCTTBGG    19    \n",
       "27 0                809r      PRIMER_RIGHT GWCKCTGTTTRTGCTTBGG    19    \n",
       "28 0                809r      PRIMER_RIGHT GWCKCTGTTTRTGCTTBGG    19    \n",
       "29 0                73f       PRIMER_LEFT  TGAAAGTHAGASCTTCYGTDAA 22    \n",
       "30 0                73f       PRIMER_LEFT  TGAAAGTHAGASCTTCYGTDAA 22    \n",
       "   degeneracy degeneracy_3prime position_start position_end Tm_avg   Tm_sd   \n",
       "1  12         2                  0              20          54.37386 1.763376\n",
       "2  12         2                  0              20          54.37386 1.763376\n",
       "3  12         2                  0              20          54.37386 1.763376\n",
       "4  12         2                  0              20          54.37386 1.763376\n",
       "5  12         2                  0              20          54.37386 1.763376\n",
       "6  12         2                  0              20          54.37386 1.763376\n",
       "7  12         2                  0              20          54.37386 1.763376\n",
       "8  24         2                 83             103          56.37333 2.279991\n",
       "9  24         2                 83             103          56.37333 2.279991\n",
       "10 24         2                 83             103          56.37333 2.279991\n",
       "11 24         2                 83             103          56.37333 2.279991\n",
       "12 24         2                 83             103          56.37333 2.279991\n",
       "13 24         2                 83             103          56.37333 2.279991\n",
       "14 24         2                 83             103          56.37333 2.279991\n",
       "15 12         2                  0              20          54.37386 1.763376\n",
       "16 12         2                  0              20          54.37386 1.763376\n",
       "17 12         2                  0              20          54.37386 1.763376\n",
       "18 12         2                  0              20          54.37386 1.763376\n",
       "19 12         2                  0              20          54.37386 1.763376\n",
       "20 12         2                  0              20          54.37386 1.763376\n",
       "21 12         2                  0              20          54.37386 1.763376\n",
       "22 24         3                 87             106          56.74707 2.521808\n",
       "23 24         3                 87             106          56.74707 2.521808\n",
       "24 24         3                 87             106          56.74707 2.521808\n",
       "25 24         3                 87             106          56.74707 2.521808\n",
       "26 24         3                 87             106          56.74707 2.521808\n",
       "27 24         3                 87             106          56.74707 2.521808\n",
       "28 24         3                 87             106          56.74707 2.521808\n",
       "29 36         3                  1              23          56.23671 1.915052\n",
       "30 36         3                  1              23          56.23671 1.915052\n",
       "   GC_avg   GC_sd    hairpin_avg hairpin_sd homodimer_avg homodimer_sd pident\n",
       "1  39.16667 3.435921 7.511429    16.79607   -27.56715     2.442498e+01 97.297\n",
       "2  39.16667 3.435921 7.511429    16.79607   -27.56715     2.442498e+01 97.297\n",
       "3  39.16667 3.435921 7.511429    16.79607   -27.56715     2.442498e+01 97.297\n",
       "4  39.16667 3.435921 7.511429    16.79607   -27.56715     2.442498e+01 97.297\n",
       "5  39.16667 3.435921 7.511429    16.79607   -27.56715     2.442498e+01 97.297\n",
       "6  39.16667 3.435921 7.511429    16.79607   -27.56715     2.442498e+01 97.297\n",
       "7  39.16667 3.435921 7.511429    16.79607   -27.56715     2.442498e+01 97.297\n",
       "8  40.83333 4.930066 0.000000     0.00000   -33.75316     2.131628e-14 97.297\n",
       "9  40.83333 4.930066 0.000000     0.00000   -33.75316     2.131628e-14 97.297\n",
       "10 40.83333 4.930066 0.000000     0.00000   -33.75316     2.131628e-14 97.297\n",
       "11 40.83333 4.930066 0.000000     0.00000   -33.75316     2.131628e-14 97.297\n",
       "12 40.83333 4.930066 0.000000     0.00000   -33.75316     2.131628e-14 97.297\n",
       "13 40.83333 4.930066 0.000000     0.00000   -33.75316     2.131628e-14 97.297\n",
       "14 40.83333 4.930066 0.000000     0.00000   -33.75316     2.131628e-14 97.297\n",
       "15 39.16667 3.435921 7.511429    16.79607   -27.56715     2.442498e+01 97.297\n",
       "16 39.16667 3.435921 7.511429    16.79607   -27.56715     2.442498e+01 97.297\n",
       "17 39.16667 3.435921 7.511429    16.79607   -27.56715     2.442498e+01 97.297\n",
       "18 39.16667 3.435921 7.511429    16.79607   -27.56715     2.442498e+01 97.297\n",
       "19 39.16667 3.435921 7.511429    16.79607   -27.56715     2.442498e+01 97.297\n",
       "20 39.16667 3.435921 7.511429    16.79607   -27.56715     2.442498e+01 97.297\n",
       "21 39.16667 3.435921 7.511429    16.79607   -27.56715     2.442498e+01 97.297\n",
       "22 50.87719 4.472824 0.000000     0.00000   -33.75316     2.131628e-14 97.297\n",
       "23 50.87719 4.472824 0.000000     0.00000   -33.75316     2.131628e-14 97.297\n",
       "24 50.87719 4.472824 0.000000     0.00000   -33.75316     2.131628e-14 97.297\n",
       "25 50.87719 4.472824 0.000000     0.00000   -33.75316     2.131628e-14 97.297\n",
       "26 50.87719 4.472824 0.000000     0.00000   -33.75316     2.131628e-14 97.297\n",
       "27 50.87719 4.472824 0.000000     0.00000   -33.75316     2.131628e-14 97.297\n",
       "28 50.87719 4.472824 0.000000     0.00000   -33.75316     2.131628e-14 97.297\n",
       "29 37.12121 3.787879 7.511429    16.79607   -27.56715     2.442498e+01 97.297\n",
       "30 37.12121 3.787879 7.511429    16.79607   -27.56715     2.442498e+01 97.297\n",
       "   sscinames                                                                                                                                                                                                                 \n",
       "1  Oscillospiraceae bacterium NSJ-54                                                                                                                                                                                         \n",
       "2  Oscillospiraceae;Hydrogenoanaerobacterium saccharovorans;Anaerotruncus sp. G3(2012);Oscillospiraceae bacterium;Marasmitruncus massiliensis;Anaerotruncus sp. X29;Anaerotruncus sp. 1XD22-93;Anaerotruncus sp. 1XD42-93    \n",
       "3  Oscillospiraceae bacterium                                                                                                                                                                                                \n",
       "4  Clostridium sp. CAG:169;Clostridium sp. CAG:242;Negativibacillus massiliensis                                                                                                                                             \n",
       "5  Ruminococcus flavefaciens;Anaerofustis stercorihominis;Oscillospiraceae bacterium;Clostridiales bacterium                                                                                                                 \n",
       "6  Eubacteriales;Candidatus Soleaferrea massiliensis AP7;Anaerotruncus massiliensis;Anaerotruncus rubiinfantis;Anaerotruncus sp. AF02-27;Anaerotruncus sp. 22A2-44;Oscillospiraceae bacterium;Clostridiaceae bacterium NSJ-31\n",
       "7  Acetanaerobacterium elongatum                                                                                                                                                                                             \n",
       "8  Oscillospiraceae bacterium NSJ-54                                                                                                                                                                                         \n",
       "9  Oscillospiraceae;Hydrogenoanaerobacterium saccharovorans;Anaerotruncus sp. G3(2012);Oscillospiraceae bacterium;Marasmitruncus massiliensis;Anaerotruncus sp. X29;Anaerotruncus sp. 1XD22-93;Anaerotruncus sp. 1XD42-93    \n",
       "10 Oscillospiraceae bacterium                                                                                                                                                                                                \n",
       "11 Clostridium sp. CAG:169;Clostridium sp. CAG:242;Negativibacillus massiliensis                                                                                                                                             \n",
       "12 Ruminococcus flavefaciens;Anaerofustis stercorihominis;Oscillospiraceae bacterium;Clostridiales bacterium                                                                                                                 \n",
       "13 Eubacteriales;Candidatus Soleaferrea massiliensis AP7;Anaerotruncus massiliensis;Anaerotruncus rubiinfantis;Anaerotruncus sp. AF02-27;Anaerotruncus sp. 22A2-44;Oscillospiraceae bacterium;Clostridiaceae bacterium NSJ-31\n",
       "14 Acetanaerobacterium elongatum                                                                                                                                                                                             \n",
       "15 Oscillospiraceae bacterium NSJ-54                                                                                                                                                                                         \n",
       "16 Oscillospiraceae;Hydrogenoanaerobacterium saccharovorans;Anaerotruncus sp. G3(2012);Oscillospiraceae bacterium;Marasmitruncus massiliensis;Anaerotruncus sp. X29;Anaerotruncus sp. 1XD22-93;Anaerotruncus sp. 1XD42-93    \n",
       "17 Oscillospiraceae bacterium                                                                                                                                                                                                \n",
       "18 Clostridium sp. CAG:169;Clostridium sp. CAG:242;Negativibacillus massiliensis                                                                                                                                             \n",
       "19 Ruminococcus flavefaciens;Anaerofustis stercorihominis;Oscillospiraceae bacterium;Clostridiales bacterium                                                                                                                 \n",
       "20 Eubacteriales;Candidatus Soleaferrea massiliensis AP7;Anaerotruncus massiliensis;Anaerotruncus rubiinfantis;Anaerotruncus sp. AF02-27;Anaerotruncus sp. 22A2-44;Oscillospiraceae bacterium;Clostridiaceae bacterium NSJ-31\n",
       "21 Acetanaerobacterium elongatum                                                                                                                                                                                             \n",
       "22 Oscillospiraceae bacterium NSJ-54                                                                                                                                                                                         \n",
       "23 Oscillospiraceae;Hydrogenoanaerobacterium saccharovorans;Anaerotruncus sp. G3(2012);Oscillospiraceae bacterium;Marasmitruncus massiliensis;Anaerotruncus sp. X29;Anaerotruncus sp. 1XD22-93;Anaerotruncus sp. 1XD42-93    \n",
       "24 Oscillospiraceae bacterium                                                                                                                                                                                                \n",
       "25 Clostridium sp. CAG:169;Clostridium sp. CAG:242;Negativibacillus massiliensis                                                                                                                                             \n",
       "26 Ruminococcus flavefaciens;Anaerofustis stercorihominis;Oscillospiraceae bacterium;Clostridiales bacterium                                                                                                                 \n",
       "27 Eubacteriales;Candidatus Soleaferrea massiliensis AP7;Anaerotruncus massiliensis;Anaerotruncus rubiinfantis;Anaerotruncus sp. AF02-27;Anaerotruncus sp. 22A2-44;Oscillospiraceae bacterium;Clostridiaceae bacterium NSJ-31\n",
       "28 Acetanaerobacterium elongatum                                                                                                                                                                                             \n",
       "29 Oscillospiraceae bacterium NSJ-54                                                                                                                                                                                         \n",
       "30 Oscillospiraceae;Hydrogenoanaerobacterium saccharovorans;Anaerotruncus sp. G3(2012);Oscillospiraceae bacterium;Marasmitruncus massiliensis;Anaerotruncus sp. X29;Anaerotruncus sp. 1XD22-93;Anaerotruncus sp. 1XD42-93    \n",
       "   max_degeneracy_3prime max_degeneracy annotation               \n",
       "1  2                     24             50S ribosomal protein L36\n",
       "2  2                     24             50S ribosomal protein L36\n",
       "3  2                     24             50S ribosomal protein L36\n",
       "4  2                     24             50S ribosomal protein L36\n",
       "5  2                     24             50S ribosomal protein L36\n",
       "6  2                     24             50S ribosomal protein L36\n",
       "7  2                     24             50S ribosomal protein L36\n",
       "8  2                     24             50S ribosomal protein L36\n",
       "9  2                     24             50S ribosomal protein L36\n",
       "10 2                     24             50S ribosomal protein L36\n",
       "11 2                     24             50S ribosomal protein L36\n",
       "12 2                     24             50S ribosomal protein L36\n",
       "13 2                     24             50S ribosomal protein L36\n",
       "14 2                     24             50S ribosomal protein L36\n",
       "15 3                     24             50S ribosomal protein L36\n",
       "16 3                     24             50S ribosomal protein L36\n",
       "17 3                     24             50S ribosomal protein L36\n",
       "18 3                     24             50S ribosomal protein L36\n",
       "19 3                     24             50S ribosomal protein L36\n",
       "20 3                     24             50S ribosomal protein L36\n",
       "21 3                     24             50S ribosomal protein L36\n",
       "22 3                     24             50S ribosomal protein L36\n",
       "23 3                     24             50S ribosomal protein L36\n",
       "24 3                     24             50S ribosomal protein L36\n",
       "25 3                     24             50S ribosomal protein L36\n",
       "26 3                     24             50S ribosomal protein L36\n",
       "27 3                     24             50S ribosomal protein L36\n",
       "28 3                     24             50S ribosomal protein L36\n",
       "29 3                     36             50S ribosomal protein L36\n",
       "30 3                     36             50S ribosomal protein L36"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# joining\n",
    "primer_info_jfj = primer_info_jf %>%\n",
    "    left_join(gene_annot %>% distinct(gene_type, cluster_id, annotation),\n",
    "              c('gene_type', 'cluster_id'))\n",
    "\n",
    "df.dims(30,50)\n",
    "primer_info_jfj\n",
    "df.dims()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Writing table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-24T13:19:40.588065Z",
     "iopub.status.busy": "2021-04-24T13:19:40.585799Z",
     "iopub.status.idle": "2021-04-24T13:19:40.612939Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "File written: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3//Anaerotruncus/best_primers.tsv \n"
     ]
    }
   ],
   "source": [
    "F = file.path(work_dir, 'best_primers.tsv')\n",
    "write_table(primer_info_jfj, F)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# sessionInfo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-24T13:19:59.295847Z",
     "iopub.status.busy": "2021-04-24T13:19:59.293939Z",
     "iopub.status.idle": "2021-04-24T13:19:59.458665Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LLG\n",
      "===\n",
      "\n",
      "Ley Lab Genome analysis pipeline (LLG)\n",
      "\n",
      "* Version: 0.2.4\n",
      "* Authors:\n",
      "  * Nick Youngblut <nyoungb2@gmail.com>\n",
      "* Maintainers:\n",
      "  * Nick Youngblut <nyoungb2@gmail.com>\n",
      "\n",
      "--- conda envs ---\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/gtdbtk.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- conda-forge::networkx\n",
      "- bioconda::gtdbtk\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/checkm.yaml <==\n",
      "channels:\n",
      "- bioconda\n",
      "dependencies:\n",
      "- python=2.7\n",
      "- pigz\n",
      "- bioconda::prodigal\n",
      "- bioconda::pplacer\n",
      "- bioconda::checkm-genome\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/quast.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- bioconda::seqkit\n",
      "- bioconda::quast>=5.0.0\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/sourmash.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- bioconda::sourmash=2.0.0a4\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/fastqc.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- bioconda::seqtk\n",
      "- bioconda::fastqc=0.11.7\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/multiqc.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- bioconda::multiqc=1.4\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/hadley.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- conda-forge::r-ape\n",
      "- conda-forge::r-dplyr\n",
      "- conda-forge::r-tidyr\n",
      "- conda-forge::r-ggplot2\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/bbmap.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::bbmap=37.78\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/ani.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::dashing\n",
      "- bioconda::fastani\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/prokka.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- bioconda::prokka\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/prodigal.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- bioconda::prodigal\n",
      "- pip:\n",
      "  - biopython-convert\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/eggnog-mapper.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::eggnog-mapper\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/phylophlan.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::phylophlan\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/drep.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::mash\n",
      "- bioconda::checkm-genome\n",
      "- bioconda::fastani\n",
      "- bioconda::drep\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/antismash.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- biopython=1.77\n",
      "- bioconda::blast=2.9\n",
      "- bioconda::antismash>=5.1.2\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/deepbgc.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::deepbgc\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/abricate.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- bioconda::abricate\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/rRNA.yaml <==\n",
      "channels:\n",
      "- qiime2/label/r2020.2\n",
      "- conda-forge\n",
      "- bioconda\n",
      "- defaults\n",
      "dependencies:\n",
      "- barrnap\n",
      "- _libgcc_mutex=0.1\n",
      "- _openmp_mutex=4.5\n",
      "- _r-mutex=1.0.1\n",
      "- alsa-lib=1.1.5\n",
      "- arb-bio-tools=6.0.6\n",
      "- attrs=19.3.0\n",
      "- backcall=0.1.0\n",
      "- bibtexparser=1.1.0\n",
      "- binutils_impl_linux-64=2.33.1\n",
      "- binutils_linux-64=2.33.1\n",
      "- bioconductor-biobase=2.42.0\n",
      "- bioconductor-biocgenerics=0.28.0\n",
      "- bioconductor-biocparallel=1.16.6\n",
      "- bioconductor-biostrings=2.50.2\n",
      "- bioconductor-dada2=1.10.0\n",
      "- bioconductor-delayedarray=0.8.0\n",
      "- bioconductor-genomeinfodb=1.18.1\n",
      "- bioconductor-genomeinfodbdata=1.2.1\n",
      "- bioconductor-genomicalignments=1.18.1\n",
      "- bioconductor-genomicranges=1.34.0\n",
      "- bioconductor-iranges=2.16.0\n",
      "- bioconductor-rsamtools=1.34.0\n",
      "- bioconductor-s4vectors=0.20.1\n",
      "- bioconductor-shortread=1.40.0\n",
      "- bioconductor-summarizedexperiment=1.12.0\n",
      "- bioconductor-xvector=0.22.0\n",
      "- bioconductor-zlibbioc=1.28.0\n",
      "- biom-format=2.1.8\n",
      "- blas=2.14\n",
      "- blast=2.9.0\n",
      "- bleach=3.1.1\n",
      "- bokeh=1.4.0\n",
      "- boost=1.68.0\n",
      "- boost-cpp=1.68.0\n",
      "- bwidget=1.9.14\n",
      "- bzip2=1.0.8\n",
      "- ca-certificates=2019.11.28\n",
      "- cachecontrol=0.12.5\n",
      "- cairo=1.16.0\n",
      "- certifi=2019.11.28\n",
      "- cffi=1.13.2\n",
      "- chardet=3.0.4\n",
      "- click=7.0\n",
      "- cryptography=2.8\n",
      "- curl=7.68.0\n",
      "- cutadapt=2.8\n",
      "- cycler=0.10.0\n",
      "- cython=0.29.15\n",
      "- dbus=1.13.6\n",
      "- deblur=1.1.0\n",
      "- decorator=4.4.1\n",
      "- defusedxml=0.6.0\n",
      "- dendropy=4.4.0\n",
      "- dnaio=0.4.1\n",
      "- emperor=1.0.0\n",
      "- entrypoints=0.3\n",
      "- expat=2.2.9\n",
      "- fastcluster=1.1.26\n",
      "- fasttree=2.1.10\n",
      "- fontconfig=2.13.1\n",
      "- freetype=2.10.0\n",
      "- future=0.18.2\n",
      "- gcc_impl_linux-64=7.3.0\n",
      "- gcc_linux-64=7.3.0\n",
      "- gettext=0.19.8.1\n",
      "- gfortran_impl_linux-64=7.3.0\n",
      "- gfortran_linux-64=7.3.0\n",
      "- giflib=5.2.1\n",
      "- glib=2.58.3\n",
      "- gmp=6.2.0\n",
      "- gneiss=0.4.6\n",
      "- gnutls=3.6.5\n",
      "- graphite2=1.3.13\n",
      "- gsl=2.5\n",
      "- gst-plugins-base=1.14.5\n",
      "- gstreamer=1.14.5\n",
      "- gxx_impl_linux-64=7.3.0\n",
      "- gxx_linux-64=7.3.0\n",
      "- h5py=2.10.0\n",
      "- harfbuzz=2.4.0\n",
      "- hdf5=1.10.5\n",
      "- hdmedians=0.13\n",
      "- hmmer=3.1b2\n",
      "- icu=58.2\n",
      "- idna=2.9\n",
      "- ijson=2.6.1\n",
      "- importlib_metadata=1.5.0\n",
      "- ipykernel=5.1.4\n",
      "- ipython=7.12.0\n",
      "- ipython_genutils=0.2.0\n",
      "- ipywidgets=7.5.1\n",
      "- iqtree=1.6.12\n",
      "- jedi=0.16.0\n",
      "- jinja2=2.11.1\n",
      "- joblib=0.14.1\n",
      "- jpeg=9c\n",
      "- jsonschema=3.2.0\n",
      "- jupyter_client=6.0.0\n",
      "- jupyter_core=4.6.3\n",
      "- kiwisolver=1.1.0\n",
      "- krb5=1.16.4\n",
      "- lcms2=2.9\n",
      "- ld_impl_linux-64=2.33.1\n",
      "- libarbdb=6.0.6\n",
      "- libblas=3.8.0\n",
      "- libcblas=3.8.0\n",
      "- libcurl=7.68.0\n",
      "- libedit=3.1.20170329\n",
      "- libffi=3.2.1\n",
      "- libgcc=7.2.0\n",
      "- libgcc-ng=9.2.0\n",
      "- libgfortran-ng=7.3.0\n",
      "- libgomp=9.2.0\n",
      "- libiconv=1.15\n",
      "- liblapack=3.8.0\n",
      "- liblapacke=3.8.0\n",
      "- libopenblas=0.3.7\n",
      "- libpng=1.6.37\n",
      "- libsodium=1.0.17\n",
      "- libssh2=1.8.2\n",
      "- libstdcxx-ng=9.2.0\n",
      "- libtiff=4.1.0\n",
      "- libuuid=2.32.1\n",
      "- libxcb=1.13\n",
      "- libxml2=2.9.9\n",
      "- lockfile=0.12.2\n",
      "- lz4-c=1.8.3\n",
      "- mafft=7.310\n",
      "- make=4.3\n",
      "- markupsafe=1.1.1\n",
      "- matplotlib=3.1.1\n",
      "- matplotlib-base=3.1.1\n",
      "- mistune=0.8.4\n",
      "- more-itertools=8.2.0\n",
      "- msgpack-python=1.0.0\n",
      "- natsort=7.0.1\n",
      "- nbconvert=5.6.1\n",
      "- nbformat=5.0.4\n",
      "- ncurses=6.1\n",
      "- nettle=3.4.1\n",
      "- networkx=2.4\n",
      "- nose=1.3.7\n",
      "- notebook=6.0.3\n",
      "- numpy=1.18.1\n",
      "- olefile=0.46\n",
      "- openjdk=11.0.1\n",
      "- openssl=1.1.1d\n",
      "- packaging=20.1\n",
      "- pandas=0.25.3\n",
      "- pandoc=2.9.2\n",
      "- pandocfilters=1.4.2\n",
      "- pango=1.40.14\n",
      "- parso=0.6.1\n",
      "- patsy=0.5.1\n",
      "- pcre=8.44\n",
      "- perl=5.26.2\n",
      "- perl-archive-tar=2.32\n",
      "- perl-carp=1.38\n",
      "- perl-common-sense=3.74\n",
      "- perl-compress-raw-bzip2=2.087\n",
      "- perl-compress-raw-zlib=2.087\n",
      "- perl-exporter=5.72\n",
      "- perl-exporter-tiny=1.002001\n",
      "- perl-extutils-makemaker=7.36\n",
      "- perl-io-compress=2.087\n",
      "- perl-io-zlib=1.10\n",
      "- perl-json=4.02\n",
      "- perl-json-xs=2.34\n",
      "- perl-list-moreutils=0.428\n",
      "- perl-list-moreutils-xs=0.428\n",
      "- perl-pathtools=3.75\n",
      "- perl-scalar-list-utils=1.52\n",
      "- perl-types-serialiser=1.0\n",
      "- perl-xsloader=0.24\n",
      "- pexpect=4.8.0\n",
      "- pickleshare=0.7.5\n",
      "- pigz=2.3.4\n",
      "- pillow=7.0.0\n",
      "- pip=20.0.2\n",
      "- pixman=0.38.0\n",
      "- pluggy=0.12.0\n",
      "- prometheus_client=0.7.1\n",
      "- prompt_toolkit=3.0.3\n",
      "- psutil=5.7.0\n",
      "- pthread-stubs=0.4\n",
      "- ptyprocess=0.6.0\n",
      "- py=1.8.1\n",
      "- pycparser=2.19\n",
      "- pygments=2.5.2\n",
      "- pyopenssl=19.1.0\n",
      "- pyparsing=2.4.6\n",
      "- pyqt=5.9.2\n",
      "- pyrsistent=0.15.7\n",
      "- pysocks=1.7.1\n",
      "- pytest=5.3.5\n",
      "- python=3.6.7\n",
      "- python-dateutil=2.8.1\n",
      "- pytz=2019.3\n",
      "- pyyaml=5.3\n",
      "- pyzmq=19.0.0\n",
      "- q2-alignment=2020.2.0\n",
      "- q2-composition=2020.2.0\n",
      "- q2-cutadapt=2020.2.0\n",
      "- q2-dada2=2020.2.0\n",
      "- q2-deblur=2020.2.0\n",
      "- q2-demux=2020.2.0\n",
      "- q2-diversity=2020.2.0\n",
      "- q2-emperor=2020.2.0\n",
      "- q2-feature-classifier=2020.2.0\n",
      "- q2-feature-table=2020.2.0\n",
      "- q2-fragment-insertion=2020.2.0\n",
      "- q2-gneiss=2020.2.0\n",
      "- q2-longitudinal=2020.2.0\n",
      "- q2-metadata=2020.2.0\n",
      "- q2-phylogeny=2020.2.0\n",
      "- q2-quality-control=2020.2.0\n",
      "- q2-quality-filter=2020.2.0\n",
      "- q2-sample-classifier=2020.2.0\n",
      "- q2-taxa=2020.2.0\n",
      "- q2-types=2020.2.0\n",
      "- q2-vsearch=2020.2.0\n",
      "- q2cli=2020.2.0\n",
      "- q2templates=2020.2.0\n",
      "- qiime2=2020.2.0\n",
      "- qt=5.9.7\n",
      "- r-assertthat=0.2.1\n",
      "- r-backports=1.1.5\n",
      "- r-base=3.5.1\n",
      "- r-bh=1.72.0_3\n",
      "- r-bitops=1.0_6\n",
      "- r-cli=2.0.2\n",
      "- r-cluster=2.1.0\n",
      "- r-colorspace=1.4_1\n",
      "- r-crayon=1.3.4\n",
      "- r-data.table=1.12.6\n",
      "- r-digest=0.6.25\n",
      "- r-ellipsis=0.3.0\n",
      "- r-fansi=0.4.1\n",
      "- r-farver=2.0.3\n",
      "- r-formatr=1.7\n",
      "- r-futile.logger=1.4.3\n",
      "- r-futile.options=1.0.1\n",
      "- r-ggplot2=3.2.1\n",
      "- r-glue=1.3.1\n",
      "- r-gtable=0.3.0\n",
      "- r-hwriter=1.3.2\n",
      "- r-labeling=0.3\n",
      "- r-lambda.r=1.2.4\n",
      "- r-lattice=0.20_40\n",
      "- r-latticeextra=0.6_28\n",
      "- r-lazyeval=0.2.2\n",
      "- r-lifecycle=0.1.0\n",
      "- r-magrittr=1.5\n",
      "- r-mass=7.3_51.5\n",
      "- r-matrix=1.2_18\n",
      "- r-matrixstats=0.55.0\n",
      "- r-mgcv=1.8_31\n",
      "- r-munsell=0.5.0\n",
      "- r-nlme=3.1_144\n",
      "- r-permute=0.9_5\n",
      "- r-pillar=1.4.3\n",
      "- r-pkgconfig=2.0.3\n",
      "- r-plyr=1.8.5\n",
      "- r-r6=2.4.1\n",
      "- r-rcolorbrewer=1.1_2\n",
      "- r-rcpp=1.0.3\n",
      "- r-rcppparallel=4.4.4\n",
      "- r-rcurl=1.98_1.1\n",
      "- r-reshape2=1.4.3\n",
      "- r-rlang=0.4.4\n",
      "- r-scales=1.1.0\n",
      "- r-snow=0.4_3\n",
      "- r-stringi=1.4.3\n",
      "- r-stringr=1.4.0\n",
      "- r-tibble=2.1.3\n",
      "- r-utf8=1.1.4\n",
      "- r-vctrs=0.2.3\n",
      "- r-vegan=2.5_6\n",
      "- r-viridislite=0.3.0\n",
      "- r-withr=2.1.2\n",
      "- r-zeallot=0.1.0\n",
      "- raxml=8.2.12\n",
      "- readline=8.0\n",
      "- requests=2.23.0\n",
      "- scikit-bio=0.5.5\n",
      "- scikit-learn=0.22.1\n",
      "- scipy=1.4.1\n",
      "- seaborn=0.10.0\n",
      "- send2trash=1.5.0\n",
      "- sepp=4.3.10\n",
      "- setuptools=45.2.0\n",
      "- sina=1.6.0\n",
      "- sip=4.19.8\n",
      "- six=1.14.0\n",
      "- sortmerna=2.0\n",
      "- sqlite=3.30.1\n",
      "- statsmodels=0.11.1\n",
      "- tbb=2019.9\n",
      "- terminado=0.8.3\n",
      "- testpath=0.4.4\n",
      "- tk=8.6.10\n",
      "- tktable=2.10\n",
      "- tornado=6.0.3\n",
      "- traitlets=4.3.3\n",
      "- tzlocal=2.0.0\n",
      "- unifrac=0.10.0\n",
      "- urllib3=1.25.7\n",
      "- vsearch=2.7.0\n",
      "- wcwidth=0.1.8\n",
      "- webencodings=0.5.1\n",
      "- wheel=0.34.2\n",
      "- widgetsnbextension=3.5.1\n",
      "- xopen=0.8.4\n",
      "- xorg-fixesproto=5.0\n",
      "- xorg-inputproto=2.3.2\n",
      "- xorg-kbproto=1.0.7\n",
      "- xorg-libice=1.0.10\n",
      "- xorg-libsm=1.2.3\n",
      "- xorg-libx11=1.6.9\n",
      "- xorg-libxau=1.0.9\n",
      "- xorg-libxdmcp=1.1.3\n",
      "- xorg-libxext=1.3.4\n",
      "- xorg-libxfixes=5.0.3\n",
      "- xorg-libxi=1.7.10\n",
      "- xorg-libxrender=0.9.10\n",
      "- xorg-libxtst=1.2.3\n",
      "- xorg-recordproto=1.14.2\n",
      "- xorg-renderproto=0.11.1\n",
      "- xorg-xextproto=7.3.0\n",
      "- xorg-xproto=7.0.31\n",
      "- xz=5.2.4\n",
      "- yaml=0.2.2\n",
      "- zeromq=4.3.2\n",
      "- zipp=3.0.0\n",
      "- zlib=1.2.11\n",
      "- zstd=1.4.4\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/traitar.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- scipy\n",
      "- pandas\n",
      "- numpy\n",
      "- matplotlib\n",
      "- pigz\n",
      "- bioconda::hmmer\n",
      "- pip:\n",
      "  - traitar3>=0.1.1\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/cctyper.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "- russel88\n",
      "dependencies:\n",
      "- pigz\n",
      "- cctyper\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/comparem.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::comparem\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/bigscape.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- python=3.6\n",
      "- hmmer\n",
      "- biopython\n",
      "- fasttree\n",
      "- numpy\n",
      "- scipy\n",
      "- networkx\n",
      "- scikit-learn=0.19.1\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/rgi.yaml <==\n",
      "name: rgi\n",
      "channels:\n",
      "  - bioconda\n",
      "  - conda-forge\n",
      "  - defaults\n",
      "dependencies:\n",
      "  - _libgcc_mutex=0.1=conda_forge\n",
      "  - _openmp_mutex=4.5=1_llvm\n",
      "  - bamtools=2.5.1=he860b03_5\n",
      "  - bedtools=2.29.2=hc088bd4_0\n",
      "  - biopython=1.72=py36h14c3975_1000\n",
      "  - blast=2.9.0=pl526h3066fca_4\n",
      "  - bowtie2=2.4.1=py36h7f0b59b_2\n",
      "  - bwa=0.7.17=hed695b0_7\n",
      "  - bzip2=1.0.8=h516909a_2\n",
      "  - ca-certificates=2020.4.5.2=hecda079_0\n",
      "  - certifi=2020.4.5.2=py36h9f0ad1d_0\n",
      "  - curl=7.69.1=h33f0ec9_0\n",
      "  - cycler=0.10.0=py_2\n",
      "  - diamond=0.8.36=h8b12597_4\n",
      "  - entrez-direct=13.3=pl526h375a9b1_0\n",
      "  - expat=2.2.9=he1b5a44_2\n",
      "  - filetype=1.0.7=pyh9f0ad1d_0\n",
      "  - freetype=2.10.2=he06d7ca_0\n",
      "  - icu=64.2=he1b5a44_1\n",
      "  - kiwisolver=1.2.0=py36hdb11119_0\n",
      "  - krb5=1.17.1=h2fd8d38_0\n",
      "  - ld_impl_linux-64=2.34=h53a641e_5\n",
      "  - libblas=3.8.0=16_openblas\n",
      "  - libcblas=3.8.0=16_openblas\n",
      "  - libcurl=7.69.1=hf7181ac_0\n",
      "  - libdeflate=1.0=h14c3975_1\n",
      "  - libedit=3.1.20191231=h46ee950_0\n",
      "  - libffi=3.2.1=he1b5a44_1007\n",
      "  - libgcc-ng=9.2.0=h24d8f2e_2\n",
      "  - libgfortran-ng=7.5.0\n",
      "  - liblapack=3.8.0=16_openblas\n",
      "  - libopenblas=0.3.9=h5ec1e0e_0\n",
      "  - libpng=1.6.37=hed695b0_1\n",
      "  - libssh2=1.9.0=hab1572f_2\n",
      "  - libstdcxx-ng=9.2.0=hdf63c60_2\n",
      "  - llvm-openmp=10.0.0=hc9558a2_0\n",
      "  - matplotlib-base=3.2.1=py36hb8e4980_0\n",
      "  - ncurses=6.1=hf484d3e_1002\n",
      "  - numpy=1.18.5=py36h7314795_0\n",
      "  - oligoarrayaux=3.8=hc9558a2_0\n",
      "  - openssl=1.1.1g=h516909a_0\n",
      "  - pandas=1.0.4=py36h830a2c2_0\n",
      "  - patsy=0.5.1=py_0\n",
      "  - pcre=8.44=he1b5a44_0\n",
      "  - perl=5.26.2=h516909a_1006\n",
      "  - perl-app-cpanminus=1.7044=pl526_1\n",
      "  - perl-archive-tar=2.32=pl526_0\n",
      "  - perl-base=2.23=pl526_1\n",
      "  - perl-business-isbn=3.004=pl526_0\n",
      "  - perl-business-isbn-data=20140910.003=pl526_0\n",
      "  - perl-carp=1.38=pl526_3\n",
      "  - perl-common-sense=3.74=pl526_2\n",
      "  - perl-compress-raw-bzip2=2.087=pl526he1b5a44_0\n",
      "  - perl-compress-raw-zlib=2.087=pl526hc9558a2_0\n",
      "  - perl-constant=1.33=pl526_1\n",
      "  - perl-data-dumper=2.173=pl526_0\n",
      "  - perl-digest-hmac=1.03=pl526_3\n",
      "  - perl-digest-md5=2.55=pl526_0\n",
      "  - perl-encode=2.88=pl526_1\n",
      "  - perl-encode-locale=1.05=pl526_6\n",
      "  - perl-exporter=5.72=pl526_1\n",
      "  - perl-exporter-tiny=1.002001=pl526_0\n",
      "  - perl-extutils-makemaker=7.36=pl526_1\n",
      "  - perl-file-listing=6.04=pl526_1\n",
      "  - perl-file-path=2.16=pl526_0\n",
      "  - perl-file-temp=0.2304=pl526_2\n",
      "  - perl-html-parser=3.72=pl526h6bb024c_5\n",
      "  - perl-html-tagset=3.20=pl526_3\n",
      "  - perl-html-tree=5.07=pl526_1\n",
      "  - perl-http-cookies=6.04=pl526_0\n",
      "  - perl-http-daemon=6.01=pl526_1\n",
      "  - perl-http-date=6.02=pl526_3\n",
      "  - perl-http-message=6.18=pl526_0\n",
      "  - perl-http-negotiate=6.01=pl526_3\n",
      "  - perl-io-compress=2.087=pl526he1b5a44_0\n",
      "  - perl-io-html=1.001=pl526_2\n",
      "  - perl-io-socket-ssl=2.066=pl526_0\n",
      "  - perl-io-zlib=1.10=pl526_2\n",
      "  - perl-json=4.02=pl526_0\n",
      "  - perl-json-xs=2.34=pl526h6bb024c_3\n",
      "  - perl-libwww-perl=6.39=pl526_0\n",
      "  - perl-list-moreutils=0.428=pl526_1\n",
      "  - perl-list-moreutils-xs=0.428=pl526_0\n",
      "  - perl-lwp-mediatypes=6.04=pl526_0\n",
      "  - perl-lwp-protocol-https=6.07=pl526_4\n",
      "  - perl-mime-base64=3.15=pl526_1\n",
      "  - perl-mozilla-ca=20180117=pl526_1\n",
      "  - perl-net-http=6.19=pl526_0\n",
      "  - perl-net-ssleay=1.88=pl526h90d6eec_0\n",
      "  - perl-ntlm=1.09=pl526_4\n",
      "  - perl-parent=0.236=pl526_1\n",
      "  - perl-pathtools=3.75=pl526h14c3975_1\n",
      "  - perl-scalar-list-utils=1.52=pl526h516909a_0\n",
      "  - perl-socket=2.027=pl526_1\n",
      "  - perl-storable=3.15=pl526h14c3975_0\n",
      "  - perl-test-requiresinternet=0.05=pl526_0\n",
      "  - perl-time-local=1.28=pl526_1\n",
      "  - perl-try-tiny=0.30=pl526_1\n",
      "  - perl-types-serialiser=1.0=pl526_2\n",
      "  - perl-uri=1.76=pl526_0\n",
      "  - perl-www-robotrules=6.02=pl526_3\n",
      "  - perl-xml-namespacesupport=1.12=pl526_0\n",
      "  - perl-xml-parser=2.44=pl526h4e0c4b3_7\n",
      "  - perl-xml-sax=1.02=pl526_0\n",
      "  - perl-xml-sax-base=1.09=pl526_0\n",
      "  - perl-xml-sax-expat=0.51=pl526_3\n",
      "  - perl-xml-simple=2.25=pl526_1\n",
      "  - perl-xsloader=0.24=pl526_0\n",
      "  - pip=20.1.1=py_1\n",
      "  - prodigal=2.6.3=h516909a_2\n",
      "  - pyahocorasick=1.4.0=py36h8c4c3a4_1\n",
      "  - pyfaidx=0.5.8=py_1\n",
      "  - pyparsing=2.4.7=pyh9f0ad1d_0\n",
      "  - python=3.6.10=h8356626_1011_cpython\n",
      "  - python-dateutil=2.8.1=py_0\n",
      "  - python_abi=3.6=1_cp36m\n",
      "  - pytz=2020.1=pyh9f0ad1d_0\n",
      "  - readline=8.0=hf8c457e_0\n",
      "  - rgi=5.1.0=py_1\n",
      "  - samtools=1.9=h8571acd_11\n",
      "  - scipy=1.4.1=py36h2d22cac_3\n",
      "  - seaborn=0.10.1=py_0\n",
      "  - setuptools=47.1.1=py36h9f0ad1d_0\n",
      "  - six=1.15.0=pyh9f0ad1d_0\n",
      "  - sqlite=3.30.1=hcee41ef_0\n",
      "  - statsmodels=0.11.1=py36h8c4c3a4_1\n",
      "  - tbb=2020.1=hc9558a2_0\n",
      "  - tk=8.6.10=hed695b0_0\n",
      "  - tornado=6.0.4=py36h8c4c3a4_1\n",
      "  - wheel=0.34.2=py_1\n",
      "  - xz=5.2.5=h516909a_0\n",
      "  - zlib=1.2.11=h516909a_1006\n",
      "prefix: /home/users/ldenies/apps/miniconda3/envs/rgi\n",
      "\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/pathofact_hmm.yaml <==\n",
      "channels:\n",
      "  - bioconda\n",
      "  - conda-forge\n",
      "  - defaults\n",
      "dependencies:\n",
      "  - _libgcc_mutex=0.1=conda_forge\n",
      "  - _openmp_mutex=4.5=0_gnu\n",
      "  - hmmer=3.2.1=he1b5a44_2\n",
      "  - libgcc-ng=9.2.0=h24d8f2e_2\n",
      "  - libgomp=9.2.0=h24d8f2e_2\n",
      "  - libstdcxx-ng=9.2.0=hdf63c60_2\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/pathofact_biopython.yaml <==\n",
      "channels:\n",
      "  - conda-forge\n",
      "  - bioconda\n",
      "  - defaults\n",
      "dependencies:\n",
      "  - _libgcc_mutex=0.1=main\n",
      "  - biopython=1.74=py36h516909a_0\n",
      "  - ca-certificates=2019.6.16=hecc5488_0\n",
      "  - certifi=2019.6.16=py36_1\n",
      "  - cycler=0.10.0=py_1\n",
      "  - dbus=1.13.6=he372182_0\n",
      "  - expat=2.2.5=he1b5a44_1003\n",
      "  - fontconfig=2.13.1=he4413a7_1000\n",
      "  - freetype=2.10.0=he983fc9_0\n",
      "  - gettext=0.19.8.1=hc5be6a0_1002\n",
      "  - glib=2.58.3=h6f030ca_1002\n",
      "  - gst-plugins-base=1.14.5=h0935bb2_0\n",
      "  - gstreamer=1.14.5=h36ae1b5_0\n",
      "  - icu=58.2=hf484d3e_1000\n",
      "  - joblib=0.13.2=py_0\n",
      "  - jpeg=9c=h14c3975_1001\n",
      "  - kiwisolver=1.1.0=py36hc9558a2_0\n",
      "  - libblas=3.8.0=11_openblas\n",
      "  - libcblas=3.8.0=11_openblas\n",
      "  - libedit=3.1.20181209=hc058e9b_0\n",
      "  - libffi=3.2.1=hd88cf55_4\n",
      "  - libgcc-ng=9.1.0=hdf63c60_0\n",
      "  - libgfortran-ng=7.3.0=hdf63c60_0\n",
      "  - libiconv=1.15=h516909a_1005\n",
      "  - liblapack=3.8.0=11_openblas\n",
      "  - libopenblas=0.3.6=h6e990d7_6\n",
      "  - libpng=1.6.37=hed695b0_0\n",
      "  - libstdcxx-ng=9.1.0=hdf63c60_0\n",
      "  - libuuid=2.32.1=h14c3975_1000\n",
      "  - libxcb=1.13=h14c3975_1002\n",
      "  - libxml2=2.9.9=h13577e0_2\n",
      "  - matplotlib=3.1.0=py36h5429711_0\n",
      "  - ncurses=6.1=he6710b0_1\n",
      "  - numpy=1.17.0=py36h95a1406_0\n",
      "  - openssl=1.1.1c=h516909a_0\n",
      "  - pandas=0.25.0=py36hb3f55d8_0\n",
      "  - patsy=0.5.1=py_0\n",
      "  - pcre=8.41=hf484d3e_1003\n",
      "  - pip=19.1.1=py36_0\n",
      "  - pthread-stubs=0.4=h14c3975_1001\n",
      "  - pyparsing=2.4.2=py_0\n",
      "  - pyqt=5.9.2=py36hcca6a23_2\n",
      "  - python=3.6.9=h265db76_0\n",
      "  - python-dateutil=2.8.0=py_0\n",
      "  - pytz=2019.2=py_0\n",
      "  - qt=5.9.7=h52cfd70_2\n",
      "  - readline=7.0=h7b6447c_5\n",
      "  - scikit-learn=0.21.3=py36hcdab131_0\n",
      "  - scipy=1.3.0=py36h921218d_1\n",
      "  - seaborn=0.9.0=py_0\n",
      "  - setuptools=41.0.1=py36_0\n",
      "  - sip=4.19.8=py36hf484d3e_1000\n",
      "  - six=1.12.0=py36_1000\n",
      "  - sqlite=3.29.0=h7b6447c_0\n",
      "  - statsmodels=0.10.1=py36hc1659b7_0\n",
      "  - tk=8.6.8=hbc83047_0\n",
      "  - tornado=6.0.3=py36h516909a_0\n",
      "  - wheel=0.33.4=py36_0\n",
      "  - xorg-libxau=1.0.9=h14c3975_0\n",
      "  - xorg-libxdmcp=1.1.3=h516909a_0\n",
      "  - xz=5.2.4=h14c3975_4\n",
      "  - zlib=1.2.11=h7b6447c_3\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/platon.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::platon\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/plasflow.yaml <==\n",
      "channels:\n",
      "  # for PlasFlow\n",
      "  - smaegol\n",
      "  # rest\n",
      "  - bioconda\n",
      "  - conda-forge\n",
      "  - defaults\n",
      "dependencies:\n",
      "  - seqkit\n",
      "  - _libgcc_mutex=0.1=conda_forge\n",
      "  - _openmp_mutex=4.5=1_llvm\n",
      "  - _r-mutex=1.0.1=anacondar_1\n",
      "  - backports=1.0=py_2\n",
      "  - backports.weakref=1.0rc1=py35_1\n",
      "  - bioconductor-biocgenerics=0.22.0=r3.3.2_0\n",
      "  - bioconductor-biostrings=2.42.1=r3.3.2_0\n",
      "  - bioconductor-iranges=2.8.2=r3.3.2_0\n",
      "  - bioconductor-s4vectors=0.12.2=r3.3.2_0\n",
      "  - bioconductor-xvector=0.14.1=r3.3.2_0\n",
      "  - bioconductor-zlibbioc=1.20.0=r3.3.2_1\n",
      "  - biopython=1.68=py35_0\n",
      "  - blas=2.16=openblas\n",
      "  - bleach=1.5.0=py35_0\n",
      "  - bzip2=1.0.8=h516909a_2\n",
      "  - ca-certificates=2020.4.5.1=hecc5488_0\n",
      "  - cairo=1.14.6=4\n",
      "  - certifi=2018.8.24=py35_1001\n",
      "  - curl=7.52.1=0\n",
      "  - fontconfig=2.12.1=6\n",
      "  - freetype=2.7=1\n",
      "  - gettext=0.19.8.1=hc5be6a0_1002\n",
      "  - glib=2.51.4=0\n",
      "  - graphite2=1.3.13=he1b5a44_1001\n",
      "  - gsl=2.6=h294904e_0\n",
      "  - harfbuzz=1.4.3=0\n",
      "  - html5lib=0.9999999=py35_0\n",
      "  - icu=58.2=hf484d3e_1000\n",
      "  - jpeg=9c=h14c3975_1001\n",
      "  - libblas=3.8.0=16_openblas\n",
      "  - libcblas=3.8.0=16_openblas\n",
      "  - libffi=3.2.1=he1b5a44_1007\n",
      "  - libgcc=7.2.0=h69d50b8_2\n",
      "  - libgcc-ng=9.2.0=h24d8f2e_2\n",
      "  - libgfortran-ng=7.3.0=hdf63c60_5\n",
      "  - libiconv=1.15=h516909a_1006\n",
      "  - liblapack=3.8.0=16_openblas\n",
      "  - liblapacke=3.8.0=16_openblas\n",
      "  - libopenblas=0.3.9=h5ec1e0e_0\n",
      "  - libpng=1.6.28=1\n",
      "  - libstdcxx-ng=9.2.0=hdf63c60_2\n",
      "  - libtiff=4.0.7=0\n",
      "  - libxml2=2.9.5=0\n",
      "  - llvm-openmp=10.0.0=hc9558a2_0\n",
      "  - markdown=3.2.1=py_0\n",
      "  - mmtf-python=1.0.2=py35_0\n",
      "  - mock=2.0.0=py35_0\n",
      "  - msgpack-python=0.5.6=py35h2d50403_3\n",
      "  - ncurses=5.9=10\n",
      "  - numpy=1.15.2=py35h99e49ec_0\n",
      "  - numpy-base=1.15.2=py35h2f8d375_0\n",
      "  - olefile=0.46=py_0\n",
      "  - openssl=1.0.2u=h516909a_0\n",
      "  - pandas=0.23.4=py35hf8a1672_0\n",
      "  - pango=1.40.4=0\n",
      "  - pbr=5.4.2=py_0\n",
      "  - pcre=8.44=he1b5a44_0\n",
      "  - pillow=4.3.0=py35_0\n",
      "  - pip=20.0.2=py_2\n",
      "  - pixman=0.34.0=h14c3975_1003\n",
      "  - protobuf=3.4.0=py35_0\n",
      "  - python=3.5.4=0\n",
      "  - python-dateutil=2.8.1=py_0\n",
      "  - pytz=2019.3=py_0\n",
      "  - r-base=3.3.2=5\n",
      "  - readline=6.2=0\n",
      "  - reportlab=3.4.0=py35_0\n",
      "  - rpy2=2.7.8=py35r3.3.2_1\n",
      "  - scikit-learn=0.20.0=py35h22eb022_1\n",
      "  - scipy=1.1.0=py35he2b7bc3_1\n",
      "  - setuptools=40.4.3=py35_0\n",
      "  - six=1.14.0=py_1\n",
      "  - sqlite=3.13.0=1\n",
      "  - tensorflow=1.3.0=py35_0\n",
      "  - tk=8.5.19=2\n",
      "  - webencodings=0.5.1=py_1\n",
      "  - werkzeug=1.0.1=pyh9f0ad1d_0\n",
      "  - wheel=0.34.2=py_1\n",
      "  - xz=5.2.5=h516909a_0\n",
      "  - zlib=1.2.8=3\n",
      "  # PlasFlow: should be installed from smaegol, NOT from pip\n",
      "  - plasflow=1.1.0"
     ]
    }
   ],
   "source": [
    "pipelineInfo('/ebio/abt3_projects/software/dev/ll_pipelines/llg/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-24T13:19:59.462928Z",
     "iopub.status.busy": "2021-04-24T13:19:59.461753Z",
     "iopub.status.idle": "2021-04-24T13:19:59.584944Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LLPRIMER\n",
      "========\n",
      "\n",
      "Ley Lab Primer generation pipeline (LLPRIMER)\n",
      "\n",
      "* Version: 0.3.0\n",
      "* Authors:\n",
      "  * Nick Youngblut <nyoungb2@gmail.com>\n",
      "* Maintainers:\n",
      "  * Nick Youngblut <nyoungb2@gmail.com>\n",
      "\n",
      "--- conda envs ---\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llprimer//bin/envs/pdp.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- python=3.7\n",
      "- intervaltree\n",
      "- prodigal\n",
      "- blast\n",
      "- bedtools\n",
      "- mafft\n",
      "- mummer=3.23\n",
      "- emboss\n",
      "- primer3=1.1.4\n",
      "- biopython<1.78\n",
      "- pybedtools\n",
      "- joblib\n",
      "- tqdm\n",
      "- openpyxl\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llprimer//bin/envs/genes.yaml <==\n",
      "channels:\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- python=3\n",
      "- numpy\n",
      "- pyfaidx\n",
      "- bioconda::seqkit\n",
      "- bioconda::fasta-splitter\n",
      "- bioconda::vsearch\n",
      "- bioconda::mmseqs2\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llprimer//bin/envs/aln.yaml <==\n",
      "channels:\n",
      "- bioconda\n",
      "- conda-forge\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::revtrans\n",
      "- bioconda::kalign3\n",
      "- bioconda::mafft\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llprimer//bin/envs/blast.yaml <==\n",
      "channels:\n",
      "- bioconda\n",
      "- conda-forge\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::seqkit\n",
      "- bioconda::biopython<1.78\n",
      "- bioconda::blast>=2.10.1\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llprimer//bin/envs/primer3.yaml <==\n",
      "channels:\n",
      "- bioconda\n",
      "- conda-forge\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::biopython<1.78\n",
      "- bioconda::primer3-py\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llprimer//bin/envs/prokka.yaml <==\n",
      "channels:\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::prokka"
     ]
    }
   ],
   "source": [
    "pipelineInfo('/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-24T13:19:59.788939Z",
     "iopub.status.busy": "2021-04-24T13:19:59.785978Z",
     "iopub.status.idle": "2021-04-24T13:20:01.356666Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "R version 3.6.3 (2020-02-29)\n",
       "Platform: x86_64-conda_cos6-linux-gnu (64-bit)\n",
       "Running under: Ubuntu 18.04.5 LTS\n",
       "\n",
       "Matrix products: default\n",
       "BLAS/LAPACK: /ebio/abt3_projects/Georg_animal_feces/envs/tidyverse/lib/libopenblasp-r0.3.9.so\n",
       "\n",
       "locale:\n",
       " [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              \n",
       " [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    \n",
       " [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   \n",
       " [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 \n",
       " [9] LC_ADDRESS=C               LC_TELEPHONE=C            \n",
       "[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       \n",
       "\n",
       "attached base packages:\n",
       "[1] stats     graphics  grDevices utils     datasets  methods   base     \n",
       "\n",
       "other attached packages:\n",
       "[1] LeyLabRMisc_0.1.6 ggplot2_3.3.1     tidyr_1.1.0       dplyr_1.0.0      \n",
       "\n",
       "loaded via a namespace (and not attached):\n",
       " [1] Rcpp_1.0.4.6     magrittr_1.5     munsell_0.5.0    tidyselect_1.1.0\n",
       " [5] uuid_0.1-4       colorspace_1.4-1 R6_2.4.1         rlang_0.4.6     \n",
       " [9] tools_3.6.3      grid_3.6.3       gtable_0.3.0     withr_2.2.0     \n",
       "[13] htmltools_0.4.0  ellipsis_0.3.1   digest_0.6.25    tibble_3.0.1    \n",
       "[17] lifecycle_0.2.0  crayon_1.3.4     IRdisplay_0.7.0  purrr_0.3.4     \n",
       "[21] repr_1.1.0       base64enc_0.1-3  vctrs_0.3.1      IRkernel_1.1    \n",
       "[25] glue_1.4.1       evaluate_0.14    stringi_1.4.6    pbdZMQ_0.3-3    \n",
       "[29] compiler_3.6.3   pillar_1.4.4     scales_1.1.1     generics_0.0.2  \n",
       "[33] jsonlite_1.6.1   pkgconfig_2.0.3 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "sessionInfo()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R [conda env:Georg_animal_feces-tidyverse]",
   "language": "R",
   "name": "conda-env-Georg_animal_feces-tidyverse-r"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.3"
  },
  "toc-autonumbering": true,
  "toc-showmarkdowntxt": false,
  "toc-showtags": false
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
