{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Goal\n",
    "\n",
    "* Primer design for clade of interest"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Var"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-25T05:11:54.244350Z",
     "iopub.status.busy": "2021-04-25T05:11:54.242087Z",
     "iopub.status.idle": "2021-04-25T05:11:54.262751Z"
    }
   },
   "outputs": [],
   "source": [
    "base_dir = '/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/'\n",
    "clade = 'Butyricimonas'\n",
    "domain = 'Bacteria'\n",
    "taxid = 574697 "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Init"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-25T05:11:55.085831Z",
     "iopub.status.busy": "2021-04-25T05:11:55.083714Z",
     "iopub.status.idle": "2021-04-25T05:11:55.103064Z"
    }
   },
   "outputs": [],
   "source": [
    "library(dplyr)\n",
    "library(tidyr)\n",
    "library(ggplot2)\n",
    "library(LeyLabRMisc)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-25T05:11:55.256130Z",
     "iopub.status.busy": "2021-04-25T05:11:55.254017Z",
     "iopub.status.idle": "2021-04-25T05:11:55.271593Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Created directory: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3//Butyricimonas \n"
     ]
    }
   ],
   "source": [
    "df.dims()\n",
    "work_dir = file.path(base_dir, clade)\n",
    "make_dir(work_dir)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Genome download\n",
    "\n",
    "* Downloading genomes from NCBI"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-12T13:07:32.874718Z",
     "iopub.status.busy": "2021-01-12T13:07:32.867938Z",
     "iopub.status.idle": "2021-01-12T13:07:32.985785Z",
     "shell.execute_reply": "2021-01-12T13:07:32.983225Z"
    }
   },
   "source": [
    "```\n",
    "OUTDIR=/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricimonas/\n",
    "mkdir -p $OUTDIR\n",
    "ncbi-genome-download -p 12 -s genbank -F fasta --genera Butyricimonas -o $OUTDIR bacteria\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Genome quality\n",
    "\n",
    "* Filtering genomes by quality"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-25T05:12:29.248476Z",
     "iopub.status.busy": "2021-04-25T05:12:29.246614Z",
     "iopub.status.idle": "2021-04-25T05:12:29.336247Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 29 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Name</th><th scope=col>Fasta</th><th scope=col>Domain</th><th scope=col>Taxid</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>GCA_000379665.1_ASM37966v1_genomic</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricimonas/genbank/bacteria/GCA_000379665.1/GCA_000379665.1_ASM37966v1_genomic.fna.gz</td><td>Bacteria</td><td>574697</td></tr>\n",
       "\t<tr><td>GCA_000519105.1_ASM51910v1_genomic</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricimonas/genbank/bacteria/GCA_000519105.1/GCA_000519105.1_ASM51910v1_genomic.fna.gz</td><td>Bacteria</td><td>574697</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>GCA_902364335.1_MGYG-HGUT-00203_genomic</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricimonas/genbank/bacteria/GCA_902364335.1/GCA_902364335.1_MGYG-HGUT-00203_genomic.fna.gz</td><td>Bacteria</td><td>574697</td></tr>\n",
       "\t<tr><td>GCA_902376195.1_MGYG-HGUT-01552_genomic</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricimonas/genbank/bacteria/GCA_902376195.1/GCA_902376195.1_MGYG-HGUT-01552_genomic.fna.gz</td><td>Bacteria</td><td>574697</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 29 × 4\n",
       "\\begin{tabular}{llll}\n",
       " Name & Fasta & Domain & Taxid\\\\\n",
       " <chr> & <chr> & <fct> & <dbl>\\\\\n",
       "\\hline\n",
       "\t GCA\\_000379665.1\\_ASM37966v1\\_genomic & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/v0.3/Butyricimonas/genbank/bacteria/GCA\\_000379665.1/GCA\\_000379665.1\\_ASM37966v1\\_genomic.fna.gz & Bacteria & 574697\\\\\n",
       "\t GCA\\_000519105.1\\_ASM51910v1\\_genomic & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/v0.3/Butyricimonas/genbank/bacteria/GCA\\_000519105.1/GCA\\_000519105.1\\_ASM51910v1\\_genomic.fna.gz & Bacteria & 574697\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t GCA\\_902364335.1\\_MGYG-HGUT-00203\\_genomic & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/v0.3/Butyricimonas/genbank/bacteria/GCA\\_902364335.1/GCA\\_902364335.1\\_MGYG-HGUT-00203\\_genomic.fna.gz & Bacteria & 574697\\\\\n",
       "\t GCA\\_902376195.1\\_MGYG-HGUT-01552\\_genomic & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/v0.3/Butyricimonas/genbank/bacteria/GCA\\_902376195.1/GCA\\_902376195.1\\_MGYG-HGUT-01552\\_genomic.fna.gz & Bacteria & 574697\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 29 × 4\n",
       "\n",
       "| Name &lt;chr&gt; | Fasta &lt;chr&gt; | Domain &lt;fct&gt; | Taxid &lt;dbl&gt; |\n",
       "|---|---|---|---|\n",
       "| GCA_000379665.1_ASM37966v1_genomic | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricimonas/genbank/bacteria/GCA_000379665.1/GCA_000379665.1_ASM37966v1_genomic.fna.gz | Bacteria | 574697 |\n",
       "| GCA_000519105.1_ASM51910v1_genomic | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricimonas/genbank/bacteria/GCA_000519105.1/GCA_000519105.1_ASM51910v1_genomic.fna.gz | Bacteria | 574697 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| GCA_902364335.1_MGYG-HGUT-00203_genomic | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricimonas/genbank/bacteria/GCA_902364335.1/GCA_902364335.1_MGYG-HGUT-00203_genomic.fna.gz | Bacteria | 574697 |\n",
       "| GCA_902376195.1_MGYG-HGUT-01552_genomic | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricimonas/genbank/bacteria/GCA_902376195.1/GCA_902376195.1_MGYG-HGUT-01552_genomic.fna.gz | Bacteria | 574697 |\n",
       "\n"
      ],
      "text/plain": [
       "   Name                                   \n",
       "1  GCA_000379665.1_ASM37966v1_genomic     \n",
       "2  GCA_000519105.1_ASM51910v1_genomic     \n",
       "⋮  ⋮                                      \n",
       "28 GCA_902364335.1_MGYG-HGUT-00203_genomic\n",
       "29 GCA_902376195.1_MGYG-HGUT-01552_genomic\n",
       "   Fasta                                                                                                                                                                                \n",
       "1  /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricimonas/genbank/bacteria/GCA_000379665.1/GCA_000379665.1_ASM37966v1_genomic.fna.gz     \n",
       "2  /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricimonas/genbank/bacteria/GCA_000519105.1/GCA_000519105.1_ASM51910v1_genomic.fna.gz     \n",
       "⋮  ⋮                                                                                                                                                                                    \n",
       "28 /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricimonas/genbank/bacteria/GCA_902364335.1/GCA_902364335.1_MGYG-HGUT-00203_genomic.fna.gz\n",
       "29 /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricimonas/genbank/bacteria/GCA_902376195.1/GCA_902376195.1_MGYG-HGUT-01552_genomic.fna.gz\n",
       "   Domain   Taxid \n",
       "1  Bacteria 574697\n",
       "2  Bacteria 574697\n",
       "⋮  ⋮        ⋮     \n",
       "28 Bacteria 574697\n",
       "29 Bacteria 574697"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "File written: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3//Butyricimonas/genbank/samples.txt \n"
     ]
    }
   ],
   "source": [
    "D = file.path(base_dir, clade, 'genbank')\n",
    "files = list_files(D, '.fna.gz')\n",
    "samps = data.frame(Name = files %>% as.character %>% basename,\n",
    "                   Fasta = files,\n",
    "                   Domain = 'Bacteria',\n",
    "                   Taxid = taxid) %>%\n",
    "    mutate(Name = gsub('\\\\.fna\\\\.gz$', '', Name),\n",
    "           Fasta = gsub('/+', '/', Fasta))\n",
    "samps\n",
    "\n",
    "# writing file\n",
    "outfile = file.path(D, 'samples.txt')\n",
    "write_table(samps, outfile)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### LLG"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Config"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-25T05:15:19.899908Z",
     "iopub.status.busy": "2021-04-25T05:15:19.898148Z",
     "iopub.status.idle": "2021-04-25T05:15:19.930202Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# table with genome --> fasta_file information\n",
      "samples_file: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/genbank/samples.txt\n",
      "\n",
      "# output location\n",
      "output_dir: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricimonas/LLG/\n",
      "\n",
      "# temporary file directory (your username will be added automatically)\n",
      "tmp_dir: /ebio/abt3_scratch/\n",
      "\n",
      "# batch processing of genomes for certain steps\n",
      "## increase to better parallelize\n",
      "batches: 2 \n",
      "\n",
      "# Domain of genomes ('Archaea' or 'Bacteria)\n",
      "## Use \"Skip\" if provided as a \"Domain\" column in the genome table\n",
      "Domain: Skip\n",
      "\n",
      "# software parameters\n",
      "# Use \"Skip\" to skip any of these steps. If no params for rule, use \"\"\n",
      "# dRep MAGs are not further analyzed, but you can de-rep & then use the de-rep genome table as input.\n",
      "params:\n",
      "  ionice: -c 3\n",
      "  # assembly assessment\n",
      "  seqkit: \"\"\n",
      "  quast: Skip #\"\"\n",
      "  multiqc_on_quast: \"\" \n",
      "  checkm: \"\"\n",
      "  # de-replication (requires checkm)\n",
      "  drep: -comp 90 -con 5 -sa 0.999\n",
      "  # taxonomy\n",
      "  sourmash:\n",
      "    compute: Skip #--scaled 10000 -k 31\n",
      "    gather: -k 31\n",
      "  gtdbtk:\n",
      "    classify_wf: --min_perc_aa 10\n",
      "  # genome pairwise ANI  \n",
      "  fastani: Skip #--fragLen 3000 --minFraction 0.2 -k 16\n",
      "  dashing: Skip #-k 31 --full-tsv\n",
      "  comparem_aai: Skip #--evalue 0.001\n",
      "  # gene annotation\n",
      "  gene_call:\n",
      "    prokka: Skip #\"\" \n",
      "    multiqc_on_prokka: \"\"\n",
      "    prodigal: Skip #\"\"\n",
      "  eggnog_mapper: \"\"\n",
      "  eggnog_mapper_annot: \"\"\n",
      "  # rRNA (16S alignment & phylogeny)\n",
      "  barrnap: Skip #--lencutoff 0.8\n",
      "  vsearch_per_genome_drep: --id 0.95    # Skip to prevent drep of 16S copies within each genome\n",
      "  qiime2_fasttree: \"\"\n",
      "  qiime2_iqtree: --p-alrt 1000 --p-abayes --p-lbp 1000  --p-substitution-model 'GTR+I+G'\n",
      "  # genome phylogeny\n",
      "  phylophlan_config: Skip #--map_dna diamond --db_aa diamond --map_aa diamond --msa mafft --trim trimal --tree1 fasttree --tree2 raxml\n",
      "  phylophlan: --diversity high --min_num_markers 50 --accurate\n",
      "  # phenotype\n",
      "  traitar: Skip #\"\"\n",
      "  # biosynthetic gene clusters (BGCs)\n",
      "  antismash: Skip #--cb-knownclusters --cb-subclusters --asf  \n",
      "  DeepBGC: Skip #--score 0.5 --classifier-score 0.5 --prodigal-meta-mode\n",
      "  # antimicrobial resistance (AMR)\n",
      "  abricate: Skip #--minid 75 --mincov 80\n",
      "  # CRISPRs\n",
      "  cctyper: Skip #--prodigal meta\n",
      "  \n",
      "# databases\n",
      "databases:\n",
      "  checkM_data: /ebio/abt3_projects/databases_no-backup/checkM/\n",
      "  sourmash: /ebio/abt3_projects/databases_no-backup/sourmash/genbank-k31.sbt.json\n",
      "  sourmash_lca: /ebio/abt3_projects/databases_no-backup/sourmash/genbank-k31.lca.json.gz\n",
      "  gtdbtk: /ebio/abt3_projects/databases_no-backup/GTDB/release95/gtdbtk/db_info.md\n",
      "  phylophlan: /ebio/abt3_projects/databases_no-backup/phylophlan/PhyloPhlan/phylophlan.faa.bz2\n",
      "  eggnog: /ebio/abt3_projects/databases_no-backup/Eggnog/v2/eggnog.db\n",
      "  eggnog_diamond: /ebio/abt3_projects/databases_no-backup/Eggnog/v2/eggnog_proteins.dmnd\n",
      "  antismash: /ebio/abt3_projects/databases_no-backup/antismash/v5/\n",
      "  deepbgc: /ebio/abt3_projects/databases_no-backup/DeepBGC/\n",
      "  traitar: /ebio/abt3_projects/databases_no-backup/pfam/traitar/\n",
      "  abricate:\n",
      "    ncbi: /ebio/abt3_projects/databases_no-backup/abricate/ncbi/sequences\n",
      "    card: /ebio/abt3_projects/databases_no-backup/abricate/card/sequences\n",
      "    resfinder: /ebio/abt3_projects/databases_no-backup/abricate/resfinder/sequences\n",
      "    argannot: /ebio/abt3_projects/databases_no-backup/abricate/argannot/sequences\n",
      "    bacmet2: /ebio/abt3_projects/databases_no-backup/abricate/bacmet2/sequences\n",
      "    vfdb: /ebio/abt3_projects/databases_no-backup/abricate/vfdb/sequences\n",
      "    megares: /ebio/abt3_projects/databases_no-backup/abricate/megares/sequences\n",
      "    plasmidfinder: /ebio/abt3_projects/databases_no-backup/abricate/plasmidfinder/sequences\n",
      "\n",
      "# snakemake pipeline\n",
      "pipeline:\n",
      "  snakemake_folder: ./\n",
      "  script_folder: ./bin/scripts/\n",
      "  use_shared_mem: True\n",
      "  name: LLG"
     ]
    }
   ],
   "source": [
    "cat_file(file.path(work_dir, 'LLG', 'config.yaml'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-12T13:14:50.416604Z",
     "iopub.status.busy": "2021-01-12T13:14:50.414179Z",
     "iopub.status.idle": "2021-01-12T13:14:50.431925Z",
     "shell.execute_reply": "2021-01-12T13:14:50.430260Z"
    }
   },
   "source": [
    "#### Run\n",
    "\n",
    "```\n",
    "(snakemake) @ rick:/ebio/abt3_projects/software/dev/ll_pipelines/llg\n",
    "$ screen -L -S llg ./snakemake_sge.sh /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricimonas/LLG/config.yaml 20 -F\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-12T13:14:10.995042Z",
     "iopub.status.busy": "2021-01-12T13:14:10.992773Z",
     "iopub.status.idle": "2021-01-12T13:14:11.009275Z",
     "shell.execute_reply": "2021-01-12T13:14:11.008255Z"
    }
   },
   "source": [
    "### Samples table of high-quality genomes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-25T05:20:08.043307Z",
     "iopub.status.busy": "2021-04-25T05:20:08.040670Z",
     "iopub.status.idle": "2021-04-25T05:20:08.100898Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 29 × 33</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Bin.Id</th><th scope=col>Marker.lineage</th><th scope=col>X..genomes</th><th scope=col>X..markers</th><th scope=col>X..marker.sets</th><th scope=col>Completeness</th><th scope=col>Contamination</th><th scope=col>Strain.heterogeneity</th><th scope=col>Genome.size..bp.</th><th scope=col>X..ambiguous.bases</th><th scope=col>⋯</th><th scope=col>X0</th><th scope=col>X1</th><th scope=col>X2</th><th scope=col>X3</th><th scope=col>X4</th><th scope=col>X5.</th><th scope=col>assembly.Id</th><th scope=col>assembler.Id</th><th scope=col>taxon.Id</th><th scope=col>File</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>GCA_000379665.1_ASM37966v1_genomic </td><td>k__Bacteria (UID2569)</td><td>434</td><td>278</td><td>186</td><td>100.00</td><td>0.00</td><td>0</td><td>4770838</td><td> 131</td><td>⋯</td><td> 0</td><td>278</td><td>0</td><td>0</td><td>0</td><td>0</td><td>|ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Butyricimonas|LLG_output|checkM|1|checkm|markers_qa_summary.tsv.1</td><td>markers_qa_summary.tsv.1</td><td>NA</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/LLG_output/checkM/1/checkm/markers_qa_summary.tsv.1</td></tr>\n",
       "\t<tr><td>GCA_001915615.1_ASM191561v1_genomic</td><td>k__Bacteria (UID2569)</td><td>434</td><td>278</td><td>186</td><td> 88.69</td><td>0.27</td><td>0</td><td>4167409</td><td>2477</td><td>⋯</td><td>44</td><td>233</td><td>1</td><td>0</td><td>0</td><td>0</td><td>|ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Butyricimonas|LLG_output|checkM|1|checkm|markers_qa_summary.tsv.2</td><td>markers_qa_summary.tsv.2</td><td>NA</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/LLG_output/checkM/1/checkm/markers_qa_summary.tsv.2</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋱</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>GCA_900184685.1_PRJEB20857_genomic     </td><td>k__Bacteria (UID2569)</td><td>434</td><td>278</td><td>186</td><td>100.00</td><td>0</td><td>0</td><td>5300003</td><td>  0</td><td>⋯</td><td>0</td><td>278</td><td>0</td><td>0</td><td>0</td><td>0</td><td>|ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Butyricimonas|LLG_output|checkM|2|checkm|markers_qa_summary.tsv.13</td><td>markers_qa_summary.tsv.13</td><td>NA</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/LLG_output/checkM/2/checkm/markers_qa_summary.tsv.13</td></tr>\n",
       "\t<tr><td>GCA_902364335.1_MGYG-HGUT-00203_genomic</td><td>k__Bacteria (UID2569)</td><td>434</td><td>278</td><td>186</td><td> 99.46</td><td>0</td><td>0</td><td>4537596</td><td>250</td><td>⋯</td><td>1</td><td>277</td><td>0</td><td>0</td><td>0</td><td>0</td><td>|ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Butyricimonas|LLG_output|checkM|2|checkm|markers_qa_summary.tsv.14</td><td>markers_qa_summary.tsv.14</td><td>NA</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/LLG_output/checkM/2/checkm/markers_qa_summary.tsv.14</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 29 × 33\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " Bin.Id & Marker.lineage & X..genomes & X..markers & X..marker.sets & Completeness & Contamination & Strain.heterogeneity & Genome.size..bp. & X..ambiguous.bases & ⋯ & X0 & X1 & X2 & X3 & X4 & X5. & assembly.Id & assembler.Id & taxon.Id & File\\\\\n",
       " <fct> & <fct> & <int> & <int> & <int> & <dbl> & <dbl> & <int> & <int> & <int> & ⋯ & <int> & <int> & <int> & <int> & <int> & <int> & <fct> & <fct> & <lgl> & <fct>\\\\\n",
       "\\hline\n",
       "\t GCA\\_000379665.1\\_ASM37966v1\\_genomic  & k\\_\\_Bacteria (UID2569) & 434 & 278 & 186 & 100.00 & 0.00 & 0 & 4770838 &  131 & ⋯ &  0 & 278 & 0 & 0 & 0 & 0 & \\textbar{}ebio\\textbar{}abt3\\_projects\\textbar{}software\\textbar{}dev\\textbar{}ll\\_pipelines\\textbar{}llprimer\\textbar{}experiments\\textbar{}HMP\\_most-wanted\\textbar{}Butyricimonas\\textbar{}LLG\\_output\\textbar{}checkM\\textbar{}1\\textbar{}checkm\\textbar{}markers\\_qa\\_summary.tsv.1 & markers\\_qa\\_summary.tsv.1 & NA & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/Butyricimonas/LLG\\_output/checkM/1/checkm/markers\\_qa\\_summary.tsv.1\\\\\n",
       "\t GCA\\_001915615.1\\_ASM191561v1\\_genomic & k\\_\\_Bacteria (UID2569) & 434 & 278 & 186 &  88.69 & 0.27 & 0 & 4167409 & 2477 & ⋯ & 44 & 233 & 1 & 0 & 0 & 0 & \\textbar{}ebio\\textbar{}abt3\\_projects\\textbar{}software\\textbar{}dev\\textbar{}ll\\_pipelines\\textbar{}llprimer\\textbar{}experiments\\textbar{}HMP\\_most-wanted\\textbar{}Butyricimonas\\textbar{}LLG\\_output\\textbar{}checkM\\textbar{}1\\textbar{}checkm\\textbar{}markers\\_qa\\_summary.tsv.2 & markers\\_qa\\_summary.tsv.2 & NA & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/Butyricimonas/LLG\\_output/checkM/1/checkm/markers\\_qa\\_summary.tsv.2\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋱ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t GCA\\_900184685.1\\_PRJEB20857\\_genomic      & k\\_\\_Bacteria (UID2569) & 434 & 278 & 186 & 100.00 & 0 & 0 & 5300003 &   0 & ⋯ & 0 & 278 & 0 & 0 & 0 & 0 & \\textbar{}ebio\\textbar{}abt3\\_projects\\textbar{}software\\textbar{}dev\\textbar{}ll\\_pipelines\\textbar{}llprimer\\textbar{}experiments\\textbar{}HMP\\_most-wanted\\textbar{}Butyricimonas\\textbar{}LLG\\_output\\textbar{}checkM\\textbar{}2\\textbar{}checkm\\textbar{}markers\\_qa\\_summary.tsv.13 & markers\\_qa\\_summary.tsv.13 & NA & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/Butyricimonas/LLG\\_output/checkM/2/checkm/markers\\_qa\\_summary.tsv.13\\\\\n",
       "\t GCA\\_902364335.1\\_MGYG-HGUT-00203\\_genomic & k\\_\\_Bacteria (UID2569) & 434 & 278 & 186 &  99.46 & 0 & 0 & 4537596 & 250 & ⋯ & 1 & 277 & 0 & 0 & 0 & 0 & \\textbar{}ebio\\textbar{}abt3\\_projects\\textbar{}software\\textbar{}dev\\textbar{}ll\\_pipelines\\textbar{}llprimer\\textbar{}experiments\\textbar{}HMP\\_most-wanted\\textbar{}Butyricimonas\\textbar{}LLG\\_output\\textbar{}checkM\\textbar{}2\\textbar{}checkm\\textbar{}markers\\_qa\\_summary.tsv.14 & markers\\_qa\\_summary.tsv.14 & NA & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/Butyricimonas/LLG\\_output/checkM/2/checkm/markers\\_qa\\_summary.tsv.14\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 29 × 33\n",
       "\n",
       "| Bin.Id &lt;fct&gt; | Marker.lineage &lt;fct&gt; | X..genomes &lt;int&gt; | X..markers &lt;int&gt; | X..marker.sets &lt;int&gt; | Completeness &lt;dbl&gt; | Contamination &lt;dbl&gt; | Strain.heterogeneity &lt;int&gt; | Genome.size..bp. &lt;int&gt; | X..ambiguous.bases &lt;int&gt; | ⋯ ⋯ | X0 &lt;int&gt; | X1 &lt;int&gt; | X2 &lt;int&gt; | X3 &lt;int&gt; | X4 &lt;int&gt; | X5. &lt;int&gt; | assembly.Id &lt;fct&gt; | assembler.Id &lt;fct&gt; | taxon.Id &lt;lgl&gt; | File &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| GCA_000379665.1_ASM37966v1_genomic  | k__Bacteria (UID2569) | 434 | 278 | 186 | 100.00 | 0.00 | 0 | 4770838 |  131 | ⋯ |  0 | 278 | 0 | 0 | 0 | 0 | |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Butyricimonas|LLG_output|checkM|1|checkm|markers_qa_summary.tsv.1 | markers_qa_summary.tsv.1 | NA | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/LLG_output/checkM/1/checkm/markers_qa_summary.tsv.1 |\n",
       "| GCA_001915615.1_ASM191561v1_genomic | k__Bacteria (UID2569) | 434 | 278 | 186 |  88.69 | 0.27 | 0 | 4167409 | 2477 | ⋯ | 44 | 233 | 1 | 0 | 0 | 0 | |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Butyricimonas|LLG_output|checkM|1|checkm|markers_qa_summary.tsv.2 | markers_qa_summary.tsv.2 | NA | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/LLG_output/checkM/1/checkm/markers_qa_summary.tsv.2 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋱ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| GCA_900184685.1_PRJEB20857_genomic      | k__Bacteria (UID2569) | 434 | 278 | 186 | 100.00 | 0 | 0 | 5300003 |   0 | ⋯ | 0 | 278 | 0 | 0 | 0 | 0 | |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Butyricimonas|LLG_output|checkM|2|checkm|markers_qa_summary.tsv.13 | markers_qa_summary.tsv.13 | NA | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/LLG_output/checkM/2/checkm/markers_qa_summary.tsv.13 |\n",
       "| GCA_902364335.1_MGYG-HGUT-00203_genomic | k__Bacteria (UID2569) | 434 | 278 | 186 |  99.46 | 0 | 0 | 4537596 | 250 | ⋯ | 1 | 277 | 0 | 0 | 0 | 0 | |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Butyricimonas|LLG_output|checkM|2|checkm|markers_qa_summary.tsv.14 | markers_qa_summary.tsv.14 | NA | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/LLG_output/checkM/2/checkm/markers_qa_summary.tsv.14 |\n",
       "\n"
      ],
      "text/plain": [
       "   Bin.Id                                  Marker.lineage        X..genomes\n",
       "1  GCA_000379665.1_ASM37966v1_genomic      k__Bacteria (UID2569) 434       \n",
       "2  GCA_001915615.1_ASM191561v1_genomic     k__Bacteria (UID2569) 434       \n",
       "⋮  ⋮                                       ⋮                     ⋮         \n",
       "28 GCA_900184685.1_PRJEB20857_genomic      k__Bacteria (UID2569) 434       \n",
       "29 GCA_902364335.1_MGYG-HGUT-00203_genomic k__Bacteria (UID2569) 434       \n",
       "   X..markers X..marker.sets Completeness Contamination Strain.heterogeneity\n",
       "1  278        186            100.00       0.00          0                   \n",
       "2  278        186             88.69       0.27          0                   \n",
       "⋮  ⋮          ⋮              ⋮            ⋮             ⋮                   \n",
       "28 278        186            100.00       0             0                   \n",
       "29 278        186             99.46       0             0                   \n",
       "   Genome.size..bp. X..ambiguous.bases ⋯ X0 X1  X2 X3 X4 X5.\n",
       "1  4770838           131               ⋯  0 278 0  0  0  0  \n",
       "2  4167409          2477               ⋯ 44 233 1  0  0  0  \n",
       "⋮  ⋮                ⋮                  ⋱ ⋮  ⋮   ⋮  ⋮  ⋮  ⋮  \n",
       "28 5300003            0                ⋯ 0  278 0  0  0  0  \n",
       "29 4537596          250                ⋯ 1  277 0  0  0  0  \n",
       "   assembly.Id                                                                                                                                          \n",
       "1  |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Butyricimonas|LLG_output|checkM|1|checkm|markers_qa_summary.tsv.1 \n",
       "2  |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Butyricimonas|LLG_output|checkM|1|checkm|markers_qa_summary.tsv.2 \n",
       "⋮  ⋮                                                                                                                                                    \n",
       "28 |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Butyricimonas|LLG_output|checkM|2|checkm|markers_qa_summary.tsv.13\n",
       "29 |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Butyricimonas|LLG_output|checkM|2|checkm|markers_qa_summary.tsv.14\n",
       "   assembler.Id              taxon.Id\n",
       "1  markers_qa_summary.tsv.1  NA      \n",
       "2  markers_qa_summary.tsv.2  NA      \n",
       "⋮  ⋮                         ⋮       \n",
       "28 markers_qa_summary.tsv.13 NA      \n",
       "29 markers_qa_summary.tsv.14 NA      \n",
       "   File                                                                                                                                                 \n",
       "1  /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/LLG_output/checkM/1/checkm/markers_qa_summary.tsv.1 \n",
       "2  /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/LLG_output/checkM/1/checkm/markers_qa_summary.tsv.2 \n",
       "⋮  ⋮                                                                                                                                                    \n",
       "28 /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/LLG_output/checkM/2/checkm/markers_qa_summary.tsv.13\n",
       "29 /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricimonas/LLG_output/checkM/2/checkm/markers_qa_summary.tsv.14"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# checkM summary\n",
    "checkm = file.path(work_dir, 'LLG', 'checkM', 'checkm_qa_summary.tsv') %>%\n",
    "    read.delim(sep='\\t') \n",
    "checkm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-25T05:20:08.104927Z",
     "iopub.status.busy": "2021-04-25T05:20:08.102943Z",
     "iopub.status.idle": "2021-04-25T05:20:08.131644Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 29 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>genome</th><th scope=col>completeness</th><th scope=col>contamination</th><th scope=col>Bin.Id</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000379665.1_ASM37966v1_genomic.fna </td><td>100.00</td><td>0.00</td><td>GCA_000379665.1_ASM37966v1_genomic </td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_001915615.1_ASM191561v1_genomic.fna</td><td> 88.69</td><td>0.27</td><td>GCA_001915615.1_ASM191561v1_genomic</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_900184685.1_PRJEB20857_genomic.fna     </td><td>100.00</td><td>0</td><td>GCA_900184685.1_PRJEB20857_genomic     </td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_902364335.1_MGYG-HGUT-00203_genomic.fna</td><td> 99.46</td><td>0</td><td>GCA_902364335.1_MGYG-HGUT-00203_genomic</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 29 × 4\n",
       "\\begin{tabular}{llll}\n",
       " genome & completeness & contamination & Bin.Id\\\\\n",
       " <fct> & <dbl> & <dbl> & <chr>\\\\\n",
       "\\hline\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_000379665.1\\_ASM37966v1\\_genomic.fna  & 100.00 & 0.00 & GCA\\_000379665.1\\_ASM37966v1\\_genomic \\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_001915615.1\\_ASM191561v1\\_genomic.fna &  88.69 & 0.27 & GCA\\_001915615.1\\_ASM191561v1\\_genomic\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_900184685.1\\_PRJEB20857\\_genomic.fna      & 100.00 & 0 & GCA\\_900184685.1\\_PRJEB20857\\_genomic     \\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_902364335.1\\_MGYG-HGUT-00203\\_genomic.fna &  99.46 & 0 & GCA\\_902364335.1\\_MGYG-HGUT-00203\\_genomic\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 29 × 4\n",
       "\n",
       "| genome &lt;fct&gt; | completeness &lt;dbl&gt; | contamination &lt;dbl&gt; | Bin.Id &lt;chr&gt; |\n",
       "|---|---|---|---|\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000379665.1_ASM37966v1_genomic.fna  | 100.00 | 0.00 | GCA_000379665.1_ASM37966v1_genomic  |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_001915615.1_ASM191561v1_genomic.fna |  88.69 | 0.27 | GCA_001915615.1_ASM191561v1_genomic |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_900184685.1_PRJEB20857_genomic.fna      | 100.00 | 0 | GCA_900184685.1_PRJEB20857_genomic      |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_902364335.1_MGYG-HGUT-00203_genomic.fna |  99.46 | 0 | GCA_902364335.1_MGYG-HGUT-00203_genomic |\n",
       "\n"
      ],
      "text/plain": [
       "   genome                                                                                           \n",
       "1  /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000379665.1_ASM37966v1_genomic.fna     \n",
       "2  /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_001915615.1_ASM191561v1_genomic.fna    \n",
       "⋮  ⋮                                                                                                \n",
       "28 /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_900184685.1_PRJEB20857_genomic.fna     \n",
       "29 /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_902364335.1_MGYG-HGUT-00203_genomic.fna\n",
       "   completeness contamination Bin.Id                                 \n",
       "1  100.00       0.00          GCA_000379665.1_ASM37966v1_genomic     \n",
       "2   88.69       0.27          GCA_001915615.1_ASM191561v1_genomic    \n",
       "⋮  ⋮            ⋮             ⋮                                      \n",
       "28 100.00       0             GCA_900184685.1_PRJEB20857_genomic     \n",
       "29  99.46       0             GCA_902364335.1_MGYG-HGUT-00203_genomic"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# dRep summary\n",
    "drep = file.path(work_dir, 'LLG', 'drep', 'checkm_markers_qa_summary.tsv') %>%\n",
    "    read.delim(sep='\\t') %>%\n",
    "    mutate(Bin.Id = gsub('.+/', '', genome),\n",
    "           Bin.Id = gsub('\\\\.fna$', '', Bin.Id))\n",
    "drep"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-25T05:20:08.219954Z",
     "iopub.status.busy": "2021-04-25T05:20:08.217948Z",
     "iopub.status.idle": "2021-04-25T05:20:08.251504Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 17 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Name</th><th scope=col>Fasta</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>GCA_003991565.1_ASM399156v1_genomic</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricimonas/LLG/drep/drep/dereplicated_genomes/GCA_003991565.1_ASM399156v1_genomic.fna</td></tr>\n",
       "\t<tr><td>GCA_900258545.1_PRJEB22910_genomic </td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricimonas/LLG/drep/drep/dereplicated_genomes/GCA_900258545.1_PRJEB22910_genomic.fna </td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>GCA_014287455.1_ASM1428745v1_genomic</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricimonas/LLG/drep/drep/dereplicated_genomes/GCA_014287455.1_ASM1428745v1_genomic.fna</td></tr>\n",
       "\t<tr><td>GCA_900184685.1_PRJEB20857_genomic  </td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricimonas/LLG/drep/drep/dereplicated_genomes/GCA_900184685.1_PRJEB20857_genomic.fna  </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 17 × 2\n",
       "\\begin{tabular}{ll}\n",
       " Name & Fasta\\\\\n",
       " <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\t GCA\\_003991565.1\\_ASM399156v1\\_genomic & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/v0.3/Butyricimonas/LLG/drep/drep/dereplicated\\_genomes/GCA\\_003991565.1\\_ASM399156v1\\_genomic.fna\\\\\n",
       "\t GCA\\_900258545.1\\_PRJEB22910\\_genomic  & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/v0.3/Butyricimonas/LLG/drep/drep/dereplicated\\_genomes/GCA\\_900258545.1\\_PRJEB22910\\_genomic.fna \\\\\n",
       "\t ⋮ & ⋮\\\\\n",
       "\t GCA\\_014287455.1\\_ASM1428745v1\\_genomic & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/v0.3/Butyricimonas/LLG/drep/drep/dereplicated\\_genomes/GCA\\_014287455.1\\_ASM1428745v1\\_genomic.fna\\\\\n",
       "\t GCA\\_900184685.1\\_PRJEB20857\\_genomic   & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/v0.3/Butyricimonas/LLG/drep/drep/dereplicated\\_genomes/GCA\\_900184685.1\\_PRJEB20857\\_genomic.fna  \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 17 × 2\n",
       "\n",
       "| Name &lt;fct&gt; | Fasta &lt;fct&gt; |\n",
       "|---|---|\n",
       "| GCA_003991565.1_ASM399156v1_genomic | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricimonas/LLG/drep/drep/dereplicated_genomes/GCA_003991565.1_ASM399156v1_genomic.fna |\n",
       "| GCA_900258545.1_PRJEB22910_genomic  | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricimonas/LLG/drep/drep/dereplicated_genomes/GCA_900258545.1_PRJEB22910_genomic.fna  |\n",
       "| ⋮ | ⋮ |\n",
       "| GCA_014287455.1_ASM1428745v1_genomic | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricimonas/LLG/drep/drep/dereplicated_genomes/GCA_014287455.1_ASM1428745v1_genomic.fna |\n",
       "| GCA_900184685.1_PRJEB20857_genomic   | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricimonas/LLG/drep/drep/dereplicated_genomes/GCA_900184685.1_PRJEB20857_genomic.fna   |\n",
       "\n"
      ],
      "text/plain": [
       "   Name                                \n",
       "1  GCA_003991565.1_ASM399156v1_genomic \n",
       "2  GCA_900258545.1_PRJEB22910_genomic  \n",
       "⋮  ⋮                                   \n",
       "16 GCA_014287455.1_ASM1428745v1_genomic\n",
       "17 GCA_900184685.1_PRJEB20857_genomic  \n",
       "   Fasta                                                                                                                                                                            \n",
       "1  /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricimonas/LLG/drep/drep/dereplicated_genomes/GCA_003991565.1_ASM399156v1_genomic.fna \n",
       "2  /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricimonas/LLG/drep/drep/dereplicated_genomes/GCA_900258545.1_PRJEB22910_genomic.fna  \n",
       "⋮  ⋮                                                                                                                                                                                \n",
       "16 /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricimonas/LLG/drep/drep/dereplicated_genomes/GCA_014287455.1_ASM1428745v1_genomic.fna\n",
       "17 /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricimonas/LLG/drep/drep/dereplicated_genomes/GCA_900184685.1_PRJEB20857_genomic.fna  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# de-replicated genomes\n",
    "drep_gen = file.path(work_dir, 'LLG', 'drep', 'dereplicated_genomes.tsv') %>%\n",
    "    read.delim(sep='\\t')\n",
    "drep_gen"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-25T05:20:08.345916Z",
     "iopub.status.busy": "2021-04-25T05:20:08.344217Z",
     "iopub.status.idle": "2021-04-25T05:20:08.391974Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 29 × 22</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>user_genome</th><th scope=col>Domain</th><th scope=col>Phylum</th><th scope=col>Class</th><th scope=col>Order</th><th scope=col>Family</th><th scope=col>Genus</th><th scope=col>Species</th><th scope=col>fastani_reference</th><th scope=col>fastani_reference_radius</th><th scope=col>⋯</th><th scope=col>fastani_af</th><th scope=col>closest_placement_reference</th><th scope=col>closest_placement_radius</th><th scope=col>closest_placement_taxonomy</th><th scope=col>closest_placement_ani</th><th scope=col>closest_placement_af</th><th scope=col>msa_percent</th><th scope=col>translation_table</th><th scope=col>red_value</th><th scope=col>warnings</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>GCA_000379665.1_ASM37966v1_genomic </td><td>d__Bacteria</td><td>p__Bacteroidota</td><td>c__Bacteroidia</td><td>o__Bacteroidales</td><td>f__Marinifilaceae</td><td>g__Butyricimonas</td><td>s__Butyricimonas synergistica</td><td>GCF_000379665.1</td><td>95</td><td>⋯</td><td>1.00</td><td>GCF_000379665.1</td><td>95</td><td>d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas synergistica</td><td>100.00</td><td>1.00</td><td>96.57</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>GCA_001915615.1_ASM191561v1_genomic</td><td>d__Bacteria</td><td>p__Bacteroidota</td><td>c__Bacteroidia</td><td>o__Bacteroidales</td><td>f__Marinifilaceae</td><td>g__Butyricimonas</td><td>s__Butyricimonas faecihominis</td><td>GCF_003851945.2</td><td>95</td><td>⋯</td><td>0.78</td><td>GCF_003851945.2</td><td>95</td><td>d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas faecihominis</td><td> 98.09</td><td>0.78</td><td>84.37</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋱</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>GCA_900184685.1_PRJEB20857_genomic     </td><td>d__Bacteria</td><td>p__Bacteroidota</td><td>c__Bacteroidia</td><td>o__Bacteroidales</td><td>f__Marinifilaceae</td><td>g__Butyricimonas</td><td>s__Butyricimonas sp900184685</td><td>GCF_900184685.1</td><td>95</td><td>⋯</td><td>1.00</td><td>GCF_900184685.1</td><td>95</td><td>d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas sp900184685</td><td>100.00</td><td>1.00</td><td>96.71</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>GCA_902364335.1_MGYG-HGUT-00203_genomic</td><td>d__Bacteria</td><td>p__Bacteroidota</td><td>c__Bacteroidia</td><td>o__Bacteroidales</td><td>f__Marinifilaceae</td><td>g__Butyricimonas</td><td>s__Butyricimonas virosa     </td><td>GCF_000519105.1</td><td>95</td><td>⋯</td><td>0.85</td><td>GCF_000519105.1</td><td>95</td><td>d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas virosa     </td><td> 97.98</td><td>0.85</td><td>96.79</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 29 × 22\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " user\\_genome & Domain & Phylum & Class & Order & Family & Genus & Species & fastani\\_reference & fastani\\_reference\\_radius & ⋯ & fastani\\_af & closest\\_placement\\_reference & closest\\_placement\\_radius & closest\\_placement\\_taxonomy & closest\\_placement\\_ani & closest\\_placement\\_af & msa\\_percent & translation\\_table & red\\_value & warnings\\\\\n",
       " <fct> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <fct> & <dbl> & ⋯ & <dbl> & <fct> & <dbl> & <fct> & <dbl> & <dbl> & <dbl> & <int> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\t GCA\\_000379665.1\\_ASM37966v1\\_genomic  & d\\_\\_Bacteria & p\\_\\_Bacteroidota & c\\_\\_Bacteroidia & o\\_\\_Bacteroidales & f\\_\\_Marinifilaceae & g\\_\\_Butyricimonas & s\\_\\_Butyricimonas synergistica & GCF\\_000379665.1 & 95 & ⋯ & 1.00 & GCF\\_000379665.1 & 95 & d\\_\\_Bacteria;p\\_\\_Bacteroidota;c\\_\\_Bacteroidia;o\\_\\_Bacteroidales;f\\_\\_Marinifilaceae;g\\_\\_Butyricimonas;s\\_\\_Butyricimonas synergistica & 100.00 & 1.00 & 96.57 & 11 & N/A & N/A\\\\\n",
       "\t GCA\\_001915615.1\\_ASM191561v1\\_genomic & d\\_\\_Bacteria & p\\_\\_Bacteroidota & c\\_\\_Bacteroidia & o\\_\\_Bacteroidales & f\\_\\_Marinifilaceae & g\\_\\_Butyricimonas & s\\_\\_Butyricimonas faecihominis & GCF\\_003851945.2 & 95 & ⋯ & 0.78 & GCF\\_003851945.2 & 95 & d\\_\\_Bacteria;p\\_\\_Bacteroidota;c\\_\\_Bacteroidia;o\\_\\_Bacteroidales;f\\_\\_Marinifilaceae;g\\_\\_Butyricimonas;s\\_\\_Butyricimonas faecihominis &  98.09 & 0.78 & 84.37 & 11 & N/A & N/A\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋱ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t GCA\\_900184685.1\\_PRJEB20857\\_genomic      & d\\_\\_Bacteria & p\\_\\_Bacteroidota & c\\_\\_Bacteroidia & o\\_\\_Bacteroidales & f\\_\\_Marinifilaceae & g\\_\\_Butyricimonas & s\\_\\_Butyricimonas sp900184685 & GCF\\_900184685.1 & 95 & ⋯ & 1.00 & GCF\\_900184685.1 & 95 & d\\_\\_Bacteria;p\\_\\_Bacteroidota;c\\_\\_Bacteroidia;o\\_\\_Bacteroidales;f\\_\\_Marinifilaceae;g\\_\\_Butyricimonas;s\\_\\_Butyricimonas sp900184685 & 100.00 & 1.00 & 96.71 & 11 & N/A & N/A\\\\\n",
       "\t GCA\\_902364335.1\\_MGYG-HGUT-00203\\_genomic & d\\_\\_Bacteria & p\\_\\_Bacteroidota & c\\_\\_Bacteroidia & o\\_\\_Bacteroidales & f\\_\\_Marinifilaceae & g\\_\\_Butyricimonas & s\\_\\_Butyricimonas virosa      & GCF\\_000519105.1 & 95 & ⋯ & 0.85 & GCF\\_000519105.1 & 95 & d\\_\\_Bacteria;p\\_\\_Bacteroidota;c\\_\\_Bacteroidia;o\\_\\_Bacteroidales;f\\_\\_Marinifilaceae;g\\_\\_Butyricimonas;s\\_\\_Butyricimonas virosa      &  97.98 & 0.85 & 96.79 & 11 & N/A & N/A\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 29 × 22\n",
       "\n",
       "| user_genome &lt;fct&gt; | Domain &lt;chr&gt; | Phylum &lt;chr&gt; | Class &lt;chr&gt; | Order &lt;chr&gt; | Family &lt;chr&gt; | Genus &lt;chr&gt; | Species &lt;chr&gt; | fastani_reference &lt;fct&gt; | fastani_reference_radius &lt;dbl&gt; | ⋯ ⋯ | fastani_af &lt;dbl&gt; | closest_placement_reference &lt;fct&gt; | closest_placement_radius &lt;dbl&gt; | closest_placement_taxonomy &lt;fct&gt; | closest_placement_ani &lt;dbl&gt; | closest_placement_af &lt;dbl&gt; | msa_percent &lt;dbl&gt; | translation_table &lt;int&gt; | red_value &lt;fct&gt; | warnings &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| GCA_000379665.1_ASM37966v1_genomic  | d__Bacteria | p__Bacteroidota | c__Bacteroidia | o__Bacteroidales | f__Marinifilaceae | g__Butyricimonas | s__Butyricimonas synergistica | GCF_000379665.1 | 95 | ⋯ | 1.00 | GCF_000379665.1 | 95 | d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas synergistica | 100.00 | 1.00 | 96.57 | 11 | N/A | N/A |\n",
       "| GCA_001915615.1_ASM191561v1_genomic | d__Bacteria | p__Bacteroidota | c__Bacteroidia | o__Bacteroidales | f__Marinifilaceae | g__Butyricimonas | s__Butyricimonas faecihominis | GCF_003851945.2 | 95 | ⋯ | 0.78 | GCF_003851945.2 | 95 | d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas faecihominis |  98.09 | 0.78 | 84.37 | 11 | N/A | N/A |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋱ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| GCA_900184685.1_PRJEB20857_genomic      | d__Bacteria | p__Bacteroidota | c__Bacteroidia | o__Bacteroidales | f__Marinifilaceae | g__Butyricimonas | s__Butyricimonas sp900184685 | GCF_900184685.1 | 95 | ⋯ | 1.00 | GCF_900184685.1 | 95 | d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas sp900184685 | 100.00 | 1.00 | 96.71 | 11 | N/A | N/A |\n",
       "| GCA_902364335.1_MGYG-HGUT-00203_genomic | d__Bacteria | p__Bacteroidota | c__Bacteroidia | o__Bacteroidales | f__Marinifilaceae | g__Butyricimonas | s__Butyricimonas virosa      | GCF_000519105.1 | 95 | ⋯ | 0.85 | GCF_000519105.1 | 95 | d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas virosa      |  97.98 | 0.85 | 96.79 | 11 | N/A | N/A |\n",
       "\n"
      ],
      "text/plain": [
       "   user_genome                             Domain      Phylum         \n",
       "1  GCA_000379665.1_ASM37966v1_genomic      d__Bacteria p__Bacteroidota\n",
       "2  GCA_001915615.1_ASM191561v1_genomic     d__Bacteria p__Bacteroidota\n",
       "⋮  ⋮                                       ⋮           ⋮              \n",
       "28 GCA_900184685.1_PRJEB20857_genomic      d__Bacteria p__Bacteroidota\n",
       "29 GCA_902364335.1_MGYG-HGUT-00203_genomic d__Bacteria p__Bacteroidota\n",
       "   Class          Order            Family            Genus           \n",
       "1  c__Bacteroidia o__Bacteroidales f__Marinifilaceae g__Butyricimonas\n",
       "2  c__Bacteroidia o__Bacteroidales f__Marinifilaceae g__Butyricimonas\n",
       "⋮  ⋮              ⋮                ⋮                 ⋮               \n",
       "28 c__Bacteroidia o__Bacteroidales f__Marinifilaceae g__Butyricimonas\n",
       "29 c__Bacteroidia o__Bacteroidales f__Marinifilaceae g__Butyricimonas\n",
       "   Species                       fastani_reference fastani_reference_radius ⋯\n",
       "1  s__Butyricimonas synergistica GCF_000379665.1   95                       ⋯\n",
       "2  s__Butyricimonas faecihominis GCF_003851945.2   95                       ⋯\n",
       "⋮  ⋮                             ⋮                 ⋮                        ⋱\n",
       "28 s__Butyricimonas sp900184685  GCF_900184685.1   95                       ⋯\n",
       "29 s__Butyricimonas virosa       GCF_000519105.1   95                       ⋯\n",
       "   fastani_af closest_placement_reference closest_placement_radius\n",
       "1  1.00       GCF_000379665.1             95                      \n",
       "2  0.78       GCF_003851945.2             95                      \n",
       "⋮  ⋮          ⋮                           ⋮                       \n",
       "28 1.00       GCF_900184685.1             95                      \n",
       "29 0.85       GCF_000519105.1             95                      \n",
       "   closest_placement_taxonomy                                                                                                  \n",
       "1  d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas synergistica\n",
       "2  d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas faecihominis\n",
       "⋮  ⋮                                                                                                                           \n",
       "28 d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas sp900184685 \n",
       "29 d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas virosa      \n",
       "   closest_placement_ani closest_placement_af msa_percent translation_table\n",
       "1  100.00                1.00                 96.57       11               \n",
       "2   98.09                0.78                 84.37       11               \n",
       "⋮  ⋮                     ⋮                    ⋮           ⋮                \n",
       "28 100.00                1.00                 96.71       11               \n",
       "29  97.98                0.85                 96.79       11               \n",
       "   red_value warnings\n",
       "1  N/A       N/A     \n",
       "2  N/A       N/A     \n",
       "⋮  ⋮         ⋮       \n",
       "28 N/A       N/A     \n",
       "29 N/A       N/A     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# GTDBTk summary\n",
    "tax = file.path(work_dir, 'LLG', 'gtdbtk', 'gtdbtk_bac_summary.tsv') %>%\n",
    "    read.delim(, sep='\\t') %>%\n",
    "    separate(classification, \n",
    "             c('Domain', 'Phylum', 'Class', 'Order', 'Family', 'Genus', 'Species'),\n",
    "             sep=';') %>%\n",
    "    select(-note, -classification_method, -pplacer_taxonomy,\n",
    "           -other_related_references.genome_id.species_name.radius.ANI.AF.)\n",
    "tax"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-25T05:20:08.495906Z",
     "iopub.status.busy": "2021-04-25T05:20:08.488176Z",
     "iopub.status.idle": "2021-04-25T05:20:08.524784Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- drep --\n",
      "intersect(x,y): 17 \n",
      "setdiff(x,y): 0 \n",
      "setdiff(y,x): 12 \n",
      "union(x,y): 29 \n",
      "-- checkm --\n",
      "intersect(x,y): 29 \n",
      "setdiff(x,y): 0 \n",
      "setdiff(y,x): 0 \n",
      "union(x,y): 29 \n",
      "-- gtdbtk --\n",
      "intersect(x,y): 29 \n",
      "setdiff(x,y): 0 \n",
      "setdiff(y,x): 0 \n",
      "union(x,y): 29 \n"
     ]
    }
   ],
   "source": [
    "# checking overlap\n",
    "cat('-- drep --\\n')\n",
    "overlap(basename(as.character(drep_gen$Fasta)), \n",
    "        basename(as.character(drep$genome)))\n",
    "cat('-- checkm --\\n')\n",
    "overlap(drep$Bin.Id, checkm$Bin.Id)\n",
    "cat('-- gtdbtk --\\n')\n",
    "overlap(drep$Bin.Id, tax$user_genome)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-25T05:20:09.025062Z",
     "iopub.status.busy": "2021-04-25T05:20:09.023219Z",
     "iopub.status.idle": "2021-04-25T05:20:09.094637Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 17 × 60</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>genome</th><th scope=col>completeness</th><th scope=col>contamination</th><th scope=col>Bin.Id</th><th scope=col>Marker.lineage</th><th scope=col>X..genomes</th><th scope=col>X..markers</th><th scope=col>X..marker.sets</th><th scope=col>Completeness</th><th scope=col>Contamination</th><th scope=col>⋯</th><th scope=col>fastani_af</th><th scope=col>closest_placement_reference</th><th scope=col>closest_placement_radius</th><th scope=col>closest_placement_taxonomy</th><th scope=col>closest_placement_ani</th><th scope=col>closest_placement_af</th><th scope=col>msa_percent</th><th scope=col>translation_table</th><th scope=col>red_value</th><th scope=col>warnings</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000379665.1_ASM37966v1_genomic.fna </td><td>100.00</td><td>0</td><td>GCA_000379665.1_ASM37966v1_genomic </td><td>k__Bacteria (UID2569)</td><td>434</td><td>278</td><td>186</td><td>100.00</td><td>0</td><td>⋯</td><td>1.00</td><td>GCF_000379665.1</td><td>95</td><td>d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas synergistica</td><td>100.00</td><td>1.00</td><td>96.57</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_002320175.1_ASM232017v1_genomic.fna</td><td> 98.39</td><td>0</td><td>GCA_002320175.1_ASM232017v1_genomic</td><td>k__Bacteria (UID2569)</td><td>434</td><td>278</td><td>186</td><td> 98.39</td><td>0</td><td>⋯</td><td>0.87</td><td>GCF_000519105.1</td><td>95</td><td>d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas virosa      </td><td> 98.27</td><td>0.87</td><td>94.37</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋱</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_014287455.1_ASM1428745v1_genomic.fna</td><td>100</td><td>0</td><td>GCA_014287455.1_ASM1428745v1_genomic</td><td>k__Bacteria (UID2569)</td><td>434</td><td>278</td><td>186</td><td>100</td><td>0</td><td>⋯</td><td>0.81</td><td>GCF_900184685.1</td><td>95</td><td>d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas sp900184685</td><td> 97.68</td><td>0.81</td><td>96.73</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_900184685.1_PRJEB20857_genomic.fna  </td><td>100</td><td>0</td><td>GCA_900184685.1_PRJEB20857_genomic  </td><td>k__Bacteria (UID2569)</td><td>434</td><td>278</td><td>186</td><td>100</td><td>0</td><td>⋯</td><td>1.00</td><td>GCF_900184685.1</td><td>95</td><td>d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas sp900184685</td><td>100.00</td><td>1.00</td><td>96.71</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 17 × 60\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " genome & completeness & contamination & Bin.Id & Marker.lineage & X..genomes & X..markers & X..marker.sets & Completeness & Contamination & ⋯ & fastani\\_af & closest\\_placement\\_reference & closest\\_placement\\_radius & closest\\_placement\\_taxonomy & closest\\_placement\\_ani & closest\\_placement\\_af & msa\\_percent & translation\\_table & red\\_value & warnings\\\\\n",
       " <fct> & <dbl> & <dbl> & <chr> & <fct> & <int> & <int> & <int> & <dbl> & <dbl> & ⋯ & <dbl> & <fct> & <dbl> & <fct> & <dbl> & <dbl> & <dbl> & <int> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_000379665.1\\_ASM37966v1\\_genomic.fna  & 100.00 & 0 & GCA\\_000379665.1\\_ASM37966v1\\_genomic  & k\\_\\_Bacteria (UID2569) & 434 & 278 & 186 & 100.00 & 0 & ⋯ & 1.00 & GCF\\_000379665.1 & 95 & d\\_\\_Bacteria;p\\_\\_Bacteroidota;c\\_\\_Bacteroidia;o\\_\\_Bacteroidales;f\\_\\_Marinifilaceae;g\\_\\_Butyricimonas;s\\_\\_Butyricimonas synergistica & 100.00 & 1.00 & 96.57 & 11 & N/A & N/A\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_002320175.1\\_ASM232017v1\\_genomic.fna &  98.39 & 0 & GCA\\_002320175.1\\_ASM232017v1\\_genomic & k\\_\\_Bacteria (UID2569) & 434 & 278 & 186 &  98.39 & 0 & ⋯ & 0.87 & GCF\\_000519105.1 & 95 & d\\_\\_Bacteria;p\\_\\_Bacteroidota;c\\_\\_Bacteroidia;o\\_\\_Bacteroidales;f\\_\\_Marinifilaceae;g\\_\\_Butyricimonas;s\\_\\_Butyricimonas virosa       &  98.27 & 0.87 & 94.37 & 11 & N/A & N/A\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋱ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_014287455.1\\_ASM1428745v1\\_genomic.fna & 100 & 0 & GCA\\_014287455.1\\_ASM1428745v1\\_genomic & k\\_\\_Bacteria (UID2569) & 434 & 278 & 186 & 100 & 0 & ⋯ & 0.81 & GCF\\_900184685.1 & 95 & d\\_\\_Bacteria;p\\_\\_Bacteroidota;c\\_\\_Bacteroidia;o\\_\\_Bacteroidales;f\\_\\_Marinifilaceae;g\\_\\_Butyricimonas;s\\_\\_Butyricimonas sp900184685 &  97.68 & 0.81 & 96.73 & 11 & N/A & N/A\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_900184685.1\\_PRJEB20857\\_genomic.fna   & 100 & 0 & GCA\\_900184685.1\\_PRJEB20857\\_genomic   & k\\_\\_Bacteria (UID2569) & 434 & 278 & 186 & 100 & 0 & ⋯ & 1.00 & GCF\\_900184685.1 & 95 & d\\_\\_Bacteria;p\\_\\_Bacteroidota;c\\_\\_Bacteroidia;o\\_\\_Bacteroidales;f\\_\\_Marinifilaceae;g\\_\\_Butyricimonas;s\\_\\_Butyricimonas sp900184685 & 100.00 & 1.00 & 96.71 & 11 & N/A & N/A\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 17 × 60\n",
       "\n",
       "| genome &lt;fct&gt; | completeness &lt;dbl&gt; | contamination &lt;dbl&gt; | Bin.Id &lt;chr&gt; | Marker.lineage &lt;fct&gt; | X..genomes &lt;int&gt; | X..markers &lt;int&gt; | X..marker.sets &lt;int&gt; | Completeness &lt;dbl&gt; | Contamination &lt;dbl&gt; | ⋯ ⋯ | fastani_af &lt;dbl&gt; | closest_placement_reference &lt;fct&gt; | closest_placement_radius &lt;dbl&gt; | closest_placement_taxonomy &lt;fct&gt; | closest_placement_ani &lt;dbl&gt; | closest_placement_af &lt;dbl&gt; | msa_percent &lt;dbl&gt; | translation_table &lt;int&gt; | red_value &lt;fct&gt; | warnings &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000379665.1_ASM37966v1_genomic.fna  | 100.00 | 0 | GCA_000379665.1_ASM37966v1_genomic  | k__Bacteria (UID2569) | 434 | 278 | 186 | 100.00 | 0 | ⋯ | 1.00 | GCF_000379665.1 | 95 | d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas synergistica | 100.00 | 1.00 | 96.57 | 11 | N/A | N/A |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_002320175.1_ASM232017v1_genomic.fna |  98.39 | 0 | GCA_002320175.1_ASM232017v1_genomic | k__Bacteria (UID2569) | 434 | 278 | 186 |  98.39 | 0 | ⋯ | 0.87 | GCF_000519105.1 | 95 | d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas virosa       |  98.27 | 0.87 | 94.37 | 11 | N/A | N/A |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋱ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_014287455.1_ASM1428745v1_genomic.fna | 100 | 0 | GCA_014287455.1_ASM1428745v1_genomic | k__Bacteria (UID2569) | 434 | 278 | 186 | 100 | 0 | ⋯ | 0.81 | GCF_900184685.1 | 95 | d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas sp900184685 |  97.68 | 0.81 | 96.73 | 11 | N/A | N/A |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_900184685.1_PRJEB20857_genomic.fna   | 100 | 0 | GCA_900184685.1_PRJEB20857_genomic   | k__Bacteria (UID2569) | 434 | 278 | 186 | 100 | 0 | ⋯ | 1.00 | GCF_900184685.1 | 95 | d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas sp900184685 | 100.00 | 1.00 | 96.71 | 11 | N/A | N/A |\n",
       "\n"
      ],
      "text/plain": [
       "   genome                                                                                        \n",
       "1  /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000379665.1_ASM37966v1_genomic.fna  \n",
       "2  /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_002320175.1_ASM232017v1_genomic.fna \n",
       "⋮  ⋮                                                                                             \n",
       "16 /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_014287455.1_ASM1428745v1_genomic.fna\n",
       "17 /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_900184685.1_PRJEB20857_genomic.fna  \n",
       "   completeness contamination Bin.Id                              \n",
       "1  100.00       0             GCA_000379665.1_ASM37966v1_genomic  \n",
       "2   98.39       0             GCA_002320175.1_ASM232017v1_genomic \n",
       "⋮  ⋮            ⋮             ⋮                                   \n",
       "16 100          0             GCA_014287455.1_ASM1428745v1_genomic\n",
       "17 100          0             GCA_900184685.1_PRJEB20857_genomic  \n",
       "   Marker.lineage        X..genomes X..markers X..marker.sets Completeness\n",
       "1  k__Bacteria (UID2569) 434        278        186            100.00      \n",
       "2  k__Bacteria (UID2569) 434        278        186             98.39      \n",
       "⋮  ⋮                     ⋮          ⋮          ⋮              ⋮           \n",
       "16 k__Bacteria (UID2569) 434        278        186            100         \n",
       "17 k__Bacteria (UID2569) 434        278        186            100         \n",
       "   Contamination ⋯ fastani_af closest_placement_reference\n",
       "1  0             ⋯ 1.00       GCF_000379665.1            \n",
       "2  0             ⋯ 0.87       GCF_000519105.1            \n",
       "⋮  ⋮             ⋱ ⋮          ⋮                          \n",
       "16 0             ⋯ 0.81       GCF_900184685.1            \n",
       "17 0             ⋯ 1.00       GCF_900184685.1            \n",
       "   closest_placement_radius\n",
       "1  95                      \n",
       "2  95                      \n",
       "⋮  ⋮                       \n",
       "16 95                      \n",
       "17 95                      \n",
       "   closest_placement_taxonomy                                                                                                  \n",
       "1  d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas synergistica\n",
       "2  d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas virosa      \n",
       "⋮  ⋮                                                                                                                           \n",
       "16 d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas sp900184685 \n",
       "17 d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas sp900184685 \n",
       "   closest_placement_ani closest_placement_af msa_percent translation_table\n",
       "1  100.00                1.00                 96.57       11               \n",
       "2   98.27                0.87                 94.37       11               \n",
       "⋮  ⋮                     ⋮                    ⋮           ⋮                \n",
       "16  97.68                0.81                 96.73       11               \n",
       "17 100.00                1.00                 96.71       11               \n",
       "   red_value warnings\n",
       "1  N/A       N/A     \n",
       "2  N/A       N/A     \n",
       "⋮  ⋮         ⋮       \n",
       "16 N/A       N/A     \n",
       "17 N/A       N/A     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# joining based on Bin.Id\n",
    "drep = drep %>%\n",
    "    inner_join(checkm, c('Bin.Id')) %>%\n",
    "    mutate(GEN = genome %>% as.character %>% basename) %>%\n",
    "    inner_join(drep_gen %>% mutate(GEN = Fasta %>% as.character %>% basename),\n",
    "               by=c('GEN')) %>%\n",
    "    inner_join(tax, c('Bin.Id'='user_genome')) #%>%\n",
    "drep"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-25T05:20:09.619333Z",
     "iopub.status.busy": "2021-04-25T05:20:09.617602Z",
     "iopub.status.idle": "2021-04-25T05:20:09.669333Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 17 × 60</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>genome</th><th scope=col>completeness</th><th scope=col>contamination</th><th scope=col>Bin.Id</th><th scope=col>Marker.lineage</th><th scope=col>X..genomes</th><th scope=col>X..markers</th><th scope=col>X..marker.sets</th><th scope=col>Completeness</th><th scope=col>Contamination</th><th scope=col>⋯</th><th scope=col>fastani_af</th><th scope=col>closest_placement_reference</th><th scope=col>closest_placement_radius</th><th scope=col>closest_placement_taxonomy</th><th scope=col>closest_placement_ani</th><th scope=col>closest_placement_af</th><th scope=col>msa_percent</th><th scope=col>translation_table</th><th scope=col>red_value</th><th scope=col>warnings</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000379665.1_ASM37966v1_genomic.fna </td><td>100.00</td><td>0</td><td>GCA_000379665.1_ASM37966v1_genomic </td><td>k__Bacteria (UID2569)</td><td>434</td><td>278</td><td>186</td><td>100.00</td><td>0</td><td>⋯</td><td>1.00</td><td>GCF_000379665.1</td><td>95</td><td>d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas synergistica</td><td>100.00</td><td>1.00</td><td>96.57</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_002320175.1_ASM232017v1_genomic.fna</td><td> 98.39</td><td>0</td><td>GCA_002320175.1_ASM232017v1_genomic</td><td>k__Bacteria (UID2569)</td><td>434</td><td>278</td><td>186</td><td> 98.39</td><td>0</td><td>⋯</td><td>0.87</td><td>GCF_000519105.1</td><td>95</td><td>d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas virosa      </td><td> 98.27</td><td>0.87</td><td>94.37</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋱</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_014287455.1_ASM1428745v1_genomic.fna</td><td>100</td><td>0</td><td>GCA_014287455.1_ASM1428745v1_genomic</td><td>k__Bacteria (UID2569)</td><td>434</td><td>278</td><td>186</td><td>100</td><td>0</td><td>⋯</td><td>0.81</td><td>GCF_900184685.1</td><td>95</td><td>d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas sp900184685</td><td> 97.68</td><td>0.81</td><td>96.73</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_900184685.1_PRJEB20857_genomic.fna  </td><td>100</td><td>0</td><td>GCA_900184685.1_PRJEB20857_genomic  </td><td>k__Bacteria (UID2569)</td><td>434</td><td>278</td><td>186</td><td>100</td><td>0</td><td>⋯</td><td>1.00</td><td>GCF_900184685.1</td><td>95</td><td>d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas sp900184685</td><td>100.00</td><td>1.00</td><td>96.71</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 17 × 60\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " genome & completeness & contamination & Bin.Id & Marker.lineage & X..genomes & X..markers & X..marker.sets & Completeness & Contamination & ⋯ & fastani\\_af & closest\\_placement\\_reference & closest\\_placement\\_radius & closest\\_placement\\_taxonomy & closest\\_placement\\_ani & closest\\_placement\\_af & msa\\_percent & translation\\_table & red\\_value & warnings\\\\\n",
       " <fct> & <dbl> & <dbl> & <chr> & <fct> & <int> & <int> & <int> & <dbl> & <dbl> & ⋯ & <dbl> & <fct> & <dbl> & <fct> & <dbl> & <dbl> & <dbl> & <int> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_000379665.1\\_ASM37966v1\\_genomic.fna  & 100.00 & 0 & GCA\\_000379665.1\\_ASM37966v1\\_genomic  & k\\_\\_Bacteria (UID2569) & 434 & 278 & 186 & 100.00 & 0 & ⋯ & 1.00 & GCF\\_000379665.1 & 95 & d\\_\\_Bacteria;p\\_\\_Bacteroidota;c\\_\\_Bacteroidia;o\\_\\_Bacteroidales;f\\_\\_Marinifilaceae;g\\_\\_Butyricimonas;s\\_\\_Butyricimonas synergistica & 100.00 & 1.00 & 96.57 & 11 & N/A & N/A\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_002320175.1\\_ASM232017v1\\_genomic.fna &  98.39 & 0 & GCA\\_002320175.1\\_ASM232017v1\\_genomic & k\\_\\_Bacteria (UID2569) & 434 & 278 & 186 &  98.39 & 0 & ⋯ & 0.87 & GCF\\_000519105.1 & 95 & d\\_\\_Bacteria;p\\_\\_Bacteroidota;c\\_\\_Bacteroidia;o\\_\\_Bacteroidales;f\\_\\_Marinifilaceae;g\\_\\_Butyricimonas;s\\_\\_Butyricimonas virosa       &  98.27 & 0.87 & 94.37 & 11 & N/A & N/A\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋱ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_014287455.1\\_ASM1428745v1\\_genomic.fna & 100 & 0 & GCA\\_014287455.1\\_ASM1428745v1\\_genomic & k\\_\\_Bacteria (UID2569) & 434 & 278 & 186 & 100 & 0 & ⋯ & 0.81 & GCF\\_900184685.1 & 95 & d\\_\\_Bacteria;p\\_\\_Bacteroidota;c\\_\\_Bacteroidia;o\\_\\_Bacteroidales;f\\_\\_Marinifilaceae;g\\_\\_Butyricimonas;s\\_\\_Butyricimonas sp900184685 &  97.68 & 0.81 & 96.73 & 11 & N/A & N/A\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_900184685.1\\_PRJEB20857\\_genomic.fna   & 100 & 0 & GCA\\_900184685.1\\_PRJEB20857\\_genomic   & k\\_\\_Bacteria (UID2569) & 434 & 278 & 186 & 100 & 0 & ⋯ & 1.00 & GCF\\_900184685.1 & 95 & d\\_\\_Bacteria;p\\_\\_Bacteroidota;c\\_\\_Bacteroidia;o\\_\\_Bacteroidales;f\\_\\_Marinifilaceae;g\\_\\_Butyricimonas;s\\_\\_Butyricimonas sp900184685 & 100.00 & 1.00 & 96.71 & 11 & N/A & N/A\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 17 × 60\n",
       "\n",
       "| genome &lt;fct&gt; | completeness &lt;dbl&gt; | contamination &lt;dbl&gt; | Bin.Id &lt;chr&gt; | Marker.lineage &lt;fct&gt; | X..genomes &lt;int&gt; | X..markers &lt;int&gt; | X..marker.sets &lt;int&gt; | Completeness &lt;dbl&gt; | Contamination &lt;dbl&gt; | ⋯ ⋯ | fastani_af &lt;dbl&gt; | closest_placement_reference &lt;fct&gt; | closest_placement_radius &lt;dbl&gt; | closest_placement_taxonomy &lt;fct&gt; | closest_placement_ani &lt;dbl&gt; | closest_placement_af &lt;dbl&gt; | msa_percent &lt;dbl&gt; | translation_table &lt;int&gt; | red_value &lt;fct&gt; | warnings &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000379665.1_ASM37966v1_genomic.fna  | 100.00 | 0 | GCA_000379665.1_ASM37966v1_genomic  | k__Bacteria (UID2569) | 434 | 278 | 186 | 100.00 | 0 | ⋯ | 1.00 | GCF_000379665.1 | 95 | d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas synergistica | 100.00 | 1.00 | 96.57 | 11 | N/A | N/A |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_002320175.1_ASM232017v1_genomic.fna |  98.39 | 0 | GCA_002320175.1_ASM232017v1_genomic | k__Bacteria (UID2569) | 434 | 278 | 186 |  98.39 | 0 | ⋯ | 0.87 | GCF_000519105.1 | 95 | d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas virosa       |  98.27 | 0.87 | 94.37 | 11 | N/A | N/A |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋱ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_014287455.1_ASM1428745v1_genomic.fna | 100 | 0 | GCA_014287455.1_ASM1428745v1_genomic | k__Bacteria (UID2569) | 434 | 278 | 186 | 100 | 0 | ⋯ | 0.81 | GCF_900184685.1 | 95 | d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas sp900184685 |  97.68 | 0.81 | 96.73 | 11 | N/A | N/A |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_900184685.1_PRJEB20857_genomic.fna   | 100 | 0 | GCA_900184685.1_PRJEB20857_genomic   | k__Bacteria (UID2569) | 434 | 278 | 186 | 100 | 0 | ⋯ | 1.00 | GCF_900184685.1 | 95 | d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas sp900184685 | 100.00 | 1.00 | 96.71 | 11 | N/A | N/A |\n",
       "\n"
      ],
      "text/plain": [
       "   genome                                                                                        \n",
       "1  /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000379665.1_ASM37966v1_genomic.fna  \n",
       "2  /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_002320175.1_ASM232017v1_genomic.fna \n",
       "⋮  ⋮                                                                                             \n",
       "16 /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_014287455.1_ASM1428745v1_genomic.fna\n",
       "17 /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_900184685.1_PRJEB20857_genomic.fna  \n",
       "   completeness contamination Bin.Id                              \n",
       "1  100.00       0             GCA_000379665.1_ASM37966v1_genomic  \n",
       "2   98.39       0             GCA_002320175.1_ASM232017v1_genomic \n",
       "⋮  ⋮            ⋮             ⋮                                   \n",
       "16 100          0             GCA_014287455.1_ASM1428745v1_genomic\n",
       "17 100          0             GCA_900184685.1_PRJEB20857_genomic  \n",
       "   Marker.lineage        X..genomes X..markers X..marker.sets Completeness\n",
       "1  k__Bacteria (UID2569) 434        278        186            100.00      \n",
       "2  k__Bacteria (UID2569) 434        278        186             98.39      \n",
       "⋮  ⋮                     ⋮          ⋮          ⋮              ⋮           \n",
       "16 k__Bacteria (UID2569) 434        278        186            100         \n",
       "17 k__Bacteria (UID2569) 434        278        186            100         \n",
       "   Contamination ⋯ fastani_af closest_placement_reference\n",
       "1  0             ⋯ 1.00       GCF_000379665.1            \n",
       "2  0             ⋯ 0.87       GCF_000519105.1            \n",
       "⋮  ⋮             ⋱ ⋮          ⋮                          \n",
       "16 0             ⋯ 0.81       GCF_900184685.1            \n",
       "17 0             ⋯ 1.00       GCF_900184685.1            \n",
       "   closest_placement_radius\n",
       "1  95                      \n",
       "2  95                      \n",
       "⋮  ⋮                       \n",
       "16 95                      \n",
       "17 95                      \n",
       "   closest_placement_taxonomy                                                                                                  \n",
       "1  d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas synergistica\n",
       "2  d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas virosa      \n",
       "⋮  ⋮                                                                                                                           \n",
       "16 d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas sp900184685 \n",
       "17 d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Marinifilaceae;g__Butyricimonas;s__Butyricimonas sp900184685 \n",
       "   closest_placement_ani closest_placement_af msa_percent translation_table\n",
       "1  100.00                1.00                 96.57       11               \n",
       "2   98.27                0.87                 94.37       11               \n",
       "⋮  ⋮                     ⋮                    ⋮           ⋮                \n",
       "16  97.68                0.81                 96.73       11               \n",
       "17 100.00                1.00                 96.71       11               \n",
       "   red_value warnings\n",
       "1  N/A       N/A     \n",
       "2  N/A       N/A     \n",
       "⋮  ⋮         ⋮       \n",
       "16 N/A       N/A     \n",
       "17 N/A       N/A     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# filtering by quality\n",
    "hq_genomes = drep %>%\n",
    "    filter(completeness >= 90,\n",
    "           contamination < 5,\n",
    "           Strain.heterogeneity < 50)\n",
    "hq_genomes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-25T05:20:10.197205Z",
     "iopub.status.busy": "2021-04-25T05:20:10.195398Z",
     "iopub.status.idle": "2021-04-25T05:20:10.227754Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A tibble: 1 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Family</th><th scope=col>Genus</th><th scope=col>n_genomes</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>f__Marinifilaceae</td><td>g__Butyricimonas</td><td>17</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 1 × 3\n",
       "\\begin{tabular}{lll}\n",
       " Family & Genus & n\\_genomes\\\\\n",
       " <chr> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t f\\_\\_Marinifilaceae & g\\_\\_Butyricimonas & 17\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 1 × 3\n",
       "\n",
       "| Family &lt;chr&gt; | Genus &lt;chr&gt; | n_genomes &lt;int&gt; |\n",
       "|---|---|---|\n",
       "| f__Marinifilaceae | g__Butyricimonas | 17 |\n",
       "\n"
      ],
      "text/plain": [
       "  Family            Genus            n_genomes\n",
       "1 f__Marinifilaceae g__Butyricimonas 17       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# summarizing the taxonomy\n",
    "df.dims(20)\n",
    "hq_genomes %>%\n",
    "    group_by(Family, Genus) %>%\n",
    "    summarize(n_genomes = n(), .groups='drop')\n",
    "df.dims()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-25T05:20:10.795233Z",
     "iopub.status.busy": "2021-04-25T05:20:10.793201Z",
     "iopub.status.idle": "2021-04-25T05:20:10.821320Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "File written: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3//Butyricimonas/samples_genomes_hq.txt \n"
     ]
    }
   ],
   "source": [
    "# writing samples table for LLPRIMER\n",
    "outfile = file.path(work_dir, 'samples_genomes_hq.txt')\n",
    "hq_genomes %>%\n",
    "    select(Bin.Id, Fasta) %>%\n",
    "    rename('Taxon' = Bin.Id) %>%\n",
    "    mutate(Taxon = gsub('_chromosome.+', '', Taxon),\n",
    "           Taxon = gsub('_bin_.+', '', Taxon),\n",
    "           Taxon = gsub('_genomic', '', Taxon),\n",
    "           Taxon = gsub('_annotated_assembly', '', Taxon),\n",
    "           Taxid = taxid,\n",
    "           Domain = domain) %>%\n",
    "    write_table(outfile)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Primer design"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Config"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-25T05:20:12.928102Z",
     "iopub.status.busy": "2021-04-25T05:20:12.926027Z",
     "iopub.status.idle": "2021-04-25T05:20:12.985718Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#-- I/O --#\n",
      "samples_file: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricimonas/samples_genomes_hq.txt\n",
      "\n",
      "# output location\n",
      "output_dir: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricimonas/primers/\n",
      "\n",
      "# temporary file directory (your username will be added automatically)\n",
      "tmp_dir: /ebio/abt3_scratch/\n",
      "\n",
      "#-- software parameters --#\n",
      "# See the README for a description\n",
      "params:\n",
      "  ionice: -c 3\n",
      "  cgp:\n",
      "    prokka: \"\"\n",
      "    mmseqs:\n",
      "      method: cluster    # or linclust (faster)\n",
      "      run: --min-seq-id 0.8 -c 0.8\n",
      "    vsearch: --id 0.94\n",
      "    core_genes:\n",
      "      cds: --perc-genomes-cds 100 --copies-per-genome-cds 1 --max-clusters-cds 500\n",
      "      rRNA: --perc-genomes-rrna 100 --copies-per-genome-rrna 10 --max-clusters-rrna 500\n",
      "    align:\n",
      "      method: linsi\n",
      "      params: --auto --maxiterate 1000\n",
      "    primer3:\n",
      "      consensus: --consensus-threshold 0.34\n",
      "      number: --num-raw-primers 1000 --num-final-primers 10\n",
      "      size: --opt-size 20 --min-size 18 --max-size 26\n",
      "      product: --opt-prod-size 150 --min-prod-size 100 --max-prod-size 250\n",
      "      Tm: --opt-tm 62 --min-tm 54 --max-tm 70 --max-tm-diff 2.5\n",
      "      PCR: --oligo-DNA 50 --salt-monovalent 50 --salt-divalent 1.5 --dNTPs 0.2\n",
      "      GC: --opt-gc 50 --min-gc 25 --max-gc 75\n",
      "      degeneracy: --max-degeneracy 128 --max-degeneracy-3prime 6 --window-3prime 5\n",
      "      internal_oligo: --make-oligo 0 --int-opt-size 20 --int-opt-tm 60\n",
      "    blast_nontarget:\n",
      "      run_locally: False\n",
      "      cds: -evalue 1e-5 -max_target_seqs 30\n",
      "      rRNA: -evalue 1e-5 -max_target_seqs 30\n",
      "    blastn: -max_target_seqs 500 -perc_identity 90\n",
      "    blast_filter: --perc-len 80 --min-amplicon-len 30 --max-amplicon-len 2000\n",
      "    seqkit_amplicon: \"\"\n",
      "\n",
      "#-- database --#\n",
      "databases:\n",
      "  blast:\n",
      "    nucl: /ebio/abt3_projects/databases_no-backup/NCBI_blastdb/nt\n",
      "    prot: /ebio/abt3_projects/databases_no-backup/NCBI_blastdb/nr\n",
      "    rRNA:\n",
      "      SSU: /ebio/abt3_projects/databases_no-backup/NCBI_rRNA/2021-01/SSU/16S_ribosomal_RNA\n",
      "      LSU: /ebio/abt3_projects/databases_no-backup/NCBI_rRNA/2021-01/LSU/LSU_prokaryote_rRNA\n",
      "    gene_names: /ebio/abt3_projects/databases_no-backup/NCBI_nr_geneIDs/2021-01/nr_accession-geneNames.pkl\n",
      "  taxonomy:\n",
      "    names: /ebio/abt3_projects/databases_no-backup/NCBI_taxonomy/2020-12-23/names.dmp\n",
      "    nodes: /ebio/abt3_projects/databases_no-backup/NCBI_taxonomy/2020-12-23/nodes.dmp\n",
      "\n",
      "#-- snakemake pipeline --#\n",
      "pipeline:\n",
      "  snakemake_folder: ./\n",
      "  script_folder: bin/scripts/\n",
      "  name: LLPRIMER"
     ]
    }
   ],
   "source": [
    "F = file.path(work_dir, 'primers', 'config.yaml')\n",
    "cat_file(F)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Run"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "(snakemake) @ rick:/ebio/abt3_projects/software/dev/ll_pipelines/llprimer\n",
    "$ screen -L -S llprimer-But ./snakemake_sge.sh /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricimonas/primers/config.yaml 50 -F --notemp\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Summary"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Primers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-28T12:42:28.359361Z",
     "iopub.status.busy": "2021-04-28T12:42:28.356498Z",
     "iopub.status.idle": "2021-04-28T12:42:28.532694Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No. of unique primers: 991 \n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 9308 × 22</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>gene_type</th><th scope=col>cluster_id</th><th scope=col>primer_set</th><th scope=col>amplicon_size_consensus</th><th scope=col>amplicon_size_avg</th><th scope=col>amplicon_size_sd</th><th scope=col>primer_id</th><th scope=col>primer_type</th><th scope=col>sequence</th><th scope=col>length</th><th scope=col>⋯</th><th scope=col>position_start</th><th scope=col>position_end</th><th scope=col>Tm_avg</th><th scope=col>Tm_sd</th><th scope=col>GC_avg</th><th scope=col>GC_sd</th><th scope=col>hairpin_avg</th><th scope=col>hairpin_sd</th><th scope=col>homodimer_avg</th><th scope=col>homodimer_sd</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>cds</td><td>1</td><td>286</td><td>201</td><td>201</td><td>0</td><td>286f</td><td>PRIMER_LEFT </td><td>TRCGYTCGKCTYTGGTDCCG</td><td>20</td><td>⋯</td><td>445</td><td>465</td><td>64.26734</td><td>2.827061</td><td>61.66667</td><td>5.527708</td><td>28.84957</td><td>27.433750</td><td>  3.172741</td><td>15.30954</td></tr>\n",
       "\t<tr><td>cds</td><td>1</td><td>286</td><td>201</td><td>201</td><td>0</td><td>286r</td><td>PRIMER_RIGHT</td><td>TVCGYAACCAWGCCAAHGCC</td><td>20</td><td>⋯</td><td>626</td><td>646</td><td>63.56097</td><td>2.428073</td><td>57.50000</td><td>4.166667</td><td>42.82996</td><td> 6.396477</td><td>-16.432855</td><td>22.12949</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋱</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>cds</td><td>500</td><td>874</td><td>117</td><td>117</td><td>0</td><td>874f</td><td>PRIMER_LEFT </td><td>ATWGAYGCSGGRCCYGAYTT  </td><td>20</td><td>⋯</td><td>225</td><td>245</td><td>62.45535</td><td>2.487909</td><td>55.00000</td><td>5.000000</td><td>23.95850</td><td>24.20188</td><td> 23.27590</td><td>16.75870</td></tr>\n",
       "\t<tr><td>cds</td><td>500</td><td>874</td><td>117</td><td>117</td><td>0</td><td>874r</td><td>PRIMER_RIGHT</td><td>YCGWATATCRTCCARKCCYCCG</td><td>22</td><td>⋯</td><td>320</td><td>342</td><td>62.67829</td><td>2.221396</td><td>56.81818</td><td>5.081973</td><td>19.97123</td><td>19.99833</td><td>-26.80071</td><td>22.36798</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 9308 × 22\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " gene\\_type & cluster\\_id & primer\\_set & amplicon\\_size\\_consensus & amplicon\\_size\\_avg & amplicon\\_size\\_sd & primer\\_id & primer\\_type & sequence & length & ⋯ & position\\_start & position\\_end & Tm\\_avg & Tm\\_sd & GC\\_avg & GC\\_sd & hairpin\\_avg & hairpin\\_sd & homodimer\\_avg & homodimer\\_sd\\\\\n",
       " <fct> & <int> & <int> & <int> & <dbl> & <dbl> & <fct> & <fct> & <fct> & <int> & ⋯ & <int> & <int> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t cds & 1 & 286 & 201 & 201 & 0 & 286f & PRIMER\\_LEFT  & TRCGYTCGKCTYTGGTDCCG & 20 & ⋯ & 445 & 465 & 64.26734 & 2.827061 & 61.66667 & 5.527708 & 28.84957 & 27.433750 &   3.172741 & 15.30954\\\\\n",
       "\t cds & 1 & 286 & 201 & 201 & 0 & 286r & PRIMER\\_RIGHT & TVCGYAACCAWGCCAAHGCC & 20 & ⋯ & 626 & 646 & 63.56097 & 2.428073 & 57.50000 & 4.166667 & 42.82996 &  6.396477 & -16.432855 & 22.12949\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋱ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t cds & 500 & 874 & 117 & 117 & 0 & 874f & PRIMER\\_LEFT  & ATWGAYGCSGGRCCYGAYTT   & 20 & ⋯ & 225 & 245 & 62.45535 & 2.487909 & 55.00000 & 5.000000 & 23.95850 & 24.20188 &  23.27590 & 16.75870\\\\\n",
       "\t cds & 500 & 874 & 117 & 117 & 0 & 874r & PRIMER\\_RIGHT & YCGWATATCRTCCARKCCYCCG & 22 & ⋯ & 320 & 342 & 62.67829 & 2.221396 & 56.81818 & 5.081973 & 19.97123 & 19.99833 & -26.80071 & 22.36798\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 9308 × 22\n",
       "\n",
       "| gene_type &lt;fct&gt; | cluster_id &lt;int&gt; | primer_set &lt;int&gt; | amplicon_size_consensus &lt;int&gt; | amplicon_size_avg &lt;dbl&gt; | amplicon_size_sd &lt;dbl&gt; | primer_id &lt;fct&gt; | primer_type &lt;fct&gt; | sequence &lt;fct&gt; | length &lt;int&gt; | ⋯ ⋯ | position_start &lt;int&gt; | position_end &lt;int&gt; | Tm_avg &lt;dbl&gt; | Tm_sd &lt;dbl&gt; | GC_avg &lt;dbl&gt; | GC_sd &lt;dbl&gt; | hairpin_avg &lt;dbl&gt; | hairpin_sd &lt;dbl&gt; | homodimer_avg &lt;dbl&gt; | homodimer_sd &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| cds | 1 | 286 | 201 | 201 | 0 | 286f | PRIMER_LEFT  | TRCGYTCGKCTYTGGTDCCG | 20 | ⋯ | 445 | 465 | 64.26734 | 2.827061 | 61.66667 | 5.527708 | 28.84957 | 27.433750 |   3.172741 | 15.30954 |\n",
       "| cds | 1 | 286 | 201 | 201 | 0 | 286r | PRIMER_RIGHT | TVCGYAACCAWGCCAAHGCC | 20 | ⋯ | 626 | 646 | 63.56097 | 2.428073 | 57.50000 | 4.166667 | 42.82996 |  6.396477 | -16.432855 | 22.12949 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋱ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| cds | 500 | 874 | 117 | 117 | 0 | 874f | PRIMER_LEFT  | ATWGAYGCSGGRCCYGAYTT   | 20 | ⋯ | 225 | 245 | 62.45535 | 2.487909 | 55.00000 | 5.000000 | 23.95850 | 24.20188 |  23.27590 | 16.75870 |\n",
       "| cds | 500 | 874 | 117 | 117 | 0 | 874r | PRIMER_RIGHT | YCGWATATCRTCCARKCCYCCG | 22 | ⋯ | 320 | 342 | 62.67829 | 2.221396 | 56.81818 | 5.081973 | 19.97123 | 19.99833 | -26.80071 | 22.36798 |\n",
       "\n"
      ],
      "text/plain": [
       "     gene_type cluster_id primer_set amplicon_size_consensus amplicon_size_avg\n",
       "1    cds       1          286        201                     201              \n",
       "2    cds       1          286        201                     201              \n",
       "⋮    ⋮         ⋮          ⋮          ⋮                       ⋮                \n",
       "9307 cds       500        874        117                     117              \n",
       "9308 cds       500        874        117                     117              \n",
       "     amplicon_size_sd primer_id primer_type  sequence               length ⋯\n",
       "1    0                286f      PRIMER_LEFT  TRCGYTCGKCTYTGGTDCCG   20     ⋯\n",
       "2    0                286r      PRIMER_RIGHT TVCGYAACCAWGCCAAHGCC   20     ⋯\n",
       "⋮    ⋮                ⋮         ⋮            ⋮                      ⋮      ⋱\n",
       "9307 0                874f      PRIMER_LEFT  ATWGAYGCSGGRCCYGAYTT   20     ⋯\n",
       "9308 0                874r      PRIMER_RIGHT YCGWATATCRTCCARKCCYCCG 22     ⋯\n",
       "     position_start position_end Tm_avg   Tm_sd    GC_avg   GC_sd   \n",
       "1    445            465          64.26734 2.827061 61.66667 5.527708\n",
       "2    626            646          63.56097 2.428073 57.50000 4.166667\n",
       "⋮    ⋮              ⋮            ⋮        ⋮        ⋮        ⋮       \n",
       "9307 225            245          62.45535 2.487909 55.00000 5.000000\n",
       "9308 320            342          62.67829 2.221396 56.81818 5.081973\n",
       "     hairpin_avg hairpin_sd homodimer_avg homodimer_sd\n",
       "1    28.84957    27.433750    3.172741    15.30954    \n",
       "2    42.82996     6.396477  -16.432855    22.12949    \n",
       "⋮    ⋮           ⋮          ⋮             ⋮           \n",
       "9307 23.95850    24.20188    23.27590     16.75870    \n",
       "9308 19.97123    19.99833   -26.80071     22.36798    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "primer_info = read.delim(file.path(work_dir, 'primers', 'cgp', 'primers_final_info.tsv'), sep='\\t')\n",
    "primer_info %>% unique_n('primers', primer_set)\n",
    "primer_info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Gene clusters"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-28T12:42:39.995035Z",
     "iopub.status.busy": "2021-04-28T12:42:39.992788Z",
     "iopub.status.idle": "2021-04-28T12:42:40.175104Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No. of clusters 484 \n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 8228 × 13</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>gene_type</th><th scope=col>cluster_id</th><th scope=col>seq_uuid</th><th scope=col>seq_orig_name</th><th scope=col>contig_id</th><th scope=col>taxon</th><th scope=col>start</th><th scope=col>end</th><th scope=col>score</th><th scope=col>strand</th><th scope=col>annotation</th><th scope=col>cluster_name</th><th scope=col>clust_id</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>cds</td><td>1</td><td>ba6e70d9efab4a67a21364b8d0c36599</td><td>APGDNBDC_03747</td><td>KB903373.1    </td><td>GCA_000379665.1_ASM37966v1 </td><td>134880</td><td>135569</td><td>.</td><td>+</td><td>hypothetical protein</td><td>566ebf93955f48b194569f670baaa7b5</td><td>1</td></tr>\n",
       "\t<tr><td>cds</td><td>1</td><td>1db21bc2cb8349a58f0c7bed9cb8e7d7</td><td>DINPFHMG_00394</td><td>DCJY01000025.1</td><td>GCA_002320175.1_ASM232017v1</td><td>   207</td><td>   947</td><td>.</td><td>-</td><td>hypothetical protein</td><td>566ebf93955f48b194569f670baaa7b5</td><td>1</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>cds</td><td>500</td><td>4d4a337be6aa4cdf8e30cbfc38b85254</td><td>DGJGCJFC_01250</td><td>JACOOH010000002.1</td><td>GCA_014287455.1_ASM1428745v1</td><td>624572</td><td>625396</td><td>.</td><td>-</td><td>Phosphoribosyl 1%2C2-cyclic phosphate phosphodiesterase</td><td>f8df3fd36e314e6f995d3bb8d84fe5ef</td><td>500</td></tr>\n",
       "\t<tr><td>cds</td><td>500</td><td>5c0e97d5e7e04f2b917dd10c4a233d61</td><td>HIBIFNEK_00668</td><td>FXXL01000007.1   </td><td>GCA_900184685.1_PRJEB20857  </td><td>762197</td><td>763021</td><td>.</td><td>-</td><td>Phosphoribosyl 1%2C2-cyclic phosphate phosphodiesterase</td><td>f8df3fd36e314e6f995d3bb8d84fe5ef</td><td>500</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 8228 × 13\n",
       "\\begin{tabular}{lllllllllllll}\n",
       " gene\\_type & cluster\\_id & seq\\_uuid & seq\\_orig\\_name & contig\\_id & taxon & start & end & score & strand & annotation & cluster\\_name & clust\\_id\\\\\n",
       " <fct> & <int> & <fct> & <fct> & <fct> & <fct> & <int> & <int> & <fct> & <fct> & <fct> & <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t cds & 1 & ba6e70d9efab4a67a21364b8d0c36599 & APGDNBDC\\_03747 & KB903373.1     & GCA\\_000379665.1\\_ASM37966v1  & 134880 & 135569 & . & + & hypothetical protein & 566ebf93955f48b194569f670baaa7b5 & 1\\\\\n",
       "\t cds & 1 & 1db21bc2cb8349a58f0c7bed9cb8e7d7 & DINPFHMG\\_00394 & DCJY01000025.1 & GCA\\_002320175.1\\_ASM232017v1 &    207 &    947 & . & - & hypothetical protein & 566ebf93955f48b194569f670baaa7b5 & 1\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t cds & 500 & 4d4a337be6aa4cdf8e30cbfc38b85254 & DGJGCJFC\\_01250 & JACOOH010000002.1 & GCA\\_014287455.1\\_ASM1428745v1 & 624572 & 625396 & . & - & Phosphoribosyl 1\\%2C2-cyclic phosphate phosphodiesterase & f8df3fd36e314e6f995d3bb8d84fe5ef & 500\\\\\n",
       "\t cds & 500 & 5c0e97d5e7e04f2b917dd10c4a233d61 & HIBIFNEK\\_00668 & FXXL01000007.1    & GCA\\_900184685.1\\_PRJEB20857   & 762197 & 763021 & . & - & Phosphoribosyl 1\\%2C2-cyclic phosphate phosphodiesterase & f8df3fd36e314e6f995d3bb8d84fe5ef & 500\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 8228 × 13\n",
       "\n",
       "| gene_type &lt;fct&gt; | cluster_id &lt;int&gt; | seq_uuid &lt;fct&gt; | seq_orig_name &lt;fct&gt; | contig_id &lt;fct&gt; | taxon &lt;fct&gt; | start &lt;int&gt; | end &lt;int&gt; | score &lt;fct&gt; | strand &lt;fct&gt; | annotation &lt;fct&gt; | cluster_name &lt;fct&gt; | clust_id &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| cds | 1 | ba6e70d9efab4a67a21364b8d0c36599 | APGDNBDC_03747 | KB903373.1     | GCA_000379665.1_ASM37966v1  | 134880 | 135569 | . | + | hypothetical protein | 566ebf93955f48b194569f670baaa7b5 | 1 |\n",
       "| cds | 1 | 1db21bc2cb8349a58f0c7bed9cb8e7d7 | DINPFHMG_00394 | DCJY01000025.1 | GCA_002320175.1_ASM232017v1 |    207 |    947 | . | - | hypothetical protein | 566ebf93955f48b194569f670baaa7b5 | 1 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| cds | 500 | 4d4a337be6aa4cdf8e30cbfc38b85254 | DGJGCJFC_01250 | JACOOH010000002.1 | GCA_014287455.1_ASM1428745v1 | 624572 | 625396 | . | - | Phosphoribosyl 1%2C2-cyclic phosphate phosphodiesterase | f8df3fd36e314e6f995d3bb8d84fe5ef | 500 |\n",
       "| cds | 500 | 5c0e97d5e7e04f2b917dd10c4a233d61 | HIBIFNEK_00668 | FXXL01000007.1    | GCA_900184685.1_PRJEB20857   | 762197 | 763021 | . | - | Phosphoribosyl 1%2C2-cyclic phosphate phosphodiesterase | f8df3fd36e314e6f995d3bb8d84fe5ef | 500 |\n",
       "\n"
      ],
      "text/plain": [
       "     gene_type cluster_id seq_uuid                         seq_orig_name \n",
       "1    cds       1          ba6e70d9efab4a67a21364b8d0c36599 APGDNBDC_03747\n",
       "2    cds       1          1db21bc2cb8349a58f0c7bed9cb8e7d7 DINPFHMG_00394\n",
       "⋮    ⋮         ⋮          ⋮                                ⋮             \n",
       "8227 cds       500        4d4a337be6aa4cdf8e30cbfc38b85254 DGJGCJFC_01250\n",
       "8228 cds       500        5c0e97d5e7e04f2b917dd10c4a233d61 HIBIFNEK_00668\n",
       "     contig_id         taxon                        start  end    score strand\n",
       "1    KB903373.1        GCA_000379665.1_ASM37966v1   134880 135569 .     +     \n",
       "2    DCJY01000025.1    GCA_002320175.1_ASM232017v1     207    947 .     -     \n",
       "⋮    ⋮                 ⋮                            ⋮      ⋮      ⋮     ⋮     \n",
       "8227 JACOOH010000002.1 GCA_014287455.1_ASM1428745v1 624572 625396 .     -     \n",
       "8228 FXXL01000007.1    GCA_900184685.1_PRJEB20857   762197 763021 .     -     \n",
       "     annotation                                             \n",
       "1    hypothetical protein                                   \n",
       "2    hypothetical protein                                   \n",
       "⋮    ⋮                                                      \n",
       "8227 Phosphoribosyl 1%2C2-cyclic phosphate phosphodiesterase\n",
       "8228 Phosphoribosyl 1%2C2-cyclic phosphate phosphodiesterase\n",
       "     cluster_name                     clust_id\n",
       "1    566ebf93955f48b194569f670baaa7b5 1       \n",
       "2    566ebf93955f48b194569f670baaa7b5 1       \n",
       "⋮    ⋮                                ⋮       \n",
       "8227 f8df3fd36e314e6f995d3bb8d84fe5ef 500     \n",
       "8228 f8df3fd36e314e6f995d3bb8d84fe5ef 500     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# target gene info\n",
    "gene_annot = read.delim(file.path(work_dir, 'primers', 'cgp', 'core_clusters_info.tsv'), \n",
    "                        sep='\\t') %>%\n",
    "    semi_join(primer_info, c('cluster_id')) \n",
    "gene_annot %>% distinct(gene_type, cluster_id) %>% nrow %>% cat('No. of clusters', ., '\\n')\n",
    "gene_annot"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Gene cluster annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-28T12:42:52.222235Z",
     "iopub.status.busy": "2021-04-28T12:42:52.220248Z",
     "iopub.status.idle": "2021-04-28T12:42:52.554379Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 14573 × 17</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>cluster_id</th><th scope=col>query</th><th scope=col>subject</th><th scope=col>subject_name</th><th scope=col>pident</th><th scope=col>length</th><th scope=col>mismatch</th><th scope=col>qstart</th><th scope=col>qend</th><th scope=col>sstart</th><th scope=col>send</th><th scope=col>evalue</th><th scope=col>slen</th><th scope=col>qlen</th><th scope=col>sscinames</th><th scope=col>staxids</th><th scope=col>pident_rank</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td>566ebf93955f48b194569f670baaa7b5</td><td>WP_118714828.1</td><td>zinc metallopeptidase [Odoribacter sp. AF15-53]</td><td>94.323</td><td>229</td><td>13</td><td>100</td><td>786</td><td>1</td><td>229</td><td>2.25e-131</td><td>229</td><td>789</td><td>Odoribacter sp. AF15-53                        </td><td>2292236        </td><td>1</td></tr>\n",
       "\t<tr><td>1</td><td>566ebf93955f48b194569f670baaa7b5</td><td>RGG45931.1    </td><td>None                                           </td><td>94.323</td><td>229</td><td>13</td><td>100</td><td>786</td><td>1</td><td>229</td><td>2.15e-130</td><td>229</td><td>789</td><td>Odoribacter sp. AM16-33;Odoribacter sp. AF21-41</td><td>2292048;2293111</td><td>1</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>500</td><td>f8df3fd36e314e6f995d3bb8d84fe5ef</td><td>WP_055130916.1</td><td>MBL fold metallo-hydrolase [Pedobacter sp. Hv1]</td><td>54.150</td><td>253</td><td>116</td><td>1</td><td>759</td><td>1</td><td>253</td><td>1.39e-104</td><td>254</td><td>762</td><td>Pedobacter sp. Hv1</td><td>1740090</td><td>30</td></tr>\n",
       "\t<tr><td>500</td><td>f8df3fd36e314e6f995d3bb8d84fe5ef</td><td>WP_110360170.1</td><td>MBL fold metallo-hydrolase [Marinifilum breve] </td><td>55.159</td><td>252</td><td>113</td><td>1</td><td>756</td><td>2</td><td>253</td><td>1.50e-104</td><td>254</td><td>762</td><td>Marinifilum breve </td><td>2184082</td><td>26</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 14573 × 17\n",
       "\\begin{tabular}{lllllllllllllllll}\n",
       " cluster\\_id & query & subject & subject\\_name & pident & length & mismatch & qstart & qend & sstart & send & evalue & slen & qlen & sscinames & staxids & pident\\_rank\\\\\n",
       " <int> & <fct> & <fct> & <fct> & <dbl> & <int> & <int> & <int> & <int> & <int> & <int> & <dbl> & <int> & <int> & <fct> & <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t 1 & 566ebf93955f48b194569f670baaa7b5 & WP\\_118714828.1 & zinc metallopeptidase {[}Odoribacter sp. AF15-53{]} & 94.323 & 229 & 13 & 100 & 786 & 1 & 229 & 2.25e-131 & 229 & 789 & Odoribacter sp. AF15-53                         & 2292236         & 1\\\\\n",
       "\t 1 & 566ebf93955f48b194569f670baaa7b5 & RGG45931.1     & None                                            & 94.323 & 229 & 13 & 100 & 786 & 1 & 229 & 2.15e-130 & 229 & 789 & Odoribacter sp. AM16-33;Odoribacter sp. AF21-41 & 2292048;2293111 & 1\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t 500 & f8df3fd36e314e6f995d3bb8d84fe5ef & WP\\_055130916.1 & MBL fold metallo-hydrolase {[}Pedobacter sp. Hv1{]} & 54.150 & 253 & 116 & 1 & 759 & 1 & 253 & 1.39e-104 & 254 & 762 & Pedobacter sp. Hv1 & 1740090 & 30\\\\\n",
       "\t 500 & f8df3fd36e314e6f995d3bb8d84fe5ef & WP\\_110360170.1 & MBL fold metallo-hydrolase {[}Marinifilum breve{]}  & 55.159 & 252 & 113 & 1 & 756 & 2 & 253 & 1.50e-104 & 254 & 762 & Marinifilum breve  & 2184082 & 26\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 14573 × 17\n",
       "\n",
       "| cluster_id &lt;int&gt; | query &lt;fct&gt; | subject &lt;fct&gt; | subject_name &lt;fct&gt; | pident &lt;dbl&gt; | length &lt;int&gt; | mismatch &lt;int&gt; | qstart &lt;int&gt; | qend &lt;int&gt; | sstart &lt;int&gt; | send &lt;int&gt; | evalue &lt;dbl&gt; | slen &lt;int&gt; | qlen &lt;int&gt; | sscinames &lt;fct&gt; | staxids &lt;fct&gt; | pident_rank &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 566ebf93955f48b194569f670baaa7b5 | WP_118714828.1 | zinc metallopeptidase [Odoribacter sp. AF15-53] | 94.323 | 229 | 13 | 100 | 786 | 1 | 229 | 2.25e-131 | 229 | 789 | Odoribacter sp. AF15-53                         | 2292236         | 1 |\n",
       "| 1 | 566ebf93955f48b194569f670baaa7b5 | RGG45931.1     | None                                            | 94.323 | 229 | 13 | 100 | 786 | 1 | 229 | 2.15e-130 | 229 | 789 | Odoribacter sp. AM16-33;Odoribacter sp. AF21-41 | 2292048;2293111 | 1 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| 500 | f8df3fd36e314e6f995d3bb8d84fe5ef | WP_055130916.1 | MBL fold metallo-hydrolase [Pedobacter sp. Hv1] | 54.150 | 253 | 116 | 1 | 759 | 1 | 253 | 1.39e-104 | 254 | 762 | Pedobacter sp. Hv1 | 1740090 | 30 |\n",
       "| 500 | f8df3fd36e314e6f995d3bb8d84fe5ef | WP_110360170.1 | MBL fold metallo-hydrolase [Marinifilum breve]  | 55.159 | 252 | 113 | 1 | 756 | 2 | 253 | 1.50e-104 | 254 | 762 | Marinifilum breve  | 2184082 | 26 |\n",
       "\n"
      ],
      "text/plain": [
       "      cluster_id query                            subject       \n",
       "1     1          566ebf93955f48b194569f670baaa7b5 WP_118714828.1\n",
       "2     1          566ebf93955f48b194569f670baaa7b5 RGG45931.1    \n",
       "⋮     ⋮          ⋮                                ⋮             \n",
       "14572 500        f8df3fd36e314e6f995d3bb8d84fe5ef WP_055130916.1\n",
       "14573 500        f8df3fd36e314e6f995d3bb8d84fe5ef WP_110360170.1\n",
       "      subject_name                                    pident length mismatch\n",
       "1     zinc metallopeptidase [Odoribacter sp. AF15-53] 94.323 229    13      \n",
       "2     None                                            94.323 229    13      \n",
       "⋮     ⋮                                               ⋮      ⋮      ⋮       \n",
       "14572 MBL fold metallo-hydrolase [Pedobacter sp. Hv1] 54.150 253    116     \n",
       "14573 MBL fold metallo-hydrolase [Marinifilum breve]  55.159 252    113     \n",
       "      qstart qend sstart send evalue    slen qlen\n",
       "1     100    786  1      229  2.25e-131 229  789 \n",
       "2     100    786  1      229  2.15e-130 229  789 \n",
       "⋮     ⋮      ⋮    ⋮      ⋮    ⋮         ⋮    ⋮   \n",
       "14572 1      759  1      253  1.39e-104 254  762 \n",
       "14573 1      756  2      253  1.50e-104 254  762 \n",
       "      sscinames                                       staxids        \n",
       "1     Odoribacter sp. AF15-53                         2292236        \n",
       "2     Odoribacter sp. AM16-33;Odoribacter sp. AF21-41 2292048;2293111\n",
       "⋮     ⋮                                               ⋮              \n",
       "14572 Pedobacter sp. Hv1                              1740090        \n",
       "14573 Marinifilum breve                               2184082        \n",
       "      pident_rank\n",
       "1     1          \n",
       "2     1          \n",
       "⋮     ⋮          \n",
       "14572 30         \n",
       "14573 26         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# non-target cds gene annotations\n",
    "gene_nontarget = read.delim(file.path(work_dir, 'primers', 'cgp', 'nontarget', 'cds_blastx.tsv'), \n",
    "                        sep='\\t') %>%\n",
    "    semi_join(primer_info, c('cluster_id')) \n",
    "gene_nontarget"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Cluster taxonomic uniqueness"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-28T12:43:12.080330Z",
     "iopub.status.busy": "2021-04-28T12:43:12.078533Z",
     "iopub.status.idle": "2021-04-28T12:43:12.103537Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 1 × 8 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Min.</th><th scope=col>1st Qu.</th><th scope=col>Median</th><th scope=col>Mean</th><th scope=col>3rd Qu.</th><th scope=col>Max.</th><th scope=col>sd</th><th scope=col>sd_err_of_mean</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>seq-ID to closest non-target</th><td>81.107</td><td>92.44225</td><td>96.508</td><td>94.98219</td><td>98.60025</td><td>99.813</td><td>6.794</td><td>2.774</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 1 × 8 of type dbl\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & Min. & 1st Qu. & Median & Mean & 3rd Qu. & Max. & sd & sd\\_err\\_of\\_mean\\\\\n",
       "\\hline\n",
       "\tseq-ID to closest non-target & 81.107 & 92.44225 & 96.508 & 94.98219 & 98.60025 & 99.813 & 6.794 & 2.774\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 1 × 8 of type dbl\n",
       "\n",
       "| <!--/--> | Min. | 1st Qu. | Median | Mean | 3rd Qu. | Max. | sd | sd_err_of_mean |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| seq-ID to closest non-target | 81.107 | 92.44225 | 96.508 | 94.98219 | 98.60025 | 99.813 | 6.794 | 2.774 |\n",
       "\n"
      ],
      "text/plain": [
       "                             Min.   1st Qu.  Median Mean     3rd Qu.  Max.  \n",
       "seq-ID to closest non-target 81.107 92.44225 96.508 94.98219 98.60025 99.813\n",
       "                             sd    sd_err_of_mean\n",
       "seq-ID to closest non-target 6.794 2.774         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# closest seqID\n",
    "gene_nontarget %>%\n",
    "    filter(pident_rank == 1) %>%\n",
    "    .$pident %>% summary_x('seq-ID to closest non-target')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Primer quality"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-28T12:43:26.716141Z",
     "iopub.status.busy": "2021-04-28T12:43:26.714063Z",
     "iopub.status.idle": "2021-04-28T12:43:26.748382Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No. of unique primer sets: 991 \n",
      "No. of unique clusters: 484 \n"
     ]
    }
   ],
   "source": [
    "# summary\n",
    "primer_info %>% unique_n('primer sets', primer_set)\n",
    "primer_info %>% distinct(gene_type, cluster_id) %>% nrow %>% cat('No. of unique clusters:', ., '\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-28T12:43:30.814721Z",
     "iopub.status.busy": "2021-04-28T12:43:30.812417Z",
     "iopub.status.idle": "2021-04-28T12:43:30.847188Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 1 × 8 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Min.</th><th scope=col>1st Qu.</th><th scope=col>Median</th><th scope=col>Mean</th><th scope=col>3rd Qu.</th><th scope=col>Max.</th><th scope=col>sd</th><th scope=col>sd_err_of_mean</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>primer sets per cluster</th><td>1</td><td>10</td><td>10</td><td>9.615702</td><td>10</td><td>10</td><td>3.646</td><td>1.489</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 1 × 8 of type dbl\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & Min. & 1st Qu. & Median & Mean & 3rd Qu. & Max. & sd & sd\\_err\\_of\\_mean\\\\\n",
       "\\hline\n",
       "\tprimer sets per cluster & 1 & 10 & 10 & 9.615702 & 10 & 10 & 3.646 & 1.489\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 1 × 8 of type dbl\n",
       "\n",
       "| <!--/--> | Min. | 1st Qu. | Median | Mean | 3rd Qu. | Max. | sd | sd_err_of_mean |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| primer sets per cluster | 1 | 10 | 10 | 9.615702 | 10 | 10 | 3.646 | 1.489 |\n",
       "\n"
      ],
      "text/plain": [
       "                        Min. 1st Qu. Median Mean     3rd Qu. Max. sd   \n",
       "primer sets per cluster 1    10      10     9.615702 10      10   3.646\n",
       "                        sd_err_of_mean\n",
       "primer sets per cluster 1.489         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# primers per cluster\n",
    "primer_info %>%\n",
    "    distinct(cluster_id, primer_set) %>%\n",
    "    group_by(cluster_id) %>%\n",
    "    summarize(n_primer_sets = n(), .groups='drop') %>%\n",
    "    .$n_primer_sets %>% summary_x('primer sets per cluster')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-28T12:43:35.521537Z",
     "iopub.status.busy": "2021-04-28T12:43:35.519528Z",
     "iopub.status.idle": "2021-04-28T12:43:35.570474Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 1 × 8 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Min.</th><th scope=col>1st Qu.</th><th scope=col>Median</th><th scope=col>Mean</th><th scope=col>3rd Qu.</th><th scope=col>Max.</th><th scope=col>sd</th><th scope=col>sd_err_of_mean</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>mean amplicon size</th><td>94.35294</td><td>127</td><td>164</td><td>167.3896</td><td>204</td><td>250</td><td>55.029</td><td>22.466</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 1 × 8 of type dbl\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & Min. & 1st Qu. & Median & Mean & 3rd Qu. & Max. & sd & sd\\_err\\_of\\_mean\\\\\n",
       "\\hline\n",
       "\tmean amplicon size & 94.35294 & 127 & 164 & 167.3896 & 204 & 250 & 55.029 & 22.466\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 1 × 8 of type dbl\n",
       "\n",
       "| <!--/--> | Min. | 1st Qu. | Median | Mean | 3rd Qu. | Max. | sd | sd_err_of_mean |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| mean amplicon size | 94.35294 | 127 | 164 | 167.3896 | 204 | 250 | 55.029 | 22.466 |\n",
       "\n"
      ],
      "text/plain": [
       "                   Min.     1st Qu. Median Mean     3rd Qu. Max. sd    \n",
       "mean amplicon size 94.35294 127     164    167.3896 204     250  55.029\n",
       "                   sd_err_of_mean\n",
       "mean amplicon size 22.466        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 1 × 8 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Min.</th><th scope=col>1st Qu.</th><th scope=col>Median</th><th scope=col>Mean</th><th scope=col>3rd Qu.</th><th scope=col>Max.</th><th scope=col>sd</th><th scope=col>sd_err_of_mean</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>degeneracy</th><td>1</td><td>8</td><td>16</td><td>21.48765</td><td>24</td><td>128</td><td>47.279</td><td>19.302</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 1 × 8 of type dbl\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & Min. & 1st Qu. & Median & Mean & 3rd Qu. & Max. & sd & sd\\_err\\_of\\_mean\\\\\n",
       "\\hline\n",
       "\tdegeneracy & 1 & 8 & 16 & 21.48765 & 24 & 128 & 47.279 & 19.302\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 1 × 8 of type dbl\n",
       "\n",
       "| <!--/--> | Min. | 1st Qu. | Median | Mean | 3rd Qu. | Max. | sd | sd_err_of_mean |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| degeneracy | 1 | 8 | 16 | 21.48765 | 24 | 128 | 47.279 | 19.302 |\n",
       "\n"
      ],
      "text/plain": [
       "           Min. 1st Qu. Median Mean     3rd Qu. Max. sd     sd_err_of_mean\n",
       "degeneracy 1    8       16     21.48765 24      128  47.279 19.302        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 1 × 8 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Min.</th><th scope=col>1st Qu.</th><th scope=col>Median</th><th scope=col>Mean</th><th scope=col>3rd Qu.</th><th scope=col>Max.</th><th scope=col>sd</th><th scope=col>sd_err_of_mean</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>degeneracy (3-prime)</th><td>1</td><td>1</td><td>2</td><td>1.905995</td><td>2</td><td>6</td><td>1.866</td><td>0.762</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 1 × 8 of type dbl\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & Min. & 1st Qu. & Median & Mean & 3rd Qu. & Max. & sd & sd\\_err\\_of\\_mean\\\\\n",
       "\\hline\n",
       "\tdegeneracy (3-prime) & 1 & 1 & 2 & 1.905995 & 2 & 6 & 1.866 & 0.762\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 1 × 8 of type dbl\n",
       "\n",
       "| <!--/--> | Min. | 1st Qu. | Median | Mean | 3rd Qu. | Max. | sd | sd_err_of_mean |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| degeneracy (3-prime) | 1 | 1 | 2 | 1.905995 | 2 | 6 | 1.866 | 0.762 |\n",
       "\n"
      ],
      "text/plain": [
       "                     Min. 1st Qu. Median Mean     3rd Qu. Max. sd   \n",
       "degeneracy (3-prime) 1    1       2      1.905995 2       6    1.866\n",
       "                     sd_err_of_mean\n",
       "degeneracy (3-prime) 0.762         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# primer quality\n",
    "primer_info %>% filter(primer_type == 'PRIMER_LEFT') %>% .$amplicon_size_avg %>% summary_x('mean amplicon size')\n",
    "primer_info %>% .$degeneracy %>% summary_x('degeneracy')\n",
    "primer_info %>% .$degeneracy_3prime %>% summary_x('degeneracy (3-prime)')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-28T13:08:10.353200Z",
     "iopub.status.busy": "2021-04-28T13:08:10.351420Z",
     "iopub.status.idle": "2021-04-28T13:08:10.417546Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 9842 × 24</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>gene_type</th><th scope=col>cluster_id</th><th scope=col>primer_set</th><th scope=col>amplicon_size_consensus</th><th scope=col>amplicon_size_avg</th><th scope=col>amplicon_size_sd</th><th scope=col>primer_id</th><th scope=col>primer_type</th><th scope=col>sequence</th><th scope=col>length</th><th scope=col>⋯</th><th scope=col>Tm_avg</th><th scope=col>Tm_sd</th><th scope=col>GC_avg</th><th scope=col>GC_sd</th><th scope=col>hairpin_avg</th><th scope=col>hairpin_sd</th><th scope=col>homodimer_avg</th><th scope=col>homodimer_sd</th><th scope=col>pident</th><th scope=col>sscinames</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>cds</td><td>1</td><td>286</td><td>201</td><td>201</td><td>0</td><td>286f</td><td>PRIMER_LEFT</td><td>TRCGYTCGKCTYTGGTDCCG</td><td>20</td><td>⋯</td><td>64.26734</td><td>2.827061</td><td>61.66667</td><td>5.527708</td><td>28.84957</td><td>27.43375</td><td>3.172741</td><td>15.30954</td><td>94.323</td><td>Odoribacter sp. AF15-53                        </td></tr>\n",
       "\t<tr><td>cds</td><td>1</td><td>286</td><td>201</td><td>201</td><td>0</td><td>286f</td><td>PRIMER_LEFT</td><td>TRCGYTCGKCTYTGGTDCCG</td><td>20</td><td>⋯</td><td>64.26734</td><td>2.827061</td><td>61.66667</td><td>5.527708</td><td>28.84957</td><td>27.43375</td><td>3.172741</td><td>15.30954</td><td>94.323</td><td>Odoribacter sp. AM16-33;Odoribacter sp. AF21-41</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋱</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>cds</td><td>500</td><td>874</td><td>117</td><td>117</td><td>0</td><td>874f</td><td>PRIMER_LEFT </td><td>ATWGAYGCSGGRCCYGAYTT  </td><td>20</td><td>⋯</td><td>62.45535</td><td>2.487909</td><td>55.00000</td><td>5.000000</td><td>23.95850</td><td>24.20188</td><td> 23.27590</td><td>16.75870</td><td>93.281</td><td>Odoribacter sp. AM16-33;Odoribacter sp. AF21-41</td></tr>\n",
       "\t<tr><td>cds</td><td>500</td><td>874</td><td>117</td><td>117</td><td>0</td><td>874r</td><td>PRIMER_RIGHT</td><td>YCGWATATCRTCCARKCCYCCG</td><td>22</td><td>⋯</td><td>62.67829</td><td>2.221396</td><td>56.81818</td><td>5.081973</td><td>19.97123</td><td>19.99833</td><td>-26.80071</td><td>22.36798</td><td>93.281</td><td>Odoribacter sp. AM16-33;Odoribacter sp. AF21-41</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 9842 × 24\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " gene\\_type & cluster\\_id & primer\\_set & amplicon\\_size\\_consensus & amplicon\\_size\\_avg & amplicon\\_size\\_sd & primer\\_id & primer\\_type & sequence & length & ⋯ & Tm\\_avg & Tm\\_sd & GC\\_avg & GC\\_sd & hairpin\\_avg & hairpin\\_sd & homodimer\\_avg & homodimer\\_sd & pident & sscinames\\\\\n",
       " <chr> & <int> & <int> & <int> & <dbl> & <dbl> & <fct> & <fct> & <fct> & <int> & ⋯ & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <fct>\\\\\n",
       "\\hline\n",
       "\t cds & 1 & 286 & 201 & 201 & 0 & 286f & PRIMER\\_LEFT & TRCGYTCGKCTYTGGTDCCG & 20 & ⋯ & 64.26734 & 2.827061 & 61.66667 & 5.527708 & 28.84957 & 27.43375 & 3.172741 & 15.30954 & 94.323 & Odoribacter sp. AF15-53                        \\\\\n",
       "\t cds & 1 & 286 & 201 & 201 & 0 & 286f & PRIMER\\_LEFT & TRCGYTCGKCTYTGGTDCCG & 20 & ⋯ & 64.26734 & 2.827061 & 61.66667 & 5.527708 & 28.84957 & 27.43375 & 3.172741 & 15.30954 & 94.323 & Odoribacter sp. AM16-33;Odoribacter sp. AF21-41\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋱ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t cds & 500 & 874 & 117 & 117 & 0 & 874f & PRIMER\\_LEFT  & ATWGAYGCSGGRCCYGAYTT   & 20 & ⋯ & 62.45535 & 2.487909 & 55.00000 & 5.000000 & 23.95850 & 24.20188 &  23.27590 & 16.75870 & 93.281 & Odoribacter sp. AM16-33;Odoribacter sp. AF21-41\\\\\n",
       "\t cds & 500 & 874 & 117 & 117 & 0 & 874r & PRIMER\\_RIGHT & YCGWATATCRTCCARKCCYCCG & 22 & ⋯ & 62.67829 & 2.221396 & 56.81818 & 5.081973 & 19.97123 & 19.99833 & -26.80071 & 22.36798 & 93.281 & Odoribacter sp. AM16-33;Odoribacter sp. AF21-41\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 9842 × 24\n",
       "\n",
       "| gene_type &lt;chr&gt; | cluster_id &lt;int&gt; | primer_set &lt;int&gt; | amplicon_size_consensus &lt;int&gt; | amplicon_size_avg &lt;dbl&gt; | amplicon_size_sd &lt;dbl&gt; | primer_id &lt;fct&gt; | primer_type &lt;fct&gt; | sequence &lt;fct&gt; | length &lt;int&gt; | ⋯ ⋯ | Tm_avg &lt;dbl&gt; | Tm_sd &lt;dbl&gt; | GC_avg &lt;dbl&gt; | GC_sd &lt;dbl&gt; | hairpin_avg &lt;dbl&gt; | hairpin_sd &lt;dbl&gt; | homodimer_avg &lt;dbl&gt; | homodimer_sd &lt;dbl&gt; | pident &lt;dbl&gt; | sscinames &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| cds | 1 | 286 | 201 | 201 | 0 | 286f | PRIMER_LEFT | TRCGYTCGKCTYTGGTDCCG | 20 | ⋯ | 64.26734 | 2.827061 | 61.66667 | 5.527708 | 28.84957 | 27.43375 | 3.172741 | 15.30954 | 94.323 | Odoribacter sp. AF15-53                         |\n",
       "| cds | 1 | 286 | 201 | 201 | 0 | 286f | PRIMER_LEFT | TRCGYTCGKCTYTGGTDCCG | 20 | ⋯ | 64.26734 | 2.827061 | 61.66667 | 5.527708 | 28.84957 | 27.43375 | 3.172741 | 15.30954 | 94.323 | Odoribacter sp. AM16-33;Odoribacter sp. AF21-41 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋱ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| cds | 500 | 874 | 117 | 117 | 0 | 874f | PRIMER_LEFT  | ATWGAYGCSGGRCCYGAYTT   | 20 | ⋯ | 62.45535 | 2.487909 | 55.00000 | 5.000000 | 23.95850 | 24.20188 |  23.27590 | 16.75870 | 93.281 | Odoribacter sp. AM16-33;Odoribacter sp. AF21-41 |\n",
       "| cds | 500 | 874 | 117 | 117 | 0 | 874r | PRIMER_RIGHT | YCGWATATCRTCCARKCCYCCG | 22 | ⋯ | 62.67829 | 2.221396 | 56.81818 | 5.081973 | 19.97123 | 19.99833 | -26.80071 | 22.36798 | 93.281 | Odoribacter sp. AM16-33;Odoribacter sp. AF21-41 |\n",
       "\n"
      ],
      "text/plain": [
       "     gene_type cluster_id primer_set amplicon_size_consensus amplicon_size_avg\n",
       "1    cds       1          286        201                     201              \n",
       "2    cds       1          286        201                     201              \n",
       "⋮    ⋮         ⋮          ⋮          ⋮                       ⋮                \n",
       "9841 cds       500        874        117                     117              \n",
       "9842 cds       500        874        117                     117              \n",
       "     amplicon_size_sd primer_id primer_type  sequence               length ⋯\n",
       "1    0                286f      PRIMER_LEFT  TRCGYTCGKCTYTGGTDCCG   20     ⋯\n",
       "2    0                286f      PRIMER_LEFT  TRCGYTCGKCTYTGGTDCCG   20     ⋯\n",
       "⋮    ⋮                ⋮         ⋮            ⋮                      ⋮      ⋱\n",
       "9841 0                874f      PRIMER_LEFT  ATWGAYGCSGGRCCYGAYTT   20     ⋯\n",
       "9842 0                874r      PRIMER_RIGHT YCGWATATCRTCCARKCCYCCG 22     ⋯\n",
       "     Tm_avg   Tm_sd    GC_avg   GC_sd    hairpin_avg hairpin_sd homodimer_avg\n",
       "1    64.26734 2.827061 61.66667 5.527708 28.84957    27.43375   3.172741     \n",
       "2    64.26734 2.827061 61.66667 5.527708 28.84957    27.43375   3.172741     \n",
       "⋮    ⋮        ⋮        ⋮        ⋮        ⋮           ⋮          ⋮            \n",
       "9841 62.45535 2.487909 55.00000 5.000000 23.95850    24.20188    23.27590    \n",
       "9842 62.67829 2.221396 56.81818 5.081973 19.97123    19.99833   -26.80071    \n",
       "     homodimer_sd pident sscinames                                      \n",
       "1    15.30954     94.323 Odoribacter sp. AF15-53                        \n",
       "2    15.30954     94.323 Odoribacter sp. AM16-33;Odoribacter sp. AF21-41\n",
       "⋮    ⋮            ⋮      ⋮                                              \n",
       "9841 16.75870     93.281 Odoribacter sp. AM16-33;Odoribacter sp. AF21-41\n",
       "9842 22.36798     93.281 Odoribacter sp. AM16-33;Odoribacter sp. AF21-41"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# adding non-target pident info\n",
    "tmp1 = gene_nontarget %>%\n",
    "    filter(pident_rank == 1) %>%\n",
    "    distinct(cluster_id, pident, sscinames) %>%\n",
    "    mutate(gene_type = 'cds')\n",
    "\n",
    "primer_info_j = primer_info %>%\n",
    "    inner_join(tmp1, c('gene_type', 'cluster_id'))\n",
    "primer_info_j"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-28T13:08:16.912515Z",
     "iopub.status.busy": "2021-04-28T13:08:16.910449Z",
     "iopub.status.idle": "2021-04-28T13:08:17.051052Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A tibble: 30 × 26</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>gene_type</th><th scope=col>cluster_id</th><th scope=col>primer_set</th><th scope=col>amplicon_size_consensus</th><th scope=col>amplicon_size_avg</th><th scope=col>amplicon_size_sd</th><th scope=col>primer_id</th><th scope=col>primer_type</th><th scope=col>sequence</th><th scope=col>length</th><th scope=col>degeneracy</th><th scope=col>degeneracy_3prime</th><th scope=col>position_start</th><th scope=col>position_end</th><th scope=col>Tm_avg</th><th scope=col>Tm_sd</th><th scope=col>GC_avg</th><th scope=col>GC_sd</th><th scope=col>hairpin_avg</th><th scope=col>hairpin_sd</th><th scope=col>homodimer_avg</th><th scope=col>homodimer_sd</th><th scope=col>pident</th><th scope=col>sscinames</th><th scope=col>max_degeneracy_3prime</th><th scope=col>max_degeneracy</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>cds</td><td>469</td><td> 677</td><td>122</td><td>122</td><td>0</td><td>677f </td><td>PRIMER_LEFT </td><td>AGTCCSGGDTATGAYCCGCA </td><td>20</td><td> 12</td><td>1</td><td> 807</td><td> 827</td><td>63.37921</td><td>1.585835</td><td>59.16667</td><td>3.435921</td><td>54.529677</td><td>10.4477986</td><td> 12.546118</td><td>13.3155804</td><td>81.107</td><td>Odoribacter sp. AF15-53</td><td>2</td><td> 64</td></tr>\n",
       "\t<tr><td>cds</td><td>469</td><td> 677</td><td>122</td><td>122</td><td>0</td><td>677r </td><td>PRIMER_RIGHT</td><td>CCBGGGGGATAYTGYGCCA  </td><td>19</td><td> 12</td><td>2</td><td> 910</td><td> 929</td><td>63.99539</td><td>2.171444</td><td>66.66667</td><td>4.472824</td><td>42.790866</td><td> 2.5680978</td><td>  3.518264</td><td>11.0215662</td><td>81.107</td><td>Odoribacter sp. AF15-53</td><td>2</td><td> 64</td></tr>\n",
       "\t<tr><td>cds</td><td>469</td><td>  84</td><td>122</td><td>122</td><td>0</td><td>84f  </td><td>PRIMER_LEFT </td><td>AGTCCSGGDTATGAYCCGCA </td><td>20</td><td> 12</td><td>1</td><td> 807</td><td> 827</td><td>63.37921</td><td>1.585835</td><td>59.16667</td><td>3.435921</td><td>54.529677</td><td>10.4477986</td><td> 12.546118</td><td>13.3155804</td><td>81.107</td><td>Odoribacter sp. AF15-53</td><td>3</td><td> 12</td></tr>\n",
       "\t<tr><td>cds</td><td>469</td><td>  84</td><td>122</td><td>122</td><td>0</td><td>84r  </td><td>PRIMER_RIGHT</td><td>CCBGGGGGATAYTGYGCCAT </td><td>20</td><td> 12</td><td>1</td><td> 909</td><td> 929</td><td>64.24558</td><td>2.069738</td><td>63.33333</td><td>4.249183</td><td>42.928759</td><td> 2.8664437</td><td>  3.518264</td><td>11.0215662</td><td>81.107</td><td>Odoribacter sp. AF15-53</td><td>3</td><td> 12</td></tr>\n",
       "\t<tr><td>cds</td><td>469</td><td> 700</td><td>119</td><td>119</td><td>0</td><td>700f </td><td>PRIMER_LEFT </td><td>AGTCCSGGDTATGAYCCGCA </td><td>20</td><td> 12</td><td>1</td><td> 807</td><td> 827</td><td>63.37921</td><td>1.585835</td><td>59.16667</td><td>3.435921</td><td>54.529677</td><td>10.4477986</td><td> 12.546118</td><td>13.3155804</td><td>81.107</td><td>Odoribacter sp. AF15-53</td><td>3</td><td> 36</td></tr>\n",
       "\t<tr><td>cds</td><td>469</td><td> 700</td><td>119</td><td>119</td><td>0</td><td>700r </td><td>PRIMER_RIGHT</td><td>GGGGGATAYTGYGCCATSGT </td><td>20</td><td>  8</td><td>2</td><td> 906</td><td> 926</td><td>62.86230</td><td>1.692288</td><td>60.00000</td><td>3.535534</td><td>41.156774</td><td> 0.1051210</td><td>  8.175715</td><td> 7.8528329</td><td>81.107</td><td>Odoribacter sp. AF15-53</td><td>3</td><td> 36</td></tr>\n",
       "\t<tr><td>cds</td><td>469</td><td> 720</td><td>119</td><td>119</td><td>0</td><td>720f </td><td>PRIMER_LEFT </td><td>AGTCCSGGDTATGAYCCGCA </td><td>20</td><td> 12</td><td>1</td><td> 807</td><td> 827</td><td>63.37921</td><td>1.585835</td><td>59.16667</td><td>3.435921</td><td>54.529677</td><td>10.4477986</td><td> 12.546118</td><td>13.3155804</td><td>81.107</td><td>Odoribacter sp. AF15-53</td><td>4</td><td> 16</td></tr>\n",
       "\t<tr><td>cds</td><td>469</td><td> 720</td><td>119</td><td>119</td><td>0</td><td>720r </td><td>PRIMER_RIGHT</td><td>GGGGGATAYTGYGCCATSGTY</td><td>21</td><td> 16</td><td>4</td><td> 905</td><td> 926</td><td>63.52770</td><td>1.607231</td><td>59.52381</td><td>4.123930</td><td>41.156774</td><td> 0.1051210</td><td>  8.175715</td><td> 7.8528329</td><td>81.107</td><td>Odoribacter sp. AF15-53</td><td>4</td><td> 16</td></tr>\n",
       "\t<tr><td>cds</td><td>469</td><td> 308</td><td>118</td><td>118</td><td>0</td><td>308f </td><td>PRIMER_LEFT </td><td>AGTCCSGGDTATGAYCCGCA </td><td>20</td><td> 12</td><td>1</td><td> 807</td><td> 827</td><td>63.37921</td><td>1.585835</td><td>59.16667</td><td>3.435921</td><td>54.529677</td><td>10.4477986</td><td> 12.546118</td><td>13.3155804</td><td>81.107</td><td>Odoribacter sp. AF15-53</td><td>4</td><td> 24</td></tr>\n",
       "\t<tr><td>cds</td><td>469</td><td> 308</td><td>118</td><td>118</td><td>0</td><td>308r </td><td>PRIMER_RIGHT</td><td>GGGGATAYTGYGCCATSGTYG</td><td>21</td><td> 16</td><td>4</td><td> 904</td><td> 925</td><td>63.09697</td><td>1.809699</td><td>59.52381</td><td>4.123930</td><td>40.972038</td><td> 0.2898574</td><td>  8.849733</td><td> 8.6056358</td><td>81.107</td><td>Odoribacter sp. AF15-53</td><td>4</td><td> 24</td></tr>\n",
       "\t<tr><td>cds</td><td>469</td><td>  25</td><td>120</td><td>120</td><td>0</td><td>25f  </td><td>PRIMER_LEFT </td><td>AGTCCSGGDTATGAYCCGCA </td><td>20</td><td> 12</td><td>1</td><td> 807</td><td> 827</td><td>63.37921</td><td>1.585835</td><td>59.16667</td><td>3.435921</td><td>54.529677</td><td>10.4477986</td><td> 12.546118</td><td>13.3155804</td><td>81.107</td><td>Odoribacter sp. AF15-53</td><td>4</td><td> 32</td></tr>\n",
       "\t<tr><td>cds</td><td>469</td><td>  25</td><td>120</td><td>120</td><td>0</td><td>25r  </td><td>PRIMER_RIGHT</td><td>BGGGGGATAYTGYGCCATSG </td><td>20</td><td> 24</td><td>2</td><td> 907</td><td> 927</td><td>63.42396</td><td>1.720398</td><td>63.33333</td><td>4.249183</td><td>42.405568</td><td> 1.7528983</td><td>  4.674529</td><td> 5.1073084</td><td>81.107</td><td>Odoribacter sp. AF15-53</td><td>4</td><td> 32</td></tr>\n",
       "\t<tr><td>cds</td><td>469</td><td> 448</td><td>119</td><td>119</td><td>0</td><td>448f </td><td>PRIMER_LEFT </td><td>GTCCSGGDTATGAYCCGCAA </td><td>20</td><td> 12</td><td>1</td><td> 808</td><td> 828</td><td>62.46664</td><td>1.557512</td><td>59.16667</td><td>3.435921</td><td>54.529677</td><td>10.4477986</td><td> 12.546118</td><td>13.3155804</td><td>81.107</td><td>Odoribacter sp. AF15-53</td><td>4</td><td> 32</td></tr>\n",
       "\t<tr><td>cds</td><td>469</td><td> 448</td><td>119</td><td>119</td><td>0</td><td>448r </td><td>PRIMER_RIGHT</td><td>BGGGGGATAYTGYGCCATSG </td><td>20</td><td> 24</td><td>2</td><td> 907</td><td> 927</td><td>63.42396</td><td>1.720398</td><td>63.33333</td><td>4.249183</td><td>42.405568</td><td> 1.7528983</td><td>  4.674529</td><td> 5.1073084</td><td>81.107</td><td>Odoribacter sp. AF15-53</td><td>4</td><td> 32</td></tr>\n",
       "\t<tr><td>cds</td><td>469</td><td> 785</td><td>125</td><td>125</td><td>0</td><td>785f </td><td>PRIMER_LEFT </td><td>TCYAGTCCSGGDTATGAYCCG</td><td>21</td><td> 24</td><td>2</td><td> 804</td><td> 825</td><td>62.16733</td><td>2.085810</td><td>58.73016</td><td>4.046841</td><td>54.810400</td><td>10.1710443</td><td> 11.909481</td><td>12.9429433</td><td>81.107</td><td>Odoribacter sp. AF15-53</td><td>4</td><td> 64</td></tr>\n",
       "\t<tr><td>cds</td><td>469</td><td> 785</td><td>125</td><td>125</td><td>0</td><td>785r </td><td>PRIMER_RIGHT</td><td>CCBGGGGGATAYTGYGCCAT </td><td>20</td><td> 12</td><td>1</td><td> 909</td><td> 929</td><td>64.24558</td><td>2.069738</td><td>63.33333</td><td>4.249183</td><td>42.928759</td><td> 2.8664437</td><td>  3.518264</td><td>11.0215662</td><td>81.107</td><td>Odoribacter sp. AF15-53</td><td>4</td><td> 64</td></tr>\n",
       "\t<tr><td>cds</td><td>469</td><td>1000</td><td>120</td><td>120</td><td>0</td><td>1000f</td><td>PRIMER_LEFT </td><td>GTCCSGGDTATGAYCCGCAA </td><td>20</td><td> 12</td><td>1</td><td> 808</td><td> 828</td><td>62.46664</td><td>1.557512</td><td>59.16667</td><td>3.435921</td><td>54.529677</td><td>10.4477986</td><td> 12.546118</td><td>13.3155804</td><td>81.107</td><td>Odoribacter sp. AF15-53</td><td>4</td><td>128</td></tr>\n",
       "\t<tr><td>cds</td><td>469</td><td>1000</td><td>120</td><td>120</td><td>0</td><td>1000r</td><td>PRIMER_RIGHT</td><td>CBGGGGGATAYTGYGCCATS </td><td>20</td><td> 24</td><td>2</td><td> 908</td><td> 928</td><td>62.87830</td><td>2.042500</td><td>63.33333</td><td>4.249183</td><td>42.928759</td><td> 2.8664437</td><td> -1.070190</td><td> 8.1980946</td><td>81.107</td><td>Odoribacter sp. AF15-53</td><td>4</td><td>128</td></tr>\n",
       "\t<tr><td>cds</td><td>469</td><td> 910</td><td>109</td><td>109</td><td>0</td><td>910f </td><td>PRIMER_LEFT </td><td>GTCCSGGDTATGAYCCGCAA </td><td>20</td><td> 12</td><td>1</td><td> 808</td><td> 828</td><td>62.46664</td><td>1.557512</td><td>59.16667</td><td>3.435921</td><td>54.529677</td><td>10.4477986</td><td> 12.546118</td><td>13.3155804</td><td>81.107</td><td>Odoribacter sp. AF15-53</td><td>6</td><td> 96</td></tr>\n",
       "\t<tr><td>cds</td><td>469</td><td> 910</td><td>109</td><td>109</td><td>0</td><td>910r </td><td>PRIMER_RIGHT</td><td>TGYGCCATSGTYGTTCTRTC </td><td>20</td><td> 16</td><td>2</td><td> 897</td><td> 917</td><td>60.27747</td><td>2.226624</td><td>52.50000</td><td>4.330127</td><td> 9.590313</td><td>16.6109088</td><td>  6.825856</td><td>12.1537464</td><td>81.107</td><td>Odoribacter sp. AF15-53</td><td>6</td><td> 96</td></tr>\n",
       "\t<tr><td>cds</td><td>118</td><td> 555</td><td>107</td><td>107</td><td>0</td><td>555f </td><td>PRIMER_LEFT </td><td>AAGGAYWTBCGNGTSGATGT </td><td>20</td><td> 96</td><td>1</td><td> 867</td><td> 887</td><td>59.21880</td><td>2.210807</td><td>48.33333</td><td>4.249183</td><td>29.190064</td><td>23.3863626</td><td> -3.187227</td><td>24.1165857</td><td>81.767</td><td>Odoribacter sp. AF15-53</td><td>2</td><td> 96</td></tr>\n",
       "\t<tr><td>cds</td><td>118</td><td> 555</td><td>107</td><td>107</td><td>0</td><td>555r </td><td>PRIMER_RIGHT</td><td>CGRCGHCCRATRATYAAYCC </td><td>20</td><td> 96</td><td>2</td><td> 954</td><td> 974</td><td>59.31237</td><td>2.922306</td><td>54.16667</td><td>6.066758</td><td>23.698622</td><td>23.9999455</td><td>  3.609712</td><td>16.5893977</td><td>81.767</td><td>Odoribacter sp. AF15-53</td><td>2</td><td> 96</td></tr>\n",
       "\t<tr><td>cds</td><td>118</td><td> 286</td><td>140</td><td>140</td><td>0</td><td>286f </td><td>PRIMER_LEFT </td><td>AAGGAYWTBCGNGTSGATGT </td><td>20</td><td> 96</td><td>1</td><td> 867</td><td> 887</td><td>59.21880</td><td>2.210807</td><td>48.33333</td><td>4.249183</td><td>29.190064</td><td>23.3863626</td><td> -3.187227</td><td>24.1165857</td><td>81.767</td><td>Odoribacter sp. AF15-53</td><td>4</td><td> 96</td></tr>\n",
       "\t<tr><td>cds</td><td>118</td><td> 286</td><td>140</td><td>140</td><td>0</td><td>286r </td><td>PRIMER_RIGHT</td><td>CGYTGYTCYTGWACCAAYCC </td><td>20</td><td> 32</td><td>2</td><td> 987</td><td>1007</td><td>59.80526</td><td>2.153698</td><td>55.00000</td><td>5.000000</td><td>11.994356</td><td>17.8257607</td><td> -6.009949</td><td>13.5798078</td><td>81.767</td><td>Odoribacter sp. AF15-53</td><td>4</td><td> 96</td></tr>\n",
       "\t<tr><td>cds</td><td>118</td><td> 287</td><td>141</td><td>141</td><td>0</td><td>287f </td><td>PRIMER_LEFT </td><td>AAGGAYWTBCGNGTSGATGT </td><td>20</td><td> 96</td><td>1</td><td> 867</td><td> 887</td><td>59.21880</td><td>2.210807</td><td>48.33333</td><td>4.249183</td><td>29.190064</td><td>23.3863626</td><td> -3.187227</td><td>24.1165857</td><td>81.767</td><td>Odoribacter sp. AF15-53</td><td>4</td><td> 96</td></tr>\n",
       "\t<tr><td>cds</td><td>118</td><td> 287</td><td>141</td><td>141</td><td>0</td><td>287r </td><td>PRIMER_RIGHT</td><td>VCGYTGYTCYTGWACCAAYC </td><td>20</td><td> 96</td><td>2</td><td> 988</td><td>1008</td><td>59.77764</td><td>2.222613</td><td>53.33333</td><td>5.527708</td><td>11.994356</td><td>17.8257607</td><td> -6.144309</td><td>13.8730856</td><td>81.767</td><td>Odoribacter sp. AF15-53</td><td>4</td><td> 96</td></tr>\n",
       "\t<tr><td>cds</td><td>118</td><td> 553</td><td>101</td><td>101</td><td>0</td><td>553f </td><td>PRIMER_LEFT </td><td>AAGGAYWTBCGNGTSGATGT </td><td>20</td><td> 96</td><td>1</td><td> 867</td><td> 887</td><td>59.21880</td><td>2.210807</td><td>48.33333</td><td>4.249183</td><td>29.190064</td><td>23.3863626</td><td> -3.187227</td><td>24.1165857</td><td>81.767</td><td>Odoribacter sp. AF15-53</td><td>4</td><td>128</td></tr>\n",
       "\t<tr><td>cds</td><td>118</td><td> 553</td><td>101</td><td>101</td><td>0</td><td>553r </td><td>PRIMER_RIGHT</td><td>CCRATRATYAAYCCNCGRTC </td><td>20</td><td>128</td><td>2</td><td> 948</td><td> 968</td><td>56.78183</td><td>2.912989</td><td>50.00000</td><td>6.123724</td><td>21.136796</td><td>18.6641357</td><td> -2.824651</td><td>22.4057285</td><td>81.767</td><td>Odoribacter sp. AF15-53</td><td>4</td><td>128</td></tr>\n",
       "\t<tr><td>cds</td><td>118</td><td> 642</td><td>230</td><td>230</td><td>0</td><td>642f </td><td>PRIMER_LEFT </td><td>CGGATRTYTTYGTDCCRGCK </td><td>20</td><td> 96</td><td>4</td><td>1237</td><td>1257</td><td>60.32781</td><td>2.703100</td><td>54.16667</td><td>6.066758</td><td>20.732458</td><td>20.8653216</td><td>-10.075399</td><td>24.5760581</td><td>81.767</td><td>Odoribacter sp. AF15-53</td><td>4</td><td>128</td></tr>\n",
       "\t<tr><td>cds</td><td>118</td><td> 642</td><td>230</td><td>230</td><td>0</td><td>642r </td><td>PRIMER_RIGHT</td><td>YYCCTTSTCRTYCCGYTTYA </td><td>20</td><td>128</td><td>4</td><td>1447</td><td>1467</td><td>58.92361</td><td>2.556469</td><td>50.00000</td><td>6.123724</td><td> 0.000000</td><td> 0.0000000</td><td>-41.763740</td><td> 0.3687041</td><td>81.767</td><td>Odoribacter sp. AF15-53</td><td>4</td><td>128</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 30 × 26\n",
       "\\begin{tabular}{llllllllllllllllllllllllll}\n",
       " gene\\_type & cluster\\_id & primer\\_set & amplicon\\_size\\_consensus & amplicon\\_size\\_avg & amplicon\\_size\\_sd & primer\\_id & primer\\_type & sequence & length & degeneracy & degeneracy\\_3prime & position\\_start & position\\_end & Tm\\_avg & Tm\\_sd & GC\\_avg & GC\\_sd & hairpin\\_avg & hairpin\\_sd & homodimer\\_avg & homodimer\\_sd & pident & sscinames & max\\_degeneracy\\_3prime & max\\_degeneracy\\\\\n",
       " <chr> & <int> & <int> & <int> & <dbl> & <dbl> & <fct> & <fct> & <fct> & <int> & <int> & <int> & <int> & <int> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <fct> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t cds & 469 &  677 & 122 & 122 & 0 & 677f  & PRIMER\\_LEFT  & AGTCCSGGDTATGAYCCGCA  & 20 &  12 & 1 &  807 &  827 & 63.37921 & 1.585835 & 59.16667 & 3.435921 & 54.529677 & 10.4477986 &  12.546118 & 13.3155804 & 81.107 & Odoribacter sp. AF15-53 & 2 &  64\\\\\n",
       "\t cds & 469 &  677 & 122 & 122 & 0 & 677r  & PRIMER\\_RIGHT & CCBGGGGGATAYTGYGCCA   & 19 &  12 & 2 &  910 &  929 & 63.99539 & 2.171444 & 66.66667 & 4.472824 & 42.790866 &  2.5680978 &   3.518264 & 11.0215662 & 81.107 & Odoribacter sp. AF15-53 & 2 &  64\\\\\n",
       "\t cds & 469 &   84 & 122 & 122 & 0 & 84f   & PRIMER\\_LEFT  & AGTCCSGGDTATGAYCCGCA  & 20 &  12 & 1 &  807 &  827 & 63.37921 & 1.585835 & 59.16667 & 3.435921 & 54.529677 & 10.4477986 &  12.546118 & 13.3155804 & 81.107 & Odoribacter sp. AF15-53 & 3 &  12\\\\\n",
       "\t cds & 469 &   84 & 122 & 122 & 0 & 84r   & PRIMER\\_RIGHT & CCBGGGGGATAYTGYGCCAT  & 20 &  12 & 1 &  909 &  929 & 64.24558 & 2.069738 & 63.33333 & 4.249183 & 42.928759 &  2.8664437 &   3.518264 & 11.0215662 & 81.107 & Odoribacter sp. AF15-53 & 3 &  12\\\\\n",
       "\t cds & 469 &  700 & 119 & 119 & 0 & 700f  & PRIMER\\_LEFT  & AGTCCSGGDTATGAYCCGCA  & 20 &  12 & 1 &  807 &  827 & 63.37921 & 1.585835 & 59.16667 & 3.435921 & 54.529677 & 10.4477986 &  12.546118 & 13.3155804 & 81.107 & Odoribacter sp. AF15-53 & 3 &  36\\\\\n",
       "\t cds & 469 &  700 & 119 & 119 & 0 & 700r  & PRIMER\\_RIGHT & GGGGGATAYTGYGCCATSGT  & 20 &   8 & 2 &  906 &  926 & 62.86230 & 1.692288 & 60.00000 & 3.535534 & 41.156774 &  0.1051210 &   8.175715 &  7.8528329 & 81.107 & Odoribacter sp. AF15-53 & 3 &  36\\\\\n",
       "\t cds & 469 &  720 & 119 & 119 & 0 & 720f  & PRIMER\\_LEFT  & AGTCCSGGDTATGAYCCGCA  & 20 &  12 & 1 &  807 &  827 & 63.37921 & 1.585835 & 59.16667 & 3.435921 & 54.529677 & 10.4477986 &  12.546118 & 13.3155804 & 81.107 & Odoribacter sp. AF15-53 & 4 &  16\\\\\n",
       "\t cds & 469 &  720 & 119 & 119 & 0 & 720r  & PRIMER\\_RIGHT & GGGGGATAYTGYGCCATSGTY & 21 &  16 & 4 &  905 &  926 & 63.52770 & 1.607231 & 59.52381 & 4.123930 & 41.156774 &  0.1051210 &   8.175715 &  7.8528329 & 81.107 & Odoribacter sp. AF15-53 & 4 &  16\\\\\n",
       "\t cds & 469 &  308 & 118 & 118 & 0 & 308f  & PRIMER\\_LEFT  & AGTCCSGGDTATGAYCCGCA  & 20 &  12 & 1 &  807 &  827 & 63.37921 & 1.585835 & 59.16667 & 3.435921 & 54.529677 & 10.4477986 &  12.546118 & 13.3155804 & 81.107 & Odoribacter sp. AF15-53 & 4 &  24\\\\\n",
       "\t cds & 469 &  308 & 118 & 118 & 0 & 308r  & PRIMER\\_RIGHT & GGGGATAYTGYGCCATSGTYG & 21 &  16 & 4 &  904 &  925 & 63.09697 & 1.809699 & 59.52381 & 4.123930 & 40.972038 &  0.2898574 &   8.849733 &  8.6056358 & 81.107 & Odoribacter sp. AF15-53 & 4 &  24\\\\\n",
       "\t cds & 469 &   25 & 120 & 120 & 0 & 25f   & PRIMER\\_LEFT  & AGTCCSGGDTATGAYCCGCA  & 20 &  12 & 1 &  807 &  827 & 63.37921 & 1.585835 & 59.16667 & 3.435921 & 54.529677 & 10.4477986 &  12.546118 & 13.3155804 & 81.107 & Odoribacter sp. AF15-53 & 4 &  32\\\\\n",
       "\t cds & 469 &   25 & 120 & 120 & 0 & 25r   & PRIMER\\_RIGHT & BGGGGGATAYTGYGCCATSG  & 20 &  24 & 2 &  907 &  927 & 63.42396 & 1.720398 & 63.33333 & 4.249183 & 42.405568 &  1.7528983 &   4.674529 &  5.1073084 & 81.107 & Odoribacter sp. AF15-53 & 4 &  32\\\\\n",
       "\t cds & 469 &  448 & 119 & 119 & 0 & 448f  & PRIMER\\_LEFT  & GTCCSGGDTATGAYCCGCAA  & 20 &  12 & 1 &  808 &  828 & 62.46664 & 1.557512 & 59.16667 & 3.435921 & 54.529677 & 10.4477986 &  12.546118 & 13.3155804 & 81.107 & Odoribacter sp. AF15-53 & 4 &  32\\\\\n",
       "\t cds & 469 &  448 & 119 & 119 & 0 & 448r  & PRIMER\\_RIGHT & BGGGGGATAYTGYGCCATSG  & 20 &  24 & 2 &  907 &  927 & 63.42396 & 1.720398 & 63.33333 & 4.249183 & 42.405568 &  1.7528983 &   4.674529 &  5.1073084 & 81.107 & Odoribacter sp. AF15-53 & 4 &  32\\\\\n",
       "\t cds & 469 &  785 & 125 & 125 & 0 & 785f  & PRIMER\\_LEFT  & TCYAGTCCSGGDTATGAYCCG & 21 &  24 & 2 &  804 &  825 & 62.16733 & 2.085810 & 58.73016 & 4.046841 & 54.810400 & 10.1710443 &  11.909481 & 12.9429433 & 81.107 & Odoribacter sp. AF15-53 & 4 &  64\\\\\n",
       "\t cds & 469 &  785 & 125 & 125 & 0 & 785r  & PRIMER\\_RIGHT & CCBGGGGGATAYTGYGCCAT  & 20 &  12 & 1 &  909 &  929 & 64.24558 & 2.069738 & 63.33333 & 4.249183 & 42.928759 &  2.8664437 &   3.518264 & 11.0215662 & 81.107 & Odoribacter sp. AF15-53 & 4 &  64\\\\\n",
       "\t cds & 469 & 1000 & 120 & 120 & 0 & 1000f & PRIMER\\_LEFT  & GTCCSGGDTATGAYCCGCAA  & 20 &  12 & 1 &  808 &  828 & 62.46664 & 1.557512 & 59.16667 & 3.435921 & 54.529677 & 10.4477986 &  12.546118 & 13.3155804 & 81.107 & Odoribacter sp. AF15-53 & 4 & 128\\\\\n",
       "\t cds & 469 & 1000 & 120 & 120 & 0 & 1000r & PRIMER\\_RIGHT & CBGGGGGATAYTGYGCCATS  & 20 &  24 & 2 &  908 &  928 & 62.87830 & 2.042500 & 63.33333 & 4.249183 & 42.928759 &  2.8664437 &  -1.070190 &  8.1980946 & 81.107 & Odoribacter sp. AF15-53 & 4 & 128\\\\\n",
       "\t cds & 469 &  910 & 109 & 109 & 0 & 910f  & PRIMER\\_LEFT  & GTCCSGGDTATGAYCCGCAA  & 20 &  12 & 1 &  808 &  828 & 62.46664 & 1.557512 & 59.16667 & 3.435921 & 54.529677 & 10.4477986 &  12.546118 & 13.3155804 & 81.107 & Odoribacter sp. AF15-53 & 6 &  96\\\\\n",
       "\t cds & 469 &  910 & 109 & 109 & 0 & 910r  & PRIMER\\_RIGHT & TGYGCCATSGTYGTTCTRTC  & 20 &  16 & 2 &  897 &  917 & 60.27747 & 2.226624 & 52.50000 & 4.330127 &  9.590313 & 16.6109088 &   6.825856 & 12.1537464 & 81.107 & Odoribacter sp. AF15-53 & 6 &  96\\\\\n",
       "\t cds & 118 &  555 & 107 & 107 & 0 & 555f  & PRIMER\\_LEFT  & AAGGAYWTBCGNGTSGATGT  & 20 &  96 & 1 &  867 &  887 & 59.21880 & 2.210807 & 48.33333 & 4.249183 & 29.190064 & 23.3863626 &  -3.187227 & 24.1165857 & 81.767 & Odoribacter sp. AF15-53 & 2 &  96\\\\\n",
       "\t cds & 118 &  555 & 107 & 107 & 0 & 555r  & PRIMER\\_RIGHT & CGRCGHCCRATRATYAAYCC  & 20 &  96 & 2 &  954 &  974 & 59.31237 & 2.922306 & 54.16667 & 6.066758 & 23.698622 & 23.9999455 &   3.609712 & 16.5893977 & 81.767 & Odoribacter sp. AF15-53 & 2 &  96\\\\\n",
       "\t cds & 118 &  286 & 140 & 140 & 0 & 286f  & PRIMER\\_LEFT  & AAGGAYWTBCGNGTSGATGT  & 20 &  96 & 1 &  867 &  887 & 59.21880 & 2.210807 & 48.33333 & 4.249183 & 29.190064 & 23.3863626 &  -3.187227 & 24.1165857 & 81.767 & Odoribacter sp. AF15-53 & 4 &  96\\\\\n",
       "\t cds & 118 &  286 & 140 & 140 & 0 & 286r  & PRIMER\\_RIGHT & CGYTGYTCYTGWACCAAYCC  & 20 &  32 & 2 &  987 & 1007 & 59.80526 & 2.153698 & 55.00000 & 5.000000 & 11.994356 & 17.8257607 &  -6.009949 & 13.5798078 & 81.767 & Odoribacter sp. AF15-53 & 4 &  96\\\\\n",
       "\t cds & 118 &  287 & 141 & 141 & 0 & 287f  & PRIMER\\_LEFT  & AAGGAYWTBCGNGTSGATGT  & 20 &  96 & 1 &  867 &  887 & 59.21880 & 2.210807 & 48.33333 & 4.249183 & 29.190064 & 23.3863626 &  -3.187227 & 24.1165857 & 81.767 & Odoribacter sp. AF15-53 & 4 &  96\\\\\n",
       "\t cds & 118 &  287 & 141 & 141 & 0 & 287r  & PRIMER\\_RIGHT & VCGYTGYTCYTGWACCAAYC  & 20 &  96 & 2 &  988 & 1008 & 59.77764 & 2.222613 & 53.33333 & 5.527708 & 11.994356 & 17.8257607 &  -6.144309 & 13.8730856 & 81.767 & Odoribacter sp. AF15-53 & 4 &  96\\\\\n",
       "\t cds & 118 &  553 & 101 & 101 & 0 & 553f  & PRIMER\\_LEFT  & AAGGAYWTBCGNGTSGATGT  & 20 &  96 & 1 &  867 &  887 & 59.21880 & 2.210807 & 48.33333 & 4.249183 & 29.190064 & 23.3863626 &  -3.187227 & 24.1165857 & 81.767 & Odoribacter sp. AF15-53 & 4 & 128\\\\\n",
       "\t cds & 118 &  553 & 101 & 101 & 0 & 553r  & PRIMER\\_RIGHT & CCRATRATYAAYCCNCGRTC  & 20 & 128 & 2 &  948 &  968 & 56.78183 & 2.912989 & 50.00000 & 6.123724 & 21.136796 & 18.6641357 &  -2.824651 & 22.4057285 & 81.767 & Odoribacter sp. AF15-53 & 4 & 128\\\\\n",
       "\t cds & 118 &  642 & 230 & 230 & 0 & 642f  & PRIMER\\_LEFT  & CGGATRTYTTYGTDCCRGCK  & 20 &  96 & 4 & 1237 & 1257 & 60.32781 & 2.703100 & 54.16667 & 6.066758 & 20.732458 & 20.8653216 & -10.075399 & 24.5760581 & 81.767 & Odoribacter sp. AF15-53 & 4 & 128\\\\\n",
       "\t cds & 118 &  642 & 230 & 230 & 0 & 642r  & PRIMER\\_RIGHT & YYCCTTSTCRTYCCGYTTYA  & 20 & 128 & 4 & 1447 & 1467 & 58.92361 & 2.556469 & 50.00000 & 6.123724 &  0.000000 &  0.0000000 & -41.763740 &  0.3687041 & 81.767 & Odoribacter sp. AF15-53 & 4 & 128\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 30 × 26\n",
       "\n",
       "| gene_type &lt;chr&gt; | cluster_id &lt;int&gt; | primer_set &lt;int&gt; | amplicon_size_consensus &lt;int&gt; | amplicon_size_avg &lt;dbl&gt; | amplicon_size_sd &lt;dbl&gt; | primer_id &lt;fct&gt; | primer_type &lt;fct&gt; | sequence &lt;fct&gt; | length &lt;int&gt; | degeneracy &lt;int&gt; | degeneracy_3prime &lt;int&gt; | position_start &lt;int&gt; | position_end &lt;int&gt; | Tm_avg &lt;dbl&gt; | Tm_sd &lt;dbl&gt; | GC_avg &lt;dbl&gt; | GC_sd &lt;dbl&gt; | hairpin_avg &lt;dbl&gt; | hairpin_sd &lt;dbl&gt; | homodimer_avg &lt;dbl&gt; | homodimer_sd &lt;dbl&gt; | pident &lt;dbl&gt; | sscinames &lt;fct&gt; | max_degeneracy_3prime &lt;int&gt; | max_degeneracy &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| cds | 469 |  677 | 122 | 122 | 0 | 677f  | PRIMER_LEFT  | AGTCCSGGDTATGAYCCGCA  | 20 |  12 | 1 |  807 |  827 | 63.37921 | 1.585835 | 59.16667 | 3.435921 | 54.529677 | 10.4477986 |  12.546118 | 13.3155804 | 81.107 | Odoribacter sp. AF15-53 | 2 |  64 |\n",
       "| cds | 469 |  677 | 122 | 122 | 0 | 677r  | PRIMER_RIGHT | CCBGGGGGATAYTGYGCCA   | 19 |  12 | 2 |  910 |  929 | 63.99539 | 2.171444 | 66.66667 | 4.472824 | 42.790866 |  2.5680978 |   3.518264 | 11.0215662 | 81.107 | Odoribacter sp. AF15-53 | 2 |  64 |\n",
       "| cds | 469 |   84 | 122 | 122 | 0 | 84f   | PRIMER_LEFT  | AGTCCSGGDTATGAYCCGCA  | 20 |  12 | 1 |  807 |  827 | 63.37921 | 1.585835 | 59.16667 | 3.435921 | 54.529677 | 10.4477986 |  12.546118 | 13.3155804 | 81.107 | Odoribacter sp. AF15-53 | 3 |  12 |\n",
       "| cds | 469 |   84 | 122 | 122 | 0 | 84r   | PRIMER_RIGHT | CCBGGGGGATAYTGYGCCAT  | 20 |  12 | 1 |  909 |  929 | 64.24558 | 2.069738 | 63.33333 | 4.249183 | 42.928759 |  2.8664437 |   3.518264 | 11.0215662 | 81.107 | Odoribacter sp. AF15-53 | 3 |  12 |\n",
       "| cds | 469 |  700 | 119 | 119 | 0 | 700f  | PRIMER_LEFT  | AGTCCSGGDTATGAYCCGCA  | 20 |  12 | 1 |  807 |  827 | 63.37921 | 1.585835 | 59.16667 | 3.435921 | 54.529677 | 10.4477986 |  12.546118 | 13.3155804 | 81.107 | Odoribacter sp. AF15-53 | 3 |  36 |\n",
       "| cds | 469 |  700 | 119 | 119 | 0 | 700r  | PRIMER_RIGHT | GGGGGATAYTGYGCCATSGT  | 20 |   8 | 2 |  906 |  926 | 62.86230 | 1.692288 | 60.00000 | 3.535534 | 41.156774 |  0.1051210 |   8.175715 |  7.8528329 | 81.107 | Odoribacter sp. AF15-53 | 3 |  36 |\n",
       "| cds | 469 |  720 | 119 | 119 | 0 | 720f  | PRIMER_LEFT  | AGTCCSGGDTATGAYCCGCA  | 20 |  12 | 1 |  807 |  827 | 63.37921 | 1.585835 | 59.16667 | 3.435921 | 54.529677 | 10.4477986 |  12.546118 | 13.3155804 | 81.107 | Odoribacter sp. AF15-53 | 4 |  16 |\n",
       "| cds | 469 |  720 | 119 | 119 | 0 | 720r  | PRIMER_RIGHT | GGGGGATAYTGYGCCATSGTY | 21 |  16 | 4 |  905 |  926 | 63.52770 | 1.607231 | 59.52381 | 4.123930 | 41.156774 |  0.1051210 |   8.175715 |  7.8528329 | 81.107 | Odoribacter sp. AF15-53 | 4 |  16 |\n",
       "| cds | 469 |  308 | 118 | 118 | 0 | 308f  | PRIMER_LEFT  | AGTCCSGGDTATGAYCCGCA  | 20 |  12 | 1 |  807 |  827 | 63.37921 | 1.585835 | 59.16667 | 3.435921 | 54.529677 | 10.4477986 |  12.546118 | 13.3155804 | 81.107 | Odoribacter sp. AF15-53 | 4 |  24 |\n",
       "| cds | 469 |  308 | 118 | 118 | 0 | 308r  | PRIMER_RIGHT | GGGGATAYTGYGCCATSGTYG | 21 |  16 | 4 |  904 |  925 | 63.09697 | 1.809699 | 59.52381 | 4.123930 | 40.972038 |  0.2898574 |   8.849733 |  8.6056358 | 81.107 | Odoribacter sp. AF15-53 | 4 |  24 |\n",
       "| cds | 469 |   25 | 120 | 120 | 0 | 25f   | PRIMER_LEFT  | AGTCCSGGDTATGAYCCGCA  | 20 |  12 | 1 |  807 |  827 | 63.37921 | 1.585835 | 59.16667 | 3.435921 | 54.529677 | 10.4477986 |  12.546118 | 13.3155804 | 81.107 | Odoribacter sp. AF15-53 | 4 |  32 |\n",
       "| cds | 469 |   25 | 120 | 120 | 0 | 25r   | PRIMER_RIGHT | BGGGGGATAYTGYGCCATSG  | 20 |  24 | 2 |  907 |  927 | 63.42396 | 1.720398 | 63.33333 | 4.249183 | 42.405568 |  1.7528983 |   4.674529 |  5.1073084 | 81.107 | Odoribacter sp. AF15-53 | 4 |  32 |\n",
       "| cds | 469 |  448 | 119 | 119 | 0 | 448f  | PRIMER_LEFT  | GTCCSGGDTATGAYCCGCAA  | 20 |  12 | 1 |  808 |  828 | 62.46664 | 1.557512 | 59.16667 | 3.435921 | 54.529677 | 10.4477986 |  12.546118 | 13.3155804 | 81.107 | Odoribacter sp. AF15-53 | 4 |  32 |\n",
       "| cds | 469 |  448 | 119 | 119 | 0 | 448r  | PRIMER_RIGHT | BGGGGGATAYTGYGCCATSG  | 20 |  24 | 2 |  907 |  927 | 63.42396 | 1.720398 | 63.33333 | 4.249183 | 42.405568 |  1.7528983 |   4.674529 |  5.1073084 | 81.107 | Odoribacter sp. AF15-53 | 4 |  32 |\n",
       "| cds | 469 |  785 | 125 | 125 | 0 | 785f  | PRIMER_LEFT  | TCYAGTCCSGGDTATGAYCCG | 21 |  24 | 2 |  804 |  825 | 62.16733 | 2.085810 | 58.73016 | 4.046841 | 54.810400 | 10.1710443 |  11.909481 | 12.9429433 | 81.107 | Odoribacter sp. AF15-53 | 4 |  64 |\n",
       "| cds | 469 |  785 | 125 | 125 | 0 | 785r  | PRIMER_RIGHT | CCBGGGGGATAYTGYGCCAT  | 20 |  12 | 1 |  909 |  929 | 64.24558 | 2.069738 | 63.33333 | 4.249183 | 42.928759 |  2.8664437 |   3.518264 | 11.0215662 | 81.107 | Odoribacter sp. AF15-53 | 4 |  64 |\n",
       "| cds | 469 | 1000 | 120 | 120 | 0 | 1000f | PRIMER_LEFT  | GTCCSGGDTATGAYCCGCAA  | 20 |  12 | 1 |  808 |  828 | 62.46664 | 1.557512 | 59.16667 | 3.435921 | 54.529677 | 10.4477986 |  12.546118 | 13.3155804 | 81.107 | Odoribacter sp. AF15-53 | 4 | 128 |\n",
       "| cds | 469 | 1000 | 120 | 120 | 0 | 1000r | PRIMER_RIGHT | CBGGGGGATAYTGYGCCATS  | 20 |  24 | 2 |  908 |  928 | 62.87830 | 2.042500 | 63.33333 | 4.249183 | 42.928759 |  2.8664437 |  -1.070190 |  8.1980946 | 81.107 | Odoribacter sp. AF15-53 | 4 | 128 |\n",
       "| cds | 469 |  910 | 109 | 109 | 0 | 910f  | PRIMER_LEFT  | GTCCSGGDTATGAYCCGCAA  | 20 |  12 | 1 |  808 |  828 | 62.46664 | 1.557512 | 59.16667 | 3.435921 | 54.529677 | 10.4477986 |  12.546118 | 13.3155804 | 81.107 | Odoribacter sp. AF15-53 | 6 |  96 |\n",
       "| cds | 469 |  910 | 109 | 109 | 0 | 910r  | PRIMER_RIGHT | TGYGCCATSGTYGTTCTRTC  | 20 |  16 | 2 |  897 |  917 | 60.27747 | 2.226624 | 52.50000 | 4.330127 |  9.590313 | 16.6109088 |   6.825856 | 12.1537464 | 81.107 | Odoribacter sp. AF15-53 | 6 |  96 |\n",
       "| cds | 118 |  555 | 107 | 107 | 0 | 555f  | PRIMER_LEFT  | AAGGAYWTBCGNGTSGATGT  | 20 |  96 | 1 |  867 |  887 | 59.21880 | 2.210807 | 48.33333 | 4.249183 | 29.190064 | 23.3863626 |  -3.187227 | 24.1165857 | 81.767 | Odoribacter sp. AF15-53 | 2 |  96 |\n",
       "| cds | 118 |  555 | 107 | 107 | 0 | 555r  | PRIMER_RIGHT | CGRCGHCCRATRATYAAYCC  | 20 |  96 | 2 |  954 |  974 | 59.31237 | 2.922306 | 54.16667 | 6.066758 | 23.698622 | 23.9999455 |   3.609712 | 16.5893977 | 81.767 | Odoribacter sp. AF15-53 | 2 |  96 |\n",
       "| cds | 118 |  286 | 140 | 140 | 0 | 286f  | PRIMER_LEFT  | AAGGAYWTBCGNGTSGATGT  | 20 |  96 | 1 |  867 |  887 | 59.21880 | 2.210807 | 48.33333 | 4.249183 | 29.190064 | 23.3863626 |  -3.187227 | 24.1165857 | 81.767 | Odoribacter sp. AF15-53 | 4 |  96 |\n",
       "| cds | 118 |  286 | 140 | 140 | 0 | 286r  | PRIMER_RIGHT | CGYTGYTCYTGWACCAAYCC  | 20 |  32 | 2 |  987 | 1007 | 59.80526 | 2.153698 | 55.00000 | 5.000000 | 11.994356 | 17.8257607 |  -6.009949 | 13.5798078 | 81.767 | Odoribacter sp. AF15-53 | 4 |  96 |\n",
       "| cds | 118 |  287 | 141 | 141 | 0 | 287f  | PRIMER_LEFT  | AAGGAYWTBCGNGTSGATGT  | 20 |  96 | 1 |  867 |  887 | 59.21880 | 2.210807 | 48.33333 | 4.249183 | 29.190064 | 23.3863626 |  -3.187227 | 24.1165857 | 81.767 | Odoribacter sp. AF15-53 | 4 |  96 |\n",
       "| cds | 118 |  287 | 141 | 141 | 0 | 287r  | PRIMER_RIGHT | VCGYTGYTCYTGWACCAAYC  | 20 |  96 | 2 |  988 | 1008 | 59.77764 | 2.222613 | 53.33333 | 5.527708 | 11.994356 | 17.8257607 |  -6.144309 | 13.8730856 | 81.767 | Odoribacter sp. AF15-53 | 4 |  96 |\n",
       "| cds | 118 |  553 | 101 | 101 | 0 | 553f  | PRIMER_LEFT  | AAGGAYWTBCGNGTSGATGT  | 20 |  96 | 1 |  867 |  887 | 59.21880 | 2.210807 | 48.33333 | 4.249183 | 29.190064 | 23.3863626 |  -3.187227 | 24.1165857 | 81.767 | Odoribacter sp. AF15-53 | 4 | 128 |\n",
       "| cds | 118 |  553 | 101 | 101 | 0 | 553r  | PRIMER_RIGHT | CCRATRATYAAYCCNCGRTC  | 20 | 128 | 2 |  948 |  968 | 56.78183 | 2.912989 | 50.00000 | 6.123724 | 21.136796 | 18.6641357 |  -2.824651 | 22.4057285 | 81.767 | Odoribacter sp. AF15-53 | 4 | 128 |\n",
       "| cds | 118 |  642 | 230 | 230 | 0 | 642f  | PRIMER_LEFT  | CGGATRTYTTYGTDCCRGCK  | 20 |  96 | 4 | 1237 | 1257 | 60.32781 | 2.703100 | 54.16667 | 6.066758 | 20.732458 | 20.8653216 | -10.075399 | 24.5760581 | 81.767 | Odoribacter sp. AF15-53 | 4 | 128 |\n",
       "| cds | 118 |  642 | 230 | 230 | 0 | 642r  | PRIMER_RIGHT | YYCCTTSTCRTYCCGYTTYA  | 20 | 128 | 4 | 1447 | 1467 | 58.92361 | 2.556469 | 50.00000 | 6.123724 |  0.000000 |  0.0000000 | -41.763740 |  0.3687041 | 81.767 | Odoribacter sp. AF15-53 | 4 | 128 |\n",
       "\n"
      ],
      "text/plain": [
       "   gene_type cluster_id primer_set amplicon_size_consensus amplicon_size_avg\n",
       "1  cds       469         677       122                     122              \n",
       "2  cds       469         677       122                     122              \n",
       "3  cds       469          84       122                     122              \n",
       "4  cds       469          84       122                     122              \n",
       "5  cds       469         700       119                     119              \n",
       "6  cds       469         700       119                     119              \n",
       "7  cds       469         720       119                     119              \n",
       "8  cds       469         720       119                     119              \n",
       "9  cds       469         308       118                     118              \n",
       "10 cds       469         308       118                     118              \n",
       "11 cds       469          25       120                     120              \n",
       "12 cds       469          25       120                     120              \n",
       "13 cds       469         448       119                     119              \n",
       "14 cds       469         448       119                     119              \n",
       "15 cds       469         785       125                     125              \n",
       "16 cds       469         785       125                     125              \n",
       "17 cds       469        1000       120                     120              \n",
       "18 cds       469        1000       120                     120              \n",
       "19 cds       469         910       109                     109              \n",
       "20 cds       469         910       109                     109              \n",
       "21 cds       118         555       107                     107              \n",
       "22 cds       118         555       107                     107              \n",
       "23 cds       118         286       140                     140              \n",
       "24 cds       118         286       140                     140              \n",
       "25 cds       118         287       141                     141              \n",
       "26 cds       118         287       141                     141              \n",
       "27 cds       118         553       101                     101              \n",
       "28 cds       118         553       101                     101              \n",
       "29 cds       118         642       230                     230              \n",
       "30 cds       118         642       230                     230              \n",
       "   amplicon_size_sd primer_id primer_type  sequence              length\n",
       "1  0                677f      PRIMER_LEFT  AGTCCSGGDTATGAYCCGCA  20    \n",
       "2  0                677r      PRIMER_RIGHT CCBGGGGGATAYTGYGCCA   19    \n",
       "3  0                84f       PRIMER_LEFT  AGTCCSGGDTATGAYCCGCA  20    \n",
       "4  0                84r       PRIMER_RIGHT CCBGGGGGATAYTGYGCCAT  20    \n",
       "5  0                700f      PRIMER_LEFT  AGTCCSGGDTATGAYCCGCA  20    \n",
       "6  0                700r      PRIMER_RIGHT GGGGGATAYTGYGCCATSGT  20    \n",
       "7  0                720f      PRIMER_LEFT  AGTCCSGGDTATGAYCCGCA  20    \n",
       "8  0                720r      PRIMER_RIGHT GGGGGATAYTGYGCCATSGTY 21    \n",
       "9  0                308f      PRIMER_LEFT  AGTCCSGGDTATGAYCCGCA  20    \n",
       "10 0                308r      PRIMER_RIGHT GGGGATAYTGYGCCATSGTYG 21    \n",
       "11 0                25f       PRIMER_LEFT  AGTCCSGGDTATGAYCCGCA  20    \n",
       "12 0                25r       PRIMER_RIGHT BGGGGGATAYTGYGCCATSG  20    \n",
       "13 0                448f      PRIMER_LEFT  GTCCSGGDTATGAYCCGCAA  20    \n",
       "14 0                448r      PRIMER_RIGHT BGGGGGATAYTGYGCCATSG  20    \n",
       "15 0                785f      PRIMER_LEFT  TCYAGTCCSGGDTATGAYCCG 21    \n",
       "16 0                785r      PRIMER_RIGHT CCBGGGGGATAYTGYGCCAT  20    \n",
       "17 0                1000f     PRIMER_LEFT  GTCCSGGDTATGAYCCGCAA  20    \n",
       "18 0                1000r     PRIMER_RIGHT CBGGGGGATAYTGYGCCATS  20    \n",
       "19 0                910f      PRIMER_LEFT  GTCCSGGDTATGAYCCGCAA  20    \n",
       "20 0                910r      PRIMER_RIGHT TGYGCCATSGTYGTTCTRTC  20    \n",
       "21 0                555f      PRIMER_LEFT  AAGGAYWTBCGNGTSGATGT  20    \n",
       "22 0                555r      PRIMER_RIGHT CGRCGHCCRATRATYAAYCC  20    \n",
       "23 0                286f      PRIMER_LEFT  AAGGAYWTBCGNGTSGATGT  20    \n",
       "24 0                286r      PRIMER_RIGHT CGYTGYTCYTGWACCAAYCC  20    \n",
       "25 0                287f      PRIMER_LEFT  AAGGAYWTBCGNGTSGATGT  20    \n",
       "26 0                287r      PRIMER_RIGHT VCGYTGYTCYTGWACCAAYC  20    \n",
       "27 0                553f      PRIMER_LEFT  AAGGAYWTBCGNGTSGATGT  20    \n",
       "28 0                553r      PRIMER_RIGHT CCRATRATYAAYCCNCGRTC  20    \n",
       "29 0                642f      PRIMER_LEFT  CGGATRTYTTYGTDCCRGCK  20    \n",
       "30 0                642r      PRIMER_RIGHT YYCCTTSTCRTYCCGYTTYA  20    \n",
       "   degeneracy degeneracy_3prime position_start position_end Tm_avg   Tm_sd   \n",
       "1   12        1                  807            827         63.37921 1.585835\n",
       "2   12        2                  910            929         63.99539 2.171444\n",
       "3   12        1                  807            827         63.37921 1.585835\n",
       "4   12        1                  909            929         64.24558 2.069738\n",
       "5   12        1                  807            827         63.37921 1.585835\n",
       "6    8        2                  906            926         62.86230 1.692288\n",
       "7   12        1                  807            827         63.37921 1.585835\n",
       "8   16        4                  905            926         63.52770 1.607231\n",
       "9   12        1                  807            827         63.37921 1.585835\n",
       "10  16        4                  904            925         63.09697 1.809699\n",
       "11  12        1                  807            827         63.37921 1.585835\n",
       "12  24        2                  907            927         63.42396 1.720398\n",
       "13  12        1                  808            828         62.46664 1.557512\n",
       "14  24        2                  907            927         63.42396 1.720398\n",
       "15  24        2                  804            825         62.16733 2.085810\n",
       "16  12        1                  909            929         64.24558 2.069738\n",
       "17  12        1                  808            828         62.46664 1.557512\n",
       "18  24        2                  908            928         62.87830 2.042500\n",
       "19  12        1                  808            828         62.46664 1.557512\n",
       "20  16        2                  897            917         60.27747 2.226624\n",
       "21  96        1                  867            887         59.21880 2.210807\n",
       "22  96        2                  954            974         59.31237 2.922306\n",
       "23  96        1                  867            887         59.21880 2.210807\n",
       "24  32        2                  987           1007         59.80526 2.153698\n",
       "25  96        1                  867            887         59.21880 2.210807\n",
       "26  96        2                  988           1008         59.77764 2.222613\n",
       "27  96        1                  867            887         59.21880 2.210807\n",
       "28 128        2                  948            968         56.78183 2.912989\n",
       "29  96        4                 1237           1257         60.32781 2.703100\n",
       "30 128        4                 1447           1467         58.92361 2.556469\n",
       "   GC_avg   GC_sd    hairpin_avg hairpin_sd homodimer_avg homodimer_sd pident\n",
       "1  59.16667 3.435921 54.529677   10.4477986  12.546118    13.3155804   81.107\n",
       "2  66.66667 4.472824 42.790866    2.5680978   3.518264    11.0215662   81.107\n",
       "3  59.16667 3.435921 54.529677   10.4477986  12.546118    13.3155804   81.107\n",
       "4  63.33333 4.249183 42.928759    2.8664437   3.518264    11.0215662   81.107\n",
       "5  59.16667 3.435921 54.529677   10.4477986  12.546118    13.3155804   81.107\n",
       "6  60.00000 3.535534 41.156774    0.1051210   8.175715     7.8528329   81.107\n",
       "7  59.16667 3.435921 54.529677   10.4477986  12.546118    13.3155804   81.107\n",
       "8  59.52381 4.123930 41.156774    0.1051210   8.175715     7.8528329   81.107\n",
       "9  59.16667 3.435921 54.529677   10.4477986  12.546118    13.3155804   81.107\n",
       "10 59.52381 4.123930 40.972038    0.2898574   8.849733     8.6056358   81.107\n",
       "11 59.16667 3.435921 54.529677   10.4477986  12.546118    13.3155804   81.107\n",
       "12 63.33333 4.249183 42.405568    1.7528983   4.674529     5.1073084   81.107\n",
       "13 59.16667 3.435921 54.529677   10.4477986  12.546118    13.3155804   81.107\n",
       "14 63.33333 4.249183 42.405568    1.7528983   4.674529     5.1073084   81.107\n",
       "15 58.73016 4.046841 54.810400   10.1710443  11.909481    12.9429433   81.107\n",
       "16 63.33333 4.249183 42.928759    2.8664437   3.518264    11.0215662   81.107\n",
       "17 59.16667 3.435921 54.529677   10.4477986  12.546118    13.3155804   81.107\n",
       "18 63.33333 4.249183 42.928759    2.8664437  -1.070190     8.1980946   81.107\n",
       "19 59.16667 3.435921 54.529677   10.4477986  12.546118    13.3155804   81.107\n",
       "20 52.50000 4.330127  9.590313   16.6109088   6.825856    12.1537464   81.107\n",
       "21 48.33333 4.249183 29.190064   23.3863626  -3.187227    24.1165857   81.767\n",
       "22 54.16667 6.066758 23.698622   23.9999455   3.609712    16.5893977   81.767\n",
       "23 48.33333 4.249183 29.190064   23.3863626  -3.187227    24.1165857   81.767\n",
       "24 55.00000 5.000000 11.994356   17.8257607  -6.009949    13.5798078   81.767\n",
       "25 48.33333 4.249183 29.190064   23.3863626  -3.187227    24.1165857   81.767\n",
       "26 53.33333 5.527708 11.994356   17.8257607  -6.144309    13.8730856   81.767\n",
       "27 48.33333 4.249183 29.190064   23.3863626  -3.187227    24.1165857   81.767\n",
       "28 50.00000 6.123724 21.136796   18.6641357  -2.824651    22.4057285   81.767\n",
       "29 54.16667 6.066758 20.732458   20.8653216 -10.075399    24.5760581   81.767\n",
       "30 50.00000 6.123724  0.000000    0.0000000 -41.763740     0.3687041   81.767\n",
       "   sscinames               max_degeneracy_3prime max_degeneracy\n",
       "1  Odoribacter sp. AF15-53 2                      64           \n",
       "2  Odoribacter sp. AF15-53 2                      64           \n",
       "3  Odoribacter sp. AF15-53 3                      12           \n",
       "4  Odoribacter sp. AF15-53 3                      12           \n",
       "5  Odoribacter sp. AF15-53 3                      36           \n",
       "6  Odoribacter sp. AF15-53 3                      36           \n",
       "7  Odoribacter sp. AF15-53 4                      16           \n",
       "8  Odoribacter sp. AF15-53 4                      16           \n",
       "9  Odoribacter sp. AF15-53 4                      24           \n",
       "10 Odoribacter sp. AF15-53 4                      24           \n",
       "11 Odoribacter sp. AF15-53 4                      32           \n",
       "12 Odoribacter sp. AF15-53 4                      32           \n",
       "13 Odoribacter sp. AF15-53 4                      32           \n",
       "14 Odoribacter sp. AF15-53 4                      32           \n",
       "15 Odoribacter sp. AF15-53 4                      64           \n",
       "16 Odoribacter sp. AF15-53 4                      64           \n",
       "17 Odoribacter sp. AF15-53 4                     128           \n",
       "18 Odoribacter sp. AF15-53 4                     128           \n",
       "19 Odoribacter sp. AF15-53 6                      96           \n",
       "20 Odoribacter sp. AF15-53 6                      96           \n",
       "21 Odoribacter sp. AF15-53 2                      96           \n",
       "22 Odoribacter sp. AF15-53 2                      96           \n",
       "23 Odoribacter sp. AF15-53 4                      96           \n",
       "24 Odoribacter sp. AF15-53 4                      96           \n",
       "25 Odoribacter sp. AF15-53 4                      96           \n",
       "26 Odoribacter sp. AF15-53 4                      96           \n",
       "27 Odoribacter sp. AF15-53 4                     128           \n",
       "28 Odoribacter sp. AF15-53 4                     128           \n",
       "29 Odoribacter sp. AF15-53 4                     128           \n",
       "30 Odoribacter sp. AF15-53 4                     128           "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# arrange by \"best\" primers\n",
    "primer_info_jf = primer_info_j %>%\n",
    "    group_by(primer_set) %>%\n",
    "    mutate(max_degeneracy_3prime = max(degeneracy_3prime),  # max of [Fwd,Rev]\n",
    "           max_degeneracy = max(degeneracy)) %>%\n",
    "    ungroup() %>%\n",
    "    arrange(pident, max_degeneracy_3prime, max_degeneracy) %>%\n",
    "    head(n=30)\n",
    "\n",
    "df.dims(30,40)\n",
    "primer_info_jf\n",
    "df.dims()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### w/ gene annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-28T13:08:54.107350Z",
     "iopub.status.busy": "2021-04-28T13:08:54.106636Z",
     "iopub.status.idle": "2021-04-28T13:08:54.228146Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A tibble: 30 × 27</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>gene_type</th><th scope=col>cluster_id</th><th scope=col>primer_set</th><th scope=col>amplicon_size_consensus</th><th scope=col>amplicon_size_avg</th><th scope=col>amplicon_size_sd</th><th scope=col>primer_id</th><th scope=col>primer_type</th><th scope=col>sequence</th><th scope=col>length</th><th scope=col>degeneracy</th><th scope=col>degeneracy_3prime</th><th scope=col>position_start</th><th scope=col>position_end</th><th scope=col>Tm_avg</th><th scope=col>Tm_sd</th><th scope=col>GC_avg</th><th scope=col>GC_sd</th><th scope=col>hairpin_avg</th><th scope=col>hairpin_sd</th><th scope=col>homodimer_avg</th><th scope=col>homodimer_sd</th><th scope=col>pident</th><th scope=col>sscinames</th><th scope=col>max_degeneracy_3prime</th><th scope=col>max_degeneracy</th><th scope=col>annotation</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>cds</td><td>469</td><td> 677</td><td>122</td><td>122</td><td>0</td><td>677f </td><td>PRIMER_LEFT </td><td>AGTCCSGGDTATGAYCCGCA </td><td>20</td><td> 12</td><td>1</td><td> 807</td><td> 827</td><td>63.37921</td><td>1.585835</td><td>59.16667</td><td>3.435921</td><td>54.529677</td><td>10.4477986</td><td> 12.546118</td><td>13.3155804</td><td>81.107</td><td>Odoribacter sp. AF15-53</td><td>2</td><td> 64</td><td>Peptidoglycan D%2CD-transpeptidase MrdA</td></tr>\n",
       "\t<tr><td>cds</td><td>469</td><td> 677</td><td>122</td><td>122</td><td>0</td><td>677r </td><td>PRIMER_RIGHT</td><td>CCBGGGGGATAYTGYGCCA  </td><td>19</td><td> 12</td><td>2</td><td> 910</td><td> 929</td><td>63.99539</td><td>2.171444</td><td>66.66667</td><td>4.472824</td><td>42.790866</td><td> 2.5680978</td><td>  3.518264</td><td>11.0215662</td><td>81.107</td><td>Odoribacter sp. AF15-53</td><td>2</td><td> 64</td><td>Peptidoglycan D%2CD-transpeptidase MrdA</td></tr>\n",
       "\t<tr><td>cds</td><td>469</td><td>  84</td><td>122</td><td>122</td><td>0</td><td>84f  </td><td>PRIMER_LEFT </td><td>AGTCCSGGDTATGAYCCGCA </td><td>20</td><td> 12</td><td>1</td><td> 807</td><td> 827</td><td>63.37921</td><td>1.585835</td><td>59.16667</td><td>3.435921</td><td>54.529677</td><td>10.4477986</td><td> 12.546118</td><td>13.3155804</td><td>81.107</td><td>Odoribacter sp. AF15-53</td><td>3</td><td> 12</td><td>Peptidoglycan D%2CD-transpeptidase MrdA</td></tr>\n",
       "\t<tr><td>cds</td><td>469</td><td>  84</td><td>122</td><td>122</td><td>0</td><td>84r  </td><td>PRIMER_RIGHT</td><td>CCBGGGGGATAYTGYGCCAT </td><td>20</td><td> 12</td><td>1</td><td> 909</td><td> 929</td><td>64.24558</td><td>2.069738</td><td>63.33333</td><td>4.249183</td><td>42.928759</td><td> 2.8664437</td><td>  3.518264</td><td>11.0215662</td><td>81.107</td><td>Odoribacter sp. AF15-53</td><td>3</td><td> 12</td><td>Peptidoglycan D%2CD-transpeptidase MrdA</td></tr>\n",
       "\t<tr><td>cds</td><td>469</td><td> 700</td><td>119</td><td>119</td><td>0</td><td>700f </td><td>PRIMER_LEFT </td><td>AGTCCSGGDTATGAYCCGCA </td><td>20</td><td> 12</td><td>1</td><td> 807</td><td> 827</td><td>63.37921</td><td>1.585835</td><td>59.16667</td><td>3.435921</td><td>54.529677</td><td>10.4477986</td><td> 12.546118</td><td>13.3155804</td><td>81.107</td><td>Odoribacter sp. AF15-53</td><td>3</td><td> 36</td><td>Peptidoglycan D%2CD-transpeptidase MrdA</td></tr>\n",
       "\t<tr><td>cds</td><td>469</td><td> 700</td><td>119</td><td>119</td><td>0</td><td>700r </td><td>PRIMER_RIGHT</td><td>GGGGGATAYTGYGCCATSGT </td><td>20</td><td>  8</td><td>2</td><td> 906</td><td> 926</td><td>62.86230</td><td>1.692288</td><td>60.00000</td><td>3.535534</td><td>41.156774</td><td> 0.1051210</td><td>  8.175715</td><td> 7.8528329</td><td>81.107</td><td>Odoribacter sp. AF15-53</td><td>3</td><td> 36</td><td>Peptidoglycan D%2CD-transpeptidase MrdA</td></tr>\n",
       "\t<tr><td>cds</td><td>469</td><td> 720</td><td>119</td><td>119</td><td>0</td><td>720f </td><td>PRIMER_LEFT </td><td>AGTCCSGGDTATGAYCCGCA </td><td>20</td><td> 12</td><td>1</td><td> 807</td><td> 827</td><td>63.37921</td><td>1.585835</td><td>59.16667</td><td>3.435921</td><td>54.529677</td><td>10.4477986</td><td> 12.546118</td><td>13.3155804</td><td>81.107</td><td>Odoribacter sp. AF15-53</td><td>4</td><td> 16</td><td>Peptidoglycan D%2CD-transpeptidase MrdA</td></tr>\n",
       "\t<tr><td>cds</td><td>469</td><td> 720</td><td>119</td><td>119</td><td>0</td><td>720r </td><td>PRIMER_RIGHT</td><td>GGGGGATAYTGYGCCATSGTY</td><td>21</td><td> 16</td><td>4</td><td> 905</td><td> 926</td><td>63.52770</td><td>1.607231</td><td>59.52381</td><td>4.123930</td><td>41.156774</td><td> 0.1051210</td><td>  8.175715</td><td> 7.8528329</td><td>81.107</td><td>Odoribacter sp. AF15-53</td><td>4</td><td> 16</td><td>Peptidoglycan D%2CD-transpeptidase MrdA</td></tr>\n",
       "\t<tr><td>cds</td><td>469</td><td> 308</td><td>118</td><td>118</td><td>0</td><td>308f </td><td>PRIMER_LEFT </td><td>AGTCCSGGDTATGAYCCGCA </td><td>20</td><td> 12</td><td>1</td><td> 807</td><td> 827</td><td>63.37921</td><td>1.585835</td><td>59.16667</td><td>3.435921</td><td>54.529677</td><td>10.4477986</td><td> 12.546118</td><td>13.3155804</td><td>81.107</td><td>Odoribacter sp. AF15-53</td><td>4</td><td> 24</td><td>Peptidoglycan D%2CD-transpeptidase MrdA</td></tr>\n",
       "\t<tr><td>cds</td><td>469</td><td> 308</td><td>118</td><td>118</td><td>0</td><td>308r </td><td>PRIMER_RIGHT</td><td>GGGGATAYTGYGCCATSGTYG</td><td>21</td><td> 16</td><td>4</td><td> 904</td><td> 925</td><td>63.09697</td><td>1.809699</td><td>59.52381</td><td>4.123930</td><td>40.972038</td><td> 0.2898574</td><td>  8.849733</td><td> 8.6056358</td><td>81.107</td><td>Odoribacter sp. AF15-53</td><td>4</td><td> 24</td><td>Peptidoglycan D%2CD-transpeptidase MrdA</td></tr>\n",
       "\t<tr><td>cds</td><td>469</td><td>  25</td><td>120</td><td>120</td><td>0</td><td>25f  </td><td>PRIMER_LEFT </td><td>AGTCCSGGDTATGAYCCGCA </td><td>20</td><td> 12</td><td>1</td><td> 807</td><td> 827</td><td>63.37921</td><td>1.585835</td><td>59.16667</td><td>3.435921</td><td>54.529677</td><td>10.4477986</td><td> 12.546118</td><td>13.3155804</td><td>81.107</td><td>Odoribacter sp. AF15-53</td><td>4</td><td> 32</td><td>Peptidoglycan D%2CD-transpeptidase MrdA</td></tr>\n",
       "\t<tr><td>cds</td><td>469</td><td>  25</td><td>120</td><td>120</td><td>0</td><td>25r  </td><td>PRIMER_RIGHT</td><td>BGGGGGATAYTGYGCCATSG </td><td>20</td><td> 24</td><td>2</td><td> 907</td><td> 927</td><td>63.42396</td><td>1.720398</td><td>63.33333</td><td>4.249183</td><td>42.405568</td><td> 1.7528983</td><td>  4.674529</td><td> 5.1073084</td><td>81.107</td><td>Odoribacter sp. AF15-53</td><td>4</td><td> 32</td><td>Peptidoglycan D%2CD-transpeptidase MrdA</td></tr>\n",
       "\t<tr><td>cds</td><td>469</td><td> 448</td><td>119</td><td>119</td><td>0</td><td>448f </td><td>PRIMER_LEFT </td><td>GTCCSGGDTATGAYCCGCAA </td><td>20</td><td> 12</td><td>1</td><td> 808</td><td> 828</td><td>62.46664</td><td>1.557512</td><td>59.16667</td><td>3.435921</td><td>54.529677</td><td>10.4477986</td><td> 12.546118</td><td>13.3155804</td><td>81.107</td><td>Odoribacter sp. AF15-53</td><td>4</td><td> 32</td><td>Peptidoglycan D%2CD-transpeptidase MrdA</td></tr>\n",
       "\t<tr><td>cds</td><td>469</td><td> 448</td><td>119</td><td>119</td><td>0</td><td>448r </td><td>PRIMER_RIGHT</td><td>BGGGGGATAYTGYGCCATSG </td><td>20</td><td> 24</td><td>2</td><td> 907</td><td> 927</td><td>63.42396</td><td>1.720398</td><td>63.33333</td><td>4.249183</td><td>42.405568</td><td> 1.7528983</td><td>  4.674529</td><td> 5.1073084</td><td>81.107</td><td>Odoribacter sp. AF15-53</td><td>4</td><td> 32</td><td>Peptidoglycan D%2CD-transpeptidase MrdA</td></tr>\n",
       "\t<tr><td>cds</td><td>469</td><td> 785</td><td>125</td><td>125</td><td>0</td><td>785f </td><td>PRIMER_LEFT </td><td>TCYAGTCCSGGDTATGAYCCG</td><td>21</td><td> 24</td><td>2</td><td> 804</td><td> 825</td><td>62.16733</td><td>2.085810</td><td>58.73016</td><td>4.046841</td><td>54.810400</td><td>10.1710443</td><td> 11.909481</td><td>12.9429433</td><td>81.107</td><td>Odoribacter sp. AF15-53</td><td>4</td><td> 64</td><td>Peptidoglycan D%2CD-transpeptidase MrdA</td></tr>\n",
       "\t<tr><td>cds</td><td>469</td><td> 785</td><td>125</td><td>125</td><td>0</td><td>785r </td><td>PRIMER_RIGHT</td><td>CCBGGGGGATAYTGYGCCAT </td><td>20</td><td> 12</td><td>1</td><td> 909</td><td> 929</td><td>64.24558</td><td>2.069738</td><td>63.33333</td><td>4.249183</td><td>42.928759</td><td> 2.8664437</td><td>  3.518264</td><td>11.0215662</td><td>81.107</td><td>Odoribacter sp. AF15-53</td><td>4</td><td> 64</td><td>Peptidoglycan D%2CD-transpeptidase MrdA</td></tr>\n",
       "\t<tr><td>cds</td><td>469</td><td>1000</td><td>120</td><td>120</td><td>0</td><td>1000f</td><td>PRIMER_LEFT </td><td>GTCCSGGDTATGAYCCGCAA </td><td>20</td><td> 12</td><td>1</td><td> 808</td><td> 828</td><td>62.46664</td><td>1.557512</td><td>59.16667</td><td>3.435921</td><td>54.529677</td><td>10.4477986</td><td> 12.546118</td><td>13.3155804</td><td>81.107</td><td>Odoribacter sp. AF15-53</td><td>4</td><td>128</td><td>Peptidoglycan D%2CD-transpeptidase MrdA</td></tr>\n",
       "\t<tr><td>cds</td><td>469</td><td>1000</td><td>120</td><td>120</td><td>0</td><td>1000r</td><td>PRIMER_RIGHT</td><td>CBGGGGGATAYTGYGCCATS </td><td>20</td><td> 24</td><td>2</td><td> 908</td><td> 928</td><td>62.87830</td><td>2.042500</td><td>63.33333</td><td>4.249183</td><td>42.928759</td><td> 2.8664437</td><td> -1.070190</td><td> 8.1980946</td><td>81.107</td><td>Odoribacter sp. AF15-53</td><td>4</td><td>128</td><td>Peptidoglycan D%2CD-transpeptidase MrdA</td></tr>\n",
       "\t<tr><td>cds</td><td>469</td><td> 910</td><td>109</td><td>109</td><td>0</td><td>910f </td><td>PRIMER_LEFT </td><td>GTCCSGGDTATGAYCCGCAA </td><td>20</td><td> 12</td><td>1</td><td> 808</td><td> 828</td><td>62.46664</td><td>1.557512</td><td>59.16667</td><td>3.435921</td><td>54.529677</td><td>10.4477986</td><td> 12.546118</td><td>13.3155804</td><td>81.107</td><td>Odoribacter sp. AF15-53</td><td>6</td><td> 96</td><td>Peptidoglycan D%2CD-transpeptidase MrdA</td></tr>\n",
       "\t<tr><td>cds</td><td>469</td><td> 910</td><td>109</td><td>109</td><td>0</td><td>910r </td><td>PRIMER_RIGHT</td><td>TGYGCCATSGTYGTTCTRTC </td><td>20</td><td> 16</td><td>2</td><td> 897</td><td> 917</td><td>60.27747</td><td>2.226624</td><td>52.50000</td><td>4.330127</td><td> 9.590313</td><td>16.6109088</td><td>  6.825856</td><td>12.1537464</td><td>81.107</td><td>Odoribacter sp. AF15-53</td><td>6</td><td> 96</td><td>Peptidoglycan D%2CD-transpeptidase MrdA</td></tr>\n",
       "\t<tr><td>cds</td><td>118</td><td> 555</td><td>107</td><td>107</td><td>0</td><td>555f </td><td>PRIMER_LEFT </td><td>AAGGAYWTBCGNGTSGATGT </td><td>20</td><td> 96</td><td>1</td><td> 867</td><td> 887</td><td>59.21880</td><td>2.210807</td><td>48.33333</td><td>4.249183</td><td>29.190064</td><td>23.3863626</td><td> -3.187227</td><td>24.1165857</td><td>81.767</td><td>Odoribacter sp. AF15-53</td><td>2</td><td> 96</td><td>hypothetical protein                   </td></tr>\n",
       "\t<tr><td>cds</td><td>118</td><td> 555</td><td>107</td><td>107</td><td>0</td><td>555r </td><td>PRIMER_RIGHT</td><td>CGRCGHCCRATRATYAAYCC </td><td>20</td><td> 96</td><td>2</td><td> 954</td><td> 974</td><td>59.31237</td><td>2.922306</td><td>54.16667</td><td>6.066758</td><td>23.698622</td><td>23.9999455</td><td>  3.609712</td><td>16.5893977</td><td>81.767</td><td>Odoribacter sp. AF15-53</td><td>2</td><td> 96</td><td>hypothetical protein                   </td></tr>\n",
       "\t<tr><td>cds</td><td>118</td><td> 286</td><td>140</td><td>140</td><td>0</td><td>286f </td><td>PRIMER_LEFT </td><td>AAGGAYWTBCGNGTSGATGT </td><td>20</td><td> 96</td><td>1</td><td> 867</td><td> 887</td><td>59.21880</td><td>2.210807</td><td>48.33333</td><td>4.249183</td><td>29.190064</td><td>23.3863626</td><td> -3.187227</td><td>24.1165857</td><td>81.767</td><td>Odoribacter sp. AF15-53</td><td>4</td><td> 96</td><td>hypothetical protein                   </td></tr>\n",
       "\t<tr><td>cds</td><td>118</td><td> 286</td><td>140</td><td>140</td><td>0</td><td>286r </td><td>PRIMER_RIGHT</td><td>CGYTGYTCYTGWACCAAYCC </td><td>20</td><td> 32</td><td>2</td><td> 987</td><td>1007</td><td>59.80526</td><td>2.153698</td><td>55.00000</td><td>5.000000</td><td>11.994356</td><td>17.8257607</td><td> -6.009949</td><td>13.5798078</td><td>81.767</td><td>Odoribacter sp. AF15-53</td><td>4</td><td> 96</td><td>hypothetical protein                   </td></tr>\n",
       "\t<tr><td>cds</td><td>118</td><td> 287</td><td>141</td><td>141</td><td>0</td><td>287f </td><td>PRIMER_LEFT </td><td>AAGGAYWTBCGNGTSGATGT </td><td>20</td><td> 96</td><td>1</td><td> 867</td><td> 887</td><td>59.21880</td><td>2.210807</td><td>48.33333</td><td>4.249183</td><td>29.190064</td><td>23.3863626</td><td> -3.187227</td><td>24.1165857</td><td>81.767</td><td>Odoribacter sp. AF15-53</td><td>4</td><td> 96</td><td>hypothetical protein                   </td></tr>\n",
       "\t<tr><td>cds</td><td>118</td><td> 287</td><td>141</td><td>141</td><td>0</td><td>287r </td><td>PRIMER_RIGHT</td><td>VCGYTGYTCYTGWACCAAYC </td><td>20</td><td> 96</td><td>2</td><td> 988</td><td>1008</td><td>59.77764</td><td>2.222613</td><td>53.33333</td><td>5.527708</td><td>11.994356</td><td>17.8257607</td><td> -6.144309</td><td>13.8730856</td><td>81.767</td><td>Odoribacter sp. AF15-53</td><td>4</td><td> 96</td><td>hypothetical protein                   </td></tr>\n",
       "\t<tr><td>cds</td><td>118</td><td> 553</td><td>101</td><td>101</td><td>0</td><td>553f </td><td>PRIMER_LEFT </td><td>AAGGAYWTBCGNGTSGATGT </td><td>20</td><td> 96</td><td>1</td><td> 867</td><td> 887</td><td>59.21880</td><td>2.210807</td><td>48.33333</td><td>4.249183</td><td>29.190064</td><td>23.3863626</td><td> -3.187227</td><td>24.1165857</td><td>81.767</td><td>Odoribacter sp. AF15-53</td><td>4</td><td>128</td><td>hypothetical protein                   </td></tr>\n",
       "\t<tr><td>cds</td><td>118</td><td> 553</td><td>101</td><td>101</td><td>0</td><td>553r </td><td>PRIMER_RIGHT</td><td>CCRATRATYAAYCCNCGRTC </td><td>20</td><td>128</td><td>2</td><td> 948</td><td> 968</td><td>56.78183</td><td>2.912989</td><td>50.00000</td><td>6.123724</td><td>21.136796</td><td>18.6641357</td><td> -2.824651</td><td>22.4057285</td><td>81.767</td><td>Odoribacter sp. AF15-53</td><td>4</td><td>128</td><td>hypothetical protein                   </td></tr>\n",
       "\t<tr><td>cds</td><td>118</td><td> 642</td><td>230</td><td>230</td><td>0</td><td>642f </td><td>PRIMER_LEFT </td><td>CGGATRTYTTYGTDCCRGCK </td><td>20</td><td> 96</td><td>4</td><td>1237</td><td>1257</td><td>60.32781</td><td>2.703100</td><td>54.16667</td><td>6.066758</td><td>20.732458</td><td>20.8653216</td><td>-10.075399</td><td>24.5760581</td><td>81.767</td><td>Odoribacter sp. AF15-53</td><td>4</td><td>128</td><td>hypothetical protein                   </td></tr>\n",
       "\t<tr><td>cds</td><td>118</td><td> 642</td><td>230</td><td>230</td><td>0</td><td>642r </td><td>PRIMER_RIGHT</td><td>YYCCTTSTCRTYCCGYTTYA </td><td>20</td><td>128</td><td>4</td><td>1447</td><td>1467</td><td>58.92361</td><td>2.556469</td><td>50.00000</td><td>6.123724</td><td> 0.000000</td><td> 0.0000000</td><td>-41.763740</td><td> 0.3687041</td><td>81.767</td><td>Odoribacter sp. AF15-53</td><td>4</td><td>128</td><td>hypothetical protein                   </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 30 × 27\n",
       "\\begin{tabular}{lllllllllllllllllllllllllll}\n",
       " gene\\_type & cluster\\_id & primer\\_set & amplicon\\_size\\_consensus & amplicon\\_size\\_avg & amplicon\\_size\\_sd & primer\\_id & primer\\_type & sequence & length & degeneracy & degeneracy\\_3prime & position\\_start & position\\_end & Tm\\_avg & Tm\\_sd & GC\\_avg & GC\\_sd & hairpin\\_avg & hairpin\\_sd & homodimer\\_avg & homodimer\\_sd & pident & sscinames & max\\_degeneracy\\_3prime & max\\_degeneracy & annotation\\\\\n",
       " <chr> & <int> & <int> & <int> & <dbl> & <dbl> & <fct> & <fct> & <fct> & <int> & <int> & <int> & <int> & <int> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <fct> & <int> & <int> & <fct>\\\\\n",
       "\\hline\n",
       "\t cds & 469 &  677 & 122 & 122 & 0 & 677f  & PRIMER\\_LEFT  & AGTCCSGGDTATGAYCCGCA  & 20 &  12 & 1 &  807 &  827 & 63.37921 & 1.585835 & 59.16667 & 3.435921 & 54.529677 & 10.4477986 &  12.546118 & 13.3155804 & 81.107 & Odoribacter sp. AF15-53 & 2 &  64 & Peptidoglycan D\\%2CD-transpeptidase MrdA\\\\\n",
       "\t cds & 469 &  677 & 122 & 122 & 0 & 677r  & PRIMER\\_RIGHT & CCBGGGGGATAYTGYGCCA   & 19 &  12 & 2 &  910 &  929 & 63.99539 & 2.171444 & 66.66667 & 4.472824 & 42.790866 &  2.5680978 &   3.518264 & 11.0215662 & 81.107 & Odoribacter sp. AF15-53 & 2 &  64 & Peptidoglycan D\\%2CD-transpeptidase MrdA\\\\\n",
       "\t cds & 469 &   84 & 122 & 122 & 0 & 84f   & PRIMER\\_LEFT  & AGTCCSGGDTATGAYCCGCA  & 20 &  12 & 1 &  807 &  827 & 63.37921 & 1.585835 & 59.16667 & 3.435921 & 54.529677 & 10.4477986 &  12.546118 & 13.3155804 & 81.107 & Odoribacter sp. AF15-53 & 3 &  12 & Peptidoglycan D\\%2CD-transpeptidase MrdA\\\\\n",
       "\t cds & 469 &   84 & 122 & 122 & 0 & 84r   & PRIMER\\_RIGHT & CCBGGGGGATAYTGYGCCAT  & 20 &  12 & 1 &  909 &  929 & 64.24558 & 2.069738 & 63.33333 & 4.249183 & 42.928759 &  2.8664437 &   3.518264 & 11.0215662 & 81.107 & Odoribacter sp. AF15-53 & 3 &  12 & Peptidoglycan D\\%2CD-transpeptidase MrdA\\\\\n",
       "\t cds & 469 &  700 & 119 & 119 & 0 & 700f  & PRIMER\\_LEFT  & AGTCCSGGDTATGAYCCGCA  & 20 &  12 & 1 &  807 &  827 & 63.37921 & 1.585835 & 59.16667 & 3.435921 & 54.529677 & 10.4477986 &  12.546118 & 13.3155804 & 81.107 & Odoribacter sp. AF15-53 & 3 &  36 & Peptidoglycan D\\%2CD-transpeptidase MrdA\\\\\n",
       "\t cds & 469 &  700 & 119 & 119 & 0 & 700r  & PRIMER\\_RIGHT & GGGGGATAYTGYGCCATSGT  & 20 &   8 & 2 &  906 &  926 & 62.86230 & 1.692288 & 60.00000 & 3.535534 & 41.156774 &  0.1051210 &   8.175715 &  7.8528329 & 81.107 & Odoribacter sp. AF15-53 & 3 &  36 & Peptidoglycan D\\%2CD-transpeptidase MrdA\\\\\n",
       "\t cds & 469 &  720 & 119 & 119 & 0 & 720f  & PRIMER\\_LEFT  & AGTCCSGGDTATGAYCCGCA  & 20 &  12 & 1 &  807 &  827 & 63.37921 & 1.585835 & 59.16667 & 3.435921 & 54.529677 & 10.4477986 &  12.546118 & 13.3155804 & 81.107 & Odoribacter sp. AF15-53 & 4 &  16 & Peptidoglycan D\\%2CD-transpeptidase MrdA\\\\\n",
       "\t cds & 469 &  720 & 119 & 119 & 0 & 720r  & PRIMER\\_RIGHT & GGGGGATAYTGYGCCATSGTY & 21 &  16 & 4 &  905 &  926 & 63.52770 & 1.607231 & 59.52381 & 4.123930 & 41.156774 &  0.1051210 &   8.175715 &  7.8528329 & 81.107 & Odoribacter sp. AF15-53 & 4 &  16 & Peptidoglycan D\\%2CD-transpeptidase MrdA\\\\\n",
       "\t cds & 469 &  308 & 118 & 118 & 0 & 308f  & PRIMER\\_LEFT  & AGTCCSGGDTATGAYCCGCA  & 20 &  12 & 1 &  807 &  827 & 63.37921 & 1.585835 & 59.16667 & 3.435921 & 54.529677 & 10.4477986 &  12.546118 & 13.3155804 & 81.107 & Odoribacter sp. AF15-53 & 4 &  24 & Peptidoglycan D\\%2CD-transpeptidase MrdA\\\\\n",
       "\t cds & 469 &  308 & 118 & 118 & 0 & 308r  & PRIMER\\_RIGHT & GGGGATAYTGYGCCATSGTYG & 21 &  16 & 4 &  904 &  925 & 63.09697 & 1.809699 & 59.52381 & 4.123930 & 40.972038 &  0.2898574 &   8.849733 &  8.6056358 & 81.107 & Odoribacter sp. AF15-53 & 4 &  24 & Peptidoglycan D\\%2CD-transpeptidase MrdA\\\\\n",
       "\t cds & 469 &   25 & 120 & 120 & 0 & 25f   & PRIMER\\_LEFT  & AGTCCSGGDTATGAYCCGCA  & 20 &  12 & 1 &  807 &  827 & 63.37921 & 1.585835 & 59.16667 & 3.435921 & 54.529677 & 10.4477986 &  12.546118 & 13.3155804 & 81.107 & Odoribacter sp. AF15-53 & 4 &  32 & Peptidoglycan D\\%2CD-transpeptidase MrdA\\\\\n",
       "\t cds & 469 &   25 & 120 & 120 & 0 & 25r   & PRIMER\\_RIGHT & BGGGGGATAYTGYGCCATSG  & 20 &  24 & 2 &  907 &  927 & 63.42396 & 1.720398 & 63.33333 & 4.249183 & 42.405568 &  1.7528983 &   4.674529 &  5.1073084 & 81.107 & Odoribacter sp. AF15-53 & 4 &  32 & Peptidoglycan D\\%2CD-transpeptidase MrdA\\\\\n",
       "\t cds & 469 &  448 & 119 & 119 & 0 & 448f  & PRIMER\\_LEFT  & GTCCSGGDTATGAYCCGCAA  & 20 &  12 & 1 &  808 &  828 & 62.46664 & 1.557512 & 59.16667 & 3.435921 & 54.529677 & 10.4477986 &  12.546118 & 13.3155804 & 81.107 & Odoribacter sp. AF15-53 & 4 &  32 & Peptidoglycan D\\%2CD-transpeptidase MrdA\\\\\n",
       "\t cds & 469 &  448 & 119 & 119 & 0 & 448r  & PRIMER\\_RIGHT & BGGGGGATAYTGYGCCATSG  & 20 &  24 & 2 &  907 &  927 & 63.42396 & 1.720398 & 63.33333 & 4.249183 & 42.405568 &  1.7528983 &   4.674529 &  5.1073084 & 81.107 & Odoribacter sp. AF15-53 & 4 &  32 & Peptidoglycan D\\%2CD-transpeptidase MrdA\\\\\n",
       "\t cds & 469 &  785 & 125 & 125 & 0 & 785f  & PRIMER\\_LEFT  & TCYAGTCCSGGDTATGAYCCG & 21 &  24 & 2 &  804 &  825 & 62.16733 & 2.085810 & 58.73016 & 4.046841 & 54.810400 & 10.1710443 &  11.909481 & 12.9429433 & 81.107 & Odoribacter sp. AF15-53 & 4 &  64 & Peptidoglycan D\\%2CD-transpeptidase MrdA\\\\\n",
       "\t cds & 469 &  785 & 125 & 125 & 0 & 785r  & PRIMER\\_RIGHT & CCBGGGGGATAYTGYGCCAT  & 20 &  12 & 1 &  909 &  929 & 64.24558 & 2.069738 & 63.33333 & 4.249183 & 42.928759 &  2.8664437 &   3.518264 & 11.0215662 & 81.107 & Odoribacter sp. AF15-53 & 4 &  64 & Peptidoglycan D\\%2CD-transpeptidase MrdA\\\\\n",
       "\t cds & 469 & 1000 & 120 & 120 & 0 & 1000f & PRIMER\\_LEFT  & GTCCSGGDTATGAYCCGCAA  & 20 &  12 & 1 &  808 &  828 & 62.46664 & 1.557512 & 59.16667 & 3.435921 & 54.529677 & 10.4477986 &  12.546118 & 13.3155804 & 81.107 & Odoribacter sp. AF15-53 & 4 & 128 & Peptidoglycan D\\%2CD-transpeptidase MrdA\\\\\n",
       "\t cds & 469 & 1000 & 120 & 120 & 0 & 1000r & PRIMER\\_RIGHT & CBGGGGGATAYTGYGCCATS  & 20 &  24 & 2 &  908 &  928 & 62.87830 & 2.042500 & 63.33333 & 4.249183 & 42.928759 &  2.8664437 &  -1.070190 &  8.1980946 & 81.107 & Odoribacter sp. AF15-53 & 4 & 128 & Peptidoglycan D\\%2CD-transpeptidase MrdA\\\\\n",
       "\t cds & 469 &  910 & 109 & 109 & 0 & 910f  & PRIMER\\_LEFT  & GTCCSGGDTATGAYCCGCAA  & 20 &  12 & 1 &  808 &  828 & 62.46664 & 1.557512 & 59.16667 & 3.435921 & 54.529677 & 10.4477986 &  12.546118 & 13.3155804 & 81.107 & Odoribacter sp. AF15-53 & 6 &  96 & Peptidoglycan D\\%2CD-transpeptidase MrdA\\\\\n",
       "\t cds & 469 &  910 & 109 & 109 & 0 & 910r  & PRIMER\\_RIGHT & TGYGCCATSGTYGTTCTRTC  & 20 &  16 & 2 &  897 &  917 & 60.27747 & 2.226624 & 52.50000 & 4.330127 &  9.590313 & 16.6109088 &   6.825856 & 12.1537464 & 81.107 & Odoribacter sp. AF15-53 & 6 &  96 & Peptidoglycan D\\%2CD-transpeptidase MrdA\\\\\n",
       "\t cds & 118 &  555 & 107 & 107 & 0 & 555f  & PRIMER\\_LEFT  & AAGGAYWTBCGNGTSGATGT  & 20 &  96 & 1 &  867 &  887 & 59.21880 & 2.210807 & 48.33333 & 4.249183 & 29.190064 & 23.3863626 &  -3.187227 & 24.1165857 & 81.767 & Odoribacter sp. AF15-53 & 2 &  96 & hypothetical protein                   \\\\\n",
       "\t cds & 118 &  555 & 107 & 107 & 0 & 555r  & PRIMER\\_RIGHT & CGRCGHCCRATRATYAAYCC  & 20 &  96 & 2 &  954 &  974 & 59.31237 & 2.922306 & 54.16667 & 6.066758 & 23.698622 & 23.9999455 &   3.609712 & 16.5893977 & 81.767 & Odoribacter sp. AF15-53 & 2 &  96 & hypothetical protein                   \\\\\n",
       "\t cds & 118 &  286 & 140 & 140 & 0 & 286f  & PRIMER\\_LEFT  & AAGGAYWTBCGNGTSGATGT  & 20 &  96 & 1 &  867 &  887 & 59.21880 & 2.210807 & 48.33333 & 4.249183 & 29.190064 & 23.3863626 &  -3.187227 & 24.1165857 & 81.767 & Odoribacter sp. AF15-53 & 4 &  96 & hypothetical protein                   \\\\\n",
       "\t cds & 118 &  286 & 140 & 140 & 0 & 286r  & PRIMER\\_RIGHT & CGYTGYTCYTGWACCAAYCC  & 20 &  32 & 2 &  987 & 1007 & 59.80526 & 2.153698 & 55.00000 & 5.000000 & 11.994356 & 17.8257607 &  -6.009949 & 13.5798078 & 81.767 & Odoribacter sp. AF15-53 & 4 &  96 & hypothetical protein                   \\\\\n",
       "\t cds & 118 &  287 & 141 & 141 & 0 & 287f  & PRIMER\\_LEFT  & AAGGAYWTBCGNGTSGATGT  & 20 &  96 & 1 &  867 &  887 & 59.21880 & 2.210807 & 48.33333 & 4.249183 & 29.190064 & 23.3863626 &  -3.187227 & 24.1165857 & 81.767 & Odoribacter sp. AF15-53 & 4 &  96 & hypothetical protein                   \\\\\n",
       "\t cds & 118 &  287 & 141 & 141 & 0 & 287r  & PRIMER\\_RIGHT & VCGYTGYTCYTGWACCAAYC  & 20 &  96 & 2 &  988 & 1008 & 59.77764 & 2.222613 & 53.33333 & 5.527708 & 11.994356 & 17.8257607 &  -6.144309 & 13.8730856 & 81.767 & Odoribacter sp. AF15-53 & 4 &  96 & hypothetical protein                   \\\\\n",
       "\t cds & 118 &  553 & 101 & 101 & 0 & 553f  & PRIMER\\_LEFT  & AAGGAYWTBCGNGTSGATGT  & 20 &  96 & 1 &  867 &  887 & 59.21880 & 2.210807 & 48.33333 & 4.249183 & 29.190064 & 23.3863626 &  -3.187227 & 24.1165857 & 81.767 & Odoribacter sp. AF15-53 & 4 & 128 & hypothetical protein                   \\\\\n",
       "\t cds & 118 &  553 & 101 & 101 & 0 & 553r  & PRIMER\\_RIGHT & CCRATRATYAAYCCNCGRTC  & 20 & 128 & 2 &  948 &  968 & 56.78183 & 2.912989 & 50.00000 & 6.123724 & 21.136796 & 18.6641357 &  -2.824651 & 22.4057285 & 81.767 & Odoribacter sp. AF15-53 & 4 & 128 & hypothetical protein                   \\\\\n",
       "\t cds & 118 &  642 & 230 & 230 & 0 & 642f  & PRIMER\\_LEFT  & CGGATRTYTTYGTDCCRGCK  & 20 &  96 & 4 & 1237 & 1257 & 60.32781 & 2.703100 & 54.16667 & 6.066758 & 20.732458 & 20.8653216 & -10.075399 & 24.5760581 & 81.767 & Odoribacter sp. AF15-53 & 4 & 128 & hypothetical protein                   \\\\\n",
       "\t cds & 118 &  642 & 230 & 230 & 0 & 642r  & PRIMER\\_RIGHT & YYCCTTSTCRTYCCGYTTYA  & 20 & 128 & 4 & 1447 & 1467 & 58.92361 & 2.556469 & 50.00000 & 6.123724 &  0.000000 &  0.0000000 & -41.763740 &  0.3687041 & 81.767 & Odoribacter sp. AF15-53 & 4 & 128 & hypothetical protein                   \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 30 × 27\n",
       "\n",
       "| gene_type &lt;chr&gt; | cluster_id &lt;int&gt; | primer_set &lt;int&gt; | amplicon_size_consensus &lt;int&gt; | amplicon_size_avg &lt;dbl&gt; | amplicon_size_sd &lt;dbl&gt; | primer_id &lt;fct&gt; | primer_type &lt;fct&gt; | sequence &lt;fct&gt; | length &lt;int&gt; | degeneracy &lt;int&gt; | degeneracy_3prime &lt;int&gt; | position_start &lt;int&gt; | position_end &lt;int&gt; | Tm_avg &lt;dbl&gt; | Tm_sd &lt;dbl&gt; | GC_avg &lt;dbl&gt; | GC_sd &lt;dbl&gt; | hairpin_avg &lt;dbl&gt; | hairpin_sd &lt;dbl&gt; | homodimer_avg &lt;dbl&gt; | homodimer_sd &lt;dbl&gt; | pident &lt;dbl&gt; | sscinames &lt;fct&gt; | max_degeneracy_3prime &lt;int&gt; | max_degeneracy &lt;int&gt; | annotation &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| cds | 469 |  677 | 122 | 122 | 0 | 677f  | PRIMER_LEFT  | AGTCCSGGDTATGAYCCGCA  | 20 |  12 | 1 |  807 |  827 | 63.37921 | 1.585835 | 59.16667 | 3.435921 | 54.529677 | 10.4477986 |  12.546118 | 13.3155804 | 81.107 | Odoribacter sp. AF15-53 | 2 |  64 | Peptidoglycan D%2CD-transpeptidase MrdA |\n",
       "| cds | 469 |  677 | 122 | 122 | 0 | 677r  | PRIMER_RIGHT | CCBGGGGGATAYTGYGCCA   | 19 |  12 | 2 |  910 |  929 | 63.99539 | 2.171444 | 66.66667 | 4.472824 | 42.790866 |  2.5680978 |   3.518264 | 11.0215662 | 81.107 | Odoribacter sp. AF15-53 | 2 |  64 | Peptidoglycan D%2CD-transpeptidase MrdA |\n",
       "| cds | 469 |   84 | 122 | 122 | 0 | 84f   | PRIMER_LEFT  | AGTCCSGGDTATGAYCCGCA  | 20 |  12 | 1 |  807 |  827 | 63.37921 | 1.585835 | 59.16667 | 3.435921 | 54.529677 | 10.4477986 |  12.546118 | 13.3155804 | 81.107 | Odoribacter sp. AF15-53 | 3 |  12 | Peptidoglycan D%2CD-transpeptidase MrdA |\n",
       "| cds | 469 |   84 | 122 | 122 | 0 | 84r   | PRIMER_RIGHT | CCBGGGGGATAYTGYGCCAT  | 20 |  12 | 1 |  909 |  929 | 64.24558 | 2.069738 | 63.33333 | 4.249183 | 42.928759 |  2.8664437 |   3.518264 | 11.0215662 | 81.107 | Odoribacter sp. AF15-53 | 3 |  12 | Peptidoglycan D%2CD-transpeptidase MrdA |\n",
       "| cds | 469 |  700 | 119 | 119 | 0 | 700f  | PRIMER_LEFT  | AGTCCSGGDTATGAYCCGCA  | 20 |  12 | 1 |  807 |  827 | 63.37921 | 1.585835 | 59.16667 | 3.435921 | 54.529677 | 10.4477986 |  12.546118 | 13.3155804 | 81.107 | Odoribacter sp. AF15-53 | 3 |  36 | Peptidoglycan D%2CD-transpeptidase MrdA |\n",
       "| cds | 469 |  700 | 119 | 119 | 0 | 700r  | PRIMER_RIGHT | GGGGGATAYTGYGCCATSGT  | 20 |   8 | 2 |  906 |  926 | 62.86230 | 1.692288 | 60.00000 | 3.535534 | 41.156774 |  0.1051210 |   8.175715 |  7.8528329 | 81.107 | Odoribacter sp. AF15-53 | 3 |  36 | Peptidoglycan D%2CD-transpeptidase MrdA |\n",
       "| cds | 469 |  720 | 119 | 119 | 0 | 720f  | PRIMER_LEFT  | AGTCCSGGDTATGAYCCGCA  | 20 |  12 | 1 |  807 |  827 | 63.37921 | 1.585835 | 59.16667 | 3.435921 | 54.529677 | 10.4477986 |  12.546118 | 13.3155804 | 81.107 | Odoribacter sp. AF15-53 | 4 |  16 | Peptidoglycan D%2CD-transpeptidase MrdA |\n",
       "| cds | 469 |  720 | 119 | 119 | 0 | 720r  | PRIMER_RIGHT | GGGGGATAYTGYGCCATSGTY | 21 |  16 | 4 |  905 |  926 | 63.52770 | 1.607231 | 59.52381 | 4.123930 | 41.156774 |  0.1051210 |   8.175715 |  7.8528329 | 81.107 | Odoribacter sp. AF15-53 | 4 |  16 | Peptidoglycan D%2CD-transpeptidase MrdA |\n",
       "| cds | 469 |  308 | 118 | 118 | 0 | 308f  | PRIMER_LEFT  | AGTCCSGGDTATGAYCCGCA  | 20 |  12 | 1 |  807 |  827 | 63.37921 | 1.585835 | 59.16667 | 3.435921 | 54.529677 | 10.4477986 |  12.546118 | 13.3155804 | 81.107 | Odoribacter sp. AF15-53 | 4 |  24 | Peptidoglycan D%2CD-transpeptidase MrdA |\n",
       "| cds | 469 |  308 | 118 | 118 | 0 | 308r  | PRIMER_RIGHT | GGGGATAYTGYGCCATSGTYG | 21 |  16 | 4 |  904 |  925 | 63.09697 | 1.809699 | 59.52381 | 4.123930 | 40.972038 |  0.2898574 |   8.849733 |  8.6056358 | 81.107 | Odoribacter sp. AF15-53 | 4 |  24 | Peptidoglycan D%2CD-transpeptidase MrdA |\n",
       "| cds | 469 |   25 | 120 | 120 | 0 | 25f   | PRIMER_LEFT  | AGTCCSGGDTATGAYCCGCA  | 20 |  12 | 1 |  807 |  827 | 63.37921 | 1.585835 | 59.16667 | 3.435921 | 54.529677 | 10.4477986 |  12.546118 | 13.3155804 | 81.107 | Odoribacter sp. AF15-53 | 4 |  32 | Peptidoglycan D%2CD-transpeptidase MrdA |\n",
       "| cds | 469 |   25 | 120 | 120 | 0 | 25r   | PRIMER_RIGHT | BGGGGGATAYTGYGCCATSG  | 20 |  24 | 2 |  907 |  927 | 63.42396 | 1.720398 | 63.33333 | 4.249183 | 42.405568 |  1.7528983 |   4.674529 |  5.1073084 | 81.107 | Odoribacter sp. AF15-53 | 4 |  32 | Peptidoglycan D%2CD-transpeptidase MrdA |\n",
       "| cds | 469 |  448 | 119 | 119 | 0 | 448f  | PRIMER_LEFT  | GTCCSGGDTATGAYCCGCAA  | 20 |  12 | 1 |  808 |  828 | 62.46664 | 1.557512 | 59.16667 | 3.435921 | 54.529677 | 10.4477986 |  12.546118 | 13.3155804 | 81.107 | Odoribacter sp. AF15-53 | 4 |  32 | Peptidoglycan D%2CD-transpeptidase MrdA |\n",
       "| cds | 469 |  448 | 119 | 119 | 0 | 448r  | PRIMER_RIGHT | BGGGGGATAYTGYGCCATSG  | 20 |  24 | 2 |  907 |  927 | 63.42396 | 1.720398 | 63.33333 | 4.249183 | 42.405568 |  1.7528983 |   4.674529 |  5.1073084 | 81.107 | Odoribacter sp. AF15-53 | 4 |  32 | Peptidoglycan D%2CD-transpeptidase MrdA |\n",
       "| cds | 469 |  785 | 125 | 125 | 0 | 785f  | PRIMER_LEFT  | TCYAGTCCSGGDTATGAYCCG | 21 |  24 | 2 |  804 |  825 | 62.16733 | 2.085810 | 58.73016 | 4.046841 | 54.810400 | 10.1710443 |  11.909481 | 12.9429433 | 81.107 | Odoribacter sp. AF15-53 | 4 |  64 | Peptidoglycan D%2CD-transpeptidase MrdA |\n",
       "| cds | 469 |  785 | 125 | 125 | 0 | 785r  | PRIMER_RIGHT | CCBGGGGGATAYTGYGCCAT  | 20 |  12 | 1 |  909 |  929 | 64.24558 | 2.069738 | 63.33333 | 4.249183 | 42.928759 |  2.8664437 |   3.518264 | 11.0215662 | 81.107 | Odoribacter sp. AF15-53 | 4 |  64 | Peptidoglycan D%2CD-transpeptidase MrdA |\n",
       "| cds | 469 | 1000 | 120 | 120 | 0 | 1000f | PRIMER_LEFT  | GTCCSGGDTATGAYCCGCAA  | 20 |  12 | 1 |  808 |  828 | 62.46664 | 1.557512 | 59.16667 | 3.435921 | 54.529677 | 10.4477986 |  12.546118 | 13.3155804 | 81.107 | Odoribacter sp. AF15-53 | 4 | 128 | Peptidoglycan D%2CD-transpeptidase MrdA |\n",
       "| cds | 469 | 1000 | 120 | 120 | 0 | 1000r | PRIMER_RIGHT | CBGGGGGATAYTGYGCCATS  | 20 |  24 | 2 |  908 |  928 | 62.87830 | 2.042500 | 63.33333 | 4.249183 | 42.928759 |  2.8664437 |  -1.070190 |  8.1980946 | 81.107 | Odoribacter sp. AF15-53 | 4 | 128 | Peptidoglycan D%2CD-transpeptidase MrdA |\n",
       "| cds | 469 |  910 | 109 | 109 | 0 | 910f  | PRIMER_LEFT  | GTCCSGGDTATGAYCCGCAA  | 20 |  12 | 1 |  808 |  828 | 62.46664 | 1.557512 | 59.16667 | 3.435921 | 54.529677 | 10.4477986 |  12.546118 | 13.3155804 | 81.107 | Odoribacter sp. AF15-53 | 6 |  96 | Peptidoglycan D%2CD-transpeptidase MrdA |\n",
       "| cds | 469 |  910 | 109 | 109 | 0 | 910r  | PRIMER_RIGHT | TGYGCCATSGTYGTTCTRTC  | 20 |  16 | 2 |  897 |  917 | 60.27747 | 2.226624 | 52.50000 | 4.330127 |  9.590313 | 16.6109088 |   6.825856 | 12.1537464 | 81.107 | Odoribacter sp. AF15-53 | 6 |  96 | Peptidoglycan D%2CD-transpeptidase MrdA |\n",
       "| cds | 118 |  555 | 107 | 107 | 0 | 555f  | PRIMER_LEFT  | AAGGAYWTBCGNGTSGATGT  | 20 |  96 | 1 |  867 |  887 | 59.21880 | 2.210807 | 48.33333 | 4.249183 | 29.190064 | 23.3863626 |  -3.187227 | 24.1165857 | 81.767 | Odoribacter sp. AF15-53 | 2 |  96 | hypothetical protein                    |\n",
       "| cds | 118 |  555 | 107 | 107 | 0 | 555r  | PRIMER_RIGHT | CGRCGHCCRATRATYAAYCC  | 20 |  96 | 2 |  954 |  974 | 59.31237 | 2.922306 | 54.16667 | 6.066758 | 23.698622 | 23.9999455 |   3.609712 | 16.5893977 | 81.767 | Odoribacter sp. AF15-53 | 2 |  96 | hypothetical protein                    |\n",
       "| cds | 118 |  286 | 140 | 140 | 0 | 286f  | PRIMER_LEFT  | AAGGAYWTBCGNGTSGATGT  | 20 |  96 | 1 |  867 |  887 | 59.21880 | 2.210807 | 48.33333 | 4.249183 | 29.190064 | 23.3863626 |  -3.187227 | 24.1165857 | 81.767 | Odoribacter sp. AF15-53 | 4 |  96 | hypothetical protein                    |\n",
       "| cds | 118 |  286 | 140 | 140 | 0 | 286r  | PRIMER_RIGHT | CGYTGYTCYTGWACCAAYCC  | 20 |  32 | 2 |  987 | 1007 | 59.80526 | 2.153698 | 55.00000 | 5.000000 | 11.994356 | 17.8257607 |  -6.009949 | 13.5798078 | 81.767 | Odoribacter sp. AF15-53 | 4 |  96 | hypothetical protein                    |\n",
       "| cds | 118 |  287 | 141 | 141 | 0 | 287f  | PRIMER_LEFT  | AAGGAYWTBCGNGTSGATGT  | 20 |  96 | 1 |  867 |  887 | 59.21880 | 2.210807 | 48.33333 | 4.249183 | 29.190064 | 23.3863626 |  -3.187227 | 24.1165857 | 81.767 | Odoribacter sp. AF15-53 | 4 |  96 | hypothetical protein                    |\n",
       "| cds | 118 |  287 | 141 | 141 | 0 | 287r  | PRIMER_RIGHT | VCGYTGYTCYTGWACCAAYC  | 20 |  96 | 2 |  988 | 1008 | 59.77764 | 2.222613 | 53.33333 | 5.527708 | 11.994356 | 17.8257607 |  -6.144309 | 13.8730856 | 81.767 | Odoribacter sp. AF15-53 | 4 |  96 | hypothetical protein                    |\n",
       "| cds | 118 |  553 | 101 | 101 | 0 | 553f  | PRIMER_LEFT  | AAGGAYWTBCGNGTSGATGT  | 20 |  96 | 1 |  867 |  887 | 59.21880 | 2.210807 | 48.33333 | 4.249183 | 29.190064 | 23.3863626 |  -3.187227 | 24.1165857 | 81.767 | Odoribacter sp. AF15-53 | 4 | 128 | hypothetical protein                    |\n",
       "| cds | 118 |  553 | 101 | 101 | 0 | 553r  | PRIMER_RIGHT | CCRATRATYAAYCCNCGRTC  | 20 | 128 | 2 |  948 |  968 | 56.78183 | 2.912989 | 50.00000 | 6.123724 | 21.136796 | 18.6641357 |  -2.824651 | 22.4057285 | 81.767 | Odoribacter sp. AF15-53 | 4 | 128 | hypothetical protein                    |\n",
       "| cds | 118 |  642 | 230 | 230 | 0 | 642f  | PRIMER_LEFT  | CGGATRTYTTYGTDCCRGCK  | 20 |  96 | 4 | 1237 | 1257 | 60.32781 | 2.703100 | 54.16667 | 6.066758 | 20.732458 | 20.8653216 | -10.075399 | 24.5760581 | 81.767 | Odoribacter sp. AF15-53 | 4 | 128 | hypothetical protein                    |\n",
       "| cds | 118 |  642 | 230 | 230 | 0 | 642r  | PRIMER_RIGHT | YYCCTTSTCRTYCCGYTTYA  | 20 | 128 | 4 | 1447 | 1467 | 58.92361 | 2.556469 | 50.00000 | 6.123724 |  0.000000 |  0.0000000 | -41.763740 |  0.3687041 | 81.767 | Odoribacter sp. AF15-53 | 4 | 128 | hypothetical protein                    |\n",
       "\n"
      ],
      "text/plain": [
       "   gene_type cluster_id primer_set amplicon_size_consensus amplicon_size_avg\n",
       "1  cds       469         677       122                     122              \n",
       "2  cds       469         677       122                     122              \n",
       "3  cds       469          84       122                     122              \n",
       "4  cds       469          84       122                     122              \n",
       "5  cds       469         700       119                     119              \n",
       "6  cds       469         700       119                     119              \n",
       "7  cds       469         720       119                     119              \n",
       "8  cds       469         720       119                     119              \n",
       "9  cds       469         308       118                     118              \n",
       "10 cds       469         308       118                     118              \n",
       "11 cds       469          25       120                     120              \n",
       "12 cds       469          25       120                     120              \n",
       "13 cds       469         448       119                     119              \n",
       "14 cds       469         448       119                     119              \n",
       "15 cds       469         785       125                     125              \n",
       "16 cds       469         785       125                     125              \n",
       "17 cds       469        1000       120                     120              \n",
       "18 cds       469        1000       120                     120              \n",
       "19 cds       469         910       109                     109              \n",
       "20 cds       469         910       109                     109              \n",
       "21 cds       118         555       107                     107              \n",
       "22 cds       118         555       107                     107              \n",
       "23 cds       118         286       140                     140              \n",
       "24 cds       118         286       140                     140              \n",
       "25 cds       118         287       141                     141              \n",
       "26 cds       118         287       141                     141              \n",
       "27 cds       118         553       101                     101              \n",
       "28 cds       118         553       101                     101              \n",
       "29 cds       118         642       230                     230              \n",
       "30 cds       118         642       230                     230              \n",
       "   amplicon_size_sd primer_id primer_type  sequence              length\n",
       "1  0                677f      PRIMER_LEFT  AGTCCSGGDTATGAYCCGCA  20    \n",
       "2  0                677r      PRIMER_RIGHT CCBGGGGGATAYTGYGCCA   19    \n",
       "3  0                84f       PRIMER_LEFT  AGTCCSGGDTATGAYCCGCA  20    \n",
       "4  0                84r       PRIMER_RIGHT CCBGGGGGATAYTGYGCCAT  20    \n",
       "5  0                700f      PRIMER_LEFT  AGTCCSGGDTATGAYCCGCA  20    \n",
       "6  0                700r      PRIMER_RIGHT GGGGGATAYTGYGCCATSGT  20    \n",
       "7  0                720f      PRIMER_LEFT  AGTCCSGGDTATGAYCCGCA  20    \n",
       "8  0                720r      PRIMER_RIGHT GGGGGATAYTGYGCCATSGTY 21    \n",
       "9  0                308f      PRIMER_LEFT  AGTCCSGGDTATGAYCCGCA  20    \n",
       "10 0                308r      PRIMER_RIGHT GGGGATAYTGYGCCATSGTYG 21    \n",
       "11 0                25f       PRIMER_LEFT  AGTCCSGGDTATGAYCCGCA  20    \n",
       "12 0                25r       PRIMER_RIGHT BGGGGGATAYTGYGCCATSG  20    \n",
       "13 0                448f      PRIMER_LEFT  GTCCSGGDTATGAYCCGCAA  20    \n",
       "14 0                448r      PRIMER_RIGHT BGGGGGATAYTGYGCCATSG  20    \n",
       "15 0                785f      PRIMER_LEFT  TCYAGTCCSGGDTATGAYCCG 21    \n",
       "16 0                785r      PRIMER_RIGHT CCBGGGGGATAYTGYGCCAT  20    \n",
       "17 0                1000f     PRIMER_LEFT  GTCCSGGDTATGAYCCGCAA  20    \n",
       "18 0                1000r     PRIMER_RIGHT CBGGGGGATAYTGYGCCATS  20    \n",
       "19 0                910f      PRIMER_LEFT  GTCCSGGDTATGAYCCGCAA  20    \n",
       "20 0                910r      PRIMER_RIGHT TGYGCCATSGTYGTTCTRTC  20    \n",
       "21 0                555f      PRIMER_LEFT  AAGGAYWTBCGNGTSGATGT  20    \n",
       "22 0                555r      PRIMER_RIGHT CGRCGHCCRATRATYAAYCC  20    \n",
       "23 0                286f      PRIMER_LEFT  AAGGAYWTBCGNGTSGATGT  20    \n",
       "24 0                286r      PRIMER_RIGHT CGYTGYTCYTGWACCAAYCC  20    \n",
       "25 0                287f      PRIMER_LEFT  AAGGAYWTBCGNGTSGATGT  20    \n",
       "26 0                287r      PRIMER_RIGHT VCGYTGYTCYTGWACCAAYC  20    \n",
       "27 0                553f      PRIMER_LEFT  AAGGAYWTBCGNGTSGATGT  20    \n",
       "28 0                553r      PRIMER_RIGHT CCRATRATYAAYCCNCGRTC  20    \n",
       "29 0                642f      PRIMER_LEFT  CGGATRTYTTYGTDCCRGCK  20    \n",
       "30 0                642r      PRIMER_RIGHT YYCCTTSTCRTYCCGYTTYA  20    \n",
       "   degeneracy degeneracy_3prime position_start position_end Tm_avg   Tm_sd   \n",
       "1   12        1                  807            827         63.37921 1.585835\n",
       "2   12        2                  910            929         63.99539 2.171444\n",
       "3   12        1                  807            827         63.37921 1.585835\n",
       "4   12        1                  909            929         64.24558 2.069738\n",
       "5   12        1                  807            827         63.37921 1.585835\n",
       "6    8        2                  906            926         62.86230 1.692288\n",
       "7   12        1                  807            827         63.37921 1.585835\n",
       "8   16        4                  905            926         63.52770 1.607231\n",
       "9   12        1                  807            827         63.37921 1.585835\n",
       "10  16        4                  904            925         63.09697 1.809699\n",
       "11  12        1                  807            827         63.37921 1.585835\n",
       "12  24        2                  907            927         63.42396 1.720398\n",
       "13  12        1                  808            828         62.46664 1.557512\n",
       "14  24        2                  907            927         63.42396 1.720398\n",
       "15  24        2                  804            825         62.16733 2.085810\n",
       "16  12        1                  909            929         64.24558 2.069738\n",
       "17  12        1                  808            828         62.46664 1.557512\n",
       "18  24        2                  908            928         62.87830 2.042500\n",
       "19  12        1                  808            828         62.46664 1.557512\n",
       "20  16        2                  897            917         60.27747 2.226624\n",
       "21  96        1                  867            887         59.21880 2.210807\n",
       "22  96        2                  954            974         59.31237 2.922306\n",
       "23  96        1                  867            887         59.21880 2.210807\n",
       "24  32        2                  987           1007         59.80526 2.153698\n",
       "25  96        1                  867            887         59.21880 2.210807\n",
       "26  96        2                  988           1008         59.77764 2.222613\n",
       "27  96        1                  867            887         59.21880 2.210807\n",
       "28 128        2                  948            968         56.78183 2.912989\n",
       "29  96        4                 1237           1257         60.32781 2.703100\n",
       "30 128        4                 1447           1467         58.92361 2.556469\n",
       "   GC_avg   GC_sd    hairpin_avg hairpin_sd homodimer_avg homodimer_sd pident\n",
       "1  59.16667 3.435921 54.529677   10.4477986  12.546118    13.3155804   81.107\n",
       "2  66.66667 4.472824 42.790866    2.5680978   3.518264    11.0215662   81.107\n",
       "3  59.16667 3.435921 54.529677   10.4477986  12.546118    13.3155804   81.107\n",
       "4  63.33333 4.249183 42.928759    2.8664437   3.518264    11.0215662   81.107\n",
       "5  59.16667 3.435921 54.529677   10.4477986  12.546118    13.3155804   81.107\n",
       "6  60.00000 3.535534 41.156774    0.1051210   8.175715     7.8528329   81.107\n",
       "7  59.16667 3.435921 54.529677   10.4477986  12.546118    13.3155804   81.107\n",
       "8  59.52381 4.123930 41.156774    0.1051210   8.175715     7.8528329   81.107\n",
       "9  59.16667 3.435921 54.529677   10.4477986  12.546118    13.3155804   81.107\n",
       "10 59.52381 4.123930 40.972038    0.2898574   8.849733     8.6056358   81.107\n",
       "11 59.16667 3.435921 54.529677   10.4477986  12.546118    13.3155804   81.107\n",
       "12 63.33333 4.249183 42.405568    1.7528983   4.674529     5.1073084   81.107\n",
       "13 59.16667 3.435921 54.529677   10.4477986  12.546118    13.3155804   81.107\n",
       "14 63.33333 4.249183 42.405568    1.7528983   4.674529     5.1073084   81.107\n",
       "15 58.73016 4.046841 54.810400   10.1710443  11.909481    12.9429433   81.107\n",
       "16 63.33333 4.249183 42.928759    2.8664437   3.518264    11.0215662   81.107\n",
       "17 59.16667 3.435921 54.529677   10.4477986  12.546118    13.3155804   81.107\n",
       "18 63.33333 4.249183 42.928759    2.8664437  -1.070190     8.1980946   81.107\n",
       "19 59.16667 3.435921 54.529677   10.4477986  12.546118    13.3155804   81.107\n",
       "20 52.50000 4.330127  9.590313   16.6109088   6.825856    12.1537464   81.107\n",
       "21 48.33333 4.249183 29.190064   23.3863626  -3.187227    24.1165857   81.767\n",
       "22 54.16667 6.066758 23.698622   23.9999455   3.609712    16.5893977   81.767\n",
       "23 48.33333 4.249183 29.190064   23.3863626  -3.187227    24.1165857   81.767\n",
       "24 55.00000 5.000000 11.994356   17.8257607  -6.009949    13.5798078   81.767\n",
       "25 48.33333 4.249183 29.190064   23.3863626  -3.187227    24.1165857   81.767\n",
       "26 53.33333 5.527708 11.994356   17.8257607  -6.144309    13.8730856   81.767\n",
       "27 48.33333 4.249183 29.190064   23.3863626  -3.187227    24.1165857   81.767\n",
       "28 50.00000 6.123724 21.136796   18.6641357  -2.824651    22.4057285   81.767\n",
       "29 54.16667 6.066758 20.732458   20.8653216 -10.075399    24.5760581   81.767\n",
       "30 50.00000 6.123724  0.000000    0.0000000 -41.763740     0.3687041   81.767\n",
       "   sscinames               max_degeneracy_3prime max_degeneracy\n",
       "1  Odoribacter sp. AF15-53 2                      64           \n",
       "2  Odoribacter sp. AF15-53 2                      64           \n",
       "3  Odoribacter sp. AF15-53 3                      12           \n",
       "4  Odoribacter sp. AF15-53 3                      12           \n",
       "5  Odoribacter sp. AF15-53 3                      36           \n",
       "6  Odoribacter sp. AF15-53 3                      36           \n",
       "7  Odoribacter sp. AF15-53 4                      16           \n",
       "8  Odoribacter sp. AF15-53 4                      16           \n",
       "9  Odoribacter sp. AF15-53 4                      24           \n",
       "10 Odoribacter sp. AF15-53 4                      24           \n",
       "11 Odoribacter sp. AF15-53 4                      32           \n",
       "12 Odoribacter sp. AF15-53 4                      32           \n",
       "13 Odoribacter sp. AF15-53 4                      32           \n",
       "14 Odoribacter sp. AF15-53 4                      32           \n",
       "15 Odoribacter sp. AF15-53 4                      64           \n",
       "16 Odoribacter sp. AF15-53 4                      64           \n",
       "17 Odoribacter sp. AF15-53 4                     128           \n",
       "18 Odoribacter sp. AF15-53 4                     128           \n",
       "19 Odoribacter sp. AF15-53 6                      96           \n",
       "20 Odoribacter sp. AF15-53 6                      96           \n",
       "21 Odoribacter sp. AF15-53 2                      96           \n",
       "22 Odoribacter sp. AF15-53 2                      96           \n",
       "23 Odoribacter sp. AF15-53 4                      96           \n",
       "24 Odoribacter sp. AF15-53 4                      96           \n",
       "25 Odoribacter sp. AF15-53 4                      96           \n",
       "26 Odoribacter sp. AF15-53 4                      96           \n",
       "27 Odoribacter sp. AF15-53 4                     128           \n",
       "28 Odoribacter sp. AF15-53 4                     128           \n",
       "29 Odoribacter sp. AF15-53 4                     128           \n",
       "30 Odoribacter sp. AF15-53 4                     128           \n",
       "   annotation                             \n",
       "1  Peptidoglycan D%2CD-transpeptidase MrdA\n",
       "2  Peptidoglycan D%2CD-transpeptidase MrdA\n",
       "3  Peptidoglycan D%2CD-transpeptidase MrdA\n",
       "4  Peptidoglycan D%2CD-transpeptidase MrdA\n",
       "5  Peptidoglycan D%2CD-transpeptidase MrdA\n",
       "6  Peptidoglycan D%2CD-transpeptidase MrdA\n",
       "7  Peptidoglycan D%2CD-transpeptidase MrdA\n",
       "8  Peptidoglycan D%2CD-transpeptidase MrdA\n",
       "9  Peptidoglycan D%2CD-transpeptidase MrdA\n",
       "10 Peptidoglycan D%2CD-transpeptidase MrdA\n",
       "11 Peptidoglycan D%2CD-transpeptidase MrdA\n",
       "12 Peptidoglycan D%2CD-transpeptidase MrdA\n",
       "13 Peptidoglycan D%2CD-transpeptidase MrdA\n",
       "14 Peptidoglycan D%2CD-transpeptidase MrdA\n",
       "15 Peptidoglycan D%2CD-transpeptidase MrdA\n",
       "16 Peptidoglycan D%2CD-transpeptidase MrdA\n",
       "17 Peptidoglycan D%2CD-transpeptidase MrdA\n",
       "18 Peptidoglycan D%2CD-transpeptidase MrdA\n",
       "19 Peptidoglycan D%2CD-transpeptidase MrdA\n",
       "20 Peptidoglycan D%2CD-transpeptidase MrdA\n",
       "21 hypothetical protein                   \n",
       "22 hypothetical protein                   \n",
       "23 hypothetical protein                   \n",
       "24 hypothetical protein                   \n",
       "25 hypothetical protein                   \n",
       "26 hypothetical protein                   \n",
       "27 hypothetical protein                   \n",
       "28 hypothetical protein                   \n",
       "29 hypothetical protein                   \n",
       "30 hypothetical protein                   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# joining\n",
    "primer_info_jfj = primer_info_jf %>%\n",
    "    left_join(gene_annot %>% distinct(gene_type, cluster_id, annotation),\n",
    "              c('gene_type', 'cluster_id'))\n",
    "\n",
    "df.dims(30,50)\n",
    "primer_info_jfj\n",
    "df.dims()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Writing table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-28T13:09:20.497439Z",
     "iopub.status.busy": "2021-04-28T13:09:20.494489Z",
     "iopub.status.idle": "2021-04-28T13:09:20.524764Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "File written: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3//Butyricimonas/best_primers.tsv \n"
     ]
    }
   ],
   "source": [
    "F = file.path(work_dir, 'best_primers.tsv')\n",
    "write_table(primer_info_jfj, F)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# sessionInfo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-28T13:09:28.434965Z",
     "iopub.status.busy": "2021-04-28T13:09:28.432966Z",
     "iopub.status.idle": "2021-04-28T13:09:28.538163Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LLG\n",
      "===\n",
      "\n",
      "Ley Lab Genome analysis pipeline (LLG)\n",
      "\n",
      "* Version: 0.2.4\n",
      "* Authors:\n",
      "  * Nick Youngblut <nyoungb2@gmail.com>\n",
      "* Maintainers:\n",
      "  * Nick Youngblut <nyoungb2@gmail.com>\n",
      "\n",
      "--- conda envs ---\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/gtdbtk.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- conda-forge::networkx\n",
      "- bioconda::gtdbtk=1.4\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/checkm.yaml <==\n",
      "channels:\n",
      "- bioconda\n",
      "dependencies:\n",
      "- python=2.7\n",
      "- pigz\n",
      "- bioconda::prodigal\n",
      "- bioconda::pplacer\n",
      "- bioconda::checkm-genome\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/quast.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- bioconda::seqkit\n",
      "- bioconda::quast>=5.0.0\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/sourmash.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- bioconda::sourmash=2.0.0a4\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/fastqc.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- bioconda::seqtk\n",
      "- bioconda::fastqc=0.11.7\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/multiqc.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- bioconda::multiqc=1.4\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/hadley.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- conda-forge::r-ape\n",
      "- conda-forge::r-dplyr\n",
      "- conda-forge::r-tidyr\n",
      "- conda-forge::r-ggplot2\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/bbmap.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::bbmap=37.78\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/ani.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::dashing\n",
      "- bioconda::fastani\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/prokka.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- bioconda::prokka\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/prodigal.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pip\n",
      "- biopython=1.77\n",
      "- :prodigal\n",
      "- pip:\n",
      "  - biopython.convert\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/eggnog-mapper.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::eggnog-mapper\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/phylophlan.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::phylophlan\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/drep.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::mash\n",
      "- bioconda::checkm-genome\n",
      "- bioconda::fastani\n",
      "- bioconda::drep\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/antismash.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- biopython=1.77\n",
      "- bioconda::blast=2.9\n",
      "- bioconda::antismash>=5.1.2\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/deepbgc.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::deepbgc\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/abricate.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- bioconda::abricate\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/rRNA.yaml <==\n",
      "channels:\n",
      "- qiime2/label/r2020.2\n",
      "- conda-forge\n",
      "- bioconda\n",
      "- defaults\n",
      "dependencies:\n",
      "- barrnap\n",
      "- _libgcc_mutex=0.1\n",
      "- _openmp_mutex=4.5\n",
      "- _r-mutex=1.0.1\n",
      "- alsa-lib=1.1.5\n",
      "- arb-bio-tools=6.0.6\n",
      "- attrs=19.3.0\n",
      "- backcall=0.1.0\n",
      "- bibtexparser=1.1.0\n",
      "- binutils_impl_linux-64=2.33.1\n",
      "- binutils_linux-64=2.33.1\n",
      "- bioconductor-biobase=2.42.0\n",
      "- bioconductor-biocgenerics=0.28.0\n",
      "- bioconductor-biocparallel=1.16.6\n",
      "- bioconductor-biostrings=2.50.2\n",
      "- bioconductor-dada2=1.10.0\n",
      "- bioconductor-delayedarray=0.8.0\n",
      "- bioconductor-genomeinfodb=1.18.1\n",
      "- bioconductor-genomeinfodbdata=1.2.1\n",
      "- bioconductor-genomicalignments=1.18.1\n",
      "- bioconductor-genomicranges=1.34.0\n",
      "- bioconductor-iranges=2.16.0\n",
      "- bioconductor-rsamtools=1.34.0\n",
      "- bioconductor-s4vectors=0.20.1\n",
      "- bioconductor-shortread=1.40.0\n",
      "- bioconductor-summarizedexperiment=1.12.0\n",
      "- bioconductor-xvector=0.22.0\n",
      "- bioconductor-zlibbioc=1.28.0\n",
      "- biom-format=2.1.8\n",
      "- blas=2.14\n",
      "- blast=2.9.0\n",
      "- bleach=3.1.1\n",
      "- bokeh=1.4.0\n",
      "- boost=1.68.0\n",
      "- boost-cpp=1.68.0\n",
      "- bwidget=1.9.14\n",
      "- bzip2=1.0.8\n",
      "- ca-certificates=2019.11.28\n",
      "- cachecontrol=0.12.5\n",
      "- cairo=1.16.0\n",
      "- certifi=2019.11.28\n",
      "- cffi=1.13.2\n",
      "- chardet=3.0.4\n",
      "- click=7.0\n",
      "- cryptography=2.8\n",
      "- curl=7.68.0\n",
      "- cutadapt=2.8\n",
      "- cycler=0.10.0\n",
      "- cython=0.29.15\n",
      "- dbus=1.13.6\n",
      "- deblur=1.1.0\n",
      "- decorator=4.4.1\n",
      "- defusedxml=0.6.0\n",
      "- dendropy=4.4.0\n",
      "- dnaio=0.4.1\n",
      "- emperor=1.0.0\n",
      "- entrypoints=0.3\n",
      "- expat=2.2.9\n",
      "- fastcluster=1.1.26\n",
      "- fasttree=2.1.10\n",
      "- fontconfig=2.13.1\n",
      "- freetype=2.10.0\n",
      "- future=0.18.2\n",
      "- gcc_impl_linux-64=7.3.0\n",
      "- gcc_linux-64=7.3.0\n",
      "- gettext=0.19.8.1\n",
      "- gfortran_impl_linux-64=7.3.0\n",
      "- gfortran_linux-64=7.3.0\n",
      "- giflib=5.2.1\n",
      "- glib=2.58.3\n",
      "- gmp=6.2.0\n",
      "- gneiss=0.4.6\n",
      "- gnutls=3.6.5\n",
      "- graphite2=1.3.13\n",
      "- gsl=2.5\n",
      "- gst-plugins-base=1.14.5\n",
      "- gstreamer=1.14.5\n",
      "- gxx_impl_linux-64=7.3.0\n",
      "- gxx_linux-64=7.3.0\n",
      "- h5py=2.10.0\n",
      "- harfbuzz=2.4.0\n",
      "- hdf5=1.10.5\n",
      "- hdmedians=0.13\n",
      "- hmmer=3.1b2\n",
      "- icu=58.2\n",
      "- idna=2.9\n",
      "- ijson=2.6.1\n",
      "- importlib_metadata=1.5.0\n",
      "- ipykernel=5.1.4\n",
      "- ipython=7.12.0\n",
      "- ipython_genutils=0.2.0\n",
      "- ipywidgets=7.5.1\n",
      "- iqtree=1.6.12\n",
      "- jedi=0.16.0\n",
      "- jinja2=2.11.1\n",
      "- joblib=0.14.1\n",
      "- jpeg=9c\n",
      "- jsonschema=3.2.0\n",
      "- jupyter_client=6.0.0\n",
      "- jupyter_core=4.6.3\n",
      "- kiwisolver=1.1.0\n",
      "- krb5=1.16.4\n",
      "- lcms2=2.9\n",
      "- ld_impl_linux-64=2.33.1\n",
      "- libarbdb=6.0.6\n",
      "- libblas=3.8.0\n",
      "- libcblas=3.8.0\n",
      "- libcurl=7.68.0\n",
      "- libedit=3.1.20170329\n",
      "- libffi=3.2.1\n",
      "- libgcc=7.2.0\n",
      "- libgcc-ng=9.2.0\n",
      "- libgfortran-ng=7.3.0\n",
      "- libgomp=9.2.0\n",
      "- libiconv=1.15\n",
      "- liblapack=3.8.0\n",
      "- liblapacke=3.8.0\n",
      "- libopenblas=0.3.7\n",
      "- libpng=1.6.37\n",
      "- libsodium=1.0.17\n",
      "- libssh2=1.8.2\n",
      "- libstdcxx-ng=9.2.0\n",
      "- libtiff=4.1.0\n",
      "- libuuid=2.32.1\n",
      "- libxcb=1.13\n",
      "- libxml2=2.9.9\n",
      "- lockfile=0.12.2\n",
      "- lz4-c=1.8.3\n",
      "- mafft=7.310\n",
      "- make=4.3\n",
      "- markupsafe=1.1.1\n",
      "- matplotlib=3.1.1\n",
      "- matplotlib-base=3.1.1\n",
      "- mistune=0.8.4\n",
      "- more-itertools=8.2.0\n",
      "- msgpack-python=1.0.0\n",
      "- natsort=7.0.1\n",
      "- nbconvert=5.6.1\n",
      "- nbformat=5.0.4\n",
      "- ncurses=6.1\n",
      "- nettle=3.4.1\n",
      "- networkx=2.4\n",
      "- nose=1.3.7\n",
      "- notebook=6.0.3\n",
      "- numpy=1.18.1\n",
      "- olefile=0.46\n",
      "- openjdk=11.0.1\n",
      "- openssl=1.1.1d\n",
      "- packaging=20.1\n",
      "- pandas=0.25.3\n",
      "- pandoc=2.9.2\n",
      "- pandocfilters=1.4.2\n",
      "- pango=1.40.14\n",
      "- parso=0.6.1\n",
      "- patsy=0.5.1\n",
      "- pcre=8.44\n",
      "- perl=5.26.2\n",
      "- perl-archive-tar=2.32\n",
      "- perl-carp=1.38\n",
      "- perl-common-sense=3.74\n",
      "- perl-compress-raw-bzip2=2.087\n",
      "- perl-compress-raw-zlib=2.087\n",
      "- perl-exporter=5.72\n",
      "- perl-exporter-tiny=1.002001\n",
      "- perl-extutils-makemaker=7.36\n",
      "- perl-io-compress=2.087\n",
      "- perl-io-zlib=1.10\n",
      "- perl-json=4.02\n",
      "- perl-json-xs=2.34\n",
      "- perl-list-moreutils=0.428\n",
      "- perl-list-moreutils-xs=0.428\n",
      "- perl-pathtools=3.75\n",
      "- perl-scalar-list-utils=1.52\n",
      "- perl-types-serialiser=1.0\n",
      "- perl-xsloader=0.24\n",
      "- pexpect=4.8.0\n",
      "- pickleshare=0.7.5\n",
      "- pigz=2.3.4\n",
      "- pillow=7.0.0\n",
      "- pip=20.0.2\n",
      "- pixman=0.38.0\n",
      "- pluggy=0.12.0\n",
      "- prometheus_client=0.7.1\n",
      "- prompt_toolkit=3.0.3\n",
      "- psutil=5.7.0\n",
      "- pthread-stubs=0.4\n",
      "- ptyprocess=0.6.0\n",
      "- py=1.8.1\n",
      "- pycparser=2.19\n",
      "- pygments=2.5.2\n",
      "- pyopenssl=19.1.0\n",
      "- pyparsing=2.4.6\n",
      "- pyqt=5.9.2\n",
      "- pyrsistent=0.15.7\n",
      "- pysocks=1.7.1\n",
      "- pytest=5.3.5\n",
      "- python=3.6.7\n",
      "- python-dateutil=2.8.1\n",
      "- pytz=2019.3\n",
      "- pyyaml=5.3\n",
      "- pyzmq=19.0.0\n",
      "- q2-alignment=2020.2.0\n",
      "- q2-composition=2020.2.0\n",
      "- q2-cutadapt=2020.2.0\n",
      "- q2-dada2=2020.2.0\n",
      "- q2-deblur=2020.2.0\n",
      "- q2-demux=2020.2.0\n",
      "- q2-diversity=2020.2.0\n",
      "- q2-emperor=2020.2.0\n",
      "- q2-feature-classifier=2020.2.0\n",
      "- q2-feature-table=2020.2.0\n",
      "- q2-fragment-insertion=2020.2.0\n",
      "- q2-gneiss=2020.2.0\n",
      "- q2-longitudinal=2020.2.0\n",
      "- q2-metadata=2020.2.0\n",
      "- q2-phylogeny=2020.2.0\n",
      "- q2-quality-control=2020.2.0\n",
      "- q2-quality-filter=2020.2.0\n",
      "- q2-sample-classifier=2020.2.0\n",
      "- q2-taxa=2020.2.0\n",
      "- q2-types=2020.2.0\n",
      "- q2-vsearch=2020.2.0\n",
      "- q2cli=2020.2.0\n",
      "- q2templates=2020.2.0\n",
      "- qiime2=2020.2.0\n",
      "- qt=5.9.7\n",
      "- r-assertthat=0.2.1\n",
      "- r-backports=1.1.5\n",
      "- r-base=3.5.1\n",
      "- r-bh=1.72.0_3\n",
      "- r-bitops=1.0_6\n",
      "- r-cli=2.0.2\n",
      "- r-cluster=2.1.0\n",
      "- r-colorspace=1.4_1\n",
      "- r-crayon=1.3.4\n",
      "- r-data.table=1.12.6\n",
      "- r-digest=0.6.25\n",
      "- r-ellipsis=0.3.0\n",
      "- r-fansi=0.4.1\n",
      "- r-farver=2.0.3\n",
      "- r-formatr=1.7\n",
      "- r-futile.logger=1.4.3\n",
      "- r-futile.options=1.0.1\n",
      "- r-ggplot2=3.2.1\n",
      "- r-glue=1.3.1\n",
      "- r-gtable=0.3.0\n",
      "- r-hwriter=1.3.2\n",
      "- r-labeling=0.3\n",
      "- r-lambda.r=1.2.4\n",
      "- r-lattice=0.20_40\n",
      "- r-latticeextra=0.6_28\n",
      "- r-lazyeval=0.2.2\n",
      "- r-lifecycle=0.1.0\n",
      "- r-magrittr=1.5\n",
      "- r-mass=7.3_51.5\n",
      "- r-matrix=1.2_18\n",
      "- r-matrixstats=0.55.0\n",
      "- r-mgcv=1.8_31\n",
      "- r-munsell=0.5.0\n",
      "- r-nlme=3.1_144\n",
      "- r-permute=0.9_5\n",
      "- r-pillar=1.4.3\n",
      "- r-pkgconfig=2.0.3\n",
      "- r-plyr=1.8.5\n",
      "- r-r6=2.4.1\n",
      "- r-rcolorbrewer=1.1_2\n",
      "- r-rcpp=1.0.3\n",
      "- r-rcppparallel=4.4.4\n",
      "- r-rcurl=1.98_1.1\n",
      "- r-reshape2=1.4.3\n",
      "- r-rlang=0.4.4\n",
      "- r-scales=1.1.0\n",
      "- r-snow=0.4_3\n",
      "- r-stringi=1.4.3\n",
      "- r-stringr=1.4.0\n",
      "- r-tibble=2.1.3\n",
      "- r-utf8=1.1.4\n",
      "- r-vctrs=0.2.3\n",
      "- r-vegan=2.5_6\n",
      "- r-viridislite=0.3.0\n",
      "- r-withr=2.1.2\n",
      "- r-zeallot=0.1.0\n",
      "- raxml=8.2.12\n",
      "- readline=8.0\n",
      "- requests=2.23.0\n",
      "- scikit-bio=0.5.5\n",
      "- scikit-learn=0.22.1\n",
      "- scipy=1.4.1\n",
      "- seaborn=0.10.0\n",
      "- send2trash=1.5.0\n",
      "- sepp=4.3.10\n",
      "- setuptools=45.2.0\n",
      "- sina=1.6.0\n",
      "- sip=4.19.8\n",
      "- six=1.14.0\n",
      "- sortmerna=2.0\n",
      "- sqlite=3.30.1\n",
      "- statsmodels=0.11.1\n",
      "- tbb=2019.9\n",
      "- terminado=0.8.3\n",
      "- testpath=0.4.4\n",
      "- tk=8.6.10\n",
      "- tktable=2.10\n",
      "- tornado=6.0.3\n",
      "- traitlets=4.3.3\n",
      "- tzlocal=2.0.0\n",
      "- unifrac=0.10.0\n",
      "- urllib3=1.25.7\n",
      "- vsearch=2.7.0\n",
      "- wcwidth=0.1.8\n",
      "- webencodings=0.5.1\n",
      "- wheel=0.34.2\n",
      "- widgetsnbextension=3.5.1\n",
      "- xopen=0.8.4\n",
      "- xorg-fixesproto=5.0\n",
      "- xorg-inputproto=2.3.2\n",
      "- xorg-kbproto=1.0.7\n",
      "- xorg-libice=1.0.10\n",
      "- xorg-libsm=1.2.3\n",
      "- xorg-libx11=1.6.9\n",
      "- xorg-libxau=1.0.9\n",
      "- xorg-libxdmcp=1.1.3\n",
      "- xorg-libxext=1.3.4\n",
      "- xorg-libxfixes=5.0.3\n",
      "- xorg-libxi=1.7.10\n",
      "- xorg-libxrender=0.9.10\n",
      "- xorg-libxtst=1.2.3\n",
      "- xorg-recordproto=1.14.2\n",
      "- xorg-renderproto=0.11.1\n",
      "- xorg-xextproto=7.3.0\n",
      "- xorg-xproto=7.0.31\n",
      "- xz=5.2.4\n",
      "- yaml=0.2.2\n",
      "- zeromq=4.3.2\n",
      "- zipp=3.0.0\n",
      "- zlib=1.2.11\n",
      "- zstd=1.4.4\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/traitar.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pip\n",
      "- scipy\n",
      "- pandas\n",
      "- numpy\n",
      "- matplotlib\n",
      "- pigz\n",
      "- bioconda::hmmer\n",
      "- pip:\n",
      "  - traitar3>=0.1.1\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/cctyper.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "- russel88\n",
      "dependencies:\n",
      "- pigz\n",
      "- cctyper\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/comparem.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::comparem\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/bigscape.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- python=3.6\n",
      "- hmmer\n",
      "- biopython\n",
      "- fasttree\n",
      "- numpy\n",
      "- scipy\n",
      "- networkx\n",
      "- scikit-learn=0.19.1\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/rgi.yaml <==\n",
      "name: rgi\n",
      "channels:\n",
      "  - bioconda\n",
      "  - conda-forge\n",
      "  - defaults\n",
      "dependencies:\n",
      "  - _libgcc_mutex=0.1=conda_forge\n",
      "  - _openmp_mutex=4.5=1_llvm\n",
      "  - bamtools=2.5.1=he860b03_5\n",
      "  - bedtools=2.29.2=hc088bd4_0\n",
      "  - biopython=1.72=py36h14c3975_1000\n",
      "  - blast=2.9.0=pl526h3066fca_4\n",
      "  - bowtie2=2.4.1=py36h7f0b59b_2\n",
      "  - bwa=0.7.17=hed695b0_7\n",
      "  - bzip2=1.0.8=h516909a_2\n",
      "  - ca-certificates=2020.4.5.2=hecda079_0\n",
      "  - certifi=2020.4.5.2=py36h9f0ad1d_0\n",
      "  - curl=7.69.1=h33f0ec9_0\n",
      "  - cycler=0.10.0=py_2\n",
      "  - diamond=0.8.36=h8b12597_4\n",
      "  - entrez-direct=13.3=pl526h375a9b1_0\n",
      "  - expat=2.2.9=he1b5a44_2\n",
      "  - filetype=1.0.7=pyh9f0ad1d_0\n",
      "  - freetype=2.10.2=he06d7ca_0\n",
      "  - icu=64.2=he1b5a44_1\n",
      "  - kiwisolver=1.2.0=py36hdb11119_0\n",
      "  - krb5=1.17.1=h2fd8d38_0\n",
      "  - ld_impl_linux-64=2.34=h53a641e_5\n",
      "  - libblas=3.8.0=16_openblas\n",
      "  - libcblas=3.8.0=16_openblas\n",
      "  - libcurl=7.69.1=hf7181ac_0\n",
      "  - libdeflate=1.0=h14c3975_1\n",
      "  - libedit=3.1.20191231=h46ee950_0\n",
      "  - libffi=3.2.1=he1b5a44_1007\n",
      "  - libgcc-ng=9.2.0=h24d8f2e_2\n",
      "  - libgfortran-ng=7.5.0\n",
      "  - liblapack=3.8.0=16_openblas\n",
      "  - libopenblas=0.3.9=h5ec1e0e_0\n",
      "  - libpng=1.6.37=hed695b0_1\n",
      "  - libssh2=1.9.0=hab1572f_2\n",
      "  - libstdcxx-ng=9.2.0=hdf63c60_2\n",
      "  - llvm-openmp=10.0.0=hc9558a2_0\n",
      "  - matplotlib-base=3.2.1=py36hb8e4980_0\n",
      "  - ncurses=6.1=hf484d3e_1002\n",
      "  - numpy=1.18.5=py36h7314795_0\n",
      "  - oligoarrayaux=3.8=hc9558a2_0\n",
      "  - openssl=1.1.1g=h516909a_0\n",
      "  - pandas=1.0.4=py36h830a2c2_0\n",
      "  - patsy=0.5.1=py_0\n",
      "  - pcre=8.44=he1b5a44_0\n",
      "  - perl=5.26.2=h516909a_1006\n",
      "  - perl-app-cpanminus=1.7044=pl526_1\n",
      "  - perl-archive-tar=2.32=pl526_0\n",
      "  - perl-base=2.23=pl526_1\n",
      "  - perl-business-isbn=3.004=pl526_0\n",
      "  - perl-business-isbn-data=20140910.003=pl526_0\n",
      "  - perl-carp=1.38=pl526_3\n",
      "  - perl-common-sense=3.74=pl526_2\n",
      "  - perl-compress-raw-bzip2=2.087=pl526he1b5a44_0\n",
      "  - perl-compress-raw-zlib=2.087=pl526hc9558a2_0\n",
      "  - perl-constant=1.33=pl526_1\n",
      "  - perl-data-dumper=2.173=pl526_0\n",
      "  - perl-digest-hmac=1.03=pl526_3\n",
      "  - perl-digest-md5=2.55=pl526_0\n",
      "  - perl-encode=2.88=pl526_1\n",
      "  - perl-encode-locale=1.05=pl526_6\n",
      "  - perl-exporter=5.72=pl526_1\n",
      "  - perl-exporter-tiny=1.002001=pl526_0\n",
      "  - perl-extutils-makemaker=7.36=pl526_1\n",
      "  - perl-file-listing=6.04=pl526_1\n",
      "  - perl-file-path=2.16=pl526_0\n",
      "  - perl-file-temp=0.2304=pl526_2\n",
      "  - perl-html-parser=3.72=pl526h6bb024c_5\n",
      "  - perl-html-tagset=3.20=pl526_3\n",
      "  - perl-html-tree=5.07=pl526_1\n",
      "  - perl-http-cookies=6.04=pl526_0\n",
      "  - perl-http-daemon=6.01=pl526_1\n",
      "  - perl-http-date=6.02=pl526_3\n",
      "  - perl-http-message=6.18=pl526_0\n",
      "  - perl-http-negotiate=6.01=pl526_3\n",
      "  - perl-io-compress=2.087=pl526he1b5a44_0\n",
      "  - perl-io-html=1.001=pl526_2\n",
      "  - perl-io-socket-ssl=2.066=pl526_0\n",
      "  - perl-io-zlib=1.10=pl526_2\n",
      "  - perl-json=4.02=pl526_0\n",
      "  - perl-json-xs=2.34=pl526h6bb024c_3\n",
      "  - perl-libwww-perl=6.39=pl526_0\n",
      "  - perl-list-moreutils=0.428=pl526_1\n",
      "  - perl-list-moreutils-xs=0.428=pl526_0\n",
      "  - perl-lwp-mediatypes=6.04=pl526_0\n",
      "  - perl-lwp-protocol-https=6.07=pl526_4\n",
      "  - perl-mime-base64=3.15=pl526_1\n",
      "  - perl-mozilla-ca=20180117=pl526_1\n",
      "  - perl-net-http=6.19=pl526_0\n",
      "  - perl-net-ssleay=1.88=pl526h90d6eec_0\n",
      "  - perl-ntlm=1.09=pl526_4\n",
      "  - perl-parent=0.236=pl526_1\n",
      "  - perl-pathtools=3.75=pl526h14c3975_1\n",
      "  - perl-scalar-list-utils=1.52=pl526h516909a_0\n",
      "  - perl-socket=2.027=pl526_1\n",
      "  - perl-storable=3.15=pl526h14c3975_0\n",
      "  - perl-test-requiresinternet=0.05=pl526_0\n",
      "  - perl-time-local=1.28=pl526_1\n",
      "  - perl-try-tiny=0.30=pl526_1\n",
      "  - perl-types-serialiser=1.0=pl526_2\n",
      "  - perl-uri=1.76=pl526_0\n",
      "  - perl-www-robotrules=6.02=pl526_3\n",
      "  - perl-xml-namespacesupport=1.12=pl526_0\n",
      "  - perl-xml-parser=2.44=pl526h4e0c4b3_7\n",
      "  - perl-xml-sax=1.02=pl526_0\n",
      "  - perl-xml-sax-base=1.09=pl526_0\n",
      "  - perl-xml-sax-expat=0.51=pl526_3\n",
      "  - perl-xml-simple=2.25=pl526_1\n",
      "  - perl-xsloader=0.24=pl526_0\n",
      "  - pip=20.1.1=py_1\n",
      "  - prodigal=2.6.3=h516909a_2\n",
      "  - pyahocorasick=1.4.0=py36h8c4c3a4_1\n",
      "  - pyfaidx=0.5.8=py_1\n",
      "  - pyparsing=2.4.7=pyh9f0ad1d_0\n",
      "  - python=3.6.10=h8356626_1011_cpython\n",
      "  - python-dateutil=2.8.1=py_0\n",
      "  - python_abi=3.6=1_cp36m\n",
      "  - pytz=2020.1=pyh9f0ad1d_0\n",
      "  - readline=8.0=hf8c457e_0\n",
      "  - rgi=5.1.0=py_1\n",
      "  - samtools=1.9=h8571acd_11\n",
      "  - scipy=1.4.1=py36h2d22cac_3\n",
      "  - seaborn=0.10.1=py_0\n",
      "  - setuptools=47.1.1=py36h9f0ad1d_0\n",
      "  - six=1.15.0=pyh9f0ad1d_0\n",
      "  - sqlite=3.30.1=hcee41ef_0\n",
      "  - statsmodels=0.11.1=py36h8c4c3a4_1\n",
      "  - tbb=2020.1=hc9558a2_0\n",
      "  - tk=8.6.10=hed695b0_0\n",
      "  - tornado=6.0.4=py36h8c4c3a4_1\n",
      "  - wheel=0.34.2=py_1\n",
      "  - xz=5.2.5=h516909a_0\n",
      "  - zlib=1.2.11=h516909a_1006\n",
      "prefix: /home/users/ldenies/apps/miniconda3/envs/rgi\n",
      "\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/pathofact_hmm.yaml <==\n",
      "channels:\n",
      "  - bioconda\n",
      "  - conda-forge\n",
      "  - defaults\n",
      "dependencies:\n",
      "  - _libgcc_mutex=0.1=conda_forge\n",
      "  - _openmp_mutex=4.5=0_gnu\n",
      "  - hmmer=3.2.1=he1b5a44_2\n",
      "  - libgcc-ng=9.2.0=h24d8f2e_2\n",
      "  - libgomp=9.2.0=h24d8f2e_2\n",
      "  - libstdcxx-ng=9.2.0=hdf63c60_2\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/pathofact_biopython.yaml <==\n",
      "channels:\n",
      "  - conda-forge\n",
      "  - bioconda\n",
      "  - defaults\n",
      "dependencies:\n",
      "  - _libgcc_mutex=0.1=main\n",
      "  - biopython=1.74=py36h516909a_0\n",
      "  - ca-certificates=2019.6.16=hecc5488_0\n",
      "  - certifi=2019.6.16=py36_1\n",
      "  - cycler=0.10.0=py_1\n",
      "  - dbus=1.13.6=he372182_0\n",
      "  - expat=2.2.5=he1b5a44_1003\n",
      "  - fontconfig=2.13.1=he4413a7_1000\n",
      "  - freetype=2.10.0=he983fc9_0\n",
      "  - gettext=0.19.8.1=hc5be6a0_1002\n",
      "  - glib=2.58.3=h6f030ca_1002\n",
      "  - gst-plugins-base=1.14.5=h0935bb2_0\n",
      "  - gstreamer=1.14.5=h36ae1b5_0\n",
      "  - icu=58.2=hf484d3e_1000\n",
      "  - joblib=0.13.2=py_0\n",
      "  - jpeg=9c=h14c3975_1001\n",
      "  - kiwisolver=1.1.0=py36hc9558a2_0\n",
      "  - libblas=3.8.0=11_openblas\n",
      "  - libcblas=3.8.0=11_openblas\n",
      "  - libedit=3.1.20181209=hc058e9b_0\n",
      "  - libffi=3.2.1=hd88cf55_4\n",
      "  - libgcc-ng=9.1.0=hdf63c60_0\n",
      "  - libgfortran-ng=7.3.0=hdf63c60_0\n",
      "  - libiconv=1.15=h516909a_1005\n",
      "  - liblapack=3.8.0=11_openblas\n",
      "  - libopenblas=0.3.6=h6e990d7_6\n",
      "  - libpng=1.6.37=hed695b0_0\n",
      "  - libstdcxx-ng=9.1.0=hdf63c60_0\n",
      "  - libuuid=2.32.1=h14c3975_1000\n",
      "  - libxcb=1.13=h14c3975_1002\n",
      "  - libxml2=2.9.9=h13577e0_2\n",
      "  - matplotlib=3.1.0=py36h5429711_0\n",
      "  - ncurses=6.1=he6710b0_1\n",
      "  - numpy=1.17.0=py36h95a1406_0\n",
      "  - openssl=1.1.1c=h516909a_0\n",
      "  - pandas=0.25.0=py36hb3f55d8_0\n",
      "  - patsy=0.5.1=py_0\n",
      "  - pcre=8.41=hf484d3e_1003\n",
      "  - pip=19.1.1=py36_0\n",
      "  - pthread-stubs=0.4=h14c3975_1001\n",
      "  - pyparsing=2.4.2=py_0\n",
      "  - pyqt=5.9.2=py36hcca6a23_2\n",
      "  - python=3.6.9=h265db76_0\n",
      "  - python-dateutil=2.8.0=py_0\n",
      "  - pytz=2019.2=py_0\n",
      "  - qt=5.9.7=h52cfd70_2\n",
      "  - readline=7.0=h7b6447c_5\n",
      "  - scikit-learn=0.21.3=py36hcdab131_0\n",
      "  - scipy=1.3.0=py36h921218d_1\n",
      "  - seaborn=0.9.0=py_0\n",
      "  - setuptools=41.0.1=py36_0\n",
      "  - sip=4.19.8=py36hf484d3e_1000\n",
      "  - six=1.12.0=py36_1000\n",
      "  - sqlite=3.29.0=h7b6447c_0\n",
      "  - statsmodels=0.10.1=py36hc1659b7_0\n",
      "  - tk=8.6.8=hbc83047_0\n",
      "  - tornado=6.0.3=py36h516909a_0\n",
      "  - wheel=0.33.4=py36_0\n",
      "  - xorg-libxau=1.0.9=h14c3975_0\n",
      "  - xorg-libxdmcp=1.1.3=h516909a_0\n",
      "  - xz=5.2.4=h14c3975_4\n",
      "  - zlib=1.2.11=h7b6447c_3\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/platon.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::platon\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/plasflow.yaml <==\n",
      "channels:\n",
      "  # for PlasFlow\n",
      "  - smaegol\n",
      "  # rest\n",
      "  - bioconda\n",
      "  - conda-forge\n",
      "  - defaults\n",
      "dependencies:\n",
      "  - seqkit\n",
      "  - _libgcc_mutex=0.1=conda_forge\n",
      "  - _openmp_mutex=4.5=1_llvm\n",
      "  - _r-mutex=1.0.1=anacondar_1\n",
      "  - backports=1.0=py_2\n",
      "  - backports.weakref=1.0rc1=py35_1\n",
      "  - bioconductor-biocgenerics=0.22.0=r3.3.2_0\n",
      "  - bioconductor-biostrings=2.42.1=r3.3.2_0\n",
      "  - bioconductor-iranges=2.8.2=r3.3.2_0\n",
      "  - bioconductor-s4vectors=0.12.2=r3.3.2_0\n",
      "  - bioconductor-xvector=0.14.1=r3.3.2_0\n",
      "  - bioconductor-zlibbioc=1.20.0=r3.3.2_1\n",
      "  - biopython=1.68=py35_0\n",
      "  - blas=2.16=openblas\n",
      "  - bleach=1.5.0=py35_0\n",
      "  - bzip2=1.0.8=h516909a_2\n",
      "  - ca-certificates=2020.4.5.1=hecc5488_0\n",
      "  - cairo=1.14.6=4\n",
      "  - certifi=2018.8.24=py35_1001\n",
      "  - curl=7.52.1=0\n",
      "  - fontconfig=2.12.1=6\n",
      "  - freetype=2.7=1\n",
      "  - gettext=0.19.8.1=hc5be6a0_1002\n",
      "  - glib=2.51.4=0\n",
      "  - graphite2=1.3.13=he1b5a44_1001\n",
      "  - gsl=2.6=h294904e_0\n",
      "  - harfbuzz=1.4.3=0\n",
      "  - html5lib=0.9999999=py35_0\n",
      "  - icu=58.2=hf484d3e_1000\n",
      "  - jpeg=9c=h14c3975_1001\n",
      "  - libblas=3.8.0=16_openblas\n",
      "  - libcblas=3.8.0=16_openblas\n",
      "  - libffi=3.2.1=he1b5a44_1007\n",
      "  - libgcc=7.2.0=h69d50b8_2\n",
      "  - libgcc-ng=9.2.0=h24d8f2e_2\n",
      "  - libgfortran-ng=7.3.0=hdf63c60_5\n",
      "  - libiconv=1.15=h516909a_1006\n",
      "  - liblapack=3.8.0=16_openblas\n",
      "  - liblapacke=3.8.0=16_openblas\n",
      "  - libopenblas=0.3.9=h5ec1e0e_0\n",
      "  - libpng=1.6.28=1\n",
      "  - libstdcxx-ng=9.2.0=hdf63c60_2\n",
      "  - libtiff=4.0.7=0\n",
      "  - libxml2=2.9.5=0\n",
      "  - llvm-openmp=10.0.0=hc9558a2_0\n",
      "  - markdown=3.2.1=py_0\n",
      "  - mmtf-python=1.0.2=py35_0\n",
      "  - mock=2.0.0=py35_0\n",
      "  - msgpack-python=0.5.6=py35h2d50403_3\n",
      "  - ncurses=5.9=10\n",
      "  - numpy=1.15.2=py35h99e49ec_0\n",
      "  - numpy-base=1.15.2=py35h2f8d375_0\n",
      "  - olefile=0.46=py_0\n",
      "  - openssl=1.0.2u=h516909a_0\n",
      "  - pandas=0.23.4=py35hf8a1672_0\n",
      "  - pango=1.40.4=0\n",
      "  - pbr=5.4.2=py_0\n",
      "  - pcre=8.44=he1b5a44_0\n",
      "  - pillow=4.3.0=py35_0\n",
      "  - pip=20.0.2=py_2\n",
      "  - pixman=0.34.0=h14c3975_1003\n",
      "  - protobuf=3.4.0=py35_0\n",
      "  - python=3.5.4=0\n",
      "  - python-dateutil=2.8.1=py_0\n",
      "  - pytz=2019.3=py_0\n",
      "  - r-base=3.3.2=5\n",
      "  - readline=6.2=0\n",
      "  - reportlab=3.4.0=py35_0\n",
      "  - rpy2=2.7.8=py35r3.3.2_1\n",
      "  - scikit-learn=0.20.0=py35h22eb022_1\n",
      "  - scipy=1.1.0=py35he2b7bc3_1\n",
      "  - setuptools=40.4.3=py35_0\n",
      "  - six=1.14.0=py_1\n",
      "  - sqlite=3.13.0=1\n",
      "  - tensorflow=1.3.0=py35_0\n",
      "  - tk=8.5.19=2\n",
      "  - webencodings=0.5.1=py_1\n",
      "  - werkzeug=1.0.1=pyh9f0ad1d_0\n",
      "  - wheel=0.34.2=py_1\n",
      "  - xz=5.2.5=h516909a_0\n",
      "  - zlib=1.2.8=3\n",
      "  # PlasFlow: should be installed from smaegol, NOT from pip\n",
      "  - plasflow=1.1.0"
     ]
    }
   ],
   "source": [
    "pipelineInfo('/ebio/abt3_projects/software/dev/ll_pipelines/llg/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-28T13:09:28.618881Z",
     "iopub.status.busy": "2021-04-28T13:09:28.616477Z",
     "iopub.status.idle": "2021-04-28T13:09:28.670099Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LLPRIMER\n",
      "========\n",
      "\n",
      "Ley Lab Primer generation pipeline (LLPRIMER)\n",
      "\n",
      "* Version: 0.3.0\n",
      "* Authors:\n",
      "  * Nick Youngblut <nyoungb2@gmail.com>\n",
      "* Maintainers:\n",
      "  * Nick Youngblut <nyoungb2@gmail.com>\n",
      "\n",
      "--- conda envs ---\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llprimer//bin/envs/pdp.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- python=3.7\n",
      "- intervaltree\n",
      "- prodigal\n",
      "- blast\n",
      "- bedtools\n",
      "- mafft\n",
      "- mummer=3.23\n",
      "- emboss\n",
      "- primer3=1.1.4\n",
      "- biopython<1.78\n",
      "- pybedtools\n",
      "- joblib\n",
      "- tqdm\n",
      "- openpyxl\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llprimer//bin/envs/genes.yaml <==\n",
      "channels:\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- python=3\n",
      "- numpy\n",
      "- pyfaidx\n",
      "- bioconda::seqkit\n",
      "- bioconda::fasta-splitter\n",
      "- bioconda::vsearch\n",
      "- bioconda::mmseqs2\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llprimer//bin/envs/aln.yaml <==\n",
      "channels:\n",
      "- bioconda\n",
      "- conda-forge\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::revtrans\n",
      "- bioconda::kalign3\n",
      "- bioconda::mafft\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llprimer//bin/envs/blast.yaml <==\n",
      "channels:\n",
      "- bioconda\n",
      "- conda-forge\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::seqkit\n",
      "- bioconda::biopython<1.78\n",
      "- bioconda::blast>=2.10.1\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llprimer//bin/envs/primer3.yaml <==\n",
      "channels:\n",
      "- bioconda\n",
      "- conda-forge\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::biopython<1.78\n",
      "- bioconda::primer3-py\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llprimer//bin/envs/prokka.yaml <==\n",
      "channels:\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::prokka"
     ]
    }
   ],
   "source": [
    "pipelineInfo('/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-28T13:09:29.340588Z",
     "iopub.status.busy": "2021-04-28T13:09:29.337567Z",
     "iopub.status.idle": "2021-04-28T13:09:29.902219Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "R version 3.6.3 (2020-02-29)\n",
       "Platform: x86_64-conda_cos6-linux-gnu (64-bit)\n",
       "Running under: Ubuntu 18.04.5 LTS\n",
       "\n",
       "Matrix products: default\n",
       "BLAS/LAPACK: /ebio/abt3_projects/Georg_animal_feces/envs/tidyverse/lib/libopenblasp-r0.3.9.so\n",
       "\n",
       "locale:\n",
       " [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              \n",
       " [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    \n",
       " [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   \n",
       " [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 \n",
       " [9] LC_ADDRESS=C               LC_TELEPHONE=C            \n",
       "[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       \n",
       "\n",
       "attached base packages:\n",
       "[1] stats     graphics  grDevices utils     datasets  methods   base     \n",
       "\n",
       "other attached packages:\n",
       "[1] LeyLabRMisc_0.1.6 ggplot2_3.3.1     tidyr_1.1.0       dplyr_1.0.0      \n",
       "\n",
       "loaded via a namespace (and not attached):\n",
       " [1] Rcpp_1.0.4.6     magrittr_1.5     munsell_0.5.0    tidyselect_1.1.0\n",
       " [5] uuid_0.1-4       colorspace_1.4-1 R6_2.4.1         rlang_0.4.6     \n",
       " [9] tools_3.6.3      grid_3.6.3       gtable_0.3.0     withr_2.2.0     \n",
       "[13] htmltools_0.4.0  ellipsis_0.3.1   digest_0.6.25    tibble_3.0.1    \n",
       "[17] lifecycle_0.2.0  crayon_1.3.4     IRdisplay_0.7.0  purrr_0.3.4     \n",
       "[21] repr_1.1.0       base64enc_0.1-3  vctrs_0.3.1      IRkernel_1.1    \n",
       "[25] glue_1.4.1       evaluate_0.14    stringi_1.4.6    pbdZMQ_0.3-3    \n",
       "[29] compiler_3.6.3   pillar_1.4.4     scales_1.1.1     generics_0.0.2  \n",
       "[33] jsonlite_1.6.1   pkgconfig_2.0.3 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "sessionInfo()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R [conda env:Georg_animal_feces-tidyverse]",
   "language": "R",
   "name": "conda-env-Georg_animal_feces-tidyverse-r"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.3"
  },
  "toc-autonumbering": true
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
