{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Goal\n",
    "\n",
    "* Primer design for clade of interest"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Var"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-28T13:22:23.149486Z",
     "iopub.status.busy": "2021-04-28T13:22:23.147379Z",
     "iopub.status.idle": "2021-04-28T13:22:23.174298Z"
    }
   },
   "outputs": [],
   "source": [
    "base_dir = '/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/'\n",
    "clade = 'Butyricicoccus'\n",
    "domain = 'Bacteria'\n",
    "taxid = 580596 "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Init"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-28T13:22:25.137764Z",
     "iopub.status.busy": "2021-04-28T13:22:25.135424Z",
     "iopub.status.idle": "2021-04-28T13:22:25.157512Z"
    }
   },
   "outputs": [],
   "source": [
    "library(dplyr)\n",
    "library(tidyr)\n",
    "library(ggplot2)\n",
    "library(LeyLabRMisc)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-28T13:22:25.273239Z",
     "iopub.status.busy": "2021-04-28T13:22:25.270954Z",
     "iopub.status.idle": "2021-04-28T13:22:25.289234Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Directory already exists: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3//Butyricicoccus \n"
     ]
    }
   ],
   "source": [
    "df.dims()\n",
    "work_dir = file.path(base_dir, clade)\n",
    "make_dir(work_dir)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Genome download\n",
    "\n",
    "* Downloading genomes from NCBI"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-12T13:07:32.874718Z",
     "iopub.status.busy": "2021-01-12T13:07:32.867938Z",
     "iopub.status.idle": "2021-01-12T13:07:32.985785Z",
     "shell.execute_reply": "2021-01-12T13:07:32.983225Z"
    }
   },
   "source": [
    "```\n",
    "OUTDIR=/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricicoccus/\n",
    "mkdir -p $OUTDIR\n",
    "ncbi-genome-download -p 12 -s genbank -F fasta --genera Butyricicoccus -o $OUTDIR bacteria\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Genome quality\n",
    "\n",
    "* Filtering genomes by quality"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-28T13:22:26.067254Z",
     "iopub.status.busy": "2021-04-28T13:22:26.065084Z",
     "iopub.status.idle": "2021-04-28T13:22:26.157681Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 36 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Name</th><th scope=col>Fasta</th><th scope=col>Domain</th><th scope=col>Taxid</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>GCA_000398925.1_Buty_pull_1_20_V1_genomic</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricicoccus/genbank/bacteria/GCA_000398925.1/GCA_000398925.1_Buty_pull_1_20_V1_genomic.fna.gz</td><td>Bacteria</td><td>580596</td></tr>\n",
       "\t<tr><td>GCA_001643385.1_ASM164338v1_genomic      </td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricicoccus/genbank/bacteria/GCA_001643385.1/GCA_001643385.1_ASM164338v1_genomic.fna.gz      </td><td>Bacteria</td><td>580596</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>GCA_900167005.1_IMG-taxon_2585428076_annotated_assembly_genomic</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricicoccus/genbank/bacteria/GCA_900167005.1/GCA_900167005.1_IMG-taxon_2585428076_annotated_assembly_genomic.fna.gz</td><td>Bacteria</td><td>580596</td></tr>\n",
       "\t<tr><td>GCA_902374965.1_MGYG-HGUT-01434_genomic                        </td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricicoccus/genbank/bacteria/GCA_902374965.1/GCA_902374965.1_MGYG-HGUT-01434_genomic.fna.gz                        </td><td>Bacteria</td><td>580596</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 36 × 4\n",
       "\\begin{tabular}{llll}\n",
       " Name & Fasta & Domain & Taxid\\\\\n",
       " <chr> & <chr> & <fct> & <dbl>\\\\\n",
       "\\hline\n",
       "\t GCA\\_000398925.1\\_Buty\\_pull\\_1\\_20\\_V1\\_genomic & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/v0.3/Butyricicoccus/genbank/bacteria/GCA\\_000398925.1/GCA\\_000398925.1\\_Buty\\_pull\\_1\\_20\\_V1\\_genomic.fna.gz & Bacteria & 580596\\\\\n",
       "\t GCA\\_001643385.1\\_ASM164338v1\\_genomic       & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/v0.3/Butyricicoccus/genbank/bacteria/GCA\\_001643385.1/GCA\\_001643385.1\\_ASM164338v1\\_genomic.fna.gz       & Bacteria & 580596\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t GCA\\_900167005.1\\_IMG-taxon\\_2585428076\\_annotated\\_assembly\\_genomic & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/v0.3/Butyricicoccus/genbank/bacteria/GCA\\_900167005.1/GCA\\_900167005.1\\_IMG-taxon\\_2585428076\\_annotated\\_assembly\\_genomic.fna.gz & Bacteria & 580596\\\\\n",
       "\t GCA\\_902374965.1\\_MGYG-HGUT-01434\\_genomic                         & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/v0.3/Butyricicoccus/genbank/bacteria/GCA\\_902374965.1/GCA\\_902374965.1\\_MGYG-HGUT-01434\\_genomic.fna.gz                         & Bacteria & 580596\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 36 × 4\n",
       "\n",
       "| Name &lt;chr&gt; | Fasta &lt;chr&gt; | Domain &lt;fct&gt; | Taxid &lt;dbl&gt; |\n",
       "|---|---|---|---|\n",
       "| GCA_000398925.1_Buty_pull_1_20_V1_genomic | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricicoccus/genbank/bacteria/GCA_000398925.1/GCA_000398925.1_Buty_pull_1_20_V1_genomic.fna.gz | Bacteria | 580596 |\n",
       "| GCA_001643385.1_ASM164338v1_genomic       | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricicoccus/genbank/bacteria/GCA_001643385.1/GCA_001643385.1_ASM164338v1_genomic.fna.gz       | Bacteria | 580596 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| GCA_900167005.1_IMG-taxon_2585428076_annotated_assembly_genomic | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricicoccus/genbank/bacteria/GCA_900167005.1/GCA_900167005.1_IMG-taxon_2585428076_annotated_assembly_genomic.fna.gz | Bacteria | 580596 |\n",
       "| GCA_902374965.1_MGYG-HGUT-01434_genomic                         | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricicoccus/genbank/bacteria/GCA_902374965.1/GCA_902374965.1_MGYG-HGUT-01434_genomic.fna.gz                         | Bacteria | 580596 |\n",
       "\n"
      ],
      "text/plain": [
       "   Name                                                           \n",
       "1  GCA_000398925.1_Buty_pull_1_20_V1_genomic                      \n",
       "2  GCA_001643385.1_ASM164338v1_genomic                            \n",
       "⋮  ⋮                                                              \n",
       "35 GCA_900167005.1_IMG-taxon_2585428076_annotated_assembly_genomic\n",
       "36 GCA_902374965.1_MGYG-HGUT-01434_genomic                        \n",
       "   Fasta                                                                                                                                                                                                         \n",
       "1  /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricicoccus/genbank/bacteria/GCA_000398925.1/GCA_000398925.1_Buty_pull_1_20_V1_genomic.fna.gz                      \n",
       "2  /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricicoccus/genbank/bacteria/GCA_001643385.1/GCA_001643385.1_ASM164338v1_genomic.fna.gz                            \n",
       "⋮  ⋮                                                                                                                                                                                                             \n",
       "35 /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricicoccus/genbank/bacteria/GCA_900167005.1/GCA_900167005.1_IMG-taxon_2585428076_annotated_assembly_genomic.fna.gz\n",
       "36 /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricicoccus/genbank/bacteria/GCA_902374965.1/GCA_902374965.1_MGYG-HGUT-01434_genomic.fna.gz                        \n",
       "   Domain   Taxid \n",
       "1  Bacteria 580596\n",
       "2  Bacteria 580596\n",
       "⋮  ⋮        ⋮     \n",
       "35 Bacteria 580596\n",
       "36 Bacteria 580596"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "File written: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3//Butyricicoccus/genbank/samples.txt \n"
     ]
    }
   ],
   "source": [
    "D = file.path(base_dir, clade, 'genbank')\n",
    "files = list_files(D, '.fna.gz')\n",
    "samps = data.frame(Name = files %>% as.character %>% basename,\n",
    "                   Fasta = files,\n",
    "                   Domain = 'Bacteria',\n",
    "                   Taxid = taxid) %>%\n",
    "    mutate(Name = gsub('\\\\.fna\\\\.gz$', '', Name),\n",
    "           Fasta = gsub('/+', '/', Fasta))\n",
    "samps\n",
    "\n",
    "# writing file\n",
    "outfile = file.path(D, 'samples.txt')\n",
    "write_table(samps, outfile)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### LLG"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Config"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-28T13:23:18.392674Z",
     "iopub.status.busy": "2021-04-28T13:23:18.390567Z",
     "iopub.status.idle": "2021-04-28T13:23:18.420430Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# table with genome --> fasta_file information\n",
      "samples_file: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricicoccus/genbank/samples.txt\n",
      "\n",
      "# output location\n",
      "output_dir: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricicoccus/LLG/\n",
      "\n",
      "# temporary file directory (your username will be added automatically)\n",
      "tmp_dir: /ebio/abt3_scratch/\n",
      "\n",
      "# batch processing of genomes for certain steps\n",
      "## increase to better parallelize\n",
      "batches: 2 \n",
      "\n",
      "# Domain of genomes ('Archaea' or 'Bacteria)\n",
      "## Use \"Skip\" if provided as a \"Domain\" column in the genome table\n",
      "Domain: Skip\n",
      "\n",
      "# software parameters\n",
      "# Use \"Skip\" to skip any of these steps. If no params for rule, use \"\"\n",
      "# dRep MAGs are not further analyzed, but you can de-rep & then use the de-rep genome table as input.\n",
      "params:\n",
      "  ionice: -c 3\n",
      "  # assembly assessment\n",
      "  seqkit: \"\"\n",
      "  quast: Skip #\"\"\n",
      "  multiqc_on_quast: \"\" \n",
      "  checkm: \"\"\n",
      "  # de-replication (requires checkm)\n",
      "  drep: -comp 90 -con 5 -sa 0.999\n",
      "  # taxonomy\n",
      "  sourmash:\n",
      "    compute: Skip #--scaled 10000 -k 31\n",
      "    gather: -k 31\n",
      "  gtdbtk:\n",
      "    classify_wf: --min_perc_aa 10\n",
      "  # genome pairwise ANI  \n",
      "  fastani: Skip #--fragLen 3000 --minFraction 0.2 -k 16\n",
      "  dashing: Skip #-k 31 --full-tsv\n",
      "  comparem_aai: Skip #--evalue 0.001\n",
      "  # gene annotation\n",
      "  gene_call:\n",
      "    prokka: Skip #\"\" \n",
      "    multiqc_on_prokka: \"\"\n",
      "    prodigal: Skip #\"\"\n",
      "  eggnog_mapper: \"\"\n",
      "  eggnog_mapper_annot: \"\"\n",
      "  # rRNA (16S alignment & phylogeny)\n",
      "  barrnap: Skip #--lencutoff 0.8\n",
      "  vsearch_per_genome_drep: --id 0.95    # Skip to prevent drep of 16S copies within each genome\n",
      "  qiime2_fasttree: \"\"\n",
      "  qiime2_iqtree: --p-alrt 1000 --p-abayes --p-lbp 1000  --p-substitution-model 'GTR+I+G'\n",
      "  # genome phylogeny\n",
      "  phylophlan_config: Skip #--map_dna diamond --db_aa diamond --map_aa diamond --msa mafft --trim trimal --tree1 fasttree --tree2 raxml\n",
      "  phylophlan: --diversity high --min_num_markers 50 --accurate\n",
      "  # phenotype\n",
      "  traitar: Skip #\"\"\n",
      "  # biosynthetic gene clusters (BGCs)\n",
      "  antismash: Skip #--cb-knownclusters --cb-subclusters --asf  \n",
      "  DeepBGC: Skip #--score 0.5 --classifier-score 0.5 --prodigal-meta-mode\n",
      "  # antimicrobial resistance (AMR)\n",
      "  abricate: Skip #--minid 75 --mincov 80\n",
      "  # CRISPRs\n",
      "  cctyper: Skip #--prodigal meta\n",
      "  \n",
      "# databases\n",
      "databases:\n",
      "  checkM_data: /ebio/abt3_projects/databases_no-backup/checkM/\n",
      "  sourmash: /ebio/abt3_projects/databases_no-backup/sourmash/genbank-k31.sbt.json\n",
      "  sourmash_lca: /ebio/abt3_projects/databases_no-backup/sourmash/genbank-k31.lca.json.gz\n",
      "  gtdbtk: /ebio/abt3_projects/databases_no-backup/GTDB/release95/gtdbtk/db_info.md\n",
      "  phylophlan: /ebio/abt3_projects/databases_no-backup/phylophlan/PhyloPhlan/phylophlan.faa.bz2\n",
      "  eggnog: /ebio/abt3_projects/databases_no-backup/Eggnog/v2/eggnog.db\n",
      "  eggnog_diamond: /ebio/abt3_projects/databases_no-backup/Eggnog/v2/eggnog_proteins.dmnd\n",
      "  antismash: /ebio/abt3_projects/databases_no-backup/antismash/v5/\n",
      "  deepbgc: /ebio/abt3_projects/databases_no-backup/DeepBGC/\n",
      "  traitar: /ebio/abt3_projects/databases_no-backup/pfam/traitar/\n",
      "  abricate:\n",
      "    ncbi: /ebio/abt3_projects/databases_no-backup/abricate/ncbi/sequences\n",
      "    card: /ebio/abt3_projects/databases_no-backup/abricate/card/sequences\n",
      "    resfinder: /ebio/abt3_projects/databases_no-backup/abricate/resfinder/sequences\n",
      "    argannot: /ebio/abt3_projects/databases_no-backup/abricate/argannot/sequences\n",
      "    bacmet2: /ebio/abt3_projects/databases_no-backup/abricate/bacmet2/sequences\n",
      "    vfdb: /ebio/abt3_projects/databases_no-backup/abricate/vfdb/sequences\n",
      "    megares: /ebio/abt3_projects/databases_no-backup/abricate/megares/sequences\n",
      "    plasmidfinder: /ebio/abt3_projects/databases_no-backup/abricate/plasmidfinder/sequences\n",
      "\n",
      "# snakemake pipeline\n",
      "pipeline:\n",
      "  snakemake_folder: ./\n",
      "  script_folder: ./bin/scripts/\n",
      "  use_shared_mem: True\n",
      "  name: LLG"
     ]
    }
   ],
   "source": [
    "cat_file(file.path(work_dir, 'LLG', 'config.yaml'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-12T13:14:50.416604Z",
     "iopub.status.busy": "2021-01-12T13:14:50.414179Z",
     "iopub.status.idle": "2021-01-12T13:14:50.431925Z",
     "shell.execute_reply": "2021-01-12T13:14:50.430260Z"
    }
   },
   "source": [
    "#### Run\n",
    "\n",
    "```\n",
    "(snakemake) @ rick:/ebio/abt3_projects/software/dev/ll_pipelines/llg\n",
    "$ screen -L -S llg ./snakemake_sge.sh /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricicoccus/LLG/config.yaml 20 -F\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-01-12T13:14:10.995042Z",
     "iopub.status.busy": "2021-01-12T13:14:10.992773Z",
     "iopub.status.idle": "2021-01-12T13:14:11.009275Z",
     "shell.execute_reply": "2021-01-12T13:14:11.008255Z"
    }
   },
   "source": [
    "### Samples table of high-quality genomes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-28T13:28:33.004241Z",
     "iopub.status.busy": "2021-04-28T13:28:33.001930Z",
     "iopub.status.idle": "2021-04-28T13:28:33.067064Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 36 × 33</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Bin.Id</th><th scope=col>Marker.lineage</th><th scope=col>X..genomes</th><th scope=col>X..markers</th><th scope=col>X..marker.sets</th><th scope=col>Completeness</th><th scope=col>Contamination</th><th scope=col>Strain.heterogeneity</th><th scope=col>Genome.size..bp.</th><th scope=col>X..ambiguous.bases</th><th scope=col>⋯</th><th scope=col>X0</th><th scope=col>X1</th><th scope=col>X2</th><th scope=col>X3</th><th scope=col>X4</th><th scope=col>X5.</th><th scope=col>assembly.Id</th><th scope=col>assembler.Id</th><th scope=col>taxon.Id</th><th scope=col>File</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>GCA_000398925.1_Buty_pull_1_20_V1_genomic</td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>98.66</td><td>0</td><td>0</td><td>3325721</td><td>26204</td><td>⋯</td><td> 2</td><td>261</td><td>0</td><td>0</td><td>0</td><td>0</td><td>|ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Butyricicoccus|LLG_output|checkM|1|checkm|markers_qa_summary.tsv.1</td><td>markers_qa_summary.tsv.1</td><td>NA</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricicoccus/LLG_output/checkM/1/checkm/markers_qa_summary.tsv.1</td></tr>\n",
       "\t<tr><td>GCA_001643435.1_ASM164343v1_genomic      </td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>67.11</td><td>0</td><td>0</td><td>1452273</td><td> 1058</td><td>⋯</td><td>78</td><td>185</td><td>0</td><td>0</td><td>0</td><td>0</td><td>|ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Butyricicoccus|LLG_output|checkM|1|checkm|markers_qa_summary.tsv.2</td><td>markers_qa_summary.tsv.2</td><td>NA</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricicoccus/LLG_output/checkM/1/checkm/markers_qa_summary.tsv.2</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋱</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>GCA_900143195.1_501571.5_genomic       </td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>98.66</td><td>0</td><td>0</td><td>3284487</td><td> 5872</td><td>⋯</td><td>2</td><td>261</td><td>0</td><td>0</td><td>0</td><td>0</td><td>|ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Butyricicoccus|LLG_output|checkM|2|checkm|markers_qa_summary.tsv.17</td><td>markers_qa_summary.tsv.17</td><td>NA</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricicoccus/LLG_output/checkM/2/checkm/markers_qa_summary.tsv.17</td></tr>\n",
       "\t<tr><td>GCA_902374965.1_MGYG-HGUT-01434_genomic</td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>98.66</td><td>0</td><td>0</td><td>3325721</td><td>26204</td><td>⋯</td><td>2</td><td>261</td><td>0</td><td>0</td><td>0</td><td>0</td><td>|ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Butyricicoccus|LLG_output|checkM|2|checkm|markers_qa_summary.tsv.18</td><td>markers_qa_summary.tsv.18</td><td>NA</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricicoccus/LLG_output/checkM/2/checkm/markers_qa_summary.tsv.18</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 36 × 33\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " Bin.Id & Marker.lineage & X..genomes & X..markers & X..marker.sets & Completeness & Contamination & Strain.heterogeneity & Genome.size..bp. & X..ambiguous.bases & ⋯ & X0 & X1 & X2 & X3 & X4 & X5. & assembly.Id & assembler.Id & taxon.Id & File\\\\\n",
       " <fct> & <fct> & <int> & <int> & <int> & <dbl> & <dbl> & <dbl> & <int> & <int> & ⋯ & <int> & <int> & <int> & <int> & <int> & <int> & <fct> & <fct> & <lgl> & <fct>\\\\\n",
       "\\hline\n",
       "\t GCA\\_000398925.1\\_Buty\\_pull\\_1\\_20\\_V1\\_genomic & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 98.66 & 0 & 0 & 3325721 & 26204 & ⋯ &  2 & 261 & 0 & 0 & 0 & 0 & \\textbar{}ebio\\textbar{}abt3\\_projects\\textbar{}software\\textbar{}dev\\textbar{}ll\\_pipelines\\textbar{}llprimer\\textbar{}experiments\\textbar{}HMP\\_most-wanted\\textbar{}Butyricicoccus\\textbar{}LLG\\_output\\textbar{}checkM\\textbar{}1\\textbar{}checkm\\textbar{}markers\\_qa\\_summary.tsv.1 & markers\\_qa\\_summary.tsv.1 & NA & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/Butyricicoccus/LLG\\_output/checkM/1/checkm/markers\\_qa\\_summary.tsv.1\\\\\n",
       "\t GCA\\_001643435.1\\_ASM164343v1\\_genomic       & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 67.11 & 0 & 0 & 1452273 &  1058 & ⋯ & 78 & 185 & 0 & 0 & 0 & 0 & \\textbar{}ebio\\textbar{}abt3\\_projects\\textbar{}software\\textbar{}dev\\textbar{}ll\\_pipelines\\textbar{}llprimer\\textbar{}experiments\\textbar{}HMP\\_most-wanted\\textbar{}Butyricicoccus\\textbar{}LLG\\_output\\textbar{}checkM\\textbar{}1\\textbar{}checkm\\textbar{}markers\\_qa\\_summary.tsv.2 & markers\\_qa\\_summary.tsv.2 & NA & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/Butyricicoccus/LLG\\_output/checkM/1/checkm/markers\\_qa\\_summary.tsv.2\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋱ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t GCA\\_900143195.1\\_501571.5\\_genomic        & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 98.66 & 0 & 0 & 3284487 &  5872 & ⋯ & 2 & 261 & 0 & 0 & 0 & 0 & \\textbar{}ebio\\textbar{}abt3\\_projects\\textbar{}software\\textbar{}dev\\textbar{}ll\\_pipelines\\textbar{}llprimer\\textbar{}experiments\\textbar{}HMP\\_most-wanted\\textbar{}Butyricicoccus\\textbar{}LLG\\_output\\textbar{}checkM\\textbar{}2\\textbar{}checkm\\textbar{}markers\\_qa\\_summary.tsv.17 & markers\\_qa\\_summary.tsv.17 & NA & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/Butyricicoccus/LLG\\_output/checkM/2/checkm/markers\\_qa\\_summary.tsv.17\\\\\n",
       "\t GCA\\_902374965.1\\_MGYG-HGUT-01434\\_genomic & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 98.66 & 0 & 0 & 3325721 & 26204 & ⋯ & 2 & 261 & 0 & 0 & 0 & 0 & \\textbar{}ebio\\textbar{}abt3\\_projects\\textbar{}software\\textbar{}dev\\textbar{}ll\\_pipelines\\textbar{}llprimer\\textbar{}experiments\\textbar{}HMP\\_most-wanted\\textbar{}Butyricicoccus\\textbar{}LLG\\_output\\textbar{}checkM\\textbar{}2\\textbar{}checkm\\textbar{}markers\\_qa\\_summary.tsv.18 & markers\\_qa\\_summary.tsv.18 & NA & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/Butyricicoccus/LLG\\_output/checkM/2/checkm/markers\\_qa\\_summary.tsv.18\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 36 × 33\n",
       "\n",
       "| Bin.Id &lt;fct&gt; | Marker.lineage &lt;fct&gt; | X..genomes &lt;int&gt; | X..markers &lt;int&gt; | X..marker.sets &lt;int&gt; | Completeness &lt;dbl&gt; | Contamination &lt;dbl&gt; | Strain.heterogeneity &lt;dbl&gt; | Genome.size..bp. &lt;int&gt; | X..ambiguous.bases &lt;int&gt; | ⋯ ⋯ | X0 &lt;int&gt; | X1 &lt;int&gt; | X2 &lt;int&gt; | X3 &lt;int&gt; | X4 &lt;int&gt; | X5. &lt;int&gt; | assembly.Id &lt;fct&gt; | assembler.Id &lt;fct&gt; | taxon.Id &lt;lgl&gt; | File &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| GCA_000398925.1_Buty_pull_1_20_V1_genomic | o__Clostridiales (UID1212) | 172 | 263 | 149 | 98.66 | 0 | 0 | 3325721 | 26204 | ⋯ |  2 | 261 | 0 | 0 | 0 | 0 | |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Butyricicoccus|LLG_output|checkM|1|checkm|markers_qa_summary.tsv.1 | markers_qa_summary.tsv.1 | NA | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricicoccus/LLG_output/checkM/1/checkm/markers_qa_summary.tsv.1 |\n",
       "| GCA_001643435.1_ASM164343v1_genomic       | o__Clostridiales (UID1212) | 172 | 263 | 149 | 67.11 | 0 | 0 | 1452273 |  1058 | ⋯ | 78 | 185 | 0 | 0 | 0 | 0 | |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Butyricicoccus|LLG_output|checkM|1|checkm|markers_qa_summary.tsv.2 | markers_qa_summary.tsv.2 | NA | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricicoccus/LLG_output/checkM/1/checkm/markers_qa_summary.tsv.2 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋱ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| GCA_900143195.1_501571.5_genomic        | o__Clostridiales (UID1212) | 172 | 263 | 149 | 98.66 | 0 | 0 | 3284487 |  5872 | ⋯ | 2 | 261 | 0 | 0 | 0 | 0 | |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Butyricicoccus|LLG_output|checkM|2|checkm|markers_qa_summary.tsv.17 | markers_qa_summary.tsv.17 | NA | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricicoccus/LLG_output/checkM/2/checkm/markers_qa_summary.tsv.17 |\n",
       "| GCA_902374965.1_MGYG-HGUT-01434_genomic | o__Clostridiales (UID1212) | 172 | 263 | 149 | 98.66 | 0 | 0 | 3325721 | 26204 | ⋯ | 2 | 261 | 0 | 0 | 0 | 0 | |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Butyricicoccus|LLG_output|checkM|2|checkm|markers_qa_summary.tsv.18 | markers_qa_summary.tsv.18 | NA | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricicoccus/LLG_output/checkM/2/checkm/markers_qa_summary.tsv.18 |\n",
       "\n"
      ],
      "text/plain": [
       "   Bin.Id                                    Marker.lineage            \n",
       "1  GCA_000398925.1_Buty_pull_1_20_V1_genomic o__Clostridiales (UID1212)\n",
       "2  GCA_001643435.1_ASM164343v1_genomic       o__Clostridiales (UID1212)\n",
       "⋮  ⋮                                         ⋮                         \n",
       "35 GCA_900143195.1_501571.5_genomic          o__Clostridiales (UID1212)\n",
       "36 GCA_902374965.1_MGYG-HGUT-01434_genomic   o__Clostridiales (UID1212)\n",
       "   X..genomes X..markers X..marker.sets Completeness Contamination\n",
       "1  172        263        149            98.66        0            \n",
       "2  172        263        149            67.11        0            \n",
       "⋮  ⋮          ⋮          ⋮              ⋮            ⋮            \n",
       "35 172        263        149            98.66        0            \n",
       "36 172        263        149            98.66        0            \n",
       "   Strain.heterogeneity Genome.size..bp. X..ambiguous.bases ⋯ X0 X1  X2 X3 X4\n",
       "1  0                    3325721          26204              ⋯  2 261 0  0  0 \n",
       "2  0                    1452273           1058              ⋯ 78 185 0  0  0 \n",
       "⋮  ⋮                    ⋮                ⋮                  ⋱ ⋮  ⋮   ⋮  ⋮  ⋮ \n",
       "35 0                    3284487           5872              ⋯ 2  261 0  0  0 \n",
       "36 0                    3325721          26204              ⋯ 2  261 0  0  0 \n",
       "   X5.\n",
       "1  0  \n",
       "2  0  \n",
       "⋮  ⋮  \n",
       "35 0  \n",
       "36 0  \n",
       "   assembly.Id                                                                                                                                           \n",
       "1  |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Butyricicoccus|LLG_output|checkM|1|checkm|markers_qa_summary.tsv.1 \n",
       "2  |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Butyricicoccus|LLG_output|checkM|1|checkm|markers_qa_summary.tsv.2 \n",
       "⋮  ⋮                                                                                                                                                     \n",
       "35 |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Butyricicoccus|LLG_output|checkM|2|checkm|markers_qa_summary.tsv.17\n",
       "36 |ebio|abt3_projects|software|dev|ll_pipelines|llprimer|experiments|HMP_most-wanted|Butyricicoccus|LLG_output|checkM|2|checkm|markers_qa_summary.tsv.18\n",
       "   assembler.Id              taxon.Id\n",
       "1  markers_qa_summary.tsv.1  NA      \n",
       "2  markers_qa_summary.tsv.2  NA      \n",
       "⋮  ⋮                         ⋮       \n",
       "35 markers_qa_summary.tsv.17 NA      \n",
       "36 markers_qa_summary.tsv.18 NA      \n",
       "   File                                                                                                                                                  \n",
       "1  /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricicoccus/LLG_output/checkM/1/checkm/markers_qa_summary.tsv.1 \n",
       "2  /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricicoccus/LLG_output/checkM/1/checkm/markers_qa_summary.tsv.2 \n",
       "⋮  ⋮                                                                                                                                                     \n",
       "35 /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricicoccus/LLG_output/checkM/2/checkm/markers_qa_summary.tsv.17\n",
       "36 /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricicoccus/LLG_output/checkM/2/checkm/markers_qa_summary.tsv.18"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# checkM summary\n",
    "checkm = file.path(work_dir, 'LLG', 'checkM', 'checkm_qa_summary.tsv') %>%\n",
    "    read.delim(sep='\\t') \n",
    "checkm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-28T13:28:33.206259Z",
     "iopub.status.busy": "2021-04-28T13:28:33.203835Z",
     "iopub.status.idle": "2021-04-28T13:28:33.241575Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 36 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>genome</th><th scope=col>completeness</th><th scope=col>contamination</th><th scope=col>Bin.Id</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000398925.1_Buty_pull_1_20_V1_genomic.fna</td><td>98.66</td><td>0</td><td>GCA_000398925.1_Buty_pull_1_20_V1_genomic</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_001643435.1_ASM164343v1_genomic.fna      </td><td>67.11</td><td>0</td><td>GCA_001643435.1_ASM164343v1_genomic      </td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_900143195.1_501571.5_genomic.fna       </td><td>98.66</td><td>0</td><td>GCA_900143195.1_501571.5_genomic       </td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_902374965.1_MGYG-HGUT-01434_genomic.fna</td><td>98.66</td><td>0</td><td>GCA_902374965.1_MGYG-HGUT-01434_genomic</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 36 × 4\n",
       "\\begin{tabular}{llll}\n",
       " genome & completeness & contamination & Bin.Id\\\\\n",
       " <fct> & <dbl> & <dbl> & <chr>\\\\\n",
       "\\hline\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_000398925.1\\_Buty\\_pull\\_1\\_20\\_V1\\_genomic.fna & 98.66 & 0 & GCA\\_000398925.1\\_Buty\\_pull\\_1\\_20\\_V1\\_genomic\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_001643435.1\\_ASM164343v1\\_genomic.fna       & 67.11 & 0 & GCA\\_001643435.1\\_ASM164343v1\\_genomic      \\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_900143195.1\\_501571.5\\_genomic.fna        & 98.66 & 0 & GCA\\_900143195.1\\_501571.5\\_genomic       \\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_902374965.1\\_MGYG-HGUT-01434\\_genomic.fna & 98.66 & 0 & GCA\\_902374965.1\\_MGYG-HGUT-01434\\_genomic\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 36 × 4\n",
       "\n",
       "| genome &lt;fct&gt; | completeness &lt;dbl&gt; | contamination &lt;dbl&gt; | Bin.Id &lt;chr&gt; |\n",
       "|---|---|---|---|\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000398925.1_Buty_pull_1_20_V1_genomic.fna | 98.66 | 0 | GCA_000398925.1_Buty_pull_1_20_V1_genomic |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_001643435.1_ASM164343v1_genomic.fna       | 67.11 | 0 | GCA_001643435.1_ASM164343v1_genomic       |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_900143195.1_501571.5_genomic.fna        | 98.66 | 0 | GCA_900143195.1_501571.5_genomic        |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_902374965.1_MGYG-HGUT-01434_genomic.fna | 98.66 | 0 | GCA_902374965.1_MGYG-HGUT-01434_genomic |\n",
       "\n"
      ],
      "text/plain": [
       "   genome                                                                                             \n",
       "1  /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000398925.1_Buty_pull_1_20_V1_genomic.fna\n",
       "2  /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_001643435.1_ASM164343v1_genomic.fna      \n",
       "⋮  ⋮                                                                                                  \n",
       "35 /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_900143195.1_501571.5_genomic.fna         \n",
       "36 /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_902374965.1_MGYG-HGUT-01434_genomic.fna  \n",
       "   completeness contamination Bin.Id                                   \n",
       "1  98.66        0             GCA_000398925.1_Buty_pull_1_20_V1_genomic\n",
       "2  67.11        0             GCA_001643435.1_ASM164343v1_genomic      \n",
       "⋮  ⋮            ⋮             ⋮                                        \n",
       "35 98.66        0             GCA_900143195.1_501571.5_genomic         \n",
       "36 98.66        0             GCA_902374965.1_MGYG-HGUT-01434_genomic  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# dRep summary\n",
    "drep = file.path(work_dir, 'LLG', 'drep', 'checkm_markers_qa_summary.tsv') %>%\n",
    "    read.delim(sep='\\t') %>%\n",
    "    mutate(Bin.Id = gsub('.+/', '', genome),\n",
    "           Bin.Id = gsub('\\\\.fna$', '', Bin.Id))\n",
    "drep"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-28T13:28:33.382271Z",
     "iopub.status.busy": "2021-04-28T13:28:33.380074Z",
     "iopub.status.idle": "2021-04-28T13:28:33.411374Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 26 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Name</th><th scope=col>Fasta</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>GCA_002157465.1_ASM215746v1_genomic </td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricicoccus/LLG_output/drep/drep/dereplicated_genomes/GCA_002157465.1_ASM215746v1_genomic.fna</td></tr>\n",
       "\t<tr><td>GCA_016297755.1_ASM1629775v1_genomic</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricicoccus/LLG/drep/drep/dereplicated_genomes/GCA_016297755.1_ASM1629775v1_genomic.fna </td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>GCA_003202935.1_ASM320293v1_genomic</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricicoccus/LLG/drep/drep/dereplicated_genomes/GCA_003202935.1_ASM320293v1_genomic.fna</td></tr>\n",
       "\t<tr><td>GCA_009881305.1_ASM988130v1_genomic</td><td>/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricicoccus/LLG/drep/drep/dereplicated_genomes/GCA_009881305.1_ASM988130v1_genomic.fna</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 26 × 2\n",
       "\\begin{tabular}{ll}\n",
       " Name & Fasta\\\\\n",
       " <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\t GCA\\_002157465.1\\_ASM215746v1\\_genomic  & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/Butyricicoccus/LLG\\_output/drep/drep/dereplicated\\_genomes/GCA\\_002157465.1\\_ASM215746v1\\_genomic.fna\\\\\n",
       "\t GCA\\_016297755.1\\_ASM1629775v1\\_genomic & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/v0.3/Butyricicoccus/LLG/drep/drep/dereplicated\\_genomes/GCA\\_016297755.1\\_ASM1629775v1\\_genomic.fna \\\\\n",
       "\t ⋮ & ⋮\\\\\n",
       "\t GCA\\_003202935.1\\_ASM320293v1\\_genomic & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/v0.3/Butyricicoccus/LLG/drep/drep/dereplicated\\_genomes/GCA\\_003202935.1\\_ASM320293v1\\_genomic.fna\\\\\n",
       "\t GCA\\_009881305.1\\_ASM988130v1\\_genomic & /ebio/abt3\\_projects/software/dev/ll\\_pipelines/llprimer/experiments/HMP\\_most-wanted/v0.3/Butyricicoccus/LLG/drep/drep/dereplicated\\_genomes/GCA\\_009881305.1\\_ASM988130v1\\_genomic.fna\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 26 × 2\n",
       "\n",
       "| Name &lt;fct&gt; | Fasta &lt;fct&gt; |\n",
       "|---|---|\n",
       "| GCA_002157465.1_ASM215746v1_genomic  | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricicoccus/LLG_output/drep/drep/dereplicated_genomes/GCA_002157465.1_ASM215746v1_genomic.fna |\n",
       "| GCA_016297755.1_ASM1629775v1_genomic | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricicoccus/LLG/drep/drep/dereplicated_genomes/GCA_016297755.1_ASM1629775v1_genomic.fna  |\n",
       "| ⋮ | ⋮ |\n",
       "| GCA_003202935.1_ASM320293v1_genomic | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricicoccus/LLG/drep/drep/dereplicated_genomes/GCA_003202935.1_ASM320293v1_genomic.fna |\n",
       "| GCA_009881305.1_ASM988130v1_genomic | /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricicoccus/LLG/drep/drep/dereplicated_genomes/GCA_009881305.1_ASM988130v1_genomic.fna |\n",
       "\n"
      ],
      "text/plain": [
       "   Name                                \n",
       "1  GCA_002157465.1_ASM215746v1_genomic \n",
       "2  GCA_016297755.1_ASM1629775v1_genomic\n",
       "⋮  ⋮                                   \n",
       "25 GCA_003202935.1_ASM320293v1_genomic \n",
       "26 GCA_009881305.1_ASM988130v1_genomic \n",
       "   Fasta                                                                                                                                                                              \n",
       "1  /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/Butyricicoccus/LLG_output/drep/drep/dereplicated_genomes/GCA_002157465.1_ASM215746v1_genomic.fna\n",
       "2  /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricicoccus/LLG/drep/drep/dereplicated_genomes/GCA_016297755.1_ASM1629775v1_genomic.fna \n",
       "⋮  ⋮                                                                                                                                                                                  \n",
       "25 /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricicoccus/LLG/drep/drep/dereplicated_genomes/GCA_003202935.1_ASM320293v1_genomic.fna  \n",
       "26 /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricicoccus/LLG/drep/drep/dereplicated_genomes/GCA_009881305.1_ASM988130v1_genomic.fna  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# de-replicated genomes\n",
    "drep_gen = file.path(work_dir, 'LLG', 'drep', 'dereplicated_genomes.tsv') %>%\n",
    "    read.delim(sep='\\t')\n",
    "drep_gen"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-28T13:28:33.522081Z",
     "iopub.status.busy": "2021-04-28T13:28:33.520349Z",
     "iopub.status.idle": "2021-04-28T13:28:33.580795Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 36 × 22</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>user_genome</th><th scope=col>Domain</th><th scope=col>Phylum</th><th scope=col>Class</th><th scope=col>Order</th><th scope=col>Family</th><th scope=col>Genus</th><th scope=col>Species</th><th scope=col>fastani_reference</th><th scope=col>fastani_reference_radius</th><th scope=col>⋯</th><th scope=col>fastani_af</th><th scope=col>closest_placement_reference</th><th scope=col>closest_placement_radius</th><th scope=col>closest_placement_taxonomy</th><th scope=col>closest_placement_ani</th><th scope=col>closest_placement_af</th><th scope=col>msa_percent</th><th scope=col>translation_table</th><th scope=col>red_value</th><th scope=col>warnings</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>⋯</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>GCA_000398925.1_Buty_pull_1_20_V1_genomic</td><td>d__Bacteria</td><td>p__Firmicutes_A</td><td>c__Clostridia</td><td>o__Oscillospirales</td><td>f__Butyricicoccaceae</td><td>g__Butyricicoccus</td><td>s__Butyricicoccus pullicaecorum   </td><td>GCF_900167005.1</td><td>95.0</td><td>⋯</td><td>1.0</td><td>GCF_900167005.1</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Butyricicoccaceae;g__Butyricicoccus;s__Butyricicoccus pullicaecorum  </td><td>99.99</td><td>1.0</td><td>97.26</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>GCA_001643435.1_ASM164343v1_genomic      </td><td>d__Bacteria</td><td>p__Firmicutes_A</td><td>c__Clostridia</td><td>o__Oscillospirales</td><td>f__Butyricicoccaceae</td><td>g__Agathobaculum </td><td>s__Agathobaculum butyriciproducens</td><td>GCF_003096535.1</td><td>95.0</td><td>⋯</td><td>0.9</td><td>GCF_003096535.1</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Butyricicoccaceae;g__Agathobaculum;s__Agathobaculum butyriciproducens</td><td>97.16</td><td>0.9</td><td>69.94</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋱</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>GCA_900143195.1_501571.5_genomic       </td><td>d__Bacteria</td><td>p__Firmicutes_A</td><td>c__Clostridia</td><td>o__Oscillospirales</td><td>f__Butyricicoccaceae</td><td>g__Butyricicoccus</td><td>s__Butyricicoccus pullicaecorum</td><td>GCF_900167005.1</td><td>95.0</td><td>⋯</td><td>0.99</td><td>GCF_900167005.1</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Butyricicoccaceae;g__Butyricicoccus;s__Butyricicoccus pullicaecorum</td><td>99.97</td><td>0.99</td><td>97.14</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>GCA_902374965.1_MGYG-HGUT-01434_genomic</td><td>d__Bacteria</td><td>p__Firmicutes_A</td><td>c__Clostridia</td><td>o__Oscillospirales</td><td>f__Butyricicoccaceae</td><td>g__Butyricicoccus</td><td>s__Butyricicoccus pullicaecorum</td><td>GCF_900167005.1</td><td>95.0</td><td>⋯</td><td>1.0 </td><td>GCF_900167005.1</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Butyricicoccaceae;g__Butyricicoccus;s__Butyricicoccus pullicaecorum</td><td>99.99</td><td>1.0 </td><td>97.26</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 36 × 22\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " user\\_genome & Domain & Phylum & Class & Order & Family & Genus & Species & fastani\\_reference & fastani\\_reference\\_radius & ⋯ & fastani\\_af & closest\\_placement\\_reference & closest\\_placement\\_radius & closest\\_placement\\_taxonomy & closest\\_placement\\_ani & closest\\_placement\\_af & msa\\_percent & translation\\_table & red\\_value & warnings\\\\\n",
       " <fct> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <fct> & <fct> & ⋯ & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <dbl> & <int> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\t GCA\\_000398925.1\\_Buty\\_pull\\_1\\_20\\_V1\\_genomic & d\\_\\_Bacteria & p\\_\\_Firmicutes\\_A & c\\_\\_Clostridia & o\\_\\_Oscillospirales & f\\_\\_Butyricicoccaceae & g\\_\\_Butyricicoccus & s\\_\\_Butyricicoccus pullicaecorum    & GCF\\_900167005.1 & 95.0 & ⋯ & 1.0 & GCF\\_900167005.1 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Butyricicoccaceae;g\\_\\_Butyricicoccus;s\\_\\_Butyricicoccus pullicaecorum   & 99.99 & 1.0 & 97.26 & 11 & N/A & N/A\\\\\n",
       "\t GCA\\_001643435.1\\_ASM164343v1\\_genomic       & d\\_\\_Bacteria & p\\_\\_Firmicutes\\_A & c\\_\\_Clostridia & o\\_\\_Oscillospirales & f\\_\\_Butyricicoccaceae & g\\_\\_Agathobaculum  & s\\_\\_Agathobaculum butyriciproducens & GCF\\_003096535.1 & 95.0 & ⋯ & 0.9 & GCF\\_003096535.1 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Butyricicoccaceae;g\\_\\_Agathobaculum;s\\_\\_Agathobaculum butyriciproducens & 97.16 & 0.9 & 69.94 & 11 & N/A & N/A\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋱ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t GCA\\_900143195.1\\_501571.5\\_genomic        & d\\_\\_Bacteria & p\\_\\_Firmicutes\\_A & c\\_\\_Clostridia & o\\_\\_Oscillospirales & f\\_\\_Butyricicoccaceae & g\\_\\_Butyricicoccus & s\\_\\_Butyricicoccus pullicaecorum & GCF\\_900167005.1 & 95.0 & ⋯ & 0.99 & GCF\\_900167005.1 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Butyricicoccaceae;g\\_\\_Butyricicoccus;s\\_\\_Butyricicoccus pullicaecorum & 99.97 & 0.99 & 97.14 & 11 & N/A & N/A\\\\\n",
       "\t GCA\\_902374965.1\\_MGYG-HGUT-01434\\_genomic & d\\_\\_Bacteria & p\\_\\_Firmicutes\\_A & c\\_\\_Clostridia & o\\_\\_Oscillospirales & f\\_\\_Butyricicoccaceae & g\\_\\_Butyricicoccus & s\\_\\_Butyricicoccus pullicaecorum & GCF\\_900167005.1 & 95.0 & ⋯ & 1.0  & GCF\\_900167005.1 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Butyricicoccaceae;g\\_\\_Butyricicoccus;s\\_\\_Butyricicoccus pullicaecorum & 99.99 & 1.0  & 97.26 & 11 & N/A & N/A\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 36 × 22\n",
       "\n",
       "| user_genome &lt;fct&gt; | Domain &lt;chr&gt; | Phylum &lt;chr&gt; | Class &lt;chr&gt; | Order &lt;chr&gt; | Family &lt;chr&gt; | Genus &lt;chr&gt; | Species &lt;chr&gt; | fastani_reference &lt;fct&gt; | fastani_reference_radius &lt;fct&gt; | ⋯ ⋯ | fastani_af &lt;fct&gt; | closest_placement_reference &lt;fct&gt; | closest_placement_radius &lt;fct&gt; | closest_placement_taxonomy &lt;fct&gt; | closest_placement_ani &lt;fct&gt; | closest_placement_af &lt;fct&gt; | msa_percent &lt;dbl&gt; | translation_table &lt;int&gt; | red_value &lt;fct&gt; | warnings &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| GCA_000398925.1_Buty_pull_1_20_V1_genomic | d__Bacteria | p__Firmicutes_A | c__Clostridia | o__Oscillospirales | f__Butyricicoccaceae | g__Butyricicoccus | s__Butyricicoccus pullicaecorum    | GCF_900167005.1 | 95.0 | ⋯ | 1.0 | GCF_900167005.1 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Butyricicoccaceae;g__Butyricicoccus;s__Butyricicoccus pullicaecorum   | 99.99 | 1.0 | 97.26 | 11 | N/A | N/A |\n",
       "| GCA_001643435.1_ASM164343v1_genomic       | d__Bacteria | p__Firmicutes_A | c__Clostridia | o__Oscillospirales | f__Butyricicoccaceae | g__Agathobaculum  | s__Agathobaculum butyriciproducens | GCF_003096535.1 | 95.0 | ⋯ | 0.9 | GCF_003096535.1 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Butyricicoccaceae;g__Agathobaculum;s__Agathobaculum butyriciproducens | 97.16 | 0.9 | 69.94 | 11 | N/A | N/A |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋱ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| GCA_900143195.1_501571.5_genomic        | d__Bacteria | p__Firmicutes_A | c__Clostridia | o__Oscillospirales | f__Butyricicoccaceae | g__Butyricicoccus | s__Butyricicoccus pullicaecorum | GCF_900167005.1 | 95.0 | ⋯ | 0.99 | GCF_900167005.1 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Butyricicoccaceae;g__Butyricicoccus;s__Butyricicoccus pullicaecorum | 99.97 | 0.99 | 97.14 | 11 | N/A | N/A |\n",
       "| GCA_902374965.1_MGYG-HGUT-01434_genomic | d__Bacteria | p__Firmicutes_A | c__Clostridia | o__Oscillospirales | f__Butyricicoccaceae | g__Butyricicoccus | s__Butyricicoccus pullicaecorum | GCF_900167005.1 | 95.0 | ⋯ | 1.0  | GCF_900167005.1 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Butyricicoccaceae;g__Butyricicoccus;s__Butyricicoccus pullicaecorum | 99.99 | 1.0  | 97.26 | 11 | N/A | N/A |\n",
       "\n"
      ],
      "text/plain": [
       "   user_genome                               Domain      Phylum         \n",
       "1  GCA_000398925.1_Buty_pull_1_20_V1_genomic d__Bacteria p__Firmicutes_A\n",
       "2  GCA_001643435.1_ASM164343v1_genomic       d__Bacteria p__Firmicutes_A\n",
       "⋮  ⋮                                         ⋮           ⋮              \n",
       "35 GCA_900143195.1_501571.5_genomic          d__Bacteria p__Firmicutes_A\n",
       "36 GCA_902374965.1_MGYG-HGUT-01434_genomic   d__Bacteria p__Firmicutes_A\n",
       "   Class         Order              Family               Genus            \n",
       "1  c__Clostridia o__Oscillospirales f__Butyricicoccaceae g__Butyricicoccus\n",
       "2  c__Clostridia o__Oscillospirales f__Butyricicoccaceae g__Agathobaculum \n",
       "⋮  ⋮             ⋮                  ⋮                    ⋮                \n",
       "35 c__Clostridia o__Oscillospirales f__Butyricicoccaceae g__Butyricicoccus\n",
       "36 c__Clostridia o__Oscillospirales f__Butyricicoccaceae g__Butyricicoccus\n",
       "   Species                            fastani_reference\n",
       "1  s__Butyricicoccus pullicaecorum    GCF_900167005.1  \n",
       "2  s__Agathobaculum butyriciproducens GCF_003096535.1  \n",
       "⋮  ⋮                                  ⋮                \n",
       "35 s__Butyricicoccus pullicaecorum    GCF_900167005.1  \n",
       "36 s__Butyricicoccus pullicaecorum    GCF_900167005.1  \n",
       "   fastani_reference_radius ⋯ fastani_af closest_placement_reference\n",
       "1  95.0                     ⋯ 1.0        GCF_900167005.1            \n",
       "2  95.0                     ⋯ 0.9        GCF_003096535.1            \n",
       "⋮  ⋮                        ⋱ ⋮          ⋮                          \n",
       "35 95.0                     ⋯ 0.99       GCF_900167005.1            \n",
       "36 95.0                     ⋯ 1.0        GCF_900167005.1            \n",
       "   closest_placement_radius\n",
       "1  95.0                    \n",
       "2  95.0                    \n",
       "⋮  ⋮                       \n",
       "35 95.0                    \n",
       "36 95.0                    \n",
       "   closest_placement_taxonomy                                                                                                           \n",
       "1  d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Butyricicoccaceae;g__Butyricicoccus;s__Butyricicoccus pullicaecorum  \n",
       "2  d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Butyricicoccaceae;g__Agathobaculum;s__Agathobaculum butyriciproducens\n",
       "⋮  ⋮                                                                                                                                    \n",
       "35 d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Butyricicoccaceae;g__Butyricicoccus;s__Butyricicoccus pullicaecorum  \n",
       "36 d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Butyricicoccaceae;g__Butyricicoccus;s__Butyricicoccus pullicaecorum  \n",
       "   closest_placement_ani closest_placement_af msa_percent translation_table\n",
       "1  99.99                 1.0                  97.26       11               \n",
       "2  97.16                 0.9                  69.94       11               \n",
       "⋮  ⋮                     ⋮                    ⋮           ⋮                \n",
       "35 99.97                 0.99                 97.14       11               \n",
       "36 99.99                 1.0                  97.26       11               \n",
       "   red_value warnings\n",
       "1  N/A       N/A     \n",
       "2  N/A       N/A     \n",
       "⋮  ⋮         ⋮       \n",
       "35 N/A       N/A     \n",
       "36 N/A       N/A     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# GTDBTk summary\n",
    "tax = file.path(work_dir, 'LLG', 'gtdbtk', 'gtdbtk_bac_summary.tsv') %>%\n",
    "    read.delim(, sep='\\t') %>%\n",
    "    separate(classification, \n",
    "             c('Domain', 'Phylum', 'Class', 'Order', 'Family', 'Genus', 'Species'),\n",
    "             sep=';') %>%\n",
    "    select(-note, -classification_method, -pplacer_taxonomy,\n",
    "           -other_related_references.genome_id.species_name.radius.ANI.AF.)\n",
    "tax"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-28T13:28:33.646126Z",
     "iopub.status.busy": "2021-04-28T13:28:33.643995Z",
     "iopub.status.idle": "2021-04-28T13:28:33.683123Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- drep --\n",
      "intersect(x,y): 26 \n",
      "setdiff(x,y): 0 \n",
      "setdiff(y,x): 10 \n",
      "union(x,y): 36 \n",
      "-- checkm --\n",
      "intersect(x,y): 36 \n",
      "setdiff(x,y): 0 \n",
      "setdiff(y,x): 0 \n",
      "union(x,y): 36 \n",
      "-- gtdbtk --\n",
      "intersect(x,y): 36 \n",
      "setdiff(x,y): 0 \n",
      "setdiff(y,x): 0 \n",
      "union(x,y): 36 \n"
     ]
    }
   ],
   "source": [
    "# checking overlap\n",
    "cat('-- drep --\\n')\n",
    "overlap(basename(as.character(drep_gen$Fasta)), \n",
    "        basename(as.character(drep$genome)))\n",
    "cat('-- checkm --\\n')\n",
    "overlap(drep$Bin.Id, checkm$Bin.Id)\n",
    "cat('-- gtdbtk --\\n')\n",
    "overlap(drep$Bin.Id, tax$user_genome)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-28T13:28:33.779966Z",
     "iopub.status.busy": "2021-04-28T13:28:33.774420Z",
     "iopub.status.idle": "2021-04-28T13:28:33.881406Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 26 × 60</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>genome</th><th scope=col>completeness</th><th scope=col>contamination</th><th scope=col>Bin.Id</th><th scope=col>Marker.lineage</th><th scope=col>X..genomes</th><th scope=col>X..markers</th><th scope=col>X..marker.sets</th><th scope=col>Completeness</th><th scope=col>Contamination</th><th scope=col>⋯</th><th scope=col>fastani_af</th><th scope=col>closest_placement_reference</th><th scope=col>closest_placement_radius</th><th scope=col>closest_placement_taxonomy</th><th scope=col>closest_placement_ani</th><th scope=col>closest_placement_af</th><th scope=col>msa_percent</th><th scope=col>translation_table</th><th scope=col>red_value</th><th scope=col>warnings</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000398925.1_Buty_pull_1_20_V1_genomic.fna</td><td>98.66</td><td>0.00</td><td>GCA_000398925.1_Buty_pull_1_20_V1_genomic</td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>98.66</td><td>0.00</td><td>⋯</td><td>1.0 </td><td>GCF_900167005.1</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Butyricicoccaceae;g__Butyricicoccus;s__Butyricicoccus pullicaecorum</td><td>99.99</td><td>1.0 </td><td>97.26</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_002157465.1_ASM215746v1_genomic.fna      </td><td>98.66</td><td>2.68</td><td>GCA_002157465.1_ASM215746v1_genomic      </td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>98.66</td><td>2.68</td><td>⋯</td><td>0.99</td><td>GCF_002157465.1</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Butyricicoccaceae;g__Butyricicoccus_A;s__Butyricicoccus_A porcorum </td><td>100.0</td><td>0.99</td><td>96.27</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋱</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_014648015.1_ASM1464801v1_genomic.fna</td><td>99.33</td><td>0.67</td><td>GCA_014648015.1_ASM1464801v1_genomic</td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>99.33</td><td>0.67</td><td>⋯</td><td>0.83</td><td>GCF_003096535.1</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Butyricicoccaceae;g__Agathobaculum;s__Agathobaculum butyriciproducens</td><td>95.65</td><td>0.83</td><td>97.10</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_016293105.1_ASM1629310v1_genomic.fna</td><td>92.88</td><td>0.67</td><td>GCA_016293105.1_ASM1629310v1_genomic</td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>92.88</td><td>0.67</td><td>⋯</td><td>0.82</td><td>GCA_002395695.1</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Butyricicoccaceae;g__Butyricicoccus_A;s__Butyricicoccus_A sp002395695</td><td>98.29</td><td>0.82</td><td>75.63</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 26 × 60\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " genome & completeness & contamination & Bin.Id & Marker.lineage & X..genomes & X..markers & X..marker.sets & Completeness & Contamination & ⋯ & fastani\\_af & closest\\_placement\\_reference & closest\\_placement\\_radius & closest\\_placement\\_taxonomy & closest\\_placement\\_ani & closest\\_placement\\_af & msa\\_percent & translation\\_table & red\\_value & warnings\\\\\n",
       " <fct> & <dbl> & <dbl> & <chr> & <fct> & <int> & <int> & <int> & <dbl> & <dbl> & ⋯ & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <dbl> & <int> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_000398925.1\\_Buty\\_pull\\_1\\_20\\_V1\\_genomic.fna & 98.66 & 0.00 & GCA\\_000398925.1\\_Buty\\_pull\\_1\\_20\\_V1\\_genomic & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 98.66 & 0.00 & ⋯ & 1.0  & GCF\\_900167005.1 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Butyricicoccaceae;g\\_\\_Butyricicoccus;s\\_\\_Butyricicoccus pullicaecorum & 99.99 & 1.0  & 97.26 & 11 & N/A & N/A\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_002157465.1\\_ASM215746v1\\_genomic.fna       & 98.66 & 2.68 & GCA\\_002157465.1\\_ASM215746v1\\_genomic       & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 98.66 & 2.68 & ⋯ & 0.99 & GCF\\_002157465.1 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Butyricicoccaceae;g\\_\\_Butyricicoccus\\_A;s\\_\\_Butyricicoccus\\_A porcorum  & 100.0 & 0.99 & 96.27 & 11 & N/A & N/A\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋱ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_014648015.1\\_ASM1464801v1\\_genomic.fna & 99.33 & 0.67 & GCA\\_014648015.1\\_ASM1464801v1\\_genomic & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 99.33 & 0.67 & ⋯ & 0.83 & GCF\\_003096535.1 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Butyricicoccaceae;g\\_\\_Agathobaculum;s\\_\\_Agathobaculum butyriciproducens & 95.65 & 0.83 & 97.10 & 11 & N/A & N/A\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_016293105.1\\_ASM1629310v1\\_genomic.fna & 92.88 & 0.67 & GCA\\_016293105.1\\_ASM1629310v1\\_genomic & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 92.88 & 0.67 & ⋯ & 0.82 & GCA\\_002395695.1 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Butyricicoccaceae;g\\_\\_Butyricicoccus\\_A;s\\_\\_Butyricicoccus\\_A sp002395695 & 98.29 & 0.82 & 75.63 & 11 & N/A & N/A\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 26 × 60\n",
       "\n",
       "| genome &lt;fct&gt; | completeness &lt;dbl&gt; | contamination &lt;dbl&gt; | Bin.Id &lt;chr&gt; | Marker.lineage &lt;fct&gt; | X..genomes &lt;int&gt; | X..markers &lt;int&gt; | X..marker.sets &lt;int&gt; | Completeness &lt;dbl&gt; | Contamination &lt;dbl&gt; | ⋯ ⋯ | fastani_af &lt;fct&gt; | closest_placement_reference &lt;fct&gt; | closest_placement_radius &lt;fct&gt; | closest_placement_taxonomy &lt;fct&gt; | closest_placement_ani &lt;fct&gt; | closest_placement_af &lt;fct&gt; | msa_percent &lt;dbl&gt; | translation_table &lt;int&gt; | red_value &lt;fct&gt; | warnings &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000398925.1_Buty_pull_1_20_V1_genomic.fna | 98.66 | 0.00 | GCA_000398925.1_Buty_pull_1_20_V1_genomic | o__Clostridiales (UID1212) | 172 | 263 | 149 | 98.66 | 0.00 | ⋯ | 1.0  | GCF_900167005.1 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Butyricicoccaceae;g__Butyricicoccus;s__Butyricicoccus pullicaecorum | 99.99 | 1.0  | 97.26 | 11 | N/A | N/A |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_002157465.1_ASM215746v1_genomic.fna       | 98.66 | 2.68 | GCA_002157465.1_ASM215746v1_genomic       | o__Clostridiales (UID1212) | 172 | 263 | 149 | 98.66 | 2.68 | ⋯ | 0.99 | GCF_002157465.1 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Butyricicoccaceae;g__Butyricicoccus_A;s__Butyricicoccus_A porcorum  | 100.0 | 0.99 | 96.27 | 11 | N/A | N/A |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋱ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_014648015.1_ASM1464801v1_genomic.fna | 99.33 | 0.67 | GCA_014648015.1_ASM1464801v1_genomic | o__Clostridiales (UID1212) | 172 | 263 | 149 | 99.33 | 0.67 | ⋯ | 0.83 | GCF_003096535.1 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Butyricicoccaceae;g__Agathobaculum;s__Agathobaculum butyriciproducens | 95.65 | 0.83 | 97.10 | 11 | N/A | N/A |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_016293105.1_ASM1629310v1_genomic.fna | 92.88 | 0.67 | GCA_016293105.1_ASM1629310v1_genomic | o__Clostridiales (UID1212) | 172 | 263 | 149 | 92.88 | 0.67 | ⋯ | 0.82 | GCA_002395695.1 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Butyricicoccaceae;g__Butyricicoccus_A;s__Butyricicoccus_A sp002395695 | 98.29 | 0.82 | 75.63 | 11 | N/A | N/A |\n",
       "\n"
      ],
      "text/plain": [
       "   genome                                                                                             \n",
       "1  /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000398925.1_Buty_pull_1_20_V1_genomic.fna\n",
       "2  /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_002157465.1_ASM215746v1_genomic.fna      \n",
       "⋮  ⋮                                                                                                  \n",
       "25 /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_014648015.1_ASM1464801v1_genomic.fna     \n",
       "26 /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_016293105.1_ASM1629310v1_genomic.fna     \n",
       "   completeness contamination Bin.Id                                   \n",
       "1  98.66        0.00          GCA_000398925.1_Buty_pull_1_20_V1_genomic\n",
       "2  98.66        2.68          GCA_002157465.1_ASM215746v1_genomic      \n",
       "⋮  ⋮            ⋮             ⋮                                        \n",
       "25 99.33        0.67          GCA_014648015.1_ASM1464801v1_genomic     \n",
       "26 92.88        0.67          GCA_016293105.1_ASM1629310v1_genomic     \n",
       "   Marker.lineage             X..genomes X..markers X..marker.sets Completeness\n",
       "1  o__Clostridiales (UID1212) 172        263        149            98.66       \n",
       "2  o__Clostridiales (UID1212) 172        263        149            98.66       \n",
       "⋮  ⋮                          ⋮          ⋮          ⋮              ⋮           \n",
       "25 o__Clostridiales (UID1212) 172        263        149            99.33       \n",
       "26 o__Clostridiales (UID1212) 172        263        149            92.88       \n",
       "   Contamination ⋯ fastani_af closest_placement_reference\n",
       "1  0.00          ⋯ 1.0        GCF_900167005.1            \n",
       "2  2.68          ⋯ 0.99       GCF_002157465.1            \n",
       "⋮  ⋮             ⋱ ⋮          ⋮                          \n",
       "25 0.67          ⋯ 0.83       GCF_003096535.1            \n",
       "26 0.67          ⋯ 0.82       GCA_002395695.1            \n",
       "   closest_placement_radius\n",
       "1  95.0                    \n",
       "2  95.0                    \n",
       "⋮  ⋮                       \n",
       "25 95.0                    \n",
       "26 95.0                    \n",
       "   closest_placement_taxonomy                                                                                                           \n",
       "1  d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Butyricicoccaceae;g__Butyricicoccus;s__Butyricicoccus pullicaecorum  \n",
       "2  d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Butyricicoccaceae;g__Butyricicoccus_A;s__Butyricicoccus_A porcorum   \n",
       "⋮  ⋮                                                                                                                                    \n",
       "25 d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Butyricicoccaceae;g__Agathobaculum;s__Agathobaculum butyriciproducens\n",
       "26 d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Butyricicoccaceae;g__Butyricicoccus_A;s__Butyricicoccus_A sp002395695\n",
       "   closest_placement_ani closest_placement_af msa_percent translation_table\n",
       "1  99.99                 1.0                  97.26       11               \n",
       "2  100.0                 0.99                 96.27       11               \n",
       "⋮  ⋮                     ⋮                    ⋮           ⋮                \n",
       "25 95.65                 0.83                 97.10       11               \n",
       "26 98.29                 0.82                 75.63       11               \n",
       "   red_value warnings\n",
       "1  N/A       N/A     \n",
       "2  N/A       N/A     \n",
       "⋮  ⋮         ⋮       \n",
       "25 N/A       N/A     \n",
       "26 N/A       N/A     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# joining based on Bin.Id\n",
    "drep = drep %>%\n",
    "    inner_join(checkm, c('Bin.Id')) %>%\n",
    "    mutate(GEN = genome %>% as.character %>% basename) %>%\n",
    "    inner_join(drep_gen %>% mutate(GEN = Fasta %>% as.character %>% basename),\n",
    "               by=c('GEN')) %>%\n",
    "    inner_join(tax, c('Bin.Id'='user_genome')) #%>%\n",
    "drep"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-28T13:28:33.909938Z",
     "iopub.status.busy": "2021-04-28T13:28:33.907470Z",
     "iopub.status.idle": "2021-04-28T13:28:33.958309Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 21 × 60</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>genome</th><th scope=col>completeness</th><th scope=col>contamination</th><th scope=col>Bin.Id</th><th scope=col>Marker.lineage</th><th scope=col>X..genomes</th><th scope=col>X..markers</th><th scope=col>X..marker.sets</th><th scope=col>Completeness</th><th scope=col>Contamination</th><th scope=col>⋯</th><th scope=col>fastani_af</th><th scope=col>closest_placement_reference</th><th scope=col>closest_placement_radius</th><th scope=col>closest_placement_taxonomy</th><th scope=col>closest_placement_ani</th><th scope=col>closest_placement_af</th><th scope=col>msa_percent</th><th scope=col>translation_table</th><th scope=col>red_value</th><th scope=col>warnings</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000398925.1_Buty_pull_1_20_V1_genomic.fna</td><td>98.66</td><td>0</td><td>GCA_000398925.1_Buty_pull_1_20_V1_genomic</td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>98.66</td><td>0</td><td>⋯</td><td>1.0</td><td>GCF_900167005.1</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Butyricicoccaceae;g__Butyricicoccus;s__Butyricicoccus pullicaecorum</td><td>99.99</td><td>1.0</td><td>97.26</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_002160385.1_ASM216038v1_genomic.fna      </td><td>98.66</td><td>0</td><td>GCA_002160385.1_ASM216038v1_genomic      </td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>98.66</td><td>0</td><td>⋯</td><td>0.8</td><td>GCF_900167005.1</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Butyricicoccaceae;g__Butyricicoccus;s__Butyricicoccus pullicaecorum</td><td>96.44</td><td>0.8</td><td>97.26</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋱</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_014648015.1_ASM1464801v1_genomic.fna</td><td>99.33</td><td>0.67</td><td>GCA_014648015.1_ASM1464801v1_genomic</td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>99.33</td><td>0.67</td><td>⋯</td><td>0.83</td><td>GCF_003096535.1</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Butyricicoccaceae;g__Agathobaculum;s__Agathobaculum butyriciproducens</td><td>95.65</td><td>0.83</td><td>97.10</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "\t<tr><td>/ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_016293105.1_ASM1629310v1_genomic.fna</td><td>92.88</td><td>0.67</td><td>GCA_016293105.1_ASM1629310v1_genomic</td><td>o__Clostridiales (UID1212)</td><td>172</td><td>263</td><td>149</td><td>92.88</td><td>0.67</td><td>⋯</td><td>0.82</td><td>GCA_002395695.1</td><td>95.0</td><td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Butyricicoccaceae;g__Butyricicoccus_A;s__Butyricicoccus_A sp002395695</td><td>98.29</td><td>0.82</td><td>75.63</td><td>11</td><td>N/A</td><td>N/A</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 21 × 60\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " genome & completeness & contamination & Bin.Id & Marker.lineage & X..genomes & X..markers & X..marker.sets & Completeness & Contamination & ⋯ & fastani\\_af & closest\\_placement\\_reference & closest\\_placement\\_radius & closest\\_placement\\_taxonomy & closest\\_placement\\_ani & closest\\_placement\\_af & msa\\_percent & translation\\_table & red\\_value & warnings\\\\\n",
       " <fct> & <dbl> & <dbl> & <chr> & <fct> & <int> & <int> & <int> & <dbl> & <dbl> & ⋯ & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <dbl> & <int> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_000398925.1\\_Buty\\_pull\\_1\\_20\\_V1\\_genomic.fna & 98.66 & 0 & GCA\\_000398925.1\\_Buty\\_pull\\_1\\_20\\_V1\\_genomic & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 98.66 & 0 & ⋯ & 1.0 & GCF\\_900167005.1 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Butyricicoccaceae;g\\_\\_Butyricicoccus;s\\_\\_Butyricicoccus pullicaecorum & 99.99 & 1.0 & 97.26 & 11 & N/A & N/A\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_002160385.1\\_ASM216038v1\\_genomic.fna       & 98.66 & 0 & GCA\\_002160385.1\\_ASM216038v1\\_genomic       & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 98.66 & 0 & ⋯ & 0.8 & GCF\\_900167005.1 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Butyricicoccaceae;g\\_\\_Butyricicoccus;s\\_\\_Butyricicoccus pullicaecorum & 96.44 & 0.8 & 97.26 & 11 & N/A & N/A\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋱ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_014648015.1\\_ASM1464801v1\\_genomic.fna & 99.33 & 0.67 & GCA\\_014648015.1\\_ASM1464801v1\\_genomic & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 99.33 & 0.67 & ⋯ & 0.83 & GCF\\_003096535.1 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Butyricicoccaceae;g\\_\\_Agathobaculum;s\\_\\_Agathobaculum butyriciproducens & 95.65 & 0.83 & 97.10 & 11 & N/A & N/A\\\\\n",
       "\t /ebio/abt3\\_scratch/nyoungblut/LLG\\_62325884640/genomes/GCA\\_016293105.1\\_ASM1629310v1\\_genomic.fna & 92.88 & 0.67 & GCA\\_016293105.1\\_ASM1629310v1\\_genomic & o\\_\\_Clostridiales (UID1212) & 172 & 263 & 149 & 92.88 & 0.67 & ⋯ & 0.82 & GCA\\_002395695.1 & 95.0 & d\\_\\_Bacteria;p\\_\\_Firmicutes\\_A;c\\_\\_Clostridia;o\\_\\_Oscillospirales;f\\_\\_Butyricicoccaceae;g\\_\\_Butyricicoccus\\_A;s\\_\\_Butyricicoccus\\_A sp002395695 & 98.29 & 0.82 & 75.63 & 11 & N/A & N/A\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 21 × 60\n",
       "\n",
       "| genome &lt;fct&gt; | completeness &lt;dbl&gt; | contamination &lt;dbl&gt; | Bin.Id &lt;chr&gt; | Marker.lineage &lt;fct&gt; | X..genomes &lt;int&gt; | X..markers &lt;int&gt; | X..marker.sets &lt;int&gt; | Completeness &lt;dbl&gt; | Contamination &lt;dbl&gt; | ⋯ ⋯ | fastani_af &lt;fct&gt; | closest_placement_reference &lt;fct&gt; | closest_placement_radius &lt;fct&gt; | closest_placement_taxonomy &lt;fct&gt; | closest_placement_ani &lt;fct&gt; | closest_placement_af &lt;fct&gt; | msa_percent &lt;dbl&gt; | translation_table &lt;int&gt; | red_value &lt;fct&gt; | warnings &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000398925.1_Buty_pull_1_20_V1_genomic.fna | 98.66 | 0 | GCA_000398925.1_Buty_pull_1_20_V1_genomic | o__Clostridiales (UID1212) | 172 | 263 | 149 | 98.66 | 0 | ⋯ | 1.0 | GCF_900167005.1 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Butyricicoccaceae;g__Butyricicoccus;s__Butyricicoccus pullicaecorum | 99.99 | 1.0 | 97.26 | 11 | N/A | N/A |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_002160385.1_ASM216038v1_genomic.fna       | 98.66 | 0 | GCA_002160385.1_ASM216038v1_genomic       | o__Clostridiales (UID1212) | 172 | 263 | 149 | 98.66 | 0 | ⋯ | 0.8 | GCF_900167005.1 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Butyricicoccaceae;g__Butyricicoccus;s__Butyricicoccus pullicaecorum | 96.44 | 0.8 | 97.26 | 11 | N/A | N/A |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋱ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_014648015.1_ASM1464801v1_genomic.fna | 99.33 | 0.67 | GCA_014648015.1_ASM1464801v1_genomic | o__Clostridiales (UID1212) | 172 | 263 | 149 | 99.33 | 0.67 | ⋯ | 0.83 | GCF_003096535.1 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Butyricicoccaceae;g__Agathobaculum;s__Agathobaculum butyriciproducens | 95.65 | 0.83 | 97.10 | 11 | N/A | N/A |\n",
       "| /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_016293105.1_ASM1629310v1_genomic.fna | 92.88 | 0.67 | GCA_016293105.1_ASM1629310v1_genomic | o__Clostridiales (UID1212) | 172 | 263 | 149 | 92.88 | 0.67 | ⋯ | 0.82 | GCA_002395695.1 | 95.0 | d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Butyricicoccaceae;g__Butyricicoccus_A;s__Butyricicoccus_A sp002395695 | 98.29 | 0.82 | 75.63 | 11 | N/A | N/A |\n",
       "\n"
      ],
      "text/plain": [
       "   genome                                                                                             \n",
       "1  /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_000398925.1_Buty_pull_1_20_V1_genomic.fna\n",
       "2  /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_002160385.1_ASM216038v1_genomic.fna      \n",
       "⋮  ⋮                                                                                                  \n",
       "20 /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_014648015.1_ASM1464801v1_genomic.fna     \n",
       "21 /ebio/abt3_scratch/nyoungblut/LLG_62325884640/genomes/GCA_016293105.1_ASM1629310v1_genomic.fna     \n",
       "   completeness contamination Bin.Id                                   \n",
       "1  98.66        0             GCA_000398925.1_Buty_pull_1_20_V1_genomic\n",
       "2  98.66        0             GCA_002160385.1_ASM216038v1_genomic      \n",
       "⋮  ⋮            ⋮             ⋮                                        \n",
       "20 99.33        0.67          GCA_014648015.1_ASM1464801v1_genomic     \n",
       "21 92.88        0.67          GCA_016293105.1_ASM1629310v1_genomic     \n",
       "   Marker.lineage             X..genomes X..markers X..marker.sets Completeness\n",
       "1  o__Clostridiales (UID1212) 172        263        149            98.66       \n",
       "2  o__Clostridiales (UID1212) 172        263        149            98.66       \n",
       "⋮  ⋮                          ⋮          ⋮          ⋮              ⋮           \n",
       "20 o__Clostridiales (UID1212) 172        263        149            99.33       \n",
       "21 o__Clostridiales (UID1212) 172        263        149            92.88       \n",
       "   Contamination ⋯ fastani_af closest_placement_reference\n",
       "1  0             ⋯ 1.0        GCF_900167005.1            \n",
       "2  0             ⋯ 0.8        GCF_900167005.1            \n",
       "⋮  ⋮             ⋱ ⋮          ⋮                          \n",
       "20 0.67          ⋯ 0.83       GCF_003096535.1            \n",
       "21 0.67          ⋯ 0.82       GCA_002395695.1            \n",
       "   closest_placement_radius\n",
       "1  95.0                    \n",
       "2  95.0                    \n",
       "⋮  ⋮                       \n",
       "20 95.0                    \n",
       "21 95.0                    \n",
       "   closest_placement_taxonomy                                                                                                           \n",
       "1  d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Butyricicoccaceae;g__Butyricicoccus;s__Butyricicoccus pullicaecorum  \n",
       "2  d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Butyricicoccaceae;g__Butyricicoccus;s__Butyricicoccus pullicaecorum  \n",
       "⋮  ⋮                                                                                                                                    \n",
       "20 d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Butyricicoccaceae;g__Agathobaculum;s__Agathobaculum butyriciproducens\n",
       "21 d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Butyricicoccaceae;g__Butyricicoccus_A;s__Butyricicoccus_A sp002395695\n",
       "   closest_placement_ani closest_placement_af msa_percent translation_table\n",
       "1  99.99                 1.0                  97.26       11               \n",
       "2  96.44                 0.8                  97.26       11               \n",
       "⋮  ⋮                     ⋮                    ⋮           ⋮                \n",
       "20 95.65                 0.83                 97.10       11               \n",
       "21 98.29                 0.82                 75.63       11               \n",
       "   red_value warnings\n",
       "1  N/A       N/A     \n",
       "2  N/A       N/A     \n",
       "⋮  ⋮         ⋮       \n",
       "20 N/A       N/A     \n",
       "21 N/A       N/A     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# filtering by quality\n",
    "hq_genomes = drep %>%\n",
    "    filter(completeness >= 90,\n",
    "           contamination < 5,\n",
    "           Strain.heterogeneity < 50,\n",
    "           X..contigs < 300,\n",
    "           Mean.contig.length..bp. > 10000)\n",
    "hq_genomes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-28T13:28:34.071219Z",
     "iopub.status.busy": "2021-04-28T13:28:34.069002Z",
     "iopub.status.idle": "2021-04-28T13:28:34.140379Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 1 × 8 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Min.</th><th scope=col>1st Qu.</th><th scope=col>Median</th><th scope=col>Mean</th><th scope=col>3rd Qu.</th><th scope=col>Max.</th><th scope=col>sd</th><th scope=col>sd_err_of_mean</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>completeness</th><td>92.88</td><td>98.66</td><td>99.33</td><td>98.67429</td><td>99.33</td><td>99.92</td><td>2.616</td><td>1.068</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 1 × 8 of type dbl\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & Min. & 1st Qu. & Median & Mean & 3rd Qu. & Max. & sd & sd\\_err\\_of\\_mean\\\\\n",
       "\\hline\n",
       "\tcompleteness & 92.88 & 98.66 & 99.33 & 98.67429 & 99.33 & 99.92 & 2.616 & 1.068\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 1 × 8 of type dbl\n",
       "\n",
       "| <!--/--> | Min. | 1st Qu. | Median | Mean | 3rd Qu. | Max. | sd | sd_err_of_mean |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| completeness | 92.88 | 98.66 | 99.33 | 98.67429 | 99.33 | 99.92 | 2.616 | 1.068 |\n",
       "\n"
      ],
      "text/plain": [
       "             Min.  1st Qu. Median Mean     3rd Qu. Max.  sd    sd_err_of_mean\n",
       "completeness 92.88 98.66   99.33  98.67429 99.33   99.92 2.616 1.068         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 1 × 8 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Min.</th><th scope=col>1st Qu.</th><th scope=col>Median</th><th scope=col>Mean</th><th scope=col>3rd Qu.</th><th scope=col>Max.</th><th scope=col>sd</th><th scope=col>sd_err_of_mean</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>contamination</th><td>0</td><td>0.67</td><td>0.67</td><td>0.6128571</td><td>0.67</td><td>1.34</td><td>0.424</td><td>0.173</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 1 × 8 of type dbl\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & Min. & 1st Qu. & Median & Mean & 3rd Qu. & Max. & sd & sd\\_err\\_of\\_mean\\\\\n",
       "\\hline\n",
       "\tcontamination & 0 & 0.67 & 0.67 & 0.6128571 & 0.67 & 1.34 & 0.424 & 0.173\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 1 × 8 of type dbl\n",
       "\n",
       "| <!--/--> | Min. | 1st Qu. | Median | Mean | 3rd Qu. | Max. | sd | sd_err_of_mean |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| contamination | 0 | 0.67 | 0.67 | 0.6128571 | 0.67 | 1.34 | 0.424 | 0.173 |\n",
       "\n"
      ],
      "text/plain": [
       "              Min. 1st Qu. Median Mean      3rd Qu. Max. sd    sd_err_of_mean\n",
       "contamination 0    0.67    0.67   0.6128571 0.67    1.34 0.424 0.173         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 1 × 8 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Min.</th><th scope=col>1st Qu.</th><th scope=col>Median</th><th scope=col>Mean</th><th scope=col>3rd Qu.</th><th scope=col>Max.</th><th scope=col>sd</th><th scope=col>sd_err_of_mean</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>No. of contigs</th><td>4</td><td>55</td><td>71</td><td>70.28571</td><td>82</td><td>143</td><td>44.834</td><td>18.303</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 1 × 8 of type dbl\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & Min. & 1st Qu. & Median & Mean & 3rd Qu. & Max. & sd & sd\\_err\\_of\\_mean\\\\\n",
       "\\hline\n",
       "\tNo. of contigs & 4 & 55 & 71 & 70.28571 & 82 & 143 & 44.834 & 18.303\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 1 × 8 of type dbl\n",
       "\n",
       "| <!--/--> | Min. | 1st Qu. | Median | Mean | 3rd Qu. | Max. | sd | sd_err_of_mean |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| No. of contigs | 4 | 55 | 71 | 70.28571 | 82 | 143 | 44.834 | 18.303 |\n",
       "\n"
      ],
      "text/plain": [
       "               Min. 1st Qu. Median Mean     3rd Qu. Max. sd     sd_err_of_mean\n",
       "No. of contigs 4    55      71     70.28571 82      143  44.834 18.303        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 1 × 8 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Min.</th><th scope=col>1st Qu.</th><th scope=col>Median</th><th scope=col>Mean</th><th scope=col>3rd Qu.</th><th scope=col>Max.</th><th scope=col>sd</th><th scope=col>sd_err_of_mean</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>N50</th><td>31284</td><td>93692</td><td>110619</td><td>280090.9</td><td>218166</td><td>2682118</td><td>1038941</td><td>424145.8</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 1 × 8 of type dbl\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & Min. & 1st Qu. & Median & Mean & 3rd Qu. & Max. & sd & sd\\_err\\_of\\_mean\\\\\n",
       "\\hline\n",
       "\tN50 & 31284 & 93692 & 110619 & 280090.9 & 218166 & 2682118 & 1038941 & 424145.8\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 1 × 8 of type dbl\n",
       "\n",
       "| <!--/--> | Min. | 1st Qu. | Median | Mean | 3rd Qu. | Max. | sd | sd_err_of_mean |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| N50 | 31284 | 93692 | 110619 | 280090.9 | 218166 | 2682118 | 1038941 | 424145.8 |\n",
       "\n"
      ],
      "text/plain": [
       "    Min.  1st Qu. Median Mean     3rd Qu. Max.    sd      sd_err_of_mean\n",
       "N50 31284 93692   110619 280090.9 218166  2682118 1038941 424145.8      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 1 × 8 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Min.</th><th scope=col>1st Qu.</th><th scope=col>Median</th><th scope=col>Mean</th><th scope=col>3rd Qu.</th><th scope=col>Max.</th><th scope=col>sd</th><th scope=col>sd_err_of_mean</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Mean contig length</th><td>18642</td><td>36058</td><td>44722</td><td>84265.19</td><td>53821</td><td>737295</td><td>282444.8</td><td>115307.6</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 1 × 8 of type dbl\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & Min. & 1st Qu. & Median & Mean & 3rd Qu. & Max. & sd & sd\\_err\\_of\\_mean\\\\\n",
       "\\hline\n",
       "\tMean contig length & 18642 & 36058 & 44722 & 84265.19 & 53821 & 737295 & 282444.8 & 115307.6\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 1 × 8 of type dbl\n",
       "\n",
       "| <!--/--> | Min. | 1st Qu. | Median | Mean | 3rd Qu. | Max. | sd | sd_err_of_mean |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Mean contig length | 18642 | 36058 | 44722 | 84265.19 | 53821 | 737295 | 282444.8 | 115307.6 |\n",
       "\n"
      ],
      "text/plain": [
       "                   Min.  1st Qu. Median Mean     3rd Qu. Max.   sd      \n",
       "Mean contig length 18642 36058   44722  84265.19 53821   737295 282444.8\n",
       "                   sd_err_of_mean\n",
       "Mean contig length 115307.6      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# summarizing the genome stats\n",
    "hq_genomes %>% summary_x('completeness', completeness)\n",
    "hq_genomes %>% summary_x('contamination', contamination)\n",
    "hq_genomes %>% summary_x('No. of contigs', X..contigs)\n",
    "hq_genomes %>% summary_x('N50', N50..contigs.)\n",
    "hq_genomes %>% summary_x('Mean contig length', Mean.contig.length..bp.)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-28T13:28:34.267921Z",
     "iopub.status.busy": "2021-04-28T13:28:34.265833Z",
     "iopub.status.idle": "2021-04-28T13:28:34.313827Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A tibble: 3 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Family</th><th scope=col>Genus</th><th scope=col>n_genomes</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>f__Butyricicoccaceae</td><td>g__Agathobaculum   </td><td>17</td></tr>\n",
       "\t<tr><td>f__Butyricicoccaceae</td><td>g__Butyricicoccus  </td><td> 3</td></tr>\n",
       "\t<tr><td>f__Butyricicoccaceae</td><td>g__Butyricicoccus_A</td><td> 1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 3 × 3\n",
       "\\begin{tabular}{lll}\n",
       " Family & Genus & n\\_genomes\\\\\n",
       " <chr> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t f\\_\\_Butyricicoccaceae & g\\_\\_Agathobaculum    & 17\\\\\n",
       "\t f\\_\\_Butyricicoccaceae & g\\_\\_Butyricicoccus   &  3\\\\\n",
       "\t f\\_\\_Butyricicoccaceae & g\\_\\_Butyricicoccus\\_A &  1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 3 × 3\n",
       "\n",
       "| Family &lt;chr&gt; | Genus &lt;chr&gt; | n_genomes &lt;int&gt; |\n",
       "|---|---|---|\n",
       "| f__Butyricicoccaceae | g__Agathobaculum    | 17 |\n",
       "| f__Butyricicoccaceae | g__Butyricicoccus   |  3 |\n",
       "| f__Butyricicoccaceae | g__Butyricicoccus_A |  1 |\n",
       "\n"
      ],
      "text/plain": [
       "  Family               Genus               n_genomes\n",
       "1 f__Butyricicoccaceae g__Agathobaculum    17       \n",
       "2 f__Butyricicoccaceae g__Butyricicoccus    3       \n",
       "3 f__Butyricicoccaceae g__Butyricicoccus_A  1       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# summarizing the taxonomy\n",
    "df.dims(20)\n",
    "hq_genomes %>%\n",
    "    group_by(Family, Genus) %>%\n",
    "    summarize(n_genomes = n(), .groups='drop')\n",
    "df.dims()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-28T13:28:34.535429Z",
     "iopub.status.busy": "2021-04-28T13:28:34.533371Z",
     "iopub.status.idle": "2021-04-28T13:28:34.566111Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "File written: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3//Butyricicoccus/samples_genomes_hq.txt \n"
     ]
    }
   ],
   "source": [
    "# writing samples table for LLPRIMER\n",
    "outfile = file.path(work_dir, 'samples_genomes_hq.txt')\n",
    "hq_genomes %>%\n",
    "    select(Bin.Id, Fasta) %>%\n",
    "    rename('Taxon' = Bin.Id) %>%\n",
    "    mutate(Taxon = gsub('_chromosome.+', '', Taxon),\n",
    "           Taxon = gsub('_bin_.+', '', Taxon),\n",
    "           Taxon = gsub('_genomic', '', Taxon),\n",
    "           Taxon = gsub('_annotated_assembly', '', Taxon),\n",
    "           Taxid = taxid,\n",
    "           Domain = domain) %>%\n",
    "    write_table(outfile)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Primer design"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Config"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-28T13:28:36.862068Z",
     "iopub.status.busy": "2021-04-28T13:28:36.860068Z",
     "iopub.status.idle": "2021-04-28T13:28:36.888231Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#-- I/O --#\n",
      "samples_file: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricicoccus/samples_genomes_hq.txt\n",
      "\n",
      "# output location\n",
      "output_dir: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricicoccus/primers/\n",
      "\n",
      "# temporary file directory (your username will be added automatically)\n",
      "tmp_dir: /ebio/abt3_scratch/\n",
      "\n",
      "#-- software parameters --#\n",
      "# See the README for a description\n",
      "params:\n",
      "  ionice: -c 3\n",
      "  cgp:\n",
      "    prokka: \"\"\n",
      "    mmseqs:\n",
      "      method: cluster    # or linclust (faster)\n",
      "      run: --min-seq-id 0.8 -c 0.8\n",
      "    vsearch: --id 0.94\n",
      "    core_genes:\n",
      "      cds: --perc-genomes-cds 100 --copies-per-genome-cds 1 --max-clusters-cds 500\n",
      "      rRNA: --perc-genomes-rrna 100 --copies-per-genome-rrna 10 --max-clusters-rrna 500\n",
      "    align:\n",
      "      method: linsi\n",
      "      params: --auto --maxiterate 1000\n",
      "    primer3:\n",
      "      consensus: --consensus-threshold 0.34\n",
      "      number: --num-raw-primers 1000 --num-final-primers 10\n",
      "      size: --opt-size 20 --min-size 18 --max-size 26\n",
      "      product: --opt-prod-size 150 --min-prod-size 100 --max-prod-size 250\n",
      "      Tm: --opt-tm 62 --min-tm 54 --max-tm 70 --max-tm-diff 2.5\n",
      "      PCR: --oligo-DNA 50 --salt-monovalent 50 --salt-divalent 1.5 --dNTPs 0.2\n",
      "      GC: --opt-gc 50 --min-gc 25 --max-gc 75\n",
      "      degeneracy: --max-degeneracy 128 --max-degeneracy-3prime 6 --window-3prime 5\n",
      "      internal_oligo: --make-oligo 0 --int-opt-size 20 --int-opt-tm 60\n",
      "    blast_nontarget:\n",
      "      run_locally: False\n",
      "      cds: -evalue 1e-5 -max_target_seqs 30\n",
      "      rRNA: -evalue 1e-5 -max_target_seqs 30\n",
      "    blastn: -max_target_seqs 500 -perc_identity 90\n",
      "    blast_filter: --perc-len 80 --min-amplicon-len 30 --max-amplicon-len 2000\n",
      "    seqkit_amplicon: \"\"\n",
      "\n",
      "#-- database --#\n",
      "databases:\n",
      "  blast:\n",
      "    nucl: /ebio/abt3_projects/databases_no-backup/NCBI_blastdb/nt\n",
      "    prot: /ebio/abt3_projects/databases_no-backup/NCBI_blastdb/nr\n",
      "    rRNA:\n",
      "      SSU: /ebio/abt3_projects/databases_no-backup/NCBI_rRNA/2021-01/SSU/16S_ribosomal_RNA\n",
      "      LSU: /ebio/abt3_projects/databases_no-backup/NCBI_rRNA/2021-01/LSU/LSU_prokaryote_rRNA\n",
      "    gene_names: /ebio/abt3_projects/databases_no-backup/NCBI_nr_geneIDs/2021-01/nr_accession-geneNames.pkl\n",
      "  taxonomy:\n",
      "    names: /ebio/abt3_projects/databases_no-backup/NCBI_taxonomy/2020-12-23/names.dmp\n",
      "    nodes: /ebio/abt3_projects/databases_no-backup/NCBI_taxonomy/2020-12-23/nodes.dmp\n",
      "\n",
      "#-- snakemake pipeline --#\n",
      "pipeline:\n",
      "  snakemake_folder: ./\n",
      "  script_folder: bin/scripts/\n",
      "  name: LLPRIMER"
     ]
    }
   ],
   "source": [
    "F = file.path(work_dir, 'primers', 'config.yaml')\n",
    "cat_file(F)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Run"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "(snakemake) @ rick:/ebio/abt3_projects/software/dev/ll_pipelines/llprimer\n",
    "$ screen -L -S llprimer-Butyc ./snakemake_sge.sh /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3/Butyricicoccus/primers/config.yaml 50 --notemp -F\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Summary"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Primers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-29T12:21:37.293766Z",
     "iopub.status.busy": "2021-04-29T12:21:37.286854Z",
     "iopub.status.idle": "2021-04-29T12:21:37.789884Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No. of unique primers: 297 \n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 706 × 22</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>gene_type</th><th scope=col>cluster_id</th><th scope=col>primer_set</th><th scope=col>amplicon_size_consensus</th><th scope=col>amplicon_size_avg</th><th scope=col>amplicon_size_sd</th><th scope=col>primer_id</th><th scope=col>primer_type</th><th scope=col>sequence</th><th scope=col>length</th><th scope=col>⋯</th><th scope=col>position_start</th><th scope=col>position_end</th><th scope=col>Tm_avg</th><th scope=col>Tm_sd</th><th scope=col>GC_avg</th><th scope=col>GC_sd</th><th scope=col>hairpin_avg</th><th scope=col>hairpin_sd</th><th scope=col>homodimer_avg</th><th scope=col>homodimer_sd</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>cds</td><td>1</td><td>116</td><td>100</td><td>100</td><td>0</td><td>116f</td><td>PRIMER_LEFT </td><td>ATYCAGTTYCTCARGGCRCT</td><td>20</td><td>⋯</td><td>909</td><td> 929</td><td>59.82351</td><td>2.199735</td><td>50.0</td><td>5.000000</td><td>20.66486</td><td>20.92350</td><td>-9.250822</td><td>20.940934</td></tr>\n",
       "\t<tr><td>cds</td><td>1</td><td>116</td><td>100</td><td>100</td><td>0</td><td>116r</td><td>PRIMER_RIGHT</td><td>CGTCCTTSKTGCCGGTRAAR</td><td>20</td><td>⋯</td><td>989</td><td>1009</td><td>61.88123</td><td>1.617130</td><td>57.5</td><td>4.330127</td><td>11.56401</td><td>17.22575</td><td> 5.615398</td><td> 5.903092</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋱</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>cds</td><td>40</td><td>973</td><td>158</td><td>158</td><td>0</td><td>973f</td><td>PRIMER_LEFT </td><td>GARTTCTTCTGCAAGCCKGG</td><td>20</td><td>⋯</td><td>687</td><td>707</td><td>59.83334</td><td>1.519704</td><td>55</td><td>3.535534</td><td>0.000000</td><td> 0.00000</td><td> -6.653505</td><td>12.39957</td></tr>\n",
       "\t<tr><td>cds</td><td>40</td><td>973</td><td>158</td><td>158</td><td>0</td><td>973r</td><td>PRIMER_RIGHT</td><td>TCKGTGGTWGCRTTGGAGTA</td><td>20</td><td>⋯</td><td>825</td><td>845</td><td>59.34475</td><td>1.864572</td><td>50</td><td>3.535534</td><td>5.098652</td><td>13.48977</td><td>-23.091678</td><td>15.40643</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 706 × 22\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " gene\\_type & cluster\\_id & primer\\_set & amplicon\\_size\\_consensus & amplicon\\_size\\_avg & amplicon\\_size\\_sd & primer\\_id & primer\\_type & sequence & length & ⋯ & position\\_start & position\\_end & Tm\\_avg & Tm\\_sd & GC\\_avg & GC\\_sd & hairpin\\_avg & hairpin\\_sd & homodimer\\_avg & homodimer\\_sd\\\\\n",
       " <fct> & <int> & <int> & <int> & <dbl> & <dbl> & <fct> & <fct> & <fct> & <int> & ⋯ & <int> & <int> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t cds & 1 & 116 & 100 & 100 & 0 & 116f & PRIMER\\_LEFT  & ATYCAGTTYCTCARGGCRCT & 20 & ⋯ & 909 &  929 & 59.82351 & 2.199735 & 50.0 & 5.000000 & 20.66486 & 20.92350 & -9.250822 & 20.940934\\\\\n",
       "\t cds & 1 & 116 & 100 & 100 & 0 & 116r & PRIMER\\_RIGHT & CGTCCTTSKTGCCGGTRAAR & 20 & ⋯ & 989 & 1009 & 61.88123 & 1.617130 & 57.5 & 4.330127 & 11.56401 & 17.22575 &  5.615398 &  5.903092\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋱ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t cds & 40 & 973 & 158 & 158 & 0 & 973f & PRIMER\\_LEFT  & GARTTCTTCTGCAAGCCKGG & 20 & ⋯ & 687 & 707 & 59.83334 & 1.519704 & 55 & 3.535534 & 0.000000 &  0.00000 &  -6.653505 & 12.39957\\\\\n",
       "\t cds & 40 & 973 & 158 & 158 & 0 & 973r & PRIMER\\_RIGHT & TCKGTGGTWGCRTTGGAGTA & 20 & ⋯ & 825 & 845 & 59.34475 & 1.864572 & 50 & 3.535534 & 5.098652 & 13.48977 & -23.091678 & 15.40643\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 706 × 22\n",
       "\n",
       "| gene_type &lt;fct&gt; | cluster_id &lt;int&gt; | primer_set &lt;int&gt; | amplicon_size_consensus &lt;int&gt; | amplicon_size_avg &lt;dbl&gt; | amplicon_size_sd &lt;dbl&gt; | primer_id &lt;fct&gt; | primer_type &lt;fct&gt; | sequence &lt;fct&gt; | length &lt;int&gt; | ⋯ ⋯ | position_start &lt;int&gt; | position_end &lt;int&gt; | Tm_avg &lt;dbl&gt; | Tm_sd &lt;dbl&gt; | GC_avg &lt;dbl&gt; | GC_sd &lt;dbl&gt; | hairpin_avg &lt;dbl&gt; | hairpin_sd &lt;dbl&gt; | homodimer_avg &lt;dbl&gt; | homodimer_sd &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| cds | 1 | 116 | 100 | 100 | 0 | 116f | PRIMER_LEFT  | ATYCAGTTYCTCARGGCRCT | 20 | ⋯ | 909 |  929 | 59.82351 | 2.199735 | 50.0 | 5.000000 | 20.66486 | 20.92350 | -9.250822 | 20.940934 |\n",
       "| cds | 1 | 116 | 100 | 100 | 0 | 116r | PRIMER_RIGHT | CGTCCTTSKTGCCGGTRAAR | 20 | ⋯ | 989 | 1009 | 61.88123 | 1.617130 | 57.5 | 4.330127 | 11.56401 | 17.22575 |  5.615398 |  5.903092 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋱ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| cds | 40 | 973 | 158 | 158 | 0 | 973f | PRIMER_LEFT  | GARTTCTTCTGCAAGCCKGG | 20 | ⋯ | 687 | 707 | 59.83334 | 1.519704 | 55 | 3.535534 | 0.000000 |  0.00000 |  -6.653505 | 12.39957 |\n",
       "| cds | 40 | 973 | 158 | 158 | 0 | 973r | PRIMER_RIGHT | TCKGTGGTWGCRTTGGAGTA | 20 | ⋯ | 825 | 845 | 59.34475 | 1.864572 | 50 | 3.535534 | 5.098652 | 13.48977 | -23.091678 | 15.40643 |\n",
       "\n"
      ],
      "text/plain": [
       "    gene_type cluster_id primer_set amplicon_size_consensus amplicon_size_avg\n",
       "1   cds       1          116        100                     100              \n",
       "2   cds       1          116        100                     100              \n",
       "⋮   ⋮         ⋮          ⋮          ⋮                       ⋮                \n",
       "705 cds       40         973        158                     158              \n",
       "706 cds       40         973        158                     158              \n",
       "    amplicon_size_sd primer_id primer_type  sequence             length ⋯\n",
       "1   0                116f      PRIMER_LEFT  ATYCAGTTYCTCARGGCRCT 20     ⋯\n",
       "2   0                116r      PRIMER_RIGHT CGTCCTTSKTGCCGGTRAAR 20     ⋯\n",
       "⋮   ⋮                ⋮         ⋮            ⋮                    ⋮      ⋱\n",
       "705 0                973f      PRIMER_LEFT  GARTTCTTCTGCAAGCCKGG 20     ⋯\n",
       "706 0                973r      PRIMER_RIGHT TCKGTGGTWGCRTTGGAGTA 20     ⋯\n",
       "    position_start position_end Tm_avg   Tm_sd    GC_avg GC_sd    hairpin_avg\n",
       "1   909             929         59.82351 2.199735 50.0   5.000000 20.66486   \n",
       "2   989            1009         61.88123 1.617130 57.5   4.330127 11.56401   \n",
       "⋮   ⋮              ⋮            ⋮        ⋮        ⋮      ⋮        ⋮          \n",
       "705 687            707          59.83334 1.519704 55     3.535534 0.000000   \n",
       "706 825            845          59.34475 1.864572 50     3.535534 5.098652   \n",
       "    hairpin_sd homodimer_avg homodimer_sd\n",
       "1   20.92350   -9.250822     20.940934   \n",
       "2   17.22575    5.615398      5.903092   \n",
       "⋮   ⋮          ⋮             ⋮           \n",
       "705  0.00000    -6.653505    12.39957    \n",
       "706 13.48977   -23.091678    15.40643    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "primer_info = read.delim(file.path(work_dir, 'primers', 'cgp', 'primers_final_info.tsv'), sep='\\t')\n",
    "primer_info %>% unique_n('primers', primer_set)\n",
    "primer_info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Gene clusters"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-29T12:21:48.349278Z",
     "iopub.status.busy": "2021-04-29T12:21:48.347200Z",
     "iopub.status.idle": "2021-04-29T12:21:48.420110Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No. of clusters 37 \n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 777 × 13</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>gene_type</th><th scope=col>cluster_id</th><th scope=col>seq_uuid</th><th scope=col>seq_orig_name</th><th scope=col>contig_id</th><th scope=col>taxon</th><th scope=col>start</th><th scope=col>end</th><th scope=col>score</th><th scope=col>strand</th><th scope=col>annotation</th><th scope=col>cluster_name</th><th scope=col>clust_id</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>cds</td><td>1</td><td>1d7bcee77c924e679b895ba1759de82d</td><td>OCDEKLAK_00702</td><td>KB976103.1    </td><td>GCA_000398925.1_Buty_pull_1_20_V1</td><td>700992</td><td>702251</td><td>.</td><td>+</td><td>Carboxynorspermidine synthase</td><td>00dd2d7561b044c2bfeaee1e6eeb3445</td><td>1</td></tr>\n",
       "\t<tr><td>cds</td><td>1</td><td>00dd2d7561b044c2bfeaee1e6eeb3445</td><td>DHLGEPAL_00507</td><td>NFKL01000012.1</td><td>GCA_002160385.1_ASM216038v1      </td><td> 12607</td><td> 13866</td><td>.</td><td>-</td><td>Carboxynorspermidine synthase</td><td>00dd2d7561b044c2bfeaee1e6eeb3445</td><td>1</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>cds</td><td>40</td><td>edf922ca11864386b62490fc5f397e04</td><td>JGLGHNAD_01418</td><td>BMQH01000006.1   </td><td>GCA_014648015.1_ASM1464801v1</td><td>83267</td><td>84679</td><td>.</td><td>-</td><td>Glycine--tRNA ligase</td><td>012d2da3e8034d61a3424db553fd69b1</td><td>40</td></tr>\n",
       "\t<tr><td>cds</td><td>40</td><td>e09f9b15a14e4a089c1cdd68c18c7601</td><td>JJLDPIFE_00529</td><td>JAEDEW010000002.1</td><td>GCA_016293105.1_ASM1629310v1</td><td>55073</td><td>56464</td><td>.</td><td>+</td><td>Glycine--tRNA ligase</td><td>012d2da3e8034d61a3424db553fd69b1</td><td>40</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 777 × 13\n",
       "\\begin{tabular}{lllllllllllll}\n",
       " gene\\_type & cluster\\_id & seq\\_uuid & seq\\_orig\\_name & contig\\_id & taxon & start & end & score & strand & annotation & cluster\\_name & clust\\_id\\\\\n",
       " <fct> & <int> & <fct> & <fct> & <fct> & <fct> & <int> & <int> & <fct> & <fct> & <fct> & <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t cds & 1 & 1d7bcee77c924e679b895ba1759de82d & OCDEKLAK\\_00702 & KB976103.1     & GCA\\_000398925.1\\_Buty\\_pull\\_1\\_20\\_V1 & 700992 & 702251 & . & + & Carboxynorspermidine synthase & 00dd2d7561b044c2bfeaee1e6eeb3445 & 1\\\\\n",
       "\t cds & 1 & 00dd2d7561b044c2bfeaee1e6eeb3445 & DHLGEPAL\\_00507 & NFKL01000012.1 & GCA\\_002160385.1\\_ASM216038v1       &  12607 &  13866 & . & - & Carboxynorspermidine synthase & 00dd2d7561b044c2bfeaee1e6eeb3445 & 1\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t cds & 40 & edf922ca11864386b62490fc5f397e04 & JGLGHNAD\\_01418 & BMQH01000006.1    & GCA\\_014648015.1\\_ASM1464801v1 & 83267 & 84679 & . & - & Glycine--tRNA ligase & 012d2da3e8034d61a3424db553fd69b1 & 40\\\\\n",
       "\t cds & 40 & e09f9b15a14e4a089c1cdd68c18c7601 & JJLDPIFE\\_00529 & JAEDEW010000002.1 & GCA\\_016293105.1\\_ASM1629310v1 & 55073 & 56464 & . & + & Glycine--tRNA ligase & 012d2da3e8034d61a3424db553fd69b1 & 40\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 777 × 13\n",
       "\n",
       "| gene_type &lt;fct&gt; | cluster_id &lt;int&gt; | seq_uuid &lt;fct&gt; | seq_orig_name &lt;fct&gt; | contig_id &lt;fct&gt; | taxon &lt;fct&gt; | start &lt;int&gt; | end &lt;int&gt; | score &lt;fct&gt; | strand &lt;fct&gt; | annotation &lt;fct&gt; | cluster_name &lt;fct&gt; | clust_id &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| cds | 1 | 1d7bcee77c924e679b895ba1759de82d | OCDEKLAK_00702 | KB976103.1     | GCA_000398925.1_Buty_pull_1_20_V1 | 700992 | 702251 | . | + | Carboxynorspermidine synthase | 00dd2d7561b044c2bfeaee1e6eeb3445 | 1 |\n",
       "| cds | 1 | 00dd2d7561b044c2bfeaee1e6eeb3445 | DHLGEPAL_00507 | NFKL01000012.1 | GCA_002160385.1_ASM216038v1       |  12607 |  13866 | . | - | Carboxynorspermidine synthase | 00dd2d7561b044c2bfeaee1e6eeb3445 | 1 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| cds | 40 | edf922ca11864386b62490fc5f397e04 | JGLGHNAD_01418 | BMQH01000006.1    | GCA_014648015.1_ASM1464801v1 | 83267 | 84679 | . | - | Glycine--tRNA ligase | 012d2da3e8034d61a3424db553fd69b1 | 40 |\n",
       "| cds | 40 | e09f9b15a14e4a089c1cdd68c18c7601 | JJLDPIFE_00529 | JAEDEW010000002.1 | GCA_016293105.1_ASM1629310v1 | 55073 | 56464 | . | + | Glycine--tRNA ligase | 012d2da3e8034d61a3424db553fd69b1 | 40 |\n",
       "\n"
      ],
      "text/plain": [
       "    gene_type cluster_id seq_uuid                         seq_orig_name \n",
       "1   cds       1          1d7bcee77c924e679b895ba1759de82d OCDEKLAK_00702\n",
       "2   cds       1          00dd2d7561b044c2bfeaee1e6eeb3445 DHLGEPAL_00507\n",
       "⋮   ⋮         ⋮          ⋮                                ⋮             \n",
       "776 cds       40         edf922ca11864386b62490fc5f397e04 JGLGHNAD_01418\n",
       "777 cds       40         e09f9b15a14e4a089c1cdd68c18c7601 JJLDPIFE_00529\n",
       "    contig_id         taxon                             start  end    score\n",
       "1   KB976103.1        GCA_000398925.1_Buty_pull_1_20_V1 700992 702251 .    \n",
       "2   NFKL01000012.1    GCA_002160385.1_ASM216038v1        12607  13866 .    \n",
       "⋮   ⋮                 ⋮                                 ⋮      ⋮      ⋮    \n",
       "776 BMQH01000006.1    GCA_014648015.1_ASM1464801v1      83267  84679  .    \n",
       "777 JAEDEW010000002.1 GCA_016293105.1_ASM1629310v1      55073  56464  .    \n",
       "    strand annotation                    cluster_name                    \n",
       "1   +      Carboxynorspermidine synthase 00dd2d7561b044c2bfeaee1e6eeb3445\n",
       "2   -      Carboxynorspermidine synthase 00dd2d7561b044c2bfeaee1e6eeb3445\n",
       "⋮   ⋮      ⋮                             ⋮                               \n",
       "776 -      Glycine--tRNA ligase          012d2da3e8034d61a3424db553fd69b1\n",
       "777 +      Glycine--tRNA ligase          012d2da3e8034d61a3424db553fd69b1\n",
       "    clust_id\n",
       "1   1       \n",
       "2   1       \n",
       "⋮   ⋮       \n",
       "776 40      \n",
       "777 40      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# target gene info\n",
    "gene_annot = read.delim(file.path(work_dir, 'primers', 'cgp', 'core_clusters_info.tsv'), \n",
    "                        sep='\\t') %>%\n",
    "    semi_join(primer_info, c('cluster_id')) \n",
    "gene_annot %>% distinct(gene_type, cluster_id) %>% nrow %>% cat('No. of clusters', ., '\\n')\n",
    "gene_annot"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Gene cluster annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-29T12:22:02.645311Z",
     "iopub.status.busy": "2021-04-29T12:22:02.643990Z",
     "iopub.status.idle": "2021-04-29T12:22:02.709704Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 1110 × 17</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>cluster_id</th><th scope=col>query</th><th scope=col>subject</th><th scope=col>subject_name</th><th scope=col>pident</th><th scope=col>length</th><th scope=col>mismatch</th><th scope=col>qstart</th><th scope=col>qend</th><th scope=col>sstart</th><th scope=col>send</th><th scope=col>evalue</th><th scope=col>slen</th><th scope=col>qlen</th><th scope=col>sscinames</th><th scope=col>staxids</th><th scope=col>pident_rank</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td>00dd2d7561b044c2bfeaee1e6eeb3445</td><td>WP_122788428.1</td><td>saccharopine dehydrogenase family protein [Intestinibacillus sp. Marseille-P6563]</td><td>92.840</td><td>419</td><td>30</td><td>1</td><td>1257</td><td>1</td><td>419</td><td>0</td><td>419</td><td>1260</td><td>Intestinibacillus sp. Marseille-P6563                                                                   </td><td>2364792              </td><td>1</td></tr>\n",
       "\t<tr><td>1</td><td>00dd2d7561b044c2bfeaee1e6eeb3445</td><td>WP_006572513.1</td><td>saccharopine dehydrogenase family protein [Pseudoflavonifractor capillosus]      </td><td>91.885</td><td>419</td><td>34</td><td>1</td><td>1257</td><td>1</td><td>419</td><td>0</td><td>419</td><td>1260</td><td>Pseudoflavonifractor capillosus;Pseudoflavonifractor capillosus ATCC 29799;uncultured Flavonifractor sp.</td><td>106588;411467;1193534</td><td>2</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>40</td><td>012d2da3e8034d61a3424db553fd69b1</td><td>WP_016317035.1</td><td>MULTISPECIES: glycine--tRNA ligase [unclassified Anaerotruncus]</td><td>78.618</td><td>463</td><td>99</td><td>1</td><td>1389</td><td> 1</td><td>463</td><td>0</td><td>463</td><td>1392</td><td>Anaerotruncus sp. G3(2012);Anaerotruncus sp. X29;Anaerotruncus sp. 1XD22-93;Anaerotruncus sp. 1XD42-93;unclassified Anaerotruncus</td><td>1235835;2304567;2320082;2320853;2641626</td><td>29</td></tr>\n",
       "\t<tr><td>40</td><td>012d2da3e8034d61a3424db553fd69b1</td><td>WP_147322456.1</td><td>MULTISPECIES: glycine--tRNA ligase [Ruminococcaceae]           </td><td>80.778</td><td>463</td><td>84</td><td>1</td><td>1389</td><td>20</td><td>477</td><td>0</td><td>477</td><td>1392</td><td>Oscillospiraceae;Provencibacterium massiliense;Harryflintia acetispora                                                           </td><td>216572;1841868;1849041                 </td><td>15</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 1110 × 17\n",
       "\\begin{tabular}{lllllllllllllllll}\n",
       " cluster\\_id & query & subject & subject\\_name & pident & length & mismatch & qstart & qend & sstart & send & evalue & slen & qlen & sscinames & staxids & pident\\_rank\\\\\n",
       " <int> & <fct> & <fct> & <fct> & <dbl> & <int> & <int> & <int> & <int> & <int> & <int> & <dbl> & <int> & <int> & <fct> & <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t 1 & 00dd2d7561b044c2bfeaee1e6eeb3445 & WP\\_122788428.1 & saccharopine dehydrogenase family protein {[}Intestinibacillus sp. Marseille-P6563{]} & 92.840 & 419 & 30 & 1 & 1257 & 1 & 419 & 0 & 419 & 1260 & Intestinibacillus sp. Marseille-P6563                                                                    & 2364792               & 1\\\\\n",
       "\t 1 & 00dd2d7561b044c2bfeaee1e6eeb3445 & WP\\_006572513.1 & saccharopine dehydrogenase family protein {[}Pseudoflavonifractor capillosus{]}       & 91.885 & 419 & 34 & 1 & 1257 & 1 & 419 & 0 & 419 & 1260 & Pseudoflavonifractor capillosus;Pseudoflavonifractor capillosus ATCC 29799;uncultured Flavonifractor sp. & 106588;411467;1193534 & 2\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t 40 & 012d2da3e8034d61a3424db553fd69b1 & WP\\_016317035.1 & MULTISPECIES: glycine--tRNA ligase {[}unclassified Anaerotruncus{]} & 78.618 & 463 & 99 & 1 & 1389 &  1 & 463 & 0 & 463 & 1392 & Anaerotruncus sp. G3(2012);Anaerotruncus sp. X29;Anaerotruncus sp. 1XD22-93;Anaerotruncus sp. 1XD42-93;unclassified Anaerotruncus & 1235835;2304567;2320082;2320853;2641626 & 29\\\\\n",
       "\t 40 & 012d2da3e8034d61a3424db553fd69b1 & WP\\_147322456.1 & MULTISPECIES: glycine--tRNA ligase {[}Ruminococcaceae{]}            & 80.778 & 463 & 84 & 1 & 1389 & 20 & 477 & 0 & 477 & 1392 & Oscillospiraceae;Provencibacterium massiliense;Harryflintia acetispora                                                            & 216572;1841868;1849041                  & 15\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 1110 × 17\n",
       "\n",
       "| cluster_id &lt;int&gt; | query &lt;fct&gt; | subject &lt;fct&gt; | subject_name &lt;fct&gt; | pident &lt;dbl&gt; | length &lt;int&gt; | mismatch &lt;int&gt; | qstart &lt;int&gt; | qend &lt;int&gt; | sstart &lt;int&gt; | send &lt;int&gt; | evalue &lt;dbl&gt; | slen &lt;int&gt; | qlen &lt;int&gt; | sscinames &lt;fct&gt; | staxids &lt;fct&gt; | pident_rank &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 00dd2d7561b044c2bfeaee1e6eeb3445 | WP_122788428.1 | saccharopine dehydrogenase family protein [Intestinibacillus sp. Marseille-P6563] | 92.840 | 419 | 30 | 1 | 1257 | 1 | 419 | 0 | 419 | 1260 | Intestinibacillus sp. Marseille-P6563                                                                    | 2364792               | 1 |\n",
       "| 1 | 00dd2d7561b044c2bfeaee1e6eeb3445 | WP_006572513.1 | saccharopine dehydrogenase family protein [Pseudoflavonifractor capillosus]       | 91.885 | 419 | 34 | 1 | 1257 | 1 | 419 | 0 | 419 | 1260 | Pseudoflavonifractor capillosus;Pseudoflavonifractor capillosus ATCC 29799;uncultured Flavonifractor sp. | 106588;411467;1193534 | 2 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| 40 | 012d2da3e8034d61a3424db553fd69b1 | WP_016317035.1 | MULTISPECIES: glycine--tRNA ligase [unclassified Anaerotruncus] | 78.618 | 463 | 99 | 1 | 1389 |  1 | 463 | 0 | 463 | 1392 | Anaerotruncus sp. G3(2012);Anaerotruncus sp. X29;Anaerotruncus sp. 1XD22-93;Anaerotruncus sp. 1XD42-93;unclassified Anaerotruncus | 1235835;2304567;2320082;2320853;2641626 | 29 |\n",
       "| 40 | 012d2da3e8034d61a3424db553fd69b1 | WP_147322456.1 | MULTISPECIES: glycine--tRNA ligase [Ruminococcaceae]            | 80.778 | 463 | 84 | 1 | 1389 | 20 | 477 | 0 | 477 | 1392 | Oscillospiraceae;Provencibacterium massiliense;Harryflintia acetispora                                                            | 216572;1841868;1849041                  | 15 |\n",
       "\n"
      ],
      "text/plain": [
       "     cluster_id query                            subject       \n",
       "1    1          00dd2d7561b044c2bfeaee1e6eeb3445 WP_122788428.1\n",
       "2    1          00dd2d7561b044c2bfeaee1e6eeb3445 WP_006572513.1\n",
       "⋮    ⋮          ⋮                                ⋮             \n",
       "1109 40         012d2da3e8034d61a3424db553fd69b1 WP_016317035.1\n",
       "1110 40         012d2da3e8034d61a3424db553fd69b1 WP_147322456.1\n",
       "     subject_name                                                                     \n",
       "1    saccharopine dehydrogenase family protein [Intestinibacillus sp. Marseille-P6563]\n",
       "2    saccharopine dehydrogenase family protein [Pseudoflavonifractor capillosus]      \n",
       "⋮    ⋮                                                                                \n",
       "1109 MULTISPECIES: glycine--tRNA ligase [unclassified Anaerotruncus]                  \n",
       "1110 MULTISPECIES: glycine--tRNA ligase [Ruminococcaceae]                             \n",
       "     pident length mismatch qstart qend sstart send evalue slen qlen\n",
       "1    92.840 419    30       1      1257 1      419  0      419  1260\n",
       "2    91.885 419    34       1      1257 1      419  0      419  1260\n",
       "⋮    ⋮      ⋮      ⋮        ⋮      ⋮    ⋮      ⋮    ⋮      ⋮    ⋮   \n",
       "1109 78.618 463    99       1      1389  1     463  0      463  1392\n",
       "1110 80.778 463    84       1      1389 20     477  0      477  1392\n",
       "     sscinames                                                                                                                        \n",
       "1    Intestinibacillus sp. Marseille-P6563                                                                                            \n",
       "2    Pseudoflavonifractor capillosus;Pseudoflavonifractor capillosus ATCC 29799;uncultured Flavonifractor sp.                         \n",
       "⋮    ⋮                                                                                                                                \n",
       "1109 Anaerotruncus sp. G3(2012);Anaerotruncus sp. X29;Anaerotruncus sp. 1XD22-93;Anaerotruncus sp. 1XD42-93;unclassified Anaerotruncus\n",
       "1110 Oscillospiraceae;Provencibacterium massiliense;Harryflintia acetispora                                                           \n",
       "     staxids                                 pident_rank\n",
       "1    2364792                                 1          \n",
       "2    106588;411467;1193534                   2          \n",
       "⋮    ⋮                                       ⋮          \n",
       "1109 1235835;2304567;2320082;2320853;2641626 29         \n",
       "1110 216572;1841868;1849041                  15         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# non-target cds gene annotations\n",
    "gene_nontarget = read.delim(file.path(work_dir, 'primers', 'cgp', 'nontarget', 'cds_blastx.tsv'), \n",
    "                        sep='\\t') %>%\n",
    "    semi_join(primer_info, c('cluster_id')) \n",
    "gene_nontarget"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Cluster taxonomic uniqueness"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-29T12:22:18.739609Z",
     "iopub.status.busy": "2021-04-29T12:22:18.737493Z",
     "iopub.status.idle": "2021-04-29T12:22:18.851590Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 1 × 8 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Min.</th><th scope=col>1st Qu.</th><th scope=col>Median</th><th scope=col>Mean</th><th scope=col>3rd Qu.</th><th scope=col>Max.</th><th scope=col>sd</th><th scope=col>sd_err_of_mean</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>seq-ID to closest non-target</th><td>85.022</td><td>95.53875</td><td>97.959</td><td>96.51311</td><td>99.35125</td><td>99.739</td><td>5.467</td><td>2.232</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 1 × 8 of type dbl\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & Min. & 1st Qu. & Median & Mean & 3rd Qu. & Max. & sd & sd\\_err\\_of\\_mean\\\\\n",
       "\\hline\n",
       "\tseq-ID to closest non-target & 85.022 & 95.53875 & 97.959 & 96.51311 & 99.35125 & 99.739 & 5.467 & 2.232\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 1 × 8 of type dbl\n",
       "\n",
       "| <!--/--> | Min. | 1st Qu. | Median | Mean | 3rd Qu. | Max. | sd | sd_err_of_mean |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| seq-ID to closest non-target | 85.022 | 95.53875 | 97.959 | 96.51311 | 99.35125 | 99.739 | 5.467 | 2.232 |\n",
       "\n"
      ],
      "text/plain": [
       "                             Min.   1st Qu.  Median Mean     3rd Qu.  Max.  \n",
       "seq-ID to closest non-target 85.022 95.53875 97.959 96.51311 99.35125 99.739\n",
       "                             sd    sd_err_of_mean\n",
       "seq-ID to closest non-target 5.467 2.232         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# closest seqID\n",
    "gene_nontarget %>%\n",
    "    filter(pident_rank == 1) %>%\n",
    "    .$pident %>% summary_x('seq-ID to closest non-target')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Primer quality"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-29T12:22:34.971510Z",
     "iopub.status.busy": "2021-04-29T12:22:34.969131Z",
     "iopub.status.idle": "2021-04-29T12:22:34.992475Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No. of unique primer sets: 297 \n",
      "No. of unique clusters: 37 \n"
     ]
    }
   ],
   "source": [
    "# summary\n",
    "primer_info %>% unique_n('primer sets', primer_set)\n",
    "primer_info %>% distinct(gene_type, cluster_id) %>% nrow %>% cat('No. of unique clusters:', ., '\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-29T12:22:38.717211Z",
     "iopub.status.busy": "2021-04-29T12:22:38.715584Z",
     "iopub.status.idle": "2021-04-29T12:22:38.780719Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 1 × 8 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Min.</th><th scope=col>1st Qu.</th><th scope=col>Median</th><th scope=col>Mean</th><th scope=col>3rd Qu.</th><th scope=col>Max.</th><th scope=col>sd</th><th scope=col>sd_err_of_mean</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>primer sets per cluster</th><td>3</td><td>10</td><td>10</td><td>9.540541</td><td>10</td><td>10</td><td>2.826</td><td>1.154</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 1 × 8 of type dbl\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & Min. & 1st Qu. & Median & Mean & 3rd Qu. & Max. & sd & sd\\_err\\_of\\_mean\\\\\n",
       "\\hline\n",
       "\tprimer sets per cluster & 3 & 10 & 10 & 9.540541 & 10 & 10 & 2.826 & 1.154\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 1 × 8 of type dbl\n",
       "\n",
       "| <!--/--> | Min. | 1st Qu. | Median | Mean | 3rd Qu. | Max. | sd | sd_err_of_mean |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| primer sets per cluster | 3 | 10 | 10 | 9.540541 | 10 | 10 | 2.826 | 1.154 |\n",
       "\n"
      ],
      "text/plain": [
       "                        Min. 1st Qu. Median Mean     3rd Qu. Max. sd   \n",
       "primer sets per cluster 3    10      10     9.540541 10      10   2.826\n",
       "                        sd_err_of_mean\n",
       "primer sets per cluster 1.154         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# primers per cluster\n",
    "primer_info %>%\n",
    "    distinct(cluster_id, primer_set) %>%\n",
    "    group_by(cluster_id) %>%\n",
    "    summarize(n_primer_sets = n(), .groups='drop') %>%\n",
    "    .$n_primer_sets %>% summary_x('primer sets per cluster')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-29T12:22:44.597967Z",
     "iopub.status.busy": "2021-04-29T12:22:44.595847Z",
     "iopub.status.idle": "2021-04-29T12:22:44.643882Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 1 × 8 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Min.</th><th scope=col>1st Qu.</th><th scope=col>Median</th><th scope=col>Mean</th><th scope=col>3rd Qu.</th><th scope=col>Max.</th><th scope=col>sd</th><th scope=col>sd_err_of_mean</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>mean amplicon size</th><td>100</td><td>125</td><td>163</td><td>167.134</td><td>208</td><td>250</td><td>54.44</td><td>22.225</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 1 × 8 of type dbl\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & Min. & 1st Qu. & Median & Mean & 3rd Qu. & Max. & sd & sd\\_err\\_of\\_mean\\\\\n",
       "\\hline\n",
       "\tmean amplicon size & 100 & 125 & 163 & 167.134 & 208 & 250 & 54.44 & 22.225\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 1 × 8 of type dbl\n",
       "\n",
       "| <!--/--> | Min. | 1st Qu. | Median | Mean | 3rd Qu. | Max. | sd | sd_err_of_mean |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| mean amplicon size | 100 | 125 | 163 | 167.134 | 208 | 250 | 54.44 | 22.225 |\n",
       "\n"
      ],
      "text/plain": [
       "                   Min. 1st Qu. Median Mean    3rd Qu. Max. sd   \n",
       "mean amplicon size 100  125     163    167.134 208     250  54.44\n",
       "                   sd_err_of_mean\n",
       "mean amplicon size 22.225        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 1 × 8 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Min.</th><th scope=col>1st Qu.</th><th scope=col>Median</th><th scope=col>Mean</th><th scope=col>3rd Qu.</th><th scope=col>Max.</th><th scope=col>sd</th><th scope=col>sd_err_of_mean</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>degeneracy</th><td>1</td><td>4</td><td>12</td><td>14.95467</td><td>16</td><td>96</td><td>35.785</td><td>14.609</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 1 × 8 of type dbl\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & Min. & 1st Qu. & Median & Mean & 3rd Qu. & Max. & sd & sd\\_err\\_of\\_mean\\\\\n",
       "\\hline\n",
       "\tdegeneracy & 1 & 4 & 12 & 14.95467 & 16 & 96 & 35.785 & 14.609\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 1 × 8 of type dbl\n",
       "\n",
       "| <!--/--> | Min. | 1st Qu. | Median | Mean | 3rd Qu. | Max. | sd | sd_err_of_mean |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| degeneracy | 1 | 4 | 12 | 14.95467 | 16 | 96 | 35.785 | 14.609 |\n",
       "\n"
      ],
      "text/plain": [
       "           Min. 1st Qu. Median Mean     3rd Qu. Max. sd     sd_err_of_mean\n",
       "degeneracy 1    4       12     14.95467 16      96   35.785 14.609        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 1 × 8 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Min.</th><th scope=col>1st Qu.</th><th scope=col>Median</th><th scope=col>Mean</th><th scope=col>3rd Qu.</th><th scope=col>Max.</th><th scope=col>sd</th><th scope=col>sd_err_of_mean</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>degeneracy (3-prime)</th><td>1</td><td>1</td><td>2</td><td>1.749292</td><td>2</td><td>6</td><td>1.874</td><td>0.765</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 1 × 8 of type dbl\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & Min. & 1st Qu. & Median & Mean & 3rd Qu. & Max. & sd & sd\\_err\\_of\\_mean\\\\\n",
       "\\hline\n",
       "\tdegeneracy (3-prime) & 1 & 1 & 2 & 1.749292 & 2 & 6 & 1.874 & 0.765\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 1 × 8 of type dbl\n",
       "\n",
       "| <!--/--> | Min. | 1st Qu. | Median | Mean | 3rd Qu. | Max. | sd | sd_err_of_mean |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| degeneracy (3-prime) | 1 | 1 | 2 | 1.749292 | 2 | 6 | 1.874 | 0.765 |\n",
       "\n"
      ],
      "text/plain": [
       "                     Min. 1st Qu. Median Mean     3rd Qu. Max. sd   \n",
       "degeneracy (3-prime) 1    1       2      1.749292 2       6    1.874\n",
       "                     sd_err_of_mean\n",
       "degeneracy (3-prime) 0.765         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# primer quality\n",
    "primer_info %>% filter(primer_type == 'PRIMER_LEFT') %>% .$amplicon_size_avg %>% summary_x('mean amplicon size')\n",
    "primer_info %>% .$degeneracy %>% summary_x('degeneracy')\n",
    "primer_info %>% .$degeneracy_3prime %>% summary_x('degeneracy (3-prime)')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-29T12:22:50.148534Z",
     "iopub.status.busy": "2021-04-29T12:22:50.146722Z",
     "iopub.status.idle": "2021-04-29T12:22:50.211294Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 866 × 24</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>gene_type</th><th scope=col>cluster_id</th><th scope=col>primer_set</th><th scope=col>amplicon_size_consensus</th><th scope=col>amplicon_size_avg</th><th scope=col>amplicon_size_sd</th><th scope=col>primer_id</th><th scope=col>primer_type</th><th scope=col>sequence</th><th scope=col>length</th><th scope=col>⋯</th><th scope=col>Tm_avg</th><th scope=col>Tm_sd</th><th scope=col>GC_avg</th><th scope=col>GC_sd</th><th scope=col>hairpin_avg</th><th scope=col>hairpin_sd</th><th scope=col>homodimer_avg</th><th scope=col>homodimer_sd</th><th scope=col>pident</th><th scope=col>sscinames</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>cds</td><td>1</td><td>116</td><td>100</td><td>100</td><td>0</td><td>116f</td><td>PRIMER_LEFT </td><td>ATYCAGTTYCTCARGGCRCT</td><td>20</td><td>⋯</td><td>59.82351</td><td>2.199735</td><td>50.0</td><td>5.000000</td><td>20.66486</td><td>20.92350</td><td>-9.250822</td><td>20.940934</td><td>92.84</td><td>Intestinibacillus sp. Marseille-P6563</td></tr>\n",
       "\t<tr><td>cds</td><td>1</td><td>116</td><td>100</td><td>100</td><td>0</td><td>116r</td><td>PRIMER_RIGHT</td><td>CGTCCTTSKTGCCGGTRAAR</td><td>20</td><td>⋯</td><td>61.88123</td><td>1.617130</td><td>57.5</td><td>4.330127</td><td>11.56401</td><td>17.22575</td><td> 5.615398</td><td> 5.903092</td><td>92.84</td><td>Intestinibacillus sp. Marseille-P6563</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋱</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>cds</td><td>40</td><td>973</td><td>158</td><td>158</td><td>0</td><td>973f</td><td>PRIMER_LEFT </td><td>GARTTCTTCTGCAAGCCKGG</td><td>20</td><td>⋯</td><td>59.83334</td><td>1.519704</td><td>55</td><td>3.535534</td><td>0.000000</td><td> 0.00000</td><td> -6.653505</td><td>12.39957</td><td>96.328</td><td>Intestinibacillus sp. Marseille-P6563</td></tr>\n",
       "\t<tr><td>cds</td><td>40</td><td>973</td><td>158</td><td>158</td><td>0</td><td>973r</td><td>PRIMER_RIGHT</td><td>TCKGTGGTWGCRTTGGAGTA</td><td>20</td><td>⋯</td><td>59.34475</td><td>1.864572</td><td>50</td><td>3.535534</td><td>5.098652</td><td>13.48977</td><td>-23.091678</td><td>15.40643</td><td>96.328</td><td>Intestinibacillus sp. Marseille-P6563</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 866 × 24\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " gene\\_type & cluster\\_id & primer\\_set & amplicon\\_size\\_consensus & amplicon\\_size\\_avg & amplicon\\_size\\_sd & primer\\_id & primer\\_type & sequence & length & ⋯ & Tm\\_avg & Tm\\_sd & GC\\_avg & GC\\_sd & hairpin\\_avg & hairpin\\_sd & homodimer\\_avg & homodimer\\_sd & pident & sscinames\\\\\n",
       " <chr> & <int> & <int> & <int> & <dbl> & <dbl> & <fct> & <fct> & <fct> & <int> & ⋯ & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <fct>\\\\\n",
       "\\hline\n",
       "\t cds & 1 & 116 & 100 & 100 & 0 & 116f & PRIMER\\_LEFT  & ATYCAGTTYCTCARGGCRCT & 20 & ⋯ & 59.82351 & 2.199735 & 50.0 & 5.000000 & 20.66486 & 20.92350 & -9.250822 & 20.940934 & 92.84 & Intestinibacillus sp. Marseille-P6563\\\\\n",
       "\t cds & 1 & 116 & 100 & 100 & 0 & 116r & PRIMER\\_RIGHT & CGTCCTTSKTGCCGGTRAAR & 20 & ⋯ & 61.88123 & 1.617130 & 57.5 & 4.330127 & 11.56401 & 17.22575 &  5.615398 &  5.903092 & 92.84 & Intestinibacillus sp. Marseille-P6563\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋱ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t cds & 40 & 973 & 158 & 158 & 0 & 973f & PRIMER\\_LEFT  & GARTTCTTCTGCAAGCCKGG & 20 & ⋯ & 59.83334 & 1.519704 & 55 & 3.535534 & 0.000000 &  0.00000 &  -6.653505 & 12.39957 & 96.328 & Intestinibacillus sp. Marseille-P6563\\\\\n",
       "\t cds & 40 & 973 & 158 & 158 & 0 & 973r & PRIMER\\_RIGHT & TCKGTGGTWGCRTTGGAGTA & 20 & ⋯ & 59.34475 & 1.864572 & 50 & 3.535534 & 5.098652 & 13.48977 & -23.091678 & 15.40643 & 96.328 & Intestinibacillus sp. Marseille-P6563\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 866 × 24\n",
       "\n",
       "| gene_type &lt;chr&gt; | cluster_id &lt;int&gt; | primer_set &lt;int&gt; | amplicon_size_consensus &lt;int&gt; | amplicon_size_avg &lt;dbl&gt; | amplicon_size_sd &lt;dbl&gt; | primer_id &lt;fct&gt; | primer_type &lt;fct&gt; | sequence &lt;fct&gt; | length &lt;int&gt; | ⋯ ⋯ | Tm_avg &lt;dbl&gt; | Tm_sd &lt;dbl&gt; | GC_avg &lt;dbl&gt; | GC_sd &lt;dbl&gt; | hairpin_avg &lt;dbl&gt; | hairpin_sd &lt;dbl&gt; | homodimer_avg &lt;dbl&gt; | homodimer_sd &lt;dbl&gt; | pident &lt;dbl&gt; | sscinames &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| cds | 1 | 116 | 100 | 100 | 0 | 116f | PRIMER_LEFT  | ATYCAGTTYCTCARGGCRCT | 20 | ⋯ | 59.82351 | 2.199735 | 50.0 | 5.000000 | 20.66486 | 20.92350 | -9.250822 | 20.940934 | 92.84 | Intestinibacillus sp. Marseille-P6563 |\n",
       "| cds | 1 | 116 | 100 | 100 | 0 | 116r | PRIMER_RIGHT | CGTCCTTSKTGCCGGTRAAR | 20 | ⋯ | 61.88123 | 1.617130 | 57.5 | 4.330127 | 11.56401 | 17.22575 |  5.615398 |  5.903092 | 92.84 | Intestinibacillus sp. Marseille-P6563 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋱ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| cds | 40 | 973 | 158 | 158 | 0 | 973f | PRIMER_LEFT  | GARTTCTTCTGCAAGCCKGG | 20 | ⋯ | 59.83334 | 1.519704 | 55 | 3.535534 | 0.000000 |  0.00000 |  -6.653505 | 12.39957 | 96.328 | Intestinibacillus sp. Marseille-P6563 |\n",
       "| cds | 40 | 973 | 158 | 158 | 0 | 973r | PRIMER_RIGHT | TCKGTGGTWGCRTTGGAGTA | 20 | ⋯ | 59.34475 | 1.864572 | 50 | 3.535534 | 5.098652 | 13.48977 | -23.091678 | 15.40643 | 96.328 | Intestinibacillus sp. Marseille-P6563 |\n",
       "\n"
      ],
      "text/plain": [
       "    gene_type cluster_id primer_set amplicon_size_consensus amplicon_size_avg\n",
       "1   cds       1          116        100                     100              \n",
       "2   cds       1          116        100                     100              \n",
       "⋮   ⋮         ⋮          ⋮          ⋮                       ⋮                \n",
       "865 cds       40         973        158                     158              \n",
       "866 cds       40         973        158                     158              \n",
       "    amplicon_size_sd primer_id primer_type  sequence             length ⋯\n",
       "1   0                116f      PRIMER_LEFT  ATYCAGTTYCTCARGGCRCT 20     ⋯\n",
       "2   0                116r      PRIMER_RIGHT CGTCCTTSKTGCCGGTRAAR 20     ⋯\n",
       "⋮   ⋮                ⋮         ⋮            ⋮                    ⋮      ⋱\n",
       "865 0                973f      PRIMER_LEFT  GARTTCTTCTGCAAGCCKGG 20     ⋯\n",
       "866 0                973r      PRIMER_RIGHT TCKGTGGTWGCRTTGGAGTA 20     ⋯\n",
       "    Tm_avg   Tm_sd    GC_avg GC_sd    hairpin_avg hairpin_sd homodimer_avg\n",
       "1   59.82351 2.199735 50.0   5.000000 20.66486    20.92350   -9.250822    \n",
       "2   61.88123 1.617130 57.5   4.330127 11.56401    17.22575    5.615398    \n",
       "⋮   ⋮        ⋮        ⋮      ⋮        ⋮           ⋮          ⋮            \n",
       "865 59.83334 1.519704 55     3.535534 0.000000     0.00000    -6.653505   \n",
       "866 59.34475 1.864572 50     3.535534 5.098652    13.48977   -23.091678   \n",
       "    homodimer_sd pident sscinames                            \n",
       "1   20.940934    92.84  Intestinibacillus sp. Marseille-P6563\n",
       "2    5.903092    92.84  Intestinibacillus sp. Marseille-P6563\n",
       "⋮   ⋮            ⋮      ⋮                                    \n",
       "865 12.39957     96.328 Intestinibacillus sp. Marseille-P6563\n",
       "866 15.40643     96.328 Intestinibacillus sp. Marseille-P6563"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# adding non-target pident info\n",
    "tmp1 = gene_nontarget %>%\n",
    "    filter(pident_rank == 1) %>%\n",
    "    distinct(cluster_id, pident, sscinames) %>%\n",
    "    mutate(gene_type = 'cds')\n",
    "\n",
    "primer_info_j = primer_info %>%\n",
    "    inner_join(tmp1, c('gene_type', 'cluster_id'))\n",
    "primer_info_j"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-29T12:22:54.610881Z",
     "iopub.status.busy": "2021-04-29T12:22:54.608595Z",
     "iopub.status.idle": "2021-04-29T12:22:54.733745Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A tibble: 30 × 26</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>gene_type</th><th scope=col>cluster_id</th><th scope=col>primer_set</th><th scope=col>amplicon_size_consensus</th><th scope=col>amplicon_size_avg</th><th scope=col>amplicon_size_sd</th><th scope=col>primer_id</th><th scope=col>primer_type</th><th scope=col>sequence</th><th scope=col>length</th><th scope=col>degeneracy</th><th scope=col>degeneracy_3prime</th><th scope=col>position_start</th><th scope=col>position_end</th><th scope=col>Tm_avg</th><th scope=col>Tm_sd</th><th scope=col>GC_avg</th><th scope=col>GC_sd</th><th scope=col>hairpin_avg</th><th scope=col>hairpin_sd</th><th scope=col>homodimer_avg</th><th scope=col>homodimer_sd</th><th scope=col>pident</th><th scope=col>sscinames</th><th scope=col>max_degeneracy_3prime</th><th scope=col>max_degeneracy</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>cds</td><td>26</td><td>540</td><td>193</td><td>193</td><td>0</td><td>540f</td><td>PRIMER_LEFT </td><td>GGYCTGACBATGGCGGAGTA </td><td>20</td><td> 6</td><td>1</td><td> 681</td><td> 701</td><td>62.39508</td><td>1.999740</td><td>60.83333</td><td>3.435921</td><td>39.17997</td><td>18.1052739</td><td>  3.475140</td><td>12.642244</td><td>85.022</td><td>Oscillospiraceae bacterium           </td><td>1</td><td>16</td></tr>\n",
       "\t<tr><td>cds</td><td>26</td><td>540</td><td>193</td><td>193</td><td>0</td><td>540r</td><td>PRIMER_RIGHT</td><td>TSGTCGAKGTRATWCGCTCC </td><td>20</td><td>16</td><td>1</td><td> 854</td><td> 874</td><td>60.26164</td><td>1.509761</td><td>55.00000</td><td>3.535534</td><td>29.23107</td><td>16.9285589</td><td>-12.584709</td><td>19.188754</td><td>85.022</td><td>Oscillospiraceae bacterium           </td><td>1</td><td>16</td></tr>\n",
       "\t<tr><td>cds</td><td>26</td><td>753</td><td>100</td><td>100</td><td>0</td><td>753f</td><td>PRIMER_LEFT </td><td>GGYAARATCGGTCTGTTCGG </td><td>20</td><td> 4</td><td>1</td><td> 441</td><td> 461</td><td>59.45219</td><td>1.761779</td><td>55.00000</td><td>3.535534</td><td>17.29587</td><td>17.2958679</td><td>-21.335645</td><td>17.154538</td><td>85.022</td><td>Oscillospiraceae bacterium           </td><td>2</td><td>16</td></tr>\n",
       "\t<tr><td>cds</td><td>26</td><td>753</td><td>100</td><td>100</td><td>0</td><td>753r</td><td>PRIMER_RIGHT</td><td>ATACCGARATRCCRCCRTGC </td><td>20</td><td>16</td><td>2</td><td> 521</td><td> 541</td><td>61.19424</td><td>2.563790</td><td>55.00000</td><td>5.000000</td><td> 0.00000</td><td> 0.0000000</td><td>-28.347864</td><td>12.236336</td><td>85.022</td><td>Oscillospiraceae bacterium           </td><td>2</td><td>16</td></tr>\n",
       "\t<tr><td>cds</td><td>26</td><td>488</td><td>181</td><td>181</td><td>0</td><td>488f</td><td>PRIMER_LEFT </td><td>AGCAYGGYGGYATYTCGGTA </td><td>20</td><td>16</td><td>1</td><td> 520</td><td> 540</td><td>62.28327</td><td>2.588591</td><td>55.00000</td><td>5.000000</td><td>18.04463</td><td>18.0446265</td><td>  7.007019</td><td>12.426061</td><td>85.022</td><td>Oscillospiraceae bacterium           </td><td>4</td><td>64</td></tr>\n",
       "\t<tr><td>cds</td><td>26</td><td>488</td><td>181</td><td>181</td><td>0</td><td>488r</td><td>PRIMER_RIGHT</td><td>TACTCCGCCATVGTCAGRCC </td><td>20</td><td> 6</td><td>2</td><td> 681</td><td> 701</td><td>62.39508</td><td>1.999740</td><td>60.83333</td><td>3.435921</td><td>31.65677</td><td>22.6781788</td><td> -2.394768</td><td>12.763173</td><td>85.022</td><td>Oscillospiraceae bacterium           </td><td>4</td><td>64</td></tr>\n",
       "\t<tr><td>cds</td><td>19</td><td>772</td><td>176</td><td>176</td><td>0</td><td>772f</td><td>PRIMER_LEFT </td><td>CGYCGYTACGGCTTCCACG  </td><td>19</td><td> 4</td><td>1</td><td> 525</td><td> 544</td><td>64.53412</td><td>1.767244</td><td>68.42105</td><td>3.721615</td><td>55.95397</td><td>10.0674735</td><td> 19.455128</td><td>14.757455</td><td>87.097</td><td>Intestinibacillus sp. Marseille-P6563</td><td>1</td><td> 8</td></tr>\n",
       "\t<tr><td>cds</td><td>19</td><td>772</td><td>176</td><td>176</td><td>0</td><td>772r</td><td>PRIMER_RIGHT</td><td>AGMGGRGTRAAGCCCATGGA </td><td>20</td><td> 8</td><td>1</td><td> 681</td><td> 701</td><td>62.68928</td><td>2.495141</td><td>57.50000</td><td>4.330127</td><td>30.14613</td><td>30.1461276</td><td> 15.601331</td><td> 0.000000</td><td>87.097</td><td>Intestinibacillus sp. Marseille-P6563</td><td>1</td><td> 8</td></tr>\n",
       "\t<tr><td>cds</td><td>19</td><td> 34</td><td>163</td><td>163</td><td>0</td><td>34f </td><td>PRIMER_LEFT </td><td>TCCATGGGCTTYACYCCKCT </td><td>20</td><td> 8</td><td>2</td><td> 681</td><td> 701</td><td>62.68928</td><td>2.495141</td><td>57.50000</td><td>4.330127</td><td>31.15315</td><td>31.1531548</td><td> 13.372744</td><td> 0.000000</td><td>87.097</td><td>Intestinibacillus sp. Marseille-P6563</td><td>2</td><td> 8</td></tr>\n",
       "\t<tr><td>cds</td><td>19</td><td> 34</td><td>163</td><td>163</td><td>0</td><td>34r </td><td>PRIMER_RIGHT</td><td>AYACRCCGGAGATACCCAGC </td><td>20</td><td> 4</td><td>1</td><td> 824</td><td> 844</td><td>62.21885</td><td>2.037360</td><td>60.00000</td><td>3.535534</td><td>34.89652</td><td> 0.0000000</td><td> 11.971233</td><td> 2.239915</td><td>87.097</td><td>Intestinibacillus sp. Marseille-P6563</td><td>2</td><td> 8</td></tr>\n",
       "\t<tr><td>cds</td><td>19</td><td>164</td><td>165</td><td>165</td><td>0</td><td>164f</td><td>PRIMER_LEFT </td><td>TCCATGGGCTTYACYCCKCT </td><td>20</td><td> 8</td><td>2</td><td> 681</td><td> 701</td><td>62.68928</td><td>2.495141</td><td>57.50000</td><td>4.330127</td><td>31.15315</td><td>31.1531548</td><td> 13.372744</td><td> 0.000000</td><td>87.097</td><td>Intestinibacillus sp. Marseille-P6563</td><td>2</td><td> 8</td></tr>\n",
       "\t<tr><td>cds</td><td>19</td><td>164</td><td>165</td><td>165</td><td>0</td><td>164r</td><td>PRIMER_RIGHT</td><td>GGAYACRCCGGAGATACCCA </td><td>20</td><td> 4</td><td>1</td><td> 826</td><td> 846</td><td>61.58367</td><td>2.056713</td><td>60.00000</td><td>3.535534</td><td>34.89652</td><td> 0.0000000</td><td> 11.971233</td><td> 2.239915</td><td>87.097</td><td>Intestinibacillus sp. Marseille-P6563</td><td>2</td><td> 8</td></tr>\n",
       "\t<tr><td>cds</td><td>19</td><td>581</td><td>164</td><td>164</td><td>0</td><td>581f</td><td>PRIMER_LEFT </td><td>CATGGGCTTYACYCCKCTGG </td><td>20</td><td> 8</td><td>2</td><td> 683</td><td> 703</td><td>62.46956</td><td>2.419663</td><td>62.50000</td><td>4.330127</td><td>31.15315</td><td>31.1531548</td><td>-17.549881</td><td>14.534374</td><td>87.097</td><td>Intestinibacillus sp. Marseille-P6563</td><td>2</td><td> 8</td></tr>\n",
       "\t<tr><td>cds</td><td>19</td><td>581</td><td>164</td><td>164</td><td>0</td><td>581r</td><td>PRIMER_RIGHT</td><td>AGGAYACRCCGGAGATACCC </td><td>20</td><td> 4</td><td>1</td><td> 827</td><td> 847</td><td>61.29900</td><td>2.066468</td><td>60.00000</td><td>3.535534</td><td>44.32892</td><td> 0.6361774</td><td> 11.971233</td><td> 2.239915</td><td>87.097</td><td>Intestinibacillus sp. Marseille-P6563</td><td>2</td><td> 8</td></tr>\n",
       "\t<tr><td>cds</td><td>19</td><td>582</td><td>165</td><td>165</td><td>0</td><td>582f</td><td>PRIMER_LEFT </td><td>CCATGGGCTTYACYCCKCTG </td><td>20</td><td> 8</td><td>2</td><td> 682</td><td> 702</td><td>62.46956</td><td>2.419663</td><td>62.50000</td><td>4.330127</td><td>31.15315</td><td>31.1531548</td><td>  3.232155</td><td> 0.000000</td><td>87.097</td><td>Intestinibacillus sp. Marseille-P6563</td><td>2</td><td> 8</td></tr>\n",
       "\t<tr><td>cds</td><td>19</td><td>582</td><td>165</td><td>165</td><td>0</td><td>582r</td><td>PRIMER_RIGHT</td><td>AGGAYACRCCGGAGATACCC </td><td>20</td><td> 4</td><td>1</td><td> 827</td><td> 847</td><td>61.29900</td><td>2.066468</td><td>60.00000</td><td>3.535534</td><td>44.32892</td><td> 0.6361774</td><td> 11.971233</td><td> 2.239915</td><td>87.097</td><td>Intestinibacillus sp. Marseille-P6563</td><td>2</td><td> 8</td></tr>\n",
       "\t<tr><td>cds</td><td>19</td><td>803</td><td>247</td><td>247</td><td>0</td><td>803f</td><td>PRIMER_LEFT </td><td>TGGGTATCTCCGGYGTRTCC </td><td>20</td><td> 4</td><td>2</td><td> 826</td><td> 846</td><td>61.58367</td><td>2.056713</td><td>60.00000</td><td>3.535534</td><td>37.24153</td><td> 0.0000000</td><td>  2.634518</td><td> 3.597188</td><td>87.097</td><td>Intestinibacillus sp. Marseille-P6563</td><td>2</td><td> 8</td></tr>\n",
       "\t<tr><td>cds</td><td>19</td><td>803</td><td>247</td><td>247</td><td>0</td><td>803r</td><td>PRIMER_RIGHT</td><td>TCSGGRTCRATCTTGATGCC </td><td>20</td><td> 8</td><td>1</td><td>1053</td><td>1073</td><td>60.54531</td><td>1.524997</td><td>55.00000</td><td>3.535534</td><td>39.05910</td><td>16.3276701</td><td>  9.466837</td><td>13.740396</td><td>87.097</td><td>Intestinibacillus sp. Marseille-P6563</td><td>2</td><td> 8</td></tr>\n",
       "\t<tr><td>cds</td><td>19</td><td>936</td><td>248</td><td>248</td><td>0</td><td>936f</td><td>PRIMER_LEFT </td><td>TGGGTATCTCCGGYGTRTCC </td><td>20</td><td> 4</td><td>2</td><td> 826</td><td> 846</td><td>61.58367</td><td>2.056713</td><td>60.00000</td><td>3.535534</td><td>37.24153</td><td> 0.0000000</td><td>  2.634518</td><td> 3.597188</td><td>87.097</td><td>Intestinibacillus sp. Marseille-P6563</td><td>2</td><td> 8</td></tr>\n",
       "\t<tr><td>cds</td><td>19</td><td>936</td><td>248</td><td>248</td><td>0</td><td>936r</td><td>PRIMER_RIGHT</td><td>CTCSGGRTCRATCTTGATGC </td><td>20</td><td> 8</td><td>1</td><td>1054</td><td>1074</td><td>59.13184</td><td>1.503151</td><td>55.00000</td><td>3.535534</td><td>37.90776</td><td>15.0985218</td><td>  6.504147</td><td>12.138283</td><td>87.097</td><td>Intestinibacillus sp. Marseille-P6563</td><td>2</td><td> 8</td></tr>\n",
       "\t<tr><td>cds</td><td>19</td><td>290</td><td>163</td><td>163</td><td>0</td><td>290f</td><td>PRIMER_LEFT </td><td>CATGGGCTTYACYCCKCTGG </td><td>20</td><td> 8</td><td>2</td><td> 683</td><td> 703</td><td>62.46956</td><td>2.419663</td><td>62.50000</td><td>4.330127</td><td>31.15315</td><td>31.1531548</td><td>-17.549881</td><td>14.534374</td><td>87.097</td><td>Intestinibacillus sp. Marseille-P6563</td><td>2</td><td>16</td></tr>\n",
       "\t<tr><td>cds</td><td>19</td><td>290</td><td>163</td><td>163</td><td>0</td><td>290r</td><td>PRIMER_RIGHT</td><td>GGAYACRCCGGAGATACCCA </td><td>20</td><td> 4</td><td>1</td><td> 826</td><td> 846</td><td>61.58367</td><td>2.056713</td><td>60.00000</td><td>3.535534</td><td>34.89652</td><td> 0.0000000</td><td> 11.971233</td><td> 2.239915</td><td>87.097</td><td>Intestinibacillus sp. Marseille-P6563</td><td>2</td><td>16</td></tr>\n",
       "\t<tr><td>cds</td><td>19</td><td>291</td><td>164</td><td>164</td><td>0</td><td>291f</td><td>PRIMER_LEFT </td><td>CCATGGGCTTYACYCCKCTG </td><td>20</td><td> 8</td><td>2</td><td> 682</td><td> 702</td><td>62.46956</td><td>2.419663</td><td>62.50000</td><td>4.330127</td><td>31.15315</td><td>31.1531548</td><td>  3.232155</td><td> 0.000000</td><td>87.097</td><td>Intestinibacillus sp. Marseille-P6563</td><td>4</td><td>16</td></tr>\n",
       "\t<tr><td>cds</td><td>19</td><td>291</td><td>164</td><td>164</td><td>0</td><td>291r</td><td>PRIMER_RIGHT</td><td>GGAYACRCCGGAGATACCCA </td><td>20</td><td> 4</td><td>1</td><td> 826</td><td> 846</td><td>61.58367</td><td>2.056713</td><td>60.00000</td><td>3.535534</td><td>34.89652</td><td> 0.0000000</td><td> 11.971233</td><td> 2.239915</td><td>87.097</td><td>Intestinibacillus sp. Marseille-P6563</td><td>4</td><td>16</td></tr>\n",
       "\t<tr><td>cds</td><td>19</td><td>955</td><td>162</td><td>162</td><td>0</td><td>955f</td><td>PRIMER_LEFT </td><td>TCCATGGGCTTYACYCCKCT </td><td>20</td><td> 8</td><td>2</td><td> 681</td><td> 701</td><td>62.68928</td><td>2.495141</td><td>57.50000</td><td>4.330127</td><td>31.15315</td><td>31.1531548</td><td> 13.372744</td><td> 0.000000</td><td>87.097</td><td>Intestinibacillus sp. Marseille-P6563</td><td>4</td><td>24</td></tr>\n",
       "\t<tr><td>cds</td><td>19</td><td>955</td><td>162</td><td>162</td><td>0</td><td>955r</td><td>PRIMER_RIGHT</td><td>YACRCCGGAGATACCCAGC  </td><td>19</td><td> 4</td><td>1</td><td> 824</td><td> 843</td><td>61.15701</td><td>1.693536</td><td>63.15789</td><td>3.721615</td><td>34.89652</td><td> 0.0000000</td><td> 11.971233</td><td> 2.239915</td><td>87.097</td><td>Intestinibacillus sp. Marseille-P6563</td><td>4</td><td>24</td></tr>\n",
       "\t<tr><td>cds</td><td>17</td><td>129</td><td>132</td><td>132</td><td>0</td><td>129f</td><td>PRIMER_LEFT </td><td>GMTGGARTTTCTSCGTCAGGT</td><td>21</td><td> 8</td><td>1</td><td>  83</td><td> 104</td><td>60.99081</td><td>1.609124</td><td>52.38095</td><td>3.367175</td><td>43.16746</td><td> 9.8501469</td><td>  7.643433</td><td> 8.130394</td><td>87.879</td><td>Intestinibacillus sp. Marseille-P6563</td><td>2</td><td>16</td></tr>\n",
       "\t<tr><td>cds</td><td>17</td><td>129</td><td>132</td><td>132</td><td>0</td><td>129r</td><td>PRIMER_RIGHT</td><td>AKGCACATCGGCARRTGRTC </td><td>20</td><td>16</td><td>2</td><td> 195</td><td> 215</td><td>61.58052</td><td>1.966489</td><td>55.00000</td><td>5.000000</td><td>48.67702</td><td> 3.6646308</td><td>  1.702271</td><td>21.058124</td><td>87.879</td><td>Intestinibacillus sp. Marseille-P6563</td><td>2</td><td>16</td></tr>\n",
       "\t<tr><td>cds</td><td>17</td><td>352</td><td>133</td><td>133</td><td>0</td><td>352f</td><td>PRIMER_LEFT </td><td>AGMTGGARTTTCTSCGTCAGG</td><td>21</td><td> 8</td><td>1</td><td>  82</td><td> 103</td><td>60.71966</td><td>1.614867</td><td>52.38095</td><td>3.367175</td><td>45.60811</td><td> 8.7545325</td><td>  6.226785</td><td>19.084320</td><td>87.879</td><td>Intestinibacillus sp. Marseille-P6563</td><td>2</td><td>16</td></tr>\n",
       "\t<tr><td>cds</td><td>17</td><td>352</td><td>133</td><td>133</td><td>0</td><td>352r</td><td>PRIMER_RIGHT</td><td>AKGCACATCGGCARRTGRTC </td><td>20</td><td>16</td><td>2</td><td> 195</td><td> 215</td><td>61.58052</td><td>1.966489</td><td>55.00000</td><td>5.000000</td><td>48.67702</td><td> 3.6646308</td><td>  1.702271</td><td>21.058124</td><td>87.879</td><td>Intestinibacillus sp. Marseille-P6563</td><td>2</td><td>16</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 30 × 26\n",
       "\\begin{tabular}{llllllllllllllllllllllllll}\n",
       " gene\\_type & cluster\\_id & primer\\_set & amplicon\\_size\\_consensus & amplicon\\_size\\_avg & amplicon\\_size\\_sd & primer\\_id & primer\\_type & sequence & length & degeneracy & degeneracy\\_3prime & position\\_start & position\\_end & Tm\\_avg & Tm\\_sd & GC\\_avg & GC\\_sd & hairpin\\_avg & hairpin\\_sd & homodimer\\_avg & homodimer\\_sd & pident & sscinames & max\\_degeneracy\\_3prime & max\\_degeneracy\\\\\n",
       " <chr> & <int> & <int> & <int> & <dbl> & <dbl> & <fct> & <fct> & <fct> & <int> & <int> & <int> & <int> & <int> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <fct> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t cds & 26 & 540 & 193 & 193 & 0 & 540f & PRIMER\\_LEFT  & GGYCTGACBATGGCGGAGTA  & 20 &  6 & 1 &  681 &  701 & 62.39508 & 1.999740 & 60.83333 & 3.435921 & 39.17997 & 18.1052739 &   3.475140 & 12.642244 & 85.022 & Oscillospiraceae bacterium            & 1 & 16\\\\\n",
       "\t cds & 26 & 540 & 193 & 193 & 0 & 540r & PRIMER\\_RIGHT & TSGTCGAKGTRATWCGCTCC  & 20 & 16 & 1 &  854 &  874 & 60.26164 & 1.509761 & 55.00000 & 3.535534 & 29.23107 & 16.9285589 & -12.584709 & 19.188754 & 85.022 & Oscillospiraceae bacterium            & 1 & 16\\\\\n",
       "\t cds & 26 & 753 & 100 & 100 & 0 & 753f & PRIMER\\_LEFT  & GGYAARATCGGTCTGTTCGG  & 20 &  4 & 1 &  441 &  461 & 59.45219 & 1.761779 & 55.00000 & 3.535534 & 17.29587 & 17.2958679 & -21.335645 & 17.154538 & 85.022 & Oscillospiraceae bacterium            & 2 & 16\\\\\n",
       "\t cds & 26 & 753 & 100 & 100 & 0 & 753r & PRIMER\\_RIGHT & ATACCGARATRCCRCCRTGC  & 20 & 16 & 2 &  521 &  541 & 61.19424 & 2.563790 & 55.00000 & 5.000000 &  0.00000 &  0.0000000 & -28.347864 & 12.236336 & 85.022 & Oscillospiraceae bacterium            & 2 & 16\\\\\n",
       "\t cds & 26 & 488 & 181 & 181 & 0 & 488f & PRIMER\\_LEFT  & AGCAYGGYGGYATYTCGGTA  & 20 & 16 & 1 &  520 &  540 & 62.28327 & 2.588591 & 55.00000 & 5.000000 & 18.04463 & 18.0446265 &   7.007019 & 12.426061 & 85.022 & Oscillospiraceae bacterium            & 4 & 64\\\\\n",
       "\t cds & 26 & 488 & 181 & 181 & 0 & 488r & PRIMER\\_RIGHT & TACTCCGCCATVGTCAGRCC  & 20 &  6 & 2 &  681 &  701 & 62.39508 & 1.999740 & 60.83333 & 3.435921 & 31.65677 & 22.6781788 &  -2.394768 & 12.763173 & 85.022 & Oscillospiraceae bacterium            & 4 & 64\\\\\n",
       "\t cds & 19 & 772 & 176 & 176 & 0 & 772f & PRIMER\\_LEFT  & CGYCGYTACGGCTTCCACG   & 19 &  4 & 1 &  525 &  544 & 64.53412 & 1.767244 & 68.42105 & 3.721615 & 55.95397 & 10.0674735 &  19.455128 & 14.757455 & 87.097 & Intestinibacillus sp. Marseille-P6563 & 1 &  8\\\\\n",
       "\t cds & 19 & 772 & 176 & 176 & 0 & 772r & PRIMER\\_RIGHT & AGMGGRGTRAAGCCCATGGA  & 20 &  8 & 1 &  681 &  701 & 62.68928 & 2.495141 & 57.50000 & 4.330127 & 30.14613 & 30.1461276 &  15.601331 &  0.000000 & 87.097 & Intestinibacillus sp. Marseille-P6563 & 1 &  8\\\\\n",
       "\t cds & 19 &  34 & 163 & 163 & 0 & 34f  & PRIMER\\_LEFT  & TCCATGGGCTTYACYCCKCT  & 20 &  8 & 2 &  681 &  701 & 62.68928 & 2.495141 & 57.50000 & 4.330127 & 31.15315 & 31.1531548 &  13.372744 &  0.000000 & 87.097 & Intestinibacillus sp. Marseille-P6563 & 2 &  8\\\\\n",
       "\t cds & 19 &  34 & 163 & 163 & 0 & 34r  & PRIMER\\_RIGHT & AYACRCCGGAGATACCCAGC  & 20 &  4 & 1 &  824 &  844 & 62.21885 & 2.037360 & 60.00000 & 3.535534 & 34.89652 &  0.0000000 &  11.971233 &  2.239915 & 87.097 & Intestinibacillus sp. Marseille-P6563 & 2 &  8\\\\\n",
       "\t cds & 19 & 164 & 165 & 165 & 0 & 164f & PRIMER\\_LEFT  & TCCATGGGCTTYACYCCKCT  & 20 &  8 & 2 &  681 &  701 & 62.68928 & 2.495141 & 57.50000 & 4.330127 & 31.15315 & 31.1531548 &  13.372744 &  0.000000 & 87.097 & Intestinibacillus sp. Marseille-P6563 & 2 &  8\\\\\n",
       "\t cds & 19 & 164 & 165 & 165 & 0 & 164r & PRIMER\\_RIGHT & GGAYACRCCGGAGATACCCA  & 20 &  4 & 1 &  826 &  846 & 61.58367 & 2.056713 & 60.00000 & 3.535534 & 34.89652 &  0.0000000 &  11.971233 &  2.239915 & 87.097 & Intestinibacillus sp. Marseille-P6563 & 2 &  8\\\\\n",
       "\t cds & 19 & 581 & 164 & 164 & 0 & 581f & PRIMER\\_LEFT  & CATGGGCTTYACYCCKCTGG  & 20 &  8 & 2 &  683 &  703 & 62.46956 & 2.419663 & 62.50000 & 4.330127 & 31.15315 & 31.1531548 & -17.549881 & 14.534374 & 87.097 & Intestinibacillus sp. Marseille-P6563 & 2 &  8\\\\\n",
       "\t cds & 19 & 581 & 164 & 164 & 0 & 581r & PRIMER\\_RIGHT & AGGAYACRCCGGAGATACCC  & 20 &  4 & 1 &  827 &  847 & 61.29900 & 2.066468 & 60.00000 & 3.535534 & 44.32892 &  0.6361774 &  11.971233 &  2.239915 & 87.097 & Intestinibacillus sp. Marseille-P6563 & 2 &  8\\\\\n",
       "\t cds & 19 & 582 & 165 & 165 & 0 & 582f & PRIMER\\_LEFT  & CCATGGGCTTYACYCCKCTG  & 20 &  8 & 2 &  682 &  702 & 62.46956 & 2.419663 & 62.50000 & 4.330127 & 31.15315 & 31.1531548 &   3.232155 &  0.000000 & 87.097 & Intestinibacillus sp. Marseille-P6563 & 2 &  8\\\\\n",
       "\t cds & 19 & 582 & 165 & 165 & 0 & 582r & PRIMER\\_RIGHT & AGGAYACRCCGGAGATACCC  & 20 &  4 & 1 &  827 &  847 & 61.29900 & 2.066468 & 60.00000 & 3.535534 & 44.32892 &  0.6361774 &  11.971233 &  2.239915 & 87.097 & Intestinibacillus sp. Marseille-P6563 & 2 &  8\\\\\n",
       "\t cds & 19 & 803 & 247 & 247 & 0 & 803f & PRIMER\\_LEFT  & TGGGTATCTCCGGYGTRTCC  & 20 &  4 & 2 &  826 &  846 & 61.58367 & 2.056713 & 60.00000 & 3.535534 & 37.24153 &  0.0000000 &   2.634518 &  3.597188 & 87.097 & Intestinibacillus sp. Marseille-P6563 & 2 &  8\\\\\n",
       "\t cds & 19 & 803 & 247 & 247 & 0 & 803r & PRIMER\\_RIGHT & TCSGGRTCRATCTTGATGCC  & 20 &  8 & 1 & 1053 & 1073 & 60.54531 & 1.524997 & 55.00000 & 3.535534 & 39.05910 & 16.3276701 &   9.466837 & 13.740396 & 87.097 & Intestinibacillus sp. Marseille-P6563 & 2 &  8\\\\\n",
       "\t cds & 19 & 936 & 248 & 248 & 0 & 936f & PRIMER\\_LEFT  & TGGGTATCTCCGGYGTRTCC  & 20 &  4 & 2 &  826 &  846 & 61.58367 & 2.056713 & 60.00000 & 3.535534 & 37.24153 &  0.0000000 &   2.634518 &  3.597188 & 87.097 & Intestinibacillus sp. Marseille-P6563 & 2 &  8\\\\\n",
       "\t cds & 19 & 936 & 248 & 248 & 0 & 936r & PRIMER\\_RIGHT & CTCSGGRTCRATCTTGATGC  & 20 &  8 & 1 & 1054 & 1074 & 59.13184 & 1.503151 & 55.00000 & 3.535534 & 37.90776 & 15.0985218 &   6.504147 & 12.138283 & 87.097 & Intestinibacillus sp. Marseille-P6563 & 2 &  8\\\\\n",
       "\t cds & 19 & 290 & 163 & 163 & 0 & 290f & PRIMER\\_LEFT  & CATGGGCTTYACYCCKCTGG  & 20 &  8 & 2 &  683 &  703 & 62.46956 & 2.419663 & 62.50000 & 4.330127 & 31.15315 & 31.1531548 & -17.549881 & 14.534374 & 87.097 & Intestinibacillus sp. Marseille-P6563 & 2 & 16\\\\\n",
       "\t cds & 19 & 290 & 163 & 163 & 0 & 290r & PRIMER\\_RIGHT & GGAYACRCCGGAGATACCCA  & 20 &  4 & 1 &  826 &  846 & 61.58367 & 2.056713 & 60.00000 & 3.535534 & 34.89652 &  0.0000000 &  11.971233 &  2.239915 & 87.097 & Intestinibacillus sp. Marseille-P6563 & 2 & 16\\\\\n",
       "\t cds & 19 & 291 & 164 & 164 & 0 & 291f & PRIMER\\_LEFT  & CCATGGGCTTYACYCCKCTG  & 20 &  8 & 2 &  682 &  702 & 62.46956 & 2.419663 & 62.50000 & 4.330127 & 31.15315 & 31.1531548 &   3.232155 &  0.000000 & 87.097 & Intestinibacillus sp. Marseille-P6563 & 4 & 16\\\\\n",
       "\t cds & 19 & 291 & 164 & 164 & 0 & 291r & PRIMER\\_RIGHT & GGAYACRCCGGAGATACCCA  & 20 &  4 & 1 &  826 &  846 & 61.58367 & 2.056713 & 60.00000 & 3.535534 & 34.89652 &  0.0000000 &  11.971233 &  2.239915 & 87.097 & Intestinibacillus sp. Marseille-P6563 & 4 & 16\\\\\n",
       "\t cds & 19 & 955 & 162 & 162 & 0 & 955f & PRIMER\\_LEFT  & TCCATGGGCTTYACYCCKCT  & 20 &  8 & 2 &  681 &  701 & 62.68928 & 2.495141 & 57.50000 & 4.330127 & 31.15315 & 31.1531548 &  13.372744 &  0.000000 & 87.097 & Intestinibacillus sp. Marseille-P6563 & 4 & 24\\\\\n",
       "\t cds & 19 & 955 & 162 & 162 & 0 & 955r & PRIMER\\_RIGHT & YACRCCGGAGATACCCAGC   & 19 &  4 & 1 &  824 &  843 & 61.15701 & 1.693536 & 63.15789 & 3.721615 & 34.89652 &  0.0000000 &  11.971233 &  2.239915 & 87.097 & Intestinibacillus sp. Marseille-P6563 & 4 & 24\\\\\n",
       "\t cds & 17 & 129 & 132 & 132 & 0 & 129f & PRIMER\\_LEFT  & GMTGGARTTTCTSCGTCAGGT & 21 &  8 & 1 &   83 &  104 & 60.99081 & 1.609124 & 52.38095 & 3.367175 & 43.16746 &  9.8501469 &   7.643433 &  8.130394 & 87.879 & Intestinibacillus sp. Marseille-P6563 & 2 & 16\\\\\n",
       "\t cds & 17 & 129 & 132 & 132 & 0 & 129r & PRIMER\\_RIGHT & AKGCACATCGGCARRTGRTC  & 20 & 16 & 2 &  195 &  215 & 61.58052 & 1.966489 & 55.00000 & 5.000000 & 48.67702 &  3.6646308 &   1.702271 & 21.058124 & 87.879 & Intestinibacillus sp. Marseille-P6563 & 2 & 16\\\\\n",
       "\t cds & 17 & 352 & 133 & 133 & 0 & 352f & PRIMER\\_LEFT  & AGMTGGARTTTCTSCGTCAGG & 21 &  8 & 1 &   82 &  103 & 60.71966 & 1.614867 & 52.38095 & 3.367175 & 45.60811 &  8.7545325 &   6.226785 & 19.084320 & 87.879 & Intestinibacillus sp. Marseille-P6563 & 2 & 16\\\\\n",
       "\t cds & 17 & 352 & 133 & 133 & 0 & 352r & PRIMER\\_RIGHT & AKGCACATCGGCARRTGRTC  & 20 & 16 & 2 &  195 &  215 & 61.58052 & 1.966489 & 55.00000 & 5.000000 & 48.67702 &  3.6646308 &   1.702271 & 21.058124 & 87.879 & Intestinibacillus sp. Marseille-P6563 & 2 & 16\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 30 × 26\n",
       "\n",
       "| gene_type &lt;chr&gt; | cluster_id &lt;int&gt; | primer_set &lt;int&gt; | amplicon_size_consensus &lt;int&gt; | amplicon_size_avg &lt;dbl&gt; | amplicon_size_sd &lt;dbl&gt; | primer_id &lt;fct&gt; | primer_type &lt;fct&gt; | sequence &lt;fct&gt; | length &lt;int&gt; | degeneracy &lt;int&gt; | degeneracy_3prime &lt;int&gt; | position_start &lt;int&gt; | position_end &lt;int&gt; | Tm_avg &lt;dbl&gt; | Tm_sd &lt;dbl&gt; | GC_avg &lt;dbl&gt; | GC_sd &lt;dbl&gt; | hairpin_avg &lt;dbl&gt; | hairpin_sd &lt;dbl&gt; | homodimer_avg &lt;dbl&gt; | homodimer_sd &lt;dbl&gt; | pident &lt;dbl&gt; | sscinames &lt;fct&gt; | max_degeneracy_3prime &lt;int&gt; | max_degeneracy &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| cds | 26 | 540 | 193 | 193 | 0 | 540f | PRIMER_LEFT  | GGYCTGACBATGGCGGAGTA  | 20 |  6 | 1 |  681 |  701 | 62.39508 | 1.999740 | 60.83333 | 3.435921 | 39.17997 | 18.1052739 |   3.475140 | 12.642244 | 85.022 | Oscillospiraceae bacterium            | 1 | 16 |\n",
       "| cds | 26 | 540 | 193 | 193 | 0 | 540r | PRIMER_RIGHT | TSGTCGAKGTRATWCGCTCC  | 20 | 16 | 1 |  854 |  874 | 60.26164 | 1.509761 | 55.00000 | 3.535534 | 29.23107 | 16.9285589 | -12.584709 | 19.188754 | 85.022 | Oscillospiraceae bacterium            | 1 | 16 |\n",
       "| cds | 26 | 753 | 100 | 100 | 0 | 753f | PRIMER_LEFT  | GGYAARATCGGTCTGTTCGG  | 20 |  4 | 1 |  441 |  461 | 59.45219 | 1.761779 | 55.00000 | 3.535534 | 17.29587 | 17.2958679 | -21.335645 | 17.154538 | 85.022 | Oscillospiraceae bacterium            | 2 | 16 |\n",
       "| cds | 26 | 753 | 100 | 100 | 0 | 753r | PRIMER_RIGHT | ATACCGARATRCCRCCRTGC  | 20 | 16 | 2 |  521 |  541 | 61.19424 | 2.563790 | 55.00000 | 5.000000 |  0.00000 |  0.0000000 | -28.347864 | 12.236336 | 85.022 | Oscillospiraceae bacterium            | 2 | 16 |\n",
       "| cds | 26 | 488 | 181 | 181 | 0 | 488f | PRIMER_LEFT  | AGCAYGGYGGYATYTCGGTA  | 20 | 16 | 1 |  520 |  540 | 62.28327 | 2.588591 | 55.00000 | 5.000000 | 18.04463 | 18.0446265 |   7.007019 | 12.426061 | 85.022 | Oscillospiraceae bacterium            | 4 | 64 |\n",
       "| cds | 26 | 488 | 181 | 181 | 0 | 488r | PRIMER_RIGHT | TACTCCGCCATVGTCAGRCC  | 20 |  6 | 2 |  681 |  701 | 62.39508 | 1.999740 | 60.83333 | 3.435921 | 31.65677 | 22.6781788 |  -2.394768 | 12.763173 | 85.022 | Oscillospiraceae bacterium            | 4 | 64 |\n",
       "| cds | 19 | 772 | 176 | 176 | 0 | 772f | PRIMER_LEFT  | CGYCGYTACGGCTTCCACG   | 19 |  4 | 1 |  525 |  544 | 64.53412 | 1.767244 | 68.42105 | 3.721615 | 55.95397 | 10.0674735 |  19.455128 | 14.757455 | 87.097 | Intestinibacillus sp. Marseille-P6563 | 1 |  8 |\n",
       "| cds | 19 | 772 | 176 | 176 | 0 | 772r | PRIMER_RIGHT | AGMGGRGTRAAGCCCATGGA  | 20 |  8 | 1 |  681 |  701 | 62.68928 | 2.495141 | 57.50000 | 4.330127 | 30.14613 | 30.1461276 |  15.601331 |  0.000000 | 87.097 | Intestinibacillus sp. Marseille-P6563 | 1 |  8 |\n",
       "| cds | 19 |  34 | 163 | 163 | 0 | 34f  | PRIMER_LEFT  | TCCATGGGCTTYACYCCKCT  | 20 |  8 | 2 |  681 |  701 | 62.68928 | 2.495141 | 57.50000 | 4.330127 | 31.15315 | 31.1531548 |  13.372744 |  0.000000 | 87.097 | Intestinibacillus sp. Marseille-P6563 | 2 |  8 |\n",
       "| cds | 19 |  34 | 163 | 163 | 0 | 34r  | PRIMER_RIGHT | AYACRCCGGAGATACCCAGC  | 20 |  4 | 1 |  824 |  844 | 62.21885 | 2.037360 | 60.00000 | 3.535534 | 34.89652 |  0.0000000 |  11.971233 |  2.239915 | 87.097 | Intestinibacillus sp. Marseille-P6563 | 2 |  8 |\n",
       "| cds | 19 | 164 | 165 | 165 | 0 | 164f | PRIMER_LEFT  | TCCATGGGCTTYACYCCKCT  | 20 |  8 | 2 |  681 |  701 | 62.68928 | 2.495141 | 57.50000 | 4.330127 | 31.15315 | 31.1531548 |  13.372744 |  0.000000 | 87.097 | Intestinibacillus sp. Marseille-P6563 | 2 |  8 |\n",
       "| cds | 19 | 164 | 165 | 165 | 0 | 164r | PRIMER_RIGHT | GGAYACRCCGGAGATACCCA  | 20 |  4 | 1 |  826 |  846 | 61.58367 | 2.056713 | 60.00000 | 3.535534 | 34.89652 |  0.0000000 |  11.971233 |  2.239915 | 87.097 | Intestinibacillus sp. Marseille-P6563 | 2 |  8 |\n",
       "| cds | 19 | 581 | 164 | 164 | 0 | 581f | PRIMER_LEFT  | CATGGGCTTYACYCCKCTGG  | 20 |  8 | 2 |  683 |  703 | 62.46956 | 2.419663 | 62.50000 | 4.330127 | 31.15315 | 31.1531548 | -17.549881 | 14.534374 | 87.097 | Intestinibacillus sp. Marseille-P6563 | 2 |  8 |\n",
       "| cds | 19 | 581 | 164 | 164 | 0 | 581r | PRIMER_RIGHT | AGGAYACRCCGGAGATACCC  | 20 |  4 | 1 |  827 |  847 | 61.29900 | 2.066468 | 60.00000 | 3.535534 | 44.32892 |  0.6361774 |  11.971233 |  2.239915 | 87.097 | Intestinibacillus sp. Marseille-P6563 | 2 |  8 |\n",
       "| cds | 19 | 582 | 165 | 165 | 0 | 582f | PRIMER_LEFT  | CCATGGGCTTYACYCCKCTG  | 20 |  8 | 2 |  682 |  702 | 62.46956 | 2.419663 | 62.50000 | 4.330127 | 31.15315 | 31.1531548 |   3.232155 |  0.000000 | 87.097 | Intestinibacillus sp. Marseille-P6563 | 2 |  8 |\n",
       "| cds | 19 | 582 | 165 | 165 | 0 | 582r | PRIMER_RIGHT | AGGAYACRCCGGAGATACCC  | 20 |  4 | 1 |  827 |  847 | 61.29900 | 2.066468 | 60.00000 | 3.535534 | 44.32892 |  0.6361774 |  11.971233 |  2.239915 | 87.097 | Intestinibacillus sp. Marseille-P6563 | 2 |  8 |\n",
       "| cds | 19 | 803 | 247 | 247 | 0 | 803f | PRIMER_LEFT  | TGGGTATCTCCGGYGTRTCC  | 20 |  4 | 2 |  826 |  846 | 61.58367 | 2.056713 | 60.00000 | 3.535534 | 37.24153 |  0.0000000 |   2.634518 |  3.597188 | 87.097 | Intestinibacillus sp. Marseille-P6563 | 2 |  8 |\n",
       "| cds | 19 | 803 | 247 | 247 | 0 | 803r | PRIMER_RIGHT | TCSGGRTCRATCTTGATGCC  | 20 |  8 | 1 | 1053 | 1073 | 60.54531 | 1.524997 | 55.00000 | 3.535534 | 39.05910 | 16.3276701 |   9.466837 | 13.740396 | 87.097 | Intestinibacillus sp. Marseille-P6563 | 2 |  8 |\n",
       "| cds | 19 | 936 | 248 | 248 | 0 | 936f | PRIMER_LEFT  | TGGGTATCTCCGGYGTRTCC  | 20 |  4 | 2 |  826 |  846 | 61.58367 | 2.056713 | 60.00000 | 3.535534 | 37.24153 |  0.0000000 |   2.634518 |  3.597188 | 87.097 | Intestinibacillus sp. Marseille-P6563 | 2 |  8 |\n",
       "| cds | 19 | 936 | 248 | 248 | 0 | 936r | PRIMER_RIGHT | CTCSGGRTCRATCTTGATGC  | 20 |  8 | 1 | 1054 | 1074 | 59.13184 | 1.503151 | 55.00000 | 3.535534 | 37.90776 | 15.0985218 |   6.504147 | 12.138283 | 87.097 | Intestinibacillus sp. Marseille-P6563 | 2 |  8 |\n",
       "| cds | 19 | 290 | 163 | 163 | 0 | 290f | PRIMER_LEFT  | CATGGGCTTYACYCCKCTGG  | 20 |  8 | 2 |  683 |  703 | 62.46956 | 2.419663 | 62.50000 | 4.330127 | 31.15315 | 31.1531548 | -17.549881 | 14.534374 | 87.097 | Intestinibacillus sp. Marseille-P6563 | 2 | 16 |\n",
       "| cds | 19 | 290 | 163 | 163 | 0 | 290r | PRIMER_RIGHT | GGAYACRCCGGAGATACCCA  | 20 |  4 | 1 |  826 |  846 | 61.58367 | 2.056713 | 60.00000 | 3.535534 | 34.89652 |  0.0000000 |  11.971233 |  2.239915 | 87.097 | Intestinibacillus sp. Marseille-P6563 | 2 | 16 |\n",
       "| cds | 19 | 291 | 164 | 164 | 0 | 291f | PRIMER_LEFT  | CCATGGGCTTYACYCCKCTG  | 20 |  8 | 2 |  682 |  702 | 62.46956 | 2.419663 | 62.50000 | 4.330127 | 31.15315 | 31.1531548 |   3.232155 |  0.000000 | 87.097 | Intestinibacillus sp. Marseille-P6563 | 4 | 16 |\n",
       "| cds | 19 | 291 | 164 | 164 | 0 | 291r | PRIMER_RIGHT | GGAYACRCCGGAGATACCCA  | 20 |  4 | 1 |  826 |  846 | 61.58367 | 2.056713 | 60.00000 | 3.535534 | 34.89652 |  0.0000000 |  11.971233 |  2.239915 | 87.097 | Intestinibacillus sp. Marseille-P6563 | 4 | 16 |\n",
       "| cds | 19 | 955 | 162 | 162 | 0 | 955f | PRIMER_LEFT  | TCCATGGGCTTYACYCCKCT  | 20 |  8 | 2 |  681 |  701 | 62.68928 | 2.495141 | 57.50000 | 4.330127 | 31.15315 | 31.1531548 |  13.372744 |  0.000000 | 87.097 | Intestinibacillus sp. Marseille-P6563 | 4 | 24 |\n",
       "| cds | 19 | 955 | 162 | 162 | 0 | 955r | PRIMER_RIGHT | YACRCCGGAGATACCCAGC   | 19 |  4 | 1 |  824 |  843 | 61.15701 | 1.693536 | 63.15789 | 3.721615 | 34.89652 |  0.0000000 |  11.971233 |  2.239915 | 87.097 | Intestinibacillus sp. Marseille-P6563 | 4 | 24 |\n",
       "| cds | 17 | 129 | 132 | 132 | 0 | 129f | PRIMER_LEFT  | GMTGGARTTTCTSCGTCAGGT | 21 |  8 | 1 |   83 |  104 | 60.99081 | 1.609124 | 52.38095 | 3.367175 | 43.16746 |  9.8501469 |   7.643433 |  8.130394 | 87.879 | Intestinibacillus sp. Marseille-P6563 | 2 | 16 |\n",
       "| cds | 17 | 129 | 132 | 132 | 0 | 129r | PRIMER_RIGHT | AKGCACATCGGCARRTGRTC  | 20 | 16 | 2 |  195 |  215 | 61.58052 | 1.966489 | 55.00000 | 5.000000 | 48.67702 |  3.6646308 |   1.702271 | 21.058124 | 87.879 | Intestinibacillus sp. Marseille-P6563 | 2 | 16 |\n",
       "| cds | 17 | 352 | 133 | 133 | 0 | 352f | PRIMER_LEFT  | AGMTGGARTTTCTSCGTCAGG | 21 |  8 | 1 |   82 |  103 | 60.71966 | 1.614867 | 52.38095 | 3.367175 | 45.60811 |  8.7545325 |   6.226785 | 19.084320 | 87.879 | Intestinibacillus sp. Marseille-P6563 | 2 | 16 |\n",
       "| cds | 17 | 352 | 133 | 133 | 0 | 352r | PRIMER_RIGHT | AKGCACATCGGCARRTGRTC  | 20 | 16 | 2 |  195 |  215 | 61.58052 | 1.966489 | 55.00000 | 5.000000 | 48.67702 |  3.6646308 |   1.702271 | 21.058124 | 87.879 | Intestinibacillus sp. Marseille-P6563 | 2 | 16 |\n",
       "\n"
      ],
      "text/plain": [
       "   gene_type cluster_id primer_set amplicon_size_consensus amplicon_size_avg\n",
       "1  cds       26         540        193                     193              \n",
       "2  cds       26         540        193                     193              \n",
       "3  cds       26         753        100                     100              \n",
       "4  cds       26         753        100                     100              \n",
       "5  cds       26         488        181                     181              \n",
       "6  cds       26         488        181                     181              \n",
       "7  cds       19         772        176                     176              \n",
       "8  cds       19         772        176                     176              \n",
       "9  cds       19          34        163                     163              \n",
       "10 cds       19          34        163                     163              \n",
       "11 cds       19         164        165                     165              \n",
       "12 cds       19         164        165                     165              \n",
       "13 cds       19         581        164                     164              \n",
       "14 cds       19         581        164                     164              \n",
       "15 cds       19         582        165                     165              \n",
       "16 cds       19         582        165                     165              \n",
       "17 cds       19         803        247                     247              \n",
       "18 cds       19         803        247                     247              \n",
       "19 cds       19         936        248                     248              \n",
       "20 cds       19         936        248                     248              \n",
       "21 cds       19         290        163                     163              \n",
       "22 cds       19         290        163                     163              \n",
       "23 cds       19         291        164                     164              \n",
       "24 cds       19         291        164                     164              \n",
       "25 cds       19         955        162                     162              \n",
       "26 cds       19         955        162                     162              \n",
       "27 cds       17         129        132                     132              \n",
       "28 cds       17         129        132                     132              \n",
       "29 cds       17         352        133                     133              \n",
       "30 cds       17         352        133                     133              \n",
       "   amplicon_size_sd primer_id primer_type  sequence              length\n",
       "1  0                540f      PRIMER_LEFT  GGYCTGACBATGGCGGAGTA  20    \n",
       "2  0                540r      PRIMER_RIGHT TSGTCGAKGTRATWCGCTCC  20    \n",
       "3  0                753f      PRIMER_LEFT  GGYAARATCGGTCTGTTCGG  20    \n",
       "4  0                753r      PRIMER_RIGHT ATACCGARATRCCRCCRTGC  20    \n",
       "5  0                488f      PRIMER_LEFT  AGCAYGGYGGYATYTCGGTA  20    \n",
       "6  0                488r      PRIMER_RIGHT TACTCCGCCATVGTCAGRCC  20    \n",
       "7  0                772f      PRIMER_LEFT  CGYCGYTACGGCTTCCACG   19    \n",
       "8  0                772r      PRIMER_RIGHT AGMGGRGTRAAGCCCATGGA  20    \n",
       "9  0                34f       PRIMER_LEFT  TCCATGGGCTTYACYCCKCT  20    \n",
       "10 0                34r       PRIMER_RIGHT AYACRCCGGAGATACCCAGC  20    \n",
       "11 0                164f      PRIMER_LEFT  TCCATGGGCTTYACYCCKCT  20    \n",
       "12 0                164r      PRIMER_RIGHT GGAYACRCCGGAGATACCCA  20    \n",
       "13 0                581f      PRIMER_LEFT  CATGGGCTTYACYCCKCTGG  20    \n",
       "14 0                581r      PRIMER_RIGHT AGGAYACRCCGGAGATACCC  20    \n",
       "15 0                582f      PRIMER_LEFT  CCATGGGCTTYACYCCKCTG  20    \n",
       "16 0                582r      PRIMER_RIGHT AGGAYACRCCGGAGATACCC  20    \n",
       "17 0                803f      PRIMER_LEFT  TGGGTATCTCCGGYGTRTCC  20    \n",
       "18 0                803r      PRIMER_RIGHT TCSGGRTCRATCTTGATGCC  20    \n",
       "19 0                936f      PRIMER_LEFT  TGGGTATCTCCGGYGTRTCC  20    \n",
       "20 0                936r      PRIMER_RIGHT CTCSGGRTCRATCTTGATGC  20    \n",
       "21 0                290f      PRIMER_LEFT  CATGGGCTTYACYCCKCTGG  20    \n",
       "22 0                290r      PRIMER_RIGHT GGAYACRCCGGAGATACCCA  20    \n",
       "23 0                291f      PRIMER_LEFT  CCATGGGCTTYACYCCKCTG  20    \n",
       "24 0                291r      PRIMER_RIGHT GGAYACRCCGGAGATACCCA  20    \n",
       "25 0                955f      PRIMER_LEFT  TCCATGGGCTTYACYCCKCT  20    \n",
       "26 0                955r      PRIMER_RIGHT YACRCCGGAGATACCCAGC   19    \n",
       "27 0                129f      PRIMER_LEFT  GMTGGARTTTCTSCGTCAGGT 21    \n",
       "28 0                129r      PRIMER_RIGHT AKGCACATCGGCARRTGRTC  20    \n",
       "29 0                352f      PRIMER_LEFT  AGMTGGARTTTCTSCGTCAGG 21    \n",
       "30 0                352r      PRIMER_RIGHT AKGCACATCGGCARRTGRTC  20    \n",
       "   degeneracy degeneracy_3prime position_start position_end Tm_avg   Tm_sd   \n",
       "1   6         1                  681            701         62.39508 1.999740\n",
       "2  16         1                  854            874         60.26164 1.509761\n",
       "3   4         1                  441            461         59.45219 1.761779\n",
       "4  16         2                  521            541         61.19424 2.563790\n",
       "5  16         1                  520            540         62.28327 2.588591\n",
       "6   6         2                  681            701         62.39508 1.999740\n",
       "7   4         1                  525            544         64.53412 1.767244\n",
       "8   8         1                  681            701         62.68928 2.495141\n",
       "9   8         2                  681            701         62.68928 2.495141\n",
       "10  4         1                  824            844         62.21885 2.037360\n",
       "11  8         2                  681            701         62.68928 2.495141\n",
       "12  4         1                  826            846         61.58367 2.056713\n",
       "13  8         2                  683            703         62.46956 2.419663\n",
       "14  4         1                  827            847         61.29900 2.066468\n",
       "15  8         2                  682            702         62.46956 2.419663\n",
       "16  4         1                  827            847         61.29900 2.066468\n",
       "17  4         2                  826            846         61.58367 2.056713\n",
       "18  8         1                 1053           1073         60.54531 1.524997\n",
       "19  4         2                  826            846         61.58367 2.056713\n",
       "20  8         1                 1054           1074         59.13184 1.503151\n",
       "21  8         2                  683            703         62.46956 2.419663\n",
       "22  4         1                  826            846         61.58367 2.056713\n",
       "23  8         2                  682            702         62.46956 2.419663\n",
       "24  4         1                  826            846         61.58367 2.056713\n",
       "25  8         2                  681            701         62.68928 2.495141\n",
       "26  4         1                  824            843         61.15701 1.693536\n",
       "27  8         1                   83            104         60.99081 1.609124\n",
       "28 16         2                  195            215         61.58052 1.966489\n",
       "29  8         1                   82            103         60.71966 1.614867\n",
       "30 16         2                  195            215         61.58052 1.966489\n",
       "   GC_avg   GC_sd    hairpin_avg hairpin_sd homodimer_avg homodimer_sd pident\n",
       "1  60.83333 3.435921 39.17997    18.1052739   3.475140    12.642244    85.022\n",
       "2  55.00000 3.535534 29.23107    16.9285589 -12.584709    19.188754    85.022\n",
       "3  55.00000 3.535534 17.29587    17.2958679 -21.335645    17.154538    85.022\n",
       "4  55.00000 5.000000  0.00000     0.0000000 -28.347864    12.236336    85.022\n",
       "5  55.00000 5.000000 18.04463    18.0446265   7.007019    12.426061    85.022\n",
       "6  60.83333 3.435921 31.65677    22.6781788  -2.394768    12.763173    85.022\n",
       "7  68.42105 3.721615 55.95397    10.0674735  19.455128    14.757455    87.097\n",
       "8  57.50000 4.330127 30.14613    30.1461276  15.601331     0.000000    87.097\n",
       "9  57.50000 4.330127 31.15315    31.1531548  13.372744     0.000000    87.097\n",
       "10 60.00000 3.535534 34.89652     0.0000000  11.971233     2.239915    87.097\n",
       "11 57.50000 4.330127 31.15315    31.1531548  13.372744     0.000000    87.097\n",
       "12 60.00000 3.535534 34.89652     0.0000000  11.971233     2.239915    87.097\n",
       "13 62.50000 4.330127 31.15315    31.1531548 -17.549881    14.534374    87.097\n",
       "14 60.00000 3.535534 44.32892     0.6361774  11.971233     2.239915    87.097\n",
       "15 62.50000 4.330127 31.15315    31.1531548   3.232155     0.000000    87.097\n",
       "16 60.00000 3.535534 44.32892     0.6361774  11.971233     2.239915    87.097\n",
       "17 60.00000 3.535534 37.24153     0.0000000   2.634518     3.597188    87.097\n",
       "18 55.00000 3.535534 39.05910    16.3276701   9.466837    13.740396    87.097\n",
       "19 60.00000 3.535534 37.24153     0.0000000   2.634518     3.597188    87.097\n",
       "20 55.00000 3.535534 37.90776    15.0985218   6.504147    12.138283    87.097\n",
       "21 62.50000 4.330127 31.15315    31.1531548 -17.549881    14.534374    87.097\n",
       "22 60.00000 3.535534 34.89652     0.0000000  11.971233     2.239915    87.097\n",
       "23 62.50000 4.330127 31.15315    31.1531548   3.232155     0.000000    87.097\n",
       "24 60.00000 3.535534 34.89652     0.0000000  11.971233     2.239915    87.097\n",
       "25 57.50000 4.330127 31.15315    31.1531548  13.372744     0.000000    87.097\n",
       "26 63.15789 3.721615 34.89652     0.0000000  11.971233     2.239915    87.097\n",
       "27 52.38095 3.367175 43.16746     9.8501469   7.643433     8.130394    87.879\n",
       "28 55.00000 5.000000 48.67702     3.6646308   1.702271    21.058124    87.879\n",
       "29 52.38095 3.367175 45.60811     8.7545325   6.226785    19.084320    87.879\n",
       "30 55.00000 5.000000 48.67702     3.6646308   1.702271    21.058124    87.879\n",
       "   sscinames                             max_degeneracy_3prime max_degeneracy\n",
       "1  Oscillospiraceae bacterium            1                     16            \n",
       "2  Oscillospiraceae bacterium            1                     16            \n",
       "3  Oscillospiraceae bacterium            2                     16            \n",
       "4  Oscillospiraceae bacterium            2                     16            \n",
       "5  Oscillospiraceae bacterium            4                     64            \n",
       "6  Oscillospiraceae bacterium            4                     64            \n",
       "7  Intestinibacillus sp. Marseille-P6563 1                      8            \n",
       "8  Intestinibacillus sp. Marseille-P6563 1                      8            \n",
       "9  Intestinibacillus sp. Marseille-P6563 2                      8            \n",
       "10 Intestinibacillus sp. Marseille-P6563 2                      8            \n",
       "11 Intestinibacillus sp. Marseille-P6563 2                      8            \n",
       "12 Intestinibacillus sp. Marseille-P6563 2                      8            \n",
       "13 Intestinibacillus sp. Marseille-P6563 2                      8            \n",
       "14 Intestinibacillus sp. Marseille-P6563 2                      8            \n",
       "15 Intestinibacillus sp. Marseille-P6563 2                      8            \n",
       "16 Intestinibacillus sp. Marseille-P6563 2                      8            \n",
       "17 Intestinibacillus sp. Marseille-P6563 2                      8            \n",
       "18 Intestinibacillus sp. Marseille-P6563 2                      8            \n",
       "19 Intestinibacillus sp. Marseille-P6563 2                      8            \n",
       "20 Intestinibacillus sp. Marseille-P6563 2                      8            \n",
       "21 Intestinibacillus sp. Marseille-P6563 2                     16            \n",
       "22 Intestinibacillus sp. Marseille-P6563 2                     16            \n",
       "23 Intestinibacillus sp. Marseille-P6563 4                     16            \n",
       "24 Intestinibacillus sp. Marseille-P6563 4                     16            \n",
       "25 Intestinibacillus sp. Marseille-P6563 4                     24            \n",
       "26 Intestinibacillus sp. Marseille-P6563 4                     24            \n",
       "27 Intestinibacillus sp. Marseille-P6563 2                     16            \n",
       "28 Intestinibacillus sp. Marseille-P6563 2                     16            \n",
       "29 Intestinibacillus sp. Marseille-P6563 2                     16            \n",
       "30 Intestinibacillus sp. Marseille-P6563 2                     16            "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# arrange by \"best\" primers\n",
    "primer_info_jf = primer_info_j %>%\n",
    "    group_by(primer_set) %>%\n",
    "    mutate(max_degeneracy_3prime = max(degeneracy_3prime),  # max of [Fwd,Rev]\n",
    "           max_degeneracy = max(degeneracy)) %>%\n",
    "    ungroup() %>%\n",
    "    arrange(pident, max_degeneracy_3prime, max_degeneracy) %>%\n",
    "    head(n=30)\n",
    "\n",
    "df.dims(30,40)\n",
    "primer_info_jf\n",
    "df.dims()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### w/ gene annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-29T12:23:25.351638Z",
     "iopub.status.busy": "2021-04-29T12:23:25.349554Z",
     "iopub.status.idle": "2021-04-29T12:23:25.461026Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A tibble: 30 × 27</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>gene_type</th><th scope=col>cluster_id</th><th scope=col>primer_set</th><th scope=col>amplicon_size_consensus</th><th scope=col>amplicon_size_avg</th><th scope=col>amplicon_size_sd</th><th scope=col>primer_id</th><th scope=col>primer_type</th><th scope=col>sequence</th><th scope=col>length</th><th scope=col>degeneracy</th><th scope=col>degeneracy_3prime</th><th scope=col>position_start</th><th scope=col>position_end</th><th scope=col>Tm_avg</th><th scope=col>Tm_sd</th><th scope=col>GC_avg</th><th scope=col>GC_sd</th><th scope=col>hairpin_avg</th><th scope=col>hairpin_sd</th><th scope=col>homodimer_avg</th><th scope=col>homodimer_sd</th><th scope=col>pident</th><th scope=col>sscinames</th><th scope=col>max_degeneracy_3prime</th><th scope=col>max_degeneracy</th><th scope=col>annotation</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>cds</td><td>26</td><td>540</td><td>193</td><td>193</td><td>0</td><td>540f</td><td>PRIMER_LEFT </td><td>GGYCTGACBATGGCGGAGTA </td><td>20</td><td> 6</td><td>1</td><td> 681</td><td> 701</td><td>62.39508</td><td>1.999740</td><td>60.83333</td><td>3.435921</td><td>39.17997</td><td>18.1052739</td><td>  3.475140</td><td>12.642244</td><td>85.022</td><td>Oscillospiraceae bacterium           </td><td>1</td><td>16</td><td>ATP synthase subunit beta</td></tr>\n",
       "\t<tr><td>cds</td><td>26</td><td>540</td><td>193</td><td>193</td><td>0</td><td>540r</td><td>PRIMER_RIGHT</td><td>TSGTCGAKGTRATWCGCTCC </td><td>20</td><td>16</td><td>1</td><td> 854</td><td> 874</td><td>60.26164</td><td>1.509761</td><td>55.00000</td><td>3.535534</td><td>29.23107</td><td>16.9285589</td><td>-12.584709</td><td>19.188754</td><td>85.022</td><td>Oscillospiraceae bacterium           </td><td>1</td><td>16</td><td>ATP synthase subunit beta</td></tr>\n",
       "\t<tr><td>cds</td><td>26</td><td>753</td><td>100</td><td>100</td><td>0</td><td>753f</td><td>PRIMER_LEFT </td><td>GGYAARATCGGTCTGTTCGG </td><td>20</td><td> 4</td><td>1</td><td> 441</td><td> 461</td><td>59.45219</td><td>1.761779</td><td>55.00000</td><td>3.535534</td><td>17.29587</td><td>17.2958679</td><td>-21.335645</td><td>17.154538</td><td>85.022</td><td>Oscillospiraceae bacterium           </td><td>2</td><td>16</td><td>ATP synthase subunit beta</td></tr>\n",
       "\t<tr><td>cds</td><td>26</td><td>753</td><td>100</td><td>100</td><td>0</td><td>753r</td><td>PRIMER_RIGHT</td><td>ATACCGARATRCCRCCRTGC </td><td>20</td><td>16</td><td>2</td><td> 521</td><td> 541</td><td>61.19424</td><td>2.563790</td><td>55.00000</td><td>5.000000</td><td> 0.00000</td><td> 0.0000000</td><td>-28.347864</td><td>12.236336</td><td>85.022</td><td>Oscillospiraceae bacterium           </td><td>2</td><td>16</td><td>ATP synthase subunit beta</td></tr>\n",
       "\t<tr><td>cds</td><td>26</td><td>488</td><td>181</td><td>181</td><td>0</td><td>488f</td><td>PRIMER_LEFT </td><td>AGCAYGGYGGYATYTCGGTA </td><td>20</td><td>16</td><td>1</td><td> 520</td><td> 540</td><td>62.28327</td><td>2.588591</td><td>55.00000</td><td>5.000000</td><td>18.04463</td><td>18.0446265</td><td>  7.007019</td><td>12.426061</td><td>85.022</td><td>Oscillospiraceae bacterium           </td><td>4</td><td>64</td><td>ATP synthase subunit beta</td></tr>\n",
       "\t<tr><td>cds</td><td>26</td><td>488</td><td>181</td><td>181</td><td>0</td><td>488r</td><td>PRIMER_RIGHT</td><td>TACTCCGCCATVGTCAGRCC </td><td>20</td><td> 6</td><td>2</td><td> 681</td><td> 701</td><td>62.39508</td><td>1.999740</td><td>60.83333</td><td>3.435921</td><td>31.65677</td><td>22.6781788</td><td> -2.394768</td><td>12.763173</td><td>85.022</td><td>Oscillospiraceae bacterium           </td><td>4</td><td>64</td><td>ATP synthase subunit beta</td></tr>\n",
       "\t<tr><td>cds</td><td>19</td><td>772</td><td>176</td><td>176</td><td>0</td><td>772f</td><td>PRIMER_LEFT </td><td>CGYCGYTACGGCTTCCACG  </td><td>19</td><td> 4</td><td>1</td><td> 525</td><td> 544</td><td>64.53412</td><td>1.767244</td><td>68.42105</td><td>3.721615</td><td>55.95397</td><td>10.0674735</td><td> 19.455128</td><td>14.757455</td><td>87.097</td><td>Intestinibacillus sp. Marseille-P6563</td><td>1</td><td> 8</td><td>Acetate kinase           </td></tr>\n",
       "\t<tr><td>cds</td><td>19</td><td>772</td><td>176</td><td>176</td><td>0</td><td>772r</td><td>PRIMER_RIGHT</td><td>AGMGGRGTRAAGCCCATGGA </td><td>20</td><td> 8</td><td>1</td><td> 681</td><td> 701</td><td>62.68928</td><td>2.495141</td><td>57.50000</td><td>4.330127</td><td>30.14613</td><td>30.1461276</td><td> 15.601331</td><td> 0.000000</td><td>87.097</td><td>Intestinibacillus sp. Marseille-P6563</td><td>1</td><td> 8</td><td>Acetate kinase           </td></tr>\n",
       "\t<tr><td>cds</td><td>19</td><td> 34</td><td>163</td><td>163</td><td>0</td><td>34f </td><td>PRIMER_LEFT </td><td>TCCATGGGCTTYACYCCKCT </td><td>20</td><td> 8</td><td>2</td><td> 681</td><td> 701</td><td>62.68928</td><td>2.495141</td><td>57.50000</td><td>4.330127</td><td>31.15315</td><td>31.1531548</td><td> 13.372744</td><td> 0.000000</td><td>87.097</td><td>Intestinibacillus sp. Marseille-P6563</td><td>2</td><td> 8</td><td>Acetate kinase           </td></tr>\n",
       "\t<tr><td>cds</td><td>19</td><td> 34</td><td>163</td><td>163</td><td>0</td><td>34r </td><td>PRIMER_RIGHT</td><td>AYACRCCGGAGATACCCAGC </td><td>20</td><td> 4</td><td>1</td><td> 824</td><td> 844</td><td>62.21885</td><td>2.037360</td><td>60.00000</td><td>3.535534</td><td>34.89652</td><td> 0.0000000</td><td> 11.971233</td><td> 2.239915</td><td>87.097</td><td>Intestinibacillus sp. Marseille-P6563</td><td>2</td><td> 8</td><td>Acetate kinase           </td></tr>\n",
       "\t<tr><td>cds</td><td>19</td><td>164</td><td>165</td><td>165</td><td>0</td><td>164f</td><td>PRIMER_LEFT </td><td>TCCATGGGCTTYACYCCKCT </td><td>20</td><td> 8</td><td>2</td><td> 681</td><td> 701</td><td>62.68928</td><td>2.495141</td><td>57.50000</td><td>4.330127</td><td>31.15315</td><td>31.1531548</td><td> 13.372744</td><td> 0.000000</td><td>87.097</td><td>Intestinibacillus sp. Marseille-P6563</td><td>2</td><td> 8</td><td>Acetate kinase           </td></tr>\n",
       "\t<tr><td>cds</td><td>19</td><td>164</td><td>165</td><td>165</td><td>0</td><td>164r</td><td>PRIMER_RIGHT</td><td>GGAYACRCCGGAGATACCCA </td><td>20</td><td> 4</td><td>1</td><td> 826</td><td> 846</td><td>61.58367</td><td>2.056713</td><td>60.00000</td><td>3.535534</td><td>34.89652</td><td> 0.0000000</td><td> 11.971233</td><td> 2.239915</td><td>87.097</td><td>Intestinibacillus sp. Marseille-P6563</td><td>2</td><td> 8</td><td>Acetate kinase           </td></tr>\n",
       "\t<tr><td>cds</td><td>19</td><td>581</td><td>164</td><td>164</td><td>0</td><td>581f</td><td>PRIMER_LEFT </td><td>CATGGGCTTYACYCCKCTGG </td><td>20</td><td> 8</td><td>2</td><td> 683</td><td> 703</td><td>62.46956</td><td>2.419663</td><td>62.50000</td><td>4.330127</td><td>31.15315</td><td>31.1531548</td><td>-17.549881</td><td>14.534374</td><td>87.097</td><td>Intestinibacillus sp. Marseille-P6563</td><td>2</td><td> 8</td><td>Acetate kinase           </td></tr>\n",
       "\t<tr><td>cds</td><td>19</td><td>581</td><td>164</td><td>164</td><td>0</td><td>581r</td><td>PRIMER_RIGHT</td><td>AGGAYACRCCGGAGATACCC </td><td>20</td><td> 4</td><td>1</td><td> 827</td><td> 847</td><td>61.29900</td><td>2.066468</td><td>60.00000</td><td>3.535534</td><td>44.32892</td><td> 0.6361774</td><td> 11.971233</td><td> 2.239915</td><td>87.097</td><td>Intestinibacillus sp. Marseille-P6563</td><td>2</td><td> 8</td><td>Acetate kinase           </td></tr>\n",
       "\t<tr><td>cds</td><td>19</td><td>582</td><td>165</td><td>165</td><td>0</td><td>582f</td><td>PRIMER_LEFT </td><td>CCATGGGCTTYACYCCKCTG </td><td>20</td><td> 8</td><td>2</td><td> 682</td><td> 702</td><td>62.46956</td><td>2.419663</td><td>62.50000</td><td>4.330127</td><td>31.15315</td><td>31.1531548</td><td>  3.232155</td><td> 0.000000</td><td>87.097</td><td>Intestinibacillus sp. Marseille-P6563</td><td>2</td><td> 8</td><td>Acetate kinase           </td></tr>\n",
       "\t<tr><td>cds</td><td>19</td><td>582</td><td>165</td><td>165</td><td>0</td><td>582r</td><td>PRIMER_RIGHT</td><td>AGGAYACRCCGGAGATACCC </td><td>20</td><td> 4</td><td>1</td><td> 827</td><td> 847</td><td>61.29900</td><td>2.066468</td><td>60.00000</td><td>3.535534</td><td>44.32892</td><td> 0.6361774</td><td> 11.971233</td><td> 2.239915</td><td>87.097</td><td>Intestinibacillus sp. Marseille-P6563</td><td>2</td><td> 8</td><td>Acetate kinase           </td></tr>\n",
       "\t<tr><td>cds</td><td>19</td><td>803</td><td>247</td><td>247</td><td>0</td><td>803f</td><td>PRIMER_LEFT </td><td>TGGGTATCTCCGGYGTRTCC </td><td>20</td><td> 4</td><td>2</td><td> 826</td><td> 846</td><td>61.58367</td><td>2.056713</td><td>60.00000</td><td>3.535534</td><td>37.24153</td><td> 0.0000000</td><td>  2.634518</td><td> 3.597188</td><td>87.097</td><td>Intestinibacillus sp. Marseille-P6563</td><td>2</td><td> 8</td><td>Acetate kinase           </td></tr>\n",
       "\t<tr><td>cds</td><td>19</td><td>803</td><td>247</td><td>247</td><td>0</td><td>803r</td><td>PRIMER_RIGHT</td><td>TCSGGRTCRATCTTGATGCC </td><td>20</td><td> 8</td><td>1</td><td>1053</td><td>1073</td><td>60.54531</td><td>1.524997</td><td>55.00000</td><td>3.535534</td><td>39.05910</td><td>16.3276701</td><td>  9.466837</td><td>13.740396</td><td>87.097</td><td>Intestinibacillus sp. Marseille-P6563</td><td>2</td><td> 8</td><td>Acetate kinase           </td></tr>\n",
       "\t<tr><td>cds</td><td>19</td><td>936</td><td>248</td><td>248</td><td>0</td><td>936f</td><td>PRIMER_LEFT </td><td>TGGGTATCTCCGGYGTRTCC </td><td>20</td><td> 4</td><td>2</td><td> 826</td><td> 846</td><td>61.58367</td><td>2.056713</td><td>60.00000</td><td>3.535534</td><td>37.24153</td><td> 0.0000000</td><td>  2.634518</td><td> 3.597188</td><td>87.097</td><td>Intestinibacillus sp. Marseille-P6563</td><td>2</td><td> 8</td><td>Acetate kinase           </td></tr>\n",
       "\t<tr><td>cds</td><td>19</td><td>936</td><td>248</td><td>248</td><td>0</td><td>936r</td><td>PRIMER_RIGHT</td><td>CTCSGGRTCRATCTTGATGC </td><td>20</td><td> 8</td><td>1</td><td>1054</td><td>1074</td><td>59.13184</td><td>1.503151</td><td>55.00000</td><td>3.535534</td><td>37.90776</td><td>15.0985218</td><td>  6.504147</td><td>12.138283</td><td>87.097</td><td>Intestinibacillus sp. Marseille-P6563</td><td>2</td><td> 8</td><td>Acetate kinase           </td></tr>\n",
       "\t<tr><td>cds</td><td>19</td><td>290</td><td>163</td><td>163</td><td>0</td><td>290f</td><td>PRIMER_LEFT </td><td>CATGGGCTTYACYCCKCTGG </td><td>20</td><td> 8</td><td>2</td><td> 683</td><td> 703</td><td>62.46956</td><td>2.419663</td><td>62.50000</td><td>4.330127</td><td>31.15315</td><td>31.1531548</td><td>-17.549881</td><td>14.534374</td><td>87.097</td><td>Intestinibacillus sp. Marseille-P6563</td><td>2</td><td>16</td><td>Acetate kinase           </td></tr>\n",
       "\t<tr><td>cds</td><td>19</td><td>290</td><td>163</td><td>163</td><td>0</td><td>290r</td><td>PRIMER_RIGHT</td><td>GGAYACRCCGGAGATACCCA </td><td>20</td><td> 4</td><td>1</td><td> 826</td><td> 846</td><td>61.58367</td><td>2.056713</td><td>60.00000</td><td>3.535534</td><td>34.89652</td><td> 0.0000000</td><td> 11.971233</td><td> 2.239915</td><td>87.097</td><td>Intestinibacillus sp. Marseille-P6563</td><td>2</td><td>16</td><td>Acetate kinase           </td></tr>\n",
       "\t<tr><td>cds</td><td>19</td><td>291</td><td>164</td><td>164</td><td>0</td><td>291f</td><td>PRIMER_LEFT </td><td>CCATGGGCTTYACYCCKCTG </td><td>20</td><td> 8</td><td>2</td><td> 682</td><td> 702</td><td>62.46956</td><td>2.419663</td><td>62.50000</td><td>4.330127</td><td>31.15315</td><td>31.1531548</td><td>  3.232155</td><td> 0.000000</td><td>87.097</td><td>Intestinibacillus sp. Marseille-P6563</td><td>4</td><td>16</td><td>Acetate kinase           </td></tr>\n",
       "\t<tr><td>cds</td><td>19</td><td>291</td><td>164</td><td>164</td><td>0</td><td>291r</td><td>PRIMER_RIGHT</td><td>GGAYACRCCGGAGATACCCA </td><td>20</td><td> 4</td><td>1</td><td> 826</td><td> 846</td><td>61.58367</td><td>2.056713</td><td>60.00000</td><td>3.535534</td><td>34.89652</td><td> 0.0000000</td><td> 11.971233</td><td> 2.239915</td><td>87.097</td><td>Intestinibacillus sp. Marseille-P6563</td><td>4</td><td>16</td><td>Acetate kinase           </td></tr>\n",
       "\t<tr><td>cds</td><td>19</td><td>955</td><td>162</td><td>162</td><td>0</td><td>955f</td><td>PRIMER_LEFT </td><td>TCCATGGGCTTYACYCCKCT </td><td>20</td><td> 8</td><td>2</td><td> 681</td><td> 701</td><td>62.68928</td><td>2.495141</td><td>57.50000</td><td>4.330127</td><td>31.15315</td><td>31.1531548</td><td> 13.372744</td><td> 0.000000</td><td>87.097</td><td>Intestinibacillus sp. Marseille-P6563</td><td>4</td><td>24</td><td>Acetate kinase           </td></tr>\n",
       "\t<tr><td>cds</td><td>19</td><td>955</td><td>162</td><td>162</td><td>0</td><td>955r</td><td>PRIMER_RIGHT</td><td>YACRCCGGAGATACCCAGC  </td><td>19</td><td> 4</td><td>1</td><td> 824</td><td> 843</td><td>61.15701</td><td>1.693536</td><td>63.15789</td><td>3.721615</td><td>34.89652</td><td> 0.0000000</td><td> 11.971233</td><td> 2.239915</td><td>87.097</td><td>Intestinibacillus sp. Marseille-P6563</td><td>4</td><td>24</td><td>Acetate kinase           </td></tr>\n",
       "\t<tr><td>cds</td><td>17</td><td>129</td><td>132</td><td>132</td><td>0</td><td>129f</td><td>PRIMER_LEFT </td><td>GMTGGARTTTCTSCGTCAGGT</td><td>21</td><td> 8</td><td>1</td><td>  83</td><td> 104</td><td>60.99081</td><td>1.609124</td><td>52.38095</td><td>3.367175</td><td>43.16746</td><td> 9.8501469</td><td>  7.643433</td><td> 8.130394</td><td>87.879</td><td>Intestinibacillus sp. Marseille-P6563</td><td>2</td><td>16</td><td>hypothetical protein     </td></tr>\n",
       "\t<tr><td>cds</td><td>17</td><td>129</td><td>132</td><td>132</td><td>0</td><td>129r</td><td>PRIMER_RIGHT</td><td>AKGCACATCGGCARRTGRTC </td><td>20</td><td>16</td><td>2</td><td> 195</td><td> 215</td><td>61.58052</td><td>1.966489</td><td>55.00000</td><td>5.000000</td><td>48.67702</td><td> 3.6646308</td><td>  1.702271</td><td>21.058124</td><td>87.879</td><td>Intestinibacillus sp. Marseille-P6563</td><td>2</td><td>16</td><td>hypothetical protein     </td></tr>\n",
       "\t<tr><td>cds</td><td>17</td><td>352</td><td>133</td><td>133</td><td>0</td><td>352f</td><td>PRIMER_LEFT </td><td>AGMTGGARTTTCTSCGTCAGG</td><td>21</td><td> 8</td><td>1</td><td>  82</td><td> 103</td><td>60.71966</td><td>1.614867</td><td>52.38095</td><td>3.367175</td><td>45.60811</td><td> 8.7545325</td><td>  6.226785</td><td>19.084320</td><td>87.879</td><td>Intestinibacillus sp. Marseille-P6563</td><td>2</td><td>16</td><td>hypothetical protein     </td></tr>\n",
       "\t<tr><td>cds</td><td>17</td><td>352</td><td>133</td><td>133</td><td>0</td><td>352r</td><td>PRIMER_RIGHT</td><td>AKGCACATCGGCARRTGRTC </td><td>20</td><td>16</td><td>2</td><td> 195</td><td> 215</td><td>61.58052</td><td>1.966489</td><td>55.00000</td><td>5.000000</td><td>48.67702</td><td> 3.6646308</td><td>  1.702271</td><td>21.058124</td><td>87.879</td><td>Intestinibacillus sp. Marseille-P6563</td><td>2</td><td>16</td><td>hypothetical protein     </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 30 × 27\n",
       "\\begin{tabular}{lllllllllllllllllllllllllll}\n",
       " gene\\_type & cluster\\_id & primer\\_set & amplicon\\_size\\_consensus & amplicon\\_size\\_avg & amplicon\\_size\\_sd & primer\\_id & primer\\_type & sequence & length & degeneracy & degeneracy\\_3prime & position\\_start & position\\_end & Tm\\_avg & Tm\\_sd & GC\\_avg & GC\\_sd & hairpin\\_avg & hairpin\\_sd & homodimer\\_avg & homodimer\\_sd & pident & sscinames & max\\_degeneracy\\_3prime & max\\_degeneracy & annotation\\\\\n",
       " <chr> & <int> & <int> & <int> & <dbl> & <dbl> & <fct> & <fct> & <fct> & <int> & <int> & <int> & <int> & <int> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <fct> & <int> & <int> & <fct>\\\\\n",
       "\\hline\n",
       "\t cds & 26 & 540 & 193 & 193 & 0 & 540f & PRIMER\\_LEFT  & GGYCTGACBATGGCGGAGTA  & 20 &  6 & 1 &  681 &  701 & 62.39508 & 1.999740 & 60.83333 & 3.435921 & 39.17997 & 18.1052739 &   3.475140 & 12.642244 & 85.022 & Oscillospiraceae bacterium            & 1 & 16 & ATP synthase subunit beta\\\\\n",
       "\t cds & 26 & 540 & 193 & 193 & 0 & 540r & PRIMER\\_RIGHT & TSGTCGAKGTRATWCGCTCC  & 20 & 16 & 1 &  854 &  874 & 60.26164 & 1.509761 & 55.00000 & 3.535534 & 29.23107 & 16.9285589 & -12.584709 & 19.188754 & 85.022 & Oscillospiraceae bacterium            & 1 & 16 & ATP synthase subunit beta\\\\\n",
       "\t cds & 26 & 753 & 100 & 100 & 0 & 753f & PRIMER\\_LEFT  & GGYAARATCGGTCTGTTCGG  & 20 &  4 & 1 &  441 &  461 & 59.45219 & 1.761779 & 55.00000 & 3.535534 & 17.29587 & 17.2958679 & -21.335645 & 17.154538 & 85.022 & Oscillospiraceae bacterium            & 2 & 16 & ATP synthase subunit beta\\\\\n",
       "\t cds & 26 & 753 & 100 & 100 & 0 & 753r & PRIMER\\_RIGHT & ATACCGARATRCCRCCRTGC  & 20 & 16 & 2 &  521 &  541 & 61.19424 & 2.563790 & 55.00000 & 5.000000 &  0.00000 &  0.0000000 & -28.347864 & 12.236336 & 85.022 & Oscillospiraceae bacterium            & 2 & 16 & ATP synthase subunit beta\\\\\n",
       "\t cds & 26 & 488 & 181 & 181 & 0 & 488f & PRIMER\\_LEFT  & AGCAYGGYGGYATYTCGGTA  & 20 & 16 & 1 &  520 &  540 & 62.28327 & 2.588591 & 55.00000 & 5.000000 & 18.04463 & 18.0446265 &   7.007019 & 12.426061 & 85.022 & Oscillospiraceae bacterium            & 4 & 64 & ATP synthase subunit beta\\\\\n",
       "\t cds & 26 & 488 & 181 & 181 & 0 & 488r & PRIMER\\_RIGHT & TACTCCGCCATVGTCAGRCC  & 20 &  6 & 2 &  681 &  701 & 62.39508 & 1.999740 & 60.83333 & 3.435921 & 31.65677 & 22.6781788 &  -2.394768 & 12.763173 & 85.022 & Oscillospiraceae bacterium            & 4 & 64 & ATP synthase subunit beta\\\\\n",
       "\t cds & 19 & 772 & 176 & 176 & 0 & 772f & PRIMER\\_LEFT  & CGYCGYTACGGCTTCCACG   & 19 &  4 & 1 &  525 &  544 & 64.53412 & 1.767244 & 68.42105 & 3.721615 & 55.95397 & 10.0674735 &  19.455128 & 14.757455 & 87.097 & Intestinibacillus sp. Marseille-P6563 & 1 &  8 & Acetate kinase           \\\\\n",
       "\t cds & 19 & 772 & 176 & 176 & 0 & 772r & PRIMER\\_RIGHT & AGMGGRGTRAAGCCCATGGA  & 20 &  8 & 1 &  681 &  701 & 62.68928 & 2.495141 & 57.50000 & 4.330127 & 30.14613 & 30.1461276 &  15.601331 &  0.000000 & 87.097 & Intestinibacillus sp. Marseille-P6563 & 1 &  8 & Acetate kinase           \\\\\n",
       "\t cds & 19 &  34 & 163 & 163 & 0 & 34f  & PRIMER\\_LEFT  & TCCATGGGCTTYACYCCKCT  & 20 &  8 & 2 &  681 &  701 & 62.68928 & 2.495141 & 57.50000 & 4.330127 & 31.15315 & 31.1531548 &  13.372744 &  0.000000 & 87.097 & Intestinibacillus sp. Marseille-P6563 & 2 &  8 & Acetate kinase           \\\\\n",
       "\t cds & 19 &  34 & 163 & 163 & 0 & 34r  & PRIMER\\_RIGHT & AYACRCCGGAGATACCCAGC  & 20 &  4 & 1 &  824 &  844 & 62.21885 & 2.037360 & 60.00000 & 3.535534 & 34.89652 &  0.0000000 &  11.971233 &  2.239915 & 87.097 & Intestinibacillus sp. Marseille-P6563 & 2 &  8 & Acetate kinase           \\\\\n",
       "\t cds & 19 & 164 & 165 & 165 & 0 & 164f & PRIMER\\_LEFT  & TCCATGGGCTTYACYCCKCT  & 20 &  8 & 2 &  681 &  701 & 62.68928 & 2.495141 & 57.50000 & 4.330127 & 31.15315 & 31.1531548 &  13.372744 &  0.000000 & 87.097 & Intestinibacillus sp. Marseille-P6563 & 2 &  8 & Acetate kinase           \\\\\n",
       "\t cds & 19 & 164 & 165 & 165 & 0 & 164r & PRIMER\\_RIGHT & GGAYACRCCGGAGATACCCA  & 20 &  4 & 1 &  826 &  846 & 61.58367 & 2.056713 & 60.00000 & 3.535534 & 34.89652 &  0.0000000 &  11.971233 &  2.239915 & 87.097 & Intestinibacillus sp. Marseille-P6563 & 2 &  8 & Acetate kinase           \\\\\n",
       "\t cds & 19 & 581 & 164 & 164 & 0 & 581f & PRIMER\\_LEFT  & CATGGGCTTYACYCCKCTGG  & 20 &  8 & 2 &  683 &  703 & 62.46956 & 2.419663 & 62.50000 & 4.330127 & 31.15315 & 31.1531548 & -17.549881 & 14.534374 & 87.097 & Intestinibacillus sp. Marseille-P6563 & 2 &  8 & Acetate kinase           \\\\\n",
       "\t cds & 19 & 581 & 164 & 164 & 0 & 581r & PRIMER\\_RIGHT & AGGAYACRCCGGAGATACCC  & 20 &  4 & 1 &  827 &  847 & 61.29900 & 2.066468 & 60.00000 & 3.535534 & 44.32892 &  0.6361774 &  11.971233 &  2.239915 & 87.097 & Intestinibacillus sp. Marseille-P6563 & 2 &  8 & Acetate kinase           \\\\\n",
       "\t cds & 19 & 582 & 165 & 165 & 0 & 582f & PRIMER\\_LEFT  & CCATGGGCTTYACYCCKCTG  & 20 &  8 & 2 &  682 &  702 & 62.46956 & 2.419663 & 62.50000 & 4.330127 & 31.15315 & 31.1531548 &   3.232155 &  0.000000 & 87.097 & Intestinibacillus sp. Marseille-P6563 & 2 &  8 & Acetate kinase           \\\\\n",
       "\t cds & 19 & 582 & 165 & 165 & 0 & 582r & PRIMER\\_RIGHT & AGGAYACRCCGGAGATACCC  & 20 &  4 & 1 &  827 &  847 & 61.29900 & 2.066468 & 60.00000 & 3.535534 & 44.32892 &  0.6361774 &  11.971233 &  2.239915 & 87.097 & Intestinibacillus sp. Marseille-P6563 & 2 &  8 & Acetate kinase           \\\\\n",
       "\t cds & 19 & 803 & 247 & 247 & 0 & 803f & PRIMER\\_LEFT  & TGGGTATCTCCGGYGTRTCC  & 20 &  4 & 2 &  826 &  846 & 61.58367 & 2.056713 & 60.00000 & 3.535534 & 37.24153 &  0.0000000 &   2.634518 &  3.597188 & 87.097 & Intestinibacillus sp. Marseille-P6563 & 2 &  8 & Acetate kinase           \\\\\n",
       "\t cds & 19 & 803 & 247 & 247 & 0 & 803r & PRIMER\\_RIGHT & TCSGGRTCRATCTTGATGCC  & 20 &  8 & 1 & 1053 & 1073 & 60.54531 & 1.524997 & 55.00000 & 3.535534 & 39.05910 & 16.3276701 &   9.466837 & 13.740396 & 87.097 & Intestinibacillus sp. Marseille-P6563 & 2 &  8 & Acetate kinase           \\\\\n",
       "\t cds & 19 & 936 & 248 & 248 & 0 & 936f & PRIMER\\_LEFT  & TGGGTATCTCCGGYGTRTCC  & 20 &  4 & 2 &  826 &  846 & 61.58367 & 2.056713 & 60.00000 & 3.535534 & 37.24153 &  0.0000000 &   2.634518 &  3.597188 & 87.097 & Intestinibacillus sp. Marseille-P6563 & 2 &  8 & Acetate kinase           \\\\\n",
       "\t cds & 19 & 936 & 248 & 248 & 0 & 936r & PRIMER\\_RIGHT & CTCSGGRTCRATCTTGATGC  & 20 &  8 & 1 & 1054 & 1074 & 59.13184 & 1.503151 & 55.00000 & 3.535534 & 37.90776 & 15.0985218 &   6.504147 & 12.138283 & 87.097 & Intestinibacillus sp. Marseille-P6563 & 2 &  8 & Acetate kinase           \\\\\n",
       "\t cds & 19 & 290 & 163 & 163 & 0 & 290f & PRIMER\\_LEFT  & CATGGGCTTYACYCCKCTGG  & 20 &  8 & 2 &  683 &  703 & 62.46956 & 2.419663 & 62.50000 & 4.330127 & 31.15315 & 31.1531548 & -17.549881 & 14.534374 & 87.097 & Intestinibacillus sp. Marseille-P6563 & 2 & 16 & Acetate kinase           \\\\\n",
       "\t cds & 19 & 290 & 163 & 163 & 0 & 290r & PRIMER\\_RIGHT & GGAYACRCCGGAGATACCCA  & 20 &  4 & 1 &  826 &  846 & 61.58367 & 2.056713 & 60.00000 & 3.535534 & 34.89652 &  0.0000000 &  11.971233 &  2.239915 & 87.097 & Intestinibacillus sp. Marseille-P6563 & 2 & 16 & Acetate kinase           \\\\\n",
       "\t cds & 19 & 291 & 164 & 164 & 0 & 291f & PRIMER\\_LEFT  & CCATGGGCTTYACYCCKCTG  & 20 &  8 & 2 &  682 &  702 & 62.46956 & 2.419663 & 62.50000 & 4.330127 & 31.15315 & 31.1531548 &   3.232155 &  0.000000 & 87.097 & Intestinibacillus sp. Marseille-P6563 & 4 & 16 & Acetate kinase           \\\\\n",
       "\t cds & 19 & 291 & 164 & 164 & 0 & 291r & PRIMER\\_RIGHT & GGAYACRCCGGAGATACCCA  & 20 &  4 & 1 &  826 &  846 & 61.58367 & 2.056713 & 60.00000 & 3.535534 & 34.89652 &  0.0000000 &  11.971233 &  2.239915 & 87.097 & Intestinibacillus sp. Marseille-P6563 & 4 & 16 & Acetate kinase           \\\\\n",
       "\t cds & 19 & 955 & 162 & 162 & 0 & 955f & PRIMER\\_LEFT  & TCCATGGGCTTYACYCCKCT  & 20 &  8 & 2 &  681 &  701 & 62.68928 & 2.495141 & 57.50000 & 4.330127 & 31.15315 & 31.1531548 &  13.372744 &  0.000000 & 87.097 & Intestinibacillus sp. Marseille-P6563 & 4 & 24 & Acetate kinase           \\\\\n",
       "\t cds & 19 & 955 & 162 & 162 & 0 & 955r & PRIMER\\_RIGHT & YACRCCGGAGATACCCAGC   & 19 &  4 & 1 &  824 &  843 & 61.15701 & 1.693536 & 63.15789 & 3.721615 & 34.89652 &  0.0000000 &  11.971233 &  2.239915 & 87.097 & Intestinibacillus sp. Marseille-P6563 & 4 & 24 & Acetate kinase           \\\\\n",
       "\t cds & 17 & 129 & 132 & 132 & 0 & 129f & PRIMER\\_LEFT  & GMTGGARTTTCTSCGTCAGGT & 21 &  8 & 1 &   83 &  104 & 60.99081 & 1.609124 & 52.38095 & 3.367175 & 43.16746 &  9.8501469 &   7.643433 &  8.130394 & 87.879 & Intestinibacillus sp. Marseille-P6563 & 2 & 16 & hypothetical protein     \\\\\n",
       "\t cds & 17 & 129 & 132 & 132 & 0 & 129r & PRIMER\\_RIGHT & AKGCACATCGGCARRTGRTC  & 20 & 16 & 2 &  195 &  215 & 61.58052 & 1.966489 & 55.00000 & 5.000000 & 48.67702 &  3.6646308 &   1.702271 & 21.058124 & 87.879 & Intestinibacillus sp. Marseille-P6563 & 2 & 16 & hypothetical protein     \\\\\n",
       "\t cds & 17 & 352 & 133 & 133 & 0 & 352f & PRIMER\\_LEFT  & AGMTGGARTTTCTSCGTCAGG & 21 &  8 & 1 &   82 &  103 & 60.71966 & 1.614867 & 52.38095 & 3.367175 & 45.60811 &  8.7545325 &   6.226785 & 19.084320 & 87.879 & Intestinibacillus sp. Marseille-P6563 & 2 & 16 & hypothetical protein     \\\\\n",
       "\t cds & 17 & 352 & 133 & 133 & 0 & 352r & PRIMER\\_RIGHT & AKGCACATCGGCARRTGRTC  & 20 & 16 & 2 &  195 &  215 & 61.58052 & 1.966489 & 55.00000 & 5.000000 & 48.67702 &  3.6646308 &   1.702271 & 21.058124 & 87.879 & Intestinibacillus sp. Marseille-P6563 & 2 & 16 & hypothetical protein     \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 30 × 27\n",
       "\n",
       "| gene_type &lt;chr&gt; | cluster_id &lt;int&gt; | primer_set &lt;int&gt; | amplicon_size_consensus &lt;int&gt; | amplicon_size_avg &lt;dbl&gt; | amplicon_size_sd &lt;dbl&gt; | primer_id &lt;fct&gt; | primer_type &lt;fct&gt; | sequence &lt;fct&gt; | length &lt;int&gt; | degeneracy &lt;int&gt; | degeneracy_3prime &lt;int&gt; | position_start &lt;int&gt; | position_end &lt;int&gt; | Tm_avg &lt;dbl&gt; | Tm_sd &lt;dbl&gt; | GC_avg &lt;dbl&gt; | GC_sd &lt;dbl&gt; | hairpin_avg &lt;dbl&gt; | hairpin_sd &lt;dbl&gt; | homodimer_avg &lt;dbl&gt; | homodimer_sd &lt;dbl&gt; | pident &lt;dbl&gt; | sscinames &lt;fct&gt; | max_degeneracy_3prime &lt;int&gt; | max_degeneracy &lt;int&gt; | annotation &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| cds | 26 | 540 | 193 | 193 | 0 | 540f | PRIMER_LEFT  | GGYCTGACBATGGCGGAGTA  | 20 |  6 | 1 |  681 |  701 | 62.39508 | 1.999740 | 60.83333 | 3.435921 | 39.17997 | 18.1052739 |   3.475140 | 12.642244 | 85.022 | Oscillospiraceae bacterium            | 1 | 16 | ATP synthase subunit beta |\n",
       "| cds | 26 | 540 | 193 | 193 | 0 | 540r | PRIMER_RIGHT | TSGTCGAKGTRATWCGCTCC  | 20 | 16 | 1 |  854 |  874 | 60.26164 | 1.509761 | 55.00000 | 3.535534 | 29.23107 | 16.9285589 | -12.584709 | 19.188754 | 85.022 | Oscillospiraceae bacterium            | 1 | 16 | ATP synthase subunit beta |\n",
       "| cds | 26 | 753 | 100 | 100 | 0 | 753f | PRIMER_LEFT  | GGYAARATCGGTCTGTTCGG  | 20 |  4 | 1 |  441 |  461 | 59.45219 | 1.761779 | 55.00000 | 3.535534 | 17.29587 | 17.2958679 | -21.335645 | 17.154538 | 85.022 | Oscillospiraceae bacterium            | 2 | 16 | ATP synthase subunit beta |\n",
       "| cds | 26 | 753 | 100 | 100 | 0 | 753r | PRIMER_RIGHT | ATACCGARATRCCRCCRTGC  | 20 | 16 | 2 |  521 |  541 | 61.19424 | 2.563790 | 55.00000 | 5.000000 |  0.00000 |  0.0000000 | -28.347864 | 12.236336 | 85.022 | Oscillospiraceae bacterium            | 2 | 16 | ATP synthase subunit beta |\n",
       "| cds | 26 | 488 | 181 | 181 | 0 | 488f | PRIMER_LEFT  | AGCAYGGYGGYATYTCGGTA  | 20 | 16 | 1 |  520 |  540 | 62.28327 | 2.588591 | 55.00000 | 5.000000 | 18.04463 | 18.0446265 |   7.007019 | 12.426061 | 85.022 | Oscillospiraceae bacterium            | 4 | 64 | ATP synthase subunit beta |\n",
       "| cds | 26 | 488 | 181 | 181 | 0 | 488r | PRIMER_RIGHT | TACTCCGCCATVGTCAGRCC  | 20 |  6 | 2 |  681 |  701 | 62.39508 | 1.999740 | 60.83333 | 3.435921 | 31.65677 | 22.6781788 |  -2.394768 | 12.763173 | 85.022 | Oscillospiraceae bacterium            | 4 | 64 | ATP synthase subunit beta |\n",
       "| cds | 19 | 772 | 176 | 176 | 0 | 772f | PRIMER_LEFT  | CGYCGYTACGGCTTCCACG   | 19 |  4 | 1 |  525 |  544 | 64.53412 | 1.767244 | 68.42105 | 3.721615 | 55.95397 | 10.0674735 |  19.455128 | 14.757455 | 87.097 | Intestinibacillus sp. Marseille-P6563 | 1 |  8 | Acetate kinase            |\n",
       "| cds | 19 | 772 | 176 | 176 | 0 | 772r | PRIMER_RIGHT | AGMGGRGTRAAGCCCATGGA  | 20 |  8 | 1 |  681 |  701 | 62.68928 | 2.495141 | 57.50000 | 4.330127 | 30.14613 | 30.1461276 |  15.601331 |  0.000000 | 87.097 | Intestinibacillus sp. Marseille-P6563 | 1 |  8 | Acetate kinase            |\n",
       "| cds | 19 |  34 | 163 | 163 | 0 | 34f  | PRIMER_LEFT  | TCCATGGGCTTYACYCCKCT  | 20 |  8 | 2 |  681 |  701 | 62.68928 | 2.495141 | 57.50000 | 4.330127 | 31.15315 | 31.1531548 |  13.372744 |  0.000000 | 87.097 | Intestinibacillus sp. Marseille-P6563 | 2 |  8 | Acetate kinase            |\n",
       "| cds | 19 |  34 | 163 | 163 | 0 | 34r  | PRIMER_RIGHT | AYACRCCGGAGATACCCAGC  | 20 |  4 | 1 |  824 |  844 | 62.21885 | 2.037360 | 60.00000 | 3.535534 | 34.89652 |  0.0000000 |  11.971233 |  2.239915 | 87.097 | Intestinibacillus sp. Marseille-P6563 | 2 |  8 | Acetate kinase            |\n",
       "| cds | 19 | 164 | 165 | 165 | 0 | 164f | PRIMER_LEFT  | TCCATGGGCTTYACYCCKCT  | 20 |  8 | 2 |  681 |  701 | 62.68928 | 2.495141 | 57.50000 | 4.330127 | 31.15315 | 31.1531548 |  13.372744 |  0.000000 | 87.097 | Intestinibacillus sp. Marseille-P6563 | 2 |  8 | Acetate kinase            |\n",
       "| cds | 19 | 164 | 165 | 165 | 0 | 164r | PRIMER_RIGHT | GGAYACRCCGGAGATACCCA  | 20 |  4 | 1 |  826 |  846 | 61.58367 | 2.056713 | 60.00000 | 3.535534 | 34.89652 |  0.0000000 |  11.971233 |  2.239915 | 87.097 | Intestinibacillus sp. Marseille-P6563 | 2 |  8 | Acetate kinase            |\n",
       "| cds | 19 | 581 | 164 | 164 | 0 | 581f | PRIMER_LEFT  | CATGGGCTTYACYCCKCTGG  | 20 |  8 | 2 |  683 |  703 | 62.46956 | 2.419663 | 62.50000 | 4.330127 | 31.15315 | 31.1531548 | -17.549881 | 14.534374 | 87.097 | Intestinibacillus sp. Marseille-P6563 | 2 |  8 | Acetate kinase            |\n",
       "| cds | 19 | 581 | 164 | 164 | 0 | 581r | PRIMER_RIGHT | AGGAYACRCCGGAGATACCC  | 20 |  4 | 1 |  827 |  847 | 61.29900 | 2.066468 | 60.00000 | 3.535534 | 44.32892 |  0.6361774 |  11.971233 |  2.239915 | 87.097 | Intestinibacillus sp. Marseille-P6563 | 2 |  8 | Acetate kinase            |\n",
       "| cds | 19 | 582 | 165 | 165 | 0 | 582f | PRIMER_LEFT  | CCATGGGCTTYACYCCKCTG  | 20 |  8 | 2 |  682 |  702 | 62.46956 | 2.419663 | 62.50000 | 4.330127 | 31.15315 | 31.1531548 |   3.232155 |  0.000000 | 87.097 | Intestinibacillus sp. Marseille-P6563 | 2 |  8 | Acetate kinase            |\n",
       "| cds | 19 | 582 | 165 | 165 | 0 | 582r | PRIMER_RIGHT | AGGAYACRCCGGAGATACCC  | 20 |  4 | 1 |  827 |  847 | 61.29900 | 2.066468 | 60.00000 | 3.535534 | 44.32892 |  0.6361774 |  11.971233 |  2.239915 | 87.097 | Intestinibacillus sp. Marseille-P6563 | 2 |  8 | Acetate kinase            |\n",
       "| cds | 19 | 803 | 247 | 247 | 0 | 803f | PRIMER_LEFT  | TGGGTATCTCCGGYGTRTCC  | 20 |  4 | 2 |  826 |  846 | 61.58367 | 2.056713 | 60.00000 | 3.535534 | 37.24153 |  0.0000000 |   2.634518 |  3.597188 | 87.097 | Intestinibacillus sp. Marseille-P6563 | 2 |  8 | Acetate kinase            |\n",
       "| cds | 19 | 803 | 247 | 247 | 0 | 803r | PRIMER_RIGHT | TCSGGRTCRATCTTGATGCC  | 20 |  8 | 1 | 1053 | 1073 | 60.54531 | 1.524997 | 55.00000 | 3.535534 | 39.05910 | 16.3276701 |   9.466837 | 13.740396 | 87.097 | Intestinibacillus sp. Marseille-P6563 | 2 |  8 | Acetate kinase            |\n",
       "| cds | 19 | 936 | 248 | 248 | 0 | 936f | PRIMER_LEFT  | TGGGTATCTCCGGYGTRTCC  | 20 |  4 | 2 |  826 |  846 | 61.58367 | 2.056713 | 60.00000 | 3.535534 | 37.24153 |  0.0000000 |   2.634518 |  3.597188 | 87.097 | Intestinibacillus sp. Marseille-P6563 | 2 |  8 | Acetate kinase            |\n",
       "| cds | 19 | 936 | 248 | 248 | 0 | 936r | PRIMER_RIGHT | CTCSGGRTCRATCTTGATGC  | 20 |  8 | 1 | 1054 | 1074 | 59.13184 | 1.503151 | 55.00000 | 3.535534 | 37.90776 | 15.0985218 |   6.504147 | 12.138283 | 87.097 | Intestinibacillus sp. Marseille-P6563 | 2 |  8 | Acetate kinase            |\n",
       "| cds | 19 | 290 | 163 | 163 | 0 | 290f | PRIMER_LEFT  | CATGGGCTTYACYCCKCTGG  | 20 |  8 | 2 |  683 |  703 | 62.46956 | 2.419663 | 62.50000 | 4.330127 | 31.15315 | 31.1531548 | -17.549881 | 14.534374 | 87.097 | Intestinibacillus sp. Marseille-P6563 | 2 | 16 | Acetate kinase            |\n",
       "| cds | 19 | 290 | 163 | 163 | 0 | 290r | PRIMER_RIGHT | GGAYACRCCGGAGATACCCA  | 20 |  4 | 1 |  826 |  846 | 61.58367 | 2.056713 | 60.00000 | 3.535534 | 34.89652 |  0.0000000 |  11.971233 |  2.239915 | 87.097 | Intestinibacillus sp. Marseille-P6563 | 2 | 16 | Acetate kinase            |\n",
       "| cds | 19 | 291 | 164 | 164 | 0 | 291f | PRIMER_LEFT  | CCATGGGCTTYACYCCKCTG  | 20 |  8 | 2 |  682 |  702 | 62.46956 | 2.419663 | 62.50000 | 4.330127 | 31.15315 | 31.1531548 |   3.232155 |  0.000000 | 87.097 | Intestinibacillus sp. Marseille-P6563 | 4 | 16 | Acetate kinase            |\n",
       "| cds | 19 | 291 | 164 | 164 | 0 | 291r | PRIMER_RIGHT | GGAYACRCCGGAGATACCCA  | 20 |  4 | 1 |  826 |  846 | 61.58367 | 2.056713 | 60.00000 | 3.535534 | 34.89652 |  0.0000000 |  11.971233 |  2.239915 | 87.097 | Intestinibacillus sp. Marseille-P6563 | 4 | 16 | Acetate kinase            |\n",
       "| cds | 19 | 955 | 162 | 162 | 0 | 955f | PRIMER_LEFT  | TCCATGGGCTTYACYCCKCT  | 20 |  8 | 2 |  681 |  701 | 62.68928 | 2.495141 | 57.50000 | 4.330127 | 31.15315 | 31.1531548 |  13.372744 |  0.000000 | 87.097 | Intestinibacillus sp. Marseille-P6563 | 4 | 24 | Acetate kinase            |\n",
       "| cds | 19 | 955 | 162 | 162 | 0 | 955r | PRIMER_RIGHT | YACRCCGGAGATACCCAGC   | 19 |  4 | 1 |  824 |  843 | 61.15701 | 1.693536 | 63.15789 | 3.721615 | 34.89652 |  0.0000000 |  11.971233 |  2.239915 | 87.097 | Intestinibacillus sp. Marseille-P6563 | 4 | 24 | Acetate kinase            |\n",
       "| cds | 17 | 129 | 132 | 132 | 0 | 129f | PRIMER_LEFT  | GMTGGARTTTCTSCGTCAGGT | 21 |  8 | 1 |   83 |  104 | 60.99081 | 1.609124 | 52.38095 | 3.367175 | 43.16746 |  9.8501469 |   7.643433 |  8.130394 | 87.879 | Intestinibacillus sp. Marseille-P6563 | 2 | 16 | hypothetical protein      |\n",
       "| cds | 17 | 129 | 132 | 132 | 0 | 129r | PRIMER_RIGHT | AKGCACATCGGCARRTGRTC  | 20 | 16 | 2 |  195 |  215 | 61.58052 | 1.966489 | 55.00000 | 5.000000 | 48.67702 |  3.6646308 |   1.702271 | 21.058124 | 87.879 | Intestinibacillus sp. Marseille-P6563 | 2 | 16 | hypothetical protein      |\n",
       "| cds | 17 | 352 | 133 | 133 | 0 | 352f | PRIMER_LEFT  | AGMTGGARTTTCTSCGTCAGG | 21 |  8 | 1 |   82 |  103 | 60.71966 | 1.614867 | 52.38095 | 3.367175 | 45.60811 |  8.7545325 |   6.226785 | 19.084320 | 87.879 | Intestinibacillus sp. Marseille-P6563 | 2 | 16 | hypothetical protein      |\n",
       "| cds | 17 | 352 | 133 | 133 | 0 | 352r | PRIMER_RIGHT | AKGCACATCGGCARRTGRTC  | 20 | 16 | 2 |  195 |  215 | 61.58052 | 1.966489 | 55.00000 | 5.000000 | 48.67702 |  3.6646308 |   1.702271 | 21.058124 | 87.879 | Intestinibacillus sp. Marseille-P6563 | 2 | 16 | hypothetical protein      |\n",
       "\n"
      ],
      "text/plain": [
       "   gene_type cluster_id primer_set amplicon_size_consensus amplicon_size_avg\n",
       "1  cds       26         540        193                     193              \n",
       "2  cds       26         540        193                     193              \n",
       "3  cds       26         753        100                     100              \n",
       "4  cds       26         753        100                     100              \n",
       "5  cds       26         488        181                     181              \n",
       "6  cds       26         488        181                     181              \n",
       "7  cds       19         772        176                     176              \n",
       "8  cds       19         772        176                     176              \n",
       "9  cds       19          34        163                     163              \n",
       "10 cds       19          34        163                     163              \n",
       "11 cds       19         164        165                     165              \n",
       "12 cds       19         164        165                     165              \n",
       "13 cds       19         581        164                     164              \n",
       "14 cds       19         581        164                     164              \n",
       "15 cds       19         582        165                     165              \n",
       "16 cds       19         582        165                     165              \n",
       "17 cds       19         803        247                     247              \n",
       "18 cds       19         803        247                     247              \n",
       "19 cds       19         936        248                     248              \n",
       "20 cds       19         936        248                     248              \n",
       "21 cds       19         290        163                     163              \n",
       "22 cds       19         290        163                     163              \n",
       "23 cds       19         291        164                     164              \n",
       "24 cds       19         291        164                     164              \n",
       "25 cds       19         955        162                     162              \n",
       "26 cds       19         955        162                     162              \n",
       "27 cds       17         129        132                     132              \n",
       "28 cds       17         129        132                     132              \n",
       "29 cds       17         352        133                     133              \n",
       "30 cds       17         352        133                     133              \n",
       "   amplicon_size_sd primer_id primer_type  sequence              length\n",
       "1  0                540f      PRIMER_LEFT  GGYCTGACBATGGCGGAGTA  20    \n",
       "2  0                540r      PRIMER_RIGHT TSGTCGAKGTRATWCGCTCC  20    \n",
       "3  0                753f      PRIMER_LEFT  GGYAARATCGGTCTGTTCGG  20    \n",
       "4  0                753r      PRIMER_RIGHT ATACCGARATRCCRCCRTGC  20    \n",
       "5  0                488f      PRIMER_LEFT  AGCAYGGYGGYATYTCGGTA  20    \n",
       "6  0                488r      PRIMER_RIGHT TACTCCGCCATVGTCAGRCC  20    \n",
       "7  0                772f      PRIMER_LEFT  CGYCGYTACGGCTTCCACG   19    \n",
       "8  0                772r      PRIMER_RIGHT AGMGGRGTRAAGCCCATGGA  20    \n",
       "9  0                34f       PRIMER_LEFT  TCCATGGGCTTYACYCCKCT  20    \n",
       "10 0                34r       PRIMER_RIGHT AYACRCCGGAGATACCCAGC  20    \n",
       "11 0                164f      PRIMER_LEFT  TCCATGGGCTTYACYCCKCT  20    \n",
       "12 0                164r      PRIMER_RIGHT GGAYACRCCGGAGATACCCA  20    \n",
       "13 0                581f      PRIMER_LEFT  CATGGGCTTYACYCCKCTGG  20    \n",
       "14 0                581r      PRIMER_RIGHT AGGAYACRCCGGAGATACCC  20    \n",
       "15 0                582f      PRIMER_LEFT  CCATGGGCTTYACYCCKCTG  20    \n",
       "16 0                582r      PRIMER_RIGHT AGGAYACRCCGGAGATACCC  20    \n",
       "17 0                803f      PRIMER_LEFT  TGGGTATCTCCGGYGTRTCC  20    \n",
       "18 0                803r      PRIMER_RIGHT TCSGGRTCRATCTTGATGCC  20    \n",
       "19 0                936f      PRIMER_LEFT  TGGGTATCTCCGGYGTRTCC  20    \n",
       "20 0                936r      PRIMER_RIGHT CTCSGGRTCRATCTTGATGC  20    \n",
       "21 0                290f      PRIMER_LEFT  CATGGGCTTYACYCCKCTGG  20    \n",
       "22 0                290r      PRIMER_RIGHT GGAYACRCCGGAGATACCCA  20    \n",
       "23 0                291f      PRIMER_LEFT  CCATGGGCTTYACYCCKCTG  20    \n",
       "24 0                291r      PRIMER_RIGHT GGAYACRCCGGAGATACCCA  20    \n",
       "25 0                955f      PRIMER_LEFT  TCCATGGGCTTYACYCCKCT  20    \n",
       "26 0                955r      PRIMER_RIGHT YACRCCGGAGATACCCAGC   19    \n",
       "27 0                129f      PRIMER_LEFT  GMTGGARTTTCTSCGTCAGGT 21    \n",
       "28 0                129r      PRIMER_RIGHT AKGCACATCGGCARRTGRTC  20    \n",
       "29 0                352f      PRIMER_LEFT  AGMTGGARTTTCTSCGTCAGG 21    \n",
       "30 0                352r      PRIMER_RIGHT AKGCACATCGGCARRTGRTC  20    \n",
       "   degeneracy degeneracy_3prime position_start position_end Tm_avg   Tm_sd   \n",
       "1   6         1                  681            701         62.39508 1.999740\n",
       "2  16         1                  854            874         60.26164 1.509761\n",
       "3   4         1                  441            461         59.45219 1.761779\n",
       "4  16         2                  521            541         61.19424 2.563790\n",
       "5  16         1                  520            540         62.28327 2.588591\n",
       "6   6         2                  681            701         62.39508 1.999740\n",
       "7   4         1                  525            544         64.53412 1.767244\n",
       "8   8         1                  681            701         62.68928 2.495141\n",
       "9   8         2                  681            701         62.68928 2.495141\n",
       "10  4         1                  824            844         62.21885 2.037360\n",
       "11  8         2                  681            701         62.68928 2.495141\n",
       "12  4         1                  826            846         61.58367 2.056713\n",
       "13  8         2                  683            703         62.46956 2.419663\n",
       "14  4         1                  827            847         61.29900 2.066468\n",
       "15  8         2                  682            702         62.46956 2.419663\n",
       "16  4         1                  827            847         61.29900 2.066468\n",
       "17  4         2                  826            846         61.58367 2.056713\n",
       "18  8         1                 1053           1073         60.54531 1.524997\n",
       "19  4         2                  826            846         61.58367 2.056713\n",
       "20  8         1                 1054           1074         59.13184 1.503151\n",
       "21  8         2                  683            703         62.46956 2.419663\n",
       "22  4         1                  826            846         61.58367 2.056713\n",
       "23  8         2                  682            702         62.46956 2.419663\n",
       "24  4         1                  826            846         61.58367 2.056713\n",
       "25  8         2                  681            701         62.68928 2.495141\n",
       "26  4         1                  824            843         61.15701 1.693536\n",
       "27  8         1                   83            104         60.99081 1.609124\n",
       "28 16         2                  195            215         61.58052 1.966489\n",
       "29  8         1                   82            103         60.71966 1.614867\n",
       "30 16         2                  195            215         61.58052 1.966489\n",
       "   GC_avg   GC_sd    hairpin_avg hairpin_sd homodimer_avg homodimer_sd pident\n",
       "1  60.83333 3.435921 39.17997    18.1052739   3.475140    12.642244    85.022\n",
       "2  55.00000 3.535534 29.23107    16.9285589 -12.584709    19.188754    85.022\n",
       "3  55.00000 3.535534 17.29587    17.2958679 -21.335645    17.154538    85.022\n",
       "4  55.00000 5.000000  0.00000     0.0000000 -28.347864    12.236336    85.022\n",
       "5  55.00000 5.000000 18.04463    18.0446265   7.007019    12.426061    85.022\n",
       "6  60.83333 3.435921 31.65677    22.6781788  -2.394768    12.763173    85.022\n",
       "7  68.42105 3.721615 55.95397    10.0674735  19.455128    14.757455    87.097\n",
       "8  57.50000 4.330127 30.14613    30.1461276  15.601331     0.000000    87.097\n",
       "9  57.50000 4.330127 31.15315    31.1531548  13.372744     0.000000    87.097\n",
       "10 60.00000 3.535534 34.89652     0.0000000  11.971233     2.239915    87.097\n",
       "11 57.50000 4.330127 31.15315    31.1531548  13.372744     0.000000    87.097\n",
       "12 60.00000 3.535534 34.89652     0.0000000  11.971233     2.239915    87.097\n",
       "13 62.50000 4.330127 31.15315    31.1531548 -17.549881    14.534374    87.097\n",
       "14 60.00000 3.535534 44.32892     0.6361774  11.971233     2.239915    87.097\n",
       "15 62.50000 4.330127 31.15315    31.1531548   3.232155     0.000000    87.097\n",
       "16 60.00000 3.535534 44.32892     0.6361774  11.971233     2.239915    87.097\n",
       "17 60.00000 3.535534 37.24153     0.0000000   2.634518     3.597188    87.097\n",
       "18 55.00000 3.535534 39.05910    16.3276701   9.466837    13.740396    87.097\n",
       "19 60.00000 3.535534 37.24153     0.0000000   2.634518     3.597188    87.097\n",
       "20 55.00000 3.535534 37.90776    15.0985218   6.504147    12.138283    87.097\n",
       "21 62.50000 4.330127 31.15315    31.1531548 -17.549881    14.534374    87.097\n",
       "22 60.00000 3.535534 34.89652     0.0000000  11.971233     2.239915    87.097\n",
       "23 62.50000 4.330127 31.15315    31.1531548   3.232155     0.000000    87.097\n",
       "24 60.00000 3.535534 34.89652     0.0000000  11.971233     2.239915    87.097\n",
       "25 57.50000 4.330127 31.15315    31.1531548  13.372744     0.000000    87.097\n",
       "26 63.15789 3.721615 34.89652     0.0000000  11.971233     2.239915    87.097\n",
       "27 52.38095 3.367175 43.16746     9.8501469   7.643433     8.130394    87.879\n",
       "28 55.00000 5.000000 48.67702     3.6646308   1.702271    21.058124    87.879\n",
       "29 52.38095 3.367175 45.60811     8.7545325   6.226785    19.084320    87.879\n",
       "30 55.00000 5.000000 48.67702     3.6646308   1.702271    21.058124    87.879\n",
       "   sscinames                             max_degeneracy_3prime max_degeneracy\n",
       "1  Oscillospiraceae bacterium            1                     16            \n",
       "2  Oscillospiraceae bacterium            1                     16            \n",
       "3  Oscillospiraceae bacterium            2                     16            \n",
       "4  Oscillospiraceae bacterium            2                     16            \n",
       "5  Oscillospiraceae bacterium            4                     64            \n",
       "6  Oscillospiraceae bacterium            4                     64            \n",
       "7  Intestinibacillus sp. Marseille-P6563 1                      8            \n",
       "8  Intestinibacillus sp. Marseille-P6563 1                      8            \n",
       "9  Intestinibacillus sp. Marseille-P6563 2                      8            \n",
       "10 Intestinibacillus sp. Marseille-P6563 2                      8            \n",
       "11 Intestinibacillus sp. Marseille-P6563 2                      8            \n",
       "12 Intestinibacillus sp. Marseille-P6563 2                      8            \n",
       "13 Intestinibacillus sp. Marseille-P6563 2                      8            \n",
       "14 Intestinibacillus sp. Marseille-P6563 2                      8            \n",
       "15 Intestinibacillus sp. Marseille-P6563 2                      8            \n",
       "16 Intestinibacillus sp. Marseille-P6563 2                      8            \n",
       "17 Intestinibacillus sp. Marseille-P6563 2                      8            \n",
       "18 Intestinibacillus sp. Marseille-P6563 2                      8            \n",
       "19 Intestinibacillus sp. Marseille-P6563 2                      8            \n",
       "20 Intestinibacillus sp. Marseille-P6563 2                      8            \n",
       "21 Intestinibacillus sp. Marseille-P6563 2                     16            \n",
       "22 Intestinibacillus sp. Marseille-P6563 2                     16            \n",
       "23 Intestinibacillus sp. Marseille-P6563 4                     16            \n",
       "24 Intestinibacillus sp. Marseille-P6563 4                     16            \n",
       "25 Intestinibacillus sp. Marseille-P6563 4                     24            \n",
       "26 Intestinibacillus sp. Marseille-P6563 4                     24            \n",
       "27 Intestinibacillus sp. Marseille-P6563 2                     16            \n",
       "28 Intestinibacillus sp. Marseille-P6563 2                     16            \n",
       "29 Intestinibacillus sp. Marseille-P6563 2                     16            \n",
       "30 Intestinibacillus sp. Marseille-P6563 2                     16            \n",
       "   annotation               \n",
       "1  ATP synthase subunit beta\n",
       "2  ATP synthase subunit beta\n",
       "3  ATP synthase subunit beta\n",
       "4  ATP synthase subunit beta\n",
       "5  ATP synthase subunit beta\n",
       "6  ATP synthase subunit beta\n",
       "7  Acetate kinase           \n",
       "8  Acetate kinase           \n",
       "9  Acetate kinase           \n",
       "10 Acetate kinase           \n",
       "11 Acetate kinase           \n",
       "12 Acetate kinase           \n",
       "13 Acetate kinase           \n",
       "14 Acetate kinase           \n",
       "15 Acetate kinase           \n",
       "16 Acetate kinase           \n",
       "17 Acetate kinase           \n",
       "18 Acetate kinase           \n",
       "19 Acetate kinase           \n",
       "20 Acetate kinase           \n",
       "21 Acetate kinase           \n",
       "22 Acetate kinase           \n",
       "23 Acetate kinase           \n",
       "24 Acetate kinase           \n",
       "25 Acetate kinase           \n",
       "26 Acetate kinase           \n",
       "27 hypothetical protein     \n",
       "28 hypothetical protein     \n",
       "29 hypothetical protein     \n",
       "30 hypothetical protein     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# joining\n",
    "primer_info_jfj = primer_info_jf %>%\n",
    "    left_join(gene_annot %>% distinct(gene_type, cluster_id, annotation),\n",
    "              c('gene_type', 'cluster_id'))\n",
    "\n",
    "df.dims(30,50)\n",
    "primer_info_jfj\n",
    "df.dims()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Writing table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-29T12:23:39.601267Z",
     "iopub.status.busy": "2021-04-29T12:23:39.591870Z",
     "iopub.status.idle": "2021-04-29T12:23:39.617944Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "File written: /ebio/abt3_projects/software/dev/ll_pipelines/llprimer/experiments/HMP_most-wanted/v0.3//Butyricicoccus/best_primers.tsv \n"
     ]
    }
   ],
   "source": [
    "F = file.path(work_dir, 'best_primers.tsv')\n",
    "write_table(primer_info_jfj, F)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# sessionInfo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-29T12:23:49.525053Z",
     "iopub.status.busy": "2021-04-29T12:23:49.522879Z",
     "iopub.status.idle": "2021-04-29T12:23:49.900713Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LLG\n",
      "===\n",
      "\n",
      "Ley Lab Genome analysis pipeline (LLG)\n",
      "\n",
      "* Version: 0.2.4\n",
      "* Authors:\n",
      "  * Nick Youngblut <nyoungb2@gmail.com>\n",
      "* Maintainers:\n",
      "  * Nick Youngblut <nyoungb2@gmail.com>\n",
      "\n",
      "--- conda envs ---\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/gtdbtk.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- conda-forge::networkx\n",
      "- bioconda::gtdbtk=1.4\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/checkm.yaml <==\n",
      "channels:\n",
      "- bioconda\n",
      "dependencies:\n",
      "- python=2.7\n",
      "- pigz\n",
      "- bioconda::prodigal\n",
      "- bioconda::pplacer\n",
      "- bioconda::checkm-genome\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/quast.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- bioconda::seqkit\n",
      "- bioconda::quast>=5.0.0\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/sourmash.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- bioconda::sourmash=2.0.0a4\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/fastqc.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- bioconda::seqtk\n",
      "- bioconda::fastqc=0.11.7\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/multiqc.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- bioconda::multiqc=1.4\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/hadley.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- conda-forge::r-ape\n",
      "- conda-forge::r-dplyr\n",
      "- conda-forge::r-tidyr\n",
      "- conda-forge::r-ggplot2\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/bbmap.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::bbmap=37.78\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/ani.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::dashing\n",
      "- bioconda::fastani\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/prokka.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- bioconda::prokka\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/prodigal.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pip\n",
      "- biopython=1.77\n",
      "- :prodigal\n",
      "- pip:\n",
      "  - biopython.convert\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/eggnog-mapper.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::eggnog-mapper\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/phylophlan.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::phylophlan\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/drep.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::mash\n",
      "- bioconda::checkm-genome\n",
      "- bioconda::fastani\n",
      "- bioconda::drep\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/antismash.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- biopython=1.77\n",
      "- bioconda::blast=2.9\n",
      "- bioconda::antismash>=5.1.2\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/deepbgc.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::deepbgc\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/abricate.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- bioconda::abricate\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/rRNA.yaml <==\n",
      "channels:\n",
      "- qiime2/label/r2020.2\n",
      "- conda-forge\n",
      "- bioconda\n",
      "- defaults\n",
      "dependencies:\n",
      "- barrnap\n",
      "- _libgcc_mutex=0.1\n",
      "- _openmp_mutex=4.5\n",
      "- _r-mutex=1.0.1\n",
      "- alsa-lib=1.1.5\n",
      "- arb-bio-tools=6.0.6\n",
      "- attrs=19.3.0\n",
      "- backcall=0.1.0\n",
      "- bibtexparser=1.1.0\n",
      "- binutils_impl_linux-64=2.33.1\n",
      "- binutils_linux-64=2.33.1\n",
      "- bioconductor-biobase=2.42.0\n",
      "- bioconductor-biocgenerics=0.28.0\n",
      "- bioconductor-biocparallel=1.16.6\n",
      "- bioconductor-biostrings=2.50.2\n",
      "- bioconductor-dada2=1.10.0\n",
      "- bioconductor-delayedarray=0.8.0\n",
      "- bioconductor-genomeinfodb=1.18.1\n",
      "- bioconductor-genomeinfodbdata=1.2.1\n",
      "- bioconductor-genomicalignments=1.18.1\n",
      "- bioconductor-genomicranges=1.34.0\n",
      "- bioconductor-iranges=2.16.0\n",
      "- bioconductor-rsamtools=1.34.0\n",
      "- bioconductor-s4vectors=0.20.1\n",
      "- bioconductor-shortread=1.40.0\n",
      "- bioconductor-summarizedexperiment=1.12.0\n",
      "- bioconductor-xvector=0.22.0\n",
      "- bioconductor-zlibbioc=1.28.0\n",
      "- biom-format=2.1.8\n",
      "- blas=2.14\n",
      "- blast=2.9.0\n",
      "- bleach=3.1.1\n",
      "- bokeh=1.4.0\n",
      "- boost=1.68.0\n",
      "- boost-cpp=1.68.0\n",
      "- bwidget=1.9.14\n",
      "- bzip2=1.0.8\n",
      "- ca-certificates=2019.11.28\n",
      "- cachecontrol=0.12.5\n",
      "- cairo=1.16.0\n",
      "- certifi=2019.11.28\n",
      "- cffi=1.13.2\n",
      "- chardet=3.0.4\n",
      "- click=7.0\n",
      "- cryptography=2.8\n",
      "- curl=7.68.0\n",
      "- cutadapt=2.8\n",
      "- cycler=0.10.0\n",
      "- cython=0.29.15\n",
      "- dbus=1.13.6\n",
      "- deblur=1.1.0\n",
      "- decorator=4.4.1\n",
      "- defusedxml=0.6.0\n",
      "- dendropy=4.4.0\n",
      "- dnaio=0.4.1\n",
      "- emperor=1.0.0\n",
      "- entrypoints=0.3\n",
      "- expat=2.2.9\n",
      "- fastcluster=1.1.26\n",
      "- fasttree=2.1.10\n",
      "- fontconfig=2.13.1\n",
      "- freetype=2.10.0\n",
      "- future=0.18.2\n",
      "- gcc_impl_linux-64=7.3.0\n",
      "- gcc_linux-64=7.3.0\n",
      "- gettext=0.19.8.1\n",
      "- gfortran_impl_linux-64=7.3.0\n",
      "- gfortran_linux-64=7.3.0\n",
      "- giflib=5.2.1\n",
      "- glib=2.58.3\n",
      "- gmp=6.2.0\n",
      "- gneiss=0.4.6\n",
      "- gnutls=3.6.5\n",
      "- graphite2=1.3.13\n",
      "- gsl=2.5\n",
      "- gst-plugins-base=1.14.5\n",
      "- gstreamer=1.14.5\n",
      "- gxx_impl_linux-64=7.3.0\n",
      "- gxx_linux-64=7.3.0\n",
      "- h5py=2.10.0\n",
      "- harfbuzz=2.4.0\n",
      "- hdf5=1.10.5\n",
      "- hdmedians=0.13\n",
      "- hmmer=3.1b2\n",
      "- icu=58.2\n",
      "- idna=2.9\n",
      "- ijson=2.6.1\n",
      "- importlib_metadata=1.5.0\n",
      "- ipykernel=5.1.4\n",
      "- ipython=7.12.0\n",
      "- ipython_genutils=0.2.0\n",
      "- ipywidgets=7.5.1\n",
      "- iqtree=1.6.12\n",
      "- jedi=0.16.0\n",
      "- jinja2=2.11.1\n",
      "- joblib=0.14.1\n",
      "- jpeg=9c\n",
      "- jsonschema=3.2.0\n",
      "- jupyter_client=6.0.0\n",
      "- jupyter_core=4.6.3\n",
      "- kiwisolver=1.1.0\n",
      "- krb5=1.16.4\n",
      "- lcms2=2.9\n",
      "- ld_impl_linux-64=2.33.1\n",
      "- libarbdb=6.0.6\n",
      "- libblas=3.8.0\n",
      "- libcblas=3.8.0\n",
      "- libcurl=7.68.0\n",
      "- libedit=3.1.20170329\n",
      "- libffi=3.2.1\n",
      "- libgcc=7.2.0\n",
      "- libgcc-ng=9.2.0\n",
      "- libgfortran-ng=7.3.0\n",
      "- libgomp=9.2.0\n",
      "- libiconv=1.15\n",
      "- liblapack=3.8.0\n",
      "- liblapacke=3.8.0\n",
      "- libopenblas=0.3.7\n",
      "- libpng=1.6.37\n",
      "- libsodium=1.0.17\n",
      "- libssh2=1.8.2\n",
      "- libstdcxx-ng=9.2.0\n",
      "- libtiff=4.1.0\n",
      "- libuuid=2.32.1\n",
      "- libxcb=1.13\n",
      "- libxml2=2.9.9\n",
      "- lockfile=0.12.2\n",
      "- lz4-c=1.8.3\n",
      "- mafft=7.310\n",
      "- make=4.3\n",
      "- markupsafe=1.1.1\n",
      "- matplotlib=3.1.1\n",
      "- matplotlib-base=3.1.1\n",
      "- mistune=0.8.4\n",
      "- more-itertools=8.2.0\n",
      "- msgpack-python=1.0.0\n",
      "- natsort=7.0.1\n",
      "- nbconvert=5.6.1\n",
      "- nbformat=5.0.4\n",
      "- ncurses=6.1\n",
      "- nettle=3.4.1\n",
      "- networkx=2.4\n",
      "- nose=1.3.7\n",
      "- notebook=6.0.3\n",
      "- numpy=1.18.1\n",
      "- olefile=0.46\n",
      "- openjdk=11.0.1\n",
      "- openssl=1.1.1d\n",
      "- packaging=20.1\n",
      "- pandas=0.25.3\n",
      "- pandoc=2.9.2\n",
      "- pandocfilters=1.4.2\n",
      "- pango=1.40.14\n",
      "- parso=0.6.1\n",
      "- patsy=0.5.1\n",
      "- pcre=8.44\n",
      "- perl=5.26.2\n",
      "- perl-archive-tar=2.32\n",
      "- perl-carp=1.38\n",
      "- perl-common-sense=3.74\n",
      "- perl-compress-raw-bzip2=2.087\n",
      "- perl-compress-raw-zlib=2.087\n",
      "- perl-exporter=5.72\n",
      "- perl-exporter-tiny=1.002001\n",
      "- perl-extutils-makemaker=7.36\n",
      "- perl-io-compress=2.087\n",
      "- perl-io-zlib=1.10\n",
      "- perl-json=4.02\n",
      "- perl-json-xs=2.34\n",
      "- perl-list-moreutils=0.428\n",
      "- perl-list-moreutils-xs=0.428\n",
      "- perl-pathtools=3.75\n",
      "- perl-scalar-list-utils=1.52\n",
      "- perl-types-serialiser=1.0\n",
      "- perl-xsloader=0.24\n",
      "- pexpect=4.8.0\n",
      "- pickleshare=0.7.5\n",
      "- pigz=2.3.4\n",
      "- pillow=7.0.0\n",
      "- pip=20.0.2\n",
      "- pixman=0.38.0\n",
      "- pluggy=0.12.0\n",
      "- prometheus_client=0.7.1\n",
      "- prompt_toolkit=3.0.3\n",
      "- psutil=5.7.0\n",
      "- pthread-stubs=0.4\n",
      "- ptyprocess=0.6.0\n",
      "- py=1.8.1\n",
      "- pycparser=2.19\n",
      "- pygments=2.5.2\n",
      "- pyopenssl=19.1.0\n",
      "- pyparsing=2.4.6\n",
      "- pyqt=5.9.2\n",
      "- pyrsistent=0.15.7\n",
      "- pysocks=1.7.1\n",
      "- pytest=5.3.5\n",
      "- python=3.6.7\n",
      "- python-dateutil=2.8.1\n",
      "- pytz=2019.3\n",
      "- pyyaml=5.3\n",
      "- pyzmq=19.0.0\n",
      "- q2-alignment=2020.2.0\n",
      "- q2-composition=2020.2.0\n",
      "- q2-cutadapt=2020.2.0\n",
      "- q2-dada2=2020.2.0\n",
      "- q2-deblur=2020.2.0\n",
      "- q2-demux=2020.2.0\n",
      "- q2-diversity=2020.2.0\n",
      "- q2-emperor=2020.2.0\n",
      "- q2-feature-classifier=2020.2.0\n",
      "- q2-feature-table=2020.2.0\n",
      "- q2-fragment-insertion=2020.2.0\n",
      "- q2-gneiss=2020.2.0\n",
      "- q2-longitudinal=2020.2.0\n",
      "- q2-metadata=2020.2.0\n",
      "- q2-phylogeny=2020.2.0\n",
      "- q2-quality-control=2020.2.0\n",
      "- q2-quality-filter=2020.2.0\n",
      "- q2-sample-classifier=2020.2.0\n",
      "- q2-taxa=2020.2.0\n",
      "- q2-types=2020.2.0\n",
      "- q2-vsearch=2020.2.0\n",
      "- q2cli=2020.2.0\n",
      "- q2templates=2020.2.0\n",
      "- qiime2=2020.2.0\n",
      "- qt=5.9.7\n",
      "- r-assertthat=0.2.1\n",
      "- r-backports=1.1.5\n",
      "- r-base=3.5.1\n",
      "- r-bh=1.72.0_3\n",
      "- r-bitops=1.0_6\n",
      "- r-cli=2.0.2\n",
      "- r-cluster=2.1.0\n",
      "- r-colorspace=1.4_1\n",
      "- r-crayon=1.3.4\n",
      "- r-data.table=1.12.6\n",
      "- r-digest=0.6.25\n",
      "- r-ellipsis=0.3.0\n",
      "- r-fansi=0.4.1\n",
      "- r-farver=2.0.3\n",
      "- r-formatr=1.7\n",
      "- r-futile.logger=1.4.3\n",
      "- r-futile.options=1.0.1\n",
      "- r-ggplot2=3.2.1\n",
      "- r-glue=1.3.1\n",
      "- r-gtable=0.3.0\n",
      "- r-hwriter=1.3.2\n",
      "- r-labeling=0.3\n",
      "- r-lambda.r=1.2.4\n",
      "- r-lattice=0.20_40\n",
      "- r-latticeextra=0.6_28\n",
      "- r-lazyeval=0.2.2\n",
      "- r-lifecycle=0.1.0\n",
      "- r-magrittr=1.5\n",
      "- r-mass=7.3_51.5\n",
      "- r-matrix=1.2_18\n",
      "- r-matrixstats=0.55.0\n",
      "- r-mgcv=1.8_31\n",
      "- r-munsell=0.5.0\n",
      "- r-nlme=3.1_144\n",
      "- r-permute=0.9_5\n",
      "- r-pillar=1.4.3\n",
      "- r-pkgconfig=2.0.3\n",
      "- r-plyr=1.8.5\n",
      "- r-r6=2.4.1\n",
      "- r-rcolorbrewer=1.1_2\n",
      "- r-rcpp=1.0.3\n",
      "- r-rcppparallel=4.4.4\n",
      "- r-rcurl=1.98_1.1\n",
      "- r-reshape2=1.4.3\n",
      "- r-rlang=0.4.4\n",
      "- r-scales=1.1.0\n",
      "- r-snow=0.4_3\n",
      "- r-stringi=1.4.3\n",
      "- r-stringr=1.4.0\n",
      "- r-tibble=2.1.3\n",
      "- r-utf8=1.1.4\n",
      "- r-vctrs=0.2.3\n",
      "- r-vegan=2.5_6\n",
      "- r-viridislite=0.3.0\n",
      "- r-withr=2.1.2\n",
      "- r-zeallot=0.1.0\n",
      "- raxml=8.2.12\n",
      "- readline=8.0\n",
      "- requests=2.23.0\n",
      "- scikit-bio=0.5.5\n",
      "- scikit-learn=0.22.1\n",
      "- scipy=1.4.1\n",
      "- seaborn=0.10.0\n",
      "- send2trash=1.5.0\n",
      "- sepp=4.3.10\n",
      "- setuptools=45.2.0\n",
      "- sina=1.6.0\n",
      "- sip=4.19.8\n",
      "- six=1.14.0\n",
      "- sortmerna=2.0\n",
      "- sqlite=3.30.1\n",
      "- statsmodels=0.11.1\n",
      "- tbb=2019.9\n",
      "- terminado=0.8.3\n",
      "- testpath=0.4.4\n",
      "- tk=8.6.10\n",
      "- tktable=2.10\n",
      "- tornado=6.0.3\n",
      "- traitlets=4.3.3\n",
      "- tzlocal=2.0.0\n",
      "- unifrac=0.10.0\n",
      "- urllib3=1.25.7\n",
      "- vsearch=2.7.0\n",
      "- wcwidth=0.1.8\n",
      "- webencodings=0.5.1\n",
      "- wheel=0.34.2\n",
      "- widgetsnbextension=3.5.1\n",
      "- xopen=0.8.4\n",
      "- xorg-fixesproto=5.0\n",
      "- xorg-inputproto=2.3.2\n",
      "- xorg-kbproto=1.0.7\n",
      "- xorg-libice=1.0.10\n",
      "- xorg-libsm=1.2.3\n",
      "- xorg-libx11=1.6.9\n",
      "- xorg-libxau=1.0.9\n",
      "- xorg-libxdmcp=1.1.3\n",
      "- xorg-libxext=1.3.4\n",
      "- xorg-libxfixes=5.0.3\n",
      "- xorg-libxi=1.7.10\n",
      "- xorg-libxrender=0.9.10\n",
      "- xorg-libxtst=1.2.3\n",
      "- xorg-recordproto=1.14.2\n",
      "- xorg-renderproto=0.11.1\n",
      "- xorg-xextproto=7.3.0\n",
      "- xorg-xproto=7.0.31\n",
      "- xz=5.2.4\n",
      "- yaml=0.2.2\n",
      "- zeromq=4.3.2\n",
      "- zipp=3.0.0\n",
      "- zlib=1.2.11\n",
      "- zstd=1.4.4\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/traitar.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pip\n",
      "- scipy\n",
      "- pandas\n",
      "- numpy\n",
      "- matplotlib\n",
      "- pigz\n",
      "- bioconda::hmmer\n",
      "- pip:\n",
      "  - traitar3>=0.1.1\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/cctyper.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "- russel88\n",
      "dependencies:\n",
      "- pigz\n",
      "- cctyper\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/comparem.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::comparem\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/bigscape.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- python=3.6\n",
      "- hmmer\n",
      "- biopython\n",
      "- fasttree\n",
      "- numpy\n",
      "- scipy\n",
      "- networkx\n",
      "- scikit-learn=0.19.1\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/rgi.yaml <==\n",
      "name: rgi\n",
      "channels:\n",
      "  - bioconda\n",
      "  - conda-forge\n",
      "  - defaults\n",
      "dependencies:\n",
      "  - _libgcc_mutex=0.1=conda_forge\n",
      "  - _openmp_mutex=4.5=1_llvm\n",
      "  - bamtools=2.5.1=he860b03_5\n",
      "  - bedtools=2.29.2=hc088bd4_0\n",
      "  - biopython=1.72=py36h14c3975_1000\n",
      "  - blast=2.9.0=pl526h3066fca_4\n",
      "  - bowtie2=2.4.1=py36h7f0b59b_2\n",
      "  - bwa=0.7.17=hed695b0_7\n",
      "  - bzip2=1.0.8=h516909a_2\n",
      "  - ca-certificates=2020.4.5.2=hecda079_0\n",
      "  - certifi=2020.4.5.2=py36h9f0ad1d_0\n",
      "  - curl=7.69.1=h33f0ec9_0\n",
      "  - cycler=0.10.0=py_2\n",
      "  - diamond=0.8.36=h8b12597_4\n",
      "  - entrez-direct=13.3=pl526h375a9b1_0\n",
      "  - expat=2.2.9=he1b5a44_2\n",
      "  - filetype=1.0.7=pyh9f0ad1d_0\n",
      "  - freetype=2.10.2=he06d7ca_0\n",
      "  - icu=64.2=he1b5a44_1\n",
      "  - kiwisolver=1.2.0=py36hdb11119_0\n",
      "  - krb5=1.17.1=h2fd8d38_0\n",
      "  - ld_impl_linux-64=2.34=h53a641e_5\n",
      "  - libblas=3.8.0=16_openblas\n",
      "  - libcblas=3.8.0=16_openblas\n",
      "  - libcurl=7.69.1=hf7181ac_0\n",
      "  - libdeflate=1.0=h14c3975_1\n",
      "  - libedit=3.1.20191231=h46ee950_0\n",
      "  - libffi=3.2.1=he1b5a44_1007\n",
      "  - libgcc-ng=9.2.0=h24d8f2e_2\n",
      "  - libgfortran-ng=7.5.0\n",
      "  - liblapack=3.8.0=16_openblas\n",
      "  - libopenblas=0.3.9=h5ec1e0e_0\n",
      "  - libpng=1.6.37=hed695b0_1\n",
      "  - libssh2=1.9.0=hab1572f_2\n",
      "  - libstdcxx-ng=9.2.0=hdf63c60_2\n",
      "  - llvm-openmp=10.0.0=hc9558a2_0\n",
      "  - matplotlib-base=3.2.1=py36hb8e4980_0\n",
      "  - ncurses=6.1=hf484d3e_1002\n",
      "  - numpy=1.18.5=py36h7314795_0\n",
      "  - oligoarrayaux=3.8=hc9558a2_0\n",
      "  - openssl=1.1.1g=h516909a_0\n",
      "  - pandas=1.0.4=py36h830a2c2_0\n",
      "  - patsy=0.5.1=py_0\n",
      "  - pcre=8.44=he1b5a44_0\n",
      "  - perl=5.26.2=h516909a_1006\n",
      "  - perl-app-cpanminus=1.7044=pl526_1\n",
      "  - perl-archive-tar=2.32=pl526_0\n",
      "  - perl-base=2.23=pl526_1\n",
      "  - perl-business-isbn=3.004=pl526_0\n",
      "  - perl-business-isbn-data=20140910.003=pl526_0\n",
      "  - perl-carp=1.38=pl526_3\n",
      "  - perl-common-sense=3.74=pl526_2\n",
      "  - perl-compress-raw-bzip2=2.087=pl526he1b5a44_0\n",
      "  - perl-compress-raw-zlib=2.087=pl526hc9558a2_0\n",
      "  - perl-constant=1.33=pl526_1\n",
      "  - perl-data-dumper=2.173=pl526_0\n",
      "  - perl-digest-hmac=1.03=pl526_3\n",
      "  - perl-digest-md5=2.55=pl526_0\n",
      "  - perl-encode=2.88=pl526_1\n",
      "  - perl-encode-locale=1.05=pl526_6\n",
      "  - perl-exporter=5.72=pl526_1\n",
      "  - perl-exporter-tiny=1.002001=pl526_0\n",
      "  - perl-extutils-makemaker=7.36=pl526_1\n",
      "  - perl-file-listing=6.04=pl526_1\n",
      "  - perl-file-path=2.16=pl526_0\n",
      "  - perl-file-temp=0.2304=pl526_2\n",
      "  - perl-html-parser=3.72=pl526h6bb024c_5\n",
      "  - perl-html-tagset=3.20=pl526_3\n",
      "  - perl-html-tree=5.07=pl526_1\n",
      "  - perl-http-cookies=6.04=pl526_0\n",
      "  - perl-http-daemon=6.01=pl526_1\n",
      "  - perl-http-date=6.02=pl526_3\n",
      "  - perl-http-message=6.18=pl526_0\n",
      "  - perl-http-negotiate=6.01=pl526_3\n",
      "  - perl-io-compress=2.087=pl526he1b5a44_0\n",
      "  - perl-io-html=1.001=pl526_2\n",
      "  - perl-io-socket-ssl=2.066=pl526_0\n",
      "  - perl-io-zlib=1.10=pl526_2\n",
      "  - perl-json=4.02=pl526_0\n",
      "  - perl-json-xs=2.34=pl526h6bb024c_3\n",
      "  - perl-libwww-perl=6.39=pl526_0\n",
      "  - perl-list-moreutils=0.428=pl526_1\n",
      "  - perl-list-moreutils-xs=0.428=pl526_0\n",
      "  - perl-lwp-mediatypes=6.04=pl526_0\n",
      "  - perl-lwp-protocol-https=6.07=pl526_4\n",
      "  - perl-mime-base64=3.15=pl526_1\n",
      "  - perl-mozilla-ca=20180117=pl526_1\n",
      "  - perl-net-http=6.19=pl526_0\n",
      "  - perl-net-ssleay=1.88=pl526h90d6eec_0\n",
      "  - perl-ntlm=1.09=pl526_4\n",
      "  - perl-parent=0.236=pl526_1\n",
      "  - perl-pathtools=3.75=pl526h14c3975_1\n",
      "  - perl-scalar-list-utils=1.52=pl526h516909a_0\n",
      "  - perl-socket=2.027=pl526_1\n",
      "  - perl-storable=3.15=pl526h14c3975_0\n",
      "  - perl-test-requiresinternet=0.05=pl526_0\n",
      "  - perl-time-local=1.28=pl526_1\n",
      "  - perl-try-tiny=0.30=pl526_1\n",
      "  - perl-types-serialiser=1.0=pl526_2\n",
      "  - perl-uri=1.76=pl526_0\n",
      "  - perl-www-robotrules=6.02=pl526_3\n",
      "  - perl-xml-namespacesupport=1.12=pl526_0\n",
      "  - perl-xml-parser=2.44=pl526h4e0c4b3_7\n",
      "  - perl-xml-sax=1.02=pl526_0\n",
      "  - perl-xml-sax-base=1.09=pl526_0\n",
      "  - perl-xml-sax-expat=0.51=pl526_3\n",
      "  - perl-xml-simple=2.25=pl526_1\n",
      "  - perl-xsloader=0.24=pl526_0\n",
      "  - pip=20.1.1=py_1\n",
      "  - prodigal=2.6.3=h516909a_2\n",
      "  - pyahocorasick=1.4.0=py36h8c4c3a4_1\n",
      "  - pyfaidx=0.5.8=py_1\n",
      "  - pyparsing=2.4.7=pyh9f0ad1d_0\n",
      "  - python=3.6.10=h8356626_1011_cpython\n",
      "  - python-dateutil=2.8.1=py_0\n",
      "  - python_abi=3.6=1_cp36m\n",
      "  - pytz=2020.1=pyh9f0ad1d_0\n",
      "  - readline=8.0=hf8c457e_0\n",
      "  - rgi=5.1.0=py_1\n",
      "  - samtools=1.9=h8571acd_11\n",
      "  - scipy=1.4.1=py36h2d22cac_3\n",
      "  - seaborn=0.10.1=py_0\n",
      "  - setuptools=47.1.1=py36h9f0ad1d_0\n",
      "  - six=1.15.0=pyh9f0ad1d_0\n",
      "  - sqlite=3.30.1=hcee41ef_0\n",
      "  - statsmodels=0.11.1=py36h8c4c3a4_1\n",
      "  - tbb=2020.1=hc9558a2_0\n",
      "  - tk=8.6.10=hed695b0_0\n",
      "  - tornado=6.0.4=py36h8c4c3a4_1\n",
      "  - wheel=0.34.2=py_1\n",
      "  - xz=5.2.5=h516909a_0\n",
      "  - zlib=1.2.11=h516909a_1006\n",
      "prefix: /home/users/ldenies/apps/miniconda3/envs/rgi\n",
      "\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/pathofact_hmm.yaml <==\n",
      "channels:\n",
      "  - bioconda\n",
      "  - conda-forge\n",
      "  - defaults\n",
      "dependencies:\n",
      "  - _libgcc_mutex=0.1=conda_forge\n",
      "  - _openmp_mutex=4.5=0_gnu\n",
      "  - hmmer=3.2.1=he1b5a44_2\n",
      "  - libgcc-ng=9.2.0=h24d8f2e_2\n",
      "  - libgomp=9.2.0=h24d8f2e_2\n",
      "  - libstdcxx-ng=9.2.0=hdf63c60_2\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/pathofact_biopython.yaml <==\n",
      "channels:\n",
      "  - conda-forge\n",
      "  - bioconda\n",
      "  - defaults\n",
      "dependencies:\n",
      "  - _libgcc_mutex=0.1=main\n",
      "  - biopython=1.74=py36h516909a_0\n",
      "  - ca-certificates=2019.6.16=hecc5488_0\n",
      "  - certifi=2019.6.16=py36_1\n",
      "  - cycler=0.10.0=py_1\n",
      "  - dbus=1.13.6=he372182_0\n",
      "  - expat=2.2.5=he1b5a44_1003\n",
      "  - fontconfig=2.13.1=he4413a7_1000\n",
      "  - freetype=2.10.0=he983fc9_0\n",
      "  - gettext=0.19.8.1=hc5be6a0_1002\n",
      "  - glib=2.58.3=h6f030ca_1002\n",
      "  - gst-plugins-base=1.14.5=h0935bb2_0\n",
      "  - gstreamer=1.14.5=h36ae1b5_0\n",
      "  - icu=58.2=hf484d3e_1000\n",
      "  - joblib=0.13.2=py_0\n",
      "  - jpeg=9c=h14c3975_1001\n",
      "  - kiwisolver=1.1.0=py36hc9558a2_0\n",
      "  - libblas=3.8.0=11_openblas\n",
      "  - libcblas=3.8.0=11_openblas\n",
      "  - libedit=3.1.20181209=hc058e9b_0\n",
      "  - libffi=3.2.1=hd88cf55_4\n",
      "  - libgcc-ng=9.1.0=hdf63c60_0\n",
      "  - libgfortran-ng=7.3.0=hdf63c60_0\n",
      "  - libiconv=1.15=h516909a_1005\n",
      "  - liblapack=3.8.0=11_openblas\n",
      "  - libopenblas=0.3.6=h6e990d7_6\n",
      "  - libpng=1.6.37=hed695b0_0\n",
      "  - libstdcxx-ng=9.1.0=hdf63c60_0\n",
      "  - libuuid=2.32.1=h14c3975_1000\n",
      "  - libxcb=1.13=h14c3975_1002\n",
      "  - libxml2=2.9.9=h13577e0_2\n",
      "  - matplotlib=3.1.0=py36h5429711_0\n",
      "  - ncurses=6.1=he6710b0_1\n",
      "  - numpy=1.17.0=py36h95a1406_0\n",
      "  - openssl=1.1.1c=h516909a_0\n",
      "  - pandas=0.25.0=py36hb3f55d8_0\n",
      "  - patsy=0.5.1=py_0\n",
      "  - pcre=8.41=hf484d3e_1003\n",
      "  - pip=19.1.1=py36_0\n",
      "  - pthread-stubs=0.4=h14c3975_1001\n",
      "  - pyparsing=2.4.2=py_0\n",
      "  - pyqt=5.9.2=py36hcca6a23_2\n",
      "  - python=3.6.9=h265db76_0\n",
      "  - python-dateutil=2.8.0=py_0\n",
      "  - pytz=2019.2=py_0\n",
      "  - qt=5.9.7=h52cfd70_2\n",
      "  - readline=7.0=h7b6447c_5\n",
      "  - scikit-learn=0.21.3=py36hcdab131_0\n",
      "  - scipy=1.3.0=py36h921218d_1\n",
      "  - seaborn=0.9.0=py_0\n",
      "  - setuptools=41.0.1=py36_0\n",
      "  - sip=4.19.8=py36hf484d3e_1000\n",
      "  - six=1.12.0=py36_1000\n",
      "  - sqlite=3.29.0=h7b6447c_0\n",
      "  - statsmodels=0.10.1=py36hc1659b7_0\n",
      "  - tk=8.6.8=hbc83047_0\n",
      "  - tornado=6.0.3=py36h516909a_0\n",
      "  - wheel=0.33.4=py36_0\n",
      "  - xorg-libxau=1.0.9=h14c3975_0\n",
      "  - xorg-libxdmcp=1.1.3=h516909a_0\n",
      "  - xz=5.2.4=h14c3975_4\n",
      "  - zlib=1.2.11=h7b6447c_3\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/platon.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::platon\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llg//bin/envs/plasflow.yaml <==\n",
      "channels:\n",
      "  # for PlasFlow\n",
      "  - smaegol\n",
      "  # rest\n",
      "  - bioconda\n",
      "  - conda-forge\n",
      "  - defaults\n",
      "dependencies:\n",
      "  - seqkit\n",
      "  - _libgcc_mutex=0.1=conda_forge\n",
      "  - _openmp_mutex=4.5=1_llvm\n",
      "  - _r-mutex=1.0.1=anacondar_1\n",
      "  - backports=1.0=py_2\n",
      "  - backports.weakref=1.0rc1=py35_1\n",
      "  - bioconductor-biocgenerics=0.22.0=r3.3.2_0\n",
      "  - bioconductor-biostrings=2.42.1=r3.3.2_0\n",
      "  - bioconductor-iranges=2.8.2=r3.3.2_0\n",
      "  - bioconductor-s4vectors=0.12.2=r3.3.2_0\n",
      "  - bioconductor-xvector=0.14.1=r3.3.2_0\n",
      "  - bioconductor-zlibbioc=1.20.0=r3.3.2_1\n",
      "  - biopython=1.68=py35_0\n",
      "  - blas=2.16=openblas\n",
      "  - bleach=1.5.0=py35_0\n",
      "  - bzip2=1.0.8=h516909a_2\n",
      "  - ca-certificates=2020.4.5.1=hecc5488_0\n",
      "  - cairo=1.14.6=4\n",
      "  - certifi=2018.8.24=py35_1001\n",
      "  - curl=7.52.1=0\n",
      "  - fontconfig=2.12.1=6\n",
      "  - freetype=2.7=1\n",
      "  - gettext=0.19.8.1=hc5be6a0_1002\n",
      "  - glib=2.51.4=0\n",
      "  - graphite2=1.3.13=he1b5a44_1001\n",
      "  - gsl=2.6=h294904e_0\n",
      "  - harfbuzz=1.4.3=0\n",
      "  - html5lib=0.9999999=py35_0\n",
      "  - icu=58.2=hf484d3e_1000\n",
      "  - jpeg=9c=h14c3975_1001\n",
      "  - libblas=3.8.0=16_openblas\n",
      "  - libcblas=3.8.0=16_openblas\n",
      "  - libffi=3.2.1=he1b5a44_1007\n",
      "  - libgcc=7.2.0=h69d50b8_2\n",
      "  - libgcc-ng=9.2.0=h24d8f2e_2\n",
      "  - libgfortran-ng=7.3.0=hdf63c60_5\n",
      "  - libiconv=1.15=h516909a_1006\n",
      "  - liblapack=3.8.0=16_openblas\n",
      "  - liblapacke=3.8.0=16_openblas\n",
      "  - libopenblas=0.3.9=h5ec1e0e_0\n",
      "  - libpng=1.6.28=1\n",
      "  - libstdcxx-ng=9.2.0=hdf63c60_2\n",
      "  - libtiff=4.0.7=0\n",
      "  - libxml2=2.9.5=0\n",
      "  - llvm-openmp=10.0.0=hc9558a2_0\n",
      "  - markdown=3.2.1=py_0\n",
      "  - mmtf-python=1.0.2=py35_0\n",
      "  - mock=2.0.0=py35_0\n",
      "  - msgpack-python=0.5.6=py35h2d50403_3\n",
      "  - ncurses=5.9=10\n",
      "  - numpy=1.15.2=py35h99e49ec_0\n",
      "  - numpy-base=1.15.2=py35h2f8d375_0\n",
      "  - olefile=0.46=py_0\n",
      "  - openssl=1.0.2u=h516909a_0\n",
      "  - pandas=0.23.4=py35hf8a1672_0\n",
      "  - pango=1.40.4=0\n",
      "  - pbr=5.4.2=py_0\n",
      "  - pcre=8.44=he1b5a44_0\n",
      "  - pillow=4.3.0=py35_0\n",
      "  - pip=20.0.2=py_2\n",
      "  - pixman=0.34.0=h14c3975_1003\n",
      "  - protobuf=3.4.0=py35_0\n",
      "  - python=3.5.4=0\n",
      "  - python-dateutil=2.8.1=py_0\n",
      "  - pytz=2019.3=py_0\n",
      "  - r-base=3.3.2=5\n",
      "  - readline=6.2=0\n",
      "  - reportlab=3.4.0=py35_0\n",
      "  - rpy2=2.7.8=py35r3.3.2_1\n",
      "  - scikit-learn=0.20.0=py35h22eb022_1\n",
      "  - scipy=1.1.0=py35he2b7bc3_1\n",
      "  - setuptools=40.4.3=py35_0\n",
      "  - six=1.14.0=py_1\n",
      "  - sqlite=3.13.0=1\n",
      "  - tensorflow=1.3.0=py35_0\n",
      "  - tk=8.5.19=2\n",
      "  - webencodings=0.5.1=py_1\n",
      "  - werkzeug=1.0.1=pyh9f0ad1d_0\n",
      "  - wheel=0.34.2=py_1\n",
      "  - xz=5.2.5=h516909a_0\n",
      "  - zlib=1.2.8=3\n",
      "  # PlasFlow: should be installed from smaegol, NOT from pip\n",
      "  - plasflow=1.1.0"
     ]
    }
   ],
   "source": [
    "pipelineInfo('/ebio/abt3_projects/software/dev/ll_pipelines/llg/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-29T12:23:49.903700Z",
     "iopub.status.busy": "2021-04-29T12:23:49.902621Z",
     "iopub.status.idle": "2021-04-29T12:23:50.060494Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LLPRIMER\n",
      "========\n",
      "\n",
      "Ley Lab Primer generation pipeline (LLPRIMER)\n",
      "\n",
      "* Version: 0.3.0\n",
      "* Authors:\n",
      "  * Nick Youngblut <nyoungb2@gmail.com>\n",
      "* Maintainers:\n",
      "  * Nick Youngblut <nyoungb2@gmail.com>\n",
      "\n",
      "--- conda envs ---\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llprimer//bin/envs/pdp.yaml <==\n",
      "channels:\n",
      "- conda-forge\n",
      "- bioconda\n",
      "dependencies:\n",
      "- python=3.7\n",
      "- intervaltree\n",
      "- prodigal\n",
      "- blast\n",
      "- bedtools\n",
      "- mafft\n",
      "- mummer=3.23\n",
      "- emboss\n",
      "- primer3=1.1.4\n",
      "- biopython<1.78\n",
      "- pybedtools\n",
      "- joblib\n",
      "- tqdm\n",
      "- openpyxl\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llprimer//bin/envs/genes.yaml <==\n",
      "channels:\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- python=3\n",
      "- numpy\n",
      "- pyfaidx\n",
      "- bioconda::seqkit\n",
      "- bioconda::fasta-splitter\n",
      "- bioconda::vsearch\n",
      "- bioconda::mmseqs2\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llprimer//bin/envs/aln.yaml <==\n",
      "channels:\n",
      "- bioconda\n",
      "- conda-forge\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::revtrans\n",
      "- bioconda::kalign3\n",
      "- bioconda::mafft\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llprimer//bin/envs/blast.yaml <==\n",
      "channels:\n",
      "- bioconda\n",
      "- conda-forge\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::seqkit\n",
      "- bioconda::biopython<1.78\n",
      "- bioconda::blast>=2.10.1\n",
      "\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llprimer//bin/envs/primer3.yaml <==\n",
      "channels:\n",
      "- bioconda\n",
      "- conda-forge\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::biopython<1.78\n",
      "- bioconda::primer3-py\n",
      "==> /ebio/abt3_projects/software/dev/ll_pipelines/llprimer//bin/envs/prokka.yaml <==\n",
      "channels:\n",
      "- bioconda\n",
      "dependencies:\n",
      "- pigz\n",
      "- bioconda::prokka"
     ]
    }
   ],
   "source": [
    "pipelineInfo('/ebio/abt3_projects/software/dev/ll_pipelines/llprimer/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-04-29T12:23:50.145041Z",
     "iopub.status.busy": "2021-04-29T12:23:50.143164Z",
     "iopub.status.idle": "2021-04-29T12:23:52.742891Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "R version 3.6.3 (2020-02-29)\n",
       "Platform: x86_64-conda_cos6-linux-gnu (64-bit)\n",
       "Running under: Ubuntu 18.04.5 LTS\n",
       "\n",
       "Matrix products: default\n",
       "BLAS/LAPACK: /ebio/abt3_projects/Georg_animal_feces/envs/tidyverse/lib/libopenblasp-r0.3.9.so\n",
       "\n",
       "locale:\n",
       " [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              \n",
       " [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    \n",
       " [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   \n",
       " [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 \n",
       " [9] LC_ADDRESS=C               LC_TELEPHONE=C            \n",
       "[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       \n",
       "\n",
       "attached base packages:\n",
       "[1] stats     graphics  grDevices utils     datasets  methods   base     \n",
       "\n",
       "other attached packages:\n",
       "[1] LeyLabRMisc_0.1.6 ggplot2_3.3.1     tidyr_1.1.0       dplyr_1.0.0      \n",
       "\n",
       "loaded via a namespace (and not attached):\n",
       " [1] Rcpp_1.0.4.6     magrittr_1.5     munsell_0.5.0    tidyselect_1.1.0\n",
       " [5] uuid_0.1-4       colorspace_1.4-1 R6_2.4.1         rlang_0.4.6     \n",
       " [9] tools_3.6.3      grid_3.6.3       gtable_0.3.0     withr_2.2.0     \n",
       "[13] htmltools_0.4.0  ellipsis_0.3.1   digest_0.6.25    tibble_3.0.1    \n",
       "[17] lifecycle_0.2.0  crayon_1.3.4     IRdisplay_0.7.0  purrr_0.3.4     \n",
       "[21] repr_1.1.0       base64enc_0.1-3  vctrs_0.3.1      IRkernel_1.1    \n",
       "[25] glue_1.4.1       evaluate_0.14    stringi_1.4.6    pbdZMQ_0.3-3    \n",
       "[29] compiler_3.6.3   pillar_1.4.4     scales_1.1.1     generics_0.0.2  \n",
       "[33] jsonlite_1.6.1   pkgconfig_2.0.3 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "sessionInfo()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R [conda env:Georg_animal_feces-tidyverse]",
   "language": "R",
   "name": "conda-env-Georg_animal_feces-tidyverse-r"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
